<html>
<head>

  <meta name="navigation" content="Comparative Genomics">
  <title>Pairwise whole-genome alignments</title>


</head>

<body>

<h1>Genomic alignments</h1>

<p>Pairwise whole genome alignments can be used to determine conservation or differences between pairs of species, or to match up regions between species, so as to study the same genomic region in multiple species.</p>

<p><a href="https://www.bx.psu.edu/~rsharris/lastz/">LastZ</a> and its predecessor <a href="https://europepmc.org/articles/PMC430961">Bla</a><a href="https://europepmc.org/articles/PMC208784">stZ</a> are used to align the genome sequences at the DNA level.</p>

<p>Genomes are compared to one another, for comparison between species, and to themselves, to identify paralogous regions. In self-alignment analyses, the trivial alignments (i.e. regions against themselves) are removed, such that only the paralogous regions remain.</p>

<p>The actual whole-genome alignments are the results of post-processing the raw LastZ (or BlastZ) results. In the first step, original blocks are chained according to their location in both genomes. The netting process chooses for the reference species the best sub-chain in each region.</p>

<p>These alignments are used to calculate <a href="/info/genome/compara/synteny.html">synteny</a> and for <a href="/info/genome/compara/Ortholog_qc_manual.html">scoring orthologue quality</a>.</p>

<h2>Alignments available</h2>

[[SCRIPT::EnsEMBL::Web::Document::HTML::Compara::draw_pairwise_alignment_list()]]

</body>
</html>
