<html>
<head>
  <meta name="navigation" content="Comparative Genomics">
  <title>EPO Pipeline</title>
</head>

<body>

<p>
In the EPO pipeline, Enredo, Pecan and Ortheus are the tip of the iceberg.
They require homology and synteny information that are derived from alignments of <em>anchor</em> sequences.
This page explains what the anchors are, how they are generated and used.
</p>

<h3 id="anchorgeneration">Anchor generation</h3>

<p>
The anchors represent short regions of conservation, typically about 100 bases in length, from a phylogenetically representative subset of the species we wish to align.
As such, anchors only have to be regenerated when the subset of species used is not representative any more.
</p>

<p>
The anchor set is generated from pairwise alignments (LastZ-Net) of a non-reference species to a reference species, so all the species chosen to generate the anchors must have a pairwise alignment with the reference species.
The pairwise alignments are stacked together based on their coordinates and the regions are realigned with <a href="https://github.com/benedictpaten/pecan">Pecan</a>.
Then <a href="http://mendel.stanford.edu/SidowLab/downloads/gerp/">GERP</a> is used to identify conserved regions, from where the anchor sequences are defined.
</p>

<p>
We consider a good set of anchors to contain several hundred thousands anchors in order to cover the genomes with enough precision.
</p>

<h3 id="anchormapping">Anchor mapping</h3>
<p>
The anchor set is mapped (currently we use exonerate) against all the genomes we wish to include in the final alignment. Overlapping anchor mappings are filtered so that any particular genomic location is associated with, at most, one anchor.  
</p>

<p>
This step is computationally expensive, but being run independently for each genome, can be modelled as a cumulative process.
Assuming the anchors have not changed, we only need to run it for the new assemblies, and can reuse the mappings of the remaining assemblies.
</p>

<h3 id="genomealignment">Genome alignment</h3>

<p>
Enredo produces a list of homologous genomic regions from the positions where the anchors have mapped. These homologous regions are then aligned with Pecan or Ortheus (Ortheus uses Pecan to do the aligning and also generates a sequence for each ancestral node in a tree which it predicts for each region).
</p>

</body>
</html>
