<html>
<head>
  <title>Gene Families</title>
</head>

<body>

<h1>Gene Families in Compara</h1>

<h2 id="Introduction">Introduction</h2>

<p>
Ensembl families are determined through classification of all Ensembl proteins along with 
metazoan sequences from <a rel="external" href="http://www.ebi.ac.uk/uniprot/">UniProtKB</a>.  
It therefore provides a way of exploring orthologues and closely related homologues across a range of animal species.
</p>

  <h2 id="overview">Pipeline overview</h2>

  <p>The pipeline consists of the following steps:</p>
  <ol>
	<li>Load members from Ensembl</li>
	<li>Load UniProt entries</li>
	<li>Run HMM search (using the HMM library described in this <a href="hmm_lib.html">document</a>, excluding the TF6 families)</li>
<h1 id="genefamilies">Gene families</h1>

<p>Ensembl families are determined through classification of all Ensembl proteins, including multiple isoforms of the same gene, along with metazoan sequences from <a href="http://www.uniprot.org/">UniProt</a>. It therefore provides a way of exploring orthologues and closely related homologues across a range of animal species.</p>

<h2 id="pipeline">Pipeline</h2>

<p>The pipeline consists of the following steps:</p>

<ol>
<li>Load proteins from Ensembl and UniProt.</li>

<li>Run HMM search (using <a href="treefam_hmm_library.md">the TreeFam HMM library</a>, excluding the TF6 families) to find the families.</li>

<li>Align the families with <a href="http://mafft.cbrc.jp/alignment/software/">Mafft</a>.</li>

<li>Annotate the family with a consensus description, based on its members' descriptions.</li>
</ol>

<h3 id="familyid">Family ID</h3>

<p>The families have been assigned the stable ID of their <a href="treefam_hmm_library.md">corresponding HMM</a>.</p>

<h3 id="consensusannotation">Consensus Annotation</h3>

<p>For each cluster obtained, a consensus annotation is automatically generated from the UniProt description lines using the following approach: </p>

<ul>
<li>If the description covers less than 40% of UniProt members in the cluster, the family description is assigned 'AMBIGUOUS'.</li>

<li>If the annotation confidence score, described below, is zero, 'UNKNOWN' is assigned.</li>

<li>Be aware that 'UNCHARACTERIZED' is a UniProt description for a protein, and does not reflect the score. </li>
</ul>

<p>The annotation confidence score is the percentage of UniProt family members with this description, or part of it. Note that only family members with 'informative' UniProt descriptions are taken into account.</p>

<h3 id="multiplealignments">Multiple Alignments</h3>

<p>Ensembl provides pre-calculated multiple sequence alignments of all members for each cluster. We provide a <a href="http://wasabiapp.org/">Wasabi viewer</a> in the browser for viewing the alignments between just the Ensembl proteins, and the Ensembl and UniProt proteins together.You can also export a text file with the alignments of all the family members - a wide range of formats is available from the control panel.</p>
<p>Alternatively, export alignments using the <a href="/info/docs/api/compara/index.html">Compara Perl API</a>.</p>


</body>
</html>
