<html>
<head>
  <title>TreeFam HMM library</title>
  </head>

  <body>

  <h1>TreeFam HMM library</h1>

  <h2 id="Introduction">Introduction</h2>
  <p>
  The TreeFam HMM library is based upon the <a href="http://www.pantherdb.org/">Panther database</a> (version 9) <a href="#fn1">[1]</a>.
  In order to define a common and comprehensive set of HMMs to be used across all the Ensembl divisions we 
  added extra families defined in the TreeFam database (<a href="http://treefam.org">version 9</a>), and built new ones to model gene-families 
  that were not previously described in either databases.
  </p>

  <h2 id="species_set">Species coverage</h2>
  <p>	
  The gene models used in this study come from the version 82 of Ensembl, and version 29 of Ensembl Genomes. 
  We selected all the genomes used in the protein-tree analysis of these databases to maximize the coverage 
  of the tree of life. We used sequences from 708 species, of which 313 fungi, 130 protists, 41 plants, 119 
  metazoans (incl. 64 chordates) and 105 bacteria.
  </p>


  <h2 id="hmm_building">HMM library</h2>

  
  <p>
      The procedure to build the HMM library is described here:
  <ol>
	<li>We first removed the redundancy between Panther 9 and TreeFam 9. This was done by:
	  <ol>
	  <li>Emitting consensus sequences from the 15,736 TreeFam 9 HMM profiles.</li>
          <li>Classifying them against the 7,181 Panther 9 profiles with PantherScore.</li>
      </ol>
      As a result:
          <ul>
              <li>13,638 TreeFam 9 families that had a hit on the Panther 9 profiles were discarded from the overall library.</li>
              <li>2,098 TreeFam families remained after this filter.</li>
          </ul>
          </li>
	<li>We classified all the available sequences (8,488,481) according to their scores to these HMM profiles using PantherScore.</li>
        <li>We used Mafft <a href="#fn2">[2]</a> to align all these clusters.</li>
        <li>We used TrimAl <a href="#fn4">[4]</a> and Notung <a href="#fn5">[5]</a> to estimate the quality of the alignments, and also computed metrics such as the percentage of gaps in the alignment.<br/>
	We identified a number of very large Panther families and decided to promote their sub-families HMMs (as defined in the Panther database) to the family rank in order to replace the Family HMM. After this step, the library is composed of 38,554 HMMs coming from Panther (5,103 from families, and 33,451 from sub-families).</li>
	<li>We reclustered all the sequences against the new library.</li>
        <li>Sequences that did not match any of the profiles were then clustered together with hcluster_sg (a hierarchical clustering software for sparse graphs <a href="#fn3">[3]</a>) using a blast all-vs-all approach (see the Protein-Tree pipeline <a href="homology_method.html">documentation</a>).</li>
	<li>Given the large number (404,108) of clusters we applied further filtering and quality-control checks. We used the following metrics and evaluated different thresholds.
	<ol>
	  <li>Taxonomic coverage (ratio of taxa present in a particular HMM profile vs all the taxa in our species tree) for each member in all the clusters.</li>
	  <li>Minimum number of genes</li>
	  <li>Minimum number of species</li>
	</ol>
	Overall, filtering the clusters by the number of different species per family and per division turned out to be the clearest way of discarding very small clusters. We used these thresholds:
	<ol>
	  <li>Clusters with at least 3 different species for protists division only</li>
	  <li>Clusters with at least 5 different species for all other divisions</li>
	</ol>
        Out of the 404,108 clusters, only 128,233 passed these filters, of which 34,119 come from Panther 9, 1,451 from Treefam 9.</li>
	
	<li>The new clusters (92,663) built with hcluster_sg that passed the filtering step were prefixed with TF6xxxxx in order to diferentiate them from previous TreeFam families.</li>
	<li>We used Mafft to align all these clusters, and built HMM profiles on them.
	We calibrated most of the profiles (the calibration of a few very big alignments has failed)</li>
  </ol>

  <h2 id="stats">Statistics</h2>
  <table>
      <tr><td style="text-align: right">Total number of families before filtering:</td><td>404,108</td></tr>
      <tr><td style="text-align: right">Total number of families after filtering:</td><td>128,233</td></tr>
  </table>

  <br/>
  <p>
  Breakdown by source and nomenclature:
  </p>
  <table>
      <tr><td style="text-align: right">Panther v9:</td><td style="text-align: right">34,119</td><td style="text-align: left">PTHRxxxxx</td></tr>
      <tr><td rowspan="2" style="vertical-align: middle; text-align: right">TreeFam v9:</td><td style="text-align: right">8</td><td style="text-align: left">TF1xxxxx</td></tr>
      <tr><td style="text-align: right">1,443</td><td style="text-align: left">TF3xxxxx</td></tr>
      <tr><td style="text-align: right">New clusters:</td><td style="text-align: right">92,663</td><td style="text-align: left">TF6xxxxx</td></tr>
  </table>
  
  <h2 id="references">Notes and References</h2>

  <ol>
      <li id="fn1">PANTHER HMM scoring tools - Version 1.03. Can be downloaded <a href="http://ftp.pantherdb.org//hmm_scoring/current_release">here</a></li>
<li id="fn2">
<a href="http://mafft.cbrc.jp/alignment/software/">Mafft</a>: Katoh K, Standley DM. "MAFFT multiple sequence alignment software version 7: improvements in performance and usability." <a href="https://europepmc.org/articles/PMC3603318">Mol Biol Evol. 2013 Apr;30(4):772-80</a><br/>
We use the version 7.113 with the command line options: <kbd>--auto</kbd>
</li>

<li id="fn3">Li, H et al., <a href="http://treesoft.svn.sourceforge.net/viewvc/treesoft/branches/lh3/hcluster/">Hcluster_sg</a>: hierarchical clustering software for sparse graphs.
</li>

<li id="fn4">"trimAl: a tool for automated alignment trimming in large-scale phylogenetic analyses."
Salvador Capella-Gutierrez; Jose M. Silla-Martinez; Toni Gabaldon. <a href="https://europepmc.org/articles/PMC2712344">Bioinformatics 2009 25: 1972-1973</a>.
</li>

<li id="fn5">
"A Inferring duplications, losses, transfers and incomplete lineage sorting with nonbinary species trees."
M. Stolzer, H. Lai, M. Xu, D. Sathaye, B. Vernot, D. Durand. <a href="https://europepmc.org/abstract/MED/22962460">2012. Bioinformatics, 28 (18): i409-i415.</a>
     </li>

  </ol>
  
</body>
</html>

