<html>
<head>

  <title>Protein trees</title>


  <meta name="author" content="web" />

  <meta name="navigation" content="Comparative Genomics" />

</head>
<body>

<h1>Protein trees</h1>
  
<p>Gene trees aim to represent the evolutionary history of gene families, which evolved from a common ancestor. Reconciliation of the gene tree against the species trees allows us to distinguish duplication and speciation events, thus inferring <a href="/info/genome/compara/homology_types.html">orthologues and paralogues</a>. </p>


<p>
There is a clear concordance with reciprocal best approaches in the
simple case of unique orthologous genes. However, the gene tree
pipeline is able to find more complex one-to-many and many-to-many
relations. This, for instance, significantly raises the number of Teleost
(bony fish) to Mammal orthologues and has even more dramatic effects on
Fly/Mammal or Worm/Mammal orthologous gene predictions. Using this
approach, we are also able to "time" duplication events which produce
paralogues by identifying the most recent common ancestor (i.e. taxonomy level) for a
given internal node of the tree.
</p>

<p>
Please note that, primarily due to storage limitations, between-species paralogy relationships are not stored in the Compara databases or listed on the website (except in some special cases described in the "Between-species paralogues" section of the <a href="/info/genome/compara/homology_types.html">homology types documentation</a>). However, these relationships can be identified by examining the gene trees, which are annotated with duplication events.
</p>



<h2 id="detailsontreebuilding">Details on tree building</h2>

<p>The gene orthology and paralogy prediction pipeline has eight basic steps:</p>

<ol>
<li>
  Load the <a href="/info/genome/genebuild/canonical.html">Ensembl canonical</a>
  translation of each gene from species in Ensembl, which is currently chosen on
  the basis of various factors such as conservation, expression, coding sequence
  length and concordance with other key resources.
</li>

<li>Run an HMM search on <a href="/info/genome/compara/hmm_lib.html">the TreeFam HMM library</a> to classify the sequences into their families.</li>

<li>Cluster the genes that did not have any match into additional families:
<ol>
<li>run NCBI Blast+ <a href="#fn1">[1]</a> (refined with SmithWaterman) on every orphaned gene against every other (both self and non-self species). We use the version 2.2.28, with the parameters: <code>-seg no -max_hsps_per_subject 1 -use_sw_tback -num_threads 1</code>.</li>

<li>Build a sparse graph of gene relations based on Blast e-values and generate clusters using hcluster_sg <a href="#fn2">[2]</a>. Hcluster_sg has a little insight about the phylogeny of the species (namely: a list of outgroup species), which helps define pertinent clusters for the ingroup species. We use yeast (Saccharomyces cerevisiae) as an outgroup (via the -C option) and the following command line arguments: <code>-m 750 -w 0 -s 0.34 -O</code>. See below for more details.</li>
</ol>
</li>

<li>Large families that would be too complex to analyse are broken down with QuickTree <a href="#fn7">[7]</a> to limit them to 1,500 genes.</li>

<li>For each cluster (family), build a multiple alignment based on the protein sequences using either a combination of multiple aligners, consensified by M-Coffee <a href="#fn3">[3]</a> or Mafft <a href="#fn4">[4]</a> when the cluster is too large, or running MCoffee takes too long. We use the version 9.03.r1318 of M-Coffee, with the aligner set: mafftgins_msa, muscle_msa, kalign_msa, t_coffee_msa, and the Mafft version 7.113 with the command line options: <code>--auto</code>.</li>

<li>For each aligned cluster, build a phylogenetic tree using TreeBeST <a href="#fn5">[5]</a> using the CDS back-translation of the protein multiple alignment from the original DNA sequences. A rooted tree with internal duplication tags is obtained at this stage, reconciling it against a species tree inferred from the <a href="https://www.ncbi.nlm.nih.gov/taxonomy">NCBI taxonomy</a> in general, or by our <a href="https://github.com/Ensembl/ensembl-compara/tree/main/pipelines/SpeciesTreeFromBusco">in-house species-tree pipeline</a> for the Vertebrates division. See below for more details.</li>

<li>From each gene tree, infer gene pairwise relations of orthology and paralogy types.</li>

<li>A <a href="stable_ids.html">stable ID</a> is assigned to each GeneTree.</li>

</ol>


<h2 id="hcluster">Clustering</h2>

<p>
hcluster_sg <a href="#fn2">[2]</a> performs hierarchical clustering under mean distance. It reads an input file that describes the similarity between two sequences, and groups two nearest nodes at each step. When two nodes are joined, the distance between the joined node and all the other nodes are updated by mean distance. This procedure is iterated until one of the three rules is met:
</p>
<ol>
<li style="list-style-type: lower-alpha">Do not merge cluster A and B if the total number of edges between A and B is smaller than <code>|A|*|B|/3</code>, where <code>|A|</code> and <code>|B|</code> are the sizes of A and B, respectively. This rule guarantees each cluster is compact.</li>
<li style="list-style-type: lower-alpha">Do not join A to any other cluster if <code>|A| &lt; 500</code>. This rule avoids huge clusters which may cause computational burden for multiple alignment and tree building as well.</li>
<li style="list-style-type: lower-alpha">Do not join A and B if both A and B contain plant genes or both A and B contain Fungi genes. This rule tries to find animal gene families. TreeFam clustering is done with outgroups.</li>
</ol>

<p>
Hcluster_sg also introduces an additional edge-breaking rule: removes an edge between cluster A and B if the number of edges between A and B is smaller than <code>|A|*|B|/10</code>. This heuristic rule removes weak relations which are quite unlikely to be joined at a later step.
</p>

<p>
As the pipeline has to be completed in time for the Ensembl releases, we limit the size of the clusters to 1,500 genes.
For larger clusters, we run Mafft <a href="#fn4">[4]</a> and QuickTree <a href="#fn7">[7]</a>.
QuickTree is a very fast and efficient way to build an unrooted phylogenetic tree. We then find the branch that best splits the cluster into two halves.
We recursively follow this approach until each sub-cluster is smaller than 1,500 genes.
</p>

<h2 id="treebest">Tree building</h2>

<p>The CDS
back-translated protein alignment (i.e., codon alignment) is used to
build five different trees (within TreeBeST <a href="#fn5">[5]</a>):</p>
<ol class="lc-roman">
<li>a maximum likelihood (ML) tree built, based on the protein alignment with the WAG model, which takes into account the species tree</li>
<li>a ML tree built using phyml, based on the codon alignment with the Hasegawa-Kishino-Yano (HKY) model, also taking into account the species tree</li>
<li>a neighbour-joining (NJ) tree using p-distance, based on the codon alignment</li>
<li>a NJ tree using dN distance, based on the codon alignment</li>
<li>a NJ tree using dS distance, based on the codon alignment</li>
</ol>

<p>For (1) and (2), TreeBeST uses a modified version of phyml release
2.4.5 which takes an input
species tree, and tries to build a gene tree that is consistent with
the topology of the species tree. This "species-guided" phyml uses the
original phyml tree-search code. However, the objective function
maximised during the tree-search is multiplied by an extra likelihood
factor not found in the original phyml. This extra likelihood factor
reflects the number of duplications and losses inferred in a gene
tree, given the topology of the species tree. The species-guided phyml
allows the gene tree to have a topology that is inconsistent with the
species tree if the alignment strongly supports this. The species tree
is based on the NCBI taxonomy tree (subject to some modifications
depending on new datasets).</p>

<p>The final tree is made by merging the five trees into one consensus
tree using the "tree merging" algorithm. This allows TreeBeST to
take advantage of the fact that DNA-based trees often are more
accurate for closely related parts of trees and protein-based trees
for distant relationships, and that a group of algorithms may
outperform others under certain scenarios. The algorithm
simultaneously merges the five input trees into a consensus tree. The
consensus topology contains clades found in any of the input trees,
where the clades chosen are those that minimise the number of
duplications and losses inferred, and have the highest bootstrap
support. Branch lengths are estimated for the final consensus tree
based on the DNA alignment, using phyml with the HKY model.</p>

<h2 id="notes">Notes and References</h2>

<ol>

<li id="fn1">
Camacho C et al,, "BLAST+: architecture and applications."  <a href="https://europepmc.org/articles/PMC2803857">BMC Bioinformatics. 2009 Dec 15;10:421.</a><br/>
We use the version 2.2.28, with the parameters: <kbd>-seg no -max_hsps_per_subject 1 -use_sw_tback -num_threads 1</kbd>
</li>

<li id="fn2">Li, H et al., <a href="https://sourceforge.net/projects/treesoft/">Hcluster_sg</a>: hierarchical clustering software for sparse graphs. <br/>
We use yeast (Saccharomyces cerevisiae) as an outgroup (via the <kbd>-C</kbd> option) and the following command line arguments: <kbd>-m 750 -w 0 -s 0.34 -O</kbd><br/>
The weights used in the graph are <kbd>MIN(100, ROUND(-LOG10(evalue)/2))</kbd>
</li>

<li id="fn3">
<a href="https://tcoffee.org/Projects/mcoffee/index.html">M-Coffee</a>:
Wallace IM, O'Sullivan O, Higgins DG, Notredame C. "M-Coffee: combining multiple sequence alignment methods with T-Coffee." <a href="https://europepmc.org/articles/PMC1410914">Nucleic Acid Research. 2006 Mar 23;34(6):1692-9.</a><br/>
We use the version 9.03.r1318, with the aligner set: <kbd>mafftgins_msa, muscle_msa, kalign_msa, t_coffee_msa</kbd>
</li>

<li id="fn4">
<a href="https://mafft.cbrc.jp/alignment/software/">Mafft</a>: Katoh K, Standley DM. "MAFFT multiple sequence alignment software version 7: improvements in performance and usability." <a href="https://europepmc.org/abstract/MED/23329690">Mol Biol Evol. 2013 Apr;30(4):772-80</a><br/>
We use the version 7.113 with the command line options: <kbd>--auto</kbd>
</li>


<li id="fn5">Li, H et al., <a href="https://github.com/Ensembl/treebest">Compara-specific version</a>, <a href="https://treesoft.sourceforge.net/treebest.shtml">Original version</a> (TreeBeST was previously known as NJTREE)</li>

<li id="fn6">Yang, Z. "PAML: a program package for phylogenetic analysis by maximum likelihood" <a href="http://abacus.gene.ucl.ac.uk/software/paml.html">Comput Appl Biosci. 1997 Oct; 13(5):555-556.</a><br/>
</li>

<li id="fn7">
<a href="https://www.sanger.ac.uk/tool/quicktree/">QuickTree</a>: Howe K, Bateman A and Durbin R, "QuickTree: building huge Neighbour-Joining trees of protein sequences." <a href="https://www.ncbi.nlm.nih.gov/pubmed/12424131">Bioinformatics (Oxford, England) 2002;18;11;1546-7</a><br/>
QuickTree builds an unrooted tree and we recursively split the cluster by finding a branch that roughly holds half of the nodes on each side.
</li>


</ol>








</body>
</html>
