<html>
<head>

  <meta name="navigation" content="Comparative Genomics">
  <title>Multiple genome alignments</title>


</head>

<body>


<h1 id="multiplegenomealignments">Multiple genome alignments</h1>

<p>Multiple alignments are calculated between groups of genomes. These are used to calculate <a href="/info/genome/compara/ancestral_sequences.html">ancestral sequences</a>, <a href="/info/genome/compara/age_of_base.html">age of base</a>, <a href="/info/genome/compara/conservation_and_constrained.html">conservation scores and constrained elements</a>.</p>

<h2 id="alignmentsavailable">Alignments available</h2>

[[SCRIPT::EnsEMBL::Web::Document::HTML::Compara::format_wga_table()]]

<h2 id="alignmentmethods">Alignment methods</h2>

<h3 id="pecanmultiplealignment">PECAN Multiple Alignment</h3>

<p><a href="https://github.com/benedictpaten/pecan">Pecan</a> is used to provide global multiple genomic alignments. First, Mercator is used to build a synteny map between the genomes and then Pecan builds alignments in these syntenic regions.</p>

<p>Pecan is a global multiple sequence alignment program that makes practical the probabilistic consistency methodology for significant numbers of sequences of practically arbitrary length. As input it takes a set of sequences and a phylogenetic tree. The parameters and heuristics it employs are highly user configurable, it is written entirely in Java and also requires the installation of Exonerate.</p>

<h3 id="epomultiplealignment">EPO Multiple Alignment</h3>

<p>The EPO (Enredo, Pecan, Ortheus) pipeline is a three step pipeline for whole-genome multiple alignments.</p>

<ol>
<li>Enredo produces colinear segments from extant genomes handling both rearrangements, deletions and duplications.</li>

<li>Pecan, as described above, is used to align these segments.</li>

<li>Finally, Ortheus is used to create genome-wide ancestral sequence reconstructions.</li>
</ol>

<p>
The pipeline requires alignments of so-called <em>anchor</em> sequences, which are explained <a href="/info/genome/compara/epo_anchors_info.html">here</a>.
Further details on all these methods can be found at: <a href="https://europepmc.org/abstract/MED/18849524">Enredo and Pecan: Genome-wide mammalian consistency-based multiple alignment with paralogs</a>
</p>

<h4 id="epo2xmultiplealignment">EPO-Extended Multiple Alignment</h4>
<p>Due to difficulties with running Ortheus on the fragmented assemblies, we have two flavours of the pipeline.</p>

<ol>
<li>The plain EPO pipeline is available on the chromosome-level genomes, listed as EPO in the table above</li>

<li>The scaffold-level genomes are then projected onto the EPO alignments using LastZ-net alignments, listed as EPO-Extended.</li>
</ol>

<p>By construction, each pair of EPO and EPO-Extended alignments represent the exact same alignment of chromosome-level genomes.</p>

<h3 id="progressivecactus">Progressive Cactus</h3>

<p>Progressive-Cactus is a next-generation aligner that stores whole-genome alignments in a graph structure. Genomes can be added incrementally, which makes it scalable to hundreds of genomes. Further details on these methods can be found in <a href="https://europepmc.org/articles/PMC3166836">Algorithms for genome multiple sequence alignment</a> and <a href="https://europepmc.org/abstract/MED/21385048">Cactus graphs for genome comparisons</a>.</p>

</body>
</html>
