<html>
<head>

  <meta name="navigation" content="Comparative Genomics">
  <title>Multiple genome alignments</title>


</head>

<body>


<h1 id="multiplegenomealignments">Multiple genome alignments</h1>

<p>Multiple alignments are calculated between groups of genomes. These are used to calculate <a href="/info/genome/compara/ancestral_sequences.html">ancestral sequences</a>, <a href="/info/genome/compara/age_of_base.html">age of base</a>, <a href="/info/genome/compara/conservation_and_constrained.html">conservation scores and constrained elements</a>.</p>

<h2 id="alignmentsavailable">Alignments available</h2>

[[SCRIPT::EnsEMBL::Web::Document::HTML::Compara::format_wga_table()]]

<h2 id="alignmentmethods">Alignment methods</h2>

<h3 id="pecanmultiplealignment">PECAN Multiple Alignment</h3>

<p>Pecan <a href="#fn1">[1]</a> is used to provide global multiple genomic alignments. First, Mercator is used to build a synteny map between the genomes and then Pecan builds alignments in these syntenic regions.</p>

<p>Pecan is a global multiple sequence alignment program that makes practical the probabilistic consistency methodology for significant numbers of sequences of practically arbitrary length. As input it takes a set of sequences and a phylogenetic tree. The parameters and heuristics it employs are highly user configurable, it is written entirely in Java and also requires the installation of exonerate <a href="#fn2">[2]</a>.</p>

<h3 id="epomultiplealignment">EPO Multiple Alignment</h3>

<p>The EPO (Enredo, Pecan, Ortheus) <a href="#fn1">[1]</a> pipeline is a three step pipeline for whole-genome multiple alignments.</p>

<ol>
<li>Enredo produces colinear segments from extant genomes handling both rearrangements, deletions and duplications.</li>

<li>Pecan, as described above, is used to align these segments.</li>

<li>Finally, Ortheus is used to create genome-wide ancestral sequence reconstructions.</li>
</ol>

<p>
The pipeline requires alignments of so-called <em>anchor</em> sequences, which are explained <a href="/info/genome/compara/epo_anchors_info.html">here</a>.
</p>

<h4 id="epo2xmultiplealignment">EPO-Extended Multiple Alignment</h4>
<p>Due to difficulties with running Ortheus on the fragmented assemblies, we have two flavours of the pipeline.</p>

<ol>
<li>The plain EPO pipeline is available on the chromosome-level genomes, listed as EPO in the table above</li>

<li>The scaffold-level genomes are then projected onto the EPO alignments using LastZ-net alignments, listed as EPO-Extended.</li>
</ol>

<p>By construction, each pair of EPO and EPO-Extended alignments represent the exact same alignment of chromosome-level genomes.</p>

<h3 id="progressivecactus">Progressive Cactus</h3>

<p>
Progressive-Cactus <a href="#fn3">[3]</a> is a next-generation aligner that stores whole-genome
alignments in a graph structure. Genomes can be added incrementally, which makes it scalable to
hundreds of genomes.
</p>

<p>The Ensembl Compara Perl API provides access to Cactus alignment data in one of two ways: via HAL file (CACTUS_HAL) or database (CACTUS_DB).</p>

<h4>Cactus alignment via HAL file</h4>

<p>
Alignments of type CACTUS_HAL are accessed via a HAL file <a href="#fn4">[4]</a>. For performance reasons,
alignments are filtered to remove blocks whose length is below a threshold set to approximately one thousandth
the size of the genomic region being accessed. Within each alignment block, aligned sequences are deduplicated
per genome, keeping only the aligned sequence with the greatest number of nucleotides for the given genome.
</p>

<h4>Cactus alignment via database</h4>

<p>
Alignments of type CACTUS_DB are preloaded from a HAL file into a MySQL database following
an approach similar to that used by cactus-hal2maf <a href="#fn3">[3]</a> (version 2.9.7).
</p>

<ol>
  <li>
  Dump a <a href="https://genome.ucsc.edu/FAQ/FAQformat.html#format5">MAF alignment</a> file for a given reference genome
  (e.g. <i class="taxonomy">Homo sapiens</i>) and sequence region (typically 500 kilobases in length) using hal2maf
  <a href="#fn4">[4]</a> (version 2.2) with command-line options: <code>--noAncestors --unique</code>
  </li>

  <li>
  Filter out aligned sequences with fewer than 5 nucleotides, and filter out alignment blocks with fewer than 20 alignment columns.
  </li>

  <li>
  Normalise the alignment to merge smaller alignment blocks using <a href="https://github.com/ComparativeGenomicsToolkit/taffy">taffy</a>
  (<a href="https://github.com/ComparativeGenomicsToolkit/taffy/commit/5221c50fbf119699db1112ce74eea90dea17ba95">commit 5221c50</a>)
  with command-line options: <code>--filterGapCausingDupes --maximumBlockLengthToMerge 8000 --maximumGapLength 1200</code>
  </li>

  <li>
  Deduplicate alignments per genome within each MAF block using the mafDuplicateFilter command of mafTools <a href="#fn5">[5]</a>
  (<a href="https://github.com/ComparativeGenomicsToolkit/mafTools/commit/259e5b47fa2ee17ff5ad1bba9cebf2992cbb7228">commit 259e5b4 of ComparativeGenomicsToolkit version</a>) with command-line option: <code>--keep-first</code>
  </li>

  <li>
  Load MAF alignment blocks into the output MySQL database.
  </li>
</ol>

<p>
CACTUS_DB alignments are also filtered by the Compara Perl API at access time,
with the minimum block length set to one hundredth the size of the accessed region.
</p>

<h2 id="references">References</h2>

<ol>
  <li id="fn1">
  Paten B, Herrero J, Beal K, Fitzgerald S, Birney E.
  "<a href="https://github.com/jherrero/enredo">Enredo</a> and <a href="https://github.com/benedictpaten/pecan">Pecan</a>:
  Genome-wide mammalian consistency-based multiple alignment with paralogs."
  <a href="https://europepmc.org/abstract/MED/18849524">Genome Res. 2008 Nov;18(11):1814-28.</a>
  </li>

  <li id="fn2">
  Slater GS, Birney E.
  "Automated generation of heuristics for biological sequence comparison."
  <a href="https://europepmc.org/article/MED/15713233">BMC Bioinformatics. 2005 Feb;6:31.</a>
  </li>

  <li id="fn3">
  Armstrong J, Hickey G, Diekhans M, et al.
  "Progressive Cactus is a multiple-genome aligner for the thousand-genome era."
  <a href="https://europepmc.org/article/pmc/pmc7673649">Nature. 2020 Nov;587(7833):246-251.</a>
  </li>

  <li id="fn4">
  Hickey G, Paten B, Earl D, Zerbino D, Haussler D.
  "HAL: a hierarchical format for storing and analyzing multiple genome alignments."
  <a href="https://europepmc.org/article/MED/23505295">Bioinformatics. 2013 May;29(10):1341-1342.</a>
  </li>

  <li id="fn5">
  Earl D, Nguyen N, Hickey G, et al.
  "Alignathon: a competitive assessment of whole-genome alignment methods."
  <a href="https://europepmc.org/article/MED/25273068">Genome Research. 2014 Dec;24(12):2077-2089.</a>
  </li>
</ol>

</body>
</html>
