<html>
<head>

  <meta name="navigation" content="Comparative Genomics">
  <title>Species trees</title>


</head>

<body>

<h1 id="speciestrees">Species trees</h1>

<p>
The Compara pipelines use two main species trees.
</p>
<ol>
    <li>The <a href="https://www.ncbi.nlm.nih.gov/taxonomy">NCBI taxonomy</a> (i.e. topology only) in:
    <ul>
        <li>the <a href="/info/genome/compara/homology_method.html">Protein-trees</a> pipeline</li>
        <li>the <a href="/info/genome/compara/ncRNA_methods.html">ncRNA-trees</a> pipeline, where the three sub-trees <em>Eutheria</em>, <em>Sauria</em>, and <em>Clupeocephala</em> are flattened</li>
        <li>the CAFE pipelines (Gene Gain/Loss trees), with branch lengths coming from  <a href="https://www.timetree.org">the TimeTree database</a>
    </ul>
    </li>
    <li>A tree with branch-lengths computed in-house (available for download <a href="https://github.com/Ensembl/ensembl-compara/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/conf/vertebrates/species_tree.branch_len.nw">here</a>) in:
    <ul>
        <li>the <a href="/info/genome/compara/multiple_genome_alignments.html">Multiple-alignment</a> pipelines</li>
        <li>the <a href="/info/genome/compara/conservation_and_constrained.html">Constrained Elements / Conservation Scores</a> pipelines</li>
    </ul>
    We pass the unmasked whole genome sequences to <a href="https://mash.readthedocs.io">Mash</a> to compute pairwise distances, and then we generate the species tree using distance-based neighbour-joining guided by taxonomy.
    </li>
</ol>

</body>
</html>
