<html>
<head>
  <title>Variant quality</title>
</head>

<body>

<h1 style="margin-top:15px">Ensembl Variation - Variant quality</h1>

<p>Ensembl carry out quality control on all imported variants, and have a summary of the evidence behind a variant.</p>

<br />

<h2 id="evidence_status">Evidence status </h2>
<p>
  We provide a simple summary of the evidence supporting a variant as a guide to its potential reliability.
</p>
<table id="evidence_status_table" class="ss" >
  <tr>
    <th><span class="_ht ht" title="Icons designed by Ensembl">Icon</span></th>
    <th class="set_min">Name</th>
    <th>Description</th>
  </tr>
  <tr>
    <td style="text-align:center"><img src="/i/val/evidence_Multiple_observations.png" title="Multiple_observations"/></td>
    <td>Multiple observations</td>
    <td>The variant has multiple independent dbSNP submissions, i.e. submissions with a different submitter handles or different discovery samples.</td>
  </tr>
  <tr class="bg2">
    <td style="text-align:center"><img src="/i/val/evidence_Frequency.png" title="Frequency"/></td>
    <td>Frequency</td>
    <td>The variant is reported to be polymorphic in at least one sample.</td>
  </tr>
  <tr>
    <td style="text-align:center"><img src="/i/val/evidence_Cited.png" title="Cited"/></td>
    <td>Cited</td>
    <td>The variant is cited in a PubMed article.</td>
  </tr>
  <tr class="bg2">
    <td style="text-align:center"><img src="/i/val/evidence_Phenotype_or_Disease.png" title="Phenotype_or_Disease"/></td>
    <td>Phenotype or Disease</td>
    <td>The variant is associated with at least one phenotype or disease.</td>
  </tr>
  <!-- Separator -->
  <tr><td colspan="3" style="border-bottom:2px dotted #CCC;padding:0px"></td></tr>
  <tr>
    <td style="text-align:center"><img src="/i/val/evidence_1000Genomes.png" title="1000Genomes"/></td>
    <td>1000 Genomes</td>
    <td>The variant was discovered in the <a rel="external" href="http://www.1000genomes.org/">1000 Genomes Project</a> (human only).</td>
  </tr>
  <tr class="bg2">
    <td style="text-align:center"><img src="/i/val/evidence_gnomAD.png" title="gnomAD"/></td>
    <td>gnomAD</td>
    <td>The variant was discovered in the <a rel="external" href="http://gnomad.broadinstitute.org/">genome Aggregation Database</a> (human only).</td>
  </tr>
  <tr class="bg2">
    <td style="text-align:center"><img src="/i/val/evidence_TOPMed.png" title="TOPMed"/></td>
    <td>TOPMed</td>
    <td>The variant was discovered in the <a rel="external" href="https://www.nhlbi.nih.gov/research/resources/nhlbi-precision-medicine-initiative/topmed">Trans-Omics for Precision Medicine</a> program (human only).</td>
  </tr>
</table>


<br />
<hr style="margin-bottom:4px" />
<h2 id="quality_control">Quality control</h2>

<p>
A quality control process is employed to check imported variant data. Suspect variants and alleles are flagged, but are not withheld from downstream annotation. Data failing the checks is available through the browser where failure reasons are prominently listed. The API does not extract failed data by default, unless the database adaptor is specifically configured to do so using 
<a href ="/info/docs/Doxygen/variation-api/classBio_1_1EnsEMBL_1_1Variation_1_1DBSQL_1_1DBAdaptor.html#add8230ebca14116047cf7d17499f9e3b">Bio::EnsEMBL::Variation::DBSQL::DBAdaptor::include_failed_variations()</a>;
<p>
 Variants for which dbSNP holds citations from PubMed are not submitted to the QC process so are not flagged as failed.
</p>

<h4>Failure reasons</h4>
<table id="quality_control_table" class="ss">

  <tr class="bg2">
    <th>QC Type</th>
    <th>Reported failure reason</th>
    <th>Checking process</th>
  </tr>

  <tr>
    <td rowspan="4" style="background-color:#E5E5E5"><b>Mapping checks</b></td>
    <td>Variant does not map to the genome </td>
    <td>Variants with flanking sequences which do not map to reference or non-reference genomic sequences are flagged as failed.</td>
  </tr>
  <tr>
    <td class="bg2">Variant maps to more than 1 location</td>
    <td class="bg2">For variants with flanking sequences mapping to a reference sequence, the number of mappings within all reference sequences is counted and those mapping more than once are flagged as failed. (Variants with a single mapping to both X and Y within a PAR region are not failed.)
                     For variants with flanking sequences which do not map to a reference sequence, the number of mappings within all non-reference sequences is counted and those mapping more than once are flagged as failed. 
    </td>
  </tr>
  <tr>
    <td>Mapped position is not compatible with reported alleles </td>
    <td>The length of the reported alleles is compared to that expected given the coordinates specified for the variant. If none of the alleles match the expected length, the variant is flagged as failed. </td>
  </tr>
  <tr>
    <td class="bg2">None of the variant alleles match the reference allele </td>
    <td class="bg2"> The sequence at the coordinates specified for the variant are extracted from the reference genome and compared to the dbSNP refSNP alleles. If the extracted sequence does not match the expected alleles, the variant is flagged as failed.</td>
  </tr>
  <tr><td colspan="3" style="background-color:#888;height:1px;padding:0px;margin:0px"></td></tr>
  <tr>
    <td rowspan="3" style="background-color:#E5E5E5"><b>Checks on the alleles of refSNPs</b></td>
    <td>Loci with no observed variant alleles in dbSNP</td>
    <td>Variants with dbSNP refSNP alleles reported as 'NOVARIATION' are flagged as failed.</td>
  </tr>
  <tr>
    <td class="bg2">Alleles contain ambiguity codes</td>
    <td class="bg2">Variants with a IUPAC ambiguity code (eg.  M, Y, R, etc ) in the dbSNP refSNP alleles are reported as failed. </td>
  </tr>
  <tr>
    <td>Alleles contain non-nucleotide characters</td>
    <td>Variants with unexpected characters in the dbSNP refSNP alleles are reported as failed.</td>
  </tr>
  <tr><td colspan="3" style="background-color:#888;height:1px;padding:0px;margin:0px"></td></tr>
  <tr>
    <td rowspan="1" style="background-color:#E5E5E5"><b>Checks on the alleles in dbSNP submissions</b></td>
    <td class="bg2">Additional submitted allele data from dbSNP does not agree with the dbSNP refSNP alleles</td>
    <td class="bg2"> Alleles from all the dbSNP submissions for the rsID are checked against the dbSNP refSNP alleles. These alleles are primarily frequency submissions but can also be from variant discovery submissions, and these are merged in the dbSNP pipeline with the pre-existing refSNP variant).  Discrepant sets of alleles are flagged as failed as this will often highlight a strand error in the submission of frequency information for a known variant. The failure is flagged at the allele submission level.
    </td>
  </tr>
  <tr><td colspan="3" style="background-color:#888;height:1px;padding:0px;margin:0px"></td></tr>
  <tr>
    <td rowspan="1" style="background-color:#E5E5E5"><b>External failure classification</b></td>
    <td> Flagged as suspect by dbSNP </td>
    <td>Variants reported as being suspect by dbSNP due to being in probable paralogous regions are imported but flagged as failed (human only).</td>
  </tr>
  <tr><td colspan="3" style="background-color:#888;height:1px;padding:0px;margin:0px"></td></tr>
  <tr>
    <td rowspan="1" style="background-color:#E5E5E5"><b>New assembly</b></td>
    <td class="bg2">Variant can not be re-mapped to the current assembly</td>
    <td class="bg2">Variants that mapped to the previous assembly, but couldn't be remapped to the current assembly are flagged as failed.</td>
  </tr>
</table>


</body>
</html>
