<html>
<head>
<title>Haplosaurus</title>
<meta name="order" content="6" />
<script type="text/javascript">

  // Function to show/hide divs
  function show_hide (param) {
    div   = document.getElementById('div_'+param);
    alink = document.getElementById('a_'+param);
    if (div.style.display=='inline') {
      div.style.display='none';
      alink.innerHTML='Show';
    }
    else {
      if (div.style.display=='none') {
        div.style.display='inline';
        alink.innerHTML='Hide';
      }
    }
  }
</script>
</head>

<body>
<style>
tr:nth-child(odd) {background-color: #f0f0f0;}
</style>

<div>

  <h1 id="top" style="color:#006">Haplosaurus</h1>
  <hr/>


  <p><span style="color:#006;font-weight:bold">Haplosaurus (haplo)</span> is a local tool implementation of the same functionality that powers the <a href="/Homo_sapiens/Transcript/Haplotypes?t=ENST00000304748">Ensembl transcript haplotypes view</a>.</p>

  <p>It takes phased genotypes from a VCF and constructs a pair of haplotype sequences for each overlapped transcript; these sequences are also translated into predicted protein haplotype sequences. Each variant haplotype sequence is aligned and compared to the reference, and an HGVS-like name is constructed representing its differences to the reference.</p>

  <p>This approach offers an advantage over VEP's analysis, which treats each input variant independently. By considering the combined change contributed by all the variant alleles across a transcript, the compound effects the variants may have are correctly accounted for.</p>

  <p><b>haplo</b> shares much of the same command line functionality with vep, and can use VEP caches, Ensembl databases, GFF and GTF files as sources of transcript data; all vep command line flags relating to this functionality work the same with <b>haplo</b>.</p>


  <br />
  <h2 id="hp_dl_install">Download and install</h2>
  <p><b>Haplosaurus</b> is part of the <a href="https://github.com/Ensembl/ensembl-vep/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/haplo" rel="external">VEP package</a>.</p>
  <p>Please follow the instructions about the <img src="/i/16/download.png" style="vertical-align:middle"/> <a href="/info/docs/tools/vep/script/vep_download.html">download and installation of VEP</a>.</p>


  <br />
  <h2 id="hp_usage">Usage</h2>

  <p>Input data must be a <b>VCF containing phased genotype data for at least one individual</b> and file must be sorted by chromosome and genomic position; no other formats are currently supported.</p>
  <p>Example of VCF input:<p/>
  <pre class="code sh_sh">##fileformat=VCFv4.2
#CHROM  POS   ID           REF     ALT QUAL  FILTER  INFO  FORMAT  IND1  IND2
12  6029429   rs1800380    C       T   .     .       .     GT      1|0   1|0
12  6029431   rs370984712  G       A   .     .       .     GT      1|0   1|0
12  12477741  rs201941751  GCACGC  G   .     .       .     GT      0|1   1|0
12  12477747  rs200271649  TGGGC   T   .     .       .     GT      0|1   1|0
21  25597309  rs540774105  G       A   .     .       .     GT      0|0   0|0
21  25597391  rs1135618    T       C   .     .       .     GT      1|1   1|0
21  25606638  rs3989369    A       G   .     .       .     GT      0|1   0|0</pre>

  <p>When using a VEP cache as the source of transcript annotation, the first time you run haplo with a particular cache it will spend some time scanning transcript locations in the cache.</p>

  <pre class="code sh_sh">./haplo -i input.vcf -o out.txt -cache</pre>


  <br />
  <h2 id="hp_output">Output</h2>

  <p>The default output format is a simple tab-delimited file reporting all observed non-reference haplotypes. It has the following fields:</p>
  <ol>
    <li>Transcript stable ID</li>
    <li>CDS haplotype name</li>
    <li>Comma-separated list of flags for CDS haplotype</li>
    <li>Protein haplotype name</li>
    <li>Comma-separated list of flags for protein haplotype</li>
    <li>Comma-separated list of frequency data for protein haplotype</li>
    <li>Comma-separated list of contributing variants</li>
    <li>Comma-separated list of sample:count that exhibit this haplotype</li>
  </ol>

  <p>Example of outputs (default format), using the VCF data displayed above:</p>


<pre class="code sh_sh"># Input
#CHROM  POS   ID           REF     ALT QUAL  FILTER  INFO  FORMAT  IND1  IND2
12  6029429   rs1800380    C       T   .     .       .     GT      1|0   1|0
12  6029431   rs370984712  G       A   .     .       .     GT      1|0   1|0

# Output
ENST00000261405 | ENST00000261405:<b>2878C>T,2880G>A</b> | | ENSP00000261405:<b>960R>*,961del{1854}</b> | <b>stop_change</b> | | rs370984712,rs1800380 | IND1:1,IND2:1</pre>

<pre class="code sh_sh"># Input
#CHROM  POS   ID           REF     ALT QUAL  FILTER  INFO  FORMAT  IND1  IND2
12  12477741  rs201941751  GCACGC  G   .     .       .     GT      0|1   1|0
12  12477747  rs200271649  TGGGC   T   .     .       .     GT      0|1   1|0

# Output
ENST00000298573 | ENST00000298573:<b>1080del{4},1085delG,1087delGTG,1092delC</b> | <b>resolved_frameshift</b>,<b>indel</b> | ENSP00000298573:<b>364delPSV</b> | <b>indel</b> | | rs200271649,rs201941751 | IND1:1,IND2:1</pre>

<pre class="code sh_sh"># Input
#CHROM  POS   ID           REF     ALT QUAL  FILTER  INFO  FORMAT  IND1  IND2
21  25597391  rs1135618    T       C   .     .       .     GT      1|1   1|0
21  25606638  rs3989369    A       G   .     .       .     GT      0|1   0|0

# Output
ENST00000352957 | ENST00000352957:<b>91T>C,612A>G</b> | | ENSP00000284967:<b>31S>P</b> | | | rs3989369,rs1135618   | IND1:1<hr style="border-top:1px solid #CCC;height:1px;margin:5px 0px"/>ENST00000307301 | ENST00000307301:<b>612A>G</b>       | | ENSP00000305682:<b>REF</b>   | | | rs1135618             | IND1:1,IND2:1
ENST00000307301 | ENST00000307301:<b>91T>C,612A>G</b> | | ENSP00000305682:<b>31S>P</b> | | | rs3989369,rs1135618   | IND1:1<hr style="border-top:1px solid #CCC;height:1px;margin:5px 0px"/>ENST00000419219 | ENST00000419219:<b>582A>G</b>       | | ENSP00000404426:<b>REF</b>   | | | rs1135618             | IND1:1,IND2:1
ENST00000419219 | ENST00000419219:<b>91T>C,582A>G</b> | | ENSP00000404426:<b>31S>P</b> | | | rs3989369,rs1135618   | IND1:1</pre>

  <p>Alternatively, JSON output matching the format of the transcript haplotype REST endpoint may be generated by using <b>--json</b>.</p>

  <p>Each transcript analysed is summarised as a JSON object written to one line of the output file.</p>

  <p>You may exclude fields in the JSON from being exported with <b>--dont_export field1,field2</b>.<br />This may be used, for example, to exclude the full haplotype sequence and aligned sequences from the output with <b>--dont_export seq,aligned_sequences</b>.</p>

  <div>
    <div style="float:left" class="info">
      <h3 id="limitations">Note</h3>
      <div class="message-pad">
        <p>JSON output does not currently include side-loaded frequency data.</p>
      </div>
    </div>  
    <div class="clear"></div>
  </div>


  <br />
  <h2 id="hp_rest">REST API</h2>

  <p>The <a href="https://rest.ensembl.org/documentation/info/transcript_haplotypes_get">transcript haplotype REST endpoint</a> returns arrays of protein_haplotypes and cds_haplotypes for a given transcript.</p>
  <p>The default haplotype record includes:</p>
  <ul>
    <li><b>population_counts</b>: the number of times the haplotype is seen in each population</li>
    <li><b>population_frequencies</b>: the frequency of the haplotype  in each population</li>
    <li><b>contributing_variants</b>:  variants contributing to the haplotype</li>
    <li><b>diffs</b>: differences between the reference and this haplotype</li>
    <li><b>hex</b>: the md5 hex of this haplotype sequence</li>
    <li><b>other_hexes</b>: the md5 hex of other related haplotype sequences (CDSHaplotypes that translate to this ProteinHaplotype or ProteinHaplotype representing 
the translation of this CDSHaplotype)</li>
    <li><b>has_indel</b>: does the haplotype contain insertions or deletions</li>
    <li><b>type</b>: the type of haplotype - cds, protein</li>
    <li><b>name</b>: a human readable name for the haplotype (sequence id + REF or a change description)</li>
    <li><b>flags</b>: [flags](#haploflags) for the haplotype</li>
    <li><b>frequency</b>: haplotype frequency in full sample set</li>
    <li><b>count</b>: haplotype count in full sample set</li>
  </ul>

  <p>The REST service does not return raw sequences, sample-haplotype assignments and the aligned sequences used to generate
differences by default.</p>


  <br />
  <h2 id="hp_flags">Flags</h2>

  <p>Haplotypes may be flagged with one or more of the following:</p>

  <ul>
    <li><b>indel</b>: haplotype contains an insertion or deletion (indel) relative to the reference.</li>
    <li><b>frameshift</b>: haplotype contains at least one indel that disrupts the reading frame of the transcript.</li>
    <li><b>resolved_frameshift</b>: haplotype contains two or more indels whose combined effect restores the reading frame of the transcript.</li>
    <li><b>stop_changed</b>: indicates either a STOP codon is gained (protein truncating variant, PTV) or the existing reference STOP codon is lost.</li>
    <li><b>deleterious_sift_or_polyphen</b>: haplotype contains at least one single amino acid substitution event flagged as deleterious (SIFT) or probably damaging (PolyPhen-2).</li>
  </ul>


  <br />
  <h2 id="hp_bioperl_ext">bioperl-ext</h2>

  <p><b>haplo</b> can make use of a fast compiled alignment algorithm from the <a href="https://github.com/bioperl/bioperl-ext">bioperl-ext</a> package; this can speed up analysis, particularly in longer transcripts where insertions and/or deletions are introduced.</p>

  <p><b>The bioperl-ext package is no longer maintained and requires some tweaking to install.</b></p>

  <p>The following instructions install the package in <b>$HOME/perl5</b>; edit <b>PREFIX=[path]</b> to change this. You may also need to edit the <b>export</b> command to point to the path created for the architecture on your machine.</p>


  <pre class="code sh_sh">git clone https://github.com/bioperl/bioperl-ext.git
cd bioperl-ext/Bio/Ext/Align/
perl -pi -e"s|(cd libs.+)CFLAGS=\\\'|\$1CFLAGS=\\\'-fPIC |" Makefile.PL
perl Makefile.PL PREFIX=~/perl5
make
make install
cd -
export PERL5LIB=${PERL5LIB}:${HOME}/perl5/lib/x86_64-linux-gnu/perl/5.22.1/</pre>

  <p>If successful the following should print <b>OK</b>:</p>

  <pre class="code sh_sh">perl -MBio::Tools::dpAlign -e"print qq{OK\n}"</pre>

  <br />
  <h2 id="hp_pub">Publication</h2>

  <div>
    <p style="margin:0px">
      William Spooner, William McLaren, Timothy Slidel, Donna K. Finch, Robin Butler, Jamie Campbell, Laura Eghobamien, David Rider,<br />
      Christine Mione Kiefer, Matthew J. Robinson, Colin Hardman, Fiona Cunningham, Tristan Vaughan, Paul Flicek & Catherine Chaillan Huntington.
    </p>
    <h3 style="margin:4px 0px">Haplosaurus computes protein haplotypes for use in precision drug design.</h3>
    <div style="margin-bottom:4px"><em>Nature Communications</em> volume 9, Article number: 4128 (<strong>2018</strong>)</div>
    <p>
      <a rel="external" href="https://doi.org/10.1038/s41467-018-06542-1" rel="external">doi:10.1038/s41467-018-06542-1</a>
    </p>
  </div>

</div>

</body>
</html>
