
<html>
<head>

<title>Schema Documentation</title>
<meta name="order" content="2" />

<script language="Javascript" type="text/javascript">
  var img_plus   = '<img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/>';
  var img_minus  = '<img src="/i/16/minus-button.png" class="sql_schema_icon" alt="hide"/>';

  // Function to show/hide the columns table
  function show_hide (param, a_text) {
  
    // Schema tables
    if (a_text === 'columns') {
      div_id   = '#div_'+param;
      alink_id = '#a_'+param;
    }  
    // Species list
    else if (a_text === 'species') {
      div_id   = '#sp_'+param;
      alink_id = '#s_'+param;
    }
    // Example tables
    else {
      div_id   = '#ex_'+param;
      alink_id = '#e_'+param;
    }
    
    if ($(div_id).is(':visible')) {
      $(alink_id).html(img_plus+' Show '+a_text);
    }
    else if ($(div_id).is(':hidden')) {
      $(alink_id).html(img_minus+' Hide '+a_text);
    }
    $(div_id).slideToggle( 300 );
  }
  
  // Function to show/hide all the tables
  function show_hide_all (link_text) {
    expand_div = '#expand';
    div_prefix = 'div_';
    $("div[id^='"+div_prefix+"']").each(function() {
      param = $(this).attr('id').substring(div_prefix.length);
      div_id   = '#'+$(this).attr('id');
      alink_id = '#a_'+param;
      
      if ($(alink_id)) {
        if ($(expand_div).val()==0) {
          $(alink_id).html(img_minus+' Hide '+link_text);
        }
        else {
          $(alink_id).html(img_plus+' Show '+link_text);
        }
      }
      $(div_id).slideToggle( 500 );
    });
    if ($(expand_div).val()==0) {
      $(expand_div).val(1);
    }
    else {
      $(expand_div).val(0);
    }
  }
</script>
</head>
<body>

<h1>Ensembl Variation - Schema documentation</h1>

<p>
This document gives a high-level description of the tables that
make up the Ensembl variation schema. Tables are listed by alphabetical order, and the purpose of each table is explained. It is intended to
allow people to familiarise themselves with the schema when
encountering it for the first time, or when they need to use some
tables that they've not used before.
</p>

<div>
  <div style="float:left">
    <p>
      This document refers to version <span style="font-weight:bold;font-size:11pt;color:#000">86</span> of the Ensembl
      variation schema. 
    </p>
    <p>
    The variation database schema diagram (PDF format) is available here:
    </p>
  </div>
  <div style="float:left;margin-left:5px">
    <a href="variation-database-schema.pdf">
      <img src="/img/variation_schema_thumb.png" style="border:1px solid navy" alt="Variation Schema Diagram" title="Variation Schema Diagram (PDF)"/>
    </a>
  </div>
  <div style="clear:both">
</div>
<br /><br />


<h3 id="top">List of the tables:</h3>
<div>

  <div class="sql_schema_table_group">
    <div class="sql_schema_table_group_header" style="border-color:#FF0000">
      <div class="sql_schema_table_group_bullet" style="background-color:#FF0000"></div>
      <h2>Attributes tables</h2>
    </div>
      <ul class="sql_schema_table_list">
        <li><a href="#attrib" class="sql_schema_link" style="font-weight:bold">attrib</a></li>
        <li><a href="#attrib_set" class="sql_schema_link" style="font-weight:bold">attrib_set</a></li>
        <li><a href="#attrib_type" class="sql_schema_link" style="font-weight:bold">attrib_type</a></li>

      </ul>
    </div>

  <div class="sql_schema_table_group">
    <div class="sql_schema_table_group_header" style="border-color:#3CB371">
      <div class="sql_schema_table_group_bullet" style="background-color:#3CB371"></div>
      <h2>Failed tables</h2>
    </div>
      <ul class="sql_schema_table_list">
        <li><a href="#failed_allele" class="sql_schema_link" style="font-weight:bold">failed_allele</a></li>
        <li><a href="#failed_description" class="sql_schema_link" style="font-weight:bold">failed_description</a></li>
        <li><a href="#failed_structural_variation" class="sql_schema_link" style="font-weight:bold">failed_structural_variation</a></li>
        <li><a href="#failed_variation" class="sql_schema_link" style="font-weight:bold">failed_variation</a></li>

      </ul>
    </div>

  <div class="sql_schema_table_group">
    <div class="sql_schema_table_group_header" style="border-color:#FF8500">
      <div class="sql_schema_table_group_bullet" style="background-color:#FF8500"></div>
      <h2>Genotype tables</h2>
    </div>
      <ul class="sql_schema_table_list">
        <li><a href="#compressed_genotype_region" class="sql_schema_link" style="font-weight:bold">compressed_genotype_region</a></li>
        <li><a href="#compressed_genotype_var" class="sql_schema_link" style="font-weight:bold">compressed_genotype_var</a></li>
        <li><a href="#population_genotype" class="sql_schema_link" style="font-weight:bold">population_genotype</a></li>
        <li><a href="#read_coverage" class="sql_schema_link" style="font-weight:bold">read_coverage</a></li>
        <li><a href="#sample_genotype_multiple_bp" class="sql_schema_link" style="font-weight:bold">sample_genotype_multiple_bp</a></li>
        <li><a href="#tmp_sample_genotype_single_bp" class="sql_schema_link" style="font-weight:bold">tmp_sample_genotype_single_bp</a></li>

      </ul>
    </div>

  <div class="sql_schema_table_group">
    <div class="sql_schema_table_group_header" style="border-color:#BC5CEC">
      <div class="sql_schema_table_group_bullet" style="background-color:#BC5CEC"></div>
      <h2>Metadata tables</h2>
    </div>
      <ul class="sql_schema_table_list">
        <li><a href="#meta" class="sql_schema_link" style="font-weight:bold">meta</a></li>
        <li><a href="#meta_coord" class="sql_schema_link" style="font-weight:bold">meta_coord</a></li>

      </ul>
    </div>
  <div style="clear:both"></div>
</div>

<div>
  <div class="sql_schema_table_group">
    <div class="sql_schema_table_group_header" style="border-color:#000">
      <div class="sql_schema_table_group_bullet" style="background-color:#000"></div>
      <h2>Other tables</h2>
    </div>
      <ul class="sql_schema_table_list">
        <li><a href="#allele_code" class="sql_schema_link" style="font-weight:bold">allele_code</a></li>
        <li><a href="#coord_system" class="sql_schema_link" style="font-weight:bold">coord_system</a></li>
        <li><a href="#genotype_code" class="sql_schema_link" style="font-weight:bold">genotype_code</a></li>
        <li><a href="#seq_region" class="sql_schema_link" style="font-weight:bold">seq_region</a></li>
        <li><a href="#submitter_handle" class="sql_schema_link" style="font-weight:bold">submitter_handle</a></li>
        <li><a href="#subsnp_handle" class="sql_schema_link" style="font-weight:bold">subsnp_handle</a></li>

      </ul>
    </div>

  <div class="sql_schema_table_group">
    <div class="sql_schema_table_group_header" style="border-color:#22949b">
      <div class="sql_schema_table_group_bullet" style="background-color:#22949b"></div>
      <h2>Phenotype tables</h2>
    </div>
      <ul class="sql_schema_table_list">
        <li><a href="#phenotype" class="sql_schema_link" style="font-weight:bold">phenotype</a></li>
        <li><a href="#phenotype_feature" class="sql_schema_link" style="font-weight:bold">phenotype_feature</a></li>
        <li><a href="#phenotype_feature_attrib" class="sql_schema_link" style="font-weight:bold">phenotype_feature_attrib</a></li>
        <li><a href="#phenotype_ontology_accession" class="sql_schema_link" style="font-weight:bold">phenotype_ontology_accession</a></li>

      </ul>
    </div>

  <div class="sql_schema_table_group">
    <div class="sql_schema_table_group_header" style="border-color:#1E90FF">
      <div class="sql_schema_table_group_bullet" style="background-color:#1E90FF"></div>
      <h2>Protein tables</h2>
    </div>
      <ul class="sql_schema_table_list">
        <li><a href="#protein_function_predictions" class="sql_schema_link" style="font-weight:bold">protein_function_predictions</a></li>
        <li><a href="#protein_function_predictions_attrib" class="sql_schema_link" style="font-weight:bold">protein_function_predictions_attrib</a></li>
        <li><a href="#translation_md5" class="sql_schema_link" style="font-weight:bold">translation_md5</a></li>

      </ul>
    </div>

  <div class="sql_schema_table_group">
    <div class="sql_schema_table_group_header" style="border-color:#FF8500">
      <div class="sql_schema_table_group_bullet" style="background-color:#FF8500"></div>
      <h2>Sample tables</h2>
    </div>
      <ul class="sql_schema_table_list">
        <li><a href="#display_group" class="sql_schema_link" style="font-weight:bold">display_group</a></li>
        <li><a href="#individual" class="sql_schema_link" style="font-weight:bold">individual</a></li>
        <li><a href="#individual_synonym" class="sql_schema_link" style="font-weight:bold">individual_synonym</a></li>
        <li><a href="#individual_type" class="sql_schema_link" style="font-weight:bold">individual_type</a></li>
        <li><a href="#population" class="sql_schema_link" style="font-weight:bold">population</a></li>
        <li><a href="#population_structure" class="sql_schema_link" style="font-weight:bold">population_structure</a></li>
        <li><a href="#population_synonym" class="sql_schema_link" style="font-weight:bold">population_synonym</a></li>
        <li><a href="#sample" class="sql_schema_link" style="font-weight:bold">sample</a></li>
        <li><a href="#sample_population" class="sql_schema_link" style="font-weight:bold">sample_population</a></li>
        <li><a href="#sample_synonym" class="sql_schema_link" style="font-weight:bold">sample_synonym</a></li>

      </ul>
    </div>
  <div style="clear:both"></div>
</div>

<div>
  <div class="sql_schema_table_group">
    <div class="sql_schema_table_group_header" style="border-color:#7CFC00">
      <div class="sql_schema_table_group_bullet" style="background-color:#7CFC00"></div>
      <h2>Source/study tables</h2>
    </div>
      <ul class="sql_schema_table_list">
        <li><a href="#associate_study" class="sql_schema_link" style="font-weight:bold">associate_study</a></li>
        <li><a href="#publication" class="sql_schema_link" style="font-weight:bold">publication</a></li>
        <li><a href="#source" class="sql_schema_link" style="font-weight:bold">source</a></li>
        <li><a href="#study" class="sql_schema_link" style="font-weight:bold">study</a></li>
        <li><a href="#variation_citation" class="sql_schema_link" style="font-weight:bold">variation_citation</a></li>

      </ul>
    </div>

  <div class="sql_schema_table_group">
    <div class="sql_schema_table_group_header" style="border-color:#01D4F7">
      <div class="sql_schema_table_group_bullet" style="background-color:#01D4F7"></div>
      <h2>Structural variation tables</h2>
    </div>
      <ul class="sql_schema_table_list">
        <li><a href="#structural_variation" class="sql_schema_link" style="font-weight:bold">structural_variation</a></li>
        <li><a href="#structural_variation_association" class="sql_schema_link" style="font-weight:bold">structural_variation_association</a></li>
        <li><a href="#structural_variation_feature" class="sql_schema_link" style="font-weight:bold">structural_variation_feature</a></li>
        <li><a href="#structural_variation_sample" class="sql_schema_link" style="font-weight:bold">structural_variation_sample</a></li>

      </ul>
    </div>

  <div class="sql_schema_table_group">
    <div class="sql_schema_table_group_header" style="border-color:#FF4DC8">
      <div class="sql_schema_table_group_bullet" style="background-color:#FF4DC8"></div>
      <h2>Variation effect tables</h2>
    </div>
      <ul class="sql_schema_table_list">
        <li><a href="#motif_feature_variation" class="sql_schema_link" style="font-weight:bold">motif_feature_variation</a></li>
        <li><a href="#regulatory_feature_variation" class="sql_schema_link" style="font-weight:bold">regulatory_feature_variation</a></li>
        <li><a href="#transcript_variation" class="sql_schema_link" style="font-weight:bold">transcript_variation</a></li>
        <li><a href="#variation_genename" class="sql_schema_link" style="font-weight:bold">variation_genename</a></li>
        <li><a href="#variation_hgvs" class="sql_schema_link" style="font-weight:bold">variation_hgvs</a></li>

      </ul>
    </div>

  <div class="sql_schema_table_group">
    <div class="sql_schema_table_group_header" style="border-color:#FFD700">
      <div class="sql_schema_table_group_bullet" style="background-color:#FFD700"></div>
      <h2>Variation set tables</h2>
    </div>
      <ul class="sql_schema_table_list">
        <li><a href="#variation_set" class="sql_schema_link" style="font-weight:bold">variation_set</a></li>
        <li><a href="#variation_set_structural_variation" class="sql_schema_link" style="font-weight:bold">variation_set_structural_variation</a></li>
        <li><a href="#variation_set_structure" class="sql_schema_link" style="font-weight:bold">variation_set_structure</a></li>
        <li><a href="#variation_set_variation" class="sql_schema_link" style="font-weight:bold">variation_set_variation</a></li>

      </ul>
    </div>
  <div style="clear:both"></div>
</div>

<div>
  <div class="sql_schema_table_group">
    <div class="sql_schema_table_group_header" style="border-color:#000">
      <div class="sql_schema_table_group_bullet" style="background-color:#000"></div>
      <h2>Variation tables</h2>
    </div>
      <ul class="sql_schema_table_list">
        <li><a href="#allele" class="sql_schema_link" style="font-weight:bold">allele</a></li>
        <li><a href="#variation" class="sql_schema_link" style="font-weight:bold">variation</a></li>
        <li><a href="#variation_attrib" class="sql_schema_link" style="font-weight:bold">variation_attrib</a></li>
        <li><a href="#variation_feature" class="sql_schema_link" style="font-weight:bold">variation_feature</a></li>
        <li><a href="#variation_synonym" class="sql_schema_link" style="font-weight:bold">variation_synonym</a></li>

      </ul>
    </div>

  <div style="clear:both" />
</div>
  <input type="button" onclick="show_hide_all('columns')" class="fbutton" value="Show/hide all"/>
  <input type="hidden" id="expand" value="0" />
  

<div class="sql_schema_group_header" style="border-color:#FF0000">
  <div id="1" class="sql_schema_group_bullet" style="background-color:#FF0000"></div>
  <h2 id="1">Attributes tables</h2>
</div>
<p class="sql_schema_group_header_desc">These tables define the variation attributes data.</p><div class="sql_schema_table">
  <div id="attrib"class="sql_schema_table_header" style="border-top-color:#FF0000">
    <div class="sql_schema_table_header_left"><span style="background-color:#FF0000"></span>attrib</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_attrib" class="help-header sql_schema_show_hide" onclick="show_hide('attrib','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">Defines various attributes used elsewhere in the database</p>

  <div id="div_attrib" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>attrib_id</b></td>
        <td class="bg1">INT(11)</td>
        <td class="bg1">-</td>
        <td class="bg1">Primary key</td>
        <td class="bg1">primary key</td>
      </tr>

      <tr>
        <td class="bg2"><b>attrib_type_id</b></td>
        <td class="bg2">SMALLINT(5)</td>
        <td class="bg2">0</td>
        <td class="bg2">Key into the <a href="#attrib_type" class="sql_schema_link">attrib_type</a> table, identifies the type of this attribute</td>
        <td class="bg2">unique key: <i>type_val_idx</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>value</b></td>
        <td class="bg1">TEXT</td>
        <td class="bg1">-</td>
        <td class="bg1">The value of this attribute</td>
        <td class="bg1">unique key: <i>type_val_idx</i></td>
      </tr>
    </table>
  </div>
<div class="sql_schema_table_examples"><p class="sql_schema_table_example_header">Example:</p><div class="sql_schema_table_example_content"><p>See below the query to display a subset of the attrib entries: </p>
      <div>
        <div class="sql_schema_table_example_query">
          <pre><span class="sql_schema_sql_highlight">SELECT</span> * <span class="sql_schema_sql_highlight">FROM</span> attrib <span class="sql_schema_sql_highlight">WHERE</span> attrib_type_id <span class="sql_schema_sql_highlight">IN</span> (469,470,471) <span class="sql_schema_sql_highlight">ORDER</span> <span class="sql_schema_sql_highlight">BY</span> attrib_id <span class="sql_schema_sql_highlight">LIMIT</span> 21;</pre>
        </div>
        <div class="sql_schema_table_example_button">
  <a id="e_attrib" class="help-header sql_schema_show_hide" onclick="show_hide('attrib','query results')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show query results
  </a></div>
        <div style="clear:both"></div>
      </div>
      
  <div id="ex_attrib" style="display:none;">
    <table class="ss sql_schema_table_example_result">
      <tr><th>attrib_id</th><th>attrib_type_id</th><th>value</th></tr>      <tr><td>1</td><td>469</td><td>SO:0001483</td></tr>      <tr class="bg2"><td>2</td><td>470</td><td>SNV</td></tr>      <tr><td>3</td><td>471</td><td>SNP</td></tr>      <tr class="bg2"><td>4</td><td>469</td><td>SO:1000002</td></tr>      <tr><td>5</td><td>470</td><td>substitution</td></tr>      <tr class="bg2"><td>6</td><td>469</td><td>SO:0001019</td></tr>      <tr><td>7</td><td>470</td><td>copy_number_variation</td></tr>      <tr class="bg2"><td>8</td><td>471</td><td>CNV</td></tr>      <tr><td>9</td><td>469</td><td>SO:0000667</td></tr>      <tr class="bg2"><td>10</td><td>470</td><td>insertion</td></tr>      <tr><td>11</td><td>469</td><td>SO:0000159</td></tr>      <tr class="bg2"><td>12</td><td>470</td><td>deletion</td></tr>      <tr><td>13</td><td>469</td><td>SO:1000032</td></tr>      <tr class="bg2"><td>14</td><td>470</td><td>indel</td></tr>      <tr><td>15</td><td>469</td><td>SO:0000705</td></tr>      <tr class="bg2"><td>16</td><td>470</td><td>tandem_repeat</td></tr>      <tr><td>17</td><td>469</td><td>SO:0001059</td></tr>      <tr class="bg2"><td>18</td><td>470</td><td>sequence_alteration</td></tr>      <tr><td>19</td><td>469</td><td>SO:0001628</td></tr>      <tr class="bg2"><td>20</td><td>470</td><td>intergenic_variant</td></tr>      <tr><td>21</td><td>471</td><td>INTERGENIC</td></tr>    </table>
  </div></div></div><table class="sql_schema_extra"><tr>    <td class="sql_schema_extra_left"">    <p style="font-weight:bold">See also:</p>
  <ul>
      <li><a href="#attrib_type">attrib_type</a></li>
      <li><a href="#attrib_set">attrib_set</a></li>
      </ul>
    </td>
  <td class="sql_schema_extra_separator"></td>
  <td class="sql_schema_extra_right" style="padding-left:25px"><p><span>List of species with populated data:</span>
  <a id="s_attrib" class="help-header sql_schema_show_hide" onclick="show_hide('attrib','species')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show species
  </a></p>
    <div id="sp_attrib" style="display:none;">      <ul>
        <li class="sql_schema_species_name">Bos taurus</li>
        <li class="sql_schema_species_name">Canis familiaris</li>
        <li class="sql_schema_species_name">Danio rerio</li>
        <li class="sql_schema_species_name">Drosophila melanogaster</li>
        <li class="sql_schema_species_name">Equus caballus</li>
        <li class="sql_schema_species_name">Felis catus</li>
        <li class="sql_schema_species_name">Gallus gallus</li>
        <li class="sql_schema_species_name">Homo sapiens</li>
        <li class="sql_schema_species_name">Macaca mulatta</li>
        <li class="sql_schema_species_name">Meleagris gallopavo</li>
        <li class="sql_schema_species_name">Monodelphis domestica</li>
        <li class="sql_schema_species_name">Mus musculus</li>
        <li class="sql_schema_species_name">Nomascus leucogenys</li>
        <li class="sql_schema_species_name">Ornithorhynchus anatinus</li>
        <li class="sql_schema_species_name">Ovis aries</li>
        <li class="sql_schema_species_name">Pan troglodytes</li>
        <li class="sql_schema_species_name">Pongo abelii</li>
        <li class="sql_schema_species_name">Rattus norvegicus</li>
        <li class="sql_schema_species_name">Saccharomyces cerevisiae</li>
        <li class="sql_schema_species_name">Sus scrofa</li>
        <li class="sql_schema_species_name">Taeniopygia guttata</li>
        <li class="sql_schema_species_name">Tetraodon nigroviridis</li>
      </ul>
    </div>
  </td></tr></table></div></div><div class="sql_schema_table">
  <div id="attrib_set"class="sql_schema_table_header" style="border-top-color:#FF0000">
    <div class="sql_schema_table_header_left"><span style="background-color:#FF0000"></span>attrib_set</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_attrib_set" class="help-header sql_schema_show_hide" onclick="show_hide('attrib_set','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">Groups related attributes together</p>

  <div id="div_attrib_set" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>attrib_set_id</b></td>
        <td class="bg1">INT(11)</td>
        <td class="bg1">0</td>
        <td class="bg1">Primary key</td>
        <td class="bg1">unique key: <i>set_idx</i></td>
      </tr>

      <tr>
        <td class="bg2"><b>attrib_id</b></td>
        <td class="bg2">INT(11)</td>
        <td class="bg2">0</td>
        <td class="bg2">Key of an attribute in this set</td>
        <td class="bg2">unique key: <i>set_idx</i><br />key: <i>attrib_idx</i></td>
      </tr>
    </table>
  </div>
<table class="sql_schema_extra"><tr>    <td class="sql_schema_extra_left"">    <p style="font-weight:bold">See also:</p>
  <ul>
      <li><a href="#attrib">attrib</a></li>
      <li><a href="#attrib_type">attrib_type</a></li>
      </ul>
    </td>
  <td class="sql_schema_extra_separator"></td>
  <td class="sql_schema_extra_right" style="padding-left:25px"><p><span>List of species with populated data:</span>
  <a id="s_attrib_set" class="help-header sql_schema_show_hide" onclick="show_hide('attrib_set','species')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show species
  </a></p>
    <div id="sp_attrib_set" style="display:none;">      <ul>
        <li class="sql_schema_species_name">Bos taurus</li>
        <li class="sql_schema_species_name">Canis familiaris</li>
        <li class="sql_schema_species_name">Danio rerio</li>
        <li class="sql_schema_species_name">Drosophila melanogaster</li>
        <li class="sql_schema_species_name">Equus caballus</li>
        <li class="sql_schema_species_name">Felis catus</li>
        <li class="sql_schema_species_name">Gallus gallus</li>
        <li class="sql_schema_species_name">Homo sapiens</li>
        <li class="sql_schema_species_name">Macaca mulatta</li>
        <li class="sql_schema_species_name">Meleagris gallopavo</li>
        <li class="sql_schema_species_name">Monodelphis domestica</li>
        <li class="sql_schema_species_name">Mus musculus</li>
        <li class="sql_schema_species_name">Nomascus leucogenys</li>
        <li class="sql_schema_species_name">Ornithorhynchus anatinus</li>
        <li class="sql_schema_species_name">Ovis aries</li>
        <li class="sql_schema_species_name">Pan troglodytes</li>
        <li class="sql_schema_species_name">Pongo abelii</li>
        <li class="sql_schema_species_name">Rattus norvegicus</li>
        <li class="sql_schema_species_name">Saccharomyces cerevisiae</li>
        <li class="sql_schema_species_name">Sus scrofa</li>
        <li class="sql_schema_species_name">Taeniopygia guttata</li>
        <li class="sql_schema_species_name">Tetraodon nigroviridis</li>
      </ul>
    </div>
  </td></tr></table></div></div><div class="sql_schema_table">
  <div id="attrib_type"class="sql_schema_table_header" style="border-top-color:#FF0000">
    <div class="sql_schema_table_header_left"><span style="background-color:#FF0000"></span>attrib_type</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_attrib_type" class="help-header sql_schema_show_hide" onclick="show_hide('attrib_type','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">Defines the set of possible attribute types used in the attrib table</p>

  <div id="div_attrib_type" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>attrib_type_id</b></td>
        <td class="bg1">SMALLINT(5)</td>
        <td class="bg1">0</td>
        <td class="bg1">Primary key</td>
        <td class="bg1">primary key</td>
      </tr>

      <tr>
        <td class="bg2"><b>code</b></td>
        <td class="bg2">VARCHAR(20)</td>
        <td class="bg2">''</td>
        <td class="bg2">A short codename for this type (indexed, so should be used for lookups)</td>
        <td class="bg2">unique key: <i>code_idx</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>name</b></td>
        <td class="bg1">VARCHAR(255)</td>
        <td class="bg1">''</td>
        <td class="bg1">The name of this type</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>description</b></td>
        <td class="bg2">TEXT</td>
        <td class="bg2">NULL</td>
        <td class="bg2">Longer description of this type</td>
        <td class="bg2"></td>
      </tr>
    </table>
  </div>
<div class="sql_schema_table_examples"><p class="sql_schema_table_example_header">Example:</p><div class="sql_schema_table_example_content"><p>See below the command to display a subset of the attrib_type entries: </p>
      <div>
        <div class="sql_schema_table_example_query">
          <pre><span class="sql_schema_sql_highlight">SELECT</span> * <span class="sql_schema_sql_highlight">FROM</span> attrib_type <span class="sql_schema_sql_highlight">WHERE</span> attrib_type_id > 468 <span class="sql_schema_sql_highlight">LIMIT</span> 10;</pre>
        </div>
        <div class="sql_schema_table_example_button">
  <a id="e_attrib_type" class="help-header sql_schema_show_hide" onclick="show_hide('attrib_type','query results')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show query results
  </a></div>
        <div style="clear:both"></div>
      </div>
      
  <div id="ex_attrib_type" style="display:none;">
    <table class="ss sql_schema_table_example_result">
      <tr><th>attrib_type_id</th><th>code</th><th>name</th><th>description</th></tr>      <tr><td>469</td><td>SO_accession</td><td>SO accession</td><td>Sequence Ontology accession</td></tr>      <tr class="bg2"><td>470</td><td>SO_term</td><td>SO term</td><td>Sequence Ontology term</td></tr>      <tr><td>471</td><td>display_term</td><td>display term</td><td>Ensembl display term</td></tr>      <tr class="bg2"><td>472</td><td>NCBI_term</td><td>NCBI term</td><td>NCBI term</td></tr>      <tr><td>473</td><td>feature_SO_term</td><td>feature SO term</td><td>Sequence Ontology term for the associated feature</td></tr>      <tr class="bg2"><td>474</td><td>rank</td><td>rank</td><td>Relative severity of this variation consequence</td></tr>      <tr><td>475</td><td>polyphen_prediction</td><td>polyphen prediction</td><td>PolyPhen-2 prediction</td></tr>      <tr class="bg2"><td>476</td><td>sift_prediction</td><td>sift prediction</td><td>SIFT prediction</td></tr>      <tr><td>477</td><td>short_name</td><td>Short name</td><td>A shorter name for an instance, e.g. a VariationSet</td></tr>      <tr class="bg2"><td>478</td><td>dbsnp_clin_sig</td><td>dbSNP/ClinVar clinical significance</td><td>The clinical significance of a variant as reported by ClinVar and dbSNP</td></tr>    </table>
  </div></div></div><table class="sql_schema_extra"><tr>    <td class="sql_schema_extra_left"">    <p style="font-weight:bold">See also:</p>
  <ul>
      <li><a href="#attrib">attrib</a></li>
      <li><a href="#attrib_set">attrib_set</a></li>
      </ul>
    </td>
  <td class="sql_schema_extra_separator"></td>
  <td class="sql_schema_extra_right" style="padding-left:25px"><p><span>List of species with populated data:</span>
  <a id="s_attrib_type" class="help-header sql_schema_show_hide" onclick="show_hide('attrib_type','species')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show species
  </a></p>
    <div id="sp_attrib_type" style="display:none;">      <ul>
        <li class="sql_schema_species_name">Bos taurus</li>
        <li class="sql_schema_species_name">Canis familiaris</li>
        <li class="sql_schema_species_name">Danio rerio</li>
        <li class="sql_schema_species_name">Drosophila melanogaster</li>
        <li class="sql_schema_species_name">Equus caballus</li>
        <li class="sql_schema_species_name">Felis catus</li>
        <li class="sql_schema_species_name">Gallus gallus</li>
        <li class="sql_schema_species_name">Homo sapiens</li>
        <li class="sql_schema_species_name">Macaca mulatta</li>
        <li class="sql_schema_species_name">Meleagris gallopavo</li>
        <li class="sql_schema_species_name">Monodelphis domestica</li>
        <li class="sql_schema_species_name">Mus musculus</li>
        <li class="sql_schema_species_name">Nomascus leucogenys</li>
        <li class="sql_schema_species_name">Ornithorhynchus anatinus</li>
        <li class="sql_schema_species_name">Ovis aries</li>
        <li class="sql_schema_species_name">Pan troglodytes</li>
        <li class="sql_schema_species_name">Pongo abelii</li>
        <li class="sql_schema_species_name">Rattus norvegicus</li>
        <li class="sql_schema_species_name">Saccharomyces cerevisiae</li>
        <li class="sql_schema_species_name">Sus scrofa</li>
        <li class="sql_schema_species_name">Taeniopygia guttata</li>
        <li class="sql_schema_species_name">Tetraodon nigroviridis</li>
      </ul>
    </div>
  </td></tr></table></div></div>

<div class="sql_schema_group_header" style="border-color:#3CB371">
  <div id="2" class="sql_schema_group_bullet" style="background-color:#3CB371"></div>
  <h2 id="2">Failed tables</h2>
</div>
<p class="sql_schema_group_header_desc">These tables define the list of variants/alleles flagged as "failed" in the Variation pipeline.<br /> The list of reasons for a variation being flagged as failed is available in the <a href="/info/genome/variation/data_description.html#quality_control">Quality Control documentation</a>.</p><div class="sql_schema_table">
  <div id="failed_allele"class="sql_schema_table_header" style="border-top-color:#3CB371">
    <div class="sql_schema_table_header_left"><span style="background-color:#3CB371"></span>failed_allele</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_failed_allele" class="help-header sql_schema_show_hide" onclick="show_hide('failed_allele','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">Contains alleles that did not pass the Ensembl filters</p>

  <div id="div_failed_allele" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>failed_allele_id</b></td>
        <td class="bg1">int(11)</td>
        <td class="bg1">-</td>
        <td class="bg1">Primary key, internal identifier.</td>
        <td class="bg1">primary key</td>
      </tr>

      <tr>
        <td class="bg2"><b>allele_id</b></td>
        <td class="bg2">int(10)</td>
        <td class="bg2">-</td>
        <td class="bg2">Foreign key references to the <a href="#allele" class="sql_schema_link">allele</a> table.</td>
        <td class="bg2">unique key: <i>allele_idx</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>failed_description_id</b></td>
        <td class="bg1">int(10)</td>
        <td class="bg1">-</td>
        <td class="bg1">Foreign key references to the <a href="#failed_description" class="sql_schema_link">failed_description</a> table.</td>
        <td class="bg1">unique key: <i>allele_idx</i></td>
      </tr>
    </table>
  </div>
<table class="sql_schema_extra"><tr>    <td class="sql_schema_extra_left"">    <p style="font-weight:bold">See also:</p>
  <ul>
      <li><a href="#failed_description">failed_description</a></li>
      <li><a href="#allele">allele</a></li>
      </ul>
    </td>
  <td class="sql_schema_extra_separator"></td>
  <td class="sql_schema_extra_right" style="padding-left:25px"><p><span>List of species with populated data:</span>
  <a id="s_failed_allele" class="help-header sql_schema_show_hide" onclick="show_hide('failed_allele','species')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show species
  </a></p>
    <div id="sp_failed_allele" style="display:none;">      <ul>
        <li class="sql_schema_species_name">Bos taurus</li>
        <li class="sql_schema_species_name">Canis familiaris</li>
        <li class="sql_schema_species_name">Danio rerio</li>
        <li class="sql_schema_species_name">Equus caballus</li>
        <li class="sql_schema_species_name">Felis catus</li>
        <li class="sql_schema_species_name">Gallus gallus</li>
        <li class="sql_schema_species_name">Homo sapiens</li>
        <li class="sql_schema_species_name">Macaca mulatta</li>
        <li class="sql_schema_species_name">Meleagris gallopavo</li>
        <li class="sql_schema_species_name">Mus musculus</li>
        <li class="sql_schema_species_name">Ovis aries</li>
        <li class="sql_schema_species_name">Pan troglodytes</li>
        <li class="sql_schema_species_name">Rattus norvegicus</li>
        <li class="sql_schema_species_name">Sus scrofa</li>
      </ul>
    </div>
  </td></tr></table></div></div><div class="sql_schema_table">
  <div id="failed_description"class="sql_schema_table_header" style="border-top-color:#3CB371">
    <div class="sql_schema_table_header_left"><span style="background-color:#3CB371"></span>failed_description</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_failed_description" class="help-header sql_schema_show_hide" onclick="show_hide('failed_description','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">This table contains descriptions of reasons for a variation being flagged as failed.</p>

  <div id="div_failed_description" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>failed_description_id</b></td>
        <td class="bg1">int(10)</td>
        <td class="bg1">-</td>
        <td class="bg1">Primary key, internal identifier.</td>
        <td class="bg1">primary key</td>
      </tr>

      <tr>
        <td class="bg2"><b>description</b></td>
        <td class="bg2">text</td>
        <td class="bg2">-</td>
        <td class="bg2">Text containing the reason why the Variation has been flagged as failed. e.g. "Variation does not map to the genome".</td>
        <td class="bg2"></td>
      </tr>
    </table>
  </div>
<div class="sql_schema_table_examples"><p class="sql_schema_table_example_header">Example:</p><div class="sql_schema_table_example_content"><p>See below the list of the descriptions available in the Ensembl variation databases: </p>
      <div>
        <div class="sql_schema_table_example_query">
          <pre><span class="sql_schema_sql_highlight">SELECT</span> * <span class="sql_schema_sql_highlight">FROM</span> failed_description;</pre>
        </div>
        <div class="sql_schema_table_example_button">
  <a id="e_failed_description" class="help-header sql_schema_show_hide" onclick="show_hide('failed_description','query results')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show query results
  </a></div>
        <div style="clear:both"></div>
      </div>
      
  <div id="ex_failed_description" style="display:none;">
    <table class="ss sql_schema_table_example_result">
      <tr><th>failed_description_id</th><th>description</th></tr>      <tr><td>1</td><td>Variant maps to more than 3 different locations</td></tr>      <tr class="bg2"><td>2</td><td>None of the variant alleles match the reference allele</td></tr>      <tr><td>3</td><td>Variant has more than 3 different alleles</td></tr>      <tr class="bg2"><td>4</td><td>Loci with no observed variant alleles in dbSNP</td></tr>      <tr><td>5</td><td>Variant does not map to the genome</td></tr>      <tr class="bg2"><td>6</td><td>Variant has no genotypes</td></tr>      <tr><td>7</td><td>Genotype frequencies do not add up to 1</td></tr>      <tr class="bg2"><td>8</td><td>Variant has no associated sequence</td></tr>      <tr><td>9</td><td>Variant submission has been withdrawn by the 1000 genomes project due to high false positive rate</td></tr>      <tr class="bg2"><td>11</td><td>Additional submitted allele data from dbSNP does not agree with the dbSNP refSNP alleles</td></tr>      <tr><td>12</td><td>Variant has more than 3 different submitted alleles</td></tr>      <tr class="bg2"><td>13</td><td>Alleles contain non-nucleotide characters</td></tr>      <tr><td>14</td><td>Alleles contain ambiguity codes</td></tr>      <tr class="bg2"><td>15</td><td>Mapped position is not compatible with reported alleles</td></tr>      <tr><td>16</td><td>Flagged as suspect by dbSNP</td></tr>      <tr class="bg2"><td>17</td><td>Variant can not be re-mapped to the current assembly</td></tr>      <tr><td>18</td><td>Supporting evidence can not be re-mapped to the current assembly</td></tr>      <tr class="bg2"><td>19</td><td>Variant maps to more than one genomic location</td></tr>      <tr><td>20</td><td>Variant at first base in sequence</td></tr>    </table>
  </div></div></div><table class="sql_schema_extra"><tr>    <td class="sql_schema_extra_left"">    <p style="font-weight:bold">See also:</p>
  <ul>
      <li><a href="#failed_variation">failed_variation</a></li>
      <li><a href="#failed_allele">failed_allele</a></li>
      <li><a href="#failed_structural_variation">failed_structural_variation</a></li>
      </ul>
    </td>
  <td class="sql_schema_extra_separator"></td>
  <td class="sql_schema_extra_right" style="padding-left:25px"><p><span>List of species with populated data:</span>
  <a id="s_failed_description" class="help-header sql_schema_show_hide" onclick="show_hide('failed_description','species')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show species
  </a></p>
    <div id="sp_failed_description" style="display:none;">      <ul>
        <li class="sql_schema_species_name">Bos taurus</li>
        <li class="sql_schema_species_name">Canis familiaris</li>
        <li class="sql_schema_species_name">Danio rerio</li>
        <li class="sql_schema_species_name">Drosophila melanogaster</li>
        <li class="sql_schema_species_name">Equus caballus</li>
        <li class="sql_schema_species_name">Felis catus</li>
        <li class="sql_schema_species_name">Gallus gallus</li>
        <li class="sql_schema_species_name">Homo sapiens</li>
        <li class="sql_schema_species_name">Macaca mulatta</li>
        <li class="sql_schema_species_name">Meleagris gallopavo</li>
        <li class="sql_schema_species_name">Monodelphis domestica</li>
        <li class="sql_schema_species_name">Mus musculus</li>
        <li class="sql_schema_species_name">Nomascus leucogenys</li>
        <li class="sql_schema_species_name">Ornithorhynchus anatinus</li>
        <li class="sql_schema_species_name">Ovis aries</li>
        <li class="sql_schema_species_name">Pan troglodytes</li>
        <li class="sql_schema_species_name">Pongo abelii</li>
        <li class="sql_schema_species_name">Rattus norvegicus</li>
        <li class="sql_schema_species_name">Saccharomyces cerevisiae</li>
        <li class="sql_schema_species_name">Sus scrofa</li>
        <li class="sql_schema_species_name">Taeniopygia guttata</li>
        <li class="sql_schema_species_name">Tetraodon nigroviridis</li>
      </ul>
    </div>
  </td></tr></table></div></div><div class="sql_schema_table">
  <div id="failed_structural_variation"class="sql_schema_table_header" style="border-top-color:#3CB371">
    <div class="sql_schema_table_header_left"><span style="background-color:#3CB371"></span>failed_structural_variation</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_failed_structural_variation" class="help-header sql_schema_show_hide" onclick="show_hide('failed_structural_variation','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">For various reasons it may be necessary to store information about a structural variation that has failed quality checks (mappings) in the Structural Variation pipeline. This table acts as a flag for such failures.</p>

  <div id="div_failed_structural_variation" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>failed_structural_variation_id</b></td>
        <td class="bg1">int(11)</td>
        <td class="bg1">-</td>
        <td class="bg1">Primary key, internal identifier.</td>
        <td class="bg1">primary key</td>
      </tr>

      <tr>
        <td class="bg2"><b>structural_variation_id</b></td>
        <td class="bg2">int(10)</td>
        <td class="bg2">-</td>
        <td class="bg2">Foreign key references to the <a href="#structural_variation" class="sql_schema_link">structural_variation</a> table.</td>
        <td class="bg2">unique key: <i>structural_variation_idx</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>failed_description_id</b></td>
        <td class="bg1">int(10)</td>
        <td class="bg1">-</td>
        <td class="bg1">Foreign key references to the <a href="#failed_description" class="sql_schema_link">failed_description</a> table.</td>
        <td class="bg1">unique key: <i>structural_variation_idx</i></td>
      </tr>
    </table>
  </div>
<table class="sql_schema_extra"><tr>    <td class="sql_schema_extra_left"">    <p style="font-weight:bold">See also:</p>
  <ul>
      <li><a href="#failed_description">failed_description</a></li>
      <li><a href="#structural_variation">structural_variation</a></li>
      </ul>
    </td>
  <td class="sql_schema_extra_separator"></td>
  <td class="sql_schema_extra_right" style="padding-left:25px"><p><span>List of species with populated data:</span>
  <a id="s_failed_structural_variation" class="help-header sql_schema_show_hide" onclick="show_hide('failed_structural_variation','species')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show species
  </a></p>
    <div id="sp_failed_structural_variation" style="display:none;">      <ul>
        <li class="sql_schema_species_name">Bos taurus</li>
        <li class="sql_schema_species_name">Canis familiaris</li>
        <li class="sql_schema_species_name">Danio rerio</li>
        <li class="sql_schema_species_name">Homo sapiens</li>
        <li class="sql_schema_species_name">Macaca mulatta</li>
        <li class="sql_schema_species_name">Mus musculus</li>
        <li class="sql_schema_species_name">Sus scrofa</li>
      </ul>
    </div>
  </td></tr></table></div></div><div class="sql_schema_table">
  <div id="failed_variation"class="sql_schema_table_header" style="border-top-color:#3CB371">
    <div class="sql_schema_table_header_left"><span style="background-color:#3CB371"></span>failed_variation</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_failed_variation" class="help-header sql_schema_show_hide" onclick="show_hide('failed_variation','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">For various reasons it may be necessary to store information about a variation that has failed quality checks in the Variation pipeline. This table acts as a flag for such failures.</p>

  <div id="div_failed_variation" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>failed_variation_id</b></td>
        <td class="bg1">int(11)</td>
        <td class="bg1">-</td>
        <td class="bg1">Primary key, internal identifier.</td>
        <td class="bg1">primary key</td>
      </tr>

      <tr>
        <td class="bg2"><b>variation_id</b></td>
        <td class="bg2">int(10)</td>
        <td class="bg2">-</td>
        <td class="bg2">Foreign key references to the <a href="#variation" class="sql_schema_link">variation</a> table.</td>
        <td class="bg2">unique key: <i>variation_idx</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>failed_description_id</b></td>
        <td class="bg1">int(10)</td>
        <td class="bg1">-</td>
        <td class="bg1">Foreign key references to the <a href="#failed_description" class="sql_schema_link">failed_description</a> table.</td>
        <td class="bg1">unique key: <i>variation_idx</i></td>
      </tr>
    </table>
  </div>
<table class="sql_schema_extra"><tr>    <td class="sql_schema_extra_left"">    <p style="font-weight:bold">See also:</p>
  <ul>
      <li><a href="#failed_description">failed_description</a></li>
      <li><a href="#variation">variation</a></li>
      </ul>
    </td>
  <td class="sql_schema_extra_separator"></td>
  <td class="sql_schema_extra_right" style="padding-left:25px"><p><span>List of species with populated data:</span>
  <a id="s_failed_variation" class="help-header sql_schema_show_hide" onclick="show_hide('failed_variation','species')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show species
  </a></p>
    <div id="sp_failed_variation" style="display:none;">      <ul>
        <li class="sql_schema_species_name">Bos taurus</li>
        <li class="sql_schema_species_name">Canis familiaris</li>
        <li class="sql_schema_species_name">Danio rerio</li>
        <li class="sql_schema_species_name">Equus caballus</li>
        <li class="sql_schema_species_name">Felis catus</li>
        <li class="sql_schema_species_name">Gallus gallus</li>
        <li class="sql_schema_species_name">Homo sapiens</li>
        <li class="sql_schema_species_name">Macaca mulatta</li>
        <li class="sql_schema_species_name">Meleagris gallopavo</li>
        <li class="sql_schema_species_name">Monodelphis domestica</li>
        <li class="sql_schema_species_name">Mus musculus</li>
        <li class="sql_schema_species_name">Ornithorhynchus anatinus</li>
        <li class="sql_schema_species_name">Ovis aries</li>
        <li class="sql_schema_species_name">Pan troglodytes</li>
        <li class="sql_schema_species_name">Pongo abelii</li>
        <li class="sql_schema_species_name">Rattus norvegicus</li>
        <li class="sql_schema_species_name">Saccharomyces cerevisiae</li>
        <li class="sql_schema_species_name">Sus scrofa</li>
        <li class="sql_schema_species_name">Taeniopygia guttata</li>
        <li class="sql_schema_species_name">Tetraodon nigroviridis</li>
      </ul>
    </div>
  </td></tr></table></div></div>

<div class="sql_schema_group_header" style="border-color:#FF8500">
  <div id="3" class="sql_schema_group_bullet" style="background-color:#FF8500"></div>
  <h2 id="3">Genotype tables</h2>
</div>
<p class="sql_schema_group_header_desc">These tables define the genotype data at the sample and population levels.</p><div class="sql_schema_table">
  <div id="compressed_genotype_region"class="sql_schema_table_header" style="border-top-color:#FF8500">
    <div class="sql_schema_table_header_left"><span style="background-color:#FF8500"></span>compressed_genotype_region</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_compressed_genotype_region" class="help-header sql_schema_show_hide" onclick="show_hide('compressed_genotype_region','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">This table holds genotypes compressed using the pack() method in Perl. These genotypes are mapped to particular genomic locations rather than variation objects. The data have been compressed to reduce table size and increase the speed of the web code when retrieving strain slices and LD data. Only data from resequenced samples are used for LD calculations are included in this table</p>

  <div id="div_compressed_genotype_region" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>sample_id</b></td>
        <td class="bg1">int(10)</td>
        <td class="bg1">-</td>
        <td class="bg1">Foreign key references to the <a href="#sample" class="sql_schema_link">sample</a> table.</td>
        <td class="bg1">key: <i>sample_idx</i></td>
      </tr>

      <tr>
        <td class="bg2"><b>seq_region_id</b></td>
        <td class="bg2">int(10)</td>
        <td class="bg2">-</td>
        <td class="bg2">Foreign key references <a href="#seq_region" class="sql_schema_link">seq_region</a> in core db. ers to the seq_region which this variant is on, which may be a chromosome, a clone, etc...</td>
        <td class="bg2">key: <i>pos_idx</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>seq_region_start</b></td>
        <td class="bg1">int(11)</td>
        <td class="bg1">-</td>
        <td class="bg1">The start position of the variation on the <a href="#seq_region" class="sql_schema_link">seq_region</a>.</td>
        <td class="bg1">key: <i>pos_idx</i></td>
      </tr>

      <tr>
        <td class="bg2"><b>seq_region_end</b></td>
        <td class="bg2">int(11)</td>
        <td class="bg2">-</td>
        <td class="bg2">The end position of the variation on the <a href="#seq_region" class="sql_schema_link">seq_region</a>.</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>seq_region_strand</b></td>
        <td class="bg1">tinyint(4)</td>
        <td class="bg1">-</td>
        <td class="bg1">The orientation of the variation on the <a href="#seq_region" class="sql_schema_link">seq_region</a>.</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>genotypes</b></td>
        <td class="bg2">blob</td>
        <td class="bg2">NULL</td>
        <td class="bg2">Encoded representation of the genotype data:<br />Each row in the compressed table stores genotypes from one individual/sample in one fixed-size region of the genome (arbitrarily defined as 100 Kb). The compressed string (using Perl's pack method) consisting of a repeating triplet of elements: a  <span style="color:#D00">distance</span> in base pairs from the previous genotype; a <span style="color:#090">variation dbID</span>; a <span style="color:#00D">genotype_code_id</span> identifier.<br />For example, a given row may have a start position of 1000, indicating the chromosomal position of the first genotype in this row. The unpacked genotypes field then may contain the following elements:<br /><b><span style="color:#D00">0</span>, <span style="color:#090">1</span>,  <span style="color:#00D">1</span>, <span style="color:#D00">20</span>, <span style="color:#090">2</span>, <span style="color:#00D">5</span>, <span style="color:#D00">35</span>, <span style="color:#090">3</span>, <span style="color:#00D">3</span>, ...</b><br />The first genotype ("<span style="color:#D00">0</span>,<span style="color:#090">1</span>,<span style="color:#00D">1</span>") has a position of 1000 + <span style="color:#D00">0</span> = 1000, and corresponds to the variation with the internal identifier <span style="color:#090">1</span> and genotype_code_id corresponding to the genotype A|G (internal ID <span style="color:#00D">1</span>).<br />The second genotype ("<span style="color:#D00">20</span>,<span style="color:#090">2</span>,<span style="color:#00D">5</span>") has a position of 1000 + <span style="color:#D00">20</span> = 1020, internal variation_id <span style="color:#090">2</span> and genotype_code_id corresponding to the genotype C|C ( internal ID <span style="color:#00D">5</span>).<br />The third genotype similarly has a position of 1055, and so on.</td>
        <td class="bg2"></td>
      </tr>
    </table>
  </div>
<table class="sql_schema_extra"><tr>    <td class="sql_schema_extra_left"">    <p style="font-weight:bold">See also:</p>
  <ul>
      <li><a href="#sample">sample</a></li>
      <li><a href="#seq_region">seq_region</a></li>
      <li><a href="#variation">variation</a></li>
      <li><a href="#genotype_code">genotype_code</a></li>
      </ul>
    </td>
  <td class="sql_schema_extra_separator"></td>
  <td class="sql_schema_extra_right" style="padding-left:25px"><p><span>List of species with populated data:</span>
  <a id="s_compressed_genotype_region" class="help-header sql_schema_show_hide" onclick="show_hide('compressed_genotype_region','species')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show species
  </a></p>
    <div id="sp_compressed_genotype_region" style="display:none;">      <ul>
        <li class="sql_schema_species_name">Canis familiaris</li>
        <li class="sql_schema_species_name">Drosophila melanogaster</li>
        <li class="sql_schema_species_name">Gallus gallus</li>
        <li class="sql_schema_species_name">Mus musculus</li>
        <li class="sql_schema_species_name">Rattus norvegicus</li>
      </ul>
    </div>
  </td></tr></table></div></div><div class="sql_schema_table">
  <div id="compressed_genotype_var"class="sql_schema_table_header" style="border-top-color:#FF8500">
    <div class="sql_schema_table_header_left"><span style="background-color:#FF8500"></span>compressed_genotype_var</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_compressed_genotype_var" class="help-header sql_schema_show_hide" onclick="show_hide('compressed_genotype_var','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">This table holds genotypes compressed using the pack() method in Perl. These genotypes are mapped directly to variation objects. The data have been compressed to reduce table size. All genotypes in the database are included in this table (included duplicates of those genotypes contained in the compressed_genotype_region table). This table is optimised for retrieval from variation.</p>

  <div id="div_compressed_genotype_var" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>variation_id</b></td>
        <td class="bg1">int(11)</td>
        <td class="bg1">-</td>
        <td class="bg1">Foreign key references to the <a href="#variation" class="sql_schema_link">variation</a> table.</td>
        <td class="bg1">key: <i>variation_idx</i></td>
      </tr>

      <tr>
        <td class="bg2"><b>subsnp_id</b></td>
        <td class="bg2">int(11)</td>
        <td class="bg2">NULL</td>
        <td class="bg2">Foreign key references to the <a href="#subsnp_handle" class="sql_schema_link">subsnp_handle</a> table.</td>
        <td class="bg2">key: <i>subsnp_idx</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>genotypes</b></td>
        <td class="bg1">blob</td>
        <td class="bg1">NULL</td>
        <td class="bg1">Encoded representation of the genotype data:<br />Each row in the compressed table stores genotypes from one subsnp of a variation (or one variation if no subsnp is defined). The compressed string (using Perl's pack method) consisting of a repeating pair of elements: an internal sample_id corresponding to a sample; a genotype_code_id identifier.</td>
        <td class="bg1"></td>
      </tr>
    </table>
  </div>
<table class="sql_schema_extra"><tr>    <td class="sql_schema_extra_left"">    <p style="font-weight:bold">See also:</p>
  <ul>
      <li><a href="#sample">sample</a></li>
      <li><a href="#variation">variation</a></li>
      <li><a href="#genotype_code">genotype_code</a></li>
      <li><a href="#subsnp_handle">subsnp_handle</a></li>
      </ul>
    </td>
  <td class="sql_schema_extra_separator"></td>
  <td class="sql_schema_extra_right" style="padding-left:25px"><p><span>List of species with populated data:</span>
  <a id="s_compressed_genotype_var" class="help-header sql_schema_show_hide" onclick="show_hide('compressed_genotype_var','species')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show species
  </a></p>
    <div id="sp_compressed_genotype_var" style="display:none;">      <ul>
        <li class="sql_schema_species_name">Bos taurus</li>
        <li class="sql_schema_species_name">Canis familiaris</li>
        <li class="sql_schema_species_name">Danio rerio</li>
        <li class="sql_schema_species_name">Drosophila melanogaster</li>
        <li class="sql_schema_species_name">Equus caballus</li>
        <li class="sql_schema_species_name">Gallus gallus</li>
        <li class="sql_schema_species_name">Homo sapiens</li>
        <li class="sql_schema_species_name">Macaca mulatta</li>
        <li class="sql_schema_species_name">Mus musculus</li>
        <li class="sql_schema_species_name">Nomascus leucogenys</li>
        <li class="sql_schema_species_name">Ornithorhynchus anatinus</li>
        <li class="sql_schema_species_name">Pan troglodytes</li>
        <li class="sql_schema_species_name">Pongo abelii</li>
        <li class="sql_schema_species_name">Rattus norvegicus</li>
        <li class="sql_schema_species_name">Saccharomyces cerevisiae</li>
        <li class="sql_schema_species_name">Sus scrofa</li>
        <li class="sql_schema_species_name">Taeniopygia guttata</li>
      </ul>
    </div>
  </td></tr></table></div></div><div class="sql_schema_table">
  <div id="population_genotype"class="sql_schema_table_header" style="border-top-color:#FF8500">
    <div class="sql_schema_table_header_left"><span style="background-color:#FF8500"></span>population_genotype</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_population_genotype" class="help-header sql_schema_show_hide" onclick="show_hide('population_genotype','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">This table stores genotypes and frequencies for variations in given populations.</p>

  <div id="div_population_genotype" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>population_genotype_id</b></td>
        <td class="bg1">int(10)</td>
        <td class="bg1">-</td>
        <td class="bg1">Primary key, internal identifier.</td>
        <td class="bg1">primary key</td>
      </tr>

      <tr>
        <td class="bg2"><b>variation_id</b></td>
        <td class="bg2">int(11)</td>
        <td class="bg2">-</td>
        <td class="bg2">Foreign key references to the <a href="#variation" class="sql_schema_link">variation</a> table.</td>
        <td class="bg2">key: <i>variation_idx</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>subsnp_id</b></td>
        <td class="bg1">int(11)</td>
        <td class="bg1">NULL</td>
        <td class="bg1">Foreign key references to the subsnp_handle table.</td>
        <td class="bg1">key: <i>subsnp_idx</i></td>
      </tr>

      <tr>
        <td class="bg2"><b>genotype_code_id</b></td>
        <td class="bg2">int(11)</td>
        <td class="bg2">NULL</td>
        <td class="bg2">Foreign key reference to the <a href="#genotype_code" class="sql_schema_link">genotype_code</a> table.</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>frequency</b></td>
        <td class="bg1">float</td>
        <td class="bg1">NULL</td>
        <td class="bg1">Frequency of the genotype in the population.</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>population_id</b></td>
        <td class="bg2">int(10)</td>
        <td class="bg2">NULL</td>
        <td class="bg2">Foreign key references to the <a href="#population" class="sql_schema_link">population</a> table.</td>
        <td class="bg2">key: <i>population_idx</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>count</b></td>
        <td class="bg1">int(10)</td>
        <td class="bg1">NULL</td>
        <td class="bg1">Number of individuals/samples who have this genotype, in this population.</td>
        <td class="bg1"></td>
      </tr>
    </table>
  </div>
<table class="sql_schema_extra"><tr>    <td class="sql_schema_extra_left"">    <p style="font-weight:bold">See also:</p>
  <ul>
      <li><a href="#population">population</a></li>
      <li><a href="#variation">variation</a></li>
      <li><a href="#subsnp_handle">subsnp_handle</a></li>
      <li><a href="#genotype_code">genotype_code</a></li>
      </ul>
    </td>
  <td class="sql_schema_extra_separator"></td>
  <td class="sql_schema_extra_right" style="padding-left:25px"><p><span>List of species with populated data:</span>
  <a id="s_population_genotype" class="help-header sql_schema_show_hide" onclick="show_hide('population_genotype','species')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show species
  </a></p>
    <div id="sp_population_genotype" style="display:none;">      <ul>
        <li class="sql_schema_species_name">Bos taurus</li>
        <li class="sql_schema_species_name">Canis familiaris</li>
        <li class="sql_schema_species_name">Danio rerio</li>
        <li class="sql_schema_species_name">Drosophila melanogaster</li>
        <li class="sql_schema_species_name">Equus caballus</li>
        <li class="sql_schema_species_name">Gallus gallus</li>
        <li class="sql_schema_species_name">Homo sapiens</li>
        <li class="sql_schema_species_name">Macaca mulatta</li>
        <li class="sql_schema_species_name">Meleagris gallopavo</li>
        <li class="sql_schema_species_name">Mus musculus</li>
        <li class="sql_schema_species_name">Ornithorhynchus anatinus</li>
        <li class="sql_schema_species_name">Ovis aries</li>
        <li class="sql_schema_species_name">Pan troglodytes</li>
        <li class="sql_schema_species_name">Rattus norvegicus</li>
        <li class="sql_schema_species_name">Saccharomyces cerevisiae</li>
        <li class="sql_schema_species_name">Sus scrofa</li>
        <li class="sql_schema_species_name">Taeniopygia guttata</li>
      </ul>
    </div>
  </td></tr></table></div></div><div class="sql_schema_table">
  <div id="read_coverage"class="sql_schema_table_header" style="border-top-color:#FF8500">
    <div class="sql_schema_table_header_left"><span style="background-color:#FF8500"></span>read_coverage</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_read_coverage" class="help-header sql_schema_show_hide" onclick="show_hide('read_coverage','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">This table stores the read coverage of resequenced samples. Each row contains sample ID, chromosomal coordinates and a read coverage level.</p>

  <div id="div_read_coverage" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>seq_region_id</b></td>
        <td class="bg1">int(10)</td>
        <td class="bg1">-</td>
        <td class="bg1">Foreign key references <a href="#seq_region" class="sql_schema_link">seq_region</a> in core db. ers to the seq_region which this variant is on, which may be a chromosome, a clone, etc...</td>
        <td class="bg1">key: <i>seq_region_idx</i></td>
      </tr>

      <tr>
        <td class="bg2"><b>seq_region_start</b></td>
        <td class="bg2">int</td>
        <td class="bg2">-</td>
        <td class="bg2">The start position of the variation on the <a href="#seq_region" class="sql_schema_link">seq_region</a>.</td>
        <td class="bg2">key: <i>seq_region_idx</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>seq_region_end</b></td>
        <td class="bg1">int</td>
        <td class="bg1">-</td>
        <td class="bg1">The end position of the variation on the <a href="#seq_region" class="sql_schema_link">seq_region</a>.</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>level</b></td>
        <td class="bg2">tinyint</td>
        <td class="bg2">-</td>
        <td class="bg2">Minimum number of reads.</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>sample_id</b></td>
        <td class="bg1">int(10)</td>
        <td class="bg1">-</td>
        <td class="bg1">Foreign key references to the <a href="#sample" class="sql_schema_link">sample</a> table.</td>
        <td class="bg1">key: <i>sample_idx</i></td>
      </tr>
    </table>
  </div>
<table class="sql_schema_extra"><tr>    <td class="sql_schema_extra_left"">    <p style="font-weight:bold">See also:</p>
  <ul>
      <li><a href="#sample">sample</a></li>
      <li><a href="#seq_region">seq_region</a></li>
      </ul>
    </td>
  <td class="sql_schema_extra_separator"></td>
  <td class="sql_schema_extra_right" style="padding-left:25px"><p><span>List of species with populated data:</span>
  <a id="s_read_coverage" class="help-header sql_schema_show_hide" onclick="show_hide('read_coverage','species')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show species
  </a></p>
    <div id="sp_read_coverage" style="display:none;">      <ul>
        <li class="sql_schema_species_name">Canis familiaris</li>
        <li class="sql_schema_species_name">Gallus gallus</li>
        <li class="sql_schema_species_name">Mus musculus</li>
        <li class="sql_schema_species_name">Rattus norvegicus</li>
      </ul>
    </div>
  </td></tr></table></div></div><div class="sql_schema_table">
  <div id="sample_genotype_multiple_bp"class="sql_schema_table_header" style="border-top-color:#FF8500">
    <div class="sql_schema_table_header_left"><span style="background-color:#FF8500"></span>sample_genotype_multiple_bp</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_sample_genotype_multiple_bp" class="help-header sql_schema_show_hide" onclick="show_hide('sample_genotype_multiple_bp','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">This table holds uncompressed genotypes for given variations.</p>

  <div id="div_sample_genotype_multiple_bp" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>variation_id</b></td>
        <td class="bg1">int(10)</td>
        <td class="bg1">-</td>
        <td class="bg1">Primary key. Foreign key references to the <a href="#variation" class="sql_schema_link">variation</a> table.</td>
        <td class="bg1">key: <i>variation_idx</i></td>
      </tr>

      <tr>
        <td class="bg2"><b>subsnp_id</b></td>
        <td class="bg2">int(15)</td>
        <td class="bg2">NULL</td>
        <td class="bg2">Foreign key references to the <a href="#subsnp_handle" class="sql_schema_link">subsnp_handle</a> table.</td>
        <td class="bg2">key: <i>subsnp_idx</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>allele_1</b></td>
        <td class="bg1">varchar(25000)</td>
        <td class="bg1">NULL</td>
        <td class="bg1">One of the alleles of the genotype, e.g. "TAG".</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>allele_2</b></td>
        <td class="bg2">varchar(25000)</td>
        <td class="bg2">NULL</td>
        <td class="bg2">The other allele of the genotype.</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>sample_id</b></td>
        <td class="bg1">int(10)</td>
        <td class="bg1">NULL</td>
        <td class="bg1">Foreign key references to the <a href="#sample" class="sql_schema_link">sample</a> table.</td>
        <td class="bg1">key: <i>sample_idx</i></td>
      </tr>
    </table>
  </div>
<table class="sql_schema_extra"><tr>    <td class="sql_schema_extra_left"">    <p style="font-weight:bold">See also:</p>
  <ul>
      <li><a href="#sample">sample</a></li>
      <li><a href="#variation">variation</a></li>
      <li><a href="#subsnp_handle">subsnp_handle</a></li>
      </ul>
    </td>
  <td class="sql_schema_extra_separator"></td>
  <td class="sql_schema_extra_right" style="padding-left:25px"><p><span>List of species with populated data:</span>
  <a id="s_sample_genotype_multiple_bp" class="help-header sql_schema_show_hide" onclick="show_hide('sample_genotype_multiple_bp','species')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show species
  </a></p>
    <div id="sp_sample_genotype_multiple_bp" style="display:none;">      <ul>
        <li class="sql_schema_species_name">Bos taurus</li>
        <li class="sql_schema_species_name">Gallus gallus</li>
        <li class="sql_schema_species_name">Homo sapiens</li>
        <li class="sql_schema_species_name">Macaca mulatta</li>
        <li class="sql_schema_species_name">Rattus norvegicus</li>
      </ul>
    </div>
  </td></tr></table></div></div><div class="sql_schema_table">
  <div id="tmp_sample_genotype_single_bp"class="sql_schema_table_header" style="border-top-color:#FF8500">
    <div class="sql_schema_table_header_left"><span style="background-color:#FF8500"></span>tmp_sample_genotype_single_bp</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_tmp_sample_genotype_single_bp" class="help-header sql_schema_show_hide" onclick="show_hide('tmp_sample_genotype_single_bp','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">This table is only needed to create master schema when run healthcheck system. Needed for other species, but human, so keep it.</p>

  <div id="div_tmp_sample_genotype_single_bp" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>variation_id</b></td>
        <td class="bg1">int(10)</td>
        <td class="bg1">-</td>
        <td class="bg1">Foreign key references to the <a href="#variation" class="sql_schema_link">variation</a> table.</td>
        <td class="bg1">key: <i>variation_idx</i></td>
      </tr>

      <tr>
        <td class="bg2"><b>subsnp_id</b></td>
        <td class="bg2">int(15)</td>
        <td class="bg2">NULL</td>
        <td class="bg2">Foreign key references to the <a href="#subsnp_handle" class="sql_schema_link">subsnp_handle</a> table.</td>
        <td class="bg2">key: <i>subsnp_idx</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>allele_1</b></td>
        <td class="bg1">char(1)</td>
        <td class="bg1">NULL</td>
        <td class="bg1">One of the alleles of the genotype, e.g. "TAG".</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>allele_2</b></td>
        <td class="bg2">char(1)</td>
        <td class="bg2">NULL</td>
        <td class="bg2">The other allele of the genotype.</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>sample_id</b></td>
        <td class="bg1">int(10)</td>
        <td class="bg1">-</td>
        <td class="bg1">Foreign key references to the <a href="#sample" class="sql_schema_link">sample</a> table.</td>
        <td class="bg1">key: <i>sample_idx</i></td>
      </tr>
    </table>
  </div>
<table class="sql_schema_extra"><tr>    <td class="sql_schema_extra_left"">    <p style="font-weight:bold">See also:</p>
  <ul>
      <li><a href="#sample">sample</a></li>
      <li><a href="#variation">variation</a></li>
      <li><a href="#subsnp_handle">subsnp_handle</a></li>
      </ul>
    </td>
  <td class="sql_schema_extra_separator"></td>
  <td class="sql_schema_extra_right" style="padding-left:25px"><p><span>List of species with populated data:</span>
  <a id="s_tmp_sample_genotype_single_bp" class="help-header sql_schema_show_hide" onclick="show_hide('tmp_sample_genotype_single_bp','species')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show species
  </a></p>
    <div id="sp_tmp_sample_genotype_single_bp" style="display:none;">      <ul>
        <li class="sql_schema_species_name">Bos taurus</li>
        <li class="sql_schema_species_name">Canis familiaris</li>
        <li class="sql_schema_species_name">Danio rerio</li>
        <li class="sql_schema_species_name">Drosophila melanogaster</li>
        <li class="sql_schema_species_name">Equus caballus</li>
        <li class="sql_schema_species_name">Gallus gallus</li>
        <li class="sql_schema_species_name">Macaca mulatta</li>
        <li class="sql_schema_species_name">Mus musculus</li>
        <li class="sql_schema_species_name">Nomascus leucogenys</li>
        <li class="sql_schema_species_name">Ornithorhynchus anatinus</li>
        <li class="sql_schema_species_name">Ovis aries</li>
        <li class="sql_schema_species_name">Pan troglodytes</li>
        <li class="sql_schema_species_name">Pongo abelii</li>
        <li class="sql_schema_species_name">Rattus norvegicus</li>
        <li class="sql_schema_species_name">Saccharomyces cerevisiae</li>
        <li class="sql_schema_species_name">Sus scrofa</li>
        <li class="sql_schema_species_name">Taeniopygia guttata</li>
        <li class="sql_schema_species_name">Tetraodon nigroviridis</li>
      </ul>
    </div>
  </td></tr></table></div></div>

<div class="sql_schema_group_header" style="border-color:#BC5CEC">
  <div id="4" class="sql_schema_group_bullet" style="background-color:#BC5CEC"></div>
  <h2 id="4">Metadata tables</h2>
</div>
<p class="sql_schema_group_header_desc">These tables define some metadata information.</p><div class="sql_schema_table">
  <div id="meta"class="sql_schema_table_header" style="border-top-color:#BC5CEC">
    <div class="sql_schema_table_header_left"><span style="background-color:#BC5CEC"></span>meta</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_meta" class="help-header sql_schema_show_hide" onclick="show_hide('meta','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">This table stores various metadata relating to the database, generally used by the Ensembl web code.</p>

  <div id="div_meta" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>meta_id</b></td>
        <td class="bg1">INT(10)</td>
        <td class="bg1">-</td>
        <td class="bg1">Primary key, internal identifier.</td>
        <td class="bg1">primary key</td>
      </tr>

      <tr>
        <td class="bg2"><b>species_id</b></td>
        <td class="bg2">INT</td>
        <td class="bg2">1</td>
        <td class="bg2">...</td>
        <td class="bg2">unique key: <i>species_key_value_idx</i><br />key: <i>species_value_idx</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>meta_key</b></td>
        <td class="bg1">varchar( 40 )</td>
        <td class="bg1">-</td>
        <td class="bg1">Name of the meta entry, e.g. "schema_version".</td>
        <td class="bg1">unique key: <i>species_key_value_idx</i></td>
      </tr>

      <tr>
        <td class="bg2"><b>meta_value</b></td>
        <td class="bg2">varchar( 255 )</td>
        <td class="bg2">-</td>
        <td class="bg2">Corresponding value of the key, e.g. "61".</td>
        <td class="bg2">unique key: <i>species_key_value_idx</i><br />key: <i>species_value_idx</i></td>
      </tr>
    </table>
  </div>
<table class="sql_schema_extra"><tr>
  <td class="sql_schema_extra_right"><p><span>List of species with populated data:</span>
  <a id="s_meta" class="help-header sql_schema_show_hide" onclick="show_hide('meta','species')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show species
  </a></p>
    <div id="sp_meta" style="display:none;">      <ul>
        <li class="sql_schema_species_name">Bos taurus</li>
        <li class="sql_schema_species_name">Canis familiaris</li>
        <li class="sql_schema_species_name">Danio rerio</li>
        <li class="sql_schema_species_name">Drosophila melanogaster</li>
        <li class="sql_schema_species_name">Equus caballus</li>
        <li class="sql_schema_species_name">Felis catus</li>
        <li class="sql_schema_species_name">Gallus gallus</li>
        <li class="sql_schema_species_name">Homo sapiens</li>
        <li class="sql_schema_species_name">Macaca mulatta</li>
        <li class="sql_schema_species_name">Meleagris gallopavo</li>
        <li class="sql_schema_species_name">Monodelphis domestica</li>
        <li class="sql_schema_species_name">Mus musculus</li>
        <li class="sql_schema_species_name">Nomascus leucogenys</li>
        <li class="sql_schema_species_name">Ornithorhynchus anatinus</li>
        <li class="sql_schema_species_name">Ovis aries</li>
        <li class="sql_schema_species_name">Pan troglodytes</li>
        <li class="sql_schema_species_name">Pongo abelii</li>
        <li class="sql_schema_species_name">Rattus norvegicus</li>
        <li class="sql_schema_species_name">Saccharomyces cerevisiae</li>
        <li class="sql_schema_species_name">Sus scrofa</li>
        <li class="sql_schema_species_name">Taeniopygia guttata</li>
        <li class="sql_schema_species_name">Tetraodon nigroviridis</li>
      </ul>
    </div>
  </td></tr></table></div></div><div class="sql_schema_table">
  <div id="meta_coord"class="sql_schema_table_header" style="border-top-color:#BC5CEC">
    <div class="sql_schema_table_header_left"><span style="background-color:#BC5CEC"></span>meta_coord</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_meta_coord" class="help-header sql_schema_show_hide" onclick="show_hide('meta_coord','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">This table gives the coordinate system used by various tables in the database.</p>

  <div id="div_meta_coord" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>table_name</b></td>
        <td class="bg1">VARCHAR(40)</td>
        <td class="bg1">-</td>
        <td class="bg1">Name of the feature table, e.g. "variation_feature".</td>
        <td class="bg1">unique: <i>table_name, coord_system_id</i></td>
      </tr>

      <tr>
        <td class="bg2"><b>coord_system_id</b></td>
        <td class="bg2">INT(10)</td>
        <td class="bg2">-</td>
        <td class="bg2">Foreign key to core database coord_system table refers to coordinate system that features from this table can be found in.</td>
        <td class="bg2">unique: <i>table_name, coord_system_id</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>max_length</b></td>
        <td class="bg1">INT</td>
        <td class="bg1">NULL</td>
        <td class="bg1">Maximum length of the feature. </td>
        <td class="bg1"></td>
      </tr>
    </table>
  </div>
<table class="sql_schema_extra"><tr>
  <td class="sql_schema_extra_right"><p><span>List of species with populated data:</span>
  <a id="s_meta_coord" class="help-header sql_schema_show_hide" onclick="show_hide('meta_coord','species')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show species
  </a></p>
    <div id="sp_meta_coord" style="display:none;">      <ul>
        <li class="sql_schema_species_name">Bos taurus</li>
        <li class="sql_schema_species_name">Canis familiaris</li>
        <li class="sql_schema_species_name">Danio rerio</li>
        <li class="sql_schema_species_name">Drosophila melanogaster</li>
        <li class="sql_schema_species_name">Equus caballus</li>
        <li class="sql_schema_species_name">Felis catus</li>
        <li class="sql_schema_species_name">Gallus gallus</li>
        <li class="sql_schema_species_name">Homo sapiens</li>
        <li class="sql_schema_species_name">Macaca mulatta</li>
        <li class="sql_schema_species_name">Meleagris gallopavo</li>
        <li class="sql_schema_species_name">Monodelphis domestica</li>
        <li class="sql_schema_species_name">Mus musculus</li>
        <li class="sql_schema_species_name">Nomascus leucogenys</li>
        <li class="sql_schema_species_name">Ornithorhynchus anatinus</li>
        <li class="sql_schema_species_name">Ovis aries</li>
        <li class="sql_schema_species_name">Pan troglodytes</li>
        <li class="sql_schema_species_name">Pongo abelii</li>
        <li class="sql_schema_species_name">Rattus norvegicus</li>
        <li class="sql_schema_species_name">Saccharomyces cerevisiae</li>
        <li class="sql_schema_species_name">Sus scrofa</li>
        <li class="sql_schema_species_name">Taeniopygia guttata</li>
        <li class="sql_schema_species_name">Tetraodon nigroviridis</li>
      </ul>
    </div>
  </td></tr></table></div></div>

<div class="sql_schema_group_header" style="border-color:#000">
  <div id="5" class="sql_schema_group_bullet" style="background-color:#000"></div>
  <h2 id="5">Other tables</h2>
</div>
<p class="sql_schema_group_header_desc">These tables define the other data associated with a variation.</p><div class="sql_schema_table">
  <div id="allele_code"class="sql_schema_table_header" style="border-top-color:#000">
    <div class="sql_schema_table_header_left"><span style="background-color:#000"></span>allele_code</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_allele_code" class="help-header sql_schema_show_hide" onclick="show_hide('allele_code','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">This table stores the relationship between the internal allele identifiers and the alleles themselves.</p>

  <div id="div_allele_code" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>allele_code_id</b></td>
        <td class="bg1">int(11)</td>
        <td class="bg1">-</td>
        <td class="bg1">Primary key, internal identifier.</td>
        <td class="bg1">primary key</td>
      </tr>

      <tr>
        <td class="bg2"><b>allele</b></td>
        <td class="bg2">varchar(60000)</td>
        <td class="bg2">NULL</td>
        <td class="bg2">String representing the allele. Has a unique constraint on the first 1000 characters (max allowed by MySQL).</td>
        <td class="bg2">unique key: <i>allele_idx</i></td>
      </tr>
    </table>
  </div>
<div class="sql_schema_table_examples"><p class="sql_schema_table_example_header">Example:</p><div class="sql_schema_table_example_content"><p>See below the first entries in the allele_code table: </p>
      <div>
        <div class="sql_schema_table_example_query">
          <pre><span class="sql_schema_sql_highlight">SELECT</span> * <span class="sql_schema_sql_highlight">FROM</span> allele_code <span class="sql_schema_sql_highlight">LIMIT</span> 6;</pre>
        </div>
        <div class="sql_schema_table_example_button">
  <a id="e_allele_code" class="help-header sql_schema_show_hide" onclick="show_hide('allele_code','query results')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show query results
  </a></div>
        <div style="clear:both"></div>
      </div>
      
  <div id="ex_allele_code" style="display:none;">
    <table class="ss sql_schema_table_example_result">
      <tr><th>allele_code_id</th><th>allele</th></tr>      <tr><td>1</td><td>T</td></tr>      <tr class="bg2"><td>2</td><td>A</td></tr>      <tr><td>3</td><td>G</td></tr>      <tr class="bg2"><td>4</td><td>C</td></tr>      <tr><td>5</td><td>-</td></tr>      <tr class="bg2"><td>6</td><td>N</td></tr>    </table>
  </div></div></div><table class="sql_schema_extra"><tr>    <td class="sql_schema_extra_left"">    <p style="font-weight:bold">See also:</p>
  <ul>
      <li><a href="#allele">allele</a></li>
      <li><a href="#genotype_code">genotype_code</a></li>
      </ul>
    </td>
  <td class="sql_schema_extra_separator"></td>
  <td class="sql_schema_extra_right" style="padding-left:25px"><p><span>List of species with populated data:</span>
  <a id="s_allele_code" class="help-header sql_schema_show_hide" onclick="show_hide('allele_code','species')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show species
  </a></p>
    <div id="sp_allele_code" style="display:none;">      <ul>
        <li class="sql_schema_species_name">Bos taurus</li>
        <li class="sql_schema_species_name">Canis familiaris</li>
        <li class="sql_schema_species_name">Danio rerio</li>
        <li class="sql_schema_species_name">Drosophila melanogaster</li>
        <li class="sql_schema_species_name">Equus caballus</li>
        <li class="sql_schema_species_name">Felis catus</li>
        <li class="sql_schema_species_name">Gallus gallus</li>
        <li class="sql_schema_species_name">Homo sapiens</li>
        <li class="sql_schema_species_name">Macaca mulatta</li>
        <li class="sql_schema_species_name">Meleagris gallopavo</li>
        <li class="sql_schema_species_name">Monodelphis domestica</li>
        <li class="sql_schema_species_name">Mus musculus</li>
        <li class="sql_schema_species_name">Nomascus leucogenys</li>
        <li class="sql_schema_species_name">Ornithorhynchus anatinus</li>
        <li class="sql_schema_species_name">Ovis aries</li>
        <li class="sql_schema_species_name">Pan troglodytes</li>
        <li class="sql_schema_species_name">Pongo abelii</li>
        <li class="sql_schema_species_name">Rattus norvegicus</li>
        <li class="sql_schema_species_name">Saccharomyces cerevisiae</li>
        <li class="sql_schema_species_name">Sus scrofa</li>
        <li class="sql_schema_species_name">Taeniopygia guttata</li>
        <li class="sql_schema_species_name">Tetraodon nigroviridis</li>
      </ul>
    </div>
  </td></tr></table></div></div><div class="sql_schema_table">
  <div id="coord_system"class="sql_schema_table_header" style="border-top-color:#000">
    <div class="sql_schema_table_header_left"><span style="background-color:#000"></span>coord_system</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_coord_system" class="help-header sql_schema_show_hide" onclick="show_hide('coord_system','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">Stores information about the available co-ordinate systems for the species identified through the species_id field. Note that for each species, there must be one co-ordinate system that has the attribute "top_level" and one that has the attribute "sequence_level".</p>

  <div id="div_coord_system" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>coord_system_id</b></td>
        <td class="bg1">INT(10)</td>
        <td class="bg1">-</td>
        <td class="bg1">Primary key, internal identifier.</td>
        <td class="bg1">primary   key</td>
      </tr>

      <tr>
        <td class="bg2"><b>species_id</b></td>
        <td class="bg2">INT(10)</td>
        <td class="bg2">1</td>
        <td class="bg2">Identifies the species for multi-species databases.</td>
        <td class="bg2">unique    key: <i>rank_idx</i><br />unique    key: <i>name_idx</i><br />key: <i>species_idx</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>name</b></td>
        <td class="bg1">VARCHAR(40)</td>
        <td class="bg1">-</td>
        <td class="bg1">Co-oridinate system name, e.g. 'chromosome', 'contig', 'scaffold' etc.</td>
        <td class="bg1">unique    key: <i>name_idx</i></td>
      </tr>

      <tr>
        <td class="bg2"><b>version</b></td>
        <td class="bg2">VARCHAR(255)</td>
        <td class="bg2">NULL</td>
        <td class="bg2">Assembly.</td>
        <td class="bg2">unique    key: <i>name_idx</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>rank</b></td>
        <td class="bg1">INT</td>
        <td class="bg1">-</td>
        <td class="bg1">Co-oridinate system rank.</td>
        <td class="bg1">unique    key: <i>rank_idx</i></td>
      </tr>

      <tr>
        <td class="bg2"><b>attrib</b></td>
        <td class="bg2">SET:<ul class="sql_schema_table_column_type">  <li>default_version</li>  <li> sequence_level</li></ul></td>
        <td class="bg2">NULL</td>
        <td class="bg2">Co-oridinate system attrib (e.g. "top_level", "sequence_level").</td>
        <td class="bg2"></td>
      </tr>
    </table>
  </div>
<table class="sql_schema_extra"><tr>    <td class="sql_schema_extra_left"">    <p style="font-weight:bold">See also:</p>
  <ul>
      <li><a href="#seq_region">seq_region</a></li>
      <li><a href="#meta_coord">meta_coord</a></li>
      <li><a href="#meta">meta</a></li>
      </ul>
    </td>
  <td class="sql_schema_extra_separator"></td>
  <td class="sql_schema_extra_right" style="padding-left:25px"><p><span>List of species with populated data:</span>
  <a id="s_coord_system" class="help-header sql_schema_show_hide" onclick="show_hide('coord_system','species')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show species
  </a></p>
    <div id="sp_coord_system" style="display:none;">      <ul>
        <li class="sql_schema_species_name">Drosophila melanogaster</li>
      </ul>
    </div>
  </td></tr></table></div></div><div class="sql_schema_table">
  <div id="genotype_code"class="sql_schema_table_header" style="border-top-color:#000">
    <div class="sql_schema_table_header_left"><span style="background-color:#000"></span>genotype_code</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_genotype_code" class="help-header sql_schema_show_hide" onclick="show_hide('genotype_code','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">This table stores genotype codes as multiple rows of allele_code identifiers, linked by genotype_code_id and ordered by haplotype_id.</p>

  <div id="div_genotype_code" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>genotype_code_id</b></td>
        <td class="bg1">int(11)</td>
        <td class="bg1">-</td>
        <td class="bg1">Internal identifier.</td>
        <td class="bg1">key: <i>genotype_code_id</i></td>
      </tr>

      <tr>
        <td class="bg2"><b>allele_code_id</b></td>
        <td class="bg2">int(11)</td>
        <td class="bg2">-</td>
        <td class="bg2">Foreign key reference to <a href="#allele_code" class="sql_schema_link">allele_code</a> table.</td>
        <td class="bg2">key: <i>allele_code_id</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>haplotype_id</b></td>
        <td class="bg1">tinyint(2)</td>
        <td class="bg1">-</td>
        <td class="bg1">Sorting order of the genotype's alleles.</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>phased</b></td>
        <td class="bg2">tinyint(2)</td>
        <td class="bg2">NULL</td>
        <td class="bg2">Indicates if this genotype is phased</td>
        <td class="bg2"></td>
      </tr>
    </table>
  </div>
<table class="sql_schema_extra"><tr>    <td class="sql_schema_extra_left"">    <p style="font-weight:bold">See also:</p>
  <ul>
      <li><a href="#allele_code">allele_code</a></li>
      <li><a href="#population_genotype">population_genotype</a></li>
      </ul>
    </td>
  <td class="sql_schema_extra_separator"></td>
  <td class="sql_schema_extra_right" style="padding-left:25px"><p><span>List of species with populated data:</span>
  <a id="s_genotype_code" class="help-header sql_schema_show_hide" onclick="show_hide('genotype_code','species')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show species
  </a></p>
    <div id="sp_genotype_code" style="display:none;">      <ul>
        <li class="sql_schema_species_name">Bos taurus</li>
        <li class="sql_schema_species_name">Canis familiaris</li>
        <li class="sql_schema_species_name">Danio rerio</li>
        <li class="sql_schema_species_name">Drosophila melanogaster</li>
        <li class="sql_schema_species_name">Equus caballus</li>
        <li class="sql_schema_species_name">Felis catus</li>
        <li class="sql_schema_species_name">Gallus gallus</li>
        <li class="sql_schema_species_name">Homo sapiens</li>
        <li class="sql_schema_species_name">Macaca mulatta</li>
        <li class="sql_schema_species_name">Meleagris gallopavo</li>
        <li class="sql_schema_species_name">Monodelphis domestica</li>
        <li class="sql_schema_species_name">Mus musculus</li>
        <li class="sql_schema_species_name">Nomascus leucogenys</li>
        <li class="sql_schema_species_name">Ornithorhynchus anatinus</li>
        <li class="sql_schema_species_name">Ovis aries</li>
        <li class="sql_schema_species_name">Pan troglodytes</li>
        <li class="sql_schema_species_name">Pongo abelii</li>
        <li class="sql_schema_species_name">Rattus norvegicus</li>
        <li class="sql_schema_species_name">Saccharomyces cerevisiae</li>
        <li class="sql_schema_species_name">Sus scrofa</li>
        <li class="sql_schema_species_name">Taeniopygia guttata</li>
        <li class="sql_schema_species_name">Tetraodon nigroviridis</li>
      </ul>
    </div>
  </td></tr></table></div></div><div class="sql_schema_table">
  <div id="seq_region"class="sql_schema_table_header" style="border-top-color:#000">
    <div class="sql_schema_table_header_left"><span style="background-color:#000"></span>seq_region</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_seq_region" class="help-header sql_schema_show_hide" onclick="show_hide('seq_region','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">This table stores the relationship between Ensembl's internal coordinate system identifiers and traditional chromosome names.</p>

  <div id="div_seq_region" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>seq_region_id</b></td>
        <td class="bg1">INT(10)</td>
        <td class="bg1">-</td>
        <td class="bg1">Primary key. Foreign key references seq_region in core db. Refers to the seq_region which this variant is on, which may be a chromosome, a clone, etc...</td>
        <td class="bg1">primary key</td>
      </tr>

      <tr>
        <td class="bg2"><b>name</b></td>
        <td class="bg2">VARCHAR(40)</td>
        <td class="bg2">-</td>
        <td class="bg2">The name of this sequence region.</td>
        <td class="bg2">unique key: <i>name_cs_idx</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>coord_system_id</b></td>
        <td class="bg1">INT(10)</td>
        <td class="bg1">-</td>
        <td class="bg1">Foreign key references to the <a href="#coord_system" class="sql_schema_link">coord_system</a> table.</td>
        <td class="bg1">unique key: <i>name_cs_idx</i><br />key: <i>cs_idx</i></td>
      </tr>
    </table>
  </div>
<table class="sql_schema_extra"><tr>    <td class="sql_schema_extra_left"">    <p style="font-weight:bold">See also:</p>
  <ul>
      <li><a href="#variation_feature">variation_feature</a></li>
      <li><a href="#compressed_genotype_region">compressed_genotype_region</a></li>
      <li><a href="#read_coverage">read_coverage</a></li>
      <li><a href="#coord_system">coord_system</a></li>
      </ul>
    </td>
  <td class="sql_schema_extra_separator"></td>
  <td class="sql_schema_extra_right" style="padding-left:25px"><p><span>List of species with populated data:</span>
  <a id="s_seq_region" class="help-header sql_schema_show_hide" onclick="show_hide('seq_region','species')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show species
  </a></p>
    <div id="sp_seq_region" style="display:none;">      <ul>
        <li class="sql_schema_species_name">Bos taurus</li>
        <li class="sql_schema_species_name">Canis familiaris</li>
        <li class="sql_schema_species_name">Danio rerio</li>
        <li class="sql_schema_species_name">Drosophila melanogaster</li>
        <li class="sql_schema_species_name">Equus caballus</li>
        <li class="sql_schema_species_name">Felis catus</li>
        <li class="sql_schema_species_name">Gallus gallus</li>
        <li class="sql_schema_species_name">Homo sapiens</li>
        <li class="sql_schema_species_name">Macaca mulatta</li>
        <li class="sql_schema_species_name">Meleagris gallopavo</li>
        <li class="sql_schema_species_name">Monodelphis domestica</li>
        <li class="sql_schema_species_name">Mus musculus</li>
        <li class="sql_schema_species_name">Nomascus leucogenys</li>
        <li class="sql_schema_species_name">Ornithorhynchus anatinus</li>
        <li class="sql_schema_species_name">Ovis aries</li>
        <li class="sql_schema_species_name">Pan troglodytes</li>
        <li class="sql_schema_species_name">Pongo abelii</li>
        <li class="sql_schema_species_name">Rattus norvegicus</li>
        <li class="sql_schema_species_name">Saccharomyces cerevisiae</li>
        <li class="sql_schema_species_name">Sus scrofa</li>
        <li class="sql_schema_species_name">Taeniopygia guttata</li>
        <li class="sql_schema_species_name">Tetraodon nigroviridis</li>
      </ul>
    </div>
  </td></tr></table></div></div><div class="sql_schema_table">
  <div id="submitter_handle"class="sql_schema_table_header" style="border-top-color:#000">
    <div class="sql_schema_table_header_left"><span style="background-color:#000"></span>submitter_handle</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_submitter_handle" class="help-header sql_schema_show_hide" onclick="show_hide('submitter_handle','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">This table holds a short string to distinguish data submitters</p>

  <div id="div_submitter_handle" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>handle_id</b></td>
        <td class="bg1">int(10)</td>
        <td class="bg1">-</td>
        <td class="bg1">Primary key, internal identifier.</td>
        <td class="bg1">primary key</td>
      </tr>

      <tr>
        <td class="bg2"><b>handle</b></td>
        <td class="bg2">varchar(25)</td>
        <td class="bg2">NULL</td>
        <td class="bg2">Short string assigned to the data submitter. </td>
        <td class="bg2">unique: <i> handle </i></td>
      </tr>
    </table>
  </div>
<table class="sql_schema_extra"><tr>    <td class="sql_schema_extra_left"">    <p style="font-weight:bold">See also:</p>
  <ul>
      <li><a href="#allele">allele</a></li>
      </ul>
    </td>
  <td class="sql_schema_extra_separator"></td>
  <td class="sql_schema_extra_right" style="padding-left:25px"><p><span>List of species with populated data:</span>
  <a id="s_submitter_handle" class="help-header sql_schema_show_hide" onclick="show_hide('submitter_handle','species')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show species
  </a></p>
    <div id="sp_submitter_handle" style="display:none;">      <ul>
        <li class="sql_schema_species_name">Bos taurus</li>
        <li class="sql_schema_species_name">Canis familiaris</li>
        <li class="sql_schema_species_name">Danio rerio</li>
        <li class="sql_schema_species_name">Equus caballus</li>
        <li class="sql_schema_species_name">Gallus gallus</li>
        <li class="sql_schema_species_name">Homo sapiens</li>
        <li class="sql_schema_species_name">Mus musculus</li>
        <li class="sql_schema_species_name">Ornithorhynchus anatinus</li>
        <li class="sql_schema_species_name">Ovis aries</li>
        <li class="sql_schema_species_name">Pan troglodytes</li>
        <li class="sql_schema_species_name">Rattus norvegicus</li>
        <li class="sql_schema_species_name">Sus scrofa</li>
        <li class="sql_schema_species_name">Taeniopygia guttata</li>
      </ul>
    </div>
  </td></tr></table></div></div><div class="sql_schema_table">
  <div id="subsnp_handle"class="sql_schema_table_header" style="border-top-color:#000">
    <div class="sql_schema_table_header_left"><span style="background-color:#000"></span>subsnp_handle</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_subsnp_handle" class="help-header sql_schema_show_hide" onclick="show_hide('subsnp_handle','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">This table contains the SubSNP(ss) ID and the name of the submitter handle of dbSNP.</p>

  <div id="div_subsnp_handle" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>subsnp_id</b></td>
        <td class="bg1">int(11)</td>
        <td class="bg1">-</td>
        <td class="bg1">Primary key. It corresponds to the subsnp identifier (ssID) from dbSNP.<br />This ssID is stored in this table without the "ss" prefix. e.g. "120258606" instead of "ss120258606".</td>
        <td class="bg1">primary key</td>
      </tr>

      <tr>
        <td class="bg2"><b>handle</b></td>
        <td class="bg2">varchar(20)</td>
        <td class="bg2">NULL</td>
        <td class="bg2">The name of the dbSNP handler who submitted the ssID.<br />Name of the synonym (a different <b>sample_id</b>).</td>
        <td class="bg2"></td>
      </tr>
    </table>
  </div>
<table class="sql_schema_extra"><tr>    <td class="sql_schema_extra_left"">    <p style="font-weight:bold">See also:</p>
  <ul>
      <li><a href="#allele">allele</a></li>
      <li><a href="#failed_variation">failed_variation</a></li>
      <li><a href="#population_genotype">population_genotype</a></li>
      <li><a href="#sample">sample</a></li>
      <li><a href="#variation_synonym">variation_synonym</a></li>
      </ul>
    </td>
  <td class="sql_schema_extra_separator"></td>
  <td class="sql_schema_extra_right" style="padding-left:25px"><p><span>List of species with populated data:</span>
  <a id="s_subsnp_handle" class="help-header sql_schema_show_hide" onclick="show_hide('subsnp_handle','species')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show species
  </a></p>
    <div id="sp_subsnp_handle" style="display:none;">      <ul>
        <li class="sql_schema_species_name">Bos taurus</li>
        <li class="sql_schema_species_name">Canis familiaris</li>
        <li class="sql_schema_species_name">Danio rerio</li>
        <li class="sql_schema_species_name">Equus caballus</li>
        <li class="sql_schema_species_name">Felis catus</li>
        <li class="sql_schema_species_name">Gallus gallus</li>
        <li class="sql_schema_species_name">Homo sapiens</li>
        <li class="sql_schema_species_name">Macaca mulatta</li>
        <li class="sql_schema_species_name">Meleagris gallopavo</li>
        <li class="sql_schema_species_name">Monodelphis domestica</li>
        <li class="sql_schema_species_name">Mus musculus</li>
        <li class="sql_schema_species_name">Ornithorhynchus anatinus</li>
        <li class="sql_schema_species_name">Ovis aries</li>
        <li class="sql_schema_species_name">Pan troglodytes</li>
        <li class="sql_schema_species_name">Pongo abelii</li>
        <li class="sql_schema_species_name">Rattus norvegicus</li>
        <li class="sql_schema_species_name">Sus scrofa</li>
        <li class="sql_schema_species_name">Taeniopygia guttata</li>
        <li class="sql_schema_species_name">Tetraodon nigroviridis</li>
      </ul>
    </div>
  </td></tr></table></div></div>

<div class="sql_schema_group_header" style="border-color:#22949b">
  <div id="6" class="sql_schema_group_bullet" style="background-color:#22949b"></div>
  <h2 id="6">Phenotype tables</h2>
</div>
<p class="sql_schema_group_header_desc">These tables store information linking entities (variants, genes, QTLs) with phenotypes and other annotations.</p><div class="sql_schema_table">
  <div id="phenotype"class="sql_schema_table_header" style="border-top-color:#22949b">
    <div class="sql_schema_table_header_left"><span style="background-color:#22949b"></span>phenotype</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_phenotype" class="help-header sql_schema_show_hide" onclick="show_hide('phenotype','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">This table stores details of the phenotypes associated with phenotype_features.</p>

  <div id="div_phenotype" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>phenotype_id</b></td>
        <td class="bg1">int(10)</td>
        <td class="bg1">-</td>
        <td class="bg1">Primary key, internal identifier.</td>
        <td class="bg1">primary key</td>
      </tr>

      <tr>
        <td class="bg2"><b>stable_id</b></td>
        <td class="bg2">varchar(255)</td>
        <td class="bg2">NULL</td>
        <td class="bg2">Ensembl stable identifier for the phenotype</td>
        <td class="bg2">key: <i>stable_idx</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>name</b></td>
        <td class="bg1">varchar(50)</td>
        <td class="bg1">NULL</td>
        <td class="bg1">Phenotype short name. e.g. "CAD".</td>
        <td class="bg1">key: <i>name_idx</i></td>
      </tr>

      <tr>
        <td class="bg2"><b>description</b></td>
        <td class="bg2">varchar(255)</td>
        <td class="bg2">NULL</td>
        <td class="bg2">varchar  Phenotype long name. e.g. "Coronary Artery Disease".</td>
        <td class="bg2">unique key: <i>desc_idx</i></td>
      </tr>
    </table>
  </div>
<table class="sql_schema_extra"><tr>    <td class="sql_schema_extra_left"">    <p style="font-weight:bold">See also:</p>
  <ul>
      <li><a href="#phenotype_feature">phenotype_feature</a></li>
      </ul>
    </td>
  <td class="sql_schema_extra_separator"></td>
  <td class="sql_schema_extra_right" style="padding-left:25px"><p><span>List of species with populated data:</span>
  <a id="s_phenotype" class="help-header sql_schema_show_hide" onclick="show_hide('phenotype','species')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show species
  </a></p>
    <div id="sp_phenotype" style="display:none;">      <ul>
        <li class="sql_schema_species_name">Bos taurus</li>
        <li class="sql_schema_species_name">Canis familiaris</li>
        <li class="sql_schema_species_name">Danio rerio</li>
        <li class="sql_schema_species_name">Equus caballus</li>
        <li class="sql_schema_species_name">Felis catus</li>
        <li class="sql_schema_species_name">Gallus gallus</li>
        <li class="sql_schema_species_name">Homo sapiens</li>
        <li class="sql_schema_species_name">Macaca mulatta</li>
        <li class="sql_schema_species_name">Meleagris gallopavo</li>
        <li class="sql_schema_species_name">Mus musculus</li>
        <li class="sql_schema_species_name">Ovis aries</li>
        <li class="sql_schema_species_name">Pan troglodytes</li>
        <li class="sql_schema_species_name">Rattus norvegicus</li>
        <li class="sql_schema_species_name">Sus scrofa</li>
      </ul>
    </div>
  </td></tr></table></div></div><div class="sql_schema_table">
  <div id="phenotype_feature"class="sql_schema_table_header" style="border-top-color:#22949b">
    <div class="sql_schema_table_header_left"><span style="background-color:#22949b"></span>phenotype_feature</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_phenotype_feature" class="help-header sql_schema_show_hide" onclick="show_hide('phenotype_feature','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">This table stores information linking entities (variants, genes, QTLs) and phenotypes.</p>

  <div id="div_phenotype_feature" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>phenotype_feature_id</b></td>
        <td class="bg1">int(11)</td>
        <td class="bg1">-</td>
        <td class="bg1">Primary key, internal identifier.</td>
        <td class="bg1">primary key</td>
      </tr>

      <tr>
        <td class="bg2"><b>phenotype_id</b></td>
        <td class="bg2">int(11)</td>
        <td class="bg2">NULL</td>
        <td class="bg2">Foreign key references to the <a href="#phenotype" class="sql_schema_link">phenotype</a> table.</td>
        <td class="bg2">key: <i>phenotype_idx</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>source_id</b></td>
        <td class="bg1">int(11)</td>
        <td class="bg1">NULL</td>
        <td class="bg1">Foreign key references to the <a href="#source" class="sql_schema_link">source</a> table.</td>
        <td class="bg1">key: <i>source_idx</i></td>
      </tr>

      <tr>
        <td class="bg2"><b>study_id</b></td>
        <td class="bg2">int(11)</td>
        <td class="bg2">NULL</td>
        <td class="bg2">Foreign key references to the <a href="#study" class="sql_schema_link">study</a> table.</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>type</b></td>
        <td class="bg1">ENUM:<ul class="sql_schema_table_column_type">  <li>Gene</li>  <li>Variation</li>  <li>StructuralVariation</li>  <li>SupportingStructuralVariation</li>  <li>QTL</li>  <li>RegulatoryFeature</li></ul></td>
        <td class="bg1">NULL</td>
        <td class="bg1">Type of object associated.</td>
        <td class="bg1">key: <i>object_idx</i><br />key: <i>type_idx</i></td>
      </tr>

      <tr>
        <td class="bg2"><b>object_id</b></td>
        <td class="bg2">varchar(255)</td>
        <td class="bg2">NULL</td>
        <td class="bg2">Stable identifier for associated object.</td>
        <td class="bg2">key: <i>object_idx</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>is_significant</b></td>
        <td class="bg1">tinyint(1)</td>
        <td class="bg1">'1'</td>
        <td class="bg1">Flag indicating if the association is statistically significant in the given study.</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>seq_region_id</b></td>
        <td class="bg2">int(11)</td>
        <td class="bg2">NULL</td>
        <td class="bg2">Foreign key references <a href="#seq_region" class="sql_schema_link">seq_region</a> in core db. Refers to the seq_region which this feature is on, which may be a chromosome, a clone, etc...</td>
        <td class="bg2">key: <i>pos_idx</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>seq_region_start</b></td>
        <td class="bg1">int(11)</td>
        <td class="bg1">NULL</td>
        <td class="bg1">The start position of the feature on the <a href="#seq_region" class="sql_schema_link">seq_region</a>.</td>
        <td class="bg1">key: <i>pos_idx</i></td>
      </tr>

      <tr>
        <td class="bg2"><b>seq_region_end</b></td>
        <td class="bg2">int(11)</td>
        <td class="bg2">NULL</td>
        <td class="bg2">The end position of the feature on the <a href="#seq_region" class="sql_schema_link">seq_region</a>.</td>
        <td class="bg2">key: <i>pos_idx</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>seq_region_strand</b></td>
        <td class="bg1">tinyint(4)</td>
        <td class="bg1">NULL</td>
        <td class="bg1">The orientation of the feature on the <a href="#seq_region" class="sql_schema_link">seq_region</a>.</td>
        <td class="bg1"></td>
      </tr>
    </table>
  </div>
<table class="sql_schema_extra"><tr>    <td class="sql_schema_extra_left"">    <p style="font-weight:bold">See also:</p>
  <ul>
      <li><a href="#variation">variation</a></li>
      <li><a href="#phenotype">phenotype</a></li>
      <li><a href="#source">source</a></li>
      <li><a href="#study">study</a></li>
      <li><a href="#seq_region">seq_region</a></li>
      </ul>
    </td>
  <td class="sql_schema_extra_separator"></td>
  <td class="sql_schema_extra_right" style="padding-left:25px"><p><span>List of species with populated data:</span>
  <a id="s_phenotype_feature" class="help-header sql_schema_show_hide" onclick="show_hide('phenotype_feature','species')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show species
  </a></p>
    <div id="sp_phenotype_feature" style="display:none;">      <ul>
        <li class="sql_schema_species_name">Bos taurus</li>
        <li class="sql_schema_species_name">Canis familiaris</li>
        <li class="sql_schema_species_name">Danio rerio</li>
        <li class="sql_schema_species_name">Equus caballus</li>
        <li class="sql_schema_species_name">Felis catus</li>
        <li class="sql_schema_species_name">Gallus gallus</li>
        <li class="sql_schema_species_name">Homo sapiens</li>
        <li class="sql_schema_species_name">Macaca mulatta</li>
        <li class="sql_schema_species_name">Meleagris gallopavo</li>
        <li class="sql_schema_species_name">Mus musculus</li>
        <li class="sql_schema_species_name">Ovis aries</li>
        <li class="sql_schema_species_name">Pan troglodytes</li>
        <li class="sql_schema_species_name">Rattus norvegicus</li>
        <li class="sql_schema_species_name">Sus scrofa</li>
      </ul>
    </div>
  </td></tr></table></div></div><div class="sql_schema_table">
  <div id="phenotype_feature_attrib"class="sql_schema_table_header" style="border-top-color:#22949b">
    <div class="sql_schema_table_header_left"><span style="background-color:#22949b"></span>phenotype_feature_attrib</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_phenotype_feature_attrib" class="help-header sql_schema_show_hide" onclick="show_hide('phenotype_feature_attrib','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">This table stores additional information on a given phenotype/object association. It is styled as an attrib table to allow for a variety of fields to be populated across different object types.</p>

  <div id="div_phenotype_feature_attrib" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>phenotype_feature_id</b></td>
        <td class="bg1">int(11)</td>
        <td class="bg1">-</td>
        <td class="bg1">Foreign key, references to the <a href="#phenotype_feature" class="sql_schema_link">phenotype_feature</a> table.</td>
        <td class="bg1">key: <i>phenotype_feature_idx</i></td>
      </tr>

      <tr>
        <td class="bg2"><b>attrib_type_id</b></td>
        <td class="bg2">int(11)</td>
        <td class="bg2">NULL</td>
        <td class="bg2">Foreign key references to the <a href="#attrib_type" class="sql_schema_link">attrib_type</a> table.</td>
        <td class="bg2">key: <i>type_value_idx</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>value</b></td>
        <td class="bg1">varchar(255)</td>
        <td class="bg1">NULL</td>
        <td class="bg1">The value of the attribute.</td>
        <td class="bg1">key: <i>type_value_idx</i></td>
      </tr>
    </table>
  </div>
<table class="sql_schema_extra"><tr>    <td class="sql_schema_extra_left"">    <p style="font-weight:bold">See also:</p>
  <ul>
      <li><a href="#phenotype_feature">phenotype_feature</a></li>
      <li><a href="#attrib_type">attrib_type</a></li>
      </ul>
    </td>
  <td class="sql_schema_extra_separator"></td>
  <td class="sql_schema_extra_right" style="padding-left:25px"><p><span>List of species with populated data:</span>
  <a id="s_phenotype_feature_attrib" class="help-header sql_schema_show_hide" onclick="show_hide('phenotype_feature_attrib','species')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show species
  </a></p>
    <div id="sp_phenotype_feature_attrib" style="display:none;">      <ul>
        <li class="sql_schema_species_name">Bos taurus</li>
        <li class="sql_schema_species_name">Canis familiaris</li>
        <li class="sql_schema_species_name">Danio rerio</li>
        <li class="sql_schema_species_name">Equus caballus</li>
        <li class="sql_schema_species_name">Felis catus</li>
        <li class="sql_schema_species_name">Gallus gallus</li>
        <li class="sql_schema_species_name">Homo sapiens</li>
        <li class="sql_schema_species_name">Macaca mulatta</li>
        <li class="sql_schema_species_name">Meleagris gallopavo</li>
        <li class="sql_schema_species_name">Mus musculus</li>
        <li class="sql_schema_species_name">Ovis aries</li>
        <li class="sql_schema_species_name">Pan troglodytes</li>
        <li class="sql_schema_species_name">Rattus norvegicus</li>
        <li class="sql_schema_species_name">Sus scrofa</li>
      </ul>
    </div>
  </td></tr></table></div></div><div class="sql_schema_table">
  <div id="phenotype_ontology_accession"class="sql_schema_table_header" style="border-top-color:#22949b">
    <div class="sql_schema_table_header_left"><span style="background-color:#22949b"></span>phenotype_ontology_accession</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_phenotype_ontology_accession" class="help-header sql_schema_show_hide" onclick="show_hide('phenotype_ontology_accession','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">This table stores accessions of phenotype ontology terms which have been linked to phenotype.descriptions</p>

  <div id="div_phenotype_ontology_accession" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>phenotype_id</b></td>
        <td class="bg1">int(11)</td>
        <td class="bg1">-</td>
        <td class="bg1">Foreign key, references to the <a href="#phenotype" class="sql_schema_link">phenotype</a> table.</td>
        <td class="bg1">primary key</td>
      </tr>

      <tr>
        <td class="bg2"><b>accession</b></td>
        <td class="bg2">varchar(255)</td>
        <td class="bg2">-</td>
        <td class="bg2">The accession of an ontology term held in the ontology database (eg. EFO:0000378) </td>
        <td class="bg2">primary key<br />key: <i>accession_idx</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>mapped_by_attrib</b></td>
        <td class="bg1">SET:<ul class="sql_schema_table_column_type">  <li>437</li>  <li>438</li>  <li>439</li>  <li>440</li>  <li>441</li>  <li>442</li>  <li>443</li>  <li>444</li></ul></td>
        <td class="bg1">NULL</td>
        <td class="bg1">The method used to annotate the phenotype.description with the ontology term</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>mapping_type</b></td>
        <td class="bg2">ENUM:<ul class="sql_schema_table_column_type">  <li>is</li>  <li>involves</li></ul></td>
        <td class="bg2">NULL</td>
        <td class="bg2">The relation defining the association between the ontology term and the phenotype.description</td>
        <td class="bg2"></td>
      </tr>
    </table>
  </div>
<table class="sql_schema_extra"><tr>    <td class="sql_schema_extra_left"">    <p style="font-weight:bold">See also:</p>
  <ul>
      <li><a href="#phenotype">phenotype</a></li>
      </ul>
    </td>
  <td class="sql_schema_extra_separator"></td>
  <td class="sql_schema_extra_right" style="padding-left:25px"><p><span>List of species with populated data:</span>
  <a id="s_phenotype_ontology_accession" class="help-header sql_schema_show_hide" onclick="show_hide('phenotype_ontology_accession','species')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show species
  </a></p>
    <div id="sp_phenotype_ontology_accession" style="display:none;">      <ul>
        <li class="sql_schema_species_name">Bos taurus</li>
        <li class="sql_schema_species_name">Gallus gallus</li>
        <li class="sql_schema_species_name">Homo sapiens</li>
        <li class="sql_schema_species_name">Macaca mulatta</li>
        <li class="sql_schema_species_name">Mus musculus</li>
        <li class="sql_schema_species_name">Rattus norvegicus</li>
      </ul>
    </div>
  </td></tr></table></div></div>

<div class="sql_schema_group_header" style="border-color:#1E90FF">
  <div id="7" class="sql_schema_group_bullet" style="background-color:#1E90FF"></div>
  <h2 id="7">Protein tables</h2>
</div>
<p class="sql_schema_group_header_desc">These tables define the protein prediction data.</p><div class="sql_schema_table">
  <div id="protein_function_predictions"class="sql_schema_table_header" style="border-top-color:#1E90FF">
    <div class="sql_schema_table_header_left"><span style="background-color:#1E90FF"></span>protein_function_predictions</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_protein_function_predictions" class="help-header sql_schema_show_hide" onclick="show_hide('protein_function_predictions','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">Contains encoded protein function predictions for every protein-coding transcript in this species.</p>

  <div id="div_protein_function_predictions" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>translation_md5_id</b></td>
        <td class="bg1">int(11)</td>
        <td class="bg1">-</td>
        <td class="bg1">Identifies the MD5 hash corresponding to the protein sequence to which  these predictions apply</td>
        <td class="bg1">primary key</td>
      </tr>

      <tr>
        <td class="bg2"><b>analysis_attrib_id</b></td>
        <td class="bg2">int(11)</td>
        <td class="bg2">-</td>
        <td class="bg2">Identifies the analysis (sift, polyphen etc.) that produced these predictions </td>
        <td class="bg2">primary key</td>
      </tr>

      <tr>
        <td class="bg1"><b>prediction_matrix</b></td>
        <td class="bg1">mediumblob</td>
        <td class="bg1">NULL</td>
        <td class="bg1">A compressed binary string containing the predictions for all possible  amino acid substitutions in this protein. See the explanation <a href="/info/genome/variation/predicted_data.html#nsSNP">here</a></td>
        <td class="bg1"></td>
      </tr>
    </table>
  </div>
<table class="sql_schema_extra"><tr>    <td class="sql_schema_extra_left"">    <p style="font-weight:bold">See also:</p>
  <ul>
      <li><a href="#translation_md5">translation_md5</a></li>
      <li><a href="#attrib">attrib</a></li>
      </ul>
    </td>
  <td class="sql_schema_extra_separator"></td>
  <td class="sql_schema_extra_right" style="padding-left:25px"><p><span>List of species with populated data:</span>
  <a id="s_protein_function_predictions" class="help-header sql_schema_show_hide" onclick="show_hide('protein_function_predictions','species')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show species
  </a></p>
    <div id="sp_protein_function_predictions" style="display:none;">      <ul>
        <li class="sql_schema_species_name">Bos taurus</li>
        <li class="sql_schema_species_name">Canis familiaris</li>
        <li class="sql_schema_species_name">Danio rerio</li>
        <li class="sql_schema_species_name">Equus caballus</li>
        <li class="sql_schema_species_name">Gallus gallus</li>
        <li class="sql_schema_species_name">Homo sapiens</li>
        <li class="sql_schema_species_name">Mus musculus</li>
        <li class="sql_schema_species_name">Ovis aries</li>
        <li class="sql_schema_species_name">Rattus norvegicus</li>
        <li class="sql_schema_species_name">Sus scrofa</li>
      </ul>
    </div>
  </td></tr></table></div></div><div class="sql_schema_table">
  <div id="protein_function_predictions_attrib"class="sql_schema_table_header" style="border-top-color:#1E90FF">
    <div class="sql_schema_table_header_left"><span style="background-color:#1E90FF"></span>protein_function_predictions_attrib</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_protein_function_predictions_attrib" class="help-header sql_schema_show_hide" onclick="show_hide('protein_function_predictions_attrib','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">Contains information on the data use in protein function predictions</p>

  <div id="div_protein_function_predictions_attrib" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>translation_md5_id</b></td>
        <td class="bg1">int(11)</td>
        <td class="bg1">-</td>
        <td class="bg1">Identifies the MD5 hash corresponding to the protein sequence to which  these data use in prediction apply</td>
        <td class="bg1">primary key</td>
      </tr>

      <tr>
        <td class="bg2"><b>analysis_attrib_id</b></td>
        <td class="bg2">int(11)</td>
        <td class="bg2">-</td>
        <td class="bg2">Identifies the analysis (sift, polyphen etc.) that produced these values </td>
        <td class="bg2">primary key</td>
      </tr>

      <tr>
        <td class="bg1"><b>attrib_type_id</b></td>
        <td class="bg1">int(11)</td>
        <td class="bg1">-</td>
        <td class="bg1">Key into the <a href="#attrib_type" class="sql_schema_link">attrib_type</a> table, identifies the type of this attribute</td>
        <td class="bg1">primary key</td>
      </tr>

      <tr>
        <td class="bg2"><b>position_values</b></td>
        <td class="bg2">blob</td>
        <td class="bg2">NULL</td>
        <td class="bg2">A compressed binary string containing data relevant to the quality of the predictions</td>
        <td class="bg2"></td>
      </tr>
    </table>
  </div>
<table class="sql_schema_extra"><tr>    <td class="sql_schema_extra_left"">    <p style="font-weight:bold">See also:</p>
  <ul>
      <li><a href="#protein_function_predictions">protein_function_predictions</a></li>
      <li><a href="#attrib_type">attrib_type</a></li>
      </ul>
    </td>
  <td class="sql_schema_extra_separator"></td>
  <td class="sql_schema_extra_right" style="padding-left:25px"><p><span>List of species with populated data:</span>
  <a id="s_protein_function_predictions_attrib" class="help-header sql_schema_show_hide" onclick="show_hide('protein_function_predictions_attrib','species')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show species
  </a></p>
    <div id="sp_protein_function_predictions_attrib" style="display:none;">      <ul>
        <li class="sql_schema_species_name">Bos taurus</li>
        <li class="sql_schema_species_name">Canis familiaris</li>
        <li class="sql_schema_species_name">Danio rerio</li>
        <li class="sql_schema_species_name">Equus caballus</li>
        <li class="sql_schema_species_name">Gallus gallus</li>
        <li class="sql_schema_species_name">Homo sapiens</li>
        <li class="sql_schema_species_name">Mus musculus</li>
        <li class="sql_schema_species_name">Ovis aries</li>
        <li class="sql_schema_species_name">Rattus norvegicus</li>
        <li class="sql_schema_species_name">Sus scrofa</li>
      </ul>
    </div>
  </td></tr></table></div></div><div class="sql_schema_table">
  <div id="translation_md5"class="sql_schema_table_header" style="border-top-color:#1E90FF">
    <div class="sql_schema_table_header_left"><span style="background-color:#1E90FF"></span>translation_md5</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_translation_md5" class="help-header sql_schema_show_hide" onclick="show_hide('translation_md5','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">Maps a hex MD5 hash of a translation sequence to an ID used for the protein function predictions</p>

  <div id="div_translation_md5" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>translation_md5_id</b></td>
        <td class="bg1">int(11)</td>
        <td class="bg1">-</td>
        <td class="bg1">Primary key</td>
        <td class="bg1">primary key</td>
      </tr>

      <tr>
        <td class="bg2"><b>translation_md5</b></td>
        <td class="bg2">char(32)</td>
        <td class="bg2">-</td>
        <td class="bg2">Hex MD5 hash of a translation sequence</td>
        <td class="bg2">unique key: <i>md5_idx</i></td>
      </tr>
    </table>
  </div>
<table class="sql_schema_extra"><tr>    <td class="sql_schema_extra_left"">    <p style="font-weight:bold">See also:</p>
  <ul>
      <li><a href="#protein_function_predictions">protein_function_predictions</a></li>
      </ul>
    </td>
  <td class="sql_schema_extra_separator"></td>
  <td class="sql_schema_extra_right" style="padding-left:25px"><p><span>List of species with populated data:</span>
  <a id="s_translation_md5" class="help-header sql_schema_show_hide" onclick="show_hide('translation_md5','species')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show species
  </a></p>
    <div id="sp_translation_md5" style="display:none;">      <ul>
        <li class="sql_schema_species_name">Bos taurus</li>
        <li class="sql_schema_species_name">Canis familiaris</li>
        <li class="sql_schema_species_name">Danio rerio</li>
        <li class="sql_schema_species_name">Equus caballus</li>
        <li class="sql_schema_species_name">Gallus gallus</li>
        <li class="sql_schema_species_name">Homo sapiens</li>
        <li class="sql_schema_species_name">Mus musculus</li>
        <li class="sql_schema_species_name">Ovis aries</li>
        <li class="sql_schema_species_name">Rattus norvegicus</li>
        <li class="sql_schema_species_name">Sus scrofa</li>
      </ul>
    </div>
  </td></tr></table></div></div>

<div class="sql_schema_group_header" style="border-color:#FF8500">
  <div id="8" class="sql_schema_group_bullet" style="background-color:#FF8500"></div>
  <h2 id="8">Sample tables</h2>
</div>
<p class="sql_schema_group_header_desc">These tables define the sample, individual and population information.</p><div class="sql_schema_table">
  <div id="display_group"class="sql_schema_table_header" style="border-top-color:#FF8500">
    <div class="sql_schema_table_header_left"><span style="background-color:#FF8500"></span>display_group</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_display_group" class="help-header sql_schema_show_hide" onclick="show_hide('display_group','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">Used to store groups of populations displayed separately on the Population Genetics page</p>

  <div id="div_display_group" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>display_group_id</b></td>
        <td class="bg1">int(10)</td>
        <td class="bg1">-</td>
        <td class="bg1">Primary key, internal identifier.</td>
        <td class="bg1">primary key</td>
      </tr>

      <tr>
        <td class="bg2"><b>display_priority</b></td>
        <td class="bg2">int(10)</td>
        <td class="bg2">-</td>
        <td class="bg2">Priority level for group (smallest number is highest on page) </td>
        <td class="bg2">unique: <i> display_priority </i></td>
      </tr>

      <tr>
        <td class="bg1"><b>display_name</b></td>
        <td class="bg1">varchar(255)</td>
        <td class="bg1">-</td>
        <td class="bg1">Name of the group to be displayed as the table header.</td>
        <td class="bg1">unique: <i> display_name </i></td>
      </tr>
    </table>
  </div>
<table class="sql_schema_extra"><tr>    <td class="sql_schema_extra_left"">    <p style="font-weight:bold">See also:</p>
  <ul>
      <li><a href="#population">population</a></li>
      </ul>
    </td>
  <td class="sql_schema_extra_separator"></td>
  <td class="sql_schema_extra_right" style="padding-left:25px"><p><span>List of species with populated data:</span>
  <a id="s_display_group" class="help-header sql_schema_show_hide" onclick="show_hide('display_group','species')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show species
  </a></p>
    <div id="sp_display_group" style="display:none;">      <ul>
        <li class="sql_schema_species_name">Bos taurus</li>
        <li class="sql_schema_species_name">Homo sapiens</li>
        <li class="sql_schema_species_name">Mus musculus</li>
        <li class="sql_schema_species_name">Ovis aries</li>
      </ul>
    </div>
  </td></tr></table></div></div><div class="sql_schema_table">
  <div id="individual"class="sql_schema_table_header" style="border-top-color:#FF8500">
    <div class="sql_schema_table_header_left"><span style="background-color:#FF8500"></span>individual</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_individual" class="help-header sql_schema_show_hide" onclick="show_hide('individual','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">Stores information about an identifiable individual, including gender and the identifiers of the individual's parents (if known).</p>

  <div id="div_individual" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>individual_id</b></td>
        <td class="bg1">int(10)</td>
        <td class="bg1">-</td>
        <td class="bg1">Primary key, internal identifier.</td>
        <td class="bg1">primary key</td>
      </tr>

      <tr>
        <td class="bg2"><b>name</b></td>
        <td class="bg2">varchar(255)</td>
        <td class="bg2">NULL</td>
        <td class="bg2">Name of the individual.</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>description</b></td>
        <td class="bg1">text</td>
        <td class="bg1">NULL</td>
        <td class="bg1">Description of the individual.</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>gender</b></td>
        <td class="bg2">ENUM:<ul class="sql_schema_table_column_type">  <li>Male</li>  <li> Female</li>  <li> Unknown</li></ul></td>
        <td class="bg2">'Unknown'</td>
        <td class="bg2">The sex of this individual.</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>father_individual_id</b></td>
        <td class="bg1">int(10)</td>
        <td class="bg1">NULL</td>
        <td class="bg1">Self referential ID, the father of this individual if known.</td>
        <td class="bg1">key: <i>father_individual_idx</i></td>
      </tr>

      <tr>
        <td class="bg2"><b>mother_individual_id</b></td>
        <td class="bg2">int(10)</td>
        <td class="bg2">NULL</td>
        <td class="bg2">Self referential ID, the mother of this individual if known.</td>
        <td class="bg2">key: <i>mother_individual_idx</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>individual_type_id</b></td>
        <td class="bg1">int(10)</td>
        <td class="bg1">0</td>
        <td class="bg1">Foreign key references to the <a href="#individual_type" class="sql_schema_link">individual_type</a> table.</td>
        <td class="bg1"></td>
      </tr>
    </table>
  </div>
<table class="sql_schema_extra"><tr>    <td class="sql_schema_extra_left"">    <p style="font-weight:bold">See also:</p>
  <ul>
      <li><a href="#individual_synonym">individual_synonym</a></li>
      <li><a href="#individual_type">individual_type</a></li>
      <li><a href="#sample">sample</a></li>
      </ul>
    </td>
  <td class="sql_schema_extra_separator"></td>
  <td class="sql_schema_extra_right" style="padding-left:25px"><p><span>List of species with populated data:</span>
  <a id="s_individual" class="help-header sql_schema_show_hide" onclick="show_hide('individual','species')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show species
  </a></p>
    <div id="sp_individual" style="display:none;">      <ul>
        <li class="sql_schema_species_name">Bos taurus</li>
        <li class="sql_schema_species_name">Canis familiaris</li>
        <li class="sql_schema_species_name">Danio rerio</li>
        <li class="sql_schema_species_name">Drosophila melanogaster</li>
        <li class="sql_schema_species_name">Equus caballus</li>
        <li class="sql_schema_species_name">Gallus gallus</li>
        <li class="sql_schema_species_name">Homo sapiens</li>
        <li class="sql_schema_species_name">Macaca mulatta</li>
        <li class="sql_schema_species_name">Monodelphis domestica</li>
        <li class="sql_schema_species_name">Mus musculus</li>
        <li class="sql_schema_species_name">Nomascus leucogenys</li>
        <li class="sql_schema_species_name">Ornithorhynchus anatinus</li>
        <li class="sql_schema_species_name">Ovis aries</li>
        <li class="sql_schema_species_name">Pan troglodytes</li>
        <li class="sql_schema_species_name">Pongo abelii</li>
        <li class="sql_schema_species_name">Rattus norvegicus</li>
        <li class="sql_schema_species_name">Saccharomyces cerevisiae</li>
        <li class="sql_schema_species_name">Sus scrofa</li>
        <li class="sql_schema_species_name">Taeniopygia guttata</li>
        <li class="sql_schema_species_name">Tetraodon nigroviridis</li>
      </ul>
    </div>
  </td></tr></table></div></div><div class="sql_schema_table">
  <div id="individual_synonym"class="sql_schema_table_header" style="border-top-color:#FF8500">
    <div class="sql_schema_table_header_left"><span style="background-color:#FF8500"></span>individual_synonym</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_individual_synonym" class="help-header sql_schema_show_hide" onclick="show_hide('individual_synonym','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">Used to store alternative names for individuals when data comes from multiple sources.</p>

  <div id="div_individual_synonym" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>synonym_id</b></td>
        <td class="bg1">int(10)</td>
        <td class="bg1">-</td>
        <td class="bg1">Primary key, internal identifier.</td>
        <td class="bg1">primary key</td>
      </tr>

      <tr>
        <td class="bg2"><b>individual_id</b></td>
        <td class="bg2">int(10)</td>
        <td class="bg2">-</td>
        <td class="bg2">Foreign key references to the <a href="#individual" class="sql_schema_link">individual</a> table.</td>
        <td class="bg2">key: <i>individual_idx</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>source_id</b></td>
        <td class="bg1">int(10)</td>
        <td class="bg1">-</td>
        <td class="bg1">Foreign key references to the <a href="#source" class="sql_schema_link">source</a> table.</td>
        <td class="bg1">key: <i>name, source_id</i></td>
      </tr>

      <tr>
        <td class="bg2"><b>name</b></td>
        <td class="bg2">varchar(255)</td>
        <td class="bg2">NULL</td>
        <td class="bg2">Name of the synonym.</td>
        <td class="bg2">key: <i>name, source_id</i></td>
      </tr>
    </table>
  </div>
<table class="sql_schema_extra"><tr>    <td class="sql_schema_extra_left"">    <p style="font-weight:bold">See also:</p>
  <ul>
      <li><a href="#individual">individual</a></li>
      <li><a href="#source">source</a></li>
      </ul>
    </td>
  <td class="sql_schema_extra_separator"></td>
  <td class="sql_schema_extra_right" style="padding-left:25px"><p><span>List of species with populated data:</span>
  <a id="s_individual_synonym" class="help-header sql_schema_show_hide" onclick="show_hide('individual_synonym','species')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show species
  </a></p>
    <div id="sp_individual_synonym" style="display:none;">      <ul>
        <li class="sql_schema_species_name">Bos taurus</li>
        <li class="sql_schema_species_name">Canis familiaris</li>
        <li class="sql_schema_species_name">Danio rerio</li>
        <li class="sql_schema_species_name">Equus caballus</li>
        <li class="sql_schema_species_name">Gallus gallus</li>
        <li class="sql_schema_species_name">Homo sapiens</li>
        <li class="sql_schema_species_name">Macaca mulatta</li>
        <li class="sql_schema_species_name">Monodelphis domestica</li>
        <li class="sql_schema_species_name">Mus musculus</li>
        <li class="sql_schema_species_name">Ornithorhynchus anatinus</li>
        <li class="sql_schema_species_name">Ovis aries</li>
        <li class="sql_schema_species_name">Pan troglodytes</li>
        <li class="sql_schema_species_name">Pongo abelii</li>
        <li class="sql_schema_species_name">Rattus norvegicus</li>
        <li class="sql_schema_species_name">Sus scrofa</li>
        <li class="sql_schema_species_name">Taeniopygia guttata</li>
        <li class="sql_schema_species_name">Tetraodon nigroviridis</li>
      </ul>
    </div>
  </td></tr></table></div></div><div class="sql_schema_table">
  <div id="individual_type"class="sql_schema_table_header" style="border-top-color:#FF8500">
    <div class="sql_schema_table_header_left"><span style="background-color:#FF8500"></span>individual_type</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_individual_type" class="help-header sql_schema_show_hide" onclick="show_hide('individual_type','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">This table gives a detailed description for each of the possible individual types: fully_inbred, partly_inbred, outbred, mutant  </p>

  <div id="div_individual_type" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>individual_type_id</b></td>
        <td class="bg1">int(0)</td>
        <td class="bg1">-</td>
        <td class="bg1">Primary key, internal identifier.</td>
        <td class="bg1">primary key</td>
      </tr>

      <tr>
        <td class="bg2"><b>name</b></td>
        <td class="bg2">varchar(255)</td>
        <td class="bg2">-</td>
        <td class="bg2">Short name of the individual type. e.g. "fully_inbred","mutant".</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>description</b></td>
        <td class="bg1">text</td>
        <td class="bg1">NULL</td>
        <td class="bg1">Long name of the individual type.</td>
        <td class="bg1"></td>
      </tr>
    </table>
  </div>
<div class="sql_schema_table_examples"><p class="sql_schema_table_example_header">Example:</p><div class="sql_schema_table_example_content"><p>See below the list of individual types: </p>
      <div>
        <div class="sql_schema_table_example_query">
          <pre><span class="sql_schema_sql_highlight">SELECT</span> * <span class="sql_schema_sql_highlight">FROM</span> individual_type;</pre>
        </div>
        <div class="sql_schema_table_example_button">
  <a id="e_individual_type" class="help-header sql_schema_show_hide" onclick="show_hide('individual_type','query results')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show query results
  </a></div>
        <div style="clear:both"></div>
      </div>
      
  <div id="ex_individual_type" style="display:none;">
    <table class="ss sql_schema_table_example_result">
      <tr><th>individual_type_id</th><th>name</th><th>description</th></tr>      <tr><td>1</td><td>fully_inbred</td><td>multiple organisms have the same genome sequence</td></tr>      <tr class="bg2"><td>2</td><td>partly_inbred</td><td>single organisms have reduced genome variability due to human intervention</td></tr>      <tr><td>3</td><td>outbred</td><td>a single organism which breeds freely</td></tr>      <tr class="bg2"><td>4</td><td>mutant</td><td>a single or multiple organisms with the same genome sequence that have a natural or experimentally induced mutation</td></tr>    </table>
  </div></div></div><table class="sql_schema_extra"><tr>    <td class="sql_schema_extra_left"">    <p style="font-weight:bold">See also:</p>
  <ul>
      <li><a href="#individual">individual</a></li>
      </ul>
    </td>
  <td class="sql_schema_extra_separator"></td>
  <td class="sql_schema_extra_right" style="padding-left:25px"><p><span>List of species with populated data:</span>
  <a id="s_individual_type" class="help-header sql_schema_show_hide" onclick="show_hide('individual_type','species')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show species
  </a></p>
    <div id="sp_individual_type" style="display:none;">      <ul>
        <li class="sql_schema_species_name">Bos taurus</li>
        <li class="sql_schema_species_name">Canis familiaris</li>
        <li class="sql_schema_species_name">Danio rerio</li>
        <li class="sql_schema_species_name">Drosophila melanogaster</li>
        <li class="sql_schema_species_name">Equus caballus</li>
        <li class="sql_schema_species_name">Felis catus</li>
        <li class="sql_schema_species_name">Gallus gallus</li>
        <li class="sql_schema_species_name">Homo sapiens</li>
        <li class="sql_schema_species_name">Macaca mulatta</li>
        <li class="sql_schema_species_name">Meleagris gallopavo</li>
        <li class="sql_schema_species_name">Monodelphis domestica</li>
        <li class="sql_schema_species_name">Mus musculus</li>
        <li class="sql_schema_species_name">Nomascus leucogenys</li>
        <li class="sql_schema_species_name">Ornithorhynchus anatinus</li>
        <li class="sql_schema_species_name">Ovis aries</li>
        <li class="sql_schema_species_name">Pan troglodytes</li>
        <li class="sql_schema_species_name">Pongo abelii</li>
        <li class="sql_schema_species_name">Rattus norvegicus</li>
        <li class="sql_schema_species_name">Saccharomyces cerevisiae</li>
        <li class="sql_schema_species_name">Sus scrofa</li>
        <li class="sql_schema_species_name">Taeniopygia guttata</li>
        <li class="sql_schema_species_name">Tetraodon nigroviridis</li>
      </ul>
    </div>
  </td></tr></table></div></div><div class="sql_schema_table">
  <div id="population"class="sql_schema_table_header" style="border-top-color:#FF8500">
    <div class="sql_schema_table_header_left"><span style="background-color:#FF8500"></span>population</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_population" class="help-header sql_schema_show_hide" onclick="show_hide('population','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">Stores information about a population. A population may be an ethnic group (e.g. Caucasian, Hispanic), assay group (e.g. 24 Europeans), phenotypic group (e.g. blue eyed, diabetes) etc. Populations may be composed of other populations by defining relationships in the population_structure table.</p>

  <div id="div_population" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>population_id</b></td>
        <td class="bg1">int(10)</td>
        <td class="bg1">-</td>
        <td class="bg1">Primary key, internal identifier.</td>
        <td class="bg1">primary key</td>
      </tr>

      <tr>
        <td class="bg2"><b>name</b></td>
        <td class="bg2">varchar(255)</td>
        <td class="bg2">NULL</td>
        <td class="bg2">Name of the population.</td>
        <td class="bg2">key: <i>name_idx</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>size</b></td>
        <td class="bg1">int(10)</td>
        <td class="bg1">NULL</td>
        <td class="bg1">Size of the population.</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>description</b></td>
        <td class="bg2">text</td>
        <td class="bg2">NULL</td>
        <td class="bg2">Description of the population.</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>collection</b></td>
        <td class="bg1">tinyint(1)</td>
        <td class="bg1">0</td>
        <td class="bg1">Flag indicating if the population is defined based on geography (0) or a collection of individuals/samples with respect to some other criteria (1).</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>freqs_from_gts</b></td>
        <td class="bg2">tinyint(1)</td>
        <td class="bg2">NULL</td>
        <td class="bg2">Flag indicating if the population frequencies can be retrieved from the allele table (0) or from the individual/sample genotypes (1).</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>display</b></td>
        <td class="bg1">ENUM:<ul class="sql_schema_table_column_type">  <li>LD</li>  <li> MARTDISPLAYABLE</li>  <li> UNDISPLAYABLE</li></ul></td>
        <td class="bg1">'UNDISPLAYABLE'</td>
        <td class="bg1">Information used by BioMart.</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>display_group_id</b></td>
        <td class="bg2">tinyint(1)</td>
        <td class="bg2">NULL</td>
        <td class="bg2">Used to group population for display on the Population Genetics page</td>
        <td class="bg2"></td>
      </tr>
    </table>
  </div>
<table class="sql_schema_extra"><tr>    <td class="sql_schema_extra_left"">    <p style="font-weight:bold">See also:</p>
  <ul>
      <li><a href="#population_synonym">population_synonym</a></li>
      <li><a href="#sample_population">sample_population</a></li>
      <li><a href="#population_structure">population_structure</a></li>
      <li><a href="#population_genotype">population_genotype</a></li>
      <li><a href="#allele">allele</a></li>
      <li><a href="#display_group">display_group</a></li>
      </ul>
    </td>
  <td class="sql_schema_extra_separator"></td>
  <td class="sql_schema_extra_right" style="padding-left:25px"><p><span>List of species with populated data:</span>
  <a id="s_population" class="help-header sql_schema_show_hide" onclick="show_hide('population','species')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show species
  </a></p>
    <div id="sp_population" style="display:none;">      <ul>
        <li class="sql_schema_species_name">Bos taurus</li>
        <li class="sql_schema_species_name">Canis familiaris</li>
        <li class="sql_schema_species_name">Danio rerio</li>
        <li class="sql_schema_species_name">Drosophila melanogaster</li>
        <li class="sql_schema_species_name">Equus caballus</li>
        <li class="sql_schema_species_name">Felis catus</li>
        <li class="sql_schema_species_name">Gallus gallus</li>
        <li class="sql_schema_species_name">Homo sapiens</li>
        <li class="sql_schema_species_name">Macaca mulatta</li>
        <li class="sql_schema_species_name">Meleagris gallopavo</li>
        <li class="sql_schema_species_name">Monodelphis domestica</li>
        <li class="sql_schema_species_name">Mus musculus</li>
        <li class="sql_schema_species_name">Ornithorhynchus anatinus</li>
        <li class="sql_schema_species_name">Ovis aries</li>
        <li class="sql_schema_species_name">Pan troglodytes</li>
        <li class="sql_schema_species_name">Pongo abelii</li>
        <li class="sql_schema_species_name">Rattus norvegicus</li>
        <li class="sql_schema_species_name">Saccharomyces cerevisiae</li>
        <li class="sql_schema_species_name">Sus scrofa</li>
        <li class="sql_schema_species_name">Taeniopygia guttata</li>
        <li class="sql_schema_species_name">Tetraodon nigroviridis</li>
      </ul>
    </div>
  </td></tr></table></div></div><div class="sql_schema_table">
  <div id="population_structure"class="sql_schema_table_header" style="border-top-color:#FF8500">
    <div class="sql_schema_table_header_left"><span style="background-color:#FF8500"></span>population_structure</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_population_structure" class="help-header sql_schema_show_hide" onclick="show_hide('population_structure','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">This table stores hierarchical relationships between populations by relating them as populations and sub-populations.</p>

  <div id="div_population_structure" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>super_population_id</b></td>
        <td class="bg1">int(10)</td>
        <td class="bg1">-</td>
        <td class="bg1">Foreign key references to the population table.</td>
        <td class="bg1">unique: <i>super_population_idx</i></td>
      </tr>

      <tr>
        <td class="bg2"><b>sub_population_id</b></td>
        <td class="bg2">int(10)</td>
        <td class="bg2">-</td>
        <td class="bg2">Foreign key references to the population table.</td>
        <td class="bg2">unique: <i>super_population_idx</i><br />key: <i>sub_population_idx</i></td>
      </tr>
    </table>
  </div>
<table class="sql_schema_extra"><tr>    <td class="sql_schema_extra_left"">    <p style="font-weight:bold">See also:</p>
  <ul>
      <li><a href="#population">population</a></li>
      </ul>
    </td>
  <td class="sql_schema_extra_separator"></td>
  <td class="sql_schema_extra_right" style="padding-left:25px"><p><span>List of species with populated data:</span>
  <a id="s_population_structure" class="help-header sql_schema_show_hide" onclick="show_hide('population_structure','species')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show species
  </a></p>
    <div id="sp_population_structure" style="display:none;">      <ul>
        <li class="sql_schema_species_name">Bos taurus</li>
        <li class="sql_schema_species_name">Canis familiaris</li>
        <li class="sql_schema_species_name">Danio rerio</li>
        <li class="sql_schema_species_name">Drosophila melanogaster</li>
        <li class="sql_schema_species_name">Equus caballus</li>
        <li class="sql_schema_species_name">Gallus gallus</li>
        <li class="sql_schema_species_name">Homo sapiens</li>
        <li class="sql_schema_species_name">Macaca mulatta</li>
        <li class="sql_schema_species_name">Meleagris gallopavo</li>
        <li class="sql_schema_species_name">Monodelphis domestica</li>
        <li class="sql_schema_species_name">Mus musculus</li>
        <li class="sql_schema_species_name">Ornithorhynchus anatinus</li>
        <li class="sql_schema_species_name">Ovis aries</li>
        <li class="sql_schema_species_name">Pan troglodytes</li>
        <li class="sql_schema_species_name">Pongo abelii</li>
        <li class="sql_schema_species_name">Rattus norvegicus</li>
        <li class="sql_schema_species_name">Sus scrofa</li>
        <li class="sql_schema_species_name">Taeniopygia guttata</li>
      </ul>
    </div>
  </td></tr></table></div></div><div class="sql_schema_table">
  <div id="population_synonym"class="sql_schema_table_header" style="border-top-color:#FF8500">
    <div class="sql_schema_table_header_left"><span style="background-color:#FF8500"></span>population_synonym</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_population_synonym" class="help-header sql_schema_show_hide" onclick="show_hide('population_synonym','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">Used to store alternative names for populations when data comes from multiple sources.</p>

  <div id="div_population_synonym" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>synonym_id</b></td>
        <td class="bg1">int(10)</td>
        <td class="bg1">-</td>
        <td class="bg1">Primary key, internal identifier.</td>
        <td class="bg1">primary key</td>
      </tr>

      <tr>
        <td class="bg2"><b>population_id</b></td>
        <td class="bg2">int(10)</td>
        <td class="bg2">-</td>
        <td class="bg2">Foreign key references to the <a href="#population" class="sql_schema_link">population</a> table.</td>
        <td class="bg2">key: <i>population_idx</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>source_id</b></td>
        <td class="bg1">int(10)</td>
        <td class="bg1">-</td>
        <td class="bg1">Foreign key references to the <a href="#source" class="sql_schema_link">source</a> table.</td>
        <td class="bg1">key: <i>name, source_id</i></td>
      </tr>

      <tr>
        <td class="bg2"><b>name</b></td>
        <td class="bg2">varchar(255)</td>
        <td class="bg2">NULL</td>
        <td class="bg2">Name of the synonym.</td>
        <td class="bg2">key: <i>name, source_id</i></td>
      </tr>
    </table>
  </div>
<table class="sql_schema_extra"><tr>    <td class="sql_schema_extra_left"">    <p style="font-weight:bold">See also:</p>
  <ul>
      <li><a href="#population">population</a></li>
      <li><a href="#source">source</a></li>
      </ul>
    </td>
  <td class="sql_schema_extra_separator"></td>
  <td class="sql_schema_extra_right" style="padding-left:25px"><p><span>List of species with populated data:</span>
  <a id="s_population_synonym" class="help-header sql_schema_show_hide" onclick="show_hide('population_synonym','species')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show species
  </a></p>
    <div id="sp_population_synonym" style="display:none;">      <ul>
        <li class="sql_schema_species_name">Bos taurus</li>
        <li class="sql_schema_species_name">Canis familiaris</li>
        <li class="sql_schema_species_name">Danio rerio</li>
        <li class="sql_schema_species_name">Equus caballus</li>
        <li class="sql_schema_species_name">Felis catus</li>
        <li class="sql_schema_species_name">Gallus gallus</li>
        <li class="sql_schema_species_name">Homo sapiens</li>
        <li class="sql_schema_species_name">Macaca mulatta</li>
        <li class="sql_schema_species_name">Meleagris gallopavo</li>
        <li class="sql_schema_species_name">Monodelphis domestica</li>
        <li class="sql_schema_species_name">Mus musculus</li>
        <li class="sql_schema_species_name">Ornithorhynchus anatinus</li>
        <li class="sql_schema_species_name">Ovis aries</li>
        <li class="sql_schema_species_name">Pan troglodytes</li>
        <li class="sql_schema_species_name">Pongo abelii</li>
        <li class="sql_schema_species_name">Rattus norvegicus</li>
        <li class="sql_schema_species_name">Sus scrofa</li>
        <li class="sql_schema_species_name">Taeniopygia guttata</li>
        <li class="sql_schema_species_name">Tetraodon nigroviridis</li>
      </ul>
    </div>
  </td></tr></table></div></div><div class="sql_schema_table">
  <div id="sample"class="sql_schema_table_header" style="border-top-color:#FF8500">
    <div class="sql_schema_table_header_left"><span style="background-color:#FF8500"></span>sample</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_sample" class="help-header sql_schema_show_hide" onclick="show_hide('sample','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">Stores information about a sample. A sample belongs to an individual. An individual can have multiple samples. A sample can belong only to one individual. A sample can be associated with a study. </p>

  <div id="div_sample" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>sample_id</b></td>
        <td class="bg1">int(10)</td>
        <td class="bg1">-</td>
        <td class="bg1">Primary key, internal identifier.</td>
        <td class="bg1">primary key</td>
      </tr>

      <tr>
        <td class="bg2"><b>individual_id</b></td>
        <td class="bg2">int(10)</td>
        <td class="bg2">-</td>
        <td class="bg2">Foreign key references to the <a href="#individual" class="sql_schema_link">individual</a> table.</td>
        <td class="bg2">key: <i>individual_idx</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>name</b></td>
        <td class="bg1">varchar(255)</td>
        <td class="bg1">NULL</td>
        <td class="bg1">Name of the sample.</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>description</b></td>
        <td class="bg2">text</td>
        <td class="bg2">NULL</td>
        <td class="bg2">Description of the sample.</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>study_id</b></td>
        <td class="bg1">int(10)</td>
        <td class="bg1">NULL</td>
        <td class="bg1">Foreign key references to the <a href="#study" class="sql_schema_link">study</a> table.</td>
        <td class="bg1">key: <i>study_idx</i></td>
      </tr>

      <tr>
        <td class="bg2"><b>display</b></td>
        <td class="bg2">ENUM:<ul class="sql_schema_table_column_type">  <li>REFERENCE</li>  <li>DEFAULT</li>  <li>DISPLAYABLE</li>  <li>UNDISPLAYABLE</li>  <li>LD</li>  <li>MARTDISPLAYABLE</li></ul></td>
        <td class="bg2">'UNDISPLAYABLE'</td>
        <td class="bg2">Information used by the website: samples with little information are filtered from some web displays.</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>has_coverage</b></td>
        <td class="bg1">tinyint(1)</td>
        <td class="bg1">'0'</td>
        <td class="bg1">Indicate if the sample has coverage data populated in the read coverage table</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>variation_set_id</b></td>
        <td class="bg2">SET:<ul class="sql_schema_table_column_type">  <li>1</li>  <li>2</li>  <li>3</li>  <li>4</li>  <li>5</li>  <li>6</li>  <li>7</li>  <li>8</li>  <li>9</li>  <li>10</li>  <li>11</li>  <li>12</li>  <li>13</li>  <li>14</li>  <li>15</li>  <li>16</li>  <li>17</li>  <li>18</li>  <li>19</li>  <li>20</li>  <li>21</li>  <li>22</li>  <li>23</li>  <li>24</li>  <li>25</li>  <li>26</li>  <li>27</li>  <li>28</li>  <li>29</li>  <li>30</li>  <li>31</li>  <li>32</li>  <li>33</li>  <li>34</li>  <li>35</li>  <li>36</li>  <li>37</li>  <li>38</li>  <li>39</li>  <li>40</li>  <li>41</li>  <li>42</li>  <li>43</li>  <li>44</li>  <li>45</li>  <li>46</li>  <li>47</li>  <li>48</li>  <li>49</li>  <li>50</li>  <li>51</li>  <li>52</li>  <li>53</li>  <li>54</li>  <li>55</li>  <li>56</li>  <li>57</li>  <li>58</li>  <li>59</li>  <li>60</li>  <li>61</li>  <li>62</li>  <li>63</li>  <li>64</li></ul></td>
        <td class="bg2">NULL</td>
        <td class="bg2">Indicates the variation sets for which a sample has genotypes</td>
        <td class="bg2"></td>
      </tr>
    </table>
  </div>
<table class="sql_schema_extra"><tr>    <td class="sql_schema_extra_left"">    <p style="font-weight:bold">See also:</p>
  <ul>
      <li><a href="#individual">individual</a></li>
      <li><a href="#study">study</a></li>
      <li><a href="#sample_population">sample_population</a></li>
      <li><a href="#sample_synonym">sample_synonym</a></li>
      <li><a href="#sample_genotype_multiple_bp">sample_genotype_multiple_bp</a></li>
      <li><a href="#read_coverage">read_coverage</a></li>
      <li><a href="#compressed_genotype_region">compressed_genotype_region</a></li>
      <li><a href="#compressed_genotype_var">compressed_genotype_var</a></li>
      <li><a href="#variation_set">variation_set</a></li>
      </ul>
    </td>
  <td class="sql_schema_extra_separator"></td>
  <td class="sql_schema_extra_right" style="padding-left:25px"><p><span>List of species with populated data:</span>
  <a id="s_sample" class="help-header sql_schema_show_hide" onclick="show_hide('sample','species')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show species
  </a></p>
    <div id="sp_sample" style="display:none;">      <ul>
        <li class="sql_schema_species_name">Bos taurus</li>
        <li class="sql_schema_species_name">Canis familiaris</li>
        <li class="sql_schema_species_name">Danio rerio</li>
        <li class="sql_schema_species_name">Drosophila melanogaster</li>
        <li class="sql_schema_species_name">Equus caballus</li>
        <li class="sql_schema_species_name">Gallus gallus</li>
        <li class="sql_schema_species_name">Homo sapiens</li>
        <li class="sql_schema_species_name">Macaca mulatta</li>
        <li class="sql_schema_species_name">Monodelphis domestica</li>
        <li class="sql_schema_species_name">Mus musculus</li>
        <li class="sql_schema_species_name">Nomascus leucogenys</li>
        <li class="sql_schema_species_name">Ornithorhynchus anatinus</li>
        <li class="sql_schema_species_name">Ovis aries</li>
        <li class="sql_schema_species_name">Pan troglodytes</li>
        <li class="sql_schema_species_name">Pongo abelii</li>
        <li class="sql_schema_species_name">Rattus norvegicus</li>
        <li class="sql_schema_species_name">Saccharomyces cerevisiae</li>
        <li class="sql_schema_species_name">Sus scrofa</li>
        <li class="sql_schema_species_name">Taeniopygia guttata</li>
        <li class="sql_schema_species_name">Tetraodon nigroviridis</li>
      </ul>
    </div>
  </td></tr></table></div></div><div class="sql_schema_table">
  <div id="sample_population"class="sql_schema_table_header" style="border-top-color:#FF8500">
    <div class="sql_schema_table_header_left"><span style="background-color:#FF8500"></span>sample_population</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_sample_population" class="help-header sql_schema_show_hide" onclick="show_hide('sample_population','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">This table resolves the many-to-many relationship between the sample and population tables; i.e. samples may belong to more than one population. Hence it is composed of rows of sample and population identifiers.</p>

  <div id="div_sample_population" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>sample_id</b></td>
        <td class="bg1">int(10)</td>
        <td class="bg1">-</td>
        <td class="bg1">Foreign key references to the <a href="#sample" class="sql_schema_link">sample</a> table.</td>
        <td class="bg1">key: <i>sample_idx</i></td>
      </tr>

      <tr>
        <td class="bg2"><b>population_id</b></td>
        <td class="bg2">int(10)</td>
        <td class="bg2">-</td>
        <td class="bg2">Foreign key references to the <a href="#population" class="sql_schema_link">population</a> table.</td>
        <td class="bg2">key: <i>population_idx</i></td>
      </tr>
    </table>
  </div>
<table class="sql_schema_extra"><tr>    <td class="sql_schema_extra_left"">    <p style="font-weight:bold">See also:</p>
  <ul>
      <li><a href="#sample">sample</a></li>
      <li><a href="#population">population</a></li>
      </ul>
    </td>
  <td class="sql_schema_extra_separator"></td>
  <td class="sql_schema_extra_right" style="padding-left:25px"><p><span>List of species with populated data:</span>
  <a id="s_sample_population" class="help-header sql_schema_show_hide" onclick="show_hide('sample_population','species')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show species
  </a></p>
    <div id="sp_sample_population" style="display:none;">      <ul>
        <li class="sql_schema_species_name">Bos taurus</li>
        <li class="sql_schema_species_name">Canis familiaris</li>
        <li class="sql_schema_species_name">Danio rerio</li>
        <li class="sql_schema_species_name">Drosophila melanogaster</li>
        <li class="sql_schema_species_name">Equus caballus</li>
        <li class="sql_schema_species_name">Gallus gallus</li>
        <li class="sql_schema_species_name">Homo sapiens</li>
        <li class="sql_schema_species_name">Macaca mulatta</li>
        <li class="sql_schema_species_name">Monodelphis domestica</li>
        <li class="sql_schema_species_name">Mus musculus</li>
        <li class="sql_schema_species_name">Ornithorhynchus anatinus</li>
        <li class="sql_schema_species_name">Ovis aries</li>
        <li class="sql_schema_species_name">Pan troglodytes</li>
        <li class="sql_schema_species_name">Pongo abelii</li>
        <li class="sql_schema_species_name">Rattus norvegicus</li>
        <li class="sql_schema_species_name">Saccharomyces cerevisiae</li>
        <li class="sql_schema_species_name">Sus scrofa</li>
        <li class="sql_schema_species_name">Taeniopygia guttata</li>
        <li class="sql_schema_species_name">Tetraodon nigroviridis</li>
      </ul>
    </div>
  </td></tr></table></div></div><div class="sql_schema_table">
  <div id="sample_synonym"class="sql_schema_table_header" style="border-top-color:#FF8500">
    <div class="sql_schema_table_header_left"><span style="background-color:#FF8500"></span>sample_synonym</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_sample_synonym" class="help-header sql_schema_show_hide" onclick="show_hide('sample_synonym','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">Used to store alternative names for samples when data comes from multiple sources.</p>

  <div id="div_sample_synonym" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>synonym_id</b></td>
        <td class="bg1">int(10)</td>
        <td class="bg1">-</td>
        <td class="bg1">Primary key, internal identifier.</td>
        <td class="bg1">primary key</td>
      </tr>

      <tr>
        <td class="bg2"><b>sample_id</b></td>
        <td class="bg2">int(10)</td>
        <td class="bg2">-</td>
        <td class="bg2">Foreign key references to the <a href="#sample" class="sql_schema_link">sample</a> table.</td>
        <td class="bg2">key: <i>sample_idx</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>source_id</b></td>
        <td class="bg1">int(10)</td>
        <td class="bg1">-</td>
        <td class="bg1">Foreign key references to the <a href="#source" class="sql_schema_link">source</a> table.</td>
        <td class="bg1">key: <i>name, source_id</i></td>
      </tr>

      <tr>
        <td class="bg2"><b>name</b></td>
        <td class="bg2">varchar(255)</td>
        <td class="bg2">NULL</td>
        <td class="bg2">Name of the synonym.</td>
        <td class="bg2">key: <i>name, source_id</i></td>
      </tr>
    </table>
  </div>
<table class="sql_schema_extra"><tr>    <td class="sql_schema_extra_left"">    <p style="font-weight:bold">See also:</p>
  <ul>
      <li><a href="#sample">sample</a></li>
      <li><a href="#source">source</a></li>
      </ul>
    </td>
</tr></table></div></div>

<div class="sql_schema_group_header" style="border-color:#7CFC00">
  <div id="9" class="sql_schema_group_bullet" style="background-color:#7CFC00"></div>
  <h2 id="9">Source/study tables</h2>
</div>
<p class="sql_schema_group_header_desc">These tables define the variation source and study information.</p><div class="sql_schema_table">
  <div id="associate_study"class="sql_schema_table_header" style="border-top-color:#7CFC00">
    <div class="sql_schema_table_header_left"><span style="background-color:#7CFC00"></span>associate_study</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_associate_study" class="help-header sql_schema_show_hide" onclick="show_hide('associate_study','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">This table contains identifiers of associated studies (e.g. NHGRI and EGA studies with the same PubMed identifier).</p>

  <div id="div_associate_study" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>study1_id</b></td>
        <td class="bg1">int(10)</td>
        <td class="bg1">-</td>
        <td class="bg1">Primary key. Foreign key references to the <a href="#study" class="sql_schema_link">study</a> table.</td>
        <td class="bg1">primary key</td>
      </tr>

      <tr>
        <td class="bg2"><b>study2_id</b></td>
        <td class="bg2">int(10)</td>
        <td class="bg2">-</td>
        <td class="bg2">Primary key. Foreign key references to the <a href="#study" class="sql_schema_link">study</a> table.</td>
        <td class="bg2">primary key</td>
      </tr>
    </table>
  </div>
<table class="sql_schema_extra"><tr>    <td class="sql_schema_extra_left"">    <p style="font-weight:bold">See also:</p>
  <ul>
      <li><a href="#study">study</a></li>
      </ul>
    </td>
  <td class="sql_schema_extra_separator"></td>
  <td class="sql_schema_extra_right" style="padding-left:25px"><p><span>List of species with populated data:</span>
  <a id="s_associate_study" class="help-header sql_schema_show_hide" onclick="show_hide('associate_study','species')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show species
  </a></p>
    <div id="sp_associate_study" style="display:none;">      <ul>
        <li class="sql_schema_species_name">Homo sapiens</li>
      </ul>
    </div>
  </td></tr></table></div></div><div class="sql_schema_table">
  <div id="publication"class="sql_schema_table_header" style="border-top-color:#7CFC00">
    <div class="sql_schema_table_header_left"><span style="background-color:#7CFC00"></span>publication</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_publication" class="help-header sql_schema_show_hide" onclick="show_hide('publication','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">This table contains details of publications citing variations. This information comes from dbSNP, UCSC and Europe PMC.</p>

  <div id="div_publication" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>publication_id</b></td>
        <td class="bg1">int(10)</td>
        <td class="bg1">-</td>
        <td class="bg1">Primary key, internal identifier.</td>
        <td class="bg1">primary key</td>
      </tr>

      <tr>
        <td class="bg2"><b>title</b></td>
        <td class="bg2">varchar(255)</td>
        <td class="bg2">NULL</td>
        <td class="bg2">Title of the publication</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>authors</b></td>
        <td class="bg1">varchar(255)</td>
        <td class="bg1">NULL</td>
        <td class="bg1">Authors of the publication</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>pmid</b></td>
        <td class="bg2">int(10)</td>
        <td class="bg2">NULL</td>
        <td class="bg2">The PubMed id for the publication if available</td>
        <td class="bg2">key: <i>pmid_idx</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>pmcid</b></td>
        <td class="bg1">varchar(255)</td>
        <td class="bg1">NULL</td>
        <td class="bg1">The PubMed Central id for the publication if available</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>year</b></td>
        <td class="bg2">int(10)</td>
        <td class="bg2">NULL</td>
        <td class="bg2">The year the publication was published</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>doi</b></td>
        <td class="bg1">varchar(50)</td>
        <td class="bg1">NULL</td>
        <td class="bg1">The DOI (Digital Object Identifier) for the publication</td>
        <td class="bg1">key: <i>doi_idx</i></td>
      </tr>

      <tr>
        <td class="bg2"><b>ucsc_id</b></td>
        <td class="bg2">varchar(50)</td>
        <td class="bg2">NULL</td>
        <td class="bg2">The external id used in the UCSC database and URL</td>
        <td class="bg2"></td>
      </tr>
    </table>
  </div>
<table class="sql_schema_extra"><tr>    <td class="sql_schema_extra_left"">    <p style="font-weight:bold">See also:</p>
  <ul>
      <li><a href="#variation_citation">variation_citation</a></li>
      </ul>
    </td>
  <td class="sql_schema_extra_separator"></td>
  <td class="sql_schema_extra_right" style="padding-left:25px"><p><span>List of species with populated data:</span>
  <a id="s_publication" class="help-header sql_schema_show_hide" onclick="show_hide('publication','species')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show species
  </a></p>
    <div id="sp_publication" style="display:none;">      <ul>
        <li class="sql_schema_species_name">Bos taurus</li>
        <li class="sql_schema_species_name">Canis familiaris</li>
        <li class="sql_schema_species_name">Equus caballus</li>
        <li class="sql_schema_species_name">Gallus gallus</li>
        <li class="sql_schema_species_name">Homo sapiens</li>
        <li class="sql_schema_species_name">Mus musculus</li>
        <li class="sql_schema_species_name">Ovis aries</li>
        <li class="sql_schema_species_name">Rattus norvegicus</li>
        <li class="sql_schema_species_name">Sus scrofa</li>
      </ul>
    </div>
  </td></tr></table></div></div><div class="sql_schema_table">
  <div id="source"class="sql_schema_table_header" style="border-top-color:#7CFC00">
    <div class="sql_schema_table_header_left"><span style="background-color:#7CFC00"></span>source</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_source" class="help-header sql_schema_show_hide" onclick="show_hide('source','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">This table contains details of the source from which a variation is derived. Most commonly this is NCBI's dbSNP; other sources include SNPs called by Ensembl.<br /> You can see the complete list, by species, <a href="/info/genome/variation/sources_documentation.html">here</a>.</p>

  <div id="div_source" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>source_id</b></td>
        <td class="bg1">int(10)</td>
        <td class="bg1">-</td>
        <td class="bg1">Primary key, internal identifier.</td>
        <td class="bg1">primary key</td>
      </tr>

      <tr>
        <td class="bg2"><b>name</b></td>
        <td class="bg2">varchar(24)</td>
        <td class="bg2">-</td>
        <td class="bg2">Name of the source. e.g. "dbSNP"</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>version</b></td>
        <td class="bg1">int</td>
        <td class="bg1">NULL</td>
        <td class="bg1">Version number of the source (if available). e.g. "132"</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>description</b></td>
        <td class="bg2">varchar(400)</td>
        <td class="bg2">NULL</td>
        <td class="bg2">Description of the source.</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>url</b></td>
        <td class="bg1">varchar(255)</td>
        <td class="bg1">NULL</td>
        <td class="bg1">URL of the source.</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>type</b></td>
        <td class="bg2">ENUM:<ul class="sql_schema_table_column_type">  <li>chip</li>  <li>lsdb</li></ul></td>
        <td class="bg2">NULL</td>
        <td class="bg2">Define the type of the source, e.g. 'chip'</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>somatic_status</b></td>
        <td class="bg1">ENUM:<ul class="sql_schema_table_column_type">  <li>germline</li>  <li>somatic</li>  <li>mixed</li></ul></td>
        <td class="bg1">'germline'</td>
        <td class="bg1">Indicates if this source includes somatic or germline mutations, or a mixture</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>data_types</b></td>
        <td class="bg2">SET:<ul class="sql_schema_table_column_type">  <li>variation</li>  <li>variation_synonym</li>  <li>structural_variation</li>  <li>phenotype_feature</li>  <li>study</li></ul></td>
        <td class="bg2">NULL</td>
        <td class="bg2">Indicates the type(s) of data provided by the source  </td>
        <td class="bg2"></td>
      </tr>
    </table>
  </div>
<div class="sql_schema_table_examples"><p class="sql_schema_table_example_header">Example:</p><div class="sql_schema_table_example_content"><p>See below the command listing all the data sources in the human variation database: </p>
      <div>
        <div class="sql_schema_table_example_query">
          <pre><span class="sql_schema_sql_highlight">SELECT</span> * <span class="sql_schema_sql_highlight">FROM</span> source <span class="sql_schema_sql_highlight">ORDER</span> <span class="sql_schema_sql_highlight">BY</span> source_id;</pre>
        </div>
        <div class="sql_schema_table_example_button">
  <a id="e_source" class="help-header sql_schema_show_hide" onclick="show_hide('source','query results')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show query results
  </a></div>
        <div style="clear:both"></div>
      </div>
      
  <div id="ex_source" style="display:none;">
    <table class="ss sql_schema_table_example_result">
      <tr><th>source_id</th><th>name</th><th>version</th><th>description</th><th>url</th><th>type</th><th>somatic_status</th><th>data_types</th></tr>      <tr><td>1</td><td>dbSNP</td><td>147</td><td>Variants (including SNPs and indels) imported from dbSNP</td><td>http://www.ncbi.nlm.nih.gov/projects/SNP/</td><td></td><td>mixed</td><td>variation</td></tr>      <tr class="bg2"><td>2</td><td>Archive dbSNP</td><td>147</td><td>Former variants names imported from dbSNP</td><td>http://www.ncbi.nlm.nih.gov/projects/SNP/</td><td></td><td>mixed</td><td>variation_synonym</td></tr>      <tr><td>3</td><td>LSDB</td><td>147</td><td>Variants dbSNP annotates as being from LSDBs</td><td>http://www.ncbi.nlm.nih.gov/projects/SNP/</td><td>lsdb</td><td>germline</td><td>variation_synonym</td></tr>      <tr class="bg2"><td>6</td><td>PhenCode</td><td>20140430</td><td>PhenCode is a collaborative project to better understand the relationship between genotype and phenotype in humans</td><td>http://phencode.bx.psu.edu/</td><td></td><td></td><td>variation,variation_synonym</td></tr>      <tr><td>7</td><td>ESP</td><td>20141103</td><td>Data from NHLBI ESP version v.0.0.30. The goal of the NHLBI GO Exome Sequencing Project is to discover novel genes and mechanisms contributing to heart, lung and blood disorders by sequencing the protein coding regions of the human genome.</td><td>http://evs.gs.washington.edu/EVS/</td><td></td><td>germline</td><td>variation</td></tr>      <tr class="bg2"><td>8</td><td>HGMD-PUBLIC</td><td>20162</td><td>Variants from HGMD-PUBLIC dataset June 2016 (mapped to GRCh38)</td><td>http://www.hgmd.cf.ac.uk/ac/index.php</td><td></td><td>germline</td><td>variation,phenotype_feature</td></tr>      <tr><td>9</td><td>PAHdb</td><td></td><td>Phenylalanine hydroxylase database</td><td>http://www.pahdb.mcgill.ca/</td><td>lsdb</td><td>germline</td><td>variation_synonym</td></tr>      <tr class="bg2"><td>10</td><td>KAT6BDB</td><td></td><td>K(lysine) acetyltransferase 6B database, BCM</td><td>https://grenada.lumc.nl/LOVD2/BCM/home.php?select_db=KAT6B</td><td>lsdb</td><td>germline</td><td>variation_synonym</td></tr>      <tr><td>11</td><td>DGVa</td><td>201607</td><td>Database of Genomic Variants Archive</td><td>http://www.ebi.ac.uk/dgva/</td><td></td><td>mixed</td><td>structural_variation,phenotype_feature,study</td></tr>      <tr class="bg2"><td>13</td><td>NHGRI-EBI GWAS catalog</td><td>20160727</td><td>Variants associated with phenotype data from the NHGRI-EBI GWAS catalog</td><td>http://www.ebi.ac.uk/gwas/</td><td></td><td>germline</td><td>phenotype_feature,study</td></tr>      <tr><td>15</td><td>EGA</td><td>201607</td><td>Variants imported from the European Genome-phenome Archive with phenotype association</td><td>http://www.ebi.ac.uk/ega/</td><td></td><td>germline</td><td>study</td></tr>      <tr class="bg2"><td>16</td><td>Uniprot</td><td>20160706</td><td>Variants with protein annotation imported from Uniprot</td><td>http://www.uniprot.org/</td><td></td><td>mixed</td><td>variation_synonym,phenotype_feature,study</td></tr>      <tr><td>18</td><td>OMIM</td><td>20160727</td><td>Variations linked to entries in the Online Mendelian Inheritance in Man (OMIM) database</td><td>http://www.omim.org/</td><td></td><td>germline</td><td>phenotype_feature,study</td></tr>      <tr class="bg2"><td>20</td><td>Illumina_CytoSNP12v1</td><td>1</td><td>Variants from the Illumina Cyto SNP-12 v1 whole genome SNP genotyping chip designed for cytogenetic analysis</td><td>http://www.illumina.com/</td><td>chip</td><td>germline</td><td>structural_variation</td></tr>      <tr><td>21</td><td>Illumina_Human660W-quad</td><td></td><td>Variants from the Illumina Human 660W-Quad whole genome SNP genotyping chip designed for association studies</td><td>http://www.illumina.com/</td><td>chip</td><td>germline</td><td>structural_variation</td></tr>      <tr class="bg2"><td>22</td><td>Illumina_Human1M-duo</td><td>3</td><td>Variants from the Illumina Human 1M-Duo v3 whole genome SNP genotyping chip designed for association studies</td><td>http://www.illumina.com/</td><td>chip</td><td>germline</td><td>structural_variation</td></tr>      <tr><td>23</td><td>Affy GenomeWideSNP_6 CNV</td><td></td><td>Copy Number Variation (CNV) probes from the Affymetrix Genome-Wide Human SNP Array 6.0</td><td>http://www.affymetrix.com/</td><td>chip</td><td>germline</td><td>structural_variation</td></tr>      <tr class="bg2"><td>26</td><td>COSMIC</td><td>77</td><td>Somatic mutations found in human cancers from the COSMIC catalogue</td><td>http://cancer.sanger.ac.uk/cosmic/</td><td></td><td>somatic</td><td>variation,phenotype_feature</td></tr>      <tr><td>32</td><td>ClinVar</td><td>201607</td><td>Variants of clinical significance imported from ClinVar</td><td>http://www.ncbi.nlm.nih.gov/clinvar/</td><td></td><td>germline</td><td>variation,variation_synonym,phenotype_feature</td></tr>      <tr class="bg2"><td>33</td><td>DDG2P</td><td>20160815</td><td>Developmental Disorders Genotype-to-Phenotype Database</td><td>http://decipher.sanger.ac.uk/</td><td></td><td>germline</td><td>phenotype_feature</td></tr>      <tr><td>34</td><td>GIANT</td><td>1</td><td>The Genetic Investigation of ANthropometric Traits (GIANT) consortium is an international collaboration that seeks to identify genetic loci that modulate human body size and shape, including height and measures of obesity.</td><td>http://www.broadinstitute.org/collaboration/giant/index.php/Main_Page</td><td></td><td>germline</td><td>phenotype_feature</td></tr>      <tr class="bg2"><td>35</td><td>MIM morbid</td><td>20160812</td><td>Online Mendelian Inheritance in Man (OMIM) database</td><td>http://www.omim.org/</td><td></td><td>germline</td><td>phenotype_feature</td></tr>      <tr><td>36</td><td>Orphanet</td><td>20160801</td><td>The portal for rare diseases and drugs</td><td>http://www.orpha.net/</td><td></td><td>germline</td><td>phenotype_feature</td></tr>      <tr class="bg2"><td>37</td><td>MAGIC</td><td>1</td><td>MAGIC (the Meta-Analyses of Glucose and Insulin-related traits Consortium) represents a collaborative effort to combine data from multiple GWAS to identify additional loci that impact on glycemic and metabolic traits</td><td>http://www.magicinvestigators.org/</td><td></td><td>germline</td><td>phenotype_feature</td></tr>      <tr><td>39</td><td>HbVar</td><td></td><td>A Database of Human Hemoglobin Variants and Thalassemias</td><td>http://globin.bx.psu.edu/hbvar/menu.html</td><td>lsdb</td><td>germline</td><td>variation_synonym</td></tr>      <tr class="bg2"><td>40</td><td>LMDD</td><td></td><td>Leiden Muscular Dystrophy Database</td><td>http://www.dmd.nl/</td><td>lsdb</td><td>germline</td><td>variation_synonym</td></tr>      <tr><td>41</td><td>OIVD</td><td></td><td>Osteogenesis Imperfecta Variant Database</td><td>https://oi.gene.le.ac.uk/home.php</td><td>lsdb</td><td>germline</td><td>variation_synonym</td></tr>      <tr class="bg2"><td>42</td><td>dbPEX</td><td></td><td>dbPEX, PEX Gene Database</td><td>http://www.dbpex.org/home.php?select_db=PEX1</td><td>lsdb</td><td>germline</td><td>variation_synonym</td></tr>      <tr><td>43</td><td>Infevers</td><td></td><td>The registry of Hereditary Auto-inflammatory Disorders Mutations </td><td>http://fmf.igh.cnrs.fr/ISSAID/infevers/index.php</td><td>lsdb</td><td>germline</td><td>variation_synonym</td></tr>      <tr class="bg2"><td>46</td><td>dbGaP</td><td>201405</td><td>The database of Genotypes and Phenotypes.</td><td>http://www.ncbi.nlm.nih.gov/gap</td><td></td><td>germline</td><td>phenotype_feature,study</td></tr>      <tr><td>47</td><td>AMDGC</td><td>1</td><td>The AMD Gene consortium is an international collaboration that seeks to identify genetic loci associated with age-related macular degeneration .</td><td>http://www.sph.umich.edu/csg/abecasis/public/amdgene2012/</td><td></td><td>germline</td><td>phenotype_feature,study</td></tr>      <tr class="bg2"><td>48</td><td>GEFOS</td><td>1</td><td>The GEnetic Factors for OSteoporosis Consortium international collaboration that seeks to identify common risk gene variants for osteoporosis</td><td>http://www.gefos.org</td><td></td><td>germline</td><td>phenotype_feature</td></tr>      <tr><td>49</td><td>IIBDGC</td><td>1</td><td>International Inflammatory Bowel Disease Genetics Consortium</td><td>http://www.ibdgenetics.org/</td><td></td><td>germline</td><td>phenotype_feature</td></tr>      <tr class="bg2"><td>50</td><td>Teslovich</td><td>1</td><td>Biological, clinical and population relevance of 95 loci for blood lipids</td><td>http://www.sph.umich.edu/csg/abecasis/public/lipids2010/</td><td></td><td>germline</td><td>phenotype_feature</td></tr>      <tr><td>52</td><td>GOA</td><td>20160812</td><td>Inferred from Mutant Phenotype (IMP), annotated by Gene Ontology Annotation</td><td>http://www.ebi.ac.uk/GOA</td><td></td><td>mixed</td><td>phenotype_feature,study</td></tr>      <tr class="bg2"><td>54</td><td>Cancer Gene Census</td><td>20160416</td><td>Catalog of genes of which mutations have been causally implicated in cancer</td><td>http://cancer.sanger.ac.uk/census</td><td></td><td>somatic</td><td>phenotype_feature,study</td></tr>      <tr><td>55</td><td>dbSNP HGVS</td><td>147</td><td>HGVS annotation from dbSNP</td><td>http://www.ncbi.nlm.nih.gov/projects/SNP/</td><td></td><td>mixed</td><td>variation_synonym</td></tr>    </table>
  </div></div></div><table class="sql_schema_extra"><tr>    <td class="sql_schema_extra_left"">    <p style="font-weight:bold">See also:</p>
  <ul>
      <li><a href="#variation">variation</a></li>
      <li><a href="#variation_synonym">variation_synonym</a></li>
      <li><a href="#variation_feature">variation_feature</a></li>
      <li><a href="#individual_synonym">individual_synonym</a></li>
      <li><a href="#phenotype_feature">phenotype_feature</a></li>
      <li><a href="#population_synonym">population_synonym</a></li>
      <li><a href="#sample_synonym">sample_synonym</a></li>
      <li><a href="#structural_variation">structural_variation</a></li>
      <li><a href="#structural_variation_feature">structural_variation_feature</a></li>
      <li><a href="#study">study</a></li>
      </ul>
    </td>
  <td class="sql_schema_extra_separator"></td>
  <td class="sql_schema_extra_right" style="padding-left:25px"><p><span>List of species with populated data:</span>
  <a id="s_source" class="help-header sql_schema_show_hide" onclick="show_hide('source','species')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show species
  </a></p>
    <div id="sp_source" style="display:none;">      <ul>
        <li class="sql_schema_species_name">Bos taurus</li>
        <li class="sql_schema_species_name">Canis familiaris</li>
        <li class="sql_schema_species_name">Danio rerio</li>
        <li class="sql_schema_species_name">Drosophila melanogaster</li>
        <li class="sql_schema_species_name">Equus caballus</li>
        <li class="sql_schema_species_name">Felis catus</li>
        <li class="sql_schema_species_name">Gallus gallus</li>
        <li class="sql_schema_species_name">Homo sapiens</li>
        <li class="sql_schema_species_name">Macaca mulatta</li>
        <li class="sql_schema_species_name">Meleagris gallopavo</li>
        <li class="sql_schema_species_name">Monodelphis domestica</li>
        <li class="sql_schema_species_name">Mus musculus</li>
        <li class="sql_schema_species_name">Nomascus leucogenys</li>
        <li class="sql_schema_species_name">Ornithorhynchus anatinus</li>
        <li class="sql_schema_species_name">Ovis aries</li>
        <li class="sql_schema_species_name">Pan troglodytes</li>
        <li class="sql_schema_species_name">Pongo abelii</li>
        <li class="sql_schema_species_name">Rattus norvegicus</li>
        <li class="sql_schema_species_name">Saccharomyces cerevisiae</li>
        <li class="sql_schema_species_name">Sus scrofa</li>
        <li class="sql_schema_species_name">Taeniopygia guttata</li>
        <li class="sql_schema_species_name">Tetraodon nigroviridis</li>
      </ul>
    </div>
  </td></tr></table></div></div><div class="sql_schema_table">
  <div id="study"class="sql_schema_table_header" style="border-top-color:#7CFC00">
    <div class="sql_schema_table_header_left"><span style="background-color:#7CFC00"></span>study</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_study" class="help-header sql_schema_show_hide" onclick="show_hide('study','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">This table contains details of the studies. The studies information can come from internal studies (DGVa, EGA) or from external studies (UniProt, NHGRI, ...).</p>

  <div id="div_study" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>study_id</b></td>
        <td class="bg1">int(10)</td>
        <td class="bg1">-</td>
        <td class="bg1">Primary key, internal identifier.</td>
        <td class="bg1">primary key</td>
      </tr>

      <tr>
        <td class="bg2"><b>source_id</b></td>
        <td class="bg2">int(10)</td>
        <td class="bg2">-</td>
        <td class="bg2">Foreign key references to the <a href="#source" class="sql_schema_link">source</a> table.</td>
        <td class="bg2">key: <i>source_idx</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>name</b></td>
        <td class="bg1">varchar(255)</td>
        <td class="bg1">null</td>
        <td class="bg1">Name of the study. e.g. "EGAS00000000001"</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>description</b></td>
        <td class="bg2">text</td>
        <td class="bg2">NULL</td>
        <td class="bg2">Description of the study.</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>url</b></td>
        <td class="bg1">varchar(255)</td>
        <td class="bg1">NULL</td>
        <td class="bg1">URL to find the study data (http or ftp).</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>external_reference</b></td>
        <td class="bg2">varchar(255)</td>
        <td class="bg2">NULL</td>
        <td class="bg2">The PubMed/id or project name associated with this study.</td>
        <td class="bg2">key: <i>external_reference_idx</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>study_type</b></td>
        <td class="bg1">varchar(255)</td>
        <td class="bg1">NULL</td>
        <td class="bg1">Displays the type of the study (e.g. genome-wide association study, control-set, case-set, curated, ...).</td>
        <td class="bg1"></td>
      </tr>
    </table>
  </div>
<table class="sql_schema_extra"><tr>    <td class="sql_schema_extra_left"">    <p style="font-weight:bold">See also:</p>
  <ul>
      <li><a href="#source">source</a></li>
      <li><a href="#phenotype_feature">phenotype_feature</a></li>
      <li><a href="#structural_variation">structural_variation</a></li>
      <li><a href="#structural_variation_feature">structural_variation_feature</a></li>
      </ul>
    </td>
  <td class="sql_schema_extra_separator"></td>
  <td class="sql_schema_extra_right" style="padding-left:25px"><p><span>List of species with populated data:</span>
  <a id="s_study" class="help-header sql_schema_show_hide" onclick="show_hide('study','species')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show species
  </a></p>
    <div id="sp_study" style="display:none;">      <ul>
        <li class="sql_schema_species_name">Bos taurus</li>
        <li class="sql_schema_species_name">Canis familiaris</li>
        <li class="sql_schema_species_name">Danio rerio</li>
        <li class="sql_schema_species_name">Equus caballus</li>
        <li class="sql_schema_species_name">Gallus gallus</li>
        <li class="sql_schema_species_name">Homo sapiens</li>
        <li class="sql_schema_species_name">Macaca mulatta</li>
        <li class="sql_schema_species_name">Meleagris gallopavo</li>
        <li class="sql_schema_species_name">Mus musculus</li>
        <li class="sql_schema_species_name">Ovis aries</li>
        <li class="sql_schema_species_name">Rattus norvegicus</li>
        <li class="sql_schema_species_name">Sus scrofa</li>
      </ul>
    </div>
  </td></tr></table></div></div><div class="sql_schema_table">
  <div id="variation_citation"class="sql_schema_table_header" style="border-top-color:#7CFC00">
    <div class="sql_schema_table_header_left"><span style="background-color:#7CFC00"></span>variation_citation</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_variation_citation" class="help-header sql_schema_show_hide" onclick="show_hide('variation_citation','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">This table links a variation to a publication</p>

  <div id="div_variation_citation" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>publication_id</b></td>
        <td class="bg1">int(10)</td>
        <td class="bg1">-</td>
        <td class="bg1">Primary key, internal identifier.</td>
        <td class="bg1">primary key</td>
      </tr>

      <tr>
        <td class="bg2"><b>variation_id</b></td>
        <td class="bg2">int(10)</td>
        <td class="bg2">-</td>
        <td class="bg2">Primary key, foreign key references <a href="#variation" class="sql_schema_link">variation</a></td>
        <td class="bg2">primary key</td>
      </tr>
    </table>
  </div>
<table class="sql_schema_extra"><tr>    <td class="sql_schema_extra_left"">    <p style="font-weight:bold">See also:</p>
  <ul>
      <li><a href="#publication">publication</a></li>
      <li><a href="#variation">variation</a></li>
      </ul>
    </td>
  <td class="sql_schema_extra_separator"></td>
  <td class="sql_schema_extra_right" style="padding-left:25px"><p><span>List of species with populated data:</span>
  <a id="s_variation_citation" class="help-header sql_schema_show_hide" onclick="show_hide('variation_citation','species')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show species
  </a></p>
    <div id="sp_variation_citation" style="display:none;">      <ul>
        <li class="sql_schema_species_name">Bos taurus</li>
        <li class="sql_schema_species_name">Canis familiaris</li>
        <li class="sql_schema_species_name">Equus caballus</li>
        <li class="sql_schema_species_name">Gallus gallus</li>
        <li class="sql_schema_species_name">Homo sapiens</li>
        <li class="sql_schema_species_name">Mus musculus</li>
        <li class="sql_schema_species_name">Ovis aries</li>
        <li class="sql_schema_species_name">Rattus norvegicus</li>
        <li class="sql_schema_species_name">Sus scrofa</li>
      </ul>
    </div>
  </td></tr></table></div></div>

<div class="sql_schema_group_header" style="border-color:#01D4F7">
  <div id="10" class="sql_schema_group_bullet" style="background-color:#01D4F7"></div>
  <h2 id="10">Structural variation tables</h2>
</div>
<p class="sql_schema_group_header_desc">These tables define the structural variation data.</p><div class="sql_schema_table">
  <div id="structural_variation"class="sql_schema_table_header" style="border-top-color:#01D4F7">
    <div class="sql_schema_table_header_left"><span style="background-color:#01D4F7"></span>structural_variation</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_structural_variation" class="help-header sql_schema_show_hide" onclick="show_hide('structural_variation','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">This table stores information about structural variation.</p>

  <div id="div_structural_variation" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>structural_variation_id</b></td>
        <td class="bg1">int(10)</td>
        <td class="bg1">-</td>
        <td class="bg1">Primary key, internal identifier.</td>
        <td class="bg1">primary key</td>
      </tr>

      <tr>
        <td class="bg2"><b>variation_name</b></td>
        <td class="bg2">varchar(255)</td>
        <td class="bg2">NULL</td>
        <td class="bg2">The external identifier or name of the variation. e.g. "esv9549".</td>
        <td class="bg2">unique: <i>variation_name</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>alias</b></td>
        <td class="bg1">varchar(255)</td>
        <td class="bg1">NULL</td>
        <td class="bg1">Other structural variation name.</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>source_id</b></td>
        <td class="bg2">int(10)</td>
        <td class="bg2">-</td>
        <td class="bg2">Foreign key references to the <a href="#source" class="sql_schema_link">source</a> table.</td>
        <td class="bg2">key: <i>source_idx</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>study_id</b></td>
        <td class="bg1">int(10)</td>
        <td class="bg1">NULL</td>
        <td class="bg1">Foreign key references to the <a href="#study" class="sql_schema_link">study</a> table.	</td>
        <td class="bg1">key: <i>study_idx</i></td>
      </tr>

      <tr>
        <td class="bg2"><b>class_attrib_id</b></td>
        <td class="bg2">int(10)</td>
        <td class="bg2">0</td>
        <td class="bg2">Foreign key references to the <a href="#attrib" class="sql_schema_link">attrib</a> table. Defines the type of structural variant.<br />  The list of structural variation classes is available <a href="/info/genome/variation/data_description.html#classes">here</a>.</td>
        <td class="bg2">key: <i>attrib_idx</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>clinical_significance</b></td>
        <td class="bg1">SET:<ul class="sql_schema_table_column_type">  <li>uncertain significance</li>  <li>not provided</li>  <li>benign</li>  <li>likely benign</li>  <li>likely pathogenic</li>  <li>pathogenic</li>  <li>drug response</li>  <li>histocompatibility</li>  <li>other</li>  <li>confers sensitivity</li>  <li>risk factor</li>  <li>association</li>  <li>protective</li></ul></td>
        <td class="bg1">NULL</td>
        <td class="bg1">A set of clinical significance classes assigned to the structural variant.<br />  The list of clinical significances is available <a href="/info/genome/variation/data_description.html#clin_significance">here</a>.</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>validation_status</b></td>
        <td class="bg2">ENUM:<ul class="sql_schema_table_column_type">  <li>validated</li>  <li>not validated</li>  <li>high quality</li></ul></td>
        <td class="bg2">NULL</td>
        <td class="bg2">Validation status of the variant.</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>is_evidence</b></td>
        <td class="bg1">TINYINT(4)</td>
        <td class="bg1">0</td>
        <td class="bg1">Flag indicating if the structural variation is a supporting evidence (1) or not (0).</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>somatic</b></td>
        <td class="bg2">TINYINT(1)</td>
        <td class="bg2">0</td>
        <td class="bg2">Flags whether this structural variation is known to be somatic or not</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>copy_number</b></td>
        <td class="bg1">TINYINT(2)</td>
        <td class="bg1">NULL</td>
        <td class="bg1">Add the copy number for the CNV supporting structural variants when available.</td>
        <td class="bg1"></td>
      </tr>
    </table>
  </div>
<table class="sql_schema_extra"><tr>    <td class="sql_schema_extra_left"">    <p style="font-weight:bold">See also:</p>
  <ul>
      <li><a href="#source">source</a></li>
      <li><a href="#study">study</a></li>
      <li><a href="#attrib">attrib</a></li>
      </ul>
    </td>
  <td class="sql_schema_extra_separator"></td>
  <td class="sql_schema_extra_right" style="padding-left:25px"><p><span>List of species with populated data:</span>
  <a id="s_structural_variation" class="help-header sql_schema_show_hide" onclick="show_hide('structural_variation','species')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show species
  </a></p>
    <div id="sp_structural_variation" style="display:none;">      <ul>
        <li class="sql_schema_species_name">Bos taurus</li>
        <li class="sql_schema_species_name">Canis familiaris</li>
        <li class="sql_schema_species_name">Danio rerio</li>
        <li class="sql_schema_species_name">Equus caballus</li>
        <li class="sql_schema_species_name">Homo sapiens</li>
        <li class="sql_schema_species_name">Macaca mulatta</li>
        <li class="sql_schema_species_name">Mus musculus</li>
        <li class="sql_schema_species_name">Ovis aries</li>
        <li class="sql_schema_species_name">Sus scrofa</li>
      </ul>
    </div>
  </td></tr></table></div></div><div class="sql_schema_table">
  <div id="structural_variation_association"class="sql_schema_table_header" style="border-top-color:#01D4F7">
    <div class="sql_schema_table_header_left"><span style="background-color:#01D4F7"></span>structural_variation_association</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_structural_variation_association" class="help-header sql_schema_show_hide" onclick="show_hide('structural_variation_association','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">This table stores the associations between structural variations and their supporting evidences.</p>

  <div id="div_structural_variation_association" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>structural_variation_id</b></td>
        <td class="bg1">int(10)</td>
        <td class="bg1">-</td>
        <td class="bg1">Primary key. Foreign key references to the <a href="#structural_variation" class="sql_schema_link">structural_variation</a> table.</td>
        <td class="bg1">primary key<br />key: <i>structural_variation_idx</i></td>
      </tr>

      <tr>
        <td class="bg2"><b>supporting_structural_variation_id</b></td>
        <td class="bg2">int(10)</td>
        <td class="bg2">-</td>
        <td class="bg2">Primary key. Foreign key references to the <a href="#structural_variation" class="sql_schema_link">structural_variation</a> table.</td>
        <td class="bg2">primary key<br />key: <i>supporting_structural_variation_idx</i></td>
      </tr>
    </table>
  </div>
<table class="sql_schema_extra"><tr>    <td class="sql_schema_extra_left"">    <p style="font-weight:bold">See also:</p>
  <ul>
      <li><a href="#structural_variation">structural_variation</a></li>
      </ul>
    </td>
  <td class="sql_schema_extra_separator"></td>
  <td class="sql_schema_extra_right" style="padding-left:25px"><p><span>List of species with populated data:</span>
  <a id="s_structural_variation_association" class="help-header sql_schema_show_hide" onclick="show_hide('structural_variation_association','species')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show species
  </a></p>
    <div id="sp_structural_variation_association" style="display:none;">      <ul>
        <li class="sql_schema_species_name">Bos taurus</li>
        <li class="sql_schema_species_name">Canis familiaris</li>
        <li class="sql_schema_species_name">Danio rerio</li>
        <li class="sql_schema_species_name">Equus caballus</li>
        <li class="sql_schema_species_name">Homo sapiens</li>
        <li class="sql_schema_species_name">Macaca mulatta</li>
        <li class="sql_schema_species_name">Mus musculus</li>
        <li class="sql_schema_species_name">Ovis aries</li>
        <li class="sql_schema_species_name">Sus scrofa</li>
      </ul>
    </div>
  </td></tr></table></div></div><div class="sql_schema_table">
  <div id="structural_variation_feature"class="sql_schema_table_header" style="border-top-color:#01D4F7">
    <div class="sql_schema_table_header_left"><span style="background-color:#01D4F7"></span>structural_variation_feature</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_structural_variation_feature" class="help-header sql_schema_show_hide" onclick="show_hide('structural_variation_feature','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">This table stores information about structural variation features (i.e. mappings of structural variations to genomic locations).</p>

  <div id="div_structural_variation_feature" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>structural_variation_feature_id</b></td>
        <td class="bg1">int(10)</td>
        <td class="bg1">-</td>
        <td class="bg1">Primary key, internal identifier.</td>
        <td class="bg1">primary key</td>
      </tr>

      <tr>
        <td class="bg2"><b>seq_region_id</b></td>
        <td class="bg2">int(10)</td>
        <td class="bg2">-</td>
        <td class="bg2">Foreign key references <a href="#seq_region" class="sql_schema_link">seq_region</a> in core db. Refers to the seq_region which this variant is on, which may be a chromosome, a clone, etc...</td>
        <td class="bg2">key: <i>pos_idx</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>outer_start</b></td>
        <td class="bg1">int</td>
        <td class="bg1">NULL</td>
        <td class="bg1">The 5' outer bound position of the variation on the <a href="#seq_region" class="sql_schema_link">seq_region</a>.</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>seq_region_start</b></td>
        <td class="bg2">int</td>
        <td class="bg2">-</td>
        <td class="bg2">The start position of the variation on the <a href="#seq_region" class="sql_schema_link">seq_region</a>.</td>
        <td class="bg2">key: <i>pos_idx</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>inner_start</b></td>
        <td class="bg1">int</td>
        <td class="bg1">NULL</td>
        <td class="bg1">The 5' inner bound position of the variation on the <a href="#seq_region" class="sql_schema_link">seq_region</a>.</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>inner_end</b></td>
        <td class="bg2">int</td>
        <td class="bg2">NULL</td>
        <td class="bg2">The 3' inner bound position of the variation on the <a href="#seq_region" class="sql_schema_link">seq_region</a>.</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>seq_region_end</b></td>
        <td class="bg1">int</td>
        <td class="bg1">-</td>
        <td class="bg1">The end position of the variation on the <a href="#seq_region" class="sql_schema_link">seq_region</a>.</td>
        <td class="bg1">key: <i>pos_idx</i></td>
      </tr>

      <tr>
        <td class="bg2"><b>outer_end</b></td>
        <td class="bg2">int</td>
        <td class="bg2">NULL</td>
        <td class="bg2">The 3' outer bound position of the variation on the <a href="#seq_region" class="sql_schema_link">seq_region</a>.</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>seq_region_strand</b></td>
        <td class="bg1">tinyint</td>
        <td class="bg1">-</td>
        <td class="bg1">The orientation of the variation on the <a href="#seq_region" class="sql_schema_link">seq_region</a>.</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>structural_variation_id</b></td>
        <td class="bg2">int(10)</td>
        <td class="bg2">-</td>
        <td class="bg2">Foreign key references to the <a href="#structural_variation" class="sql_schema_link">structural_variation</a> table.</td>
        <td class="bg2">key: <i>structural_variation_idx</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>variation_name</b></td>
        <td class="bg1">varchar(255)</td>
        <td class="bg1">NULL</td>
        <td class="bg1">A denormalisation taken from the structural_variation table. This is the name or identifier that is used for displaying the feature (e.g. "esv9549").</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>source_id</b></td>
        <td class="bg2">int(10)</td>
        <td class="bg2">-</td>
        <td class="bg2">Foreign key references to the <a href="#source" class="sql_schema_link">source</a> table.</td>
        <td class="bg2">key: <i>source_idx</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>study_id</b></td>
        <td class="bg1">int(10)</td>
        <td class="bg1">NULL</td>
        <td class="bg1">Foreign key references to the <a href="#study" class="sql_schema_link">study</a> table</td>
        <td class="bg1">key: <i>study_idx</i></td>
      </tr>

      <tr>
        <td class="bg2"><b>class_attrib_id</b></td>
        <td class="bg2">int(10)</td>
        <td class="bg2">0</td>
        <td class="bg2">Foreign key references to the <a href="#attrib" class="sql_schema_link">attrib</a> table. Defines the type of structural variant.<br />  The list of structural variation classes is available <a href="/info/genome/variation/data_description.html#classes">here</a>.</td>
        <td class="bg2">key: <i>attrib_idx</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>allele_string</b></td>
        <td class="bg1">longtext</td>
        <td class="bg1">NULL</td>
        <td class="bg1">The variant allele, where known.</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>is_evidence</b></td>
        <td class="bg2">tinyint(1)</td>
        <td class="bg2">0</td>
        <td class="bg2">Flag indicating if the structural variation is a supporting evidence (1) or not (0).</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>variation_set_id</b></td>
        <td class="bg1">SET:<ul class="sql_schema_table_column_type">  <li>1</li>  <li>2</li>  <li>3</li>  <li>4</li>  <li>5</li>  <li>6</li>  <li>7</li>  <li>8</li>  <li>9</li>  <li>10</li>  <li>11</li>  <li>12</li>  <li>13</li>  <li>14</li>  <li>15</li>  <li>16</li>  <li>17</li>  <li>18</li>  <li>19</li>  <li>20</li>  <li>21</li>  <li>22</li>  <li>23</li>  <li>24</li>  <li>25</li>  <li>26</li>  <li>27</li>  <li>28</li>  <li>29</li>  <li>30</li>  <li>31</li>  <li>32</li>  <li>33</li>  <li>34</li>  <li>35</li>  <li>36</li>  <li>37</li>  <li>38</li>  <li>39</li>  <li>40</li>  <li>41</li>  <li>42</li>  <li>43</li>  <li>44</li>  <li>45</li>  <li>46</li>  <li>47</li>  <li>48</li>  <li>49</li>  <li>50</li>  <li>51</li>  <li>52</li>  <li>53</li>  <li>54</li>  <li>55</li>  <li>56</li>  <li>57</li>  <li>58</li>  <li>59</li>  <li>60</li>  <li>61</li>  <li>62</li>  <li>63</li>  <li>64 </li></ul></td>
        <td class="bg1">''</td>
        <td class="bg1">The structural variation feature can belong to a <a href="#variation_set" class="sql_schema_link">variation_set</a>.</td>
        <td class="bg1">key: <i>variation_set_idx</i></td>
      </tr>

      <tr>
        <td class="bg2"><b>somatic</b></td>
        <td class="bg2">TINYINT(1)</td>
        <td class="bg2">0</td>
        <td class="bg2">Flags whether this structural variation is known to be somatic or not</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>breakpoint_order</b></td>
        <td class="bg1">TINYINT(4)</td>
        <td class="bg1">NULL</td>
        <td class="bg1">Defines the order of the breakpoints when several events/mutation occurred for a structural variation (e.g. somatic mutations)</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>length</b></td>
        <td class="bg2">int(10)</td>
        <td class="bg2">NULL</td>
        <td class="bg2">Length of the structural variant. Used for the variants with a class "insertion", when the size of the insertion is known.</td>
        <td class="bg2"></td>
      </tr>
    </table>
  </div>
<table class="sql_schema_extra"><tr>    <td class="sql_schema_extra_left"">    <p style="font-weight:bold">See also:</p>
  <ul>
      <li><a href="#structural_variation">structural_variation</a></li>
      <li><a href="#source">source</a></li>
      <li><a href="#study">study</a></li>
      <li><a href="#seq_region">seq_region</a></li>
      <li><a href="#attrib">attrib</a></li>
      <li><a href="#variation_set">variation_set</a></li>
      </ul>
    </td>
  <td class="sql_schema_extra_separator"></td>
  <td class="sql_schema_extra_right" style="padding-left:25px"><p><span>List of species with populated data:</span>
  <a id="s_structural_variation_feature" class="help-header sql_schema_show_hide" onclick="show_hide('structural_variation_feature','species')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show species
  </a></p>
    <div id="sp_structural_variation_feature" style="display:none;">      <ul>
        <li class="sql_schema_species_name">Bos taurus</li>
        <li class="sql_schema_species_name">Canis familiaris</li>
        <li class="sql_schema_species_name">Danio rerio</li>
        <li class="sql_schema_species_name">Equus caballus</li>
        <li class="sql_schema_species_name">Homo sapiens</li>
        <li class="sql_schema_species_name">Macaca mulatta</li>
        <li class="sql_schema_species_name">Mus musculus</li>
        <li class="sql_schema_species_name">Ovis aries</li>
        <li class="sql_schema_species_name">Sus scrofa</li>
      </ul>
    </div>
  </td></tr></table></div></div><div class="sql_schema_table">
  <div id="structural_variation_sample"class="sql_schema_table_header" style="border-top-color:#01D4F7">
    <div class="sql_schema_table_header_left"><span style="background-color:#01D4F7"></span>structural_variation_sample</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_structural_variation_sample" class="help-header sql_schema_show_hide" onclick="show_hide('structural_variation_sample','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">This table stores sample and strain information for structural variants and their supporting evidences.</p>

  <div id="div_structural_variation_sample" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>structural_variation_sample_id</b></td>
        <td class="bg1">int(10)</td>
        <td class="bg1">-</td>
        <td class="bg1">Primary key, internal identifier.</td>
        <td class="bg1">primary key</td>
      </tr>

      <tr>
        <td class="bg2"><b>structural_variation_id</b></td>
        <td class="bg2">int(10)</td>
        <td class="bg2">-</td>
        <td class="bg2">Foreign key references to the <a href="#structural_variation" class="sql_schema_link">structural_variation</a> table.</td>
        <td class="bg2">key: <i>structural_variation_idx</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>sample_id</b></td>
        <td class="bg1">int(10)</td>
        <td class="bg1">NULL</td>
        <td class="bg1">Foreign key references to the <a href="#sample" class="sql_schema_link">sample</a> table. Defines the individual or sample name.</td>
        <td class="bg1">key: <i>sample_idx</i></td>
      </tr>
    </table>
  </div>
<table class="sql_schema_extra"><tr>    <td class="sql_schema_extra_left"">    <p style="font-weight:bold">See also:</p>
  <ul>
      <li><a href="#structural_variation">structural_variation</a></li>
      <li><a href="#sample">sample</a></li>
      <li><a href="#individudal">individudal</a></li>
      </ul>
    </td>
  <td class="sql_schema_extra_separator"></td>
  <td class="sql_schema_extra_right" style="padding-left:25px"><p><span>List of species with populated data:</span>
  <a id="s_structural_variation_sample" class="help-header sql_schema_show_hide" onclick="show_hide('structural_variation_sample','species')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show species
  </a></p>
    <div id="sp_structural_variation_sample" style="display:none;">      <ul>
        <li class="sql_schema_species_name">Bos taurus</li>
        <li class="sql_schema_species_name">Canis familiaris</li>
        <li class="sql_schema_species_name">Danio rerio</li>
        <li class="sql_schema_species_name">Equus caballus</li>
        <li class="sql_schema_species_name">Homo sapiens</li>
        <li class="sql_schema_species_name">Macaca mulatta</li>
        <li class="sql_schema_species_name">Mus musculus</li>
        <li class="sql_schema_species_name">Ovis aries</li>
        <li class="sql_schema_species_name">Sus scrofa</li>
      </ul>
    </div>
  </td></tr></table></div></div>

<div class="sql_schema_group_header" style="border-color:#FF4DC8">
  <div id="11" class="sql_schema_group_bullet" style="background-color:#FF4DC8"></div>
  <h2 id="11">Variation effect tables</h2>
</div>
<p class="sql_schema_group_header_desc">These tables define the variation effect prediction data in different Ensembl features.</p><div class="sql_schema_table">
  <div id="motif_feature_variation"class="sql_schema_table_header" style="border-top-color:#FF4DC8">
    <div class="sql_schema_table_header_left"><span style="background-color:#FF4DC8"></span>motif_feature_variation</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_motif_feature_variation" class="help-header sql_schema_show_hide" onclick="show_hide('motif_feature_variation','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">This table relates a single allele of a variation_feature to a motif feature (see Regulation documentation). It contains the consequence of the allele.</p>

  <div id="div_motif_feature_variation" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>motif_feature_variation_id</b></td>
        <td class="bg1">int(11)</td>
        <td class="bg1">-</td>
        <td class="bg1">Primary key, internal identifier.</td>
        <td class="bg1">primary key</td>
      </tr>

      <tr>
        <td class="bg2"><b>variation_feature_id</b></td>
        <td class="bg2">int(11)</td>
        <td class="bg2">-</td>
        <td class="bg2">Foreign key references to the <a href="#variation_feature" class="sql_schema_link">variation_feature</a> table.</td>
        <td class="bg2">key: <i>variation_feature_idx</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>feature_stable_id</b></td>
        <td class="bg1">varchar(128)</td>
        <td class="bg1">NULL</td>
        <td class="bg1">Foreign key to regulation databases. Unique stable id of related regulatory_feature.</td>
        <td class="bg1">key: <i>somatic_feature_idx</i></td>
      </tr>

      <tr>
        <td class="bg2"><b>motif_feature_id</b></td>
        <td class="bg2">int(11)</td>
        <td class="bg2">-</td>
        <td class="bg2">Foreign key to regulation databases. Internal id of related motif_feature.</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>allele_string</b></td>
        <td class="bg1">text</td>
        <td class="bg1">NULL</td>
        <td class="bg1">Shows the reference sequence and variant sequence of this allele.</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>somatic</b></td>
        <td class="bg2">tinyint(1)</td>
        <td class="bg2">0</td>
        <td class="bg2">Flags if the associated variation is known to be somatic.</td>
        <td class="bg2">key: <i>somatic_feature_idx</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>consequence_types</b></td>
        <td class="bg1">SET:<ul class="sql_schema_table_column_type">  <li>TF_binding_site_variant</li>  <li>TFBS_ablation</li>  <li>TFBS_fusion</li>  <li>TFBS_amplification</li>  <li>TFBS_translocation </li></ul></td>
        <td class="bg1">NULL</td>
        <td class="bg1">The consequence(s) of the variant allele on this motif_feature.<br /> The list of consequence descriptions is available <a href="/info/genome/variation/predicted_data.html#consequences">here</a>.</td>
        <td class="bg1">key: <i>consequence_type_idx</i></td>
      </tr>

      <tr>
        <td class="bg2"><b>motif_name</b></td>
        <td class="bg2">varchar(60)</td>
        <td class="bg2">NULL</td>
        <td class="bg2">The display label of the motif.</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>motif_start</b></td>
        <td class="bg1">int(11)</td>
        <td class="bg1">NULL</td>
        <td class="bg1">The start position of the variation in the motif.</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>motif_end</b></td>
        <td class="bg2">int(11)</td>
        <td class="bg2">NULL</td>
        <td class="bg2">The end position of the variation in the motif.</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>motif_score_delta</b></td>
        <td class="bg1">float</td>
        <td class="bg1">NULL</td>
        <td class="bg1">The deviation from the score (that is derived from alignment software (e.g. MOODS)) caused by the variation.</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>in_informative_position</b></td>
        <td class="bg2">tinyint(1)</td>
        <td class="bg2">0</td>
        <td class="bg2">Flags if the variation is in an informative position.</td>
        <td class="bg2"></td>
      </tr>
    </table>
  </div>
<table class="sql_schema_extra"><tr>    <td class="sql_schema_extra_left"">    <p style="font-weight:bold">See also:</p>
  <ul>
      <li><a href="#variation_feature">variation_feature</a></li>
      </ul>
    </td>
  <td class="sql_schema_extra_separator"></td>
  <td class="sql_schema_extra_right" style="padding-left:25px"><p><span>List of species with populated data:</span>
  <a id="s_motif_feature_variation" class="help-header sql_schema_show_hide" onclick="show_hide('motif_feature_variation','species')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show species
  </a></p>
    <div id="sp_motif_feature_variation" style="display:none;">      <ul>
        <li class="sql_schema_species_name">Homo sapiens</li>
        <li class="sql_schema_species_name">Mus musculus</li>
      </ul>
    </div>
  </td></tr></table></div></div><div class="sql_schema_table">
  <div id="regulatory_feature_variation"class="sql_schema_table_header" style="border-top-color:#FF4DC8">
    <div class="sql_schema_table_header_left"><span style="background-color:#FF4DC8"></span>regulatory_feature_variation</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_regulatory_feature_variation" class="help-header sql_schema_show_hide" onclick="show_hide('regulatory_feature_variation','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">This table relates a single allele of a variation_feature to a regulatory feature (see Regulation documentation). It contains the consequence of the allele.</p>

  <div id="div_regulatory_feature_variation" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>regulatory_feature_variation_id</b></td>
        <td class="bg1">int(11)</td>
        <td class="bg1">-</td>
        <td class="bg1">Primary key, internal identifier.</td>
        <td class="bg1">primary key</td>
      </tr>

      <tr>
        <td class="bg2"><b>variation_feature_id</b></td>
        <td class="bg2">int(11)</td>
        <td class="bg2">-</td>
        <td class="bg2">Foreign key references to the <a href="#variation_feature" class="sql_schema_link">variation_feature</a> table.</td>
        <td class="bg2">key: <i>variation_feature_idx</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>feature_stable_id</b></td>
        <td class="bg1">varchar(128)</td>
        <td class="bg1">NULL</td>
        <td class="bg1">Foreign key to regulation databases. Unique stable id of related regulatory_feature.</td>
        <td class="bg1">key: <i>somatic_feature_idx</i></td>
      </tr>

      <tr>
        <td class="bg2"><b>feature_type</b></td>
        <td class="bg2">text</td>
        <td class="bg2">NULL</td>
        <td class="bg2">The name of the feature type.</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>allele_string</b></td>
        <td class="bg1">text</td>
        <td class="bg1">NULL</td>
        <td class="bg1">Shows the reference sequence and variant sequence of this allele.</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>somatic</b></td>
        <td class="bg2">tinyint(1)</td>
        <td class="bg2">0</td>
        <td class="bg2">Flags if the associated variation is known to be somatic.</td>
        <td class="bg2">key: <i>somatic_feature_idx</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>consequence_types</b></td>
        <td class="bg1">SET:<ul class="sql_schema_table_column_type">  <li>regulatory_region_variant</li>  <li>regulatory_region_ablation</li>  <li>regulatory_region_fusion</li>  <li>regulatory_region_amplification</li>  <li>regulatory_region_translocation </li></ul></td>
        <td class="bg1">NULL</td>
        <td class="bg1">The consequence(s) of the variant allele on this regulatory feature.<br /> The list of consequence descriptions is available <a href="/info/genome/variation/predicted_data.html#consequences">here</a>.</td>
        <td class="bg1">key: <i>consequence_type_idx</i></td>
      </tr>
    </table>
  </div>
<table class="sql_schema_extra"><tr>    <td class="sql_schema_extra_left"">    <p style="font-weight:bold">See also:</p>
  <ul>
      <li><a href="#variation_feature">variation_feature</a></li>
      </ul>
    </td>
  <td class="sql_schema_extra_separator"></td>
  <td class="sql_schema_extra_right" style="padding-left:25px"><p><span>List of species with populated data:</span>
  <a id="s_regulatory_feature_variation" class="help-header sql_schema_show_hide" onclick="show_hide('regulatory_feature_variation','species')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show species
  </a></p>
    <div id="sp_regulatory_feature_variation" style="display:none;">      <ul>
        <li class="sql_schema_species_name">Homo sapiens</li>
        <li class="sql_schema_species_name">Mus musculus</li>
      </ul>
    </div>
  </td></tr></table></div></div><div class="sql_schema_table">
  <div id="transcript_variation"class="sql_schema_table_header" style="border-top-color:#FF4DC8">
    <div class="sql_schema_table_header_left"><span style="background-color:#FF4DC8"></span>transcript_variation</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_transcript_variation" class="help-header sql_schema_show_hide" onclick="show_hide('transcript_variation','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">This table relates a single allele of a variation_feature to a transcript (see Core documentation). It contains the consequence of the allele e.g. intron_variant, non_synonymous_codon, stop_lost etc, along with the change in amino acid in the resulting protein if applicable.</p>

  <div id="div_transcript_variation" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>transcript_variation_id</b></td>
        <td class="bg1">int(11)</td>
        <td class="bg1">-</td>
        <td class="bg1">Primary key, internal identifier.</td>
        <td class="bg1">primary key</td>
      </tr>

      <tr>
        <td class="bg2"><b>feature_stable_id</b></td>
        <td class="bg2">varchar(128)</td>
        <td class="bg2">NULL</td>
        <td class="bg2">Foreign key to core databases. Unique stable id of related transcript.</td>
        <td class="bg2">key: <i>somatic_feature_idx</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>variation_feature_id</b></td>
        <td class="bg1">int(11)</td>
        <td class="bg1">-</td>
        <td class="bg1">Foreign key references to the <a href="#variation_feature" class="sql_schema_link">variation_feature</a> table.</td>
        <td class="bg1">key: <i>variation_feature_idx</i></td>
      </tr>

      <tr>
        <td class="bg2"><b>allele_string</b></td>
        <td class="bg2">text</td>
        <td class="bg2">NULL</td>
        <td class="bg2">Shows the reference sequence and variant sequence of this allele</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>somatic</b></td>
        <td class="bg1">tinyint(1)</td>
        <td class="bg1">0</td>
        <td class="bg1">Flags if the associated variation is known to be somatic</td>
        <td class="bg1">key: <i>somatic_feature_idx</i></td>
      </tr>

      <tr>
        <td class="bg2"><b>consequence_types</b></td>
        <td class="bg2">SET:<ul class="sql_schema_table_column_type">  <li>splice_acceptor_variant</li>  <li>splice_donor_variant</li>  <li>stop_lost</li>  <li>coding_sequence_variant</li>  <li>missense_variant</li>  <li>stop_gained</li>  <li>synonymous_variant</li>  <li>frameshift_variant</li>  <li>non_coding_transcript_variant</li>  <li>non_coding_transcript_exon_variant</li>  <li>mature_miRNA_variant</li>  <li>NMD_transcript_variant</li>  <li>5_prime_UTR_variant</li>  <li>3_prime_UTR_variant</li>  <li>incomplete_terminal_codon_variant</li>  <li>intron_variant</li>  <li>splice_region_variant</li>  <li>downstream_gene_variant</li>  <li>upstream_gene_variant</li>  <li>start_lost</li>  <li>stop_retained_variant</li>  <li>inframe_insertion</li>  <li>inframe_deletion</li>  <li>transcript_ablation</li>  <li>transcript_fusion</li>  <li>transcript_amplification</li>  <li>transcript_translocation</li>  <li>TFBS_ablation</li>  <li>TFBS_fusion</li>  <li>TFBS_amplification</li>  <li>TFBS_translocation</li>  <li>regulatory_region_ablation</li>  <li>regulatory_region_fusion</li>  <li>regulatory_region_amplification</li>  <li>regulatory_region_translocation</li>  <li>feature_elongation</li>  <li>feature_truncation</li>  <li>protein_altering_variant </li></ul></td>
        <td class="bg2">NULL</td>
        <td class="bg2">The consequence(s) of the variant allele on this transcript.<br /> The list of consequence descriptions is available <a href="/info/genome/variation/predicted_data.html#consequences">here</a>. </td>
        <td class="bg2">key: <i>consequence_type_idx</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>cds_start</b></td>
        <td class="bg1">int(11)</td>
        <td class="bg1">NULL</td>
        <td class="bg1">The start position of variation in cds coordinates.</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>cds_end</b></td>
        <td class="bg2">int(11)</td>
        <td class="bg2">NULL</td>
        <td class="bg2">The end position of variation in cds coordinates.</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>cdna_start</b></td>
        <td class="bg1">int(11)</td>
        <td class="bg1">NULL</td>
        <td class="bg1">The start position of variation in cdna coordinates.</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>cdna_end</b></td>
        <td class="bg2">int(11)</td>
        <td class="bg2">NULL</td>
        <td class="bg2">The end position of variation in cdna coordinates.</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>translation_start</b></td>
        <td class="bg1">int(11)</td>
        <td class="bg1">NULL</td>
        <td class="bg1">The start position of variation on peptide.</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>translation_end</b></td>
        <td class="bg2">int(11)</td>
        <td class="bg2">NULL</td>
        <td class="bg2">The end position of variation on peptide.</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>distance_to_transcript</b></td>
        <td class="bg1">int(11)</td>
        <td class="bg1">NULL</td>
        <td class="bg1">Only for upstream or downstream variants, it gives the distance from the start or the end of the transcript</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>codon_allele_string</b></td>
        <td class="bg2">text</td>
        <td class="bg2">NULL</td>
        <td class="bg2">The reference and variant codons</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>pep_allele_string</b></td>
        <td class="bg1">text</td>
        <td class="bg1">NULL</td>
        <td class="bg1">The reference and variant peptides</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>hgvs_genomic</b></td>
        <td class="bg2">text</td>
        <td class="bg2">NULL</td>
        <td class="bg2">HGVS representation of this allele with respect to the genomic sequence</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>hgvs_transcript</b></td>
        <td class="bg1">text</td>
        <td class="bg1">NULL</td>
        <td class="bg1">HGVS representation of this allele with respect to the [coding or non-coding] transcript</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>hgvs_protein</b></td>
        <td class="bg2">text</td>
        <td class="bg2">NULL</td>
        <td class="bg2">HGVS representation of this allele with respect to the protein</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>polyphen_prediction</b></td>
        <td class="bg1">ENUM:<ul class="sql_schema_table_column_type">  <li>unknown</li>  <li> benign</li>  <li> possibly damaging</li>  <li> probably damaging</li></ul></td>
        <td class="bg1">NULL</td>
        <td class="bg1">The PolyPhen prediction for the effect of this allele on the protein</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>polyphen_score</b></td>
        <td class="bg2">float</td>
        <td class="bg2">NULL</td>
        <td class="bg2">The PolyPhen score corresponding to the prediction </td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>sift_prediction</b></td>
        <td class="bg1">ENUM:<ul class="sql_schema_table_column_type">  <li>tolerated</li>  <li> deleterious</li>  <li> tolerated - low confidence</li>  <li> deleterious - low confidence</li></ul></td>
        <td class="bg1">NULL</td>
        <td class="bg1">The SIFT prediction for the effect of this allele on the protein </td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>sift_score</b></td>
        <td class="bg2">float</td>
        <td class="bg2">NULL</td>
        <td class="bg2">The SIFT score corresponding to this prediction</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>display</b></td>
        <td class="bg1">int(1)</td>
        <td class="bg1">1</td>
        <td class="bg1">Flags whether this transcript_variation should be displayed in browser tracks and returned by default by the API</td>
        <td class="bg1"></td>
      </tr>
    </table>
  </div>
<table class="sql_schema_extra"><tr>    <td class="sql_schema_extra_left"">    <p style="font-weight:bold">See also:</p>
  <ul>
      <li><a href="#variation_feature">variation_feature</a></li>
      </ul>
    </td>
  <td class="sql_schema_extra_separator"></td>
  <td class="sql_schema_extra_right" style="padding-left:25px"><p><span>List of species with populated data:</span>
  <a id="s_transcript_variation" class="help-header sql_schema_show_hide" onclick="show_hide('transcript_variation','species')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show species
  </a></p>
    <div id="sp_transcript_variation" style="display:none;">      <ul>
        <li class="sql_schema_species_name">Bos taurus</li>
        <li class="sql_schema_species_name">Canis familiaris</li>
        <li class="sql_schema_species_name">Danio rerio</li>
        <li class="sql_schema_species_name">Drosophila melanogaster</li>
        <li class="sql_schema_species_name">Equus caballus</li>
        <li class="sql_schema_species_name">Felis catus</li>
        <li class="sql_schema_species_name">Gallus gallus</li>
        <li class="sql_schema_species_name">Homo sapiens</li>
        <li class="sql_schema_species_name">Macaca mulatta</li>
        <li class="sql_schema_species_name">Meleagris gallopavo</li>
        <li class="sql_schema_species_name">Monodelphis domestica</li>
        <li class="sql_schema_species_name">Mus musculus</li>
        <li class="sql_schema_species_name">Nomascus leucogenys</li>
        <li class="sql_schema_species_name">Ornithorhynchus anatinus</li>
        <li class="sql_schema_species_name">Ovis aries</li>
        <li class="sql_schema_species_name">Pan troglodytes</li>
        <li class="sql_schema_species_name">Pongo abelii</li>
        <li class="sql_schema_species_name">Rattus norvegicus</li>
        <li class="sql_schema_species_name">Saccharomyces cerevisiae</li>
        <li class="sql_schema_species_name">Sus scrofa</li>
        <li class="sql_schema_species_name">Taeniopygia guttata</li>
        <li class="sql_schema_species_name">Tetraodon nigroviridis</li>
      </ul>
    </div>
  </td></tr></table></div></div><div class="sql_schema_table">
  <div id="variation_genename"class="sql_schema_table_header" style="border-top-color:#FF4DC8">
    <div class="sql_schema_table_header_left"><span style="background-color:#FF4DC8"></span>variation_genename</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_variation_genename" class="help-header sql_schema_show_hide" onclick="show_hide('variation_genename','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">This table is used in web index creation. It links a variation_id to the names of the genes the variation is within</p>

  <div id="div_variation_genename" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>variation_id</b></td>
        <td class="bg1">int(10)</td>
        <td class="bg1">-</td>
        <td class="bg1">Primary key, foreign key references <a href="#variation" class="sql_schema_link">variation</a></td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>gene_name</b></td>
        <td class="bg2">varchar(255)</td>
        <td class="bg2">-</td>
        <td class="bg2">Primary key, display name of gene</td>
        <td class="bg2"></td>
      </tr>
    </table>
  </div>
<table class="sql_schema_extra"><tr>    <td class="sql_schema_extra_left"">    <p style="font-weight:bold">See also:</p>
  <ul>
      <li><a href="#variation">variation</a></li>
      </ul>
    </td>
  <td class="sql_schema_extra_separator"></td>
  <td class="sql_schema_extra_right" style="padding-left:25px"><p><span>List of species with populated data:</span>
  <a id="s_variation_genename" class="help-header sql_schema_show_hide" onclick="show_hide('variation_genename','species')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show species
  </a></p>
    <div id="sp_variation_genename" style="display:none;">      <ul>
        <li class="sql_schema_species_name">Bos taurus</li>
        <li class="sql_schema_species_name">Canis familiaris</li>
        <li class="sql_schema_species_name">Danio rerio</li>
        <li class="sql_schema_species_name">Drosophila melanogaster</li>
        <li class="sql_schema_species_name">Equus caballus</li>
        <li class="sql_schema_species_name">Felis catus</li>
        <li class="sql_schema_species_name">Gallus gallus</li>
        <li class="sql_schema_species_name">Homo sapiens</li>
        <li class="sql_schema_species_name">Macaca mulatta</li>
        <li class="sql_schema_species_name">Meleagris gallopavo</li>
        <li class="sql_schema_species_name">Monodelphis domestica</li>
        <li class="sql_schema_species_name">Mus musculus</li>
        <li class="sql_schema_species_name">Nomascus leucogenys</li>
        <li class="sql_schema_species_name">Ornithorhynchus anatinus</li>
        <li class="sql_schema_species_name">Ovis aries</li>
        <li class="sql_schema_species_name">Pan troglodytes</li>
        <li class="sql_schema_species_name">Pongo abelii</li>
        <li class="sql_schema_species_name">Rattus norvegicus</li>
        <li class="sql_schema_species_name">Sus scrofa</li>
        <li class="sql_schema_species_name">Taeniopygia guttata</li>
        <li class="sql_schema_species_name">Tetraodon nigroviridis</li>
      </ul>
    </div>
  </td></tr></table></div></div><div class="sql_schema_table">
  <div id="variation_hgvs"class="sql_schema_table_header" style="border-top-color:#FF4DC8">
    <div class="sql_schema_table_header_left"><span style="background-color:#FF4DC8"></span>variation_hgvs</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_variation_hgvs" class="help-header sql_schema_show_hide" onclick="show_hide('variation_hgvs','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">This table is used in web index creation. It links a variation_id to all possible transcript and protein level change descriptions in HGVS annotation.</p>

  <div id="div_variation_hgvs" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>variation_id</b></td>
        <td class="bg1">int(10)</td>
        <td class="bg1">-</td>
        <td class="bg1">Primary key, foreign key references <a href="#variation" class="sql_schema_link">variation</a></td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>hgvs_name</b></td>
        <td class="bg2">varchar(255)</td>
        <td class="bg2">-</td>
        <td class="bg2">Primary key, HGVS change description</td>
        <td class="bg2"></td>
      </tr>
    </table>
  </div>
<table class="sql_schema_extra"><tr>    <td class="sql_schema_extra_left"">    <p style="font-weight:bold">See also:</p>
  <ul>
      <li><a href="#variation">variation</a></li>
      </ul>
    </td>
  <td class="sql_schema_extra_separator"></td>
  <td class="sql_schema_extra_right" style="padding-left:25px"><p><span>List of species with populated data:</span>
  <a id="s_variation_hgvs" class="help-header sql_schema_show_hide" onclick="show_hide('variation_hgvs','species')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show species
  </a></p>
    <div id="sp_variation_hgvs" style="display:none;">      <ul>
        <li class="sql_schema_species_name">Bos taurus</li>
        <li class="sql_schema_species_name">Canis familiaris</li>
        <li class="sql_schema_species_name">Danio rerio</li>
        <li class="sql_schema_species_name">Drosophila melanogaster</li>
        <li class="sql_schema_species_name">Equus caballus</li>
        <li class="sql_schema_species_name">Felis catus</li>
        <li class="sql_schema_species_name">Gallus gallus</li>
        <li class="sql_schema_species_name">Homo sapiens</li>
        <li class="sql_schema_species_name">Macaca mulatta</li>
        <li class="sql_schema_species_name">Meleagris gallopavo</li>
        <li class="sql_schema_species_name">Monodelphis domestica</li>
        <li class="sql_schema_species_name">Mus musculus</li>
        <li class="sql_schema_species_name">Nomascus leucogenys</li>
        <li class="sql_schema_species_name">Ornithorhynchus anatinus</li>
        <li class="sql_schema_species_name">Ovis aries</li>
        <li class="sql_schema_species_name">Pan troglodytes</li>
        <li class="sql_schema_species_name">Pongo abelii</li>
        <li class="sql_schema_species_name">Rattus norvegicus</li>
        <li class="sql_schema_species_name">Saccharomyces cerevisiae</li>
        <li class="sql_schema_species_name">Sus scrofa</li>
        <li class="sql_schema_species_name">Taeniopygia guttata</li>
        <li class="sql_schema_species_name">Tetraodon nigroviridis</li>
      </ul>
    </div>
  </td></tr></table></div></div>

<div class="sql_schema_group_header" style="border-color:#FFD700">
  <div id="12" class="sql_schema_group_bullet" style="background-color:#FFD700"></div>
  <h2 id="12">Variation set tables</h2>
</div>
<p class="sql_schema_group_header_desc">These tables define the variation and structural variation set data. The complete list of variation sets with their descriptions is available <a href="/info/genome/variation/data_description.html#variation_sets">here</a>.</p><div class="sql_schema_table">
  <div id="variation_set"class="sql_schema_table_header" style="border-top-color:#FFD700">
    <div class="sql_schema_table_header_left"><span style="background-color:#FFD700"></span>variation_set</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_variation_set" class="help-header sql_schema_show_hide" onclick="show_hide('variation_set','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">This table contains the name of sets and subsets of variations stored in the database. It usually represents the name of the project or subproject where a group of variations has been identified.</p>

  <div id="div_variation_set" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>variation_set_id</b></td>
        <td class="bg1">int(10)</td>
        <td class="bg1">-</td>
        <td class="bg1">Primary key, internal identifier.</td>
        <td class="bg1">primary key</td>
      </tr>

      <tr>
        <td class="bg2"><b>name</b></td>
        <td class="bg2">VARCHAR(255)</td>
        <td class="bg2">NULL</td>
        <td class="bg2">Name of the set e.g. "Phenotype-associated variations".</td>
        <td class="bg2">key: <i>name_idx</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>description</b></td>
        <td class="bg1">TEXT</td>
        <td class="bg1">NULL</td>
        <td class="bg1">Description of the set.</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>short_name_attrib_id</b></td>
        <td class="bg2">INT(10)</td>
        <td class="bg2">NULL</td>
        <td class="bg2">Foreign key references to the <a href="#attrib" class="sql_schema_link">attrib</a> table. Short name used for web purpose.</td>
        <td class="bg2"></td>
      </tr>
    </table>
  </div>
<div class="sql_schema_table_examples"><p class="sql_schema_table_example_header">Example:</p><div class="sql_schema_table_example_content"><p>See below the command to display the list of variation set entries, e.g. for human: </p>
      <div>
        <div class="sql_schema_table_example_query">
          <pre><span class="sql_schema_sql_highlight">SELECT</span> * <span class="sql_schema_sql_highlight">FROM</span> variation_set;</pre>
        </div>
        <div class="sql_schema_table_example_button">
  <a id="e_variation_set" class="help-header sql_schema_show_hide" onclick="show_hide('variation_set','query results')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show query results
  </a></div>
        <div style="clear:both"></div>
      </div>
      
  <div id="ex_variation_set" style="display:none;">
    <table class="ss sql_schema_table_example_result">
      <tr><th>variation_set_id</th><th>name</th><th>description</th><th>short_name_attrib_id</th></tr>      <tr><td>1</td><td>All failed variations</td><td>Variations that have failed the Ensembl QC checks</td><td>179</td></tr>      <tr class="bg2"><td>2</td><td>Genotyping chip variants</td><td>Variants which have assays on commercial chips held in ensembl</td><td>360</td></tr>      <tr><td>3</td><td>Illumina_CytoSNP12v1</td><td>Variants from the Illumina Cyto SNP-12 v1 whole genome SNP genotyping chip designed for cytogenetic analysis</td><td>339</td></tr>      <tr class="bg2"><td>4</td><td>Affy GenomeWideSNP_6.0</td><td>Variants from the Affymetrix Genome-Wide Human SNP Array 6.0</td><td>333</td></tr>      <tr><td>5</td><td>Illumina_Human660W-quad</td><td>Variants from the Illumina Human660W-quad whole genome genotyping array designed for association studies</td><td>334</td></tr>      <tr class="bg2"><td>6</td><td>Illumina_Human1M-duo</td><td>Variants from the Illumina Human1M-duo v3 whole genome genotyping array designed for association studies</td><td>335</td></tr>      <tr><td>7</td><td>Affy GeneChip 500K</td><td>Variants from the Affymetrix GeneChip Human Mapping 500K Array Set</td><td>332</td></tr>      <tr class="bg2"><td>8</td><td>Illumina_Cardio-Metabo_Chip</td><td>Variants from the Illumina Cardio-Metabo_Chip genotyping array designed to target variants of interest for metabolic and cardiovascular disease traits</td><td>337</td></tr>      <tr><td>9</td><td>Illumina_HumanOmni1-Quad</td><td>Variants from the Illumina HumanOmni1-Quad whole genome genotyping array designed for association studies</td><td>338</td></tr>      <tr class="bg2"><td>10</td><td>Illumina_HumanHap650Y</td><td>Variants from the Illumina HumanHap650Y v3.0 whole genome genotyping array designed for association studies</td><td>340</td></tr>      <tr><td>11</td><td>Illumina_HumanOmni2.5</td><td>Variants from the Illumina HumanOmni2.5 4v1 whole genome genotyping array designed for association studies</td><td>341</td></tr>      <tr class="bg2"><td>12</td><td>Illumina_Human610_Quad</td><td>Variants from the Illumina Human610_Quad v1_B whole genome genotyping array designed for association studies</td><td>342</td></tr>      <tr><td>13</td><td>Illumina_HumanHap550</td><td>Variants from the Illumina Human550 v3.0 whole genome genotyping array designed for association studies</td><td>343</td></tr>      <tr class="bg2"><td>14</td><td>Illumina_HumanOmni5</td><td>Variants from the Illumina HumanOmni5v1 whole genome genotyping array designed for association studies</td><td>354</td></tr>      <tr><td>15</td><td>Illumina_ExomeChip</td><td>Variants from the Illumina ExomeChip genotyping array designed to target variants within exons</td><td>358</td></tr>      <tr class="bg2"><td>16</td><td>Illumina_ImmunoChip</td><td>Variants from the Illumina ImmunoChip genotyping array designed to target variants of interest for autoimmune and inflammatory diseases</td><td>359</td></tr>      <tr><td>17</td><td>HumanOmniExpress</td><td>Variants from the Illumina HumanOmniExpress 12v1-1_a whole genome genotyping array</td><td>373</td></tr>      <tr class="bg2"><td>18</td><td>All HapMap</td><td>Variants which have been assayed by The International HapMap Project [http://hapmap.ncbi.nlm.nih.gov/]</td><td>176</td></tr>      <tr><td>19</td><td>HapMap - CEU</td><td>Variants which have been assayed by The International HapMap Project from CEU individuals</td><td>257</td></tr>      <tr class="bg2"><td>20</td><td>HapMap - HCB</td><td>Variants which have been assayed by The International HapMap Project from HCB individuals</td><td>258</td></tr>      <tr><td>21</td><td>HapMap - JPT</td><td>Variants which have been assayed by The International HapMap Project from JPT individuals</td><td>259</td></tr>      <tr class="bg2"><td>22</td><td>HapMap - YRI</td><td>Variants which have been assayed by The International HapMap Project from YRI individuals</td><td>260</td></tr>      <tr><td>23</td><td>All phenotype/disease-associated variants</td><td>Variants that have been associated with a phenotype or a disease</td><td>195</td></tr>      <tr class="bg2"><td>24</td><td>OMIM phenotype variants</td><td>Variations linked to entries in the Online Mendelian Inheritance in Man (OMIM) database</td><td>194</td></tr>      <tr><td>25</td><td>Uniprot phenotype variants</td><td>Variations with phenotype annotations provided by Uniprot</td><td>196</td></tr>      <tr class="bg2"><td>26</td><td>NHGRI-EBI catalog phenotype variants</td><td>Variants associated with phenotype data from the NHGRI-EBI GWAS catalog [http://www.ebi.ac.uk/gwas/]</td><td>193</td></tr>      <tr><td>27</td><td>PhenCode</td><td>Variants from the PhenCode Project</td><td>355</td></tr>      <tr class="bg2"><td>28</td><td>COSMIC phenotype variants</td><td>Phenotype annotations of somatic mutations found in human cancers from the COSMIC project</td><td>197</td></tr>      <tr><td>29</td><td>HGMD-PUBLIC variants</td><td>Variants annotated by HGMD</td><td>191</td></tr>      <tr class="bg2"><td>30</td><td>All ClinVar</td><td>Variants with ClinVar annotation</td><td>374</td></tr>      <tr><td>31</td><td>Clinically associated variants</td><td>Variants described by ClinVar as being probable-pathogenic, pathogenic, drug-response or histocompatibility</td><td>345</td></tr>      <tr class="bg2"><td>32</td><td>LSDB</td><td>Variants dbSNP annotates as being from LSDBs</td><td>395</td></tr>      <tr><td>33</td><td>dbPEX</td><td>Variants from the PEX Gene Database</td><td>396</td></tr>      <tr class="bg2"><td>34</td><td>KAT6BDB</td><td>Variants from the K(lysine) acetyltransferase 6B database, BCM</td><td>399</td></tr>      <tr><td>35</td><td>HbVar</td><td>Variants for the Human Hemoglobin Variants and Thalassemias database</td><td>397</td></tr>      <tr class="bg2"><td>36</td><td>LMDD</td><td>Variants from the Leiden Muscular Dystrophy Database</td><td>400</td></tr>      <tr><td>37</td><td>OIVD</td><td>Variants from the Osteogenesis Imperfecta Variant Database</td><td>401</td></tr>      <tr class="bg2"><td>38</td><td>PAHdb</td><td>Variants from the Phenylalanine hydroxylase database</td><td>402</td></tr>      <tr><td>39</td><td>Infevers</td><td>Variants from the registry of Hereditary Auto-inflammatory Disorders Mutations</td><td>398</td></tr>      <tr class="bg2"><td>40</td><td>HumanCoreExome-12</td><td>Variants from the Illumina HumanCoreExome-12 v1 genotyping chip.</td><td>390</td></tr>      <tr><td>42</td><td>1000 Genomes 3 - All</td><td>Variants genotyped by the 1000 Genomes project (phase 3)</td><td>404</td></tr>      <tr class="bg2"><td>43</td><td>1000 Genomes 3 - EUR - common</td><td>Variants genotyped in European individuals by the 1000 Genomes project (phase 1) with frequency of at least 1%</td><td>415</td></tr>      <tr><td>44</td><td>1000 Genomes 3 - SAS - common</td><td>Variants genotyped in East Asian individuals by the 1000 Genomes project (phase 1) with frequency of at least 1%</td><td>414</td></tr>      <tr class="bg2"><td>45</td><td>1000 Genomes 3 - EAS - common</td><td>Variants genotyped in East Asian individuals by the 1000 Genomes project (phase 1) with frequency of at least 1%</td><td>413</td></tr>      <tr><td>47</td><td>1000 Genomes 3 - AMR - common</td><td>Variants genotyped in admixed American individuals by the 1000 Genomes project (phase 1) with frequency of at least 1%</td><td>412</td></tr>      <tr class="bg2"><td>48</td><td>1000 Genomes 3 - AFR - common</td><td>Variants genotyped in African individuals by the 1000 Genomes project (phase 3) with frequency of at least 1%</td><td>411</td></tr>      <tr><td>49</td><td>1000 Genomes 3 - All - common</td><td>Variants genotyped by the 1000 Genomes project (phase 1) with frequency of at least 1%)</td><td>410</td></tr>      <tr class="bg2"><td>50</td><td>1000 Genomes 3 - EUR</td><td>Variants genotyped in European individuals by the 1000 Genomes project (phase 3)</td><td>409</td></tr>      <tr><td>51</td><td>1000 Genomes 3 - SAS</td><td>Variants genotyped in South Asian individuals by the 1000 Genomes project (phase 3)</td><td>408</td></tr>      <tr class="bg2"><td>52</td><td>1000 Genomes 3 - EAS</td><td>Variants genotyped in East Asian individuals by the 1000 Genomes project (phase 3)</td><td>407</td></tr>    </table>
  </div></div></div><table class="sql_schema_extra"><tr>    <td class="sql_schema_extra_left"">    <p style="font-weight:bold">See also:</p>
  <ul>
      <li><a href="#variation_set_variation">variation_set_variation</a></li>
      <li><a href="#variation_set_structure">variation_set_structure</a></li>
      <li><a href="#attrib">attrib</a></li>
      </ul>
    </td>
  <td class="sql_schema_extra_separator"></td>
  <td class="sql_schema_extra_right" style="padding-left:25px"><p><span>List of species with populated data:</span>
  <a id="s_variation_set" class="help-header sql_schema_show_hide" onclick="show_hide('variation_set','species')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show species
  </a></p>
    <div id="sp_variation_set" style="display:none;">      <ul>
        <li class="sql_schema_species_name">Bos taurus</li>
        <li class="sql_schema_species_name">Canis familiaris</li>
        <li class="sql_schema_species_name">Danio rerio</li>
        <li class="sql_schema_species_name">Equus caballus</li>
        <li class="sql_schema_species_name">Felis catus</li>
        <li class="sql_schema_species_name">Gallus gallus</li>
        <li class="sql_schema_species_name">Homo sapiens</li>
        <li class="sql_schema_species_name">Macaca mulatta</li>
        <li class="sql_schema_species_name">Meleagris gallopavo</li>
        <li class="sql_schema_species_name">Monodelphis domestica</li>
        <li class="sql_schema_species_name">Mus musculus</li>
        <li class="sql_schema_species_name">Nomascus leucogenys</li>
        <li class="sql_schema_species_name">Ornithorhynchus anatinus</li>
        <li class="sql_schema_species_name">Ovis aries</li>
        <li class="sql_schema_species_name">Pan troglodytes</li>
        <li class="sql_schema_species_name">Pongo abelii</li>
        <li class="sql_schema_species_name">Rattus norvegicus</li>
        <li class="sql_schema_species_name">Sus scrofa</li>
        <li class="sql_schema_species_name">Taeniopygia guttata</li>
        <li class="sql_schema_species_name">Tetraodon nigroviridis</li>
      </ul>
    </div>
  </td></tr></table></div></div><div class="sql_schema_table">
  <div id="variation_set_structural_variation"class="sql_schema_table_header" style="border-top-color:#FFD700">
    <div class="sql_schema_table_header_left"><span style="background-color:#FFD700"></span>variation_set_structural_variation</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_variation_set_structural_variation" class="help-header sql_schema_show_hide" onclick="show_hide('variation_set_structural_variation','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">A table for mapping structural variations to variation_sets.</p>

  <div id="div_variation_set_structural_variation" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>structural_variation_id</b></td>
        <td class="bg1">int(10)</td>
        <td class="bg1">-</td>
        <td class="bg1">Primary key. Foreign key references to the <a href="#structural_variation" class="sql_schema_link">structural_variation</a> table.</td>
        <td class="bg1">primary key</td>
      </tr>

      <tr>
        <td class="bg2"><b>variation_set_id</b></td>
        <td class="bg2">int(10)</td>
        <td class="bg2">-</td>
        <td class="bg2">Primary key. Foreign key references to the <a href="#variation_set" class="sql_schema_link">variation_set</a> table.</td>
        <td class="bg2">primary key</td>
      </tr>
    </table>
  </div>
<table class="sql_schema_extra"><tr>    <td class="sql_schema_extra_left"">    <p style="font-weight:bold">See also:</p>
  <ul>
      <li><a href="#structural_variation">structural_variation</a></li>
      <li><a href="#variation_set">variation_set</a></li>
      </ul>
    </td>
  <td class="sql_schema_extra_separator"></td>
  <td class="sql_schema_extra_right" style="padding-left:25px"><p><span>List of species with populated data:</span>
  <a id="s_variation_set_structural_variation" class="help-header sql_schema_show_hide" onclick="show_hide('variation_set_structural_variation','species')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show species
  </a></p>
    <div id="sp_variation_set_structural_variation" style="display:none;">      <ul>
        <li class="sql_schema_species_name">Homo sapiens</li>
      </ul>
    </div>
  </td></tr></table></div></div><div class="sql_schema_table">
  <div id="variation_set_structure"class="sql_schema_table_header" style="border-top-color:#FFD700">
    <div class="sql_schema_table_header_left"><span style="background-color:#FFD700"></span>variation_set_structure</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_variation_set_structure" class="help-header sql_schema_show_hide" onclick="show_hide('variation_set_structure','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">This table stores hierarchical relationships between variation sets by relating them as variation sets and variation subsets.</p>

  <div id="div_variation_set_structure" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>variation_set_super</b></td>
        <td class="bg1">int(10)</td>
        <td class="bg1">-</td>
        <td class="bg1">Primary key. Foreign key references to the <a href="#variation_set" class="sql_schema_link">variation_set</a> table.</td>
        <td class="bg1">primary key<br />key: <i>sub_idx</i></td>
      </tr>

      <tr>
        <td class="bg2"><b>variation_set_sub</b></td>
        <td class="bg2">int(10)</td>
        <td class="bg2">-</td>
        <td class="bg2">Primary key. Foreign key references to the <a href="#variation_set" class="sql_schema_link">variation_set</a> table.</td>
        <td class="bg2">primary key<br />key: <i>sub_idx</i></td>
      </tr>
    </table>
  </div>
<table class="sql_schema_extra"><tr>    <td class="sql_schema_extra_left"">    <p style="font-weight:bold">See also:</p>
  <ul>
      <li><a href="#variation_set">variation_set</a></li>
      </ul>
    </td>
  <td class="sql_schema_extra_separator"></td>
  <td class="sql_schema_extra_right" style="padding-left:25px"><p><span>List of species with populated data:</span>
  <a id="s_variation_set_structure" class="help-header sql_schema_show_hide" onclick="show_hide('variation_set_structure','species')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show species
  </a></p>
    <div id="sp_variation_set_structure" style="display:none;">      <ul>
        <li class="sql_schema_species_name">Bos taurus</li>
        <li class="sql_schema_species_name">Equus caballus</li>
        <li class="sql_schema_species_name">Gallus gallus</li>
        <li class="sql_schema_species_name">Homo sapiens</li>
        <li class="sql_schema_species_name">Ovis aries</li>
        <li class="sql_schema_species_name">Sus scrofa</li>
      </ul>
    </div>
  </td></tr></table></div></div><div class="sql_schema_table">
  <div id="variation_set_variation"class="sql_schema_table_header" style="border-top-color:#FFD700">
    <div class="sql_schema_table_header_left"><span style="background-color:#FFD700"></span>variation_set_variation</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_variation_set_variation" class="help-header sql_schema_show_hide" onclick="show_hide('variation_set_variation','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">A table for mapping variations to variation_sets.</p>

  <div id="div_variation_set_variation" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>variation_id</b></td>
        <td class="bg1">int(10)</td>
        <td class="bg1">-</td>
        <td class="bg1">Primary key. Foreign key references to the <a href="#variation" class="sql_schema_link">variation</a> table.</td>
        <td class="bg1">primary key<br />key: <i>variation_set_idx</i></td>
      </tr>

      <tr>
        <td class="bg2"><b>variation_set_id</b></td>
        <td class="bg2">int(10)</td>
        <td class="bg2">-</td>
        <td class="bg2">Primary key. Foreign key references to the <a href="#variation_set" class="sql_schema_link">variation_set</a> table.</td>
        <td class="bg2">primary key<br />key: <i>variation_set_idx</i></td>
      </tr>
    </table>
  </div>
<table class="sql_schema_extra"><tr>    <td class="sql_schema_extra_left"">    <p style="font-weight:bold">See also:</p>
  <ul>
      <li><a href="#variation">variation</a></li>
      <li><a href="#variation_set">variation_set</a></li>
      </ul>
    </td>
  <td class="sql_schema_extra_separator"></td>
  <td class="sql_schema_extra_right" style="padding-left:25px"><p><span>List of species with populated data:</span>
  <a id="s_variation_set_variation" class="help-header sql_schema_show_hide" onclick="show_hide('variation_set_variation','species')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show species
  </a></p>
    <div id="sp_variation_set_variation" style="display:none;">      <ul>
        <li class="sql_schema_species_name">Bos taurus</li>
        <li class="sql_schema_species_name">Canis familiaris</li>
        <li class="sql_schema_species_name">Danio rerio</li>
        <li class="sql_schema_species_name">Equus caballus</li>
        <li class="sql_schema_species_name">Felis catus</li>
        <li class="sql_schema_species_name">Gallus gallus</li>
        <li class="sql_schema_species_name">Homo sapiens</li>
        <li class="sql_schema_species_name">Macaca mulatta</li>
        <li class="sql_schema_species_name">Meleagris gallopavo</li>
        <li class="sql_schema_species_name">Monodelphis domestica</li>
        <li class="sql_schema_species_name">Mus musculus</li>
        <li class="sql_schema_species_name">Ornithorhynchus anatinus</li>
        <li class="sql_schema_species_name">Ovis aries</li>
        <li class="sql_schema_species_name">Pan troglodytes</li>
        <li class="sql_schema_species_name">Pongo abelii</li>
        <li class="sql_schema_species_name">Rattus norvegicus</li>
        <li class="sql_schema_species_name">Sus scrofa</li>
        <li class="sql_schema_species_name">Taeniopygia guttata</li>
        <li class="sql_schema_species_name">Tetraodon nigroviridis</li>
      </ul>
    </div>
  </td></tr></table></div></div>

<div class="sql_schema_group_header" style="border-color:#000">
  <div id="13" class="sql_schema_group_bullet" style="background-color:#000"></div>
  <h2 id="13">Variation tables</h2>
</div>
<p class="sql_schema_group_header_desc">These tables define the central variation data.</p><div class="sql_schema_table">
  <div id="allele"class="sql_schema_table_header" style="border-top-color:#000">
    <div class="sql_schema_table_header_left"><span style="background-color:#000"></span>allele</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_allele" class="help-header sql_schema_show_hide" onclick="show_hide('allele','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">This table stores information about each of a variation's alleles, along with population frequencies.</p>

  <div id="div_allele" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>allele_id</b></td>
        <td class="bg1">int(11)</td>
        <td class="bg1">-</td>
        <td class="bg1">Primary key, internal identifier.</td>
        <td class="bg1">primary key</td>
      </tr>

      <tr>
        <td class="bg2"><b>variation_id</b></td>
        <td class="bg2">int(11)</td>
        <td class="bg2">-</td>
        <td class="bg2">Foreign key references to the <a href="#variation" class="sql_schema_link">variation</a> table.</td>
        <td class="bg2">key: <i>variation_idx</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>subsnp_id</b></td>
        <td class="bg1">int(11)</td>
        <td class="bg1">NULL</td>
        <td class="bg1">Foreign key references to the <a href="#subsnp_handle" class="sql_schema_link">subsnp_handle</a> table.</td>
        <td class="bg1">key: <i>subsnp_idx</i></td>
      </tr>

      <tr>
        <td class="bg2"><b>allele_code_id</b></td>
        <td class="bg2">int(11)</td>
        <td class="bg2">-</td>
        <td class="bg2">Foreign key reference to <a href="#allele_code" class="sql_schema_link">allele_code</a> table.</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>population_id</b></td>
        <td class="bg1">int(11)</td>
        <td class="bg1">NULL</td>
        <td class="bg1">Foreign key references to the <a href="#population" class="sql_schema_link">population</a> table.</td>
        <td class="bg1">key: <i>population_idx</i></td>
      </tr>

      <tr>
        <td class="bg2"><b>frequency</b></td>
        <td class="bg2">float</td>
        <td class="bg2">NULL</td>
        <td class="bg2">Frequency of this allele in the population.</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>count</b></td>
        <td class="bg1">int(11)</td>
        <td class="bg1">NULL</td>
        <td class="bg1">Number of individuals/samples in the population where this allele is found.</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>frequency_submitter_handle</b></td>
        <td class="bg2">int(10)</td>
        <td class="bg2">NULL</td>
        <td class="bg2">dbSNP handle for submitter of frequency data [may be different to submitter of observed variant]</td>
        <td class="bg2"></td>
      </tr>
    </table>
  </div>
<table class="sql_schema_extra"><tr>    <td class="sql_schema_extra_left"">    <p style="font-weight:bold">See also:</p>
  <ul>
      <li><a href="#variation">variation</a></li>
      <li><a href="#subsnp_handle">subsnp_handle</a></li>
      <li><a href="#allele_code">allele_code</a></li>
      <li><a href="#population">population</a></li>
      <li><a href="#submitter_handle">submitter_handle</a></li>
      </ul>
    </td>
  <td class="sql_schema_extra_separator"></td>
  <td class="sql_schema_extra_right" style="padding-left:25px"><p><span>List of species with populated data:</span>
  <a id="s_allele" class="help-header sql_schema_show_hide" onclick="show_hide('allele','species')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show species
  </a></p>
    <div id="sp_allele" style="display:none;">      <ul>
        <li class="sql_schema_species_name">Bos taurus</li>
        <li class="sql_schema_species_name">Canis familiaris</li>
        <li class="sql_schema_species_name">Danio rerio</li>
        <li class="sql_schema_species_name">Drosophila melanogaster</li>
        <li class="sql_schema_species_name">Equus caballus</li>
        <li class="sql_schema_species_name">Felis catus</li>
        <li class="sql_schema_species_name">Gallus gallus</li>
        <li class="sql_schema_species_name">Homo sapiens</li>
        <li class="sql_schema_species_name">Macaca mulatta</li>
        <li class="sql_schema_species_name">Meleagris gallopavo</li>
        <li class="sql_schema_species_name">Monodelphis domestica</li>
        <li class="sql_schema_species_name">Mus musculus</li>
        <li class="sql_schema_species_name">Nomascus leucogenys</li>
        <li class="sql_schema_species_name">Ornithorhynchus anatinus</li>
        <li class="sql_schema_species_name">Ovis aries</li>
        <li class="sql_schema_species_name">Pan troglodytes</li>
        <li class="sql_schema_species_name">Pongo abelii</li>
        <li class="sql_schema_species_name">Rattus norvegicus</li>
        <li class="sql_schema_species_name">Saccharomyces cerevisiae</li>
        <li class="sql_schema_species_name">Sus scrofa</li>
        <li class="sql_schema_species_name">Taeniopygia guttata</li>
        <li class="sql_schema_species_name">Tetraodon nigroviridis</li>
      </ul>
    </div>
  </td></tr></table></div></div><div class="sql_schema_table">
  <div id="variation"class="sql_schema_table_header" style="border-top-color:#000">
    <div class="sql_schema_table_header_left"><span style="background-color:#000"></span>variation</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_variation" class="help-header sql_schema_show_hide" onclick="show_hide('variation','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">This is the schema's generic representation of a variation, defined as a genetic feature that varies between individuals of the same species.  The most common type is the single nucleotide variation (SNP) though the schema also accommodates copy number variations (CNVs) and structural variations (SVs).<br /> In Ensembl, a variation is defined by its flanking sequence rather than its mapped location on a chromosome; a variation may in fact have multiple mappings across a genome,  although this fails our <a href="/info/genome/variation/data_description.html#quality_control">Quality Control</a>.<br />  This table stores a variation's name (commonly an ID of the form e.g. rs123456, assigned by dbSNP), along with an ancestral (or reference) allele.</p>

  <div id="div_variation" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>variation_id</b></td>
        <td class="bg1">int(10)</td>
        <td class="bg1">-</td>
        <td class="bg1">Primary key, internal identifier.</td>
        <td class="bg1">primary key</td>
      </tr>

      <tr>
        <td class="bg2"><b>source_id</b></td>
        <td class="bg2">int(10)</td>
        <td class="bg2">-</td>
        <td class="bg2">Foreign key references to the <a href="#source" class="sql_schema_link">source</a> table.</td>
        <td class="bg2">key: <i>source_idx</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>name</b></td>
        <td class="bg1">varchar(255)</td>
        <td class="bg1">NULL</td>
        <td class="bg1">Name of the variation. e.g. "rs1333049".</td>
        <td class="bg1">unique: <i> name </i></td>
      </tr>

      <tr>
        <td class="bg2"><b>ancestral_allele</b></td>
        <td class="bg2">varchar(255)</td>
        <td class="bg2">NULL</td>
        <td class="bg2">Taken from dbSNP to show ancestral allele for the variation.</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>flipped</b></td>
        <td class="bg1">tinyint(1)</td>
        <td class="bg1">NULL</td>
        <td class="bg1">This is set to 1 if the variant is flipped from the negative to the positive strand during import.</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>class_attrib_id</b></td>
        <td class="bg2">int(10)</td>
        <td class="bg2">0</td>
        <td class="bg2">Class of the variation, key into the <a href="#attrib" class="sql_schema_link">attrib</a> table.<br /> The list of variation classes is available <a href="/info/genome/variation/data_description.html#classes">here</a>.</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>somatic</b></td>
        <td class="bg1">tinyint(1)</td>
        <td class="bg1">0</td>
        <td class="bg1">Flags whether this variation is known to be somatic or not</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>minor_allele</b></td>
        <td class="bg2">varchar(50)</td>
        <td class="bg2">NULL</td>
        <td class="bg2">The minor allele of this variant, as reported by dbSNP</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>minor_allele_freq</b></td>
        <td class="bg1">float</td>
        <td class="bg1">NULL</td>
        <td class="bg1">The 'global' frequency of the minor allele of this variant, as reported by dbSNP</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>minor_allele_count</b></td>
        <td class="bg2">int(10)</td>
        <td class="bg2">NULL</td>
        <td class="bg2">The number of samples the minor allele of this variant is found in, as reported by dbSNP</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>clinical_significance</b></td>
        <td class="bg1">SET:<ul class="sql_schema_table_column_type">  <li>uncertain significance</li>  <li>not provided</li>  <li>benign</li>  <li>likely benign</li>  <li>likely pathogenic</li>  <li>pathogenic</li>  <li>drug response</li>  <li>histocompatibility</li>  <li>other</li>  <li>confers sensitivity</li>  <li>risk factor</li>  <li>association</li>  <li>protective</li></ul></td>
        <td class="bg1">NULL</td>
        <td class="bg1">A set of clinical significance classes assigned to the variant.<br />  The list of clinical significances is available <a href="/info/genome/variation/data_description.html#clin_significance">here</a>.</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>evidence_attribs</b></td>
        <td class="bg2">SET:<ul class="sql_schema_table_column_type">  <li>367</li>  <li>368</li>  <li>369</li>  <li>370</li>  <li>371</li>  <li>372</li>  <li>418</li>  <li>421</li></ul></td>
        <td class="bg2">NULL</td>
        <td class="bg2">A summary of the evidence supporting a variant as a guide to its potential reliability. See the evidence descriptions <a href="/info/genome/variation/data_description.html#evidence_status">here</a>.</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>display</b></td>
        <td class="bg1">int(1)</td>
        <td class="bg1">1</td>
        <td class="bg1">Flags whether this variation should be displayed in browser tracks and returned by default by the API</td>
        <td class="bg1"></td>
      </tr>
    </table>
  </div>
<table class="sql_schema_extra"><tr>    <td class="sql_schema_extra_left"">    <p style="font-weight:bold">See also:</p>
  <ul>
      <li><a href="#variation_synonym">variation_synonym</a></li>
      <li><a href="#failed_variation">failed_variation</a></li>
      <li><a href="#variation_feature">variation_feature</a></li>
      <li><a href="#allele">allele</a></li>
      <li><a href="#sample_genotype_multiple_bp">sample_genotype_multiple_bp</a></li>
      <li><a href="#compressed_genotype_var">compressed_genotype_var</a></li>
      <li><a href="#attrib">attrib</a></li>
      <li><a href="#source">source</a></li>
      </ul>
    </td>
  <td class="sql_schema_extra_separator"></td>
  <td class="sql_schema_extra_right" style="padding-left:25px"><p><span>List of species with populated data:</span>
  <a id="s_variation" class="help-header sql_schema_show_hide" onclick="show_hide('variation','species')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show species
  </a></p>
    <div id="sp_variation" style="display:none;">      <ul>
        <li class="sql_schema_species_name">Bos taurus</li>
        <li class="sql_schema_species_name">Canis familiaris</li>
        <li class="sql_schema_species_name">Danio rerio</li>
        <li class="sql_schema_species_name">Drosophila melanogaster</li>
        <li class="sql_schema_species_name">Equus caballus</li>
        <li class="sql_schema_species_name">Felis catus</li>
        <li class="sql_schema_species_name">Gallus gallus</li>
        <li class="sql_schema_species_name">Homo sapiens</li>
        <li class="sql_schema_species_name">Macaca mulatta</li>
        <li class="sql_schema_species_name">Meleagris gallopavo</li>
        <li class="sql_schema_species_name">Monodelphis domestica</li>
        <li class="sql_schema_species_name">Mus musculus</li>
        <li class="sql_schema_species_name">Nomascus leucogenys</li>
        <li class="sql_schema_species_name">Ornithorhynchus anatinus</li>
        <li class="sql_schema_species_name">Ovis aries</li>
        <li class="sql_schema_species_name">Pan troglodytes</li>
        <li class="sql_schema_species_name">Pongo abelii</li>
        <li class="sql_schema_species_name">Rattus norvegicus</li>
        <li class="sql_schema_species_name">Saccharomyces cerevisiae</li>
        <li class="sql_schema_species_name">Sus scrofa</li>
        <li class="sql_schema_species_name">Taeniopygia guttata</li>
        <li class="sql_schema_species_name">Tetraodon nigroviridis</li>
      </ul>
    </div>
  </td></tr></table></div></div><div class="sql_schema_table">
  <div id="variation_attrib"class="sql_schema_table_header" style="border-top-color:#000">
    <div class="sql_schema_table_header_left"><span style="background-color:#000"></span>variation_attrib</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_variation_attrib" class="help-header sql_schema_show_hide" onclick="show_hide('variation_attrib','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">This table stores miscellaneous attributes associated with a variation entry.</p>

  <div id="div_variation_attrib" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>variation_id</b></td>
        <td class="bg1">INT(11)</td>
        <td class="bg1">-</td>
        <td class="bg1">Foreign key references <a href="#variation" class="sql_schema_link">variation</a> table</td>
        <td class="bg1">key: <i>variation_idx</i></td>
      </tr>

      <tr>
        <td class="bg2"><b>attrib_id</b></td>
        <td class="bg2">INT(11)</td>
        <td class="bg2">NULL</td>
        <td class="bg2">Foreign key references <a href="#attrib" class="sql_schema_link">attrib</a> table, describes the attribute</td>
        <td class="bg2">key: <i>attrib_value_idx</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>value</b></td>
        <td class="bg1">VARCHAR(255)</td>
        <td class="bg1">NULL</td>
        <td class="bg1">Attribute value</td>
        <td class="bg1">key: <i>attrib_value_idx</i></td>
      </tr>
    </table>
  </div>
<table class="sql_schema_extra"><tr>    <td class="sql_schema_extra_left"">    <p style="font-weight:bold">See also:</p>
  <ul>
      <li><a href="#variation">variation</a></li>
      <li><a href="#attrib">attrib</a></li>
      </ul>
    </td>
  <td class="sql_schema_extra_separator"></td>
  <td class="sql_schema_extra_right" style="padding-left:25px"><p><span>List of species with populated data:</span>
  <a id="s_variation_attrib" class="help-header sql_schema_show_hide" onclick="show_hide('variation_attrib','species')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show species
  </a></p>
    <div id="sp_variation_attrib" style="display:none;">      <ul>
        <li class="sql_schema_species_name">Danio rerio</li>
        <li class="sql_schema_species_name">Gallus gallus</li>
        <li class="sql_schema_species_name">Homo sapiens</li>
        <li class="sql_schema_species_name">Meleagris gallopavo</li>
        <li class="sql_schema_species_name">Mus musculus</li>
        <li class="sql_schema_species_name">Ovis aries</li>
        <li class="sql_schema_species_name">Sus scrofa</li>
      </ul>
    </div>
  </td></tr></table></div></div><div class="sql_schema_table">
  <div id="variation_feature"class="sql_schema_table_header" style="border-top-color:#000">
    <div class="sql_schema_table_header_left"><span style="background-color:#000"></span>variation_feature</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_variation_feature" class="help-header sql_schema_show_hide" onclick="show_hide('variation_feature','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">This table represents mappings of variations to genomic locations. It stores an allele string representing the different possible alleles that are found at that locus e.g. "A/T" for a SNP, as well as a "worst case" consequence of the mutation. It also acts as part of the relationship between variations and transcripts.</p>

  <div id="div_variation_feature" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>variation_feature_id</b></td>
        <td class="bg1">int(10)</td>
        <td class="bg1">-</td>
        <td class="bg1">Primary key, internal identifier.</td>
        <td class="bg1">primary key</td>
      </tr>

      <tr>
        <td class="bg2"><b>seq_region_id</b></td>
        <td class="bg2">int(10)</td>
        <td class="bg2">-</td>
        <td class="bg2">Foreign key references <a href="#seq_region" class="sql_schema_link">seq_region</a> in core db. Refers to the seq_region which this variant is on, which may be a chromosome, a clone, etc...</td>
        <td class="bg2">key: <i>pos_idx</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>seq_region_start</b></td>
        <td class="bg1">int</td>
        <td class="bg1">-</td>
        <td class="bg1">The start position of the variation on the <a href="#seq_region" class="sql_schema_link">seq_region</a>.</td>
        <td class="bg1">key: <i>pos_idx</i></td>
      </tr>

      <tr>
        <td class="bg2"><b>seq_region_end</b></td>
        <td class="bg2">int</td>
        <td class="bg2">-</td>
        <td class="bg2">The end position of the variation on the <a href="#seq_region" class="sql_schema_link">seq_region</a>.</td>
        <td class="bg2">key: <i>pos_idx</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>seq_region_strand</b></td>
        <td class="bg1">tinyint</td>
        <td class="bg1">-</td>
        <td class="bg1">The orientation of the variation on the <a href="#seq_region" class="sql_schema_link">seq_region</a>.</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>variation_id</b></td>
        <td class="bg2">int(10)</td>
        <td class="bg2">-</td>
        <td class="bg2">Foreign key references to the <a href="#variation" class="sql_schema_link">variation</a> table.</td>
        <td class="bg2">key: <i>variation_idx</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>allele_string</b></td>
        <td class="bg1">varchar(50000)</td>
        <td class="bg1">NULL</td>
        <td class="bg1">This is a denormalised string taken from the alleles in the allele table associated with this variation. The reference allele (i.e. one on the reference genome comes first).</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>variation_name</b></td>
        <td class="bg2">varchar(255)</td>
        <td class="bg2">NULL</td>
        <td class="bg2">A denormalisation taken from the variation table. This is the name or identifier that is used for displaying the feature.</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>map_weight</b></td>
        <td class="bg1">int</td>
        <td class="bg1">-</td>
        <td class="bg1">The number of times that this variation has mapped to the genome. This is a denormalisation as this particular feature is one example of a mapped location. This can be used to limit the the features that come back from a query.</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>flags</b></td>
        <td class="bg2">SET('genotyped')</td>
        <td class="bg2">NULL</td>
        <td class="bg2">Flag to filter the selection of variations.</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>source_id</b></td>
        <td class="bg1">int(10)</td>
        <td class="bg1">-</td>
        <td class="bg1">Foreign key references to the source table.</td>
        <td class="bg1">key: <i>source_idx</i></td>
      </tr>

      <tr>
        <td class="bg2"><b>consequence_types</b></td>
        <td class="bg2">SET:<ul class="sql_schema_table_column_type">  <li>intergenic_variant</li>  <li>splice_acceptor_variant</li>  <li>splice_donor_variant</li>  <li>stop_lost</li>  <li>coding_sequence_variant</li>  <li>missense_variant</li>  <li>stop_gained</li>  <li>synonymous_variant</li>  <li>frameshift_variant</li>  <li>non_coding_transcript_variant</li>  <li>non_coding_transcript_exon_variant</li>  <li>mature_miRNA_variant</li>  <li>NMD_transcript_variant</li>  <li>5_prime_UTR_variant</li>  <li>3_prime_UTR_variant</li>  <li>incomplete_terminal_codon_variant</li>  <li>intron_variant</li>  <li>splice_region_variant</li>  <li>downstream_gene_variant</li>  <li>upstream_gene_variant</li>  <li>start_lost</li>  <li>stop_retained_variant</li>  <li>inframe_insertion</li>  <li>inframe_deletion</li>  <li>transcript_ablation</li>  <li>transcript_fusion</li>  <li>transcript_amplification</li>  <li>transcript_translocation</li>  <li>TFBS_ablation</li>  <li>TFBS_fusion</li>  <li>TFBS_amplification</li>  <li>TFBS_translocation</li>  <li>regulatory_region_ablation</li>  <li>regulatory_region_fusion</li>  <li>regulatory_region_amplification</li>  <li>regulatory_region_translocation</li>  <li>feature_elongation</li>  <li>feature_truncation</li>  <li>regulatory_region_variant</li>  <li>TF_binding_site_variant</li>  <li>protein_altering_variant </li></ul></td>
        <td class="bg2">'intergenic_variant'</td>
        <td class="bg2">The SO term(s) of all unique observed consequence types of this variation feature.<br /> The list of consequence descriptions is available <a href="/info/genome/variation/predicted_data.html#consequences">here</a>.</td>
        <td class="bg2">key: <i>consequence_type_idx</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>variation_set_id</b></td>
        <td class="bg1">SET:<ul class="sql_schema_table_column_type">  <li>1</li>  <li>2</li>  <li>3</li>  <li>4</li>  <li>5</li>  <li>6</li>  <li>7</li>  <li>8</li>  <li>9</li>  <li>10</li>  <li>11</li>  <li>12</li>  <li>13</li>  <li>14</li>  <li>15</li>  <li>16</li>  <li>17</li>  <li>18</li>  <li>19</li>  <li>20</li>  <li>21</li>  <li>22</li>  <li>23</li>  <li>24</li>  <li>25</li>  <li>26</li>  <li>27</li>  <li>28</li>  <li>29</li>  <li>30</li>  <li>31</li>  <li>32</li>  <li>33</li>  <li>34</li>  <li>35</li>  <li>36</li>  <li>37</li>  <li>38</li>  <li>39</li>  <li>40</li>  <li>41</li>  <li>42</li>  <li>43</li>  <li>44</li>  <li>45</li>  <li>46</li>  <li>47</li>  <li>48</li>  <li>49</li>  <li>50</li>  <li>51</li>  <li>52</li>  <li>53</li>  <li>54</li>  <li>55</li>  <li>56</li>  <li>57</li>  <li>58</li>  <li>59</li>  <li>60</li>  <li>61</li>  <li>62</li>  <li>63</li>  <li>64 </li></ul></td>
        <td class="bg1">''</td>
        <td class="bg1">The variation feature can belong to a <a href="#variation_set" class="sql_schema_link">variation_set</a>.</td>
        <td class="bg1">key: <i>variation_set_idx</i></td>
      </tr>

      <tr>
        <td class="bg2"><b>class_attrib_id</b></td>
        <td class="bg2">int(10)</td>
        <td class="bg2">0</td>
        <td class="bg2">Class of the variation, key in the <a href="#attrib" class="sql_schema_link">attrib</a> table.<br /> The list of variation classes is available <a href="/info/genome/variation/data_description.html#classes">here</a>.</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>somatic</b></td>
        <td class="bg1">tinyint(1)</td>
        <td class="bg1">0</td>
        <td class="bg1">Flags whether this variation_feature is somatic or germline</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>minor_allele</b></td>
        <td class="bg2">varchar(50)</td>
        <td class="bg2">NULL</td>
        <td class="bg2">The minor allele of this variant, as reported by dbSNP</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>minor_allele_freq</b></td>
        <td class="bg1">float</td>
        <td class="bg1">NULL</td>
        <td class="bg1">The 'global' frequency of the minor allele of this variant, as reported by dbSNP</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>minor_allele_count</b></td>
        <td class="bg2">int(10)</td>
        <td class="bg2">NULL</td>
        <td class="bg2">The number of samples the minor allele of this variant is found in, as reported by dbSNP</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>alignment_quality</b></td>
        <td class="bg1">double</td>
        <td class="bg1">NULL</td>
        <td class="bg1">Quality of alignment for variants mapped by flanks rather than position justified.</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>evidence_attribs</b></td>
        <td class="bg2">SET:<ul class="sql_schema_table_column_type">  <li>367</li>  <li>368</li>  <li>369</li>  <li>370</li>  <li>371</li>  <li>372</li>  <li>418</li>  <li>421</li></ul></td>
        <td class="bg2">NULL</td>
        <td class="bg2">A summary of the evidence supporting a variant as a guide to its potential reliability. See the evidence descriptions <a href="/info/genome/variation/data_description.html#evidence_status">here</a>.</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>clinical_significance</b></td>
        <td class="bg1">SET:<ul class="sql_schema_table_column_type">  <li>uncertain significance</li>  <li>not provided</li>  <li>benign</li>  <li>likely benign</li>  <li>likely pathogenic</li>  <li>pathogenic</li>  <li>drug response</li>  <li>histocompatibility</li>  <li>other</li>  <li>confers sensitivity</li>  <li>risk factor</li>  <li>association</li>  <li>protective</li></ul></td>
        <td class="bg1">NULL</td>
        <td class="bg1">A set of clinical significance classes assigned to the variant.<br />  The list of clinical significances is available <a href="/info/genome/variation/data_description.html#clin_significance">here</a>.</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>display</b></td>
        <td class="bg2">int(1)</td>
        <td class="bg2">1</td>
        <td class="bg2">Flags whether this variation should be displayed in browser tracks and returned by default by the API</td>
        <td class="bg2"></td>
      </tr>
    </table>
  </div>
<table class="sql_schema_extra"><tr>    <td class="sql_schema_extra_left"">    <p style="font-weight:bold">See also:</p>
  <ul>
      <li><a href="#variation">variation</a></li>
      <li><a href="#transcript_variation">transcript_variation</a></li>
      <li><a href="#seq_region">seq_region</a></li>
      <li><a href="#attrib">attrib</a></li>
      <li><a href="#variation_set">variation_set</a></li>
      </ul>
    </td>
  <td class="sql_schema_extra_separator"></td>
  <td class="sql_schema_extra_right" style="padding-left:25px"><p><span>List of species with populated data:</span>
  <a id="s_variation_feature" class="help-header sql_schema_show_hide" onclick="show_hide('variation_feature','species')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show species
  </a></p>
    <div id="sp_variation_feature" style="display:none;">      <ul>
        <li class="sql_schema_species_name">Bos taurus</li>
        <li class="sql_schema_species_name">Canis familiaris</li>
        <li class="sql_schema_species_name">Danio rerio</li>
        <li class="sql_schema_species_name">Drosophila melanogaster</li>
        <li class="sql_schema_species_name">Equus caballus</li>
        <li class="sql_schema_species_name">Felis catus</li>
        <li class="sql_schema_species_name">Gallus gallus</li>
        <li class="sql_schema_species_name">Homo sapiens</li>
        <li class="sql_schema_species_name">Macaca mulatta</li>
        <li class="sql_schema_species_name">Meleagris gallopavo</li>
        <li class="sql_schema_species_name">Monodelphis domestica</li>
        <li class="sql_schema_species_name">Mus musculus</li>
        <li class="sql_schema_species_name">Nomascus leucogenys</li>
        <li class="sql_schema_species_name">Ornithorhynchus anatinus</li>
        <li class="sql_schema_species_name">Ovis aries</li>
        <li class="sql_schema_species_name">Pan troglodytes</li>
        <li class="sql_schema_species_name">Pongo abelii</li>
        <li class="sql_schema_species_name">Rattus norvegicus</li>
        <li class="sql_schema_species_name">Saccharomyces cerevisiae</li>
        <li class="sql_schema_species_name">Sus scrofa</li>
        <li class="sql_schema_species_name">Taeniopygia guttata</li>
        <li class="sql_schema_species_name">Tetraodon nigroviridis</li>
      </ul>
    </div>
  </td></tr></table></div></div><div class="sql_schema_table">
  <div id="variation_synonym"class="sql_schema_table_header" style="border-top-color:#000">
    <div class="sql_schema_table_header_left"><span style="background-color:#000"></span>variation_synonym</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_variation_synonym" class="help-header sql_schema_show_hide" onclick="show_hide('variation_synonym','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">This table allows for a variation to have multiple IDs, generally given by multiple sources.</p>

  <div id="div_variation_synonym" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>variation_synonym_id</b></td>
        <td class="bg1">int(10)</td>
        <td class="bg1">-</td>
        <td class="bg1">Primary key, internal identifier.</td>
        <td class="bg1">primary key</td>
      </tr>

      <tr>
        <td class="bg2"><b>variation_id</b></td>
        <td class="bg2">int(10)</td>
        <td class="bg2">-</td>
        <td class="bg2">Foreign key references to the variation table.</td>
        <td class="bg2">key: <i>variation_idx</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>subsnp_id</b></td>
        <td class="bg1">int(15)</td>
        <td class="bg1">NULL</td>
        <td class="bg1">Foreign key references to the subsnp_handle table.</td>
        <td class="bg1">key: <i>subsnp_idx</i></td>
      </tr>

      <tr>
        <td class="bg2"><b>source_id</b></td>
        <td class="bg2">int(10)</td>
        <td class="bg2">-</td>
        <td class="bg2">Foreign key references to the source table.</td>
        <td class="bg2">unique: <i>name, source_id</i><br />key: <i>source_idx</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>name</b></td>
        <td class="bg1">varchar(255)</td>
        <td class="bg1">NULL</td>
        <td class="bg1">Name of the synonym variation. e.g. 'rs1333049'.</td>
        <td class="bg1">unique: <i>name, source_id</i></td>
      </tr>
    </table>
  </div>
<table class="sql_schema_extra"><tr>    <td class="sql_schema_extra_left"">    <p style="font-weight:bold">See also:</p>
  <ul>
      <li><a href="#source">source</a></li>
      <li><a href="#variation">variation</a></li>
      <li><a href="#subsnp_handle">subsnp_handle</a></li>
      </ul>
    </td>
  <td class="sql_schema_extra_separator"></td>
  <td class="sql_schema_extra_right" style="padding-left:25px"><p><span>List of species with populated data:</span>
  <a id="s_variation_synonym" class="help-header sql_schema_show_hide" onclick="show_hide('variation_synonym','species')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show species
  </a></p>
    <div id="sp_variation_synonym" style="display:none;">      <ul>
        <li class="sql_schema_species_name">Bos taurus</li>
        <li class="sql_schema_species_name">Canis familiaris</li>
        <li class="sql_schema_species_name">Danio rerio</li>
        <li class="sql_schema_species_name">Equus caballus</li>
        <li class="sql_schema_species_name">Felis catus</li>
        <li class="sql_schema_species_name">Gallus gallus</li>
        <li class="sql_schema_species_name">Homo sapiens</li>
        <li class="sql_schema_species_name">Mus musculus</li>
        <li class="sql_schema_species_name">Nomascus leucogenys</li>
        <li class="sql_schema_species_name">Ornithorhynchus anatinus</li>
        <li class="sql_schema_species_name">Ovis aries</li>
        <li class="sql_schema_species_name">Rattus norvegicus</li>
        <li class="sql_schema_species_name">Sus scrofa</li>
        <li class="sql_schema_species_name">Taeniopygia guttata</li>
        <li class="sql_schema_species_name">Tetraodon nigroviridis</li>
      </ul>
    </div>
  </td></tr></table></div></div>

    </div>
  </div>
</body>
</html>
