<html>
<head>
<title>BLAST/BLAT setup instructions</title>
</head>
<body>

<h1>BLAST/BLAT setup instructions</h1>

<p>As with our old code, the Tools system combines BLAST and BLAT sequence searches
into one interface. However Ensembl now uses NCBI BLAST by default, though the Tools 
system can be configured to use wu-blast instead.</p>

<h2>Setting up BLAT</h2>

<ol>
<li>Install the BLAT executable, available from
the <a href="http://hgdownload.cse.ucsc.edu/admin/exe/">UCSC</a>.</li>
<li>Download the required FASTA files from the <a href="[[SPECIESDEFS::ENSEMBL_FTP_URL]]/current_fasta/">Ensembl FTP site</a>, 
and use them to create 2bit files for rapid indexing, as described in the 
<a href="http://genome.ucsc.edu/goldenpath/help/blatSpec.html">UCSC BLAT specifications</a> N.B. you can use the same Ensembl pipeline, described below, to produce both BLAT and BLAST indices.</li>
<li>Configure the locations of these files in <kbd>public-plugins/mirror/conf/SiteDefs.pm</kbd>:
<p><pre>
    $SiteDefs::ENSEMBL_BLAT_BIN_PATH    = '/usr/local/bin/gfClient';
    $SiteDefs::ENSEMBL_BLAT_TWOBIT_DIR  = '/usr/local/ensembl/tools_data/blat';
</pre></p>
</li>
</ol>

<h2>Setting up NCBI-BLAST</h2>

<ol>
<li>Install a suitable version of the BLAST software, available from
the <a href="http://ftp.ncbi.nlm.nih.gov/blast/executables/LATEST/">NCBI FTP site</a>.</li>
<li>Download necessary files (dna and genes) from <a href="[[SPECIESDEFS::ENSEMBL_FTP_URL]]/current_ncbi_blast/" target="_blank">Latest Blast FTP files</a></li>
<li>Configure your BLAST paths in SiteDefs.pm:
<p><pre>
$SiteDefs::ENSEMBL_NCBIBLAST_BIN_PATH = '/path/to/ncbi-blast/bin'; # path to blast executables  
$SiteDefs::ENSEMBL_NCBIBLAST_MATRIX = '/path/to/ncbi-blast/data'; # path to blast matrix files 
$SiteDefs::ENSEMBL_NCBIBLAST_DATA_PATH_DNA = "/path/to/blast/genomic"; # path for the blast DNA index files
$SiteDefs::ENSEMBL_NCBIBLAST_DATA_PATH = "/path/to/genes"; # path for the blast index files (other than DNA)
$SiteDefs::ENSEMBL_REPEATMASK_BIN_PATH = '/path/to/RepeatMasker'; # path to RepeatMasker executable
</pre></p>
</li>
</ol>


<!-- <h2>Setting up WU-BLAST</h2>-->

<h2>Excluding data sources</h2>

<p>By default, the BLAST interface shows all data sources used by Ensembl. If you do not
have all these datasources for your species, you will need to make a copy of the ENSEMBL_BLAST_DATASOURCES_ALL
code block in public-plugins/tools/conf/ini-files/MULTI.ini and edit it to your requirements. E.g.</p>
<p>Original version in tools plugin:</p>
<p><pre>
[ENSEMBL_BLAST_DATASOURCES_ALL]
ORDER           = [LATESTGP LATESTGP_MASKED LATESTGP_SOFT CDNA_ALL CDNA_ABINITIO NCRNA PEP_ALL PEP_ABINITIO] ; order preserved
LATESTGP        = dna Genomic sequence
LATESTGP_MASKED = dna Genomic sequence (hard masked)
LATESTGP_SOFT   = dna Genomic sequence (soft masked)
CDNA_ALL        = dna cDNAs (transcripts/splice variants)
CDNA_ABINITIO   = dna Ab-initio cDNAs (Genscan/SNAP)
NCRNA           = dna Ensembl Non-coding RNA genes
PEP_ALL         = peptide Proteins (GENCODE/Ensembl)
PEP_ABINITIO    = peptide Ab-initio Peptides (Genscan/SNAP)
</pre></p>

<p>Edited version in your plugin:</p>
<p><pre>
[ENSEMBL_BLAST_DATASOURCES_ALL]
ORDER           = [LATESTGP LATESTGP_MASKED LATESTGP_SOFT CDNA_ALL PEP_ALL PEP_ABINITIO] ; order preserved
LATESTGP        = dna Genomic sequence
LATESTGP_MASKED = dna Genomic sequence (hard masked)
LATESTGP_SOFT   = dna Genomic sequence (soft masked)
CDNA_ALL        = dna cDNAs (transcripts/splice variants)
PEP_ALL         = peptide Proteins (GENCODE/Ensembl)
PEP_ABINITIO    = peptide Ab-initio Peptides (Genscan/SNAP)
</pre></p>


</body>
</html>
