<html>
<head>
<title>Downloads - [[PERL::sprintf('%s %s', $_->species_defs->ENSEMBL_SITE_NAME, $_->species_defs->SITE_RELEASE_VERSION || $_->species_defs->ENSEMBL_VERSION)]]</title>
</head>
<body>

<table class="blobs">
<tr>

<td class="fourcol">
<h2 class="first"><a href="/info/data/export.html">Download a sequence or region</a></h2>
<div class="centered"><a href="/info/data/export.html"><img src="/img/download_sequence.gif" alt="" title="Find out more" /></a></div>
<p>Click on the 'Export data' button in the lefthand menu of most pages to export:</p>
<ul>
<li>FASTA sequence</li>
<li>GTF or GFF features</li>
</ul>
<p>...and more!</p>
</td>


<td class="fourcol">
<h2 class="first"><a href="/info/docs/api/">Fetch data programmatically</a></h2>
<div class="centered"><a href="/info/docs/api/"><img src="/img/download_api.gif" alt="" title="Find out more" /></a></div>
<p>Write your own Perl scripts to retrieve small-to-medium datasets. All our data, as well as added functionality, is available 
through the Ensembl Perl API.</p>
<p>Use the API to retrieve gene and transcript sets, fetch alignments between sequences, compare allele frequencies and much more!</p>
<p>You can also use our <a href='http://rest.ensembl.org/'>REST API</a> to retrieve data to process in the programming language of your choice.</p>
</td>


</tr>
</table>


</body>
</html>
