<html>
<head>
<title>Input form</title>
<meta name="order" content="1" />
</head>

<body>

  
  <div style="float:right"><img src="/img/vep_logo.png"/></div>

  <h1 id="top"><span style="color:#006;padding-right:15px">Variant Effect Predictor</span><span style="color:#666"><img src="/i/16/pencil.png"/> Input form</span></h1>
  <hr/>

<p> When you reach the VEP web interface, you will be presented with a form to
enter your data and alter various options.</p>

<h2 id="input">Data input</h2>

<div class="column-wrapper">
    <div class="column-two">
      <div class="column-left">
          <ol>
              <li> First select the correct species for your data. Ensembl hosts
              many vertebrate genomes; genomes for plants, protists and fungi
              can be found at <a href="http://www.ensemblgenomes.org/">Ensembl
              Genomes</a>. </li>
              
              <li> You can optionally choose a name for the data you upload -
              this can make it easier for you to identify jobs and files that
              you have uploaded to the VEP at a later point. </li>
              
              <li> <p>You have three options for uploading your data:
                  
                  <ul>
                      <li><b>File upload</b> - click the "Choose file" button
                      and locate the file on your system</li>
                      <li><b>Paste file</b> - simply copy and paste the contents
                      of your file into the large text box</li>
                      <li><b>File URL</b> - point the VEP to a file hosted on a
                      publically accessible address. This can be either a
                      <b>http://</b> or <b>ftp://</b> address.</li>
                  </ul>
                  
                  Once you have uploaded some data, you can select it as the
                  input for future jobs by choosing the data from the drop down
                  menu.</p>
                  
                  <p> The format of your data is automatically detected; see the
                  examples or the <a href="../vep_formats.html#input">input
                  format</a> documentation. </p>
              </li>
              
              <li> For pasted data you can get an instant preview of the results of your
              first variant by clicking the button that appears when you paste
              your data. This quickly shows you the consequence type, the IDs of
              any overlapping variants, genes, transcripts and regulatory
              features, as well as SIFT and PolyPhen predictions. To see the full
              results set submit your job as normal. </li>
              
              <li>
                  <p> For some species you can select which transcript database
                  to use. The default is to use Ensembl transcripts, which offer
                  the most rich annotation through VEP. </p>
                  <p> GENCODE Basic is a subset of the GENCODE gene set, and is
                  intended to provide a simplified, high-quality subset of the
                  GENCODE transcript annotations that will be useful to the
                  majority of users. GENCODE Basic includes all genes in the
                  GENCODE gene set, with a representative subset of the
                  transcripts (splice variants). </p>
                  <p>You can also select to use RefSeq transcripts from the <a
                  href="/info/docs/api/core/index.html#est">otherfeatures
                  database</a>; note though that these transcripts are simply
                  aligned to the reference genome and the database is missing
                  much of the annotation found when using the main Ensembl
                  database (e.g. protein domains, CCDS identifiers).</p>
              </li>
          </ol>
      </div>
    </div>
      
    <div class="column-two">
      <div class="column-right" style="padding-left:2em;">
          <div class="plain-box">
              <img src="/img/vep_web_input.png">
          </div>
      </div>
    </div>
</div>


<hr/>
<h2 id="ident">Identifiers</h2>

<div class="column-wrapper">
    <div class="column-two">
        <div class="column-left">
            VEP can provide additional identifiers for genes, transcripts,
            proteins and variants.
            
            <ul>
                <li><b>Gene symbol</b>
                  <p>Add the gene symbol for the gene to
                  the output. This will typically be, for example, the <a
                  href="http://www.genenames.org/" rel="external">HGNC</a>
                  identifier for genes in human. Equivalent to <a
                  href="/info/docs/tools/vep/script/vep_options.html#opt_symbol">--symbol</a>
                  in the VEP script.</p>
                </li>

                 <li><b>Transcript version</b>
                   <p>Add the transcript version to the transcript identifier. Equivalent to <a 
                   href="/info/docs/tools/vep/script/vep_options.html#opt_transcript_version">--transcript_version</a>.</p> 
 
                <li><b>CCDS</b>
                  <p>Add the <a href="/info/genome/genebuild/ccds.html">Consensus CDS</a>
                  transcript identifier where available. Equivalent to <a
                  href="/info/docs/tools/vep/script/vep_options.html#opt_ccds">--ccds</a>.</p>
                </li>
                
                <li><b>Protein</b>
                  <p>Add the Ensembl protein identifer (ENSP). Equivalent to <a
                  href="/info/docs/tools/vep/script/vep_options.html#opt_protein">--protein</a>.</p>
                </li>
                
                <li><b>UniProt</b>
                  <p>Add identifiers for translated protein
                  products from three <a href="http://www.ebi.ac.uk/uniprot"
                  rel="external">UniProt</a>-related databases (SWISSPROT, TREMBL
                  and UniParc). Equivalent to <a
                  href="/info/docs/tools/vep/script/vep_options.html#opt_uniprot">--uniprot</a>.</p>
                </li>
                
                <li><b>HGVS</b>
                  <p>Generate <a href="http://varnomen.hgvs.org/" rel="external">HGVS</a>
                  identifiers for your input variants relative to the transcript
                  coding sequence (HGVSc) and the protein sequence (HGVSp).
                  Equivalent to <a href="/info/docs/tools/vep/script/vep_options.html#opt_hgvs">--hgvs</a>.</p>
                </li>
            </ul>
        </div>
    </div>
      
    <div class="column-two">
      <div class="column-right" style="padding-left:2em">
          <div class="plain-box" style="padding-bottom:1em">
              <img src="/img/vep_web_ident.png">
          </div>
      </div>
    </div>
</div>


<hr/>
<h2 id="freq">Variants and frequency data</h2>

<div class="column-wrapper">
    <div class="column-two">
        <div class="column-left">
            VEP can also search the Ensembl database for
            known variants that are co-located with variants from your input
            data.
            
            <ul>
                <li>
                    <p><b>Find co-located known variants</b> - report known
                    variants from the Ensembl Variation database that overlap
                    with your input. A list of variant sources imported can be
                    viewed <a
                    href="/info/genome/variation/species/species_data_types.html#sources">here</a>.
                    Note that this feature is only available for species with an
                    Ensembl Variation database. Equivalent to <a
                    href="/info/docs/tools/vep/script/vep_options.html#opt_check_existing">--check_existing</a>.
                    </p>
                    <p> VEP will by default compares the alleles of
                    your input variant to that of the existing variant;
                    VEP will only report the existing
                    variant ID if none of the alleles in your input variant are
                    novel. </p>
                    <p> For example, if your input variant has alleles A/G,
                    and the existing variant has alleles A/T, then the existing
                    variant will not be reported. If instead your input variant
                    has alleles A/T, then the existing variant will be reported.
                    </p>
                    <p> To disable this allele matching, select the option
                    "Yes but don't compare alleles" for the option
                    "Find co-located known variants".</p>
                    <p> For known variants VEP can also provide PubMed IDs
                    of publications citing the variant (equivalent to <a
                    href="/info/docs/tools/vep/script/vep_options.html#opt_uniprot">--pubmed</a>).</p>
                </li>
                <li><b>Variant synonyms</b>
                    <p>Report known synonyms for co-located variants.</p>
                </li>
                <li><b>Frequency data for co-located variants</b>
                    <p>VEP can also report allele frequency (AF)
                    data for existing variants from several major genotyping
                    projects, the <a href="http://www.1000genomes.org/"
                    rel="external">1000 Genomes Project</a>,
                    and <a href="http://gnomad.broadinstitute.org/" rel="external">gnomAD</a>;
                    this only applies when you have selected human as your species.
                        <ul>
                            <li><b>1000 Genomes global</b> - the combined phase
                            3 population (i.e. all individuals from all
                            populations). Equivalent to <a
                            href="/info/docs/tools/vep/script/vep_options.html#opt_af">--af</a>
                            </li>
                            <li><b>1000 Genomes continental</b> - the four
                            continent-level populations - AFR (African), AMR
                            (American), ASN (Asian) and EUR (European). Equivalent to <a
                            href="/info/docs/tools/vep/script/vep_options.html#opt_af_1kg">--af_1kg</a>
                            </li>
                            <li><b>gnomAD exomes</b> - combined, AFR, AMR, ASJ, EAS, FIN, NFE, OTH, SAS populations. Equivalent to <a
                            href="/info/docs/tools/vep/script/vep_options.html#opt_af_gnomade">--af_gnomade</a>
                            </li>
                            <li><b>gnomAD genomes</b> - combined, AFR, AMR, AMI, ASJ, EAS, FIN, MID, NFE, OTH, SAS populations. Equivalent to <a
                              href="/info/docs/tools/vep/script/vep_options.html#opt_af_gnomadg">--af_gnomadg</a>
                            </li>
                        </ul>
                    </p>
                </li>
                <li><b>PubMed IDs for citations of co-located variants</b>
                    <p>Report the PubMed IDs of any publications that cite the co-located variant(s).</p>
                </li>
                <li><b>Include flagged variants</b>
                    <p>Variants flagged as failed by the Ensembl Variation quality control.</p>
                </li>
            </ul>
      </div>
    </div>
      
    <div class="column-two">
      <div class="column-right" style="padding-left:2em">
          <div class="plain-box" style="padding-bottom:1em">
              <img src="/img/vep_web_freq.png">
          </div>
      </div>
    </div>
</div>

<hr/>
<h2 id="annotation">Additional annotations</h2>

<div class="column-wrapper">
    <div class="column-two">
        <div class="column-left">
            <ul>
                <li><b>Transcript biotype</b>
                  <p>Add the <a  href="/Help/Faq?id=468">transcript biotype</a> to the output.
                  Equivalent to <a href="/info/docs/tools/vep/script/vep_options.html#opt_biotype">--biotype</a> in
                  the VEP script.</p>
                </li>

                <li><b>Exon and intron numbers</b>
                  <p>Report the exon or intron number that a variant falls in as NUMBER / TOTAL, i.e. exon 2/5
                  means the variant falls in the 2nd of 5 exons in the transcript. Equivalent to <a
                  href="/info/docs/tools/vep/script/vep_options.html#opt_numbers">--numbers</a>.</p>
                </li>
                
                <li> <b>Transcript support level</b>
                  <p>Report the <a href="/Help/Glossary?id=492">transcript support level</a> of the overlapped
                  transcript. Equivalent to
                  <a href="/info/docs/tools/vep/script/vep_options.html#opt_tsl">--tsl</a>.</p></li>
                
                <li> <b>APPRIS</b>
                  <p>Report the <a href="/Help/Glossary?id=521">APPRIS</a> score of the overlapped
                  transcript. Equivalent to
                  <a href="/info/docs/tools/vep/script/vep_options.html#opt_appris">--appris</a>.</p>
                </li>
                
                <li><b>Identify canonical transcripts</b>
                  <p>Add a flag to the output indicating if the reported transcript is the <a
                  href="/Help/Glossary?id=346">canonical transcript</a> for the
                  gene. Equivalent to <a
                  href="/info/docs/tools/vep/script/vep_options.html#opt_canonical">--canonical</a>.</p>
                </li>
                
                <li><b>Upstream/Downstream distance (bp)</b>
                  <p>Change the distance to assign the upstream and downstream consequences.
                  Equivalent to  <a
                  href="/info/docs/tools/vep/script/vep_options.html#opt_distance">--distance</a>.</p>
                </li>

                <li><b>NMD</b>
                  <p>Predicts if a stop_gained variant allows the transcript to escape nonsense-mediated 
                  mRNA decay based on certain rules. This functionality is provided by the 
                  <a href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/NMD.pm">NMD</a> plugin.</p>
                </li>

                <li><b>miRNA structure</b>
                  <p>Determines where in the secondary structure of a miRNA a variant falls.
                  Equivalent to <a
                  href="/info/docs/tools/vep/script/vep_options.html#opt_mirna">--mirna</a>.</p>
		            </li>
                
                <hr />
                
                <li><b>Protein matches</b>
                  <p>Shows the variant location on PDBe and AlphaFold protein structures in interactive 3d displays, where available.
                  Report <a href="/Help/Glossary?id=18">protein domains</a> from <a href="https://www.ebi.ac.uk/pdbe/" rel="external">PDBe</a>, <a
                  href="https://www.ebi.ac.uk/interpro" rel="external">Pfam</a>, <a
                  href="http://prosite.expasy.org/" rel="external">Prosite</a> and
                  <a href="http://www.ebi.ac.uk/interpro/"
                  rel="external">InterPro</a> that overlap input variants.
                  Equivalent to <a
                  href="/info/docs/tools/vep/script/vep_options.html#opt_domains">--domains</a>.</p>
                </li>

                <li><b>IntAct</b>
                  <p>Reports relevant data for variants that falls within molecular interaction site as reprted by
                  <a href="https://www.ebi.ac.uk/intact/home">IntAct</a> database. This functionality is provided by the
                  <a href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/IntAct.pm">IntAct</a> plugin.</p>
                </li>
                
                <hr />
                
                <li><b>Get regulatory region consequences</b>
                  <p>In addition to predicting consequences with overlapping transcripts, VEP
                  can find overlaps with known regulatory regions as determined in
                  the <a href="/info/genome/funcgen/index.html">Ensembl Regulatory build</a>.<br />
                  Using this option, VEP will also report if a
                  variant falls in a transcription factor binding motif, and give
                  a score that reflects whether the altered motif sequence is more
                  or less similar to the consensus.<br />
                  Get regulatory consequences is equivalent to <a
                  href="/info/docs/tools/vep/script/vep_options.html#opt_regulatory">--regulatory</a>.</p>
                </li>

                <hr />

                <li><b>Phenotypes</b>
                  <p>
                    Report the phenotypic data overlapping the genomic features. This functionality is provided by the 
                    <a href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/Phenotypes.pm">Phenotypes</a> plugin.<br />
                    For more information on the imported phenotypic data for genes, variation and QTLs see 
                    <a href="/info/genome/variation/phenotype/sources_phenotype_documentation.html">our phenotype documentation</a>.<br />
                    <b>Note:</b> This web functionality is not reporting cancer phenotypic data this release. However the cancer phenotypic data is available in the command line version.
                  </p>
                </li>

                <li><b>Gene Ontology</b>
                  <p>
                    Add terms to describe any overlapping gene's function, the cellular component in which the function is performed and the biological processes to which this contributes.
                  </p>
                </li>

                <li><b>DisGeNET</b>
                  <p>
                    Report Variant-Disease-PMID associations from the <a href="https://www.disgenet.org/" rel="external">DisGeNET</a> database. This functionality is provided by the
                    <a href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/DisGeNET.pm">DisGeNET</a> plugin.<br />
		    <b>Note:</b> A <a href="https://www.disgenet.org/legal">license</a> is needed for commercial use.
                  </p>
                </li>

                <li><b>Mastermind</b>
                  <p>
                    Uses the <a href="https://www.genomenon.com/mastermind" rel="external">Mastermind Genomic Search Engine</a> to report variants that have clinical evidence cited in the medical literature. This functionality is provided by the
                    <a href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/Mastermind.pm">Mastermind</a> plugin.<br />
                    <b>Note:</b> This web functionality is only reporting the URL to the Mastermind Genomic Search Engine webpage.
                  </p>
                </li>
            </ul>
        </div>
    </div>
      
    <div class="column-two">
      <div class="column-right" style="padding-left:2em">
          <div class="plain-box" style="padding-bottom:1em">
              <img src="/img/vep_web_annotation.png">
          </div>
      </div>
    </div>
</div>


<hr/>
<h2 id="prediction">Predictions</h2>

<div class="column-wrapper">
    <div class="column-two">
        <div class="column-left">
            <ul>
                <li><b>SIFT predictions</b>
                  <p><a rel="external" href="http://sift.bii.a-star.edu.sg/">SIFT</a> predicts whether an amino
                  acid substitution affects protein function based on sequence
                  homology and the physical properties of amino acids. Only
                  available in popular species. For both SIFT and PolyPhen VEP
                  can report either a score between 0 and 1, a prediction in
                  words, or both. Equivalent to <a
                  href="/info/docs/tools/vep/script/vep_options.html#opt_sift">--sift</a>.</p>
                </li>
                
                <li><b>PolyPhen predictions</b>
                  <p><a rel="external" href="http://genetics.bwh.harvard.edu/pph2/">PolyPhen</a> is a
                  tool which predicts possible impact of an amino acid
                  substitution on the structure and function of a human protein
                  using straightforward physical and comparative considerations.
                  Equivalent to <a
                  href="/info/docs/tools/vep/script/vep_options.html#opt_polyphen">--polyphen</a>.</p>
                </li>
                
                <li><b>dbNSFP</b>
                  <p>Retrieves data for missense variants from <a rel="external" href="https://sites.google.com/site/jpopgen/dbNSFP">dbNSFP</a>.
                  Equivalent to the VEP plugin <a rel="external" 
                  href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/dbNSFP.pm">dbNSFP</a>.</p>
                </li>
                
                <li><b>LoFtool</b>
                  <p>Provides a rank of genic intolerance and consequent susceptibility to 
                  disease based on the ratio of Loss-of-function (LoF)to synonymous mutations.
                  Equivalent to the VEP plugin <a rel="external" 
                  href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/LoFtool.pm">LoFtool</a>.</p>
                </li>
                
                <li><b>EVE</b>
                  <p>Adds information from <a rel="external"
                  href="https://evemodel.org/">EVE</a> (evolutionary model of variant effect).
                  Equivalent to the VEP plugin <a rel="external"
                  href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/EVE.pm">EVE</a>.</p>
                </li>
                
                <hr />
                
                <li><b>dbscSNV</b>
                  <p>Retrieves data for splicing variants from <a rel="external" href="https://sites.google.com/site/jpopgen/dbNSFP">dbscSNV</a>.
                  Equivalent to the VEP plugin <a rel="external" 
                  href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/dbscSNV.pm">dbscSNV</a>.</p>
                </li>
                
                <li><b>MaxEntScan</b>
                  <p>Get splice site predictions from <a rel="external" href="http://genes.mit.edu/burgelab/maxent/Xmaxentscan_scoreseq.html">MaxEntScan</a>.
                  Equivalent to the VEP plugin <a rel="external" 
                  href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/MaxEntScan.pm">MaxEntScan</a>.</p>
                </li>

                <li><b>SpliceAI</b>
                  <p>
                    Pre-calculated annotations from <a href="https://github.com/Illumina/SpliceAI" rel="external">SpliceAI</a> a deep neural network,
                    developed by Illumina, Inc that predicts splice junctions from an arbitrary pre-mRNA transcript sequence. Used for non-commercial purposes.
                    This functionality is provided by the
                    <a href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/SpliceAI.pm">SpliceAI</a> plugin.<br />
                    The pre-calculated annotations for all possible single nucleotide substitutions can be retrieved from:
                    <ul>
                        <li>Ensembl/GENCODE v24 canonical transcripts
                          <p>Masked scores</p>
                        </li>
                        <li>Ensembl/GENCODE v37 MANE transcripts
                          <p>Raw scores</p>
                        </li>
                    </ul>
                  <b>Note:</b> The pre-calculated annotations for 1 base insertions, and 1-4 base deletions are only available for Ensembl/GENCODE v24 canonical transcripts.
                  </p>
                </li>

                <hr />
                
                <li><b>BLOSUM62</b>
                  <p>Looks up the BLOSUM 62 substitution matrix score for the reference and alternative 
                  amino acids predicted for a missense mutation.
                  Equivalent to the VEP plugin <a rel="external" 
                  href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/Blosum62.pm">Blosum62</a>.</p>
                </li>
                
                <li><b>Ancestral allele</b>
                  <p>Retrieves ancestral allele sequences from a FASTA file. Ensembl produces <a rel="external" href="[[SPECIESDEFS::ENSEMBL_FTP_URL]]/current_fasta/ancestral_alleles/">FASTA file dumps</a> of the ancestral sequences of key species.
                  Equivalent to the VEP plugin <a rel="external" 
                  href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/AncestralAllele.pm">AncestralAllele</a>.</p>
                </li>
            </ul>
        </div>
    </div>
      
    <div class="column-two">
      <div class="column-right" style="padding-left:2em">
          <div class="plain-box" style="padding-bottom:1em">
              <img src="/img/vep_web_prediction.png">
          </div>
      </div>
    </div>
</div>


<hr/>
<h2 id="filter">Filtering options</h2>

<div class="column-wrapper">
    <div class="column-two">
        <div class="column-left">
            VEP allows you to pre-filter your results e.g. by MAF or
            consequence type. Note that it is also possible to perform
            equivalent operations on the results page for VEP, so if you
            aren't sure, don't use any of these options! 
            <ul>
                <li><b>By frequency</b>
                  <p>Filter variants by minor allele frequency (MAF). Two options are provided:
                    <ul>
                        <li><b>Exclude common variants</b>
                          <p>Filter out variants that are co-located with an existing variant that has a
                          frequency greater than 0.01 (1%) in the 1000 Genomes
                          global population. Equivalent to <a
                          href="/info/docs/tools/vep/script/vep_options.html#opt_filter_common">--filter_common</a>
                          in the VEP script.</p>
                        </li>
                        <li><b>Advanced filtering</b>
                          <p>Enabling this option allows you to specify a population and frequency to
                          compare to, as well whether matching variants should be
                          included or excluded from the results.</p>
                        </li>
                    </ul>
                   </p>
                </li>
                
                <li><b>Return results for variants in coding regions only</b>
                  <p>Exclude variants that don't fall in a coding region of a
                  transcript. Equivalent to <a
                  href="/info/docs/tools/vep/script/vep_options.html#opt_coding_only">--coding_only</a>.</p>
                </li>
                
                <li><b>Restrict results</b>
                  <p>For many variants VEP will report multiple consequence types - typically this is because
                  the variant overlaps more than one transcript. For each of these
                  options VEP uses consequence ranks that are subjectively
                  determined by Ensembl. <a
                  href="/info/genome/variation/prediction/predicted_data.html#consequence_type_table">This
                  table</a> gives all of the consquence types predicted by
                  Ensembl, ordered by rank. Note that enabling one of these
                  options not only loses potentially relevant data, but in some
                  cases may be scientifically misleading. Options:
                    <ul>
                        <li><b>Show one selected consequence</b>
                        <p>Pick one consequence type across all those predicted for the
                          variant; the output will include transcript- or
                          feature-specific information. Consequences are chosen by
                          the canonical, biotype status and length of the
                          transcript, along with the ranking of the consequence
                          type according to <a
                          href="/info/genome/variation/prediction/predicted_data.html#consequence_type_table">this
                          table</a>. This is the best method to use if you are
                          interested only in one consequence per variant.
                          Equivalent to <a
                          href="/info/docs/tools/vep/script/vep_options.html#opt_per_gene">--pick</a>.</p>
                        </li>
                        <li><b>Show one selected consequence per gene</b>
                          <p>Pick one consequence type for each gene using the same
                          criteria as above. Note that if a variant overlaps more
                          than one gene, output for each gene will be reported.
                          Equivalent to <a
                          href="/info/docs/tools/vep/script/vep_options.html#opt_per_gene">--per_gene</a>.</p>
                        </li>
                        <li><b>Show only list of consequences per variant</b>
                          <p>Give a comma-separated list of all observed consequence
                          types for each variant. No transcript-specific or
                          gene-specific output will be given. Equivalent to <a
                          href="/info/docs/tools/vep/script/vep_options.html#opt_summary">--summary</a>.</p>
                        </li>
                        <li><b>Show most severe per variant</b>
                          <p>Only the most severe of all observed consequence types is reported for
                          each variant. No transcript-specific or gene-specific
                          output will be given. Equivalent to <a
                          href="/info/docs/tools/vep/script/vep_options.html#opt_most_severe">--most_severe</a>.</p>
                        </li>
                    </ul>
                </li>
            </ul>
        </div>
    </div>
      
    <div class="column-two">
      <div class="column-right" style="padding-left:2em">
          <div class="plain-box" style="padding-bottom:1em">
              <img src="/img/vep_web_filter.png">
          </div>
      </div>
    </div>
</div>


<hr/>
<h2 id="advanced">Advanced options</h2>

<div class="column-wrapper">
    <div class="column-two">
        <div class="column-left">
          <p>The VEP web interface allows you to use/setup advanced options:</p>
          <ul>
            <li><b>Buffer size</b>
              <p>By default VEP process the variants by blocks of 5000 (i.e. what we call "buffer size").<br />
In some cases, reducing the size of the blocks (buffer size) could prevent memory issues for large VEP queries (e.g. use of regulatory data, many plugins or custom annotations).<br />
This is why the maximum buffer size is automatically set to 500 on the VEP Web interface when the "Regulatory data" option is selected.</p></li>

<li><b>Right align variants prior to consequence calculation</b>
  <p>By default VEP performs consequence calculation at the given input coordinates.<br />
Optionally, VEP can shift insertions and deletions found within repeated regions as far as possible in the 3' direction, normalising output.
</ul>
        </div>
    </div>

    <div class="column-two">
      <div class="column-right" style="padding-left:2em">
          <div class="plain-box" style="padding-bottom:1em">
              <img src="/img/vep_web_advanced.png">
          </div>
      </div>
    </div>
</div>


<hr/>
<h2 id="jobs">Jobs</h2>

<div class="column-wrapper">
    <div class="column-two">
      <div class="column-left">
          <p>Once you have clicked &quot;Run&quot;, your input will be checked
          and submitted to the VEP as a job. All jobs associated with your
          session or account are shown in the &quot;Recent Tickets&quot; table.
          You may submit multiple jobs simultaneously.</p> <p>The
          &quot;Jobs&quot; column of the table shows the current status of the
          job.</p>

          <ul>
            <li><b>Queued</b> - your job is waiting to be submitted to the
            system</li>
            <li><b>Running</b> - your job is currently running</li>
            <li><b>Done</b> - your job is finished - click the [View results]
            link to be taken to the results page</li>
            <li><b>Failed</b> - there is a
            problem with your job - click the magnifying glass icon <img
            src="/i/16/search.png"> to see more details</li>
          </ul>

          <p>You may delete a job by clicking the trash can icon <img
          src="/i/16/trash.png">. If you are logged in to Ensembl, you can save
          the job by clicking the save icon <img src="/i/16/save.png">.</p>
          
          <p> You may also resubmit a job (for example, to re-run with the same
          data but change some parameters) by clicking the edit icon <img
          src="/i/16/pencil.png">.</p>

          <p> You can see a summary of the options that you selected for your
          VEP job by clicking on the magnifying glass icon <img
          src="/i/16/search.png">.</p>
      </div>
    </div>
      
    <div class="column-two">
      <div class="column-right" style="padding-left:2em;">
          <!--<div class="plain-box">-->
              <img src="/img/vep_web_jobs.png">
          <!--</div>-->
      </div>
    </div>
</div>

</body>
</html>
