<html>
<head>
<title>Input form</title>
<meta name="order" content="1" />
</head>

<body>

<style>
._afterimage._ht.job-species {
    width: 16px;
    height: 16px;
}
</style>

  
  <div style="float:right"><img src="/img/vep_logo.png"/></div>

  <h1 id="top"><span style="color:#006;padding-right:15px">Variant Effect Predictor</span><span style="color:#666"><img src="/i/16/pencil.png"/> Input form</span></h1>
  <hr/>

<p> When you reach the VEP web interface, you will be presented with a form to
enter your data and alter various options.</p>

<p> Note that the listed options change depending on the selected species. </p>

<h2 id="input">Data input</h2>

<div class="column-wrapper">
    <div class="column-two">
      <div class="column-left">
          <ol>
              <li> First select the correct species for your data. Ensembl hosts
              many vertebrate genomes; genomes for plants, protists and fungi
              can be found at <a href="http://www.ensemblgenomes.org/">Ensembl
              Genomes</a>. </li>
              
              <li> You can optionally choose a name for the data you upload -
              this can make it easier for you to identify jobs and files that
              you have uploaded to the VEP at a later point. </li>
              
              <li> <p>You have three options for uploading your data:
                  
                  <ul>
                      <li><b>File upload</b> - click the "Choose file" button
                      and locate the file on your system. Please ensure your files
			                are sorted by location if you are using VCF or other input 
			                formats using location. This greatly improves speed of analysis</li>
                      <li><b>Paste file</b> - simply copy and paste the contents
                      of your file into the large text box</li>
                      <li><b>File URL</b> - point the VEP to a file hosted on a
                      publically accessible address. This can be either a
                      <b>http://</b> or <b>ftp://</b> address.</li>
                  </ul>
                  
                  Once you have uploaded some data, you can select it as the
                  input for future jobs by choosing the data from the drop down
                  menu.</p>
                  
                  <p> The format of your data is automatically detected; see the
                  examples or the <a href="../vep_formats.html#input">input
                  format</a> documentation. </p>
              </li>
              
              <li> For pasted data you can get an instant preview of the results of your
              first variant by clicking the button that appears when you paste
              your data. This quickly shows you the consequence type, the IDs of
              any overlapping variants, genes, transcripts and regulatory
              features, as well as SIFT and PolyPhen predictions. To see the full
              results set submit your job as normal. </li>
              
              <li>
                  <p> For some species you can select which transcript database
                  to use. The default is to use Ensembl transcripts, which offer
                  the most rich annotation through VEP. </p>
                  <p> GENCODE Basic is a subset of the GENCODE gene set, and is
                  intended to provide a simplified, high-quality subset of the
                  GENCODE transcript annotations that will be useful to the
                  majority of users. GENCODE Basic includes all genes in the
                  GENCODE gene set, with a representative subset of the
                  transcripts (splice variants). </p>
                  <p> GENCODE Primary is a new transcript subset which covers
                  all human exons in a minimal set of transcripts. 
                  This aims to enable annotation of all potential variant 
                  consequences without duplication across multiple transcripts.
                  </p>
                  <p>You can also select to use RefSeq transcripts from the <a
                  href="/info/docs/api/core/index.html#est">otherfeatures
                  database</a>; note though that these transcripts are simply
                  aligned to the reference genome and the database is missing
                  much of the annotation found when using the main Ensembl
                  database (e.g. protein domains, CCDS identifiers).</p>
              </li>
          </ol>
      </div>
    </div>
      
    <div class="column-two">
      <div class="column-right" style="padding-left:2em">
        <form style="scale: 0.6; transform-origin: top left; pointer-events: none; width: fit-content" class="plain-box vep-form _tool_form bgcolour check _check">
          <fieldset>
            <div class="form-field _adjustable_height" style="min-height: 0px;"><label class="ff-label">Species:</label><div class="taxon_selector_form js_panel _sdd ff-right">
              <div class="list-wrapper">
                <ul class="list"><li><img src="/i/species/Homo_sapiens.png" class="nosprite"><span class="ss-selected">Homo_sapiens</span><span class="ss-selection-delete">x</span></li></ul>
                <div class="_vep_assembly italic">Assembly: <span>GRCh38.p13</span></div>
                <div class="links"><a href="#">Change species</a><div class="italic assembly_msg">If you are looking for VEP for Human GRCh37, please go to <a href="#" rel="external">GRCh37 website</a>.</div></div>
              </div>
              <div class="checkboxes"><input type="checkbox">Homo_sapiens<br></div>
            </div></div>

            <div class="form-field"><label class="ff-label">Name for this job (optional):</label><div class="ff-right"><input type="text" class="_string optional ftext"></div></div>

            <div class="form-field ff-multi"><label class="ff-label">Input data:</label>
              <div class="ff-right"><div><b>Either paste data:</b></div></div><div class="ff-right"><textarea cols="40" class="_text vep-input ftextarea optional focused" rows="10" style="background-position: 0px 29px;">rs699&#13;&#10;rs171&#13;&#10;rs665</textarea><input value="Run instant VEP for current line â€º" type="button" class="quick-vep-button fbutton hidden" style="display: inline-block; margin-top: 30px;"></div>

              <div class="ff-right"><div><span class="small"><b>Examples:&nbsp;</b><a href="#" class="_example_input" rel="ensembl">Ensembl default</a>, <a href="#" class="_example_input" rel="vcf">VCF</a>, <a href="#" class="_example_input" rel="id">Variant identifiers</a>, <a href="#" class="_example_input" rel="hgvs">HGVS notations</a>, <a href="#" class="_example_input" rel="spdi">SPDI</a></span></div></div>

              <div class="ff-right vep_left_input"><span class="_ht ht"><b>Or upload file:</b></span></div><div class="ff-right"><input class="ffile" type="file"></div>

              <div class="ff-right vep_left_label vep_left_input"><div><b>Or provide file URL:</b></div></div><div class="ff-right"><input type="text" size="30" class="ftext optional url _url"></div>
            </div>
            <div class="form-field _stt_rfq" style="display: block;"><label class="ff-label"><span class="_ht ht">Transcript database to use</span>:</label><div class="ff-right"><p class="ff-checklist"><input type="radio" class="_stt" checked="checked"><label class="ff-checklist-label">Ensembl<span style="display: inline;">/GENCODE</span> transcripts</label></p><p class="ff-checklist" style="display: block;"><input type="radio" class="_stt"><label class="ff-checklist-label">Ensembl/GENCODE basic transcripts</label></p><p class="ff-checklist" style="display: block;"><input type="radio" class="_stt"><label class="ff-checklist-label">Ensembl/GENCODE primary transcripts</label><p class="ff-checklist"><input class="_stt" type="radio"><label class="ff-checklist-label">RefSeq transcripts</label></p><p class="ff-checklist"><input class="_stt" type="radio"><label class="ff-checklist-label">Ensembl<span style="display: inline;">/GENCODE</span> and RefSeq transcripts</label></p></div></div></fieldset></form>
      </div>
    </div>
</div>


<hr/>
<h2 id="ident">Identifiers</h2>

<div class="column-wrapper">
    <div class="column-two">
        <div class="column-left">
            VEP can provide additional identifiers for genes, transcripts,
            proteins and variants.
            
            <ul>
                <li><b>Gene symbol</b>
                  <p>Add the gene symbol for the gene to
                  the output. This will typically be, for example, the <a
                  href="http://www.genenames.org/" rel="external">HGNC</a>
                  identifier for genes in human. Equivalent to <a
                  href="/info/docs/tools/vep/script/vep_options.html#opt_symbol">--symbol</a>
                  in the VEP script.</p>
                </li>

                 <li><b>Transcript version</b>
                   <p>Add the transcript version to the transcript identifier. Equivalent to <a 
                   href="/info/docs/tools/vep/script/vep_options.html#opt_transcript_version">--transcript_version</a>.</p> 
 
                <li><b>CCDS</b>
                  <p>Add the <a href="/info/genome/genebuild/ccds.html">Consensus CDS</a>
                  transcript identifier where available. Equivalent to <a
                  href="/info/docs/tools/vep/script/vep_options.html#opt_ccds">--ccds</a>.</p>
                </li>
                
                <li><b>Protein</b>
                  <p>Add the Ensembl protein identifer (ENSP). Equivalent to <a
                  href="/info/docs/tools/vep/script/vep_options.html#opt_protein">--protein</a>.</p>
                </li>
                
                <li><b>UniProt</b>
                  <p>Add identifiers for translated protein
                  products from three <a href="http://www.ebi.ac.uk/uniprot"
                  rel="external">UniProt</a>-related databases (SWISSPROT, TREMBL
                  and UniParc). Equivalent to <a
                  href="/info/docs/tools/vep/script/vep_options.html#opt_uniprot">--uniprot</a>.</p>
                </li>
                
                <li><b>HGVS</b>
                  <p>Generate <a href="http://varnomen.hgvs.org/" rel="external">HGVS</a>
                  identifiers for your input variants relative to the transcript
                  coding sequence (HGVSc) and the protein sequence (HGVSp).
                  Equivalent to <a href="/info/docs/tools/vep/script/vep_options.html#opt_hgvs">--hgvs</a>.</p>
                </li>
            </ul>
        </div>
    </div>
      
    <div class="column-two">
      <div class="column-right" style="padding-left:2em">
        <form style="scale: 0.8; transform-origin: top left; pointer-events: none; width: 150%" class="_tool_form _check check vep-form bgcolour">
          <fieldset class="extra-options-fieldset" style="margin: 0; padding: 8px"><div class="extra_configs_wrapper" style="margin: 0"><div class="extra_configs_button">
            <a href="#Identifiers" class="toggle _slide_toggle set_cookie open" rel="__togg_Identifiers">Identifiers</a>
            <span class="extra_configs_info">Additional identifiers for genes, transcripts and variants</span></div>

            <div class="toggleable hidden extra_configs __togg_Identifiers" style="display: block;"><fieldset><h2>Identifiers</h2>
              <div class="form-field"><label class="ff-label"><span class="ht _ht">Gene symbol</span>:</label><div class="ff-right"><p class="ff-checklist"><input type="checkbox" checked="checked"></p></div></div>
              <div class="form-field"><label class="ff-label"><span class="_ht ht">Transcript version</span>:</label><div class="ff-right"><p class="ff-checklist"><input checked="checked" type="checkbox"></p></div></div>
              <div class="form-field" style="display: block;"><label class="ff-label"><span class="_ht ht">CCDS</span>:</label><div class="ff-right"><p class="ff-checklist"><input type="checkbox"></p></div></div>
              <div class="form-field"><label class="ff-label"><span class="ht _ht">Protein</span>:</label><div class="ff-right"><p class="ff-checklist"><input type="checkbox"></p></div></div>
              <div class="form-field"><label class="ff-label"><span class="ht _ht">UniProt</span>:</label><div class="ff-right"><p class="ff-checklist"><input type="checkbox"></p></div></div>
              <div class="form-field"><label class="ff-label"><span class="_ht ht">HGVS</span>:</label><div class="ff-right"><p class="ff-checklist"><input type="checkbox"></p></div></div>
            </fieldset></div>
          </div></fieldset>
        </form>
      </div>
    </div>
</div>


<hr/>
<h2 id="freq">Variants and frequency data</h2>

<div class="column-wrapper">
    <div class="column-two">
        <div class="column-left">
            VEP can also search the Ensembl database for
            known variants that are co-located with variants from your input
            data.
            
            <ul>
                <li>
                    <p><b>Find co-located known variants</b> - report known
                    variants from the Ensembl Variation database that overlap
                    with your input. A list of variant sources imported can be
                    viewed <a
                    href="/info/genome/variation/species/species_data_types.html#sources">here</a>.
                    Note that this feature is only available for species with an
                    Ensembl Variation database. Equivalent to <a
                    href="/info/docs/tools/vep/script/vep_options.html#opt_check_existing">--check_existing</a>.
                    </p>
                    <p> VEP will by default compares the alleles of
                    your input variant to that of the existing variant;
                    VEP will only report the existing
                    variant ID if none of the alleles in your input variant are
                    novel. </p>
                    <p> For example, if your input variant has alleles A/G,
                    and the existing variant has alleles A/T, then the existing
                    variant will not be reported. If instead your input variant
                    has alleles A/T, then the existing variant will be reported.
                    </p>
                    <p> To disable this allele matching, select the option
                    "Yes but don't compare alleles" for the option
                    "Find co-located known variants".</p>
                    <p> For known variants VEP can also provide PubMed IDs
                    of publications citing the variant (equivalent to <a
                    href="/info/docs/tools/vep/script/vep_options.html#opt_uniprot">--pubmed</a>).</p>
                </li>
                <li><b>Variant synonyms</b>
                    <p>Report known synonyms for co-located variants.</p>
                </li>
                <li><b>Frequency data for co-located variants</b>
                    <p>VEP can also report allele frequency (AF)
                    data for existing variants from several major genotyping
                    projects, the <a href="http://www.1000genomes.org/"
                    rel="external">1000 Genomes Project</a>,
                    and <a href="http://gnomad.broadinstitute.org/" rel="external">gnomAD</a>;
                    this only applies when you have selected human as your species.
                        <ul>
                            <li><b>1000 Genomes global</b> - the combined phase
                            3 population (i.e. all individuals from all
                            populations). Equivalent to <a
                            href="/info/docs/tools/vep/script/vep_options.html#opt_af">--af</a>
                            </li>
                            <li><b>1000 Genomes continental</b> - the four
                            continent-level populations - AFR (African), AMR
                            (American), ASN (Asian) and EUR (European). Equivalent to <a
                            href="/info/docs/tools/vep/script/vep_options.html#opt_af_1kg">--af_1kg</a>
                            </li>
                            <li><b>gnomAD exomes</b> - combined, AFR, AMR, ASJ, EAS, FIN, NFE, OTH, SAS populations. Equivalent to <a
                            href="/info/docs/tools/vep/script/vep_options.html#opt_af_gnomade">--af_gnomade</a>
                            </li>
                            <li><b>gnomAD genomes</b> - combined, AFR, AMR, AMI, ASJ, EAS, FIN, MID, NFE, OTH, SAS populations. Equivalent to <a
                              href="/info/docs/tools/vep/script/vep_options.html#opt_af_gnomadg">--af_gnomadg</a>
                            </li>
                        </ul>
                    </p>
                </li>
                <li><b>PubMed IDs for citations of co-located variants</b>
                    <p>Report the PubMed IDs of any publications that cite the co-located variant(s).</p>
                </li>
                <li><b>Include flagged variants</b>
                    <p>Variants flagged as failed by the Ensembl Variation quality control.</p>
                </li>

                <li><b>Paralogue variants</b>
                  <p>
                    Retrieves variants that overlap genomic coordinates corresponding to aligned amino acid positions in paralogous proteins. This functionality is provided by the
                    <a href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/Paralogues.pm">Paralogues</a> plugin.<br />
                  </p>
                </li>

                <li><b>Open Targets Genetics</b>
                  <p>
                    Returns locus-to-gene (L2G) scores to predict causal genes at GWAS loci from
                    <a href="https://genetics.opentargets.org" rel="external">Open Targets Genetics</a>. This functionality is provided by the
                    <a href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/OpenTargets.pm">OpenTargets</a> plugin.<br />
                  </p>
                </li>

            </ul>
      </div>
    </div>

    <div class="column-two">
      <div class="column-right" style="padding-left:2em">
        <form style="scale: 0.8; transform-origin: top left; pointer-events: none; width: 150%" class="_tool_form _check check vep-form bgcolour"><fieldset class="extra-options-fieldset" style="margin: 0; padding: 8px">
          <div class="_stt_var extra_configs_wrapper" style="margin:0; display: block;"><div class="extra_configs_button"><a class="set_cookie toggle _slide_toggle open" rel="__togg_Variants_and_frequency_data" href="#Variants_and_frequency_data">Variants and frequency data</a><span class="extra_configs_info">Co-located variants and frequency data</span></div><div class="__togg_Variants_and_frequency_data toggleable hidden extra_configs" style="display: block;">

          <fieldset>
            <h2>Variants and frequency data</h2>
            <div class="form-field _stt_var" style="display: block;"><label class="ff-label"><span class="ht _ht">Find co-located known variants</span>:</label><div class="ff-right"><select class="_stt fselect"><option selected="selected">Yes</option></select></div></div>
            <div class="_stt_yes form-field" style="display: block;"><label class="ff-label"><span class="ht _ht">Variant synonyms</span>:</label><div class="ff-right"><p class="ff-checklist"><input type="checkbox"></p></div></div><div>
            <div class="_stt_allele _stt_yes form-field"><label class="ff-label">Frequency data for co-located variants:</label><div class="ff-right"><p class="ff-checklist"><input type="checkbox" checked="checked"><label class="ff-checklist-label _ht ht">1000 Genomes global minor allele frequency</label></p><p class="ff-checklist"><input type="checkbox"><label class="ff-checklist-label ht _ht">1000 Genomes continental allele frequencies</label></p><p class="ff-checklist"><input type="checkbox"><label class="_ht ht ff-checklist-label">gnomAD (exomes) allele frequencies</label></p><p class="ff-checklist"><input type="checkbox"><label class="_ht ht ff-checklist-label">gnomAD (genomes) allele frequencies</label></p></div></div>
            <div class="form-field _stt_yes _stt_allele"><label class="ff-label"><span class="ht _ht">PubMed IDs for citations of co-located variants</span>:</label><div class="ff-right"><p class="ff-checklist"><input  type="checkbox" checked="checked"></p></div></div></div>
            <div class="form-field"><label class="ff-label"><span class="ht _ht">Include flagged variants</span>:</label><div class="ff-right"><p class="ff-checklist"><input type="checkbox"></p></div></div>
            <div class="form-field"><label class="ff-label"><span class="_ht ht">Paralogue variants</span>:</label><div class="ff-right"><p class="ff-checklist"><input type="radio" checked="checked" value="no"><label class="ff-checklist-label">Disabled</label></p><p class="ff-checklist"><input class="_stt plugin_enable" value="plugin_Paralogues" type="radio" name="plugin_Paralogues"><label class="ff-checklist-label">Enabled</label></p></div></div>
            <div class="form-field"><label class="ff-label"><span class="_ht ht">Open Targets Genetics</span>:</label><div class="ff-right"><p class="ff-checklist"><input class="plugin_enable _stt" type="checkbox"></p></div></div>
          </fieldset></div></div>
        </fieldset></form>
      </div>
    </div>
</div>

<hr/>
<h2 id="annotation">Additional annotations</h2>

<div class="column-wrapper">
    <div class="column-two">
        <div class="column-left">
            <ul>

                <li><b>DosageSensitivity</b>
                  <p>Retrieves haploinsufficiency and triplosensitivity probability scores
                  for affected genes from a dosage sensitivity catalogue published in paper -
                  <a href="https://www.sciencedirect.com/science/article/pii/S0092867422007887">https://www.sciencedirect.com/science/article/pii/S0092867422007887</a>.</p>
                </li>

                <li><b>LOEUF</b>
                  <p>LOEUF stands for the 'loss-of-function observed/expected upper bound fraction'. 
                  This plugin adds constraint scores derived from gnomAD to VEP.
                  Equivalent to the VEP plugin <a rel="external"
                  href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/LOEUF.pm">LOEUF</a>.</p>
                </li>

                <hr />

                <li><b>Transcript biotype</b>
                  <p>Add the <a  href="/Help/Faq?id=468">transcript biotype</a> to the output.
                  Equivalent to <a href="/info/docs/tools/vep/script/vep_options.html#opt_biotype">--biotype</a> in
                  the VEP script.</p>
                </li>

                <li><b>Exon and intron numbers</b>
                  <p>Report the exon or intron number that a variant falls in as NUMBER / TOTAL, i.e. exon 2/5
                  means the variant falls in the 2nd of 5 exons in the transcript. Equivalent to <a
                  href="/info/docs/tools/vep/script/vep_options.html#opt_numbers">--numbers</a>.</p>
                </li>
                
                <li> <b>Transcript support level</b>
                  <p>Report the <a href="/Help/Glossary?id=492">transcript support level</a> of the overlapped
                  transcript. Equivalent to
                  <a href="/info/docs/tools/vep/script/vep_options.html#opt_tsl">--tsl</a>.</p></li>
                
                <li> <b>APPRIS</b>
                  <p>Report the <a href="/Help/Glossary?id=521">APPRIS</a> score of the overlapped
                  transcript. Equivalent to
                  <a href="/info/docs/tools/vep/script/vep_options.html#opt_appris">--appris</a>.</p>
                </li>
                
                <li><b>Identify canonical transcripts</b>
                  <p>Add a flag to the output indicating if the reported transcript is the <a
                  href="/Help/Glossary?id=346">canonical transcript</a> for the
                  gene. Equivalent to <a
                  href="/info/docs/tools/vep/script/vep_options.html#opt_canonical">--canonical</a>.</p>
                </li>
                
                <li><b>Upstream/Downstream distance (bp)</b>
                  <p>Change the distance to assign the upstream and downstream consequences.
                  Equivalent to  <a
                  href="/info/docs/tools/vep/script/vep_options.html#opt_distance">--distance</a>.</p>
                </li>

                <li><b>miRNA structure</b>
                  <p>Determines where in the secondary structure of a miRNA a variant falls (only for Ensembl/GENCODE transcripts).
                  Equivalent to <a
                  href="/info/docs/tools/vep/script/vep_options.html#opt_mirna">--mirna</a>.</p>
		            </li>

                <li><b>NMD</b>
                  <p>Predicts if a stop_gained variant allows the transcript to escape nonsense-mediated 
                  mRNA decay based on certain rules. This functionality is provided by the 
                  <a href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/NMD.pm">NMD</a> plugin.</p>
                </li>

                <li><b>UTRAnnotator</b>
                  <p>Annotates the effect of 5' UTR variant especially for
                  variant creating/disrupting upstream ORFs. This functionality
                  is provided by the 
                  <a href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/UTRAnnotator.pm">UTRAnnotator</a> plugin.</p>
                </li>

                <li><b>RiboseqORFs</b>
                  <p>Annotates the consequences of variants overlapping Ribo-seq ORFs.
                  This functionality is provided by the
                  <a href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/RiboseqORFs.pm">RiboseqORFs</a> plugin.</p>
                </li>

                <hr />
                
                <li><b>Protein matches</b>
                  <p>Shows the variant location on PDBe and AlphaFold protein structures in interactive 3d displays, where available.
                  Report <a href="/Help/Glossary?id=18">protein domains</a> from <a href="https://www.ebi.ac.uk/pdbe/" rel="external">PDBe</a>, <a
                  href="https://www.ebi.ac.uk/interpro" rel="external">Pfam</a>, <a
                  href="http://prosite.expasy.org/" rel="external">Prosite</a> and
                  <a href="http://www.ebi.ac.uk/interpro/"
                  rel="external">InterPro</a> that overlap input variants.
                  Equivalent to <a
                  href="/info/docs/tools/vep/script/vep_options.html#opt_domains">--domains</a>.</p>
                </li>

                <li><b>mutfunc</b>
                  <p>
                    <a rel="external" href="http://www.mutfunc.com">mutfunc</a>
                    predicts destabilization effect of protein structure,
                    interaction, regulatory region, etc. caused by a variant.
                    This functionality is provided by the
                    <a href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/mutfunc.pm">mutfunc</a> plugin.</p>
                  </p>
                </li>

                <hr />

                <li><b>IntAct</b>
                  <p>Reports relevant data for variants that falls within molecular interaction site as reprted by
                  <a href="https://www.ebi.ac.uk/intact/home">IntAct</a> database. This functionality is provided by the
                  <a href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/IntAct.pm">IntAct</a> plugin.</p>
                </li>

                <li><b>MaveDB</b>
                  <p><a rel="external" href="https://mavedb.org">MaveDB</a>
                  holds experimentally determined measures of variant effect.
                  This functionality is provided by the
                  <a href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/MaveDB.pm">MaveDB</a> plugin.</p>
                </li>

                <hr />
                
                <li><b>Get regulatory region consequences</b>
                  <p>In addition to predicting consequences with overlapping transcripts, VEP
                  can find overlaps with known regulatory regions as determined in
                  the <a href="/info/genome/funcgen/index.html">Ensembl Regulatory build</a>.<br />
                  Using this option, VEP will also report if a
                  variant falls in a transcription factor binding motif, and give
                  a score that reflects whether the altered motif sequence is more
                  or less similar to the consensus.<br />
                  Get regulatory consequences is equivalent to <a
                  href="/info/docs/tools/vep/script/vep_options.html#opt_regulatory">--regulatory</a>.</p>
                </li>

                <hr />

                <li><b>Enformer</b>
                  <p>Predictions of variant impact on gene expression.
                  This functionality is provided by the  <a rel="external" 
                  href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/Enformer.pm">Enformer</a> plugin.</p>
                </li>

                <hr />

                <li><b>Phenotypes</b>
                  <p>
                    Report the phenotypic data overlapping the genomic features. This functionality is provided by the 
                    <a href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/Phenotypes.pm">Phenotypes</a> plugin.<br />
                    For more information on the imported phenotypic data for genes, variation and QTLs see 
                    <a href="/info/genome/variation/phenotype/sources_phenotype_documentation.html">our phenotype documentation</a>.<br />
                    <b>Note:</b> This web functionality is not reporting cancer phenotypic data this release. However the cancer phenotypic data is available in the command line version.
                  </p>
                </li>

                <li><b>Gene Ontology</b>
                  <p>
                    Add terms to describe any overlapping gene's function, the cellular component in which the function is performed and the biological processes to which this contributes.
                  </p>
                </li>

                <li><b>Geno2MP</b>
                  <p>
                    <a rel="external" href="https://geno2mp.gs.washington.edu">Geno2MP</a>
                    is a web-accessible database of rare variant genotypes
                    linked to individual-level phenotypic profiles defined by
                    human phenotype ontology (HPO) terms.
                    This functionality is provided by the
                    <a href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/Geno2MP.pm">Geno2MP</a> plugin.</p>
                  </p>
                </li>

                <li><b>DisGeNET</b>
                  <p>
                    Report Variant-Disease-PMID associations from the <a href="https://www.disgenet.org/" rel="external">DisGeNET</a> database. This functionality is provided by the
                    <a href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/DisGeNET.pm">DisGeNET</a> plugin.<br />
		    <b>Note:</b> A <a href="https://www.disgenet.org/legal">license</a> is needed for commercial use.
                  </p>
                </li>

                <li><b>Mastermind</b>
                  <p>
                    Uses the <a href="https://www.genomenon.com/mastermind" rel="external">Mastermind Genomic Search Engine</a> to report variants that have clinical evidence cited in the medical literature. This functionality is provided by the
                    <a href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/Mastermind.pm">Mastermind</a> plugin.<br />
                    <b>Note:</b> This web functionality is only reporting the URL to the Mastermind Genomic Search Engine webpage.
                  </p>
                </li>

            </ul>
        </div>
    </div>
      
    <div class="column-two">
      <div class="column-right" style="padding-left:2em">
        <form style="scale: 0.8; transform-origin: top left; pointer-events: none; width: 150%" class="_tool_form _check check vep-form bgcolour"><fieldset class="extra-options-fieldset" style="margin: 0; padding: 8px">
          <div class="extra_configs_wrapper" style="margin: 0"><div class="extra_configs_button"><a class="toggle _slide_toggle set_cookie open" rel="__togg_Additional_annotations" href="#Additional_annotations">Additional annotations</a><span class="extra_configs_info">Addtional transcript, protein and regulatory annotations</span></div><div class="hidden toggleable extra_configs __togg_Additional_annotations" style="display: block;">
            <fieldset>
              <h2>Gene tolerance to change</h2>
              <div class="form-field"><label class="ff-label"><span class="ht _ht">DosageSensitivity</span>:</label><div class="ff-right"><p class="ff-checklist"><input type="checkbox"></p></div></div>
              <div class="form-field"><label class="ff-label"><span class="ht _ht">LOEUF</span>:</label><div class="ff-right"><p class="ff-checklist"><input type="checkbox"></p></div></div>
            </fieldset>

            <fieldset>
              <h2>Transcript annotation</h2>
              <div class="form-field"><label class="ff-label"><span class="ht _ht">Transcript biotype</span>:</label><div class="ff-right"><p class="ff-checklist"><input checked="checked" type="checkbox"></p></div></div>
              <div class="form-field"><label class="ff-label"><span class="ht _ht">Exon and intron numbers</span>:</label><div class="ff-right"><p class="ff-checklist"><input type="checkbox"></p></div></div>
              <div class="form-field" style="display: block;"><label class="ff-label">Transcript support level:</label><div class="ff-right"><p class="ff-checklist"><input type="checkbox" checked="checked"></p></div></div>
              <div class="form-field" style="display: block;"><label class="ff-label"><span class="_ht ht">APPRIS</span>:</label><div class="ff-right"><p class="ff-checklist"><input type="checkbox" checked="checked"></p></div></div>
              <div class="form-field" style="display: block;"><label class="ff-label"><span class="_ht ht">MANE</span>:</label><div class="ff-right"><p class="ff-checklist"><input type="checkbox" checked="checked"></p></div></div>
              <div class="_stt_merged form-field _stt_core _stt_gencode_basic _stt_gencode_primary " style="display: block;"><label class="ff-label">Identify canonical transcripts:</label><div class="ff-right"><p class="ff-checklist"><input type="checkbox"></p></div></div>
              <div class="form-field"><label class="ff-label"><span class="ht _ht">Upstream/Downstream distance (bp)</span>:</label><div class="ff-right"><input value="5000" class="optional _string ftext" type="text"></div></div>
              <div class="form-field"><label class="ff-label"><span class="_ht ht">miRNA structure</span>:</label><div class="ff-right"><p class="ff-checklist"><input type="checkbox" ></p></div></div>
              <div class="form-field"><label class="ff-label"><span class="ht _ht">NMD</span>:</label><div class="ff-right"><p class="ff-checklist"><input type="checkbox" class="_stt plugin_enable"></p></div></div>
              <div class="form-field"><label class="ff-label"><span class="ht _ht">UTRAnnotator</span>:</label><div class="ff-right"><p class="ff-checklist"><input type="checkbox" class="plugin_enable _stt"></p></div></div>
              <div class="form-field"><label class="ff-label"><span class="ht _ht">RiboseqORFs</span>:</label><div class="ff-right"><p class="ff-checklist"><input type="checkbox" class="plugin_enable _stt"></p></div></div>
            </fieldset>

            <fieldset>
              <h2>Protein annotation</h2>
              <div class="_stt_merged form-field _stt_core _stt_gencode_basic _stt_gencode_primary" style="display: block;"><label class="ff-label"><span class="ht _ht">Protein matches</span>:</label><div class="ff-right"><p class="ff-checklist"><input type="checkbox"></p></div></div>
              <div class="_stt_Mus_musculus form-field _stt_Homo_sapiens" style="display: block;"><label class="ff-label"><span class="_ht ht">mutfunc</span>:</label><div class="ff-right"><p class="ff-checklist"><input type="radio" checked="checked" class="_stt plugin_enable"><label class="ff-checklist-label">Disabled</label></p><p class="ff-checklist"><input type="radio" class="plugin_enable _stt"><label class="ff-checklist-label">Enabled</label></p></div></div>
            </fieldset>

            <fieldset>
              <h2>Functional effect</h2>
              <div class="_stt_Mus_musculus form-field _stt_Homo_sapiens" style="display: block;"><label class="ff-label"><span class="_ht ht">IntAct</span>:</label><div class="ff-right"><p class="ff-checklist"><input type="radio" checked="checked" class="_stt plugin_enable"><label class="ff-checklist-label">Disabled</label></p><p class="ff-checklist"><input type="radio" class="plugin_enable _stt"><label class="ff-checklist-label">Enabled</label></p></div></div>
              <div class="_stt_merged form-field _stt_core _stt_gencode_basic _stt_gencode_primary" style="display: block;"><label class="ff-label"><span class="ht _ht">MaveDB</span>:</label><div class="ff-right"><p class="ff-checklist"><input type="checkbox"></p></div></div>
            </fieldset>

            <fieldset>
              <h2>Regulatory data</h2>
              <div class="form-field _stt_var" style="display: block;"><label class="ff-label"><span class="ht _ht">Get regulatory region consequences</span>:</label><div class="ff-right"><select class="_stt fselect"><option selected="selected">Yes</option></select></div></div>
            </fieldset>

            <fieldset class="_stt_Homo_sapiens">
              <h2>Regulatory impact</h2>
              <div class="_stt_Homo_sapiens form-field"><label class="ff-label"><span class="ht _ht">Enformer</span>:</label><div class="ff-right"><p class="ff-checklist"><input class="plugin_enable _stt" type="checkbox"></p></div></div>
            </fieldset>
                                    
            <fieldset style="display: block;">
              <h2>Phenotype data and citations</h2>
              <div class="form-field"><label class="ff-label"><span class="_ht ht">Phenotypes</span>:</label><div class="ff-right"><p class="ff-checklist"><input class="plugin_enable _stt" type="checkbox"></p></div></div>
              <div class="form-field"><label class="ff-label"><span class="_ht ht">Gene Ontology</span>:</label><div class="ff-right"><p class="ff-checklist"><input class="_stt plugin_enable" type="checkbox"></p></div></div>
              <div class="form-field"><label class="ff-label"><span class="_ht ht">DisGeNET</span>:</label><div class="ff-right"><p class="ff-checklist"><input type="checkbox" class="plugin_enable _stt"></p></div></div>
              <div class="form-field"><label class="ff-label"><span class="ht _ht">Mastermind</span>:</label><div class="ff-right"><p class="ff-checklist"><input type="checkbox" class="plugin_enable _stt"></p></div></div></fieldset>
            </div>
          </div>
        </fieldset></form>
      </div>
    </div>
</div>


<hr/>
<h2 id="prediction">Predictions</h2>

<div class="column-wrapper">
    <div class="column-two">
        <div class="column-left">
            <ul>
                <li><b>SIFT predictions</b>
                  <p><a rel="external" href="http://sift.bii.a-star.edu.sg/">SIFT</a> predicts whether an amino
                  acid substitution affects protein function based on sequence
                  homology and the physical properties of amino acids. Only
                  available in popular species. For both SIFT and PolyPhen VEP
                  can report either a score between 0 and 1, a prediction in
                  words, or both. Equivalent to <a
                  href="/info/docs/tools/vep/script/vep_options.html#opt_sift">--sift</a>.</p>
                </li>
                
                <li><b>PolyPhen predictions</b>
                  <p><a rel="external" href="http://genetics.bwh.harvard.edu/pph2/">PolyPhen</a> is a
                  tool which predicts possible impact of an amino acid
                  substitution on the structure and function of a human protein
                  using straightforward physical and comparative considerations.
                  Equivalent to <a
                  href="/info/docs/tools/vep/script/vep_options.html#opt_polyphen">--polyphen</a>.</p>
                </li>

                <li><b>dbNSFP</b>
                  <p>Retrieves data for missense variants from <a rel="external" href="https://sites.google.com/site/jpopgen/dbNSFP">dbNSFP</a>.
                  Equivalent to the VEP plugin <a rel="external" 
                  href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/dbNSFP.pm">dbNSFP</a>.</p>
                </li>

                <li><b>AlphaMissense</b>
                  <p>
                    <a rel="external" href="https://www.science.org/doi/10.1126/science.adg7492">AlphaMissense</a> is a deep learning
                    model developed by Google DeepMind that predicts the pathogenicity of single nucleotide missense variants.
                    Equivalent to the VEP plugin <a rel="external"
                    href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/AlphaMissense.pm">AlphaMissense</a>.
                  </p>
                </li>

                <li><b>CADD</b>
                  <p>
                    <a rel="external" href="https://cadd.gs.washington.edu">Combined Annotation Dependent Depletion (CADD)</a> is a tool for
                    scoring the deleteriousness of single nucleotide variants
                    and insertion/deletion variants in the human genome. CADD
                    integrates multiple annotations into one metric by
                    contrasting variants that survived natural selection with
                    simulated mutations. CADD is only available here for
                    non-commercial use. See
                    <a rel="external" href="https://cadd.gs.washington.edu">CADD website</a> for more information.
                    Equivalent to the VEP plugin <a rel="external" 
                  href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/CADD.pm">CADD</a>.
                  </p>
                </li>

                <li><b>REVEL</b>
                  <p>
                    <a rel="external" href="https://sites.google.com/site/revelgenomics">Rare Exome Variant Ensemble Learner (REVEL)</a> is an ensemble method for predicting the pathogenicity of missense variants based on a combination of scores from multiple individual tools. REVEL is only available here for
                    non-commercial use.
                    Equivalent to the VEP plugin <a rel="external" 
                  href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/REVEL.pm">REVEL</a>.
                  </p>
                </li>

                <li><b>ClinPred</b>
                  <p>
                    <a rel="external" href="https://sites.google.com/site/clinpred">ClinPred</a> is a prediction tool to identify disease-relevant nonsynonymous single nucleotide variants. The predictor incorporates existing pathogenicity scores and benefits from normal population allele frequencies. ClinPred is only available here for
                    non-commercial use.
                    Equivalent to the VEP plugin <a rel="external" 
                  href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/ClinPred.pm">ClinPred</a>.
                  </p>
                </li>

                <li><b>EVE</b>
                  <p>Adds information from <a rel="external"
                  href="https://evemodel.org/">EVE</a> (evolutionary model of variant effect).
                  Equivalent to the VEP plugin <a rel="external"
                  href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/EVE.pm">EVE</a>.</p>
                </li>

                <hr />
                
                <li><b>dbscSNV</b>
                  <p>Retrieves data for splicing variants from <a rel="external" href="https://sites.google.com/site/jpopgen/dbNSFP">dbscSNV</a>.
                  Equivalent to the VEP plugin <a rel="external" 
                  href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/dbscSNV.pm">dbscSNV</a>.</p>
                </li>
                
                <li><b>MaxEntScan</b>
                  <p>Get splice site predictions from <a rel="external" href="http://genes.mit.edu/burgelab/maxent/Xmaxentscan_scoreseq.html">MaxEntScan</a>.
                  Equivalent to the VEP plugin <a rel="external" 
                  href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/MaxEntScan.pm">MaxEntScan</a>.</p>
                </li>

                <li><b>SpliceAI</b>
                  <p>
                    Pre-calculated annotations from <a href="https://github.com/Illumina/SpliceAI" rel="external">SpliceAI</a> a deep neural network,
                    developed by Illumina, Inc that predicts splice junctions from an arbitrary pre-mRNA transcript sequence. Used for non-commercial purposes.
                    This functionality is provided by the
                    <a href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/SpliceAI.pm">SpliceAI</a> plugin.<br />
                    The pre-calculated annotations for all possible single nucleotide substitutions can be retrieved from:
                    <ul>
                        <li>Ensembl/GENCODE v24 canonical transcripts
                          <p>Masked scores</p>
                        </li>
                        <li>Ensembl/GENCODE v37 MANE transcripts
                          <p>Raw scores</p>
                        </li>
                    </ul>
                  <b>Note:</b> The pre-calculated annotations for 1 base insertions, and 1-4 base deletions are only available for Ensembl/GENCODE v24 canonical transcripts.
                  </p>
                </li>

                <hr />
                
                <li><b>BLOSUM62</b>
                  <p>Looks up the BLOSUM 62 substitution matrix score for the reference and alternative 
                  amino acids predicted for a missense mutation.
                  Equivalent to the VEP plugin <a rel="external" 
                  href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/Blosum62.pm">Blosum62</a>.</p>
                </li>

                
                <li><b>Ancestral allele</b>
                  <p>Retrieves ancestral allele sequences from a FASTA file. Ensembl produces <a rel="external" href="[[SPECIESDEFS::ENSEMBL_FTP_URL]]/current_fasta/ancestral_alleles/">FASTA file dumps</a> of the ancestral sequences of key species.
                  Equivalent to the VEP plugin <a rel="external" 
                  href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/AncestralAllele.pm">AncestralAllele</a>.</p>
                </li>


            </ul>
        </div>
    </div>
      
    <div class="column-two">
      <div class="column-right" style="padding-left:2em">
        <form style="scale: 0.8; transform-origin: top left; pointer-events: none; width: 150%" class="_tool_form _check check vep-form bgcolour"><fieldset class="extra-options-fieldset" style="margin: 0; padding: 8px">
          <div class="extra_configs_wrapper" style="margin: 0"><div class="extra_configs_button"><a href="#Predictions" rel="__togg_Predictions" class="set_cookie toggle _slide_toggle open">Predictions</a><span class="extra_configs_info">Variant predictions, e.g. SIFT, PolyPhen</span></div><div class="extra_configs toggleable __togg_Predictions hidden" style="display: block;">
            <fieldset class="_stt_pphn _stt_sift" style="display: block;">
              <h2>Pathogenicity predictions</h2><input type="hidden">
              <div class="_stt_sift form-field" style="display: block;"><label class="ff-label"><span class="_ht ht">SIFT</span>:</label><div class="ff-right"><select class="fselect"><option selected="selected">Prediction and score</option></select></div></div>
              <div class="_stt_pphn form-field"><label class="ff-label"><span class="ht _ht">PolyPhen</span>:</label><div class="ff-right"><select class="fselect"><option selected="selected">Prediction and score</option></select></div></div>
              <div class="_stt_Homo_sapiens form-field"><label class="ff-label"><span class="_ht ht">dbNSFP</span>:</label><div class="ff-right"><p class="ff-checklist"><input class="_stt plugin_enable" type="radio" checked="checked"><label class="ff-checklist-label">Disabled</label></p><p class="ff-checklist"><input type="radio" class="plugin_enable _stt"><label class="ff-checklist-label">Enabled</label></p></div></div><div class="_stt_plugin_dbNSFP ff-multi form-field" style="display: none;"><label class="ff-label"><span class="_ht ht">Fields to include</span>:</label><div class="ff-right"><select class="fselect" multiple="multiple" style="height:150px;"><option>aapos</option></select></div><div class="ff-notes">Field descriptions in <a rel="external" href="#">dbNSFP README</a></div></div><div class="form-field _stt_plugin_dbNSFP" style="display: none;"><label class="ff-label"><span class="_ht ht">Transcript match</span>:</label><div class="ff-right"><p class="ff-checklist"><input type="checkbox"></p></div></div>
              <div class="form-field _stt_Homo_sapiens"><label class="ff-label"><span class="_ht ht">AlphaMissense</span>:</label><div class="ff-right"><p class="ff-checklist"><input class="plugin_enable _stt" type="checkbox"></p></div></div>
              <div class="form-field _stt_Homo_sapiens"><label class="ff-label"><span class="_ht ht">CADD</span>:</label><div class="ff-right"><p class="ff-checklist"><input class="plugin_enable _stt" type="checkbox"></p></div></div>
              <div class="form-field _stt_Homo_sapiens"><label class="ff-label"><span class="_ht ht">REVEL</span>:</label><div class="ff-right"><p class="ff-checklist"><input class="plugin_enable _stt" type="checkbox"></p></div></div>
              <div class="form-field _stt_Homo_sapiens"><label class="ff-label"><span class="_ht ht">ClinPred</span>:</label><div class="ff-right"><p class="ff-checklist"><input class="plugin_enable _stt" type="checkbox"></p></div></div>
              <div class="_stt_Homo_sapiens form-field"><label class="ff-label"><span class="ht _ht">EVE</span>:</label><div class="ff-right"><p class="ff-checklist"><input class="plugin_enable _stt" type="checkbox"></p></div></div>
            </fieldset>
            <fieldset class="_stt_Homo_sapiens">
              <h2>Splicing predictions</h2>
              <div class="_stt_Homo_sapiens form-field"><label class="ff-label"><span class="ht _ht">dbscSNV</span>:</label><div class="ff-right"><p class="ff-checklist"><input class="plugin_enable _stt" type="checkbox"></p></div></div>
              <div class="form-field _stt_Homo_sapiens"><label class="ff-label"><span class="ht _ht">MaxEntScan</span>:</label><div class="ff-right"><p class="ff-checklist"><input class="_stt plugin_enable" type="checkbox"></p></div></div>
              <div class="form-field"><label class="ff-label"><span class="_ht ht">SpliceAI</span>:</label><div class="ff-right"><p class="ff-checklist"><input checked="checked" class="_stt plugin_enable" type="radio"><label class="ff-checklist-label">Disabled</label></p><p class="ff-checklist"><input type="radio" class="plugin_enable _stt"><label class="ff-checklist-label">Enabled</label></p></div></div><div class="form-field _stt_plugin_SpliceAI" style="display: none;"><label class="ff-label">Select analysis mode:</label><div class="ff-right"><select class="fselect"><option selected="selected">Masked scores (Ensembl/GENCODE v24 canonical transcripts)</option><option>Raw scores (Ensembl/GENCODE v37 MANE transcripts, SNV only)</option></select></div></div>
            </fieldset>
            <fieldset>
              <h2>Conservation</h2>
              <div class="form-field"><label class="ff-label"><span class="_ht ht">BLOSUM62</span>:</label><div class="ff-right"><p class="ff-checklist"><input type="checkbox" class="plugin_enable _stt"></p></div></div>
              <div class="form-field _stt_Homo_sapiens"><label class="ff-label"><span class="ht _ht">Ancestral allele</span>:</label><div class="ff-right"><p class="ff-checklist"><input class="plugin_enable _stt" type="checkbox"></p></div></div>
            </fieldset>
          </div></div>
        </form>
      </div>
    </div>
</div>


<hr/>
<h2 id="filter">Filtering options</h2>

<div class="column-wrapper">
    <div class="column-two">
        <div class="column-left">
            VEP allows you to pre-filter your results e.g. by MAF or
            consequence type. Note that it is also possible to perform
            equivalent operations on the results page for VEP, so if you
            aren't sure, don't use any of these options! 
            <ul>
                <li><b>By frequency</b>
                  <p>Filter variants by minor allele frequency (MAF). Two options are provided:
                    <ul>
                        <li><b>Exclude common variants</b>
                          <p>Filter out variants that are co-located with an existing variant that has a
                          frequency greater than 0.01 (1%) in the 1000 Genomes
                          global population. Equivalent to <a
                          href="/info/docs/tools/vep/script/vep_options.html#opt_filter_common">--filter_common</a>
                          in the VEP script.</p>
                        </li>
                        <li><b>Advanced filtering</b>
                          <p>Enabling this option allows you to specify a population and frequency to
                          compare to, as well whether matching variants should be
                          included or excluded from the results.</p>
                        </li>
                    </ul>
                   </p>
                </li>
                
                <li><b>Return results for variants in coding regions only</b>
                  <p>Exclude variants that don't fall in a coding region of a
                  transcript. Equivalent to <a
                  href="/info/docs/tools/vep/script/vep_options.html#opt_coding_only">--coding_only</a>.</p>
                </li>
                
                <li><b>Restrict results</b>
                  <p>For many variants VEP will report multiple consequence types - typically this is because
                  the variant overlaps more than one transcript. For each of these
                  options VEP uses consequence ranks that are subjectively
                  determined by Ensembl. <a
                  href="/info/genome/variation/prediction/predicted_data.html#consequence_type_table">This
                  table</a> gives all of the consquence types predicted by
                  Ensembl, ordered by rank. Note that enabling one of these
                  options not only loses potentially relevant data, but in some
                  cases may be scientifically misleading. Options:
                    <ul>
                        <li><b>Show one selected consequence</b>
                        <p>Pick one consequence type across all those predicted for the
                          variant; the output will include transcript- or
                          feature-specific information. Consequences are chosen by
                          the canonical, biotype status and length of the
                          transcript, along with the ranking of the consequence
                          type according to <a
                          href="/info/genome/variation/prediction/predicted_data.html#consequence_type_table">this
                          table</a>. This is the best method to use if you are
                          interested only in one consequence per variant.
                          Equivalent to <a
                          href="/info/docs/tools/vep/script/vep_options.html#opt_per_gene">--pick</a>.</p>
                        </li>
                        <li><b>Show one selected consequence per gene</b>
                          <p>Pick one consequence type for each gene using the same
                          criteria as above. Note that if a variant overlaps more
                          than one gene, output for each gene will be reported.
                          Equivalent to <a
                          href="/info/docs/tools/vep/script/vep_options.html#opt_per_gene">--per_gene</a>.</p>
                        </li>
                        <li><b>Show only list of consequences per variant</b>
                          <p>Give a comma-separated list of all observed consequence
                          types for each variant. No transcript-specific or
                          gene-specific output will be given. Equivalent to <a
                          href="/info/docs/tools/vep/script/vep_options.html#opt_summary">--summary</a>.</p>
                        </li>
                        <li><b>Show most severe per variant</b>
                          <p>Only the most severe of all observed consequence types is reported for
                          each variant. No transcript-specific or gene-specific
                          output will be given. Equivalent to <a
                          href="/info/docs/tools/vep/script/vep_options.html#opt_most_severe">--most_severe</a>.</p>
                        </li>
                    </ul>
                </li>
            </ul>
        </div>
    </div>
      
    <div class="column-two">
      <div class="column-right" style="padding-left:2em">
        <form style="scale: 0.8; transform-origin: top left; pointer-events: none; width: 150%" class="_tool_form _check check vep-form bgcolour"><fieldset class="extra-options-fieldset" style="margin: 0; padding: 8px">
          <div class="extra_configs_wrapper" style="margin: 0"><div class="extra_configs_button"><a class="set_cookie _slide_toggle toggle open" rel="__togg_Filtering_options" href="#Filtering_options">Filtering options</a><span class="extra_configs_info">Pre-filter results by frequency or consequence type</span></div><div class="extra_configs toggleable hidden __togg_Filtering_options" style="display: block;">
            <fieldset>
              <h2>Filters</h2>
              <div class="form-field ff-multi _stt_Homo_sapiens"><label class="ff-label"><span class="_ht ht">Filter by frequency</span>:</label><div class="ff-right"><p class="ff-checklist"><input type="radio" class="_stt"><label class="ff-checklist-label">No filtering</label></p><p class="ff-checklist"><input type="radio" class="_stt"><label class="ff-checklist-label">Exclude common variants</label></p><p class="ff-checklist"><input class="_stt" type="radio" checked="checked"><label class="ff-checklist-label">Advanced filtering</label></p></div><div class="ff-inline ff-right _stt_advanced"><select class="fselect"><option selected="selected">Exclude</option><option>Include only</option></select><select class="fselect"><option selected="selected">variants with MAF greater than</option><option>variants with MAF less than</option></select><input value="0.01" class="ftext optional _string" type="text"><select class="fselect"><option selected="selected">in 1000 genomes (1KG) combined population</option></select></div></div>
              <div class="form-field"><label class="ff-label"><span class="_ht ht">Return results for variants in coding regions only</span>:</label><div class="ff-right"><p class="ff-checklist"><input type="checkbox"></p></div></div>
              <div class="form-field ff-multi"><label class="ff-label"><span class="ht _ht">Restrict results</span>:</label><div class="ff-right"><select class="fselect"><option selected="selected">Show all results</option></select></div><div class="ff-notes"><b>NB:</b> Restricting results may exclude biologically important data!</div></div>
            </fieldset>
          </div></div>
        </fieldset></form>
      </div>
    </div>
</div>


<hr/>
<h2 id="advanced">Advanced options</h2>

<div class="column-wrapper">
    <div class="column-two">
        <div class="column-left">
          <p>The VEP web interface allows you to use/setup advanced options:</p>
          <ul>
            <li><b>Buffer size</b>
              <p>By default VEP process the variants by blocks of 5000 (i.e. what we call "buffer size").<br />
In some cases, reducing the size of the blocks (buffer size) could prevent memory issues for large VEP queries (e.g. use of regulatory data, many plugins or custom annotations).<br />
This is why the maximum buffer size is automatically set to 500 on the VEP Web interface when the "Regulatory data" option is selected.</p></li>

<li><b>Right align variants prior to consequence calculation</b>
  <p>By default VEP performs consequence calculation at the given input coordinates.<br />
Optionally, VEP can shift insertions and deletions found within repeated regions as far as possible in the 3' direction, normalising output.
</ul>
        </div>
    </div>

    <div class="column-two" style="min-height: 250px;">
      <div class="column-right" style="padding-left:2em">
        <form style="scale: 0.8; transform-origin: top left; pointer-events: none; width: 150%" class="_tool_form _check check vep-form bgcolour"><fieldset class="extra-options-fieldset" style="margin: 0; padding: 8px">
          <div class="extra_configs_wrapper" style="margin: 0"><div class="extra_configs_button"><a href="#Advanced_options" rel="__togg_Advanced_options" class="toggle _slide_toggle set_cookie open">Advanced options</a><span class="extra_configs_info">Additional enhancements</span></div><div class="__togg_Advanced_options toggleable hidden extra_configs" style="display: block;">
            <fieldset>
              <h2>Advanced options</h2>
              <div class="ff-multi form-field"><label class="ff-label"><span class="_ht ht">Buffer size</span>:</label><div class="ff-right"><select class="fselect"><option selected="selected">5000</option></select></div><div class="ff-notes"><div class="_stt_Homo_sapiens _stt_Mus_musculus" style="display: block;"><b>NB:</b> When the <b>Regulatory data</b> option is selected then due to the large amount of regulatory data available, the <b>maximum buffer size</b> is automatically reduced from the default value of <b>5000</b> to <b>500</b>. This reduces the memory requirement but might increase the run time. If you find that your jobs are still failing due to memory limitations then you can select a value <b>lower than 500</b>.</div></div></div>
              <div class="form-field"><label class="ff-label"><span class="ht _ht">Right align variants prior to consequence calculation</span>:</label><div class="ff-right"><select class="fselect"><option selected="selected">Right align relative to transcript</option></select></div></div>
            </fieldset>
          </div></div>
        </fieldset></form>
      </div>
    </div>
</div>


<hr/>
<h2 id="jobs">Jobs</h2>

<div class="column-wrapper">
    <div class="column-two">
      <div class="column-left">
          <p>Once you have clicked <b>Run</b>, your input will be checked
          and submitted to the VEP as a job. All jobs associated with your
          session or account are shown in the <b>Recent Tickets</b> table.
          You may submit multiple jobs simultaneously.</p> <p>The
          <b>Jobs</b> column of the table shows the current status of the
          job.</p>

          <ul>
            <li><span class="_ht job-status-submitted job-status"><a>Queued</a></span>
            - your job is waiting to be submitted to the system</li>
            <li><span class="_ht job-status-running job-status"><a>Running</a></span>
            - your job is currently running</li>
            <li><span class="_ht job-status-done job-status"><a>Done</a></span>
            - your job is finished - click the [View results]
            link to be taken to the results page</li>
            <li><span class="_ht job-status-failed job-status"><a>Failed</a></span>
            - there is a problem with your job - click the magnifying glass icon
            <img src="/i/16/search.png"> to see more details</li>
          </ul>

          The following actions are available for each job:
          <ul>
            <li><b><img src="/i/16/save.png"> Save icon:</b> save the job (you
            need to login with an Ensembl account).</li>
            <li><b><img src="/i/16/pencil.png"> Edit icon:</b> resubmit a job
            (for example, to slightly tweak the data or parameters before
            re-running).</li>
            <li><b><img src="/i/16/search.png"> Magnifying glass icon:</b> see
            summary of the options that you selected for your VEP job, as well
            as data versions associated with this run.</li>
            <li><b><img src="/i/16/share.png"> Share icon:</b> display URL to
            share with other users. You can also disable URL sharing here.</li>
            <li><b><img src="/i/16/trash.png"> Trash can icon:</b> delete a
            job.</li>
          </ul>
      </div>
    </div>
      
    <div class="column-two">
      <div class="column-right" style="padding-left:2em">
        <div class="dataTables_wrapper" style="scale: 0.6; transform-origin: top left; pointer-events: none; width: 150%"><div class="dataTables_top"><div class="dataTables_length"><label>Show <select data-role="none" size="1"><option selected="selected">All</option></select> entries</label></div><div class="col_toggle"><div class="toggle">Show/hide columns</div></div><div class="dataTables_filter"><label><input data-role="none" type="text"><div class="overlay">Filter</div></label></div><div class="invisible"></div></div><table class="autocenter ss data_table" style="width: 100%" cellpadding="0" cellspacing="0">
          <thead style="">
            <tr class="ss_header" style="white-space: nowrap; width: 100%;">
              <th class="sort_string sorting" title="Analysis" rowspan="1" colspan="1" style="white-space: normal; width: 10px;">Analysis</th>
              <th class="sort_none sorting_disabled" title="Jobs" rowspan="1" colspan="1" style="white-space: normal">Jobs</th>
              <th class="sort_numeric_hidden sorting" title="Submitted at (GMT)" rowspan="1" colspan="1" style="white-space: normal; width: 10px;">Submitted at (GMT)</th>
              <th class="sort_none sorting_disabled" rowspan="1" colspan="1" style="white-space: normal; width: 10px;"></th>
            </tr>
          </thead>
          <tbody>
            <tr class="_ticket_8Cca81zmCDOPs02S bg1"><td>Variant Effect Predictor</td><td><img class="ticket_img" src="//static.ensembl.org/i/species/Bos_taurus.png" style="width:16px; height:16px;" alt="" class="_ht job-species"><span class="right-margin">VEP analysis of pasted data in Bos_taurus</span><span class="_ht job-status-done job-status"><a>Done</a></span><a class="small left-margin">[View results]</a><span class="job-sprites" style="position: absolute"><a class="_ticket_view job-sprite _change_location"><span class="view_icon sprite _ht"></span></a><a class="_ticket_edit job-sprite _change_location"><span class="_ht edit_icon sprite"></span></a><a class="job-sprite _json_link"><span class="sprite delete_icon _ht"></span></a></span></p></td><td><span class="hidden">20230511170040</span>11/05/2023, 17:00</td><td><div class="ticket-sprites"><span class="_ht save_icon sprite sprite_disabled"></span><a class="_change_location _ticket_edit"><span class="sprite edit_icon _ht"></span></a><div class="ticket-share-icon share_icon _ticket_share hidden sprite"><form class="top-margin"><p><label><input type="checkbox" checked="checked">&nbsp;Share ticket via URL</label></p><p class="_ticket_share_url"><input class="ticket-share-input" type="text"></p></form></div><a class="_json_link"><span class="delete_icon sprite _ht"></span></a></div></td></tr>
            <tr class="_ticket_NzFhqZNWDwWVXfZJ bg2"><td>Variant Effect Predictor</td><td><img class="_ht job-species" style="height:16px; width:16px;" src="//static.ensembl.org/i/species/Ovis_aries.png" alt=""><span class="right-margin">VEP analysis of pasted data in Ovis_aries</span><span class="_ht job-status job-status-done"><a>Done</a></span><a class="left-margin small">[View results]</a></p></td><td><span class="hidden">20230511165532</span>11/05/2023, 16:55</td><td><div class="ticket-sprites"><span class="_ht save_icon sprite sprite_disabled"></span><a href="/Multi/Tools/VEP/Edit?db=core;tl=NzFhqZNWDwWVXfZJ" class="_change_location _ticket_edit"><span class="_ht edit_icon sprite"></span></a><div class="share_icon ticket-share-icon sprite hidden _ticket_share"><form class="top-margin"><p><label><input type="checkbox" checked="checked">&nbsp;Share ticket via URL</label></p><p class="_ticket_share_url"><input class="ticket-share-input" type="text"></p></form></div><a class="_json_link" href="/Multi/Json/Tools/VEP/delete?db=core;tl=NzFhqZNWDwWVXfZJ"><span class="delete_icon sprite _ht"></span><span class="_confirm hidden">This will delete the following job permanently: VEP analysis of pasted data in Homo_sapiens</span></a></div></td></tr>
            <tr class="_ticket_1uyqKkUHWWUzEbyk bg1"><td>Variant Effect Predictor</td><td><img class="_ht job-species" style="width:16px; height:16px;" src="//static.ensembl.org/i/species/Homo_sapiens.png" alt=""><span class="right-margin">VEP analysis of pasted data in Homo_sapiens</span><span class="job-status job-status-failed _ht"><a>Failed</a></span></p></td><td><span class="hidden">20230420095705</span>11/05/2023, 16:54</td><td><div class="ticket-sprites"><span class="save_icon _ht sprite sprite_disabled"></span><a class="_ticket_edit _change_location"><span class="edit_icon _ht sprite"></span></a><div class="ticket-share-icon share_icon _ticket_share hidden sprite"><form class="top-margin"><p><label><input type="checkbox" checked="checked">&nbsp;Share ticket via URL</label></p><p class="_ticket_share_url"><input class="ticket-share-input" type="text"></p></form></div><a class="_json_link"><span class="_ht sprite delete_icon"></span></a></div></td></tr></tr>
            <tr class="_ticket_1uyqKkUHWWUzEbyk bg2"><td>Variant Effect Predictor</td><td><img class="_ht job-species" style="width:16px; height:16px;" src="//static.ensembl.org/i/species/Homo_sapiens.png" alt=""><span class="right-margin">VEP analysis of pasted data in Homo_sapiens</span><span class="job-status job-status-running _ht"><a>Running</a></span></p></td><td><span class="hidden">20230420095705</span>11/05/2023, 16:51</td><td><div class="ticket-sprites"><span class="save_icon _ht sprite sprite_disabled"></span><a class="_ticket_edit _change_location"><span class="edit_icon _ht sprite"></span></a><div class="ticket-share-icon share_icon _ticket_share hidden sprite"><form class="top-margin"><p><label><input type="checkbox" checked="checked">&nbsp;Share ticket via URL</label></p><p class="_ticket_share_url"><input class="ticket-share-input" type="text"></p></form></div><a class="_json_link"><span class="_ht sprite delete_icon"></span></a></div></td></tr></tr>
            <tr class="_ticket_1uyqKkUHWWUzEbyk bg1"><td>Variant Effect Predictor</td><td><img class="_ht job-species" style="width:16px; height:16px;" src="//static.ensembl.org/i/species/Homo_sapiens.png" alt=""><span class="right-margin">VEP analysis of pasted data in Homo_sapiens</span><span class="job-status job-status-submitted _ht"><a>Queued</a></span></p></td><td><span class="hidden">20230420095705</span>11/05/2023, 16:49</td><td><div class="ticket-sprites"><span class="save_icon _ht sprite sprite_disabled"></span><a class="_ticket_edit _change_location"><span class="edit_icon _ht sprite"></span></a><div class="ticket-share-icon share_icon _ticket_share hidden sprite"><form class="top-margin"><p><label><input type="checkbox" checked="checked">&nbsp;Share ticket via URL</label></p><p class="_ticket_share_url"><input class="ticket-share-input" type="text"></p></form></div><a class="_json_link"><span class="_ht sprite delete_icon"></span></a></div></td></tr></tr>
          </tbody>
        </table><div class="dataTables_bottom" style="display: none;"><div class="dataTables_info" style="display: none;">Showing 1 to 63 of 63 entries</div><div class="dataTables_paginate paging_full_numbers" style="display: none;"><span class="first paginate_button paginate_button_disabled">&lt;&lt;</span><span class="previous paginate_button paginate_button_disabled">&lt;</span><span></span><span class="next paginate_button paginate_button_disabled">&gt;</span><span class="last paginate_button paginate_button_disabled">&gt;&gt;</span></div><div class="invisible"></div></div></div>
      </div>
    </div>
</div>

</body>
</html>
