
<html>
  <title>VEP documentation</title>
  <body>
<style>
img.opaque-hover:hover {
  opacity:0.6;
  filter:alpha(opacity=60);
}
img.opaque-hover {
  opacity:1.0;
  filter:alpha(opacity=100);
}
</style>

<div>
  <div style="float:right"><img src="/img/vep_logo.png"/></div>

  <h1 id="top"><span style="color:#006;padding-right:15px">Ensembl Variant Effect Predictor</span><span style="color:#666"><img src="/i/16/globe.png"/> Web interface</span></h1>
  <hr/>
  <div>
    <div style="padding-left:2em; float:right;">
      <a href="/[[SPECIESDEFS::ENSEMBL_PRIMARY_SPECIES]]/Tools/VEP" class="_ht" title="Launch the VEP">
        <img style="padding: 5px;" class="opaque-hover" src="/img/vep_launch.png">
      </a>
    </div>
    
    <div>
      <p> Use the Ensembl VEP online to analyse your variants through a
      simple point-and-click interface. </p>
      
      <p> The web interface allows you to access the key features of
      the Ensembl VEP without using the command line. Interactively filter
      your results to find the data you want. Download your results in
      multiple data formats, easily share your results with others,
      and integrate your variation data with the powerful
      Ensembl web browser. </p>
      
      <p> If you use the Ensembl VEP in your work, please cite <b>McLaren et.
      al.</b> (<a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-016-0974-4"
      rel="external">doi:10.1186/s13059-016-0974-4</a>) </p>

      <p>See our latest tutorial
      <a href="https://onlinelibrary.wiley.com/doi/10.1002/humu.24298" rel="external">
      Annotating and prioritizing genomic variants using the Ensembl Variant Effect Predictor â€” A tutorial</a>
      for more information on using the Ensembl VEP web interface.</p>
      
      <p> Any questions? Send an email to the Ensembl developer's mailing
      list, <a href="mailto:dev@ensembl.org">dev@ensembl.org</a> 
      or contact the Ensembl Helpdesk at <a
      href="mailto:helpdesk@ensembl.org">helpdesk@ensembl.org</a>.</p>
    </div>
  </div>
  
  <div style="clear:both; padding-top: 20px; ">
    <div class="tinted-box">
      <div style="float:right; font-size: 16px;"> 
        <img src="/i/16/download.png"> <a href="VEP_web_documentation.pdf">Download documentation in PDF format</a>
      </div>
      
      <h1 id="contents"><img src="/i/16/documentation.png"> Documentation contents</h1>
      
      <div class="vep_doc_col" style="float:left;margin-right:40px">
      
        <img src="/i/16/pencil.png"> <a href="input.html" class="notext" style="font-size:16px;font-weight:bold">Input form</a>
        <ul>
          <li><a href="input.html#input">Data input</a></li>
          <li><a href="input.html#ident">Identifiers</a></li>
          <li><a href="input.html#freq">Variants and frequency data</a></li>
          <li><a href="input.html#annotation">Additional annotations</a></li>
          <li><a href="input.html#prediction">Predictions</a></li>
          <li><a href="input.html#filter">Filtering options</a></li>
          <li><a href="input.html#advanced">Advanced options</a></li>
          <li><a href="input.html#jobs">Jobs</a></li>
        </ul>
      </div>
      
      
      <div class="vep_doc_col" style="float:left;margin-right:40px">
        <img src="/i/16/copy.png"> <a href="results.html" class="notext" style="font-size:16px;font-weight:bold">Results</a>
        <ul>
          <li><a href="results.html#summary">Results summary</a></li>
          <li><a href="results.html#table">Results preview table</a></li>
          <li><a href="results.html#nav">Navigating results</a></li>
          <li><a href="results.html#filter">Filtering results</a></li>
          <li><a href="results.html#download">Downloading results</a></li>
        </ul>
      </div>
      
      <div class="vep_doc_col" style="float:left;margin-right:40px">
        <img src="/i/16/copy.png"> <a href="../vep_formats.html" class="notext" style="font-size:16px;font-weight:bold">Data formats</a>
        <ul>
          <li><a href="../vep_formats.html#input">Input</a></li>
          <li><a href="../vep_formats.html#output">Output</a></li>
        </ul>
      </div>
      
      <div style="float:left">
        <img src="/i/16/help.png"> <a href="../vep_faq.html" class="notext" style="font-size:16px;font-weight:bold">FAQ</a>
        <ul>
          <li><a href="../vep_faq.html#general">General questions</a></li>
          <li><a href="../vep_faq.html#web">Web Ensembl VEP questions</a></li>
          <li><a href="../vep_faq.html#script">Ensembl VEP script questions</a></li>
        </ul>
      </div>
    </div>
  </div>
</div>
<div style="clear:both"><h1 style="page-break-after: always">&nbsp</h1></div>
<style>
._afterimage._ht.job-species {
    width: 16px;
    height: 16px;
}
</style>

  
  <div style="float:right"><img src="/img/vep_logo.png"/></div>

  <h1 id="top"><span style="color:#006;padding-right:15px">Ensembl Variant Effect Predictor</span><span style="color:#666"><img src="/i/16/pencil.png"/> Input form</span></h1>
  <hr/>

<p> When you reach the Ensembl VEP web interface, you will be presented with a form to
enter your data and alter various options.</p>

<p> Note that the listed options change depending on the selected species. </p>

<h2 id="input">Data input</h2>

<div class="column-wrapper">
    <div class="column-two">
      <div class="column-left">
          <ol>
              <li> First select the correct species for your data. Ensembl hosts
              many vertebrate genomes; genomes for plants, protists and fungi
              can be found at <a href="http://www.ensemblgenomes.org/">Ensembl
              Genomes</a>. </li>
              
              <li> You can optionally choose a name for the data you upload -
              this can make it easier for you to identify jobs and files that
              you have uploaded to the Ensembl VEP at a later point. </li>
              
              <li> <p>You have three options for uploading your data:
                  
                  <ul>
                      <li><b>File upload</b> - click the "Choose file" button
                      and locate the file on your system. Please ensure your files
			                are sorted by location if you are using VCF or other input 
			                formats using location. This greatly improves speed of analysis</li>
                      <li><b>Paste file</b> - simply copy and paste the contents
                      of your file into the large text box</li>
                      <li><b>File URL</b> - point the Ensembl VEP to a file hosted on a
                      publically accessible address. This can be either a
                      <b>http://</b> or <b>ftp://</b> address.</li>
                  </ul>
                  
                  Once you have uploaded some data, you can select it as the
                  input for future jobs by choosing the data from the drop down
                  menu.</p>
                  
                  <p> The format of your data is automatically detected; see the
                  examples or the <a href="../vep_formats.html#input">input
                  format</a> documentation. </p>
              </li>
              
              <li> For pasted data you can get an instant preview of the results of your
              first variant by clicking the button that appears when you paste
              your data. This quickly shows you the consequence type, the IDs of
              any overlapping variants, genes, transcripts and regulatory
              features, as well as SIFT and PolyPhen predictions. To see the full
              results set submit your job as normal. </li>
              
              <li>
                  <p> For some species you can select which transcript database
                  to use. The default is to use Ensembl transcripts, which offer
                  the most rich annotation through Ensembl VEP. </p>
                  <p> GENCODE Basic is a subset of the GENCODE gene set, and is
                  intended to provide a simplified, high-quality subset of the
                  GENCODE transcript annotations that will be useful to the
                  majority of users. GENCODE Basic includes all genes in the
                  GENCODE gene set, with a representative subset of the
                  transcripts (splice variants). </p>
                  <p> GENCODE Primary is a new transcript subset which covers
                  all human exons in a minimal set of transcripts. 
                  This aims to enable annotation of all potential variant 
                  consequences without duplication across multiple transcripts.
                  </p>
                  <p>You can also select to use RefSeq transcripts from the <a
                  href="/info/docs/api/core/index.html#est">otherfeatures
                  database</a>; note though that these transcripts are simply
                  aligned to the reference genome and the database is missing
                  much of the annotation found when using the main Ensembl
                  database (e.g. protein domains, CCDS identifiers).</p>
              </li>
          </ol>
      </div>
    </div>
      
    <div class="column-two">
      <div class="column-right" style="padding-left:2em">
        <form style="scale: 0.6; transform-origin: top left; pointer-events: none; width: fit-content" class="plain-box vep-form _tool_form bgcolour check _check">
          <fieldset>
            <div class="form-field _adjustable_height" style="min-height: 0px;"><label class="ff-label">Species:</label><div class="taxon_selector_form js_panel _sdd ff-right">
              <div class="list-wrapper">
                <ul class="list"><li><img src="/i/species/Homo_sapiens.png" class="nosprite"><span class="ss-selected">Homo_sapiens</span><span class="ss-selection-delete">x</span></li></ul>
                <div class="_vep_assembly italic">Assembly: <span>GRCh38.p13</span></div>
                <div class="links"><a href="#">Change species</a><div class="italic assembly_msg">If you are looking for Ensembl VEP for Human GRCh37, please go to <a href="#" rel="external">GRCh37 website</a>.</div></div>
              </div>
              <div class="checkboxes"><input type="checkbox">Homo_sapiens<br></div>
            </div></div>

            <div class="form-field"><label class="ff-label">Name for this job (optional):</label><div class="ff-right"><input type="text" class="_string optional ftext"></div></div>

            <div class="form-field ff-multi"><label class="ff-label">Input data:</label>
              <div class="ff-right"><div><b>Either paste data:</b></div></div><div class="ff-right"><textarea cols="40" class="_text vep-input ftextarea optional focused" rows="10" style="background-position: 0px 29px;">rs699&#13;&#10;rs171&#13;&#10;rs665</textarea><input value="Run instant Ensembl VEP for current line â€º" type="button" class="quick-vep-button fbutton hidden" style="display: inline-block; margin-top: 30px;"></div>

              <div class="ff-right"><div><span class="small"><b>Examples:&nbsp;</b><a href="#" class="_example_input" rel="ensembl">Ensembl default</a>, <a href="#" class="_example_input" rel="vcf">VCF</a>, <a href="#" class="_example_input" rel="id">Variant identifiers</a>, <a href="#" class="_example_input" rel="hgvs">HGVS notations</a>, <a href="#" class="_example_input" rel="spdi">SPDI</a></span></div></div>

              <div class="ff-right vep_left_input"><span class="_ht ht"><b>Or upload file:</b></span></div><div class="ff-right"><input class="ffile" type="file"></div>

              <div class="ff-right vep_left_label vep_left_input"><div><b>Or provide file URL:</b></div></div><div class="ff-right"><input type="text" size="30" class="ftext optional url _url"></div>
            </div>
            <div class="form-field _stt_rfq" style="display: block;"><label class="ff-label"><span class="_ht ht">Transcript database to use</span>:</label><div class="ff-right"><p class="ff-checklist"><input type="radio" class="_stt" checked="checked"><label class="ff-checklist-label">Ensembl<span style="display: inline;">/GENCODE</span> transcripts</label></p><p class="ff-checklist" style="display: block;"><input type="radio" class="_stt"><label class="ff-checklist-label">Ensembl/GENCODE basic transcripts</label></p><p class="ff-checklist" style="display: block;"><input type="radio" class="_stt"><label class="ff-checklist-label">Ensembl/GENCODE primary transcripts</label><p class="ff-checklist"><input class="_stt" type="radio"><label class="ff-checklist-label">RefSeq transcripts</label></p><p class="ff-checklist"><input class="_stt" type="radio"><label class="ff-checklist-label">Ensembl<span style="display: inline;">/GENCODE</span> and RefSeq transcripts</label></p></div></div></fieldset></form>
      </div>
    </div>
</div>


<hr/>
<h2 id="ident">Identifiers</h2>

<div class="column-wrapper">
    <div class="column-two">
        <div class="column-left">
            Ensembl VEP can provide additional identifiers for genes, transcripts,
            proteins and variants.
            
            <ul>
                <li><b>Gene symbol</b>
                  <p>Add the gene symbol for the gene to
                  the output. This will typically be, for example, the <a
                  href="http://www.genenames.org/" rel="external">HGNC</a>
                  identifier for genes in human. Equivalent to <a
                  href="/info/docs/tools/vep/script/vep_options.html#opt_symbol">--symbol</a>
                  in the Ensembl VEP script.</p>
                </li>

                 <li><b>Transcript version</b>
                   <p>Add the transcript version to the transcript identifier. Equivalent to <a 
                   href="/info/docs/tools/vep/script/vep_options.html#opt_transcript_version">--transcript_version</a>.</p> 
 
                <li><b>CCDS</b>
                  <p>Add the <a href="/info/genome/genebuild/ccds.html">Consensus CDS</a>
                  transcript identifier where available. Equivalent to <a
                  href="/info/docs/tools/vep/script/vep_options.html#opt_ccds">--ccds</a>.</p>
                </li>
                
                <li><b>Protein</b>
                  <p>Add the Ensembl protein identifer (ENSP). Equivalent to <a
                  href="/info/docs/tools/vep/script/vep_options.html#opt_protein">--protein</a>.</p>
                </li>
                
                <li><b>UniProt</b>
                  <p>Add identifiers for translated protein
                  products from three <a href="http://www.ebi.ac.uk/uniprot"
                  rel="external">UniProt</a>-related databases (SWISSPROT, TREMBL
                  and UniParc). Equivalent to <a
                  href="/info/docs/tools/vep/script/vep_options.html#opt_uniprot">--uniprot</a>.</p>
                </li>
                
                <li><b>HGVS</b>
                  <p>Generate <a href="http://varnomen.hgvs.org/" rel="external">HGVS</a>
                  identifiers for your input variants relative to the transcript
                  coding sequence (HGVSc) and the protein sequence (HGVSp).
                  Equivalent to <a href="/info/docs/tools/vep/script/vep_options.html#opt_hgvs">--hgvs</a>.</p>
                </li>
            </ul>
        </div>
    </div>
      
    <div class="column-two">
      <div class="column-right" style="padding-left:2em">
        <form style="scale: 0.8; transform-origin: top left; pointer-events: none; width: 150%" class="_tool_form _check check vep-form bgcolour">
          <fieldset class="extra-options-fieldset" style="margin: 0; padding: 8px"><div class="extra_configs_wrapper" style="margin: 0"><div class="extra_configs_button">
            <a href="#Identifiers" class="toggle _slide_toggle set_cookie open" rel="__togg_Identifiers">Identifiers</a>
            <span class="extra_configs_info">Additional identifiers for genes, transcripts and variants</span></div>

            <div class="toggleable hidden extra_configs __togg_Identifiers" style="display: block;"><fieldset><h2>Identifiers</h2>
              <div class="form-field"><label class="ff-label"><span class="ht _ht">Gene symbol</span>:</label><div class="ff-right"><p class="ff-checklist"><input type="checkbox" checked="checked"></p></div></div>
              <div class="form-field"><label class="ff-label"><span class="_ht ht">Transcript version</span>:</label><div class="ff-right"><p class="ff-checklist"><input checked="checked" type="checkbox"></p></div></div>
              <div class="form-field" style="display: block;"><label class="ff-label"><span class="_ht ht">CCDS</span>:</label><div class="ff-right"><p class="ff-checklist"><input type="checkbox"></p></div></div>
              <div class="form-field"><label class="ff-label"><span class="ht _ht">Protein</span>:</label><div class="ff-right"><p class="ff-checklist"><input type="checkbox"></p></div></div>
              <div class="form-field"><label class="ff-label"><span class="ht _ht">UniProt</span>:</label><div class="ff-right"><p class="ff-checklist"><input type="checkbox"></p></div></div>
              <div class="form-field"><label class="ff-label"><span class="_ht ht">HGVS</span>:</label><div class="ff-right"><p class="ff-checklist"><input type="checkbox"></p></div></div>
            </fieldset></div>
          </div></fieldset>
        </form>
      </div>
    </div>
</div>


<hr/>
<h2 id="freq">Variants and frequency data</h2>

<div class="column-wrapper">
    <div class="column-two">
        <div class="column-left">
            Ensembl VEP can also search the Ensembl database for
            known variants that are co-located with variants from your input
            data.
            
            <ul>
                <li>
                    <p><b>Find co-located known variants</b> - report known
                    variants from the Ensembl Variation database that overlap
                    with your input. A list of variant sources imported can be
                    viewed <a
                    href="/info/genome/variation/species/species_data_types.html#sources">here</a>.
                    Note that this feature is only available for species with an
                    Ensembl Variation database. Equivalent to <a
                    href="/info/docs/tools/vep/script/vep_options.html#opt_check_existing">--check_existing</a>.
                    </p>
                    <p> Ensembl VEP will by default compares the alleles of
                    your input variant to that of the existing variant;
                    Ensembl VEP will only report the existing
                    variant ID if none of the alleles in your input variant are
                    novel. </p>
                    <p> For example, if your input variant has alleles A/G,
                    and the existing variant has alleles A/T, then the existing
                    variant will not be reported. If instead your input variant
                    has alleles A/T, then the existing variant will be reported.
                    </p>
                    <p> To disable this allele matching, select the option
                    "Yes but don't compare alleles" for the option
                    "Find co-located known variants".</p>
                    <p> For known variants Ensembl VEP can also provide PubMed IDs
                    of publications citing the variant (equivalent to <a
                    href="/info/docs/tools/vep/script/vep_options.html#opt_uniprot">--pubmed</a>).</p>
                </li>
                <li><b>Variant synonyms</b>
                    <p>Report known synonyms for co-located variants.</p>
                </li>
                <li><b>Frequency data for co-located variants</b>
                    <p>Ensembl VEP can also report allele frequency (AF)
                    data for existing variants from several major genotyping
                    projects, the <a href="http://www.1000genomes.org/"
                    rel="external">1000 Genomes Project</a>,
                    and <a href="http://gnomad.broadinstitute.org/" rel="external">gnomAD</a>;
                    this only applies when you have selected human as your species.
                        <ul>
                            <li><b>1000 Genomes global</b> - the combined phase
                            3 population (i.e. all individuals from all
                            populations). Equivalent to <a
                            href="/info/docs/tools/vep/script/vep_options.html#opt_af">--af</a>
                            </li>
                            <li><b>1000 Genomes continental</b> - the four
                            continent-level populations - AFR (African), AMR
                            (American), ASN (Asian) and EUR (European). Equivalent to <a
                            href="/info/docs/tools/vep/script/vep_options.html#opt_af_1kg">--af_1kg</a>
                            </li>
                            <li><b>gnomAD exomes</b> - combined, AFR, AMR, ASJ, EAS, FIN, NFE, OTH, SAS populations. Equivalent to <a
                            href="/info/docs/tools/vep/script/vep_options.html#opt_af_gnomade">--af_gnomade</a>
                            </li>
                            <li><b>gnomAD genomes</b> - combined, AFR, AMR, AMI, ASJ, EAS, FIN, MID, NFE, OTH, SAS populations. Equivalent to <a
                              href="/info/docs/tools/vep/script/vep_options.html#opt_af_gnomadg">--af_gnomadg</a>
                            </li>
                        </ul>
                    </p>
                </li>
                <li><b>PubMed IDs for citations of co-located variants</b>
                    <p>Report the PubMed IDs of any publications that cite the co-located variant(s).</p>
                </li>
                <li><b>Include flagged variants</b>
                    <p>Variants flagged as failed by the Ensembl Variation quality control.</p>
                </li>

                <li><b>Paralogue variants</b>
                  <p>
                    Retrieves variants that overlap genomic coordinates corresponding to aligned amino acid positions in paralogous proteins. This functionality is provided by the
                    <a href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/Paralogues.pm">Paralogues</a> plugin.<br />
                  </p>
                </li>

                <li><b>Open Targets Genetics</b>
                  <p>
                    Returns locus-to-gene (L2G) scores to predict causal genes at GWAS loci from
                    <a href="https://genetics.opentargets.org" rel="external">Open Targets Genetics</a>. This functionality is provided by the
                    <a href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/OpenTargets.pm">OpenTargets</a> plugin.<br />
                  </p>
                </li>

            </ul>
      </div>
    </div>

    <div class="column-two">
      <div class="column-right" style="padding-left:2em">
        <form style="scale: 0.8; transform-origin: top left; pointer-events: none; width: 150%" class="_tool_form _check check vep-form bgcolour"><fieldset class="extra-options-fieldset" style="margin: 0; padding: 8px">
          <div class="_stt_var extra_configs_wrapper" style="margin:0; display: block;"><div class="extra_configs_button"><a class="set_cookie toggle _slide_toggle open" rel="__togg_Variants_and_frequency_data" href="#Variants_and_frequency_data">Variants and frequency data</a><span class="extra_configs_info">Co-located variants and frequency data</span></div><div class="__togg_Variants_and_frequency_data toggleable hidden extra_configs" style="display: block;">

          <fieldset>
            <h2>Variants and frequency data</h2>
            <div class="form-field _stt_var" style="display: block;"><label class="ff-label"><span class="ht _ht">Find co-located known variants</span>:</label><div class="ff-right"><select class="_stt fselect"><option selected="selected">Yes</option></select></div></div>
            <div class="_stt_yes form-field" style="display: block;"><label class="ff-label"><span class="ht _ht">Variant synonyms</span>:</label><div class="ff-right"><p class="ff-checklist"><input type="checkbox"></p></div></div><div>
            <div class="_stt_allele _stt_yes form-field"><label class="ff-label">Frequency data for co-located variants:</label><div class="ff-right"><p class="ff-checklist"><input type="checkbox" checked="checked"><label class="ff-checklist-label _ht ht">1000 Genomes global minor allele frequency</label></p><p class="ff-checklist"><input type="checkbox"><label class="ff-checklist-label ht _ht">1000 Genomes continental allele frequencies</label></p><p class="ff-checklist"><input type="checkbox"><label class="_ht ht ff-checklist-label">gnomAD (exomes) allele frequencies</label></p><p class="ff-checklist"><input type="checkbox"><label class="_ht ht ff-checklist-label">gnomAD (genomes) allele frequencies</label></p></div></div>
            <div class="form-field _stt_yes _stt_allele"><label class="ff-label"><span class="ht _ht">PubMed IDs for citations of co-located variants</span>:</label><div class="ff-right"><p class="ff-checklist"><input  type="checkbox" checked="checked"></p></div></div></div>
            <div class="form-field"><label class="ff-label"><span class="ht _ht">Include flagged variants</span>:</label><div class="ff-right"><p class="ff-checklist"><input type="checkbox"></p></div></div>
            <div class="form-field"><label class="ff-label"><span class="_ht ht">Paralogue variants</span>:</label><div class="ff-right"><p class="ff-checklist"><input type="radio" checked="checked" value="no"><label class="ff-checklist-label">Disabled</label></p><p class="ff-checklist"><input class="_stt plugin_enable" value="plugin_Paralogues" type="radio" name="plugin_Paralogues"><label class="ff-checklist-label">Enabled</label></p></div></div>
            <div class="form-field"><label class="ff-label"><span class="_ht ht">Open Targets Genetics</span>:</label><div class="ff-right"><p class="ff-checklist"><input class="plugin_enable _stt" type="checkbox"></p></div></div>
          </fieldset></div></div>
        </fieldset></form>
      </div>
    </div>
</div>

<hr/>
<h2 id="annotation">Additional annotations</h2>

<div class="column-wrapper">
    <div class="column-two">
        <div class="column-left">
            <ul>

                <li><b>DosageSensitivity</b>
                  <p>Retrieves haploinsufficiency and triplosensitivity probability scores
                  for affected genes from a dosage sensitivity catalogue published in paper -
                  <a href="https://www.sciencedirect.com/science/article/pii/S0092867422007887">https://www.sciencedirect.com/science/article/pii/S0092867422007887</a>.</p>
                </li>

                <li><b>LOEUF</b>
                  <p>LOEUF stands for the 'loss-of-function observed/expected upper bound fraction'. 
                  This plugin adds constraint scores derived from gnomAD to Ensembl VEP.
                  Equivalent to the Ensembl VEP plugin <a rel="external"
                  href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/LOEUF.pm">LOEUF</a>.</p>
                </li>

                <hr />

                <li><b>Transcript biotype</b>
                  <p>Add the <a  href="/Help/Faq?id=468">transcript biotype</a> to the output.
                  Equivalent to <a href="/info/docs/tools/vep/script/vep_options.html#opt_biotype">--biotype</a> in
                  the Ensembl VEP script.</p>
                </li>

                <li><b>Exon and intron numbers</b>
                  <p>Report the exon or intron number that a variant falls in as NUMBER / TOTAL, i.e. exon 2/5
                  means the variant falls in the 2nd of 5 exons in the transcript. Equivalent to <a
                  href="/info/docs/tools/vep/script/vep_options.html#opt_numbers">--numbers</a>.</p>
                </li>
                
                <li> <b>Transcript support level</b>
                  <p>Report the <a href="/Help/Glossary?id=492">transcript support level</a> of the overlapped
                  transcript. Equivalent to
                  <a href="/info/docs/tools/vep/script/vep_options.html#opt_tsl">--tsl</a>.</p></li>
                
                <li> <b>APPRIS</b>
                  <p>Report the <a href="/Help/Glossary?id=521">APPRIS</a> score of the overlapped
                  transcript. Equivalent to
                  <a href="/info/docs/tools/vep/script/vep_options.html#opt_appris">--appris</a>.</p>
                </li>
                
                <li><b>Identify canonical transcripts</b>
                  <p>Add a flag to the output indicating if the reported transcript is the <a
                  href="/Help/Glossary?id=346">canonical transcript</a> for the
                  gene. Equivalent to <a
                  href="/info/docs/tools/vep/script/vep_options.html#opt_canonical">--canonical</a>.</p>
                </li>
                
                <li><b>Upstream/Downstream distance (bp)</b>
                  <p>Change the distance to assign the upstream and downstream consequences.
                  Equivalent to  <a
                  href="/info/docs/tools/vep/script/vep_options.html#opt_distance">--distance</a>.</p>
                </li>

                <li><b>miRNA structure</b>
                  <p>Determines where in the secondary structure of a miRNA a variant falls (only for Ensembl/GENCODE transcripts).
                  Equivalent to <a
                  href="/info/docs/tools/vep/script/vep_options.html#opt_mirna">--mirna</a>.</p>
		            </li>

                <li><b>NMD</b>
                  <p>Predicts if a stop_gained variant allows the transcript to escape nonsense-mediated 
                  mRNA decay based on certain rules. This functionality is provided by the 
                  <a href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/NMD.pm">NMD</a> plugin.</p>
                </li>

                <li><b>UTRAnnotator</b>
                  <p>Annotates the effect of 5' UTR variant especially for
                  variant creating/disrupting upstream ORFs. This functionality
                  is provided by the 
                  <a href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/UTRAnnotator.pm">UTRAnnotator</a> plugin.</p>
                </li>

                <li><b>RiboseqORFs</b>
                  <p>Annotates the consequences of variants overlapping Ribo-seq ORFs.
                  This functionality is provided by the
                  <a href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/RiboseqORFs.pm">RiboseqORFs</a> plugin.</p>
                </li>

                <hr />
                
                <li><b>Protein matches</b>
                  <p>Shows the variant location on PDBe and AlphaFold protein structures in interactive 3d displays, where available.
                  Report <a href="/Help/Glossary?id=18">protein domains</a> from <a href="https://www.ebi.ac.uk/pdbe/" rel="external">PDBe</a>, <a
                  href="https://www.ebi.ac.uk/interpro" rel="external">Pfam</a>, <a
                  href="http://prosite.expasy.org/" rel="external">Prosite</a> and
                  <a href="http://www.ebi.ac.uk/interpro/"
                  rel="external">InterPro</a> that overlap input variants.
                  Equivalent to <a
                  href="/info/docs/tools/vep/script/vep_options.html#opt_domains">--domains</a>.</p>
                </li>

                <li><b>mutfunc</b>
                  <p>
                    <a rel="external" href="http://www.mutfunc.com">mutfunc</a>
                    predicts destabilization effect of protein structure,
                    interaction, regulatory region, etc. caused by a variant.
                    This functionality is provided by the
                    <a href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/mutfunc.pm">mutfunc</a> plugin.</p>
                  </p>
                </li>

                <hr />

                <li><b>IntAct</b>
                  <p>Reports relevant data for variants that falls within molecular interaction site as reprted by
                  <a href="https://www.ebi.ac.uk/intact/home">IntAct</a> database. This functionality is provided by the
                  <a href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/IntAct.pm">IntAct</a> plugin.</p>
                </li>

                <li><b>MaveDB</b>
                  <p><a rel="external" href="https://mavedb.org">MaveDB</a>
                  holds experimentally determined measures of variant effect.
                  This functionality is provided by the
                  <a href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/MaveDB.pm">MaveDB</a> plugin.</p>
                </li>

                <hr />
                
                <li><b>Get regulatory region consequences</b>
                  <p>In addition to predicting consequences with overlapping transcripts, Ensembl VEP
                  can find overlaps with known regulatory regions as determined in
                  the <a href="/info/genome/funcgen/index.html">Ensembl Regulatory build</a>.<br />
                  Using this option, Ensembl VEP will also report if a
                  variant falls in a transcription factor binding motif, and give
                  a score that reflects whether the altered motif sequence is more
                  or less similar to the consensus.<br />
                  Get regulatory consequences is equivalent to <a
                  href="/info/docs/tools/vep/script/vep_options.html#opt_regulatory">--regulatory</a>.</p>
                </li>

                <hr />

                <li><b>Enformer</b>
                  <p>Predictions of variant impact on gene expression.
                  This functionality is provided by the  <a rel="external" 
                  href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/Enformer.pm">Enformer</a> plugin.</p>
                </li>

                <hr />

                <li><b>Phenotypes</b>
                  <p>
                    Report the phenotypic data overlapping the genomic features. This functionality is provided by the 
                    <a href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/Phenotypes.pm">Phenotypes</a> plugin.<br />
                    For more information on the imported phenotypic data for genes, variation and QTLs see 
                    <a href="/info/genome/variation/phenotype/sources_phenotype_documentation.html">our phenotype documentation</a>.<br />
                    <b>Note:</b> This web functionality is not reporting cancer phenotypic data this release. However the cancer phenotypic data is available in the command line version.
                  </p>
                </li>

                <li><b>Gene Ontology</b>
                  <p>
                    Add terms to describe any overlapping gene's function, the cellular component in which the function is performed and the biological processes to which this contributes.
                  </p>
                </li>

                <li><b>Geno2MP</b>
                  <p>
                    <a rel="external" href="https://geno2mp.gs.washington.edu">Geno2MP</a>
                    is a web-accessible database of rare variant genotypes
                    linked to individual-level phenotypic profiles defined by
                    human phenotype ontology (HPO) terms.
                    This functionality is provided by the
                    <a href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/Geno2MP.pm">Geno2MP</a> plugin.</p>
                  </p>
                </li>

                <li><b>Mastermind</b>
                  <p>
                    Uses the <a href="https://www.genomenon.com/mastermind" rel="external">Mastermind Genomic Search Engine</a> to report variants that have clinical evidence cited in the medical literature. This functionality is provided by the
                    <a href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/Mastermind.pm">Mastermind</a> plugin.<br />
                    <b>Note:</b> This web functionality is only reporting the URL to the Mastermind Genomic Search Engine webpage.
                  </p>
                </li>

            </ul>
        </div>
    </div>
      
    <div class="column-two">
      <div class="column-right" style="padding-left:2em">
        <form style="scale: 0.8; transform-origin: top left; pointer-events: none; width: 150%" class="_tool_form _check check vep-form bgcolour"><fieldset class="extra-options-fieldset" style="margin: 0; padding: 8px">
          <div class="extra_configs_wrapper" style="margin: 0"><div class="extra_configs_button"><a class="toggle _slide_toggle set_cookie open" rel="__togg_Additional_annotations" href="#Additional_annotations">Additional annotations</a><span class="extra_configs_info">Addtional transcript, protein and regulatory annotations</span></div><div class="hidden toggleable extra_configs __togg_Additional_annotations" style="display: block;">
            <fieldset>
              <h2>Gene tolerance to change</h2>
              <div class="form-field"><label class="ff-label"><span class="ht _ht">DosageSensitivity</span>:</label><div class="ff-right"><p class="ff-checklist"><input type="checkbox"></p></div></div>
              <div class="form-field"><label class="ff-label"><span class="ht _ht">LOEUF</span>:</label><div class="ff-right"><p class="ff-checklist"><input type="checkbox"></p></div></div>
            </fieldset>

            <fieldset>
              <h2>Transcript annotation</h2>
              <div class="form-field"><label class="ff-label"><span class="ht _ht">Transcript biotype</span>:</label><div class="ff-right"><p class="ff-checklist"><input checked="checked" type="checkbox"></p></div></div>
              <div class="form-field"><label class="ff-label"><span class="ht _ht">Exon and intron numbers</span>:</label><div class="ff-right"><p class="ff-checklist"><input type="checkbox"></p></div></div>
              <div class="form-field" style="display: block;"><label class="ff-label">Transcript support level:</label><div class="ff-right"><p class="ff-checklist"><input type="checkbox" checked="checked"></p></div></div>
              <div class="form-field" style="display: block;"><label class="ff-label"><span class="_ht ht">APPRIS</span>:</label><div class="ff-right"><p class="ff-checklist"><input type="checkbox" checked="checked"></p></div></div>
              <div class="form-field" style="display: block;"><label class="ff-label"><span class="_ht ht">MANE</span>:</label><div class="ff-right"><p class="ff-checklist"><input type="checkbox" checked="checked"></p></div></div>
              <div class="_stt_merged form-field _stt_core _stt_gencode_basic _stt_gencode_primary " style="display: block;"><label class="ff-label">Identify canonical transcripts:</label><div class="ff-right"><p class="ff-checklist"><input type="checkbox"></p></div></div>
              <div class="form-field"><label class="ff-label"><span class="ht _ht">Upstream/Downstream distance (bp)</span>:</label><div class="ff-right"><input value="5000" class="optional _string ftext" type="text"></div></div>
              <div class="form-field"><label class="ff-label"><span class="_ht ht">miRNA structure</span>:</label><div class="ff-right"><p class="ff-checklist"><input type="checkbox" ></p></div></div>
              <div class="form-field"><label class="ff-label"><span class="ht _ht">NMD</span>:</label><div class="ff-right"><p class="ff-checklist"><input type="checkbox" class="_stt plugin_enable"></p></div></div>
              <div class="form-field"><label class="ff-label"><span class="ht _ht">UTRAnnotator</span>:</label><div class="ff-right"><p class="ff-checklist"><input type="checkbox" class="plugin_enable _stt"></p></div></div>
              <div class="form-field"><label class="ff-label"><span class="ht _ht">RiboseqORFs</span>:</label><div class="ff-right"><p class="ff-checklist"><input type="checkbox" class="plugin_enable _stt"></p></div></div>
            </fieldset>

            <fieldset>
              <h2>Protein annotation</h2>
              <div class="_stt_merged form-field _stt_core _stt_gencode_basic _stt_gencode_primary" style="display: block;"><label class="ff-label"><span class="ht _ht">Protein matches</span>:</label><div class="ff-right"><p class="ff-checklist"><input type="checkbox"></p></div></div>
              <div class="_stt_Mus_musculus form-field _stt_Homo_sapiens" style="display: block;"><label class="ff-label"><span class="_ht ht">mutfunc</span>:</label><div class="ff-right"><p class="ff-checklist"><input type="radio" checked="checked" class="_stt plugin_enable"><label class="ff-checklist-label">Disabled</label></p><p class="ff-checklist"><input type="radio" class="plugin_enable _stt"><label class="ff-checklist-label">Enabled</label></p></div></div>
            </fieldset>

            <fieldset>
              <h2>Functional effect</h2>
              <div class="_stt_Mus_musculus form-field _stt_Homo_sapiens" style="display: block;"><label class="ff-label"><span class="_ht ht">IntAct</span>:</label><div class="ff-right"><p class="ff-checklist"><input type="radio" checked="checked" class="_stt plugin_enable"><label class="ff-checklist-label">Disabled</label></p><p class="ff-checklist"><input type="radio" class="plugin_enable _stt"><label class="ff-checklist-label">Enabled</label></p></div></div>
              <div class="_stt_merged form-field _stt_core _stt_gencode_basic _stt_gencode_primary" style="display: block;"><label class="ff-label"><span class="ht _ht">MaveDB</span>:</label><div class="ff-right"><p class="ff-checklist"><input type="checkbox"></p></div></div>
            </fieldset>

            <fieldset>
              <h2>Regulatory data</h2>
              <div class="form-field _stt_var" style="display: block;"><label class="ff-label"><span class="ht _ht">Get regulatory region consequences</span>:</label><div class="ff-right"><select class="_stt fselect"><option selected="selected">Yes</option></select></div></div>
            </fieldset>

            <fieldset class="_stt_Homo_sapiens">
              <h2>Regulatory impact</h2>
              <div class="_stt_Homo_sapiens form-field"><label class="ff-label"><span class="ht _ht">Enformer</span>:</label><div class="ff-right"><p class="ff-checklist"><input class="plugin_enable _stt" type="checkbox"></p></div></div>
            </fieldset>
                                    
            <fieldset style="display: block;">
              <h2>Phenotype data and citations</h2>
              <div class="form-field"><label class="ff-label"><span class="_ht ht">Phenotypes</span>:</label><div class="ff-right"><p class="ff-checklist"><input class="plugin_enable _stt" type="checkbox"></p></div></div>
              <div class="form-field"><label class="ff-label"><span class="_ht ht">Gene Ontology</span>:</label><div class="ff-right"><p class="ff-checklist"><input class="_stt plugin_enable" type="checkbox"></p></div></div>
              <div class="form-field"><label class="ff-label"><span class="ht _ht">Mastermind</span>:</label><div class="ff-right"><p class="ff-checklist"><input type="checkbox" class="plugin_enable _stt"></p></div></div></fieldset>
            </div>
          </div>
        </fieldset></form>
      </div>
    </div>
</div>


<hr/>
<h2 id="prediction">Predictions</h2>

<div class="column-wrapper">
    <div class="column-two">
        <div class="column-left">
            <ul>
                <li><b>SIFT predictions</b>
                  <p><a rel="external" href="http://sift.bii.a-star.edu.sg/">SIFT</a> predicts whether an amino
                  acid substitution affects protein function based on sequence
                  homology and the physical properties of amino acids. Only
                  available in popular species. For both SIFT and PolyPhen Ensembl VEP
                  can report either a score between 0 and 1, a prediction in
                  words, or both. Equivalent to <a
                  href="/info/docs/tools/vep/script/vep_options.html#opt_sift">--sift</a>.</p>
                </li>
                
                <li><b>PolyPhen predictions</b>
                  <p><a rel="external" href="http://genetics.bwh.harvard.edu/pph2/">PolyPhen</a> is a
                  tool which predicts possible impact of an amino acid
                  substitution on the structure and function of a human protein
                  using straightforward physical and comparative considerations.
                  Equivalent to <a
                  href="/info/docs/tools/vep/script/vep_options.html#opt_polyphen">--polyphen</a>.</p>
                </li>

                <li><b>dbNSFP</b>
                  <p>Retrieves data for missense variants from <a rel="external" href="https://sites.google.com/site/jpopgen/dbNSFP">dbNSFP</a>.
                  Equivalent to the Ensembl VEP plugin <a rel="external" 
                  href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/dbNSFP.pm">dbNSFP</a>.</p>
                </li>

                <li><b>AlphaMissense</b>
                  <p>
                    <a rel="external" href="https://www.science.org/doi/10.1126/science.adg7492">AlphaMissense</a> is a deep learning
                    model developed by Google DeepMind that predicts the pathogenicity of single nucleotide missense variants.
                    Equivalent to the Ensembl VEP plugin <a rel="external"
                    href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/AlphaMissense.pm">AlphaMissense</a>.
                  </p>
                </li>

                <li><b>CADD</b>
                  <p>
                    <a rel="external" href="https://cadd.gs.washington.edu">Combined Annotation Dependent Depletion (CADD)</a> is a tool for
                    scoring the deleteriousness of single nucleotide variants
                    and insertion/deletion variants in the human genome. CADD
                    integrates multiple annotations into one metric by
                    contrasting variants that survived natural selection with
                    simulated mutations. CADD is only available here for
                    non-commercial use. See
                    <a rel="external" href="https://cadd.gs.washington.edu">CADD website</a> for more information.
                    Equivalent to the Ensembl VEP plugin <a rel="external" 
                  href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/CADD.pm">CADD</a>.
                  </p>
                </li>

                <li><b>REVEL</b>
                  <p>
                    <a rel="external" href="https://sites.google.com/site/revelgenomics">Rare Exome Variant Ensemble Learner (REVEL)</a> is an ensemble method for predicting the pathogenicity of missense variants based on a combination of scores from multiple individual tools. REVEL is only available here for
                    non-commercial use.
                    Equivalent to the Ensembl VEP plugin <a rel="external" 
                  href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/REVEL.pm">REVEL</a>.
                  </p>
                </li>

                <li><b>ClinPred</b>
                  <p>
                    <a rel="external" href="https://sites.google.com/site/clinpred">ClinPred</a> is a prediction tool to identify disease-relevant nonsynonymous single nucleotide variants. The predictor incorporates existing pathogenicity scores and benefits from normal population allele frequencies. ClinPred is only available here for
                    non-commercial use.
                    Equivalent to the Ensembl VEP plugin <a rel="external" 
                  href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/ClinPred.pm">ClinPred</a>.
                  </p>
                </li>

                <li><b>EVE</b>
                  <p>Adds information from <a rel="external"
                  href="https://evemodel.org/">EVE</a> (evolutionary model of variant effect).
                  Equivalent to the Ensembl VEP plugin <a rel="external"
                  href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/EVE.pm">EVE</a>.</p>
                </li>

                <hr />
                
                <li><b>dbscSNV</b>
                  <p>Retrieves data for splicing variants from <a rel="external" href="https://sites.google.com/site/jpopgen/dbNSFP">dbscSNV</a>.
                  Equivalent to the Ensembl VEP plugin <a rel="external" 
                  href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/dbscSNV.pm">dbscSNV</a>.</p>
                </li>
                
                <li><b>MaxEntScan</b>
                  <p>Get splice site predictions from <a rel="external" href="http://genes.mit.edu/burgelab/maxent/Xmaxentscan_scoreseq.html">MaxEntScan</a>.
                  Equivalent to the Ensembl VEP plugin <a rel="external" 
                  href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/MaxEntScan.pm">MaxEntScan</a>.</p>
                </li>

                <li><b>SpliceAI</b>
                  <p>
                    Pre-calculated annotations from <a href="https://github.com/Illumina/SpliceAI" rel="external">SpliceAI</a> a deep neural network,
                    developed by Illumina, Inc that predicts splice junctions from an arbitrary pre-mRNA transcript sequence. Used for non-commercial purposes.
                    This functionality is provided by the
                    <a href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/SpliceAI.pm">SpliceAI</a> plugin.<br />
                    The pre-calculated annotations for all possible single nucleotide substitutions can be retrieved from:
                    <ul>
                        <li>Ensembl/GENCODE v24 canonical transcripts
                          <p>Masked scores</p>
                        </li>
                        <li>Ensembl/GENCODE v37 MANE transcripts
                          <p>Raw scores</p>
                        </li>
                    </ul>
                  <b>Note:</b> The pre-calculated annotations for 1 base insertions, and 1-4 base deletions are only available for Ensembl/GENCODE v24 canonical transcripts.
                  </p>
                </li>

                <hr />
                
                <li><b>BLOSUM62</b>
                  <p>Looks up the BLOSUM 62 substitution matrix score for the reference and alternative 
                  amino acids predicted for a missense mutation.
                  Equivalent to the Ensembl VEP plugin <a rel="external" 
                  href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/Blosum62.pm">Blosum62</a>.</p>
                </li>

                
                <li><b>Ancestral allele</b>
                  <p>Retrieves ancestral allele sequences from a FASTA file. Ensembl produces <a rel="external" href="[[SPECIESDEFS::ENSEMBL_FTP_URL]]/current_fasta/ancestral_alleles/">FASTA file dumps</a> of the ancestral sequences of key species.
                  Equivalent to the Ensembl VEP plugin <a rel="external" 
                  href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/AncestralAllele.pm">AncestralAllele</a>.</p>
                </li>


            </ul>
        </div>
    </div>
      
    <div class="column-two">
      <div class="column-right" style="padding-left:2em">
        <form style="scale: 0.8; transform-origin: top left; pointer-events: none; width: 150%" class="_tool_form _check check vep-form bgcolour"><fieldset class="extra-options-fieldset" style="margin: 0; padding: 8px">
          <div class="extra_configs_wrapper" style="margin: 0"><div class="extra_configs_button"><a href="#Predictions" rel="__togg_Predictions" class="set_cookie toggle _slide_toggle open">Predictions</a><span class="extra_configs_info">Variant predictions, e.g. SIFT, PolyPhen</span></div><div class="extra_configs toggleable __togg_Predictions hidden" style="display: block;">
            <fieldset class="_stt_pphn _stt_sift" style="display: block;">
              <h2>Pathogenicity predictions</h2><input type="hidden">
              <div class="_stt_sift form-field" style="display: block;"><label class="ff-label"><span class="_ht ht">SIFT</span>:</label><div class="ff-right"><select class="fselect"><option selected="selected">Prediction and score</option></select></div></div>
              <div class="_stt_pphn form-field"><label class="ff-label"><span class="ht _ht">PolyPhen</span>:</label><div class="ff-right"><select class="fselect"><option selected="selected">Prediction and score</option></select></div></div>
              <div class="_stt_Homo_sapiens form-field"><label class="ff-label"><span class="_ht ht">dbNSFP</span>:</label><div class="ff-right"><p class="ff-checklist"><input class="_stt plugin_enable" type="radio" checked="checked"><label class="ff-checklist-label">Disabled</label></p><p class="ff-checklist"><input type="radio" class="plugin_enable _stt"><label class="ff-checklist-label">Enabled</label></p></div></div><div class="_stt_plugin_dbNSFP ff-multi form-field" style="display: none;"><label class="ff-label"><span class="_ht ht">Fields to include</span>:</label><div class="ff-right"><select class="fselect" multiple="multiple" style="height:150px;"><option>aapos</option></select></div><div class="ff-notes">Field descriptions in <a rel="external" href="#">dbNSFP README</a></div></div><div class="form-field _stt_plugin_dbNSFP" style="display: none;"><label class="ff-label"><span class="_ht ht">Transcript match</span>:</label><div class="ff-right"><p class="ff-checklist"><input type="checkbox"></p></div></div>
              <div class="form-field _stt_Homo_sapiens"><label class="ff-label"><span class="_ht ht">AlphaMissense</span>:</label><div class="ff-right"><p class="ff-checklist"><input class="plugin_enable _stt" type="checkbox"></p></div></div>
              <div class="form-field _stt_Homo_sapiens"><label class="ff-label"><span class="_ht ht">CADD</span>:</label><div class="ff-right"><p class="ff-checklist"><input class="plugin_enable _stt" type="checkbox"></p></div></div>
              <div class="form-field _stt_Homo_sapiens"><label class="ff-label"><span class="_ht ht">REVEL</span>:</label><div class="ff-right"><p class="ff-checklist"><input class="plugin_enable _stt" type="checkbox"></p></div></div>
              <div class="form-field _stt_Homo_sapiens"><label class="ff-label"><span class="_ht ht">ClinPred</span>:</label><div class="ff-right"><p class="ff-checklist"><input class="plugin_enable _stt" type="checkbox"></p></div></div>
              <div class="_stt_Homo_sapiens form-field"><label class="ff-label"><span class="ht _ht">EVE</span>:</label><div class="ff-right"><p class="ff-checklist"><input class="plugin_enable _stt" type="checkbox"></p></div></div>
            </fieldset>
            <fieldset class="_stt_Homo_sapiens">
              <h2>Splicing predictions</h2>
              <div class="_stt_Homo_sapiens form-field"><label class="ff-label"><span class="ht _ht">dbscSNV</span>:</label><div class="ff-right"><p class="ff-checklist"><input class="plugin_enable _stt" type="checkbox"></p></div></div>
              <div class="form-field _stt_Homo_sapiens"><label class="ff-label"><span class="ht _ht">MaxEntScan</span>:</label><div class="ff-right"><p class="ff-checklist"><input class="_stt plugin_enable" type="checkbox"></p></div></div>
              <div class="form-field"><label class="ff-label"><span class="_ht ht">SpliceAI</span>:</label><div class="ff-right"><p class="ff-checklist"><input checked="checked" class="_stt plugin_enable" type="radio"><label class="ff-checklist-label">Disabled</label></p><p class="ff-checklist"><input type="radio" class="plugin_enable _stt"><label class="ff-checklist-label">Enabled</label></p></div></div><div class="form-field _stt_plugin_SpliceAI" style="display: none;"><label class="ff-label">Select analysis mode:</label><div class="ff-right"><select class="fselect"><option selected="selected">Masked scores (Ensembl/GENCODE v24 canonical transcripts)</option><option>Raw scores (Ensembl/GENCODE v37 MANE transcripts, SNV only)</option></select></div></div>
            </fieldset>
            <fieldset>
              <h2>Conservation</h2>
              <div class="form-field"><label class="ff-label"><span class="_ht ht">BLOSUM62</span>:</label><div class="ff-right"><p class="ff-checklist"><input type="checkbox" class="plugin_enable _stt"></p></div></div>
              <div class="form-field _stt_Homo_sapiens"><label class="ff-label"><span class="ht _ht">Ancestral allele</span>:</label><div class="ff-right"><p class="ff-checklist"><input class="plugin_enable _stt" type="checkbox"></p></div></div>
            </fieldset>
          </div></div>
        </form>
      </div>
    </div>
</div>


<hr/>
<h2 id="filter">Filtering options</h2>

<div class="column-wrapper">
    <div class="column-two">
        <div class="column-left">
            Ensembl VEP allows you to pre-filter your results e.g. by MAF or
            consequence type. Note that it is also possible to perform
            equivalent operations on the results page for Ensembl VEP, so if you
            aren't sure, don't use any of these options! 
            <ul>
                <li><b>By frequency</b>
                  <p>Filter variants by minor allele frequency (MAF). Two options are provided:
                    <ul>
                        <li><b>Exclude common variants</b>
                          <p>Filter out variants that are co-located with an existing variant that has a
                          frequency greater than 0.01 (1%) in the 1000 Genomes
                          global population. Equivalent to <a
                          href="/info/docs/tools/vep/script/vep_options.html#opt_filter_common">--filter_common</a>
                          in the Ensembl VEP script.</p>
                        </li>
                        <li><b>Advanced filtering</b>
                          <p>Enabling this option allows you to specify a population and frequency to
                          compare to, as well whether matching variants should be
                          included or excluded from the results.</p>
                        </li>
                    </ul>
                   </p>
                </li>
                
                <li><b>Return results for variants in coding regions only</b>
                  <p>Exclude variants that don't fall in a coding region of a
                  transcript. Equivalent to <a
                  href="/info/docs/tools/vep/script/vep_options.html#opt_coding_only">--coding_only</a>.</p>
                </li>
                
                <li><b>Restrict results</b>
                  <p>For many variants Ensembl VEP will report multiple consequence types - typically this is because
                  the variant overlaps more than one transcript. For each of these
                  options Ensembl VEP uses consequence ranks that are subjectively
                  determined by Ensembl. <a
                  href="/info/genome/variation/prediction/predicted_data.html#consequence_type_table">This
                  table</a> gives all of the consquence types predicted by
                  Ensembl, ordered by rank. Note that enabling one of these
                  options not only loses potentially relevant data, but in some
                  cases may be scientifically misleading. Options:
                    <ul>
                        <li><b>Show one selected consequence</b>
                        <p>Pick one consequence type across all those predicted for the
                          variant; the output will include transcript- or
                          feature-specific information. Consequences are chosen by
                          the canonical, biotype status and length of the
                          transcript, along with the ranking of the consequence
                          type according to <a
                          href="/info/genome/variation/prediction/predicted_data.html#consequence_type_table">this
                          table</a>. This is the best method to use if you are
                          interested only in one consequence per variant.
                          Equivalent to <a
                          href="/info/docs/tools/vep/script/vep_options.html#opt_per_gene">--pick</a>.</p>
                        </li>
                        <li><b>Show one selected consequence per gene</b>
                          <p>Pick one consequence type for each gene using the same
                          criteria as above. Note that if a variant overlaps more
                          than one gene, output for each gene will be reported.
                          Equivalent to <a
                          href="/info/docs/tools/vep/script/vep_options.html#opt_per_gene">--per_gene</a>.</p>
                        </li>
                        <li><b>Show only list of consequences per variant</b>
                          <p>Give a comma-separated list of all observed consequence
                          types for each variant. No transcript-specific or
                          gene-specific output will be given. Equivalent to <a
                          href="/info/docs/tools/vep/script/vep_options.html#opt_summary">--summary</a>.</p>
                        </li>
                        <li><b>Show most severe per variant</b>
                          <p>Only the most severe of all observed consequence types is reported for
                          each variant. No transcript-specific or gene-specific
                          output will be given. Equivalent to <a
                          href="/info/docs/tools/vep/script/vep_options.html#opt_most_severe">--most_severe</a>.</p>
                        </li>
                    </ul>
                </li>
            </ul>
        </div>
    </div>
      
    <div class="column-two">
      <div class="column-right" style="padding-left:2em">
        <form style="scale: 0.8; transform-origin: top left; pointer-events: none; width: 150%" class="_tool_form _check check vep-form bgcolour"><fieldset class="extra-options-fieldset" style="margin: 0; padding: 8px">
          <div class="extra_configs_wrapper" style="margin: 0"><div class="extra_configs_button"><a class="set_cookie _slide_toggle toggle open" rel="__togg_Filtering_options" href="#Filtering_options">Filtering options</a><span class="extra_configs_info">Pre-filter results by frequency or consequence type</span></div><div class="extra_configs toggleable hidden __togg_Filtering_options" style="display: block;">
            <fieldset>
              <h2>Filters</h2>
              <div class="form-field ff-multi _stt_Homo_sapiens"><label class="ff-label"><span class="_ht ht">Filter by frequency</span>:</label><div class="ff-right"><p class="ff-checklist"><input type="radio" class="_stt"><label class="ff-checklist-label">No filtering</label></p><p class="ff-checklist"><input type="radio" class="_stt"><label class="ff-checklist-label">Exclude common variants</label></p><p class="ff-checklist"><input class="_stt" type="radio" checked="checked"><label class="ff-checklist-label">Advanced filtering</label></p></div><div class="ff-inline ff-right _stt_advanced"><select class="fselect"><option selected="selected">Exclude</option><option>Include only</option></select><select class="fselect"><option selected="selected">variants with MAF greater than</option><option>variants with MAF less than</option></select><input value="0.01" class="ftext optional _string" type="text"><select class="fselect"><option selected="selected">in 1000 genomes (1KG) combined population</option></select></div></div>
              <div class="form-field"><label class="ff-label"><span class="_ht ht">Return results for variants in coding regions only</span>:</label><div class="ff-right"><p class="ff-checklist"><input type="checkbox"></p></div></div>
              <div class="form-field ff-multi"><label class="ff-label"><span class="ht _ht">Restrict results</span>:</label><div class="ff-right"><select class="fselect"><option selected="selected">Show all results</option></select></div><div class="ff-notes"><b>NB:</b> Restricting results may exclude biologically important data!</div></div>
            </fieldset>
          </div></div>
        </fieldset></form>
      </div>
    </div>
</div>


<hr/>
<h2 id="advanced">Advanced options</h2>

<div class="column-wrapper">
    <div class="column-two">
        <div class="column-left">
          <p>The Ensembl VEP web interface allows you to use/setup advanced options:</p>
          <ul>
            <li><b>Buffer size</b>
              <p>By default Ensembl VEP process the variants by blocks of 5000 (i.e. what we call "buffer size").<br />
In some cases, reducing the size of the blocks (buffer size) could prevent memory issues for large Ensembl VEP queries (e.g. use of regulatory data, many plugins or custom annotations).<br />
This is why the maximum buffer size is automatically set to 500 on the Ensembl VEP Web interface when the "Regulatory data" option is selected.</p></li>

<li><b>Right align variants prior to consequence calculation</b>
  <p>By default Ensembl VEP performs consequence calculation at the given input coordinates.<br />
Optionally, Ensembl VEP can shift insertions and deletions found within repeated regions as far as possible in the 3' direction, normalising output.
</ul>
        </div>
    </div>

    <div class="column-two" style="min-height: 250px;">
      <div class="column-right" style="padding-left:2em">
        <form style="scale: 0.8; transform-origin: top left; pointer-events: none; width: 150%" class="_tool_form _check check vep-form bgcolour"><fieldset class="extra-options-fieldset" style="margin: 0; padding: 8px">
          <div class="extra_configs_wrapper" style="margin: 0"><div class="extra_configs_button"><a href="#Advanced_options" rel="__togg_Advanced_options" class="toggle _slide_toggle set_cookie open">Advanced options</a><span class="extra_configs_info">Additional enhancements</span></div><div class="__togg_Advanced_options toggleable hidden extra_configs" style="display: block;">
            <fieldset>
              <h2>Advanced options</h2>
              <div class="ff-multi form-field"><label class="ff-label"><span class="_ht ht">Buffer size</span>:</label><div class="ff-right"><select class="fselect"><option selected="selected">5000</option></select></div><div class="ff-notes"><div class="_stt_Homo_sapiens _stt_Mus_musculus" style="display: block;"><b>NB:</b> When the <b>Regulatory data</b> option is selected then due to the large amount of regulatory data available, the <b>maximum buffer size</b> is automatically reduced from the default value of <b>5000</b> to <b>500</b>. This reduces the memory requirement but might increase the run time. If you find that your jobs are still failing due to memory limitations then you can select a value <b>lower than 500</b>.</div></div></div>
              <div class="form-field"><label class="ff-label"><span class="ht _ht">Right align variants prior to consequence calculation</span>:</label><div class="ff-right"><select class="fselect"><option selected="selected">Right align relative to transcript</option></select></div></div>
            </fieldset>
          </div></div>
        </fieldset></form>
      </div>
    </div>
</div>


<hr/>
<h2 id="jobs">Jobs</h2>

<div class="column-wrapper">
    <div class="column-two">
      <div class="column-left">
          <p>Once you have clicked <b>Run</b>, your input will be checked
          and submitted to the Ensembl VEP as a job. All jobs associated with your
          session or account are shown in the <b>Recent Tickets</b> table.
          You may submit multiple jobs simultaneously.</p> <p>The
          <b>Jobs</b> column of the table shows the current status of the
          job.</p>

          <ul>
            <li><span class="_ht job-status-submitted job-status"><a>Queued</a></span>
            - your job is waiting to be submitted to the system</li>
            <li><span class="_ht job-status-running job-status"><a>Running</a></span>
            - your job is currently running</li>
            <li><span class="_ht job-status-done job-status"><a>Done</a></span>
            - your job is finished - click the [View results]
            link to be taken to the results page</li>
            <li><span class="_ht job-status-failed job-status"><a>Failed</a></span>
            - there is a problem with your job - click the magnifying glass icon
            <img src="/i/16/search.png"> to see more details</li>
          </ul>

          The following actions are available for each job:
          <ul>
            <li><b><img src="/i/16/save.png"> Save icon:</b> save the job (you
            need to login with an Ensembl account).</li>
            <li><b><img src="/i/16/pencil.png"> Edit icon:</b> resubmit a job
            (for example, to slightly tweak the data or parameters before
            re-running).</li>
            <li><b><img src="/i/16/search.png"> Magnifying glass icon:</b> see
            summary of the options that you selected for your Ensembl VEP job, as well
            as data versions associated with this run.</li>
            <li><b><img src="/i/16/share.png"> Share icon:</b> display URL to
            share with other users. You can also disable URL sharing here.</li>
            <li><b><img src="/i/16/trash.png"> Trash can icon:</b> delete a
            job.</li>
          </ul>
      </div>
    </div>
      
    <div class="column-two">
      <div class="column-right" style="padding-left:2em">
        <div class="dataTables_wrapper" style="scale: 0.6; transform-origin: top left; pointer-events: none; width: 150%"><div class="dataTables_top"><div class="dataTables_length"><label>Show <select data-role="none" size="1"><option selected="selected">All</option></select> entries</label></div><div class="col_toggle"><div class="toggle">Show/hide columns</div></div><div class="dataTables_filter"><label><input data-role="none" type="text"><div class="overlay">Filter</div></label></div><div class="invisible"></div></div><table class="autocenter ss data_table" style="width: 100%" cellpadding="0" cellspacing="0">
            <tr class="ss_header" style="white-space: nowrap; width: 100%;">
              <th class="sort_string sorting" title="Analysis" rowspan="1" colspan="1" style="white-space: normal; width: 10px;">Analysis</th>
              <th class="sort_none sorting_disabled" title="Jobs" rowspan="1" colspan="1" style="white-space: normal">Jobs</th>
              <th class="sort_numeric_hidden sorting" title="Submitted at (GMT)" rowspan="1" colspan="1" style="white-space: normal; width: 10px;">Submitted at (GMT)</th>
              <th class="sort_none sorting_disabled" rowspan="1" colspan="1" style="white-space: normal; width: 10px;"></th>
            </tr>
          <tbody>
            <tr class="_ticket_8Cca81zmCDOPs02S bg1"><td>Ensembl Variant Effect Predictor</td><td><img class="ticket_img" src="//static.ensembl.org/i/species/Bos_taurus.png" style="width:16px; height:16px;" alt="" class="_ht job-species"><span class="right-margin">Ensembl VEP analysis of pasted data in Bos_taurus</span><span class="_ht job-status-done job-status"><a>Done</a></span><a class="small left-margin">[View results]</a><span class="job-sprites" style="position: absolute"><a class="_ticket_view job-sprite _change_location"><span class="view_icon sprite _ht"></span></a><a class="_ticket_edit job-sprite _change_location"><span class="_ht edit_icon sprite"></span></a><a class="job-sprite _json_link"><span class="sprite delete_icon _ht"></span></a></span></p></td><td><span class="hidden">20230511170040</span>11/05/2023, 17:00</td><td><div class="ticket-sprites"><span class="_ht save_icon sprite sprite_disabled"></span><a class="_change_location _ticket_edit"><span class="sprite edit_icon _ht"></span></a><div class="ticket-share-icon share_icon _ticket_share hidden sprite"><form class="top-margin"><p><label><input type="checkbox" checked="checked">&nbsp;Share ticket via URL</label></p><p class="_ticket_share_url"><input class="ticket-share-input" type="text"></p></form></div><a class="_json_link"><span class="delete_icon sprite _ht"></span></a></div></td></tr>
            <tr class="_ticket_NzFhqZNWDwWVXfZJ bg2"><td>Ensembl Variant Effect Predictor</td><td><img class="_ht job-species" style="height:16px; width:16px;" src="//static.ensembl.org/i/species/Ovis_aries.png" alt=""><span class="right-margin">Ensembl VEP analysis of pasted data in Ovis_aries</span><span class="_ht job-status job-status-done"><a>Done</a></span><a class="left-margin small">[View results]</a></p></td><td><span class="hidden">20230511165532</span>11/05/2023, 16:55</td><td><div class="ticket-sprites"><span class="_ht save_icon sprite sprite_disabled"></span><a href="/Multi/Tools/VEP/Edit?db=core;tl=NzFhqZNWDwWVXfZJ" class="_change_location _ticket_edit"><span class="_ht edit_icon sprite"></span></a><div class="share_icon ticket-share-icon sprite hidden _ticket_share"><form class="top-margin"><p><label><input type="checkbox" checked="checked">&nbsp;Share ticket via URL</label></p><p class="_ticket_share_url"><input class="ticket-share-input" type="text"></p></form></div><a class="_json_link" href="/Multi/Json/Tools/VEP/delete?db=core;tl=NzFhqZNWDwWVXfZJ"><span class="delete_icon sprite _ht"></span><span class="_confirm hidden">This will delete the following job permanently: Ensembl VEP analysis of pasted data in Homo_sapiens</span></a></div></td></tr>
            <tr class="_ticket_1uyqKkUHWWUzEbyk bg1"><td>Ensembl Variant Effect Predictor</td><td><img class="_ht job-species" style="width:16px; height:16px;" src="//static.ensembl.org/i/species/Homo_sapiens.png" alt=""><span class="right-margin">Ensembl VEP analysis of pasted data in Homo_sapiens</span><span class="job-status job-status-failed _ht"><a>Failed</a></span></p></td><td><span class="hidden">20230420095705</span>11/05/2023, 16:54</td><td><div class="ticket-sprites"><span class="save_icon _ht sprite sprite_disabled"></span><a class="_ticket_edit _change_location"><span class="edit_icon _ht sprite"></span></a><div class="ticket-share-icon share_icon _ticket_share hidden sprite"><form class="top-margin"><p><label><input type="checkbox" checked="checked">&nbsp;Share ticket via URL</label></p><p class="_ticket_share_url"><input class="ticket-share-input" type="text"></p></form></div><a class="_json_link"><span class="_ht sprite delete_icon"></span></a></div></td></tr></tr>
            <tr class="_ticket_1uyqKkUHWWUzEbyk bg2"><td>Ensembl Variant Effect Predictor</td><td><img class="_ht job-species" style="width:16px; height:16px;" src="//static.ensembl.org/i/species/Homo_sapiens.png" alt=""><span class="right-margin">Ensembl VEP analysis of pasted data in Homo_sapiens</span><span class="job-status job-status-running _ht"><a>Running</a></span></p></td><td><span class="hidden">20230420095705</span>11/05/2023, 16:51</td><td><div class="ticket-sprites"><span class="save_icon _ht sprite sprite_disabled"></span><a class="_ticket_edit _change_location"><span class="edit_icon _ht sprite"></span></a><div class="ticket-share-icon share_icon _ticket_share hidden sprite"><form class="top-margin"><p><label><input type="checkbox" checked="checked">&nbsp;Share ticket via URL</label></p><p class="_ticket_share_url"><input class="ticket-share-input" type="text"></p></form></div><a class="_json_link"><span class="_ht sprite delete_icon"></span></a></div></td></tr></tr>
            <tr class="_ticket_1uyqKkUHWWUzEbyk bg1"><td>Ensembl Variant Effect Predictor</td><td><img class="_ht job-species" style="width:16px; height:16px;" src="//static.ensembl.org/i/species/Homo_sapiens.png" alt=""><span class="right-margin">Ensembl VEP analysis of pasted data in Homo_sapiens</span><span class="job-status job-status-submitted _ht"><a>Queued</a></span></p></td><td><span class="hidden">20230420095705</span>11/05/2023, 16:49</td><td><div class="ticket-sprites"><span class="save_icon _ht sprite sprite_disabled"></span><a class="_ticket_edit _change_location"><span class="edit_icon _ht sprite"></span></a><div class="ticket-share-icon share_icon _ticket_share hidden sprite"><form class="top-margin"><p><label><input type="checkbox" checked="checked">&nbsp;Share ticket via URL</label></p><p class="_ticket_share_url"><input class="ticket-share-input" type="text"></p></form></div><a class="_json_link"><span class="_ht sprite delete_icon"></span></a></div></td></tr></tr>
          </tbody>
        </table><div class="dataTables_bottom" style="display: none;"><div class="dataTables_info" style="display: none;">Showing 1 to 63 of 63 entries</div><div class="dataTables_paginate paging_full_numbers" style="display: none;"><span class="first paginate_button paginate_button_disabled">&lt;&lt;</span><span class="previous paginate_button paginate_button_disabled">&lt;</span><span></span><span class="next paginate_button paginate_button_disabled">&gt;</span><span class="last paginate_button paginate_button_disabled">&gt;&gt;</span></div><div class="invisible"></div></div></div>
      </div>
    </div>
</div>

<div style="clear:both"><h1 style="page-break-after: always">&nbsp</h1></div>    
<style>
tr:nth-child(odd) {background-color: #f0f0f0;}
</style>

  <img src="/img/vep_logo.png" class="float-right" style="width:111px;height:50px" />	
 
  <h1 id="top"><span style="color:#006;padding-right:15px">Ensembl Variant Effect Predictor</span><span style="color:#666"><img src="/i/16/copy.png"/> Data formats</span></h1>
  <hr />

    <!-- Input formats -->

    <h2 id="input">Input</h2>
    
    <p> Both the web and command line interfaces to Ensembl VEP can use the same input formats.</p>
    <p>Supported input formats:</p>
    
    <table class="ss">
      <tr>
        <th>Format</th>
        <th>Variant example</th>
        <th><a href="#sv">Structural variant example</a></th>
      </tr>
      <tr>
        <td><a href="#default">Default Ensembl VEP input</a></td>
        <td><kbd>1 881907 881906 -/C +</kbd></td>
        <td><kbd>1 160283 471362 DUP +</kbd></td>
      </tr>
      <tr>
        <td><a href="#vcf">VCF</a></td>
        <td><kbd>1 65568 . A C . . .</kbd></td>
        <td><kbd>1 7936271 . N N[12:58877476[ . . SVTYPE=BND</kbd></td>
      </tr>
      <tr>
        <td><a href="#hgvs">HGVS identifiers</a></td>
        <td><kbd>ENST00000618231.3:c.9G>C</kbd></td>
        <td style="text-align: center"><font style="color: darkgray">&#10007; Not supported</font></td>
      </tr>
      <tr>
        <td><a href="#id">Variant identifiers</a></td>
        <td><kbd>rs699</kbd></td>
        <td><kbd>nsv1000164</kbd></td>
      </tr>
      <tr>
        <td><a href="#spdi">Genomic SPDI notation</a></td>
        <td><kbd>NC_000016.10:68684738:G:A</kbd></td>
        <td style="text-align: center"><font style="color: darkgray">&#10007; Not supported</font></td>
      </tr>
      <tr>
        <td><a href="#region">REST-style regions</a></td>
        <td><kbd>14:19584687-19584687:-1/T</kbd></td>
        <td><kbd>21:25587759-25587769/DEL</kbd></td>
      </tr>
    </table>

    <div>
      <div class="info" style="float:left">
        <h3>Note</h3>
        <div class="message-pad">
          <p>Ensembl VEP can read compressed (gzipped) input file of any format listed above, e.g.:</p>
          <pre class="code sh_sh"> ./vep -i input.vcf.gz -o output.txt</pre>
        </div>
      </div>
      <div style="clear:both"></div>
    </div>
      

    <br />
    <hr /> 
    <h2 id="default"> Default Ensembl VEP input</h2>
    
    <p>The default format is a simple <strong>whitespace-separated</strong> format
    (columns may be separated by space or tab characters),
    containing five required columns plus an optional identifier column:
    </p>
    
    <ol>
        <li><strong>chromosome</strong> - just the name or number, with no 'chr'
        prefix</li>
        <li><strong>start</strong></li>
        <li><strong>end</strong></li>
        <li><strong>allele</strong> - pair of alleles separated by a '/', with the
        reference allele first (or <a href="#sv">structural variant type)</a></li>
        <li><strong>strand</strong> - defined as + (forward) or - (reverse). The alleles will be reverse complemented for mapping to the genome if the minus strand is provided.</li>
        <li><strong>identifier</strong> - this identifier will be used the
        output. If not provided, Ensembl VEP will construct an identifier from the given
        coordinates and alleles.</li>
    </ol>
    
    <pre class="code sh_sh">
1   881907    881906    -/C   +
2   946507    946507    G/C   +
5   140532    140532    T/C   +
8   150029    150029    A/T   +    var2
12  1017956   1017956   T/A   +
14  19584687  19584687  C/T   -
19  66520     66520     G/A   +    var1</pre>
    
    <p>An insertion (of any size) is indicated by start coordinate = end coordinate
    + 1. For example, an insertion of 'C' between nucleotides 12600 and 12601 on the
    forward strand of chromosome 8 is indicated as follows:</p>
    
    <pre class="code sh_sh">8   12601     12600     -/C   +</pre>
    <p>A deletion is indicated by the exact nucleotide coordinates. For example, a
    three base pair deletion of nucleotides 12600, 12601, and 12602 of the reverse
    strand of chromosome 8 will be:</p>
    
    <pre class="code sh_sh">8   12600     12602     CGT/- -</pre>
    
    <p>
      Structural variants are also supported by indicating a
      <a href="#sv">structural variant type</a> instead of the allele:
    </p>
  
    <pre class="code sh_sh">
1    20000     30000     CN4  +  cnv4
1    160283    471362    DUP  +  dup
1    1385015   1387562   DEL  +  del1
12   1017956   1017956   INV  +  inv
21   25587759  25587769  CN0  +  del2</pre>
    
    <br />
    <hr />
    <h2 id="vcf"> VCF</h2>
    
    <p> <a rel="external" href="http://www.1000genomes.org/wiki/Analysis/vcf4.0">VCF (Variant Call Format)
    version 4.0</a> is supported. This is a common format produced by many variant calling tools
    and is the recommended format for use with Ensembl VEP: </p>

  <pre class="code sh_sh">
#CHROM  POS        ID      REF  ALT            QUAL  FILTER  INFO  FORMAT
1       65568      .       A    C              .     .       .     .
1       230710048  rs699   A    G              .     .       .     .
2       265023     .       C    T              .     .       .     .
3       319780     .       GA   G              .     .       .     .
20      3          .       C    CAAG,CAAGAAG   .     PASS    .     .
21      43762120   rs1300  T    A,C,G          .     .       .     .</pre>
    
    <p>Structural variants are also supported depending on
      <a href="#sv">structural variant type.</a></p>
    
    <p> Users using VCF should note a peculiarity in the difference between how
    Ensembl and VCF describe unbalanced variants. For any unbalanced variant (i.e.
    insertion, deletion or unbalanced substitution), the VCF specification requires
    that the base immediately before the variant should be included in both the
    reference and variant alleles. This also affects the reported position i.e. the
    reported position will be one base before the actual site of the variant. </p>
    
    <p> In order to parse this correctly, Ensembl VEP needs to convert such variants
    into Ensembl-type coordinates, and it does this by removing the additional base
    and adjusting the coordinates accordingly. This means that if an identifier is
    not supplied for a variant (in the 3rd column of the VCF), then the identifier
    constructed and the position reported in Ensembl VEP output file will differ from
    the input. </p>
    
    <p> This problem can be overcome with the following: </p>
    <ol>
      <li> ensuring each variant has a unique
    identifier specified in the 3rd column of the VCF </li>
      <li> using VCF format as output (<a href="script/vep_options.html#opt_vcf">--vcf</a>) - this preserves the formatting of your input coordinates and alleles </li>
      <li> using <a href="script/vep_options.html#opt_minimal">--minimal</a> and <a href="script/vep_options.html#opt_allele_number">--allele_number</a> (see <a href="#complex_vcf">Complex VCF entries</a>).</li> 
    </ol>
    
    <p> The following examples illustrate how VCF describes a variant and how it is
    handled internally by Ensembl VEP. Consider the following aligned sequences (for the
    purposes of discussion on chromosome 20): </p>
    
    <pre class="code sh_sh">Ref: a t C g a // C is the reference base
1 : a t G g a // C base is a G in individual 1
2 : a t - g a // C base is deleted w.r.t. the reference in individual 2
3 : a t CAg a // A base is inserted w.r.t. the reference sequence in individual 3</pre>
    
    <h4> Individual 1 </h4>
    
    <p> The first individual shows a simple balanced substitution of G for C at base
    3. This is described in a compatible manner in VCF and Ensembl styles. Firstly,
    in VCF: </p>
    
    <pre class="code sh_sh">20   3   .   C   G   .   PASS   .</pre>
    
    <p> And in Ensembl format: </p>
    
    <pre class="code sh_sh"> 20   3   3   C/G   +</pre>
    
    <h4> Individual 2 </h4>
    
    <p> The second individual has the 3rd base deleted relative to the reference. In
    VCF, both the reference and variant allele columns must include the preceding
    base (T) and the reported position is that of the preceding base: </p>
    
    <pre class="code sh_sh">20   2   .   TC   T   .   PASS   .</pre>
    
    <p> In Ensembl format, the preceding base is not included, and the start/end
    coordinates represent the region of the sequence deleted. A "-" character is
    used to indicate that the base is deleted in the variant sequence: </p>
    
    <pre class="code sh_sh">20   3   3   C/-   +</pre>
    
    <p> The upshot of this is that while in the VCF input file the position of the
    variant is reported as 2, in the output file in Ensembl VEP default format the position will be
    reported as 3. If no identifier is provided in the third column of the VCF, then
    the constructed identifier will be: </p>
    
    <pre class="code sh_sh">20_3_C/-</pre>
    
    <h4> Individual 3 </h4>
    
    <p> The third individual has an "A" inserted between the 3rd and 4th bases of
    the sequence relative to the reference. In VCF, as for the deletion, the base
    before the insertion is included in both the reference and variant allele
    columns, and the reported position is that of the preceding base: </p>
    
    <pre class="code sh_sh">20   3   .   C   CA   .   PASS   .</pre>
    
    <p> In Ensembl format, again the preceding base is not included, and the
    start/end positions are "swapped" to indicate that this is an insertion.
    Similarly to a deletion, a "-" is used to indicate no sequence in the reference:
    </p>
    
    <pre class="code sh_sh"> 20   4   3   -/A   +</pre>
    
    <p> Again, the output will appear different, and the constructed identifier may
    not be what is expected: </p>
    
    <pre class="code sh_sh">20_3_-/A</pre>

    <p> Using VCF format output, or adding unique identifiers to the input (in the
    third VCF column), can mitigate this issue. </p>
   

    <br /> 
    <h3 id="complex_vcf"> Complex VCF entries </h3>

    <p> For VCF entries with multiple alternate alleles, Ensembl VEP will only trim the
    leading base from alleles if <b>all</b> REF and ALT alleles start with the same
    base: </p>

    <pre class="code sh_sh">20   3   .   C   CAAG,CAAGAAG   .   PASS   .</pre>

    <p> This will be considered internally by Ensembl VEP as equivalent to: </p>

    <pre class="code sh_sh">20   4   3   -/AAG/AAGAAG   +</pre>

    <p> Now consider the case where a single VCF line contains a representation
    of both a SNV and an insertion: </p>

    <pre class="code sh_sh">20   3   .   C   CAAAG,G   .   PASS   .</pre>

    <p> Here the input alleles will remain unchanged, and Ensembl VEP will consider the first
    REF/ALT pair as a substitution of C for CAAG, and the second as a C/G SNV:</p>

    <pre class="code sh_sh">20   3   3   C/CAAG/G   +</pre>

    <p> To modify this behaviour, with the commandline tool you can use <a href="script/vep_options.html#opt_minimal">--minimal</a>.
    This flag forces Ensembl VEP to consider each REF/ALT pair independently, trimming
    identical leading and trailing bases from each as appropriate. Since this can
    lead to confusing output regarding coordinates etc, it is not the default
    behaviour. It is recommended to use the
    <a href="script/vep_options.html#opt_allele_number">--allele_number</a> flag to track
    the correspondence between alleles as input and how they appear in the output.</p>

    <br />
    <hr />
    <h2 id="hgvs"> HGVS identifiers</h2>
    
    <p> See <a rel="external" href="//varnomen.hgvs.org">https://varnomen.hgvs.org</a> for
    details. These must be relative to genomic or Ensembl transcript coordinates.</p>
    <p>It also is possible to use RefSeq transcripts, if they match the reference genome
     See <a href="script/vep_other.html#hgvs">HGVS documentation</a>
    </p>
    
    <p> Examples: </p>
    
    <pre class="code sh_sh">ENST00000618231.3:c.9G>C
ENST00000471631.1:c.28_33delTCGCGG
ENST00000285667.3:c.1047_1048insC
5:g.140532G>C</pre>
    
    <p> Examples using RefSeq identifiers (using <a
    href="script/vep_options.html#opt_refseq">--refseq</a> in the command line or select
    the 'RefSeq transcripts' on the web interface: </p>
    
    <pre class="code sh_sh">NM_153681.2:c.7C>T
NM_005239.6:c.190G>A
NM_001025204.2:c.336G>A</pre>
    
    <p> HGVS protein notations may also be used, provided that they
    unambiguously map to a single genomic change. Due to redundancy in the amino
    acid code, it is not always possible to work out the corresponding genomic
    sequence change for a given protein sequence change. The following example
    is for a permissable protein notation in dog <i>(Canis familiaris)</i>:</p>
    
    <pre class="code sh_sh">ENSCAFP00000040171.1:p.Thr92Asn</pre>

    <h4>Ambiguous gene-based descriptions</h4>
    <p>It is possible to use ambiguous descriptions listing only gene symbol or UniProt accession and protein change (e.g. PHF21B:p.Tyr124Cys,  P01019:p.Ala268Val), as seen in the literature, though this is not recommended as it can produce multiple different variants at genomic level. The transcripts for a gene are considered in the following order:
      <ol>
        <li><a href="/info/genome/genebuild/mane.html">MANE Select transcript status</a></li>
        <li><a href="/info/genome/genebuild/mane.html">MANE Plus Clinical transcript status</a></li>
        <li>canonical status of transcript</li>
        <li><a href="/Help/Glossary?id=521">APPRIS isoform annotation</a></li>
        <li><a href="/Help/Glossary?id=492">transcript support level</a></li>
        <li>biotype of transcript ("protein_coding" preferred)</li>
        <li>CCDS status of transcript</li>
        <li>consequence rank according to <a href="/info/genome/variation/prediction/predicted_data.html#consequences">this table</a></li>
        <li>translated, transcript or feature length (longer preferred)</li>
      </ol>
      and the first compatible transcript is used to map variants to the genome for annotation.</p>

    <br />
    <hr />
    <h2 id="id"> Variant identifiers</h2>
    
    <p> These should be dbSNP rsIDs (such as <kbd>rs699</kbd>), or any synonym for a variant present in
    the Ensembl Variation database. Structural variant identifiers (like <kbd>nsv1000164</kbd> and
    <kbd>esv1850194</kbd>) are also supported.</p>

    <p> See <a
    href="/info/genome/variation/species/sources_documentation.html">here</a> for a list of
    identifier sources in Ensembl. </p>

    <p> Examples: </p>

    <pre class="code sh_sh">rs1156485833
rs1258750482
rs867704559
esv1815690
nsv1000164</pre>
   
    <div>
      <div class="info" style="float:left">
        <h3>Note</h3>
        <div class="message-pad">
          <p>Ensembl VEP is optimised for the analysis of variants in VCF files (sorted by location) or other position-based formats. Using variant identifiers can take 2-5 times longer as a database lookup to find the variant location is required.</p>
        </div>
      </div>
      <div style="clear:both"></div>
    </div> 

    <br />
    <hr />
    <h2 id="spdi"> Genomic SPDI notation</h2>

    <p> Genomic SPDI notation which uses four fields delimited by colons S:P:D:I (Sequence:Position:Deletion:Insertion) is also supported. In SPDI notation, the position refers to the base before the variant, not the base of the variant itsef.</p> 
    <p>See <a href="https://www.ncbi.nlm.nih.gov/variation/notation/">here</a> for details.</p>

    <p> Examples: </p>
    
    <pre class="code sh_sh">NC_000016.10:68684738:G:A
NC_000017.11:43092199:GCTTTT:
NC_000013.11:32315789::C
NC_000016.10:68644746:AA:GTA
16:68684738:2:AC</pre>


    <br />
    <hr />
    <h2 id="region"> REST-style regions</h2>
    <p>
      The Ensembl VEP region REST endoint requires variants are described as
      <code>[chr]:[start]-[end]:[strand]/[allele]</code>.
    </p>
    <p>
      This follows the same conventions as the
      <a href="#default">default input format</a>, with the key difference being
      that this format does not require the reference (REF) allele to be
      included; this will be looked up  using either
      a provided FASTA file (preferred) or Ensembl core database. Strand is
      optional and defaults to 1 (forward strand).
    </p>

    <pre class="code sh_sh"># SNP
5:140532-140532:1/C

# SNP (reverse strand)
14:19584687-19584687:-1/T

# insertion
1:881907-881906:1/C

# 5bp deletion
2:946507-946511:1/-</pre>

  <p>
    Structural variants are also supported by indicating a
    <a href="#sv">structural variant type</a> in the place of the
    <code>[allele]</code>:
  </p>
  <pre class="code sh_sh"># structural variant: deletion
21:25587759-25587769/DEL
    
# structural variant: inversion
21:25587759-25587769/INV</pre>

<br />
    <hr />
    <h2 id="sv"> Structural variant types</h2>

    <p>Ensembl VEP also predicts molecular consequences for structural variants using the
      following input formats:</p>
    <ul>
      <li><a href="#default">Default Ensembl VEP input</a></li>
      <li><a href="#region">REST-style regions</a></li>
      <li> <a href="#id">Variant identifiers</a></li>
      <li> <a href="#vcf">VCF</a></li>
    </ul>

    <p> To recognise a variant as a structural variant, the allele string
      (or <code>SVTYPE</code> in the INFO column of the VCF format) must be set
      to one of the currently supported values: </p>

    <ul>
        <li><b>INS</b> - insertion</li>
        <ul>
          <li><b>INS:ME</b> - insertion of mobile element</li>
          <li><b>INS:ME:ALU</b> - insertion of ALU element</li>
          <li><b>INS:ME:HERV</b> - insertion of HERV element</li>
          <li><b>INS:ME:LINE1</b> - insertion of LINE1 element</li>
          <li><b>INS:ME:SVA</b> - insertion of SVA element</li>
        </ul>

        <li><b>DEL</b> - deletion</li>
        <ul>
          <li><b>DEL:ME</b> - deletion of mobile element</li>
          <li><b>DEL:ME:ALU</b> - deletion of ALU element</li>
          <li><b>DEL:ME:HERV</b> - deletion of HERV element</li>
          <li><b>DEL:ME:LINE1</b> - deletion of LINE1 element</li>
          <li><b>DEL:ME:SVA</b> - deletion of SVA element</li>
        </ul>

        <li><b>DUP</b> - duplication</li>
        <ul>
          <li><b>DUP:TANDEM</b> - tandem duplication</li>
          <li><b>TDUP</b> - tandem duplication</li>
        </ul>
        <li><b>INV</b> - inversion</li>

        <li><b>CNV</b> - copy number variation</li>
        <ul>
          <li>
            The copy number value can be specified like so:
            <ul>
              <li><kbd>CN0</kbd></li>
              <li><kbd>CN=4</kbd></li>
              <li><kbd>CN3,CN4,CN6</kbd></li>
              <li><kbd>CN=0,CN=2,CN=4</kbd></li>
            </ul>
          </li>
          <li><b>CNV:TR</b> - tandem repeats</li>
          <ul>
            <li> Requires <code>INFO</code> fields describing the tandem repeat, such as <code>RUS</code> and <code>RN</code> â€“ check <a href="https://samtools.github.io/hts-specs/VCFv4.4.pdf" rel="external">VCF 4.4 specification, section 5.7</a>
            <li> Currently, the <code>CIRUC</code> and <code>CIRB</code> <code>INFO</code> fields are ignored when calculating alternative alleles in tandem repeats
          </ul>
        </ul>

        <li><b>BND</b> - chromosome breakpoints</li>
        <ul>
          <li> Breakpoint variants are composed by one or more breakends
          <li>
            In VCF, breakend replacements are inserted into the <code>ALT</code>
            column and need to meet the
            <a rel="external" href="http://samtools.github.io/hts-specs/">HTS specifications</a>,
            such as <kbd>TG[12:58877476[</kbd>
          </li>
          <li>Single breakends can be specified in <code>ALT</code>, such as <kbd>T.</kbd> and <kbd>.G</kbd></li>
          <li>Multiple, comma-separated alternative breakends can be specified in <code>ALT</code>, such as <kbd>A[22:22893780[,A[X:10932343[</kbd></li>
          <li>Breakends are only supported for VCF input format</li>
        </ul>
    </ul>

    <p>More information on how Ensembl VEP processes structural variants can be found <a href="script/vep_example.html#StructVar">here</a>.</p>

    <h3> Examples of structural variants encoded in VCF format </h3>

    <pre class="code sh_sh">
#CHROM  POS        ID   REF  ALT             QUAL  FILTER  INFO
1       160283     dup  .    &lt;DUP&gt;           .     .       SVTYPE=DUP;END=471362
1       1385015    del  .    &lt;DEL&gt;           .     .       SVTYPE=DEL;END=1387562
1       7936271    bnd  N    N[12:58877476[  .     .       SVTYPE=BND</pre>

    <p> See the <a
    rel="external" href="http://www.1000genomes.org/wiki/Analysis/Variant%20Call%20Format/VCF%20%28Variant%20Call%20Format%29%20version%204.0/encoding-structural-variants">VCF
    definition document</a> for more detail on how to describe
    structural variants in VCF format. </p>

    <!-- Output formats -->
    <br />
    <hr style="margin-bottom:0"/>
    <hr style="margin-top:0"/>
    <h2 id="output">Output</h2>

    <p>Ensembl VEP can return the results in different formats:</p>
    <ul>
      <li><a href="#defaultout">Default Ensembl VEP output</a></li>
      <li><a href="#tab">Tab-delimited output</a></li>
      <li><a href="#vcfout">VCF</a></li>
      <li><a href="#json">JSON output</a></li>
    </ul>
   
    <p>Along with the results Ensembl VEP computes and returns some <a href="#stats">statistics</a>.</p>


    <br />
    <hr />
    <h2 id="defaultout"> Default Ensembl VEP output</h2>
 
    <p> The default output format ("VEP" format when downloading from the web interface) is a 14 column tab-delimited file. Empty values are denoted by '-'. The output
    columns are:</p>
    
    <ol>
        <li><strong>Uploaded variation</strong> - as chromosome_start_alleles</li>
        <li><strong>Location</strong> - in standard coordinate format (chr:start or chr:start-end)</li>
        <li><strong>Allele</strong> - the variant allele used to calculate the consequence</li>
        <li><strong>Gene</strong> - Ensembl stable ID of affected gene</li>
        <li><strong>Feature</strong> - Ensembl stable ID of feature</li>
        <li><strong>Feature type</strong> - type of feature. Currently one of Transcript, RegulatoryFeature, MotifFeature.</li>
        <li><strong>Consequence</strong> - <a href="/info/genome/variation/prediction/predicted_data.html#consequences">consequence type</a> of this variant</li>
        <li><strong>Position in cDNA</strong> - relative position of base pair in cDNA sequence</li>
        <li><strong>Position in CDS</strong> - relative position of base pair in coding sequence</li>
        <li><strong>Position in protein</strong> - relative position of amino acid in protein</li>
        <li><strong>Amino acid change</strong> - only given if the variant affects the protein-coding sequence</li>
        <li><strong>Codon change</strong> - the alternative codons with the variant base in upper case</li>
        <li><strong>Co-located variation</strong> - identifier of any <a href="/info/docs/tools/vep/script/vep_other.html#colocated">existing variants</a>. Switch on with <a href="script/vep_options.html#opt_check_existing">--check_existing</a></li>
        <li><strong>Extra</strong> - this column contains extra information as key=value pairs separated by ";", see below.</li>
    </ol>


    <h3 style="margin:20px 0px 30px">Other output fields:</h3>

    <div id="div_other_fields" >
      <ul>
        <li><b>REF_ALLELE</b> - the reference allele (after minimisation)</li>
        <li><b>UPLOADED_ALLELE</b> - the uploaded allele string (before minimisation)</li>
        <li><b>IMPACT</b> - the impact modifier for the consequence type</li>
        <li><b>VARIANT_CLASS</b> - Sequence Ontology <a href="/info/genome/variation/prediction/classification.html#classes">variant class</a></li>
        <li><b>SYMBOL</b> - the gene symbol</li>
        <li><b>SYMBOL_SOURCE</b> - the source of the gene symbol</li>
        <li><b>STRAND</b> - the DNA strand (1 or -1) on which the transcript/feature lies</li>
        <li><b>ENSP</b> - the Ensembl protein identifier of the affected transcript</li>
        <li>
            <b>FLAGS</b> - transcript quality flags:
            <ul>
                <li><i>cds_start_NF:</i> CDS 5' incomplete</li>
                <li><i>cds_end_NF:</i> CDS 3' incomplete</li>
            </ul>
        </li>
        <li><b>SWISSPROT</b> - Best match UniProtKB/Swiss-Prot accession of protein product
        <li><b>TREMBL</b> - Best match UniProtKB/TrEMBL accession of protein product</li>
        <li><b>UNIPARC</b> - Best match UniParc accession of protein product</li>
        <li><b>HGVSc</b> - the HGVS coding sequence name</li>
        <li><b>HGVSp</b> - the HGVS protein sequence name</li>
        <li><b>HGVSg</b> - the HGVS genomic sequence name</li>
        <li><b>HGVS_OFFSET</b> - Indicates by how many bases the HGVS notations for this variant have been <a href="script/vep_options.html#opt_shift_hgvs">shifted</a>. Value must be greater than 0.</li>
        <li><b>NEAREST</b> - Identifier(s) of nearest transcription start site</li>
        <li><b>SIFT</b> - the SIFT prediction and/or score, with both given as prediction(score)</li>
        <li><b>PolyPhen</b> - the PolyPhen prediction and/or score</li>
        <li><b>MOTIF_NAME</b> - the source and identifier of a transcription factor binding profile aligned at this position</li>
        <li><b>MOTIF_POS</b> - The relative position of the variation in the aligned TFBP</li>
        <li><b>HIGH_INF_POS</b> - a flag indicating if the variant falls in a high information position of a transcription factor binding profile (TFBP)</li>
        <li><b>MOTIF_SCORE_CHANGE</b> - The difference in motif score of the reference and variant sequences for the TFBP</li>
        <li><b>CELL_TYPE</b> - List of cell types and classifications for regulatory feature</li>
        <li><b>CANONICAL</b> - a flag indicating if the transcript is denoted as the canonical transcript for this gene</li>
        <li><b>CCDS</b> - the CCDS identifer for this transcript, where applicable</li>
        <li><b>INTRON</b> - the intron number (out of total number)</li>
        <li><b>EXON</b> - the exon number (out of total number)</li>
        <li><b>DOMAINS</b> - the source and identifer of any overlapping protein domains</li>
        <li><b>DISTANCE</b> - Shortest distance from variant to transcript. <i>Note: DISTANCE of 0 is possible for insertions happening just before or after a transcript because variant coordinates are considered to be the flanking bases where insertion happens.</i></li>
        <li><b>IND</b> - individual name</li>
        <li><b>ZYG</b> - zygosity of individual genotype at this locus</li>
        <li><b>SV</b> - IDs of overlapping structural variants</li>
        <li><b>FREQS</b> - Frequencies of overlapping variants used in filtering</li>
        <li><b>AF</b> - Frequency of existing variant in 1000 Genomes</li>
        <li><b>AFR_AF</b> - Frequency of existing variant in 1000 Genomes combined African population</li>
        <li><b>AMR_AF</b> - Frequency of existing variant in 1000 Genomes combined American population</li>
        <li><b>ASN_AF</b> - Frequency of existing variant in 1000 Genomes combined Asian population</li>
        <li><b>EUR_AF</b> - Frequency of existing variant in 1000 Genomes combined European population</li>
        <li><b>EAS_AF</b> - Frequency of existing variant in 1000 Genomes combined East Asian population</li>
        <li><b>SAS_AF</b> - Frequency of existing variant in 1000 Genomes combined South Asian population</li>
        <li><b>gnomADe_AF</b> - Frequency of existing variant in gnomAD exomes combined population</li>
        <li><b>gnomADe_AFR_AF</b> - Frequency of existing variant in gnomAD exomes African/American population</li>
        <li><b>gnomADe_AMR_AF</b> - Frequency of existing variant in gnomAD exomes American population</li>
        <li><b>gnomADe_ASJ_AF</b> - Frequency of existing variant in gnomAD exomes Ashkenazi Jewish population</li>
        <li><b>gnomADe_EAS_AF</b> - Frequency of existing variant in gnomAD exomes East Asian population</li>
        <li><b>gnomADe_FIN_AF</b> - Frequency of existing variant in gnomAD exomes Finnish population</li>
        <li><b>gnomADg_MID_AF</b> - Frequency of existing variant in gnomAD exomes Mid-eastern population</li>
        <li><b>gnomADe_NFE_AF</b> - Frequency of existing variant in gnomAD exomes Non-Finnish European population</li>
        <li><b>gnomADe_REMAINING_AF</b> - Frequency of existing variant in gnomAD exomes combined remaining combined populations</li>
        <li><b>gnomADe_SAS_AF</b> - Frequency of existing variant in gnomAD exomes South Asian population</li>
        <li><b>gnomADg_AF</b> - Frequency of existing variant in gnomAD genomes combined population</li>
        <li><b>gnomADg_AFR_AF</b> - Frequency of existing variant in gnomAD genomes African/American population</li>
        <li><b>gnomADg_AMI_AF</b> - Frequency of existing variant in gnomAD genomes Amish population</li>
        <li><b>gnomADg_AMR_AF</b> - Frequency of existing variant in gnomAD genomes American population</li>
        <li><b>gnomADg_ASJ_AF</b> - Frequency of existing variant in gnomAD genomes Ashkenazi Jewish population</li>
        <li><b>gnomADg_EAS_AF</b> - Frequency of existing variant in gnomAD genomes East Asian population</li>
        <li><b>gnomADg_FIN_AF</b> - Frequency of existing variant in gnomAD genomes Finnish population</li>
        <li><b>gnomADg_MID_AF</b> - Frequency of existing variant in gnomAD genomes Mid-eastern population</li>
        <li><b>gnomADg_NFE_AF</b> - Frequency of existing variant in gnomAD genomes Non-Finnish European population</li>
        <li><b>gnomADg_REMAINING_AF</b> - Frequency of existing variant in gnomAD genomes combined remaining combined populations</li>
        <li><b>gnomADg_SAS_AF</b> - Frequency of existing variant in gnomAD genomes South Asian population</li>
        <li><b>MAX_AF</b> - Maximum observed allele frequency in 1000 Genomes, ESP and gnomAD</li>
        <li><b>MAX_AF_POPS</b> - Populations in which maximum allele frequency was observed</li>
        <li><b>CLIN_SIG</b> - ClinVar clinical significance of the dbSNP variant</li>
        <li><b>BIOTYPE</b> - Biotype of transcript or regulatory feature</li>
        <li><b>APPRIS</b> - Annotates alternatively spliced transcripts as primary or alternate based on a range of computational methods. NB: not available for GRCh37</li>
        <li><b>TSL</b> - Transcript support level. NB: not available for GRCh37</li>
        <li><b>GENCODE_PRIMARY</b> - Reports if transcript belongs to GENCODE primary subset</li>
        <li><b>PUBMED</b> - Pubmed ID(s) of publications that cite existing variant</li>
        <li><b>SOMATIC</b> - Somatic status of existing variant(s); multiple values correspond to multiple values in the Existing_variation field</li>
        <li><b>PHENO</b> - Indicates if existing variant is associated with a phenotype, disease or trait; multiple values correspond to multiple values in the Existing_variation field</li>
        <li><b>GENE_PHENO</b> - Indicates if overlapped gene is associated with a phenotype, disease or trait</li>
        <li><b>ALLELE_NUM</b> - Allele number from input; 0 is reference, 1 is first alternate etc</li>
        <li><b>MINIMISED</b> - Alleles in this variant have been converted to minimal representation before consequence calculation</li>
        <li><b>PICK</b> - indicates if this block of consequence data was picked by <a href="script/vep_options.html#opt_flag_pick">--flag_pick</a> or <a href="script/vep_options.html#opt_flag_pick_allele">--flag_pick_allele</a></li>
        <li><b>BAM_EDIT</b> - Indicates success or failure of edit using BAM file</li>
        <li><b>GIVEN_REF</b> - Reference allele from input</li>
        <li><b>USED_REF</b> - Reference allele as used to get consequences</li>

        <li id="refseq_match"><b>REFSEQ_MATCH</b> - the RefSeq transcript match status; contains a number of flags indicating whether this RefSeq transcript matches the underlying reference sequence and/or an Ensembl transcript (<a href="script/vep_other.html#refseq">more information</a>). 
          <ul>
            <li><i>rseq_3p_mismatch:</i> signifies a mismatch between the RefSeq transcript and the underlying primary genome assembly sequence. Specifically, there is a mismatch in the 3' UTR of the RefSeq model with respect to the primary genome assembly (e.g. GRCh37/GRCh38). </li>
            <li><i>rseq_5p_mismatch:</i> signifies a mismatch between the RefSeq transcript and the underlying primary genome assembly sequence. Specifically, there is a mismatch in the 5' UTR of the RefSeq model with respect to the primary genome assembly. </li>
            <li><i>rseq_cds_mismatch:</i> signifies a mismatch between the RefSeq transcript and the underlying primary genome assembly sequence. Specifically, there is a mismatch in the CDS of the RefSeq model with respect to the primary genome assembly. </li>
            <li><i>rseq_ens_match_cds:</i> signifies that for the RefSeq transcript there is an overlapping Ensembl model that is identical across the CDS region only. A CDS match is defined as follows: the CDS and peptide sequences are identical and the genomic coordinates of every translatable exon match. Useful related attributes are: rseq_ens_match_wt and rseq_ens_no_match.</li>
            <li><i>rseq_ens_match_wt:</i> signifies that for the RefSeq transcript there is an overlapping Ensembl model that is identical across the whole transcript. A whole transcript match is defined as follows: 1) In the case that both models are coding, the transcript, CDS and peptide sequences are all identical and the genomic coordinates of every exon match. 2) In the case that both transcripts are non-coding the transcript sequences and the genomic coordinates of every exon are identical. No comparison is made between a coding and a non-coding transcript. Useful related attributes are: rseq_ens_match_cds and rseq_ens_no_match.</li>
            <li><i>rseq_ens_no_match:</i> signifies that for the RefSeq transcript there is no overlapping Ensembl model that is identical across either the whole transcript or the CDS. This is caused by differences between the transcript, CDS or peptide sequences or between the exon genomic coordinates. Useful related attributes are: rseq_ens_match_wt and rseq_ens_match_cds.</li>
            <li><i>rseq_mrna_match:</i> signifies an exact match between the RefSeq transcript and the underlying primary genome assembly sequence (based on a match between the transcript stable id and an accession in the RefSeq mRNA file). An exact match occurs when the underlying genomic sequence of the model can be perfectly aligned to the mRNA sequence post polyA clipping.</li>
            <li><i>rseq_mrna_nonmatch:</i> signifies a non-match between the RefSeq transcript and the underlying primary genome assembly sequence. A non-match is deemed to have occurred if the underlying genomic sequence does not have a perfect alignment to the mRNA sequence post polyA clipping. It can also signify that no comparison was possible as the model stable id may not have had a corresponding entry in the RefSeq mRNA file (sometimes happens when accessions are retired or changed). When a non-match occurs one or several of the following transcript attributes will also be present to provide more detail on the nature of the non-match: rseq_5p_mismatch, rseq_cds_mismatch, rseq_3p_mismatch, rseq_nctran_mismatch, rseq_no_comparison</li>
            <li><i>rseq_nctran_mismatch:</i> signifies a mismatch between the RefSeq transcript and the underlying primary genome assembly sequence. This is a comparison between the entire underlying genomic sequence of the RefSeq model to the mRNA in the case of RefSeq models that are non-coding. </li>
            <li><i>rseq_no_comparison:</i> signifies that no alignment was carried out between the underlying primary genome assembly sequence and a corresponding RefSeq mRNA. The reason for this is generally that no corresponding, unversioned accession was found in the RefSeq mRNA file for the transcript stable id. This sometimes happens when accessions are retired or replaced. A second possibility is that the sequences were too long and problematic to align (though this is rare).</li>
          </ul>
        </li>
        <li><b>OverlapBP</b> - Number of base pairs overlapping with the corresponding structural variation feature</li>
        <li><b>OverlapPC</b> - Percentage of corresponding structural variation feature overlapped by the given input</li>
        <li><b>CHECK_REF</b> - Reports variants where the input reference does not match the expected reference</li>
        <li><b>AMBIGUITY</b> - IUPAC allele ambiguity code</li>
      </ul>
    </div>
   
    <p>Example of Ensembl VEP default output format:</p> 
    <pre class="code sh_sh">
11_224088_C/A    11:224088   A  ENSG00000142082  ENST00000525319  Transcript         missense_variant           742  716  239  T/N  aCc/aAc  -  SIFT=deleterious(0);PolyPhen=unknown(0)
11_224088_C/A    11:224088   A  ENSG00000142082  ENST00000534381  Transcript         5_prime_UTR_variant        -    -    -    -    -        -  -
11_224088_C/A    11:224088   A  ENSG00000142082  ENST00000529055  Transcript         downstream_variant         -    -    -    -    -        -  -
11_224585_G/A    11:224585   A  ENSG00000142082  ENST00000529937  Transcript         intron_variant             -    -    -    -    -        -  HGVSc=ENST00000529937.1:c.136-346G>A
22_16084370_G/A  22:16084370 A  -                ENSR00000615113  RegulatoryFeature  regulatory_region_variant  -    -    -    -    -        -  -</pre>
    
    <p> The Ensembl VEP command line tool will also add a header to the output file. This contains
    information about the databases connected to, and also a key describing the
    key/value pairs used in the extra column. </p>
    
    <pre class="code sh_sh">
## ENSEMBL VARIANT EFFECT PREDICTOR v[[SPECIESDEFS::ENSEMBL_VERSION]].0
## Output produced at 2017-03-21 14:51:27
## Connected to homo_sapiens_core_[[SPECIESDEFS::ENSEMBL_VERSION]]_38 on ensembldb.ensembl.org
## Using cache in /homes/user/.vep/homo_sapiens/[[SPECIESDEFS::ENSEMBL_VERSION]]_GRCh38
## Using API version [[SPECIESDEFS::ENSEMBL_VERSION]], DB version [[SPECIESDEFS::ENSEMBL_VERSION]]
## polyphen version 2.2.2
## sift version sift5.2.2
## COSMIC version 78
## ESP version 20141103
## gencode version GENCODE 25
## genebuild version 2014-07
## HGMD-PUBLIC version 20162
## regbuild version 16
## assembly version GRCh38.p7
## ClinVar version 201610
## dbSNP version 147
## Column descriptions:
## Uploaded_variation : Identifier of uploaded variant
## Location : Location of variant in standard coordinate format (chr:start or chr:start-end)
## Allele : The variant allele used to calculate the consequence
## Gene : Stable ID of affected gene
## Feature : Stable ID of feature
## Feature_type : Type of feature - Transcript, RegulatoryFeature or MotifFeature
## Consequence : Consequence type
## cDNA_position : Relative position of base pair in cDNA sequence
## CDS_position : Relative position of base pair in coding sequence
## Protein_position : Relative position of amino acid in protein
## Amino_acids : Reference and variant amino acids
## Codons : Reference and variant codon sequence
## Existing_variation : Identifier(s) of co-located known variants
## Extra column keys:
## IMPACT : Subjective impact classification of consequence type
## DISTANCE : Shortest distance from variant to transcript
## STRAND : Strand of the feature (1/-1)
## FLAGS : Transcript quality flags</pre>


    <br />
    <hr />
    <h2 id="tab"> Tab-delimited output</h2>

    <p> The <a href="script/vep_options.html#opt_tab">--tab</a> specifies
     a tab-delimited output file.<br />
    <b>This differs from the default output format in that each individual field from the 
    "Extra" field is written to a separate tab-delimited column</b>.<br />
    This makes the output more suitable for import into spreadsheet programs such as Excel.<br />
    Furthermore the header is the same as the one for the Ensembl VEP default output format and this is 
    also the format used when selecting the "TXT" option on the Ensembl VEP web interface.</p>

    <p>Example of tab-delimited output format:</p>
    <pre class="code sh_sh">
#Uploaded_variation  Location   Allele  Gene             Feature          Feature_type  Consequence                            cDNA_position  CDS_position  Protein_position  Amino_acids  Codons   Existing_variation  IMPACT    DISTANCE  STRAND  FLAGS 
11_224088_C/A        11:224088  A       ENSG00000142082  ENST00000525319  Transcript    missense_variant                       742            716           239               S/I          aGc/aTc  -                   MODERATE  -         -1      - 
11_224088_C/A        11:224088  A       ENSG00000142082  ENST00000534381  Transcript    downstream_gene_variant                -              -             -                 -            -        -                   MODIFIER  1674      -1      -
11_224088_C/A        11:224088  A       ENSG00000142082  ENST00000529055  Transcript    downstream_gene_variant                -              -             -                 -            -        -                   MODIFIER  134       -1      -
11_224585_G/A        11:224585  A       ENSG00000142082  ENST00000529937  Transcript    intron_variant,NMD_transcript_variant  -              -             -                 -            -        -                   MODIFIER  -         -1      -</pre>
    
    <div>
      <div class="info" style="float:left">
        <h3>Note</h3>
        <div class="message-pad">
          <p>The Existing_variation column is only populated if we use option <a href="script/vep_options.html#opt_check_existing">--check_existing</a> or any other option that switches it on (--af_1kg silently switches on --check_existing).</p>
        </div>
      </div>
      <div style="clear:both"></div>
    </div>

    <p> The choice and order of columns in the output may be configured using <a href="script/vep_options.html#opt_fields">--fields</a>. For instance:<br />
      <pre class="code sh_sh">./vep -i examples/homo_sapiens_GRCh38.vcf --cache --force_overwrite <b>--tab --fields "Uploaded variation,Location,Allele,Gene"</b></pre>
    </p>
    

    <br />
    <hr />
    <h2 id="vcfout"> VCF output</h2>
    
    <p> The Ensembl VEP commandline tool can also generate VCF output using the <a href="script/vep_options.html#opt_vcf">--vcf</a> flag.</p>
    <p>Main information about the VCF output format:</p>
    <ul>
      <li>Consequences are added in the INFO field of the VCF file, using the key "<b>CSQ</b>" 
          (you can change it using <a href="script/vep_options.html#opt_vcf_info_field">--vcf_info_field</a>).</li>
      <li>Data fields are encoded separated by the character "<b>&#124;</b>" (pipe). The order of fields is written in the VCF header. 
          Unpopulated fields are represented by an empty string.</li>
      <li>Output fields in the "CSQ" INFO field can be configured by using <a href="script/vep_options.html#opt_fields">--fields</a>.</li>
      <li>Each prediction, for a given variant, is separated by the character "<b>,</b>" in the CSQ INFO field (e.g. when a variant overlaps more than 1 transcript)</li>
    </ul>
    <br />

    <p>
    Here is a list of the (default) fields you can find within the CSQ field:
    <pre class="code sh_sh">Allele|Consequence|IMPACT|SYMBOL|Gene|Feature_type|Feature|BIOTYPE|EXON|INTRON|HGVSc|HGVSp|cDNA_position|CDS_position|Protein_position|Amino_acids|Codons|Existing_variation|DISTANCE|STRAND|FLAGS|SYMBOL_SOURCE|HGNC_ID</pre>
    </p>

    <div>
      <div class="info" style="float:left">
        <h3>Note</h3>
        <div class="message-pad">
          <p>Some fields are not reported by default and need configuring including HGVS, Symbol and Existing_variation.</p>
        </div>
      </div>
      <div style="clear:both"></div>
    </div>

    <p>Example command using the <a href="script/vep_options.html#opt_vcf">--vcf</a> and <a href="script/vep_options.html#opt_fields">--fields</a> options:
    <pre class="code sh_sh">./vep -i examples/homo_sapiens_GRCh38.vcf --cache --force_overwrite <b>--vcf --fields "Allele,Consequence,Feature_type,Feature"</b></pre>
    
    <p> VCFs produced by Ensembl VEP can be filtered using <a
    href="script/vep_filter.html">filter_vep.pl</a> in the same way as standard format
    output files. </p>
    
    <p> If the input format was VCF, the file will remain unchanged save for the
    addition of the CSQ field and the header (unless using any filtering). If an
    existing CSQ field is found, it will be replaced by the new annotation
    (use <a href="script/vep_options.html#opt_keep_csq">--keep_csq</a> to
    preserve it). </p>
    
    <p> Custom data added with <a
    href="script/vep_options.html#opt_custom">--custom</a> are added as separate
    fields, using the key specified for each data file.</p>
    
    <p> Commas in fields are replaced with ampersands (&amp;) to preserve VCF
    format. </p>
    
    <pre class="code sh_sh">##INFO=&lt;ID=CSQ,Number=.,Type=String,Description="Consequence annotations from Ensembl VEP. Format: Allele|Consequence|IMPACT|SYMBOL|Gene|Feature_type|Feature|BIOTYPE|EXON|INTRON|HGVSc|HGVSp|cDNA_position|CDS_position|Protein_position"&gt;
#CHROM  POS       ID          REF  ALT  QUAL  FILTER  INFO
21      26978790  rs75377686  T    C    .     .       CSQ=C|missense_variant|MODERATE|MRPL39|ENSG00000154719|Transcript|ENST00000419219|protein_coding|2/8||ENST00000419219.1:c.251A>G|ENSP00000404426.1:p.Asn84Ser|260|251|84</pre>
    
    
    <br />
    <hr />
    <h2 id="json"> JSON output</h2>
    
    <p> Ensembl VEP can output serialised <a
    href="http://json.org/" rel="external">JSON</a> objects using the <a
    href="script/vep_options.html#opt_json">--json</a> flag. JSON is a
    serialisation format that can be parsed and processed easily by many
    packages and programming languages; it is used as the default output format
    for <a href="http://rest.ensembl.org" rel="external">Ensembl's REST
    server</a>. </p>
    
    <p> Each input variant is reported as a single JSON object which constitutes
    one line of the output file. The JSON object is structured somewhat
    differently to the other  output formats, in that per-variant fields
    (e.g. co-located existing variant details) are reported only once.
    Consequences are grouped under the feature type that they affect
    (Transcript, Regulatory Feature, etc). The original input line (e.g. from
    VCF input) is reported under the "input" key in order to aid aligning input
    with output. When using a cache file, frequencies for co-located variants are reported by default (see
    <a href="./script/vep_options.html#opt_af_1kg">--af_1kg</a>,
    <a href="./script/vep_options.html#opt_af_gnomade">--af_gnomade</a>).
    </p>
    
    <p> Here follows an example of JSON output (prettified and redacted for display here): </p>
    
    <pre class="code sh_sh">{
  "input": "1 1918090 test1 A G . . .",
  "id": "test1",
  "seq_region_name": "1",
  "start": 1918090,
  "end": 1918090,
  "strand": 1,
  "allele_string": "A/G",
  "most_severe_consequence": "missense_variant",
  "colocated_variants": [
    {
      "id": "COSV57068665",
      "seq_region_name": "1",
      "start": 1918090,
      "end": 1918090,
      "strand": 1,
      "allele_string": "COSMIC_MUTATION"
    },
    {
      "id": "rs28640257",
      "seq_region_name": "1",
      "start": 1918090,
      "end": 1918090,
      "strand": 1,
      "allele_string": "A/G/T",
      "minor_allele": "G",
      "minor_allele_freq": 0.352,
      "frequencies": {
        "G": {
          "amr": 0.5072,
          "gnomade_sas": 0.3635,
          "gnomade": 0.481,
          "gnomade_remaining": 0.4536,
          "gnomade_asj": 0.3939,
          "gnomade_nfe": 0.5042,
          "gnomade_afr": 0.0975,
          "afr": 0.053,
          "gnomade_amr": 0.5568,
          "gnomade_fin": 0.4751,
          "sas": 0.3906,
          "gnomade_eas": 0.4516,
          "eur": 0.4901,
          "eas": 0.4623,
          "gnomade_mid: "0.3306"
        }
      }
    }
  ],
  "transcript_consequences": [
    {
      "variant_allele": "G",
      "consequence_terms": [
        "missense_variant"
      ],
      "gene_id": "ENSG00000178821",
      "transcript_id": "ENST00000310991",
      "strand": -1,
      "cdna_start": 436,
      "cdna_end": 436,
      "cds_start": 422,
      "cds_end": 422,
      "protein_start": 141,
      "protein_end": 141,
      "codons": "aTg/aCg",
      "amino_acids": "M/T",
      "polyphen_prediction": "benign",
      "polyphen_score": 0.001,
      "sift_prediction": "tolerated",
      "sift_score": 0.22,
      "hgvsp": "ENSP00000311122.3:p.Met141Thr",
      "hgvsc": "ENST00000310991.8:c.422T>C"
      }
  ],
  "regulatory_feature_consequences": [
    {
      "variant_allele": "G",
      "consequence_terms": [
        "regulatory_region_variant"
      ],
      "regulatory_feature_id": "ENSR00000000255"
    }
  ]
}</pre>

    <p> In accordance with JSON conventions, all keys (except alleles) are lower-case. Some keys
    also have different names and structures to those found in the other Ensembl VEP
    output formats: </p>
    
    <table class="ss">
      <tr>
        <th>Key</th>
        <th>JSON equivalent(s)</th>
        <th>Notes</th>
      </tr>
      <tr>
        <td>Consequence</td>
        <td>consequence_terms</td>
        <td></td>
      </tr>
      <tr>
        <td>Gene</td>
        <td>gene_id</td>
        <td></td>
      </tr>
      <tr>
        <td>Feature</td>
        <td>transcript_id, regulatory_feature_id, motif_feature_id</td>
        <td>Consequences are grouped under the feature type they affect</td>
      </tr>
      <tr>
        <td>ALLELE</td>
        <td>variant_allele</td>
        <td></td>
      </tr>
      <tr>
        <td>SYMBOL</td>
        <td>gene_symbol</td>
        <td></td>
      </tr>
      <tr>
        <td>SYMBOL_SOURCE</td>
        <td>gene_symbol_source</td>
        <td></td>
      </tr>
      <tr>
        <td>ENSP</td>
        <td>protein_id</td>
        <td></td>
      </tr>
      <tr>
        <td>OverlapBP</td>
        <td>bp_overlap</td>
        <td></td>
      </tr>
      <tr>
        <td>OverlapPC</td>
        <td>percentage_overlap</td>
        <td></td>
      </tr>
      <tr>
        <td>Uploaded_variation</td>
        <td>id</td>
        <td></td>
      </tr>
      <tr>
        <td>Location</td>
        <td>seq_region_name, start, end, strand</td>
        <td>The variant's location field is broken down into constituent coordinate parts for clarity. "seq_region_name" is used in place of "chr" or "chromosome" for consistency with other parts of Ensembl's REST API</td>
      </tr>
      <tr>
        <td>*_maf</td>
        <td>*_allele, *_maf</td>
        <td></td>
      </tr>
      <tr>
        <td>cDNA_position</td>
        <td>cdna_start, cdna_end</td>
        <td></td>
      </tr>
      <tr>
        <td>CDS_position</td>
        <td>cds_start, cds_end</td>
        <td></td>
      </tr>
      <tr>
        <td>Protein_position</td>
        <td>protein_start, protein_end</td>
        <td></td>
      </tr>
      <tr>
        <td>SIFT</td>
        <td>sift_prediction, sift_score</td>
        <td></td>
      </tr>
      <tr>
        <td>PolyPhen</td>
        <td>polyphen_prediction, polyphen_score</td>
        <td></td>
      </tr>
    </table>
   
    <br />
    <hr />
    <h2 id="stats"> Statistics</h2>
    
    <p> Ensembl VEP writes an HTML file containing statistics pertaining to the
    results of your job; it is named <b>[output_file]_summary.html</b> (with the
    default options the file will be named
    <b>variant_effect_output.txt_summary.html</b>). To view it, please open
    the file in your web browser. </p>
    
    <ul>
      <li> To prevent Ensembl VEP writing a stats file, use <a
      href="script/vep_options.html#opt_no_stats">--no_stats</a>.</li>

      <li> To get a machine-readable text file in place of the HTML file,
      use <a href="script/vep_options.html#opt_stats_text">--stats_text</a>.
      You can get both a HTML file and plain text file by using both
      <a href="script/vep_options.html#opt_stats_text">--stats_text</a>
      and <a href="script/vep_options.html#opt_stats_html">--stats_html</a>.
      </li>

      <li>To change the name of the stats file from the default, use <a
      href="script/vep_options.html#opt_stats_file">--stats_file [file]</a>. </li>
    </ul>
    <p> The page contains several sections: </p>
    
    <h4> General statistics </h4>
    
    <p> This section contains two tables. The first describes the cache and/or
    database used, the version of Ensembl VEP, species, command line parameters,
    input/output files and run time. The second table contains information about
    the number of variants, and the number of genes, transcripts and regulatory
    features overlapped by the input. </p>
    
    <h4> Charts and tables </h4>
    
    <p> There then follows several charts, most with accompanying tables. Tables
    and charts are interactive; clicking on a row to highlight it in the table
    will highlight the relevant segment in the chart, and vice versa. </p>
    
    <div class="tinted-box" style="width: 200px; padding-bottom: 5px; float: left; margin-right: 20px;">
        <a href="/img/vep_stats_1.png"><img style="width: 200px; border: 1px solid grey" src="/img/vep_stats_1.png">
        General statistics</a>
    </div>
    
    <div class="tinted-box" style="width: 200px; padding-bottom: 5px; float: left; margin-right: 20px;">
        <a href="/img/vep_stats_2.png"><img style="width: 200px; border: 1px solid grey" src="/img/vep_stats_2.png">
        Summary of called consequence types</a>
    </div>
    
    <div class="tinted-box" style="width: 200px; padding-bottom: 5px; float: left; margin-right: 20px;">
        <a href="/img/vep_stats_3.png"><img style="width: 200px; border: 1px solid grey" src="/img/vep_stats_3.png">
        Distribution of variants across chromosomes</a>
    </div>

<div style="clear:both"><h1 style="page-break-after: always">&nbsp</h1></div><div>
  <div style="float:right;"><img src="/img/vep_logo.png"/></div>
	
  <h1 id="top"><span style="color:#006;padding-right:15px">Ensembl Variant Effect Predictor</span><span style="color:#666"><img src="/i/16/help.png"/> FAQ</span></h1>
  <hr/>
    
    <p> For any questions not covered here, please send an email to the Ensembl
    <a href="/info/about/contact/index.html">developer's mailing list</a>
    (public) or contact the
    <a href="/Help/Contact" class="popup">Ensembl Helpdesk</a> (private).
    Also you can report issues through our (public) Github repositories. For general Ensembl VEP issues you should use 
    <a href="https://github.com/Ensembl/ensembl-vep/issues">ensembl-vep</a> repository and for specific plugins you should use 
    <a href="https://github.com/Ensembl/VEP_plugins/issues">VEP_plugins</a> repository.
    </p>

<h2 id="general">General questions</h2>

<h3 id="indel-vcf">Q: Why has my insertion/deletion variant encoded in VCF
disappeared from the output? </h3>

<p> Ensembl treats unbalanced variants differently to VCF - your variant
hasn't disappeared, it may have just changed slightly! You can solve this by
giving your variants a unique identifier in the third column of the VCF file.
See <a href="vep_formats.html#vcf">here</a> for a full discussion.</p>

<p>&nbsp;</p>

<h3 id="no-co-located-vars"> Q: Why don't I see any co-located variants when
using species X? </h3>

<p> Not all species have variants and not all species that do are in the Ensembl variation resource
- see <a href="/info/genome/variation/species/species_data_types.html#source">this
  document</a> for details.
The <a href="script/vep_options.html#opt_custom"> custom </a>option can be used in the commandline interface to include more variant sets</p>
</p>

<p>&nbsp;</p>

<h3 id="multi-vars-mapped"> Q: Why do I see multiple known variants mapped to my input variant? </h3>

<p> Ensembl VEP compares your input to known variants from the Ensembl variation database.
In some cases one input variant can match multiple known variants:
<ul>
  <li>Germline variants from dbSNP and somatic mutations from COSMIC may be found at the same locus</li>
  <li>Some sources, e.g. HGMD, do not provide public access to allele-specific data, so an HGMD variant with unknown alleles may colocate with one from dbSNP with known alleles</li>
  <li>Multiple alternate alleles from your input may match different variants as they are described in dbSNP</li>
</ul>

See <a href="script/vep_other.html#colocated">here</a> for a full discussion.</p>

<p>&nbsp;</p>

<h3 id="no-freq"> Q: Ensembl VEP is not assigning a frequency to my input variant - why? </h3>

<p> Ensembl VEP's cache contains frequency data only for specific studies. See <a href="script/vep_other.html#colocated">here</a> for a full discussion. 
The <a href="script/vep_options.html#opt_custom"> custom </a>option can be used in the commandline interface to include more frequency sets</p>

<p>&nbsp;</p>

<h3 id="many-lines"> Q: Why do I see so many lines of output for each variant in my input? </h3>

<p> While it would be convenient to have a simple, one word answer to the
question "What is the consequence of this variant?", in reality biology is not
 this simple! Many genes have more than one transcript, so Ensembl VEP
provides a prediction for each transcript that a variant overlaps. Ensembl VEP
has options to help select results according to your requirements; the <a
href="script/vep_options.html#opt_canonical">--canonical</a> and <a
href="script/vep_options.html#mane">--mane</a> options indicate which
transcripts are canonical and belong to the human MANE set respectively, while <a
href="script/vep_options.html#opt_pick">--pick</a>, <a
href="script/vep_options.html#opt_per_gene">--per_gene</a>, <a
href="script/vep_options.html#opt_summary">--summary</a> and <a
href="script/vep_options.html#opt_most_severe">--most_severe</a> allow you to
give a more summary level assessment per variant.</p>

<p> Furthermore, several "compound" consequences are also possible - if, for
example, a variant falls in the final few bases of an exon, it may be considered
to affect a splicing site, in addition to possibly affecting the coding
sequence. </p>


<p>&nbsp;</p>

<h3 id="reduce-mem"> Q: How do I reduce Ensembl VEP's memory requirement?</h3>

<p> There are a number of ways to do this-
<ol>
  <li>Ensure your input file is sorted by location. This can greatly reduce memory
      requirements and runtime </li>
  <li>Consider reducing the buffer size. This reduces the number of variants annotated
      together in a batch and can be modified in both command line and web interfaces.
      Reducing buffer size may increase run time.</li>
  <li>Ensure you are only using the options you need, rather than --everything.
      Some data-rich options, such as regulatory annotation have an impact on memory use</li>
</ol>
</p>

<p>&nbsp;</p>


<h3 id="cite-vep"> Q: How to cite Ensembl VEP?</h3>

<p> If you use Ensembl VEP, please cite our
<a href="/info/docs/tools/vep/index.html#publication">latest publication</a>
to continue to support Ensembl VEP development. </p>

<hr/>

<h2 id="web">Ensembl VEP web interface questions</h2>

<h3 id="web-vep"> Q: How do I access the web version of the Ensembl Variant
Effect Predictor? </h3>

<p> You can find the Ensembl VEP web tool on the <a href="/info/docs/tools/index.html">Tools</a> page.</p>

<p>&nbsp;</p>


<h3 id="different-output"> Q: Why is the output I get for my input file
different when I use the Ensembl VEP web and command line interfaces? </h3>

<p> Ensure that you are passing equivalent arguments to the command line tool that you
are using in the web interface. If you are sure this is still a problem, please
report it on the <a href="http://lists.ensembl.org/mailman/listinfo/dev" rel="external">ensembl-dev</a> mailing list.</p>

<p>&nbsp;</p>


<h3 id="web-vep-tutorial"> Q: Is there a tutorial for the web tool?</h3>

<p>Yes, see our latest tutorial <a href="https://onlinelibrary.wiley.com/doi/10.1002/humu.24298" rel="external">
Annotating and prioritizing genomic variants using the Ensembl Variant Effect Predictor â€” A tutorial</a>
for more information on using the Ensembl VEP web interface.</p>
<hr/>

<h2 id="script">Ensembl VEP command line tool questions</h2>

<h3 id="faster-vep"> Q: How can I make Ensembl VEP run faster? </h3>

<p> There are a number of factors that influence how fast Ensembl VEP runs. Have a look at our <a href="script/vep_other.html#faster">handy guide</a> for tips on improving runtime. </p>

<p>&nbsp;</p>

<h3 id="different-input-var"> Q: Why am I not seeing the same variant from my input in the output? </h3>
<p> Since the Ensembl 110 release, Ensembl VEP by default will minimise the input allele for display in the output. To see the exact allele representation you provided, use the<a
href="script/vep_options.html#opt_uploaded_allele"> --uploaded_allele</a> option. </p>

<p>&nbsp;</p>

<h3 id="hgvs-n"> Q: Why do I see "N" as the reference allele in my HGVS strings? </h3>
<h3 id="sequence.pm"> Q: Why do I get errors related with Sequence.pm?</h3>

<pre class="code sh_sh">substr outside of string at /nfs/users/nfs_w/wm2/Perl/ensembl-variation/modules/Bio/EnsEMBL/Variation/Utils/Sequence.pm line 511.
Use of uninitialized value $ref_allele in string eq at /nfs/users/nfs_w/wm2/Perl/ensembl-variation/modules/Bio/EnsEMBL/Variation/Utils/Sequence.pm line 514.
Use of uninitialized value in concatenation (.) or string at /nfs/users/nfs_w/wm2/Perl/ensembl-variation/modules/Bio/EnsEMBL/Variation/Utils/Sequence.pm line 643.</pre>

<p> Both of these error types are usually seen when using a <a
href="script/vep_cache.html#fasta">FASTA file</a> for retrieving sequence.
There are a couple of steps you can take to try to remedy them:</p>

<ol>
  <li>
    The index alongside the FASTA can become corrupted. Delete
    [fastafile].index and re-run Ensembl VEP to regenerate it. By default
    this file is located in your $HOME/.vep/[species]/[version]_[assembly]
    directory.
  </li>

  <li>
    The FASTA file itself may have been corrupted during download;
    delete the fasta file and the index and re-download (you can use the
    <a href="script/vep_download.html#install">Ensembl VEP installer</a> to do this).
  </li>

  <li>
    Older versions of BioPerl (1.2.3 in particular is known to have this)
    cannot properly index large FASTA files. Make sure you are using a later
    (>=1.6) version of BioPerl. The <a href="script/vep_download.html#install">Ensembl VEP
    installer</a> installs 1.6.924 for you.
  </li>
</ol>

<p> If you still see problems after taking these steps, or if you were not
using a FASTA file in the first place, please <a href="#top">contact us</a>. </p>

<p>&nbsp;</p>

<h3 id="chr-not-found"> Q: Why are chromosomes not found in annotation sources or synonyms?</h3>

<pre class="code sh_sh">WARNING: Chromosome 21 not found in annotation sources or synonyms on line 160</pre>

<p> This can occur if the chromosome names differ between your input variant and any annotation source that you are using (cache, database, GFF/GTF file, FASTA file, custom annotation file). To circumvent this you may provide a <a href="script/vep_options.html#opt_synonyms">synonyms file</a>. A synonym file is included in Ensembl VEP's cache files, so if you have one of these for your species you can use it as follows:</p>

<pre class="code sh_sh">./vep -i input.vcf -cache -synonyms ~/.vep/homo_sapiens/[[SPECIESDEFS::ENSEMBL_VERSION]]_GRCh38/chr_synonyms.txt</pre>

<p> The file consists of lines containing pairs of tab-separated synonyms. Order is not important as synonyms can be used in both "directions". </p>

<p>&nbsp;</p>

<h3 id="feature_type_warnings"> Q: Why do I get feature_type warnings from my GFF/GTF file?</h3>

<pre class="code sh_sh sh_sourceCode"><span class="sh_variable">WARNING: </span>Ignoring 'five_prime_utr' feature_type from Homo_sapiens.GRCh38.111.gtf.gz GFF/GTF file. This feature_type is not supported in Ensembl VEP.</span></pre>

<p> This can occur if you are using GFF/GTF file and the file contains a type that is not supported by Ensembl VEP. Those lines are simply ignored. However, in cases where the transcript model is incomplete the full model may be ignored.</p>
  
<p>Please try to use supported feature types as mentioned <a href="https://www.ensembl.org/info/docs/tools/vep/script/vep_cache.html#gff">here</a></p>

<p>&nbsp;</p>


<h3 id="gnomad-frequences"> Q: Can I get gnomAD exomes and genomes frequencies in Ensembl VEP? </h3>

<p> Yes, see <a href="script/vep_example.html#gnomad">this guide</a>. </p>

<p>&nbsp;</p>

<h3 id="no-server-connection"> Q: Why do I have issues connecting to Ensembl databases? </h3>

<pre class="code sh_sh">
Could not connect to database homo_sapiens_core_63_37 as user anonymous using [DBI:mysql:database=homo_sapiens_core_63_37;host=ensembldb.ensembl.org;port=5306] as a locator:
Unknown MySQL server host 'ensembldb.ensembl.org' (2) at $HOME/src/ensembl/modules/Bio/EnsEMBL/DBSQL/DBConnection.pm line 290.

-------------------- EXCEPTION --------------------
MSG: Could not connect to database homo_sapiens_core_63_37 as user anonymous using [DBI:mysql:database=homo_sapiens_core_63_37;host=ensembldb.ensembl.org;port=5306] as a locator:
Unknown MySQL server host 'ensembldb.ensembl.org' (2)
</pre>

<p> If you select the database option rather than using a cache
Ensembl VEP will try to connect to the public MySQL
server at ensembldb.ensembl.org. Occasionally the server may break connection
with your process, which causes this error. This can happen when the server is
busy, or due to various network issues. Consider using the 
<a href="script/vep_cache.html#cache">caching system</a>. 
Using a cache and fasta file is the most effcient way to run Ensembl VEP
</p>

<p>&nbsp;</p>

<h3 id="vep-windows"> Q: Can I use Ensembl VEP on Windows? </h3>

<p> Yes - see the <a href="script/vep_download.html#windows">documentation</a> for a
few different ways to get the Ensembl VEP running on Windows. </p>

<p>&nbsp;</p>

<h3 id="vep-custom-assemblies"> Q: Can I use Ensembl VEP with species and assemblies which are not available in Ensembl? </h3>

<p> Yes - you can run Ensembl VEP on any species you have data for by providing a custom
gene annotation in <a href="script/vep_cache.html#gff">GFF/GTF</a> and genome sequence in
<a href="script/vep_cache.html#fasta">FASTA</a> file, like so:
<pre class="code sh_sh">./vep -i input.vcf --gff data.gff.gz --fasta genome.fa.gz</pre>
</p>

<p>&nbsp;</p>

<h3 id="vep-custom-assemblies"> Q: Can I use Ensembl VEP with T2T-CHM13 and other human assemblies? </h3>

<p> Yes - you can run Ensembl VEP using <a href="https://humanpangenome.org/" rel="external">Human Pangenome Reference Consortium (HPRC)</a> data by following the instructions on how to <a href="script/vep_example.html#pangenomes">use Ensembl VEP with HPRC assemblies</a>.

</p>

<p>&nbsp;</p>

<h3 id="download-sift-polyphen"> Q: Can I download all of the SIFT and/or PolyPhen predictions? </h3>

<p> The Ensembl Variation database and the human Ensembl VEP cache file contain
precalculated SIFT and PolyPhen-2 predictions for every possible amino acid change
in every translated protein product in Ensembl. Since these data are huge, we
store them in a <a
href="/info/genome/variation/prediction/protein_function.html#nsSNP_data_format">compressed format</a>. </p>

<p> There are different approaches to download SIFT/PolyPhen-2 data: </p>

<ul>
  <li> Using the <a href="script/vep_plugins.html#PolyPhen_SIFT">PolyPhen_SIFT plugin</a>: </li>
  <ul>
    <li> For any species with predictions in our Ensembl databases, the plugin is able to download the predictions data into a local SQLite database for offline use. PolyPhen predictions are only available for human data. </li>
    <li> We also provide a downloadble SQLite database containing PolyPhen/SIFT predictions based on <a href="https://humanpangenome.org" rel="external">Human Pangenome Reference Consortium (HPRC)</a> and GRCh38 assemblies. For more information, refer to <a href="script/vep_example.html#PolyPhen_SIFT">Missense deleteriousness predictions</a> in HPRC assemblies.</li>
  </ul>
  <li> Using our <b>Perl API</b>: </li>
  <ul>
    <li> Fetch a <a
href="/info/docs/Doxygen/variation-api/classBio_1_1EnsEMBL_1_1Variation_1_1ProteinFunctionPredictionMatrix.html"><kbd>ProteinFunctionPredictionMatrix</kbd></a> for your protein of interest and then call its
<a href="/info/docs/Doxygen/variation-api/classBio_1_1EnsEMBL_1_1Variation_1_1ProteinFunctionPredictionMatrix.html#af6346bcc5c4c28281e3c9a009ffcb15d"><kbd>get_prediction()</kbd></a> method to get the score for a particular position and amino
acid, looping over all possible amino acids for your position. </li>
    <li> You would need to work out which peptide position your codon maps to, but
there are methods in the <a
href="/info/docs/Doxygen/variation-api/classBio_1_1EnsEMBL_1_1Variation_1_1TranscriptVariation.html"><kbd>TranscriptVariation</kbd></a>
class that should help you (probably <a href="/info/docs/Doxygen/variation-api/classBio_1_1EnsEMBL_1_1Variation_1_1BaseTranscriptVariation.html#acfc84f951919f445624545afbd5da0a8"><kbd>translation_start()</kbd></a> and <a href="/info/docs/Doxygen/variation-api/classBio_1_1EnsEMBL_1_1Variation_1_1BaseTranscriptVariation.html#ae32d5894a6466c7bf0fa02e6d200d038"><kbd>translation_end()</kbd></a>).</li>
  </ul>
</ul>

</div>

<div style="clear:both"><h1 style="page-break-after: always">&nbsp</h1></div></body></html>
