<html>
<head>
<title>Regulation Database and API</title>
<meta name="order" content="13" />
</head>

<body>

<h1>Ensembl Funcgen Summary</h1>

<div style="display:table;font-size:14px;margin:20px 0px 0px">
  <div style="display:table-row">
    <div style="display:table-cell">
      <div class="round-box plain-box " style="width: 600px">
        <h3>Deprecation note</h3>
        <p>As of release 113, it is no longer recommended to access funcgen data via the Perl API or the funcgen DB.
        See <a href="/info/genome/funcgen/data/accessing-regulation.html">Accessing regulatory data</a> for alternative options.</p>
        <p>If the Perl API or funcgen database is important to your research, please let us know about your use case by emailing <a href="mailto:helpdesk@ensembl.org">helpdesk@ensembl.org</a>.<p>
      </div>
    </div>
  </div>
</div>

<h2>Database Description</h2>

<p>
The Ensembl regulation database and API are developed and maintained by the Ensembl Regulation team. The main focus of the 'funcgen' database is on experimental data which may elucidate transcriptional regulation.  The schema was originally designed to support ChIP on Chip tiling array experiments but soon expanded to support other sequencing-based technologies.  The original requirements supported use as a local data storage and analysis platform, as such the regulation data bases supports multiple coordinate systems. There is also support for micro array designs and externally curated data sets. See <a href='/info/genome/funcgen/index.html'>here</a> for a summary of the data base content.
</p>

<ul style="vertical-align:top">
  <li style="list-style-image: url(/i/16/documentation.png);vertical-align:top">
    <a href="funcgen_schema.html" style="font-weight:bold;vertical-align:top">Schema documentation</a>: page describing every column of every table of the Ensembl Funcgen database schema.
  </li>
  <li style="list-style-image: url(/i/16/database.png);vertical-align:top">
    <span style="font-weight:bold;vertical-align:top">Database diagram </span>: The Regulation Database schema is available below and is broken into several individual parts
    <br>
    <a href="regulatory.png">
       <img src="regulatory_thumb.png" style="border:1px solid navy;vertical-align:top" alt="Regulatory Build Schema Diagram" title="Regulatory Build Schema Diagram">
    </a>
    <a href="probe.png">
       <img src="probe_thumb.png" style="border:1px solid navy;vertical-align:top" alt="Probemapping Schema Diagram" title="Probemapping Schema Diagram">
    </a>
    <a href="motif.png">
       <img src="motif_thumb.png" style="border:1px solid navy;vertical-align:top" alt="Motif Schema Diagram" title="Motif Schema Diagram">
    </a>
  </li>
</ul>

<!-- The database content includes:

</p>

<ul>
<li>The <a href='/info/docs/index.html'>Ensembl Regulatory Build</a> - Regulatory features generated from a combined analysis of all of the above.</li>
<li>Experimental data and meta data from functional genomics experiments using the following technologies:
  <ul>
  <li>Chromatin immunoprecipitation using tiling arrays (ChIP-chip)</li>
  <li>Chromatin immunoprecipitation sequencing</li>
  <li>Other techniques identifying open chromatin e.g. DNase sequencing, FAIRE</li>
  </ul>
</li>
<li>Experimental data on epigenomic modifications and DNA binding proteins:
  <ul>
  <li>Histone modifications</li>
  <li>Transcription factors e.g. CCCTC binding factor (CTCF) etc.</li>
  <li>RNAPollII</li>
  <li><a href='http://genome.ucsc.edu/cgi-bin/hgFileUi?db=hg19&g=wgEncodeHaibMethylRrbs'>DNA Methylation data</a> from Reduced Representation Bisulfite Sequencing (RRBS). Data from the <a href='http://www.genome.gov/10005107'>ENCODE</a> project  </ul>
</li>
<li>Externally curated data with a regulatory focus:
  <ul>
  <li><a href='http://www.cisred.org/'>cisRED</a> - Regulatory motifs</li>
  <li><a href='http://www.ebi.ac.uk/enright-srv/microcosm/htdocs/targets/v5/consset.html'>miRanda</a> - miRNA targets</li>
  <li><a href='http://enhancer.lbl.gov/'>VISTA</a> - Enhancers</li>
  </ul>
</li>
<li>Array designs
  <ul>
  <li>For various array types e.g. Affy IVT/ST, Illumina Whole Genome/Infinium, Codelink, Agilent etc</li>
  <li>Ensembl transcript annotations for Expression arrays.
  </ul>
</li>
</ul>
 -->

<h2 id="api">Perl API</h2>

<p>
A comprehensive Perl Application Program Interface (API) provides efficient access to the Ensembl Funcgen database.
</p>

<ul>
  <li style="list-style-image: url(/i/16/download.png);vertical-align:top">
    <a href="/info/docs/api/api_installation.html" style="font-weight:bold;vertical-align:top">API Installation</a>: A step-by-step installation guide for all Perl Ensembl APIs.
  </li>
  <li style="list-style-image: url(/i/16/documentation.png);vertical-align:top">
    <a href="/info/docs/Doxygen/funcgen-api/index.html" style="font-weight:bold;vertical-align:top">Funcgen API Documentation</a>: A complete reference to the objects and methods used in the Funcgen database API, through the Doxygen tool.
  </li>
  <li style="list-style-image: url(/i/16/documentation.png);vertical-align:top">
    <a href="/info/docs/api/funcgen/regulation_tutorial.html" style="font-weight:bold;vertical-align:top">Funcgen API Tutorial</a>: An introduction to the underlying concepts of the Funcgen database API.
  </li>
</ul>


<!-- <li>Guidelines on how to set up the eFG analysis environment are available from the eFG perl package: ensembl-funcgen/scripts/introduction_to_efg.txt</li> -->

<h2 id="support">Ensembl Software Support</h2>

<p>Ensembl is an open project and we would like to encourage
correspondence and discussions on any subject on any aspect of Ensembl.
Please see the
<a href="/info/about/contact/index.html">Ensembl Contacts</a> page for suitable options getting in touch with us.
</p>


</body>
</html>
