<html>
<head>
<title>Non-coding RNAs</title>
</head>

<body>

<h1>Non-coding RNAs</h1>

<h2>Non-coding RNA Overview</h2>

<p>
<strong>Non-coding RNAs</strong> (ncRNAs) are involved in many
biological processes and are increasingly seen as important. As is the
case with proteins, it is the overall structure of the molecule which
imparts function. However, while similar protein structures are often
reflected in a conserved amino acid sequence, sequences underlying RNA
secondary structure are very variable; this makes ncRNAs difficult to
detect using sequence alone.
</p>

<p>
Because of this, we use a variety of techniques to detect
ncRNAs. First, a combination of sensitive BLAST searches are used to
identify likely targets, then a covariance model search is used to
measure the probability that the targets can fold into the structures
required. Other ncRNAs are added as part of the raw compute stage.
</p>

<p>
The following non-coding RNA gene types are annotated, along with
pseudogenes
</p>

<dl>

<dt>tRNA</dt><dd>transfer RNA</dd>
<dt>Mt-tRNA</dt><dd>transfer RNA located in the mitochondrial genome</dd>
<dt>rRNA</dt><dd>ribosomal RNA</dd>
<dt>scRNA</dt><dd>small cytoplasmic RNA</dd>
<dt>snRNA</dt><dd>small nuclear RNA</dd> 
<dt>snoRNA</dt><dd>small nucleolar RNA</dd> 
<dt>miRNA</dt><dd>microRNA precursors</dd> 
<dt>misc_RNA</dt><dd>miscellaneous other RNA</dd>
<dt>lincRNA</dt><dd>Long intergenic non-coding RNAs</dd>

</dl>

<h2 id="Details">Annotation Details</h2>

<p>
Most <strong>ncRNAs</strong> are annotated by aligning genomic
sequence against <a rel="external"
href="http://rfam.sanger.ac.uk/">RFAM</a> using <a rel="external"
href="http://blast.wustl.edu/">BLASTN</a>. The BLAST hits are
clustered and filtered by E value and are used to seed Infernal
searches of the locus with the corresponding RFAM covariance
models. The purpose of this is to reduce the search space required, as
to scan the entire genome with all the RFAM covariance models would be
extremely CPU-intensive. The resulting BLAST hits are then used as
supporting evidence for ncRNA genes.
</p>

<p>
<strong>miRNAs</strong> are predicted by BLASTN of genomic sequence
slices against <a rel="external"
href="http://www.mirbase.org/">miRBase</a> sequences. All species are
used. The BLAST hits are clustered and filtered by E value and the
aligned genomic sequence is then checked for possible secondary
structure using RNAFold. If evidence is found that the genomic
sequence could form a stable hairpin structure, the locus is used to
create a miRNA gene model. The resulting BLAST hit is used as
supporting evidence for the miRNA gene.
</p>

<p>
Note: The miRNA identifier and name are only associated to the
resulting Ensembl miRNA if they are of the same species.
</p>

<p>
<strong>tRNAs</strong> are annotated as part of the raw compute
process using <a rel="external"
href="http://lowelab.ucsc.edu/tRNAscan-SE/">tRNAscan-SE</a>.
</p>

<p>
<strong>lincRNA</strong> (Long intergenic non-coding RNAs)
Ensembl gene annotation, cDNA alignments and chromatin-state map data 
from the Ensembl regulatory build are used to predict lincRNAs for human 
and mouse. We do not import the lincRNAs identified by Guttman et al [1], 
but their publication guided us to our current approach for automatically annotating lincRNAs. 
First, regions of chromatin methylation (H3K4me3 and H3K36me3) 
outside known protein-coding loci are identified. Next, cDNAs which 
overlap with H3K4me3 or H3K36me3 features are identified as candidate 
lincRNAs. A final evaluation step investigates if each candidate 
lincRNA has any protein-coding potential. Any candidate lincRNA 
containing a substantial open reading frame (ORF) covering 35% or more 
of its length and containing PFAM/tigrfam protein domains will be 
rejected. Candidate lincRNAs that pass the final evaluation step are 
included in the human or mouse gene set as lincRNA genes.
</p>

<p>References</p>

<ol>
<li><p>Guttman M, Amit I, Garber M, French C, Lin MF, Feldser D, Huarte M,
Zuk O, Carey BW, Cassady JP, Cabili MN, Jaenisch R, Mikkelsen TS,
Jacks T, Hacohen N, Bernstein BE, Kellis M, Regev A, Rinn JL, Lander ES<br />
<strong>Chromatin signature reveals over a thousand highly conserved
large non-coding RNAs in mammals.</strong><br />
<em>Nature</em> 2009 458:223-227<br />
<a href="http://dx.doi.org/10.1038/nature07672"
target="external">doi:10.1038/nature07672</a></p>
</li>

</ol>

</body>
</html>
