<html>
<head>
<title>Compara API Tutorial</title>
</head>

<body>

<h1>Compara API Tutorial</h1>


<h2 id="intro">Introduction</h2>

<p>
This tutorial is an introduction to the Ensembl Compara API. Knowledge of the 
<a href="/info/docs/api/core_tutorial/index.html">Ensembl Core API</a> and of the
<a href="/info/docs/api/general_instructions.html">coding conventions</a> 
used in the Ensembl APIs is assumed.
</p>
<p>
Documentation about the Compara database schema is available
<a href="compara_schema.html">here</a>, and while is not necessary for this tutorial,
an understanding of the database tables may help as many of the adaptor modules are table-specific.
</p>

<h2>Species and databases in Ensembl Compara</h2>
<p>
The compara API can be used to access the main Ensembl database as well as the non-vertebrate Ensembl Genomes databases.
Details of how to access these databases with your registry are found in the 
<a href="/info/docs/api/general_instructions.html">general instructions</a> for the Ensembl APIs.
There is no comparative genomics data in the Ensembl GRCh37 database.
</p>

<p>
When using Compara adaptors, we specify the species as <kbd>'Multi'</kbd> for vertebrates; 
for non vertebrates, we name the division, such as <kbd>'Plants'</kbd> or <kbd>'Fungi'</kbd>.
The <a href="https://plants.ensembl.org/info/genome/compara/pan_compara.html">pan-taxonomic comparative analysis</a> 
can be accessed by specifying as 'pan_homology'.
</p>


<h2>Main Ensembl Compara objects</h2>

<p>
The Ensembl Compara databases uses a number of objects that share some similarities to objects in other Ensembl APIs, but are not exactly the same. 
Here are some examples of object types and how they differ from their counterparts in other APIs:
</p>

<ul>
<li><a href="/info/docs/Doxygen/compara-api/classBio_1_1EnsEMBL_1_1Compara_1_1GenomeDB.html">GenomeDB</a> objects represent a genome in the database, usually a single species but it may also be a strain of a species. It contains information about the taxonomy and is usually fetched with the
<a href="/info/docs/Doxygen/compara-api/classBio_1_1EnsEMBL_1_1Compara_1_1DBSQL_1_1GenomeDBAdaptor.html">GenomeDBAdaptor</a> 
and the species name.
Unlike the Core, Variation and Regulation databases, where each genome has its own database and this is specified in the adaptor,
the compara databases are multi-genome, which means each genome must be represented as an object within the database.</li>
<li><a href="/info/docs/Doxygen/compara-api/classBio_1_1EnsEMBL_1_1Compara_1_1DnaFrag.html">DnaFrag</a> objects are similar to the 
seq_regions attached to Features in other databases, and represent chromosomes, contigs or scaffolds.</li>
<li><a href="/info/docs/Doxygen/compara-api/classBio_1_1EnsEMBL_1_1Compara_1_1SpeciesSet.html">SpeciesSets</a>
are groups of species.</li>
<li>A <a href="/info/docs/Doxygen/compara-api/classBio_1_1EnsEMBL_1_1Compara_1_1Method.html">Method</a> is a type of analysis that can be done on a SpeciesSet.</a>
<li>A <a href="/info/docs/Doxygen/compara-api/classBio_1_1EnsEMBL_1_1Compara_1_1MethodLinkSpeciesSet.html">MethodLinkSpeciesSet</a> combines a Method with a SpeciesSet to specify a particular analysis.</li>
</ul>

<h3>MethodLinkSpeciesSets</h3>

<p>
The compara API works by specifying which type of analysis and set of species you're working with using the 
<a href="/info/docs/Doxygen/compara-api/classBio_1_1EnsEMBL_1_1Compara_1_1MethodLinkSpeciesSet.html">MethodLinkSpeciesSet</a> object.
This is fetched using the <a href="/info/docs/Doxygen/compara-api/classBio_1_1EnsEMBL_1_1Compara_1_1DBSQL_1_1MethodLinkSpeciesSetAdaptor.html">MethodLinkSpeciesSetAdaptor</a>.
</p>
<p>
A MethodLinkSpeciesSet is obtained using its species and its Method. 
The Method represents the type of analysis, and includes gene homology methods and whole genome alignment methods.
Below is a non-exhaustive list of methods available:
</p>
    <table width="800">
        <tr><th align="left">Method name</th><th align="left">Description</th></tr>
        <tr><td><kbd>LASTZ_NET</kbd></td><td>Pairwise whole genome alignment using LastZ</td></tr>
        <tr><td><kbd>EPO</kbd></td><td>Multiple whole genome alignment (WGA), with ancestral inference, using Enredo, Pecan and Ortheus</td></tr>
        <tr><td><kbd>EPO_EXTENDED</kbd></td><td>Multiple WGA for lower quality genomes, extending EPO using LASTZ</td></tr>
        <tr><td><kbd>PECAN</kbd></td><td>Multiple WGA using Mercator and Pecan</td></tr>
        <tr><td><kbd>CACTUS_HAL</kbd></td><td>Multiple WGA using progressiveCactus (* additional setup required - see <a href="https://github.com/Ensembl/ensembl-compara/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/README.md#api-to-access-hal-alignments-progressive-cactus">here</a> for details)</td></tr>
        <tr><td><kbd>ENSEMBL_ORTHOLOGUES</kbd></td><td>Orthologue calls between a pair of species</td></tr>
        <tr><td><kbd>ENSEMBL_PARALOGUES</kbd></td><td>Paralogue calls for a species</td></tr>
        <tr><td><kbd>PROTEIN_TREES</kbd></td><td>Gene trees for protein coding genes</td></tr>
        <tr><td><kbd>NC_TREES</kbd></td><td>Gene trees for non-coding RNAs</td></tr>
    </table>
    <br>
</p>
<p>
For pairwise MethodLinkSpeciesSets, such as pairwise orthologues or LASTZ_NET pairwise whole genome alignments,
you can get the MethodLinkSpeciesSet using the species name (known as the <kbd>registry_alias</kbd>), 
which can be the common name or the latin name, or by the <a href="/info/docs/Doxygen/compara-api/classBio_1_1EnsEMBL_1_1Compara_1_1GenomeDB.html">GenomeDB</a>.
</p>
<p>
For multiple alignment MethodLinkSpeciesSets, such as EPO, PECAN or CACTUS_HAL, 
these can be fetched using the taxonomy-based name of the group, such as "mammals", "amniotes" or "murinae".
The <a href="/info/genome/compara/multiple_genome_alignments.html">documentation for the alignments</a> 
includes the terms you need to fetch these alignments with the MethodLinkSpeciesSetAdaptor.
</p>
<p>
It is possible to find the MethodLinkSpeciesSet IDs, and fetch using these, but we do not recommend this,
particularly for the multiple alignments, as when new species are added to these alignments, 
new IDs are assigned, so your scripts will not work in new releases. 
</p>
<p>
The following script fetches the LastZ-net alignment between human and mouse, 
using the MethodLinkSpeciesSet and GenomeDBs:
</p>

<pre class="code sh_perl">
use strict;
use warnings;
use Bio::EnsEMBL::Registry;

my $registry = 'Bio::EnsEMBL::Registry';

$registry->load_registry_from_db(
  -host => 'ensembldb.ensembl.org',
  -user => 'anonymous'
);

# Get the GenomeDB adaptor
my $genome_db_adaptor = $registry->get_adaptor( 'Multi', 'compara', 'GenomeDB' );

# Fetch GenomeDB objects for human and mouse:
my $human_genome_db = $genome_db_adaptor->fetch_by_name_assembly('homo_sapiens');
my $mouse_genome_db = $genome_db_adaptor->fetch_by_name_assembly('mus_musculus');

# Get the MethodLinkSpeciesSet adaptor
my $method_link_species_set_adaptor = $registry->get_adaptor( 'Multi', 'compara', 'MethodLinkSpeciesSet');

# Fetch the MethodLinkSpeciesSet object corresponding to LASTZ_NET alignments between human and mouse genomic sequences
my $human_mouse_lastz_net_mlss = $method_link_species_set_adaptor->fetch_by_method_link_type_GenomeDBs( "LASTZ_NET", [$human_genome_db, $mouse_genome_db] );
</pre>
<div class="code sh_perl" style="margin-bottom:15px">
  <div style="float:left;font-weight:bold">Used objects:</div>
  <div style="float:left;margin-left:10px;border-left:1px dotted #CCC">
    <span style="padding-left:0.9em">Adaptor objects</span>
    <ul style="margin-bottom:0px">
      <li><a style="text-decoration:none" href="/info/docs/Doxygen/compara-api/classBio_1_1EnsEMBL_1_1Compara_1_1DBSQL_1_1GenomeDBAdaptor.html">GenomeDBAdaptor</a></li>
      <li><a style="text-decoration:none" href="/info/docs/Doxygen/compara-api/classBio_1_1EnsEMBL_1_1Compara_1_1DBSQL_1_1MethodLinkSpeciesSetAdaptor.html">MethodLinkSpeciesSetAdaptor</a></li>
    </ul>
  </div>
  <div style="float:left;margin-left:10px;border-left:1px dotted #CCC">
    <span style="padding-left:0.9em">Main objects</span>
    <ul style="margin-bottom:0px">
      <li><a style="text-decoration:none" href="/info/docs/Doxygen/compara-api/classBio_1_1EnsEMBL_1_1Compara_1_1GenomeDB.html">GenomeDB</a></li>
      <li><a href="/info/docs/Doxygen/compara-api/classBio_1_1EnsEMBL_1_1Compara_1_1MethodLinkSpeciesSet.html">MethodLinkSpeciesSet</a></li>
    </ul>
  </div>
  <div style="clear:both"></div>
</div>

<div style="width:800px" class="info">
  <h3>Registry</h3>
  <div class="message-pad">
	<p>What is this object $registry? Make sure you have defined it in all your scripts. Learn more in the <a href="/info/docs/api/general_instructions.html#connectin">general instructions page</a>.
	</p>
	</div>
</div>

<h3>Ensembl Compara Adaptors</h3>

<p>
Below is a non-exhaustive list of Ensembl Compara adaptors that are most often used:
</p>

<ul>
<li><a style="text-decoration:none" href="/info/docs/Doxygen/compara-api/classBio_1_1EnsEMBL_1_1Compara_1_1DBSQL_1_1GenomeDBAdaptor.html"> GenomeDBAdaptor</a> to fetch Bio::EnsEMBL::Compara::<a style="text-decoration:none" href="/info/docs/Doxygen/compara-api/classBio_1_1EnsEMBL_1_1Compara_1_1GenomeDB.html">GenomeDB</a> objects</li>
    <li><a style="text-decoration:none" href="/info/docs/Doxygen/compara-api/classBio_1_1EnsEMBL_1_1Compara_1_1DBSQL_1_1DnaFragAdaptor.html">DnaFragAdaptor</a> to fetch Bio::EnsEMBL::Compara::<a style="text-decoration:none" href="/info/docs/Doxygen/compara-api/classBio_1_1EnsEMBL_1_1Compara_1_1DnaFrag.html">DnaFrag</a> objects</li>
    <li><a style="text-decoration:none" href="/info/docs/Doxygen/compara-api/classBio_1_1EnsEMBL_1_1Compara_1_1DBSQL_1_1GenomicAlignBlockAdaptor.html">GenomicAlignBlockAdaptor</a> to fetch Bio::EnsEMBL::Compara::<a style="text-decoration:none" href="/info/docs/Doxygen/compara-api/classBio_1_1EnsEMBL_1_1Compara_1_1GenomicAlignBlock.html">GenomicAlignBlock</a> objects</li>
    <li><a style="text-decoration:none" href="/info/docs/Doxygen/compara-api/classBio_1_1EnsEMBL_1_1Compara_1_1DBSQL_1_1SyntenyRegionAdaptor.html">SyntenyRegionAdaptor</a> to fetch Bio::EnsEMBL::Compara::<a style="text-decoration:none" href="/info/docs/Doxygen/compara-api/classBio_1_1EnsEMBL_1_1Compara_1_1SyntenyRegion.html">SyntenyRegion</a> objects</li>
    <li><a style="text-decoration:none" href="/info/docs/Doxygen/compara-api/classBio_1_1EnsEMBL_1_1Compara_1_1DBSQL_1_1GeneMemberAdaptor.html">GeneMemberAdaptor</a> to fetch Bio::EnsEMBL::Compara::<a style="text-decoration:none" href="/info/docs/Doxygen/compara-api/classBio_1_1EnsEMBL_1_1Compara_1_1GeneMember.html">GeneMember</a> objects</li>
    <li><a style="text-decoration:none" href="/info/docs/Doxygen/compara-api/classBio_1_1EnsEMBL_1_1Compara_1_1DBSQL_1_1GeneTreeAdaptor.html">GeneTreeAdaptor</a> to fetch Bio::EnsEMBL::Compara::<a style="text-decoration:none" href="/info/docs/Doxygen/compara-api/classBio_1_1EnsEMBL_1_1Compara_1_1GeneTree.html">GeneTree</a> objects</li>
    <li><a style="text-decoration:none" href="/info/docs/Doxygen/compara-api/classBio_1_1EnsEMBL_1_1Compara_1_1DBSQL_1_1HomologyAdaptor.html">HomologyAdaptor</a> to fetch Bio::EnsEMBL::Compara::<a style="text-decoration:none" href="/info/docs/Doxygen/compara-api/classBio_1_1EnsEMBL_1_1Compara_1_1Homology.html">Homology</a> objects</li>
</ul>

<p>
Only some of these adaptors will be used for illustration as part of this tutorial through commented perl scripts code.
</p>



<h2 id="genomic_align">Whole Genome Alignments</h2>

<p>
The Compara database contains a number of different types of whole genome alignments, 
including <a href="/info/genome/compara/multiple_genome_alignments.html">multiple alignments</a> 
and <a href="http://www.ensembl.org/info/genome/compara/analyses.html">pairwise alignments</a>. 
</p>

<h3>GenomicAlignBlock objects</h3>

<p>
<a href="/info/docs/Doxygen/compara-api/classBio_1_1EnsEMBL_1_1Compara_1_1GenomicAlignBlock.html">GenomicAlignBlocks</a> 
are the preferred way to store and fetch genomic alignments.
A GenomicAlignBlock contains several <a href="/info/docs/Doxygen/compara-api/classBio_1_1EnsEMBL_1_1Compara_1_1GenomicAlign.html">GenomicAlign</a> objects.
Every GenomicAlign object corresponds to a piece of genomic sequence from one genome 
aligned with another GenomicAlign from another genome in the same GenomicAlignBlock.
A GenomicAlign object is always in relation with other GenomicAlign objects
and this relation is defined through the GenomicAlignBlock object.
Therefore the usual way to fetch genomic alignments is by fetching GenomicAlignBlock objects.
</p>

<p>
You can fetch GenomicAlignBlocks using <kbd>Slice</kbd> objects. 
When you fetch GenomicAlignBlocks this way, the blocks will not just represent your Slice, 
but the whole block of alignment.
This means that you will need to restrict the block once you have fetched it.
You will always get an array of blocks.
</p>

<pre class="code sh_perl">
use strict;
use warnings;
use Bio::EnsEMBL::Registry;

my $registry = 'Bio::EnsEMBL::Registry';

$registry->load_registry_from_db(
  -host => 'ensembldb.ensembl.org',
  -user => 'anonymous'
);

# Define the query species and the coordinates of the Slice
my $query_species = 'human';
my $seq_region = '14';
my $seq_region_start = 75000000;
my $seq_region_end   = 75010000;

# Get the SliceAdaptor and fetch a slice
my $slice_adaptor = $registry->get_adaptor( $query_species, 'core', 'Slice' );
my $query_slice = $slice_adaptor->fetch_by_region( 'toplevel', $seq_region, $seq_region_start, $seq_region_end );

# Get the GenomeDB adaptor
my $genome_db_adaptor = $registry->get_adaptor( 'Multi', 'compara', 'GenomeDB' );

# Fetch GenomeDB objects for human and mouse:
my $human_genome_db = $genome_db_adaptor->fetch_by_name_assembly('homo_sapiens');
my $mouse_genome_db = $genome_db_adaptor->fetch_by_name_assembly('mus_musculus');

# Get the MethodLinkSpeciesSetAdaptor
my $method_link_species_set_adaptor = $registry->get_adaptor( 'Multi', 'compara', 'MethodLinkSpeciesSet');

# Fetch the MethodLinkSpeciesSet object corresponding to LASTZ_NET alignments between human and mouse genomic sequences
my $human_mouse_lastz_net_mlss = $method_link_species_set_adaptor->fetch_by_method_link_type_GenomeDBs( "LASTZ_NET", [$human_genome_db, $mouse_genome_db] );

# Get the GenomicAlignBlockAdaptor
my $genomic_align_block_adaptor = $registry->get_adaptor( 'Multi', 'compara', 'GenomicAlignBlock' );

# Fetch all the GenomicAlignBlocks corresponding to this Slice from the pairwise alignments (LASTZ_NET) between human and mouse
my @genomic_align_blocks = @{ $genomic_align_block_adaptor->fetch_all_by_MethodLinkSpeciesSet_Slice( $human_mouse_lastz_net_mlss, $query_slice ) };

# We will then (usually) need to restrict the blocks to the required positions in the reference sequence

foreach my $genomic_align_block( @genomic_align_blocks ) {
    my $restricted_gab = $genomic_align_block->restrict_between_reference_positions($seq_region_start, $seq_region_end);
}
</pre>
<div class="code sh_perl" style="margin-bottom:15px">
  <div style="float:left;font-weight:bold">Used objects:</div>
  <div style="float:left;margin-left:10px;border-left:1px dotted #CCC">
    <span style="padding-left:0.9em">Adaptor objects</span>
    <ul style="margin-bottom:0px">
      <li><a style="text-decoration:none" href="/info/docs/Doxygen/compara-api/classBio_1_1EnsEMBL_1_1Compara_1_1DBSQL_1_1GenomeDBAdaptor.html">GenomeDBAdaptor</a></li>
      <li><a style="text-decoration:none" href="/info/docs/Doxygen/compara-api/classBio_1_1EnsEMBL_1_1Compara_1_1DBSQL_1_1MethodLinkSpeciesSetAdaptor.html">MethodLinkSpeciesSetAdaptor</a></li>
      <li><a style="text-decoration:none" href="/info/docs/Doxygen/core-api/classBio_1_1EnsEMBL_1_1DBSQL_1_1SliceAdaptor.html">SliceAdaptor</a> <i>(Core API)</i></li>
      <li><a style="text-decoration:none" href="/info/docs/Doxygen/compara-api/classBio_1_1EnsEMBL_1_1Compara_1_1DBSQL_1_1GenomicAlignBlockAdaptor.html">GenomicAlignBlockAdaptor</a></li>
    </ul>
  </div>
  <div style="float:left;margin-left:10px;border-left:1px dotted #CCC">
    <span style="padding-left:0.9em">Main objects</span>
    <ul style="margin-bottom:0px">
      <li><a style="text-decoration:none" href="/info/docs/Doxygen/compara-api/classBio_1_1EnsEMBL_1_1Compara_1_1GenomeDB.html">GenomeDB</a></li>
      <li><a href="/info/docs/Doxygen/compara-api/classBio_1_1EnsEMBL_1_1Compara_1_1MethodLinkSpeciesSet.html">MethodLinkSpeciesSet</a></li>
      <li><a style="text-decoration:none" href="/info/docs/Doxygen/core-api/classBio_1_1EnsEMBL_1_1Slice.html">Slice</a> <i>(Core API)</i></li>
      <li><a style="text-decoration:none" href="/info/docs/Doxygen/compara-api/classBio_1_1EnsEMBL_1_1Compara_1_1GenomicAlignBlock.html">GenomicAlignBlock</a></li>
    </ul>
  </div>
  <div style="clear:both"></div>
</div>

<h3>Print alignment</h3>

<p>
Once you've retrieved you alignment, you can use modules in BioPerl to print your alignment,
such as Bio::SimpleAlign, from the GenomicAlignBlocks. 
The following code can be added to the end of the previous script to get SimpleAligns and print them to a file:
<p>
<pre class="code sh_perl">
my $all_aligns;

# Get a Bio::SimpleAlign object from every GenomicAlignBlock
foreach my $this_genomic_align_block (@genomic_align_blocks) {
	my $restricted_gab = $this_genomic_align_block->restrict_between_reference_positions($seq_region_start, $seq_region_end);
    my $simple_align = $restricted_gab->get_SimpleAlign;
    push(@$all_aligns, $simple_align);
}

# print all the genomic alignments using a Bio::AlignIO object
my $output_format = "clustalw";
my $alignIO = Bio::AlignIO->newFh(
    -interleaved => 0,
    -fh => \*STDOUT,
    -format => $output_format,
    -idlength => 10
);

foreach my $this_align (@$all_aligns) {
    print $alignIO $this_align;
}
</pre>
<div class="code sh_perl" style="margin-bottom:15px">
  <div style="float:left;font-weight:bold">Used objects:</div>
  <div style="float:left;margin-left:10px;border-left:1px dotted #CCC">
    <span style="padding-left:0.9em">Adaptor objects</span>
    <ul style="margin-bottom:0px">
    </ul>
  </div>
  <div style="float:left;margin-left:10px;border-left:1px dotted #CCC">
    <span style="padding-left:0.9em">Main objects</span>
    <ul style="margin-bottom:0px">

      <li><a style="text-decoration:none" href="/info/docs/Doxygen/compara-api/classBio_1_1EnsEMBL_1_1Compara_1_1GenomicAlignBlock.html">GenomicAlignBlock</a></li>
    </ul>
  </div>
  <div style="clear:both"></div>
</div>

<h3>Print coordinates</h3>
<p>
From a GenomicAlign object, you can create a Slice for that species, 
which you can then use to fetch coordinates or sequence, or perform any other Slice functions, like find overlapping features.
Taking again the script where we fetched the GenomicAlignBlocks, we can add the following code to get coordinates:
</p>
<pre class="code sh_perl">
foreach my $genomic_align_block( @genomic_align_blocks ) {
    my $restricted_gab = $genomic_align_block->restrict_between_reference_positions($seq_region_start, $seq_region_end);
    
    # fetch the GenomicAligns and move through
    my @genomic_aligns = @ { $restricted_gab->get_all_GenomicAligns };
    foreach my $genomic_align (@genomic_aligns) {
    	my $species = $genomic_align->genome_db->get_scientific_name;
    	my $slice = $genomic_align->get_Slice;
    	print $species, "\t", $slice->seq_region_name, ":", $slice->seq_region_start, "-", $slice->seq_region_end, "\t";
    }
    print "\n";
}
</pre>
<div class="code sh_perl" style="margin-bottom:15px">
  <div style="float:left;font-weight:bold">Used objects:</div>
  <div style="float:left;margin-left:10px;border-left:1px dotted #CCC">
    <span style="padding-left:0.9em">Adaptor objects</span>
    <ul style="margin-bottom:0px">
    </ul>
  </div>
  <div style="float:left;margin-left:10px;border-left:1px dotted #CCC">
    <span style="padding-left:0.9em">Main objects</span>
    <ul style="margin-bottom:0px">
      <li><a style="text-decoration:none" href="/info/docs/Doxygen/compara-api/classBio_1_1EnsEMBL_1_1Compara_1_1GenomicAlignBlock.html">GenomicAlignBlock</a></li>
      <li><a style="text-decoration:none" href="/info/docs/Doxygen/compara-api/classBio_1_1EnsEMBL_1_1Compara_1_1GenomicAlign.html">GenomicAlign</a></li>
      <li><a style="text-decoration:none" href="/info/docs/Doxygen/core-api/classBio_1_1EnsEMBL_1_1Slice.html">Slice</a> <i>(Core API)</i></li>
    </ul>
  </div>
  <div style="clear:both"></div>
</div>

<h2 id="gene_resources">Gene trees, Homologies and Protein clusters</h2>

<p>
All the gene trees and homologies refer to GeneMembers and SeqMembers. 
Homology objects store orthologous and paralogous relationships between members, 
GeneTree objects represent the evolutionary history of a set of members.
</p>


<h3>*Member objects</h3>

<p>
A member represent either a gene (GeneMember) or a sequence-bearing locus, e.g. a protein or a transcript (SeqMember). Most of them
are defined in the corresponding Ensembl core database. For instance, the sequence for the human gene ENSG00000004059
is stored in the human core database.
</p>

<p>
The fetch_by_stable_id_GenomeDB method of the corresponding *MemberAdaptor returns Members by their stable_id and genome.
Here is a simple example:
</p>

<pre class="code sh_perl">
use strict;
use warnings;
use Bio::EnsEMBL::Registry;

my $registry = 'Bio::EnsEMBL::Registry';

$registry->load_registry_from_db(
  -host => 'ensembldb.ensembl.org',
  -user => 'anonymous'
);

# get the GenomeDB adaptor
my $genome_db_adaptor = $registry->get_adaptor('Multi','compara','GenomeDB');

# fetch GenomeDB object for human
my $human_genome_db = $genome_db_adaptor->fetch_by_name_assembly('homo_sapiens');

# get the MemberAdaptor
my $genemember_adaptor = $registry->get_adaptor('Multi','compara','GeneMember');

# fetch a human Member
my $member = $genemember_adaptor->fetch_by_stable_id_GenomeDB('ENSG00000004059', $human_genome_db);

# print out some information about the Member
print $member->source_name, " (", $member->dnafrag->name, ":", $member->dnafrag_start, "-", $member->dnafrag_end, "): ", $member->description, "\n";
</pre>
<div class="code sh_perl" style="margin-bottom:15px">
  <div style="float:left;font-weight:bold">Used objects:</div>
  <div style="float:left;margin-left:10px;border-left:1px dotted #CCC">
    <span style="padding-left:0.9em">Adaptor objects</span>
    <ul style="margin-bottom:0px">
      <li><a style="text-decoration:none" href="/info/docs/Doxygen/compara-api/classBio_1_1EnsEMBL_1_1Compara_1_1DBSQL_1_1GeneMemberAdaptor.html">GeneMemberAdaptor</a></li>
    </ul>
  </div>
  <div style="float:left;margin-left:10px;border-left:1px dotted #CCC">
    <span style="padding-left:0.9em">Main objects</span>
    <ul style="margin-bottom:0px">Ensembl
      <li><a style="text-decoration:none" href="/info/docs/Doxygen/compara-api/classBio_1_1EnsEMBL_1_1Compara_1_1GeneMember.html">GeneMember</a></li>
    </ul>
  </div>
  <div style="clear:both"></div>
</div>

<p>
You can fetch the corresponding <a href="/info/docs/api/core_tutorial/index.html">Ensembl Core API</a> 
objects from all of the *Member objects, as well as coordinates. The <kbd>taxon</kbd> method returns an 
<a href="http://www.ensembl.org/info/docs/Doxygen/compara-api/classBio_1_1EnsEMBL_1_1Compara_1_1NCBITaxon.html">NCBITaxon</a> 
object, which contains information about the species and taxonomy.
</p>

<p>
*Member objects have a <kbd>source_name</kbd>, which describes where the member comes from:
</p>
<dl>
<dt>for GeneMember</dt>
<dd>
  <ul>
  <li>ENSEMBLGENE, derived from an Ensembl gene</li>
  <li>EXTERNALGENE, loaded from an external source (currently unused in the live databases)</li>
  </ul>
</dd>
<dt>for SeqMember</dt>
<dd>
  <ul>
  <li>ENSEMBLPEP, derived from an Ensembl translation</li>
  <li>ENSEMBLTRANS, derived from an Ensembl transcript</li>
  <li>Uniprot/SWISSPROT, derived from a Uniprot/Swissprot entry</li>
  <li>Uniprot/SPTREMBL, derived from a Uniprot/SP-TrEMBL entry</li>
  </ul>
</dd>
</dl>

<pre class="code sh_perl">
use strict;
use warnings;
use Bio::EnsEMBL::Registry;

my $registry = 'Bio::EnsEMBL::Registry';

$registry->load_registry_from_db(
  -host => 'ensembldb.ensembl.org',
  -user => 'anonymous'
);

# get the GenomeDB adaptor
my $genome_db_adaptor = $registry->get_adaptor('Multi','compara','GenomeDB');

# fetch GenomeDB object for human
my $human_genome_db = $genome_db_adaptor->fetch_by_name_assembly('homo_sapiens');

# get the MemberAdaptor
my $genemember_adaptor = $registry->get_adaptor('Multi','compara','GeneMember');

# fetch a human Member
my $member = $genemember_adaptor->fetch_by_stable_id_GenomeDB('ENSG00000004059', $human_genome_db);

my $taxon = $member->taxon;
print "common_name ", $taxon->get_common_name,"\ngenus ", $taxon->genus,"\nspecies ", $taxon->species, "\nbinomial ", $taxon->scientific_name,  "\nclassification ", $taxon->classification,"\n";
</pre>
<div class="code sh_perl" style="margin-bottom:15px">
  <div style="float:left;font-weight:bold">Used objects:</div>
  <div style="float:left;margin-left:10px;border-left:1px dotted #CCC">
    <span style="padding-left:0.9em">Adaptor objects</span>
    <ul style="margin-bottom:0px">
      <li><a style="text-decoration:none" href="/info/docs/Doxygen/compara-api/classBio_1_1EnsEMBL_1_1Compara_1_1DBSQL_1_1GeneMemberAdaptor.html">GeneMemberAdaptor</a></li>
    </ul>
  </div>
  <div style="float:left;margin-left:10px;border-left:1px dotted #CCC">
    <span style="padding-left:0.9em">Main objects</span>
    <ul style="margin-bottom:0px">Ensembl
      <li><a style="text-decoration:none" href="/info/docs/Doxygen/compara-api/classBio_1_1EnsEMBL_1_1Compara_1_1GeneMember.html">GeneMember</a></li>
      <li><a style="text-decoration:none" href="/info/docs/Doxygen/compara-api/classBio_1_1EnsEMBL_1_1Compara_1_1NCBITaxon.html">NCBITaxon</a></li>
    </ul>
  </div>
  <div style="clear:both"></div>
</div>

<h3>GeneTree Objects</h3>

<p>
<a style="text-decoration:none" href="/info/docs/Doxygen/compara-api/classBio_1_1EnsEMBL_1_1Compara_1_1GeneTree.html">GeneTree</a> 
objects give us the phylogenetic context for a set of genes, as well as their alignment.
</p>
<p>
In general, you would want to fetch the gene tree for a given gene of interest. The 
<a style="text-decoration:none" href="/info/docs/Doxygen/compara-api/classBio_1_1EnsEMBL_1_1Compara_1_1DBSQL_1_1GeneTreeAdaptor.html">GeneTreeAdaptor</a> 
has a fetching method called <kbd>fetch_all_by_Member()</kbd>. You will need the GeneMember object for your
query gene, therefore you will fetch the GeneMember first like in this example:
</p>

<pre class="code sh_perl">
use strict;
use warnings;
use Bio::EnsEMBL::Registry;

my $registry = 'Bio::EnsEMBL::Registry';

$registry->load_registry_from_db(
  -host => 'ensembldb.ensembl.org',
  -user => 'anonymous'
);

# first, let's use a GenomeDBAdaptor to get the GenomeDB of interest
my $genome_db_adaptor = $registry->get_adaptor( 'Multi', 'compara', 'GenomeDB' );
my $human_genome_db = $genome_db_adaptor->fetch_by_name_assembly('homo_sapiens');

# next, get our human GeneMember of interest from a GeneMemberAdaptor
my $genemem_adapt = $registry->get_adaptor( 'Multi', 'compara', 'GeneMember' );
my $genemem = $genemem_adapt->fetch_by_stable_id_GenomeDB('ENSG00000238344', $human_genome_db);

# then, set up a GeneTreeAdaptor and fetch the default tree for our GeneMember
my $genetree_adapt = $registry->get_adaptor( 'Multi', 'compara', 'GeneTree' );
my $genetree = $genetree_adapt->fetch_default_for_Member($genemem);

# look at all members of the tree
print "Members of tree:\n";
my @members = @{ $genetree->get_all_Members };
foreach my $m ( @members ) {
    print $m->name, " (", $m->genome_db->name, ")\n";
}

# print the full tree in Newick format
print $genetree->newick_format() . "\n";
</pre>
<div class="code sh_perl" style="margin-bottom:15px">
  <div style="float:left;font-weight:bold">Used objects:</div>
  <div style="float:left;margin-left:10px;border-left:1px dotted #CCC">
    <span style="padding-left:0.9em">Adaptor objects</span>
    <ul style="margin-bottom:0px">
      <li><a style="text-decoration:none" href="/info/docs/Doxygen/compara-api/classBio_1_1EnsEMBL_1_1Compara_1_1DBSQL_1_1GeneMemberAdaptor.html">GeneMemberAdaptor</a></li>
      <li><a style="text-decoration:none" href="/info/docs/Doxygen/compara-api/classBio_1_1EnsEMBL_1_1Compara_1_1DBSQL_1_1GeneTreeAdaptor.html">GeneTreeAdaptor</a></li>
    </ul>
  </div>
  <div style="float:left;margin-left:10px;border-left:1px dotted #CCC">
    <span style="padding-left:0.9em">Main objects</span>
    <ul style="margin-bottom:0px">Ensembl
      <li><a style="text-decoration:none" href="/info/docs/Doxygen/compara-api/classBio_1_1EnsEMBL_1_1Compara_1_1GeneMember.html">GeneMember</a></li>
      <li><a style="text-decoration:none" href="/info/docs/Doxygen/compara-api/classBio_1_1EnsEMBL_1_1Compara_1_1GeneTree.html">GeneTree</a></li>
    </ul>
  </div>
  <div style="clear:both"></div>
</div>

<p>
GeneTree objects not only hold the structure of the phylogeny, they also hold the gene alignment upon which the tree was based. This can be printed out using the following:
</p>

<pre class="code sh_perl">
    $genetree->print_alignment_to_file('/path/to/file', -format=>'clustalw');
</pre>

<h3>Homology Objects</h3>

<p>
An <a style="text-decoration:none" href="/info/docs/Doxygen/compara-api/classBio_1_1EnsEMBL_1_1Compara_1_1Homology.html">Homology</a> 
object represents either an orthologous or paralogous relationships between two members.
</p>

<p>
Typically you want to get homologies for a given gene. As with the GeneTreeAdaptor, the 
<a style="text-decoration:none" href="/info/docs/Doxygen/compara-api/classBio_1_1EnsEMBL_1_1Compara_1_1DBSQL_1_1GeneMemberAdaptor.html">HomologyAdaptor</a> 
has a fetching method called <kbd>fetch_all_by_Member()</kbd>.
This has optional arguments to only fetch by a particular Method, MethodLinkSpeciesSet and target species.
</p>
<p>
When you get all Homologies, you will get an array where each item is a Homology representing the relationship between exactly two genes.
One is the query gene, the gene you used as input, and the other is the target gene, the homologue.
Even if you specify a single species, if there is a one-to-many or many-to-many relationship, each of these will be one homology.
</p>

<pre class="code sh_perl">
use strict;
use warnings;
use Bio::EnsEMBL::Registry;

my $registry = 'Bio::EnsEMBL::Registry';

$registry->load_registry_from_db(
  -host => 'ensembldb.ensembl.org',
  -user => 'anonymous'
);

# get human GenomeDB object
my $genome_db_adaptor = $registry->get_adaptor('Multi', 'compara', 'GenomeDB');
my $human_genome_db = $genome_db_adaptor->fetch_by_name_assembly('homo_sapiens');

# get a GeneMember object
my $gene_member_adaptor = $registry->get_adaptor('Multi', 'compara', 'GeneMember');
my $gene_member = $gene_member_adaptor->fetch_by_stable_id_GenomeDB('ENSG00000004059', $human_genome_db);

# get the homologies where the member is involved
my $homology_adaptor = $registry->get_adaptor('Multi', 'compara', 'Homology');
my @homologies = @ { $homology_adaptor->fetch_all_by_Member($gene_member, -TARGET_SPECIES=>"mus_musculus") };

# the homology_adaptor will always return an array, even if it only has one homology in it, so we move through the array
foreach my $homology (@homologies) {
	
	# Get the GeneMembers and print their species and ID
	my @homologous_genes = @ { $homology->get_all_GeneMembers };
		foreach my $gene (@homologous_genes){
			print $gene->taxon->get_common_name, ": ", $gene->stable_id, "\n";
		}
	# Print the homology relationship
	print $homology->description," ", $homology->taxonomy_level,"\n";
}
</pre>
<div class="code sh_perl" style="margin-bottom:15px">
  <div style="float:left;font-weight:bold">Used objects:</div>
  <div style="float:left;margin-left:10px;border-left:1px dotted #CCC">
    <span style="padding-left:0.9em">Adaptor objects</span>
    <ul style="margin-bottom:0px">
      <li><a style="text-decoration:none" href="/info/docs/Doxygen/compara-api/classBio_1_1EnsEMBL_1_1Compara_1_1DBSQL_1_1GeneMemberAdaptor.html">GeneMemberAdaptor</a></li>
      <li><a style="text-decoration:none" href="/info/docs/Doxygen/compara-api/classBio_1_1EnsEMBL_1_1Compara_1_1DBSQL_1_1HomologyAdaptor.html">HomologyAdaptor</a></li>
    </ul>
  </div>
  <div style="float:left;margin-left:10px;border-left:1px dotted #CCC">
    <span style="padding-left:0.9em">Main objects</span>
    <ul style="margin-bottom:0px">Ensembl
      <li><a style="text-decoration:none" href="/info/docs/Doxygen/compara-api/classBio_1_1EnsEMBL_1_1Compara_1_1GeneMember.html">GeneMember</a></li>
      <li><a style="text-decoration:none" href="/info/docs/Doxygen/compara-api/classBio_1_1EnsEMBL_1_1Compara_1_1Homology.html">Homology</a></li>
      <li><a style="text-decoration:none" href="/info/docs/Doxygen/compara-api/classBio_1_1EnsEMBL_1_1Compara_1_1NCBITaxon.html">NCBITaxon</a></li>
    </ul>
  </div>
  <div style="clear:both"></div>
</div>

<p>
You can get the original alignment used to define an homology:
</p>

<pre class="code sh_perl">
    $homology->print_alignment_to_file('/path/to/file', -format=>'fasta');
</pre>

<h2 id="efficiency">Memory management and code efficiency</h2>
<p>There are a myriad of ways to ensure optimal performance of your code. Here, we provide just a few tricks that pertain specifically to Ensembl APIs.</p>
<h4>Fetching from adaptors</h4>
<p>
In all examples above, we've used a <kbd>foreach</kbd> to loop through the results of a <kbd>fetch_all_by_YY</kbd> call. This is fine for small datasets. However, adaptors can
also support a <kbd>while</kbd> + <kbd>shift</kbd> combo for memory efficiency:
</p>

<pre class="code sh_perl">
my $members = $genetree->get_all_Members;
while ( my $m = shift @$members ) {
    # do something with member
}
</pre>

<h4>Releasing trees from memory</h4>
<p>
    When working with many GeneTree objects, memory can quickly get out of hand. This is because our current object model uses a cyclic graph of Perl references.
    As a consequence, the usual garbage-collector is not able to release the memory used by a gene tree when you lose its reference (unlike most of the Ensembl objects). 
    This means that you will have to call <kbd>release_tree()</kbd> on each tree after using it.
</p>

<h4>Preloading data</h4>
<p>
Most of the objects do lazy-loading of related objects via queries to the database.  
This system is sub-optimal when there are a lot of objects to fetch or if the server is distant. 
Our <kbd>Bio::EnsEMBL::Compara::Utils::Preloader</kbd> module provides several methods to do a bulk-loading of objects in a minimum number of queries.
This will result in a higher memory usage, but faster processing of data. 
</p>

<br>

<h2 id="help">Further help</h2>

<p>
For additional information or help mail the <a href="http://lists.ensembl.org/mailman/listinfo/dev" rel="external">ensembl-dev</a> mailing list.
You will need to subscribe to this mailing list to use it.
More information on subscribing to any Ensembl mailing list is available from the 
<a href="http://www.ensembl.org/info/about/contact/">Ensembl Contacts</a> page.
</p>

</body>
</html>
