
<html>
<head>

<title>Schema Documentation</title>
<meta name="order" content="2" />

<script language="Javascript" type="text/javascript">
  var img_plus   = '<img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/>';
  var img_minus  = '<img src="/i/16/minus-button.png" class="sql_schema_icon" alt="hide"/>';

  // Function to show/hide the columns table
  function show_hide (param, a_text) {
  
    // Schema tables
    if (a_text === 'columns') {
      div_id   = '#div_'+param;
      alink_id = '#a_'+param;
    }  
    // Species list
    else if (a_text === 'species') {
      div_id   = '#sp_'+param;
      alink_id = '#s_'+param;
    }
    // Example tables
    else {
      div_id   = '#ex_'+param;
      alink_id = '#e_'+param;
    }
    
    if ($(div_id).is(':visible')) {
      $(alink_id).html(img_plus+' Show '+a_text);
    }
    else if ($(div_id).is(':hidden')) {
      $(alink_id).html(img_minus+' Hide '+a_text);
    }
    $(div_id).slideToggle( 300 );
  }
  
  // Function to show/hide all the tables
  function show_hide_all (link_text) {
    expand_div = '#expand';
    div_prefix = 'div_';
    $("div[id^='"+div_prefix+"']").each(function() {
      param = $(this).attr('id').substring(div_prefix.length);
      div_id   = '#'+$(this).attr('id');
      alink_id = '#a_'+param;
      
      if ($(alink_id)) {
        if ($(expand_div).val()==0) {
          $(alink_id).html(img_minus+' Hide '+link_text);
        }
        else {
          $(alink_id).html(img_plus+' Show '+link_text);
        }
      }
      $(div_id).slideToggle( 500 );
    });
    if ($(expand_div).val()==0) {
      $(expand_div).val(1);
    }
    else {
      $(expand_div).val(0);
    }
  }
</script>
</head>
<body>

<h1>Ensembl Compara Schema Documentation</h1>
<h2>Introduction</h2>
<p>
This document describes the tables that make up the Ensembl Compara schema. Tables are listed grouped in different categories, and the purpose of each table is explained. Several examples are also given. They are intended to allow people to familiarise themselves with the schema. The overall diagram can be found <a target="_blank" href="./diagrams/Compara.png">here</a>.
</p>



<h3 id="top">List of the tables:</h3>
<div>

  <div class="sql_schema_table_group">
    <div class="sql_schema_table_group_header" style="border-color:#3CB371">
      <div class="sql_schema_table_group_bullet" style="background-color:#3CB371"></div>
      <h2><a class="sql_schema_link" href="#header_1">Dataset description</a></h2>
    </div>
      <ul class="sql_schema_table_list">
        <li><a href="#meta" class="sql_schema_link" style="font-weight:bold">meta</a></li>
        <li><a href="#species_set_header" class="sql_schema_link" style="font-weight:bold">species_set_header</a></li>
        <li><a href="#species_set" class="sql_schema_link" style="font-weight:bold">species_set</a></li>
        <li><a href="#species_set_tag" class="sql_schema_link" style="font-weight:bold">species_set_tag</a></li>
        <li><a href="#method_link" class="sql_schema_link" style="font-weight:bold">method_link</a></li>
        <li><a href="#method_link_species_set" class="sql_schema_link" style="font-weight:bold">method_link_species_set</a></li>
        <li><a href="#method_link_species_set_tag" class="sql_schema_link" style="font-weight:bold">method_link_species_set_tag</a></li>
        <li><a href="#method_link_species_set_attr" class="sql_schema_link" style="font-weight:bold">method_link_species_set_attr</a></li>

      </ul>
    </div>

  <div class="sql_schema_table_group">
    <div class="sql_schema_table_group_header" style="border-color:#24DA06">
      <div class="sql_schema_table_group_bullet" style="background-color:#24DA06"></div>
      <h2><a class="sql_schema_link" href="#header_2">Taxonomy and species-tree</a></h2>
    </div>
      <ul class="sql_schema_table_list">
        <li><a href="#ncbi_taxa_node" class="sql_schema_link" style="font-weight:bold">ncbi_taxa_node</a></li>
        <li><a href="#ncbi_taxa_name" class="sql_schema_link" style="font-weight:bold">ncbi_taxa_name</a></li>
        <li><a href="#species_tree_node" class="sql_schema_link" style="font-weight:bold">species_tree_node</a></li>
        <li><a href="#species_tree_root" class="sql_schema_link" style="font-weight:bold">species_tree_root</a></li>
        <li><a href="#species_tree_node_tag" class="sql_schema_link" style="font-weight:bold">species_tree_node_tag</a></li>
        <li><a href="#species_tree_node_attr" class="sql_schema_link" style="font-weight:bold">species_tree_node_attr</a></li>

      </ul>
    </div>

  <div class="sql_schema_table_group">
    <div class="sql_schema_table_group_header" style="border-color:#808000">
      <div class="sql_schema_table_group_bullet" style="background-color:#808000"></div>
      <h2><a class="sql_schema_link" href="#header_3">Genomes</a></h2>
    </div>
      <ul class="sql_schema_table_list">
        <li><a href="#genome_db" class="sql_schema_link" style="font-weight:bold">genome_db</a></li>
        <li><a href="#dnafrag" class="sql_schema_link" style="font-weight:bold">dnafrag</a></li>
        <li><a href="#dnafrag_alt_region" class="sql_schema_link" style="font-weight:bold">dnafrag_alt_region</a></li>
        <li><a href="#sequence" class="sql_schema_link" style="font-weight:bold">sequence</a></li>
        <li><a href="#gene_member" class="sql_schema_link" style="font-weight:bold">gene_member</a></li>
        <li><a href="#seq_member" class="sql_schema_link" style="font-weight:bold">seq_member</a></li>
        <li><a href="#exon_boundaries" class="sql_schema_link" style="font-weight:bold">exon_boundaries</a></li>
        <li><a href="#other_member_sequence" class="sql_schema_link" style="font-weight:bold">other_member_sequence</a></li>

      </ul>
    </div>

  <div class="sql_schema_table_group">
    <div class="sql_schema_table_group_header" style="border-color:#FF6666">
      <div class="sql_schema_table_group_bullet" style="background-color:#FF6666"></div>
      <h2><a class="sql_schema_link" href="#header_4">Synteny</a></h2>
    </div>
      <ul class="sql_schema_table_list">
        <li><a href="#synteny_region" class="sql_schema_link" style="font-weight:bold">synteny_region</a></li>
        <li><a href="#dnafrag_region" class="sql_schema_link" style="font-weight:bold">dnafrag_region</a></li>

      </ul>
    </div>
  <div style="clear:both"></div>
</div>

<div>
  <div class="sql_schema_table_group">
    <div class="sql_schema_table_group_header" style="border-color:#FF8500">
      <div class="sql_schema_table_group_bullet" style="background-color:#FF8500"></div>
      <h2><a class="sql_schema_link" href="#header_5">Genomic alignments</a></h2>
    </div>
      <ul class="sql_schema_table_list">
        <li><a href="#genomic_align_block" class="sql_schema_link" style="font-weight:bold">genomic_align_block</a></li>
        <li><a href="#genomic_align_tree" class="sql_schema_link" style="font-weight:bold">genomic_align_tree</a></li>
        <li><a href="#genomic_align" class="sql_schema_link" style="font-weight:bold">genomic_align</a></li>

      </ul>
    </div>

  <div class="sql_schema_table_group">
    <div class="sql_schema_table_group_header" style="border-color:#C70C09">
      <div class="sql_schema_table_group_bullet" style="background-color:#C70C09"></div>
      <h2><a class="sql_schema_link" href="#header_6">Conservation</a></h2>
    </div>
      <ul class="sql_schema_table_list">
        <li><a href="#conservation_score" class="sql_schema_link" style="font-weight:bold">conservation_score</a></li>
        <li><a href="#constrained_element" class="sql_schema_link" style="font-weight:bold">constrained_element</a></li>

      </ul>
    </div>

  <div class="sql_schema_table_group">
    <div class="sql_schema_table_group_header" style="border-color:#1E90FF">
      <div class="sql_schema_table_group_bullet" style="background-color:#1E90FF"></div>
      <h2><a class="sql_schema_link" href="#header_7">Gene trees and homologies</a></h2>
    </div>
      <ul class="sql_schema_table_list">
        <li><a href="#peptide_align_feature" class="sql_schema_link" style="font-weight:bold">peptide_align_feature</a></li>
        <li><a href="#gene_align" class="sql_schema_link" style="font-weight:bold">gene_align</a></li>
        <li><a href="#gene_align_member" class="sql_schema_link" style="font-weight:bold">gene_align_member</a></li>
        <li><a href="#gene_tree_node" class="sql_schema_link" style="font-weight:bold">gene_tree_node</a></li>
        <li><a href="#gene_tree_root" class="sql_schema_link" style="font-weight:bold">gene_tree_root</a></li>
        <li><a href="#gene_tree_node_tag" class="sql_schema_link" style="font-weight:bold">gene_tree_node_tag</a></li>
        <li><a href="#gene_tree_root_tag" class="sql_schema_link" style="font-weight:bold">gene_tree_root_tag</a></li>
        <li><a href="#gene_tree_root_attr" class="sql_schema_link" style="font-weight:bold">gene_tree_root_attr</a></li>
        <li><a href="#gene_tree_node_attr" class="sql_schema_link" style="font-weight:bold">gene_tree_node_attr</a></li>
        <li><a href="#gene_tree_object_store" class="sql_schema_link" style="font-weight:bold">gene_tree_object_store</a></li>
        <li><a href="#homology" class="sql_schema_link" style="font-weight:bold">homology</a></li>
        <li><a href="#homology_member" class="sql_schema_link" style="font-weight:bold">homology_member</a></li>

      </ul>
    </div>

  <div class="sql_schema_table_group">
    <div class="sql_schema_table_group_header" style="border-color:#FFCC66">
      <div class="sql_schema_table_group_bullet" style="background-color:#FFCC66"></div>
      <h2><a class="sql_schema_link" href="#header_8">Extra annotations on members</a></h2>
    </div>
      <ul class="sql_schema_table_list">
        <li><a href="#gene_member_hom_stats" class="sql_schema_link" style="font-weight:bold">gene_member_hom_stats</a></li>
        <li><a href="#seq_member_projection_stable_id" class="sql_schema_link" style="font-weight:bold">seq_member_projection_stable_id</a></li>
        <li><a href="#seq_member_projection" class="sql_schema_link" style="font-weight:bold">seq_member_projection</a></li>
        <li><a href="#external_db" class="sql_schema_link" style="font-weight:bold">external_db</a></li>
        <li><a href="#member_xref" class="sql_schema_link" style="font-weight:bold">member_xref</a></li>
        <li><a href="#gene_member_qc" class="sql_schema_link" style="font-weight:bold">gene_member_qc</a></li>

      </ul>
    </div>
  <div style="clear:both"></div>
</div>

<div>
  <div class="sql_schema_table_group">
    <div class="sql_schema_table_group_header" style="border-color:#BC5CEC">
      <div class="sql_schema_table_group_bullet" style="background-color:#BC5CEC"></div>
      <h2><a class="sql_schema_link" href="#header_9">Protein families</a></h2>
    </div>
      <ul class="sql_schema_table_list">
        <li><a href="#family" class="sql_schema_link" style="font-weight:bold">family</a></li>
        <li><a href="#family_member" class="sql_schema_link" style="font-weight:bold">family_member</a></li>

      </ul>
    </div>

  <div class="sql_schema_table_group">
    <div class="sql_schema_table_group_header" style="border-color:#66CCFF">
      <div class="sql_schema_table_group_bullet" style="background-color:#66CCFF"></div>
      <h2><a class="sql_schema_link" href="#header_10">Profile HMMs</a></h2>
    </div>
      <ul class="sql_schema_table_list">
        <li><a href="#hmm_profile" class="sql_schema_link" style="font-weight:bold">hmm_profile</a></li>
        <li><a href="#hmm_annot" class="sql_schema_link" style="font-weight:bold">hmm_annot</a></li>
        <li><a href="#hmm_curated_annot" class="sql_schema_link" style="font-weight:bold">hmm_curated_annot</a></li>

      </ul>
    </div>

  <div class="sql_schema_table_group">
    <div class="sql_schema_table_group_header" style="border-color:#AAAAAA">
      <div class="sql_schema_table_group_bullet" style="background-color:#AAAAAA"></div>
      <h2><a class="sql_schema_link" href="#header_11">Stable-ID mapping</a></h2>
    </div>
      <ul class="sql_schema_table_list">
        <li><a href="#mapping_session" class="sql_schema_link" style="font-weight:bold">mapping_session</a></li>
        <li><a href="#stable_id_history" class="sql_schema_link" style="font-weight:bold">stable_id_history</a></li>

      </ul>
    </div>

  <div class="sql_schema_table_group">
    <div class="sql_schema_table_group_header" style="border-color:#01D4F7">
      <div class="sql_schema_table_group_bullet" style="background-color:#01D4F7"></div>
      <h2><a class="sql_schema_link" href="#header_12">Gene gain/loss trees</a></h2>
    </div>
      <ul class="sql_schema_table_list">
        <li><a href="#CAFE_gene_family" class="sql_schema_link" style="font-weight:bold">CAFE_gene_family</a></li>
        <li><a href="#CAFE_species_gene" class="sql_schema_link" style="font-weight:bold">CAFE_species_gene</a></li>

      </ul>
    </div>

  <div style="clear:both" />
</div>
  <input type="button" onclick="show_hide_all('columns')" class="fbutton" value="Show/hide all"/>
  <input type="hidden" id="expand" value="0" />
  

<div id="header_1" class="sql_schema_group_header" style="border-color:#3CB371">
  <div class="sql_schema_group_bullet" style="background-color:#3CB371"></div>
  <h2>Dataset description</h2>
</div>
<p class="sql_schema_group_header_desc">These are general tables used in the Compara schema</p><div align="center"><a href="diagrams/Compara.dataset_description.png"><img src="diagrams/Compara.dataset_description.png" width="198px" /></a></div><div class="sql_schema_table">
  <div id="meta" class="sql_schema_table_header" style="border-top-color:#3CB371">
    <div class="sql_schema_table_header_left"><span style="background-color:#3CB371"></span>meta</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_meta" class="help-header sql_schema_show_hide" onclick="show_hide('meta','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">This table stores meta information about the compara database</p>

  <div id="div_meta" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>meta_id</b></td>
        <td class="bg1">INT</td>
        <td class="bg1">-</td>
        <td class="bg1">Internal unique ID for the table</td>
        <td class="bg1">primary   key</td>
      </tr>

      <tr>
        <td class="bg2"><b>species_id</b></td>
        <td class="bg2">INT</td>
        <td class="bg2">1</td>
        <td class="bg2">Only used in core databases</td>
        <td class="bg2">unique    key: <i>species_key_value_idx</i><br />key: <i>species_value_idx</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>meta_key</b></td>
        <td class="bg1">VARCHAR(40)</td>
        <td class="bg1">-</td>
        <td class="bg1">Key for the key/value pair</td>
        <td class="bg1">unique    key: <i>species_key_value_idx</i></td>
      </tr>

      <tr>
        <td class="bg2"><b>meta_value</b></td>
        <td class="bg2">TEXT</td>
        <td class="bg2">-</td>
        <td class="bg2">Value for the key/value pair</td>
        <td class="bg2">unique    key: <i>species_key_value_idx</i><br />key: <i>species_value_idx</i></td>
      </tr>
    </table>
  </div>
<div class="sql_schema_table_examples"><p class="sql_schema_table_example_header">Example:</p><div class="sql_schema_table_example_content"><p>This query defines which API version must be used to access this database. </p>
      <div>
        <div class="sql_schema_table_example_query">
          <pre><span class="sql_schema_sql_highlight">SELECT</span> * <span class="sql_schema_sql_highlight">FROM</span> meta <span class="sql_schema_sql_highlight">WHERE</span> meta_key = &quot;schema_version&quot;;</pre>
        </div>
        <div class="sql_schema_table_example_button">
  <a id="e_meta" class="help-header sql_schema_show_hide" onclick="show_hide('meta','query results')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show query results
  </a></div>
        <div style="clear:both"></div>
      </div>
      
  <div id="ex_meta" style="display:none;">
    <table class="ss sql_schema_table_example_result">
      <tr><th>meta_id</th><th>species_id</th><th>meta_key</th><th>meta_value</th></tr>      <tr><td>5</td><td>NULL</td><td>schema_version</td><td>108</td></tr>    </table>
  </div></div></div></div></div><div class="sql_schema_table">
  <div id="species_set_header" class="sql_schema_table_header" style="border-top-color:#3CB371">
    <div class="sql_schema_table_header_left"><span style="background-color:#3CB371"></span>species_set_header</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_species_set_header" class="help-header sql_schema_show_hide" onclick="show_hide('species_set_header','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">Header for the <a href="#species_set" class="sql_schema_link">species_set</a> table which groups or sets of species which are used in the <a href="#method_link_species_set" class="sql_schema_link">method_link_species_set</a> table.</p>

  <div id="div_species_set_header" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>species_set_id</b></td>
        <td class="bg1">INT</td>
        <td class="bg1">-</td>
        <td class="bg1">Internal (non-unique) ID for the table</td>
        <td class="bg1">primary key</td>
      </tr>

      <tr>
        <td class="bg2"><b>name</b></td>
        <td class="bg2">varchar(255)</td>
        <td class="bg2">''</td>
        <td class="bg2">Name of the species set (e.g. &quot;H.sap&quot; for one species, &quot;H.sap-M.mus&quot; for two species and &quot;amniotes&quot; for larger sets)</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>size</b></td>
        <td class="bg1">INT</td>
        <td class="bg1">-</td>
        <td class="bg1">Number of species in the set</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>first_release</b></td>
        <td class="bg2">smallint</td>
        <td class="bg2">NULL</td>
        <td class="bg2">The first release this set genome was present in</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>last_release</b></td>
        <td class="bg1">smallint</td>
        <td class="bg1">NULL</td>
        <td class="bg1">The last release this set was present in, or NULL if it is still current</td>
        <td class="bg1"></td>
      </tr>
    </table>
  </div>
<div class="sql_schema_table_examples"><p class="sql_schema_table_example_header">Example:</p><div class="sql_schema_table_example_content"><p>This query shows the first 10 species_sets having human </p>
      <div>
        <div class="sql_schema_table_example_query">
          <pre><span class="sql_schema_sql_highlight">SELECT</span> ssh.species_set_id, ssh.name, ssh.size, gdb.genome_db_id, gdb.name, gdb.assembly <span class="sql_schema_sql_highlight">FROM</span> species_set_header ssh <span class="sql_schema_sql_highlight">JOIN</span> species_set ss USING(species_set_id) <span class="sql_schema_sql_highlight">JOIN</span> genome_db gdb USING(genome_db_id) <span class="sql_schema_sql_highlight">WHERE</span> ssh.name=&quot;primates&quot;;  </pre>
        </div>
        <div class="sql_schema_table_example_button">
  <a id="e_species_set_header" class="help-header sql_schema_show_hide" onclick="show_hide('species_set_header','query results')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show query results
  </a></div>
        <div style="clear:both"></div>
      </div>
      
  <div id="ex_species_set_header" style="display:none;">
    <table class="ss sql_schema_table_example_result">
      <tr><th>ssh.species_set_id</th><th>ssh.name</th><th>ssh.size</th><th>gdb.genome_db_id</th><th>gdb.name</th><th>gdb.assembly</th></tr>      <tr><td>83234</td><td>primates</td><td>24</td><td>124</td><td>otolemur_garnettii</td><td>OtoGar3</td></tr>      <tr class="bg2"><td>83234</td><td>primates</td><td>24</td><td>150</td><td>homo_sapiens</td><td>GRCh38</td></tr>      <tr><td>83234</td><td>primates</td><td>24</td><td>153</td><td>chlorocebus_sabaeus</td><td>ChlSab1.1</td></tr>      <tr class="bg2"><td>83234</td><td>primates</td><td>24</td><td>197</td><td>cercocebus_atys</td><td>Caty_1.0</td></tr>      <tr><td>83234</td><td>primates</td><td>24</td><td>199</td><td>nomascus_leucogenys</td><td>Nleu_3.0</td></tr>      <tr class="bg2"><td>83234</td><td>primates</td><td>24</td><td>200</td><td>macaca_nemestrina</td><td>Mnem_1.0</td></tr>      <tr><td>83234</td><td>primates</td><td>24</td><td>201</td><td>mandrillus_leucophaeus</td><td>Mleu.le_1.0</td></tr>      <tr class="bg2"><td>83234</td><td>primates</td><td>24</td><td>203</td><td>propithecus_coquereli</td><td>Pcoq_1.0</td></tr>      <tr><td>83234</td><td>primates</td><td>24</td><td>204</td><td>rhinopithecus_bieti</td><td>ASM169854v1</td></tr>      <tr class="bg2"><td>83234</td><td>primates</td><td>24</td><td>205</td><td>saimiri_boliviensis_boliviensis</td><td>SaiBol1.0</td></tr>      <tr><td>83234</td><td>primates</td><td>24</td><td>206</td><td>microcebus_murinus</td><td>Mmur_3.0</td></tr>      <tr class="bg2"><td>83234</td><td>primates</td><td>24</td><td>207</td><td>cebus_capucinus</td><td>Cebus_imitator-1.0</td></tr>      <tr><td>83234</td><td>primates</td><td>24</td><td>209</td><td>gorilla_gorilla</td><td>gorGor4</td></tr>      <tr class="bg2"><td>83234</td><td>primates</td><td>24</td><td>210</td><td>pan_paniscus</td><td>panpan1.1</td></tr>      <tr><td>83234</td><td>primates</td><td>24</td><td>216</td><td>rhinopithecus_roxellana</td><td>Rrox_v1</td></tr>      <tr class="bg2"><td>83234</td><td>primates</td><td>24</td><td>217</td><td>carlito_syrichta</td><td>Tarsius_syrichta-2.0.1</td></tr>      <tr><td>83234</td><td>primates</td><td>24</td><td>219</td><td>aotus_nancymaae</td><td>Anan_2.0</td></tr>      <tr class="bg2"><td>83234</td><td>primates</td><td>24</td><td>221</td><td>pan_troglodytes</td><td>Pan_tro_3.0</td></tr>      <tr><td>83234</td><td>primates</td><td>24</td><td>302</td><td>prolemur_simus</td><td>Prosim_1.0</td></tr>      <tr class="bg2"><td>83234</td><td>primates</td><td>24</td><td>361</td><td>macaca_mulatta</td><td>Mmul_10</td></tr>      <tr><td>83234</td><td>primates</td><td>24</td><td>465</td><td>macaca_fascicularis</td><td>Macaca_fascicularis_6.0</td></tr>      <tr class="bg2"><td>83234</td><td>primates</td><td>24</td><td>472</td><td>callithrix_jacchus</td><td>mCalJac1.pat.X</td></tr>      <tr><td>83234</td><td>primates</td><td>24</td><td>474</td><td>papio_anubis</td><td>Panubis1.0</td></tr>      <tr class="bg2"><td>83234</td><td>primates</td><td>24</td><td>476</td><td>pongo_abelii</td><td>Susie_PABv2</td></tr>      <tr><td>84239</td><td>primates</td><td>10</td><td>150</td><td>homo_sapiens</td><td>GRCh38</td></tr>      <tr class="bg2"><td>84239</td><td>primates</td><td>10</td><td>153</td><td>chlorocebus_sabaeus</td><td>ChlSab1.1</td></tr>      <tr><td>84239</td><td>primates</td><td>10</td><td>199</td><td>nomascus_leucogenys</td><td>Nleu_3.0</td></tr>      <tr class="bg2"><td>84239</td><td>primates</td><td>10</td><td>206</td><td>microcebus_murinus</td><td>Mmur_3.0</td></tr>      <tr><td>84239</td><td>primates</td><td>10</td><td>209</td><td>gorilla_gorilla</td><td>gorGor4</td></tr>      <tr class="bg2"><td>84239</td><td>primates</td><td>10</td><td>210</td><td>pan_paniscus</td><td>panpan1.1</td></tr>      <tr><td>84239</td><td>primates</td><td>10</td><td>221</td><td>pan_troglodytes</td><td>Pan_tro_3.0</td></tr>      <tr class="bg2"><td>84239</td><td>primates</td><td>10</td><td>361</td><td>macaca_mulatta</td><td>Mmul_10</td></tr>      <tr><td>84239</td><td>primates</td><td>10</td><td>465</td><td>macaca_fascicularis</td><td>Macaca_fascicularis_6.0</td></tr>      <tr class="bg2"><td>84239</td><td>primates</td><td>10</td><td>476</td><td>pongo_abelii</td><td>Susie_PABv2</td></tr>    </table>
  </div></div></div><table class="sql_schema_extra"><tr>    <td class="sql_schema_extra_left">    <p style="font-weight:bold">See also:</p>
  <ul>
      <li><a href="#method_link_species_set">method_link_species_set</a></li>
      <li><a href="#genome_db">genome_db</a></li>
      <li><a href="#species_set">species_set</a></li>
      </ul>
    </td>
</tr></table></div></div><div class="sql_schema_table">
  <div id="species_set" class="sql_schema_table_header" style="border-top-color:#3CB371">
    <div class="sql_schema_table_header_left"><span style="background-color:#3CB371"></span>species_set</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_species_set" class="help-header sql_schema_show_hide" onclick="show_hide('species_set','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">Describes the content of each species-set (<a href="#species_set_header" class="sql_schema_link">species_set_header</a>) as a set of <a href="#genome_db" class="sql_schema_link">genome_db</a> objects</p>

  <div id="div_species_set" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>species_set_id</b></td>
        <td class="bg1">INT</td>
        <td class="bg1">-</td>
        <td class="bg1">Internal ID for the table, foreign key to <a href="#species_set_header" class="sql_schema_link">species_set_header</a></td>
        <td class="bg1">primary key</td>
      </tr>

      <tr>
        <td class="bg2"><b>genome_db_id</b></td>
        <td class="bg2">INT</td>
        <td class="bg2">-</td>
        <td class="bg2">External reference to genome_db_id in the <a href="#genome_db" class="sql_schema_link">genome_db</a> table</td>
        <td class="bg2">primary key</td>
      </tr>
    </table>
  </div>
<div class="sql_schema_table_examples"><p class="sql_schema_table_example_header">Example:</p><div class="sql_schema_table_example_content"><p>This query shows the first 10 species_sets having human </p>
      <div>
        <div class="sql_schema_table_example_query">
          <pre><span class="sql_schema_sql_highlight">SELECT</span> species_set_id, GROUP_CONCAT(name) <span class="sql_schema_sql_highlight">AS</span> species <span class="sql_schema_sql_highlight">FROM</span> species_set <span class="sql_schema_sql_highlight">JOIN</span> genome_db USING(genome_db_id) <span class="sql_schema_sql_highlight">GROUP</span> <span class="sql_schema_sql_highlight">BY</span> species_set_id HAVING species LIKE &#39;%homo_sapiens%&#39; <span class="sql_schema_sql_highlight">ORDER</span> <span class="sql_schema_sql_highlight">BY</span> species_set_id <span class="sql_schema_sql_highlight">LIMIT</span> 10;</pre>
        </div>
        <div class="sql_schema_table_example_button">
  <a id="e_species_set" class="help-header sql_schema_show_hide" onclick="show_hide('species_set','query results')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show query results
  </a></div>
        <div style="clear:both"></div>
      </div>
      
  <div id="ex_species_set" style="display:none;">
    <table class="ss sql_schema_table_example_result">
      <tr><th>species_set_id</th><th>species</th></tr>      <tr><td>35674</td><td>choloepus_hoffmanni,homo_sapiens</td></tr>      <tr class="bg2"><td>35676</td><td>oryctolagus_cuniculus,homo_sapiens</td></tr>      <tr><td>35678</td><td>otolemur_garnettii,homo_sapiens</td></tr>      <tr class="bg2"><td>35680</td><td>dasypus_novemcinctus,homo_sapiens</td></tr>      <tr><td>35682</td><td>erinaceus_europaeus,homo_sapiens</td></tr>      <tr class="bg2"><td>35684</td><td>echinops_telfairi,homo_sapiens</td></tr>      <tr><td>35686</td><td>loxodonta_africana,homo_sapiens</td></tr>      <tr class="bg2"><td>35687</td><td>notamacropus_eugenii,homo_sapiens</td></tr>      <tr><td>35689</td><td>ochotona_princeps,homo_sapiens</td></tr>      <tr class="bg2"><td>35690</td><td>procavia_capensis,homo_sapiens</td></tr>    </table>
  </div></div></div><table class="sql_schema_extra"><tr>    <td class="sql_schema_extra_left">    <p style="font-weight:bold">See also:</p>
  <ul>
      <li><a href="#method_link_species_set">method_link_species_set</a></li>
      <li><a href="#genome_db">genome_db</a></li>
      <li><a href="#species_set_header">species_set_header</a></li>
      </ul>
    </td>
</tr></table></div></div><div class="sql_schema_table">
  <div id="species_set_tag" class="sql_schema_table_header" style="border-top-color:#3CB371">
    <div class="sql_schema_table_header_left"><span style="background-color:#3CB371"></span>species_set_tag</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_species_set_tag" class="help-header sql_schema_show_hide" onclick="show_hide('species_set_tag','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">This table contains descriptive tags for the species_set_ids in the species_set table. It is used to store options on clades and group of species. It has been initially developed for the gene tree view.</p>

  <div id="div_species_set_tag" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>species_set_id</b></td>
        <td class="bg1">INT</td>
        <td class="bg1">-</td>
        <td class="bg1">External reference to species_set_id in the <a href="#species_set" class="sql_schema_link">species_set</a> table</td>
        <td class="bg1">primary key</td>
      </tr>

      <tr>
        <td class="bg2"><b>tag</b></td>
        <td class="bg2">varchar(50)</td>
        <td class="bg2">-</td>
        <td class="bg2">Tag name</td>
        <td class="bg2">primary key<br />key: <i>tag</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>value</b></td>
        <td class="bg1">mediumtext</td>
        <td class="bg1">-</td>
        <td class="bg1">Tag value</td>
        <td class="bg1"></td>
      </tr>
    </table>
  </div>
<table class="sql_schema_extra"><tr>    <td class="sql_schema_extra_left">    <p style="font-weight:bold">See also:</p>
  <ul>
      <li><a href="#species_set">species_set</a></li>
      </ul>
    </td>
</tr></table></div></div><div class="sql_schema_table">
  <div id="method_link" class="sql_schema_table_header" style="border-top-color:#3CB371">
    <div class="sql_schema_table_header_left"><span style="background-color:#3CB371"></span>method_link</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_method_link" class="help-header sql_schema_show_hide" onclick="show_hide('method_link','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">This table specifies which kind of link can exist between entities in compara (dna/dna alignment, synteny regions, homologous gene pairs, etc...)<br/>NOTE: We use method_link_ids between 1 and 100 for DNA-DNA alignments, between 101 and 200 for genomic syntenies, between 201 and 300 for protein homologies, between 301 and 400 for protein families and between 401 and 500 for protein and ncRNA trees. Each category corresponds to data stored in different tables.</p>

  <div id="div_method_link" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>method_link_id</b></td>
        <td class="bg1">INT</td>
        <td class="bg1">-</td>
        <td class="bg1">Internal unique ID</td>
        <td class="bg1">primary key</td>
      </tr>

      <tr>
        <td class="bg2"><b>type</b></td>
        <td class="bg2">varchar(50)</td>
        <td class="bg2">''</td>
        <td class="bg2">The code used to refer to this linking method</td>
        <td class="bg2">unique key: <i>type</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>class</b></td>
        <td class="bg1">varchar(50)</td>
        <td class="bg1">''</td>
        <td class="bg1">Description of type of data associated with the \&quot;type\&quot; field. Used to match similar types</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>display_name</b></td>
        <td class="bg2">varchar(255)</td>
        <td class="bg2">''</td>
        <td class="bg2">Plain English description of this method</td>
        <td class="bg2"></td>
      </tr>
    </table>
  </div>
<div class="sql_schema_table_examples"><p class="sql_schema_table_example_header">Example:</p><div class="sql_schema_table_example_content"><p>These are our current entries: </p>
      <div>
        <div class="sql_schema_table_example_query">
          <pre><span class="sql_schema_sql_highlight">SELECT</span> * <span class="sql_schema_sql_highlight">FROM</span> method_link;</pre>
        </div>
        <div class="sql_schema_table_example_button">
  <a id="e_method_link" class="help-header sql_schema_show_hide" onclick="show_hide('method_link','query results')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show query results
  </a></div>
        <div style="clear:both"></div>
      </div>
      
  <div id="ex_method_link" style="display:none;">
    <table class="ss sql_schema_table_example_result">
      <tr><th>method_link_id</th><th>type</th><th>class</th><th>display_name</th></tr>      <tr><td>10</td><td>PECAN</td><td>GenomicAlignBlock.multiple_alignment</td><td>Mercator-Pecan</td></tr>      <tr class="bg2"><td>11</td><td>GERP_CONSTRAINED_ELEMENT</td><td>ConstrainedElement.constrained_element</td><td>GERP Constrained Elements</td></tr>      <tr><td>13</td><td>EPO</td><td>GenomicAlignTree.ancestral_alignment</td><td>EPO</td></tr>      <tr class="bg2"><td>14</td><td>EPO_EXTENDED</td><td>GenomicAlignTree.tree_alignment</td><td>EPO-Extended</td></tr>      <tr><td>16</td><td>LASTZ_NET</td><td>GenomicAlignBlock.pairwise_alignment</td><td>LastZ</td></tr>      <tr class="bg2"><td>19</td><td>LASTZ_PATCH</td><td>GenomicAlignBlock.pairwise_alignment</td><td>LastZ-patch</td></tr>      <tr><td>101</td><td>SYNTENY</td><td>SyntenyRegion.synteny</td><td>synteny</td></tr>      <tr class="bg2"><td>201</td><td>ENSEMBL_ORTHOLOGUES</td><td>Homology.homology</td><td>orthologues</td></tr>      <tr><td>202</td><td>ENSEMBL_PARALOGUES</td><td>Homology.homology</td><td>paralogues</td></tr>      <tr class="bg2"><td>205</td><td>ENSEMBL_PROJECTIONS</td><td>Homology.homology</td><td>patch projections</td></tr>      <tr><td>401</td><td>PROTEIN_TREES</td><td>ProteinTree.protein_tree_node</td><td>protein-trees</td></tr>      <tr class="bg2"><td>402</td><td>NC_TREES</td><td>NCTree.nc_tree_node</td><td>ncRNA-trees</td></tr>      <tr><td>501</td><td>GERP_CONSERVATION_SCORE</td><td>ConservationScore.conservation_score</td><td>GERP Conservation Scores</td></tr>      <tr class="bg2"><td>600</td><td>SPECIES_TREE</td><td>SpeciesTree.species_tree_root</td><td>species-tree</td></tr>    </table>
  </div></div></div><table class="sql_schema_extra"><tr>    <td class="sql_schema_extra_left">    <p style="font-weight:bold">See also:</p>
  <ul>
      <li><a href="#method_link_species_set">method_link_species_set</a></li>
      <li><a href="#species_set">species_set</a></li>
      </ul>
    </td>
</tr></table></div></div><div class="sql_schema_table">
  <div id="method_link_species_set" class="sql_schema_table_header" style="border-top-color:#3CB371">
    <div class="sql_schema_table_header_left"><span style="background-color:#3CB371"></span>method_link_species_set</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_method_link_species_set" class="help-header sql_schema_show_hide" onclick="show_hide('method_link_species_set','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">This table contains information about the comparisons stored in the database. A given method_link_species_set_id exist for each comparison made and relates a method_link_id in <a href="#method_link" class="sql_schema_link">method_link</a> with a set of species (species_set_id) in the <a href="#species_set" class="sql_schema_link">species_set</a> table.</p>

  <div id="div_method_link_species_set" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>method_link_species_set_id</b></td>
        <td class="bg1">INT</td>
        <td class="bg1">-</td>
        <td class="bg1">Internal unique ID</td>
        <td class="bg1">primary key</td>
      </tr>

      <tr>
        <td class="bg2"><b>method_link_id</b></td>
        <td class="bg2">INT</td>
        <td class="bg2">-</td>
        <td class="bg2">External reference to method_link_id in the <a href="#method_link" class="sql_schema_link">method_link</a> table</td>
        <td class="bg2">unique key: <i>method_link_id</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>species_set_id</b></td>
        <td class="bg1">INT</td>
        <td class="bg1">-</td>
        <td class="bg1">External reference to species_set_id in the <a href="#species_set" class="sql_schema_link">species_set</a> table</td>
        <td class="bg1">unique key: <i>method_link_id</i></td>
      </tr>

      <tr>
        <td class="bg2"><b>name</b></td>
        <td class="bg2">varchar(255)</td>
        <td class="bg2">''</td>
        <td class="bg2">Human-readable description for this method_link_species_set</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>source</b></td>
        <td class="bg1">varchar(255)</td>
        <td class="bg1">'ensembl'</td>
        <td class="bg1">Source of the data. Currently either &quot;ensembl&quot; or &quot;ucsc&quot; if data were imported from UCSC</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>url</b></td>
        <td class="bg2">varchar(255)</td>
        <td class="bg2">''</td>
        <td class="bg2">A URL where you can find the orignal data if they were imported</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>first_release</b></td>
        <td class="bg1">smallint</td>
        <td class="bg1">NULL</td>
        <td class="bg1">The first release this analysis was present in</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>last_release</b></td>
        <td class="bg2">smallint</td>
        <td class="bg2">NULL</td>
        <td class="bg2">The last release this analysis was present in, or NULL if it is still current</td>
        <td class="bg2"></td>
      </tr>
    </table>
  </div>
<div class="sql_schema_table_examples"><p class="sql_schema_table_example_header">Example:</p><div class="sql_schema_table_example_content"><p>This query shows all the EPO alignments in this database: </p>
      <div>
        <div class="sql_schema_table_example_query">
          <pre><span class="sql_schema_sql_highlight">SELECT</span> * <span class="sql_schema_sql_highlight">FROM</span> method_link_species_set <span class="sql_schema_sql_highlight">WHERE</span> method_link_id = 13;</pre>
        </div>
        <div class="sql_schema_table_example_button">
  <a id="e_method_link_species_set" class="help-header sql_schema_show_hide" onclick="show_hide('method_link_species_set','query results')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show query results
  </a></div>
        <div style="clear:both"></div>
      </div>
      
  <div id="ex_method_link_species_set" style="display:none;">
    <table class="ss sql_schema_table_example_result">
      <tr><th>method_link_species_set_id</th><th>method_link_id</th><th>species_set_id</th><th>name</th><th>source</th><th>url</th><th>first_release</th><th>last_release</th></tr>      <tr><td>2001</td><td>13</td><td>83217</td><td>21 murinae EPO</td><td>ensembl</td><td></td><td>105</td><td>NULL</td></tr>      <tr class="bg2"><td>1995</td><td>13</td><td>83235</td><td>43 eutherian mammals EPO</td><td>ensembl</td><td></td><td>105</td><td>NULL</td></tr>      <tr><td>2006</td><td>13</td><td>84239</td><td>10 primates EPO</td><td>ensembl</td><td></td><td>106</td><td>NULL</td></tr>      <tr class="bg2"><td>2007</td><td>13</td><td>84240</td><td>39 fish EPO</td><td>ensembl</td><td></td><td>106</td><td>NULL</td></tr>      <tr><td>2038</td><td>13</td><td>85062</td><td>17 sauropsids EPO</td><td>ensembl</td><td></td><td>107</td><td>NULL</td></tr>    </table>
  </div></div></div><table class="sql_schema_extra"><tr>    <td class="sql_schema_extra_left">    <p style="font-weight:bold">See also:</p>
  <ul>
      <li><a href="#method_link">method_link</a></li>
      <li><a href="#species_set">species_set</a></li>
      </ul>
    </td>
</tr></table></div></div><div class="sql_schema_table">
  <div id="method_link_species_set_tag" class="sql_schema_table_header" style="border-top-color:#3CB371">
    <div class="sql_schema_table_header_left"><span style="background-color:#3CB371"></span>method_link_species_set_tag</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_method_link_species_set_tag" class="help-header sql_schema_show_hide" onclick="show_hide('method_link_species_set_tag','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">Contains serveral tag/value data associated with method_link_species_set entries</p>

  <div id="div_method_link_species_set_tag" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>method_link_species_set_id</b></td>
        <td class="bg1">INT</td>
        <td class="bg1">-</td>
        <td class="bg1">External reference to method_link_species_set_id in the <a href="#method_link_species_set" class="sql_schema_link">method_link_species_set</a> table</td>
        <td class="bg1">primary key</td>
      </tr>

      <tr>
        <td class="bg2"><b>tag</b></td>
        <td class="bg2">varchar(50)</td>
        <td class="bg2">-</td>
        <td class="bg2">Tag name</td>
        <td class="bg2">primary key<br />key: <i>tag</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>value</b></td>
        <td class="bg1">mediumtext</td>
        <td class="bg1">-</td>
        <td class="bg1">Tag value</td>
        <td class="bg1"></td>
      </tr>
    </table>
  </div>
<table class="sql_schema_extra"><tr>    <td class="sql_schema_extra_left">    <p style="font-weight:bold">See also:</p>
  <ul>
      <li><a href="#method_link_species_set">method_link_species_set</a></li>
      </ul>
    </td>
</tr></table></div></div><div class="sql_schema_table">
  <div id="method_link_species_set_attr" class="sql_schema_table_header" style="border-top-color:#3CB371">
    <div class="sql_schema_table_header_left"><span style="background-color:#3CB371"></span>method_link_species_set_attr</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_method_link_species_set_attr" class="help-header sql_schema_show_hide" onclick="show_hide('method_link_species_set_attr','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">This table contains the distribution of the gene order conservation scores </p>

  <div id="div_method_link_species_set_attr" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>method_link_species_set_id</b></td>
        <td class="bg1">INT</td>
        <td class="bg1">-</td>
        <td class="bg1">internal unique ID for the orthologs</td>
        <td class="bg1">primary key</td>
      </tr>

      <tr>
        <td class="bg2"><b>n_goc_0</b></td>
        <td class="bg2">int</td>
        <td class="bg2">NULL</td>
        <td class="bg2">the number of orthologs with no gene order conservation among their neighbours</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>n_goc_25</b></td>
        <td class="bg1">int</td>
        <td class="bg1">NULL</td>
        <td class="bg1">the number of orthologs with 25% gene order conservation among their neighbours</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>n_goc_50</b></td>
        <td class="bg2">int</td>
        <td class="bg2">NULL</td>
        <td class="bg2">the number of orthologs with 50% gene order conservation among their neighbours</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>n_goc_75</b></td>
        <td class="bg1">int</td>
        <td class="bg1">NULL</td>
        <td class="bg1">the number of orthologs with 75% gene order conservation among their neighbours</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>n_goc_100</b></td>
        <td class="bg2">int</td>
        <td class="bg2">NULL</td>
        <td class="bg2">the number of orthologs with 100% gene order conservation among their neighbours</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>perc_orth_above_goc_thresh</b></td>
        <td class="bg1">float</td>
        <td class="bg1">NULL</td>
        <td class="bg1">the percentage of orthologs above the goc threshold </td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>goc_quality_threshold</b></td>
        <td class="bg2">int</td>
        <td class="bg2">NULL</td>
        <td class="bg2">the chosen threshold for &quot;high quality&quot; orthologs based on gene order conservation</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>wga_quality_threshold</b></td>
        <td class="bg1">int</td>
        <td class="bg1">NULL</td>
        <td class="bg1">the chosen threshold for &quot;high quality&quot; orthologs based on the whole genome alignments coverage of homologous pairs</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>perc_orth_above_wga_thresh</b></td>
        <td class="bg2">float</td>
        <td class="bg2">NULL</td>
        <td class="bg2">the percentage of orthologs above the wga threshold</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>threshold_on_ds</b></td>
        <td class="bg1">int</td>
        <td class="bg1">NULL</td>
        <td class="bg1">the threshold_on_ds</td>
        <td class="bg1"></td>
      </tr>
    </table>
  </div>
<table class="sql_schema_extra"><tr>    <td class="sql_schema_extra_left">    <p style="font-weight:bold">See also:</p>
  <ul>
      <li><a href="#method_link_species_set">method_link_species_set</a></li>
      <li><a href="#homology">homology</a></li>
      </ul>
    </td>
</tr></table></div></div>

<div id="header_2" class="sql_schema_group_header" style="border-color:#24DA06">
  <div class="sql_schema_group_bullet" style="background-color:#24DA06"></div>
  <h2>Taxonomy and species-tree</h2>
</div>
<p class="sql_schema_group_header_desc">Species-tree used in the Compara analyses (incl. new annotations generated in-house), and the NCBI taxonomy (which often used as a template for species-trees)</p><div align="center"><a href="diagrams/Compara.taxonomy_and_species_tree.png"><img src="diagrams/Compara.taxonomy_and_species_tree.png" width="283px" /></a></div><div class="sql_schema_table">
  <div id="ncbi_taxa_node" class="sql_schema_table_header" style="border-top-color:#24DA06">
    <div class="sql_schema_table_header_left"><span style="background-color:#24DA06"></span>ncbi_taxa_node</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_ncbi_taxa_node" class="help-header sql_schema_show_hide" onclick="show_hide('ncbi_taxa_node','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">This table contains all taxa used in this database, which mirror the data and tree structure from NCBI Taxonomy database (for more details see ensembl-compara/script/taxonomy/README-taxonomy which explain our import process)</p>

  <div id="div_ncbi_taxa_node" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>taxon_id</b></td>
        <td class="bg1">int(10)</td>
        <td class="bg1">-</td>
        <td class="bg1">The NCBI Taxonomy ID</td>
        <td class="bg1">primary key</td>
      </tr>

      <tr>
        <td class="bg2"><b>parent_id</b></td>
        <td class="bg2">int(10)</td>
        <td class="bg2">-</td>
        <td class="bg2">The parent taxonomy ID for this node (refers to ncbi_taxa_node.taxon_id)</td>
        <td class="bg2">key: <i>parent_id</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>rank</b></td>
        <td class="bg1">char(32)</td>
        <td class="bg1">''</td>
        <td class="bg1">E.g. kingdom, family, genus, etc.</td>
        <td class="bg1">key: <i>rank</i></td>
      </tr>

      <tr>
        <td class="bg2"><b>genbank_hidden_flag</b></td>
        <td class="bg2">tinyint(1)</td>
        <td class="bg2">0</td>
        <td class="bg2">Boolean value which defines whether this rank is used or not in the abbreviated lineage</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>left_index</b></td>
        <td class="bg1">int(10)</td>
        <td class="bg1">0</td>
        <td class="bg1">Sub-set left index. All sub-nodes have left_index and right_index values larger than this left_index</td>
        <td class="bg1">key: <i>left_index</i></td>
      </tr>

      <tr>
        <td class="bg2"><b>right_index</b></td>
        <td class="bg2">int(10)</td>
        <td class="bg2">0</td>
        <td class="bg2">Sub-set right index. All sub-nodes have left_index and right_index values smaller than this right_index</td>
        <td class="bg2">key: <i>right_index</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>root_id</b></td>
        <td class="bg1">int(10)</td>
        <td class="bg1">1</td>
        <td class="bg1">The root taxonomy ID for this node (refers to ncbi_taxa_node.taxon_id)</td>
        <td class="bg1"></td>
      </tr>
    </table>
  </div>
<div class="sql_schema_table_examples"><p class="sql_schema_table_example_header">Example:</p><div class="sql_schema_table_example_content"><p>This examples shows how to get the lineage for Homo sapiens: </p>
      <div>
        <div class="sql_schema_table_example_query">
          <pre><span class="sql_schema_sql_highlight">SELECT</span> n2.taxon_id, n2.parent_id, na.name, n2.rank, n2.left_index, n2.right_index <span class="sql_schema_sql_highlight">FROM</span> ncbi_taxa_node n1 <span class="sql_schema_sql_highlight">JOIN</span> (ncbi_taxa_node n2 <span class="sql_schema_sql_highlight">LEFT</span> <span class="sql_schema_sql_highlight">JOIN</span> ncbi_taxa_name na <span class="sql_schema_sql_highlight">ON</span> n2.taxon_id = na.taxon_id <span class="sql_schema_sql_highlight">AND</span> na.name_class = &quot;scientific name&quot;)  <span class="sql_schema_sql_highlight">ON</span> n2.left_index &lt;= n1.left_index <span class="sql_schema_sql_highlight">AND</span> n2.right_index &gt;= n1.right_index <span class="sql_schema_sql_highlight">WHERE</span> n1.taxon_id = 9606 <span class="sql_schema_sql_highlight">ORDER</span> <span class="sql_schema_sql_highlight">BY</span> left_index;</pre>
        </div>
        <div class="sql_schema_table_example_button">
  <a id="e_ncbi_taxa_node" class="help-header sql_schema_show_hide" onclick="show_hide('ncbi_taxa_node','query results')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show query results
  </a></div>
        <div style="clear:both"></div>
      </div>
      
  <div id="ex_ncbi_taxa_node" style="display:none;">
    <table class="ss sql_schema_table_example_result">
      <tr><th>n2.taxon_id</th><th>n2.parent_id</th><th>na.name</th><th>n2.rank</th><th>n2.left_index</th><th>n2.right_index</th></tr>      <tr><td>1</td><td>0</td><td>root</td><td>no rank</td><td>1</td><td>4839748</td></tr>      <tr class="bg2"><td>131567</td><td>1</td><td>cellular organisms</td><td>no rank</td><td>454784</td><td>4803281</td></tr>      <tr><td>2759</td><td>131567</td><td>Eukaryota</td><td>superkingdom</td><td>1546141</td><td>4803280</td></tr>      <tr class="bg2"><td>33154</td><td>2759</td><td>Opisthokonta</td><td>clade</td><td>2053414</td><td>4709947</td></tr>      <tr><td>33208</td><td>33154</td><td>Metazoa</td><td>kingdom</td><td>2416905</td><td>4709246</td></tr>      <tr class="bg2"><td>6072</td><td>33208</td><td>Eumetazoa</td><td>clade</td><td>2428610</td><td>4709233</td></tr>      <tr><td>33213</td><td>6072</td><td>Bilateria</td><td>clade</td><td>2456723</td><td>4709232</td></tr>      <tr class="bg2"><td>33511</td><td>33213</td><td>Deuterostomia</td><td>clade</td><td>4467860</td><td>4707861</td></tr>      <tr><td>7711</td><td>33511</td><td>Chordata</td><td>phylum</td><td>4480907</td><td>4707468</td></tr>      <tr class="bg2"><td>89593</td><td>7711</td><td>Craniata</td><td>subphylum</td><td>4483204</td><td>4707429</td></tr>      <tr><td>7742</td><td>89593</td><td>Vertebrata</td><td>clade</td><td>4483205</td><td>4707428</td></tr>      <tr class="bg2"><td>7776</td><td>7742</td><td>Gnathostomata</td><td>clade</td><td>4483206</td><td>4706229</td></tr>      <tr><td>117570</td><td>7776</td><td>Teleostomi</td><td>clade</td><td>4487755</td><td>4706228</td></tr>      <tr class="bg2"><td>117571</td><td>117570</td><td>Euteleostomi</td><td>clade</td><td>4487756</td><td>4706227</td></tr>      <tr><td>8287</td><td>117571</td><td>Sarcopterygii</td><td>superclass</td><td>4591507</td><td>4706226</td></tr>      <tr class="bg2"><td>1338369</td><td>8287</td><td>Dipnotetrapodomorpha</td><td>clade</td><td>4591520</td><td>4706225</td></tr>      <tr><td>32523</td><td>1338369</td><td>Tetrapoda</td><td>clade</td><td>4591583</td><td>4706224</td></tr>      <tr class="bg2"><td>32524</td><td>32523</td><td>Amniota</td><td>clade</td><td>4616682</td><td>4706223</td></tr>      <tr><td>40674</td><td>32524</td><td>Mammalia</td><td>class</td><td>4679913</td><td>4706222</td></tr>      <tr class="bg2"><td>32525</td><td>40674</td><td>Theria</td><td>clade</td><td>4679940</td><td>4706199</td></tr>      <tr><td>9347</td><td>32525</td><td>Eutheria</td><td>clade</td><td>4681311</td><td>4706198</td></tr>      <tr class="bg2"><td>1437010</td><td>9347</td><td>Boreoeutheria</td><td>clade</td><td>4682028</td><td>4706173</td></tr>      <tr><td>314146</td><td>1437010</td><td>Euarchontoglires</td><td>superorder</td><td>4693269</td><td>4706168</td></tr>      <tr class="bg2"><td>9443</td><td>314146</td><td>Primates</td><td>order</td><td>4693344</td><td>4695391</td></tr>      <tr><td>376913</td><td>9443</td><td>Haplorrhini</td><td>suborder</td><td>4693877</td><td>4695390</td></tr>      <tr class="bg2"><td>314293</td><td>376913</td><td>Simiiformes</td><td>infraorder</td><td>4693878</td><td>4695347</td></tr>      <tr><td>9526</td><td>314293</td><td>Catarrhini</td><td>parvorder</td><td>4694543</td><td>4695346</td></tr>      <tr class="bg2"><td>314295</td><td>9526</td><td>Hominoidea</td><td>superfamily</td><td>4695160</td><td>4695345</td></tr>      <tr><td>9604</td><td>314295</td><td>Hominidae</td><td>family</td><td>4695265</td><td>4695344</td></tr>      <tr class="bg2"><td>207598</td><td>9604</td><td>Homininae</td><td>subfamily</td><td>4695266</td><td>4695317</td></tr>      <tr><td>9605</td><td>207598</td><td>Homo</td><td>genus</td><td>4695299</td><td>4695316</td></tr>      <tr class="bg2"><td>9606</td><td>9605</td><td>Homo sapiens</td><td>species</td><td>4695300</td><td>4695305</td></tr>    </table>
  </div></div></div><table class="sql_schema_extra"><tr>    <td class="sql_schema_extra_left">    <p style="font-weight:bold">See also:</p>
  <ul>
      <li><a href="#ncbi_taxa_name">ncbi_taxa_name</a></li>
      </ul>
    </td>
</tr></table></div></div><div class="sql_schema_table">
  <div id="ncbi_taxa_name" class="sql_schema_table_header" style="border-top-color:#24DA06">
    <div class="sql_schema_table_header_left"><span style="background-color:#24DA06"></span>ncbi_taxa_name</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_ncbi_taxa_name" class="help-header sql_schema_show_hide" onclick="show_hide('ncbi_taxa_name','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">This table contains different names, aliases and meta data for the taxa used in Ensembl.</p>

  <div id="div_ncbi_taxa_name" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>taxon_id</b></td>
        <td class="bg1">int(10)</td>
        <td class="bg1">-</td>
        <td class="bg1">External reference to taxon_id in <a href="#ncbi_taxa_node" class="sql_schema_link">ncbi_taxa_node</a></td>
        <td class="bg1">key: <i>taxon_id</i></td>
      </tr>

      <tr>
        <td class="bg2"><b>name</b></td>
        <td class="bg2">varchar(255)</td>
        <td class="bg2">-</td>
        <td class="bg2">Information assigned to this taxon_id</td>
        <td class="bg2">key: <i>name</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>name_class</b></td>
        <td class="bg1">varchar(50)</td>
        <td class="bg1">-</td>
        <td class="bg1">Type of information. e.g. common name, genbank_synonym, scientif name, etc.</td>
        <td class="bg1">key: <i>name_class</i></td>
      </tr>
    </table>
  </div>
<div class="sql_schema_table_examples"><p class="sql_schema_table_example_header">Example:</p><div class="sql_schema_table_example_content"><p>Here is an example on how to get the taxonomic ID for a species: </p>
      <div>
        <div class="sql_schema_table_example_query">
          <pre><span class="sql_schema_sql_highlight">SELECT</span> * <span class="sql_schema_sql_highlight">FROM</span> ncbi_taxa_name <span class="sql_schema_sql_highlight">WHERE</span> name_class = &quot;scientific name&quot; <span class="sql_schema_sql_highlight">AND</span> name = &quot;Homo sapiens&quot;;</pre>
        </div>
        <div class="sql_schema_table_example_button">
  <a id="e_ncbi_taxa_name" class="help-header sql_schema_show_hide" onclick="show_hide('ncbi_taxa_name','query results')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show query results
  </a></div>
        <div style="clear:both"></div>
      </div>
      
  <div id="ex_ncbi_taxa_name" style="display:none;">
    <table class="ss sql_schema_table_example_result">
      <tr><th>taxon_id</th><th>name</th><th>name_class</th></tr>      <tr><td>9606</td><td>Homo sapiens</td><td>scientific name</td></tr>    </table>
  </div></div></div><table class="sql_schema_extra"><tr>    <td class="sql_schema_extra_left">    <p style="font-weight:bold">See also:</p>
  <ul>
      <li><a href="#ncbi_taxa_node">ncbi_taxa_node</a></li>
      </ul>
    </td>
</tr></table></div></div><div class="sql_schema_table">
  <div id="species_tree_node" class="sql_schema_table_header" style="border-top-color:#24DA06">
    <div class="sql_schema_table_header_left"><span style="background-color:#24DA06"></span>species_tree_node</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_species_tree_node" class="help-header sql_schema_show_hide" onclick="show_hide('species_tree_node','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">This table contains the nodes of the species tree used in the gene gain/loss analysis</p>

  <div id="div_species_tree_node" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>node_id</b></td>
        <td class="bg1">bigint</td>
        <td class="bg1">-</td>
        <td class="bg1">Internal unique ID</td>
        <td class="bg1">primary key</td>
      </tr>

      <tr>
        <td class="bg2"><b>parent_id</b></td>
        <td class="bg2">bigint</td>
        <td class="bg2">NULL</td>
        <td class="bg2">Link to the parent node</td>
        <td class="bg2">key: <i>parent_id</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>root_id</b></td>
        <td class="bg1">bigint</td>
        <td class="bg1">NULL</td>
        <td class="bg1">Link to the root node</td>
        <td class="bg1">key: <i>root_id</i></td>
      </tr>

      <tr>
        <td class="bg2"><b>left_index</b></td>
        <td class="bg2">INT</td>
        <td class="bg2">0</td>
        <td class="bg2">Internal index</td>
        <td class="bg2">key: <i>root_id</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>right_index</b></td>
        <td class="bg1">INT</td>
        <td class="bg1">0</td>
        <td class="bg1">Internal index</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>distance_to_parent</b></td>
        <td class="bg2">double</td>
        <td class="bg2">'1'</td>
        <td class="bg2">Phylogenetic distance between this node and its parent</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>taxon_id</b></td>
        <td class="bg1">INT</td>
        <td class="bg1">NULL</td>
        <td class="bg1">Link to NCBI taxon node</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>genome_db_id</b></td>
        <td class="bg2">INT</td>
        <td class="bg2">NULL</td>
        <td class="bg2">Link to the genome_db</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>node_name</b></td>
        <td class="bg1">VARCHAR(255)</td>
        <td class="bg1">NULL</td>
        <td class="bg1">A name that can be set to the taxon name or any other arbitrary name</td>
        <td class="bg1"></td>
      </tr>
    </table>
  </div>
<table class="sql_schema_extra"><tr>    <td class="sql_schema_extra_left">    <p style="font-weight:bold">See also:</p>
  <ul>
      <li><a href="#species_tree_node_tag">species_tree_node_tag</a></li>
      <li><a href="#species_tree_node_attr">species_tree_node_attr</a></li>
      <li><a href="#species_tree_root">species_tree_root</a></li>
      <li><a href="#CAFE_gene_family">CAFE_gene_family</a></li>
      <li><a href="#CAFE_species_gene">CAFE_species_gene</a></li>
      </ul>
    </td>
</tr></table></div></div><div class="sql_schema_table">
  <div id="species_tree_root" class="sql_schema_table_header" style="border-top-color:#24DA06">
    <div class="sql_schema_table_header_left"><span style="background-color:#24DA06"></span>species_tree_root</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_species_tree_root" class="help-header sql_schema_show_hide" onclick="show_hide('species_tree_root','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">This table stores species trees used in compara. Each tree is made of species_tree_node's</p>

  <div id="div_species_tree_root" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>root_id</b></td>
        <td class="bg1">bigint</td>
        <td class="bg1">-</td>
        <td class="bg1">Internal unique ID</td>
        <td class="bg1">primary key</td>
      </tr>

      <tr>
        <td class="bg2"><b>method_link_species_set_id</b></td>
        <td class="bg2">INT</td>
        <td class="bg2">-</td>
        <td class="bg2">External reference to method_link_species_set_id in the <a href="#method_link_species_set" class="sql_schema_link">method_link_species_set</a> table</td>
        <td class="bg2">unique: <i>key</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>label</b></td>
        <td class="bg1">VARCHAR(256)</td>
        <td class="bg1">'default'</td>
        <td class="bg1">Label to differentiate different trees with the same mlss_id</td>
        <td class="bg1">unique: <i>key</i></td>
      </tr>
    </table>
  </div>
<div class="sql_schema_table_examples"><p class="sql_schema_table_example_header">Example:</p><div class="sql_schema_table_example_content"><p>Retrieve all the species trees stored in the database </p>
      <div>
        <div class="sql_schema_table_example_query">
          <pre><span class="sql_schema_sql_highlight">SELECT</span> * <span class="sql_schema_sql_highlight">FROM</span> species_tree_root</pre>
        </div>
        <div class="sql_schema_table_example_button">
  <a id="e_species_tree_root" class="help-header sql_schema_show_hide" onclick="show_hide('species_tree_root','query results')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show query results
  </a></div>
        <div style="clear:both"></div>
      </div>
      
  <div id="ex_species_tree_root" style="display:none;">
    <table class="ss sql_schema_table_example_result">
      <tr><th>root_id</th><th>method_link_species_set_id</th><th>label</th></tr>      <tr><td>195600000</td><td>1956</td><td>default</td></tr>      <tr class="bg2"><td>200600019</td><td>1994</td><td>default</td></tr>      <tr><td>199500000</td><td>1995</td><td>default</td></tr>      <tr class="bg2"><td>199500085</td><td>1996</td><td>default</td></tr>      <tr><td>200100000</td><td>2001</td><td>default</td></tr>      <tr class="bg2"><td>200600000</td><td>2006</td><td>default</td></tr>      <tr><td>200700000</td><td>2007</td><td>default</td></tr>      <tr class="bg2"><td>200700077</td><td>2008</td><td>default</td></tr>      <tr><td>203800000</td><td>2038</td><td>default</td></tr>      <tr class="bg2"><td>203800033</td><td>2039</td><td>default</td></tr>      <tr><td>204100000</td><td>2041</td><td>default</td></tr>      <tr class="bg2"><td>4016200339</td><td>40162</td><td>cafe</td></tr>      <tr><td>4016200000</td><td>40162</td><td>default</td></tr>      <tr class="bg2"><td>4015100000</td><td>40151</td><td>default</td></tr>      <tr><td>4015700000</td><td>40157</td><td>default</td></tr>      <tr class="bg2"><td>4015900000</td><td>40159</td><td>default</td></tr>      <tr><td>4015300000</td><td>40153</td><td>default</td></tr>      <tr class="bg2"><td>4016300563</td><td>40163</td><td>cafe</td></tr>      <tr><td>4016300000</td><td>40163</td><td>default</td></tr>      <tr class="bg2"><td>4016300224</td><td>40163</td><td>full_species_tree</td></tr>      <tr><td>4016300574</td><td>60025</td><td>Ensembl</td></tr>      <tr class="bg2"><td>4016301027</td><td>60025</td><td>NCBI Taxonomy</td></tr>    </table>
  </div></div></div><table class="sql_schema_extra"><tr>    <td class="sql_schema_extra_left">    <p style="font-weight:bold">See also:</p>
  <ul>
      <li><a href="#species_tree_node">species_tree_node</a></li>
      <li><a href="#method_link_species_set">method_link_species_set</a></li>
      </ul>
    </td>
</tr></table></div></div><div class="sql_schema_table">
  <div id="species_tree_node_tag" class="sql_schema_table_header" style="border-top-color:#24DA06">
    <div class="sql_schema_table_header_left"><span style="background-color:#24DA06"></span>species_tree_node_tag</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_species_tree_node_tag" class="help-header sql_schema_show_hide" onclick="show_hide('species_tree_node_tag','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">This table contains tag/value data for species_tree_nodes</p>

  <div id="div_species_tree_node_tag" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>node_id</b></td>
        <td class="bg1">bigint</td>
        <td class="bg1">-</td>
        <td class="bg1">Internal unique ID</td>
        <td class="bg1">key: <i>node_id_tag</i></td>
      </tr>

      <tr>
        <td class="bg2"><b>tag</b></td>
        <td class="bg2">varchar(50)</td>
        <td class="bg2">-</td>
        <td class="bg2">Tag name for the tag/value pair</td>
        <td class="bg2">key: <i>node_id_tag</i><br />key: <i>tag</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>value</b></td>
        <td class="bg1">mediumtext</td>
        <td class="bg1">-</td>
        <td class="bg1">Value for the tag/value pair</td>
        <td class="bg1"></td>
      </tr>
    </table>
  </div>
<table class="sql_schema_extra"><tr>    <td class="sql_schema_extra_left">    <p style="font-weight:bold">See also:</p>
  <ul>
      <li><a href="#species_tree_node">species_tree_node</a></li>
      <li><a href="#species_tree_root">species_tree_root</a></li>
      </ul>
    </td>
</tr></table></div></div><div class="sql_schema_table">
  <div id="species_tree_node_attr" class="sql_schema_table_header" style="border-top-color:#24DA06">
    <div class="sql_schema_table_header_left"><span style="background-color:#24DA06"></span>species_tree_node_attr</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_species_tree_node_attr" class="help-header sql_schema_show_hide" onclick="show_hide('species_tree_node_attr','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">This table contains tag/value data for species_tree_nodes</p>

  <div id="div_species_tree_node_attr" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>node_id</b></td>
        <td class="bg1">bigint</td>
        <td class="bg1">-</td>
        <td class="bg1">Internal unique ID</td>
        <td class="bg1">primary key</td>
      </tr>

      <tr>
        <td class="bg2"><b>nb_long_genes</b></td>
        <td class="bg2">int</td>
        <td class="bg2">NULL</td>
        <td class="bg2">the number of genes longer than the avg length of their orthologs</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>nb_short_genes</b></td>
        <td class="bg1">int</td>
        <td class="bg1">NULL</td>
        <td class="bg1">the number of genes shorter than the avg length of their orthologs</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>avg_dupscore</b></td>
        <td class="bg2">float</td>
        <td class="bg2">NULL</td>
        <td class="bg2">the average duplication score</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>avg_dupscore_nondub</b></td>
        <td class="bg1">float</td>
        <td class="bg1">NULL</td>
        <td class="bg1">avg_dupscore_nondub</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>nb_dubious_nodes</b></td>
        <td class="bg2">int</td>
        <td class="bg2">NULL</td>
        <td class="bg2">nb_dubious_nodes</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>nb_dup_nodes</b></td>
        <td class="bg1">int</td>
        <td class="bg1">NULL</td>
        <td class="bg1">nb_dup_nodes</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>nb_genes</b></td>
        <td class="bg2">int</td>
        <td class="bg2">NULL</td>
        <td class="bg2">the number of genes</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>nb_genes_in_tree</b></td>
        <td class="bg1">int</td>
        <td class="bg1">NULL</td>
        <td class="bg1">the number of genes in the tree </td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>nb_genes_in_tree_multi_species</b></td>
        <td class="bg2">int</td>
        <td class="bg2">NULL</td>
        <td class="bg2">nb_genes_in_tree_multi_species</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>nb_genes_in_tree_single_species</b></td>
        <td class="bg1">int</td>
        <td class="bg1">NULL</td>
        <td class="bg1">nb_genes_in_tree_single_species</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>nb_nodes</b></td>
        <td class="bg2">int</td>
        <td class="bg2">NULL</td>
        <td class="bg2">the number of nodes</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>nb_orphan_genes</b></td>
        <td class="bg1">int</td>
        <td class="bg1">NULL</td>
        <td class="bg1">nb_orphan_genes</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>nb_seq</b></td>
        <td class="bg2">int</td>
        <td class="bg2">NULL</td>
        <td class="bg2">the number of sequences</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>nb_spec_nodes</b></td>
        <td class="bg1">int</td>
        <td class="bg1">NULL</td>
        <td class="bg1">nb_spec_nodes</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>nb_gene_splits</b></td>
        <td class="bg2">int</td>
        <td class="bg2">NULL</td>
        <td class="bg2">the number of split gene events</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>nb_split_genes</b></td>
        <td class="bg1">int</td>
        <td class="bg1">NULL</td>
        <td class="bg1">the number of split genes</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>root_avg_gene</b></td>
        <td class="bg2">float</td>
        <td class="bg2">NULL</td>
        <td class="bg2">root_avg_gene</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>root_avg_gene_per_spec</b></td>
        <td class="bg1">float</td>
        <td class="bg1">NULL</td>
        <td class="bg1">root_avg_gene_per_spec</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>root_avg_spec</b></td>
        <td class="bg2">float</td>
        <td class="bg2">NULL</td>
        <td class="bg2">root_avg_spec</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>root_max_gene</b></td>
        <td class="bg1">int</td>
        <td class="bg1">NULL</td>
        <td class="bg1">root_max_gene</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>root_max_spec</b></td>
        <td class="bg2">int</td>
        <td class="bg2">NULL</td>
        <td class="bg2">root_max_spec</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>root_min_gene</b></td>
        <td class="bg1">int</td>
        <td class="bg1">NULL</td>
        <td class="bg1">root_min_gene</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>root_min_spec</b></td>
        <td class="bg2">int</td>
        <td class="bg2">NULL</td>
        <td class="bg2">root_min_spec</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>root_nb_genes</b></td>
        <td class="bg1">int</td>
        <td class="bg1">NULL</td>
        <td class="bg1">root_nb_genes</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>root_nb_trees</b></td>
        <td class="bg2">int</td>
        <td class="bg2">NULL</td>
        <td class="bg2">root_nb_trees</td>
        <td class="bg2"></td>
      </tr>
    </table>
  </div>
<table class="sql_schema_extra"><tr>    <td class="sql_schema_extra_left">    <p style="font-weight:bold">See also:</p>
  <ul>
      <li><a href="#species_tree_node">species_tree_node</a></li>
      <li><a href="#species_tree_root">species_tree_root</a></li>
      <li><a href="#gene_member_qc">gene_member_qc</a></li>
      </ul>
    </td>
</tr></table></div></div>

<div id="header_3" class="sql_schema_group_header" style="border-color:#808000">
  <div class="sql_schema_group_bullet" style="background-color:#808000"></div>
  <h2>Genomes</h2>
</div>
<p class="sql_schema_group_header_desc">Description of the genomes (assembly, sequences, genes, etc)</p><div align="center"><a href="diagrams/Compara.genomes.png"><img src="diagrams/Compara.genomes.png" width="253px" /></a></div><div class="sql_schema_table">
  <div id="genome_db" class="sql_schema_table_header" style="border-top-color:#808000">
    <div class="sql_schema_table_header_left"><span style="background-color:#808000"></span>genome_db</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_genome_db" class="help-header sql_schema_show_hide" onclick="show_hide('genome_db','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">This table contains information about the version of the genome assemblies used in this database</p>

  <div id="div_genome_db" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>genome_db_id</b></td>
        <td class="bg1">INT</td>
        <td class="bg1">-</td>
        <td class="bg1">Internal unique ID for this table</td>
        <td class="bg1">primary key</td>
      </tr>

      <tr>
        <td class="bg2"><b>taxon_id</b></td>
        <td class="bg2">INT</td>
        <td class="bg2">NULL</td>
        <td class="bg2">External reference to taxon_id in the <a href="#ncbi_taxa_node" class="sql_schema_link">ncbi_taxa_node</a> table</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>name</b></td>
        <td class="bg1">varchar(128)</td>
        <td class="bg1">''</td>
        <td class="bg1">Species name</td>
        <td class="bg1">unique key: <i>name</i></td>
      </tr>

      <tr>
        <td class="bg2"><b>assembly</b></td>
        <td class="bg2">varchar(100)</td>
        <td class="bg2">''</td>
        <td class="bg2">Assembly version of the genome</td>
        <td class="bg2">unique key: <i>name</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>genebuild</b></td>
        <td class="bg1">varchar(255)</td>
        <td class="bg1">''</td>
        <td class="bg1">Version of the genebuild</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>has_karyotype</b></td>
        <td class="bg2">tinyint(1)</td>
        <td class="bg2">0</td>
        <td class="bg2">Whether the genome has a karyotype</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>is_good_for_alignment</b></td>
        <td class="bg1">TINYINT(1)</td>
        <td class="bg1">0</td>
        <td class="bg1">Whether the genome is good enough to be used in multiple alignments</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>genome_component</b></td>
        <td class="bg2">varchar(5)</td>
        <td class="bg2">NULL</td>
        <td class="bg2">Only used for polyploid genomes: the name of the genome component</td>
        <td class="bg2">unique key: <i>name</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>strain_name</b></td>
        <td class="bg1">varchar(100)</td>
        <td class="bg1">NULL</td>
        <td class="bg1">Name of the particular strain this GenomeDB refers to</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>display_name</b></td>
        <td class="bg2">varchar(255)</td>
        <td class="bg2">NULL</td>
        <td class="bg2">Named used for display purposes. Imported from the core databases</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>locator</b></td>
        <td class="bg1">varchar(400)</td>
        <td class="bg1">NULL</td>
        <td class="bg1">Used for production purposes or for user configuration in in-house installation.</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>first_release</b></td>
        <td class="bg2">smallint</td>
        <td class="bg2">NULL</td>
        <td class="bg2">The first release this genome was present in</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>last_release</b></td>
        <td class="bg1">smallint</td>
        <td class="bg1">NULL</td>
        <td class="bg1">The last release this genome was present in, or NULL if it is still current</td>
        <td class="bg1"></td>
      </tr>
    </table>
  </div>
<div class="sql_schema_table_examples"><p class="sql_schema_table_example_header">Example:</p><div class="sql_schema_table_example_content"><p>This query shows the entries for human and chicken </p>
      <div>
        <div class="sql_schema_table_example_query">
          <pre><span class="sql_schema_sql_highlight">SELECT</span> * <span class="sql_schema_sql_highlight">FROM</span> genome_db <span class="sql_schema_sql_highlight">WHERE</span> name <span class="sql_schema_sql_highlight">IN</span> (&quot;Homo_sapiens&quot;, &quot;Gallus_gallus&quot;);</pre>
        </div>
        <div class="sql_schema_table_example_button">
  <a id="e_genome_db" class="help-header sql_schema_show_hide" onclick="show_hide('genome_db','query results')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show query results
  </a></div>
        <div style="clear:both"></div>
      </div>
      
  <div id="ex_genome_db" style="display:none;">
    <table class="ss sql_schema_table_example_result">
      <tr><th>genome_db_id</th><th>taxon_id</th><th>name</th><th>assembly</th><th>genebuild</th><th>has_karyotype</th><th>is_good_for_alignment</th><th>genome_component</th><th>strain_name</th><th>display_name</th><th>locator</th><th>first_release</th><th>last_release</th></tr>      <tr><td>481</td><td>9031</td><td>gallus_gallus</td><td>bGalGal1.mat.broiler.GRCg7b</td><td>2021-09-Ensembl</td><td>1</td><td>1</td><td>NULL</td><td>reference breed</td><td>Chicken</td><td>NULL</td><td>107</td><td>NULL</td></tr>      <tr class="bg2"><td>150</td><td>9606</td><td>homo_sapiens</td><td>GRCh38</td><td>2014-01-Ensembl</td><td>1</td><td>1</td><td>NULL</td><td>NULL</td><td>Human</td><td>NULL</td><td>76</td><td>NULL</td></tr>    </table>
  </div></div></div><table class="sql_schema_extra"><tr>    <td class="sql_schema_extra_left">    <p style="font-weight:bold">See also:</p>
  <ul>
      <li><a href="#ncbi_taxa_node">ncbi_taxa_node</a></li>
      </ul>
    </td>
</tr></table></div></div><div class="sql_schema_table">
  <div id="dnafrag" class="sql_schema_table_header" style="border-top-color:#808000">
    <div class="sql_schema_table_header_left"><span style="background-color:#808000"></span>dnafrag</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_dnafrag" class="help-header sql_schema_show_hide" onclick="show_hide('dnafrag','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">This table defines the genomic sequences used in the comparative genomics analyisis. It is used by the <a href="#genomic_align_block" class="sql_schema_link">genomic_align_block</a> table to define aligned sequences. It is also used by the <a href="#dnafrag_region" class="sql_schema_link">dnafrag_region</a> table to define syntenic regions.<br />NOTE: Index &lt;name&gt; has genome_db_id in the first place because unless fetching all dnafrags or fetching by dnafrag_id, genome_db_id appears always in the WHERE clause. Unique key &lt;name&gt; is used to ensure that Bio::EnsEMBL::Compara::DBSQL::DnaFragAdaptor->fetch_by_GenomeDB_and_name will always fetch a single row. This can be used in the EnsEMBL Compara DB because we store top-level dnafrags only.</p>

  <div id="div_dnafrag" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>dnafrag_id</b></td>
        <td class="bg1">bigint</td>
        <td class="bg1">-</td>
        <td class="bg1">Internal unique ID</td>
        <td class="bg1">primary key</td>
      </tr>

      <tr>
        <td class="bg2"><b>length</b></td>
        <td class="bg2">int</td>
        <td class="bg2">0</td>
        <td class="bg2">The total length of the dnafrag</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>name</b></td>
        <td class="bg1">varchar(255)</td>
        <td class="bg1">''</td>
        <td class="bg1">Name of the DNA sequence (e.g., the name of the chromosome)</td>
        <td class="bg1">unique key: <i>name</i></td>
      </tr>

      <tr>
        <td class="bg2"><b>genome_db_id</b></td>
        <td class="bg2">INT</td>
        <td class="bg2">-</td>
        <td class="bg2">External reference to genome_db_id in the <a href="#genome_db" class="sql_schema_link">genome_db</a> table</td>
        <td class="bg2">unique key: <i>name</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>coord_system_name</b></td>
        <td class="bg1">varchar(40)</td>
        <td class="bg1">''</td>
        <td class="bg1">Refers to the coord system in which this dnafrag has been defined</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>is_reference</b></td>
        <td class="bg2">tinyint(1)</td>
        <td class="bg2">1</td>
        <td class="bg2">Boolean, whether dnafrag is reference (1) or non-reference (0) eg haplotype</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>cellular_component</b></td>
        <td class="bg1">ENUM:<ul class="sql_schema_table_column_type">  <li>NUC</li>  <li> MT</li>  <li> PT</li>  <li> OTHER</li></ul></td>
        <td class="bg1">'NUC'</td>
        <td class="bg1">Either &quot;NUC&quot;, &quot;MT&quot;, &quot;PT&quot; or &quot;OTHER&quot;. Represents which organelle genome the dnafrag is part of</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>codon_table_id</b></td>
        <td class="bg2">TINYINT</td>
        <td class="bg2">1</td>
        <td class="bg2">Integer. The numeric identifier of the codon-table that applies to this dnafrag (https://www.ncbi.nlm.nih.gov/Taxonomy/Utils/wprintgc.cgi)</td>
        <td class="bg2"></td>
      </tr>
    </table>
  </div>
<div class="sql_schema_table_examples"><p class="sql_schema_table_example_header">Example:</p><div class="sql_schema_table_example_content"><p>This query shows the chromosome 14 of the Human genome (genome_db.genome_db_id = 150 refers to Human genome in this example) which is 107349540 nucleotides long. </p>
      <div>
        <div class="sql_schema_table_example_query">
          <pre><span class="sql_schema_sql_highlight">SELECT</span> dnafrag.* <span class="sql_schema_sql_highlight">FROM</span> dnafrag <span class="sql_schema_sql_highlight">LEFT</span> <span class="sql_schema_sql_highlight">JOIN</span> genome_db <span class="sql_schema_sql_highlight">USING</span> (genome_db_id) <span class="sql_schema_sql_highlight">WHERE</span> dnafrag.name = &quot;14&quot; <span class="sql_schema_sql_highlight">AND</span> genome_db.name = &quot;homo_sapiens&quot;;</pre>
        </div>
        <div class="sql_schema_table_example_button">
  <a id="e_dnafrag" class="help-header sql_schema_show_hide" onclick="show_hide('dnafrag','query results')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show query results
  </a></div>
        <div style="clear:both"></div>
      </div>
      
  <div id="ex_dnafrag" style="display:none;">
    <table class="ss sql_schema_table_example_result">
      <tr><th>dnafrag_id</th><th>length</th><th>name</th><th>genome_db_id</th><th>coord_system_name</th><th>cellular_component</th><th>is_reference</th><th>codon_table_id</th></tr>      <tr><td>13955524</td><td>107043718</td><td>14</td><td>150</td><td>chromosome</td><td>NUC</td><td>1</td><td>1</td></tr>    </table>
  </div></div></div><table class="sql_schema_extra"><tr>    <td class="sql_schema_extra_left">    <p style="font-weight:bold">See also:</p>
  <ul>
      <li><a href="#genomic_align_block">genomic_align_block</a></li>
      <li><a href="#dnafrag_region">dnafrag_region</a></li>
      <li><a href="#dnafrag_alt_region">dnafrag_alt_region</a></li>
      <li><a href="#genome_db">genome_db</a></li>
      </ul>
    </td>
</tr></table></div></div><div class="sql_schema_table">
  <div id="dnafrag_alt_region" class="sql_schema_table_header" style="border-top-color:#808000">
    <div class="sql_schema_table_header_left"><span style="background-color:#808000"></span>dnafrag_alt_region</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_dnafrag_alt_region" class="help-header sql_schema_show_hide" onclick="show_hide('dnafrag_alt_region','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">This table lists the position of the alternative sequences on non-reference dnafrags (haplotypes and assembly patches). On these dnafrags, only the region between these two coordinates differ from the base (reference) dnafrag.</p>

  <div id="div_dnafrag_alt_region" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>dnafrag_id</b></td>
        <td class="bg1">BIGINT</td>
        <td class="bg1">-</td>
        <td class="bg1">External reference to dnafrag_id in the <a href="#dnafrag" class="sql_schema_link">dnafrag</a> table</td>
        <td class="bg1">primary key</td>
      </tr>

      <tr>
        <td class="bg2"><b>dnafrag_start</b></td>
        <td class="bg2">INT</td>
        <td class="bg2">-</td>
        <td class="bg2">Position of the first nucleotide from this dnafrag that differs from the reference dnafrag</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>dnafrag_end</b></td>
        <td class="bg1">INT</td>
        <td class="bg1">-</td>
        <td class="bg1">Position of the last nucleotide from this dnafrag that differs from the reference dnafrag</td>
        <td class="bg1"></td>
      </tr>
    </table>
  </div>
<div class="sql_schema_table_examples"><p class="sql_schema_table_example_header">Example:</p><div class="sql_schema_table_example_content"><p>This query lists some haplotypes of the human chromosome 20, incl. the length of the complete patched chromosome and the coordinates of the region that differs. </p>
      <div>
        <div class="sql_schema_table_example_query">
          <pre><span class="sql_schema_sql_highlight">SELECT</span> dnafrag_id, name, length, dnafrag_start, dnafrag_end <span class="sql_schema_sql_highlight">FROM</span> dnafrag_alt_regi<span class="sql_schema_sql_highlight">ON</span> <span class="sql_schema_sql_highlight">JOIN</span> dnafrag <span class="sql_schema_sql_highlight">USING</span> (dnafrag_id) <span class="sql_schema_sql_highlight">WHERE</span> name LIKE &quot;CHR\_HSCHR20\_%&quot;;</pre>
        </div>
        <div class="sql_schema_table_example_button">
  <a id="e_dnafrag_alt_region" class="help-header sql_schema_show_hide" onclick="show_hide('dnafrag_alt_region','query results')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show query results
  </a></div>
        <div style="clear:both"></div>
      </div>
      
  <div id="ex_dnafrag_alt_region" style="display:none;">
    <table class="ss sql_schema_table_example_result">
      <tr><th>dnafrag_id</th><th>name</th><th>length</th><th>dnafrag_start</th><th>dnafrag_end</th></tr>      <tr><td>13955256</td><td>CHR_HSCHR20_1_CTG3</td><td>64447812</td><td>64095055</td><td>64278487</td></tr>      <tr class="bg2"><td>13955314</td><td>CHR_HSCHR20_1_CTG4</td><td>64446453</td><td>63414721</td><td>63473381</td></tr>      <tr><td>13955469</td><td>CHR_HSCHR20_1_CTG1</td><td>64450183</td><td>17771102</td><td>17899487</td></tr>      <tr class="bg2"><td>13955496</td><td>CHR_HSCHR20_1_CTG2</td><td>64441059</td><td>61886306</td><td>62005079</td></tr>    </table>
  </div></div></div><table class="sql_schema_extra"><tr>    <td class="sql_schema_extra_left">    <p style="font-weight:bold">See also:</p>
  <ul>
      <li><a href="#dnafrag">dnafrag</a></li>
      </ul>
    </td>
</tr></table></div></div><div class="sql_schema_table">
  <div id="sequence" class="sql_schema_table_header" style="border-top-color:#808000">
    <div class="sql_schema_table_header_left"><span style="background-color:#808000"></span>sequence</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_sequence" class="help-header sql_schema_show_hide" onclick="show_hide('sequence','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">This table contains the sequences of the seq_member entries</p>

  <div id="div_sequence" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>sequence_id</b></td>
        <td class="bg1">INT</td>
        <td class="bg1">-</td>
        <td class="bg1">Internal unique ID</td>
        <td class="bg1">primary key</td>
      </tr>

      <tr>
        <td class="bg2"><b>length</b></td>
        <td class="bg2">INT</td>
        <td class="bg2">-</td>
        <td class="bg2">Length of the sequence</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>sequence</b></td>
        <td class="bg1">longtext</td>
        <td class="bg1">-</td>
        <td class="bg1">The actual sequence</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>md5sum</b></td>
        <td class="bg2">CHAR(32)</td>
        <td class="bg2">-</td>
        <td class="bg2">md5sum</td>
        <td class="bg2">key: <i>md5sum</i></td>
      </tr>
    </table>
  </div>
</div></div><div class="sql_schema_table">
  <div id="gene_member" class="sql_schema_table_header" style="border-top-color:#808000">
    <div class="sql_schema_table_header_left"><span style="background-color:#808000"></span>gene_member</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_gene_member" class="help-header sql_schema_show_hide" onclick="show_hide('gene_member','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">This table links sequences to the EnsEMBL core DB or to external DBs.</p>

  <div id="div_gene_member" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>gene_member_id</b></td>
        <td class="bg1">INT</td>
        <td class="bg1">-</td>
        <td class="bg1">Internal unique ID</td>
        <td class="bg1">primary key</td>
      </tr>

      <tr>
        <td class="bg2"><b>stable_id</b></td>
        <td class="bg2">varchar(128)</td>
        <td class="bg2">-</td>
        <td class="bg2">EnsEMBL stable ID</td>
        <td class="bg2">unique: <i>key</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>version</b></td>
        <td class="bg1">INT</td>
        <td class="bg1">0</td>
        <td class="bg1">Version of the stable ID (see EnsEMBL core DB)</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>source_name</b></td>
        <td class="bg2">ENUM:<ul class="sql_schema_table_column_type">  <li>ENSEMBLGENE</li>  <li> EXTERNALGENE</li></ul></td>
        <td class="bg2">-</td>
        <td class="bg2">The source of the member</td>
        <td class="bg2">key: <i>source_name</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>taxon_id</b></td>
        <td class="bg1">INT</td>
        <td class="bg1">-</td>
        <td class="bg1">External reference to taxon_id in the <a href="#ncbi_taxa_node" class="sql_schema_link">ncbi_taxa_node</a> table</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>genome_db_id</b></td>
        <td class="bg2">INT</td>
        <td class="bg2">NULL</td>
        <td class="bg2">External reference to genome_db_id in the <a href="#genome_db" class="sql_schema_link">genome_db</a> table</td>
        <td class="bg2">key: <i>genome_db_id_biotype</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>biotype_group</b></td>
        <td class="bg1">ENUM:<ul class="sql_schema_table_column_type">  <li>coding</li>  <li>pseudogene</li>  <li>snoncoding</li>  <li>lnoncoding</li>  <li>mnoncoding</li>  <li>LRG</li>  <li>undefined</li>  <li>no_group</li>  <li>current_notdumped</li>  <li>notcurrent</li></ul></td>
        <td class="bg1">'coding'</td>
        <td class="bg1">Biotype of this gene.</td>
        <td class="bg1">key: <i>biotype_dnafrag_id_start_end</i><br />key: <i>genome_db_id_biotype</i></td>
      </tr>

      <tr>
        <td class="bg2"><b>canonical_member_id</b></td>
        <td class="bg2">INT</td>
        <td class="bg2">NULL</td>
        <td class="bg2">External reference to seq_member_id in the <a href="#seq_member" class="sql_schema_link">seq_member</a> table to allow linkage from a gene to its canonical peptide</td>
        <td class="bg2">key: <i>canonical_member_id</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>description</b></td>
        <td class="bg1">text</td>
        <td class="bg1">NULL</td>
        <td class="bg1">The description of the gene/protein as described in the core database or from the Uniprot entry</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>dnafrag_id</b></td>
        <td class="bg2">bigint</td>
        <td class="bg2">NULL</td>
        <td class="bg2">External reference to dnafrag_id in the <a href="#dnafrag" class="sql_schema_link">dnafrag</a> table. It shows the dnafrag the member is on.</td>
        <td class="bg2">key: <i>dnafrag_id_start</i><br />key: <i>dnafrag_id_end</i><br />key: <i>biotype_dnafrag_id_start_end</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>dnafrag_start</b></td>
        <td class="bg1">INT</td>
        <td class="bg1">NULL</td>
        <td class="bg1">Starting position within the dnafrag defined by dnafrag_id</td>
        <td class="bg1">key: <i>dnafrag_id_start</i><br />key: <i>biotype_dnafrag_id_start_end</i></td>
      </tr>

      <tr>
        <td class="bg2"><b>dnafrag_end</b></td>
        <td class="bg2">INT</td>
        <td class="bg2">NULL</td>
        <td class="bg2">Ending position within the dnafrag defined by dnafrag_id</td>
        <td class="bg2">key: <i>dnafrag_id_end</i><br />key: <i>biotype_dnafrag_id_start_end</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>dnafrag_strand</b></td>
        <td class="bg1">TINYINT</td>
        <td class="bg1">NULL</td>
        <td class="bg1">Strand in the dnafrag defined by dnafrag_id</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>display_label</b></td>
        <td class="bg2">varchar(128)</td>
        <td class="bg2">NULL</td>
        <td class="bg2">Display name (imported from the core database)</td>
        <td class="bg2"></td>
      </tr>
    </table>
  </div>
<div class="sql_schema_table_examples"><p class="sql_schema_table_example_header">Example:</p><div class="sql_schema_table_example_content"><p>The following query refers to the human (ncbi_taxa_node.taxon_id = 9606 or genome_db_id = 150) gene ENSG00000176105 </p>
      <div>
        <div class="sql_schema_table_example_query">
          <pre><span class="sql_schema_sql_highlight">SELECT</span> * <span class="sql_schema_sql_highlight">FROM</span> gene_member <span class="sql_schema_sql_highlight">WHERE</span> stable_id = &quot;ENSG00000176105&quot;;</pre>
        </div>
        <div class="sql_schema_table_example_button">
  <a id="e_gene_member" class="help-header sql_schema_show_hide" onclick="show_hide('gene_member','query results')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show query results
  </a></div>
        <div style="clear:both"></div>
      </div>
      
  <div id="ex_gene_member" style="display:none;">
    <table class="ss sql_schema_table_example_result">
      <tr><th>gene_member_id</th><th>stable_id</th><th>version</th><th>source_name</th><th>taxon_id</th><th>genome_db_id</th><th>biotype_group</th><th>canonical_member_id</th><th>description</th><th>dnafrag_id</th><th>dnafrag_start</th><th>dnafrag_end</th><th>dnafrag_strand</th><th>display_label</th></tr>      <tr><td>915944340</td><td>ENSG00000176105</td><td>14</td><td>ENSEMBLGENE</td><td>9606</td><td>150</td><td>coding</td><td>953120404</td><td>YES proto-oncogene 1, Src family tyrosine kinase [Source:HGNC Symbol;Acc:HGNC:12841]</td><td>13955525</td><td>721588</td><td>812546</td><td>-1</td><td>YES1</td></tr>    </table>
  </div></div></div><table class="sql_schema_extra"><tr>    <td class="sql_schema_extra_left">    <p style="font-weight:bold">See also:</p>
  <ul>
      <li><a href="#sequence">sequence</a></li>
      <li><a href="#ncbi_taxa_node">ncbi_taxa_node</a></li>
      <li><a href="#genome_db">genome_db</a></li>
      <li><a href="#seq_member">seq_member</a></li>
      <li><a href="#dnafrag">dnafrag</a></li>
      </ul>
    </td>
</tr></table></div></div><div class="sql_schema_table">
  <div id="seq_member" class="sql_schema_table_header" style="border-top-color:#808000">
    <div class="sql_schema_table_header_left"><span style="background-color:#808000"></span>seq_member</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_seq_member" class="help-header sql_schema_show_hide" onclick="show_hide('seq_member','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">This table links sequences to the EnsEMBL core DB or to external DBs.</p>

  <div id="div_seq_member" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>seq_member_id</b></td>
        <td class="bg1">INT</td>
        <td class="bg1">-</td>
        <td class="bg1">Internal unique ID</td>
        <td class="bg1">primary key<br />key: <i>seq_member_gene_member_id_end</i></td>
      </tr>

      <tr>
        <td class="bg2"><b>stable_id</b></td>
        <td class="bg2">varchar(128)</td>
        <td class="bg2">-</td>
        <td class="bg2">EnsEMBL stable ID or external ID (for Uniprot/SWISSPROT and Uniprot/SPTREMBL)</td>
        <td class="bg2">unique: <i>key</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>version</b></td>
        <td class="bg1">INT</td>
        <td class="bg1">0</td>
        <td class="bg1">Version of the stable ID (see EnsEMBL core DB)</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>source_name</b></td>
        <td class="bg2">ENUM:<ul class="sql_schema_table_column_type">  <li>ENSEMBLPEP</li>  <li>ENSEMBLTRANS</li>  <li>Uniprot/SPTREMBL</li>  <li>Uniprot/SWISSPROT</li>  <li>EXTERNALPEP</li>  <li>EXTERNALTRANS</li>  <li>EXTERNALCDS</li></ul></td>
        <td class="bg2">-</td>
        <td class="bg2">The source of the member</td>
        <td class="bg2">key: <i>source_name</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>taxon_id</b></td>
        <td class="bg1">INT</td>
        <td class="bg1">-</td>
        <td class="bg1">External reference to taxon_id in the <a href="#ncbi_taxa_node" class="sql_schema_link">ncbi_taxa_node</a> table</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>genome_db_id</b></td>
        <td class="bg2">INT</td>
        <td class="bg2">NULL</td>
        <td class="bg2">External reference to genome_db_id in the <a href="#genome_db" class="sql_schema_link">genome_db</a> table</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>sequence_id</b></td>
        <td class="bg1">INT</td>
        <td class="bg1">NULL</td>
        <td class="bg1">External reference to sequence_id in the <a href="#sequence" class="sql_schema_link">sequence</a> table. May be 0 when the sequence is not available in the <a href="#sequence" class="sql_schema_link">sequence</a> table, e.g. for a gene instance</td>
        <td class="bg1">key: <i>sequence_id</i></td>
      </tr>

      <tr>
        <td class="bg2"><b>gene_member_id</b></td>
        <td class="bg2">INT</td>
        <td class="bg2">NULL</td>
        <td class="bg2">External reference to gene_member_id in the <a href="#gene_member" class="sql_schema_link">gene_member</a> table to allow linkage from peptides and transcripts to genes</td>
        <td class="bg2">key: <i>gene_member_id</i><br />key: <i>seq_member_gene_member_id_end</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>has_transcript_edits</b></td>
        <td class="bg1">tinyint(1)</td>
        <td class="bg1">0</td>
        <td class="bg1">Boolean. Whether there are SeqEdits that modify the transcript sequence. When this happens, the (exon) coordinates don&#39;t match the transcript sequence</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>has_translation_edits</b></td>
        <td class="bg2">tinyint(1)</td>
        <td class="bg2">0</td>
        <td class="bg2">Boolean. Whether there are SeqEdits that modify the protein sequence. When this happens, the protein sequence doesn&#39;t match the transcript sequence</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>description</b></td>
        <td class="bg1">text</td>
        <td class="bg1">NULL</td>
        <td class="bg1">The description of the gene/protein as described in the core database or from the Uniprot entry</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>dnafrag_id</b></td>
        <td class="bg2">bigint</td>
        <td class="bg2">NULL</td>
        <td class="bg2">External reference to dnafrag_id in the <a href="#dnafrag" class="sql_schema_link">dnafrag</a> table. It shows the dnafrag the member is on.</td>
        <td class="bg2">key: <i>dnafrag_id_start</i><br />key: <i>dnafrag_id_end</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>dnafrag_start</b></td>
        <td class="bg1">INT</td>
        <td class="bg1">NULL</td>
        <td class="bg1">Starting position within the dnafrag defined by dnafrag_id</td>
        <td class="bg1">key: <i>dnafrag_id_start</i></td>
      </tr>

      <tr>
        <td class="bg2"><b>dnafrag_end</b></td>
        <td class="bg2">INT</td>
        <td class="bg2">NULL</td>
        <td class="bg2">Ending position within the dnafrag defined by dnafrag_id</td>
        <td class="bg2">key: <i>dnafrag_id_end</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>dnafrag_strand</b></td>
        <td class="bg1">TINYINT</td>
        <td class="bg1">NULL</td>
        <td class="bg1">Strand in the dnafrag defined by dnafrag_id</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>display_label</b></td>
        <td class="bg2">varchar(128)</td>
        <td class="bg2">NULL</td>
        <td class="bg2">Display name (imported from the core database)</td>
        <td class="bg2"></td>
      </tr>
    </table>
  </div>
<div class="sql_schema_table_examples"><p class="sql_schema_table_example_header">Example:</p><div class="sql_schema_table_example_content"><p>The following query refers to the human (ncbi_taxa_node.taxon_id = 9606 or genome_db_id = 150) peptide ENSP00000324740 </p>
      <div>
        <div class="sql_schema_table_example_query">
          <pre><span class="sql_schema_sql_highlight">SELECT</span> * <span class="sql_schema_sql_highlight">FROM</span> seq_member <span class="sql_schema_sql_highlight">WHERE</span> stable_id = &quot;ENSP00000324740&quot;;</pre>
        </div>
        <div class="sql_schema_table_example_button">
  <a id="e_seq_member" class="help-header sql_schema_show_hide" onclick="show_hide('seq_member','query results')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show query results
  </a></div>
        <div style="clear:both"></div>
      </div>
      
  <div id="ex_seq_member" style="display:none;">
    <table class="ss sql_schema_table_example_result">
      <tr><th>seq_member_id</th><th>stable_id</th><th>version</th><th>source_name</th><th>taxon_id</th><th>genome_db_id</th><th>sequence_id</th><th>gene_member_id</th><th>has_transcript_edits</th><th>has_translation_edits</th><th>description</th><th>dnafrag_id</th><th>dnafrag_start</th><th>dnafrag_end</th><th>dnafrag_strand</th><th>display_label</th></tr>      <tr><td>953120404</td><td>ENSP00000324740</td><td>5</td><td>ENSEMBLPEP</td><td>9606</td><td>150</td><td>934884733</td><td>915944340</td><td>0</td><td>0</td><td>NULL</td><td>13955525</td><td>721588</td><td>812242</td><td>-1</td><td>YES1-201</td></tr>    </table>
  </div></div></div><table class="sql_schema_extra"><tr>    <td class="sql_schema_extra_left">    <p style="font-weight:bold">See also:</p>
  <ul>
      <li><a href="#sequence">sequence</a></li>
      <li><a href="#ncbi_taxa_node">ncbi_taxa_node</a></li>
      <li><a href="#genome_db">genome_db</a></li>
      <li><a href="#gene_member">gene_member</a></li>
      <li><a href="#dnafrag">dnafrag</a></li>
      </ul>
    </td>
</tr></table></div></div><div class="sql_schema_table">
  <div id="exon_boundaries" class="sql_schema_table_header" style="border-top-color:#808000">
    <div class="sql_schema_table_header_left"><span style="background-color:#808000"></span>exon_boundaries</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_exon_boundaries" class="help-header sql_schema_show_hide" onclick="show_hide('exon_boundaries','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">This table stores the exon coordinates of a seq_member. Coordinates are assumed to be on the dnafrag of the seq_member</p>

  <div id="div_exon_boundaries" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>gene_member_id</b></td>
        <td class="bg1">INT</td>
        <td class="bg1">-</td>
        <td class="bg1">External reference to gene_member_id in the <a href="#gene_member" class="sql_schema_link">gene_member</a> table to allow querying all the exons of all the translations of a gene</td>
        <td class="bg1">index: <i>gene_member_id</i></td>
      </tr>

      <tr>
        <td class="bg2"><b>seq_member_id</b></td>
        <td class="bg2">INT</td>
        <td class="bg2">-</td>
        <td class="bg2">External reference to seq_member_id in the <a href="#seq_member" class="sql_schema_link">seq_member</a> table to indicate which translation the exons refer to</td>
        <td class="bg2">index: <i>seq_member_id</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>dnafrag_start</b></td>
        <td class="bg1">INT</td>
        <td class="bg1">-</td>
        <td class="bg1">Starting position within the dnafrag defined by the dnafrag_id of the seq_member</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>dnafrag_end</b></td>
        <td class="bg2">INT</td>
        <td class="bg2">-</td>
        <td class="bg2">Ending position within the dnafrag defined by the dnafrag_id of the seq_member</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>sequence_length</b></td>
        <td class="bg1">INT</td>
        <td class="bg1">-</td>
        <td class="bg1">Length of the chunk of the sequence that corresponds to this exon</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>left_over</b></td>
        <td class="bg2">TINYINT</td>
        <td class="bg2">0</td>
        <td class="bg2">Phase information (0, 1 or 2) used to produce the &quot;exon_bounded&quot; sequence</td>
        <td class="bg2"></td>
      </tr>
    </table>
  </div>
<table class="sql_schema_extra"><tr>    <td class="sql_schema_extra_left">    <p style="font-weight:bold">See also:</p>
  <ul>
      <li><a href="#seq_member">seq_member</a></li>
      <li><a href="#gene_member">gene_member</a></li>
      <li><a href="#dnafrag">dnafrag</a></li>
      </ul>
    </td>
</tr></table></div></div><div class="sql_schema_table">
  <div id="other_member_sequence" class="sql_schema_table_header" style="border-top-color:#808000">
    <div class="sql_schema_table_header_left"><span style="background-color:#808000"></span>other_member_sequence</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_other_member_sequence" class="help-header sql_schema_show_hide" onclick="show_hide('other_member_sequence','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">This table includes alternative sequences for Member, like sequences with flanking regions</p>

  <div id="div_other_member_sequence" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>seq_member_id</b></td>
        <td class="bg1">INT</td>
        <td class="bg1">-</td>
        <td class="bg1">External reference to seq_member_id in the <a href="#seq_member" class="sql_schema_link">seq_member</a> table</td>
        <td class="bg1">primary key</td>
      </tr>

      <tr>
        <td class="bg2"><b>seq_type</b></td>
        <td class="bg2">VARCHAR(40)</td>
        <td class="bg2">-</td>
        <td class="bg2">A short description of this alternative sequence</td>
        <td class="bg2">primary key</td>
      </tr>

      <tr>
        <td class="bg1"><b>length</b></td>
        <td class="bg1">INT</td>
        <td class="bg1">-</td>
        <td class="bg1">The length of this sequence</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>sequence</b></td>
        <td class="bg2">mediumtext</td>
        <td class="bg2">-</td>
        <td class="bg2">The actual sequence</td>
        <td class="bg2"></td>
      </tr>
    </table>
  </div>
<table class="sql_schema_extra"><tr>    <td class="sql_schema_extra_left">    <p style="font-weight:bold">See also:</p>
  <ul>
      <li><a href="#seq_member">seq_member</a></li>
      <li><a href="#sequence">sequence</a></li>
      </ul>
    </td>
</tr></table></div></div>

<div id="header_4" class="sql_schema_group_header" style="border-color:#FF6666">
  <div class="sql_schema_group_bullet" style="background-color:#FF6666"></div>
  <h2>Synteny</h2>
</div>
<p class="sql_schema_group_header_desc">These tables store information about genomic alignments in the Compara schema</p><div align="center"><a href="diagrams/Compara.synteny.png"><img src="diagrams/Compara.synteny.png" width="500px" /></a></div><div class="sql_schema_table">
  <div id="synteny_region" class="sql_schema_table_header" style="border-top-color:#FF6666">
    <div class="sql_schema_table_header_left"><span style="background-color:#FF6666"></span>synteny_region</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_synteny_region" class="help-header sql_schema_show_hide" onclick="show_hide('synteny_region','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">Contains all the syntenic relationships found and the relative orientation of both syntenic regions.</p>

  <div id="div_synteny_region" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>synteny_region_id</b></td>
        <td class="bg1">INT</td>
        <td class="bg1">-</td>
        <td class="bg1">Internal unique ID</td>
        <td class="bg1">primary key</td>
      </tr>

      <tr>
        <td class="bg2"><b>method_link_species_set_id</b></td>
        <td class="bg2">INT</td>
        <td class="bg2">-</td>
        <td class="bg2">External reference to method_link_species_set_id in the <a href="#method_link_species_set" class="sql_schema_link">method_link_species_set</a> table</td>
        <td class="bg2">key: <i>method_link_species_set_id</i></td>
      </tr>
    </table>
  </div>
<div class="sql_schema_table_examples"><p class="sql_schema_table_example_header">Example:</p><div class="sql_schema_table_example_content"><p>This query shows the 4 first syntenic regions between the Human and Opossum genomes by linking with the <a href="#method_link_species_set" class="sql_schema_link">method_link_species_set</a> table  </p>
      <div>
        <div class="sql_schema_table_example_query">
          <pre><span class="sql_schema_sql_highlight">SELECT</span> synteny_region.* <span class="sql_schema_sql_highlight">FROM</span> synteny_regi<span class="sql_schema_sql_highlight">ON</span> <span class="sql_schema_sql_highlight">JOIN</span> method_link_species_set <span class="sql_schema_sql_highlight">USING</span> (method_link_species_set_id) <span class="sql_schema_sql_highlight">JOIN</span> species_set_header <span class="sql_schema_sql_highlight">USING</span> (species_set_id) <span class="sql_schema_sql_highlight">WHERE</span> species_set_header.name = &quot;Hsap-Mdom&quot; <span class="sql_schema_sql_highlight">LIMIT</span> 4;</pre>
        </div>
        <div class="sql_schema_table_example_button">
  <a id="e_synteny_region" class="help-header sql_schema_show_hide" onclick="show_hide('synteny_region','query results')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show query results
  </a></div>
        <div style="clear:both"></div>
      </div>
      
  <div id="ex_synteny_region" style="display:none;">
    <table class="ss sql_schema_table_example_result">
      <tr><th>synteny_region_id</th><th>method_link_species_set_id</th></tr>      <tr><td>126977</td><td>10273</td></tr>      <tr class="bg2"><td>126978</td><td>10273</td></tr>      <tr><td>126979</td><td>10273</td></tr>      <tr class="bg2"><td>126980</td><td>10273</td></tr>    </table>
  </div></div></div><table class="sql_schema_extra"><tr>    <td class="sql_schema_extra_left">    <p style="font-weight:bold">See also:</p>
  <ul>
      <li><a href="#method_link_species_set">method_link_species_set</a></li>
      </ul>
    </td>
</tr></table></div></div><div class="sql_schema_table">
  <div id="dnafrag_region" class="sql_schema_table_header" style="border-top-color:#FF6666">
    <div class="sql_schema_table_header_left"><span style="background-color:#FF6666"></span>dnafrag_region</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_dnafrag_region" class="help-header sql_schema_show_hide" onclick="show_hide('dnafrag_region','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">This table contains the genomic regions corresponding to every synteny relationship found. There are two genomic regions for every synteny relationship.</p>

  <div id="div_dnafrag_region" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>synteny_region_id</b></td>
        <td class="bg1">INT</td>
        <td class="bg1">0</td>
        <td class="bg1">External reference to synteny_region_id in the <a href="#synteny_region" class="sql_schema_link">synteny_region</a> table</td>
        <td class="bg1">key: <i>synteny</i><br />key: <i>synteny_reversed</i></td>
      </tr>

      <tr>
        <td class="bg2"><b>dnafrag_id</b></td>
        <td class="bg2">bigint</td>
        <td class="bg2">0</td>
        <td class="bg2">External reference to dnafrag_id in the <a href="#dnafrag" class="sql_schema_link">dnafrag</a> table</td>
        <td class="bg2">key: <i>synteny</i><br />key: <i>synteny_reversed</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>dnafrag_start</b></td>
        <td class="bg1">INT</td>
        <td class="bg1">0</td>
        <td class="bg1">Position of the first nucleotide from this dnafrag which is in synteny</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>dnafrag_end</b></td>
        <td class="bg2">INT</td>
        <td class="bg2">0</td>
        <td class="bg2">Position of the last nucleotide from this dnafrag which is in synteny</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>dnafrag_strand</b></td>
        <td class="bg1">TINYINT</td>
        <td class="bg1">0</td>
        <td class="bg1">Strand of this region</td>
        <td class="bg1"></td>
      </tr>
    </table>
  </div>
<div class="sql_schema_table_examples"><p class="sql_schema_table_example_header">Example 1:</p><div class="sql_schema_table_example_content"><p>Return two top dnafrag regions </p>
      <div>
        <div class="sql_schema_table_example_query">
          <pre><span class="sql_schema_sql_highlight">SELECT</span> * <span class="sql_schema_sql_highlight">FROM</span> dnafrag_regi<span class="sql_schema_sql_highlight">ON</span> <span class="sql_schema_sql_highlight">ORDER</span> <span class="sql_schema_sql_highlight">BY</span> synteny_region_id <span class="sql_schema_sql_highlight">LIMIT</span> 2;</pre>
        </div>
        <div class="sql_schema_table_example_button">
  <a id="e_dnafrag_region1" class="help-header sql_schema_show_hide" onclick="show_hide('dnafrag_region1','query results')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show query results
  </a></div>
        <div style="clear:both"></div>
      </div>
      
  <div id="ex_dnafrag_region1" style="display:none;">
    <table class="ss sql_schema_table_example_result">
      <tr><th>synteny_region_id</th><th>dnafrag_id</th><th>dnafrag_start</th><th>dnafrag_end</th><th>dnafrag_strand</th></tr>      <tr><td>51817</td><td>12966551</td><td>10692525</td><td>19644506</td><td>1</td></tr>      <tr class="bg2"><td>51817</td><td>13955520</td><td>6409596</td><td>14809562</td><td>1</td></tr>    </table>
  </div></div></div><div class="sql_schema_table_examples"><p class="sql_schema_table_example_header">Example 2:</p><div class="sql_schema_table_example_content"><p>Same two dnafrag regions, but annotated </p>
      <div>
        <div class="sql_schema_table_example_query">
          <pre><span class="sql_schema_sql_highlight">SELECT</span> genome_db.name <span class="sql_schema_sql_highlight">AS</span> species_name, dnafrag.name, dnafrag_start, dnafrag_end, dnafrag_str<span class="sql_schema_sql_highlight">AND</span> <span class="sql_schema_sql_highlight">FROM</span> dnafrag_regi<span class="sql_schema_sql_highlight">ON</span> <span class="sql_schema_sql_highlight">JOIN</span> dnafrag <span class="sql_schema_sql_highlight">USING</span> (dnafrag_id) <span class="sql_schema_sql_highlight">JOIN</span> genome_db <span class="sql_schema_sql_highlight">USING</span> (genome_db_id) <span class="sql_schema_sql_highlight">ORDER</span> <span class="sql_schema_sql_highlight">BY</span> synteny_region_id <span class="sql_schema_sql_highlight">LIMIT</span> 2;</pre>
        </div>
        <div class="sql_schema_table_example_button">
  <a id="e_dnafrag_region2" class="help-header sql_schema_show_hide" onclick="show_hide('dnafrag_region2','query results')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show query results
  </a></div>
        <div style="clear:both"></div>
      </div>
      
  <div id="ex_dnafrag_region2" style="display:none;">
    <table class="ss sql_schema_table_example_result">
      <tr><th>species_name</th><th>dnafrag.name</th><th>dnafrag_start</th><th>dnafrag_end</th><th>dnafrag_strand</th></tr>      <tr><td>oryctolagus_cuniculus</td><td>16</td><td>10692525</td><td>19644506</td><td>1</td></tr>      <tr class="bg2"><td>homo_sapiens</td><td>10</td><td>6409596</td><td>14809562</td><td>1</td></tr>    </table>
  </div></div></div><table class="sql_schema_extra"><tr>    <td class="sql_schema_extra_left">    <p style="font-weight:bold">See also:</p>
  <ul>
      <li><a href="#synteny_region">synteny_region</a></li>
      <li><a href="#dnafrag">dnafrag</a></li>
      </ul>
    </td>
</tr></table></div></div>

<div id="header_5" class="sql_schema_group_header" style="border-color:#FF8500">
  <div class="sql_schema_group_bullet" style="background-color:#FF8500"></div>
  <h2>Genomic alignments</h2>
</div>
<p class="sql_schema_group_header_desc">Whole-genome alignments (as blocks and trees)</p><div align="center"><a href="diagrams/Compara.genomic_alignments.png"><img src="diagrams/Compara.genomic_alignments.png" width="420px" /></a></div><div class="sql_schema_table">
  <div id="genomic_align_block" class="sql_schema_table_header" style="border-top-color:#FF8500">
    <div class="sql_schema_table_header_left"><span style="background-color:#FF8500"></span>genomic_align_block</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_genomic_align_block" class="help-header sql_schema_show_hide" onclick="show_hide('genomic_align_block','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">This table is the key table for the genomic alignments. The software used to align the genomic blocks is refered as an external key to the <a href="#method_link" class="sql_schema_link">method_link</a> table. Nevertheless, actual aligned sequences are defined in the <a href="#genomic_align" class="sql_schema_link">genomic_align</a> table.<br />Tree alignments (EPO alignments) are best accessed through the <a href="#genomic_align_tree" class="sql_schema_link">genomic_align_tree</a> table although the alignments are also indexed in this table. This allows the user to also access the tree alignments as normal multiple alignments.<br />NOTE: All queries in the API uses the primary key as rows are always fetched using the genomic_align_block_id. The key 'method_link_species_set_id' is used by MART when fetching all the genomic_align_blocks corresponding to a given method_link_species_set_id</p>

  <div id="div_genomic_align_block" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>genomic_align_block_id</b></td>
        <td class="bg1">bigint</td>
        <td class="bg1">-</td>
        <td class="bg1">Internal unique ID</td>
        <td class="bg1">primary key</td>
      </tr>

      <tr>
        <td class="bg2"><b>method_link_species_set_id</b></td>
        <td class="bg2">INT</td>
        <td class="bg2">0</td>
        <td class="bg2">External reference to method_link_species_set_id in the <a href="#method_link_species_set" class="sql_schema_link">method_link_species_set</a> table</td>
        <td class="bg2">key: <i>method_link_species_set_id</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>score</b></td>
        <td class="bg1">double</td>
        <td class="bg1">NULL</td>
        <td class="bg1">Score returned by the homology search program</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>perc_id</b></td>
        <td class="bg2">TINYINT</td>
        <td class="bg2">NULL</td>
        <td class="bg2">Used for pairwise comparison. Defines the percentage of identity between both sequences</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>length</b></td>
        <td class="bg1">INT</td>
        <td class="bg1">-</td>
        <td class="bg1">Total length of the alignment</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>group_id</b></td>
        <td class="bg2">bigint</td>
        <td class="bg2">NULL</td>
        <td class="bg2">Used to group alignments</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>level_id</b></td>
        <td class="bg1">TINYINT</td>
        <td class="bg1">0</td>
        <td class="bg1">Level of orthologous layer. 1 corresponds to the principal layer of orthologous sequences found (the largest), 2 and over are additional layers. Use for building the syntenies (based on level_id = 1 only). Note that level_ids are not computed on whole chromosomes but rather on chunks. This means that level_ids can be inconsistent within an alignment-net.</td>
        <td class="bg1"></td>
      </tr>
    </table>
  </div>
<div class="sql_schema_table_examples"><p class="sql_schema_table_example_header">Example:</p><div class="sql_schema_table_example_content"><p>The following query refers to the LastZ alignment between medaka and zebrafish: </p>
      <div>
        <div class="sql_schema_table_example_query">
          <pre><span class="sql_schema_sql_highlight">SELECT</span> genomic_align_block.* <span class="sql_schema_sql_highlight">FROM</span> genomic_align_block <span class="sql_schema_sql_highlight">JOIN</span> method_link_species_set <span class="sql_schema_sql_highlight">USING</span> (method_link_species_set_id) <span class="sql_schema_sql_highlight">JOIN</span> species_set_header <span class="sql_schema_sql_highlight">USING</span> (species_set_id) <span class="sql_schema_sql_highlight">WHERE</span> method_link_id = 16 <span class="sql_schema_sql_highlight">AND</span> species_set_header.name = &quot;Drer-Olat&quot; <span class="sql_schema_sql_highlight">ORDER</span> <span class="sql_schema_sql_highlight">BY</span> genomic_align_block_id <span class="sql_schema_sql_highlight">LIMIT</span> 4;</pre>
        </div>
        <div class="sql_schema_table_example_button">
  <a id="e_genomic_align_block" class="help-header sql_schema_show_hide" onclick="show_hide('genomic_align_block','query results')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show query results
  </a></div>
        <div style="clear:both"></div>
      </div>
      
  <div id="ex_genomic_align_block" style="display:none;">
    <table class="ss sql_schema_table_example_result">
      <tr><th>genomic_align_block_id</th><th>method_link_species_set_id</th><th>score</th><th>perc_id</th><th>length</th><th>group_id</th><th>level_id</th><th>direction</th></tr>      <tr><td>12850000000001</td><td>1285</td><td>8094</td><td>NULL</td><td>185</td><td>12850031204617</td><td>1</td><td>1</td></tr>      <tr class="bg2"><td>12850000000003</td><td>1285</td><td>20570</td><td>NULL</td><td>218</td><td>12850029126909</td><td>1</td><td>1</td></tr>      <tr><td>12850000000018</td><td>1285</td><td>20570</td><td>NULL</td><td>295</td><td>12850029126909</td><td>1</td><td>1</td></tr>      <tr class="bg2"><td>12850000000069</td><td>1285</td><td>10359</td><td>NULL</td><td>196</td><td>12850029126988</td><td>1</td><td>1</td></tr>    </table>
  </div></div></div><table class="sql_schema_extra"><tr>    <td class="sql_schema_extra_left">    <p style="font-weight:bold">See also:</p>
  <ul>
      <li><a href="#method_link_species_set">method_link_species_set</a></li>
      <li><a href="#genomic_align_tree">genomic_align_tree</a></li>
      <li><a href="#method_link">method_link</a></li>
      <li><a href="#genomic_align">genomic_align</a></li>
      </ul>
    </td>
</tr></table></div></div><div class="sql_schema_table">
  <div id="genomic_align_tree" class="sql_schema_table_header" style="border-top-color:#FF8500">
    <div class="sql_schema_table_header_left"><span style="background-color:#FF8500"></span>genomic_align_tree</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_genomic_align_tree" class="help-header sql_schema_show_hide" onclick="show_hide('genomic_align_tree','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">This table is used to index tree alignments, e.g. EPO alignments. These alignments include inferred ancestral sequences. The tree required to index these sequences is stored in this table. This table stores the structure of the tree. Each node links to an entry in the <a href="#genomic_align" class="sql_schema_link">genomic_align</a> table.<br />NOTE: Left_index and right_index are used to speed up fetching trees from the database. Any given node has its left_index larger than the left_index of its parent node and its right index smaller than the right_index of its parent node. In other words, all descendent nodes of a given node can be obtained by fetching all the node with a left_index (or right_index or both) between the left_index and the right_index of that node.</p>

  <div id="div_genomic_align_tree" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>node_id</b></td>
        <td class="bg1">bigint</td>
        <td class="bg1">-</td>
        <td class="bg1">Internal unique ID</td>
        <td class="bg1">primary key</td>
      </tr>

      <tr>
        <td class="bg2"><b>parent_id</b></td>
        <td class="bg2">bigint</td>
        <td class="bg2">NULL</td>
        <td class="bg2">Link to the parent node</td>
        <td class="bg2">key: <i>parent_id</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>root_id</b></td>
        <td class="bg1">bigint</td>
        <td class="bg1">0</td>
        <td class="bg1">Link to root node</td>
        <td class="bg1">key: <i>left_index</i></td>
      </tr>

      <tr>
        <td class="bg2"><b>left_index</b></td>
        <td class="bg2">INT</td>
        <td class="bg2">0</td>
        <td class="bg2">Internal index. See above</td>
        <td class="bg2">key: <i>left_index</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>right_index</b></td>
        <td class="bg1">INT</td>
        <td class="bg1">0</td>
        <td class="bg1">Internal index. See above</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>left_node_id</b></td>
        <td class="bg2">bigint</td>
        <td class="bg2">NULL</td>
        <td class="bg2">Link to the node on the left side of this node</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>right_node_id</b></td>
        <td class="bg1">bigint</td>
        <td class="bg1">NULL</td>
        <td class="bg1">Link to the node on the right side of this node</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>distance_to_parent</b></td>
        <td class="bg2">double</td>
        <td class="bg2">1</td>
        <td class="bg2">Phylogenetic distance between this node and its parent</td>
        <td class="bg2"></td>
      </tr>
    </table>
  </div>
<div class="sql_schema_table_examples"><p class="sql_schema_table_example_header">Example 1:</p><div class="sql_schema_table_example_content"><p>The following query corresponds to the root of a tree, because parent_id = 0 and root_id = node_id </p>
      <div>
        <div class="sql_schema_table_example_query">
          <pre><span class="sql_schema_sql_highlight">SELECT</span> * <span class="sql_schema_sql_highlight">FROM</span> genomic_align_tree <span class="sql_schema_sql_highlight">WHERE</span> node_id = root_id <span class="sql_schema_sql_highlight">LIMIT</span> 1;</pre>
        </div>
        <div class="sql_schema_table_example_button">
  <a id="e_genomic_align_tree1" class="help-header sql_schema_show_hide" onclick="show_hide('genomic_align_tree1','query results')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show query results
  </a></div>
        <div style="clear:both"></div>
      </div>
      
  <div id="ex_genomic_align_tree1" style="display:none;">
    <table class="ss sql_schema_table_example_result">
      <tr><th>node_id</th><th>parent_id</th><th>root_id</th><th>left_index</th><th>right_index</th><th>left_node_id</th><th>right_node_id</th><th>distance_to_parent</th></tr>      <tr><td>19560000000001</td><td>NULL</td><td>19560000000001</td><td>1</td><td>6</td><td>NULL</td><td>NULL</td><td>0.05368889</td></tr>    </table>
  </div></div></div><div class="sql_schema_table_examples"><p class="sql_schema_table_example_header">Example 2:</p><div class="sql_schema_table_example_content"><p>Grab the first two trees that have exactly 5 nodes </p>
      <div>
        <div class="sql_schema_table_example_query">
          <pre><span class="sql_schema_sql_highlight">SELECT</span> * <span class="sql_schema_sql_highlight">FROM</span> genomic_align_tree <span class="sql_schema_sql_highlight">WHERE</span> root_id <span class="sql_schema_sql_highlight">IN</span> (<span class="sql_schema_sql_highlight">SELECT</span> root_id <span class="sql_schema_sql_highlight">FROM</span> genomic_align_tree <span class="sql_schema_sql_highlight">GROUP</span> <span class="sql_schema_sql_highlight">BY</span> root_id HAVING COUNT(*) = 5) <span class="sql_schema_sql_highlight">LIMIT</span> 10;</pre>
        </div>
        <div class="sql_schema_table_example_button">
  <a id="e_genomic_align_tree2" class="help-header sql_schema_show_hide" onclick="show_hide('genomic_align_tree2','query results')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show query results
  </a></div>
        <div style="clear:both"></div>
      </div>
      
  <div id="ex_genomic_align_tree2" style="display:none;">
    <table class="ss sql_schema_table_example_result">
      <tr><th>node_id</th><th>parent_id</th><th>root_id</th><th>left_index</th><th>right_index</th><th>left_node_id</th><th>right_node_id</th><th>distance_to_parent</th></tr>      <tr><td>19560000000528</td><td>NULL</td><td>19560000000528</td><td>1</td><td>10</td><td>NULL</td><td>NULL</td><td>0</td></tr>      <tr class="bg2"><td>19560000000530</td><td>19560000000528</td><td>19560000000528</td><td>8</td><td>9</td><td>19560000185888</td><td>19560000154443</td><td>0.0855434</td></tr>      <tr><td>19560000000535</td><td>19560000000528</td><td>19560000000528</td><td>2</td><td>7</td><td>19560000185890</td><td>19560000154439</td><td>0.06279142</td></tr>      <tr class="bg2"><td>19560000000537</td><td>19560000000535</td><td>19560000000528</td><td>5</td><td>6</td><td>19560000185891</td><td>19560000154440</td><td>0.0278214</td></tr>      <tr><td>19560000000540</td><td>19560000000535</td><td>19560000000528</td><td>3</td><td>4</td><td>19560000185892</td><td>19560000154442</td><td>0.02754917</td></tr>      <tr class="bg2"><td>19560000000755</td><td>NULL</td><td>19560000000755</td><td>1</td><td>10</td><td>NULL</td><td>NULL</td><td>0</td></tr>      <tr><td>19560000000756</td><td>19560000000755</td><td>19560000000755</td><td>2</td><td>7</td><td>19560000258883</td><td>19560000002315</td><td>0.06279142</td></tr>      <tr class="bg2"><td>19560000000757</td><td>19560000000756</td><td>19560000000755</td><td>5</td><td>6</td><td>19560000258884</td><td>19560000002316</td><td>0.0278214</td></tr>      <tr><td>19560000000758</td><td>19560000000756</td><td>19560000000755</td><td>3</td><td>4</td><td>19560000258885</td><td>19560000002317</td><td>0.02754917</td></tr>      <tr class="bg2"><td>19560000000759</td><td>19560000000755</td><td>19560000000755</td><td>8</td><td>9</td><td>19560000258881</td><td>19560000002313</td><td>0.0855434</td></tr>    </table>
  </div></div></div><table class="sql_schema_extra"><tr>    <td class="sql_schema_extra_left">    <p style="font-weight:bold">See also:</p>
  <ul>
      <li><a href="#genomic_align">genomic_align</a></li>
      </ul>
    </td>
</tr></table></div></div><div class="sql_schema_table">
  <div id="genomic_align" class="sql_schema_table_header" style="border-top-color:#FF8500">
    <div class="sql_schema_table_header_left"><span style="background-color:#FF8500"></span>genomic_align</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_genomic_align" class="help-header sql_schema_show_hide" onclick="show_hide('genomic_align','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">This table contains the coordinates and all the information needed to rebuild genomic alignments. Every entry corresponds to one of the aligned sequences. It also contains an external key to the <a href="#method_link_species_set" class="sql_schema_link">method_link_species_set</a> which refers to the software and set of species used for getting the corresponding alignment. The aligned sequence is defined by an external reference to the <a href="#dnafrag" class="sql_schema_link">dnafrag</a> table, the starting and ending position within this dnafrag, the strand and a cigar_line.<br /> The original aligned sequence is not stored but it can be retrieved using the <b>cigar_line</b> field and the original sequence. The cigar line defines the sequence of matches/mismatches and deletions (or gaps). For example, this cigar line <b>2MD3M2D2M</b> will mean that the alignment contains 2 matches/mismatches, 1 deletion (number 1 is omitted in order to save some space), 3 matches/mismatches, 2 deletions and 2 matches/mismatches. If the original sequence is <code>AACGCTT</code>, the aligned sequence will be:<br /> <table> <caption>cigar line: 2MD3M2D2M</caption> <thead><tr> <th>M</th> <th>M</th> <th>D</th> <th>M</th> <th>M</th> <th>M</th> <th>D</th> <th>D</th> <th>M</th> <th>M</th> </tr> </thead> <tbody> <tr> <td>A</td> <td>A</td> <td>-</td> <td>C</td> <td>G</td> <td>C</td> <td>-</td> <td>-</td> <td>T</td> <td>T</td> </tr> </tbody> </table></p>

  <div id="div_genomic_align" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>genomic_align_id</b></td>
        <td class="bg1">bigint</td>
        <td class="bg1">-</td>
        <td class="bg1">Unique internal ID</td>
        <td class="bg1">primary key</td>
      </tr>

      <tr>
        <td class="bg2"><b>genomic_align_block_id</b></td>
        <td class="bg2">bigint</td>
        <td class="bg2">-</td>
        <td class="bg2">External reference to genomic_align_block_id in the <a href="#genomic_align_block" class="sql_schema_link">genomic_align_block</a> table</td>
        <td class="bg2">key: <i>genomic_align_block_id</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>method_link_species_set_id</b></td>
        <td class="bg1">INT</td>
        <td class="bg1">0</td>
        <td class="bg1">External reference to method_link_species_set_id in the <a href="#method_link_species_set" class="sql_schema_link">method_link_species_set</a> table. This information is redundant because it also appears in the <a href="#genomic_align_block" class="sql_schema_link">genomic_align_block</a> table but it is used to speed up the queries</td>
        <td class="bg1">key: <i>method_link_species_set_id</i><br />key: <i>dnafrag</i></td>
      </tr>

      <tr>
        <td class="bg2"><b>dnafrag_id</b></td>
        <td class="bg2">bigint</td>
        <td class="bg2">0</td>
        <td class="bg2">External reference to dnafrag_id in the <a href="#dnafrag" class="sql_schema_link">dnafrag</a> table</td>
        <td class="bg2">key: <i>dnafrag</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>dnafrag_start</b></td>
        <td class="bg1">INT</td>
        <td class="bg1">0</td>
        <td class="bg1">Starting position within the dnafrag defined by dnafrag_id</td>
        <td class="bg1">key: <i>dnafrag</i></td>
      </tr>

      <tr>
        <td class="bg2"><b>dnafrag_end</b></td>
        <td class="bg2">INT</td>
        <td class="bg2">0</td>
        <td class="bg2">Ending position within the dnafrag defined by dnafrag_id</td>
        <td class="bg2">key: <i>dnafrag</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>dnafrag_strand</b></td>
        <td class="bg1">TINYINT</td>
        <td class="bg1">0</td>
        <td class="bg1">Strand in the dnafrag defined by dnafrag_id</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>cigar_line</b></td>
        <td class="bg2">mediumtext</td>
        <td class="bg2">-</td>
        <td class="bg2">Internal description of the aligned sequence</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>visible</b></td>
        <td class="bg1">TINYINT</td>
        <td class="bg1">1</td>
        <td class="bg1">Used in self alignments to ensure only one Bio::EnsEMBL::Compara::GenomicAlignBlock is visible when you have more than 1 block covering the same region</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>node_id</b></td>
        <td class="bg2">bigint</td>
        <td class="bg2">NULL</td>
        <td class="bg2">External reference to node_id in the <a href="#genomic_align_tree" class="sql_schema_link">genomic_align_tree</a> table</td>
        <td class="bg2">key: <i>node_id</i></td>
      </tr>
    </table>
  </div>
<div class="sql_schema_table_examples"><p class="sql_schema_table_example_header">Example 1:</p><div class="sql_schema_table_example_content"><p>The following query corresponds to the 4x2 sequences included in the alignment described above (see <a href="#genomic_align_block" class="sql_schema_link">genomic_align_block</a> table description). </p>
      <div>
        <div class="sql_schema_table_example_query">
          <pre><span class="sql_schema_sql_highlight">SELECT</span> genomic_align.* <span class="sql_schema_sql_highlight">FROM</span> genomic_align_block <span class="sql_schema_sql_highlight">JOIN</span> method_link_species_set <span class="sql_schema_sql_highlight">USING</span> (method_link_species_set_id) <span class="sql_schema_sql_highlight">JOIN</span> species_set_header <span class="sql_schema_sql_highlight">USING</span> (species_set_id) <span class="sql_schema_sql_highlight">JOIN</span> genomic_align <span class="sql_schema_sql_highlight">USING</span> (genomic_align_block_id) <span class="sql_schema_sql_highlight">WHERE</span> method_link_id = 16 <span class="sql_schema_sql_highlight">AND</span> species_set_header.name = &quot;Drer-Olat&quot; <span class="sql_schema_sql_highlight">ORDER</span> <span class="sql_schema_sql_highlight">BY</span> genomic_align_block_id <span class="sql_schema_sql_highlight">LIMIT</span> 8;</pre>
        </div>
        <div class="sql_schema_table_example_button">
  <a id="e_genomic_align1" class="help-header sql_schema_show_hide" onclick="show_hide('genomic_align1','query results')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show query results
  </a></div>
        <div style="clear:both"></div>
      </div>
      
  <div id="ex_genomic_align1" style="display:none;">
    <table class="ss sql_schema_table_example_result">
      <tr><th>genomic_align_id</th><th>genomic_align_block_id</th><th>method_link_species_set_id</th><th>dnafrag_id</th><th>dnafrag_start</th><th>dnafrag_end</th><th>dnafrag_strand</th><th>cigar_line</th><th>visible</th><th>node_id</th></tr>      <tr><td>12850000000028</td><td>12850000000001</td><td>1285</td><td>16642790</td><td>5660</td><td>5837</td><td>1</td><td>129M7D49M</td><td>1</td><td>NULL</td></tr>      <tr class="bg2"><td>12850000000035</td><td>12850000000001</td><td>1285</td><td>17458342</td><td>11305494</td><td>11305675</td><td>-1</td><td>146M2D28MD8M</td><td>1</td><td>NULL</td></tr>      <tr><td>12850000000001</td><td>12850000000003</td><td>1285</td><td>16642780</td><td>885</td><td>1102</td><td>1</td><td>218M</td><td>1</td><td>NULL</td></tr>      <tr class="bg2"><td>12850000000013</td><td>12850000000003</td><td>1285</td><td>17458332</td><td>25290534</td><td>25290751</td><td>1</td><td>218M</td><td>1</td><td>NULL</td></tr>      <tr><td>12850000000061</td><td>12850000000018</td><td>1285</td><td>16642780</td><td>1183</td><td>1432</td><td>1</td><td>13MD11M41D156M3D70M</td><td>1</td><td>NULL</td></tr>      <tr class="bg2"><td>12850000000099</td><td>12850000000018</td><td>1285</td><td>17458332</td><td>25291008</td><td>25291302</td><td>1</td><td>295M</td><td>1</td><td>NULL</td></tr>      <tr><td>12850000000177</td><td>12850000000069</td><td>1285</td><td>16642780</td><td>11774</td><td>11969</td><td>1</td><td>196M</td><td>1</td><td>NULL</td></tr>      <tr class="bg2"><td>12850000000188</td><td>12850000000069</td><td>1285</td><td>17458332</td><td>25296080</td><td>25296270</td><td>1</td><td>168M3D7M2D16M</td><td>1</td><td>NULL</td></tr>    </table>
  </div></div></div><div class="sql_schema_table_examples"><p class="sql_schema_table_example_header">Example 2:</p><div class="sql_schema_table_example_content"><p>Here is a better way to get this by joining the <a href="#dnafrag" class="sql_schema_link">dnafrag</a> and <a href="#genome_db" class="sql_schema_link">genome_db</a> tables: </p>
      <div>
        <div class="sql_schema_table_example_query">
          <pre><span class="sql_schema_sql_highlight">SELECT</span> genome_db.name, dnafrag.name, dnafrag_start, dnafrag_end, dnafrag_str<span class="sql_schema_sql_highlight">AND</span> str, cigar_line <span class="sql_schema_sql_highlight">FROM</span> genomic_align_block <span class="sql_schema_sql_highlight">JOIN</span> method_link_species_set <span class="sql_schema_sql_highlight">USING</span> (method_link_species_set_id) <span class="sql_schema_sql_highlight">JOIN</span> species_set_header <span class="sql_schema_sql_highlight">USING</span> (species_set_id) <span class="sql_schema_sql_highlight">JOIN</span> genomic_align <span class="sql_schema_sql_highlight">USING</span> (genomic_align_block_id) <span class="sql_schema_sql_highlight">JOIN</span> dnafrag <span class="sql_schema_sql_highlight">USING</span> (dnafrag_id) <span class="sql_schema_sql_highlight">JOIN</span> genome_db <span class="sql_schema_sql_highlight">USING</span> (genome_db_id) <span class="sql_schema_sql_highlight">WHERE</span> method_link_id = 16 <span class="sql_schema_sql_highlight">AND</span> species_set_header.name = &quot;Drer-Olat&quot; <span class="sql_schema_sql_highlight">ORDER</span> <span class="sql_schema_sql_highlight">BY</span> genomic_align_block_id <span class="sql_schema_sql_highlight">LIMIT</span> 8;</pre>
        </div>
        <div class="sql_schema_table_example_button">
  <a id="e_genomic_align2" class="help-header sql_schema_show_hide" onclick="show_hide('genomic_align2','query results')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show query results
  </a></div>
        <div style="clear:both"></div>
      </div>
      
  <div id="ex_genomic_align2" style="display:none;">
    <table class="ss sql_schema_table_example_result">
      <tr><th>genome_db.name</th><th>dnafrag.name</th><th>dnafrag_start</th><th>dnafrag_end</th><th>dnafrag_strandstr</th><th>cigar_line</th></tr>      <tr><td>danio_rerio</td><td>KN150510.1</td><td>5660</td><td>5837</td><td>1</td><td>129M7D49M</td></tr>      <tr class="bg2"><td>oryzias_latipes</td><td>24</td><td>11305494</td><td>11305675</td><td>-1</td><td>146M2D28MD8M</td></tr>      <tr><td>danio_rerio</td><td>KN150174.1</td><td>885</td><td>1102</td><td>1</td><td>218M</td></tr>      <tr class="bg2"><td>oryzias_latipes</td><td>14</td><td>25290534</td><td>25290751</td><td>1</td><td>218M</td></tr>      <tr><td>danio_rerio</td><td>KN150174.1</td><td>1183</td><td>1432</td><td>1</td><td>13MD11M41D156M3D70M</td></tr>      <tr class="bg2"><td>oryzias_latipes</td><td>14</td><td>25291008</td><td>25291302</td><td>1</td><td>295M</td></tr>      <tr><td>danio_rerio</td><td>KN150174.1</td><td>11774</td><td>11969</td><td>1</td><td>196M</td></tr>      <tr class="bg2"><td>oryzias_latipes</td><td>14</td><td>25296080</td><td>25296270</td><td>1</td><td>168M3D7M2D16M</td></tr>    </table>
  </div></div></div><table class="sql_schema_extra"><tr>    <td class="sql_schema_extra_left">    <p style="font-weight:bold">See also:</p>
  <ul>
      <li><a href="#genomic_align_block">genomic_align_block</a></li>
      <li><a href="#method_link_species_set">method_link_species_set</a></li>
      <li><a href="#dnafrag">dnafrag</a></li>
      <li><a href="#genomic_align_tree">genomic_align_tree</a></li>
      <li><a href="#genome_db">genome_db</a></li>
      </ul>
    </td>
</tr></table></div></div>

<div id="header_6" class="sql_schema_group_header" style="border-color:#C70C09">
  <div class="sql_schema_group_bullet" style="background-color:#C70C09"></div>
  <h2>Conservation</h2>
</div>
<p class="sql_schema_group_header_desc">Evolutionary conservation (scores and regions)</p><div align="center"><a href="diagrams/Compara.conservation.png"><img src="diagrams/Compara.conservation.png" width="500px" /></a></div><div class="sql_schema_table">
  <div id="conservation_score" class="sql_schema_table_header" style="border-top-color:#C70C09">
    <div class="sql_schema_table_header_left"><span style="background-color:#C70C09"></span>conservation_score</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_conservation_score" class="help-header sql_schema_show_hide" onclick="show_hide('conservation_score','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">This table contains conservation scores calculated from the whole-genome multiple alignments stored in the <a href="#genomic_align_block" class="sql_schema_link">genomic_align_block</a> table. Several scores are stored per row. expected_score and diff_score are binary columns and you need to use the Perl API to access these data.</p>

  <div id="div_conservation_score" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>genomic_align_block_id</b></td>
        <td class="bg1">bigint</td>
        <td class="bg1">-</td>
        <td class="bg1">External reference to genomic_align_block_id in the <a href="#genomic_align_block" class="sql_schema_link">genomic_align_block</a> table</td>
        <td class="bg1">key: <i>genomic_align_block_id, window_size</i></td>
      </tr>

      <tr>
        <td class="bg2"><b>window_size</b></td>
        <td class="bg2">smallint</td>
        <td class="bg2">-</td>
        <td class="bg2">The scores are stored at different resolution levels. This column defines the window size used to calculate the average score</td>
        <td class="bg2">key: <i>genomic_align_block_id, window_size</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>position</b></td>
        <td class="bg1">int</td>
        <td class="bg1">-</td>
        <td class="bg1">Position of the first score (in alignment coordinates)</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>expected_score</b></td>
        <td class="bg2">blob</td>
        <td class="bg2">NULL</td>
        <td class="bg2">Expected score. The observed score can be determined using the diff_score and the expected_score</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>diff_score</b></td>
        <td class="bg1">blob</td>
        <td class="bg1">NULL</td>
        <td class="bg1">The difference between the expected and observed variation, i.e. the conservation score</td>
        <td class="bg1"></td>
      </tr>
    </table>
  </div>
<table class="sql_schema_extra"><tr>    <td class="sql_schema_extra_left">    <p style="font-weight:bold">See also:</p>
  <ul>
      <li><a href="#genomic_align_block">genomic_align_block</a></li>
      </ul>
    </td>
</tr></table></div></div><div class="sql_schema_table">
  <div id="constrained_element" class="sql_schema_table_header" style="border-top-color:#C70C09">
    <div class="sql_schema_table_header_left"><span style="background-color:#C70C09"></span>constrained_element</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_constrained_element" class="help-header sql_schema_show_hide" onclick="show_hide('constrained_element','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">This table contains constrained elements calculated from the whole-genome multiple alignments stored in the <a href="#genomic_align_block" class="sql_schema_link">genomic_align_block</a> table</p>

  <div id="div_constrained_element" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>constrained_element_id</b></td>
        <td class="bg1">bigint</td>
        <td class="bg1">-</td>
        <td class="bg1">Internal (but unique) ID</td>
        <td class="bg1">key: <i>constrained_element_id_idx</i></td>
      </tr>

      <tr>
        <td class="bg2"><b>dnafrag_id</b></td>
        <td class="bg2">bigint</td>
        <td class="bg2">-</td>
        <td class="bg2">External reference to dnafrag_id in the <a href="#dnafrag" class="sql_schema_link">dnafrag</a> table</td>
        <td class="bg2">key: <i>mlssid_dfid_dfstart_dfend_idx</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>dnafrag_start</b></td>
        <td class="bg1">INT</td>
        <td class="bg1">-</td>
        <td class="bg1">Start of the constrained element</td>
        <td class="bg1">key: <i>mlssid_dfid_dfstart_dfend_idx</i></td>
      </tr>

      <tr>
        <td class="bg2"><b>dnafrag_end</b></td>
        <td class="bg2">INT</td>
        <td class="bg2">-</td>
        <td class="bg2">End of the constrained element</td>
        <td class="bg2">key: <i>mlssid_dfid_dfstart_dfend_idx</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>dnafrag_strand</b></td>
        <td class="bg1">TINYINT</td>
        <td class="bg1">-</td>
        <td class="bg1">Strand of the constrained element</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>method_link_species_set_id</b></td>
        <td class="bg2">INT</td>
        <td class="bg2">-</td>
        <td class="bg2">External reference to method_link_species_set_id in the <a href="#method_link_species_set" class="sql_schema_link">method_link_species_set</a> table</td>
        <td class="bg2">key: <i>mlssid_dfid_dfstart_dfend_idx</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>p_value</b></td>
        <td class="bg1">double</td>
        <td class="bg1">0</td>
        <td class="bg1">p-value derived from Gerp</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>score</b></td>
        <td class="bg2">double</td>
        <td class="bg2">0</td>
        <td class="bg2">Score derived from Gerp</td>
        <td class="bg2"></td>
      </tr>
    </table>
  </div>
<div class="sql_schema_table_examples"><p class="sql_schema_table_example_header">Example 1:</p><div class="sql_schema_table_example_content"><p>Example entry for a constrained_element: </p>
      <div>
        <div class="sql_schema_table_example_query">
          <pre><span class="sql_schema_sql_highlight">SELECT</span> * <span class="sql_schema_sql_highlight">FROM</span> constrained_element <span class="sql_schema_sql_highlight">ORDER</span> <span class="sql_schema_sql_highlight">BY</span> constrained_element_id <span class="sql_schema_sql_highlight">LIMIT</span> 1;</pre>
        </div>
        <div class="sql_schema_table_example_button">
  <a id="e_constrained_element1" class="help-header sql_schema_show_hide" onclick="show_hide('constrained_element1','query results')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show query results
  </a></div>
        <div style="clear:both"></div>
      </div>
      
  <div id="ex_constrained_element1" style="display:none;">
    <table class="ss sql_schema_table_example_result">
      <tr><th>constrained_element_id</th><th>dnafrag_id</th><th>dnafrag_start</th><th>dnafrag_end</th><th>dnafrag_strand</th><th>method_link_species_set_id</th><th>p_value</th><th>score</th></tr>      <tr><td>19570000000001</td><td>15500984</td><td>122009022</td><td>122009276</td><td>1</td><td>1957</td><td>7.93566e-15</td><td>46.2</td></tr>    </table>
  </div></div></div><div class="sql_schema_table_examples"><p class="sql_schema_table_example_header">Example 2:</p><div class="sql_schema_table_example_content"><p>There are 2 other elements in the same constrained_element: </p>
      <div>
        <div class="sql_schema_table_example_query">
          <pre><span class="sql_schema_sql_highlight">SELECT</span> * <span class="sql_schema_sql_highlight">FROM</span> constrained_element <span class="sql_schema_sql_highlight">JOIN</span> (<span class="sql_schema_sql_highlight">SELECT</span> MIN(constrained_element_id) <span class="sql_schema_sql_highlight">AS</span> constrained_element_id <span class="sql_schema_sql_highlight">FROM</span> constrained_element) t <span class="sql_schema_sql_highlight">USING</span> (constrained_element_id);</pre>
        </div>
        <div class="sql_schema_table_example_button">
  <a id="e_constrained_element2" class="help-header sql_schema_show_hide" onclick="show_hide('constrained_element2','query results')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show query results
  </a></div>
        <div style="clear:both"></div>
      </div>
      
  <div id="ex_constrained_element2" style="display:none;">
    <table class="ss sql_schema_table_example_result">
      <tr><th>constrained_element_id</th><th>dnafrag_id</th><th>dnafrag_start</th><th>dnafrag_end</th><th>dnafrag_strand</th><th>method_link_species_set_id</th><th>p_value</th><th>score</th></tr>      <tr><td>19570000000001</td><td>15500984</td><td>122009022</td><td>122009276</td><td>1</td><td>1957</td><td>7.93566e-15</td><td>46.2</td></tr>      <tr class="bg2"><td>19570000000001</td><td>17543211</td><td>157596443</td><td>157596698</td><td>-1</td><td>1957</td><td>7.93566e-15</td><td>46.2</td></tr>      <tr><td>19570000000001</td><td>18580243</td><td>80283540</td><td>80283795</td><td>1</td><td>1957</td><td>7.93566e-15</td><td>46.2</td></tr>      <tr class="bg2"><td>19570000000001</td><td>23887356</td><td>120603918</td><td>120604172</td><td>1</td><td>1957</td><td>7.93566e-15</td><td>46.2</td></tr>      <tr><td>19570000000001</td><td>24015457</td><td>1827044</td><td>1827298</td><td>1</td><td>1957</td><td>7.93566e-15</td><td>46.2</td></tr>      <tr class="bg2"><td>19570000000001</td><td>24322762</td><td>934861</td><td>935115</td><td>-1</td><td>1957</td><td>7.93566e-15</td><td>46.2</td></tr>      <tr><td>19570000000001</td><td>24489353</td><td>1286476</td><td>1286730</td><td>-1</td><td>1957</td><td>7.93566e-15</td><td>46.2</td></tr>      <tr class="bg2"><td>19570000000001</td><td>24494544</td><td>474235</td><td>474489</td><td>-1</td><td>1957</td><td>7.93566e-15</td><td>46.2</td></tr>      <tr><td>19570000000001</td><td>24614392</td><td>30740</td><td>30994</td><td>1</td><td>1957</td><td>7.93566e-15</td><td>46.2</td></tr>      <tr class="bg2"><td>19570000000001</td><td>24763386</td><td>290195</td><td>290449</td><td>-1</td><td>1957</td><td>7.93566e-15</td><td>46.2</td></tr>      <tr><td>19570000000001</td><td>24804900</td><td>1283383</td><td>1283637</td><td>-1</td><td>1957</td><td>7.93566e-15</td><td>46.2</td></tr>      <tr class="bg2"><td>19570000000001</td><td>24897782</td><td>1282590</td><td>1282844</td><td>-1</td><td>1957</td><td>7.93566e-15</td><td>46.2</td></tr>      <tr><td>19570000000001</td><td>24979990</td><td>1885228</td><td>1885482</td><td>-1</td><td>1957</td><td>7.93566e-15</td><td>46.2</td></tr>      <tr class="bg2"><td>19570000000001</td><td>24980520</td><td>1509265</td><td>1509519</td><td>-1</td><td>1957</td><td>7.93566e-15</td><td>46.2</td></tr>      <tr><td>19570000000001</td><td>24985496</td><td>8663421</td><td>8663675</td><td>1</td><td>1957</td><td>7.93566e-15</td><td>46.2</td></tr>      <tr class="bg2"><td>19570000000001</td><td>30042676</td><td>24983966</td><td>24984220</td><td>-1</td><td>1957</td><td>7.93566e-15</td><td>46.2</td></tr>    </table>
  </div></div></div><table class="sql_schema_extra"><tr>    <td class="sql_schema_extra_left">    <p style="font-weight:bold">See also:</p>
  <ul>
      <li><a href="#dnafrag">dnafrag</a></li>
      <li><a href="#genomic_align_block">genomic_align_block</a></li>
      <li><a href="#method_link_species_set">method_link_species_set</a></li>
      </ul>
    </td>
</tr></table></div></div>

<div id="header_7" class="sql_schema_group_header" style="border-color:#1E90FF">
  <div class="sql_schema_group_bullet" style="background-color:#1E90FF"></div>
  <h2>Gene trees and homologies</h2>
</div>
<p class="sql_schema_group_header_desc">These tables store information about gene alignments, trees and homologies</p><div align="center"><a href="diagrams/Compara.gene_trees_and_homologies.png"><img src="diagrams/Compara.gene_trees_and_homologies.png" width="300px" /></a></div><div class="sql_schema_table">
  <div id="peptide_align_feature" class="sql_schema_table_header" style="border-top-color:#1E90FF">
    <div class="sql_schema_table_header_left"><span style="background-color:#1E90FF"></span>peptide_align_feature</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_peptide_align_feature" class="help-header sql_schema_show_hide" onclick="show_hide('peptide_align_feature','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">: This table stores the raw local alignment results of peptide to peptide alignments returned by a BLAST run. The hits are actually stored in species-specific tables rather than in a single table. For example, human has the genome_db_id 150, and all the hits that have a human gene as a query are stored in peptide_align_feature</p>

  <div id="div_peptide_align_feature" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>peptide_align_feature_id</b></td>
        <td class="bg1">bigint</td>
        <td class="bg1">-</td>
        <td class="bg1">Internal unique ID</td>
        <td class="bg1">primary key</td>
      </tr>

      <tr>
        <td class="bg2"><b>qmember_id</b></td>
        <td class="bg2">INT</td>
        <td class="bg2">-</td>
        <td class="bg2">External reference to seq_member_id in the <a href="#seq_member" class="sql_schema_link">seq_member</a> table for the query peptide</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>hmember_id</b></td>
        <td class="bg1">INT</td>
        <td class="bg1">-</td>
        <td class="bg1">External reference to seq_member_id in the <a href="#seq_member" class="sql_schema_link">seq_member</a> table for the hit peptide</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>qgenome_db_id</b></td>
        <td class="bg2">INT</td>
        <td class="bg2">NULL</td>
        <td class="bg2">External reference to genome_db_id in the <a href="#genome_db" class="sql_schema_link">genome_db</a> table for the query peptide (for query optimization)</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>hgenome_db_id</b></td>
        <td class="bg1">INT</td>
        <td class="bg1">NULL</td>
        <td class="bg1">External reference to genome_db_id in the <a href="#genome_db" class="sql_schema_link">genome_db</a> table for the hit peptide (for query optimization)</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>qstart</b></td>
        <td class="bg2">INT</td>
        <td class="bg2">0</td>
        <td class="bg2">Starting position in the query peptide sequence</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>qend</b></td>
        <td class="bg1">INT</td>
        <td class="bg1">0</td>
        <td class="bg1">Ending position in the query peptide sequence</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>hstart</b></td>
        <td class="bg2">int</td>
        <td class="bg2">0</td>
        <td class="bg2">Starting position in the hit peptide sequence</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>hend</b></td>
        <td class="bg1">int</td>
        <td class="bg1">0</td>
        <td class="bg1">Ending position in the hit peptide sequence</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>score</b></td>
        <td class="bg2">double(16,4)</td>
        <td class="bg2">0.0000</td>
        <td class="bg2">Blast score for this HSP</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>evalue</b></td>
        <td class="bg1">double</td>
        <td class="bg1">-</td>
        <td class="bg1">Blast evalue for this HSP</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>align_length</b></td>
        <td class="bg2">INT</td>
        <td class="bg2">-</td>
        <td class="bg2">Alignment length of HSP</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>identical_matches</b></td>
        <td class="bg1">INT</td>
        <td class="bg1">-</td>
        <td class="bg1">Blast HSP match score</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>perc_ident</b></td>
        <td class="bg2">TINYINT</td>
        <td class="bg2">-</td>
        <td class="bg2">Percent identical matches in the HSP length</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>positive_matches</b></td>
        <td class="bg1">INT</td>
        <td class="bg1">-</td>
        <td class="bg1">Blast HSP positive score</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>perc_pos</b></td>
        <td class="bg2">TINYINT</td>
        <td class="bg2">-</td>
        <td class="bg2">Percent positive matches in the HSP length</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>hit_rank</b></td>
        <td class="bg1">SMALLINT</td>
        <td class="bg1">-</td>
        <td class="bg1">Rank in blast result</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>cigar_line</b></td>
        <td class="bg2">mediumtext</td>
        <td class="bg2">NULL</td>
        <td class="bg2">Cigar string coding the actual alignment</td>
        <td class="bg2"></td>
      </tr>
    </table>
  </div>
<div class="sql_schema_table_examples"><p class="sql_schema_table_example_header">Example 1:</p><div class="sql_schema_table_example_content"><p>Example of peptide_align_feature entry: sql                              SELECT * FROM peptide_align_feature WHERE hgenome_db_id = 111 LIMIT 1;</p></div></div><div class="sql_schema_table_examples"><p class="sql_schema_table_example_header">Example 2:</p><div class="sql_schema_table_example_content"><p>The following query corresponds to a particular hit found between a Homo sapiens protein and a Anolis carolinensis protein: sql                              SELECT g1.name as qgenome, m1.stable_id as qstable_id, g2.name as hgenome, m2.stable_id as hstable_id, score, evalue FROM peptide_align_feature JOIN seq_member m1 ON (qmember_id = m1.seq_member_id) JOIN seq_member m2 ON (hmember_id = m2.seq_member_id) JOIN genome_db g1 ON (qgenome_db_id = g1.genome_db_id) JOIN genome_db g2 ON (hgenome_db_id = g2.genome_db_id) WHERE hgenome_db_id = 111 LIMIT 1;</p></div></div><table class="sql_schema_extra"><tr>    <td class="sql_schema_extra_left">    <p style="font-weight:bold">See also:</p>
  <ul>
      <li><a href="#seq_member">seq_member</a></li>
      <li><a href="#genome_db">genome_db</a></li>
      </ul>
    </td>
</tr></table></div></div><div class="sql_schema_table">
  <div id="gene_align" class="sql_schema_table_header" style="border-top-color:#1E90FF">
    <div class="sql_schema_table_header_left"><span style="background-color:#1E90FF"></span>gene_align</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_gene_align" class="help-header sql_schema_show_hide" onclick="show_hide('gene_align','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">This table stores information about alignments for members</p>

  <div id="div_gene_align" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>gene_align_id</b></td>
        <td class="bg1">INT</td>
        <td class="bg1">-</td>
        <td class="bg1">Internal unique ID</td>
        <td class="bg1">primary key</td>
      </tr>

      <tr>
        <td class="bg2"><b>seq_type</b></td>
        <td class="bg2">varchar(40)</td>
        <td class="bg2">NULL</td>
        <td class="bg2">Short description of this alignment</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>aln_method</b></td>
        <td class="bg1">varchar(40)</td>
        <td class="bg1">''</td>
        <td class="bg1">The alignment method used</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>aln_length</b></td>
        <td class="bg2">INT</td>
        <td class="bg2">0</td>
        <td class="bg2">The total length of the alignment</td>
        <td class="bg2"></td>
      </tr>
    </table>
  </div>
<table class="sql_schema_extra"><tr>    <td class="sql_schema_extra_left">    <p style="font-weight:bold">See also:</p>
  <ul>
      <li><a href="#seq_member">seq_member</a></li>
      <li><a href="#gene_align_member">gene_align_member</a></li>
      </ul>
    </td>
</tr></table></div></div><div class="sql_schema_table">
  <div id="gene_align_member" class="sql_schema_table_header" style="border-top-color:#1E90FF">
    <div class="sql_schema_table_header_left"><span style="background-color:#1E90FF"></span>gene_align_member</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_gene_align_member" class="help-header sql_schema_show_hide" onclick="show_hide('gene_align_member','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">This table allows certain nodes (leaves) to have aligned protein member_scores attached to them</p>

  <div id="div_gene_align_member" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>gene_align_id</b></td>
        <td class="bg1">INT</td>
        <td class="bg1">-</td>
        <td class="bg1">External reference to gene_align_id in the <a href="#gene_align" class="sql_schema_link">gene_align</a> table</td>
        <td class="bg1">primary key</td>
      </tr>

      <tr>
        <td class="bg2"><b>seq_member_id</b></td>
        <td class="bg2">INT</td>
        <td class="bg2">-</td>
        <td class="bg2">External reference to seq_member_id in the <a href="#seq_member" class="sql_schema_link">seq_member</a> table in many-to-1 relation (single member per node)</td>
        <td class="bg2">primary key<br />key: <i>seq_member_id</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>cigar_line</b></td>
        <td class="bg1">mediumtext</td>
        <td class="bg1">NULL</td>
        <td class="bg1">String with the alignment score values</td>
        <td class="bg1"></td>
      </tr>
    </table>
  </div>
<table class="sql_schema_extra"><tr>    <td class="sql_schema_extra_left">    <p style="font-weight:bold">See also:</p>
  <ul>
      <li><a href="#seq_member">seq_member</a></li>
      <li><a href="#gene_align">gene_align</a></li>
      </ul>
    </td>
</tr></table></div></div><div class="sql_schema_table">
  <div id="gene_tree_node" class="sql_schema_table_header" style="border-top-color:#1E90FF">
    <div class="sql_schema_table_header_left"><span style="background-color:#1E90FF"></span>gene_tree_node</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_gene_tree_node" class="help-header sql_schema_show_hide" onclick="show_hide('gene_tree_node','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">This table holds the gene tree data structure, such as root, relation between parent and child, leaves, etc... In our data structure, all the trees of a given clusterset are arbitrarily connected to the same root. This eases to store and query in the same database the data from independant tree building analysis. Hence the "biological roots" of the trees are the children nodes of the main clusterset root. See the examples below.</p>

  <div id="div_gene_tree_node" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>node_id</b></td>
        <td class="bg1">INT</td>
        <td class="bg1">-</td>
        <td class="bg1">Internal unique ID</td>
        <td class="bg1">primary key</td>
      </tr>

      <tr>
        <td class="bg2"><b>parent_id</b></td>
        <td class="bg2">INT</td>
        <td class="bg2">NULL</td>
        <td class="bg2">Link to the parent node</td>
        <td class="bg2">key: <i>parent_id</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>root_id</b></td>
        <td class="bg1">INT</td>
        <td class="bg1">NULL</td>
        <td class="bg1">Link to the root node</td>
        <td class="bg1">key: <i>root_id_left_index</i></td>
      </tr>

      <tr>
        <td class="bg2"><b>left_index</b></td>
        <td class="bg2">INT</td>
        <td class="bg2">0</td>
        <td class="bg2">Internal index. See above</td>
        <td class="bg2">key: <i>root_id_left_index</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>right_index</b></td>
        <td class="bg1">INT</td>
        <td class="bg1">0</td>
        <td class="bg1">Internal index. See above</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>distance_to_parent</b></td>
        <td class="bg2">double</td>
        <td class="bg2">1</td>
        <td class="bg2">Phylogenetic distance between this node and its parent</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>seq_member_id</b></td>
        <td class="bg1">INT</td>
        <td class="bg1">NULL</td>
        <td class="bg1">External reference to seq_member_id in the <a href="#seq_member" class="sql_schema_link">seq_member</a> table to allow linkage from trees to peptides/transcripts.</td>
        <td class="bg1">key: <i>seq_member_id</i></td>
      </tr>
    </table>
  </div>
<div class="sql_schema_table_examples"><p class="sql_schema_table_example_header">Example:</p><div class="sql_schema_table_example_content"><p>The following query returns the root nodes of the independant protein trees stored in the database </p>
      <div>
        <div class="sql_schema_table_example_query">
          <pre><span class="sql_schema_sql_highlight">SELECT</span> gtn.node_id <span class="sql_schema_sql_highlight">FROM</span> gene_tree_node gtn <span class="sql_schema_sql_highlight">LEFT</span> <span class="sql_schema_sql_highlight">JOIN</span> gene_tree_root gtr <span class="sql_schema_sql_highlight">ON</span> (gtn.parent_id = gtr.root_id) <span class="sql_schema_sql_highlight">WHERE</span> gtr.tree_type = &#39;clusterset&#39; <span class="sql_schema_sql_highlight">AND</span> gtr.member_type = &#39;protein&#39; <span class="sql_schema_sql_highlight">LIMIT</span> 10;</pre>
        </div>
        <div class="sql_schema_table_example_button">
  <a id="e_gene_tree_node" class="help-header sql_schema_show_hide" onclick="show_hide('gene_tree_node','query results')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show query results
  </a></div>
        <div style="clear:both"></div>
      </div>
      
  <div id="ex_gene_tree_node" style="display:none;">
    <table class="ss sql_schema_table_example_result">
      <tr><th>gtn.node_id</th></tr>      <tr><td>2</td></tr>      <tr class="bg2"><td>14272</td></tr>      <tr><td>49629</td></tr>      <tr class="bg2"><td>56928</td></tr>      <tr><td>75004</td></tr>      <tr class="bg2"><td>80086</td></tr>      <tr><td>90049</td></tr>      <tr class="bg2"><td>99162</td></tr>      <tr><td>103230</td></tr>      <tr class="bg2"><td>107159</td></tr>    </table>
  </div></div></div><table class="sql_schema_extra"><tr>    <td class="sql_schema_extra_left">    <p style="font-weight:bold">See also:</p>
  <ul>
      <li><a href="#gene_tree_root">gene_tree_root</a></li>
      <li><a href="#seq_member">seq_member</a></li>
      <li><a href="#gene_tree_node_attr">gene_tree_node_attr</a></li>
      <li><a href="#gene_tree_node_tag">gene_tree_node_tag</a></li>
      </ul>
    </td>
</tr></table></div></div><div class="sql_schema_table">
  <div id="gene_tree_root" class="sql_schema_table_header" style="border-top-color:#1E90FF">
    <div class="sql_schema_table_header_left"><span style="background-color:#1E90FF"></span>gene_tree_root</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_gene_tree_root" class="help-header sql_schema_show_hide" onclick="show_hide('gene_tree_root','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">Header table for gene_trees. The database is able to contain several sets of trees computed on the same genes. We call these analysis "clustersets" and they can be distinguished with the clusterset_id field. Traditionally, the compara databases have contained only one clusterset (clusterset_id=1), but currently (starting on release 66) we have at least 2 (one for protein trees and one for ncRNA trees). See the examples below.</p>

  <div id="div_gene_tree_root" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>root_id</b></td>
        <td class="bg1">INT</td>
        <td class="bg1">-</td>
        <td class="bg1">Internal unique ID</td>
        <td class="bg1">primary key</td>
      </tr>

      <tr>
        <td class="bg2"><b>member_type</b></td>
        <td class="bg2">ENUM:<ul class="sql_schema_table_column_type">  <li>protein</li>  <li> ncrna</li></ul></td>
        <td class="bg2">-</td>
        <td class="bg2">The type of members used in the tree</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>tree_type</b></td>
        <td class="bg1">ENUM:<ul class="sql_schema_table_column_type">  <li>clusterset</li>  <li> supertree</li>  <li> tree</li></ul></td>
        <td class="bg1">-</td>
        <td class="bg1">The type of the tree</td>
        <td class="bg1">key: <i>tree_type</i></td>
      </tr>

      <tr>
        <td class="bg2"><b>clusterset_id</b></td>
        <td class="bg2">VARCHAR(50)</td>
        <td class="bg2">'default'</td>
        <td class="bg2">Name for the set of clusters/trees</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>method_link_species_set_id</b></td>
        <td class="bg1">INT</td>
        <td class="bg1">-</td>
        <td class="bg1">External reference to method_link_species_set_id in the method_link_species_set table</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>species_tree_root_id</b></td>
        <td class="bg2">BIGINT</td>
        <td class="bg2">NULL</td>
        <td class="bg2">External reference to root_id in the species_tree_root table</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>gene_align_id</b></td>
        <td class="bg1">INT</td>
        <td class="bg1">NULL</td>
        <td class="bg1">External reference to gene_align_id in the <a href="#gene_align" class="sql_schema_link">gene_align</a> table</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>ref_root_id</b></td>
        <td class="bg2">INT</td>
        <td class="bg2">NULL</td>
        <td class="bg2">External reference to default (merged) root_id for this tree</td>
        <td class="bg2">key: <i>ref_root_id</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>stable_id</b></td>
        <td class="bg1">VARCHAR(40)</td>
        <td class="bg1">NULL</td>
        <td class="bg1">Unique, stable ID for the tree (follows the pattern: label(5).release_introduced(4).unique_id(10))</td>
        <td class="bg1">unique: <i>key</i></td>
      </tr>

      <tr>
        <td class="bg2"><b>version</b></td>
        <td class="bg2">INT</td>
        <td class="bg2">NULL</td>
        <td class="bg2">Version of the stable ID (changes only when members move to/from existing trees)</td>
        <td class="bg2"></td>
      </tr>
    </table>
  </div>
<div class="sql_schema_table_examples"><p class="sql_schema_table_example_header">Example 1:</p><div class="sql_schema_table_example_content"><p>The following query retrieves all the node_id of the current clustersets </p>
      <div>
        <div class="sql_schema_table_example_query">
          <pre><span class="sql_schema_sql_highlight">SELECT</span> * <span class="sql_schema_sql_highlight">FROM</span> gene_tree_root <span class="sql_schema_sql_highlight">WHERE</span> tree_type = &#39;clusterset&#39;;</pre>
        </div>
        <div class="sql_schema_table_example_button">
  <a id="e_gene_tree_root1" class="help-header sql_schema_show_hide" onclick="show_hide('gene_tree_root1','query results')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show query results
  </a></div>
        <div style="clear:both"></div>
      </div>
      
  <div id="ex_gene_tree_root1" style="display:none;">
    <table class="ss sql_schema_table_example_result">
      <tr><th>root_id</th><th>member_type</th><th>tree_type</th><th>clusterset_id</th><th>method_link_species_set_id</th><th>species_tree_root_id</th><th>gene_align_id</th><th>ref_root_id</th><th>stable_id</th><th>version</th></tr>      <tr><td>1</td><td>protein</td><td>clusterset</td><td>default</td><td>40162</td><td>NULL</td><td>NULL</td><td>NULL</td><td>NULL</td><td>NULL</td></tr>      <tr class="bg2"><td>4381894</td><td>protein</td><td>clusterset</td><td>treebest</td><td>40162</td><td>NULL</td><td>NULL</td><td>NULL</td><td>NULL</td><td>NULL</td></tr>      <tr><td>4381895</td><td>protein</td><td>clusterset</td><td>phyml-aa</td><td>40162</td><td>NULL</td><td>NULL</td><td>NULL</td><td>NULL</td><td>NULL</td></tr>      <tr class="bg2"><td>4381896</td><td>protein</td><td>clusterset</td><td>phyml-nt</td><td>40162</td><td>NULL</td><td>NULL</td><td>NULL</td><td>NULL</td><td>NULL</td></tr>      <tr><td>4381897</td><td>protein</td><td>clusterset</td><td>nj-dn</td><td>40162</td><td>NULL</td><td>NULL</td><td>NULL</td><td>NULL</td><td>NULL</td></tr>      <tr class="bg2"><td>4381898</td><td>protein</td><td>clusterset</td><td>nj-ds</td><td>40162</td><td>NULL</td><td>NULL</td><td>NULL</td><td>NULL</td><td>NULL</td></tr>      <tr><td>4381899</td><td>protein</td><td>clusterset</td><td>nj-mm</td><td>40162</td><td>NULL</td><td>NULL</td><td>NULL</td><td>NULL</td><td>NULL</td></tr>      <tr class="bg2"><td>4381900</td><td>protein</td><td>clusterset</td><td>raxml</td><td>40162</td><td>NULL</td><td>NULL</td><td>NULL</td><td>NULL</td><td>NULL</td></tr>      <tr><td>4381901</td><td>protein</td><td>clusterset</td><td>raxml_parsimony</td><td>40162</td><td>NULL</td><td>NULL</td><td>NULL</td><td>NULL</td><td>NULL</td></tr>      <tr class="bg2"><td>4381902</td><td>protein</td><td>clusterset</td><td>raxml_bl</td><td>40162</td><td>NULL</td><td>NULL</td><td>NULL</td><td>NULL</td><td>NULL</td></tr>      <tr><td>4381903</td><td>protein</td><td>clusterset</td><td>notung</td><td>40162</td><td>NULL</td><td>NULL</td><td>NULL</td><td>NULL</td><td>NULL</td></tr>      <tr class="bg2"><td>4381904</td><td>protein</td><td>clusterset</td><td>treerecs</td><td>40162</td><td>NULL</td><td>NULL</td><td>NULL</td><td>NULL</td><td>NULL</td></tr>      <tr><td>4381905</td><td>protein</td><td>clusterset</td><td>copy</td><td>40162</td><td>NULL</td><td>NULL</td><td>NULL</td><td>NULL</td><td>NULL</td></tr>      <tr class="bg2"><td>4381906</td><td>protein</td><td>clusterset</td><td>raxml_update</td><td>40162</td><td>NULL</td><td>NULL</td><td>NULL</td><td>NULL</td><td>NULL</td></tr>      <tr><td>2000000001</td><td>protein</td><td>clusterset</td><td>pig_breeds</td><td>40151</td><td>NULL</td><td>NULL</td><td>NULL</td><td>NULL</td><td>NULL</td></tr>      <tr class="bg2"><td>2000361782</td><td>protein</td><td>clusterset</td><td>pig_breeds_treebest</td><td>40151</td><td>NULL</td><td>NULL</td><td>NULL</td><td>NULL</td><td>NULL</td></tr>      <tr><td>2000361783</td><td>protein</td><td>clusterset</td><td>pig_breeds_phyml-aa</td><td>40151</td><td>NULL</td><td>NULL</td><td>NULL</td><td>NULL</td><td>NULL</td></tr>      <tr class="bg2"><td>2000361784</td><td>protein</td><td>clusterset</td><td>pig_breeds_phyml-nt</td><td>40151</td><td>NULL</td><td>NULL</td><td>NULL</td><td>NULL</td><td>NULL</td></tr>      <tr><td>2000361785</td><td>protein</td><td>clusterset</td><td>pig_breeds_nj-dn</td><td>40151</td><td>NULL</td><td>NULL</td><td>NULL</td><td>NULL</td><td>NULL</td></tr>      <tr class="bg2"><td>2000361786</td><td>protein</td><td>clusterset</td><td>pig_breeds_nj-ds</td><td>40151</td><td>NULL</td><td>NULL</td><td>NULL</td><td>NULL</td><td>NULL</td></tr>      <tr><td>2000361787</td><td>protein</td><td>clusterset</td><td>pig_breeds_nj-mm</td><td>40151</td><td>NULL</td><td>NULL</td><td>NULL</td><td>NULL</td><td>NULL</td></tr>      <tr class="bg2"><td>2000361788</td><td>protein</td><td>clusterset</td><td>pig_breeds_raxml</td><td>40151</td><td>NULL</td><td>NULL</td><td>NULL</td><td>NULL</td><td>NULL</td></tr>      <tr><td>2000361789</td><td>protein</td><td>clusterset</td><td>pig_breeds_raxml_parsimony</td><td>40151</td><td>NULL</td><td>NULL</td><td>NULL</td><td>NULL</td><td>NULL</td></tr>      <tr class="bg2"><td>2000361790</td><td>protein</td><td>clusterset</td><td>pig_breeds_raxml_bl</td><td>40151</td><td>NULL</td><td>NULL</td><td>NULL</td><td>NULL</td><td>NULL</td></tr>      <tr><td>2000361791</td><td>protein</td><td>clusterset</td><td>pig_breeds_notung</td><td>40151</td><td>NULL</td><td>NULL</td><td>NULL</td><td>NULL</td><td>NULL</td></tr>      <tr class="bg2"><td>2000361792</td><td>protein</td><td>clusterset</td><td>pig_breeds_treerecs</td><td>40151</td><td>NULL</td><td>NULL</td><td>NULL</td><td>NULL</td><td>NULL</td></tr>      <tr><td>2000361793</td><td>protein</td><td>clusterset</td><td>pig_breeds_copy</td><td>40151</td><td>NULL</td><td>NULL</td><td>NULL</td><td>NULL</td><td>NULL</td></tr>      <tr class="bg2"><td>2000361794</td><td>protein</td><td>clusterset</td><td>pig_breeds_raxml_update</td><td>40151</td><td>NULL</td><td>NULL</td><td>NULL</td><td>NULL</td><td>NULL</td></tr>      <tr><td>1800000001</td><td>protein</td><td>clusterset</td><td>murinae</td><td>40157</td><td>NULL</td><td>NULL</td><td>NULL</td><td>NULL</td><td>NULL</td></tr>      <tr class="bg2"><td>1800503868</td><td>protein</td><td>clusterset</td><td>mur_treebest</td><td>40157</td><td>NULL</td><td>NULL</td><td>NULL</td><td>NULL</td><td>NULL</td></tr>      <tr><td>1800503869</td><td>protein</td><td>clusterset</td><td>mur_phyml-aa</td><td>40157</td><td>NULL</td><td>NULL</td><td>NULL</td><td>NULL</td><td>NULL</td></tr>      <tr class="bg2"><td>1800503870</td><td>protein</td><td>clusterset</td><td>mur_phyml-nt</td><td>40157</td><td>NULL</td><td>NULL</td><td>NULL</td><td>NULL</td><td>NULL</td></tr>      <tr><td>1800503871</td><td>protein</td><td>clusterset</td><td>mur_nj-dn</td><td>40157</td><td>NULL</td><td>NULL</td><td>NULL</td><td>NULL</td><td>NULL</td></tr>      <tr class="bg2"><td>1800503872</td><td>protein</td><td>clusterset</td><td>mur_nj-ds</td><td>40157</td><td>NULL</td><td>NULL</td><td>NULL</td><td>NULL</td><td>NULL</td></tr>      <tr><td>1800503873</td><td>protein</td><td>clusterset</td><td>mur_nj-mm</td><td>40157</td><td>NULL</td><td>NULL</td><td>NULL</td><td>NULL</td><td>NULL</td></tr>      <tr class="bg2"><td>1800503874</td><td>protein</td><td>clusterset</td><td>mur_raxml</td><td>40157</td><td>NULL</td><td>NULL</td><td>NULL</td><td>NULL</td><td>NULL</td></tr>      <tr><td>1800503875</td><td>protein</td><td>clusterset</td><td>mur_raxml_parsimony</td><td>40157</td><td>NULL</td><td>NULL</td><td>NULL</td><td>NULL</td><td>NULL</td></tr>      <tr class="bg2"><td>1800503876</td><td>protein</td><td>clusterset</td><td>mur_raxml_bl</td><td>40157</td><td>NULL</td><td>NULL</td><td>NULL</td><td>NULL</td><td>NULL</td></tr>      <tr><td>1800503877</td><td>protein</td><td>clusterset</td><td>mur_notung</td><td>40157</td><td>NULL</td><td>NULL</td><td>NULL</td><td>NULL</td><td>NULL</td></tr>      <tr class="bg2"><td>1800503878</td><td>protein</td><td>clusterset</td><td>mur_treerecs</td><td>40157</td><td>NULL</td><td>NULL</td><td>NULL</td><td>NULL</td><td>NULL</td></tr>      <tr><td>1800503879</td><td>protein</td><td>clusterset</td><td>mur_copy</td><td>40157</td><td>NULL</td><td>NULL</td><td>NULL</td><td>NULL</td><td>NULL</td></tr>      <tr class="bg2"><td>1800503880</td><td>protein</td><td>clusterset</td><td>mur_raxml_update</td><td>40157</td><td>NULL</td><td>NULL</td><td>NULL</td><td>NULL</td><td>NULL</td></tr>      <tr><td>1900000001</td><td>ncrna</td><td>clusterset</td><td>murinae</td><td>40159</td><td>NULL</td><td>NULL</td><td>NULL</td><td>NULL</td><td>NULL</td></tr>      <tr class="bg2"><td>1900267755</td><td>ncrna</td><td>clusterset</td><td>mur_pg_it_nj</td><td>40159</td><td>NULL</td><td>NULL</td><td>NULL</td><td>NULL</td><td>NULL</td></tr>      <tr><td>1900267756</td><td>ncrna</td><td>clusterset</td><td>mur_ml_it_10</td><td>40159</td><td>NULL</td><td>NULL</td><td>NULL</td><td>NULL</td><td>NULL</td></tr>      <tr class="bg2"><td>1900267757</td><td>ncrna</td><td>clusterset</td><td>mur_pg_it_phyml</td><td>40159</td><td>NULL</td><td>NULL</td><td>NULL</td><td>NULL</td><td>NULL</td></tr>      <tr><td>1900267758</td><td>ncrna</td><td>clusterset</td><td>mur_ss_it_s16</td><td>40159</td><td>NULL</td><td>NULL</td><td>NULL</td><td>NULL</td><td>NULL</td></tr>      <tr class="bg2"><td>1900267759</td><td>ncrna</td><td>clusterset</td><td>mur_ss_it_s6a</td><td>40159</td><td>NULL</td><td>NULL</td><td>NULL</td><td>NULL</td><td>NULL</td></tr>      <tr><td>1900267760</td><td>ncrna</td><td>clusterset</td><td>mur_ss_it_s16a</td><td>40159</td><td>NULL</td><td>NULL</td><td>NULL</td><td>NULL</td><td>NULL</td></tr>      <tr class="bg2"><td>1900267761</td><td>ncrna</td><td>clusterset</td><td>mur_ss_it_s6b</td><td>40159</td><td>NULL</td><td>NULL</td><td>NULL</td><td>NULL</td><td>NULL</td></tr>    </table>
  </div></div></div><div class="sql_schema_table_examples"><p class="sql_schema_table_example_header">Example 2:</p><div class="sql_schema_table_example_content"><p>To get the number of trees of each type </p>
      <div>
        <div class="sql_schema_table_example_query">
          <pre><span class="sql_schema_sql_highlight">SELECT</span> member_type, tree_type, COUNT(*) <span class="sql_schema_sql_highlight">FROM</span> gene_tree_root <span class="sql_schema_sql_highlight">GROUP</span> <span class="sql_schema_sql_highlight">BY</span> member_type, tree_type;</pre>
        </div>
        <div class="sql_schema_table_example_button">
  <a id="e_gene_tree_root2" class="help-header sql_schema_show_hide" onclick="show_hide('gene_tree_root2','query results')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show query results
  </a></div>
        <div style="clear:both"></div>
      </div>
      
  <div id="ex_gene_tree_root2" style="display:none;">
    <table class="ss sql_schema_table_example_result">
      <tr><th>member_type</th><th>tree_type</th><th>COUNT(*)</th></tr>      <tr><td>protein</td><td>clusterset</td><td>42</td></tr>      <tr class="bg2"><td>protein</td><td>supertree</td><td>3884</td></tr>      <tr><td>protein</td><td>tree</td><td>309625</td></tr>      <tr class="bg2"><td>ncrna</td><td>clusterset</td><td>66</td></tr>      <tr><td>ncrna</td><td>supertree</td><td>84</td></tr>      <tr class="bg2"><td>ncrna</td><td>tree</td><td>118254</td></tr>    </table>
  </div></div></div><table class="sql_schema_extra"><tr>    <td class="sql_schema_extra_left">    <p style="font-weight:bold">See also:</p>
  <ul>
      <li><a href="#gene_tree_node">gene_tree_node</a></li>
      <li><a href="#gene_tree_root_tag">gene_tree_root_tag</a></li>
      <li><a href="#seq_member">seq_member</a></li>
      <li><a href="#method_link_species_set">method_link_species_set</a></li>
      <li><a href="#gene_align">gene_align</a></li>
      </ul>
    </td>
</tr></table></div></div><div class="sql_schema_table">
  <div id="gene_tree_node_tag" class="sql_schema_table_header" style="border-top-color:#1E90FF">
    <div class="sql_schema_table_header_left"><span style="background-color:#1E90FF"></span>gene_tree_node_tag</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_gene_tree_node_tag" class="help-header sql_schema_show_hide" onclick="show_hide('gene_tree_node_tag','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">This table contains several tag/value data attached to node_ids</p>

  <div id="div_gene_tree_node_tag" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>node_id</b></td>
        <td class="bg1">INT</td>
        <td class="bg1">-</td>
        <td class="bg1">External reference to node_id in the <a href="#gene_tree_node" class="sql_schema_link">gene_tree_node</a> table</td>
        <td class="bg1">key: <i>node_id_tag</i></td>
      </tr>

      <tr>
        <td class="bg2"><b>tag</b></td>
        <td class="bg2">varchar(50)</td>
        <td class="bg2">-</td>
        <td class="bg2">Tag name for the tag/value pair</td>
        <td class="bg2">key: <i>node_id_tag</i><br />key: <i>tag</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>value</b></td>
        <td class="bg1">mediumtext</td>
        <td class="bg1">-</td>
        <td class="bg1">Value for the tag/value pair</td>
        <td class="bg1"></td>
      </tr>
    </table>
  </div>
<table class="sql_schema_extra"><tr>    <td class="sql_schema_extra_left">    <p style="font-weight:bold">See also:</p>
  <ul>
      <li><a href="#gene_tree_node">gene_tree_node</a></li>
      <li><a href="#gene_tree_node_attr">gene_tree_node_attr</a></li>
      </ul>
    </td>
</tr></table></div></div><div class="sql_schema_table">
  <div id="gene_tree_root_tag" class="sql_schema_table_header" style="border-top-color:#1E90FF">
    <div class="sql_schema_table_header_left"><span style="background-color:#1E90FF"></span>gene_tree_root_tag</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_gene_tree_root_tag" class="help-header sql_schema_show_hide" onclick="show_hide('gene_tree_root_tag','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">This table contains several tag/value data for gene_tree_roots</p>

  <div id="div_gene_tree_root_tag" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>root_id</b></td>
        <td class="bg1">INT</td>
        <td class="bg1">-</td>
        <td class="bg1">External reference to root_id in the <a href="#gene_tree_root" class="sql_schema_link">gene_tree_root</a> table</td>
        <td class="bg1">key: <i>root_id_tag</i></td>
      </tr>

      <tr>
        <td class="bg2"><b>tag</b></td>
        <td class="bg2">VARCHAR(255)</td>
        <td class="bg2">-</td>
        <td class="bg2">Tag name for the tag/value pair</td>
        <td class="bg2">key: <i>root_id_tag</i><br />key: <i>tag</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>value</b></td>
        <td class="bg1">mediumtext</td>
        <td class="bg1">-</td>
        <td class="bg1">Value for the tag/value pair</td>
        <td class="bg1"></td>
      </tr>
    </table>
  </div>
<table class="sql_schema_extra"><tr>    <td class="sql_schema_extra_left">    <p style="font-weight:bold">See also:</p>
  <ul>
      <li><a href="#gene_tree_root">gene_tree_root</a></li>
      </ul>
    </td>
</tr></table></div></div><div class="sql_schema_table">
  <div id="gene_tree_root_attr" class="sql_schema_table_header" style="border-top-color:#1E90FF">
    <div class="sql_schema_table_header_left"><span style="background-color:#1E90FF"></span>gene_tree_root_attr</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_gene_tree_root_attr" class="help-header sql_schema_show_hide" onclick="show_hide('gene_tree_root_attr','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">This table contains several gene tree attributes data attached to root_ids</p>

  <div id="div_gene_tree_root_attr" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>root_id</b></td>
        <td class="bg1">INT</td>
        <td class="bg1">-</td>
        <td class="bg1">External reference to root_id in the <a href="#gene_tree_root" class="sql_schema_link">gene_tree_root</a> table.</td>
        <td class="bg1">primary key</td>
      </tr>

      <tr>
        <td class="bg2"><b>aln_after_filter_length</b></td>
        <td class="bg2">INT</td>
        <td class="bg2">NULL</td>
        <td class="bg2">Alignment length after filtering.</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>aln_length</b></td>
        <td class="bg1">INT</td>
        <td class="bg1">NULL</td>
        <td class="bg1">Alignment length before filtering.</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>aln_num_residues</b></td>
        <td class="bg2">INT</td>
        <td class="bg2">NULL</td>
        <td class="bg2">Total number of residues in the whole alignment.</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>aln_percent_identity</b></td>
        <td class="bg1">FLOAT(5)</td>
        <td class="bg1">NULL</td>
        <td class="bg1">Alignment identity.</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>best_fit_model_family</b></td>
        <td class="bg2">VARCHAR(10)</td>
        <td class="bg2">NULL</td>
        <td class="bg2">Best Amino Acid replacement evolution model (WAG, JTT, etc).</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>best_fit_model_parameter</b></td>
        <td class="bg1">VARCHAR(5)</td>
        <td class="bg1">NULL</td>
        <td class="bg1">Best paremeters used in the model (I, G, IG, IGF, etc).</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>gene_count</b></td>
        <td class="bg2">INT</td>
        <td class="bg2">NULL</td>
        <td class="bg2">Number of sequences present in the alignment.</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>k_score</b></td>
        <td class="bg1">FLOAT(5)</td>
        <td class="bg1">NULL</td>
        <td class="bg1">Tree distance metric.</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>k_score_rank</b></td>
        <td class="bg2">INT</td>
        <td class="bg2">NULL</td>
        <td class="bg2">Rank of the tree in the comparison.</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>mcoffee_scores_gene_align_id</b></td>
        <td class="bg1">INT</td>
        <td class="bg1">NULL</td>
        <td class="bg1">Gene alignment ID, used to fetch the mcoffee scores used in the alignment.</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>aln_n_removed_columns</b></td>
        <td class="bg2">INT</td>
        <td class="bg2">NULL</td>
        <td class="bg2">Number of colunms that were removed by the alignment filtering process.</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>aln_num_of_patterns</b></td>
        <td class="bg1">INT</td>
        <td class="bg1">NULL</td>
        <td class="bg1">Number of different patterns present in the alignment (used by ExaML).</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>aln_shrinking_factor</b></td>
        <td class="bg2">FLOAT(2)</td>
        <td class="bg2">NULL</td>
        <td class="bg2">Factor used to measure how much the alignments were filtered (factor: 0..1).</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>spec_count</b></td>
        <td class="bg1">INT</td>
        <td class="bg1">NULL</td>
        <td class="bg1">Number of different species present in the cluster</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>tree_max_branch</b></td>
        <td class="bg2">FLOAT</td>
        <td class="bg2">NULL</td>
        <td class="bg2">Maximum branch length.</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>tree_max_length</b></td>
        <td class="bg1">FLOAT(5)</td>
        <td class="bg1">NULL</td>
        <td class="bg1">Maximum tree length.</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>tree_num_dup_nodes</b></td>
        <td class="bg2">INT</td>
        <td class="bg2">NULL</td>
        <td class="bg2">Number of duplication nodes.</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>tree_num_leaves</b></td>
        <td class="bg1">INT</td>
        <td class="bg1">NULL</td>
        <td class="bg1">Number of leaves in a tree.</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>tree_num_spec_nodes</b></td>
        <td class="bg2">INT</td>
        <td class="bg2">NULL</td>
        <td class="bg2">Number of speciation events.</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>lca_node_id</b></td>
        <td class="bg1">BIGINT</td>
        <td class="bg1">NULL</td>
        <td class="bg1">Lowest common ancestor (species_tree node_id).</td>
        <td class="bg1">key: <i>lca_node_id</i></td>
      </tr>

      <tr>
        <td class="bg2"><b>taxonomic_coverage</b></td>
        <td class="bg2">FLOAT(5)</td>
        <td class="bg2">NULL</td>
        <td class="bg2">Taxonomic coverage of the species present in the gene tree over all the species for that particular node on the species tree.</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>ratio_species_genes</b></td>
        <td class="bg1">FLOAT(5)</td>
        <td class="bg1">NULL</td>
        <td class="bg1">Ration of the number of species over the number of genes in a tree.</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>model_name</b></td>
        <td class="bg2">VARCHAR(40)</td>
        <td class="bg2">NULL</td>
        <td class="bg2">HMM model name (cluster table_id).</td>
        <td class="bg2"></td>
      </tr>
    </table>
  </div>
<table class="sql_schema_extra"><tr>    <td class="sql_schema_extra_left">    <p style="font-weight:bold">See also:</p>
  <ul>
      <li><a href="#gene_tree_root">gene_tree_root</a></li>
      <li><a href="#gene_tree_root_tag">gene_tree_root_tag</a></li>
      </ul>
    </td>
</tr></table></div></div><div class="sql_schema_table">
  <div id="gene_tree_node_attr" class="sql_schema_table_header" style="border-top-color:#1E90FF">
    <div class="sql_schema_table_header_left"><span style="background-color:#1E90FF"></span>gene_tree_node_attr</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_gene_tree_node_attr" class="help-header sql_schema_show_hide" onclick="show_hide('gene_tree_node_attr','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">This table contains several gene tree attributes data attached to node_ids</p>

  <div id="div_gene_tree_node_attr" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>node_id</b></td>
        <td class="bg1">INT</td>
        <td class="bg1">-</td>
        <td class="bg1">External reference to node_id in the <a href="#gene_tree_node" class="sql_schema_link">gene_tree_node</a> table</td>
        <td class="bg1">primary key</td>
      </tr>

      <tr>
        <td class="bg2"><b>node_type</b></td>
        <td class="bg2">ENUM:<ul class="sql_schema_table_column_type">  <li>duplication</li>  <li> dubious</li>  <li> speciation</li>  <li> sub-speciation</li>  <li> gene_split</li></ul></td>
        <td class="bg2">NULL</td>
        <td class="bg2">Type of homology</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>species_tree_node_id</b></td>
        <td class="bg1">BIGINT</td>
        <td class="bg1">NULL</td>
        <td class="bg1">Taxon / Ancestral species annotation (given as the node_id in the species tree)</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>bootstrap</b></td>
        <td class="bg2">TINYINT</td>
        <td class="bg2">NULL</td>
        <td class="bg2">The bootstrap value of the node in the tree</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>duplication_confidence_score</b></td>
        <td class="bg1">DOUBLE(5,4)</td>
        <td class="bg1">NULL</td>
        <td class="bg1">The calculated confidence score for duplications</td>
        <td class="bg1"></td>
      </tr>
    </table>
  </div>
<table class="sql_schema_extra"><tr>    <td class="sql_schema_extra_left">    <p style="font-weight:bold">See also:</p>
  <ul>
      <li><a href="#gene_tree_node">gene_tree_node</a></li>
      <li><a href="#gene_tree_node_tag">gene_tree_node_tag</a></li>
      </ul>
    </td>
</tr></table></div></div><div class="sql_schema_table">
  <div id="gene_tree_object_store" class="sql_schema_table_header" style="border-top-color:#1E90FF">
    <div class="sql_schema_table_header_left"><span style="background-color:#1E90FF"></span>gene_tree_object_store</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_gene_tree_object_store" class="help-header sql_schema_show_hide" onclick="show_hide('gene_tree_object_store','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">This table contains arbitrary data related to gene-trees. Commonly used for precomputed tracks / layers</p>

  <div id="div_gene_tree_object_store" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>root_id</b></td>
        <td class="bg1">INT</td>
        <td class="bg1">-</td>
        <td class="bg1">External reference to root_id in the <a href="#gene_tree_root" class="sql_schema_link">gene_tree_root</a> table</td>
        <td class="bg1">primary key</td>
      </tr>

      <tr>
        <td class="bg2"><b>data_label</b></td>
        <td class="bg2">VARCHAR(255)</td>
        <td class="bg2">-</td>
        <td class="bg2">A label that uniquely identifies the data (for a given root_id)</td>
        <td class="bg2">primary key</td>
      </tr>

      <tr>
        <td class="bg1"><b>compressed_data</b></td>
        <td class="bg1">MEDIUMBLOB</td>
        <td class="bg1">-</td>
        <td class="bg1">The data, compressed with ZLib. Can be uncompressed with MySQL&#39;s UNCOMPRESS</td>
        <td class="bg1"></td>
      </tr>
    </table>
  </div>
<table class="sql_schema_extra"><tr>    <td class="sql_schema_extra_left">    <p style="font-weight:bold">See also:</p>
  <ul>
      <li><a href="#gene_tree_root">gene_tree_root</a></li>
      </ul>
    </td>
</tr></table></div></div><div class="sql_schema_table">
  <div id="homology" class="sql_schema_table_header" style="border-top-color:#1E90FF">
    <div class="sql_schema_table_header_left"><span style="background-color:#1E90FF"></span>homology</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_homology" class="help-header sql_schema_show_hide" onclick="show_hide('homology','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">This table contains all the genomic homologies. There are two homology_member entries for each homology entry for now, but both the schema and the API can handle more than just pairwise relationships. <br />dN, dS, N, S and lnL are statistical values given by the codeml program of the <a href="http://abacus.gene.ucl.ac.uk/software/paml.html">Phylogenetic Analysis by Maximum Likelihood (PAML)</a> package.</p>

  <div id="div_homology" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>homology_id</b></td>
        <td class="bg1">bigint</td>
        <td class="bg1">-</td>
        <td class="bg1">Unique internal ID</td>
        <td class="bg1">primary key</td>
      </tr>

      <tr>
        <td class="bg2"><b>method_link_species_set_id</b></td>
        <td class="bg2">INT</td>
        <td class="bg2">-</td>
        <td class="bg2">External reference to method_link_species_set_id in the <a href="#method_link_species_set" class="sql_schema_link">method_link_species_set</a> table</td>
        <td class="bg2">key: <i>method_link_species_set_id</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>description</b></td>
        <td class="bg1">ENUM:<ul class="sql_schema_table_column_type">  <li>ortholog_one2one</li>  <li>ortholog_one2many</li>  <li>ortholog_many2many</li>  <li>within_species_paralog</li>  <li>other_paralog</li>  <li>gene_split</li>  <li>between_species_paralog</li>  <li>alt_allele</li>  <li>homoeolog_one2one</li>  <li>homoeolog_one2many</li>  <li>homoeolog_many2many</li></ul></td>
        <td class="bg1">-</td>
        <td class="bg1">A normalized, short description of the homology relationship</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>is_tree_compliant</b></td>
        <td class="bg2">tinyint(1)</td>
        <td class="bg2">0</td>
        <td class="bg2">Whether the homology is fully compliant with the tree and the definition of orthology / paralogy</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>dn</b></td>
        <td class="bg1">float(10,5)</td>
        <td class="bg1">NULL</td>
        <td class="bg1">The non-synonymous mutation rate</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>ds</b></td>
        <td class="bg2">float(10,5)</td>
        <td class="bg2">NULL</td>
        <td class="bg2">The synonymous mutation rate</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>n</b></td>
        <td class="bg1">float(10,1)</td>
        <td class="bg1">NULL</td>
        <td class="bg1">The estimated number of non-synonymous mutations</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>s</b></td>
        <td class="bg2">float(10,1)</td>
        <td class="bg2">NULL</td>
        <td class="bg2">The estimated number of synonymous mutations</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>lnl</b></td>
        <td class="bg1">float(10,3)</td>
        <td class="bg1">NULL</td>
        <td class="bg1">The negative log likelihood of the estimation</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>species_tree_node_id</b></td>
        <td class="bg2">bigint</td>
        <td class="bg2">NULL</td>
        <td class="bg2">The node_id of the species-tree node to which the homology is attached</td>
        <td class="bg2">key: <i>species_tree_node_id</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>gene_tree_node_id</b></td>
        <td class="bg1">INT</td>
        <td class="bg1">NULL</td>
        <td class="bg1">The node_id of the gene-tree node from which the homology is derived</td>
        <td class="bg1">key: <i>gene_tree_node_id</i></td>
      </tr>

      <tr>
        <td class="bg2"><b>gene_tree_root_id</b></td>
        <td class="bg2">INT</td>
        <td class="bg2">NULL</td>
        <td class="bg2">The root_id of the gene tree from which the homology is derived</td>
        <td class="bg2">key: <i>gene_tree_root_id</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>goc_score</b></td>
        <td class="bg1">tinyint</td>
        <td class="bg1">NULL</td>
        <td class="bg1">Gene order conservation score</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>wga_coverage</b></td>
        <td class="bg2">DECIMAL(5,2)</td>
        <td class="bg2">NULL</td>
        <td class="bg2">Whole genome alignment coverage of the homology</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>is_high_confidence</b></td>
        <td class="bg1">tinyint(1)</td>
        <td class="bg1">NULL</td>
        <td class="bg1">Whether the homology is considered &quot;high-confidence&quot;</td>
        <td class="bg1"></td>
      </tr>
    </table>
  </div>
<div class="sql_schema_table_examples"><p class="sql_schema_table_example_header">Example 1:</p><div class="sql_schema_table_example_content"><p>The following query defines a pair of paralogous xenopous genes. See <a href="#homology_member" class="sql_schema_link">homology_member</a> for more details </p>
      <div>
        <div class="sql_schema_table_example_query">
          <pre><span class="sql_schema_sql_highlight">SELECT</span> homology.* <span class="sql_schema_sql_highlight">FROM</span> homology <span class="sql_schema_sql_highlight">JOIN</span> method_link_species_set <span class="sql_schema_sql_highlight">USING</span> (method_link_species_set_id) <span class="sql_schema_sql_highlight">WHERE</span> name=&quot;Xtro paralogues&quot; <span class="sql_schema_sql_highlight">LIMIT</span> 1;</pre>
        </div>
        <div class="sql_schema_table_example_button">
  <a id="e_homology1" class="help-header sql_schema_show_hide" onclick="show_hide('homology1','query results')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show query results
  </a></div>
        <div style="clear:both"></div>
      </div>
      
  <div id="ex_homology1" style="display:none;">
    <table class="ss sql_schema_table_example_result">
      <tr><th>homology_id</th><th>method_link_species_set_id</th><th>description</th><th>is_tree_compliant</th><th>dn</th><th>ds</th><th>n</th><th>s</th><th>lnl</th><th>species_tree_node_id</th><th>gene_tree_node_id</th><th>gene_tree_root_id</th><th>goc_score</th><th>wga_coverage</th><th>is_high_confidence</th></tr>      <tr><td>52728496</td><td>149080</td><td>other_paralog</td><td>1</td><td>NULL</td><td>NULL</td><td>NULL</td><td>NULL</td><td>NULL</td><td>4016200011</td><td>7027638</td><td>3</td><td>NULL</td><td>NULL</td><td>NULL</td></tr>    </table>
  </div></div></div><div class="sql_schema_table_examples"><p class="sql_schema_table_example_header">Example 2:</p><div class="sql_schema_table_example_content"><p>See species_names that participate in this particular homology entry </p>
      <div>
        <div class="sql_schema_table_example_query">
          <pre><span class="sql_schema_sql_highlight">SELECT</span> homology_id, description, GROUP_CONCAT(genome_db.name) <span class="sql_schema_sql_highlight">AS</span> species <span class="sql_schema_sql_highlight">FROM</span> homology <span class="sql_schema_sql_highlight">JOIN</span> method_link_species_set <span class="sql_schema_sql_highlight">USING</span> (method_link_species_set_id) <span class="sql_schema_sql_highlight">JOIN</span> species_set <span class="sql_schema_sql_highlight">USING</span> (species_set_id) <span class="sql_schema_sql_highlight">JOIN</span> genome_db USING(genome_db_id) <span class="sql_schema_sql_highlight">WHERE</span> method_link_id=201 <span class="sql_schema_sql_highlight">AND</span> homology_id&lt;200000000  <span class="sql_schema_sql_highlight">GROUP</span> <span class="sql_schema_sql_highlight">BY</span> homology_id <span class="sql_schema_sql_highlight">LIMIT</span> 4;</pre>
        </div>
        <div class="sql_schema_table_example_button">
  <a id="e_homology2" class="help-header sql_schema_show_hide" onclick="show_hide('homology2','query results')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show query results
  </a></div>
        <div style="clear:both"></div>
      </div>
      
  <div id="ex_homology2" style="display:none;">
    <table class="ss sql_schema_table_example_result">
      <tr><th>homology_id</th><th>description</th><th>species</th></tr>      <tr><td>53839224</td><td>ortholog_one2one</td><td>gorilla_gorilla,pongo_abelii</td></tr>      <tr class="bg2"><td>53839225</td><td>ortholog_one2many</td><td>gorilla_gorilla,pongo_abelii</td></tr>      <tr><td>53839226</td><td>ortholog_one2many</td><td>gorilla_gorilla,pongo_abelii</td></tr>      <tr class="bg2"><td>53839227</td><td>ortholog_one2one</td><td>gorilla_gorilla,pongo_abelii</td></tr>    </table>
  </div></div></div><table class="sql_schema_extra"><tr>    <td class="sql_schema_extra_left">    <p style="font-weight:bold">See also:</p>
  <ul>
      <li><a href="#homology_member">homology_member</a></li>
      <li><a href="#method_link_species_set">method_link_species_set</a></li>
      </ul>
    </td>
</tr></table></div></div><div class="sql_schema_table">
  <div id="homology_member" class="sql_schema_table_header" style="border-top-color:#1E90FF">
    <div class="sql_schema_table_header_left"><span style="background-color:#1E90FF"></span>homology_member</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_homology_member" class="help-header sql_schema_show_hide" onclick="show_hide('homology_member','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">This table contains the sequences corresponding to every genomic homology relationship found. There are two homology_member entries for each pairwise homology entry. As written in the homology table section, both schema and API can deal with more than pairwise relationships.<br /> The original alignment is not stored but it can be retrieved using the cigar_line field and the original sequences. The cigar line defines the sequence of matches or mismatches and deletions in the alignment. <ul> <li><b>First peptide sequence</b>: SERCQVVVISIGPISVLSMILDFY</li> <li><b>Second peptide sequence</b>: SDRCQVLVISILSMIGLDFY</li> <li><b>First corresponding cigar line</b>: 20MD4M</li> <li><b>Second corresponding cigar line</b>: 11M5D9M</li> </ul> The alignment will be:<br /> <table class="spreadsheet"> <caption>Example of alignment reconstruction</caption> <tbody> <tr> <th>First peptide cigar line</th> <td>M</td> <td>M</td> <td>M</td> <td>M</td> <td>M</td> <td>M</td> <td>M</td> <td>M</td> <td>M</td> <td>M</td> <td>M</td> <td>M</td> <td>M</td> <td>M</td> <td>M</td> <td>M</td> <td>M</td> <td>M</td> <td>M</td> <td>M</td> <td>D</td> <td>M</td> <td>M</td> <td>M</td> <td>M</td> </tr> <tr> <th>First aligned peptide</th> <td>S</td> <td>E</td> <td>R</td> <td>C</td> <td>Q</td> <td>V</td> <td>V</td> <td>V</td> <td>I</td> <td>S</td> <td>I</td> <td>G</td> <td>P</td> <td>I</td> <td>S</td> <td>V</td> <td>L</td> <td>S</td> <td>M</td> <td>I</td> <td>-</td> <td>L</td> <td>D</td> <td>F</td> <td>Y</td> </tr> <tr> <th>Second aligned peptide</th> <td>S</td> <td>D</td> <td>R</td> <td>C</td> <td>Q</td> <td>V</td> <td>L</td> <td>V</td> <td>I</td> <td>S</td> <td>I</td> <td>-</td> <td>-</td> <td>-</td> <td>-</td> <td>-</td> <td>L</td> <td>S</td> <td>M</td> <td>I</td> <td>G</td> <td>L</td> <td>D</td> <td>F</td> <td>Y</td> </tr> <tr> <th>Second peptide cigar line</th> <td>M</td> <td>M</td> <td>M</td> <td>M</td> <td>M</td> <td>M</td> <td>M</td> <td>M</td> <td>M</td> <td>M</td> <td>M</td> <td>D</td> <td>D</td> <td>D</td> <td>D</td> <td>D</td> <td>M</td> <td>M</td> <td>M</td> <td>M</td> <td>M</td> <td>M</td> <td>M</td> <td>M</td> <td>M</td> </tr> </tbody> </table></p>

  <div id="div_homology_member" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>homology_id</b></td>
        <td class="bg1">bigint</td>
        <td class="bg1">-</td>
        <td class="bg1">External reference to homology_id in the <a href="#homology" class="sql_schema_link">homology</a> table</td>
        <td class="bg1">primary key</td>
      </tr>

      <tr>
        <td class="bg2"><b>gene_member_id</b></td>
        <td class="bg2">INT</td>
        <td class="bg2">-</td>
        <td class="bg2">External reference to gene_member_id in the <a href="#gene_member" class="sql_schema_link">gene_member</a> table. Refers to the corresponding &quot;ENSMBLGENE&quot; entry</td>
        <td class="bg2">primary key<br />key: <i>gene_member_id</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>seq_member_id</b></td>
        <td class="bg1">INT</td>
        <td class="bg1">NULL</td>
        <td class="bg1">External reference to seq_member_id in the <a href="#seq_member" class="sql_schema_link">seq_member</a> table. Refers to the corresponding &quot;ENSEMBLPEP&quot; entry</td>
        <td class="bg1">key: <i>seq_member_id</i></td>
      </tr>

      <tr>
        <td class="bg2"><b>cigar_line</b></td>
        <td class="bg2">mediumtext</td>
        <td class="bg2">NULL</td>
        <td class="bg2">An internal description of the alignment. It contains mathces/mismatches (M) and delations (D) and refers to the corresponding seq_member_id sequence</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>perc_cov</b></td>
        <td class="bg1">float</td>
        <td class="bg1">0</td>
        <td class="bg1">Defines the percentage of the peptide which has been aligned</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>perc_id</b></td>
        <td class="bg2">float</td>
        <td class="bg2">0</td>
        <td class="bg2">Defines the percentage of identity between both homologues</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>perc_pos</b></td>
        <td class="bg1">float</td>
        <td class="bg1">0</td>
        <td class="bg1">Defines the percentage of positivity (similarity) between both homologues</td>
        <td class="bg1"></td>
      </tr>
    </table>
  </div>
<div class="sql_schema_table_examples"><p class="sql_schema_table_example_header">Example:</p><div class="sql_schema_table_example_content"><p>The following query refers to the two homologue sequences from the first xenopus' paralogy object. Gene and peptide sequence of the second homologue can retrieved in the same way. </p>
      <div>
        <div class="sql_schema_table_example_query">
          <pre><span class="sql_schema_sql_highlight">SELECT</span> homology_member.* <span class="sql_schema_sql_highlight">FROM</span> homology_member <span class="sql_schema_sql_highlight">JOIN</span> homology <span class="sql_schema_sql_highlight">USING</span> (homology_id) <span class="sql_schema_sql_highlight">JOIN</span> method_link_species_set <span class="sql_schema_sql_highlight">USING</span> (method_link_species_set_id) <span class="sql_schema_sql_highlight">WHERE</span> name=&quot;Xtro paralogues&quot; <span class="sql_schema_sql_highlight">LIMIT</span> 2;</pre>
        </div>
        <div class="sql_schema_table_example_button">
  <a id="e_homology_member" class="help-header sql_schema_show_hide" onclick="show_hide('homology_member','query results')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show query results
  </a></div>
        <div style="clear:both"></div>
      </div>
      
  <div id="ex_homology_member" style="display:none;">
    <table class="ss sql_schema_table_example_result">
      <tr><th>homology_id</th><th>gene_member_id</th><th>seq_member_id</th><th>cigar_line</th><th>perc_cov</th><th>perc_id</th><th>perc_pos</th></tr>      <tr><td>52728496</td><td>915779297</td><td>952795836</td><td>2M3D5MD6MDMD2M3D4M3D8MD9MD33MD11MD3MD5M2D45MD4MD63MD10MD41MD5MDMD9M2D33M4D7M</td><td>90.228</td><td>39.7394</td><td>59.9349</td></tr>      <tr class="bg2"><td>52728496</td><td>915796546</td><td>952835206</td><td>MD10MDMD3MD7M2D6MD4MD12MD34MD6MD7MD5M2D41MD5MD27MD37MD4MD45MD5MD4MD3M2D42M6D</td><td>89.644</td><td>39.4822</td><td>59.5469</td></tr>    </table>
  </div></div></div><table class="sql_schema_extra"><tr>    <td class="sql_schema_extra_left">    <p style="font-weight:bold">See also:</p>
  <ul>
      <li><a href="#gene_member">gene_member</a></li>
      <li><a href="#seq_member">seq_member</a></li>
      <li><a href="#homology">homology</a></li>
      </ul>
    </td>
</tr></table></div></div>

<div id="header_8" class="sql_schema_group_header" style="border-color:#FFCC66">
  <div class="sql_schema_group_bullet" style="background-color:#FFCC66"></div>
  <h2>Extra annotations on members</h2>
</div>
<p class="sql_schema_group_header_desc">Various member (gene and proteins) related information stored in the database, either loaded from Core databases or aggregated from Compara analyses</p><div align="center"><a href="diagrams/Compara.extra_annotations_on_members.png"><img src="diagrams/Compara.extra_annotations_on_members.png" width="231px" /></a></div><div class="sql_schema_table">
  <div id="gene_member_hom_stats" class="sql_schema_table_header" style="border-top-color:#FFCC66">
    <div class="sql_schema_table_header_left"><span style="background-color:#FFCC66"></span>gene_member_hom_stats</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_gene_member_hom_stats" class="help-header sql_schema_show_hide" onclick="show_hide('gene_member_hom_stats','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">This table contains for each gene_member some statistics about the homology pipelines we've run</p>

  <div id="div_gene_member_hom_stats" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>gene_member_id</b></td>
        <td class="bg1">INT</td>
        <td class="bg1">-</td>
        <td class="bg1">External reference to gene_member_id in the <a href="#gene_member" class="sql_schema_link">gene_member</a> table</td>
        <td class="bg1">primary key</td>
      </tr>

      <tr>
        <td class="bg2"><b>collection</b></td>
        <td class="bg2">varchar(40)</td>
        <td class="bg2">-</td>
        <td class="bg2">Name of the collection this row of statistics refers to (usual values are &quot;ensembl&quot;, &quot;mouse&quot;, etc)</td>
        <td class="bg2">primary key</td>
      </tr>

      <tr>
        <td class="bg1"><b>families</b></td>
        <td class="bg1">INT</td>
        <td class="bg1">0</td>
        <td class="bg1">The number of families associated with this member</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>gene_trees</b></td>
        <td class="bg2">tinyint(1)</td>
        <td class="bg2">0</td>
        <td class="bg2">If this member is part of a gene tree</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>gene_gain_loss_trees</b></td>
        <td class="bg1">tinyint(1)</td>
        <td class="bg1">0</td>
        <td class="bg1">If this member is part of a gene gain/loss tree</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>orthologues</b></td>
        <td class="bg2">INT</td>
        <td class="bg2">0</td>
        <td class="bg2">The number of orthologues for this member</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>paralogues</b></td>
        <td class="bg1">INT</td>
        <td class="bg1">0</td>
        <td class="bg1">The number of paralogues for this member</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>homoeologues</b></td>
        <td class="bg2">INT</td>
        <td class="bg2">0</td>
        <td class="bg2">The number of homoeologues for this member</td>
        <td class="bg2"></td>
      </tr>
    </table>
  </div>
<table class="sql_schema_extra"><tr>    <td class="sql_schema_extra_left">    <p style="font-weight:bold">See also:</p>
  <ul>
      <li><a href="#gene_member">gene_member</a></li>
      </ul>
    </td>
</tr></table></div></div><div class="sql_schema_table">
  <div id="seq_member_projection_stable_id" class="sql_schema_table_header" style="border-top-color:#FFCC66">
    <div class="sql_schema_table_header_left"><span style="background-color:#FFCC66"></span>seq_member_projection_stable_id</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_seq_member_projection_stable_id" class="help-header sql_schema_show_hide" onclick="show_hide('seq_member_projection_stable_id','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">This table stores data about projected transcripts (in the gene-annotation process), which is used to help the clustering. This table links to the source stable_id and is used until the source members are loaded</p>

  <div id="div_seq_member_projection_stable_id" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>target_seq_member_id</b></td>
        <td class="bg1">INT</td>
        <td class="bg1">-</td>
        <td class="bg1">External reference to seq_member_id in the <a href="#seq_member" class="sql_schema_link">seq_member</a> table. Shows the target of the projection, i.e. this transcript was annotated by projection of source_stable_id</td>
        <td class="bg1">primary key</td>
      </tr>

      <tr>
        <td class="bg2"><b>source_stable_id</b></td>
        <td class="bg2">VARCHAR(128)</td>
        <td class="bg2">-</td>
        <td class="bg2">The stable ID of the source of the projection</td>
        <td class="bg2">index: <i>source_stable_id</i></td>
      </tr>
    </table>
  </div>
<div class="sql_schema_table_examples"><p class="sql_schema_table_example_header">Example:</p><div class="sql_schema_table_example_content"><p>The following query shows the projections of the mouse gene Pdk3 to all the other species </p>
      <div>
        <div class="sql_schema_table_example_query">
          <pre><span class="sql_schema_sql_highlight">SELECT</span> ss.stable_id, gs.name, source_stable_id <span class="sql_schema_sql_highlight">FROM</span> seq_member ss <span class="sql_schema_sql_highlight">JOIN</span> genome_db gs <span class="sql_schema_sql_highlight">USING</span> (genome_db_id) <span class="sql_schema_sql_highlight">JOIN</span> seq_member_projection_stable_id <span class="sql_schema_sql_highlight">ON</span> seq_member_id = target_seq_member_id <span class="sql_schema_sql_highlight">WHERE</span> source_stable_id = &quot;ENSMUST00000045748&quot;</pre>
        </div>
        <div class="sql_schema_table_example_button">
  <a id="e_seq_member_projection_stable_id" class="help-header sql_schema_show_hide" onclick="show_hide('seq_member_projection_stable_id','query results')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show query results
  </a></div>
        <div style="clear:both"></div>
      </div>
      
  <div id="ex_seq_member_projection_stable_id" style="display:none;">
    <table class="ss sql_schema_table_example_result">
      <tr><th>ss.stable_id</th><th>gs.name</th><th>source_stable_id</th></tr>      <tr><td>MGP_AJ_P0096917</td><td>mus_musculus_aj</td><td>ENSMUST00000045748</td></tr>      <tr class="bg2"><td>MGP_NZOHlLtJ_P0097695</td><td>mus_musculus_nzohlltj</td><td>ENSMUST00000045748</td></tr>      <tr><td>MGP_CAROLIEiJ_P0092103</td><td>mus_caroli</td><td>ENSMUST00000045748</td></tr>      <tr class="bg2"><td>MGP_SPRETEiJ_P0095872</td><td>mus_spretus</td><td>ENSMUST00000045748</td></tr>      <tr><td>MGP_BALBcJ_P0096836</td><td>mus_musculus_balbcj</td><td>ENSMUST00000045748</td></tr>      <tr class="bg2"><td>MGP_C57BL6NJ_P0097338</td><td>mus_musculus_c57bl6nj</td><td>ENSMUST00000045748</td></tr>      <tr><td>MGP_WSBEiJ_P0095354</td><td>mus_musculus_wsbeij</td><td>ENSMUST00000045748</td></tr>      <tr class="bg2"><td>MGP_CASTEiJ_P0096968</td><td>mus_musculus_casteij</td><td>ENSMUST00000045748</td></tr>      <tr><td>MGP_129S1SvImJ_P0096820</td><td>mus_musculus_129s1svimj</td><td>ENSMUST00000045748</td></tr>      <tr class="bg2"><td>MGP_LPJ_P0096544</td><td>mus_musculus_lpj</td><td>ENSMUST00000045748</td></tr>      <tr><td>MGP_PWKPhJ_P0096374</td><td>mus_musculus_pwkphj</td><td>ENSMUST00000045748</td></tr>      <tr class="bg2"><td>MGP_AKRJ_P0096852</td><td>mus_musculus_akrj</td><td>ENSMUST00000045748</td></tr>      <tr><td>MGP_C3HHeJ_P0096387</td><td>mus_musculus_c3hhej</td><td>ENSMUST00000045748</td></tr>      <tr class="bg2"><td>MGP_PahariEiJ_P0089728</td><td>mus_pahari</td><td>ENSMUST00000045748</td></tr>      <tr><td>MGP_CBAJ_P0096302</td><td>mus_musculus_cbaj</td><td>ENSMUST00000045748</td></tr>      <tr class="bg2"><td>MGP_NODShiLtJ_P0096296</td><td>mus_musculus_nodshiltj</td><td>ENSMUST00000045748</td></tr>      <tr><td>MGP_DBA2J_P0096515</td><td>mus_musculus_dba2j</td><td>ENSMUST00000045748</td></tr>      <tr class="bg2"><td>MGP_FVBNJ_P0096348</td><td>mus_musculus_fvbnj</td><td>ENSMUST00000045748</td></tr>      <tr><td>ENSPEMP00000016688</td><td>peromyscus_maniculatus_bairdii</td><td>ENSMUST00000045748</td></tr>    </table>
  </div></div></div><table class="sql_schema_extra"><tr>    <td class="sql_schema_extra_left">    <p style="font-weight:bold">See also:</p>
  <ul>
      <li><a href="#seq_member">seq_member</a></li>
      <li><a href="#seq_member_projection">seq_member_projection</a></li>
      </ul>
    </td>
</tr></table></div></div><div class="sql_schema_table">
  <div id="seq_member_projection" class="sql_schema_table_header" style="border-top-color:#FFCC66">
    <div class="sql_schema_table_header_left"><span style="background-color:#FFCC66"></span>seq_member_projection</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_seq_member_projection" class="help-header sql_schema_show_hide" onclick="show_hide('seq_member_projection','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">This table stores data about projected transcripts (in the gene-annotation process), which is used to help the clustering. This table can only be used when both genomes have been loaded. Thus we first populate <a href="#seq_member_projection_stable_id" class="sql_schema_link">seq_member_projection_stable_id</a> and then copy the data whilst transforming the stable_id into a seq_member_id</p>

  <div id="div_seq_member_projection" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>target_seq_member_id</b></td>
        <td class="bg1">INT</td>
        <td class="bg1">-</td>
        <td class="bg1">External reference to seq_member_id in the <a href="#seq_member" class="sql_schema_link">seq_member</a> table. Shows the target of the projection, i.e. this transcript was annotated by projection of source_seq_member_id</td>
        <td class="bg1">primary key</td>
      </tr>

      <tr>
        <td class="bg2"><b>source_seq_member_id</b></td>
        <td class="bg2">INT</td>
        <td class="bg2">-</td>
        <td class="bg2">External reference to seq_member_id in the <a href="#seq_member" class="sql_schema_link">seq_member</a> table. Shows the source of the projection</td>
        <td class="bg2">key: <i>source_seq_member_id</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>identity</b></td>
        <td class="bg1">float(5,2)</td>
        <td class="bg1">NULL</td>
        <td class="bg1">(can be missing). The percentage of identity between the two members.</td>
        <td class="bg1"></td>
      </tr>
    </table>
  </div>
<div class="sql_schema_table_examples"><p class="sql_schema_table_example_header">Example:</p><div class="sql_schema_table_example_content"><p>The following query shows the projections of the gene AGO2 with the Anole lizard and the zebrafinch sql       SELECT ss.stable_id, gs.name, st.stable_id, gt.name, identity FROM seq_member ss JOIN genome_db gs USING (genome_db_id) JOIN seq_member_projection ON ss.seq_member_id = source_seq_member_id JOIN (seq_member st JOIN genome_db gt USING (genome_db_id)) ON st.seq_member_id=target_seq_member_id WHERE ss.stable_id IN (&quot;ENSACAP00000000183&quot;, &quot;ENSTGUP00000014905&quot;);</p></div></div><table class="sql_schema_extra"><tr>    <td class="sql_schema_extra_left">    <p style="font-weight:bold">See also:</p>
  <ul>
      <li><a href="#seq_member">seq_member</a></li>
      <li><a href="#seq_member_projection_stable_id">seq_member_projection_stable_id</a></li>
      </ul>
    </td>
</tr></table></div></div><div class="sql_schema_table">
  <div id="external_db" class="sql_schema_table_header" style="border-top-color:#FFCC66">
    <div class="sql_schema_table_header_left"><span style="background-color:#FFCC66"></span>external_db</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_external_db" class="help-header sql_schema_show_hide" onclick="show_hide('external_db','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">This table stores data about the external databases in which the objects described in the <a href="#member_xref" class="sql_schema_link">member_xref</a> table are stored.</p>

  <div id="div_external_db" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>external_db_id</b></td>
        <td class="bg1">int(10)</td>
        <td class="bg1">-</td>
        <td class="bg1">Internal unique ID</td>
        <td class="bg1">primary key</td>
      </tr>

      <tr>
        <td class="bg2"><b>db_name</b></td>
        <td class="bg2">varchar(100)</td>
        <td class="bg2">-</td>
        <td class="bg2">External database name</td>
        <td class="bg2">unique key: <i>db_name_db_release_idx</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>db_release</b></td>
        <td class="bg1">varchar(255)</td>
        <td class="bg1">NULL</td>
        <td class="bg1">Release information about the external database</td>
        <td class="bg1">unique key: <i>db_name_db_release_idx</i></td>
      </tr>

      <tr>
        <td class="bg2"><b>status</b></td>
        <td class="bg2">ENUM:<ul class="sql_schema_table_column_type">  <li>KNOWNXREF</li>  <li>KNOWN</li>  <li>XREF</li>  <li>PRED</li>  <li>ORTH</li>  <li>PSEUDO</li></ul></td>
        <td class="bg2">-</td>
        <td class="bg2">Status, e.g. &#39;KNOWNXREF&#39;,&#39;KNOWN&#39;,&#39;XREF&#39;,&#39;PRED&#39;,&#39;ORTH&#39;,&#39;PSEUDO&#39;.</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>priority</b></td>
        <td class="bg1">int(11)</td>
        <td class="bg1">-</td>
        <td class="bg1">Determines which one of the xrefs will be used as the gene name.</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>db_display_name</b></td>
        <td class="bg2">varchar(255)</td>
        <td class="bg2">NULL</td>
        <td class="bg2">Database display name</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>type</b></td>
        <td class="bg1">ENUM:<ul class="sql_schema_table_column_type">  <li>ARRAY</li>  <li>ALT_TRANS</li>  <li>ALT_GENE</li>  <li>MISC</li>  <li>LIT</li>  <li>PRIMARY_DB_SYNONYM</li>  <li>ENSEMBL</li></ul></td>
        <td class="bg1">NULL</td>
        <td class="bg1">Type, e.g. &#39;ARRAY&#39;, &#39;ALT_TRANS&#39;, &#39;ALT_GENE&#39;, &#39;MISC&#39;, &#39;LIT&#39;, &#39;PRIMARY_DB_SYNONYM&#39;, &#39;ENSEMBL&#39;</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>secondary_db_name</b></td>
        <td class="bg2">varchar(255)</td>
        <td class="bg2">NULL</td>
        <td class="bg2">Secondary database name</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>secondary_db_table</b></td>
        <td class="bg1">varchar(255)</td>
        <td class="bg1">NULL</td>
        <td class="bg1">Secondary database table</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>description</b></td>
        <td class="bg2">text</td>
        <td class="bg2">NULL</td>
        <td class="bg2">Description of the external database</td>
        <td class="bg2"></td>
      </tr>
    </table>
  </div>
<table class="sql_schema_extra"><tr>    <td class="sql_schema_extra_left">    <p style="font-weight:bold">See also:</p>
  <ul>
      <li><a href="#member_xref">member_xref</a></li>
      </ul>
    </td>
</tr></table></div></div><div class="sql_schema_table">
  <div id="member_xref" class="sql_schema_table_header" style="border-top-color:#FFCC66">
    <div class="sql_schema_table_header_left"><span style="background-color:#FFCC66"></span>member_xref</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_member_xref" class="help-header sql_schema_show_hide" onclick="show_hide('member_xref','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">This table stores cross-references for gene members derived from the core databases. It is used by Bio::EnsEMBL::Compara::DBSQL::XrefMemberAdaptor and provides the data used in highlighting gene trees by GO and InterPro annotation" </p>

  <div id="div_member_xref" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>gene_member_id</b></td>
        <td class="bg1">INT</td>
        <td class="bg1">-</td>
        <td class="bg1">External reference to gene_member_id in the <a href="#gene_member" class="sql_schema_link">gene_member</a> table. Indicates the gene to which the xref applies.</td>
        <td class="bg1">primary key</td>
      </tr>

      <tr>
        <td class="bg2"><b>dbprimary_acc</b></td>
        <td class="bg2">varchar(10)</td>
        <td class="bg2">-</td>
        <td class="bg2">Accession of xref (e.g. GO term, InterPro accession)</td>
        <td class="bg2">primary key</td>
      </tr>

      <tr>
        <td class="bg1"><b>external_db_id</b></td>
        <td class="bg1">int(10)</td>
        <td class="bg1">-</td>
        <td class="bg1">External reference to external_db_id in the <a href="#external_db" class="sql_schema_link">external_db</a> table. Indicates to which external database the xref belongs.</td>
        <td class="bg1">primary key</td>
      </tr>
    </table>
  </div>
<table class="sql_schema_extra"><tr>    <td class="sql_schema_extra_left">    <p style="font-weight:bold">See also:</p>
  <ul>
      <li><a href="#external_db">external_db</a></li>
      <li><a href="#gene_member">gene_member</a></li>
      </ul>
    </td>
</tr></table></div></div><div class="sql_schema_table">
  <div id="gene_member_qc" class="sql_schema_table_header" style="border-top-color:#FFCC66">
    <div class="sql_schema_table_header_left"><span style="background-color:#FFCC66"></span>gene_member_qc</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_gene_member_qc" class="help-header sql_schema_show_hide" onclick="show_hide('gene_member_qc','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">This table contains gene quality information from the geneset_QC pipeline</p>

  <div id="div_gene_member_qc" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>gene_member_stable_id</b></td>
        <td class="bg1">varchar(128)</td>
        <td class="bg1">-</td>
        <td class="bg1">EnsEMBL stable ID</td>
        <td class="bg1">key: <i>gene_member_stable_id</i></td>
      </tr>

      <tr>
        <td class="bg2"><b>genome_db_id</b></td>
        <td class="bg2">INT</td>
        <td class="bg2">-</td>
        <td class="bg2">Internal unique ID for this table</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>seq_member_id</b></td>
        <td class="bg1">INT</td>
        <td class="bg1">NULL</td>
        <td class="bg1">canonical seq_member_id</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>n_species</b></td>
        <td class="bg2">INT</td>
        <td class="bg2">NULL</td>
        <td class="bg2">-n_species</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>n_orth</b></td>
        <td class="bg1">INT</td>
        <td class="bg1">NULL</td>
        <td class="bg1">-n_orth</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>avg_cov</b></td>
        <td class="bg2">FLOAT</td>
        <td class="bg2">NULL</td>
        <td class="bg2">-avg_cov</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>status</b></td>
        <td class="bg1">varchar(50)</td>
        <td class="bg1">-</td>
        <td class="bg1">&quot;orphaned-gene&quot;, &quot;split-gene&quot;, &quot;long-gene&quot; or &quot;short-gene&quot;</td>
        <td class="bg1"></td>
      </tr>
    </table>
  </div>
</div></div>

<div id="header_9" class="sql_schema_group_header" style="border-color:#BC5CEC">
  <div class="sql_schema_group_bullet" style="background-color:#BC5CEC"></div>
  <h2>Protein families</h2>
</div>
<p class="sql_schema_group_header_desc">Protein families (sets of homologous protein sequences)</p><div align="center"><a href="diagrams/Compara.protein_families.png"><img src="diagrams/Compara.protein_families.png" width="815px" /></a></div><div class="sql_schema_table">
  <div id="family" class="sql_schema_table_header" style="border-top-color:#BC5CEC">
    <div class="sql_schema_table_header_left"><span style="background-color:#BC5CEC"></span>family</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_family" class="help-header sql_schema_show_hide" onclick="show_hide('family','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">This table contains all the group homologies found. There are several family_member entries for each family entry.</p>

  <div id="div_family" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>family_id</b></td>
        <td class="bg1">INT</td>
        <td class="bg1">-</td>
        <td class="bg1">Internal unique ID</td>
        <td class="bg1">primary key</td>
      </tr>

      <tr>
        <td class="bg2"><b>stable_id</b></td>
        <td class="bg2">varchar(40)</td>
        <td class="bg2">-</td>
        <td class="bg2">Stable family ID. NOTE: stable_id are currently not stable. We are working in getting IDs stable between releases.</td>
        <td class="bg2">unique: <i>key</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>version</b></td>
        <td class="bg1">INT</td>
        <td class="bg1">-</td>
        <td class="bg1">Version of the stable_id (may only change when members migrate between this family and another one; stays the same otherwise)</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>method_link_species_set_id</b></td>
        <td class="bg2">INT</td>
        <td class="bg2">-</td>
        <td class="bg2">External reference to method_link_species_set_id in the method_link_species_set table</td>
        <td class="bg2">key: <i>method_link_species_set_id</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>description</b></td>
        <td class="bg1">TEXT</td>
        <td class="bg1">NULL</td>
        <td class="bg1">Description of the family as found using the Longest Common String (LCS) of the descriptions of the member proteins.</td>
        <td class="bg1">key: <i>description(255)</i></td>
      </tr>

      <tr>
        <td class="bg2"><b>description_score</b></td>
        <td class="bg2">double</td>
        <td class="bg2">NULL</td>
        <td class="bg2">Scores the accuracy of the annotation (max. 100)</td>
        <td class="bg2"></td>
      </tr>
    </table>
  </div>
<table class="sql_schema_extra"><tr>    <td class="sql_schema_extra_left">    <p style="font-weight:bold">See also:</p>
  <ul>
      <li><a href="#method_link_species_set">method_link_species_set</a></li>
      </ul>
    </td>
</tr></table></div></div><div class="sql_schema_table">
  <div id="family_member" class="sql_schema_table_header" style="border-top-color:#BC5CEC">
    <div class="sql_schema_table_header_left"><span style="background-color:#BC5CEC"></span>family_member</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_family_member" class="help-header sql_schema_show_hide" onclick="show_hide('family_member','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">This table contains the proteins corresponding to protein family relationship found. There are several family_member entries for each family entry</p>

  <div id="div_family_member" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>family_id</b></td>
        <td class="bg1">INT</td>
        <td class="bg1">-</td>
        <td class="bg1">External reference to family_id in the <a href="#family" class="sql_schema_link">family</a> table</td>
        <td class="bg1">primary key</td>
      </tr>

      <tr>
        <td class="bg2"><b>seq_member_id</b></td>
        <td class="bg2">INT</td>
        <td class="bg2">-</td>
        <td class="bg2">External reference to the seq_member_id in the <a href="#seq_member" class="sql_schema_link">seq_member</a> table</td>
        <td class="bg2">primary key<br />key: <i>seq_member_id</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>cigar_line</b></td>
        <td class="bg1">mediumtext</td>
        <td class="bg1">NULL</td>
        <td class="bg1">Internal description of the multiple alignment (see the description in the <a href="#homology_member" class="sql_schema_link">homology_member</a> table)</td>
        <td class="bg1"></td>
      </tr>
    </table>
  </div>
<table class="sql_schema_extra"><tr>    <td class="sql_schema_extra_left">    <p style="font-weight:bold">See also:</p>
  <ul>
      <li><a href="#family">family</a></li>
      <li><a href="#seq_member">seq_member</a></li>
      <li><a href="#homology_member">homology_member</a></li>
      </ul>
    </td>
</tr></table></div></div>

<div id="header_10" class="sql_schema_group_header" style="border-color:#66CCFF">
  <div class="sql_schema_group_bullet" style="background-color:#66CCFF"></div>
  <h2>Profile HMMs</h2>
</div>
<p class="sql_schema_group_header_desc">The profile HMMs and their hits</p><div align="center"><a href="diagrams/Compara.profile_hmms.png"><img src="diagrams/Compara.profile_hmms.png" width="401px" /></a></div><div class="sql_schema_table">
  <div id="hmm_profile" class="sql_schema_table_header" style="border-top-color:#66CCFF">
    <div class="sql_schema_table_header_left"><span style="background-color:#66CCFF"></span>hmm_profile</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_hmm_profile" class="help-header sql_schema_show_hide" onclick="show_hide('hmm_profile','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">This table stores different HMM-based profiles used and produced by gene trees</p>

  <div id="div_hmm_profile" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>model_id</b></td>
        <td class="bg1">varchar(40)</td>
        <td class="bg1">-</td>
        <td class="bg1">Model ID of the profile. This is the stable_id of the gene-tree, or the external ID in case of imported models (such as RF00001)</td>
        <td class="bg1">primary key</td>
      </tr>

      <tr>
        <td class="bg2"><b>name</b></td>
        <td class="bg2">varchar(40)</td>
        <td class="bg2">NULL</td>
        <td class="bg2">Name of the model, if available (such as 5S_rRNA for RF00001)</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>type</b></td>
        <td class="bg1">varchar(40)</td>
        <td class="bg1">-</td>
        <td class="bg1">Short description of the origin of the profile (usually, one of &quot;tree_hmm_aa_v3&quot;, &quot;tree_hmm_dna_v3&quot;, &quot;infernal&quot; or &quot;infernal-refined&quot;)</td>
        <td class="bg1">primary key</td>
      </tr>

      <tr>
        <td class="bg2"><b>compressed_profile</b></td>
        <td class="bg2">mediumblob</td>
        <td class="bg2">NULL</td>
        <td class="bg2">The HMM profile, compressed with zlib. It can be decompressed with the MySQL function UNCOMPRESS()</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>consensus</b></td>
        <td class="bg1">mediumtext</td>
        <td class="bg1">NULL</td>
        <td class="bg1">The consensus sequence derived from the profile</td>
        <td class="bg1"></td>
      </tr>
    </table>
  </div>
</div></div><div class="sql_schema_table">
  <div id="hmm_annot" class="sql_schema_table_header" style="border-top-color:#66CCFF">
    <div class="sql_schema_table_header_left"><span style="background-color:#66CCFF"></span>hmm_annot</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_hmm_annot" class="help-header sql_schema_show_hide" onclick="show_hide('hmm_annot','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">This table stores the HMM annotation of the seq_members</p>

  <div id="div_hmm_annot" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>seq_member_id</b></td>
        <td class="bg1">INT</td>
        <td class="bg1">-</td>
        <td class="bg1">External reference to a seq_member_id in the <a href="#seq_member" class="sql_schema_link">seq_member</a> table</td>
        <td class="bg1">primary key</td>
      </tr>

      <tr>
        <td class="bg2"><b>model_id</b></td>
        <td class="bg2">varchar(40)</td>
        <td class="bg2">NULL</td>
        <td class="bg2">External reference to the internal numeric ID of a HMM profile in <a href="#hmm_profile" class="sql_schema_link">hmm_profile</a></td>
        <td class="bg2">key: <i>model_id</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>evalue</b></td>
        <td class="bg1">float</td>
        <td class="bg1">NULL</td>
        <td class="bg1">The e-value of the hit</td>
        <td class="bg1"></td>
      </tr>
    </table>
  </div>
<table class="sql_schema_extra"><tr>    <td class="sql_schema_extra_left">    <p style="font-weight:bold">See also:</p>
  <ul>
      <li><a href="#seq_member">seq_member</a></li>
      <li><a href="#hmm_profile">hmm_profile</a></li>
      </ul>
    </td>
</tr></table></div></div><div class="sql_schema_table">
  <div id="hmm_curated_annot" class="sql_schema_table_header" style="border-top-color:#66CCFF">
    <div class="sql_schema_table_header_left"><span style="background-color:#66CCFF"></span>hmm_curated_annot</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_hmm_curated_annot" class="help-header sql_schema_show_hide" onclick="show_hide('hmm_curated_annot','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">This table stores the curated / forced HMM annotation of the seq_members</p>

  <div id="div_hmm_curated_annot" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>seq_member_stable_id</b></td>
        <td class="bg1">varchar(40)</td>
        <td class="bg1">-</td>
        <td class="bg1">External reference to a seq_member_id in the <a href="#seq_member" class="sql_schema_link">seq_member</a> table</td>
        <td class="bg1">primary key</td>
      </tr>

      <tr>
        <td class="bg2"><b>model_id</b></td>
        <td class="bg2">varchar(40)</td>
        <td class="bg2">NULL</td>
        <td class="bg2">External reference to the internal numeric ID of a HMM profile in <a href="#hmm_profile" class="sql_schema_link">hmm_profile</a></td>
        <td class="bg2">key: <i>model_id</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>library_version</b></td>
        <td class="bg1">varchar(40)</td>
        <td class="bg1">-</td>
        <td class="bg1">Name of the HMM library against the curation has been done</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>annot_date</b></td>
        <td class="bg2">TIMESTAMP</td>
        <td class="bg2">CURRENT_TIMESTAMP</td>
        <td class="bg2">When did the curation happened</td>
        <td class="bg2"></td>
      </tr>

      <tr>
        <td class="bg1"><b>reason</b></td>
        <td class="bg1">MEDIUMTEXT</td>
        <td class="bg1">NULL</td>
        <td class="bg1">Why are we forcing this curation</td>
        <td class="bg1"></td>
      </tr>
    </table>
  </div>
<table class="sql_schema_extra"><tr>    <td class="sql_schema_extra_left">    <p style="font-weight:bold">See also:</p>
  <ul>
      <li><a href="#seq_member">seq_member</a></li>
      <li><a href="#hmm_profile">hmm_profile</a></li>
      </ul>
    </td>
</tr></table></div></div>

<div id="header_11" class="sql_schema_group_header" style="border-color:#AAAAAA">
  <div class="sql_schema_group_bullet" style="background-color:#AAAAAA"></div>
  <h2>Stable-ID mapping</h2>
</div>
<p class="sql_schema_group_header_desc">History of the gene-tree and family IDs across different versions of Ensembl</p><div align="center"><a href="diagrams/Compara.stable_id_mapping.png"><img src="diagrams/Compara.stable_id_mapping.png" width="500px" /></a></div><div class="sql_schema_table">
  <div id="mapping_session" class="sql_schema_table_header" style="border-top-color:#AAAAAA">
    <div class="sql_schema_table_header_left"><span style="background-color:#AAAAAA"></span>mapping_session</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_mapping_session" class="help-header sql_schema_show_hide" onclick="show_hide('mapping_session','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">This table contains one entry per stable_id mapping session (either for Families or for Protein Trees), which contains the type, the date of the mapping, and which releases were linked together. A single mapping_session is the event when mapping between two given releases for a particular class type ('family' or 'tree') is loaded. The whole event is thought to happen momentarily at 'when_mapped' (used for sorting in historical order).</p>

  <div id="div_mapping_session" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>mapping_session_id</b></td>
        <td class="bg1">INT</td>
        <td class="bg1">-</td>
        <td class="bg1">Internal unique ID</td>
        <td class="bg1">primary key</td>
      </tr>

      <tr>
        <td class="bg2"><b>type</b></td>
        <td class="bg2">ENUM:<ul class="sql_schema_table_column_type">  <li>family</li>  <li> tree</li>  <li> hmm</li></ul></td>
        <td class="bg2">NULL</td>
        <td class="bg2">Type of stable_ids that were mapped during this session</td>
        <td class="bg2">unique: <i>key</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>when_mapped</b></td>
        <td class="bg1">TIMESTAMP</td>
        <td class="bg1">CURRENT_TIMESTAMP</td>
        <td class="bg1">Normally, we use the date of creation of the mapping file being loaded. This prevents the date from chaging even if we accidentally remove the entry and have to re-load it.</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>rel_from</b></td>
        <td class="bg2">INT</td>
        <td class="bg2">NULL</td>
        <td class="bg2">rel.number from which the stable_ids were mapped during this session. rel_from &amp;lt; rel_to</td>
        <td class="bg2">unique: <i>key</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>rel_to</b></td>
        <td class="bg1">INT</td>
        <td class="bg1">NULL</td>
        <td class="bg1">rel.number to which the stable_ids were mapped during this session. rel_from &amp;lt; rel_to</td>
        <td class="bg1">unique: <i>key</i></td>
      </tr>

      <tr>
        <td class="bg2"><b>prefix</b></td>
        <td class="bg2">CHAR(4)</td>
        <td class="bg2">-</td>
        <td class="bg2">Prefix</td>
        <td class="bg2">unique: <i>key</i></td>
      </tr>
    </table>
  </div>
</div></div><div class="sql_schema_table">
  <div id="stable_id_history" class="sql_schema_table_header" style="border-top-color:#AAAAAA">
    <div class="sql_schema_table_header_left"><span style="background-color:#AAAAAA"></span>stable_id_history</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_stable_id_history" class="help-header sql_schema_show_hide" onclick="show_hide('stable_id_history','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">This table keeps the history of stable_id changes from one release to another. The primary key 'object' describes a set of members migrating from stable_id_from to stable_id_to. Their volume (related to the 'shared_size' of the new class) is reflected by the fractional 'contribution' field. Since both stable_ids are listed in the primary key, they are not allowed to be NULLs. We shall treat empty strings as NULLs. If stable_id_from is empty, it means these members are newcomers into the new release. If stable_id_to is empty, it means these previously known members are disappearing in the new release. If both neither stable_id_from nor stable_id_to is empty, these members are truly migrating.</p>

  <div id="div_stable_id_history" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>mapping_session_id</b></td>
        <td class="bg1">INT</td>
        <td class="bg1">-</td>
        <td class="bg1">Reference to mapping_session.mapping_session_id. All the stable_ids of a given mapping should have the same session_id</td>
        <td class="bg1">primary key</td>
      </tr>

      <tr>
        <td class="bg2"><b>stable_id_from</b></td>
        <td class="bg2">VARCHAR(40)</td>
        <td class="bg2">''</td>
        <td class="bg2">The previous stable ID</td>
        <td class="bg2">primary key</td>
      </tr>

      <tr>
        <td class="bg1"><b>version_from</b></td>
        <td class="bg1">INT</td>
        <td class="bg1">NULL</td>
        <td class="bg1">The version number of the previous stable ID (specific to each stable ID; not to be confused with the release number)</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>stable_id_to</b></td>
        <td class="bg2">VARCHAR(40)</td>
        <td class="bg2">''</td>
        <td class="bg2">The new stable ID</td>
        <td class="bg2">primary key</td>
      </tr>

      <tr>
        <td class="bg1"><b>version_to</b></td>
        <td class="bg1">INT</td>
        <td class="bg1">NULL</td>
        <td class="bg1">The new version number</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>contribution</b></td>
        <td class="bg2">FLOAT</td>
        <td class="bg2">NULL</td>
        <td class="bg2">Percentage of of the new object (tree / family) that comes from the previous one</td>
        <td class="bg2"></td>
      </tr>
    </table>
  </div>
</div></div>

<div id="header_12" class="sql_schema_group_header" style="border-color:#01D4F7">
  <div class="sql_schema_group_bullet" style="background-color:#01D4F7"></div>
  <h2>Gene gain/loss trees</h2>
</div>
<p class="sql_schema_group_header_desc">Analysis of gain and loss across gene-families</p><div align="center"><a href="diagrams/Compara.gene_gain_loss_trees.png"><img src="diagrams/Compara.gene_gain_loss_trees.png" width="818px" /></a></div><div class="sql_schema_table">
  <div id="CAFE_gene_family" class="sql_schema_table_header" style="border-top-color:#01D4F7">
    <div class="sql_schema_table_header_left"><span style="background-color:#01D4F7"></span>CAFE_gene_family</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_CAFE_gene_family" class="help-header sql_schema_show_hide" onclick="show_hide('CAFE_gene_family','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">This table holds information about each CAFE gene family</p>

  <div id="div_CAFE_gene_family" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>cafe_gene_family_id</b></td>
        <td class="bg1">INT</td>
        <td class="bg1">-</td>
        <td class="bg1">Internal unique ID</td>
        <td class="bg1">primary key</td>
      </tr>

      <tr>
        <td class="bg2"><b>root_id</b></td>
        <td class="bg2">bigint</td>
        <td class="bg2">-</td>
        <td class="bg2">External reference to root_id in the <a href="#species_tree_root" class="sql_schema_link">species_tree_root</a> table</td>
        <td class="bg2">key: <i>root_id</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>lca_id</b></td>
        <td class="bg1">bigint</td>
        <td class="bg1">-</td>
        <td class="bg1">External reference to the lowest common ancestor node in the <a href="#species_tree_node" class="sql_schema_link">species_tree_node</a> table</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>gene_tree_root_id</b></td>
        <td class="bg2">INT</td>
        <td class="bg2">-</td>
        <td class="bg2">External reference to root_id in the <a href="#gene_tree_root" class="sql_schema_link">gene_tree_root</a> table</td>
        <td class="bg2">key: <i>gene_tree_root_id</i></td>
      </tr>

      <tr>
        <td class="bg1"><b>pvalue_avg</b></td>
        <td class="bg1">double(5,4)</td>
        <td class="bg1">NULL</td>
        <td class="bg1">The average pvalue for the gene family as reported by CAFE</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>lambdas</b></td>
        <td class="bg2">varchar(100)</td>
        <td class="bg2">NULL</td>
        <td class="bg2">The lambda/s values reported/used by CAFE</td>
        <td class="bg2"></td>
      </tr>
    </table>
  </div>
<table class="sql_schema_extra"><tr>    <td class="sql_schema_extra_left">    <p style="font-weight:bold">See also:</p>
  <ul>
      <li><a href="#species_tree_node">species_tree_node</a></li>
      <li><a href="#species_tree_root">species_tree_root</a></li>
      <li><a href="#CAFE_species_gene">CAFE_species_gene</a></li>
      <li><a href="#gene_tree_root">gene_tree_root</a></li>
      </ul>
    </td>
</tr></table></div></div><div class="sql_schema_table">
  <div id="CAFE_species_gene" class="sql_schema_table_header" style="border-top-color:#01D4F7">
    <div class="sql_schema_table_header_left"><span style="background-color:#01D4F7"></span>CAFE_species_gene</div>
    <div class="sql_schema_table_header_right">
  
  <a id="a_CAFE_species_gene" class="help-header sql_schema_show_hide" onclick="show_hide('CAFE_species_gene','columns')">
    <img src="/i/16/plus-button.png" class="sql_schema_icon" alt="show"/> Show columns
  </a>
      <span class="sql_schema_table_separator"> </span> <a href="#top" class="sql_schema_link">[Back to top]</a>
    </div>
    <div style="clear:both"></div>
  </div>
<div class="sql_schema_table_content">  <p class="sql_schema_table_desc">This table stores per <a href="#species_tree_node" class="sql_schema_link">species_tree_node</a> information about expansions/contractions of each <a href="#CAFE_gene_family" class="sql_schema_link">CAFE_gene_family</a></p>

  <div id="div_CAFE_species_gene" style="display:none">
    <table class="ss sql_schema_table_column">
      <tr class="center">
        <th>Column</th>
        <th>Type</th>
        <th class="val">Default value</th>
        <th class="desc">Description</th>
        <th class="index">Index</th>
      </tr>

      <tr>
        <td class="bg1"><b>cafe_gene_family_id</b></td>
        <td class="bg1">INT</td>
        <td class="bg1">-</td>
        <td class="bg1">External reference to cafe_gene_family_id in the <a href="#CAFE_gene_family" class="sql_schema_link">CAFE_gene_family</a> table.</td>
        <td class="bg1">primary key</td>
      </tr>

      <tr>
        <td class="bg2"><b>node_id</b></td>
        <td class="bg2">bigint</td>
        <td class="bg2">-</td>
        <td class="bg2">External reference to node_id in the <a href="#species_tree_node" class="sql_schema_link">species_tree_node</a> table</td>
        <td class="bg2">primary key</td>
      </tr>

      <tr>
        <td class="bg1"><b>n_members</b></td>
        <td class="bg1">INT</td>
        <td class="bg1">-</td>
        <td class="bg1">The number of members for the node as reported by CAFE</td>
        <td class="bg1"></td>
      </tr>

      <tr>
        <td class="bg2"><b>pvalue</b></td>
        <td class="bg2">double(5,4)</td>
        <td class="bg2">NULL</td>
        <td class="bg2">The pvalue of the node as reported by CAFE</td>
        <td class="bg2"></td>
      </tr>
    </table>
  </div>
<table class="sql_schema_extra"><tr>    <td class="sql_schema_extra_left">    <p style="font-weight:bold">See also:</p>
  <ul>
      <li><a href="#species_tree_node">species_tree_node</a></li>
      <li><a href="#CAFE_gene_family">CAFE_gene_family</a></li>
      </ul>
    </td>
</tr></table></div></div>

    </div>
  </div>
</body>
</html>
