<p>
  We display 17,1796 sequence variants from four different data sources. There is overlap between the sample sets used in variant identification, but different analysis methods were employed.
</p>



<h2>European Nucleotide Archive (ENA)</h2>

<p>The ENA team have developed a <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3526318/">LoFreq</a>-based <a href="https://github.com/enasequence/covid-sequence-analysis-workflow/blob/master/workflow.nf">pipeline</a> to call variants from SARS-CoV-2 data submitted to the archive. 
This estimates the proportion of each variant allele within the sequenced sample. 
4,852 variants from the preliminary analysis of the first available 5,115 sequencing runs ( 2020/08/17) are displayed here</p>


<h3>Data filtering</h3>

<p>As this is an early version of the ENA variant data strict, basic filters have been applied to reduce the proportion of lower confidence sites. We have removed:</p>
<p>
  <ul>
    <li>Sequencing runs with more than 40 variant calls </li>
    <li>Variants where no sample has a frequency of 20% or more for the non-reference allele</li>
    <li>Variants where all samples show strand bias
  </ul>
</p>

<h3>Data representation</h3>

<p>We represent only the alleles seen in each sample and not the frequencies observed within each sample. Variants were called for each sequencing run individually and for display it is assumed that sites at which a variant was not called in a sample match the reference. This will provide a more accurate estimation of the frequency of each allele across the entire sample set.</p>

<h2>European Variation Archive (EVA)</h2>
<p>
14,806 variant loci from the first SARS-CoV-2 
<a href="https://ftp.ebi.ac.uk/pub/databases/eva/covid19_releases/release_1/">data release</a> 
(20/07/2021) from the EVA are available. 
Variant records from different submissions have been clustered by location and type and stable RefSNP accessions have been assigned. 
These records include variant location, alleles and RefSNP identifier.
</p>


<h2>Nextstrain</h2>

<p>
  SARS-CoV-2 mutation data is made available as a phylogenetic tree by <a href='https://nextstrain.org/'>Nextstrain</a>[Hadfield 2018], an open-source project to harness the scientific and public health potential of pathogen genome data.
The Nextstrain project uses subsamples of the available genome sequence data to provide a real-time snapshot of evolving SARS-CoV-2 populations and creates visualisations to interrogate these over time and by region. 
2,332 variants from the data version available on 2020/04/08 is displayed here.
</p>

<p>
  The SARS-CoV-2 genome sequence data used in the Nextstrain analysis were collected and made available by <a href='https://www.gisaid.org/'>GISAID</a>. Many groups contributed pre-publication sequence data to GISAID; please see the <a href='https://www.gisaid.org/'>GISAID site</a> for a full list of attributions.
</p>

<p>
  <b>See <a href='https://nextstrain.org/ncov/global'>Nextstrain</a> for the latest data and analysis</b>
</p>

<h2> COG UK </h2>
The <a href="http://sars2.cvr.gla.ac.uk/cog-uk/"> COG-UK Mutation Explorer </a> provides information on sequence variants discovered in data generated by the COVID-19 Genomics (COG-UK) Consortium. 
Reports focus on mutations in the spike gene and other variants of known or potential importance and are described by gene name and protein change. To link these variants to genomic information we map the descriptions to genome locations and display where novel (291 variants) or include the gene-protein based names as synonyms in records from other sources (1097 variants). 
HGVS-style names such as "S:p.N501Y" can be entered into the search function on the SARS-CoV-2 species page to retrieve variant records.
Data from 2021/04/14 is displayed here.


<h2> Variant reliability annotation</h2>
<p>Some loci are flagged as a further guide to quality:</p>
<p>
  <ul>
    <li>Variants seen in more than one sample in either set have an evidence status of 'Multiple observations'</li>
    <li>Variants at sites recommended for masking by <a href="https://virological.org/t/issues-with-sars-cov-2-sequencing-data/473/12">De Maio et al</a> (version 2020-07-29) have a flag of ‘Suspect reference location’</li>
  </ul>
</p>
