<p>SARS-CoV-2 mutation data is made available as a phylogenic tree by Nextstrain, which is an open-source project to harness the scientific and public health potential of pathogen genome data. The Nextstrain project uses subsamples of the available genome sequence data to provide a real-time snapshot of evolving SARS-CoV-2 populations and creates visualisations to interrogate these over time and by region. The 2020-04-08 version of http://data.nextstrain.org/ncov.json is used here.
The SARS-CoV-2 genome sequence data used in the Nextstrain analysis were collected and made available by GISAID. Many groups contributed pre-publication sequence data to GISAID;  please see metadata.tsv for a full list of attributions.
See https://nextstrain.org/ncov/global for the latest data and analysis.

Hadfield et al., Nextstrain: real-time tracking of pathogen evolution, Bioinformatics (2018)
Sagulenko et al, TreeTime: Maximum-likelihood phylodynamic analysis, Virus Evolution (2017)

</p>
