<p>
  We display 6,123 sequence variants derived by two different groups using two different analysis methods. There is overlap between the sample sets used in the two analyses.
</p>

<h4>Nextstrain</h4>

<p>
  SARS-CoV-2 mutation data is made available as a phylogenetic tree by <a href='https://nextstrain.org/'>Nextstrain</a>[Hadfield 2018], an open-source project to harness the scientific and public health potential of pathogen genome data. The Nextstrain project uses subsamples of the available genome sequence data to provide a real-time snapshot of evolving SARS-CoV-2 populations and creates visualisations to interrogate these over time and by region. A version of data available on 2020-04-08 is used.
</p>

<p>
  The SARS-CoV-2 genome sequence data used in the Nextstrain analysis were collected and made available by <a href='https://www.gisaid.org/'>GISAID</a>. Many groups contributed pre-publication sequence data to GISAID; please see the <a href='https://www.gisaid.org/'>GISAID site</a> for a full list of attributions.
</p>

<p>
  <b>See <a href='https://nextstrain.org/ncov/global'>Nextstrain</a> for the latest data and analysis</b>
</p>



<h4>European Nucleotide Archive (ENA)</h4>

<p>The ENA team has developed a <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3526318/">LoFreq</a>-based <a href="https://github.com/enasequence/covid-sequence-analysis-workflow/blob/master/workflow.nf">pipeline</a> to call variants from SARS-CoV-2 data submitted to the archive. This estimates the proportion of each variant allele within the sequenced sample. The results of preliminary analysis of the first available 5115 sequencing runs, created on 2020-08-17, is displayed here</p>

<p>Data filtering</p>

<p>As this is an early version of the ENA variant data strict, basic filters have been applied to reduce the proportion of lower confidence sites. We have removed:</p>
<p>
  <ul>
    <li>Sequencing runs with more than 40 variant calls </li>
    <li>Variants where no sample has a frequency of 20% or more for the non-reference allele</li>
    <li>Variants where all samples show strand bias
  </ul>
</p>

<p>Data representation</p>

<p>We represent only the alleles seen in each sample and not the frequencies observed within each sample. Variants were called for each sequencing run individually and for display it is assumed that sites at which a variant was not called in a sample match the reference. This will provide a more accurate estimation of the frequency of each allele across the entire sample set.</p>

<p>Some loci are flagged as a further guide to quality:</p>
<p>
  <ul>
    <li>Variants seen in more than one sample in either set have an evidence status of 'Multiple observations'</li>
    <li>Variants at sites recommended for masking by <a href="https://virological.org/t/issues-with-sars-cov-2-sequencing-data/473/12">De Maio et al</a> (version 2020-07-29) have a flag of ‘Suspect reference location’</li>
  </ul>
</p>
