<html>
<head>
  <title>Phenotype annotations</title>
</head>

<body>

<h1 style="margin-top:15px">Ensembl Variation - Phenotype and disease annotations</h1>

<p>Phenotypes are imported from a number of <a href="./sources_phenotype_documentation.html">external sources</a> and displayed in Ensembl. On top of these, for human we display the clinical significance of variants, and map phenotype terms using ontologies.</p>

<h2 id="clin_significance">Clinical significance</h2>

<p>See below the list of the clinical significance terms you can find in the human Ensembl Variation database:</p>

<!-- Data clin_significance - start -->
<!-- Data clin_significance - end -->

<br />
<hr style="margin-bottom:4px" />
<p></p>
<h2 id="phenotype_classes">Phenotype classes</h2>
<p>We group phenotype terms into the classes below: </p>
<!-- Data pheno_class - start -->

<table class="ss" style="width:auto">
  <tr> <th>Classes</th> <th  style="border-left:1px solid #BBB"> Example phenotypes</th> <th  style="border-left:1px solid #BBB">Example variants</th></tr>  <tr>
    <td><b>trait</b></td>
    <td style="border-left:1px solid #BBB">'BARDET-BIEDL SYNDROME 1' from <a href="https://www.ncbi.nlm.nih.gov/clinvar/">ClinVar</a></td>
    <td style="border-left:1px solid #BBB"><a href="/Homo_sapiens/Variation/Phenotype?v=rs771454836">rs771454836</a></td>
  </tr>  <tr class="bg2">
    <td><b>tumour</b></td>
    <td style="border-left:1px solid #BBB">'Prostate tumour' from <a href="https://cancer.sanger.ac.uk/cosmic">COSMIC</a></td>
    <td style="border-left:1px solid #BBB"><a href="/Homo_sapiens/Variation/Phenotype?v=COSV65031212">COSV65031212</a></td>
  </tr>  <tr>
    <td><b>non_specified</b></td>
    <td style="border-left:1px solid #BBB">'not specified' from <a href="https://www.ncbi.nlm.nih.gov/clinvar/">ClinVar</a></td>
    <td style="border-left:1px solid #BBB"><a href="/Homo_sapiens/Variation/Phenotype?v=rs2137519559">rs2137519559</a></td>
  </tr>
</table>
<!-- Data pheno_class - end -->

<p>These can be used to retrieve corresponding subsets of phenotype annotations for genes or in a specific region via the <a href="https://rest.ensembl.org/">REST</a> API.
 Phenotype pages are displayed for all classes except 'non_specified' class.</p>

<br />
<hr style="margin-bottom:4px" />
<p></p>
<h2 id="phenotype_ontologies">Phenotype/disease ontologies</h2>
<p>We import ontology terms related to phenotypes, traits and diseases from a variety of sources
using an automated process. Ontologies used are:</p>
<table class="ss" style="width:auto">
  <thead>
    <tr><th colspan="2">Ontology</th><th style="border-left:1px solid #BBB">Version/Downloaded</th></tr>
  </thead>
  <tbody>
    <tr>
      <td><b>CMO</b></td>
      <td><a href="https://www.ebi.ac.uk/ols/ontologies/cmo">Clinical Measurement Ontology</a></td>
      <td style="border-left:1px solid #BBB"><span id="cmo_version">2019-02-19</span></td>
    </tr>
    <tr class="bg2">
      <td><b>EFO</b></a></td>
      <td><a href="https://www.ebi.ac.uk/efo">Experimental Factor Ontology</a></td>
      <td style="border-left:1px solid #BBB"><span id="efo_version">3.49.0</span></td>
    </tr>
    <tr>
      <td><b>HPO</b></td>
      <td><a href="http://www.human-phenotype-ontology.org">Human Phenotype Ontology</a></td>
      <td style="border-left:1px solid #BBB"><span id="hp_version">2022-06-11</span></td>
    </tr>
    <tr class="bg2">
      <td><b>MP</b></td>
      <td><a href="http://www.informatics.jax.org/vocab/mp_ontology/">Mammalian Phenotype Ontology</a></td>
      <td style="border-left:1px solid #BBB"><span id="mp_version"></span></td>
    </tr>
    <tr>
      <td><b>VT</b></td>
      <td><a href="https://www.ebi.ac.uk/ols/ontologies/vt">Vertebrate Trait Ontology</a></td>
      <td style="border-left:1px solid #BBB"><span id="vt_version"></span></td>
    </tr>
    
  </tbody>
</table>

<p></p>
<p>
<h3>Descriptions are linked to ontology terms using:</h3>
<ul>
<li> Mappings provided by association data sources such as Orphanet, the NHGRI-EBI GWAS catalog and ClinVar</li>
<li> Annotations of OMIM terms created by HPO</li>
<li> Annotations of OMIM terms created by Orphanet</li>
<li><a href="http://www.ebi.ac.uk/ols/">Ontology LookUp Service</a> searches of full or truncated descriptions for exact matches to terms or synomyms</li>
<li><a href="http://www.ebi.ac.uk/spot/zooma/">Zooma</a> searches of annotations curated by the <a href="http://www.ebi.ac.uk/eva/">European Variation Archive</a> team</li>
</ul>

<p>
<h3>References</h3>
<ul>

<li> 
<p>Sebastian Köhler, Sandra C Doelken, Christopher J. Mungall, Sebastian Bauer, Helen V. Firth, Isabelle Bailleul-Forestier, Graeme C. M. Black, Danielle L. Brown, Michael Brudno, Jennifer Campbell, David R. FitzPatrick, Janan T. Eppig, Andrew P. Jackson, Kathleen Freson, Marta Girdea, Ingo Helbig, Jane A. Hurst, Johanna Jähn, Laird G. Jackson, Anne M. Kelly, David H. Ledbetter, Sahar Mansour, Christa L. Martin, Celia Moss, Andrew Mumford, Willem H. Ouwehand, Soo-Mi Park, Erin Rooney Riggs, Richard H. Scott, Sanjay Sisodiya, Steven Van Vooren, Ronald J. Wapner, Andrew O. M. Wilkie, Caroline F. Wright, Anneke T. Vulto-van Silfhout, Nicole de Leeuw, Bert B. A. de Vries, Nicole L. Washingthon, Cynthia L. Smith, Monte Westerfield, Paul Schofield, Barbara J. Ruef, Georgios V. Gkoutos, Melissa Haendel, Damian Smedley, Suzanna E. Lewis, and Peter N. Robinson</br>
<strong>The Human Phenotype Ontology project: linking molecular biology and disease through phenotype data</strong></br>
<em>Nucl. Acids Res. (1 January 2014) 42 (D1): D966-D974</em><br />
<a href="http://nar.oxfordjournals.org/content/42/D1/D966.long">doi:10.1093/nar/gkt1026</a></p>
</li>

<li> 
<p> Malone J, Holloway E, Adamusiak T, Kapushesky M, Zheng J, Kolesnikov N, Zhukova, A, Brazma A, Parkinson H.</br> 
<strong>Modeling sample variables with an Experimental Factor Ontology</strong></br>
<em>Bioinformatics (2010) 26 (8): 1112-1118 </em><br />
<a href="http://bioinformatics.oxfordjournals.org/content/26/8/1112.long">doi:10.1093/bioinformatics</a><p>
</li>

</body>
</html>
