<html>
<head>
<meta name="order" content="3" />
<title>Accessing Ensembl Data</title>
</head>

<body>

<h1>Accessing Ensembl Data</h1>

<p>Ensembl data is available through a number of routes - which you choose depends on
the amount and type of data you wish to fetch. Please note that Ensembl coordinates always have a one-based start.</p>

<h2>Small quantities of data</h2>
<p>Many of the pages displaying Ensembl genomic data offer an <a href="export.html">export</a> 
option, suitable for small amounts of data, e.g. a single gene sequence.</p>
<h2>Large datasets/complex analyses</h2>
<p>If you require larger amounts of data (e.g. all the genes on a chromosome) or are 
conducting a more detailed analysis, we recommend using our 
publicly-accessible <a href="mysql.html">MySQL server</a>. We provide a 
<a href="/info/docs/api/">Perl API</a> to these databases, so they can be 
scripted against without needing to know the database schema.</p>
<p>If you are not familiar with Perl, we recommend using our <a href=[[SPECIESDEFS::ENSEMBL_REST_URL]]>REST server</a>.
Various REST endpoints provide access to vast amounts of Ensembl data.</p>
<h2>Complex cross-database queries</h2>
<p>More complex datasets can be retrieved using the <a href="biomart/">BioMart</a> data-mining tool.</p>
<h2>Whole databases</h2>
<p>If required, entire databases can be downloaded from our <a href="ftp/">FTP site</a> 
in a variety of formats, from flat files to MySQL dumps.</p>

<div class="tinted-box center"><p>All data produced by the Ensembl project is
<a href="/info/about/legal/">freely available</a> for your own use.</p></div>

</body>
</html>
