<html>
<head>
<meta name="order" content="1" />
<title>Variation</title>
</head>

<body>

  <h1 style="margin-top:15px">About Ensembl Variation</h1>


  <p style="font-weight:bold">
    The Ensembl Variation database stores areas of the genome that differ
    between individual genomes ("variants") and, where available,
    associated disease and phenotype information.
  </p>
  <p>
    There are different types of variants for several <a href="./species/species_data_types.html#sources">species</a>:
  </p>
  <ul style="margin-top:2px">
    <li style="margin-top:4px">single nucleotide polymorphisms (SNPs)</li>
    <li>short nucleotide insertions and/or deletions</li>
    <li>longer variants classified as structural variants (including CNVs)</li>
  </ul>

  <p>Explore the links below to learn more about Ensembl Variation:</p>

  <div style="display:table;font-size:14px;margin:10px 0px 40px">
    <div style="display:table-row">

       <div style="display:table-cell">
        <div class="round-box plain-box bordered" style="margin-right:16px">
          <h3>Data access and tools</h3>
          <ul>
            <li><a href="./tools/data_access.html">Data access</a></li>
            <li><a href="./tools/variant_tools.html">Variant tools</a></li>
          </ul>
        </div>
      </div>


      <div style="display:table-cell">
        <div class="round-box plain-box bordered" style="margin-right:16px">
          <h3>Phenotype data</h3>
          <ul>
            <li><a href="./phenotype/phenotype_annotation.html">Phenotype and disease annotations</a></li>
            <li><a href="./phenotype/sources_phenotype_documentation.html">Phenotype sources</a></li>
          </ul>
        </div>
      </div>

    </div>
    <div style="display:table-row">

      <div style="display:table-cell">
        <div class="round-box plain-box bordered" style="margin-right:16px">
          <h3>Data prediction</h3>
          <ul>
            <li><a href="./prediction/predicted_data.html">Calculated consequences</a></li>
            <li><a href="./prediction/protein_function.html">Pathogenicity prediction</a></li>
            <li><a href="./prediction/classification.html">Variant classification</a></li>
            <li><a href="./prediction/variant_quality.html">Variant quality</a></li>
          </ul>
        </div>
      </div>

      <div style="display:table-cell">
        <div class="round-box plain-box bordered" style="margin-right:16px">
          <h3>Species data</h3>
          <ul>
            <li><a href="./species/sources_documentation.html">Data sources</a></li>
            <li><a href="./species/species_data_types.html">Data types and species list</a></li>
            <li><a href="./species/species_detailed_counts.html">Detailed species data count</a></li>
            <li><a href="./species/populations.html">Population allele frequencies and genotypes</a></li>
            <li><a href="./species/sets.html">Variant sets</a></li>
          </ul>
        </div>
      </div>

    </div>
  </div>



  <hr style="margin-bottom:4px" />
  <p></p>
  <h2 id="examples">Examples</h2>

  <p>
    For several different species in Ensembl,
    we import variant data (<span class="ht" title="Single Nucleotide Polymorphisms">SNPs</span>, <span class="ht" title="Copy Number Variations">CNVs</span>, allele frequencies, genotypes, etc) from a variety of
    <a href="./species/sources_documentation.html">sources</a> (e.g. dbSNP). Imported variants and alleles are subjected to a <a href="./prediction/variant_quality.html">quality control</a> process to flag suspect data.
  </p>
  <p>
   We classify the variants into different
    <a href="./prediction/classification.html">classes</a> and calculate the <a href="./prediction/predicted_data.html">predicted consequence(s)</a>
    of the variant and we have also created variant <a href="./species/sets.html">sets</a> to help people retrieve a specific
    group of variants from a particular dataset.
  </p>
  <p>
    In human, we calculate the linkage disequilibrium for each variant, by <a href="./species/populations.html">population</a>.
  </p>

  <p>See some examples of imported data on the Ensembl website (Human):</p>
  <div class="portal" style="margin-bottom:5px;width:auto">
    <div>
      <a href="/Homo_sapiens/Variation/Context?v=rs1333049" title="Location of a variant in the genome"><img src="/i/96/var_genomic_context.png" alt="Location of a variant in the genome"/></a>
    </div>
    <div>
      <a href="/Homo_sapiens/Variation/Population?v=rs699" title="Population genotypes and frequencies of a variant"><img src="/i/96/var_population_genetics.png" alt="Population genotypes and frequencies of a variant"/></a>
    </div>
    <div>
      <a href="/Homo_sapiens/Variation/Sample?v=rs699" title="Sample genotypes of a variant"><img src="/i/96/var_sample_genotypes.png" alt="Sample genotypes of a variant"/></a>
    </div>
    <div>
      <a href="/Homo_sapiens/Variation/Phenotype?v=rs1333049" title="Phenotype(s) associated with a variant"><img src="/i/96/var_phenotype_data.png"alt="Phenotype(s) associated with a variant"/></a>
    </div>
    <div>
      <a href="/Homo_sapiens/Variation/Citations?v=rs1333049" title="Citations"><img src="/i/96/var_citations.png" alt="Citations"/></a>
    </div>

  </div>
  <div class="clear"></div>


<hr style="margin-bottom:4px" />
<p></p>
<h2 id="references">References</h2>

<ul>

  <li>
   Fiona Cunningham, James E Allen, Jamie Allen, Jorge Alvarez-Jarreta, M Ridwan Amode, Irina M Armean, 
   Olanrewaju Austine-Orimoloye, Andrey G Azov, If Barnes, Ruth Bennett, Andrew Berry, Jyothish Bhai, Alexandra Bignell,   Konstantinos Billis, Sanjay Boddu, Lucy Brooks, Mehrnaz Charkhchi, Carla Cummins, Luca Da Rin Fioretto, 
   Claire Davidson, Kamalkumar Dodiya, Sarah Donaldson, Bilal El Houdaigui, Tamara El Naboulsi, Reham Fatima, 
   Carlos Garcia Giron, Thiago Genez, Jose Gonzalez Martinez, Cristina Guijarro-Clarke, Arthur Gymer, Matthew Hardy, 
   Zoe Hollis, Thibaut Hourlier, Toby Hunt, Thomas Juettemann, Vinay Kaikala, Mike Kay, Ilias Lavidas, Tuan Le, 
   Diana Lemos, José Carlos Marugán, Shamika Mohanan, Aleena Mushtaq, Marc Naven, Denye N Ogeh, Anne Parker, 
   Andrew Parton, Malcolm Perry, Ivana Piližota, Irina Prosovetskaia, Manoj Pandian Sakthivel, Ahamed Imran Abdul Salam,   Bianca M Schmitt, Helen Schuilenburg, Dan Sheppard, José G Pérez-Silva, William Stark, Emily Steed, Kyösti Sutinen, 
   Ranjit Sukumaran, Dulika Sumathipala, Marie-Marthe Suner, Michal Szpak, Anja Thormann, Francesca Floriana Tricomi, 
   David Urbina-Gómez, Andres Veidenberg, Thomas A Walsh, Brandon Walts, Natalie Willhoft, Andrea Winterbottom, 
   Elizabeth Wass, Marc Chakiachvili, Bethany Flint, Adam Frankish, Stefano Giorgetti, Leanne Haggerty, Sarah E Hunt, 
   Garth R IIsley, Jane E Loveland, Fergal J Martin, Benjamin Moore, Jonathan M Mudge, Matthieu Muffato, Emily Perry, 
   Magali Ruffier, John Tate, David Thybert, Stephen J Trevanion, Sarah Dyer, Peter W Harrison, Kevin L Howe, 
   Andrew D Yates, Daniel R Zerbino, Paul Flicek <br />

    <strong>Ensembl 2022</strong><br />
    <em>Nucleic Acids Res. 2022, vol. 50(1):988–995</em><br />
     PubMed PMID: 34791404</br >
    <a href="https://doi.org/10.1093/nar/gkab1049" rel="external">doi:10.1093/nar/gkab1049</a>
  </li>

  <li>
    Sarah E Hunt, William McLaren, Laurent Gil, Anja Thormann, Helen Schuilenburg, Dan Sheppard, Andrew Parton, 
    Irina M Armean, Stephen J Trevanion, Paul Flicek, Fiona Cunningham<br />
     <strong>Ensembl variation resources</strong><br />
    <em>Database</em> Volume 2018<br />
    <a href="https://doi.org/10.1093/database/bay119" rel="external">doi:10.1093/database/bay119</a>
  </li>

  <li>
    William Spooner, William McLaren, Timothy Slidel, Donna K. Finch, Robin Butler, Jamie Campbell, Laura Eghobamien, 
    David Rider, Christine Mione Kiefer, Matthew J. Robinson, Colin Hardman, Fiona Cunningham, Tristan Vaughan, 
    Paul Flicek and Catherine Chaillan Huntington<br />
     <strong>Haplosaurus computes protein haplotypes for use in precision drug design.</strong><br />
    <em>Nature Communications</em> 9:4128(2018)<br />
    <a href="https://doi.org/10.1038/s41467-018-06542-1" rel="external">doi:10.1038/s41467-018-06542-1</a>
  </li>

  <li>
    <p>
      McLaren W, Gil L, Hunt SE, Riat HS, Ritchie GRS, Thormann A, Flicek P and Cunningham F.<br />
      <strong>The Ensembl Variant Effect Predictor</strong><br />
      <em>Genome Biology</em> 17:122(2016)<br />
      <a href="http://dx.doi.org/10.1186/s13059-016-0974-4" rel="external">doi:10.1186/s13059-016-0974-4</a>
    </p>
  </li>
</ul>

</body>
</html>
