<html>
<head>
<meta name="order" content="1" />
<title>Variation</title>
</head>

<body>

  <h1 style="margin-top:15px">About Ensembl Variation</h1>


  <p style="font-weight:bold">
    The Ensembl Variation database stores areas of the genome that differ
    between individual genomes ("variants") and, where available,
    associated disease and phenotype information.
  </p>
  <p>
    There are different types of variants for several <a href="./species/species_data_types.html#sources">species</a>:
  </p>
  <ul style="margin-top:2px">
    <li style="margin-top:4px">single nucleotide polymorphisms (SNPs)</li>
    <li>short nucleotide insertions and/or deletions</li>
    <li>longer variants classified as structural variants (including CNVs)</li>
  </ul>

  <p>Explore the links below to learn more about Ensembl Variation:</p>

  <div style="display:table;font-size:14px;margin:10px 0px 40px">
    <div style="display:table-row">

       <div style="display:table-cell">
        <div class="round-box plain-box bordered" style="margin-right:16px">
          <h3>Data access and tools</h3>
          <ul>
            <li><a href="./tools/data_access.html">Data access</a></li>
            <li><a href="./tools/variant_tools.html">Variant tools</a></li>
          </ul>
        </div>
      </div>


      <div style="display:table-cell">
        <div class="round-box plain-box bordered" style="margin-right:16px">
          <h3>Phenotype data</h3>
          <ul>
            <li><a href="./phenotype/phenotype_annotation.html">Phenotype and disease annotations</a></li>
            <li><a href="./phenotype/sources_phenotype_documentation.html">Phenotype sources</a></li>
          </ul>
        </div>
      </div>

    </div>
    <div style="display:table-row">

      <div style="display:table-cell">
        <div class="round-box plain-box bordered" style="margin-right:16px">
          <h3>Data prediction</h3>
          <ul>
            <li><a href="./prediction/predicted_data.html">Calculated consequences</a></li>
            <li><a href="./prediction/protein_function.html">Protein function prediction</a></li>
            <li><a href="./prediction/classification.html">Variant classification</a></li>
            <li><a href="./prediction/variant_quality.html">Variant quality</a></li>
          </ul>
        </div>
      </div>

      <div style="display:table-cell">
        <div class="round-box plain-box bordered" style="margin-right:16px">
          <h3>Species data</h3>
          <ul>
            <li><a href="./species/sources_documentation.html">Data sources</a></li>
            <li><a href="./species/species_data_types.html">Data types and species list</a></li>
            <li><a href="./species/species_detailed_counts.html">Detailed species data count</a></li>
            <li><a href="./species/populations.html">Population allele frequencies and genotypes</a></li>
            <li><a href="./species/sets.html">Variant sets</a></li>
          </ul>
        </div>
      </div>

    </div>
  </div>



  <hr style="margin-bottom:4px" />
  <p></p>
  <h2 id="examples">Examples</h2>

  <p>
    For several different species in Ensembl,
    we import variant data (<span class="ht" title="Single Nucleotide Polymorphisms">SNPs</span>, <span class="ht" title="Copy Number Variations">CNVs</span>, allele frequencies, genotypes, etc) from a variety of
    <a href="./species/sources_documentation.html">sources</a> (e.g. dbSNP). Imported variants and alleles are subjected to a <a href="./prediction/variant_quality.html">quality control</a> process to flag suspect data.
  </p>
  <p>
   We classify the variants into different
    <a href="./prediction/classification.html">classes</a> and calculate the <a href="./prediction/predicted_data.html">predicted consequence(s)</a>
    of the variant and we have also created variant <a href="./species/sets.html">sets</a> to help people retrieve a specific
    group of variants from a particular dataset.
  </p>
  <p>
    In human, we calculate the linkage disequilibrium for each variant, by <a href="./species/populations.html">population</a>.
  </p>

  <p>See some examples of imported data on the Ensembl website (Human):</p>
  <div class="portal" style="margin-bottom:5px;width:auto">
    <div>
      <a href="/Homo_sapiens/Variation/Context?v=rs1333049" title="Location of a variant in the genome"><img src="/i/96/var_genomic_context.png" alt="Location of a variant in the genome"/></a>
    </div>
    <div>
      <a href="/Homo_sapiens/Variation/Population?v=rs699" title="Population genotypes and frequencies of a variant"><img src="/i/96/var_population_genetics.png" alt="Population genotypes and frequencies of a variant"/></a>
    </div>
    <div>
      <a href="/Homo_sapiens/Variation/Sample?v=rs699" title="Sample genotypes of a variant"><img src="/i/96/var_sample_genotypes.png" alt="Sample genotypes of a variant"/></a>
    </div>
    <div>
      <a href="/Homo_sapiens/Variation/Phenotype?v=rs1333049" title="Phenotype(s) associated with a variant"><img src="/i/96/var_phenotype_data.png"alt="Phenotype(s) associated with a variant"/></a>
    </div>
    <div>
      <a href="/Homo_sapiens/Variation/Citations?v=rs1333049" title="Citations"><img src="/i/96/var_citations.png" alt="Citations"/></a>
    </div>

  </div>
  <div class="clear"></div>


<hr style="margin-bottom:4px" />
<p></p>
<h2 id="references">References</h2>

<ul>

  <li>
    Daniel R Zerbino, Premanand Achuthan, Wasiu Akanni, M Ridwan Amode, Daniel Barrell, 
    Jyothish Bhai, Konstantinos Billis, Carla Cummins, Astrid Gall, Carlos Garc&iacute;a Gir&oacute;n, 
    Laurent Gil, Leo Gordon, Leanne Haggerty, Erin Haskell, Thibaut Hourlier, Osagie G Izuogu, 
    Sophie H Janacek, Thomas Juettemann, Jimmy Kiang To, Matthew R Laird, Ilias Lavidas, 
    Zhicheng Liu, Jane E Loveland, Thomas Maurel, William McLaren, Benjamin Moore, Jonathan Mudge, 
    Daniel N Murphy, Victoria Newman, Michael Nuhn, Denye Ogeh, Chuang Kee Ong, Anne Parker, 
    Mateus Patricio, Harpreet Singh Riat, Helen Schuilenburg, Dan Sheppard, Helen Sparrow, 
    Kieron Taylor, Anja Thormann, Alessandro Vullo, Brandon Walts, Amonida Zadissa, Adam Frankish, 
    Sarah E Hunt, Myrto Kostadima, Nicholas Langridge, Fergal J Martin, Matthieu Muffato, 
    Emily Perry, Magali Ruffier, Dan M Staines, Stephen J Trevanion, Bronwen L Aken, Fiona Cunningham, 
    Andrew Yates, and Paul Flicek<br />

    <strong>Ensembl 2018</strong><br />
    <em>Nucleic Acids Research</em><br />
    <a href="https://doi.org/10.1093/nar/gkx1098" rel="external">doi:10.1093/nar/gkx1098</a>    
  </li>

  <li>
    <p>
      McLaren W, Gil L, Hunt SE, Riat HS, Ritchie GRS, Thormann A, Flicek P and Cunningham F.<br />
      <strong>The Ensembl Variant Effect Predictor</strong><br />
      <em>Genome Biology</em> 17:122(2016)<br />
      <a href="http://dx.doi.org/10.1186/s13059-016-0974-4" rel="external">doi:10.1186/s13059-016-0974-4</a>
    </p>
  </li>

  <li>
    <p>
    Rios D, McLaren WM, Chen Y, Birney E, Stabenau A, Flicek P, Cunningham F.<br />
    <strong>A Database and API for variation, dense genotyping and resequencing data</strong><br />
    <em>BMC Bioinformatics</em> 11:238 (2010)<br />
    <a href="http://dx.doi.org/10.1186/1471-2105-11-238"
    rel="external">doi:10.1186/1471-2105-11-238</a></p>
  </li>

  <li>
    <p>Chen Y, Cunningham F, Rios D, McLaren WM, Smith J, Pritchard B, Spudich GM, Brent S, Kulesha E, Marin-Garcia P, Smedley D, Birney E, Flicek P.<br />
    <strong>Ensembl Variation Resources</strong><br />
    <em>BMC Genomics</em> 11(1):293 (2010)<br />
    <a href="http://dx.doi.org/10.1186/1471-2164-11-293"
    rel="external">doi:10.1186/1471-2164-11-293</a></p>
  </li>
</ul>

</body>
</html>
