<html>
<head>
<title>Annotation of Ig/Tcr Segment Genes</title>
</head>

<body>

<h1>Annotation of Ig/Tcr Segment Genes</h1>

<p>
Ensembl Ig Segment genes are annotations of the Immunoglobulin and
T-cell receptor genes present in germ-line genomic DNA. During B-cell
and T-cell maturation, these genes are rearranged and brought together
by V-D-J recombination to form functional Immunoglobulin and T-cell
receptor genes.
</p>

<p>
The protein and nucleotide databases contain many sequences of mature
Immunoglobulin/T-cell receptor genes. Inferring gene structures by
aligning these to the genome using the standard
Ensembl <a href="genome_annotation.html">genebuild</a> pipeline gives
unsatisfactory results, for two main reasons:
</p>

<ul>

<li>
The non-coding DNA that separates pairs of gene segments brought
together by V-D-J recombination  is not an intron, and the gene
boundaries display different sequence signals to splice sites.
Spliced alignment programs such as 
<a rel="external"
href="http://dx.doi.org/10.1101/gr.1865504">GeneWise</a> therefore
require retuning in order to predict the gene structure correctly.
</li>

<li>
Some genes comprise multiple exons. Delineating a "spliced" alignment
of a mature Immunoglobulin/T-cell receptor sequence to the genome into
its constituent segments is therefore not straightforward.
</li>

</ul>

<p>
For this reason, Immunoglobulin/T-cell receptor gene segments have
been annotated differently to other Ensembl genes. Ensembl makes use
of the 
<a rel="external" href="http://www.imgt.org/">IMGT database</a>, which
provides annotation of individual gene segments on reference
sequences. Ensembl extracts individual gene sequences and aligns them
to the genome using the
<a rel="external"
href="http://dx.doi.org/10.1186/1471-2105-6-31">Exonerate</a>
alignment tool.
</p>

<p>
To reflect their role in the V-D-J recombination process, segments are
annotated as belonging to one of four classes: V segment, D segment, J
segment and C segment.  At present, Ensembl Ig gene annotations are
only available for human and mouse.
</p>

<h2>References</h2>

<p>
<span class="ref-author">Lefranc M.-P., Giudicelli V., Kaas Q., Duprat
E., Jabado-Michaloud J., Scaviner D., Ginestoux C., Cl&eacute;ment O.,
Chaume D. and Lefranc G.</span><br />  IMGT, the international
ImMunoGeneTics information system.<br />
<span class="ref-pub">Nucl. Acids Res.</span> 2005 33:D593-D597
</p>

<p>
<span class="ref-author">Slater G.S. and Birney E.</span><br />
Automated generation of heuristics for biological sequence
comparison.<br />
<span class="ref-pub">BMC Bioinformatics</span> 2005 6:31<br />
<a href="http://dx.doi.org/10.1186/1471-2105-6-31"
rel="external">doi:10.1186/1471-2105-6-31</a>
</p>

<p>
<span class="ref-author">Ewan Birney, Michele Clamp and Richard
Durbin</span><br />
<strong>GeneWise and Genomewise</strong><br />
<span class="ref-pub">Genome Research</span> 2004 14(5):988-995.
<a href="http://dx.doi.org/10.1101/gr.1865504"
rel="external">doi:10.1101/gr.1865504</a><br />
</p>

</body>
</html>
