<html>
<head>
  <title>Download and install</title>
  <meta name="order" content="2" />
  <script type="text/javascript">
    // Function to show/hide divs
    function show_hide (param) {
      div   = document.getElementById('div_'+param);
      alink = document.getElementById('a_'+param);
      if (div.style.display=='inline') {
        div.style.display='none';
        alink.innerHTML= alink.innerHTML.replace("Hide","Show");
      }
      else {
        if (div.style.display=='none') {
          div.style.display='inline';
          alink.innerHTML= alink.innerHTML.replace("Show", "Hide");
        }
      }
    }
  </script>
</head>

<body>

<style>
tr:nth-child(odd) {background-color: #f0f0f0;}
code {background-color: rgb(240, 240, 240); font-family: Courier New, Courier, monospace;}
</style>
<div>
  <div style="float:right"><img src="/img/vep_logo.png"/></div>

  <h1 id="top"><span style="color:#006;padding-right:15px">Variant Effect Predictor</span><span style="color:#666"><img src="/i/16/download.png"> Download and install</span></h1>
  <hr/>

  <h2 id="download">Download</h2>

  <p>Download ensembl-vep package (see below the different ways to download it) and then follow the <a href="#installer">installation instructions</a>.</p>

  <br />
  <h3>Using Git</h3>
  <h4><img src="/i/bullet_1.gif" style="vertical-align:baseline"/> Clone the Git repository</h4>
  <p> Use git to download the ensembl-vep package:</p>

  <pre class="code sh_sh">git clone https://github.com/Ensembl/ensembl-vep.git
cd ensembl-vep</pre>


<br />
  <h4><img src="/i/bullet_1.gif" style="vertical-align:baseline"/> Update to a newer version</h4>
  <p>To update from a previous version: </p>

  <pre class="code sh_sh">cd ensembl-vep
git pull
git checkout release/[[SPECIESDEFS::ENSEMBL_VERSION]]
perl INSTALL.pl</pre>

  <br />
  <h4><img src="/i/bullet_1.gif" style="vertical-align:baseline"/> Use an older version</h4>
  <p> To use an older version (this example shows how to set up release 87): </p>

  <pre class="code sh_sh">cd ensembl-vep
git checkout release/87
perl INSTALL.pl</pre>


  <br />
  <h3>Download the Zipped package file</h3>
  <p> Users without the git utility installed may download a zip file from GitHub, though we would always recommend using git if possible.</p>

  <pre class="code sh_sh" id="nogit">curl -L -O https://github.com/Ensembl/ensembl-vep/archive/release/[[SPECIESDEFS::ENSEMBL_VERSION]].zip
unzip [[SPECIESDEFS::ENSEMBL_VERSION]].zip
cd ensembl-vep-release-[[SPECIESDEFS::ENSEMBL_VERSION]]/</pre>

  <br />
  <h3 id="toolsversions"> Previous versions (ensembl-tools)</h3>
  <p>
    Previously VEP was available as part of the ensembl-tools package (see the <a href="http://e87.ensembl.org/info/docs/tools/vep/script/index.html">Ensembl archive site</a> for documentation). The following downloads are available for archival purposes. <a class="button" href="#toolsversions" onclick="show_hide('toolsversions');" id="a_toolsversions">Show versions</a>
  </p>

  <div id="div_toolsversions" style="display:none;">
  <ul>
    <li><a href="https://github.com/Ensembl/ensembl-tools/archive/release/87.zip">Download version 87</a> (Ensembl 87) </li>
    <li><a href="https://github.com/Ensembl/ensembl-tools/archive/release/86.zip">Download version 86</a> (Ensembl 86) </li>
    <li><a href="https://github.com/Ensembl/ensembl-tools/archive/release/85.zip">Download version 85</a> (Ensembl 85) </li>
    <li><a href="https://github.com/Ensembl/ensembl-tools/archive/release/84.zip">Download version 84</a> (Ensembl 84) </li>
    <li><a href="https://github.com/Ensembl/ensembl-tools/archive/release/83.zip">Download version 83</a> (Ensembl 83) </li>
    <li><a href="https://github.com/Ensembl/ensembl-tools/archive/release/82.zip">Download version 82</a> (Ensembl 82) </li>
    <li><a href="https://github.com/Ensembl/ensembl-tools/archive/release/81.zip">Download version 81</a> (Ensembl 81) </li>
    <li><a href="https://github.com/Ensembl/ensembl-tools/archive/release/80.zip">Download version 80</a> (Ensembl 80) </li>
    <li><a href="https://github.com/Ensembl/ensembl-tools/archive/release/79.zip">Download version 79</a> (Ensembl 79) </li>
    <li><a href="https://github.com/Ensembl/ensembl-tools/archive/release/78.zip">Download version 78</a> (Ensembl 78) </li>
    <li><a href="https://github.com/Ensembl/ensembl-tools/archive/release/77.zip">Download version 77</a> (Ensembl 77) </li>
    <li><a href="https://github.com/Ensembl/ensembl-tools/archive/release/76.zip">Download version 76</a> (Ensembl 76) </li>
    <li><a href="https://github.com/Ensembl/ensembl-tools/archive/release/75.zip">Download version 75</a> (Ensembl 75) </li>
    <li><a href="https://github.com/Ensembl/ensembl-tools/archive/release/74.zip">Download version 74</a> (Ensembl 74) </li>
    <li><a href="https://github.com/Ensembl/ensembl-tools/archive/release/73.zip">Download version 73</a> (Ensembl 73) </li>
    <li><a href="https://github.com/Ensembl/ensembl-tools/archive/release/72.zip">Download version 72</a> (Ensembl 72) </li>
    <li><a href="https://github.com/Ensembl/ensembl-tools/archive/release/71.zip">Download version 71</a> (Ensembl 71) </li>
    <li><a href="https://github.com/Ensembl/ensembl-tools/archive/release/70.zip">Download version 2.8</a> (Ensembl 70) </li>
    <li><a href="https://github.com/Ensembl/ensembl-tools/archive/release/69.zip">Download version 2.7</a> (Ensembl 69) </li>
    <li><a href="https://github.com/Ensembl/ensembl-tools/archive/release/68.zip">Download version 2.6</a> (Ensembl 68) </li>
    <li><a href="https://github.com/Ensembl/ensembl-tools/archive/release/67.zip">Download version 2.5</a> (Ensembl 67) </li>
    <li><a href="https://github.com/Ensembl/ensembl-tools/archive/release/66.zip">Download version 2.4</a> (Ensembl 66) </li>
    <li><a href="https://github.com/Ensembl/ensembl-tools/archive/release/65.zip">Download version 2.3</a> (Ensembl 65) </li>
    <li><a href="https://github.com/Ensembl/ensembl-tools/archive/release/64.zip">Download version 2.2</a> (Ensembl 64 - ensembl-tools/scripts/variant_effect_predictor) </li>
    <li><a href="https://github.com/Ensembl/ensembl-variation/archive/release/63.zip">Download version 2.1</a> (Ensembl 63) </li>
    <li><a href="https://github.com/Ensembl/ensembl-variation/archive/release/62.zip">Download version 2.0</a> (Ensembl 62 - ensembl-variation/scripts/examples) </li>
  </ul>
  </div><hr/>

  <h2 id="new">What's new</h2>

  <h4>New in version [[SPECIESDEFS::ENSEMBL_VERSION]] <i>([[SPECIESDEFS::ENSEMBL_RELEASE_DATE]])</i></h4>

  <ul>
    <li>The VEP parser is now more permissive for the GFF files (ID attribute only required for genes and transcripts)</li>
    <li>Add new option <a href="vep_options.html#opt_show_ref_allele">--show_ref_allele</a> to include the allele reference in the VEP default output and the tab output formats</li>
    <li>Add a warning message when the VEP annotations INFO field hasn't been found/recognised in the VCF input file</li>
    <li>VEP Docker image:
      <ul>
        <li>Reduce the size of the VEP Docker image by about 45%.</li>
        <li>Include the Linkage disequilibrium script in the VEP Docker image, making possible to run the LD plugin</li>
      </ul>
    </li>
    <li>New VEP plugins:
      <ul>
        <li>
          <a href="//github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/ReferenceQuality.pm" rel="external">Reference quality</a>
        </li>
        <li>
          <a href="//github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/POSTGAP.pm" rel="external">OpenTargets results (POSTGAP)</a>
        </li>
        <li>
          <a href="//github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/SingleLetterAA.pm" rel="external">Single letter amino acid for HGVS</a>
        </li>
      </ul>
    </li>
  </ul>


  <br />
  <h4 id="history">Previous version history: <a class="button" href="#history" onclick="show_hide('history');" id="a_history" style="font-weight:normal">Show</a></h4>

  <div id="div_history" style="display:none">

  <h4>New in version 94 <i>(October 2018)</i></h4>

  <ul>
    <li>RefSeq transcript version updated.</li>
    <li>Minor updates on the <a href="/Tools/VEP">VEP web tool</a> interface.</li>
    <li>When the input data format is not specified on the command line, VEP attempts to detect it. The assumed format is now reported in verbose mode (<a href="vep_options.html#opt_verbose">--verbose</a>).</li>
    <li>VEP assigns assigned the consequence types <i>TF_binding_site_variant</i>, <i>TFBS_ablation</i>, <i>TFBS_fusion</i>, <i>TFBS_amplification</i> and <i>TFBS_translocation</i> to human and mouse variants which overlapped motif features. These annotations will not be available in VEP caches for human in release 94 so must be added as a <a href="/info/docs/tools/vep/script/vep_custom.html">custom annotation</a>.</li>
  </ul>


  <br />
  <h4>New in version 93 <i>(July 2018)</i></h4>

  <ul>
    <li>Update the JSON output format (allele frequencies) for the <a href="//rest.ensembl.org/#VEP" rel="external">Ensembl REST - VEP</a> endpoints. <a href="//github.com/Ensembl/ensembl-rest/wiki/Change-log#70---2018-06" rel="external">See more information</a>.</li>
    <li>The new Ensembl release brings more frequency data from <a href="http://gnomad.broadinstitute.org/" rel="external">gnomAD</a>.</li>
    <li>Add the possibility to print the content of the FILTER column (from the VCF custom annotation files) in the output.</li>
    <li>Include the <a href="//github.com/Ensembl/ensembl-xs" rel="external">Ensembl/ensembl-xs</a> repository in Docker image to speed up the VEP container.</li>
    <li>Add a new consequence 'extended_intronic_splice_region_variant' in the <a href="//github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/SpliceRegion.pm" rel="external">SpliceRegion</a> VEP plugin.</li>
  </ul>
 

  <br />
  <h4>New in version 92 <i>(April 2018)</i></h4>

  <ul>
    <li>New VEP plugin <a href="//www.ncbi.nlm.nih.gov/pubmed/27666373" rel="external">REVEL</a> (see <a href="//github.com/Ensembl/VEP_plugins/blob/release/92/REVEL.pm" rel="external">REVEL plugin</a>).</li>
    <li>Get ambiguity code with <a href="vep_options.html#opt_ambiguity">--ambiguity</a>.</li>
    <li><a href="vep_cache.html#gff">GFF/GTF files</a> with exons assigned to multiple transcripts are now supported.</li>
    <li>Improved 1000 Genomes Project frequencies.</li>
  </ul>


  <br />
  <h4>New in version 91 <i>(December 2017)</i></h4>

  <ul>
    <li>New input format "<a href="../vep_formats.html#region">region</a>" allows REST-style input to VEP.</li>
    <li>Replace your input variant reference allele with the correct one from the genome with <a href="vep_options.html#opt_lookup_ref">--lookup_ref</a>.</li>
    <li>Add version numbers to Ensembl transcripts with <a href="vep_options.html#opt_transcript_version">--transcript_version</a>.</li>
  </ul>
 

  <br />
  <h4>New in version 90 <i>(August 2017)</i></h4>

  <ul>
    <li><a href="http://gnomad.broadinstitute.org/" rel="external">gnomAD</a> exomes allele frequencies now available with <a href="vep_options.html#opt_af_gnomad">--af_gnomad</a>, replacing ExAC. gnomAD genomes and ExAC are <a href="vep_example.html#gnomad">available via custom annotation</a>.</li>
    <li>VEP is now available as a <a href="#docker">Docker image</a>.</li>
    <li>RefSeq transcripts in VEP cache files are now <a href="vep_other.html#refseq">"corrected" from the reference genome sequence</a>.</li>
    <li>VEP's algorithm for matching colocated known variants has been overhauled - <a href="vep_other.html#colocated">details</a>.</li>
    <li>Change VEP's default (5kb) up/downstream distance with <a href="vep_options.html#opt_distance">--distance</a>. This supercedes the functionality of the UpDownDistance VEP plugin.</li>
    <li>Feed input directly to VEP with <a href="vep_options.html#opt_input_data">--input_data</a>.</li>
    <li>Suppress header output with <a href="vep_options.html#opt_no_headers">--no_headers</a>.</li>
    <li>Detailed <a href="#bigfile">installation instructions for Bio::DB::BigFile</a> to access bigWig custom annotation files.</li>
  </ul>


  <br />
  <h4> New in version 89 <i>(May 2017)</i> </h4>

  <ul>
    <li>exclude known variants with unknown (null) alleles with <a href="vep_options.html#opt_exclude_null_alleles">--exclude_null_alleles</a>.</li>
    <li>write compressed output with <a href="vep_options.html#opt_compress_output">--compress_output</a>.</li>
    <li>improved matching of alleles in <a href="vep_custom.html">custom VCF files</a>.</li>
    <li>API perldoc documentation added.</li>
  </ul>

 
  <br />
  <h4> New in version 88 <i>(March 2017)</i> </h4>

  <ul>
    <li>ensembl-vep is now the officially supported version of VEP</li>
    <li>Documentation updated to reflect switch to ensembl-vep. See the <a href="http://e87.ensembl.org/info/docs/tools/vep/script/index.html">Ensembl archive site</a> for documentation of the obsolete ensembl-tools VEP.</li>
    <li>The VEP script is now named simply <code>vep</code> (formerly <code>variant_effect_predictor.pl</code> or <code>vep.pl</code>)</li>
    <li>Directly use tabix-indexed <a href="vep_cache.html#gff">GFF/GTF files as annotation sources</a></li>
    <li>Allele-specific reporting of frequencies (<a href="vep_options.html#opt_af">--af</a> and more) and <a href="vep_custom.html">custom VCF annotations</a></li>
    <li><a href="vep_options.html#opt_check_existing">--check_existing</a> now compares alleles by default, disable with <a href="vep_options.html">--no_check_alleles</a></li>
    <li>Report the highest allele frequency observed in any population from 1000 genomes, ESP or ExAC using <a href="vep_options.html#opt_max_af">--max_af</a></li>
    <li>Get genomic HGVS nomenclature with <a href="vep_options.html#opt_hgvsg">--hgvsg</a></li>
    <li>Find the gene or transcript with the nearest transcription start site (TSS) to each input variant with <a href="vep_options.html#opt_nearest">--nearest</a></li>
    <li><a href="vep_filter.html">filter_vep</a> supports field/field comparisons e.g. AFR_AF > #EUR_AF</li>
    <li>Exclude predicted (XM and XR) transcripts when using RefSeq or merged cache with <a href="vep_options.html#opt_exclude_predicted">--exclude_predicted</a></li>
    <li>Filter transcripts used for annotation with <a href="vep_options.html#opt_transcript_filter">--transcript_filter</a></li>
    <li>pileup input format no longer supported</li>
  </ul>

  <hr/>
  <h4>ensembl-tools</h4>
  <p> Versions of VEP up to and including 87 were released as part of the ensembl-tools package. See <a href="#toolsversions">download links</a> above. </p>

  <h4> New in version 87 <i>(December 2016)</i> </h4>

  <ul>
    <li><a rel="external" href="https://github.com/Ensembl/ensembl-vep">Shiny new code</a> available for beta testing!</li>
    <li>Some minor speed optimisations</li>
    <li>Improve checks for valid chromosome names in input</li>
    <li><a rel="external" href="https://github.com/Ensembl/ensembl-vep#haplo">Haplosaurus</a> beta released - generate whole-transcript haplotype sequences from phased genotype data</li>
  </ul>

  <h4> New in version 86 <i>(October 2016)</i> </h4>

  <ul>
    <li>Chromosome synonyms supported when using VEP caches; may be loaded manually with <a href="vep_options.html#opt_synonyms">--synonyms</a></li>
  </ul>

  <h4> New in version 85 <i>(July 2016)</i> </h4>

  <ul>
    <li><a href="vep_options.html#opt_pick">--pick</a> now uses translated length instead of genomic transcript length</li>
    <li>Support for epigenomes in regulatory features</li>
  </ul>

  <h4> New in version 84 <i>(March 2016)</i> </h4>

  <ul>
    <li>Add <a href="/info/docs/tools/vep/vep_formats.html#tab">tab-delimited</a> output option</li>
    <li>Add <a href="/info/docs/tools/vep/vep_formats.html#output">transcript flags</a> indicating if the transcript is 5'- or 3'-incomplete</li>
    <li>Improve annotation of long variants where invariant parts of the alternate allele overlap splice regions</li>
  </ul>

  <h4> New in version 83 <i>(December 2015)</i> </h4>

  <ul>
    <li>
      Speed:
      <ul>
        <li>Basic consequence calculations up to 2x faster than version 82</li>
        <li>HGVS calculations up to 10x faster</li>
        <li>FASTA sequence retrieval implements caching</li>
      </ul>
    </li>
    <li>Add <a href="http://exac.broadinstitute.org/" rel="external">ExAC project</a> frequencies with <a href="vep_options.html#opt_maf_exac">--maf_exac</a></li>
    <li><a href="/Help/Glossary?id=521">APPRIS</a> isoform annotations now available with <a href="vep_options.html#opt_appris">--appris</a> and used by <a href="vep_options.html#opt_pick">--pick</a> and others to prioritise VEP annotations</li>
  </ul>

  <h4> New in version 82 <i>(September 2015)</i> </h4>

  <ul>
    <li><a href="#installer">Faster FASTA file access</a> using Bio::DB::HTS/htslib and bgzipped FASTA files</li>
    <li><a href="vep_options.html#opt_gene_phenotype">Flag genes</a> with phenotype associations</li>
    <li>Some plugins now available for use via the <a href="/Tools/VEP">web</a> and <a rel="external" href="http://rest.ensembl.org/#VEP">REST</a> interfaces</li>
  </ul>

  <h4> New in version 81 <i>(July 2015)</i> </h4>

  <ul>
    <li>Plugin registry means plugins can be installed from the <a href="#installer">VEP installer</a></li>
    <li>GFF format now supported by VEP's <a href="vep_cache.html#gtf">cache converter</a></li>
    <li>Fixes and improvements for sequence retrieval from FASTA files</li>
  </ul>

  <h4> New in version 80 <i>(May 2015)</i> </h4>

  <ul>
    <li><a href="../vep_formats.html#output">Flag added</a> indicating if an overlapping known variant is associated with a phenotype, disease or trait</li>
    <li>HGVS notations are now 3'-shifted by default (use <a href="vep_options.html#opt_shift_hgvs">--shift_hgvs</a> to force enable/disable)</li>
    <li>Source version information added to caches; see output file headers or use <a href="vep_options.html#opt_show_cache_info">--show_cache_info</a></li>
    <li>Get the variant class using <a href="vep_options.html#opt_variant_class">--variant_class</a></li>
    <li>CCDS status added to categories used by <a href="vep_options.html#opt_pick">--pick</a> flag (and <a href="vep_other.html#pick">others</a>)</li>
  </ul>

  <h4> New in version 79 <i>(March 2015)</i> </h4>

  <ul>
    <li>Focus on performance and stability: ~100% faster than version 78 and a new test suite</li>
    <li>New guide to <a href="vep_other.html#faster">getting VEP running faster</a></li>
    <li>1000 Genomes Phase 3 data available in GRCh37 cache download (GRCh38 coming soon, see <a href="vep_example.html#1kg_p3">docs</a> to access now)</li>
    <li><a href="../vep_formats.html#vcfout">VCF output</a> has changed slightly to match output from other tools</li>
    <li>Impact modifier added for each consequence type</li>
  </ul>

  <h4> New in version 78 <i>(December 2014)</i> </h4>

  <ul>
    <li>Customise <a href="vep_options.html#opt_pick">--pick</a> using
    <a href="vep_options.html#opt_pick_order">--pick_order</a></li>
    <li>Get <a href="/Help/Glossary?id=492">transcript support level</a>
    using <a href="vep_options.html#opt_tsl">--tsl</a></li>
  </ul>

  <h4> New in version 77 <i>(October 2014)</i> </h4>

  <ul>
    <li>Get the <a href="http://www.sequenceontology.org/" rel="external">SO</a>
    feature type of regulatory features using <a
    href="vep_options.html#opt_regulatory">--regulatory</a> and <a
    href="vep_options.html#opt_biotype">--biotype</a></li>
  </ul>

  <h4> New in version 76 <i>(August 2014)</i> </h4>

  <ul>
    <li>VEP now supports caches from multiple assemblies (<a
    href="vep_options.html#opt_assembly">--assembly</a>) on the same software
    version - e.g. <a href="vep_other.html#assembly">human builds GRCh37 and
    GRCh38</a></li>
    <li>Protein identifiers from UniProt (SWISSPROT, TrEMBL and UniParc) now
    available using <a href="vep_options.html#opt_uniprot">--uniprot</a></li>
    <li>VEP can generate <a href="../vep_formats.html#json">JSON output</a>
    using <a href="vep_options.html#opt_json">--json</a></li>
    <li>Two new analysis set options - <a
    href="vep_options.html#opt_gencode_basic">--gencode_basic</a> and the merged
    Ensembl/RefSeq cache (<a
    href="vep_options.html#opt_merged">--merged</a>)</li>
    <li>Non-RefSeq transcripts now excluded by default when using the RefSeq or
    merged cache; use <a href="vep_options.html#opt_all_refseq">--all_refseq</a>
    to include them</li>
    <li>Let VEP pick one consequence per variant allele using <a
    href="vep_options.html#opt_pick_allele">--pick_allele</a></li>
    <li>Allele now included alongside frequency for 1000 Genomes (<a
    href="vep_options.html#opt_maf_1kg">--maf_1kg</a>) and ESP (<a
    href="vep_options.html#opt_maf_esp">--maf_esp</a>) data</li>
    <li>Not strictly script-related, but the <a
    href="http://rest.ensembl.org/#Variation" rel="external">VEP REST API</a>
    has come out of beta!</li>
  </ul>

  <h4> New in version 75 <i>(February 2014)</i> </h4>

  <ul>
    <li>let VEP pick one consequence per variant for you using <a
    href="vep_options.html#opt_pick">--pick</a>; includes all
    transcript-specific data</li>
    <li><a href="vep_options.html#opt_symbol">gene symbol</a> available in
    RefSeq cache and when using <a
    href="vep_options.html#opt_refseq">--refseq</a></li>
    <li>Installation and use of RefSeq cache improved - remember to use <a
    href="vep_options.html#opt_refseq">--refseq</a> with your RefSeq cache!</li>
    <li>Added <a href="vep_options.html#opt_cache_version">--cache_version</a>
    option, primarily to aid Ensembl Genomes users.</li>
  </ul>

  <h4> New in version 74 <i>(December 2013)</i> </h4>

  <ul>
    <li>retrieve the <a href="http://genetics.bwh.harvard.edu/pph2/dokuwiki/overview#prediction" rel="external">humDiv PolyPhen prediction</a> instead of humVar using <a href="vep_options.html#opt_humdiv">--humdiv</a></li>
    <li>source for gene symbol available with <a href="vep_options.html#opt_symbol">--symbol</a></li>
  </ul>

  <h4> New in version 73 <i>(August 2013)</i> </h4>

  <ul>
    <li>NHLBI-ESP frequencies available in cache (<a href="vep_options.html#opt_maf_esp">--maf_esp</a>)</li>
    <li>Pubmed IDs for cited existing variants available in cache (<a href="vep_options.html#opt_pubmed">--pubmed</a>)</li>
    <li><a href="vep_cache.html#convert">Convert your cache to use tabix</a> - much faster when retrieving co-located existing variants!</li>
        <li>The <a href="vep_download.html#installer">installer</a> can now update the VEP to the latest version and install <a href="vep_cache.html#fasta">FASTA files</a></li>
    <li><b>--hgnc</b> replaced by <a href="vep_options.html#opt_symbol">--symbol</a> for non-human compatibility</li>
        <li>HGVS strings are now part <a rel="external" href="http://en.wikipedia.org/wiki/Percent-encoding" rel="exeternal">URI-escaped</a> to avoid "=" sign clashes</li>
    <li>use <a href="vep_options.html#opt_allele_number">--allele_number</a> to identify input alleles by their order in the VCF ALT field</li>
    <li>use <a href="vep_options.html#opt_total_length">--total_length</a> to give the total length of cDNA, CDS and protein sequences</li>
        <li>add data from VCF INFO fields when using <a href="vep_custom.html">custom annotations</a></li>
  </ul>

  <h4> New in version 72 <i>(June 2013)</i> </h4>

  <ul>
    <li>Speed and stability improvements when using forking</li>
    <li>Filter VEP results using <a href="vep_filter.html">filter_vep.pl</a></li>
  </ul>

  <h4> New in version 71 <i>(April 2013)</i> </h4>

  <ul>
    <li>SIFT predictions now available for Chicken, Cow, Dog, Human, Mouse,
    Pig, Rat and Zebrafish</li>
    <li>View <a href="../vep_formats.html#stats">summary statistics</a> for VEP runs in [output]_summary.html</li>
    <li>Generate HTML output using <a href="vep_options.html#opt_html">--html</a></li>
    <li>Support for simple tab-delimited format for input of structural
    variant data</li>
    <li>Cache now contains clinical significance statuses from dbSNP for
    human variants</li>
    <li><p><b>NOTE:</b> VEP version numbers have now (from release 71)
    changed to match Ensembl release numbers.</p></li>
  </ul>

  <h4> New in version 2.8 <i>(December 2012)</i> </h4>

  <ul>
    <li>Easily filter out common human variants with <a href="vep_options.html#opt_filter_common">--filter_common</a></li>
    <li>1000 Genomes continental population frequencies now stored in cache
    files</li>
  </ul>

  <h4> New in version 2.7 <i>(October 2012)</i> </h4>

  <ul>
    <li>build VEP cache files offline from GTF and FASTA files</li>
    <li>support for using FASTA files for sequence lookup in HGVS notations
    in offline/cache modes</li>
  </ul>

  <h4> New in version 2.6 <i>(July 2012)</i> </h4>

  <ul>
    <li>support for <a href="../vep_formats.html#sv">structural variant</a> consequences</li>
    <li>Sequence Ontology (SO) consequence terms now default</li>
    <li>script runtime 3-4x faster when using <a href="vep_options.html#forking">forking</a></li>
    <li>1000 Genomes global MAF available in cache files</li>
    <li>improved memory usage</li>
  </ul>

  <h4> New in version 2.5 <i>(May 2012)</i> </h4>

  <ul>
    <li>SIFT and PolyPhen predictions now available for RefSeq transcripts</li>
    <li>retrieve cell type-specific regulatory consequences</li>
    <li>consequences can be retrieved based on a single individual's genotype in
    a VCF input file</li>
    <li>find overlapping structural variants</li>
    <li>Condel support removed from main script and moved to a plugin</li>
  </ul>

  <h4> New in version 2.4 <i>(February 2012)</i> </h4>

  <ul>
    <li>offline mode and new installer script make it easy to use the VEP
    without the usual dependencies</li>
    <li>output columns configurable using the <a href="vep_options.html#opt_fields">--fields</a> flag</li>
    <li>VCF output support expanded, can now carry all fields</li>
    <li>output affected exon and intron numbers with <a href="vep_options.html#opt_numbers">--numbers</a></li>
    <li>output overlapping protein domains using <a href="vep_options.html#opt_domains">--domains</a></li>
    <li>enhanced support for LRGs</li>
    <li>plugins now work on variants called as intergenic</li>
  </ul>

  <h4> New in version 2.3 <i>(December 2011)</i> </h4>

  <ul>
    <li>add custom annotations from tabix-indexed files (BED, GFF, GTF, VCF,
    bigWig)</li>
    <li>add new functionality to the VEP with user-written plugins</li>
    <li>filter input on consequence type</li>
  </ul>

  <h4> New in version 2.2 <i>(September 2011)</i> </h4>

  <ul>
    <li>SIFT, PolyPhen and Condel predictions and regulatory features now
    accessible from the <a href="vep_cache.html">cache</a></li>
    <li>support for calling consequences against <a href="vep_other.html#refseq">RefSeq</a>
    transcripts</li>
    <li>variant identifiers (e.g. dbSNP rsIDs) and <a href="vep_other.html#hgvs">HGVS
    notations</a> supported as input format</li>
    <li>variants can now be <a href="vep_options.html#filt">filtered</a> by frequency in HapMap
    and 1000 genomes populations</li>
    <li>script can be used to convert files between formats
    (Ensembl/VCF/Pileup/HGVS to Ensembl/VCF/Pileup)</li>
    <li>large amount of code moved to API modules to ensure consistency between
    web and script VEP</li>
    <li>memory usage optimisations</li>
    <li>VEP script moved to <a href="https://github.com/Ensembl/ensembl-tools/tree/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/scripts/variant_effect_predictor" rel="external">ensembl-tools repo</a></li>
    <li>Added <a href="vep_options.html#opt_canonical">--canonical</a>, <a href="vep_options.html#opt_per_gene">--per_gene</a> and <a href="vep_options.html#opt_no_intergenic">--no_intergenic</a> options</li>
  </ul>

  <h4> New in version 2.1 <i>(June 2011)</i> </h4>

  <ul>
    <li>ability to use local file <a href="vep_cache.html#cache">cache</a> in place of or
    alongside connecting to an Ensembl database</li>
    <li>significant improvements to speed of script</li>
    <li>whole-genome mode now default (no disadvantage for smaller
    datasets)</li>
    <li>improved status output with progress bars</li>
    <li>regulatory region consequences now reinstated and improved</li>
    <li>modification to output file - Transcript column is now Feature, and is
    followed by a Feature_type column</li>
  </ul>

  <h4> New in version 2.0 <i>(April 2011)</i> </h4>

  <ul>
    <li>support for SIFT, PolyPhen and Condel missense predictions in human</li>
    <li>per-allele and compound consequence types</li>
    <li>support for Sequence Ontology (SO) and NCBI consequence terms</li>
    <li>
      modified output format
      <ul>
        <li>support for new output fields in Extra column</li>
        <li>header section contains information on database and software versions</li>
        <li>codon change shown in output</li>
        <li>CDS position shown in output</li>
        <li>option to output Ensembl protein identifiers</li>
        <li>option to output HGVS nomenclature for variants</li>
      </ul>
    </li>
    <li>support for gzipped input files</li>
    <li>enhanced configuration options, including the ability to read configuration from a file</li>
    <li>verbose output now much more useful</li>
    <li>whole-genome mode now more stable</li>
    <li>finding existing co-located variations now ~5x faster</li>
  </ul>

    </div>

    <hr/>
  <h2 id="requirements">Requirements</h2>

  VEP requires:
  <ul>
    <li><b>gcc</b>, <b>g++</b> and <b>make</b></li>
    <li><b>Perl</b> version <b>5.10</b> or above recommended (tested on 5.10, 5.14, 5.18, 5.22, 5.26)</li>
    <li><b>Perl</b> packages:
      <ul>
        <li><a href="https://metacpan.org/pod/Archive::Zip" rel="external">Archive::Zip</a></li>
        <li><a href="https://metacpan.org/pod/DBD::mysql" rel="external">DBD::mysql</a></li>
        <li><a href="https://metacpan.org/pod/DBI" rel="external">DBI</a></li>
      </ul>
      See <a href="http://www.cpan.org/modules/INSTALL.html" rel="external">this guide</a> for more information on how to install perl modules.<br/>
      <a href="#additional">Additional libraries</a> can be installed for extra features and enhancements but they are not required to run VEP in most of the use cases.
    </li>
  </ul>

  <p>VEP's INSTALL.pl script will install required components of Ensembl API for you, but VEP may also be used with any pre-existing API installations you have, <b>provided their versions match the version of VEP you are using</b>. </p>

  <p>VEP has been developed for UNIX-like environments and works well on Linux (e.g. Ubuntu, Debian, Mint) and Mac OSX.<br />It can also be used on <a href="#windows"><img style="vertical-align:bottom" src="/i/16/windows.png"/> Windows</a> systems with a more involved installation process. </p>

  <hr/>


  <h2 id="installer">Installation</h2>

  <p> VEP's INSTALL.pl makes it easy to set up your environment for
    using the VEP. It will download and configure a minimal set of the Ensembl
    API for use by the VEP, and can also download <a
    href="vep_cache.html#cache">cache files</a>, <a
    href="vep_cache.html#fasta">FASTA files</a> and <a href="vep_plugins.html">plugins</a>. </p>

  <p>Run the following, and follow any prompts as they appear:</p>

  <pre class="code sh_sh">perl INSTALL.pl</pre>

  <p><a href="#additional">Additional non-essential components</a> and enhancements must be installed manually.</p>
  
  <br />
  <h3> Software components installed </h3>

  <ul>
    <li><a href="https://github.com/bioperl/bioperl-live" rel="external">BioPerl</a></li>
    <li><a href="https://github.com/Ensembl/ensembl" rel="external">ensembl</a></li>
    <li><a href="https://github.com/Ensembl/ensembl-io" rel="external">ensembl-io</a></li>
    <li><a href="https://github.com/Ensembl/ensembl-variation" rel="external">ensembl-variation</a></li>
    <li><a href="https://github.com/Ensembl/ensembl-funcgen" rel="external">ensembl-funcgen</a></li>
    <li><a href="https://github.com/Ensembl/Bio-DB-HTS" rel="external">Bio::DB::HTS</a></li>
  </ul>

  <p> If you already have the latest version of the API installed you do not
  need to run the installer, although it can be used to simply
  update your API version (with post-release patches applied), and
  retrieve cache and FASTA files. The installer downloads the API within the
  VEP directory and will not affect any other Ensembl API installations. </p>

  <p> The script will also attempt to install a Perl::XS module,
  <a rel="external" href="https://github.com/Ensembl/Bio-HTS">Bio::DB::HTS</a>, for
  rapid access to bgzipped FASTA files. If this fails, you may add the
  --NO_HTSLIB flag when running the installer; VEP will fall back to using
  Bio::DB::Fasta for this functionality (<a
  href="vep_cache.html#fasta">more details</a>).</p>

  <br />
  <h3 id="run_install">Running the installer</h2>

  <p> The installer is run on the command line as follows: </p>

  <pre class="code sh_sh"> perl INSTALL.pl [options] </pre>

  <p> Follow on-screen prompts and note warnings of any files which will be deleted/overwritten</p>

  <p> You should not need to add any options, but configuration of the
  installer is possible with the following flags: </p>

  <table class="ss">
    <tr>
      <th>Flag</th>
      <th>Alternate</th>
      <th>Description</th>
    </tr>
    <tr>
      <td><pre>--ASSEMBLY</pre></td>
      <td><pre>-y</pre></td>
      <td>
        Assembly version to use when using <code>--AUTO</code>. Most species have only one
        assembly available on each software release; currently this is only
        required for <a href="vep_other.html#assembly">human on release 76</a>
        onwards.
      </td>
    </tr>
    <tr>
      <td><pre>--AUTO</pre></td>
      <td><pre>-a</pre></td>
      <td>
        Run installer without prompts. Use the following options to specify
        parts to install:
        <ul>
          <li><b>a</b> (API + Bio::DB::HTS/htslib)</li>
          <li><b>l</b> (Bio::DB::HTS/htslib only)</li>
          <li><b>c</b> (cache)</li>
          <li><b>f</b> (FASTA)</li>
          <li><b>p</b> (plugins)</li>
        </ul>
        <p>e.g. for API and cache:</p>
        <pre class="code sh_sh">perl INSTALL.pl --AUTO ac</pre>
      </td>
    </tr>
    <tr>
      <td><pre>--CACHE_VERSION [version]</pre></td>
      <td>&nbsp;</td>
      <td>
        By default the installer will download the latest version of VEP caches
        and FASTA files (currently [[SPECIESDEFS::ENSEMBL_VERSION]]). You can
        force the script to install a different version, but there is no guarantee that
        a version of the API will be compatible with a different version of the cache.
        Use <code>--VERSION</code> to set the API version separately.
      </td>
    </tr>
    <tr>
      <td><pre>--CACHEDIR [dir]</pre></td>
      <td><pre>-c</pre></td>
      <td>
        <p>By default the script will install the cache files in the ".vep"
        subdirectory in your home area. This option
        configures where cache files are installed. The <a
        href="vep_options.html#opt_dir">--dir</a> flag must be passed when
        running the VEP if a non-default directory is given:</p>
        <pre class="code sh_sh">./vep --dir [dir]</pre>
      </td>
    </tr>
    <tr>
      <td><pre>--DESTDIR [dir]</pre></td>
      <td><pre>-d</pre></td>
      <td>
        <p>By default the script will install the API modules in a
        subdirectory of the current directory named "Bio". Using this option
        you can configure where the Bio directory is created. If something
        other than the default is used, this directory must either be added
        to your PERL5LIB environment variable when running the VEP, or
        included using perl's -I flag:</p>
        <pre class="code sh_sh">perl -I [dir] vep</pre>
      </td>
    </tr>
    <tr>
      <td><pre>--NO_HTSLIB</pre></td>
      <td><pre>-l</pre></td>
      <td>Don't attempt to install Bio::DB::HTS/htslib</td>
    </tr>
    <tr>
      <td><pre>--NO_TEST</pre></td>
      <td>&nbsp;</td>
      <td>Don't run API tests - useful if you know a harmless failure will prevent continuation of the installer</td>
    </tr>
    <tr>
      <td><pre>--NO_UPDATE</pre></td>
      <td><pre>-n</pre></td>
      <td>
        By default the script will check for new versions or updates of the VEP.
        Using this option will skip this check.
      </td>
    </tr>
    <tr>
      <td><pre>--PLUGINS</pre></td>
      <td><pre>-g</pre></td>
      <td>
        <p>Comma-separated list of plugins to install when using <code>--AUTO</code>. To
        install all available plugins, use <code>--PLUGINS all</code>.<br />To list available
        plugins, use:</p>
        <pre class="code sh_sh">perl INSTALL.pl -a p --PLUGINS list</pre>
      </td>
    </tr>
     <tr>
      <td><pre>--PREFER_BIN</pre></td>
      <td><pre>-p</pre></td>
      <td>Use this if the installer fails with out of memory errors.</td>
    </tr>
    <tr>
      <td><pre>--SPECIES</pre></td>
      <td><pre>-s</pre></td>
      <td>
        Comma-separated list of species to install when using <code>--AUTO</code>. To
        install the RefSeq cache, add "_refseq" to the species name, e.g.
        "homo_sapiens_refseq", or "_merged" to install the merged Ensembl/RefSeq
        cache. Remember to use <a
        href="vep_options.html#opt_refseq">--refseq</a> or <a
        href="vep_options.html#opt_merged">--merged</a> when running the VEP
        with the relevant cache!
      </td>
    </tr>
    <tr>
      <td><pre>--VERSION [version]</pre></td>
      <td><pre>-v</pre></td>
      <td>
        By default the latest version of the Ensembl API
        (currently [[SPECIESDEFS::ENSEMBL_VERSION]]) will be installed. You can force the script
        to install a different version. This flag will also
        set the data version (cache, FASTA) to install unless set separately
        with <code>--CACHE_VERSION</code>.
      </td>
    </tr>
    <tr>
      <td><pre>--QUIET</pre></td>
      <td><pre>-q</pre></td>
      <td>Don't write any status output when using <code>--AUTO</code>.</td>
    </tr>
  </table>




  <hr/>
  <h2 id="additional">Additional components</h2>

  <p> INSTALL.pl will set up the minimum requirements for VEP. Some features and enhancements, however, require the installation of additional components. Most are perl modules that are easily installed using cpanm; see <a href="http://www.cpan.org/modules/INSTALL.html" rel="external">this guide</a> for more information on how to install perl modules. </p>

  <p> Typically, you will use cpanm to install modules locally in your home directories; this shows how to set up a path for perl modules and install one there: </p>

  <pre class="code sh_sh">mkdir -p $HOME/cpanm
export PERL5LIB=$PERL5LIB:$HOME/cpanm/lib/perl5
cpanm -l $HOME/cpanm Set::IntervalTree</pre>

  <p> To make the change to <code>PERL5LIB</code> permanent, it is recommended to add the <code>export</code> line to your <code>$HOME/.bashrc</code> or <code>$HOME/.profile</code>. </p>

  <ul>
    <li>
      Additional features
      <ul>
        <li><a href="http://search.cpan.org/dist/JSON/" rel="external">JSON</a> - required to produce <a href="vep_options.html#opt_json">JSON format output</a></a></li>
        <li><a href="http://search.cpan.org/~benbooth/Set-IntervalTree/lib/Set/IntervalTree.pm" rel="external">Set::IntervalTree</a> - used to find overlaps between entities in coordinate space. Required to use <a href="vep_options.html#opt_nearest">--nearest</a></li>
        <li><a href="http://search.cpan.org/~lds/Bio-BigFile-1.07/lib/Bio/DB/BigFile.pm" rel="external">Bio::DB::BigFile</a> - required to use bigWig format <a href="vep_custom.html">custom annotation files</a>. See <a href="#bigfile">Bio::DB::BigFile instructions</a>.</li>
      </ul>
    </li>
    <li>
      Speed enhancements - these modules can improve VEP runtime
      <ul>
        <li><a href="http://search.cpan.org/~nwclark/PerlIO-gzip-0.19/gzip.pm">PerlIO::gzip</a> - marginal gains in compressed file parsing as used by VEP cache</li>
        <li><a href="https://github.com/Ensembl/ensembl-xs" rel="external">ensembl-xs</a> - provides pre-compiled replacements for frequently used routines in VEP. Requires manual installation, see <a href="https://github.com/Ensembl/ensembl-xs" rel="external">README</a> for details</li>
      </ul>
    </li>
  </ul>

  <br />
  <h3 id="bigfile">Bio::DB::BigFile</h3>

  <p> In order for VEP to be able to access bigWig format custom annotation files, the Bio::DB::BigFile perl module is required. Installation involves downloading and compiling the <a href="https://github.com/ucscGenomeBrowser/kent" rel="external">kent source tree</a>. The current version of the kent source tree does not work correctly with Bio::DB::BigFile, so it is necessary to install an archive version known to work (v335). </p>

  <ol>
    <li>
      <p>Download and unpack the kent source tree</p>
      <pre class="code sh_sh">wget https://github.com/ucscGenomeBrowser/kent/archive/v335_base.tar.gz
tar xzf v335_base.tar.gz</pre>
    </li>
    <li>
      <p>Set up some environment variables; these are required only temporarily for this installation process</p>
      <pre class="code sh_sh">export KENT_SRC=$PWD/kent-335_base/src
export MACHTYPE=$(uname -m)
export CFLAGS="-fPIC"
export MYSQLINC=`mysql_config --include | sed -e 's/^-I//g'`
export MYSQLLIBS=`mysql_config --libs`</pre>
    </li>
    <li>
      <p>Modify kent build parameters</p>
      <pre class="code sh_sh">cd $KENT_SRC/lib
echo 'CFLAGS="-fPIC"' > ../inc/localEnvironment.mk</pre>
    </li>
    <li>
      <p>Build kent source</p>
      <pre class="code sh_sh">make clean && make
cd ../jkOwnLib
make clean && make</pre>
      <p>If either of these steps fail, you may have some missing dependencies. Known common missing dependencies are libpng and libssl; these may be installed, for example, with <code>apt-get</code> on Ubuntu. If you do not have sudo access you may have to ask your sysadmin to install any missing dependencies.</p>
      <pre class="code sh_sh">sudo apt-get install libpng-dev libssl-dev</pre>
      <p>On Mac OSX you may use <a href="https://brew.sh/" rel="external"><code>brew</code></a>; the openssl libraries also need to be symbolically linked to a different path:</p>
      <pre class="code sh_sh">brew install libpng openssl
cd /usr/local/include
ln -s ../opt/openssl/include/openssl .
cd -</pre>
    </li>
    <li>
      <p>On some systems (e.g. Mac OSX), a compiled file is placed in a path that Bio::DB::BigFile cannot find. You can correct this with:</p>
      <pre class="code sh_sh">ln -s $KENT_SRC/lib/x86_64/* $KENT_SRC/lib/</pre>
    </li>
    <li>
      <p>We'll now use cpanm to install the perl module for Bio::DB::BigFile itself. See <a href="#additional">above</a> for guidance on this. In this example we're going to install the module to a path within your home directory. In order to do this we must modify the paths that perl looks in to find modules by adding to the <code>PERL5LIB</code> environment module. To make this change permanent you must add the <code>export</code> line to your <code>$HOME/.bashrc</code> or <code>$HOME/.profile</code>.</p>
      <pre class="code sh_sh">mkdir -p $HOME/cpanm
export PERL5LIB=$PERL5LIB:$HOME/cpanm/lib/perl5
cpanm -l $HOME/cpanm Bio::DB::BigFile</pre>
      <p>If you are prompted for the path to the kent source tree, that means something didn't go right in the compilation above. Double check that <code>$KENT_SRC/lib/jkweb.a</code> exists and is not found instead at e.g. <code>$KENT_SRC/lib/x86_64/jkweb.a</code>. You may copy or link the file (and the other files in that directory) to the former path.</p>
      <pre class="code sh_sh">ln -s $KENT_SRC/lib/x86_64/* $KENT_SRC/lib/</pre>
    </li>
    <li>
      <p>You should now be able to successfully run the appropriate test in the VEP package:</p>
      <pre class="code sh_sh">perl -Imodules t/AnnotationSource_File_BigWig.t</pre>
    </li>
  </ol>



  <hr/>
  <h2 id="macos">Using VEP in Mac OS</h2>
  
  <p>Installing VEP on Mac OS is slightly trickier than other Linux-based systems, and will require additional dependancies.<br />
  These instructions will guide you through the setup of <b>Perlbrew</b>, <b>Homebrew</b>, <b>MySQL</b> and other dependancies that will allow for a clean installation of VEP on your Mac OS system.</p>

  <p>These instructions have been tested on <b>macOS High Sierra (10.13)</b> and <b>macOS Sierra (10.12)</b>.<br />Older versions may require additional tweaks, however we shall endeavor to keep these instructions up to date for future versions of MacOS.</p>

  <br />
  <h3>Prerequisite Setup</h3>

  <p>List of prerequisites: <b>XCode</b>, <b>GCC</b>, <b>Perlbrew</b>, <b>Cpanm</b>, <b>Homebrew</b>, <b>mysql</b>, <b>DBI</b>, <b>DBD::mysql</b></p>

  <h4>XCode and GCC</h4>
  
  <p>VEP requires XCode and GCC for installation purposes. Fortunately, recent versions of macOS will look for (and attempt to install if required) both of these when you run the following command:</p>

  <pre class="code sh_sh">gcc -v</pre>

  <h4>Perlbrew</h4>
  
  <p>We recommend using Perlbrew to install a new version of Perl on your mac, to prevent messing with the vendor perl too much. This can be done with the following command:</p>

  <pre class="code sh_sh">curl -L http://install.perlbrew.pl | bash

echo 'source $HOME/perl5/perlbrew/etc/bashrc' >> ~/.bash_profile</pre>

 <p>At this point, PLEASE RESTART YOUR TERMINAL WINDOW to allow for the perlbrew changes to take effect.</p>

 <p>We recommend installing Perl version <b>5.26.2</b> to run VEP, and installing cpanm to handle the installation of perl modules.<br />
 These steps can be completed with the commands:</p>

  <pre class="code sh_sh">perlbrew install -j 5 --as 5.26.2 --thread --64all -Duseshrplib perl-5.26.2 --notest
perlbrew switch 5.26.2
perlbrew install-cpanm</pre>

  <h4>Homebrew</h4>
  
  <p>This package management system for Mac OS would make the installation of the next prerequisite (i.e. xs) easier.</p>

  <pre class="code sh_sh">/usr/bin/ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)"</pre>

  <h4>xz</h4>

  <p>VEP requires the installation of xz, a data-compression utility. The easiest way to install the xz package is through homebrew:</p>

  <pre class="code sh_sh">brew install xz</pre>

  <h4>MySQL</h4>
  
  <p>In order to connect to the Ensembl databases, a collection of MySQL related dependancies are required. Fortunately, these can be installed neatly with <b>Homebrew</b> and <b>Cpanm</b>:</p>

  <pre class="code sh_sh">brew install mysql
cpanm DBI
cpanm DBD::mysql</pre>

  <br />
  <h3>Installing BioPerl</h3>

  <p>On some versions of macOS, the VEP installer fails to cleanly install BioPerl, so a manual install will prevent issues:</p>

  <pre class="code sh_sh">curl -O https://cpan.metacpan.org/authors/id/C/CJ/CJFIELDS/BioPerl-1.6.924.tar.gz
tar zxvf BioPerl-1.6.924.tar.gz
echo 'export PERL5LIB=${PERL5LIB}:##PATH_TO##/bioperl-1.6.924' >> ~/.bash_profile</pre>
  
  <p> where <i>##PATH_TO##/bioperl-1.6.924</i> refers to the location of the newly unzipped BioPerl directory.</p>
  
  <br />
  <h3>Final Dependancies</h3>

  <p>Installing the following Perl modules with cpanm will allow for full VEP functionality:</p>

  <pre class="code sh_sh">cpanm Test::Differences Test::Exception Test::Perl::Critic Archive::Zip PadWalker Error Devel::Cycle Role::Tiny::With Module::Build

export DYLD_LIBRARY_PATH=/usr/local/mysql/lib/:$DYLD_LIBRARY_PATH</pre>


  <br />
  <h3>Installing VEP</h3>

  <p>And that should be that! You should now be able to install VEP using the installer:</p>

  <pre class="code sh_sh">git clone https://github.com/ensembl/ensembl-vep
cd ensembl-vep
perl INSTALL.pl --NO_TEST</pre>



  <hr/>
  <h2 id="windows">Using VEP in Windows <img style="padding-left:2px" src="/i/16/windows.png"/></h2>

  <p> VEP was developed as a command-line tool, and as a Perl script its
  natural environment is a Linux system. However, there are several ways you can
  use VEP on a Windows machine. </p>

  <p> You may also consider using VEP's web or REST interfaces. </p>

  <br />
  <h4 id="virtual">Virtual machines</h4>

  <p> Using a virtual machine you can run a virtual Linux system in a window on
  your machine. There are two ways to do this: </p>

  <ol>
    <li>Use the <a
    href="/info/data/virtual_machine.html">Ensembl virtual machine image</a></li>
    <li>Use <a href="#docker">Docker</a></li>
  </ol>

  <br />
  <h4 id="dwim">DWIMperl</h4>

  <p> DWIMperl has a Windows package that contains base requirements for setting up VEP. </p>

  <ol>
    <li>Download and install <a href="http://dwimperl.com/windows.html" rel="external">DWIMperl for Windows</a></li>
    <li>Download and unpack the <a href="https://github.com/Ensembl/ensembl-vep/archive/release/[[SPECIESDEFS::ENSEMBL_VERSION]].zip">zip of the ensembl-vep package</a></li>
    <li>Open a Command Prompt (search for Command Prompt in the Start Menu)</li>
    <li>
      <p>Navigate to the directory where you unpacked the VEP package, e.g.</p>
      <pre class="code sh_sh">cd Downloads/ensembl-vep-release-[[SPECIESDEFS::ENSEMBL_VERSION]]</pre>
    </li>
    <li>
      <p>Run INSTALL.pl with --NO_HTSLIB and --NO_TEST; you will see some warnings about the "which" command not being available (these will also appear when running VEP and can be ignored).</p>
      <pre class="code sh_sh">perl INSTALL.pl --NO_HTSLIB --NO_TEST</pre>
    </li>
  </ol>


  <hr/>
  <h2 id="docker">Docker</h2>

  <p> <a href="https://www.docker.com/" rel="external">Docker</a> allows you to run applications in virtualised "containers". A docker image for VEP is available from DockerHub:  
  <a class="button" href="https://hub.docker.com/r/ensemblorg/ensembl-vep/" target="_blank">VEP in DockerHub</a></p>

  <p>Commands to download the VEP Docker image (need to download and install <a href="https://www.docker.com/" rel="external">docker</a> beforehand):</p>
  <pre class="code sh_sh">docker pull ensemblorg/ensembl-vep
docker run -t -i ensemblorg/ensembl-vep ./vep</pre>

  <p> Currently no <a href="https://docs.docker.com/engine/tutorials/dockervolumes/" rel="external">volumes</a> are pre-configured for the container; this is required if you wish to download data (e.g. cache files) that persists across sessions. </p>

  <p> The following is a brief example showing how to use a directory on your local (host) machine to store cache data for VEP. </p>

  <pre class="code sh_sh"># Create a directory on your machine:
mkdir $HOME/vep_data

# Make sure that the created directory on your machine has read and write access granted
# so the docker container can write in the directory (VEP output):
chmod a+rwx $HOME/vep_data    

docker run -t -i -v $HOME/vep_data:/opt/vep/.vep ensemblorg/ensembl-vep perl INSTALL.pl</pre>

<br />
<br />
<h3>Cache and Plugins installation</h3>
  <p>
    You will now be prompted by the installer if you wish to re-install the API.
    Type "n" followed by enter to skip to cache installation.
    You will be presented with a list of species; type the number for your species/assembly of interest and press enter.
    Your data will now download and unpack; this may take a while.
  </p>
  <p>
    If you wish to retrieve HGVS annotations it is recommended to also download the FASTA file for your species.
    To do this, at the next prompt type "0" and press enter. You may skip the plugin installation also.
  </p>

  <p>
    The above process may also be performed in one command; for example, to set up the cache and corresponding FASTA for human GRCh38:
  </p>

  <pre class="code sh_sh">docker run -t -i -v $HOME/vep_data:/opt/vep/.vep ensemblorg/ensembl-vep perl INSTALL.pl -a cf -s homo_sapiens -y GRCh38</pre>

  <p>If you wish to include the <a href="https://github.com/Ensembl/VEP_plugins" rel="external">VEP plugins</a>, add the '<b>p</b>' value to the <code>-a</code> flag:</p>
  <pre class="code sh_sh">docker run -t -i -v $HOME/vep_data:/opt/vep/.vep ensemblorg/ensembl-vep perl INSTALL.pl -a cfp -s homo_sapiens -y GRCh38</pre>
  
  <br />
  <p>
    The installer has now downloaded this data to <b>$HOME/vep_data</b> (and <b>$HOME/vep_data/Plugins</b> for the VEP plugins).
    VEP will automatically detect caches downloaded in this folder as it is mapped to VEP's default directory within the Docker instance.
  </p>

  <pre class="code sh_sh">docker run -t -i -v $HOME/vep_data:/opt/vep/.vep ensemblorg/ensembl-vep
./vep -i examples/homo_sapiens_GRCh38.vcf -cache</pre>

  
  <div>
    <div style="float:left" class="warning">
      <h3 id="read_write_access">Read/Write access from the Docker container</h3>
      <div class="message-pad">
        <p>In some distributions (e.g. CentOS, Fedora, Red Hat Enterprise Linux) Docker daemon requires <b>root</b> privileges (i.e. needs to prefix the command with <b>sudo</b>), which might cause read/write issues to the mounted volume.</p>
        <p>One solution is to use the option <b>:Z</b> within the Docker <b>-v</b> option (only from docker 1.7.0):
          <pre class="code sh_sh">sudo docker run -t -i -v $HOME/vep_data:/opt/vep/.vep:Z ensemblorg/ensembl-vep</pre>
        </p>
        <p>An other solution is to change the read/write access of the mounted volume (<b>$HOME/vep_data</b>):
          <pre class="code sh_sh">chmod -R a+rwx $HOME/vep_data</pre>
        </p>
      </div>
    </div>
    <div class="clear"></div>
  </div>

<br />
<br />
<h3>Update from a previous version</h3>

<ol>
  <li><p>Update your docker container</p>
    <pre class="code sh_sh">
# List containers
docker ps -a
# e.g.
CONTAINER ID     IMAGE                        COMMAND          CREATED              STATUS                         PORTS          NAMES
d64055ffe9e9     ensemblorg/ensembl-vep       "/bin/bash"      About a minute ago   Exited (0) 59 seconds ago                     tender_ritchie

# Stop and remove old container
docker stop tender_ritchie
docker rm tender_ritchie

# Update the container
docker pull ensemblorg/ensembl-vep</pre>
  </li>
  <li><p>Update your cache</p>
    <pre class="code sh_sh"># Install the new cache through the VEP INSTALL.pl script (see "Cache installation" section above)
docker run -t -i -v $HOME/vep_data:/opt/vep/.vep ensemblorg/ensembl-vep perl INSTALL.pl

# Or you can install the cache manually
cd $HOME/vep_data
curl -O ftp://ftp.ensembl.org/pub/release-[[SPECIESDEFS::ENSEMBL_VERSION]]/variation/VEP/homo_sapiens_vep_[[SPECIESDEFS::ENSEMBL_VERSION]]_GRCh38.tar.gz
tar xzf homo_sapiens_vep_[[SPECIESDEFS::ENSEMBL_VERSION]]_GRCh38.tar.gz</pre>
  </li>
</ol>

  </div>
</div>

</body>
</html>
