
<html>
  <title>VEP documentation</title>
  <body>
<style>
img.opaque-hover:hover {
    opacity:0.6;
    filter:alpha(opacity=60);
}
img.opaque-hover {
    opacity:1.0;
    filter:alpha(opacity=100);
}
</style>

  <div style="float:right"><img src="/img/vep_logo.png"/></div>
	
  <h1 id="top"><span style="color:#006">Ensembl Variant Effect Predictor (Ensembl VEP)</span></h1>
  <hr/>

  <p>
    <span style="color:#006;font-weight:bold">Ensembl VEP predicts the effect of your variants</span> (SNPs, insertions,
    deletions, CNVs or structural variants) on gene transcripts and
    protein sequence, as well as regulatory regions.
    It reports reference data including gene and variant phenotype associations and population allele frequencies 
    to facilitate variant prioritisation and interpretation. 
  </p>
  <span>Simply input the coordinates of your variants and the nucleotide changes to find out the:</span>
  <ul>
    <li><b>Genes</b> and <b>Transcripts</b> overlapping the variant, which it may impact</li>
    <li><b>Predicted consequence</b> of your variants on the protein (e.g. stop gained, missense,, frameshift), transcript ( e.g. splice_donor_variant) or regulatory element ( e.g. regulatory_region_variant, TF_binding_site_variant) see <a href="/info/genome/variation/prediction/predicted_data.html">variant consequences</a></li>
    <li><b>Known variants</b> that match yours, and associated allele frequencies from projects including <b>gnomAD</b> and the <b>1000 Genomes Project</b>, citation and links to information in other resources</li>
    <li>Predictions of variant deleteriousness from tools including <b>AlphaMissense</b>, <b>CADD</b>, <b>SIFT</b>,  <b>REVEL</b> and  <b>SpliceAI</b></li>
    <li><b>Functional assay</b>  results from sources including  <b>MaveDB</b>

    <li>... And more! See the current <a href="/info/docs/tools/vep/online/input.html#ident">analysis options</a>, <a href="/info/docs/tools/vep/script/vep_cache.html#cache_content">data versions</a>.</li>
  </ul>

  <h2><img
  src="/i/16/star.png"/> <a href="/info/docs/tools/vep/script/vep_download.html#new">What's new in release [[SPECIESDEFS::ENSEMBL_VERSION]]?</a></h2>

  <br />
  <h2 id="vep_interfaces">Ensembl VEP interfaces</h2>
  <div class="bottom-margin" style="display:table">

    <!-- Web interface -->
    [[PERL::$_->species_defs->ENSEMBL_VEP_ENABLED
    ? q(
    <div style="display:table-cell;padding-right:2em">
      <div class="tinted-box" style="padding-bottom:16px;margin-bottom:-1px">
        <div><h1 style="text-align:center;padding-top:0px;margin-bottom:16px">Web interface</h1></div>
        <div style="float:left;margin-right:15px">
          <a href="/Tools/VEP" class="_ht" title="Launch the web interface"><img src="/img/vep_globe.png" style="border:2px solid #FFF;border-radius:5px;background:#336"/></a>
        </div>

        <div style="float:left;color:#333">
          <ul style="padding-left:1.5em;margin-bottom:2em">
            <li>Point-and-click interface</li>
            <li>Suits smaller volumes of data</li>
          </ul>
          <div style="font-weight:bold;margin-left:5px">
            <img src="/i/16/documentation.png" style="vertical-align:bottom"/> <a href="/info/docs/tools/vep/online/index.html">Documentation</a>
          </div>
        </div>
      </div>
      <div class="tinted-box" style="padding-top:10px;text-align:center;background-color:#FFF;height:105px">
        <a href="/Tools/VEP" class="_ht" title="Launch the web interface"><img src="/img/vep_launch_small.png" class="opaque-hover" style="width:200px"/></a>
      </div>
    </div>
    ): q()
    ]]

    <!-- Perl script -->
    <div style="display:table-cell;padding-right:2em">
      <div class="tinted-box" style="padding-bottom:16px;margin-bottom:-1px">
        <div><h1 style="text-align:center;padding-top:0px;margin-bottom:16px">Command line tool</h1></div>
        <div style="float:left;margin-right:15px">
          <a href="/info/docs/tools/vep/script/index.html" class="_ht" title="View Ensembl VEP documentation"><img src="/img/vep_cmd.png" style="border:2px solid #FFF;border-radius:5px;background:white"/></a>
        </div>
        <div style="float:left;color:#333">
          <ul style="padding-left:1.5em;margin-bottom:2em">
            <li>More options and flexibility</li>
            <li>For large volumes of data</li>
          </ul>
          <div style="font-weight:bold;margin-left:5px">
            <img src="/i/16/documentation.png" style="vertical-align:bottom"/> <a href="/info/docs/tools/vep/script/index.html">Documentation</a>
          </div>
        </div>
      </div>
      <div class="tinted-box" style="padding-top:10px;text-align:center;background-color:#FFF;height:105px">
        <div style="padding-top:10px">
          <h3><img src="/i/16/copy.png" style="vertical-align:bottom"/> <a href="https://github.com/Ensembl/ensembl-vep" rel="external">Clone from GitHub</a></h3>
          <h3><img src="/i/16/download.png" style="vertical-align:bottom"/> <a href="https://github.com/Ensembl/ensembl-vep/archive/release/[[SPECIESDEFS::ENSEMBL_VERSION]].zip" rel="external">Download (zip)</a></h3>
          <h3><img src="/i/16/copy.png" style="vertical-align:bottom"/> <a href="https://hub.docker.com/r/ensemblorg/ensembl-vep" rel="external">Pull Docker image from DockerHub</a></h3>
        </div>
      </div>
    </div>

    <!-- REST API -->
    <div style="display:table-cell">
      <div class="tinted-box" style="padding-bottom:16px;margin-bottom:-1px">
        <div><h1 style="text-align:center;padding-top:0px;margin-bottom:16px">REST API</h1></div>
        <div style="float:left;margin-right:5px">
          <a href="https://rest.ensembl.org/#VEP" class="_ht" title="Go to REST documentation" rel="external"><img src="/img/restlogo_sq.png" style="padding:1px;border:1px solid #b0bbd8;border-radius:5px;background:white;text-decoration:none"/></a>
        </div>
        <div style="float:left;color:#333">
          <ul style="padding-left:1.5em;margin-bottom:2em">
            <li>Language-independent API</li>
            <li>Simple URL-based queries</li>
          </ul>
          <div style="font-weight:bold;margin-left:5px">
            <img src="/i/16/documentation.png" style="vertical-align:bottom"/> <a href="https://rest.ensembl.org/#VEP" rel="external">Documentation</a>
          </div>
        </div>
      </div>
      <div class="tinted-box" style="padding-top:10px;text-align:center;background-color:#FFF;height:105px">
        <div style="padding-top:40px">
          <h3><img src="/i/16/globe.png" style="vertical-align:bottom"/> <a href="https://rest.ensembl.org/#VEP" title="VEP REST endpoint" rel="external">Ensembl VEP REST API</a></h3>
        </div>
      </div>
    </div>
  </div>


  <br />
  <hr />
  <h2 id="publication">Publication</h2>

  <p>If you use Ensembl VEP, please cite our most recent publication to help us continue to support development:</p>

  <div>
    <div style="float:left;margin-bottom:20px" class="info">
      <h3 id="cite_us">Cite us</h3>
      <div class="message-pad">
        <p>McLaren W, Gil L, Hunt SE, Riat HS, Ritchie GR, Thormann A, Flicek P, Cunningham F.<br />
          <strong>The Ensembl Variant Effect Predictor.</strong><br />
          <em>Genome Biology</em> Jun 6;17(1):122. (<strong>2016</strong>)<br />
          <a rel="external" href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-016-0974-4" rel="external">doi:10.1186/s13059-016-0974-4 </a>
        </p>
      </div>
    </div>
    <div style="clear:both"></div>
  </div>


  <br />
  <hr />
  <h2 id="vep_tools"><img src="/i/16/tool.png" style="vertical-align:bottom;margin-right:5px"/>tools related to Ensembl VEP</h2>
  
  <div style="display:table;margin-top:10px">

    <!-- Variant recoder -->
    <div style="display:table-cell;padding-right:2em">
      <div class="tinted-box" style="padding-bottom:16px;margin-bottom:-1px">
        <div><h1 style="text-align:center;padding-top:0px;margin-bottom:16px">Variant recoder</h1></div>
        <div style="color:#333">
          <p>Translate a <b>variant identifier</b> or <b>HGVS notation</b> to all possible variant IDs and HGVS</p>
          <div><img src="/i/16/documentation.png" style="vertical-align:bottom"> <a href="/info/docs/tools/vep/recoder/index.html">Documentation</a></div>
        </div>
      </div>
      <div class="tinted-box" style="background-color:#FFF;padding-top:20px;padding-bottom:6px">
        <table>
          <tr>
            <td><h3>Command line:</h3></td>
            <td>
              <h3><img src="/i/16/copy.png" style="vertical-align:bottom"/> <a href="https://github.com/Ensembl/ensembl-vep" rel="external">Clone from GitHub</a><span style="padding:0px 8px">|</span><img src="/i/16/download.png" style="vertical-align:bottom"/> <a href="https://github.com/Ensembl/ensembl-vep/archive/release/[[SPECIESDEFS::ENSEMBL_VERSION]].zip" rel="external">Download (zip)</a></h3>
            </td>
          </tr>
          <tr>
            <td><h3>REST API:</h3></td>
            <td>
              <h3><img src="/i/16/globe.png" style="vertical-align:bottom"/> <a href="http://rest.ensembl.org/documentation/info/variant_recoder" title="Variant Recoder REST endpoint" rel="external">Variant Recoder REST API</a></h3>
            </td>
          </tr>
        </table>
      </div>
    </div>


    <!-- Haplosaurus -->
    <div style="display:table-cell">
      <div class="tinted-box" style="padding-bottom:16px;margin-bottom:-1px">
        <div><h1 style="text-align:center;padding-top:0px;margin-bottom:16px">Haplosaurus</h1></div>
        <div style="color:#333">
          <p>Computes observed transcript <b>haplotype</b> sequences based on <b>phased genotype</b> data</p>
          <div><img src="/i/16/documentation.png" style="vertical-align:bottom"> <a href="/info/docs/tools/vep/haplo/index.html">Documentation</a></div>
        </div>
      </div>
      <div class="tinted-box" style="background-color:#FFF;padding-top:20px;padding-bottom:6px">
        <table>
          <tr>
            <td><h3>Command line:</h3></td>
            <td>
              <h3><img src="/i/16/copy.png" style="vertical-align:bottom"/> <a href="https://github.com/Ensembl/ensembl-vep" rel="external">Clone from GitHub</a><span style="padding:0px 8px">|</span><img src="/i/16/download.png" style="vertical-align:bottom"/> <a href="https://github.com/Ensembl/ensembl-vep/archive/release/[[SPECIESDEFS::ENSEMBL_VERSION]].zip" rel="external">Download (zip)</a></h3> 
            </td>
          </tr>
          <tr>
            <td><h3>REST API:</h3></td>
            <td>
              <h3><img src="/i/16/globe.png" style="vertical-align:bottom"/> <a href="http://rest.ensembl.org/documentation/info/transcript_haplotypes_get" title="Haplosaurus REST endpoint" rel="external">Haplosaurus REST API</a></h3>
            </td>
          </tr>
        </table>
      </div>
    </div>

  </div>    

</div>
<div style="clear:both"><h1 style="page-break-after: always">&nbsp</h1></div><div>
		
  <div style="float:right"><img src="/img/vep_logo.png"/></div>
  
  <h1 id="top"><span style="color:#006;padding-right:15px">Ensembl Variant Effect Predictor</span><span style="color:#666"><img src="/i/16/book.png"/> Tutorial</span></h1>
  <hr/>
   
    <div>
    <div style="float:left" class="info">
      <h3 id="read_write_access">Note</h3>
      <div class="message-pad">
        <p>
          If you're using a UNIX or Mac system, you can dive straight
          into this tutorial by opening your favourite terminal application. If you're
          on Windows you might like to have a look at the <a href="vep_download.html#windows"><img style="vertical-align:bottom" src="/i/16/windows.png"/> guide for Windows
          users</a> before starting.
        </p>
      </div>
    </div>
    <div class="clear"></div>
  </div>
 
  <br />
  <h2 id="install">Install Ensembl VEP</h2>
    
    <p> Have you downloaded Ensembl VEP yet? Use git to clone it:</p>

    <pre class="code sh_sh">git clone https://github.com/Ensembl/ensembl-vep
cd ensembl-vep</pre>
    
    <p> Ensembl VEP uses "cache files" or a remote database to read genomic data.
    Using cache files gives the best performance - let's set one up using the
    installer: </p>
    
    <pre class="code sh_sh">perl INSTALL.pl

Hello! This installer is configured to install v[[SPECIESDEFS::ENSEMBL_VERSION]] of the Ensembl API for use by VEP.
It will not affect any existing installations of the Ensembl API that you may have.

It will also download and install cache files from Ensembl's FTP server.

Checking for installed versions of the Ensembl API...done
It looks like you already have v[[SPECIESDEFS::ENSEMBL_VERSION]] of the API installed.
You shouldn't need to install the API

Skip to the next step (n) to install cache files

Do you want to continue installing the API (y/n)?</pre>
    
    <p> If you haven't yet installed the API, type "y" followed by enter,
    otherwise type "n" (perhaps if you ran the installer before). At the next
    prompt, type "y" to install cache files</p>
    
    <pre class="code sh_sh">Do you want to continue installing the API (y/n)? n
 - skipping API installation

Ensembl VEP can either connect to remote or local databases, or use local cache files.
Cache files will be stored in /nfs/users/nfs_w/wm2/.vep
Do you want to install any cache files (y/n)? y

Downloading list of available cache files
The following species/files are available; which do you want (can specify multiple separated by spaces):
1 : ailuropoda_melanoleuca_vep_[[SPECIESDEFS::ENSEMBL_VERSION]]_ailMel1.tar.gz
2 : anas_platyrhynchos_vep_[[SPECIESDEFS::ENSEMBL_VERSION]]_BGI_duck_1.0.tar.gz
3 : anolis_carolinensis_vep_[[SPECIESDEFS::ENSEMBL_VERSION]]_AnoCar2.0.tar.gz
...
42 : homo_sapiens_vep_[[SPECIESDEFS::ENSEMBL_VERSION]]_GRCh38.tar.gz
...

?</pre>
    <p> Type "42" (or the relevant number for homo_sapiens and GRCh38) to
    install the cache for the latest human assembly. This will take a little
    while to download and unpack! By default Ensembl VEP assumes you are working in
    human; it's easy to switch to any other
    species using <a href="vep_options.html#opt_species">--species
    [species]</a>. </p>
    
    <pre class="code sh_sh">? 42
 - downloading [[SPECIESDEFS::ENSEMBL_FTP_URL]]/release-[[SPECIESDEFS::ENSEMBL_VERSION]]/variation/vep/homo_sapiens_vep_[[SPECIESDEFS::ENSEMBL_VERSION]]_GRCh38.tar.gz
 - unpacking homo_sapiens_vep_[[SPECIESDEFS::ENSEMBL_VERSION]]_GRCh38.tar.gz

Success</pre>
    
    <p> By default Ensembl VEP installs cache files in a folder in your home area
    (<b>$HOME/.vep</b>); you can easily change this using the <b>-d</b> flag
    when running the installer. See the <a
    href="vep_download.html#installer">installer documentation</a> for more
    details. </p>
  
    <br /> 
    <hr />
    <h2 id="run">Run Ensembl VEP</h2>
 
    <p> Ensembl VEP needs some input containing variant positions to run. In their
    most basic form, this should just be a chromosomal location and a pair of
    alleles (reference and alternate). Ensembl VEP can also use common formats such
    as VCF and HGVS as input. Have a look at the <a href="../vep_formats.html#input">Data
    formats</a> page for more information. </p>
    
    <p> We can now use our cache file to run Ensembl VEP on the supplied example
    file <b>examples/homo_sapiens_GRCh38.vcf</b>, which is a VCF file containing variants from
    the 1000 Genomes Project, remapped to GRCh38: </p>
    
    <pre class="code sh_sh">./vep -i examples/homo_sapiens_GRCh38.vcf --cache

2013-07-31 09:17:54 - Read existing cache info
2013-07-31 09:17:54 - Starting...
ERROR: Output file variant_effect_output.txt already exists. Specify a different output file
with --output_file or overwrite existing file with --force_overwrite</pre>
    
    <p> You may see this error message if you've already run Ensembl VEP in the same directory.
    It will not overwrite your existing files unless you request this.
    So let's tell it to using <a
    href="vep_options.html#opt_force_overwrite">--force_overwrite</a> </p>
    
    <pre class="code sh_sh">./vep -i examples/homo_sapiens_GRCh38.vcf --cache --force_overwrite</pre>
    
    <p> By default Ensembl VEP writes to a file named "variant_effect_output.txt" -
    you can change this file name using <a
    href="vep_options.html#opt_output_file">-o</a>. Let's have a look at the
    output. </p>
    
    <pre class="code sh_sh">head variant_effect_output.txt

## ENSEMBL VARIANT EFFECT PREDICTOR v[[SPECIESDEFS::ENSEMBL_VERSION]].0
## Output produced at 2017-03-21 14:51:27
## Connected to homo_sapiens_core_[[SPECIESDEFS::ENSEMBL_VERSION]]_38 on ensembldb.ensembl.org
## Using cache in /homes/user/.vep/homo_sapiens/[[SPECIESDEFS::ENSEMBL_VERSION]]_GRCh38
## Using API version [[SPECIESDEFS::ENSEMBL_VERSION]], DB version [[SPECIESDEFS::ENSEMBL_VERSION]]
## polyphen version 2.2.2
## sift version sift5.2.2
## COSMIC version 78
## ESP version 20141103
## gencode version GENCODE 25
## genebuild version 2014-07
## HGMD-PUBLIC version 20162
## regbuild version 16
## assembly version GRCh38.p7
## ClinVar version 201610
## dbSNP version 147
## Column descriptions:
## Uploaded_variation : Identifier of uploaded variant
## Location : Location of variant in standard coordinate format (chr:start or chr:start-end)
## Allele : The variant allele used to calculate the consequence
## Gene : Stable ID of affected gene
## Feature : Stable ID of feature
## Feature_type : Type of feature - Transcript, RegulatoryFeature or MotifFeature
## Consequence : Consequence type
## cDNA_position : Relative position of base pair in cDNA sequence
## CDS_position : Relative position of base pair in coding sequence
## Protein_position : Relative position of amino acid in protein
## Amino_acids : Reference and variant amino acids
## Codons : Reference and variant codon sequence
## Existing_variation : Identifier(s) of co-located known variants
## Extra column keys:
## IMPACT : Subjective impact classification of consequence type
## DISTANCE : Shortest distance from variant to transcript
## STRAND : Strand of the feature (1/-1)
## FLAGS : Transcript quality flags
#Uploaded_variation  Location     Allele  Gene             Feature          Feature_type  Consequence        ...
rs7289170            22:17181903  G       ENSG00000093072  ENST00000262607  Transcript    synonymous_variant ...
rs7289170            22:17181903  G       ENSG00000093072  ENST00000330232  Transcript    synonymous_variant ...</pre>
    
    <p> The lines starting with "#" are header or meta information lines. The
    final one of these (highlighted in blue above) gives the column names for
    the data that follows. To see more information about Ensembl VEP's output
    format, see the <a href="../vep_formats.html#output">Data formats</a> page.
    </p>
    
    <p> We can see two lines of output here, both for the uploaded variant named
    rs7289170. In many cases, a variant will fall in more than one transcript.
    Typically this is where a single gene has multiple splicing variants.
    Here our variant has a consequence for
    the transcripts ENST00000262607 and ENST00000330232. </p>
    
    <p> In the consequence column, we can see the term
    'synonymous_variant'. This term is from the 
    <a rel="external" href="http://www.sequenceontology.org/">Sequence Ontology (SO)</a>.
    which describes the predicted molecular effects of sequence variants on genomic features.
    See our <a
    href="/info/genome/variation/prediction/predicted_data.html#consequences">predicted
    data</a> page for a guide to the consequence types used in Ensembl VEP. </p>
    
    <p> Let's try something a little more interesting. SIFT is an algorithm for
    predicting whether a given change in a protein sequence will be deleterious
    to the function of that protein. Ensembl VEP can give SIFT predictions for most
    of the missense variants in human and other highly studied species. To do this, simply add <a
    href="vep_options.html#opt_sift">--sift b</a> (the b means we want
    <b>b</b>oth the prediction and the score):</p>
    
    <pre class="code sh_sh">./vep -i examples/homo_sapiens_GRCh38.vcf --cache --force_overwrite --sift b</pre>
    
    <p> SIFT calls variants either "deleterious" or "tolerated". We can use the
     <a href="vep_filter.html">filtering tool</a> to find only those that
    SIFT considers deleterious: </p>
    
    <pre class="code sh_sh">./filter_vep -i variant_effect_output.txt -filter "SIFT is deleterious" | grep -v "##" | head -n5

#Uploaded_variation  Location     Allele  Gene             Feature          ...  Extra
rs2231495            22:17188416  C       ENSG00000093072  ENST00000262607  ...  SIFT=deleterious(0.05)
rs2231495            22:17188416  C       ENSG00000093072  ENST00000399837  ...  SIFT=deleterious(0.05)
rs2231495            22:17188416  C       ENSG00000093072  ENST00000399839  ...  SIFT=deleterious(0.05)
rs115736959          22:19973143  A       ENSG00000099889  ENST00000263207  ...  SIFT=deleterious(0.01)</pre>
    
    <p> Note that the SIFT score appears in the "Extra" column, as a key/value
    pair. This column can contain multiple key/value pairs depending on the
    options you give to Ensembl VEP. See the <a
    href="../vep_formats.html#output">Data formats</a> page for more information on
    the fields in the Extra column. </p>
    
    <p>You can also configure how Ensembl VEP writes its output using the <a
    href="vep_options.html#opt_fields">--fields</a> flag. </p>
    
    <p> You'll also see that we have multiple results for the same gene,
    ENSG00000093072. Let's say we're only interested in what is considered the
    canonical transcript for this gene (<a
    href="vep_options.html#opt_canonical">--canonical</a>), and that we want to
    know what the commonly used gene symbol from HGNC is for this gene (<a
    href="vep_options.html#opt_symbol">--symbol</a>). We can also use a UNIX
    pipe to pass the output from Ensembl VEP directly into the filtering tool:
    </p>
    
    <pre class="code sh_sh">./vep -i examples/homo_sapiens_GRCh38.vcf --cache --force_overwrite --sift b --canonical --symbol --tab --fields Uploaded_variation,SYMBOL,CANONICAL,SIFT -o STDOUT | \
./filter_vep --filter "CANONICAL is YES and SIFT is deleterious"

...

#Uploaded_variation  SYMBOL  CANONICAL  SIFT
rs2231495            CECR1   YES        deleterious(0.05)
rs115736959          ARVCF   YES        deleterious(0.01)
rs116398106          ARVCF   YES        deleterious(0)
rs116782322          ARVCF   YES        deleterious(0)
...                  ...     ...        ...
rs115264708          PHF21B  YES        deleterious(0.03)</pre>
    
    <p> So now we can see all of the variants that have a deleterious effect on
    canonical transcripts, and the symbol for their genes. Nice! </p>

    <p> For <a href="/info/genome/variation/species/species_data_types.html#sources">species with an Ensembl database of variants</a>, Ensembl VEP can be configured to annotate your input with identifiers and frequency data from variants co-located with your input data. For human, Ensembl VEP's cache contains frequency data from the 1000 Genomes Project and gnomAD. Since our input file is from the 1000 Genomes Project, let's add frequency data using <a href="vep_options.html#opt_af_1kg">--af_1kg</a>:</p>

    <pre class="code sh_sh">./vep -i examples/homo_sapiens_GRCh38.vcf --cache --force_overwrite --af_1kg -o STDOUT | grep -v "##" | head -n2

#Uploaded_variation  Location     Allele  Gene             Feature          ...  Existing_variation  Extra
rs7289170            22:17181903  G       ENSG00000093072  ENST00000262607  ...  rs7289170           IMPACT=LOW;STRAND=-1;AFR_AF=0.2390;AMR_AF=0.2003;EAS_AF=0.0456;EUR_AF=0.3211;SAS_AF=0.1401</pre>
    
    <p> We can see frequency data for the AFR, AMR, EAS, EUR and SAS continental population groupings; these represent the frequency of the alternate (ALT) allele from our input (G in the case of rs7289170). Note that the Existing_variation column is populated by the identifier of the variant found in the Ensembl VEP cache (and that it corresponds to the identifier from our input in Uploaded_variation). To retrieve only this information and not the frequency data, we could have used <a href="vep_options.html#opt_check_existing">--check_existing</a> (--af_1kg silently switches on --check_existing). </p>
   
    <br /> 
    <hr />
    <h2> Over to you! </h2>
    
    <p> This has been just a short introduction to the capabilities of Ensembl VEP -
    have a look through some more of the <a href="vep_options.html">options</a>,
    see them all on the command line using <a href="vep_options.html#opt_help">--help</a>, or try using the
    shortcut <a href="vep_options.html#opt_everything">--everything</a> which switches on almost all available output
    fields! Try out the different options in the <a
    href="vep_filter.html">filtering tool</a>, and if you're feeling
    adventurous why not use some of your <a href="vep_custom.html">own data to
    annotate your variants</a> or have a go with a <a
    href="vep_plugins.html">plugin</a> or two. </p>
	
	</div>
</div>

<div style="clear:both"><h1 style="page-break-after: always">&nbsp</h1></div>
<style>
tr:nth-child(odd) {background-color: #f0f0f0;}
code {background-color: rgb(240, 240, 240); font-family: Courier New, Courier, monospace;}
</style>
<div>
  <div style="float:right"><img src="/img/vep_logo.png"/></div>

  <h1 id="top"><span style="color:#006;padding-right:15px">Ensembl Variant Effect Predictor</span><span style="color:#666"><img src="/i/16/download.png"> Download and install</span></h1>
  <hr/>

  <div>
    <div style="float:left" class="info ">
      <h3 id="container">Note</h3>
      <div class="message-pad">
        <p>To address potential installation challenges with Ensembl VEP dependencies on newer systems, we recommend using <a href="#docker">Docker</a> or <a href="#singularity">Singularity</a> containers.

        <p>These containerised environments enable to access all the latest Ensembl VEP features without the hassle of installing dependencies directly on your system. Additionally, they are highly compatible with computer clusters.</p>
      </div>
    </div>
    <div class="clear"></div>
  </div>

  <h2 id="download">Download</h2>

  <p>Download ensembl-vep package (see below the different ways to download it) and then follow the <a href="#installer">installation instructions</a>.</p>

  <br />
  <h3>Using Git</h3>
  <h4><img src="/i/bullet_1.gif" style="vertical-align:baseline"/> Clone the Git repository</h4>
  <p> Use git to download the ensembl-vep package:</p>

  <pre class="code sh_sh">git clone https://github.com/Ensembl/ensembl-vep.git
cd ensembl-vep</pre>


<br />
  <h4><img src="/i/bullet_1.gif" style="vertical-align:baseline"/> Update to a newer version</h4>
  <p>To update from a previous version: </p>

  <pre class="code sh_sh">cd ensembl-vep
git pull
git checkout release/[[SPECIESDEFS::ENSEMBL_VERSION]]
perl INSTALL.pl</pre>

  <br />
  <h4><img src="/i/bullet_1.gif" style="vertical-align:baseline"/> Use an older version</h4>
  <p> To use an older version (this example shows how to set up release 87): </p>

  <pre class="code sh_sh">cd ensembl-vep
git checkout release/87
perl INSTALL.pl</pre>


  <br />
  <h3>Download the Zipped package file</h3>
  <p> Users without the git utility installed may download a zip file from GitHub, though we would always recommend using git if possible.</p>

  <pre class="code sh_sh" id="nogit">curl -L -O https://github.com/Ensembl/ensembl-vep/archive/release/[[SPECIESDEFS::ENSEMBL_VERSION]].zip
unzip [[SPECIESDEFS::ENSEMBL_VERSION]].zip
cd ensembl-vep-release-[[SPECIESDEFS::ENSEMBL_VERSION]]/</pre>

  <br />
  <h3 id="toolsversions"> Previous versions (ensembl-tools)</h3>
  <p>
    Previously, Ensembl  VEP was available as part of the ensembl-tools package (see the <a href="http://e87.ensembl.org/info/docs/tools/vep/script/index.html">Ensembl archive site</a> for documentation). The following downloads are available for archival purposes. <a class="button" href="#toolsversions" onclick="show_hide('toolsversions');" id="a_toolsversions">Show versions</a>
  </p>

  <div id="div_toolsversions" style="display:none;">
  <ul>
    <li><a href="https://github.com/Ensembl/ensembl-tools/archive/release/87.zip">Download version 87</a> (Ensembl 87) </li>
    <li><a href="https://github.com/Ensembl/ensembl-tools/archive/release/86.zip">Download version 86</a> (Ensembl 86) </li>
    <li><a href="https://github.com/Ensembl/ensembl-tools/archive/release/85.zip">Download version 85</a> (Ensembl 85) </li>
    <li><a href="https://github.com/Ensembl/ensembl-tools/archive/release/84.zip">Download version 84</a> (Ensembl 84) </li>
    <li><a href="https://github.com/Ensembl/ensembl-tools/archive/release/83.zip">Download version 83</a> (Ensembl 83) </li>
    <li><a href="https://github.com/Ensembl/ensembl-tools/archive/release/82.zip">Download version 82</a> (Ensembl 82) </li>
    <li><a href="https://github.com/Ensembl/ensembl-tools/archive/release/81.zip">Download version 81</a> (Ensembl 81) </li>
    <li><a href="https://github.com/Ensembl/ensembl-tools/archive/release/80.zip">Download version 80</a> (Ensembl 80) </li>
    <li><a href="https://github.com/Ensembl/ensembl-tools/archive/release/79.zip">Download version 79</a> (Ensembl 79) </li>
    <li><a href="https://github.com/Ensembl/ensembl-tools/archive/release/78.zip">Download version 78</a> (Ensembl 78) </li>
    <li><a href="https://github.com/Ensembl/ensembl-tools/archive/release/77.zip">Download version 77</a> (Ensembl 77) </li>
    <li><a href="https://github.com/Ensembl/ensembl-tools/archive/release/76.zip">Download version 76</a> (Ensembl 76) </li>
    <li><a href="https://github.com/Ensembl/ensembl-tools/archive/release/75.zip">Download version 75</a> (Ensembl 75) </li>
    <li><a href="https://github.com/Ensembl/ensembl-tools/archive/release/74.zip">Download version 74</a> (Ensembl 74) </li>
    <li><a href="https://github.com/Ensembl/ensembl-tools/archive/release/73.zip">Download version 73</a> (Ensembl 73) </li>
    <li><a href="https://github.com/Ensembl/ensembl-tools/archive/release/72.zip">Download version 72</a> (Ensembl 72) </li>
    <li><a href="https://github.com/Ensembl/ensembl-tools/archive/release/71.zip">Download version 71</a> (Ensembl 71) </li>
    <li><a href="https://github.com/Ensembl/ensembl-tools/archive/release/70.zip">Download version 2.8</a> (Ensembl 70) </li>
    <li><a href="https://github.com/Ensembl/ensembl-tools/archive/release/69.zip">Download version 2.7</a> (Ensembl 69) </li>
    <li><a href="https://github.com/Ensembl/ensembl-tools/archive/release/68.zip">Download version 2.6</a> (Ensembl 68) </li>
    <li><a href="https://github.com/Ensembl/ensembl-tools/archive/release/67.zip">Download version 2.5</a> (Ensembl 67) </li>
    <li><a href="https://github.com/Ensembl/ensembl-tools/archive/release/66.zip">Download version 2.4</a> (Ensembl 66) </li>
    <li><a href="https://github.com/Ensembl/ensembl-tools/archive/release/65.zip">Download version 2.3</a> (Ensembl 65) </li>
    <li><a href="https://github.com/Ensembl/ensembl-tools/archive/release/64.zip">Download version 2.2</a> (Ensembl 64 - ensembl-tools/scripts/variant_effect_predictor) </li>
    <li><a href="https://github.com/Ensembl/ensembl-variation/archive/release/63.zip">Download version 2.1</a> (Ensembl 63) </li>
    <li><a href="https://github.com/Ensembl/ensembl-variation/archive/release/62.zip">Download version 2.0</a> (Ensembl 62 - ensembl-variation/scripts/examples) </li>
  </ul>
  </div><hr/>

  <h2 id="new">What's new?</h2>

<div class="toolbox" style="float:none;padding:6px">

    <h4>New in version [[SPECIESDEFS::ENSEMBL_VERSION]] <i>([[SPECIESDEFS::ENSEMBL_RELEASE_DATE]])</i></h4>

    <ul>
      <li>Added Ensembl VEP support for annotating your structural variants with allele frequencies from gnomAD and clinical significance (CLINSIG) from ClinVar.</li>
      <li>Added Ensembl VEP and Ensembl Variation API support for the new <a href="https://doi.org/10.1093/nar/gkae1090" rel="external">ClinVar somatic classifications.</a></li>
      <li>We have enabled support for <a href="https://www.gencodegenes.org/pages/promoter_windows/" rel="external">GENCODE promoters,</a> variants falling within them can now be annotated with details of the promoter</li>
      <li>New plugin (on CLI):</li>
      <ul>
        <li><a href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/MechPredict.pm" rel="external">MechPredict</a></li>
      </ul>
    </ul>
</div>


  <br />
  <h4 id="history">Previous version history - from version 88: <a class="button" href="#history" onclick="show_hide('history');" id="a_history" style="font-weight:normal">Show</a></h4>

  <div id="div_history" style="display:none">

 <h4>New in version 114 <i>(May 2025)</i></h4>

  <ul>
      <li>MAVE data has been updated from the latest version of MaveDB, representing a nearly 6.5 fold increase in variants covered (~1.2 million to ~7.7 million).</li>
      <li>Support for https protocol when downloading FTP files and adding GitHub Token to increase rate limit in Ensembl VEP install script.</li>
      <li>Allele frequency from NIH AllOfUs study is now available in the web Ensembl VEP.</li>
      <li>Plugin support added to REST for: </li>
      <ul>
        <li><a href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/Paralogues.pm" rel="external">Paralogues</a></li>
      </ul>
      <li>Plugin data version updated: </li>
      <ul>
        <li><a href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/dbNSFP.pm" rel="external">dbNSFP</a> (from 4.7c to 4.9c)</li>
        <li><a href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/LOEUF.pm" rel="external">LOEUF</a> (from gnomAD v2.1.1 to gnomAD v4.1)</li>
      </ul>
      <li>Plugin deprecated: </li>
      <ul>
        <li><a href="https://github.com/Ensembl/VEP_plugins/blob/release/113/DisGeNET.pm" rel="external">DisGeNET</a></li>
        <li><a href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/Mastermind.pm" rel="external">Mastermind</a> (Only from REST)</li>
      </ul>

  </ul>
</div>

    <h4>New in version 113 <i>(October 2024)</i></h4>
    <div class="left-margin" style="border-left:2px solid #7b8baf">
  <ul>
        <li>gnomAD frequency data updated to v4.1 for both genomes and exomes.</li>
        <li>Support for GENCODE primary transcript set added. See, <a href="vep_options.html#opt_gencode_primary">--gencode_primary</a> and <a href="vep_options.html#opt_flag_gencode_primary">--flag_gencode_primary</a>.</li>
        <li>Support added for <a href="vep_options.html#opt_mane">--mane</a>, <a href="vep_options.html#opt_mane_select">--mane_select</a>, and <a href="vep_options.html#opt_canonical">--canonical</a> when GFF/GTF file used as annotation source.</li>
        <li>Nextflow Ensembl VEP now supports other input data formats besides VCF. For supported formats see - <a href="../vep_formats.html#input">Data formats</a>.</li>
  
        <li>Plugin support added to REST and Web for: </li>
        <ul>
          <li><a href="//github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/RiboseqORFs.pm" rel="external">RiboseqORFs</a></li>
          <li><a href="//github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/REVEL.pm" rel="external">REVEL</a></li>
          <li><a href="//github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/ClinPred.pm" rel="external">ClinPred</a></li>
        </ul>
        <li>Plugin support added to Web for: </li>
        <ul>
          <li><a href="//github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/Paralogues.pm" rel="external">Paralogues</a></li>
        </ul>
        <li>Plugin support added to REST for: </li>
        <ul>
          <li><a href="//github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/LOEUF.pm" rel="external">LOEUF</a></li>
        </ul>
        <li>Plugin data version updated for CADD (v1.6 to v1.7) and dbNSFP (4.5c to 4.7c).</li>
  
  </ul>
  </div>
  
    <h4>New in version 112 <i>(May 2024)</i></h4>
    <div class="left-margin" style="border-left:2px solid #7b8baf">
  <ul>
      <li>Enhanced Structural Variant Support:</li>
      <ul>
        <li>Added support for CNV:TR</li>
        <li>Enabled the use of chromosome synonyms in breakends</li>
        <li>Report consequences for each breakend and enable the input of single breakends</li>
      </ul>

      <li>New plugins (supported on CLI, Web and REST): </li>
      <ul>
        <li><a href="//github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/AlphaMissense.pm" rel="external">AlphaMissense</a> - annotates missense variants with the pre-computed AlphaMissense pathogenicity scores. AlphaMissense is a deep learning model developed by Google DeepMind that predicts the pathogenicity of single nucleotide missense variants. </li>
      </ul>

      <li>New plugins (supported on CLI and Web): </li>
      <ul>
        <li><a href="//github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/RiboseqORFs.pm" rel="external">RiboseqORFs</a> - uses a standardized catalog of human Ribo-seq ORFs to re-calculate consequences for variants located in these translated regions</li>
      </ul>

    <li>New plugins (supported on CLI): </li>
      <ul>
        <li><a href="//github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/Paralogues.pm" rel="external">Paralogues</a> - fetches variants overlapping the genomic coordinates of amino acids aligned between paralogue proteins</li>
        <li><a href="//github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/AVADA.pm" rel="external">AVADA</a> - Automatic VAriant evidence DAtabase is a novel machine learning tool that uses natural language processing to automatically identify pathogenic genetic variant evidence in full-text primary literature about monogenic disease and convert it to genomic coordinates</li>
        <li><a href="//github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/GeneBe.pm" rel="external">GeneBe</a> - A plugin kindly contributed by the GeneBe team, it retrieves automatic ACMG variant classification data from https://genebe.net/</li>
        <li><a href="//github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/PhenotypeOrthologous.pm" rel="external">PhenotypeOrthologous</a> A VEP plugin that retrieves phenotype information associated with orthologous genes from model organisms</li>
      </ul>

      <li>Plugin support added to REST and Web for: </li>
      <ul>
        <li><a href="//github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/CADD.pm" rel="external">CADD_SV</a></li>
        <li><a href="//github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/CADD.pm" rel="external">CADD</a> scores for <i>Sus scrofa</i></li>
        <li><a href="//github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/DosageSensitivity.pm" rel="external">Dosage Sensitivity</a></li>
        <li><a href="//github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/Enformer.pm" rel="external">Enformer</a></li>
      </ul>

  </ul>
  </div>

  <h4>New in version 111 <i>(January 2024)</i></h4>
    <div class="left-margin" style="border-left:2px solid #7b8baf">
  <ul>
    <li>New option <a href="vep_options.html#opt_individual_zyg">--individual_zyg</a> returns a single list of individuals and their zygosity (instead of a separate line of output for each individual and variant combination like in <a href="vep_options.html#opt_individual">--individual</a>)</li>
    <li><a href="vep_custom.html">Custom annotation</a> has been improved with the following options:</li>
      <ul>
        <li><a href="vep_custom.html#opt_num_records">num_records</a> to limit the number of matching records (50 by default)</li>
        <li><a href="vep_custom.html#opt_summary_stats">summary_stats</a> to calculate summary statistics (min, mean, max, count, sum) using annotation scores (not used by default)</li>
      </ul>

    <li>New plugin (supported on CLI, REST and web): </li>
      <ul>
        <li><a href="//github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/OpenTargets.pm" rel="external">OpenTargets</a> - adds locus-to-gene (L2G) scores to predict causal genes at GWAS loci from Open Targets Genetics</li>
      </ul>

    <li>New plugin (supported on CLI and REST): </li>
      <ul>
        <li><a href="//github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/Enformer.pm" rel="external">Enformer</a> - adds pre-calculated predictions of variant impact on gene expression</li>
      </ul>

    <li>New plugins (supported on CLI): </li>
      <ul>
        <li><a href="//github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/BayesDel.pm" rel="external">BayesDel</a> - adds a deleteriousness meta-score combining multiple deleteriousness predictors</li>
        <li><a href="//github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/DeNovo.pm" rel="external">DeNovo</a> - identifies de novo variants in a VCF file. This plugin requires a pedigree (.ped) file</li>
        <li><a href="//github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/SpliceVault.pm" rel="external">SpliceVault</a> - predicts exon-skipping events and activated cryptic splice sites based on the most common mis-splicing events around a splice site</li>
        <li><a href="//github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/DosageSensitivity.pm" rel="external">DosageSensitivity</a> - annotates the likelihood of a gene being haploinsufficient or triplosensitive</li>
        <li><a href="//github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/VARITY.pm" rel="external">VARITY</a> - adds pre-calculated pathogenicity scores of rare human missense variants</li>
      </ul>
  </ul>

  </div>

    <h4>New in version 110 <i>(July 2023)</i></h4>
  <div class="left-margin" style="border-left:2px solid #7b8baf">
  <ul>
    <li>New plugins (supported on CLI): </li>
      <ul>
        <li><a href="//github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/TranscriptAnnotator.pm" rel="external">TranscriptAnnotator</a> - an Ensembl VEP plugin that annotates variant-transcript pairs</li>
      </ul>
    <li>New Plugins (supported on CLI, REST and web): </li>
      <ul>
        <li><a href="//github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/Geno2MP.pm" rel="external">Geno2MP</a> - adds information from Geno2MP, a web-accessible database of rare variant genotypes linked to phenotypic information</li>
        <li><a href="//github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/MaveDB.pm" rel="external">MaveDB</a> - adds information from MaveDB, a database that holds experimentally determined measures of variant effect</li>
      </ul>
  </ul>

  </div>

    <h4>New in version 109 <i>(February 2023)</i></h4>
  <div class="left-margin" style="border-left:2px solid #7b8baf">
  <ul>
    <li><a href="https://hub.docker.com/r/ensemblorg/ensembl-vep">Ensembl VEP Docker image</a> now includes all Ensembl VEP plugins</li>
    <li>New plugin (supported on CLI): </li>
      <ul>
        <li><a href="//github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/GWAS.pm" rel="external">GWAS</a> - reports genome-wide association study data from GWAS catalog</li>
      </ul>
    <li>Plugins now available in REST and web: </li>
      <ul>
        <li><a href="//github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/UTRAnnotator.pm" rel="external">UTRAnnotator</a> - annotates the effect of 5' UTR variant especially for variant creating/disrupting upstream ORFs</li>
      </ul>
    <li>Plugins now available in REST: </li>
      <ul>
        <li><a href="//github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/NMD.pm" rel="external">NMD</a> - predicts if a variant allows transcript to escape nonsense-mediated mRNA decay based on certain rules</li>
      </ul>
    <li>Plugin LOEUF replaces Loftool in the web with more recent ‘loss-of-function’ score for variants</li>
    <li>Deprecated Plugins: </li>
      <ul>
        <li><a href="//github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/miRNA.pm" rel="external">miRNA</a> - this plugin was fully deprecated in favour of --mirna flag (in web and REST)</li>
        <li><a href="//github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/ExAC.pm" rel="external">ExAC</a> - this plugin was deprecated given that Ensembl VEP cache includes ExAC data as part of gnomAD</li>
      </ul>
    <li>SIFT version has been updated from 5.2.2 to 6.2.1 (except for human GRCh37)</li>
    <li>PolyPhen-2 version has been updated from 2.2.2 to 2.2.3 (except for human GRCh37)</li>
  </ul>

  </div>

    <h4>New in version 108 <i>(October 2022)</i></h4>
  <div class="left-margin" style="border-left:2px solid #7b8baf">
  <ul>
    <li>New plugin (supported on CLI, REST, and web): </li>
      <ul>
        <li><a href="//github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/mutfunc.pm" rel="external">mutfunc</a> - predicts destabilization of protein structure, interaction and others features by a variant (GRCh38 only)</li>
      </ul>
    <li>Plugin feature extension: </li>
      <ul>
        <li><a href="//github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/IntAct.pm" rel="external">IntAct</a> - 4 new species are now supported - rat, chicken (red jungle fowl), yeast, and arabidopsis</li>
      </ul>
  </ul>
  
  </div>

    <h4>New in version 107 <i>(July 2022)</i></h4>
  <div class="left-margin" style="border-left:2px solid #7b8baf">
  <ul>
    <li>New plugin (supported on CLI, REST, and web): </li>
      <ul>
        <li><a href="//github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/EVE.pm" rel="external">EVE</a> - annotates human variants using EVA classification method based solely on evolutionary sequences (GRCh38 only)</li>
      </ul>
    <li>Plugins now available in REST and web (already available in CLI): </li>
      <ul>
        <li><a href="//github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/GO.pm" rel="external">GO</a> - retrieves Gene Ontology terms associated with transcripts/translations</li>
        <li><a href="//github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/IntAct.pm" rel="external">IntAct</a> - annotates human variants which fall in interaction sites, as described in the IntAct database</li>
      </ul>
      <li>Plugins now available in web (already available in CLI): </li>
      <ul>
        <li><a href="//github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/NMD.pm" rel="external">NMD</a> - predicts if a stop_gained variant allows transcript to escape nonsense-mediated mRNA decay based on certain rules</li>
      </ul>
    <li>Readthrough transcripts are now removed from cache</li>
    <li>Transcripts of biotype ‘artifact’ which are artifactual duplication are now removed from cache and not accessible using database</li>
    <li>gnomaAD allele frequencies are now available for exomes and genomes separately through —af_gnomade and —af_gnomadg options respectively. The —af_gnomad option have same function as --af_gnomade.</li>
  </ul>

</div>

    <h4>New in version 106 <i>(April 2022)</i></h4>
  <div class="left-margin" style="border-left:2px solid #7b8baf">
  <ul>
    <li>New plugins for command line use: </li>
      <ul>
        <li><a href="//github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/IntAct.pm" rel="external">IntAct</a> - annotates human variants which fall in interaction sites, as described in the IntAct database</li>
        <li><a href="//github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/CAPICE.pm" rel="external">CAPICE</a> - integrates scored from a machine-learning-based method for prioritizing pathogenic variants (GRCh37 only)</li>
      </ul>
    <li>Nextflow pipeline:</li>
      <ul>
        <li>A new configurable pipeline is available to run Ensembl VEP efficiently on large scale VCF</li>
      </ul>
  </ul>

  </div>

    <h4>New in version 105 <i>(December 2021)</i></h4>
  <div class="left-margin" style="border-left:2px solid #7b8baf">
    <ul>
      <li>3 new Sequence Ontology terms are reported for more detailed splice consequence annotation </li>
        <ul>
          <li>splice_donor_5th_base_variant (<a rel="external" href="http://www.sequenceontology.org/miso/current_release/term/SO:0001787">SO:0001787</a>)</li>
          <li>splice_donor_region_variant   (<a rel="external" href="http://www.sequenceontology.org/miso/current_release/term/SO:0002170">SO:0002170</a>)</li>
          <li>splice_polypyrimidine_tract_variant (<a rel="external" href="http://www.sequenceontology.org/miso/current_release/term/SO:0002169">SO:0002169</a>)</li>
        </ul>
      <li>New plugins </li>
        <ul>
          <li><a href="//github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/ClinPred.pm" rel="external">ClinPred</a> - adds pre-calculated scores from ClinPred which helps identify disease-relevant missense variants</li>
          <li><a href="//github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/NMD.pm" rel="external">NMD</a> - predicts whether a stop-gained variant will allow a transcript to escape nonsense-mediated decay</li>
        </ul>
      <li>Condel scores are no longer available via the Ensembl VEP web interface as they have not been updated since 2014 and newer scores like CADD and REVEL are available</li>
    </ul>

  </div>

    <h4>New in version 104 <i>(May 2021)</i></h4>
  <div class="left-margin" style="border-left:2px solid #7b8baf">
    <ul>
      <li>Human GRCh37 cache files now include dbSNP 154! </li>
      <li><a href="vep_options.html#opt_var_synonyms">--var_synonyms</a> output structure has been altered when used with <a href="vep_options.html#opt_json">--json</a></li>
    </ul>
    <ul><li>Ensembl VEP Plugins:</li>
     <ul>
      <li><a href="//github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/dbNSFP.pm" rel="external">dbNSFP</a> - now supports matching by peptides</li>
      <li><a href="//github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/SpliceAI.pm" rel="external">SpliceAI</a> - now compares gene symbols to improve score accuracy</li>
     </ul>
    </ul>
  </div>

    <h4>New in version 103 <i>(February 2021)</i></h4>
  <div class="left-margin" style="border-left:2px solid #7b8baf">
    <ul>
      <li><b>New:</b> Variant Recoder is now available as a web tool</li>
      <li>Variant Recoder output is now allele specific</li>
    </ul>
    <ul><li>Web Ensembl VEP Options:</li>
      <ul>
        <li>Variant Synonyms are now available through the web interface</li>
        <li>MasterMind results are available through the REST and web interfaces</li>
      </ul>
    </ul>
    <ul><li>Ensembl VEP Options:</li>
      <ul>
        <li><a href="vep_options.html#opt_mane">--mane</a> : Now provides additional MANE Plus Clinical annotations alongside MANE Select</li>
        <li><a href="vep_options.html#opt_mane_select">--mane_select</a> : Returns MANE Select annotations </li>
      </ul>
    </ul>

  </div>



      <h4>New in version 102 <i>(November 2020)</i></h4>
    <div class="left-margin" style="border-left:2px solid #7b8baf">
      <ul>
        <li>Ensembl VEP options:
          <ul>
            <li><a href="vep_options.html#opt_uniprot">--uniprot</a>: Now we report precise Ensembl translation to UniProt isoform mappings.</li>
            <li><a href="vep_options.html#opt_spdi">--spdi</a> - <b>new</b>: Add genomic <a rel="external" href="https://www.ncbi.nlm.nih.gov/variation/notation/",
            target="_blank">SPDI</a> notation.</li>
          </ul>
        </li>
        <li> Web Ensembl VEP options:
          <ul>
  	  <li>Shifting variants in the 3' direction with <a href="vep_options.html#opt_shift_3prime">--shift_3prime</a> and <a href="vep_options.html#opt_shift_genomic">--shift_genomic</a> is now supported through the web interface.</li>
            <li><a href="//github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/SpliceAI.pm" rel="external">SpliceAI</a> - <b>new</b>: SpliceAI pre-calculated scores are available through the web interface.</li>
          </ul>
        </li>
        <li>Ensembl VEP filter options:
          <ul>
            <li><a href="vep_filter.html#opt_softfilter">--soft_filter</a> - <b>new</b>: Option to only flag the failing variation in the FILTER column and keep the entries in the output VCF file.</li>
          </ul>
        </li>
      </ul>

    </div>
    <br/>




    <h4>New in version version 101 <i>(August 2020)</i></h4>
  <div class="left-margin" style="border-left:2px solid #7b8baf">
    <ul>
      <li>New options:
        <ul>
          <li><a href="vep_options.html#opt_var_synonyms">--var_synonyms</a>: Report known synonyms for colocated variants. Must be used with <a href="vep_options.html#opt_cache">--cache</a>.</li>
        </ul>
      </li>
      <li>Ensembl VEP plugins:
        <ul>
          <li><a href="//github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/neXtProt.pm" rel="external">neXtProt</a> - <b>new</b>: neXtProt retrieves comprehensive human-centric protein-related data for missense variants </li>
        </ul>
      </li>
    </ul>

  </div>
  <br/>

    <h4>New in version 100 <i>(April 2020)</i></h4>

    <div class="left-margin" style="border-left:2px solid #7b8baf">
      <ul>
        <li>Human GRCh37 variant and phenotype data has been updated with multiple data sets including dbSNP153, ClinVar's 201912 release and COSMIC release 90</li>
        <li>The GRCh37 RefSeq transcript set has been updated to NCBI's 1st November 2019 release (initially annotated on GCF_000001405.25)!
        <li>New options:
          <ul>
            <li><a href="vep_options.html#opt_shift_3prime">--shift_3prime</a>: Right aligns all variants relative to their associated transcripts prior to consequence calculation</li>
            <li><a href="vep_options.html#opt_shift_genomic">--shift_genomic</a>: Right aligns all variants, including intergenic variants, before consequence calculation and updates the <i>Location</i> field</li>
          </ul>
        </li>
        <li>Ensembl VEP plugins:
          <ul>
            <li><a href="//github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/SpliceAI.pm" rel="external">SpliceAI</a> - <b>new</b>: SpliceAI is a deep neural network, developed by Illumina, Inc that predicts splice junctions from an arbitrary pre-mRNA transcript sequence.</li>
          </ul>
        </li>
      </ul>
    </div>
    <br/>


    <h4>New in version 99 <i>(January 2020)</i></h4>

    <div class="left-margin" style="border-left:2px solid #7b8baf">
      <ul>
        <li>Human GRCh38 cache files now contain variants from dbSNP153</li>
        <li>New options have been added to REST:
          <ul>
            <li>vcf_string: Ensembl VEP can now provide a VCF-like string representing the input variant </li>
            <li>transcript_version: Add version numbers to Ensembl transcript identifiers </li>
            <li>SpliceRegion: Provides granular predictions of splicing effects (<a target="_blank" href="https://raw.githubusercontent.com/ensembl-variation/VEP_plugins/master/SpliceRegion.pm">Details</a>)  </li>
            <li>LoF: LOFTEE implements a set of filters to predict LoF (loss-of-function) variants. (<a target="_blank" href="https://github.com/konradjk/loftee/blob/master/README.md">Details</a>) </li>
          </ul>
        </li>
      </ul>
    </div>
    <br/>

  <h4>New in version 98 <i>(September 2019)</i></h4>

  <div class="left-margin" style="border-left:2px solid #7b8baf">
    <ul>
      <li>Human GRCh38 cache files now contain variants from dbSNP152</li>
      <li>This employs a new clustering strategy which may result in different rsIDs being reported as known variants for some insertions and deletions - for more information see <a href='http://www.ensembl.info/2019/08/29/coming-soon-to-an-ensembl-near-you-dbsnp-2-0/'> here </a></li>
      <li><a href="vep_options.html#opt_clin_sig_allele">--clin_sig_allele</a> has been updated to be used by default</li>
      <li>New options:
        <ul>
          <li><a href="vep_options.html#opt_custom_multi_allelic">--custom_multi_allelic</a>: prevents Ensembl VEP from assuming that comma separated lists in custom annotations are allele specific</li>
        </ul>
      </li>
      <li>MANE attributes are now included within Ensembl VEP cache files, web Ensembl VEP and REST</li>
      <li>Ensembl VEP plugins:
        <ul>
          <li><a href="//github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/satMutMPRA.pm" rel="external">satMutMPRA</a> - <b>new</b>: measures variant effects on gene RNA expression for 21 regulatory elements</li>
        </ul>
      </li>
      <li> Ensembl VEP Installer: </li>
        <ul>
          <li>HTSLib v1.9 is now installed by default (previously v1.3.2)</li>
          <li>Bio::DB::HTS v2.11 is now installed by default (previously v2.9)</li>
          <li>New option 'PLUGINSDIR' allows you to specify the installation directory for plugins</li>    
        </ul>
    </ul>
  </div>

  <br/>
  
  <h4>New in version 97 <i>(July 2019)</i></h4>

  <div class="left-margin" style="border-left:2px solid #7b8baf">
    <ul>
      <li>Allele-specific clinical significance reported (it was previously variant-specific).</li>
      <li>New options:
        <ul>
          <li><a href="vep_options.html#opt_clin_sig_allele">--clin_sig_allele</a>: report allele specific clinical significance.</li>
          <li><a href="vep_options.html#opt_mane">--mane</a>: report if a transcript is the MANE Select.</li>
          <li><a href="vep_options.html#opt_max_sv_size">--max_sv_size</a>: extend the maximum Structural Variant size Ensembl VEP can process.</li>
          <li><a href="vep_options.html#opt_no_check_variants_order">--no_check_variants_order</a>: permit the use of unsorted input files (WARNING - this is slow and requires more memory).</li>
          <li><a href="vep_options.html#opt_overlaps">--overlaps</a>: report the proportion and length of a transcript overlapped by a structural variant in VCF format.</li>          
        </ul>
      </li>
      <li>Include the <a href="vep_options.html#opt_mane">--mane</a> option into the <a href="vep_options.html#opt_everything">--everything</a> group option.</li>
      <li>Update <a href="vep_options.html#opt_pick">--pick</a> and <a href="vep_options.html#opt_pick_order">--pick_order</a> to support MANE Select transcripts.</li>
      <li>Check if the input variants are ordered: non ordered variants slow down Ensembl VEP and require more memory.</li>
      <li>Skip annotation of complex and long structural variants and display a warning message.</li>
      <li>Variant recoder: add an option <a href="/info/docs/tools/vep/recoder/index.html#opt_vcf_string">--vcf_string</a> to return results in VCF format.</li>
      <li>Ensembl VEP plugins:
        <ul>
          <li><a href="//github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/FunMotifs.pm" rel="external">FunMotifs</a> - <b>new</b>: provide information about overlapping tissue-specific transcription factor motifs.</li>
          <li><a href="//github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/Mastermind.pm" rel="external">Mastermind</a> - <b>new</b>: reports variants that have clinical evidence cited in the medical literature.</li>
          <li><a href="//github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/StructuralVariantOverlap.pm" rel="external">StructuralVariantOverlap</a> - <b>new</b>: provide information from overlapping structural variants.</li>
          <li><a href="//github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/G2P.pm" rel="external">G2P</a> - <b>update</b>: now the plugin can be run offline.</li>
          <li><a href="//github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/Phenotypes.pm" rel="external">Phenotypes</a> - <b>update</b>: change the format of the data file (from BED to GVF).</li>
        </ul>
      </li>
      <li>Ensembl VEP web tool: the transcript identifiers are now returned with versions unless otherwise specified.</li>
      <li>Ensembl VEP installer: tabix-indexed variant cache files are now installed by default.</li>
    </ul>
  </div>


  <br />

  <h4>New in version 96 <i>(April 2019)</i></h4>

  <div class="left-margin" style="border-left:2px solid #7b8baf">
    <ul>
      <li>Add <b>SPDI</b> format for Ensembl VEP (input) and Variant Recoder (input and output).</li>
      <li>Update Ensembl VEP cache with <b>gnomAD 2.1</b> (human).</li>
      <li>Update the Docker Ensembl VEP base image to <b>Ubuntu 18.04</b>.</li>
      <li>Retire deprecated flags: --gmaf, --maf_1kg, --maf_esp, --maf_exac, --check_alleles, --html, --gvf.</li>
      <li>Retire legacy code about the pileup input format, which is no longer supported.
      <li>Deprecate the installation flag "--VERSION"</li>
      <li>Force numbers to be encoded as numbers in JSON output</li>
      <li>Ensembl VEP plugins:
        <ul>
          <li><a href="//github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/NearestExonJB.pm" rel="external">NearestExonJB</a> - <b>new</b>: find the nearest exon junction boundary to a coding sequence variant.</li>
          <li><a href="//github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/Conservation.pm" rel="external">Conservation</a> - update: can use BigWig files instead of the Ensembl Compara database.</li>
          <li><a href="//github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/dbNSFP.pm" rel="external">dbNSFP</a> - update: support of the dbNSFP data version 4.</li>
          <li><a href="//github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/Phenotypes.pm" rel="external">Phenotypes</a> - update: possibility to report the phenotype description(s) and other information.</li>
          <li><a href="//github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/PostGAP.pm" rel="external">PostGAP</a> - update: replace the plugin name POSTGAP to PostGAP.</li>
        </ul>
      </li>
    </ul>
  </div>  


  <br />
  <h4>New in version 95 <i>(January 2019)</i></h4>  

  <div class="left-margin" style="border-left:2px solid #7b8baf">
    <ul>
      <li>The Ensembl VEP parser is now more permissive for the GFF files (ID attribute only required for genes and transcripts)</li>
      <li>Add new option <a href="vep_options.html#opt_show_ref_allele">--show_ref_allele</a> to include the allele reference in the VEP default output and the tab output formats</li>
      <li>Add a warning message when the Ensembl VEP annotations INFO field hasn't been found/recognised in the VCF input file</li>
      <li>Ensembl VEP Docker image:
        <ul>
          <li>Reduce the size of the Ensembl VEP Docker image by about 45%.</li>
          <li>Include the Linkage disequilibrium script in the Ensembl VEP Docker image, making possible to run the LD plugin</li>
        </ul>
      </li>
      <li>New Ensembl VEP plugins:
        <ul>
          <li>
            <a href="//github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/ReferenceQuality.pm" rel="external">Reference quality</a>
          </li>
          <li>
            <a href="//github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/POSTGAP.pm" rel="external">OpenTargets results (POSTGAP)</a>
          </li>
          <li>
            <a href="//github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/SingleLetterAA.pm" rel="external">Single letter amino acid for HGVS</a>
          </li>
        </ul>
      </li>
    </ul>
  </div>


  <br />
  <h4>New in version 94 <i>(October 2018)</i></h4>

  <div class="left-margin" style="border-left:2px solid #7b8baf">
    <ul>
      <li>RefSeq transcript version updated.</li>
      <li>Minor updates on the <a href="/Tools/VEP">Ensembl VEP web tool</a> interface.</li>
      <li>When the input data format is not specified on the command line, Ensembl VEP attempts to detect it. The assumed format is now reported in verbose mode (<a href="vep_options.html#opt_verbose">--verbose</a>).</li>
      <li>Ensembl VEP assigns assigned the consequence types <i>TF_binding_site_variant</i>, <i>TFBS_ablation</i>, <i>TFBS_fusion</i>, <i>TFBS_amplification</i> and <i>TFBS_translocation</i> to human and mouse variants which overlapped motif features. These annotations will not be available in VEP caches for human in release 94 so must be added as a <a href="/info/docs/tools/vep/script/vep_custom.html">custom annotation</a>.</li>
    </ul>
  </div>


  <br />
  <h4>New in version 93 <i>(July 2018)</i></h4>

  <div class="left-margin" style="border-left:2px solid #7b8baf">
    <ul>
      <li>Update the JSON output format (allele frequencies) for the <a href="//rest.ensembl.org/#VEP" rel="external">Ensembl REST - Ensembl VEP</a> endpoints. <a href="//github.com/Ensembl/ensembl-rest/wiki/Change-log#70---2018-06" rel="external">See more information</a>.</li>
      <li>The new Ensembl release brings more frequency data from <a href="http://gnomad.broadinstitute.org/" rel="external">gnomAD</a>.</li>
      <li>Add the possibility to print the content of the FILTER column (from the VCF custom annotation files) in the output.</li>
      <li>Include the <a href="//github.com/Ensembl/ensembl-xs" rel="external">Ensembl/ensembl-xs</a> repository in Docker image to speed up the Ensembl VEP container.</li>
      <li>Add a new consequence 'extended_intronic_splice_region_variant' in the <a href="//github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/SpliceRegion.pm" rel="external">SpliceRegion</a> Ensembl VEP plugin.</li>
    </ul>
  </div>


  <br />
  <h4>New in version 92 <i>(April 2018)</i></h4>

  <div class="left-margin" style="border-left:2px solid #7b8baf">
    <ul>
      <li>New Ensembl VEP plugin <a href="//www.ncbi.nlm.nih.gov/pubmed/27666373" rel="external">REVEL</a> (see <a href="//github.com/Ensembl/VEP_plugins/blob/release/92/REVEL.pm" rel="external">REVEL plugin</a>).</li>
      <li>Get ambiguity code with <a href="vep_options.html#opt_ambiguity">--ambiguity</a>.</li>
      <li><a href="vep_cache.html#gff">GFF/GTF files</a> with exons assigned to multiple transcripts are now supported.</li>
      <li>Improved 1000 Genomes Project frequencies.</li>
    </ul>
  </div>


  <br />
  <h4>New in version 91 <i>(December 2017)</i></h4>

  <div class="left-margin" style="border-left:2px solid #7b8baf">
    <ul>
      <li>New input format "<a href="../vep_formats.html#region">region</a>" allows REST-style input to Ensembl VEP.</li>
      <li>Replace your input variant reference allele with the correct one from the genome with <a href="vep_options.html#opt_lookup_ref">--lookup_ref</a>.</li>
      <li>Add version numbers to Ensembl transcripts with <a href="vep_options.html#opt_transcript_version">--transcript_version</a>.</li>
    </ul>
  </div>


  <br />
  <h4>New in version 90 <i>(August 2017)</i></h4>

  <div class="left-margin" style="border-left:2px solid #7b8baf">
    <ul>
      <li><a href="http://gnomad.broadinstitute.org/" rel="external">gnomAD</a> exomes allele frequencies now available with <a href="vep_options.html#opt_af_gnomade">--af_gnomad</a>, replacing ExAC. gnomAD genomes and ExAC are <a href="vep_example.html#gnomad">available via custom annotation</a>.</li>
      <li>Ensembl VEP is now available as a <a href="#docker">Docker image</a>.</li>
      <li>RefSeq transcripts in Ensembl VEP cache files are now <a href="vep_other.html#refseq">"corrected" from the reference genome sequence</a>.</li>
      <li>Ensembl VEP's algorithm for matching colocated known variants has been overhauled - <a href="vep_other.html#colocated">details</a>.</li>
      <li>Change Ensembl VEP's default (5kb) up/downstream distance with <a href="vep_options.html#opt_distance">--distance</a>. This supercedes the functionality of the UpDownDistance Ensembl VEP plugin.</li>
      <li>Feed input directly to Ensembl VEP with <a href="vep_options.html#opt_input_data">--input_data</a>.</li>
      <li>Suppress header output with <a href="vep_options.html#opt_no_headers">--no_headers</a>.</li>
      <li>Detailed <a href="#bigfile">installation instructions for Bio::DB::BigFile</a> to access bigWig custom annotation files.</li>
    </ul>
  </div>  


  <br />
  <h4> New in version 89 <i>(May 2017)</i> </h4>

  <div class="left-margin" style="border-left:2px solid #7b8baf">
    <ul>
      <li>exclude known variants with unknown (null) alleles with <a href="vep_options.html#opt_exclude_null_alleles">--exclude_null_alleles</a>.</li>
      <li>write compressed output with <a href="vep_options.html#opt_compress_output">--compress_output</a>.</li>
      <li>improved matching of alleles in <a href="vep_custom.html">custom VCF files</a>.</li>
      <li>API perldoc documentation added.</li>
    </ul>
  </div>

 
  <br />
  <h4> New in version 88 <i>(March 2017)</i> </h4>

  <div class="left-margin" style="border-left:2px solid #7b8baf">
    <ul>
      <li>ensembl-vep is now the officially supported version of Ensembl VEP</li>
      <li>Documentation updated to reflect switch to ensembl-vep. See the <a href="http://e87.ensembl.org/info/docs/tools/vep/script/index.html">Ensembl archive site</a> for documentation of the obsolete ensembl-tools Ensembl VEP.</li>
      <li>The Ensembl VEP script is now named simply <code>vep</code> (formerly <code>variant_effect_predictor.pl</code> or <code>vep.pl</code>)</li>
      <li>Directly use tabix-indexed <a href="vep_cache.html#gff">GFF/GTF files as annotation sources</a></li>
      <li>Allele-specific reporting of frequencies (<a href="vep_options.html#opt_af">--af</a> and more) and <a href="vep_custom.html">custom VCF annotations</a></li>
      <li><a href="vep_options.html#opt_check_existing">--check_existing</a> now compares alleles by default, disable with <a href="vep_options.html">--no_check_alleles</a></li>
      <li>Report the highest allele frequency observed in any population from 1000 genomes, ESP or ExAC using <a href="vep_options.html#opt_max_af">--max_af</a></li>
      <li>Get genomic HGVS nomenclature with <a href="vep_options.html#opt_hgvsg">--hgvsg</a></li>
      <li>Find the gene or transcript with the nearest transcription start site (TSS) to each input variant with <a href="vep_options.html#opt_nearest">--nearest</a></li>
      <li><a href="vep_filter.html">filter_vep</a> supports field/field comparisons e.g. AFR_AF > #EUR_AF</li>
      <li>Exclude predicted (XM and XR) transcripts when using RefSeq or merged cache with <a href="vep_options.html#opt_exclude_predicted">--exclude_predicted</a></li>
      <li>Filter transcripts used for annotation with <a href="vep_options.html#opt_transcript_filter">--transcript_filter</a></li>
      <li>pileup input format no longer supported</li>
    </ul>
  </div>

</div>

<h4 id="old_vep_tool">Older versions (ensembl-tools) - until version 87: <a class="button" href="#old_vep_tool" onclick="show_hide('old_vep_tool');" id="a_old_vep_tool" style="font-weight:normal">Show</a></h4>
 
<div id="div_old_vep_tool" style="display:none">
  <p> Versions of Ensembl VEP up to and including 87 were released as part of the ensembl-tools package. See <a href="#toolsversions">download links</a> above. </p>

  <h4> New in version 87 <i>(December 2016)</i> </h4>

  <ul>
    <li><a rel="external" href="https://github.com/Ensembl/ensembl-vep">Shiny new code</a> available for beta testing!</li>
    <li>Some minor speed optimisations</li>
    <li>Improve checks for valid chromosome names in input</li>
    <li><a rel="external" href="https://github.com/Ensembl/ensembl-vep#haplo">Haplosaurus</a> beta released - generate whole-transcript haplotype sequences from phased genotype data</li>
  </ul>

  <h4> New in version 86 <i>(October 2016)</i> </h4>

  <ul>
    <li>Chromosome synonyms supported when using Ensembl VEP caches; may be loaded manually with <a href="vep_options.html#opt_synonyms">--synonyms</a></li>
  </ul>

  <h4> New in version 85 <i>(July 2016)</i> </h4>

  <ul>
    <li><a href="vep_options.html#opt_pick">--pick</a> now uses translated length instead of genomic transcript length</li>
    <li>Support for epigenomes in regulatory features</li>
  </ul>

  <h4> New in version 84 <i>(March 2016)</i> </h4>

  <ul>
    <li>Add <a href="/info/docs/tools/vep/vep_formats.html#tab">tab-delimited</a> output option</li>
    <li>Add <a href="/info/docs/tools/vep/vep_formats.html#output">transcript flags</a> indicating if the transcript is 5'- or 3'-incomplete</li>
    <li>Improve annotation of long variants where invariant parts of the alternate allele overlap splice regions</li>
  </ul>

  <h4> New in version 83 <i>(December 2015)</i> </h4>

  <ul>
    <li>
      Speed:
      <ul>
        <li>Basic consequence calculations up to 2x faster than version 82</li>
        <li>HGVS calculations up to 10x faster</li>
        <li>FASTA sequence retrieval implements caching</li>
      </ul>
    </li>
    <li>Add <a href="http://exac.broadinstitute.org/" rel="external">ExAC project</a> frequencies with <a href="vep_options.html#opt_af_exac">--af_exac</a></li>
    <li><a href="/Help/Glossary?id=521">APPRIS</a> isoform annotations now available with <a href="vep_options.html#opt_appris">--appris</a> and used by <a href="vep_options.html#opt_pick">--pick</a> and others to prioritise VEP annotations</li>
  </ul>

  <h4> New in version 82 <i>(September 2015)</i> </h4>

  <ul>
    <li><a href="#installer">Faster FASTA file access</a> using Bio::DB::HTS/htslib and bgzipped FASTA files</li>
    <li><a href="vep_options.html#opt_gene_phenotype">Flag genes</a> with phenotype associations</li>
    <li>Some plugins now available for use via the <a href="/Tools/VEP">web</a> and <a rel="external" href="http://rest.ensembl.org/#VEP">REST</a> interfaces</li>
  </ul>

  <h4> New in version 81 <i>(July 2015)</i> </h4>

  <ul>
    <li>Plugin registry means plugins can be installed from the <a href="#installer">Ensembl VEP installer</a></li>
    <li>GFF format now supported by Ensembl VEP's <a href="vep_cache.html#gtf">cache converter</a></li>
    <li>Fixes and improvements for sequence retrieval from FASTA files</li>
  </ul>

  <h4> New in version 80 <i>(May 2015)</i> </h4>

  <ul>
    <li><a href="../vep_formats.html#output">Flag added</a> indicating if an overlapping known variant is associated with a phenotype, disease or trait</li>
    <li>HGVS notations are now 3'-shifted by default (use <a href="vep_options.html#opt_shift_hgvs">--shift_hgvs</a> to force enable/disable)</li>
    <li>Source version information added to caches; see output file headers or use <a href="vep_options.html#opt_show_cache_info">--show_cache_info</a></li>
    <li>Get the variant class using <a href="vep_options.html#opt_variant_class">--variant_class</a></li>
    <li>CCDS status added to categories used by <a href="vep_options.html#opt_pick">--pick</a> flag (and <a href="vep_other.html#pick">others</a>)</li>
  </ul>

  <h4> New in version 79 <i>(March 2015)</i> </h4>

  <ul>
    <li>Focus on performance and stability: ~100% faster than version 78 and a new test suite</li>
    <li>New guide to <a href="vep_other.html#faster">getting Ensembl VEP running faster</a></li>
    <li>1000 Genomes Phase 3 data available in GRCh37 cache download (GRCh38 coming soon, see <a href="vep_example.html#1kg_p3">docs</a> to access now)</li>
    <li><a href="../vep_formats.html#vcfout">VCF output</a> has changed slightly to match output from other tools</li>
    <li>Impact modifier added for each consequence type</li>
  </ul>

  <h4> New in version 78 <i>(December 2014)</i> </h4>

  <ul>
    <li>Customise <a href="vep_options.html#opt_pick">--pick</a> using
    <a href="vep_options.html#opt_pick_order">--pick_order</a></li>
    <li>Get <a href="/Help/Glossary?id=492">transcript support level</a>
    using <a href="vep_options.html#opt_tsl">--tsl</a></li>
  </ul>

  <h4> New in version 77 <i>(October 2014)</i> </h4>

  <ul>
    <li>Get the <a href="http://www.sequenceontology.org/" rel="external">SO</a>
    feature type of regulatory features using <a
    href="vep_options.html#opt_regulatory">--regulatory</a> and <a
    href="vep_options.html#opt_biotype">--biotype</a></li>
  </ul>

  <h4> New in version 76 <i>(August 2014)</i> </h4>

  <ul>
    <li>Ensembl VEP now supports caches from multiple assemblies (<a
    href="vep_options.html#opt_assembly">--assembly</a>) on the same software
    version - e.g. <a href="vep_other.html#assembly">human builds GRCh37 and
    GRCh38</a></li>
    <li>Protein identifiers from UniProt (SWISSPROT, TrEMBL and UniParc) now
    available using <a href="vep_options.html#opt_uniprot">--uniprot</a></li>
    <li>Ensembl VEP can generate <a href="../vep_formats.html#json">JSON output</a>
    using <a href="vep_options.html#opt_json">--json</a></li>
    <li>Two new analysis set options - <a
    href="vep_options.html#opt_gencode_basic">--gencode_basic</a> and the merged
    Ensembl/RefSeq cache (<a
    href="vep_options.html#opt_merged">--merged</a>)</li>
    <li>Non-RefSeq transcripts now excluded by default when using the RefSeq or
    merged cache; use <a href="vep_options.html#opt_all_refseq">--all_refseq</a>
    to include them</li>
    <li>Let Ensembl VEP pick one consequence per variant allele using <a
    href="vep_options.html#opt_pick_allele">--pick_allele</a></li>
    <li>Allele now included alongside frequency for 1000 Genomes (<a
    href="vep_options.html#opt_af_1kg">--af_1kg</a>) and ESP (<a
    href="vep_options.html#opt_af_esp">--af_esp</a>) data</li>
    <li>Not strictly script-related, but the <a
    href="http://rest.ensembl.org/#Variation" rel="external">Ensembl VEP REST API</a>
    has come out of beta!</li>
  </ul>

  <h4> New in version 75 <i>(February 2014)</i> </h4>

  <ul>
    <li>let Ensembl VEP pick one consequence per variant for you using <a
    href="vep_options.html#opt_pick">--pick</a>; includes all
    transcript-specific data</li>
    <li><a href="vep_options.html#opt_symbol">gene symbol</a> available in
    RefSeq cache and when using <a
    href="vep_options.html#opt_refseq">--refseq</a></li>
    <li>Installation and use of RefSeq cache improved - remember to use <a
    href="vep_options.html#opt_refseq">--refseq</a> with your RefSeq cache!</li>
    <li>Added <a href="vep_options.html#opt_cache_version">--cache_version</a>
    option, primarily to aid Ensembl Genomes users.</li>
  </ul>

  <h4> New in version 74 <i>(December 2013)</i> </h4>

  <ul>
    <li>retrieve the <a href="http://genetics.bwh.harvard.edu/pph2/dokuwiki/overview#prediction" rel="external">humDiv PolyPhen prediction</a> instead of humVar using <a href="vep_options.html#opt_humdiv">--humdiv</a></li>
    <li>source for gene symbol available with <a href="vep_options.html#opt_symbol">--symbol</a></li>
  </ul>

  <h4> New in version 73 <i>(August 2013)</i> </h4>

  <ul>
    <li>NHLBI-ESP frequencies available in cache (<a href="vep_options.html#opt_af_esp">--af_esp</a>)</li>
    <li>Pubmed IDs for cited existing variants available in cache (<a href="vep_options.html#opt_pubmed">--pubmed</a>)</li>
    <li><a href="vep_cache.html#convert">Convert your cache to use tabix</a> - much faster when retrieving co-located existing variants!</li>
        <li>The <a href="vep_download.html#installer">installer</a> can now update the Ensembl VEP to the latest version and install <a href="vep_cache.html#fasta">FASTA files</a></li>
    <li><b>--hgnc</b> replaced by <a href="vep_options.html#opt_symbol">--symbol</a> for non-human compatibility</li>
        <li>HGVS strings are now part <a rel="external" href="http://en.wikipedia.org/wiki/Percent-encoding" rel="exeternal">URI-escaped</a> to avoid "=" sign clashes</li>
    <li>use <a href="vep_options.html#opt_allele_number">--allele_number</a> to identify input alleles by their order in the VCF ALT field</li>
    <li>use <a href="vep_options.html#opt_total_length">--total_length</a> to give the total length of cDNA, CDS and protein sequences</li>
        <li>add data from VCF INFO fields when using <a href="vep_custom.html">custom annotations</a></li>
  </ul>

  <h4> New in version 72 <i>(June 2013)</i> </h4>

  <ul>
    <li>Speed and stability improvements when using forking</li>
    <li>Filter Ensembl VEP results using <a href="vep_filter.html">filter_vep.pl</a></li>
  </ul>

  <h4> New in version 71 <i>(April 2013)</i> </h4>

  <ul>
    <li>SIFT predictions now available for Chicken, Cow, Dog, Human, Mouse,
    Pig, Rat and Zebrafish</li>
    <li>View <a href="../vep_formats.html#stats">summary statistics</a> for Ensembl VEP runs in [output]_summary.html</li>
    <li>Generate HTML output using <a href="vep_options.html#opt_html">--html</a></li>
    <li>Support for simple tab-delimited format for input of structural
    variant data</li>
    <li>Cache now contains clinical significance statuses from dbSNP for
    human variants</li>
    <li><p><b>NOTE:</b> Ensembl VEP version numbers have now (from release 71)
    changed to match Ensembl release numbers.</p></li>
  </ul>

  <h4> New in version 2.8 <i>(December 2012)</i> </h4>

  <ul>
    <li>Easily filter out common human variants with <a href="vep_options.html#opt_filter_common">--filter_common</a></li>
    <li>1000 Genomes continental population frequencies now stored in cache
    files</li>
  </ul>

  <h4> New in version 2.7 <i>(October 2012)</i> </h4>

  <ul>
    <li>build Ensembl VEP cache files offline from GTF and FASTA files</li>
    <li>support for using FASTA files for sequence lookup in HGVS notations
    in offline/cache modes</li>
  </ul>

  <h4> New in version 2.6 <i>(July 2012)</i> </h4>

  <ul>
    <li>support for <a href="../vep_formats.html#sv">structural variant</a> consequences</li>
    <li>Sequence Ontology (SO) consequence terms now default</li>
    <li>script runtime 3-4x faster when using <a href="vep_options.html#forking">forking</a></li>
    <li>1000 Genomes global MAF available in cache files</li>
    <li>improved memory usage</li>
  </ul>

  <h4> New in version 2.5 <i>(May 2012)</i> </h4>

  <ul>
    <li>SIFT and PolyPhen predictions now available for RefSeq transcripts</li>
    <li>retrieve cell type-specific regulatory consequences</li>
    <li>consequences can be retrieved based on a single individual's genotype in
    a VCF input file</li>
    <li>find overlapping structural variants</li>
    <li>Condel support removed from main script and moved to a plugin</li>
  </ul>

  <h4> New in version 2.4 <i>(February 2012)</i> </h4>

  <ul>
    <li>offline mode and new installer script make it easy to use the Ensembl VEP
    without the usual dependencies</li>
    <li>output columns configurable using the <a href="vep_options.html#opt_fields">--fields</a> flag</li>
    <li>VCF output support expanded, can now carry all fields</li>
    <li>output affected exon and intron numbers with <a href="vep_options.html#opt_numbers">--numbers</a></li>
    <li>output overlapping protein domains using <a href="vep_options.html#opt_domains">--domains</a></li>
    <li>enhanced support for LRGs</li>
    <li>plugins now work on variants called as intergenic</li>
  </ul>

  <h4> New in version 2.3 <i>(December 2011)</i> </h4>

  <ul>
    <li>add custom annotations from tabix-indexed files (BED, GFF, GTF, VCF,
    bigWig)</li>
    <li>add new functionality to the Ensembl VEP with user-written plugins</li>
    <li>filter input on consequence type</li>
  </ul>

  <h4> New in version 2.2 <i>(September 2011)</i> </h4>

  <ul>
    <li>SIFT, PolyPhen and Condel predictions and regulatory features now
    accessible from the <a href="vep_cache.html">cache</a></li>
    <li>support for calling consequences against <a href="vep_other.html#refseq">RefSeq</a>
    transcripts</li>
    <li>variant identifiers (e.g. dbSNP rsIDs) and <a href="vep_other.html#hgvs">HGVS
    notations</a> supported as input format</li>
    <li>variants can now be <a href="vep_options.html#filt">filtered</a> by frequency in HapMap
    and 1000 genomes populations</li>
    <li>script can be used to convert files between formats
    (Ensembl/VCF/Pileup/HGVS to Ensembl/VCF/Pileup)</li>
    <li>large amount of code moved to API modules to ensure consistency between
    web and script Ensembl VEP</li>
    <li>memory usage optimisations</li>
    <li>Ensembl VEP script moved to <a href="https://github.com/Ensembl/ensembl-tools/tree/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/scripts/variant_effect_predictor" rel="external">ensembl-tools repo</a></li>
    <li>Added <a href="vep_options.html#opt_canonical">--canonical</a>, <a href="vep_options.html#opt_per_gene">--per_gene</a> and <a href="vep_options.html#opt_no_intergenic">--no_intergenic</a> options</li>
  </ul>

  <h4> New in version 2.1 <i>(June 2011)</i> </h4>

  <ul>
    <li>ability to use local file <a href="vep_cache.html#cache">cache</a> in place of or
    alongside connecting to an Ensembl database</li>
    <li>significant improvements to speed of script</li>
    <li>whole-genome mode now default (no disadvantage for smaller
    datasets)</li>
    <li>improved status output with progress bars</li>
    <li>regulatory region consequences now reinstated and improved</li>
    <li>modification to output file - Transcript column is now Feature, and is
    followed by a Feature_type column</li>
  </ul>

  <h4> New in version 2.0 <i>(April 2011)</i> </h4>

  <ul>
    <li>support for SIFT, PolyPhen and Condel missense predictions in human</li>
    <li>per-allele and compound consequence types</li>
    <li>support for Sequence Ontology (SO) and NCBI consequence terms</li>
    <li>
      modified output format
      <ul>
        <li>support for new output fields in Extra column</li>
        <li>header section contains information on database and software versions</li>
        <li>codon change shown in output</li>
        <li>CDS position shown in output</li>
        <li>option to output Ensembl protein identifiers</li>
        <li>option to output HGVS nomenclature for variants</li>
      </ul>
    </li>
    <li>support for gzipped input files</li>
    <li>enhanced configuration options, including the ability to read configuration from a file</li>
    <li>verbose output now much more useful</li>
    <li>whole-genome mode now more stable</li>
    <li>finding existing co-located variations now ~5x faster</li>
  </ul>

</div>

    <hr/>
  <h2 id="requirements">Requirements</h2>

  Ensembl VEP requires:
  <ul>
    <li><b>gcc</b>, <b>g++</b> and <b>make</b></li>
    <li><b>Perl</b> version <b>5.22</b> or above recommended (tested on 5.22, 5.26, 5.32, 5.34, 5.38)</li>
    <li><b>Perl</b> packages:
      <ul>
        <li><a href="https://metacpan.org/pod/Archive::Zip" rel="external">Archive::Zip</a></li>
        <li><a href="https://metacpan.org/release/DVEEDEN/DBD-mysql-4.050/view/lib/DBD/mysql.pm" rel="external">DBD::mysql</a> (version &lt;=4.050)</li>
        <li><a href="https://metacpan.org/pod/DBI" rel="external">DBI</a></li>
      </ul>
      See <a href="http://www.cpan.org/modules/INSTALL.html" rel="external">this guide</a> for more information on how to install perl modules.<br/>
      <a href="#additional">Additional libraries</a> can be installed for extra features and enhancements but they are not required to run Ensembl VEP in most of the use cases.
    </li>
  </ul>

  <p>Ensembl VEP's INSTALL.pl script will install required components of Ensembl API for you, but Ensembl VEP may also be used with any pre-existing API installations you have, <b>provided their versions match the version of VEP you are using</b>. </p>

  <p>Ensembl VEP is available in the following platforms:<p>
  <ul>
    <li>Linux (e.g., Ubuntu, Debian, Mint)</li>
    <li><a href="#macos">macOS</a></li>
    <li><a href="#windows"><img style="vertical-align:bottom" src="/i/16/windows.png"/> Windows</a> (requires a more involved installation process)</li>
  </ul>

  <p>Ensembl VEP is also available as <a href="#docker">Docker</a> and <a href="#singularity">Singularity</a> images, allowing to skip the complex installation steps.</p>

  <hr/>


  <h2 id="installer">Installation</h2>

  <p> Ensembl VEP's INSTALL.pl makes it easy to set up your environment for
    using the Ensembl VEP. It will download and configure a minimal set of the Ensembl
    API for use by the Ensembl VEP, and can also download <a
    href="vep_cache.html#cache">cache files</a>, <a
    href="vep_cache.html#fasta">FASTA files</a> and <a href="vep_plugins.html">plugins</a>. </p>

  <p>Run the following, and follow any prompts as they appear:</p>

  <pre class="code sh_sh">perl INSTALL.pl</pre>

  <p><a href="#additional">Additional non-essential components</a> and enhancements must be installed manually.</p>
  
  <br />
  <h3> Software components installed </h3>

  <ul>
    <li><a href="https://github.com/bioperl/bioperl-live" rel="external">BioPerl</a></li>
    <li><a href="https://github.com/Ensembl/ensembl" rel="external">ensembl</a></li>
    <li><a href="https://github.com/Ensembl/ensembl-io" rel="external">ensembl-io</a></li>
    <li><a href="https://github.com/Ensembl/ensembl-variation" rel="external">ensembl-variation</a></li>
    <li><a href="https://github.com/Ensembl/ensembl-funcgen" rel="external">ensembl-funcgen</a></li>
    <li><a href="https://github.com/Ensembl/Bio-DB-HTS" rel="external">Bio::DB::HTS</a></li>
  </ul>

  <p> If you already have the latest version of the API installed you do not
  need to run the installer, although it can be used to simply
  update your API version (with post-release patches applied), and
  retrieve cache and FASTA files. The installer downloads the API within the
  Ensembl VEP directory and will not affect any other Ensembl API installations. </p>

  <p> The script will also attempt to install a Perl::XS module,
  <a rel="external" href="https://github.com/Ensembl/Bio-HTS">Bio::DB::HTS</a>, for
  rapid access to bgzipped FASTA files. If this fails, you may add the
  --NO_HTSLIB flag when running the installer; Ensembl VEP will fall back to using
  Bio::DB::Fasta for this functionality (<a
  href="vep_cache.html#fasta">more details</a>).</p>

  <br />
  <h3 id="run_install">Running the installer</h2>

  <p> The installer is run on the command line as follows: </p>

  <pre class="code sh_sh"> perl INSTALL.pl [options] </pre>

  <p> Follow on-screen prompts and note warnings of any files which will be deleted/overwritten</p>

  <p> You should not need to add any options, but configuration of the installer
  is possible with the flags below. Options can also be set by exporting
  <b>environment variables</b> prefixed with <kbd>VEP_</kbd> before running the
  installer (for instance, <kbd>export VEP_NO_HTSLIB=1</kbd> and
  <kbd>export VEP_DIR_PLUGINS="/plugins"</kbd>).</p>

  <table class="ss">
    <tr>
      <th>Flag</th>
      <th>Alternate</th>
      <th>Description</th>
    </tr>
    <tr id="opt_assembly">
      <td><pre>--ASSEMBLY</pre></td>
      <td><pre>-y</pre></td>
      <td>
        Assembly version to use when using <code>--AUTO</code>. Most species have only one
        assembly available on each software release; currently this is only
        required for <a href="vep_other.html#assembly">human on release 76</a>
        onwards.
      </td>
    </tr>
    <tr id="opt_auto">
      <td><pre>--AUTO</pre></td>
      <td><pre>-a</pre></td>
      <td>
        Run installer without prompts. Use the following options to specify
        parts to install:
        <ul>
          <li><b>a</b> (API + Bio::DB::HTS/htslib)</li>
          <li><b>l</b> (Bio::DB::HTS/htslib only)</li>
          <li><b>c</b> (cache)</li>
          <li><b>f</b> (FASTA)</li>
          <li><b>p</b> (plugins) &mdash; Require the use of the <a href="#opt_plugins">--PLUGINS</a> flag to list the plugin(s) to install.</li>
        </ul>
        <p>e.g. for API and cache:</p>
        <pre class="code sh_sh">perl INSTALL.pl --AUTO ac</pre>
      </td>
    </tr>
    <tr id="opt_cache_version">
      <td><pre>--CACHE_VERSION [version]</pre></td>
      <td>&nbsp;</td>
      <td>
        By default the installer will download the latest version of Ensembl VEP caches
        and FASTA files (currently [[SPECIESDEFS::ENSEMBL_VERSION]]). You can
        force the script to install a different version, but there is no guarantee that
        a version of the API will be compatible with a different version of the cache.
      </td>
    </tr>
    <tr id="opt_cachedir">
      <td><pre>--CACHEDIR [dir]</pre></td>
      <td><pre>-c</pre></td>
      <td>
        <p>By default the script will install the cache files in the ".vep"
        subdirectory in your home area. This option
        configures where cache files are installed.</p>
        <p>The <a href="vep_options.html#opt_dir_cache">--dir_cache</a> flag must be passed when
        running Ensembl VEP if a non-default cache directory is given:</p>
        <pre class="code sh_sh">./vep --dir_cache [dir]</pre>
      </td>
    </tr>
    <tr id="opt_destdir">
      <td><pre>--DESTDIR [dir]</pre></td>
      <td><pre>-d</pre></td>
      <td>
        <p>By default the script will install the API modules in a
        subdirectory of the current directory named "Bio". Using this option
        you can configure where the Bio directory is created. If something
        other than the default is used, this directory must either be added
        to your PERL5LIB environment variable when running Ensembl VEP, or
        included using perl's -I flag:</p>
        <pre class="code sh_sh">perl -I [dir] vep</pre>
      </td>
    </tr>
    <tr id="opt_no_htslib">
      <td><pre>--NO_HTSLIB</pre></td>
      <td><pre>-l</pre></td>
      <td>Don't attempt to install Bio::DB::HTS/htslib</td>
    </tr>
    <tr id="opt_no_test">
      <td><pre>--NO_TEST</pre></td>
      <td>&nbsp;</td>
      <td>Don't run API tests - useful if you know a harmless failure will prevent continuation of the installer</td>
    </tr>
    <tr id="opt_no_update">
      <td><pre>--NO_UPDATE</pre></td>
      <td><pre>-n</pre></td>
      <td>
        By default the script will check for new versions or updates of Ensembl VEP.
        Using this option will skip this check.
      </td>
    </tr>
    <tr id="opt_plugins">
      <td><pre>--PLUGINS</pre></td>
      <td><pre>-g</pre></td>
      <td>
        <p>Comma-separated list of plugins to install when using <code>--AUTO</code>. To install all available plugins, use <code>--PLUGINS all</code>.</p>
        <pre class="code sh_sh"># List the available plugins:
perl INSTALL.pl -a p --PLUGINS list
# Download/install all the available plugins:
perl INSTALL.pl -a p --PLUGINS all
# Download/install a defined list of plugins, e.g.:
perl INSTALL.pl -a p --PLUGINS dbNSFP,CADD,G2P</pre>
      </td>
    </tr>
    <tr id="opt_pluginsdir">
      <td><pre>--PLUGINSDIR [dir]</pre></td>
      <td><pre>-r</pre></td>
      <td>
        <p>By default the script will install the plugins files in the "Plugins" subdirectory of the <code>--CACHEDIR</code> directory.
        This option configures where the plugins files are installed.</p>
        <p>The <a href="vep_options.html#opt_dir_plugins">--dir_plugins</a> flag must be passed when
        running Ensembl VEP if a non-default plugins directory is given:</p>
        <pre class="code sh_sh">./vep --dir_plugins [dir]</pre>
      </td>
    </tr>
     <tr id="opt_prefer_bin">
      <td><pre>--PREFER_BIN</pre></td>
      <td><pre>-p</pre></td>
      <td>Use this if the installer fails with "out of memory" errors.</td>
    </tr>
    <tr id="opt_species">
      <td><pre>--SPECIES</pre></td>
      <td><pre>-s</pre></td>
      <td>
        <p>Comma-separated list of species to install when using <code>--AUTO</code>. To
        install the RefSeq cache, add "_refseq" to the species name, e.g.
        "homo_sapiens_refseq", or "_merged" to install the merged Ensembl/RefSeq
        cache. Remember to use <a
        href="vep_options.html#opt_refseq">--refseq</a> or <a
        href="vep_options.html#opt_merged">--merged</a> when running the VEP
        with the relevant cache!</p>
        <p>Use <code>all</code> to install data for all available species.</p>
      </td>
    </tr>
    <tr id="opt_use_https_proto">
      <td><pre>--USE_HTTPS_PROTO</pre></td>
      <td><pre></pre></td>
      <td>Download cache and FASTA file using HTTPs protocol instead of FTP. Useful for networks where FTP port is blocked by firewall.</td>
    </tr>
    <tr id="opt_github_token">
      <td><pre>--GITHUBTOKEN</pre></td>
      <td><pre></pre></td>
      <td>Set token to use for authentication when querying GitHub API. Authenticated user have increased rate-limit. NOTE: use token with read-only access.</td>
    </tr>
    <tr id="opt_quiet">
      <td><pre>--QUIET</pre></td>
      <td><pre>-q</pre></td>
      <td>Don't write any status output when using <code>--AUTO</code>.</td>
    </tr>
  </table>




  <hr/>
  <h2 id="additional">Additional components</h2>

  <p> INSTALL.pl will set up the minimum requirements Ensembl VEP. Some features and enhancements, however, require the installation of additional components. Most are perl modules that are easily installed using cpanm; see <a href="http://www.cpan.org/modules/INSTALL.html" rel="external">this guide</a> for more information on how to install perl modules. </p>

  <p> Typically, you will use cpanm to install modules locally in your home directories; this shows how to set up a path for perl modules and install one there: </p>

  <pre class="code sh_sh">mkdir -p $HOME/cpanm
export PERL5LIB=$PERL5LIB:$HOME/cpanm/lib/perl5
cpanm -l $HOME/cpanm Set::IntervalTree</pre>

  <p> To make the change to <code>PERL5LIB</code> permanent, it is recommended to add the <code>export</code> line to your <code>$HOME/.bashrc</code> or <code>$HOME/.profile</code>. </p>

  <ul>
    <li>
      Additional features
      <ul>
        <li><a href="http://search.cpan.org/dist/JSON/" rel="external">JSON</a> - required to produce <a href="vep_options.html#opt_json">JSON format output</a></a></li>
        <li><a href="http://search.cpan.org/~benbooth/Set-IntervalTree/lib/Set/IntervalTree.pm" rel="external">Set::IntervalTree</a> - used to find overlaps between entities in coordinate space. Required to use <a href="vep_options.html#opt_nearest">--nearest</a></li>
        <li><a href="http://search.cpan.org/~lds/Bio-BigFile-1.07/lib/Bio/DB/BigFile.pm" rel="external">Bio::DB::BigFile</a> - required to use bigWig format <a href="vep_custom.html">custom annotation files</a>. See <a href="#bigfile">Bio::DB::BigFile instructions</a>.</li>
      </ul>
    </li>
    <li>
      Speed enhancements - these modules can improve Ensembl VEP runtime
      <ul>
        <li><a href="http://search.cpan.org/~nwclark/PerlIO-gzip-0.19/gzip.pm">PerlIO::gzip</a> - marginal gains in compressed file parsing as used by the Ensembl VEP cache</li>
        <li><a href="https://github.com/Ensembl/ensembl-xs" rel="external">ensembl-xs</a> - provides pre-compiled replacements for frequently used routines in Ensembl VEP. Requires manual installation, see <a href="https://github.com/Ensembl/ensembl-xs" rel="external">README</a> for details</li>
      </ul>
    </li>
  </ul>

  <br />
  <h3 id="bigfile">Bio::DB::BigFile</h3>

  <p> In order for Ensembl VEP to be able to access bigWig format custom annotation files, the Bio::DB::BigFile perl module is required. Installation involves downloading and compiling the <a href="https://github.com/ucscGenomeBrowser/kent" rel="external">kent source tree</a>. The current version of the kent source tree does not work correctly with Bio::DB::BigFile, so it is necessary to install an archive version known to work (v335). </p>

  <ol>
    <li>
      <p>Download and unpack the kent source tree</p>
      <pre class="code sh_sh">wget https://github.com/ucscGenomeBrowser/kent/archive/v335_base.tar.gz
tar xzf v335_base.tar.gz</pre>
    </li>
    <li>
      <p>Set up some environment variables; these are required only temporarily for this installation process</p>
      <pre class="code sh_sh">export KENT_SRC=$PWD/kent-335_base/src
export MACHTYPE=$(uname -m)
export CFLAGS="-fPIC"
export MYSQLINC=`mysql_config --include | sed -e 's/^-I//g'`
export MYSQLLIBS=`mysql_config --libs`</pre>
    </li>
    <li>
      <p>Modify kent build parameters</p>
      <pre class="code sh_sh">cd $KENT_SRC/lib
echo 'CFLAGS="-fPIC"' > ../inc/localEnvironment.mk</pre>
    </li>
    <li>
      <p>Build kent source</p>
      <pre class="code sh_sh">make clean && make
cd ../jkOwnLib
make clean && make</pre>
      <p>If either of these steps fail, you may have some missing dependencies. Known common missing dependencies are libpng and libssl; these may be installed, for example, with <code>apt-get</code> on Ubuntu. If you do not have sudo access you may have to ask your sysadmin to install any missing dependencies.</p>
      <pre class="code sh_sh">sudo apt-get install libpng-dev libssl-dev</pre>
      <p>On macOS you may use <a href="https://brew.sh/" rel="external"><code>brew</code></a>; the openssl libraries also need to be symbolically linked to a different path:</p>
      <pre class="code sh_sh">brew install libpng openssl
cd /usr/local/include
ln -s ../opt/openssl/include/openssl .
cd -</pre>
    </li>
    <li>
      <p>On some systems (e.g. macOS), a compiled file is placed in a path that Bio::DB::BigFile cannot find. You can correct this with:</p>
      <pre class="code sh_sh">ln -s $KENT_SRC/lib/x86_64/* $KENT_SRC/lib/</pre>
    </li>
    <li>
      <p>We'll now use cpanm to install the perl module for Bio::DB::BigFile itself. See <a href="#additional">above</a> for guidance on this. In this example we're going to install the module to a path within your home directory. In order to do this we must modify the paths that perl looks in to find modules by adding to the <code>PERL5LIB</code> environment module. To make this change permanent you must add the <code>export</code> line to your <code>$HOME/.bashrc</code> or <code>$HOME/.profile</code>.</p>
      <pre class="code sh_sh">mkdir -p $HOME/cpanm
export PERL5LIB=$PERL5LIB:$HOME/cpanm/lib/perl5
cpanm -l $HOME/cpanm Bio::DB::BigFile</pre>
      <p>If you are prompted for the path to the kent source tree, that means something didn't go right in the compilation above. Double check that <code>$KENT_SRC/lib/jkweb.a</code> exists and is not found instead at e.g. <code>$KENT_SRC/lib/x86_64/jkweb.a</code>. You may copy or link the file (and the other files in that directory) to the former path.</p>
      <pre class="code sh_sh">ln -s $KENT_SRC/lib/x86_64/* $KENT_SRC/lib/</pre>
    </li>
    <li>
      <p>You should now be able to successfully run the appropriate test in the Ensembl VEP package:</p>
      <pre class="code sh_sh">perl -Imodules t/AnnotationSource_File_BigWig.t</pre>
    </li>
  </ol>



  <hr/>
  <h2 id="macos">Using Ensembl VEP in macOS</h2>

  <p>Installing Ensembl VEP on macOS is slightly trickier than other Linux-based systems, and will require additional dependancies.<br />
  These instructions will guide you through the setup of <b>Perlbrew</b>, <b>Homebrew</b>, <b>MySQL</b> and other dependancies that will allow for a clean installation of Ensembl VEP on your macOS system.</p>

  <p>These instructions have been tested on <b>macOS High Sierra (10.13)</b> and <b>macOS Sierra (10.12)</b>.<br />Older versions may require additional tweaks, however we shall endeavouXcoder to keep these instructions up to date for future versions of MacOS.</p>

  <div>
    <div style="float:left" class="warning">
      <h3 id="apple_silicon_macs">Installation issues with M-series Macs</h3>
      <div class="message-pad">
        <p>We advise using the <a href="#docker">Docker</a> or <a href="#singularity">Singularity</a> images for Ensembl VEP if you are having issues installing Ensembl VEP in Apple Silicon (M1 upwards) Macs.</p>
      </div>
    </div>
    <div class="clear"></div>
  </div>

  <br />
  <h3>Prerequisite Setup</h3>

  <p>List of prerequisites: <b>Xcode</b>, <b>GCC</b>, <b>Perlbrew</b>, <b>Cpanm</b>, <b>Homebrew</b>, <b>mysql</b>, <b>DBI</b>, <b>DBD::mysql</b> (version &lt;=4.050)</p>

  <h4>Xcode and GCC</h4>
  
  <p>Ensembl VEP requires Xcode and GCC for installation purposes. Fortunately, recent versions of macOS will look for (and attempt to install if required) both of these when you run the following command:</p>

  <pre class="code sh_sh">gcc -v</pre>

  <h4>Perlbrew</h4>
  
  <p>We recommend using Perlbrew to install a new version of Perl on your mac, to prevent messing with the vendor perl too much. This can be done with the following command:</p>

  <pre class="code sh_sh">curl -L http://install.perlbrew.pl | bash

echo 'source $HOME/perl5/perlbrew/etc/bashrc' >> ~/.bash_profile</pre>

 <p>At this point, PLEASE RESTART YOUR TERMINAL WINDOW to allow for the perlbrew changes to take effect.</p>

 <p>We recommend installing Perl version <b>5.26.2</b> to run Ensembl VEP, and installing cpanm to handle the installation of perl modules.<br />
 These steps can be completed with the commands:</p>

  <pre class="code sh_sh">perlbrew install -j 5 --as 5.26.2 --thread --64all -Duseshrplib perl-5.26.2 --notest
perlbrew switch 5.26.2
perlbrew install-cpanm</pre>

  <h4>Homebrew</h4>
  
  <p>This package management system for macOS would make the installation of the next prerequisite (i.e. xs) easier.</p>

  <pre class="code sh_sh">/usr/bin/ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)"</pre>

  <h4>xz</h4>

  <p>Ensembl VEP requires the installation of xz, a data-compression utility. The easiest way to install the xz package is through homebrew:</p>

  <pre class="code sh_sh">brew install xz</pre>

  <h4>MySQL</h4>
  
  <p>In order to connect to the Ensembl databases, a collection of MySQL related dependancies are required. Fortunately, these can be installed neatly with <b>Homebrew</b> and <b>Cpanm</b>:</p>

  <pre class="code sh_sh">brew install mysql
cpanm DBI
cpanm DBD::mysql@4.050</pre>

  <br />
  <h3>Installing BioPerl</h3>

  <p>On some versions of macOS, the Ensembl VEP installer fails to cleanly install BioPerl, so a manual install will prevent issues:</p>

  <pre class="code sh_sh">curl -O https://cpan.metacpan.org/authors/id/C/CJ/CJFIELDS/BioPerl-1.6.924.tar.gz
tar zxvf BioPerl-1.6.924.tar.gz
echo 'export PERL5LIB=${PERL5LIB}:##PATH_TO##/bioperl-1.6.924' >> ~/.bash_profile</pre>
  
  <p> where <i>##PATH_TO##/bioperl-1.6.924</i> refers to the location of the newly unzipped BioPerl directory.</p>
  
  <br />
  <h3>Final Dependancies</h3>

  <p>Installing the following Perl modules with cpanm will allow for full Ensembl VEP functionality:</p>

  <pre class="code sh_sh">cpanm Test::Differences Test::Exception Test::Perl::Critic Archive::Zip PadWalker Error Devel::Cycle Role::Tiny::With Module::Build LWP List::MoreUtils

export DYLD_LIBRARY_PATH=/usr/local/mysql/lib/:$DYLD_LIBRARY_PATH</pre>


  <br />
  <h3>Installing Ensembl VEP</h3>

  <p>And that should be that! You should now be able to install Ensembl VEP using the installer:</p>

  <pre class="code sh_sh">git clone https://github.com/ensembl/ensembl-vep
cd ensembl-vep
perl INSTALL.pl --NO_TEST</pre>



  <hr/>
  <h2 id="windows">Using Ensembl VEP in Windows <img style="padding-left:2px" src="/i/16/windows.png"/></h2>

  <p> Ensembl VEP was developed as a command-line tool, and as a Perl script its
  natural environment is a Linux system. However, there are several ways you can
  use Ensembl VEP on a Windows machine. </p>

  <p> You may also consider using Ensembl VEP's web or REST interfaces. </p>

  <br />
  <h4 id="virtual">Virtual machines</h4>

  <p> Using a virtual machine you can run a virtual Linux system in a window on
  your machine. There are two ways to do this: </p>

  <ol>
    <li>Use the <a
    href="/info/data/virtual_machine.html">Ensembl virtual machine image</a></li>
    <li>Use <a href="#docker">Docker</a></li>
  </ol>

  <br />
  <h4 id="wperl">Perl</h4>

  <p> If Perl is installed on Windows, Ensembl VEP can be setup. However this may require installation of
      dependent modules. We recommend using <a href="#docker">Docker</a> to run Ensembl VEP on Windows. </p>

  <ol>
    <li>Check Perl is installed</a></li>
    <li>Download and unpack the <a href="https://github.com/Ensembl/ensembl-vep/archive/release/[[SPECIESDEFS::ENSEMBL_VERSION]].zip">zip of the ensembl-vep package</a></li>
    <li>Open a Command Prompt (search for Command Prompt in the Start Menu)</li>
    <li>
      <p>Navigate to the directory where you unpacked the Ensembl VEP package, e.g.</p>
      <pre class="code sh_sh">cd Downloads/ensembl-vep-release-[[SPECIESDEFS::ENSEMBL_VERSION]]</pre>
    </li>
    <li>
      <p>Run INSTALL.pl with --NO_HTSLIB and --NO_TEST; you will see some warnings about the "which" command not being available (these will also appear when running Ensembl VEP and can be ignored).</p>
      <pre class="code sh_sh">perl INSTALL.pl --NO_HTSLIB --NO_TEST</pre>
    </li>
  </ol>


  <hr/>
  <h2 id="docker">Docker</h2>

  <p> <a href="https://www.docker.com/" rel="external">Docker</a> allows running applications in virtualised <i>containers</i>. The Ensembl VEP Docker image is available from DockerHub:
  <a class="button" href="https://hub.docker.com/r/ensemblorg/ensembl-vep/" target="_blank">Ensembl VEP in DockerHub</a></p>

  <p>After installing <a href="https://www.docker.com/" rel="external">Docker</a>, download the Ensembl VEP Docker image:</p>
  <pre class="code sh_sh">docker pull ensemblorg/ensembl-vep</pre>

  <p> To download cache files and other data with Ensembl VEP Docker, we recommend <a href="https://docs.docker.com/engine/tutorials/dockervolumes/" rel="external">mounting a directory</a> from your local (host) machine to folder <code>/data</code> from the Docker image. For instance: </p>

  <pre class="code sh_sh">
mkdir $HOME/vep_data
docker run -t -i -v $HOME/vep_data:/data ensemblorg/ensembl-vep</pre>

  <p> In the example above, data in <code>$HOME/vep_data</code> will be accessible by both the local machine and Ensembl VEP Docker. The Ensembl VEP API, plugins and their dependencies (e.g. Perl APIs, Bio::DB::HTS, htslib, ...) are already installed in the image. </p>

  <div>
    <div style="float:left" class="warning">
      <h3 id="read_write_access">Read/Write access from the container</h3>
      <div class="message-pad">
        <p>In some distributions (e.g. CentOS, Fedora, Red Hat Enterprise Linux) Docker daemon requires <b>root</b> privileges (i.e. needs to prefix the command with <b>sudo</b>), which might cause read/write issues to the mounted volume.</p>
        <p>One solution is to use the option <b>:Z</b> within the Docker <b>-v</b> option (only from docker 1.7.0):
          <pre class="code sh_sh">sudo docker run -t -i -v $HOME/vep_data:/data:Z ensemblorg/ensembl-vep</pre>
        </p>
        <p>An other solution is to change the read/write access of the mounted volume (<code>$HOME/vep_data</code>):
          <pre class="code sh_sh">chmod -R a+rwx $HOME/vep_data</pre>
        </p>
      </div>
    </div>
    <div class="clear"></div>
  </div>

<br />
<h3 id="docker_cache">Cache and FASTA files installation</h3>

  <p> You can run the INSTALL.pl script to install the cache and FASTA files: </p>
  <pre class="code sh_sh">docker run -t -i -v $HOME/vep_data:/data ensemblorg/ensembl-vep INSTALL.pl</pre>

  <ul>
    <li>
      You will be asked to install cache data.
      Type the comma-separated numbers for the species/assembly of interest and press <kbd>enter</kbd>.
      Your data will download and unpack; this may take a while.
    </li>
    <li>
    If you wish to retrieve HGVS annotations, please download the FASTA files for your species.
    To do this, at the next prompt type <kbd>0</kbd> and press <kbd>enter</kbd>.
    </li>
  </ul>

  <p>
    The above process may also be performed in one command; for example, to set up the cache and corresponding FASTA for human GRCh38:
  </p>

  <pre class="code sh_sh">docker run -t -i -v $HOME/vep_data:/data ensemblorg/ensembl-vep INSTALL.pl -a cf -s homo_sapiens -y GRCh38</pre>

  <p>
    The installer downloads Ensembl VEP data to the mounted directory (e.g., <code>$HOME/vep_data</code>).
    The downloaded data will be automatically detected as long as its folder is mounted when running VEP:
  </p>

  <pre class="code sh_sh">docker run -v $HOME/vep_data:/data ensemblorg/ensembl-vep vep -i examples/homo_sapiens_GRCh38.vcf --cache</pre>

<br />
<h3 id="docker_setup">Running Ensembl VEP with data from local folder</h3>

<p>Here is an example on running Ensembl VEP with data from folder <code>$HOME/vep_data</code> in the local machine (provided that the cache has been downloaded to that folder):</p>

<pre class="code sh_sh">
docker run -v $HOME/vep_data:/data ensemblorg/ensembl-vep \
  vep --cache --offline --format vcf --vcf --force_overwrite \
      --input_file input/my_input.vcf \
      --output_file output/my_output.vcf \
      --custom file=custom/my_extra_data.bed,short_name=BED_DATA,format=bed,type=exact,coords=1 \
      --plugin NMD</pre>

<p>Please avoid using absolute paths to data as the paths inside the container differ from your local machine.</p>

<br />
<h3>Update from a previous version</h3>

<ol>
  <li><p>Update your Docker container</p>
    <pre class="code sh_sh">
docker pull ensemblorg/ensembl-vep</pre>
  </li>
  <li><p>Update your cache</p>
    <pre class="code sh_sh"># Install the new cache through the Ensembl VEP INSTALL.pl script (see "Cache installation" section above)
docker run -t -i -v $HOME/vep_data:/data ensemblorg/ensembl-vep INSTALL.pl -a c

# Or install the cache manually
cd $HOME/vep_data
curl -O [[SPECIESDEFS::ENSEMBL_FTP_URL]]/release-[[SPECIESDEFS::ENSEMBL_VERSION]]/variation/vep/homo_sapiens_vep_[[SPECIESDEFS::ENSEMBL_VERSION]]_GRCh38.tar.gz
tar xzf homo_sapiens_vep_[[SPECIESDEFS::ENSEMBL_VERSION]]_GRCh38.tar.gz</pre>
  </li>
</ol>

  </div>

  <hr/>
  <h2 id="singularity">Singularity</h2>

  <p>
    Due to root requirements for the Docker daemon, using the <a href="https://hub.docker.com/r/ensemblorg/ensembl-vep/" rel="external">Docker container for Ensembl VEP</a> is not always possible to HPC users.
    Singularity, an alternative containerisation tool, does not assume that you have a system where you are the root user. This has led to increased popularity in HPC contexts due to increased access rights flexibility.
  </p>

  <p>After installing <a href="https://sylabs.io/singularity/" rel="external">Singularity</a>, Ensembl VEP may be used with Singularity based on the VEP Docker image from DockerHub:</p>
  <pre class="code sh_sh">singularity pull --name vep.sif docker://ensemblorg/ensembl-vep</pre>

  <p> The following is a brief example showing how to use a directory on your local (host) machine to store cache data for VEP. </p>

  <pre class="code sh_sh">
mkdir $HOME/vep_data
singularity exec vep.sif vep --dir $HOME/vep_data --help</pre>

  <p> The Ensembl VEP API, plugins and their dependencies (e.g. Perl APIs, Bio::DB::HTS, htslib, ...) are already installed in the image. </p>

<br />
<h3 id="singularity_cache">Cache and FASTA files installation</h3>
  <p>
    You can run the INSTALL.pl script to install the Cache data and FASTA files.
    For example, to set up the cache and corresponding FASTA for human GRCh38 in your local folder <code>$HOME/vep_data</code>:
  </p>
  <pre class="code sh_sh">singularity exec vep.sif INSTALL.pl -c $HOME/vep_data -a cf -s homo_sapiens -y GRCh38</pre>

  <p>
    The installer downloads data to the specified directory (e.g., <code>$HOME/vep_data</code>).
    When running Ensembl VEP via Singularity, point to this directory using <code>--dir</code>:
  </p>

  <pre class="code sh_sh">singularity exec vep.sif vep --dir $HOME/vep_data -i examples/homo_sapiens_GRCh38.vcf --cache</pre>

<br />
<h3 id="singularity_setup">Running Ensembl VEP with data from local folder</h3>

<p>Here is an example on running Ensembl VEP with data from folder <code>$HOME/vep_data</code> in the local machine (provided that the cache has been downloaded to that folder):</p>

<pre class="code sh_sh">
singularity exec vep.sif \
  vep --dir $HOME/vep_data \
      --cache --offline --format vcf --vcf --force_overwrite \
      --input_file input/my_input.vcf \
      --output_file output/my_output.vcf \
      --custom file=custom/my_extra_data.bed,short_name=BED_DATA,format=bed,type=exact,coords=1 \
      --plugin NMD</pre>

<br />
<h3>Update from a previous version</h3>

<ol>
  <li><p>Update your docker container</p>
    <pre class="code sh_sh">singularity pull --name vep.sif docker://ensemblorg/ensembl-vep</pre>
  </li>
  <li><p>Update your cache</p>
    <pre class="code sh_sh"># Install the new cache through the VEP INSTALL.pl script (see "Cache installation" section above)
singularity exec vep.sif INSTALL.pl -c $HOME/vep_data -a c

# Or install the cache manually
cd $HOME/vep_data
curl -O [[SPECIESDEFS::ENSEMBL_FTP_URL]]/release-[[SPECIESDEFS::ENSEMBL_VERSION]]/variation/vep/homo_sapiens_vep_[[SPECIESDEFS::ENSEMBL_VERSION]]_GRCh38.tar.gz
tar xzf homo_sapiens_vep_[[SPECIESDEFS::ENSEMBL_VERSION]]_GRCh38.tar.gz</pre>
  </li>
</ol>


  <hr/>
  <h2 id="nextflow">Nextflow</h2>

  <p>We offer a <a href="https://github.com/Ensembl/ensembl-vep/tree/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/nextflow" rel="external">Nextflow Ensembl VEP pipeline</a> that aims to run Ensembl VEP using simple parallelisation. The pipeline is deployable on an individual Linux machine or on computing clusters running LSF, SLURM or other workload managers.</p>

  <p>The process can be summarised briefly by the following steps:</p>

  <ul>
    <li>Splitting the input data into multiple files using a given number of bins</li>
    <li>Running Ensembl VEP on the split files in parallel</li>
    <li>Merging Ensembl VEP outputs into a single file</li>
  </ul>

  <p>To run the pipeline in a system with <a href="https://nextflow.io" rel="external">Nexflow</a> installed, you will need to prepare a <a href="vep_options.html#opt_config"><b>vep.ini</b> config file</a>. Here are some examples commands to run the Nextflow Ensembl VEP pipeline:</p>

  <pre class="code sh_sh">
# Run Nextflow Ensembl VEP using local Ensembl VEP installation
# NB: Nextflow automatically downloads the GitHub repository
nextflow run Ensembl/ensembl-vep -r main \
  --input input.vcf \
  --vep_config vep.ini

# Run latest Ensembl VEP version using Docker
nextflow run Ensembl/ensembl-vep -r main \
  -profile docker \
  --input input.vcf \
  --vep_config vep.ini

# Run Ensembl VEP [[SPECIESDEFS::ENSEMBL_VERSION]].0 using Docker
nextflow run Ensembl/ensembl-vep -r main \
  -profile docker \
  --input input.vcf \
  --vep_config vep.ini \
  --vep_version [[SPECIESDEFS::ENSEMBL_VERSION]].0

# Run Ensembl VEP [[SPECIESDEFS::ENSEMBL_VERSION]].0 using SLURM and Singularity
nextflow run Ensembl/ensembl-vep -r main \
  -profile slurm,singularity \
  --input input.vcf \
  --vep_config vep.ini \
  --vep_version [[SPECIESDEFS::ENSEMBL_VERSION]].0
</pre>

  <p>For a full list of supported profiles, as well as more instructions on setting up and running the pipeline, please refer to the <a href="https://github.com/Ensembl/ensembl-vep/tree/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/nextflow" rel="external">Nextflow Ensembl VEP instructions</a>.</p>

</div>
  </div>
</div>

<div style="clear:both"><h1 style="page-break-after: always">&nbsp</h1></div>    
<style>
tr:nth-child(odd) {background-color: #f0f0f0;}
</style>

  <img src="/img/vep_logo.png" class="float-right" style="width:111px;height:50px" />	
 
  <h1 id="top"><span style="color:#006;padding-right:15px">Ensembl Variant Effect Predictor</span><span style="color:#666"><img src="/i/16/copy.png"/> Data formats</span></h1>
  <hr />

    <!-- Input formats -->

    <h2 id="input">Input</h2>
    
    <p> Both the web and command line interfaces to Ensembl VEP can use the same input formats.</p>
    <p>Supported input formats:</p>
    
    <table class="ss">
      <tr>
        <th>Format</th>
        <th>Variant example</th>
        <th><a href="#sv">Structural variant example</a></th>
      </tr>
      <tr>
        <td><a href="#default">Default Ensembl VEP input</a></td>
        <td><kbd>1 881907 881906 -/C +</kbd></td>
        <td><kbd>1 160283 471362 DUP +</kbd></td>
      </tr>
      <tr>
        <td><a href="#vcf">VCF</a></td>
        <td><kbd>1 65568 . A C . . .</kbd></td>
        <td><kbd>1 7936271 . N N[12:58877476[ . . SVTYPE=BND</kbd></td>
      </tr>
      <tr>
        <td><a href="#hgvs">HGVS identifiers</a></td>
        <td><kbd>ENST00000618231.3:c.9G>C</kbd></td>
        <td style="text-align: center"><font style="color: darkgray">&#10007; Not supported</font></td>
      </tr>
      <tr>
        <td><a href="#id">Variant identifiers</a></td>
        <td><kbd>rs699</kbd></td>
        <td><kbd>nsv1000164</kbd></td>
      </tr>
      <tr>
        <td><a href="#spdi">Genomic SPDI notation</a></td>
        <td><kbd>NC_000016.10:68684738:G:A</kbd></td>
        <td style="text-align: center"><font style="color: darkgray">&#10007; Not supported</font></td>
      </tr>
      <tr>
        <td><a href="#region">REST-style regions</a></td>
        <td><kbd>14:19584687-19584687:-1/T</kbd></td>
        <td><kbd>21:25587759-25587769/DEL</kbd></td>
      </tr>
    </table>

    <div>
      <div class="info" style="float:left">
        <h3>Note</h3>
        <div class="message-pad">
          <p>Ensembl VEP can read compressed (gzipped) input file of any format listed above, e.g.:</p>
          <pre class="code sh_sh"> ./vep -i input.vcf.gz -o output.txt</pre>
        </div>
      </div>
      <div style="clear:both"></div>
    </div>
      

    <br />
    <hr /> 
    <h2 id="default"> Default Ensembl VEP input</h2>
    
    <p>The default format is a simple <strong>whitespace-separated</strong> format
    (columns may be separated by space or tab characters),
    containing five required columns plus an optional identifier column:
    </p>
    
    <ol>
        <li><strong>chromosome</strong> - just the name or number, with no 'chr'
        prefix</li>
        <li><strong>start</strong></li>
        <li><strong>end</strong></li>
        <li><strong>allele</strong> - pair of alleles separated by a '/', with the
        reference allele first (or <a href="#sv">structural variant type)</a></li>
        <li><strong>strand</strong> - defined as + (forward) or - (reverse). The alleles will be reverse complemented for mapping to the genome if the minus strand is provided.</li>
        <li><strong>identifier</strong> - this identifier will be used the
        output. If not provided, Ensembl VEP will construct an identifier from the given
        coordinates and alleles.</li>
    </ol>
    
    <pre class="code sh_sh">
1   881907    881906    -/C   +
2   946507    946507    G/C   +
5   140532    140532    T/C   +
8   150029    150029    A/T   +    var2
12  1017956   1017956   T/A   +
14  19584687  19584687  C/T   -
19  66520     66520     G/A   +    var1</pre>
    
    <p>An insertion (of any size) is indicated by start coordinate = end coordinate
    + 1. For example, an insertion of 'C' between nucleotides 12600 and 12601 on the
    forward strand of chromosome 8 is indicated as follows:</p>
    
    <pre class="code sh_sh">8   12601     12600     -/C   +</pre>
    <p>A deletion is indicated by the exact nucleotide coordinates. For example, a
    three base pair deletion of nucleotides 12600, 12601, and 12602 of the reverse
    strand of chromosome 8 will be:</p>
    
    <pre class="code sh_sh">8   12600     12602     CGT/- -</pre>
    
    <p>
      Structural variants are also supported by indicating a
      <a href="#sv">structural variant type</a> instead of the allele:
    </p>
  
    <pre class="code sh_sh">
1    20000     30000     CN4  +  cnv4
1    160283    471362    DUP  +  dup
1    1385015   1387562   DEL  +  del1
12   1017956   1017956   INV  +  inv
21   25587759  25587769  CN0  +  del2</pre>
    
    <br />
    <hr />
    <h2 id="vcf"> VCF</h2>
    
    <p> <a rel="external" href="http://www.1000genomes.org/wiki/Analysis/vcf4.0">VCF (Variant Call Format)
    version 4.0</a> is supported. This is a common format produced by many variant calling tools
    and is the recommended format for use with Ensembl VEP: </p>

  <pre class="code sh_sh">
#CHROM  POS        ID      REF  ALT            QUAL  FILTER  INFO  FORMAT
1       65568      .       A    C              .     .       .     .
1       230710048  rs699   A    G              .     .       .     .
2       265023     .       C    T              .     .       .     .
3       319780     .       GA   G              .     .       .     .
20      3          .       C    CAAG,CAAGAAG   .     PASS    .     .
21      43762120   rs1300  T    A,C,G          .     .       .     .</pre>
    
    <p>Structural variants are also supported depending on
      <a href="#sv">structural variant type.</a></p>
    
    <p> Users using VCF should note a peculiarity in the difference between how
    Ensembl and VCF describe unbalanced variants. For any unbalanced variant (i.e.
    insertion, deletion or unbalanced substitution), the VCF specification requires
    that the base immediately before the variant should be included in both the
    reference and variant alleles. This also affects the reported position i.e. the
    reported position will be one base before the actual site of the variant. </p>
    
    <p> In order to parse this correctly, Ensembl VEP needs to convert such variants
    into Ensembl-type coordinates, and it does this by removing the additional base
    and adjusting the coordinates accordingly. This means that if an identifier is
    not supplied for a variant (in the 3rd column of the VCF), then the identifier
    constructed and the position reported in Ensembl VEP output file will differ from
    the input. </p>
    
    <p> This problem can be overcome with the following: </p>
    <ol>
      <li> ensuring each variant has a unique
    identifier specified in the 3rd column of the VCF </li>
      <li> using VCF format as output (<a href="script/vep_options.html#opt_vcf">--vcf</a>) - this preserves the formatting of your input coordinates and alleles </li>
      <li> using <a href="script/vep_options.html#opt_minimal">--minimal</a> and <a href="script/vep_options.html#opt_allele_number">--allele_number</a> (see <a href="#complex_vcf">Complex VCF entries</a>).</li> 
    </ol>
    
    <p> The following examples illustrate how VCF describes a variant and how it is
    handled internally by Ensembl VEP. Consider the following aligned sequences (for the
    purposes of discussion on chromosome 20): </p>
    
    <pre class="code sh_sh">Ref: a t C g a // C is the reference base
1 : a t G g a // C base is a G in individual 1
2 : a t - g a // C base is deleted w.r.t. the reference in individual 2
3 : a t CAg a // A base is inserted w.r.t. the reference sequence in individual 3</pre>
    
    <h4> Individual 1 </h4>
    
    <p> The first individual shows a simple balanced substitution of G for C at base
    3. This is described in a compatible manner in VCF and Ensembl styles. Firstly,
    in VCF: </p>
    
    <pre class="code sh_sh">20   3   .   C   G   .   PASS   .</pre>
    
    <p> And in Ensembl format: </p>
    
    <pre class="code sh_sh"> 20   3   3   C/G   +</pre>
    
    <h4> Individual 2 </h4>
    
    <p> The second individual has the 3rd base deleted relative to the reference. In
    VCF, both the reference and variant allele columns must include the preceding
    base (T) and the reported position is that of the preceding base: </p>
    
    <pre class="code sh_sh">20   2   .   TC   T   .   PASS   .</pre>
    
    <p> In Ensembl format, the preceding base is not included, and the start/end
    coordinates represent the region of the sequence deleted. A "-" character is
    used to indicate that the base is deleted in the variant sequence: </p>
    
    <pre class="code sh_sh">20   3   3   C/-   +</pre>
    
    <p> The upshot of this is that while in the VCF input file the position of the
    variant is reported as 2, in the output file in Ensembl VEP default format the position will be
    reported as 3. If no identifier is provided in the third column of the VCF, then
    the constructed identifier will be: </p>
    
    <pre class="code sh_sh">20_3_C/-</pre>
    
    <h4> Individual 3 </h4>
    
    <p> The third individual has an "A" inserted between the 3rd and 4th bases of
    the sequence relative to the reference. In VCF, as for the deletion, the base
    before the insertion is included in both the reference and variant allele
    columns, and the reported position is that of the preceding base: </p>
    
    <pre class="code sh_sh">20   3   .   C   CA   .   PASS   .</pre>
    
    <p> In Ensembl format, again the preceding base is not included, and the
    start/end positions are "swapped" to indicate that this is an insertion.
    Similarly to a deletion, a "-" is used to indicate no sequence in the reference:
    </p>
    
    <pre class="code sh_sh"> 20   4   3   -/A   +</pre>
    
    <p> Again, the output will appear different, and the constructed identifier may
    not be what is expected: </p>
    
    <pre class="code sh_sh">20_3_-/A</pre>

    <p> Using VCF format output, or adding unique identifiers to the input (in the
    third VCF column), can mitigate this issue. </p>
   

    <br /> 
    <h3 id="complex_vcf"> Complex VCF entries </h3>

    <p> For VCF entries with multiple alternate alleles, Ensembl VEP will only trim the
    leading base from alleles if <b>all</b> REF and ALT alleles start with the same
    base: </p>

    <pre class="code sh_sh">20   3   .   C   CAAG,CAAGAAG   .   PASS   .</pre>

    <p> This will be considered internally by Ensembl VEP as equivalent to: </p>

    <pre class="code sh_sh">20   4   3   -/AAG/AAGAAG   +</pre>

    <p> Now consider the case where a single VCF line contains a representation
    of both a SNV and an insertion: </p>

    <pre class="code sh_sh">20   3   .   C   CAAAG,G   .   PASS   .</pre>

    <p> Here the input alleles will remain unchanged, and Ensembl VEP will consider the first
    REF/ALT pair as a substitution of C for CAAG, and the second as a C/G SNV:</p>

    <pre class="code sh_sh">20   3   3   C/CAAG/G   +</pre>

    <p> To modify this behaviour, with the commandline tool you can use <a href="script/vep_options.html#opt_minimal">--minimal</a>.
    This flag forces Ensembl VEP to consider each REF/ALT pair independently, trimming
    identical leading and trailing bases from each as appropriate. Since this can
    lead to confusing output regarding coordinates etc, it is not the default
    behaviour. It is recommended to use the
    <a href="script/vep_options.html#opt_allele_number">--allele_number</a> flag to track
    the correspondence between alleles as input and how they appear in the output.</p>

    <br />
    <hr />
    <h2 id="hgvs"> HGVS identifiers</h2>
    
    <p> See <a rel="external" href="//varnomen.hgvs.org">https://varnomen.hgvs.org</a> for
    details. These must be relative to genomic or Ensembl transcript coordinates.</p>
    <p>It also is possible to use RefSeq transcripts, if they match the reference genome
     See <a href="script/vep_other.html#hgvs">HGVS documentation</a>
    </p>
    
    <p> Examples: </p>
    
    <pre class="code sh_sh">ENST00000618231.3:c.9G>C
ENST00000471631.1:c.28_33delTCGCGG
ENST00000285667.3:c.1047_1048insC
5:g.140532G>C</pre>
    
    <p> Examples using RefSeq identifiers (using <a
    href="script/vep_options.html#opt_refseq">--refseq</a> in the command line or select
    the 'RefSeq transcripts' on the web interface: </p>
    
    <pre class="code sh_sh">NM_153681.2:c.7C>T
NM_005239.6:c.190G>A
NM_001025204.2:c.336G>A</pre>
    
    <p> HGVS protein notations may also be used, provided that they
    unambiguously map to a single genomic change. Due to redundancy in the amino
    acid code, it is not always possible to work out the corresponding genomic
    sequence change for a given protein sequence change. The following example
    is for a permissable protein notation in dog <i>(Canis familiaris)</i>:</p>
    
    <pre class="code sh_sh">ENSCAFP00000040171.1:p.Thr92Asn</pre>

    <h4>Ambiguous gene-based descriptions</h4>
    <p>It is possible to use ambiguous descriptions listing only gene symbol or UniProt accession and protein change (e.g. PHF21B:p.Tyr124Cys,  P01019:p.Ala268Val), as seen in the literature, though this is not recommended as it can produce multiple different variants at genomic level. The transcripts for a gene are considered in the following order:
      <ol>
        <li><a href="/info/genome/genebuild/mane.html">MANE Select transcript status</a></li>
        <li><a href="/info/genome/genebuild/mane.html">MANE Plus Clinical transcript status</a></li>
        <li>canonical status of transcript</li>
        <li><a href="/Help/Glossary?id=521">APPRIS isoform annotation</a></li>
        <li><a href="/Help/Glossary?id=492">transcript support level</a></li>
        <li>biotype of transcript ("protein_coding" preferred)</li>
        <li>CCDS status of transcript</li>
        <li>consequence rank according to <a href="/info/genome/variation/prediction/predicted_data.html#consequences">this table</a></li>
        <li>translated, transcript or feature length (longer preferred)</li>
      </ol>
      and the first compatible transcript is used to map variants to the genome for annotation.</p>

    <br />
    <hr />
    <h2 id="id"> Variant identifiers</h2>
    
    <p> These should be dbSNP rsIDs (such as <kbd>rs699</kbd>), or any synonym for a variant present in
    the Ensembl Variation database. Structural variant identifiers (like <kbd>nsv1000164</kbd> and
    <kbd>esv1850194</kbd>) are also supported.</p>

    <p> See <a
    href="/info/genome/variation/species/sources_documentation.html">here</a> for a list of
    identifier sources in Ensembl. </p>

    <p> Examples: </p>

    <pre class="code sh_sh">rs1156485833
rs1258750482
rs867704559
esv1815690
nsv1000164</pre>
   
    <div>
      <div class="info" style="float:left">
        <h3>Note</h3>
        <div class="message-pad">
          <p>Ensembl VEP is optimised for the analysis of variants in VCF files (sorted by location) or other position-based formats. Using variant identifiers can take 2-5 times longer as a database lookup to find the variant location is required.</p>
        </div>
      </div>
      <div style="clear:both"></div>
    </div> 

    <br />
    <hr />
    <h2 id="spdi"> Genomic SPDI notation</h2>

    <p> Genomic SPDI notation which uses four fields delimited by colons S:P:D:I (Sequence:Position:Deletion:Insertion) is also supported. In SPDI notation, the position refers to the base before the variant, not the base of the variant itsef.</p> 
    <p>See <a href="https://www.ncbi.nlm.nih.gov/variation/notation/">here</a> for details.</p>

    <p> Examples: </p>
    
    <pre class="code sh_sh">NC_000016.10:68684738:G:A
NC_000017.11:43092199:GCTTTT:
NC_000013.11:32315789::C
NC_000016.10:68644746:AA:GTA
16:68684738:2:AC</pre>


    <br />
    <hr />
    <h2 id="region"> REST-style regions</h2>
    <p>
      The Ensembl VEP region REST endoint requires variants are described as
      <code>[chr]:[start]-[end]:[strand]/[allele]</code>.
    </p>
    <p>
      This follows the same conventions as the
      <a href="#default">default input format</a>, with the key difference being
      that this format does not require the reference (REF) allele to be
      included; this will be looked up  using either
      a provided FASTA file (preferred) or Ensembl core database. Strand is
      optional and defaults to 1 (forward strand).
    </p>

    <pre class="code sh_sh"># SNP
5:140532-140532:1/C

# SNP (reverse strand)
14:19584687-19584687:-1/T

# insertion
1:881907-881906:1/C

# 5bp deletion
2:946507-946511:1/-</pre>

  <p>
    Structural variants are also supported by indicating a
    <a href="#sv">structural variant type</a> in the place of the
    <code>[allele]</code>:
  </p>
  <pre class="code sh_sh"># structural variant: deletion
21:25587759-25587769/DEL
    
# structural variant: inversion
21:25587759-25587769/INV</pre>

<br />
    <hr />
    <h2 id="sv"> Structural variant types</h2>

    <p>Ensembl VEP also predicts molecular consequences for structural variants using the
      following input formats:</p>
    <ul>
      <li><a href="#default">Default Ensembl VEP input</a></li>
      <li><a href="#region">REST-style regions</a></li>
      <li> <a href="#id">Variant identifiers</a></li>
      <li> <a href="#vcf">VCF</a></li>
    </ul>

    <p> To recognise a variant as a structural variant, the allele string
      (or <code>SVTYPE</code> in the INFO column of the VCF format) must be set
      to one of the currently supported values: </p>

    <ul>
        <li><b>INS</b> - insertion</li>
        <ul>
          <li><b>INS:ME</b> - insertion of mobile element</li>
          <li><b>INS:ME:ALU</b> - insertion of ALU element</li>
          <li><b>INS:ME:HERV</b> - insertion of HERV element</li>
          <li><b>INS:ME:LINE1</b> - insertion of LINE1 element</li>
          <li><b>INS:ME:SVA</b> - insertion of SVA element</li>
        </ul>

        <li><b>DEL</b> - deletion</li>
        <ul>
          <li><b>DEL:ME</b> - deletion of mobile element</li>
          <li><b>DEL:ME:ALU</b> - deletion of ALU element</li>
          <li><b>DEL:ME:HERV</b> - deletion of HERV element</li>
          <li><b>DEL:ME:LINE1</b> - deletion of LINE1 element</li>
          <li><b>DEL:ME:SVA</b> - deletion of SVA element</li>
        </ul>

        <li><b>DUP</b> - duplication</li>
        <ul>
          <li><b>DUP:TANDEM</b> - tandem duplication</li>
          <li><b>TDUP</b> - tandem duplication</li>
        </ul>
        <li><b>INV</b> - inversion</li>

        <li><b>CNV</b> - copy number variation</li>
        <ul>
          <li>
            The copy number value can be specified like so:
            <ul>
              <li><kbd>CN0</kbd></li>
              <li><kbd>CN=4</kbd></li>
              <li><kbd>CN3,CN4,CN6</kbd></li>
              <li><kbd>CN=0,CN=2,CN=4</kbd></li>
            </ul>
          </li>
          <li><b>CNV:TR</b> - tandem repeats</li>
          <ul>
            <li> Requires <code>INFO</code> fields describing the tandem repeat, such as <code>RUS</code> and <code>RN</code> – check <a href="https://samtools.github.io/hts-specs/VCFv4.4.pdf" rel="external">VCF 4.4 specification, section 5.7</a>
            <li> Currently, the <code>CIRUC</code> and <code>CIRB</code> <code>INFO</code> fields are ignored when calculating alternative alleles in tandem repeats
          </ul>
        </ul>

        <li><b>BND</b> - chromosome breakpoints</li>
        <ul>
          <li> Breakpoint variants are composed by one or more breakends
          <li>
            In VCF, breakend replacements are inserted into the <code>ALT</code>
            column and need to meet the
            <a rel="external" href="http://samtools.github.io/hts-specs/">HTS specifications</a>,
            such as <kbd>TG[12:58877476[</kbd>
          </li>
          <li>Single breakends can be specified in <code>ALT</code>, such as <kbd>T.</kbd> and <kbd>.G</kbd></li>
          <li>Multiple, comma-separated alternative breakends can be specified in <code>ALT</code>, such as <kbd>A[22:22893780[,A[X:10932343[</kbd></li>
          <li>Breakends are only supported for VCF input format</li>
        </ul>
    </ul>

    <p>More information on how Ensembl VEP processes structural variants can be found <a href="script/vep_example.html#StructVar">here</a>.</p>

    <h3> Examples of structural variants encoded in VCF format </h3>

    <pre class="code sh_sh">
#CHROM  POS        ID   REF  ALT             QUAL  FILTER  INFO
1       160283     dup  .    &lt;DUP&gt;           .     .       SVTYPE=DUP;END=471362
1       1385015    del  .    &lt;DEL&gt;           .     .       SVTYPE=DEL;END=1387562
1       7936271    bnd  N    N[12:58877476[  .     .       SVTYPE=BND</pre>

    <p> See the <a
    rel="external" href="http://www.1000genomes.org/wiki/Analysis/Variant%20Call%20Format/VCF%20%28Variant%20Call%20Format%29%20version%204.0/encoding-structural-variants">VCF
    definition document</a> for more detail on how to describe
    structural variants in VCF format. </p>

    <!-- Output formats -->
    <br />
    <hr style="margin-bottom:0"/>
    <hr style="margin-top:0"/>
    <h2 id="output">Output</h2>

    <p>Ensembl VEP can return the results in different formats:</p>
    <ul>
      <li><a href="#defaultout">Default Ensembl VEP output</a></li>
      <li><a href="#tab">Tab-delimited output</a></li>
      <li><a href="#vcfout">VCF</a></li>
      <li><a href="#json">JSON output</a></li>
    </ul>
   
    <p>Along with the results Ensembl VEP computes and returns some <a href="#stats">statistics</a>.</p>


    <br />
    <hr />
    <h2 id="defaultout"> Default Ensembl VEP output</h2>
 
    <p> The default output format ("VEP" format when downloading from the web interface) is a 14 column tab-delimited file. Empty values are denoted by '-'. The output
    columns are:</p>
    
    <ol>
        <li><strong>Uploaded variation</strong> - as chromosome_start_alleles</li>
        <li><strong>Location</strong> - in standard coordinate format (chr:start or chr:start-end)</li>
        <li><strong>Allele</strong> - the variant allele used to calculate the consequence</li>
        <li><strong>Gene</strong> - Ensembl stable ID of affected gene</li>
        <li><strong>Feature</strong> - Ensembl stable ID of feature</li>
        <li><strong>Feature type</strong> - type of feature. Currently one of Transcript, RegulatoryFeature, MotifFeature.</li>
        <li><strong>Consequence</strong> - <a href="/info/genome/variation/prediction/predicted_data.html#consequences">consequence type</a> of this variant</li>
        <li><strong>Position in cDNA</strong> - relative position of base pair in cDNA sequence</li>
        <li><strong>Position in CDS</strong> - relative position of base pair in coding sequence</li>
        <li><strong>Position in protein</strong> - relative position of amino acid in protein</li>
        <li><strong>Amino acid change</strong> - only given if the variant affects the protein-coding sequence</li>
        <li><strong>Codon change</strong> - the alternative codons with the variant base in upper case</li>
        <li><strong>Co-located variation</strong> - identifier of any <a href="/info/docs/tools/vep/script/vep_other.html#colocated">existing variants</a>. Switch on with <a href="script/vep_options.html#opt_check_existing">--check_existing</a></li>
        <li><strong>Extra</strong> - this column contains extra information as key=value pairs separated by ";", see below.</li>
    </ol>


    <h3 style="margin:20px 0px 30px">Other output fields:</h3>

    <div id="div_other_fields" >
      <ul>
        <li><b>REF_ALLELE</b> - the reference allele (after minimisation)</li>
        <li><b>UPLOADED_ALLELE</b> - the uploaded allele string (before minimisation)</li>
        <li><b>IMPACT</b> - the impact modifier for the consequence type</li>
        <li><b>VARIANT_CLASS</b> - Sequence Ontology <a href="/info/genome/variation/prediction/classification.html#classes">variant class</a></li>
        <li><b>SYMBOL</b> - the gene symbol</li>
        <li><b>SYMBOL_SOURCE</b> - the source of the gene symbol</li>
        <li><b>STRAND</b> - the DNA strand (1 or -1) on which the transcript/feature lies</li>
        <li><b>ENSP</b> - the Ensembl protein identifier of the affected transcript</li>
        <li>
            <b>FLAGS</b> - transcript quality flags:
            <ul>
                <li><i>cds_start_NF:</i> CDS 5' incomplete</li>
                <li><i>cds_end_NF:</i> CDS 3' incomplete</li>
            </ul>
        </li>
        <li><b>SWISSPROT</b> - Best match UniProtKB/Swiss-Prot accession of protein product
        <li><b>TREMBL</b> - Best match UniProtKB/TrEMBL accession of protein product</li>
        <li><b>UNIPARC</b> - Best match UniParc accession of protein product</li>
        <li><b>HGVSc</b> - the HGVS coding sequence name</li>
        <li><b>HGVSp</b> - the HGVS protein sequence name</li>
        <li><b>HGVSg</b> - the HGVS genomic sequence name</li>
        <li><b>HGVS_OFFSET</b> - Indicates by how many bases the HGVS notations for this variant have been <a href="script/vep_options.html#opt_shift_hgvs">shifted</a>. Value must be greater than 0.</li>
        <li><b>NEAREST</b> - Identifier(s) of nearest transcription start site</li>
        <li><b>SIFT</b> - the SIFT prediction and/or score, with both given as prediction(score)</li>
        <li><b>PolyPhen</b> - the PolyPhen prediction and/or score</li>
        <li><b>MOTIF_NAME</b> - the source and identifier of a transcription factor binding profile aligned at this position</li>
        <li><b>MOTIF_POS</b> - The relative position of the variation in the aligned TFBP</li>
        <li><b>HIGH_INF_POS</b> - a flag indicating if the variant falls in a high information position of a transcription factor binding profile (TFBP)</li>
        <li><b>MOTIF_SCORE_CHANGE</b> - The difference in motif score of the reference and variant sequences for the TFBP</li>
        <li><b>CELL_TYPE</b> - List of cell types and classifications for regulatory feature</li>
        <li><b>CANONICAL</b> - a flag indicating if the transcript is denoted as the canonical transcript for this gene</li>
        <li><b>CCDS</b> - the CCDS identifer for this transcript, where applicable</li>
        <li><b>INTRON</b> - the intron number (out of total number)</li>
        <li><b>EXON</b> - the exon number (out of total number)</li>
        <li><b>DOMAINS</b> - the source and identifer of any overlapping protein domains</li>
        <li><b>DISTANCE</b> - Shortest distance from variant to transcript. <i>Note: DISTANCE of 0 is possible for insertions happening just before or after a transcript because variant coordinates are considered to be the flanking bases where insertion happens.</i></li>
        <li><b>IND</b> - individual name</li>
        <li><b>ZYG</b> - zygosity of individual genotype at this locus</li>
        <li><b>SV</b> - IDs of overlapping structural variants</li>
        <li><b>FREQS</b> - Frequencies of overlapping variants used in filtering</li>
        <li><b>AF</b> - Frequency of existing variant in 1000 Genomes</li>
        <li><b>AFR_AF</b> - Frequency of existing variant in 1000 Genomes combined African population</li>
        <li><b>AMR_AF</b> - Frequency of existing variant in 1000 Genomes combined American population</li>
        <li><b>ASN_AF</b> - Frequency of existing variant in 1000 Genomes combined Asian population</li>
        <li><b>EUR_AF</b> - Frequency of existing variant in 1000 Genomes combined European population</li>
        <li><b>EAS_AF</b> - Frequency of existing variant in 1000 Genomes combined East Asian population</li>
        <li><b>SAS_AF</b> - Frequency of existing variant in 1000 Genomes combined South Asian population</li>
        <li><b>gnomADe_AF</b> - Frequency of existing variant in gnomAD exomes combined population</li>
        <li><b>gnomADe_AFR_AF</b> - Frequency of existing variant in gnomAD exomes African/American population</li>
        <li><b>gnomADe_AMR_AF</b> - Frequency of existing variant in gnomAD exomes American population</li>
        <li><b>gnomADe_ASJ_AF</b> - Frequency of existing variant in gnomAD exomes Ashkenazi Jewish population</li>
        <li><b>gnomADe_EAS_AF</b> - Frequency of existing variant in gnomAD exomes East Asian population</li>
        <li><b>gnomADe_FIN_AF</b> - Frequency of existing variant in gnomAD exomes Finnish population</li>
        <li><b>gnomADg_MID_AF</b> - Frequency of existing variant in gnomAD exomes Mid-eastern population</li>
        <li><b>gnomADe_NFE_AF</b> - Frequency of existing variant in gnomAD exomes Non-Finnish European population</li>
        <li><b>gnomADe_REMAINING_AF</b> - Frequency of existing variant in gnomAD exomes combined remaining combined populations</li>
        <li><b>gnomADe_SAS_AF</b> - Frequency of existing variant in gnomAD exomes South Asian population</li>
        <li><b>gnomADg_AF</b> - Frequency of existing variant in gnomAD genomes combined population</li>
        <li><b>gnomADg_AFR_AF</b> - Frequency of existing variant in gnomAD genomes African/American population</li>
        <li><b>gnomADg_AMI_AF</b> - Frequency of existing variant in gnomAD genomes Amish population</li>
        <li><b>gnomADg_AMR_AF</b> - Frequency of existing variant in gnomAD genomes American population</li>
        <li><b>gnomADg_ASJ_AF</b> - Frequency of existing variant in gnomAD genomes Ashkenazi Jewish population</li>
        <li><b>gnomADg_EAS_AF</b> - Frequency of existing variant in gnomAD genomes East Asian population</li>
        <li><b>gnomADg_FIN_AF</b> - Frequency of existing variant in gnomAD genomes Finnish population</li>
        <li><b>gnomADg_MID_AF</b> - Frequency of existing variant in gnomAD genomes Mid-eastern population</li>
        <li><b>gnomADg_NFE_AF</b> - Frequency of existing variant in gnomAD genomes Non-Finnish European population</li>
        <li><b>gnomADg_REMAINING_AF</b> - Frequency of existing variant in gnomAD genomes combined remaining combined populations</li>
        <li><b>gnomADg_SAS_AF</b> - Frequency of existing variant in gnomAD genomes South Asian population</li>
        <li><b>MAX_AF</b> - Maximum observed allele frequency in 1000 Genomes, ESP and gnomAD</li>
        <li><b>MAX_AF_POPS</b> - Populations in which maximum allele frequency was observed</li>
        <li><b>CLIN_SIG</b> - ClinVar clinical significance of the dbSNP variant</li>
        <li><b>BIOTYPE</b> - Biotype of transcript or regulatory feature</li>
        <li><b>APPRIS</b> - Annotates alternatively spliced transcripts as primary or alternate based on a range of computational methods. NB: not available for GRCh37</li>
        <li><b>TSL</b> - Transcript support level. NB: not available for GRCh37</li>
        <li><b>GENCODE_PRIMARY</b> - Reports if transcript belongs to GENCODE primary subset</li>
        <li><b>PUBMED</b> - Pubmed ID(s) of publications that cite existing variant</li>
        <li><b>SOMATIC</b> - Somatic status of existing variant(s); multiple values correspond to multiple values in the Existing_variation field</li>
        <li><b>PHENO</b> - Indicates if existing variant is associated with a phenotype, disease or trait; multiple values correspond to multiple values in the Existing_variation field</li>
        <li><b>GENE_PHENO</b> - Indicates if overlapped gene is associated with a phenotype, disease or trait</li>
        <li><b>ALLELE_NUM</b> - Allele number from input; 0 is reference, 1 is first alternate etc</li>
        <li><b>MINIMISED</b> - Alleles in this variant have been converted to minimal representation before consequence calculation</li>
        <li><b>PICK</b> - indicates if this block of consequence data was picked by <a href="script/vep_options.html#opt_flag_pick">--flag_pick</a> or <a href="script/vep_options.html#opt_flag_pick_allele">--flag_pick_allele</a></li>
        <li><b>BAM_EDIT</b> - Indicates success or failure of edit using BAM file</li>
        <li><b>GIVEN_REF</b> - Reference allele from input</li>
        <li><b>USED_REF</b> - Reference allele as used to get consequences</li>

        <li id="refseq_match"><b>REFSEQ_MATCH</b> - the RefSeq transcript match status; contains a number of flags indicating whether this RefSeq transcript matches the underlying reference sequence and/or an Ensembl transcript (<a href="script/vep_other.html#refseq">more information</a>). 
          <ul>
            <li><i>rseq_3p_mismatch:</i> signifies a mismatch between the RefSeq transcript and the underlying primary genome assembly sequence. Specifically, there is a mismatch in the 3' UTR of the RefSeq model with respect to the primary genome assembly (e.g. GRCh37/GRCh38). </li>
            <li><i>rseq_5p_mismatch:</i> signifies a mismatch between the RefSeq transcript and the underlying primary genome assembly sequence. Specifically, there is a mismatch in the 5' UTR of the RefSeq model with respect to the primary genome assembly. </li>
            <li><i>rseq_cds_mismatch:</i> signifies a mismatch between the RefSeq transcript and the underlying primary genome assembly sequence. Specifically, there is a mismatch in the CDS of the RefSeq model with respect to the primary genome assembly. </li>
            <li><i>rseq_ens_match_cds:</i> signifies that for the RefSeq transcript there is an overlapping Ensembl model that is identical across the CDS region only. A CDS match is defined as follows: the CDS and peptide sequences are identical and the genomic coordinates of every translatable exon match. Useful related attributes are: rseq_ens_match_wt and rseq_ens_no_match.</li>
            <li><i>rseq_ens_match_wt:</i> signifies that for the RefSeq transcript there is an overlapping Ensembl model that is identical across the whole transcript. A whole transcript match is defined as follows: 1) In the case that both models are coding, the transcript, CDS and peptide sequences are all identical and the genomic coordinates of every exon match. 2) In the case that both transcripts are non-coding the transcript sequences and the genomic coordinates of every exon are identical. No comparison is made between a coding and a non-coding transcript. Useful related attributes are: rseq_ens_match_cds and rseq_ens_no_match.</li>
            <li><i>rseq_ens_no_match:</i> signifies that for the RefSeq transcript there is no overlapping Ensembl model that is identical across either the whole transcript or the CDS. This is caused by differences between the transcript, CDS or peptide sequences or between the exon genomic coordinates. Useful related attributes are: rseq_ens_match_wt and rseq_ens_match_cds.</li>
            <li><i>rseq_mrna_match:</i> signifies an exact match between the RefSeq transcript and the underlying primary genome assembly sequence (based on a match between the transcript stable id and an accession in the RefSeq mRNA file). An exact match occurs when the underlying genomic sequence of the model can be perfectly aligned to the mRNA sequence post polyA clipping.</li>
            <li><i>rseq_mrna_nonmatch:</i> signifies a non-match between the RefSeq transcript and the underlying primary genome assembly sequence. A non-match is deemed to have occurred if the underlying genomic sequence does not have a perfect alignment to the mRNA sequence post polyA clipping. It can also signify that no comparison was possible as the model stable id may not have had a corresponding entry in the RefSeq mRNA file (sometimes happens when accessions are retired or changed). When a non-match occurs one or several of the following transcript attributes will also be present to provide more detail on the nature of the non-match: rseq_5p_mismatch, rseq_cds_mismatch, rseq_3p_mismatch, rseq_nctran_mismatch, rseq_no_comparison</li>
            <li><i>rseq_nctran_mismatch:</i> signifies a mismatch between the RefSeq transcript and the underlying primary genome assembly sequence. This is a comparison between the entire underlying genomic sequence of the RefSeq model to the mRNA in the case of RefSeq models that are non-coding. </li>
            <li><i>rseq_no_comparison:</i> signifies that no alignment was carried out between the underlying primary genome assembly sequence and a corresponding RefSeq mRNA. The reason for this is generally that no corresponding, unversioned accession was found in the RefSeq mRNA file for the transcript stable id. This sometimes happens when accessions are retired or replaced. A second possibility is that the sequences were too long and problematic to align (though this is rare).</li>
          </ul>
        </li>
        <li><b>OverlapBP</b> - Number of base pairs overlapping with the corresponding structural variation feature</li>
        <li><b>OverlapPC</b> - Percentage of corresponding structural variation feature overlapped by the given input</li>
        <li><b>CHECK_REF</b> - Reports variants where the input reference does not match the expected reference</li>
        <li><b>AMBIGUITY</b> - IUPAC allele ambiguity code</li>
      </ul>
    </div>
   
    <p>Example of Ensembl VEP default output format:</p> 
    <pre class="code sh_sh">
11_224088_C/A    11:224088   A  ENSG00000142082  ENST00000525319  Transcript         missense_variant           742  716  239  T/N  aCc/aAc  -  SIFT=deleterious(0);PolyPhen=unknown(0)
11_224088_C/A    11:224088   A  ENSG00000142082  ENST00000534381  Transcript         5_prime_UTR_variant        -    -    -    -    -        -  -
11_224088_C/A    11:224088   A  ENSG00000142082  ENST00000529055  Transcript         downstream_variant         -    -    -    -    -        -  -
11_224585_G/A    11:224585   A  ENSG00000142082  ENST00000529937  Transcript         intron_variant             -    -    -    -    -        -  HGVSc=ENST00000529937.1:c.136-346G>A
22_16084370_G/A  22:16084370 A  -                ENSR00000615113  RegulatoryFeature  regulatory_region_variant  -    -    -    -    -        -  -</pre>
    
    <p> The Ensembl VEP command line tool will also add a header to the output file. This contains
    information about the databases connected to, and also a key describing the
    key/value pairs used in the extra column. </p>
    
    <pre class="code sh_sh">
## ENSEMBL VARIANT EFFECT PREDICTOR v[[SPECIESDEFS::ENSEMBL_VERSION]].0
## Output produced at 2017-03-21 14:51:27
## Connected to homo_sapiens_core_[[SPECIESDEFS::ENSEMBL_VERSION]]_38 on ensembldb.ensembl.org
## Using cache in /homes/user/.vep/homo_sapiens/[[SPECIESDEFS::ENSEMBL_VERSION]]_GRCh38
## Using API version [[SPECIESDEFS::ENSEMBL_VERSION]], DB version [[SPECIESDEFS::ENSEMBL_VERSION]]
## polyphen version 2.2.2
## sift version sift5.2.2
## COSMIC version 78
## ESP version 20141103
## gencode version GENCODE 25
## genebuild version 2014-07
## HGMD-PUBLIC version 20162
## regbuild version 16
## assembly version GRCh38.p7
## ClinVar version 201610
## dbSNP version 147
## Column descriptions:
## Uploaded_variation : Identifier of uploaded variant
## Location : Location of variant in standard coordinate format (chr:start or chr:start-end)
## Allele : The variant allele used to calculate the consequence
## Gene : Stable ID of affected gene
## Feature : Stable ID of feature
## Feature_type : Type of feature - Transcript, RegulatoryFeature or MotifFeature
## Consequence : Consequence type
## cDNA_position : Relative position of base pair in cDNA sequence
## CDS_position : Relative position of base pair in coding sequence
## Protein_position : Relative position of amino acid in protein
## Amino_acids : Reference and variant amino acids
## Codons : Reference and variant codon sequence
## Existing_variation : Identifier(s) of co-located known variants
## Extra column keys:
## IMPACT : Subjective impact classification of consequence type
## DISTANCE : Shortest distance from variant to transcript
## STRAND : Strand of the feature (1/-1)
## FLAGS : Transcript quality flags</pre>


    <br />
    <hr />
    <h2 id="tab"> Tab-delimited output</h2>

    <p> The <a href="script/vep_options.html#opt_tab">--tab</a> specifies
     a tab-delimited output file.<br />
    <b>This differs from the default output format in that each individual field from the 
    "Extra" field is written to a separate tab-delimited column</b>.<br />
    This makes the output more suitable for import into spreadsheet programs such as Excel.<br />
    Furthermore the header is the same as the one for the Ensembl VEP default output format and this is 
    also the format used when selecting the "TXT" option on the Ensembl VEP web interface.</p>

    <p>Example of tab-delimited output format:</p>
    <pre class="code sh_sh">
#Uploaded_variation  Location   Allele  Gene             Feature          Feature_type  Consequence                            cDNA_position  CDS_position  Protein_position  Amino_acids  Codons   Existing_variation  IMPACT    DISTANCE  STRAND  FLAGS 
11_224088_C/A        11:224088  A       ENSG00000142082  ENST00000525319  Transcript    missense_variant                       742            716           239               S/I          aGc/aTc  -                   MODERATE  -         -1      - 
11_224088_C/A        11:224088  A       ENSG00000142082  ENST00000534381  Transcript    downstream_gene_variant                -              -             -                 -            -        -                   MODIFIER  1674      -1      -
11_224088_C/A        11:224088  A       ENSG00000142082  ENST00000529055  Transcript    downstream_gene_variant                -              -             -                 -            -        -                   MODIFIER  134       -1      -
11_224585_G/A        11:224585  A       ENSG00000142082  ENST00000529937  Transcript    intron_variant,NMD_transcript_variant  -              -             -                 -            -        -                   MODIFIER  -         -1      -</pre>
    
    <div>
      <div class="info" style="float:left">
        <h3>Note</h3>
        <div class="message-pad">
          <p>The Existing_variation column is only populated if we use option <a href="script/vep_options.html#opt_check_existing">--check_existing</a> or any other option that switches it on (--af_1kg silently switches on --check_existing).</p>
        </div>
      </div>
      <div style="clear:both"></div>
    </div>

    <p> The choice and order of columns in the output may be configured using <a href="script/vep_options.html#opt_fields">--fields</a>. For instance:<br />
      <pre class="code sh_sh">./vep -i examples/homo_sapiens_GRCh38.vcf --cache --force_overwrite <b>--tab --fields "Uploaded variation,Location,Allele,Gene"</b></pre>
    </p>
    

    <br />
    <hr />
    <h2 id="vcfout"> VCF output</h2>
    
    <p> The Ensembl VEP commandline tool can also generate VCF output using the <a href="script/vep_options.html#opt_vcf">--vcf</a> flag.</p>
    <p>Main information about the VCF output format:</p>
    <ul>
      <li>Consequences are added in the INFO field of the VCF file, using the key "<b>CSQ</b>" 
          (you can change it using <a href="script/vep_options.html#opt_vcf_info_field">--vcf_info_field</a>).</li>
      <li>Data fields are encoded separated by the character "<b>&#124;</b>" (pipe). The order of fields is written in the VCF header. 
          Unpopulated fields are represented by an empty string.</li>
      <li>Output fields in the "CSQ" INFO field can be configured by using <a href="script/vep_options.html#opt_fields">--fields</a>.</li>
      <li>Each prediction, for a given variant, is separated by the character "<b>,</b>" in the CSQ INFO field (e.g. when a variant overlaps more than 1 transcript)</li>
    </ul>
    <br />

    <p>
    Here is a list of the (default) fields you can find within the CSQ field:
    <pre class="code sh_sh">Allele|Consequence|IMPACT|SYMBOL|Gene|Feature_type|Feature|BIOTYPE|EXON|INTRON|HGVSc|HGVSp|cDNA_position|CDS_position|Protein_position|Amino_acids|Codons|Existing_variation|DISTANCE|STRAND|FLAGS|SYMBOL_SOURCE|HGNC_ID</pre>
    </p>

    <div>
      <div class="info" style="float:left">
        <h3>Note</h3>
        <div class="message-pad">
          <p>Some fields are not reported by default and need configuring including HGVS, Symbol and Existing_variation.</p>
        </div>
      </div>
      <div style="clear:both"></div>
    </div>

    <p>Example command using the <a href="script/vep_options.html#opt_vcf">--vcf</a> and <a href="script/vep_options.html#opt_fields">--fields</a> options:
    <pre class="code sh_sh">./vep -i examples/homo_sapiens_GRCh38.vcf --cache --force_overwrite <b>--vcf --fields "Allele,Consequence,Feature_type,Feature"</b></pre>
    
    <p> VCFs produced by Ensembl VEP can be filtered using <a
    href="script/vep_filter.html">filter_vep.pl</a> in the same way as standard format
    output files. </p>
    
    <p> If the input format was VCF, the file will remain unchanged save for the
    addition of the CSQ field and the header (unless using any filtering). If an
    existing CSQ field is found, it will be replaced by the new annotation
    (use <a href="script/vep_options.html#opt_keep_csq">--keep_csq</a> to
    preserve it). </p>
    
    <p> Custom data added with <a
    href="script/vep_options.html#opt_custom">--custom</a> are added as separate
    fields, using the key specified for each data file.</p>
    
    <p> Commas in fields are replaced with ampersands (&amp;) to preserve VCF
    format. </p>
    
    <pre class="code sh_sh">##INFO=&lt;ID=CSQ,Number=.,Type=String,Description="Consequence annotations from Ensembl VEP. Format: Allele|Consequence|IMPACT|SYMBOL|Gene|Feature_type|Feature|BIOTYPE|EXON|INTRON|HGVSc|HGVSp|cDNA_position|CDS_position|Protein_position"&gt;
#CHROM  POS       ID          REF  ALT  QUAL  FILTER  INFO
21      26978790  rs75377686  T    C    .     .       CSQ=C|missense_variant|MODERATE|MRPL39|ENSG00000154719|Transcript|ENST00000419219|protein_coding|2/8||ENST00000419219.1:c.251A>G|ENSP00000404426.1:p.Asn84Ser|260|251|84</pre>
    
    
    <br />
    <hr />
    <h2 id="json"> JSON output</h2>
    
    <p> Ensembl VEP can output serialised <a
    href="http://json.org/" rel="external">JSON</a> objects using the <a
    href="script/vep_options.html#opt_json">--json</a> flag. JSON is a
    serialisation format that can be parsed and processed easily by many
    packages and programming languages; it is used as the default output format
    for <a href="http://rest.ensembl.org" rel="external">Ensembl's REST
    server</a>. </p>
    
    <p> Each input variant is reported as a single JSON object which constitutes
    one line of the output file. The JSON object is structured somewhat
    differently to the other  output formats, in that per-variant fields
    (e.g. co-located existing variant details) are reported only once.
    Consequences are grouped under the feature type that they affect
    (Transcript, Regulatory Feature, etc). The original input line (e.g. from
    VCF input) is reported under the "input" key in order to aid aligning input
    with output. When using a cache file, frequencies for co-located variants are reported by default (see
    <a href="./script/vep_options.html#opt_af_1kg">--af_1kg</a>,
    <a href="./script/vep_options.html#opt_af_gnomade">--af_gnomade</a>).
    </p>
    
    <p> Here follows an example of JSON output (prettified and redacted for display here): </p>
    
    <pre class="code sh_sh">{
  "input": "1 1918090 test1 A G . . .",
  "id": "test1",
  "seq_region_name": "1",
  "start": 1918090,
  "end": 1918090,
  "strand": 1,
  "allele_string": "A/G",
  "most_severe_consequence": "missense_variant",
  "colocated_variants": [
    {
      "id": "COSV57068665",
      "seq_region_name": "1",
      "start": 1918090,
      "end": 1918090,
      "strand": 1,
      "allele_string": "COSMIC_MUTATION"
    },
    {
      "id": "rs28640257",
      "seq_region_name": "1",
      "start": 1918090,
      "end": 1918090,
      "strand": 1,
      "allele_string": "A/G/T",
      "minor_allele": "G",
      "minor_allele_freq": 0.352,
      "frequencies": {
        "G": {
          "amr": 0.5072,
          "gnomade_sas": 0.3635,
          "gnomade": 0.481,
          "gnomade_remaining": 0.4536,
          "gnomade_asj": 0.3939,
          "gnomade_nfe": 0.5042,
          "gnomade_afr": 0.0975,
          "afr": 0.053,
          "gnomade_amr": 0.5568,
          "gnomade_fin": 0.4751,
          "sas": 0.3906,
          "gnomade_eas": 0.4516,
          "eur": 0.4901,
          "eas": 0.4623,
          "gnomade_mid: "0.3306"
        }
      }
    }
  ],
  "transcript_consequences": [
    {
      "variant_allele": "G",
      "consequence_terms": [
        "missense_variant"
      ],
      "gene_id": "ENSG00000178821",
      "transcript_id": "ENST00000310991",
      "strand": -1,
      "cdna_start": 436,
      "cdna_end": 436,
      "cds_start": 422,
      "cds_end": 422,
      "protein_start": 141,
      "protein_end": 141,
      "codons": "aTg/aCg",
      "amino_acids": "M/T",
      "polyphen_prediction": "benign",
      "polyphen_score": 0.001,
      "sift_prediction": "tolerated",
      "sift_score": 0.22,
      "hgvsp": "ENSP00000311122.3:p.Met141Thr",
      "hgvsc": "ENST00000310991.8:c.422T>C"
      }
  ],
  "regulatory_feature_consequences": [
    {
      "variant_allele": "G",
      "consequence_terms": [
        "regulatory_region_variant"
      ],
      "regulatory_feature_id": "ENSR00000000255"
    }
  ]
}</pre>

    <p> In accordance with JSON conventions, all keys (except alleles) are lower-case. Some keys
    also have different names and structures to those found in the other Ensembl VEP
    output formats: </p>
    
    <table class="ss">
      <tr>
        <th>Key</th>
        <th>JSON equivalent(s)</th>
        <th>Notes</th>
      </tr>
      <tr>
        <td>Consequence</td>
        <td>consequence_terms</td>
        <td></td>
      </tr>
      <tr>
        <td>Gene</td>
        <td>gene_id</td>
        <td></td>
      </tr>
      <tr>
        <td>Feature</td>
        <td>transcript_id, regulatory_feature_id, motif_feature_id</td>
        <td>Consequences are grouped under the feature type they affect</td>
      </tr>
      <tr>
        <td>ALLELE</td>
        <td>variant_allele</td>
        <td></td>
      </tr>
      <tr>
        <td>SYMBOL</td>
        <td>gene_symbol</td>
        <td></td>
      </tr>
      <tr>
        <td>SYMBOL_SOURCE</td>
        <td>gene_symbol_source</td>
        <td></td>
      </tr>
      <tr>
        <td>ENSP</td>
        <td>protein_id</td>
        <td></td>
      </tr>
      <tr>
        <td>OverlapBP</td>
        <td>bp_overlap</td>
        <td></td>
      </tr>
      <tr>
        <td>OverlapPC</td>
        <td>percentage_overlap</td>
        <td></td>
      </tr>
      <tr>
        <td>Uploaded_variation</td>
        <td>id</td>
        <td></td>
      </tr>
      <tr>
        <td>Location</td>
        <td>seq_region_name, start, end, strand</td>
        <td>The variant's location field is broken down into constituent coordinate parts for clarity. "seq_region_name" is used in place of "chr" or "chromosome" for consistency with other parts of Ensembl's REST API</td>
      </tr>
      <tr>
        <td>*_maf</td>
        <td>*_allele, *_maf</td>
        <td></td>
      </tr>
      <tr>
        <td>cDNA_position</td>
        <td>cdna_start, cdna_end</td>
        <td></td>
      </tr>
      <tr>
        <td>CDS_position</td>
        <td>cds_start, cds_end</td>
        <td></td>
      </tr>
      <tr>
        <td>Protein_position</td>
        <td>protein_start, protein_end</td>
        <td></td>
      </tr>
      <tr>
        <td>SIFT</td>
        <td>sift_prediction, sift_score</td>
        <td></td>
      </tr>
      <tr>
        <td>PolyPhen</td>
        <td>polyphen_prediction, polyphen_score</td>
        <td></td>
      </tr>
    </table>
   
    <br />
    <hr />
    <h2 id="stats"> Statistics</h2>
    
    <p> Ensembl VEP writes an HTML file containing statistics pertaining to the
    results of your job; it is named <b>[output_file]_summary.html</b> (with the
    default options the file will be named
    <b>variant_effect_output.txt_summary.html</b>). To view it, please open
    the file in your web browser. </p>
    
    <ul>
      <li> To prevent Ensembl VEP writing a stats file, use <a
      href="script/vep_options.html#opt_no_stats">--no_stats</a>.</li>

      <li> To get a machine-readable text file in place of the HTML file,
      use <a href="script/vep_options.html#opt_stats_text">--stats_text</a>.
      You can get both a HTML file and plain text file by using both
      <a href="script/vep_options.html#opt_stats_text">--stats_text</a>
      and <a href="script/vep_options.html#opt_stats_html">--stats_html</a>.
      </li>

      <li>To change the name of the stats file from the default, use <a
      href="script/vep_options.html#opt_stats_file">--stats_file [file]</a>. </li>
    </ul>
    <p> The page contains several sections: </p>
    
    <h4> General statistics </h4>
    
    <p> This section contains two tables. The first describes the cache and/or
    database used, the version of Ensembl VEP, species, command line parameters,
    input/output files and run time. The second table contains information about
    the number of variants, and the number of genes, transcripts and regulatory
    features overlapped by the input. </p>
    
    <h4> Charts and tables </h4>
    
    <p> There then follows several charts, most with accompanying tables. Tables
    and charts are interactive; clicking on a row to highlight it in the table
    will highlight the relevant segment in the chart, and vice versa. </p>
    
    <div class="tinted-box" style="width: 200px; padding-bottom: 5px; float: left; margin-right: 20px;">
        <a href="/img/vep_stats_1.png"><img style="width: 200px; border: 1px solid grey" src="/img/vep_stats_1.png">
        General statistics</a>
    </div>
    
    <div class="tinted-box" style="width: 200px; padding-bottom: 5px; float: left; margin-right: 20px;">
        <a href="/img/vep_stats_2.png"><img style="width: 200px; border: 1px solid grey" src="/img/vep_stats_2.png">
        Summary of called consequence types</a>
    </div>
    
    <div class="tinted-box" style="width: 200px; padding-bottom: 5px; float: left; margin-right: 20px;">
        <a href="/img/vep_stats_3.png"><img style="width: 200px; border: 1px solid grey" src="/img/vep_stats_3.png">
        Distribution of variants across chromosomes</a>
    </div>

<div style="clear:both"><h1 style="page-break-after: always">&nbsp</h1></div>    
<style>
tr:nth-child(odd) {background-color: #f0f0f0;}
td {
  line-height: 1.25em;
  word-wrap: break-word;
}
th { word-wrap: break-word; }
.info {width: inherit !important;}
pre {
  overflow-x: auto;
  white-space: pre-wrap;
  word-break: break-word;
}
</style>

<div>
  
  <div style="float:right"><img src="/img/vep_logo.png"/></div>
  
  <h1 id="top"><span style="color:#006;padding-right:15px">Ensembl Variant Effect Predictor</span><span style="color:#666"><img src="/i/16/setting.png"/> Running Ensembl VEP</span></h1>
  <hr/>
  
  <p>Ensembl VEP is run on the command line as follows (assuming you are in the ensembl-vep directory): </p>
  
  <div>
    <div style="float:left">
      <pre class="code sh_sh"> ./vep [options] </pre>
    </div>
    <div style="clear:both"></div>
  </div>
  
  <p> where [options] represent a set of flags and options.
  A basic set of flags can be listed using <a href="#opt_help">--help</a>: </p>

  <div>
    <div style="float:left">
      <pre class="code sh_sh"> ./vep --help </pre>
    </div>
    <div style="clear:both"></div>
  </div>
  
  Ensembl VEP can be run in the following modes:
  <ul>
  <li><p> For optimum performance, download a cache file for your species of interest, using
  either the <a href="vep_download.html#installer"><img style="vertical-align:bottom" src="/i/16/download.png"/> installer </a> or by
  following the <a href="vep_cache.html#pre">VEP Cache documentation</a>, and run Ensembl VEP with either
  the <a href="#opt_cache">--cache</a> or <a href="#opt_offline">--offline</a>
  option. </p></li>
  
  <li><p> By connecting to the public Ensembl database servers in place of a cache. 
      This can be adequate when annotating small files, but the database servers can become busy and slow.
      To enable this option, use <a href="#opt_database">--database</a>. </p></li>

  <li><p> To run Ensembl VEP using your own species and assembly, please use
      a <a href="#opt_fasta">--fasta</a> file and <a href="#opt_gff">--gff</a>
      or <a href="#opt_gtf">--gtf</a> annotation. </p></li>
  </ul>

  <p> To run Ensembl VEP with default options, use the following command: </p>
  
  <div>
      <div style="float:left">
      <pre class="code sh_sh"> ./vep --cache -i input.txt -o output.txt </pre>
    </div>
    <div style="clear:both"></div>
  </div>

  <p> where <kbd>input.txt</kbd> contains data in one of the compatible
  <a href="../vep_formats.html#input"> input formats</a>
  and <kbd>output.txt</kbd> is the <a href="../vep_formats.html#output">output file</a> to be created.
  </p>
  
  <p> Options can be passed as the full string (e.g. <a
  href="vep_options.html#opt_format">--format</a>), or as the shortest unique
  string among the options (e.g. <a
  href="vep_options.html#opt_form">--form</a> for <a
  href="vep_options.html#opt_format">--format</a>, since there is another
  option <a
  href="vep_options.html#opt_force_overwrite">--force_overwrite</a>). </p>

  <p> You may use one or two hypen ("-") characters before each option name; <b>--cache</b> or <b>-cache</b>.</p>

  <p> Ensembl VEP options can also be read from: </p>
  <ul>
    <li>
      <b>Configuration files</b> using <a href="#opt_config">--config</a>.
      Options set in configuration files are overriden if specified on the
      command line.
    </li>
    <li>
      <b>Environment variables</b> that start with prefix <kbd>VEP_</kbd>. For
      instance, you can set the cache flag with <kbd>export VEP_CACHE=1</kbd>
      and the input flag with <kbd>export VEP_INPUT="/path/to/input.txt"</kbd>
      before running <kbd>./vep</kbd>. Options set in environment variables are
      overriden if specified in configuration files or on the command line.
    </li>
  </ul>

  <div id="nv_species"> 
    <div class="info" style="float:left">
      <h3>Running Ensembl VEP on non-vertebrate species</h3>
      <div class="message-pad">
        <p>To use Ensembl VEP on non-vertebrate species, you need to use extra options:</p>
        <ul>
          <li> <p> <b>Database</b> access: add the option <a href="#opt_genomes">--genomes</a> to point to the correct database server. </p>

          <pre class="code sh_sh"> ./vep -i input.txt -o output.txt --species triticum_aestivum --database --genomes </pre>

          <p>Some of the non-vetebrate species databases (mainly bacteria and protists) are composed of a collection of species.
          For these databases the flag "<a href="#opt_is_multispecies">--is_multispecies</a> 1" is required.</p>

          <pre class="code sh_sh"> ./vep -i input.txt -o output.txt --species protists_euglenozoa1 --database --genomes --is_multispecies 1 </pre>
          </li>

        <li> <p> <b>Cache</b> access: use the option "<a href="#opt_genomes">--cache_version</a> <i>eg_version</i>" where <i>eg_version</i> is the 
        <a href="http://ensemblgenomes.org/">Ensembl genomes</a> version number, which differs from the Ensembl vertebrates/VEP version numbers
        ([[SPECIESDEFS::ENSEMBL_VERSION]]).</p>

        <pre class="code sh_sh"> ./vep -i input.txt -o output.txt --species triticum_aestivum --cache --cache_version 42</pre>

        <p>More information about the Ensembl VEP cache is available <a href="vep_cache.html#cache">here</a>.</p>
        </li>
        </ul>
      </div>
    </div>
    <div style="clear:both"></div>
  </div>
  

  <br/> 
  <hr/>
  <h2 id="basic">Basic options</h2>
  
  <table class="ss" style="table-layout: fixed">
    <tr>
      <th width="20%">Flag</th>
      <th width="8%">Alternate</th>
      <th>Description</th>
      <th width="10%">Incompatible with</th>
    </tr>
    <tr id="opt_help">
      <td><pre>--help</pre></td>
      <td>&nbsp;</td>
      <td>Display help message and quit</td>
      <td>&nbsp;</td>
    </tr>
    <tr id="opt_quiet">
      <td><pre>--quiet</pre></td>
      <td><pre>-q</pre></td>
      <td>Suppress warning messages.<i>Not used by
      default</i></td>
      <td><a href="#opt_verbose">--verbose</a></td> 
    </tr>
    <tr id="opt_verbose">
      <td><pre>--verbose</pre></td>
      <td><pre>-v</pre></td>
      <td>Print out a bit more information while running. <i>Not used by
      default</i></td>
      <td><a href="#opt_quiet">--quiet</a></td>
    </tr>    
    <tr id="opt_config">
      <td><pre>--config [filename]</pre></td>
      <td>&nbsp;</td>
      <td>
        <p>Load configuration options from a config file. The config file
        should consist of whitespace-separated pairs of option names and
        settings e.g.:</p>
  <pre class="code sh_sh">output_file   my_output.txt
species       mus_musculus
format        vcf
host          useastdb.ensembl.org</pre>
        A config file can also be implicitly read; save the file as
        <b>$HOME/.vep/vep.ini</b> (or equivalent directory if using
        <a href="vep_options.html#opt_dir">--dir</a>). Any options in this file will be overridden by
        those specified in a config file using <a href="vep_options.html#opt_config">--config</a>, and in
        turn by any options specified on the command line. You
        can create a quick version file of this by setting the flags as
        normal and running Ensembl VEP in verbose (<b>-v</b>) mode. This
        will output lines that can be copied to a config file that can
        be loaded in on the next run using <a href="vep_options.html#opt_config">--config</a>. <i>Not used
        by default</i>
      </td>
      <td>&nbsp;</td>
    </tr>
    <tr id="opt_everything">
      <td><pre>--everything</pre></td>
      <td><pre>-e</pre></td>
      <td>Shortcut flag to switch on all of the following:
        <p>
          <a href="#opt_sift">--sift b</a>,
          <a href="#opt_polyphen">--polyphen b</a>,
          <a href="#opt_ccds">--ccds</a>,
          <a href="#opt_hgvs">--hgvs</a>,
          <a href="#opt_symbol">--symbol</a>,
          <a href="#opt_numbers">--numbers</a>,
          <a href="#opt_domains">--domains</a>,
          <a href="#opt_regulatory">--regulatory</a>,
          <a href="#opt_canonical">--canonical</a>,
          <a href="#opt_protein">--protein</a>,
          <a href="#opt_biotype">--biotype</a>,
          <a href="#opt_af">--af</a>,
          <a href="#opt_af_1kg">--af_1kg</a>,
          <a href="#opt_af_esp">--af_esp</a>,
          <a href="#opt_af_gnomade">--af_gnomade</a>,
          <a href="#opt_af_gnomadg">--af_gnomadg</a>,
          <a href="#opt_max_af">--max_af</a>,
          <a href="#opt_pubmed">--pubmed</a>,
          <a href="#opt_uniprot">--uniprot</a>,
          <a href="#opt_mane">--mane</a>,
          <a href="#opt_tsl">--tsl</a>,
          <a href="#opt_appris">--appris</a>,
          <a href="#opt_variant_class">--variant_class</a>,
          <a href="#opt_gene_phenotype">--gene_phenotype</a>,
          <a href="#opt_mirna">--mirna</a>
        </ul>
      </td>
      <td>&nbsp;</td>
    </tr>
    <tr id="opt_species">
      <td><pre>--species [species]</pre></td>
      <td>&nbsp;</td>
      <td>
        Species for your data. This can be the latin name e.g.
        "homo_sapiens" or any Ensembl alias e.g. "mouse". Specifying the
        latin name can speed up initial database connection as the
        registry does not have to load all available database aliases on
        the server. <i>Default = "homo_sapiens"</i>
      </td>
      <td>&nbsp;</td>
    </tr>
    <tr id="opt_assembly">
      <td><pre>--assembly [name]</pre></td>
      <td><pre>-a</pre></td>
      <td>
        Select the assembly version to use if more than one available. If using
        the cache, you must have the appropriate assembly's cache file
        installed. If not specified and you have only 1 assembly version
        installed, this will be chosen by default. <i>Default = use found
        assembly version</i>
      </td>
      <td>&nbsp;</td>
    </tr>
    <tr id="opt_input_file">
      <td><pre>--input_file [filename]</pre></td>
      <td><pre>-i</pre></td>
      <td>Input file name. If not specified, Ensembl VEP will attempt to read from STDIN. Can use compressed file (gzipped).</td>
      <td>&nbsp;</td>
    </tr>
    <tr id="opt_input_data">
      <td><pre>--input_data [string]</pre></td>
      <td><pre>--id</pre></td>
      <td><p>Raw input data as a string. May be used, for example, to input a single rsID
        or HGVS notation quickly to Ensembl VEP:</p>
        <pre class="code sh_sh">--input_data rs699</pre>
      </td>
      <td>&nbsp;</td>
    </tr>
    <tr id="opt_format">
      <td><pre>--format [format]</pre></td>
      <td>&nbsp;</td>
      <td>
        <p>
        <a href="../vep_formats.html#input">Input file format</a> - one of
        "ensembl", "vcf", "hgvs", "id", "region", "spdi".<br />
        By default, Ensembl VEP auto-detects the input file format. Using this option you
        can specify the input file is Ensembl, VCF, IDs, HGVS, SPDI or region format.
        Can use compressed version (gzipped) of any file format listed above.
        <i>Auto-detects format by default</i>
        </p>
      </td>
      <td>&nbsp;</td>
    </tr>
    <tr id="opt_output_file">
      <td><pre>--output_file [filename]</pre></td>
      <td><pre>-o</pre></td>
      <td>
        Output file name. Results can write to STDOUT by specifying
        'STDOUT' as the output file name - this will force quiet mode.
        <i>Default = "variant_effect_output.txt"</i>
      </td>
      <td>&nbsp;</td>
    </tr>
    <tr id="opt_force_overwrite">
      <td><pre>--force_overwrite</pre></td>
      <td><pre>--force</pre></td>
      <td>
        By default, Ensembl VEP will fail with an error if the output
        file already exists. You can force the overwrite of the existing
        file by using this flag. <i>Not used by default</i>
      </td>
      <td>&nbsp;</td>
    </tr>
    <tr id="opt_no_stats">
      <td><pre>--no_stats</pre></td>
      <td>&nbsp;</td>
      <td>
        Don't generate a stats file. Provides marginal gains in run time.
      </td>
      <td>&nbsp;</td>
    </tr>
    <tr id="opt_stats_file">
      <td><pre>--stats_file [filename]</pre></td>
      <td><pre>--sf</pre></td>
      <td>
        <a href="../vep_formats.html#stats">Summary stats file</a> name. This
        file contains a summary of the Ensembl VEP run. If stats are returned in an HTML
        file (default), the filename should end in <kbd>.html</kbd> or
        <kbd>.htm</kbd>. <i>Default = "variant_effect_output.txt_summary.html"</i>
      </td>
      <td>&nbsp;</td>
    </tr>
    <tr id="opt_stats_html">
      <td><pre>--stats_html</pre></td>
      <td>&nbsp;</td>
      <td>
        Generate a <a href="../vep_formats.html#stats">HTML stats file</a>
        (default).
      </td>
      <td>&nbsp;</td>
    </tr>
    <tr id="opt_stats_text">
      <td><pre>--stats_text</pre></td>
      <td>&nbsp;</td>
      <td>
        Generate a plain text stats file. Can be combined with
        <kbd><a href="#opt_stats_html">--stats_html</a></kbd> to generate
        both plain text and HTML stats files.
      </td>
      <td>&nbsp;</td>
    </tr>
    <tr id="opt_warning_file">
      <td><pre>--warning_file [filename]</pre></td>
      <td>&nbsp;</td>
      <td>
        File name to write warnings and errors to.
        <i>Default = STDERR (standard error)</i>
      </td>
      <td>&nbsp;</td>
    </tr>
    <tr id="opt_skipped_variants_file">
      <td><pre>--skipped_variants_file [filename]</pre></td>
      <td>&nbsp;</td>
      <td>
        File name to write skipped variants to.
        <i>Default = STDERR (standard error)</i>
      </td>
      <td>&nbsp;</td>
    </tr>
    <tr id="opt_max_sv_size">
      <td><pre>--max_sv_size</pre></td>
      <td>&nbsp;</td>
      <td>
        Extend the maximum Structural Variant size Ensembl VEP can process. <i>Default =
          10000000</i>
      </td>
      <td>&nbsp;</td>
    </tr>
    <tr id="opt_no_check_variants_order">
      <td><pre>--no_check_variants_order</pre></td>
      <td>&nbsp;</td>
      <td>
        Permit the use of unsorted input files. However running Ensembl VEP on unsorted input files slows down the tool and requires more memory.
      </td>
      <td>&nbsp;</td>
    </tr>
    <tr id="opt_fork">
      <td><pre>--fork [num_forks]</pre></td>
      <td>&nbsp;</td>
      <td>
        Enable <a href="vep_other.html#faster">forking</a>, using the specified
        number of forks. Forking can dramatically improve runtime. <i>Not used by default</i>
      </td>
      <td>&nbsp;</td>
    </tr>
    <tr id="opt_safe">
      <td><pre>--safe</pre></td>
      <td>&nbsp;</td>
      <td>
        By default, an Ensembl VEP run is successful even when a plugin reports issues.
        Use this flag to ensure Ensembl VEP fails if a plugin raises warnings or generates compilation errors.
        This is particularly useful to ensure plugins run successfully when using Ensembl VEP in pipelines.
        <i>Not used by default</i>
      </td>
      <td>&nbsp;</td>
    </tr>
  </table>
  
  
  

  <br/>  
  <hr/>
    
  <h2 id="cacheopt">Cache options</h2>
  
  <table class="ss" style="table-layout: fixed">
    <tr>
      <th width="20%">Flag</th>
      <th width="8%">Alternate</th>
      <th>Description</th>
      <th width="12%">Output fields</th>
      <th width="10%">Incompatible with</th>
    </tr>
    <tr id="opt_cache">
      <td><pre>--cache</pre></td>
      <td>&nbsp;</td>
      <td>
        Enables use of the <a href="vep_cache.html#cache">cache</a>. Add <a
        href="#opt_refseq">--refseq</a> or <a href="#opt_merged">--merged</a> to use the refseq or merged cache, (if installed).
      </td>
      <td>&nbsp;</td>
      <td>
        <div style="padding:2px 0px"><a href="#opt_database">--database</a></div>
      </td>
    </tr>
    <tr id="opt_dir">
      <td><pre>--dir [directory]</pre></td>
      <td>&nbsp;</td>
      <td>
        Specify the base cache/plugin directory to use. <i>Default =
        "$HOME/.vep/"</i>
      </td>
      <td >&nbsp;</td>
      <td >&nbsp;</td>
    </tr>
    <tr id="opt_dir_cache">
      <td><pre>--dir_cache [directory]</pre></td>
      <td>&nbsp;</td>
      <td>
        Specify the cache directory to use. <i>Default =
        "$HOME/.vep/"</i>
      </td>
      <td>&nbsp;</td>
      <td>&nbsp;</td>
    </tr>
    <tr id="opt_dir_plugins">
      <td><pre>--dir_plugins [directory]</pre></td>
      <td>&nbsp;</td>
      <td>
        Specify the plugin directory to use. <i>Default =
        "$HOME/.vep/"</i>
      </td>
      <td >&nbsp;</td>
      <td >&nbsp;</td>
    </tr>
    <tr id="opt_offline">
      <td><pre>--offline</pre></td>
      <td>&nbsp;</td>
      <td>
        Enable <a href="vep_cache.html#offline">offline mode</a>. No database
        connections will be made, and a cache file or <a href="#opt_gff">GFF</a>/<a href="#opt_gtf">GTF</a> file is required for annotation. 
        Add <a href="#opt_refseq">--refseq</a> to use the refseq cache (if
        installed). <i>Not used by default</i>
      </td>
      <td>&nbsp;</td>
      <td>
        <div style="padding:2px 0px"><a href="#opt_database">--database</a></div>
        <div style="padding:2px 0px"><a href="#opt_check_svs">--check_svs</a></div>
        <div style="padding:2px 0px"><a href="#opt_lrg">--lrg</a></div>
      </td>
    </tr>
    <tr id="opt_fasta">
      <td><pre>--fasta [file|dir]</pre></td>
      <td><pre>--fa</pre></td>
      <td>
        Specify a FASTA file or a directory containing FASTA files to
        use to look up reference sequence. The first time you run Ensembl VEP
        with this parameter an index will be built which can take
        a few minutes. This is required if fetching HGVS annotations (<a
        href="#opt_hgvs">--hgvs</a>) or checking reference sequences (<a
        href="#opt_check_ref">--check_ref</a>) in offline mode (<a
        href="#opt_offline">--offline</a>), and optional with some
        performance increase in cache mode (<a
        href="#opt_cache">--cache</a>). See <a
        href="vep_cache.html#fasta">documentation</a> for more details.
        <i>Not used by default</i>
      </td>
      <td >&nbsp;</td>
      <td >&nbsp;</td>
    </tr>
    <tr id="opt_refseq">
      <td><pre>--refseq</pre></td>
      <td>&nbsp;</td>
      <td>
        <p> Specify this option if you have installed the RefSeq
        cache in order for Ensembl VEP to pick up the alternate cache directory. This cache contains
        transcript objects corresponding to RefSeq transcripts.
        Consequence output will be given relative to these transcripts in place
        of the default Ensembl transcripts (see <a
        href="vep_other.html#refseq">documentation</a>)</p>
        </p>
      </td>
      <td>REFSEQ_MATCH, BAM_EDIT</td>
      <td>
        <div style="padding:2px 0px"><a href="#opt_gencode_basic">--gencode_basic</a></div>
        <div style="padding:2px 0px"><a href="#opt_gencode_primary">--gencode_primary</a></div>
        <div style="padding:2px 0px"><a href="#opt_merged">--merged</a></div>
      </td>
    </tr>
    <tr id="opt_merged">
      <td><pre>--merged</pre></td>
      <td>&nbsp;</td>
      <td>
        <p> Use the merged Ensembl and RefSeq cache. Consequences are flagged
        with the SOURCE of each transcript used. <p>
      </td>
      <td>REFSEQ_MATCH, BAM_EDIT, SOURCE</td>
      <td>
        <div style="padding:2px 0px"><a href="#opt_refseq">--refseq</a></div>
      </td>
    </tr>
    <tr id="opt_cache_version">
      <td><pre>--cache_version</pre></td>
      <td>&nbsp;</td>
      <td>
        Use a different cache version than the assumed default (the Ensembl VEP
        version). This should be used with Ensembl Genomes caches since their
        version numbers do not match Ensembl versions. For example, the
        VEP/Ensembl version may be 88 and the Ensembl Genomes version 35. <i>Not
        used by default</i>
      </td>
      <td >&nbsp;</td>
      <td >&nbsp;</td>
    </tr>
    <tr id="opt_show_cache_info">
      <td><pre>--show_cache_info</pre></td>
      <td>&nbsp;</td>
      <td>
        Show source version information for selected cache and quit
      </td>
      <td >&nbsp;</td>
      <td >&nbsp;</td>
    </tr>
    <tr id="opt_buffer_size">
      <td><pre>--buffer_size [number]</pre></td>
      <td>&nbsp;</td>
      <td>
        Sets the internal buffer size, corresponding to the number of
        variants that are read in to memory simultaneously. Set this
        lower to use less memory at the expense of longer run time, and
        higher to use more memory with a faster run time. <i>Default =
        5000</i>
      </td>
      <td >&nbsp;</td>
      <td >&nbsp;</td>
    </tr>
  </table>


  <br/>
  <hr/>  
  <h2 id="other">Other annotation sources</h2>
  
  <table class="ss" style="table-layout: fixed">
    <tr>
      <th width="20%">Flag</th>
      <th width="10%">Alternate</th>
      <th>Description</th>
      <th width="15%">Output fields</th>
    </tr>
    <tr id="opt_plugin">
      <td><pre>--plugin [plugin name]</pre></td>
      <td>&nbsp;</td>
      <td>
        Use named plugin. Plugin modules should be installed in the
        Plugins subdirectory of the Ensembl VEP cache directory (defaults to
        $HOME/.vep/). Multiple plugins can be used by supplying the
        <a href="vep_options.html#opt_plugin">--plugin</a> flag multiple times. See <a
        href="vep_plugins.html">plugin documentation</a>. <i>Not used by
        default</i>
      </td>
      <td>Plugin-dependent</td>
    </tr>
    <tr id="opt_custom">
      <td><pre>--custom file=[filename]</pre></td>
      <td>&nbsp;</td>
      <td>
        Add custom annotation to the output. Files must be tabix indexed
        or in the bigWig format. Multiple files can be specified by
        supplying the <a href="vep_options.html#opt_custom">--custom</a> flag multiple times. <a
        href="vep_custom.html">See here</a> for full details. <i>Not
        used by default</i>
      </td>
      <td>SOURCE, Custom file dependent</td>
    </tr>
    <tr id="opt_gff">
      <td><pre>--gff [filename]</pre></td>
      <td>&nbsp;</td>
      <td>
        Use <a href="vep_cache.html#gff">GFF transcript annotations</a> in [filename] as an annotation source. Requires a <a href="#opt_fasta">FASTA file</a> of genomic sequence. <i>Not
        used by default</i>
      </td>
      <td>SOURCE</td>
    </tr>
    <tr id="opt_gtf">
      <td><pre>--gtf [filename]</pre></td>
      <td>&nbsp;</td>
      <td>
        Use <a href="vep_cache.html#gff">GTF transcript annotations</a> in [filename] as an annotation source. Requires a <a href="#opt_fasta">FASTA file</a> of genomic sequence. <i>Not
        used by default</i>
      </td>
      <td>SOURCE</td>
    </tr>
    <tr id="opt_bam">
      <td><pre>--bam [filename]</pre></td>
      <td>&nbsp;</td>
      <td>
        <b>ADVANCED</b> Use BAM file of sequence alignments to correct transcript models not derived from reference genome sequence. Used to correct <a href="vep_other.html#refseq_bam">RefSeq transcript models</a>. Enables <a href="#opt_use_transcript_ref">--use_transcript_ref</a>; add <a href="#opt_use_given_ref">--use_given_ref</a> to override this behaviour.
        <i>Not used by default</i>
      </td>
      <td>BAM_EDIT</td>
    </tr>
    <tr id="opt_use_transcript_ref">
      <td><pre>--use_transcript_ref</pre></td>
      <td>&nbsp;</td>
      <td>
        By default Ensembl VEP uses the reference allele provided in the input file to calculate consequences for the provided alternate allele(s). Use this flag to force Ensembl VEP to replace the provided reference allele with sequence derived from the overlapped transcript. This is especially relevant when using the RefSeq cache, see <a href="vep_other.html#refseq_bam">documentation</a> for more details. The <a href="../vep_formats.html#output">GIVEN_REF and USED_REF fields</a> are set in the output to indicate any change.
        <i>Not used by default</i>
      </td>
      <td>GIVEN_REF, USED_REF</td>
    </tr>
    <tr id="opt_use_given_ref">
      <td><pre>--use_given_ref</pre></td>
      <td>&nbsp;</td>
      <td>
        Using <a href="#opt_bam">--bam</a> or a <a href="vep_other.html#refseq_bam">BAM-edited RefSeq cache</a> by default enables <a href="#opt_use_transcript_ref">--use_transcript_ref</a>; add this flag to override this behaviour and use the provided reference allele from the input.
        <i>Not used by default</i>
      </td>
      <td>&nbsp;</td>
    </tr>
    <tr id="opt_custom_multi_allelic">
      <td><pre>--custom_multi_allelic</pre></td>
      <td>&nbsp;</td>
      <td>
        By default, comma separated lists found within the INFO field of custom annotation VCFs are assumed to be allele specific. For example, a variant with allele_string A/G/C with associated custom annotation 'single,double,triple' will associate triple with C, double with G and single with A. This flag instructs Ensembl VEP to return all annotations for all alleles.
        <i>Not used by default</i>
      </td>
      <td>&nbsp;</td>
    </tr>
  </table>
    
 
  <br/>
  <hr/>
  <h2 id="format">Output format options</h2>

  <table class="ss" style="table-layout: fixed">
    <tr>
      <th width="20%">Flag</th>
      <th width="10%">Alternate</th>
      <th>Description</th>
      <th width="10%">Output fields</th>
      <th width="10%">Incompatible with</th>
    </tr>
    <tr id="opt_vcf">
      <td><pre>--vcf</pre></td>
      <td>&nbsp;</td>
      <td>
        <p> Writes output in <a href="../vep_formats.html#vcfout">VCF format</a>.
        Consequences are added in the INFO field of the VCF file, using the key
        "CSQ". Data fields are encoded separated by "&#124;"; the order of
        fields is written in the VCF header. Output fields in the "CSQ" INFO field can be selected by
        using <a href="#opt_fields">--fields</a>.</p>
        <p> If the input format
        was VCF, the file will remain unchanged save for the addition of the CSQ
        field (unless using any filtering).<br/><br/> Custom data added with <a
        href="#opt_custom">--custom</a> are added as separate fields, using the
        key specified for each data file.<br/><br/> Commas in fields are
        replaced with ampersands (&amp;) to preserve VCF format.</p>
        <i>Not used by default</i>
      </td>
      <td>&nbsp;</td>
      <td>
        <div style="padding:2px 0px"><a href="#opt_json">--json</a></div>
        <div style="padding:2px 0px"><a href="#opt_tab">--tab</a></div>
        <div style="padding:2px 0px"><a href="#opt_summary">--summary</a></div>
        <div style="padding:2px 0px"><a href="#opt_most_severe">--most_severe</a></div>
        <div style="padding:2px 0px"><a href="#opt_ga4gh_vrs">--ga4gh_vrs</a></div>
      </td>
    </tr>
    <tr id="opt_tab">
      <td><pre>--tab</pre></td>
      <td>&nbsp;</td>
      <td>
        Writes output in <a href="../vep_formats.html#tab">tab-delimited format</a>.
        <i>Not used by default</i>
      </td>
      <td>&nbsp;</td>
      <td>
        <div style="padding:2px 0px"><a href="#opt_json">--json</a></div>
        <div style="padding:2px 0px"><a href="#opt_vcf">--vcf</a></div>
      </td>
    </tr>
    <tr id="opt_json">
      <td><pre>--json</pre></td>
      <td>&nbsp;</td>
      <td>
        Writes output in <a href="../vep_formats.html#json">JSON format</a>.
        <i>Not used by default</i>
      </td>
      <td>&nbsp;</td>
      <td>
        <div style="padding:2px 0px"><a href="#opt_tab">--tab</a></div>
        <div style="padding:2px 0px"><a href="#opt_vcf">--vcf</a></div>
      </td>
    </tr>  
    <tr id="opt_compress_output">
      <td><pre>--compress_output [gzip|bgzip]</pre></td>
      <td>&nbsp;</td>
      <td>
        Writes output compressed using either gzip or bgzip.
        <i>Not used by default</i>
      </td>
      <td >&nbsp;</td>
      <td >&nbsp;</td>
    </tr>
    <tr id="opt_fields">
      <td><pre>--fields [list]</pre></td>
      <td>&nbsp;</td>
      <td>
        <p>Configure the output format using a comma separated list of
        fields.<br />
        Can only be used with <a href="../vep_formats.html#tab">tab</a> (<a href="#opt_tab">--tab</a>) or <a href="../vep_formats.html#vcfout">VCF format</a> (<a href="#opt_vcf">--vcf</a>) output.<br />
        For the tab format output, the selected fields may be those present in the default <a href="../vep_formats.html#output">output columns</a>, or any of
        those that appear in the Extra column (including those added by plugins or custom annotations) if the appropriate output is available (e.g. use <a href="#opt_show_ref_allele">--show_ref_allele</a> to access 'REF_ALLELE'). Output remains tab-delimited.<br />
        For the VCF format output, the selected fields are those present within the "CSQ" INFO field.
        </p>
        <p>Example of command for the tab output:</p>
        <pre class="code sh_sh">--tab --fields "Uploaded_variation,Location,Allele,Gene"</pre>
        <p>Example of command for the VCF format output:</p>
        <pre class="code sh_sh">--vcf --fields "Allele,Consequence,Feature_type,Feature"</pre>
        <i>Not used by default</i>
      </td>
      <td >&nbsp;</td>
      <td >&nbsp;</td>
    </tr>
    <tr id="opt_minimal">
      <td><pre>--minimal</pre></td>
      <td>&nbsp;</td>
      <td>
        Convert alleles to their most minimal representation before
        consequence calculation i.e. sequence that is identical between
        each pair of reference and alternate alleles is trimmed off from
        both ends, with coordinates adjusted accordingly.<br />
        Note this may
        lead to discrepancies between input coordinates and coordinates
        reported by Ensembl VEP relative to transcript sequences; to avoid
        issues, use <a href="#opt_allele_number">--allele_number</a>
        and/or ensure that your input variants have unique identifiers.
        The MINIMISED flag is set in the Ensembl VEP output where relevant.
        For an insertion/deletion, the allele is minimised by default. 
        To access the input allele before minimisation, use 
        <a href="#opt_uploaded_allele">--uploaded_allele</a>.<br/> 
        <i>Not used by default</i>
      </td>
      <td>MINIMISED</td>
      <td>
        <div style="padding:2px 0px"><a href="#opt_individual">--individual</a></div>
      </td> 
    </tr>
  </table>


 
  <br/> 
  <hr/>
  <h2 id="output">Output options</h2>
  
  <table class="ss" style="table-layout: fixed">
    <tr>
      <th width="20%">Flag</th>
      <th width="8%">Alternate</th>
      <th>Description</th>
      <th width="10%">Output fields</th>
      <th width="10%">Incompatible with</th>
    </tr>
    <tr id="opt_variant_class">
      <td><pre>--variant_class</pre></td>
      <td>&nbsp;</td>
      <td>
        Output the Sequence Ontology <a
        href="/info/genome/variation/prediction/classification.html#classes">variant
        class</a>. <i>Not used by default</i>
      </td>
      <td>VARIANT_CLASS</td>
      <td>&nbsp;</td>
    </tr>
    <tr id="opt_sift">
      <td><pre>--sift [p<b>|</b>s<b>|</b>b]</pre></td>
      <td>&nbsp;</td>
      <td>
        <span style="color:red;">Species limited</span>
        <a rel="external" href="http://sift.bii.a-star.edu.sg/" target="_blank">SIFT</a>
        predicts whether an amino acid substitution affects protein
        function based on sequence homology and the physical properties
        of amino acids. Ensembl VEP can output the <b>p</b>rediction term,
        <b>s</b>core or <b>b</b>oth. <i>Not used by default</i>
      </td>
      <td>SIFT</td>
      <td>
        <div style="padding:2px 0px"><a href="#opt_most_severe">--most_severe</a></div>
        <div style="padding:2px 0px"><a href="#opt_summary">--summary</a></div>
      </td>
    </tr>
    <tr id="opt_polyphen">
      <td><pre>--polyphen [p<b>|</b>s<b>|</b>b]</pre></td>
      <td>&nbsp;</td>
      <td>
        <span style="color:red;">Human only</span>
        <a rel="external" href="http://genetics.bwh.harvard.edu/pph2/"
        target="_blank">PolyPhen</a> is a tool which predicts possible
        impact of an amino acid substitution on the structure and
        function of a human protein using straightforward physical and
        comparative considerations. Ensembl VEP can output the
        <b>p</b>rediction term, <b>s</b>core or <b>b</b>oth. Ensembl VEP uses the
        humVar score by default - use <a href="#opt_humdiv">--humdiv</a> to
        retrieve the humDiv score. <i>Not used by default</i>
      </td>
      <td>PolyPhen</td>
      <td>
        <div style="padding:2px 0px"><a href="#opt_most_severe">--most_severe</a></div>
        <div style="padding:2px 0px"><a href="#opt_summary">--summary</a></div>
      </td>
    </tr>
    <tr id="opt_humdiv">
      <td><pre>--humdiv</pre></td>
      <td>&nbsp;</td>
      <td>
        <span style="color:red;">Human only</span>
        Retrieve the <a
        href="http://genetics.bwh.harvard.edu/pph2/dokuwiki/overview#prediction"
        rel="external">humDiv PolyPhen prediction</a> instead of the default
        humVar. <i>Not used by default</i>
      </td>
      <td>PolyPhen</td>
      <td>&nbsp;</td>
    </tr>
    <tr id="opt_nearest">
      <td><pre>--nearest [transcript|gene|symbol]</pre></td>
      <td>&nbsp;</td>
      <td>
        <p> Retrieve the transcript or gene with the nearest protein-coding transcription start site (TSS)
        to each input variant. Use "transcript" to retrieve the transcript
        stable ID, "gene" to retrieve the gene stable ID, or "symbol" to retrieve
        the gene symbol. Note that the nearest TSS may not belong to a transcript
        that overlaps the input variant, and more than one may be reported in the
        case where two are equidistant from the input coordinates.</p>

        <p> Currently only available when using a
        <a href="vep_cache.html#cache">cache</a> annotation source, and <a href="vep_download.html#additional">requires
        the Set::IntervalTree perl module</a>.</p>
        <i>Not used by default</i>
      </td>
      <td>NEAREST</td>
      <td>&nbsp;</td>
    </tr>
    <tr id="opt_distance">
      <td><pre>--distance [bp_distance(,downstream_distance)]</pre></td>
      <td>&nbsp;</td>
      <td>
        Modify the distance up and/or downstream between a variant and a transcript
        for which Ensembl VEP will assign the upstream_gene_variant or downstream_gene_variant
        consequences. Giving one distance will modify both up- and downstream distances;
        prodiving two separated by commas will set the up- (5') and down- (3') stream
        distances respectively.
        <i>Default: 5000</i>
      </td>
      <td >&nbsp;</td>
      <td >&nbsp;</td>
    </tr>
    <tr id="opt_overlaps">
      <td><pre>--overlaps</pre></td>
      <td>&nbsp;</td>
      <td>
        Report the proportion and length of a transcript overlapped by a structural variant in VCF format.
      </td>
      <td >&nbsp;</td>
      <td >&nbsp;</td>
    </tr>
    <tr id="opt_gene_phenotype">
      <td><pre>--gene_phenotype</pre></td>
      <td>&nbsp;</td>
      <td>
        Indicates if the overlapped gene is associated with a phenotype, disease
        or trait. See <a
        href="/info/genome/variation/phenotype/sources_phenotype_documentation.html">list
        of phenotype sources</a>. <i>Not used by default</i>
      </td>
      <td>GENE_PHENO</td>
      <td>&nbsp;</td>
    </tr>
    <tr id="opt_regulatory">
      <td><pre>--regulatory</pre></td>
      <td>&nbsp;</td>
      <td>
        Look for overlaps with regulatory regions. Ensembl VEP can also
        report if a variant falls in a high information position within a
        transcription factor binding site. Output lines have a Feature
        type of RegulatoryFeature or MotifFeature. <i>Not used by
        default</i>
      </td>
      <td>MOTIF_NAME, MOTIF_POS, HIGH_INF_POS, MOTIF_SCORE_CHANGE</td>
      <td>&nbsp;</td>
    </tr>
    <tr id="opt_cell_type">
      <td><pre>--cell_type</pre></td>
      <td>&nbsp;</td>
      <td>
        Report only regulatory regions that are found in the given cell
        type(s). Can be a single cell type or a comma-separated list.
        The functional type in each cell type is reported under
        CELL_TYPE in the output. To retrieve a list of cell types, use
        <a href="vep_options.html#opt_cell_type">--cell_type list</a>. <i>Not used by default</i>
      </td>
      <td>CELL_TYPE</td>
      <td>&nbsp;</td>
    </tr>
    <tr id="opt_individual">
      <td><pre>--individual [all|ind list]</pre></td>
      <td>&nbsp;</td>
      <td>
        Consider only alternate alleles present in the genotypes of the
        specified individual(s). May be a single individual, a
        comma-separated list or "all" to assess all individuals
        separately. Individual variant combinations homozygous for the
        given reference allele will not be reported. Each individual and
        variant combination is given on a separate line of output. Only
        works with VCF files containing individual genotype data;
        individual IDs are taken from column headers. <i>Not used by
        default</i>
      </td>
      <td>IND, ZYG</td>
      <td>
        <div style="padding:2px 0px"><a href="#opt_minimal">--minimal</a></div>
        <div style="padding:2px 0px"><a href="#opt_individual_zyg">--individual_zyg</a></div>
      </td>
    </tr>
    <tr id="opt_individual_zyg">
      <td><pre>--individual_zyg [all|ind list]</pre></td>
      <td>&nbsp;</td>
      <td>
        Consider alternate and reference alleles present in the genotypes of the
        specified individual(s). May be a single individual, a
        comma-separated list or "all" to assess all individuals
        separately. Returns a list of individuals and their zygosity.
        Only works with VCF files containing individual genotype data;
        individual IDs are taken from column headers. <i>Not used by
        default</i>
      </td>
      <td>ZYG</td>
      <td>
        <div style="padding:2px 0px"><a href="#opt_individual">--individual</a></div>
      </td>
    </tr>
    <tr id="opt_phased">
      <td><pre>--phased</pre></td>
      <td>&nbsp;</td>
      <td>
        Force VCF genotypes to be interpreted as phased. For use with
        plugins that depend on phased data. <i>Not used by
        default</i>
      </td>
      <td >&nbsp;</td>
      <td >&nbsp;</td>
    </tr>
    <tr id="opt_allele_number">
      <td><pre>--allele_number</pre></td>
      <td>&nbsp;</td>
      <td>
        Identify allele number from VCF input, where 1 = first ALT
        allele, 2 = second ALT allele etc. Useful when using <a href="#opt_minimal">--minimal</a>
        <i>Not used by default</i>
      </td>
      <td>ALLELE_NUM</td>
      <td>&nbsp;</td>
    </tr>
    <tr id="opt_show_ref_allele">
      <td><pre>--show_ref_allele</pre></td>
      <td>&nbsp;</td>
      <td>
        Adds the reference allele in the output (after minimisation). Mainly useful for the Ensembl VEP "default" and tab-delimited output formats.
        <i>Not used by default</i>
      </td>
      <td>REF_ALLELE</td>
      <td>&nbsp;</td>
    </tr> 
    <tr id="opt_uploaded_allele">
      <td><pre>--uploaded_allele</pre></td>
      <td>&nbsp;</td>
      <td>
        Adds the uploaded allele string in the output (before minimisation). 
      </td>
      <td>UPLOADED_ALLELE</td>
      <td>&nbsp;</td>
    </tr> 
    <tr id="opt_total_length">
      <td><pre>--total_length</pre></td>
      <td>&nbsp;</td>
      <td>
        Give cDNA, CDS and protein positions as Position/Length. <i>Not
        used by default</i>
      </td>
      <td >&nbsp;</td>
      <td >&nbsp;</td>
    </tr>
    <tr id="opt_numbers">
      <td><pre>--numbers</pre></td>
      <td>&nbsp;</td>
      <td>
        Adds affected exon and intron numbering to to output. Format is
        Number/Total. <i>Not used by default</i>
      </td>
      <td>EXON, INTRON</td>
      <td>
        <div style="padding:2px 0px"><a href="#opt_most_severe">--most_severe</a></div>
        <div style="padding:2px 0px"><a href="#opt_summary">--summary</a></div>
      </td>
    </tr>
    <tr id="opt_mirna">
      <td><pre>--mirna</pre></td>
      <td>&nbsp;</td>
      <td>
        Reports where the variant lies in the miRNA secondary structure (only for Ensembl/GENCODE transcripts). <i>Not used by default</i>
      </td>
      <td>&nbsp;</td>
      <td>&nbsp;</td>
    </tr>
    <tr id="opt_no_escape">
      <td><pre>--no_escape</pre></td>
      <td>&nbsp;</td>
      <td>
        Don't URI escape HGVSp field. <i>Default = escape</i>
      </td>
      <td >&nbsp;</td>
      <td >&nbsp;</td>
    </tr>
    <tr id="opt_keep_csq">
      <td><pre>--keep_csq</pre></td>
      <td>&nbsp;</td>
      <td>
        Don't overwrite existing CSQ entry in
        <a href="../vep_formats.html#vcfout">VCF INFO field</a>. <i>Overwrites
        by default</i>
      </td>
      <td >&nbsp;</td>
      <td >&nbsp;</td>
    </tr>
    <tr id="opt_vcf_info_field">
      <td><pre>--vcf_info_field [CSQ|ANN|(other)]</pre></td>
      <td>&nbsp;</td>
      <td>
        Change the name of the INFO key that Ensembl VEP write the consequences to
        in its <a href="../vep_formats.html#vcfout">VCF output</a>. Use "ANN"
        for compatibility with other tools such as <a
        href="http://snpeff.sourceforge.net/" rel="external">snpEff</a>.
        <i>Default: CSQ</i>
      </td>
      <td >&nbsp;</td>
      <td >&nbsp;</td>
    </tr>
    <tr id="opt_terms">
      <td><pre>--terms [SO|display|NCBI]</pre></td>
      <td><pre>-t</pre></td>
      <td>
        The type of consequence terms to output. The Ensembl terms are
        described <a
        href="/info/genome/variation/prediction/predicted_data.html#consequences"
        target="_blank">here</a>. The <a
        rel="external" href="http://www.sequenceontology.org/" target="_blank">Sequence
        Ontology</a> is a joint effort by genome annotation centres to
        standardise descriptions of biological sequences. <i>Default =
        "SO"</i>
      </td>
      <td >&nbsp;</td>
      <td >&nbsp;</td>
    </tr>
    <tr id="opt_no_headers">
      <td><pre>--no_headers</pre></td>
      <td>&nbsp;</td>
      <td>
        Don't write header lines in output files. <i>Default = add headers</i>
      </td>
      <td >&nbsp;</td>
      <td >&nbsp;</td>
    </tr>
    <tr id="opt_shift_3prime">
      <td><pre>--shift_3prime [0<b>|</b>1]</pre></td>
      <td>&nbsp;</td>
      <td> Right aligns all variants relative to their associated transcripts prior to consequence calculation. <br/>
	   An example using this option can be found <a
        href="/info/docs/tools/vep/script/vep_other.html#shifting"
        target="_blank">here</a>.<br/> <i>Default = 0</i>
      </td>
      <td></td>
      <td>
        <div style="padding:2px 0px"><a href="#opt_shift_hgvs">--shift_hgvs</a></div>
      </td>
    </tr>
    <tr id="opt_shift_genomic">
      <td><pre>--shift_genomic [0<b>|</b>1]</pre></td>
      <td>&nbsp;</td>
      <td> Right aligns all variants, including intergenic variants, before consequence calculation and updates the <i>Location</i> field. <br/>
           An example using this option can be found <a
        href="/info/docs/tools/vep/script/vep_other.html#shifting"
        target="_blank">here</a>.<br/> <i>Default = 0</i>
      </td>
      <td></td>
      <td>
        <div style="padding:2px 0px"><a href="#opt_shift_hgvs">--shift_hgvs</a></div>
      </td>
    </tr>
    <tr id="opt_shift_length">
      <td><pre>--shift_length</pre></td>
      <td>&nbsp;</td>
      <td> Reports the distance each variant has been shifted when used in conjuction with <a href="#opt_shift_3prime">--shift_3prime</a>
      </td>
      <td>&nbsp;</td>
      <td>&nbsp;</td>
    </tr>
  </table>
    
  
  <br/> 
  <hr/>
  <h2 id="ident">Identifiers</h2>
  
  <table class="ss" style="table-layout: fixed">
    <tr>
      <th width="20%">Flag</th>
      <th width="8%">Alternate</th>
      <th>Description</th>
      <th width="10%">Output fields</th>
      <th width="10%">Incompatible with</th>
    </tr>
    <tr id="opt_hgvs">
      <td><pre>--hgvs</pre></td>
      <td>&nbsp;</td>
      <td>
        Add <a rel="external" href="//varnomen.hgvs.org"
        target="_blank">HGVS</a> nomenclature based on Ensembl stable
        identifiers to the output. Both coding and protein sequence
        names are added where appropriate. To generate HGVS identifiers when
        using <a href="#opt_cache">--cache</a> or <a
        href="#opt_offline">--offline</a> you must use a FASTA file and <a
        href="#opt_fasta">--fasta</a>. HGVS notations given on Ensembl
        identifiers are <a href="/info/genome/stable_ids/index.html">
        versioned</a>. <i>Not used by default</i>
      </td>
      <td>HGVSc, HGVSp, HGVS_OFFSET</td>
      <td>&nbsp;</td>
    </tr>
    <tr id="opt_hgvsg">
      <td><pre>--hgvsg</pre></td>
      <td>&nbsp;</td>
      <td>
        Add genomic <a rel="external" href="//varnomen.hgvs.org"
        target="_blank">HGVS</a> nomenclature based on the input chromosome name.
        To generate HGVS identifiers when
        using <a href="#opt_cache">--cache</a> or <a
        href="#opt_offline">--offline</a> you must use a FASTA file and <a
        href="#opt_fasta">--fasta</a>. <i>Not used by default</i>
      </td>
      <td>HGVSg</td>
      <td>&nbsp;</td>
    </tr>
    <tr id="opt_hgvsg_use_accession">
      <td><pre>--hgvsg_use_accession</pre></td>
      <td>&nbsp;</td>
      <td>
        Force <a href="#opt_hgvsg">--hgvsg</a> to return RefSeq reference sequence.
        For example, reports NC_000002.11 for human chromosome 2 (build GRCh38).
      </td>
      <td>HGVSg</td>
      <td>&nbsp;</td>
    </tr>
    <tr id="opt_hgvsp_use_prediction">
      <td><pre>--hgvsp_use_prediction</pre></td>
      <td>&nbsp;</td>
      <td>
        Force <a href="#opt_hgvs">--hgvs</a> to return the HGVSp notation in predicted format.
        For example, ENSP00000233741.4:p.Thr367AsnfsTer13 will be returned as 
        ENSP00000233741.4:p.(Thr367AsnfsTer13).
      </td>
      <td>HGVSp</td>
      <td>&nbsp;</td>
    </tr>
    <tr id="opt_ambiguous_hgvs">
      <td><pre>--ambiguous_hgvs [0|1]</pre></td>
      <td>&nbsp;</td>
      <td>
        Allow input HGVSp to resolve to all genomic locations. Otherwise, most likely transcript will be selected. <i>Default: 0 (most likely transcript selected)</i>
      </td>
      <td>&nbsp;</td>
      <td>&nbsp;</td>
    </tr>
    <tr id="opt_spdi">
      <td><pre>--spdi</pre></td>
      <td>&nbsp;</td>
      <td>
      Add genomic <a rel="external" href="https://www.ncbi.nlm.nih.gov/variation/notation/",
      target="_blank">SPDI</a> notation. To generate SPDI when using <a href="#opt_cache">--cache</a> or <a
      href="#opt_offline">--offline</a> you must use a FASTA file and <a
      href="#opt_fasta">--fasta</a>. <i>Not used by default</i>
      </td>
      <td>SPDI</td>
      <td>&nbsp;</td>
    </tr>
    <tr id="opt_ga4gh_vrs">
      <td><pre>--ga4gh_vrs</pre></td>
      <td>&nbsp;</td>
      <td>
      Add <a rel="external" href="https://vrs.ga4gh.org/",
      target="_blank">GA4GH Variation Representation Specification (VRS)</a> notation. To generate GA4GH VRS when using <a href="#opt_cache">--cache</a> or <a
      href="#opt_offline">--offline</a> you must use a FASTA file and <a
      href="#opt_fasta">--fasta</a>. <i>Not used by default</i>
      </td>
      <td>GA4GH_VRS</td>
      <td>
        <div style="padding:2px 0px"><a href="#opt_vcf">--vcf</a></div>
      </td>
    </tr>
    <tr id="opt_shift_hgvs">
      <td><pre>--shift_hgvs [0|1]</pre></td>
      <td>&nbsp;</td>
      <td>
        Enable or disable 3' shifting of HGVS notations. HGVS nomenclature 
        requires an ambiguous sequence change to be described at the most 
        3' possible location. When enabled, this causes "shifting" 
        to the most 3' possible coordinates (relative to the transcript 
        sequence and strand) before the HGVS notations are calculated; 
        the flag HGVS_OFFSET is set to the number of bases by which the 
        variant has shifted, relative to the input genomic coordinates. 
        If HGVS_OFFSET is equals to 0, no value will be added to 
        HGVS_OFFSET column. To disable the changing of location at transcript
        level set --shift_hgvs to 0. <i>Default: 1 (shift)</i>
      </td>
      <td >&nbsp;</td>
      <td >
        <div style="padding:2px 0px"><a href="#opt_shift_3prime">--shift_3prime</a></div>
        <div style="padding:2px 0px"><a href="#opt_shift_genomic">--shift_genomic</a></div>
      </td>
    </tr>
    <tr id="opt_transcript_version">
      <td><pre>--transcript_version</pre></td>
      <td>&nbsp;</td>
      <td>
        Add version numbers to Ensembl transcript identifiers</i>
      </td>
      <td >&nbsp;</td>
      <td >&nbsp;</td>
    </tr>
    <tr id="opt_gene_version">
      <td><pre>--gene_version</pre></td>
      <td>&nbsp;</td>
      <td>
        Add version numbers to Ensembl gene identifiers</i>
      </td>
      <td >&nbsp;</td>
      <td >&nbsp;</td>
    </tr>
    <tr id="opt_protein">
      <td><pre>--protein</pre></td>
      <td>&nbsp;</td>
      <td>
        Add the Ensembl protein identifier to the output where
        appropriate. <i>Not used by default</i>
      </td>
      <td>ENSP</td>
      <td>
        <div style="padding:2px 0px"><a href="#opt_most_severe">--most_severe</a></div>
        <div style="padding:2px 0px"><a href="#opt_summary">--summary</a></div>
      </td>
    </tr>
    <tr id="opt_symbol">
      <td><pre>--symbol</pre></td>
      <td>&nbsp;</td>
      <td>
        Adds the gene symbol (e.g. HGNC) (where available) to the output. Some gene symbol, e.g. HGNC, are only available in merged and Ensembl caches and therefore should <b>not</b> be used with the <a href="#opt_refseq">--refseq</a> cache option.  
        <i>Not used by default</i>
      </td>
      <td>SYMBOL, SYMBOL_SOURCE, HGNC_ID</td>
      <td>
        <div style="padding:2px 0px"><a href="#opt_most_severe">--most_severe</a></div>
        <div style="padding:2px 0px"><a href="#opt_summary">--summary</a></div>
      </td>
    </tr>
    <tr id="opt_ccds">
      <td><pre>--ccds</pre></td>
      <td>&nbsp;</td>
      <td>
        Adds the CCDS transcript identifer (where available) to the
        output. <i>Not used by default</i>
      </td>
      <td>CCDS</td>
      <td>
        <div style="padding:2px 0px"><a href="#opt_most_severe">--most_severe</a></div>
        <div style="padding:2px 0px"><a href="#opt_summary">--summary</a></div>
      </td>
    </tr>
    <tr id="opt_uniprot">
      <td><pre>--uniprot</pre></td>
      <td>&nbsp;</td>
      <td>
        Adds best match accessions for translated protein products from three <a
        href="http://www.ebi.ac.uk/uniprot" rel="external">UniProt</a>-related
        databases (SWISSPROT, TREMBL and UniParc) to the output. <i>Not used by
        default</i>
      </td>
      <td>SWISSPROT, TREMBL, UNIPARC, UNIPROT_ISOFORM</td>
      <td>
        <div style="padding:2px 0px"><a href="#opt_most_severe">--most_severe</a></div>
        <div style="padding:2px 0px"><a href="#opt_summary">--summary</a></div>
      </td>
    </tr>
    <tr id="opt_tsl">
      <td><pre>--tsl</pre></td>
      <td>&nbsp;</td>
      <td>
        Adds the <a href="/Help/Glossary?id=492">transcript support level</a>
        for this transcript to the output. <i>Not used by default</i>
        <div class="info" style="max-width:350px;margin:5px 0px 5px 10px">
          <h3>Note</h3>
          <div class="message-pad" style="padding:5px 10px;background-color:#FFF">
            Only available for human on the GRCh38 assembly
          </div>
        </div>
      </td>
      <td>TSL</td>
      <td>
        <div style="padding:2px 0px"><a href="#opt_most_severe">--most_severe</a></div>
        <div style="padding:2px 0px"><a href="#opt_summary">--summary</a></div>
      </td>
    </tr>
    <tr id="opt_appris">
      <td><pre>--appris</pre></td>
      <td>&nbsp;</td>
      <td>
        Adds the <a href="/Help/Glossary?id=521">APPRIS</a> isoform annotation
        for this transcript to the output. <i>Not used by default</i>
        <div class="info" style="max-width:350px;margin:5px 0px 5px 10px">
          <h3>Note</h3>
          <div class="message-pad" style="padding:5px 10px;background-color:#FFF">
            Only available for human on the GRCh38 assembly
          </div>
        </div>
      </td>
      <td>APPRIS</td>
      <td>
        <div style="padding:2px 0px"><a href="#opt_most_severe">--most_severe</a></div>
        <div style="padding:2px 0px"><a href="#opt_summary">--summary</a></div>
      </td>
    </tr>
    <tr id="opt_canonical">
      <td><pre>--canonical</pre></td>
      <td>&nbsp;</td>
      <td>
        Adds a flag indicating if the transcript is the canonical
        transcript for the gene. <i>Not used by default</i>
      </td>
      <td>CANONICAL</td>
      <td>
        <div style="padding:2px 0px"><a href="#opt_most_severe">--most_severe</a></div>
        <div style="padding:2px 0px"><a href="#opt_summary">--summary</a></div>
      </td>
    </tr>
    <tr id="opt_mane">
      <td><pre>--mane</pre></td>
      <td>&nbsp;</td>
      <td>
        Adds a flag indicating if the transcript is the <a href="/info/genome/genebuild/mane.html">MANE Select</a> or <a href="/info/genome/genebuild/mane.html">MANE Plus Clinical</a> 
        transcript for the gene. If <a href="#opt_cache">--cache</a> or <a href="#opt_database">--database</a> annotation source is used, the alternative transcript stable ID is also added. 
        <i>Not used by default</i>
        <div class="info" style="max-width:350px;margin:5px 0px 5px 10px">
          <h3>Note</h3>
          <div class="message-pad" style="padding:5px 10px;background-color:#FFF">
            Only available for human on the GRCh38 assembly
          </div>
        </div>
      </td>
      <td>MANE, MANE_SELECT, MANE_PLUS_CLINICAL</td>
      <td>
        <div style="padding:2px 0px"><a href="#opt_most_severe">--most_severe</a></div>
        <div style="padding:2px 0px"><a href="#opt_summary">--summary</a></div>
      </td>
    </tr>
    <tr id="opt_mane_select">
      <td><pre>--mane_select</pre></td>
      <td>&nbsp;</td>
      <td>
        Adds a flag indicating if the transcript is the <a href="/info/genome/genebuild/mane.html">MANE Select</a>
        transcript for the gene. If <a href="#opt_cache">--cache</a> or <a href="#opt_database">--database</a> annotation source is used, the alternative transcript stable ID is also added. 
        <i>Not used by default</i>
        <div class="info" style="max-width:350px;margin:5px 0px 5px 10px">
          <h3>Note</h3>
          <div class="message-pad" style="padding:5px 10px;background-color:#FFF">
            Only available for human on the GRCh38 assembly
          </div>
        </div>
      </td>
      <td>MANE, MANE_SELECT</td>
      <td>
        <div style="padding:2px 0px"><a href="#opt_most_severe">--most_severe</a></div>
        <div style="padding:2px 0px"><a href="#opt_summary">--summary</a></div>
      </td>
    </tr>
    <tr id="opt_biotype">
      <td><pre>--biotype</pre></td>
      <td>&nbsp;</td>
      <td>
        Adds the biotype of the transcript or regulatory feature. <i>Not used by
        default</i>
      </td>
      <td>BIOTYPE</td>
      <td>
        <div style="padding:2px 0px"><a href="#opt_most_severe">--most_severe</a></div>
        <div style="padding:2px 0px"><a href="#opt_summary">--summary</a></div>
      </td>
    </tr>
    <tr id="opt_domains">
      <td><pre>--domains</pre></td>
      <td>&nbsp;</td>
      <td>
        Adds names of overlapping protein domains to output. <i>Not used
        by default</i>
      </td>
      <td>DOMAINS</td>
      <td>
        <div style="padding:2px 0px"><a href="#opt_most_severe">--most_severe</a></div>
        <div style="padding:2px 0px"><a href="#opt_summary">--summary</a></div>
      </td>
    </tr>
    <tr id="opt_xref_refseq">
      <td><pre>--xref_refseq</pre></td>
      <td>&nbsp;</td>
      <td>
        Output aligned RefSeq mRNA identifier for transcript. <i>Not used by default</i>
        <div class="info" style="margin:5px 0px 5px 10px">
          <h3>Note</h3>
          <div class="message-pad" style="padding:5px 10px;background-color:#FFF">
            The RefSeq and Ensembl transcripts aligned in this way MAY NOT,
            AND FREQUENTLY WILL NOT, match exactly in sequence, exon
            structure and protein product
          </div>
        </div>
      </td>
      <td>RefSeq</td>
      <td>
        <div style="padding:2px 0px"><a href="#opt_most_severe">--most_severe</a></div>
        <div style="padding:2px 0px"><a href="#opt_summary">--summary</a></div>
      </td>
    </tr>
    <tr id="opt_synonyms">
      <td><pre>--synonyms [file]</pre></td>
      <td>&nbsp;</td>
      <td>
        Load a file of chromosome synonyms. File should be tab-delimited
        with the primary identifier in column 1 and the synonym in column 2.
        Synonyms allow different chromosome identifiers to be used in the
        input file and any annotation source (cache, database, GFF, custom file, FASTA file).
        <i>Not used by default</i>
      </td>
      <td>&nbsp;</td>
      <td>&nbsp;</td>
    </tr>
  </table>
    
  
  <br/>
  <hr/>
  <h2 id="existing">Co-located variants</h2>
  
  <table class="ss" style="table-layout: fixed">
    <tr>
      <th width="20%">Flag</th>
      <th width="8%">Alternate</th>
      <th>Description</th>
      <th width="10%">Output fields</th>
      <th width="10%">Incompatible with</th>
    </tr>
    <tr id="opt_check_existing">
      <td><pre>--check_existing</pre></td>
      <td>&nbsp;</td>
      <td>
        Checks for the existence of known variants that are co-located
        with your input. By default the alleles are compared and variants on an allele-specific
        basis - to compare only coordinates,
        use <a href="#opt_no_check_alleles">--no_check_alleles</a>.
        <br/><br/>Some databases may contain variants with unknown (null) alleles and these
        are included by default; to exclude them use <a href="#opt_exclude_null_alleles">
        --exclude_null_alleles</a>.
        <br/><br/>See <a href="vep_other.html#colocated">this page</a> for more details.
        <br/><br/><i>Not used by default</i>
      </td>
      <td>Existing_variation, CLIN_SIG, SOMATIC, PHENO</td>
      <td>&nbsp;</td>
    </tr>
    <tr id="opt_check_svs">
      <td><pre>--check_svs</pre></td>
      <td>&nbsp;</td>
      <td>
        Checks for the existence of structural variants that overlap your
        input. Currently requires database access. <i>Not used by
        default</i>
      </td>
      <td>SV</td>
      <td><a href="#opt_offline">--offline</a></td>
    </tr>
    <tr id="opt_clin_sig_allele">
      <td><pre>--clin_sig_allele [1|0]</pre></td>
      <td>&nbsp;</td>
      <td>
        Return allele specific clinical significance. Setting this option to 0 will provide all known clinical significance values at the given locus.
        <i>Default: 1 (Provide allele-specific annotations)</i>
      </td>
      <td>CLIN_SIG</td>
      <td>&nbsp;</td>
    </tr>
    <tr id="opt_clinvar_somatic">
      <td><pre>--clinvar_somatic_classification</pre></td>
      <td>&nbsp;</td>
      <td>
        Return the somatic classification of a variant as reported by ClinVar.
        <i>Not used by default</i>
      </td>
      <td>SOMATIC_CLASSIFICATION</td>
      <td>&nbsp;</td>
    </tr>
    <tr id="opt_exclude_null_alleles">
      <td><pre>--exclude_null_alleles</pre></td>
      <td>&nbsp;</td>
      <td>
        Do not include variants with unknown alleles when checking for co-located variants.
        Our human database contains variants from HGMD and COSMIC for which the
        alleles are not publically available; by default these are included when using
        <a href="#opt_check_existing">--check_existing</a>, use this flag to exclude them.
        <i>Not used by default</i>
      </td>
      <td >&nbsp;</td>
      <td >&nbsp;</td>
    </tr>
    <tr id="opt_no_check_alleles">
      <td><pre>--no_check_alleles</pre></td>
      <td>&nbsp;</td>
      <td>
        When checking for existing variants, by default Ensembl VEP only reports a co-located
        variant if none of the input alleles are novel. For example,
        if your input variant has alleles A/G, and an existing co-located
        variant has alleles A/C, the co-located variant will not be
        reported.<br/><br/>Strand is also taken into account - in the
        same example, if the input variant has alleles T/G but on the
        negative strand, then the co-located variant <b>will</b> be
        reported since its alleles match the reverse complement of
        input variant.
        <br/><br/>Use this flag to disable this behaviour and compare using coordinates alone.
        <i>Not used by default</i>
      </td>
      <td >&nbsp;</td>
      <td >&nbsp;</td>
    </tr>
    <tr id="opt_af">
      <td><pre>--af</pre></td>
      <td>&nbsp;</td>
      <td>
        Add the global allele frequency (AF) from 1000 Genomes
        Phase 3 data for any known co-located variant to the output. For this and all --af_* flags, the frequency reported is for the <b>input allele</b> only, not necessarily the non-reference or derived allele. <i>Not used
        by default</i>
      </td>
      <td>AF</td>
      <td>&nbsp;</td>
    </tr>
    <tr id="opt_max_af">
      <td><pre>--max_af</pre></td>
      <td>&nbsp;</td>
      <td>
        Report the highest allele frequency observed in any population from 1000 genomes, ESP or gnomAD. <i>Not used
        by default</i>
      </td>
      <td>MAX_AF, MAX_AF_POPS</td>
      <td>
        <div style="padding:2px 0px"><a href="#opt_database">--database</a></div>
      </td>
    </tr>
    <tr id="opt_af_1kg">
      <td><pre>--af_1kg</pre></td>
      <td>&nbsp;</td>
      <td>
        Add allele frequency from continental populations (AFR,AMR,EAS,EUR,SAS) of <a href="http://www.1000genomes.org/" rel="external">1000
        Genomes Phase 3</a> to the output. Must be used with <a
        href="#opt_cache">--cache</a>. <i>Not used by default</i>
      </td>
      <td>AFR_AF, AMR_AF, EAS_AF, EUR_AF, SAS_AF</td>
      <td>
        <div style="padding:2px 0px"><a href="#opt_database">--database</a></div>
      </td>
    </tr>
    <tr id="opt_af_esp">
      <td><pre>--af_esp</pre></td>
      <td>&nbsp;</td>
      <td>
        Include allele frequency from <a href="http://evs.gs.washington.edu/EVS/" rel="external">NHLBI-ESP</a> populations. Must be used with
        <a href="#opt_cache">--cache</a>. <i>Deprecated.</i>
      </td>
      <td>AA_AF, EA_AF</td>
      <td>
        <div style="padding:2px 0px"><a href="#opt_database">--database</a></div>
      </td>
    </tr>
    <tr id="opt_af_gnomade">
      <td><pre>--af_gnomade</pre></td>
      <td><pre>--af_gnomad</pre></td>
      <td>
        Include allele frequency from <a href="http://gnomad.broadinstitute.org/" rel="external">Genome Aggregation Database (gnomAD)</a> exome populations. Note only data from the gnomAD exomes are included; to retrieve data from the additional genomes data set,
        see <a href="vep_example.html#gnomad">this guide</a>. Must be used with
        <a href="#opt_cache">--cache</a> <i>Not used by default</i>
      </td>
      <td>gnomADe_AF, gnomADe_AFR_AF, gnomADe_AMR_AF, gnomADe_ASJ_AF, gnomADe_EAS_AF, gnomADe_FIN_AF, gnomADe_NFE_AF, gnomADe_OTH_AF, gnomADe_SAS_AF</td>
      <td>
        <div style="padding:2px 0px"><a href="#opt_database">--database</a></div>
        <div style="padding:2px 0px"><a href="#opt_af_gnomad">--af_gnomad</a></div>
      </td>
    </tr>
    <tr id="opt_af_gnomadg">
      <td><pre>--af_gnomadg</pre></td>
      <td>&nbsp;</td>
      <td>
        Include allele frequency from <a href="http://gnomad.broadinstitute.org/" rel="external">Genome Aggregation Database (gnomAD)</a> genome populations. Note only data from the gnomAD genomes are included; to retrieve data from the additional genomes data set,
        see <a href="vep_example.html#gnomad">this guide</a>. Must be used with
        <a href="#opt_cache">--cache</a> <i>Not used by default</i>
      </td>
      <td>gnomADg_AF, gnomADg_AFR_AF, gnomADg_AMI_AF, gnomADg_AMR_AF, gnomADg_ASJ_AF, gnomADg_EAS_AF, gnomADg_FIN_AF, gnomADg_MID_AF, gnomADg_NFE_AF, gnomADg_OTH_AF, gnomADg_SAS_AF</td>
      <td>
        <div style="padding:2px 0px"><a href="#opt_database">--database</a></div>
      </td>
    </tr>
    <tr id="opt_af_exac">
      <td><pre>--af_exac</pre></td>
      <td>&nbsp;</td>
      <td>
        Include allele frequency from <a href="http://exac.broadinstitute.org/" rel="external">ExAC project</a> populations. Must be used with
        <a href="#opt_cache">--cache</a>. <i>Deprecated.</i>
      </td>
      <td>ExAC_AF, ExAC_Adj_AF, ExAC_AFR_AF, ExAC_AMR_AF, ExAC_EAS_AF, ExAC_FIN_AF, ExAC_NFE_AF, ExAC_OTH_AF, ExAC_SAS_AF</td>
      <td>
        <div style="padding:2px 0px"><a href="#opt_database">--database</a></div>
      </td>
    </tr>
    <tr id="opt_pubmed">
      <td><pre>--pubmed</pre></td>
      <td>&nbsp;</td>
      <td>
        Report Pubmed IDs for publications that cite existing variant.
        Must be used with <a href="#opt_cache">--cache</a>. <i>Not used
        by default</i>
      </td>
      <td>PUBMED</td>
      <td>
        <div style="padding:2px 0px"><a href="#opt_database">--database</a></div>
      </td>
    </tr>
    <tr id="opt_var_synonyms">
      <td><pre>--var_synonyms</pre></td>
      <td>&nbsp;</td>
      <td>
        Report known synonyms for co-located variants.
        Must be used with <a href="#opt_cache">--cache</a>. <i>Not used
        by default</i>
      </td>
      <td>VAR_SYNONYMS</td>
      <td>
        <div style="padding:2px 0px"><a href="#opt_database">--database</a></div>
      </td>
    </tr>
    <tr id="opt_failed">
      <td><pre>--failed [0|1]</pre></td>
      <td>&nbsp;</td>
      <td>
        When checking for co-located variants, by default Ensembl VEP
        will exclude variants that have been flagged as failed. Set this
        flag to include such variants. <i>Default: 0 (exclude)</i>
      </td>
      <td >&nbsp;</td>
      <td >&nbsp;</td>
    </tr>
  </table>
  
  
  <br/>
  <hr/> 
  <h2 id="filt">Filtering and QC options</h2>
    
    <p> <b>NOTE:</b> The filtering options here filter your results
    <b>before</b> they are written to your output file. Using Ensembl VEP's <a
    href="vep_filter.html">filtering script</a>, it is possible to filter your
    results <b>after</b> Ensembl VEP has run. This way you can retain all of the
    results and run multiple filter sets on the same results to find different
    data of interest. </p>
  
  <table class="ss" style="table-layout: fixed">
    <tr>
      <th width="20%">Flag</th>
      <th width="8%">Alternate</th>
      <th>Description</th>
      <th width="10%">Output fields</th>
      <th width="10%">Incompatible with</th>
    </tr>
    <tr id="opt_gencode_basic">
      <td><pre>--gencode_basic</pre></td>
      <td>&nbsp;</td>
      <td>
        Limit your analysis to transcripts belonging to the GENCODE
        basic set. This set has fragmented or problematic transcripts removed. <i>Not used by default</i>
      </td>
      <td>&nbsp;</td>
      <td>
        <div style="padding:2px 0px"><a href="#opt_gencode_primary">--gencode_primary</a></div>
        <div style="padding:2px 0px"><a href="#opt_refseq">--refseq</a></div>
      </td>
    </tr>
    <tr id="opt_gencode_primary">
      <td><pre>--gencode_primary</pre></td>
      <td>&nbsp;</td>
      <td>
        Limit your analysis to transcripts belonging to the GENCODE
        primary set. This set covers all human exons in a minimal set of transcripts. <i>Not used by default</i>
        <div class="info" style="max-width:350px;margin:5px 0px 5px 10px">
          <h3>Note</h3>
          <div class="message-pad" style="padding:5px 10px;background-color:#FFF">
            Only available for human on the GRCh38 assembly
          </div>
        </div>
      </td>
      <td>&nbsp;</td>
      <td>
        <div style="padding:2px 0px"><a href="#opt_gencode_basic">--gencode_basic</a></div>
        <div style="padding:2px 0px"><a href="#opt_refseq">--refseq</a></div>
      </td>
    </tr>
    <tr id="opt_exclude_predicted">
      <td><pre>--exclude_predicted</pre></td>
      <td>&nbsp;</td>
      <td>
        When using the RefSeq or merged cache, exclude predicted transcripts
        (i.e. those with identifiers beginning with "XM_" or "XR_").
        <div class="info" style="max-width:350px;margin:5px 0px 5px 10px">
          <h3>Note</h3>
          <div class="message-pad" style="padding:5px 10px;background-color:#FFF">
            We do not support predicted RefSeq transcripts for GRCh37.
          </div>
        </div>
      </td>
      <td >&nbsp;</td>
      <td >&nbsp;</td>
    </tr>
    <tr id="opt_transcript_filter">
      <td><pre>--transcript_filter</pre></td>
      <td>&nbsp;</td>
      <td>
        <p><b>ADVANCED</b> Filter transcripts according to any arbitrary set of rules. Uses similar notation to <a href="vep_filter.html">filter_vep</a>.</p>
        <p>You may filter on any key defined in the root of the transcript object; most commonly this will be "stable_id":</p>
        <pre class="code sh_sh">--transcript_filter "stable_id match N[MR]_"</pre>
        <p>or, a list of stable ids in file acting as a allowlist or a blocklist:</p>
        <pre class="code sh_sh">--transcript_filter "not stable_id in blocklist.txt"</pre>
      </td>
      <td >&nbsp;</td>
      <td >&nbsp;</td>
    </tr>
    <tr id="opt_check_ref">
      <td><pre>--check_ref</pre></td>
      <td>&nbsp;</td>
      <td>
        Force Ensembl VEP to check the supplied reference allele against
        the sequence stored in the Ensembl Core database or supplied <a href="#opt_fasta">FASTA file</a>. Lines that do
        not match are skipped. Checking is done on the minimised sequence. Example chr13 32900399 . AGT A . the As are removed and the reference sequence is checked from 32900400 to see if it matches GT<i>Not used by default</i>
      </td>
      <td>&nbsp;</td>
      <td>
        <div style="padding:2px 0px"><a href="#opt_lookup_ref">--lookup_ref</a></div>
      </td>
    </tr>
    <tr id="opt_lookup_ref">
      <td><pre>--lookup_ref</pre></td>
      <td>&nbsp;</td>
      <td>
        Force overwrite the supplied reference allele with
        the sequence stored in the Ensembl Core database or supplied <a href="#opt_fasta">FASTA file</a>. <i>Not used by default</i>
      </td>
      <td>&nbsp;</td>
      <td>
        <div style="padding:2px 0px"><a href="#opt_check_ref">--check_ref</a></div>
      </td>
    </tr>
    <tr id="dont_skip">
      <td><pre>--dont_skip</pre></td>
      <td>&nbsp;</td>
      <td>
        Don't skip input variants that fail validation, e.g. those that
        fall on unrecognised sequences.<br />Combining <a href="#opt_check_ref">--check_ref</a> with <a href="#opt_dont_skip">--dont_skip</a> will add a CHECK_REF output field when the given reference does not match the underlying reference sequence.
      </td>
      <td>CHECK_REF</td>
      <td>&nbsp;</td>
    </tr>
    <tr id="opt_allow_non_variant">
      <td><pre>--allow_non_variant</pre></td>
      <td>&nbsp;</td>
      <td>
        When using VCF format as input and output, by default Ensembl VEP
        will skip non-variant lines of input (where the ALT allele is
        null). Enabling this option the lines will be printed in the VCF
        output with no consequence data added.
      </td>
      <td >&nbsp;</td>
      <td >&nbsp;</td>
    </tr>
    <tr id="opt_chr">
      <td><pre>--chr [list]</pre></td>
      <td>&nbsp;</td>
      <td>
        Select a subset of chromosomes to analyse from your file. Any
        data not on this chromosome in the input will be skipped. The
        list can be comma separated, with "-" characters representing an
        interval.<br />For example, to include chromosomes 1, 2, 3, 10 and X
        you could use <a href="vep_options.html#opt_chr">--chr 1-3,10,X</a> <i>Not used by
        default</i>
      </td>
      <td >&nbsp;</td>
      <td >&nbsp;</td>
    </tr>
    <tr id="opt_coding_only">
      <td><pre>--coding_only</pre></td>
      <td>&nbsp;</td>
      <td>
        Only return consequences that fall in the coding regions of
        transcripts. <i>Not used by default</i>
      </td>
      <td>&nbsp;</td>
      <td>
        <div style="padding:2px 0px"><a href="#opt_most_severe">--most_severe</a></div>
        <div style="padding:2px 0px"><a href="#opt_summary">--summary</a></div>
      </td>
    </tr>
    <tr id="opt_no_intergenic">
      <td><pre>--no_intergenic</pre></td>
      <td>&nbsp;</td>
      <td>
        Do not include intergenic consequences in the output. <i>Not
        used by default</i>
      </td>
      <td>&nbsp;</td>
      <td>
        <div style="padding:2px 0px"><a href="#opt_most_severe">--most_severe</a></div>
        <div style="padding:2px 0px"><a href="#opt_summary">--summary</a></div>
      </td>
    </tr>
    <tr id="opt_pick">
      <td><pre>--pick</pre></td>
      <td>&nbsp;</td>
      <td>
        Pick one line or block of consequence data per variant, including
        transcript-specific columns.<br />Consequences are chosen according to the
        criteria described <a href="vep_other.html#pick">here</a>, and the
        order the criteria are applied may be customised with
        <a href="#opt_pick_order">--pick_order</a>.
        This is the best method to use if you are interested only in
        one consequence per variant. <i>Not used by default</i>
      </td>
      <td>&nbsp;</td>
      <td>
        <div style="padding:2px 0px"><a href="#opt_most_severe">--most_severe</a></div>
        <div style="padding:2px 0px"><a href="#opt_summary">--summary</a></div>
      </td>
    </tr>
    <tr id="opt_pick_allele">
      <td><pre>--pick_allele</pre></td>
      <td>&nbsp;</td>
      <td>
        Like <a href="#opt_pick">--pick</a>, but chooses one line or block of
        consequence data per variant allele. Will only differ in behaviour from
        <b>--pick</b> when the input variant has multiple alternate alleles.
        <i>Not used by default</i>
      </td>
      <td>&nbsp;</td>
      <td>
        <div style="padding:2px 0px"><a href="#opt_most_severe">--most_severe</a></div>
        <div style="padding:2px 0px"><a href="#opt_summary">--summary</a></div>
      </td>
    </tr>
    <tr id="opt_per_gene">
      <td><pre>--per_gene</pre></td>
      <td>&nbsp;</td>
      <td>
        Output only the most severe consequence per gene. The transcript
        selected is arbitrary if more than one has the same predicted
        consequence. Uses the same ranking system as
        <a href="#opt_pick">--pick</a>. <i>Not used by default</i>
      </td>
      <td >&nbsp;</td>
      <td >&nbsp;</td>
    </tr>
    <tr id="opt_pick_allele_gene">
      <td><pre>--pick_allele_gene</pre></td>
      <td>&nbsp;</td>
      <td>
        Like <a href="#opt_pick">--pick_allele</a>, but chooses one line or block of
        consequence data per variant allele <b>and</b> gene combination.
        <i>Not used by default</i>
      </td>
      <td >&nbsp;</td>
      <td >&nbsp;</td>
    </tr>
    <tr id="opt_flag_pick">
      <td><pre>--flag_pick</pre></td>
      <td>&nbsp;</td>
      <td>
        As per <a href="#opt_pick">--pick</a>, but adds the PICK flag to the
        chosen block of consequence data and retains others. <i>Not used by
        default</i>
      </td>
      <td>PICK</td>
      <td>
        <div style="padding:2px 0px"><a href="#opt_most_severe">--most_severe</a></div>
        <div style="padding:2px 0px"><a href="#opt_summary">--summary</a></div>
      </td>
    </tr>
    <tr id="opt_flag_pick_allele">
      <td><pre>--flag_pick_allele</pre></td>
      <td>&nbsp;</td>
      <td>
        As per <a href="#opt_pick_allele">--pick_allele</a>, but adds the PICK
        flag to the chosen block of consequence data and retains others. <i>Not used by default</i>
      </td>
      <td>PICK</td>
      <td>
        <div style="padding:2px 0px"><a href="#opt_most_severe">--most_severe</a></div>
        <div style="padding:2px 0px"><a href="#opt_summary">--summary</a></div>
      </td>
    </tr>
    <tr id="opt_flag_pick_allele_gene">
      <td><pre>--flag_pick_allele_gene</pre></td>
      <td>&nbsp;</td>
      <td>
        As per <a href="#opt_pick_allele_gene">--pick_allele_gene</a>, but adds the PICK
        flag to the chosen block of consequence data and retains others. <i>Not used by default</i>
      </td>
      <td>PICK</td>
      <td>&nbsp;</td>
    </tr>
    <tr id="opt_pick_order">
      <td><pre>--pick_order [c1,c2,...,cN]</pre></td>
      <td>&nbsp;</td>
      <td>
        <p>Customise the order of criteria (and the list of criteria) applied when choosing a block of
        annotation data with one of the following options: <a href="#opt_pick">--pick</a>, <a href="#opt_pick_allele">--pick_allele</a>, <a href="#opt_per_gene">--per_gene</a>, <a href="#opt_pick_allele_gene">--pick_allele_gene</a>, <a href="#opt_flag_pick">--flag_pick</a>, <a href="#opt_flag_pick_allele">--flag_pick_allele</a>, <a href="#opt_flag_pick_allele_gene">--flag_pick_allele_gene</a>. See
        <a href="vep_other.html#pick_options">this page</a> for the default order.<br />
        Valid criteria are: <i>mane_select, mane_plus_clinical, canonical, appris, tsl, biotype, ccds, rank, length, ensembl, refseq</i>. e.g.:</p>
        <pre class="code sh_sh">--pick --pick_order tsl,appris,rank</pre>
      </td>
      <td >&nbsp;</td>
      <td >&nbsp;</td>
    </tr>
    <tr id="opt_most_severe">
      <td><pre>--most_severe</pre></td>
      <td>&nbsp;</td>
      <td>
        Output only the most severe consequence per variant.
        Transcript-specific columns will be left blank. Consequence ranks are
        given in <a
        href="/info/genome/variation/prediction/predicted_data.html#consequences">this
        table</a>.<br />
        To include regulatory consequences, use the <b>--regulatory</b> option in combination with this flag.<br /> 
        <i>Not used by default</i>
      </td>
      <td>&nbsp;</td>
      <td>
        <div style="padding:2px 0px"><a href="#opt_appris">--appris</a></div>
        <div style="padding:2px 0px"><a href="#opt_biotype">--biotype</a></div>
        <div style="padding:2px 0px"><a href="#opt_canonical">--canonical</a></div>
        <div style="padding:2px 0px"><a href="#opt_ccds">--ccds</a></div>
        <div style="padding:2px 0px"><a href="#opt_coding_only">--coding_only</a></div>
        <div style="padding:2px 0px"><a href="#opt_domains">--domains</a></div>
        <div style="padding:2px 0px"><a href="#opt_flag_pick">--flag_pick</a></div>
        <div style="padding:2px 0px"><a href="#opt_flag_pick_allele">--flag_pick_allele</a></div>
        <div style="padding:2px 0px"><a href="#opt_no_intergenic">--no_intergenic</a></div>
        <div style="padding:2px 0px"><a href="#opt_numbers">--numbers</a></div>
        <div style="padding:2px 0px"><a href="#opt_pick">--pick</a></div>
        <div style="padding:2px 0px"><a href="#opt_pick_allele">--pick_allele</a></div>
        <div style="padding:2px 0px"><a href="#opt_polyphen">--polyphen</a></div>
        <div style="padding:2px 0px"><a href="#opt_protein">--protein</a></div>
        <div style="padding:2px 0px"><a href="#opt_sift">--sift</a></div>
        <div style="padding:2px 0px"><a href="#opt_summary">--summary</a></div>
        <div style="padding:2px 0px"><a href="#opt_symbol">--symbol</a></div>
        <div style="padding:2px 0px"><a href="#opt_tsl">--tsl</a></div>
        <div style="padding:2px 0px"><a href="#opt_uniprot">--uniprot</a></div>
        <div style="padding:2px 0px"><a href="#opt_xref_refseq">--xref_refseq</a></div>
        <div style="padding:2px 0px"><a href="#opt_mane">--mane</a></div>
        <div style="padding:2px 0px"><a href="#opt_mane_select">--mane_select</a></div>
        <div style="padding:2px 0px"><a href="#opt_vcf">--vcf</a></div>
      </td>
    </tr>
    <tr id="opt_summary">
      <td><pre>--summary</pre></td>
      <td>&nbsp;</td>
      <td>
        Output only a comma-separated list of all observed consequences
        per variant. Transcript-specific columns will be left blank.
        <i>Not used by default</i>
      </td>
      <td>&nbsp;</td>
      <td>
        <div style="padding:2px 0px"><a href="#opt_appris">--appris</a></div>
        <div style="padding:2px 0px"><a href="#opt_biotype">--biotype</a></div>
        <div style="padding:2px 0px"><a href="#opt_canonical">--canonical</a></div>
        <div style="padding:2px 0px"><a href="#opt_ccds">--ccds</a></div>
        <div style="padding:2px 0px"><a href="#opt_coding_only">--coding_only</a></div>
        <div style="padding:2px 0px"><a href="#opt_domains">--domains</a></div>
        <div style="padding:2px 0px"><a href="#opt_flag_pick">--flag_pick</a></div>
        <div style="padding:2px 0px"><a href="#opt_flag_pick_allele">--flag_pick_allele</a></div>
        <div style="padding:2px 0px"><a href="#opt_most_severe">--most_severe</a></div>
        <div style="padding:2px 0px"><a href="#opt_no_intergenic">--no_intergenic</a></div>
        <div style="padding:2px 0px"><a href="#opt_numbers">--numbers</a></div>
        <div style="padding:2px 0px"><a href="#opt_pick">--pick</a></div>
        <div style="padding:2px 0px"><a href="#opt_pick_allele">--pick_allele</a></div>
        <div style="padding:2px 0px"><a href="#opt_polyphen">--polyphen</a></div>
        <div style="padding:2px 0px"><a href="#opt_protein">--protein</a></div>
        <div style="padding:2px 0px"><a href="#opt_sift">--sift</a></div>
        <div style="padding:2px 0px"><a href="#opt_symbol">--symbol</a></div>
        <div style="padding:2px 0px"><a href="#opt_tsl">--tsl</a></div>
        <div style="padding:2px 0px"><a href="#opt_uniprot">--uniprot</a></div>
        <div style="padding:2px 0px"><a href="#opt_xref_refseq">--xref_refseq</a></div>
        <div style="padding:2px 0px"><a href="#opt_mane">--mane</a></div>
        <div style="padding:2px 0px"><a href="#opt_mane_select">--mane_select</a></div>
        <div style="padding:2px 0px"><a href="#opt_vcf">--vcf</a></div>
      </td>
    </tr>
    <tr id="opt_flag_gencode_primary">
      <td><pre>--flag_gencode_primary</pre></td>
      <td>&nbsp;</td>
      <td>
        Flags transcripts as GENCODE primary using a boolean value. <i>Not used by default</i>
        <div class="info" style="max-width:350px;margin:5px 0px 5px 10px">
          <h3>Note</h3>
          <div class="message-pad" style="padding:5px 10px;background-color:#FFF">
            Only available for human on the GRCh38 assembly
          </div>
        </div>
      </td>
      <td>GENCODE_PRIMARY</td>
      <td>&nbsp;</td>
    </tr>
    <tr id="opt_filter_common">
      <td><pre>--filter_common</pre></td>
      <td>&nbsp;</td>
      <td>
        Shortcut flag for the filters below - this will exclude variants
        that have a co-located existing variant with global AF &gt;
        0.01 (1%). May be modified using any of the following freq_*
        filters. <i>Not used by default</i>
      </td>
      <td>FREQS</td>
      <td>&nbsp;</td>
    </tr>
    <tr id="opt_check_frequency">
      <td><pre>--check_frequency</pre></td>
      <td>&nbsp;</td>
      <td>
        Turns on frequency filtering. Use this to include or exclude
        variants based on the frequency of co-located existing variants
        in the Ensembl Variation database. You must also specify all of the --freq_* flags below.
        Frequencies used in filtering are added to the output under the FREQS
        key in the Extra field. <i>Not used by default</i>
      </td>
      <td>FREQS</td>
      <td>&nbsp;</td>
    </tr>
    <tr id="opt_freq_pop">
      <td><pre>--freq_pop [pop]</pre></td>
      <td>&nbsp;</td>
      <td>
        Name of the population to use in frequency filter. This must be one of the following:
        <p>
          <table class="ss" style="width:auto">
            <thead>
              <tr>
                <th>Name</th><th>Description</th>
              </tr>
            </thead>
            <tbody>
              <tr class="bg2">
                <td>1KG_ALL</td><td>1000 genomes combined population (global)</td>
              </tr>
              <tr>
                <td>1KG_AFR</td><td>1000 genomes combined African population</td>
              </tr>
              <tr class="bg1">
                <td>1KG_AMR</td><td>1000 genomes combined American population</td>
              </tr>
              <tr>
                <td>1KG_EAS</td><td>1000 genomes combined East Asian population</td>
              </tr>
              <tr class="bg1">
                <td>1KG_EUR</td><td>1000 genomes combined European population</td>
              </tr>
              <tr>
                <td>1KG_SAS</td><td>1000 genomes combined South Asian population</td>
              </tr>

              <tr class="bg2">
                <td>gnomADe</td><td>gnomAD exomes combined population</td>
              </tr>
              <tr>
                <td>gnomADe_AFR</td><td>gnomAD exomes African/African American population</td>
              </tr>
              <tr class="bg1">
                <td>gnomADe_AMR</td><td>gnomAD exomes Latino population</td>
              </tr>
              <tr>
                <td>gnomADe_ASJ</td><td>gnomAD exomes Ashkenazi Jewish population</td>
              </tr>
              <tr class="bg1">
                <td>gnomADe_EAS</td><td>gnomAD exomes East Asian population</td>
              </tr>
              <tr>
                <td>gnomADe_FIN</td><td>gnomAD exomes Finnish population</td>
              </tr>
              <tr class="bg1">
                <td>gnomADe_NFE</td><td>gnomAD exomes non-Finnish European population</td>
              </tr>
              <tr>
                <td>gnomADe_OTH</td><td>gnomAD exomes other population</td>
              </tr>
              <tr class="bg1">
                <td>gnomADe_SAS</td><td>gnomAD exomes South Asian population</td>
              </tr>
              <tr class="bg2">
                <td>gnomADg</td><td>gnomAD genomes combined population</td>
              </tr>
              <tr class="bg1">
                <td>gnomADg_AFR</td><td>gnomAD genomes African/African American population</td>
              </tr>
              <tr>
                <td>gnomADg_AMR</td><td>gnomAD genomes Latino population</td>
              </tr>
              <tr class="bg1">
                <td>gnomADg_AMI</td><td>gnomAD genomes Amish population</td>
              </tr>
              <tr>
                <td>gnomADg_ASJ</td><td>gnomAD genomes Ashkenazi Jewish population</td>
              </tr>
              <tr class="bg1">
                <td>gnomADg_EAS</td><td>gnomAD genomes East Asian population</td>
              </tr>
              <tr>
                <td>gnomADg_FIN</td><td>gnomAD genomes Finnish population</td>
              </tr>
              <tr class="bg1">
                <td>gnomADg_MID</td><td>gnomAD genomes Mid-eastern population</td>
              </tr>
              <tr>
                <td>gnomADg_NFE</td><td>gnomAD genomes non-Finnish European population</td>
              </tr>
              <tr class="bg1">
                <td>gnomADg_OTH</td><td>gnomAD genomes other population</td>
              </tr>
              <tr>
                <td>gnomADg_SAS</td><td>gnomAD genomes South Asian population</td>
              </tr>
            </tbody>
          </table>
        </p>
      </td>
      <td >&nbsp;</td>
      <td >&nbsp;</td>
    </tr>
    <tr id="opt_freq_freq">
      <td><pre>--freq_freq [freq]</pre></td>
      <td>&nbsp;</td>
      <td>
        Allele frequency to use for filtering. Must be a float
        value between 0 and 1
      </td>
      <td >&nbsp;</td>
      <td >&nbsp;</td>
    </tr>
    <tr id="opt_freq_gt_lt">
      <td><pre>--freq_gt_lt [gt<b>|</b>lt]</pre></td>
      <td>&nbsp;</td>
      <td>
        Specify whether the frequency of the co-located variant must be
        greater than (<b>gt</b>) or less than (<b>lt</b>) the value
        specified with <a href="#opt_freq_freq">--freq_freq</a>
      </td>
      <td >&nbsp;</td>
      <td >&nbsp;</td>
    </tr>
    <tr id="opt_freq_filter">
      <td><pre>--freq_filter [exclude<b>|</b>include]</pre></td>
      <td>&nbsp;</td>
      <td>
        Specify whether to <b>exclude</b> or <b>include</b> only
        variants that pass the frequency filter
      </td>
      <td >&nbsp;</td>
      <td >&nbsp;</td>
    </tr>    
  </table>
 

  <br/>
  <hr/>
  <h2 id="dbopt">Database options</h2>
  
  <table class="ss">
    <tr>
      <th width="20%">Flag</th>
      <th width="8%">Alternate</th>
      <th>Description</th>
      <th width="10%">Output fields</th>
      <th width="10%">Incompatible with</th>
    </tr>
    <tr id="opt_database">
      <td><pre>--database</pre></td>
      <td>&nbsp;</td>
      <td>Enable Ensembl VEP to use local or remote databases.</td>
      <td>&nbsp;</td>
      <td>
        <div style="padding:2px 0px"><a href="#opt_af_1kg">--af_1kg</a></div>
        <div style="padding:2px 0px"><a href="#opt_af_esp">--af_esp</a></div>
        <div style="padding:2px 0px"><a href="#opt_af_exac">--af_exac</a></div>
        <div style="padding:2px 0px"><a href="#opt_af_gnomade">--af_gnomad</a></div>
        <div style="padding:2px 0px"><a href="#opt_af_gnomade">--af_gnomade</a></div>
        <div style="padding:2px 0px"><a href="#opt_af_gnomadg">--af_gnomadg</a></div>
        <div style="padding:2px 0px"><a href="#opt_cache">--cache</a></div>
        <div style="padding:2px 0px"><a href="#opt_max_af">--max_af</a></div>
        <div style="padding:2px 0px"><a href="#opt_offline">--offline</a></div>
        <div style="padding:2px 0px"><a href="#opt_pubmed">--pubmed</a></div>
        <div style="padding:2px 0px"><a href="#opt_var_synonyms">--var_synonyms</a></div>
      </td>
    </tr>
    <tr id="opt_host">
      <td><pre>--host [hostname]</pre></td>
      <td>&nbsp;</td>
      <td>
        Manually define the database host to connect to. Users in the US
        may find connection and transfer speeds quicker using our East
        coast mirror, useastdb.ensembl.org. <i>Default =
        "ensembldb.ensembl.org"</i>
      </td>
      <td >&nbsp;</td>
      <td >&nbsp;</td>
    </tr>
    <tr id="opt_user">
      <td><pre>--user [username]</pre></td>
      <td><pre>-u</pre></td>
      <td>Manually define the database username. <i>Default =
      "anonymous"</i></td>
      <td >&nbsp;</td>
      <td >&nbsp;</td>
    </tr>
    <tr id="opt_password">
      <td><pre>--password [password]</pre></td>
      <td><pre>--pass</pre></td>
      <td>Manually define the database password. <i>Not used by
      default</i></td>
      <td >&nbsp;</td>
      <td >&nbsp;</td>
    </tr>
    <tr id="opt_port">
      <td><pre>--port [number]</pre></td>
      <td>&nbsp;</td>
      <td>Manually define the database port. <i>Default = 5306</i></td>
      <td >&nbsp;</td>
      <td >&nbsp;</td>
    </tr>
    <tr id="opt_genomes">
      <td><pre>--genomes</pre></td>
      <td>&nbsp;</td>
      <td>
        Override the default connection settings with those for the
        Ensembl Genomes public MySQL server. Required when using any of
        the <a rel="external" href="https://www.ensemblgenomes.org/"
        target="_blank">Ensembl Genomes</a> species. <i>Not used by
        default</i>
      </td>
      <td >&nbsp;</td>
      <td >&nbsp;</td>
    </tr>
    <tr id="opt_is_multispecies">
      <td><pre>--is_multispecies [0|1]</pre></td>
      <td>&nbsp;</td>
      <td>
        Some of the <a rel="external" href="https://www.ensemblgenomes.org/"
        target="_blank">Ensembl Genomes</a> databases (mainly bacteria and protists) are composed of a collection of close species.
        It updates the database connection settings (i.e. the database name) if the value is set to 1. <i>Default: 0</i>
      </td>
      <td >&nbsp;</td>
      <td >&nbsp;</td>
    </tr>
    <tr id="opt_lrg">
      <td><pre>--lrg</pre></td>
      <td>&nbsp;</td>
      <td>
        Map input variants to LRG coordinates (or to chromosome coordinates if
        given in LRG coordinates), and provide consequences on both LRG and
        chromosomal transcripts. <i>Not used by default</i>
      </td>
      <td>&nbsp;</td>
      <td>
        <div style="padding:2px 0px"><a href="#opt_offline">--offline</a></div>
      </td>
    </tr>
    <tr id="opt_db_version">
      <td><pre>--db_version [number]</pre></td>
      <td>&nbsp;</td>
      <td>
        Force Ensembl VEP to connect to a specific version of the Ensembl
        databases. Not recommended as there may be conflicts
        between software and database versions. <i>Not used by
        default</i>
      </td>
      <td >&nbsp;</td>
      <td >&nbsp;</td>
    </tr>
    <tr id="opt_registry">
      <td><pre>--registry [filename]</pre></td>
      <td>&nbsp;</td>
      <td>
        Defining a registry file overwrites other connection settings
        and uses those found in the specified registry file to connect.
        <i>Not used by default</i>
      </td>
      <td >&nbsp;</td>
      <td >&nbsp;</td>
    </tr>
  </table>
</div>

<div style="clear:both"><h1 style="page-break-after: always">&nbsp</h1></div><style>
tr:nth-child(odd) {background-color: #f0f0f0;}
</style>

<div>

  <div style="float:right"><img src="/img/vep_logo.png"/></div>

  <h1 id="top"><span style="color:#006;padding-right:15px">Ensembl Variant Effect Predictor</span><span style="color:#666"><img src="/i/16/database.png"> Annotation sources</span></h1>
  <hr/>

  <p> Ensembl VEP can use a variety of annotation sources to retrieve the transcript
  models used to predict consequence types. </p>

  <ul>
    <li><b><a href="#cache">Cache</a></b> - a downloadable file containing all transcript models, regulatory features and variant data for a species</li>
    <li><b><a href="#gff">GFF or GTF</a></b> - use transcript models defined in a tabix-indexed GFF or GTF file
      <ul><li>
        Requires a <b><a href="vep_cache.html#fasta">FASTA</a></b> file in
        <a href="vep_options.html#opt_offline">--offline</a> mode or if the desired species or assembly is not part of the
        <a href="/info/about/species.html">Ensembl species list</a>.
      </li></ul>
    </li>
    <li><b><a href="#database">Database</a></b> - connect to a MySQL database server hosting Ensembl databases</li>
  </ul>

  <p> Data from VCF, BED and bigWig files can also be incorporated by Ensembl VEP's <img src="/i/16/page-user.png" style="vertical-align:bottom"/> <a href="vep_custom.html">Custom annotation</a> feature.</p>

  <div style="background-color:#f3f9f4; color:#333; border-radius:5px; font-size:16px; border: 1px solid #91c89c; margin: 10px 0 10px 0; padding: 10px;">
    Using a cache is the most efficient way to use Ensembl VEP; we would
    encourage you to use a cache wherever possible. Caches are easy to
    download and set up using the <a
    href="vep_download.html#installer">installer</a>. Follow the <a
    href="vep_tutorial.html">tutorial</a> for a simple guide.
  </div>


  <br/>
  <hr/>
  <h2 id="cache">Caches</h2>

  <p> Using a cache (<a href="vep_options.html#opt_cache">--cache</a>) is
  the fastest and most efficient way to use Ensembl VEP, as in most cases
  only a single initial network connection is made and most data is read from
  local disk. Use <a href="#offline">offline</a> mode to eliminate all network
  connections for speed and/or privacy.</p>

  <div>
    <div style="float:left" class="warning">
      <h3 id="read_write_access">Cache version</h3>
      <div class="message-pad">
        <p>
          We strongly recommend that you download/use the cache version which corresponds to your Ensembl VEP installation, <br/>
          i.e. cache version <b>[[SPECIESDEFS::ENSEMBL_VERSION]]</b> should be used with the Ensembl VEP tool version <b>[[SPECIESDEFS::ENSEMBL_VERSION]]</b>.
        </p>
        <p>
          This is mainly due to the fact that the cache (data content and structure) is generated every Ensembl release, regarding the data and API updates for this release, therefore the cache data format might differ between versions (and be incompatible with a newer version of the Ensembl VEP tool).
        </p>
      </div>
    </div>
    <div class="clear"></div>
  </div>


  <br />
  <h3 id="pre">Downloading caches</h3>

  <p> Cache files are created for every species for each Ensembl release.
  They can be automatically downloaded and configured using <a
  href="vep_download.html#installer">INSTALL.pl</a>. <p>

  <p> If interested in RefSeq transcripts you may download an alternate cache
  file (e.g. homo_sapiens_refseq), or a merged file of RefSeq and Ensembl
  transcripts (eg homo_sapiens_merged); remember to specify <a
  href="vep_options.html#opt_refseq">--refseq</a> or <a
  href="vep_options.html#opt_merged">--merged</a> when running Ensembl VEP to use
  the relevant cache. See <a href="vep_other.html#refseq">documentation</a> for full details.</p>

  <br />
  <h3 id="manual_cache">Manually downloading caches</h3>

  <p> It is also simple to download and set up caches without using the installer.
  By default, Ensembl VEP searches for caches in $HOME/.vep; to use a different directory when running Ensembl VEP, use <a href="vep_options.html#opt_dir_cache">--dir_cache</a>.</p>

  <ul>
      <p><b>Indexed cache</b> (<a href="[[SPECIESDEFS::ENSEMBL_FTP_URL]]/release-[[SPECIESDEFS::ENSEMBL_VERSION]]/variation/indexed_vep_cache/">[[SPECIESDEFS::ENSEMBL_FTP_URL]]/release-[[SPECIESDEFS::ENSEMBL_VERSION]]/variation/indexed_vep_cache/</a>)
        <p>Essential for human and other species with large sets of variant data - requires <a href="https://github.com/Ensembl/Bio-DB-HTS" rel="external">Bio::DB::HTS</a> (setup by INSTALL.pl) or <a href="https://github.com/samtools/tabix" rel="external">tabix</a>, e.g.:</p>
      <pre class="code sh_sh left-margin">cd $HOME/.vep
curl -O [[SPECIESDEFS::ENSEMBL_FTP_URL]]/release-[[SPECIESDEFS::ENSEMBL_VERSION]]/variation/indexed_vep_cache/homo_sapiens_vep_[[SPECIESDEFS::ENSEMBL_VERSION]]_GRCh38.tar.gz
tar xzf homo_sapiens_vep_[[SPECIESDEFS::ENSEMBL_VERSION]]_GRCh38.tar.gz</pre>
  </ul>

  <p><img src="/i/16/download.png" style="vertical-align:bottom"/> <b>FTP directories with indexed cache data:</b></p>

  <table class="ss" style="width:auto">
    <tr>
      <th style="border-bottom:1px solid #f0f0f0">Ensembl: </th>
      <td>
        <a href="[[SPECIESDEFS::ENSEMBL_FTP_URL]]/current_variation/indexed_vep_cache/">Vertebrates</a>
      </td>
    </tr>
    <tr>
      <th style="padding-right:10px">Ensembl Genomes: </th>
      <td>
        <a href="[[SPECIESDEFS::ENSEMBL_GENOMES_FTP_URL]]/bacteria/current/variation/indexed_vep_cache/">Bacteria</a>
        <span style="padding:0px 4px">|</span>
        <a href="[[SPECIESDEFS::ENSEMBL_GENOMES_FTP_URL]]/fungi/current/variation/indexed_vep_cache/">Fungi</a>
        <span style="padding:0px 4px">|</span>
        <a href="[[SPECIESDEFS::ENSEMBL_GENOMES_FTP_URL]]/metazoa/current/variation/indexed_vep_cache/">Metazoa</a>
        <span style="padding:0px 4px">|</span>
        <a href="[[SPECIESDEFS::ENSEMBL_GENOMES_FTP_URL]]/plants/current/variation/indexed_vep_cache/">Plants</a>
        <span style="padding:0px 4px">|</span>
        <a href="[[SPECIESDEFS::ENSEMBL_GENOMES_FTP_URL]]/protists/current/variation/indexed_vep_cache/">Protists</a>
      </td>
    </tr>
  </table>
  <p><b>NB:</b> When using Ensembl Genomes caches, you should use the <a
  href="vep_options.html#opt_cache_version">--cache_version</a> option to
  specify the relevant Ensembl Genomes version number as these differ from the
  concurrent Ensembl VEP version numbers.</p>

  <h4 id="pangenomes">HPRC and alternative assemblies</h4>

  <p>Ensembl VEP caches are also available for Human Pangenome Reference Consortium (HPRC) data at the <a href="https://projects.ensembl.org/hprc/" rel="external">Ensembl HPRC data page</a>. Click <a href="vep_example.html#pangenomes">here</a> for more information on how to annotate variants on HPRC assemblies.</p>

  <br />
  <h3 id="cache_content">Data in the cache</h3>

  <p>The data content of Ensembl VEP caches vary by species. This table shows the contents of the default human cache files in release [[SPECIESDEFS::ENSEMBL_VERSION]].</p>

  <table class="ss" style="width:auto">
    <thead>
      <tr><th>Source</th><th>Version (GRCh38)</th><th>Version (GRCh37)</th></tr>
    </thead>
    <tbody>
      <tr>
        <td><b>Ensembl database version</b></td>
        <td>[[SPECIESDEFS::ENSEMBL_VERSION]]</td>
        <td>[[SPECIESDEFS::ENSEMBL_VERSION]]</td>
      </tr>
      <tr>
        <td><b>Genome assembly</b></td>
        <td>GRCh38.p14</td>
        <td>GRCh37.p13</td>
      </tr>
      <tr>
        <td><b>MANE Version</b></td>
        <td>v1.4</td>
        <td>n/a</td>
      </tr>
      <tr>
        <td><b>GENCODE</b></td>
        <td>49</td>
        <td>19</td>
      </tr>
      <tr>
        <td><b>RefSeq</b></td>
        <td>GCF_000001405.40-RS_2024_08<br />(GCF_000001405.40_GRCh38.p14_genomic.gff)</td>
        <td>105.20220307<br />(GCF_000001405.25_GRCh37.p13_genomic.gff)</td>
      </tr>
      <tr>
        <td><b>Regulatory build</b></td>
        <td>1.0</td>
        <td>1.0</td>
      </tr>
      <tr>
        <td><b>PolyPhen-2</b></td>
        <td>2.2.3</td>
        <td>2.2.2</td>
      </tr>
      <tr>
        <td><b>SIFT</b></td>
        <td>6.2.1</td>
        <td>5.2.2</td>
      </tr>
      <tr>
        <td><b>dbSNP</b></td>
        <td>156</td>
        <td>156</td>
      </tr>
      <tr>
        <td><b>COSMIC</b></td>
        <td>101</td>
        <td>98</td>
      </tr>
      <tr>
        <td><b>HGMD-PUBLIC</b></td>
        <td>2020.4</td>
        <td>2020.4</td>
      </tr>
      <tr>
        <td><b>ClinVar</b></td>
        <td>2025-02</td>
        <td>2023-06</td>
      </tr>
      <tr>
        <td><b>1000 Genomes</b></td>
        <td>Phase 3 (remapped)</td>
        <td>Phase 3</td>
      </tr>
      <tr>
        <td><b>gnomAD exomes</b></td>
        <td>v4.1</td>
        <td>v4.1</td>
      </tr>
      <tr>
        <td><b>gnomAD genomes</b></td>
        <td>v4.1</td>
        <td>v4.1</td>
      </tr>
    </tbody>
  </table>


  <div style="width:auto" class="warning">
  
    <h3 id="limitations">Limitations of the cache</h3>

    <div class="message-pad">
      <p> The cache <b>stores</b> the following information: </p>

      <ul>
        <li>Transcript location, sequence, exons and other attributes</li>
        <li>Gene, protein, HGNC and other identifiers for each transcript (where
        applicable, <a href="vep_other.html#refseq_limitations">limitations apply to RefSeq caches</a>)</li>
        <li>Locations, alleles and frequencies of <a href="#existing_variants">existing variants</a> (see note below).</li>
        <li>Regulatory regions</li>
        <li>Predictions and scores for SIFT, PolyPhen-2</li>
      </ul>

      <p> The cache <b>does not store</b> any information pertaining to, and therefore cannot be
      used for, the following: </p>

      <ul>
        <li>HGVS names (<a href="vep_options.html#opt_hgvs">--hgvs</a>, <a href="vep_options.html#opt_hgvsg">--hgvsg</a>) - to retrieve these you must additionally point
        to a <a href="#fasta">FASTA</a> file containing the reference sequence for your species
        (<a href="vep_options.html#opt_fasta">--fasta</a>)</li>
        <li>Using HGVS notation as input (<a href="vep_options.html#opt_format">--format hgvs</a>)</li>
        <li>Using variant identifiers as input (<a href="vep_options.html#opt_format">--format id</a>)</li>
        <li>Finding overlapping structural variants (<a href="vep_options.html#opt_check_sv">--check_sv</a>)</li>
      </ul>

      <p> Enabling one of these options with <a href="vep_options.html#opt_cache">--cache</a> will cause Ensembl VEP
      to warn you in its status output with something like the following: </p>

      <pre class="code sh_sh"> 2011-06-16 16:24:51 - INFO: Database will be accessed when using --hgvs </pre>
    </div>
  </div>

      
  <div style="width:auto" class="info">
    <h3 id="existing_variants">Existing variants</h3>
    <div class="message-pad">
      <p>Here <b>existing variants</b> referes to those variants that have been loaded to <a href="#database">Ensembl variation database</a> from accessioning resources. 
        For example, for human, you can see the source of data in the above table. We load variants from accessioning resources such as dbSNP, COSMIC, and HGMD-PUBLIC. 
      </p>
      <p>
        Note that <b>gnomAD</b> is not a variant accessioning body. What it means is that any gnomAD variant that are not accessioned will not be avialable in the
        cache. For example, gnomAD v4.1 was released in April 2024, but will not be available in the cache until the variants have been submitted to dbSNP for accessioning 
        and made available in a dbSNP release. If you run the variant <a href="https://gnomad.broadinstitute.org/variant/5-32100960-ATAAG-A?dataset=gnomad_r4">5-32100960-ATAAG-A</a> 
        using 113 cache you would not get any frequency information because it was not accessioned at the time of Ensembl 113 release -
      </p>
      <pre class="code sh_sh">./vep --id "5 32100960 . ATAAG A" --af_gnomadg --af_gnomade --check_existing --cache --cache_version 113 --fasta genome.fa.gz
#Uploaded_variation	Location	Allele	Gene	Feature	Feature_type	Consequence	cDNA_position	CDS_position	Protein_position	Amino_acids	Codons	Existing_variation	Extra
5_32100961_TAAG/-	5:32100961-32100964	-	ENSG00000133401	ENST00000397559	Transcript	splice_donor_variant,non_coding_transcript_exon_variant	95-?	-	-	-	-	-	IMPACT=HIGH;STRAND=1</pre>
        
      <p><b>Alternative:</b> In such cases you can use gnomAD VCF file with <a href="vep_custom.html">--custom</a> option.</p>
    </div>
  </div>

  <br />
  <h3 id="offline">Data privacy and offline mode</h3>

  <p>When using the public database servers, Ensembl VEP requests
    transcript and variation data that overlap the loci in your input file. As
    such, these coordinates are transmitted over the network to a public server,
    which may not be appropriate for the analysis of sensitive or private data.
  </p>
  <div>
    <div style="float:left" class="info">
      <h3 id="read_write_access">Note</h3>
      <div class="message-pad">
        <p><b>Only</b> the <b>coordinates</b> are transmitted to the server; no other information is sent.</p>
      </div>
    </div>
    <div class="clear"></div>
  </div>

  <p>To use offline mode that does not use any network connections, use the flag <a href="vep_options.html#opt_offline">--offline</a>.</p>

  <p> The <a href="#limitations">limitations</a> described above apply
    absolutely when using offline mode. For example, if you specify <a
    href="vep_options.html#opt_offline">--offline</a> and <a
    href="vep_options.html#opt_format">--format id</a>, Ensembl VEP will report
    an error and refuse to run: </p>

  <pre class="code sh_sh">ERROR: Cannot use ID format in offline mode</pre>

  <p> All other features, including the ability to use <a
    href="vep_custom.html">custom annotations</a> and <a
    href="vep_plugins.html">plugins</a>, are accessible in offline mode. </p>


  <br/>
  <hr/>
  <h2 id="gff">GFF/GTF files</h2>

  <p>Ensembl VEP can use transcript annotations defined in <a href="https://github.com/The-Sequence-Ontology/Specifications/blob/master/gff3.md" rel="external">GFF</a> or <a href="/info/website/upload/gff.html">GTF</a> files. The files must be bgzipped and indexed with tabix and a <a href="#fasta">FASTA</a> file containing the genomic sequence is required in order to generate transcript models. This allows you to annotate variants from any species and assembly with these data. </p>

  <p> Your GFF or GTF file must be sorted in chromosomal order. Ensembl VEP does not use header lines so it is safe to remove them.</p>

  <pre class="code sh_sh">grep -v "#" data.gff | sort -k1,1 -k4,4n -k5,5n -t$'\t' | bgzip -c > data.gff.gz
tabix -p gff data.gff.gz
./vep -i input.vcf --gff data.gff.gz --fasta genome.fa.gz</pre>

  <p>You may use any number of GFF/GTF files in this way, providing they refer to the same genome. You may also use them in concert with annotations from a cache or database source; annotations are distinguished by the SOURCE field in the output.</p>

  <ul>
    <li>
      <p><b>GFF file</b></p>
      <p>Example of command line with GFF, using flag <a href="vep_options.html#opt_gff">--gff</a> :</p>
      <pre class="code sh_sh">./vep -i input.vcf --cache --gff data.gff.gz --fasta genome.fa.gz</pre>

      <p><b>NOTE:</b> If you wish to customise the name of the GFF as it appears in the SOURCE field and Ensembl VEP output header, use the <a href="vep_custom.html#custom_options">longer <kbd>--custom</kbd> annotation form</a>:</p>

      <pre class="code sh_sh">--custom file=data.gff.gz,short_name=frequency,format=gff</pre>
    </li>

    <li style="margin-top:2em">
      <p><b>GTF file</b></p>
      <p>Example of command line with GTF, using flag <a href="vep_options.html#opt_gtf">--gtf</a> :</p>
      <pre class="code sh_sh">./vep -i input.vcf --cache --gtf data.gtf.gz --fasta genome.fa.gz</pre>

      <p><b>NOTE:</b> If you wish to customise the name of the GFF as it appears in the SOURCE field and Ensembl VEP output header, use the <a href="vep_custom.html#custom_options">longer <kbd>--custom</kbd> annotation form</a>:</p>

      <pre class="code sh_sh">--custom file=data.gtf.gz,short_name=frequency,format=gtf</pre>
    </li>
  </ul>


  <br/>
  <h3 id="gff_req">GFF format expectations</h3>

  <p>Ensembl VEP has been tested on GFF files generated by Ensembl and NCBI (RefSeq). Due to inconsistency in the GFF specification and adherence to it, not all GFF files will be compatible with Ensembl VEP and not all transcript biotypes may be supported. Additionally, Ensembl VEP does not support GFF files with embedded FASTA sequence.</p>

  <br />
  <p id="gff_type"><b>Column "type" (3rd column):</b></p>

  <p> The following entity/feature types are supported by Ensembl VEP.</p>

  <p>
    <a class="button" href="#gfftypes" onclick="show_hide('gfftypes');" id="a_gfftypes">Show supported types</a>
  </p>

  <div id="div_gfftypes" style="display:none;">
    <div class="column-left" style="float:left">
      <ul>
        <li>aberrant_processed_transcript</li>
        <li>CDS</li>
        <li>C_gene_segment</li>
        <li>D_gene_segment</li>
        <li>exon</li>
        <li>gene</li>
        <li>J_gene_segment</li>
        <li>lincRNA</li>
        <li>lincRNA_gene</li>
        <li>miRNA</li>
        <li>miRNA_gene</li>
        <li>mRNA</li>
        <li>mt_gene</li>
        <li>ncRNA</li>
        <li>NMD_transcript_variant</li>
        <li>primary_transcript</li>
      </ul>
    </div>
    <div class="column-right" style="float:left">
      <ul>
        <li>processed_pseudogene</li>
        <li>processed_transcript</li>
        <li>pseudogene</li>
        <li>pseudogenic_transcript</li>
        <li>RNA</li>
        <li>rRNA</li>
        <li>rRNA_gene</li>
        <li>snoRNA</li>
        <li>snoRNA_gene</li>
        <li>snRNA</li>
        <li>snRNA_gene</li>
        <li>supercontig</li>
        <li>transcript</li>
        <li>tRNA</li>
        <li>VD_gene_segment</li>
        <li>V_gene_segment</li>
      </ul>
    </div>
    <div class="clear"></div>
  </div>

  <p>Lines of other types will be ignored; if this leads to an incomplete transcript model, the whole transcript model may be discarded. If unsupported types are used you will see a warning like the following -</p>
  <pre class="code sh_sh sh_sourceCode"><span class="sh_variable">WARNING: </span>Ignoring 'five_prime_utr' feature_type from Homo_sapiens.GRCh38.111.gtf.gz GFF/GTF file. This feature_type is not supported in Ensembl VEP.</span></pre>

  <br />
  <p id="gff_9th"><b>Expected parameters in the 9th column:</b></p>
  <ul>
    <li><b>ID</b><br />
      <p>Only required for the genes and transcripts entities.</p>
    </li>
    <li><b>parent</b>/<b>Parent</b><br />
      <p>
        - Entities in the GFF are expected to be linked using a key named "<b>parent</b>" or "<b>Parent</b>" in the attributes (9th) column of the GFF.<br />
        - Unlinked entities (i.e. those with no parents <b>or</b> children) are discarded.<br />
        - Sibling entities (those that share the same parent) may have overlapping coordinates, e.g. for exon and CDS entities.
      </p>
    </li>
    <li><b>biotype</b><br />
      <p>Transcripts require a Sequence Ontology biotype to be defined in order to be used.<br />
         The simplest way to define this is using an attribute named "<b>biotype</b>" on the transcript entity.
         Other configurations are supported in order for Ensembl VEP to use GFF files from NCBI and other sources.
      </p>
    </li>
  </ul>
  <p>Here is an example:</p>
  <pre class="code sh_sh">
##gff-version 3.2.1
##sequence-region 1 1 10000
1 Ensembl gene        1000  5000  . + . ID=gene1;Name=GENE1
1 Ensembl transcript  1100  4900  . + . ID=transcript1;Name=GENE1-001;Parent=gene1;biotype=protein_coding
1 Ensembl exon        1200  1300  . + . ID=exon1;Name=GENE1-001_1;Parent=transcript1
1 Ensembl exon        1500  3000  . + . ID=exon2;Name=GENE1-001_2;Parent=transcript1
1 Ensembl exon        3500  4000  . + . ID=exon3;Name=GENE1-001_2;Parent=transcript1
1 Ensembl CDS         1300  3800  . + . ID=cds1;Name=CDS0001;Parent=transcript1</pre>


  <br />
  <h3 id="gtf_req">GTF format expectations</h3>

  <p> The following GTF entity types will be extracted:</p>

  <ul>
    <li>cds (or CDS)</li>
    <li>stop_codon</li>
    <li>exon</li>
    <li>gene</li>
    <li>transcript</li>
  </ul>

  <p> Entities are linked by an attribute named for the <b>parent</b> entity type e.g. exon is linked to transcript by transcript_id, transcript is linked to gene by gene_id. </p>

  <p> Transcript biotypes are defined in attributes named "<b>biotype</b>", "<b>transcript_biotype</b>" or "<b>transcript_type</b>". If none of these exist, Ensembl VEP will attempt to interpret the source field (2nd column) of the GTF as the biotype. </p>

  <p>Here is an example:</p>
  <pre class="code sh_sh">
1 Ensembl gene        1000  5000  . + . gene_id "gene1"; gene_name "GENE1";
1 Ensembl transcript  1100  4900  . + . gene_id "gene1"; transcript_id "transcript1"; gene_name "GENE1"; transcript_name "GENE1-001"; transcript_biotype "protein_coding";
1 Ensembl exon        1200  1300  . + . gene_id "gene1"; transcript_id "transcript1"; exon_number "exon1"; exon_id "GENE1-001_1";
1 Ensembl exon        1500  3000  . + . gene_id "gene1"; transcript_id "transcript1"; exon_number "exon2"; exon_id "GENE1-001_2";
1 Ensembl exon        3500  4000  . + . gene_id "gene1"; transcript_id "transcript1"; exon_number "exon3"; exon_id "GENE1-001_2";
1 Ensembl CDS         1300  3800  . + . gene_id "gene1"; transcript_id "transcript1"; exon_number "exon2"; ccds_id "CDS0001";</pre>

  <br />
  <h3 id="gff_synonyms">Chromosome synonyms</h3>

  <p> If the chromosome names used in your GFF/GTF differ from those used in the FASTA or your input VCF, you may see warnings like this when running Ensembl VEP:</p>

  <pre class="code sh_sh">WARNING: Chromosome 21 not found in annotation sources or synonyms on line 160</pre>

  <p> To circumvent this you may provide Ensembl VEP with a <a href="vep_options.html#opt_synonyms">synonyms file</a>. A synonym file is included in Ensembl VEP's cache files, so if you have one of these for your species you can use it as follows:</p>

  <pre class="code sh_sh">./vep -i input.vcf -cache -gff data.gff.gz -fasta genome.fa.gz -synonyms ~/.vep/homo_sapiens/[[SPECIESDEFS::ENSEMBL_VERSION]]_GRCh38/chr_synonyms.txt</pre>

  <br />
  <div style="width:auto" class="warning">

    <h3 id="gff_limitations">Limitations of the cache</h3>

    <div class="message-pad">

      <p> Using a GFF or GTF file as the gene annotation source limits access to some auxiliary information available when using a <a href="#cache">cache</a>. Currently most external reference data such as gene symbols, transcript identifiers and protein domains are inaccessible when using only a GFF/GTF file.</p>

      <p> Ensembl VEP's flexibility does allow some annotation types to be replaced. The following table illustrates some examples and alternative means to retrieve equivalent data. </p>

      <table class="ss" style="width:auto">
        <thead>
          <tr>
            <th style="min-width:320px">Data type</th><th style="min-width:500px">Alternative</th>
          </tr>
        </thead>
        <tbody>
          <tr>
           <td>SIFT and PolyPhen-2 predictions (<a href="vep_options.html#opt_sift">--sift</a>, <a href="vep_options.html#opt_polyphen">--polyphen</a>)</td>
            <td>Use the <a href="vep_plugins.html#PolyPhen_SIFT">PolyPhen_SIFT plugin</a></td>
          </tr>
          <tr>
            <td>Co-located variants (<a href="vep_options.html#opt_check_existing">--check_existing</a>, --af* flags)</td>
            <td>A couple of options are available:
              <ol>
                <li>Use a <a href="vep_custom.html">VCF with --custom</a> to retrieve variant IDs, frequency and other data</li>
                <li>Add <a href="vep_options.html#opt_cache">--cache</a> to use variants in the <a href="#cache">cache</a>. <b>*</b></li>
              </ol>
            </td>
          </tr>
          <tr>
            <td>Regulatory consequences (<a href="vep_options.html#opt_regulatory">--regulatory</a>)</td>
            <td>
              Add <a href="vep_options.html#opt_cache">--cache</a> to use regulatory features in the <a href="#cache">cache</a>. <b>*</b>
            </td>
          </tr>
        </tbody>
      </table>

      <p><b>*</b> Note this will also instruct Ensembl VEP to annotate input variants against transcript models retrieved from the cache <b>as well as</b> those from the GFF/GTF file. It is possible to use <a href="vep_options.html#opt_transcript_filter">--transcript_filter</a> to include only the transcripts from your GFF/GTF file:</p>

      <pre class="code sh_sh">./vep -i input.vcf -cache --custom file=data.gff.gz,short_name=myGFF,format=gff --fasta genome.fa.gz --transcript_filter "_source_cache is myGFF"</pre>
    </div>
  </div>


  <br/>
  <hr/>
  <h2 id="fasta">FASTA files</h2>

  <p> By pointing Ensembl VEP to a FASTA file (or directory containing several
  files), it is possible to retrieve reference sequence locally when using
  <a href="vep_options.html#opt_cache">--cache</a> or <a
  href="vep_options.html#opt_offline">--offline</a>. This enables Ensembl VEP to:
  <ul>
    <li> Retrieve HGVS notations (<a href="vep_options.html#opt_hgvs">--hgvs</a>) </li>
    <li> Check the reference sequence given in input data (<a
    href="vep_options.html#opt_check_ref">--check_ref</a>) </li>
    <li> Construct
    transcript models from a GFF or GTF file without accessing a database
    (specially useful for performance reasons or if using data from
    species/assembly not part of
    <a href="/info/about/species.html">Ensembl species list</a>)</li>
  </ul></p>

  <p> FASTA files from Ensembl can be set up using the <a
  href="vep_download.html#installer">installer</a>; files set up using the
  installer are automatically detected when using <a
  href="vep_options.html#opt_cache">--cache</a> or <a
  href="vep_options.html#opt_offline">--offline</a>; you should not need to
  use <a href="vep_options.html#opt_fasta">--fasta</a> to manually specify
  them. </p>

  <p>
    The following plugins do require the fasta file to be explicitly passed as a command line argument (i.e. <kbd>--fasta /VEP_DIR/your_downloaded.fasta</kbd>)
    <ul>
      <li>CSN</li>
      <li>GeneSplicer</li>
      <li>MaxEntScan</li>
    </ul>
  </p>

  <p>To enable this, Ensembl VEP uses one of two modules:</p>
  <ul>
    <li>
      The <a rel="external" href="https://github.com/Ensembl/Bio-DB-HTS">Bio::DB::HTS</a> Perl XS
      module with <a href="http://www.htslib.org">HTSlib</a>. This module
      uses compiled C code and can access compressed
      (bgzipped) or uncompressed FASTA files. It is set up by the <a
      href="vep_download.html#installer">installer</a>.
    </li>
    <li>
      The <a rel="external" href="http://search.cpan.org/~cjfields/BioPerl-1.6.924/Bio/DB/Fasta.pm">Bio::DB::Fasta</a>
      module. This may be used on systems where installation of the Bio::DB::HTS
      module has not been possible. It can access only uncompressed FASTA
      files. It is also set up by the installer
      and comes as part of the BioPerl package.
    </li>
  </ul>

  <p> The first time you run Ensembl VEP
  with a specific FASTA file, an index will be built. This can take a few
  minutes, depending on the size of the FASTA file and the speed of your
  system. On subsequent runs the index does not need to be rebuilt (if the
  FASTA file has been modified, Ensembl VEP will force a rebuild of the index).
  </p>

  <br />
  <p><img src="/i/16/download.png" style="vertical-align:bottom"/> <b>FASTA FTP directories</b></p>

  <p> Suitable reference FASTA files are available to download from the Ensembl
  FTP server. See the <a href="/info/data/ftp/index.html">Downloads</a> page
  for details. </p>

  <p>You should preferably use the installer as described above to
  fetch these files; manual instructions are provided for reference.
  In most cases it is best to download the single large
  "primary_assembly" file for your species. You should use the unmasked
  (without <kbd>_rm</kbd> or <kbd>_sm</kbd> in the name) sequences. </p>

  <p> Note that Ensembl VEP requires that the file be either unzipped
  (<kbd>Bio::DB::Fasta</kbd>) or unzipped and then recompressed with bgzip
  (<kbd>Bio::DB::HTS::Faidx</kbd>) to run; when unzipped these files can be very
  large (25GB for human). An example set of commands for setting up the data for
  human follows: </p>

  <pre class="code sh_sh">curl -O [[SPECIESDEFS::ENSEMBL_FTP_URL]]/release-[[SPECIESDEFS::ENSEMBL_VERSION]]/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.primary_assembly.fa.gz
gzip -d Homo_sapiens.GRCh38.dna.primary_assembly.fa.gz
bgzip Homo_sapiens.GRCh38.dna.primary_assembly.fa
./vep -i input.vcf --offline --hgvs --fasta Homo_sapiens.GRCh38.dna.primary_assembly.fa.gz</pre>


  <br/>
  <hr/>
  <h2 id="database">Databases</h2>

  <p> Ensembl VEP can use remote or local database servers to retrieve annotations. </p>

  <ul>
    <li>Using <a href="vep_options.html#opt_cache">--cache</a> (without <a href="vep_options.html#opt_offline">--offline</a>) uses the local cache on disk to fetch most annotations, but allows database connections for some features (see <a href="#limitations">cache limitations</a>)</li>
    <li>Using <a href="vep_options.html#opt_database">--database</a> tells Ensembl VEP to retrieve <b>all</b> annotations from the database. <b>Please only use this for small input files or when using a local database server!</b></li>
  </ul>


  <br />
  <h3 id="public">Public database servers</h3>

  <p> By default, Ensembl VEP is configured to connect to the public
  MySQL instance at ensembldb.ensembl.org. If you are in the USA (or
  geographically closer to the east coast of the USA than to the Ensembl
  data centre in Cambridge, UK), a mirror server is available at
  useastdb.ensembl.org. To use the mirror, use the flag <code><a
  href="vep_options.html#opt_host">--host</a> useastdb.ensembl.org</code> </p>

  <p> Data for Ensembl Genomes species (e.g. plants, fungi, microbes) is available
  through a different public MySQL server. The appropriate connection parameters can be
  automatically loaded by using the flag <a
  href="vep_options.html#opt_genomes">--genomes</a> </p>

  <p> If you have a very small data set (100s of variants), using the public database
  servers should provide adequate performance. If you have larger data sets, or
  wish to use Ensembl VEP in a batch manner, consider one of the alternatives below. </p>


  <br />
  <h3 id="local">Using a local database</h3>

    <p> It is possible to set up a local MySQL mirror with the databases for
    your species of interest installed. For instructions on installing a local
    mirror, see <a href="/info/docs/webcode/mirror/install/ensembl-data.html"
    target="_blank">here</a>. You will need a MySQL server that you can connect
    to from the machine where you will run Ensembl VEP (this can be the same
    machine). For most annotation functionality, you will only need the
    Core database (e.g. homo_sapiens_core_[[SPECIESDEFS::ENSEMBL_VERSION]]_38)
    installed. In order to find co-located variants or to use SIFT or
    PolyPhen-2, it is also necessary to install the relevant variation database
    (e.g. homo_sapiens_variation_[[SPECIESDEFS::ENSEMBL_VERSION]]_38). </p>

    <p> Note that unless you have custom data to insert in the database, in most
    cases it will be much more efficient to use a <a href="#cache">pre-built
    cache</a> in place of a local database. </p>

    <p> To connect to your mirror, you can either set the connection parameters
    using <a href="vep_options.html#opt_host">--host</a>, <a
    href="vep_options.html#opt_port">--port</a>, <a
    href="vep_options.html#opt_user">--user</a> and <a
    href="vep_options.html#opt_password">--password</a>, or use a registry file.
    Registry files contain all the connection parameters for your database, as
    well as any species aliases you wish to set up: </p>

    <pre class="code sh_perl sh_sourceCode">
use Bio::EnsEMBL::DBSQL::DBAdaptor;
use Bio::EnsEMBL::Variation::DBSQL::DBAdaptor;
use Bio::EnsEMBL::Registry;

Bio::EnsEMBL::DBSQL::DBAdaptor->new(
  '-species' => "Homo_sapiens",
  '-group'   => "core",
  '-port'    => 5306,
  '-host'    => 'ensembldb.ensembl.org',
  '-user'    => 'anonymous',
  '-pass'    => '',
  '-dbname'  => 'homo_sapiens_core_[[SPECIESDEFS::ENSEMBL_VERSION]]_38'
);

Bio::EnsEMBL::Variation::DBSQL::DBAdaptor->new(
  '-species' => "Homo_sapiens",
  '-group'   => "variation",
  '-port'    => 5306,
  '-host'    => 'ensembldb.ensembl.org',
  '-user'    => 'anonymous',
  '-pass'    => '',
  '-dbname'  => 'homo_sapiens_variation_[[SPECIESDEFS::ENSEMBL_VERSION]]_38'
);

Bio::EnsEMBL::Registry->add_alias("Homo_sapiens","human");</pre>

  <p> For more information on the registry and registry files, see <a 
  href="/info/docs/api/registry.html" target="_blank">here</a>. </p>


  <br/>
  <hr/>
  <h2 id="technical">Cache - technical information</h2>

  <p> <span style="color:red;">ADVANCED</span> The cache consists of
    compressed files containing listrefs of serialised objects. These objects
    are initially created from the database as if using the Ensembl API
    normally. In order to reduce the size of the cache and allow the
    serialisation to occur, some changes are made to the objects before they are
    dumped to disk. This means that they will not behave in exactly the same way
    as an object retrieved from the database when writing, for example, a plugin
    that uses the cache. </p>

  <p>The following hash keys are deleted from each transcript object: </p>

  <ul>
    <li><b>analysis</b></li>
    <li><b>created_date</b></li>
    <li><b>dbentries</b> : this contains the external references retrieved
      when calling $transcript->get_all_DBEntries(); hence this call
      on a cached object will return no entries</li>
    <li><b>description</b></li>
    <li><b>display_xref</b></li>
    <li><b>edits_enabled</b></li>
    <li><b>external_db</b></li>
    <li><b>external_display_name</b></li>
    <li><b>external_name</b></li>
    <li><b>external_status</b></li>
    <li><b>is_current</b></li>
    <li><b>modified_date</b></li>
    <li><b>status</b></li>
    <li><b>transcript_mapper</b> : used to convert between genomic, cdna,
	cds and protein coordinates. A copy of this is cached separately by Ensembl VEP as
        <p>
	  <pre class="code sh_perl sh_sourceCode">$transcript->{_variation_effect_feature_cache}->{mapper}</pre>
        </p>
    </li>
  </ul>

  <p> As mentioned above, a special hash key "_variation_effect_feature_cache"
    is created on the transcript object and used to cache things used by Ensembl VEP
    in predicting consequences, things which might otherwise have to be fetched
    from the database. Some of these are stored in place of equivalent keys that
    are deleted as described above. The following keys and data are stored: </p>

  <ul>
    <li><b>introns</b> : listref of intron objects for the transcript. The adaptor,
        analysis, dbID, next, prev and seqname keys are stripped from each intron object</li>
    <li><b>translateable_seq</b> : as returned by
	<p><pre class="code sh_perl sh_sourceCode">$transcript->translateable_seq</pre></p></li>
    <li><b>mapper</b> : transcript mapper as described above</li>
    <li><b>peptide</b> : the translated sequence as a string, as returned by
	<p><pre class="code sh_perl sh_sourceCode">$transcript->translate->seq</pre></p></li>
    <li><b>protein_features</b> : protein domains for the transcript's translation as returned by
        <p><pre class="code sh_perl sh_sourceCode">$transcript->translation->get_all_ProteinFeatures</pre></p>
	Each protein feature is stripped of all keys but: start, end, analysis, hseqname</li>
    <li><b>codon_table</b> : the codon table ID used to translate the transcript, as returned by
	<p><pre class="code sh_perl sh_sourceCode">$transcript->slice->get_all_Attributes('codon_table')->[0]</pre></p></li>
    <li><b>protein_function_predictions</b> : a hashref containing the keys "sift"
	and "polyphen"; each one contains a protein function prediction matrix as returned by e.g.
	<p><pre class="code sh_perl sh_sourceCode">$protein_function_prediction_matrix_adaptor->fetch_by_analysis_translation_md5('sift', md5_hex($transcript-{_variation_effect_feature_cache}->{peptide}))</pre></p></li>
  </ul>

  <p>Similarly, some further data is cached directly on the transcript object under the following keys:</p>

  <ul>
    <li><b>_gene</b> : gene object. This object has all keys but the following deleted: start, end, strand, stable_id</li>
    <li><b>_gene_symbol</b> : the gene symbol</li>
    <li><b>_ccds</b> : the CCDS identifier for the transcript</li>
    <li><b>_refseq</b> : the "NM" RefSeq mRNA identifier for the transcript</li>
    <li><b>_protein</b> : the Ensembl stable identifier of the translation</li>
    <li><b>_source_cache</b> : the source of the transcript object. Only defined in the merged cache (values: Ensembl, RefSeq) or when using a GFF/GTF file (value: short name or filename)</li>
  </ul>
</div>

<div style="clear:both"><h1 style="page-break-after: always">&nbsp</h1></div>    
<style>
tr:nth-child(odd) {background-color: #f0f0f0;}
</style>

<div>
	
  <div style="float:right"><img src="/img/vep_logo.png"/></div>
	
  <h1 id="top"><span style="color:#006;padding-right:15px"> Ensembl Variant Effect Predictor</span><span style="color:#666"><img src="/i/16/search.png"/> Filtering results</span></h1>
  <hr/>
    
		
	
    <p> The Ensembl VEP package includes a tool, filter_vep, to filter results files on a variety of attributes.</p>
    
    <p> It operates on standard, tab-delimited or VCF formatted output (NB only VCF
    output produced by Ensembl VEP or in the same format can be used). </p>
    
    <h2 id="filter_run">Running filter_vep</h2>
    
    <p> Run as follows: </p>
    
    <pre class="code sh_sh">./vep -i in.vcf -o out.txt -cache -everything
./filter_vep -i out.txt -o out_filtered.txt -filter "[filter_text]"</pre>
    
    <p> filter_vep can also read from STDIN and write to STDOUT, and so may be
    used in a UNIX pipe: </p>
    
    <pre class="code sh_sh">./vep -i in.vcf -o stdout -cache -check_existing | ./filter_vep -filter "not Existing_variation" -o out.txt</pre>
    
    <p> The above command removes known variants from the output </p>
    
    <hr/>
    <h2 id="filter_opt">Options</h2>
    
	<table class="ss" style="width:75%;">
		<tr>
			<th>Flag</th>
			<th>Alternate</th>
			<th>Description</th>
		</tr>
		<tr>
			<td><pre>--help</pre></td>
			<td><pre>-h</pre></td>
			<td>Print usage message and exit</td>
		</tr>
		<tr>
			<td><pre>--input_file [file]</pre></td>
			<td><pre>-i</pre></td>
			<td>
                Specify the input file (i.e. the Ensembl VEP results file). If no input
                file is specified, filter_vep will attempt to read from STDIN.
                Input may be gzipped - to read a gzipped file 
                use <code>--gz</code>
			</td>
		</tr>
		<tr>
			<td><pre>--format [format]</pre></td>
			<td>&nbsp;</td>
			<td>
        <p>Specify input file format:</p>
        <ul>
          <li><b>tab</b> (i.e. the Ensembl VEP results file)</li>
          <li><b>vcf</b></li>
        </ul>
			</td>
		</tr>
		<tr>
			<td><pre>--output_file [file]</pre></td>
			<td><pre>-o</pre></td>
			<td>
                Specify the output file to write to. If no output file is
                specified, the filter_vep will write to STDOUT
			</td>
		</tr>
		<tr>
			<td><pre>--force_overwrite</pre></td>
			<td>&nbsp;</td>
			<td>
                Force an output file of the same name to be overwritten
			</td>
		</tr>
		<tr>
			<td><pre>--filter [filters]</pre></td>
			<td><pre>-f</pre></td>
			<td>
                Add filter (see below). Multiple <code>--filter</code> flags may be used, and
                are treated as logical ANDs, i.e. all filters must pass for a
                line to be printed
			</td>
		</tr>
    <tr id="opt_softfilter">
      <td><pre>--soft_filter</pre></td>
      <td></td>
      <td>
         Variants not passing given filters will be flagged in the FILTER column
         of the VCF file, and will not be removed from output.
      </td>
    </tr>
		<tr>
			<td><pre>--list</pre></td>
			<td><pre>-l</pre></td>
			<td>
                List allowed fields from the input file
			</td>
		</tr>
		<tr>
			<td><pre>--count</pre></td>
			<td><pre>-c</pre></td>
			<td>
                Print only a count of matched lines
			</td>
		</tr>
		<tr>
			<td><pre>--only_matched</pre></td>
			<td>&nbsp;</td>
			<td>
                In VCF files, the CSQ field that contains the consequence data
                will often contain more than one "block" of consequence data,
                where each block corresponds to a variant/feature overlap. Using
                <code>--only_matched</code> will remove blocks that do not pass the filters.
                By default, filter_vep prints out the entire VCF line if any of
                the blocks pass the filters.
			</td>
		</tr>
                <tr>
                  <td><pre>--vcf_info_field [key]</pre></td>
                  <td>&nbsp;</td>
                  <td>
                    <p>With VCF input files, by default filter_vep expects to find Ensembl VEP annotations encoded
                    in the CSQ INFO key; Ensembl VEP itself can be configured to write to
                    a different key (with the equivalent
                    <a href="vep_options.html#opt_vcf_info_field">--vcf_info_field</a>
                    flag).</p>
                    Use this flag to change the INFO key Ensembl VEP expects to decode:<br />
                    e.g. use the command "<code>--vcf_info_field ANN</code>" if the Ensembl VEP annotations are stored in the INFO key "ANN".
                  </td>
                </tr>		
                <tr>
			<td><pre>--ontology</pre></td>
			<td><pre>-y</pre></td>
			<td>
        Use <a rel="external"
href="http://www.sequenceontology.org/">Sequence Ontology</a> to match
consequence terms. Use with
        operator "is" to match against all child terms of your value. e.g.
        "Consequence is coding_sequence_variant" will match missense_variant,
        synonymous_variant etc. Requires database connection; defaults to
        connecting to ensembldb.ensembl.org. Use <code>--host</code>, <code>--port</code>, <code>--user</code>,
        <code>--password</code>, <code>--version</code> as per <code>vep</code> to change
        connection parameters.
			</td>
		</tr>
    </table>
    
    <hr/>
    <h2 id="filter_write">Writing filters</h2>
    
    <p> Filter strings consist of three components <b>that must be separated by whitespace</b>: </p>
    
    <ol>
        <li><b>Field</b> : A field name from the Ensembl VEP results file. This can be
        any field in the "main" columns of the output, or any in the "Extra"
        final column. For VCF files, this is any field defined in the
        "##INFO=&lt;ID=CSQ" header. You can list available fields using
        --list. Field names are not case sensitive, and you may use the first
        few characters of a field name if they resolve uniquely to one field name.</li>
        <li><b>Operator</b> : The operator defines the comparison carried out.
        </li>
        <li><b>Value</b> : The value to which the content of the field is
        compared. May be prefixed with "#" to represent the value of another field.</li>
    </ol>
    
    <p> Examples: </p>
    
    <pre class="code sh_sh"># match entries where Feature (Transcript) is "ENST00000307301"
--filter "Feature is ENST00000307301"

# match entries where Protein_position is less than 10
--filter "Protein_position < 10"

# match entries where Consequence contains "stream" (this will match upstream and downstream)
--filter "Consequence matches stream"</pre>
        
    <p> For certain fields you may only be interested in whether a value exists for that field;
    in this case the operator and value can be left out: </p>
    
    <pre class="code sh_sh"># filter for MANE transcripts
--filter "MANE"

# match entries where the gene symbol is defined
--filter "SYMBOL"</pre>

    <p> The value component may be another field; to represent this, prefix the name of the field to be used as a value with "#": </p>
    
    <pre class="code sh_sh"># match entries where AFR_AF is greater than EUR_AF
--filter "AFR_AF > #EUR_AF"</pre>
    
    <p> Filter strings can be linked together by the logical operators "or"
    and "and", and inverted by prefixing with "not": </p>
    
    <pre class="code sh_sh"># filter for missense variants in CCDS transcripts where the variant falls in a protein domain
--filter "Consequence is missense_variant and CCDS and DOMAINS"

# find variants where the allele frequency is greater than 10% in either AFR or EUR populations
--filter "AFR_AF > 0.1 or EUR_AF > 0.1"

# filter out known variants
--filter "not Existing_variation"</pre>

    <p> Filter logic may be constrained using parentheses, to any arbitrary level: </p>

    <pre class="code sh_sh"># find variants with AF > 0.1 in AFR or EUR but not EAS or SAS
--filter "(AFR_AF > 0.1 or EUR_AF > 0.1) and (EAS_AF < 0.1 and SAS_AF < 0.1)"</pre>
    
    <p> For fields that contain string and number components, filter_vep will
    try and match the relevant part based on the operator in use. For example,
    using <a href="vep_options.html#opt_sift">--sift b</a> in Ensembl VEP gives strings that look like
    "tolerated(0.46)". This will give a match to either of the following
    filters: </p>
    
    <pre class="code sh_sh"># match string part
--filter "SIFT is tolerated"

# match number part
--filter "SIFT < 0.5"</pre>

    <p> Note that for numeric fields, such as the *AF allele frequency fields, filter_vep does not consider the absence of a value for that field as equivalent to a 0 value.
    For example, if you wish to find rare variants by finding those where the allele frequency is less than 1% <b>or</b> absent, you should use the following:</p>

    <pre class="code sh_sh">--filter "AF < 0.01 or not AF"</pre>
    
    <p> For the Consequence field it is possible to use the <a rel="external"
    href="http://www.sequenceontology.org/">Sequence Ontology</a> to match terms
    ontologically; for example, to match all coding consequences (e.g.
    missense_variant, synonymous_variant):</p>
    
    <pre class="code sh_sh">--ontology --filter "Consequence is coding_sequence_variant"</pre>
    
    <hr/>
    <h2 id="filter_op">Operators</h2>
    
    <ul>
        <li><p><b>is</b> (synonyms: = , eq) : Match exactly</p>
        <pre class="code sh_sh"># get only transcript consequences
--filter "Feature_type is Transcript"</pre></li>
        <li><p><b>!=</b> (synonym: ne) : Does not match exactly</p>
        <pre class="code sh_sh"># filter out tolerated SIFT predictions
--filter "SIFT != tolerated"</pre></li>
        <li><p><b>match</b> (synonyms: matches , re , regex) : Match string using
        regular expression. You may include any regular expression notation,
        e.g. "\d" for any numerical character</p>
        <pre class="code sh_sh"># match stop_gained, stop_lost and stop_retained
--filter "Consequence match stop"</pre></li>
        <li><p><b>&lt;</b> (synonym: lt) : Less than. Note an absent value is not considered to be equivalent to 0.</p>
        <pre class="code sh_sh"># find SIFT scores less than 0.1
--filter "SIFT < 0.1"</pre></li>
        <li><p><b>&gt;</b> (synonym: gt) : Greater than</p>
        <pre class="code sh_sh"># find variants not in the first exon
--filter "Exon > 1"</pre></li>
        <li><b>&lt;=</b> (synonym: lte) : Less than or equal to. Note an absent value is not considered to be equivalent to 0.</li>
        <li><b>&gt;=</b> (synonym: gte) : Greater than or equal to</li>
        <li><b>exists</b> (synonyms: ex , defined) : Field is defined - equivalent to using no operator and value</li>
        <li><p><b>in</b> : Find in list or file. Value may be either a
        comma-separated list or a file containing values on separate lines. Each
        list item is compared using the "is" operator.</p>
        <pre class="code sh_sh"># find variants in a list of gene names
--filter "SYMBOL in BRCA1,BRCA2"

# filter using a file of MotifFeatures
--filter "Feature in /data/files/motifs_list.txt"</pre></li>
    </ul>
    
    
</div>

<div style="clear:both"><h1 style="page-break-after: always">&nbsp</h1></div><div>
  
  <div style="float:right"><img src="/img/vep_logo.png"/></div>
  
  <h1 id="top"><span style="color:#006;padding-right:15px">Ensembl Variant Effect Predictor</span><span style="color:#666"><img src="/i/16/page-user.png"> Custom annotations</span></h1>
  <hr/> 
        
  <p> Ensembl VEP can integrate custom annotation from standard format
  files into your results by using the <a href="vep_options.html#opt_custom">--custom</a> flag. </p>
    
  <p> These files may be hosted locally or remotely, with no limit to the
  number or size of the files. The files must be indexed using the <a
  rel="external" href="http://samtools.sourceforge.net/tabix.shtml" target="_blank">tabix</a>
  utility (BED, GFF, GTF, VCF); bigWig files contain their own indices. </p>
  
  <p> Annotations typically appear as key=value pairs in the Extra column of the VEP
  output; they will also appear in the INFO column if using VCF format output.
  The value for a particular annotation is defined as the identifier for each
  feature; if not available, an identifier derived from the coordinates of the
  annotation is used. Annotations will appear in each line of output for the
  variant where multiple lines exist. </p>
  
  <h2 id="custom_formats">Data formats</h2>
  
  <p> Ensembl VEP supports the following annotation formats: </p>
  
  <style> tr:nth-child(2n + 1) { background-color: #f0f0f0; } </style>
  <table class="ss"><tbody>
    <tr>
      <th>Format</th>
      <th>Type</th>
      <th>Description</th>
      <th>Notes</th>
    </tr>
    <tr id="opt_gff">
      <td><a href="vep_cache.html#gff">GFF<br>GTF</a></td>
      <td>Gene/transcript annotations & GENCODE promoters</td>
      <td>Formats to describe genes and other genomic features &mdash; format specifications: <a rel="external" href="https://github.com/The-Sequence-Ontology/Specifications/blob/master/gff3.md">GFF3</a> and <a href="/info/website/upload/gff.html">GTF</a></td>
      <td>Requires a <b><a href="vep_cache.html#fasta">FASTA</a></b> file in offline mode or if the desired species or assembly is not part of the <a href="/info/about/species.html">Ensembl species list</a>.</td>
    </tr>
    <tr id="opt_vcf">
      <td><a rel="external" href="http://www.1000genomes.org/wiki/Analysis/Variant%20Call%20Format/vcf-variant-call-format-version-41" target="_blank">VCF</a></td>
      <td>Variant data</td>
      <td>A format used to describe genomic variants</td>
      <td>Ensembl VEP uses the 3rd column as the identifier. INFO and FILTER fields from
          records may be added to the Ensembl VEP output.</td>
    </tr>
    <tr id="opt_bed">
      <td><a href="/info/website/upload/bed.html" target="_blank">BED</a></td>
      <td>Basic/uninterpreted data</td>
      <td>A simple tab-delimited format containing 3-12 columns of data.
          The first 3 columns contain the coordinates of the feature.</td>
      <td>Ensembl VEP uses the 4th column (if available) as the feature identifier.</td>
    </tr>
    <tr id="opt_bigWig">
      <td><a rel="external" href="http://genome.ucsc.edu/goldenPath/help/bigWig.html"
        target="_blank">bigWig</a></td>
      <td>Basic/uninterpreted data</td>
      <td>A format for storage of dense continuous data.</td>
      <td>Ensembl VEP uses the value for the given position as the identifier.
          BigWig files contain their own indices, and do not need to be indexed by tabix.
          Requires <a href="vep_download.html#bigfile">Bio::DB::BigFile</a>.
      </td>
    </tr>
  </tbody></table>

  <p> Any other files can be easily converted to be compatible with Ensembl VEP;
  the easiest format to produce is a BED-like file containing coordinates and
  an (optional) identifier: </p>
  
  <pre class="code sh_sh">chr1    10000    11000    Feature1
chr3    25000    26000    Feature2
chrX    99000    99001    Feature3</pre>
  
  <p> Chromosomes can be denoted by either e.g. "chr7" or "7", "chrX" or "X". </p>
  
    <hr/>
  <h2 id="custom_prep">Preparing files</h2>
  
  <p> Custom annotation files must be prepared in a particular way in order to
  work with tabix and therefore with Ensembl VEP. Files must be stripped of comment lines,
  sorted in chromosome and position order, compressed using bgzip
  and finally indexed using tabix. Here are some examples of that process for:</p>
  
  <ul>
    <li><b>GFF file</b>
      <pre class="code sh_sh" style="margin:5px 0px 10px">grep -v "#" myData.gff | sort -k1,1 -k4,4n -k5,5n -t$'\t' | bgzip -c > myData.gff.gz
tabix -p gff myData.gff.gz</pre>
      <div>
        <div class="warning" style="float:left">
          <h3>Note</h3>
          <div class="message-pad">
            <p>Ensembl VEP expects few extra fields in the 9th column of the GFF file. See the <a href="vep_cache.html#gff_req">extra fields list</a>.</p>
          </div>
        </div>
        <div style="clear:both"></div>
      </div>
    </li>
    <li><b>BED file</b>
  <pre class="code sh_sh" style="margin-top:5px">grep -v "#" myData.bed | sort -k1,1 -k2,2n -k3,3n -t$'\t' | bgzip -c > myData.bed.gz
tabix -p bed myData.bed.gz</pre>
  </ul>  

  <p> The tabix utility has several preset filetypes that it can process, and
  it can also process any arbitrary filetype containing at least a chromosome
  and position column. See the <a
  rel="external" href="http://samtools.sourceforge.net/tabix.shtml"
  target="_blank">documentation</a> for details. </p>
  
  <p> If you are going to use the file remotely (i.e. over HTTP or FTP
  protocol), you should ensure the file is world-readable on your server. </p>
  
  <hr/>
  <h2 id="custom_options">Options</h2>

<div class="navbar" onclick="show_hide('custom_compatibility');" id="custom_compatibility" style="padding-left:5px;margin-bottom:0px;cursor:pointer"><img id="img_custom_compatibility" src="/i/16/plus-button.png" style="vertical-align:middle"/> <span style="position:absolute;left:28px;top:9px">Using positional options in <kbd>--custom</kbd> with VEP 109 and earlier (compatible with VEP [[SPECIESDEFS::ENSEMBL_VERSION]])</span></div>
  <div id="div_custom_compatibility" style="display:none">

  <br/>
  <p> Each custom file that you configure Ensembl VEP to use can be configured.
  Beyond the filepath, there are further options, each of which is specified
  in a comma-separated list, like this: </p>

  <pre class="code sh_sh">
./vep [...] --custom <b>Filename</b>,<b>Short_name</b>,<b>File_type</b>,<b>Annotation_type</b>,<b>Force_report_coordinates</b>,<b>VCF_fields</b></pre>

  <p> The options are as follows: </p>
  
  <ul>
    <li><b>Filename</b> :
    <div style="margin:5px 0px 12px">The path to the file. For tabix indexed files, the
    Ensembl VEP will check that both the file and the corresponding .tbi file exist.
    For remote files, Ensembl VEP will check that the tabix index is accessible on startup.</div></li>

    <li><b>Short name</b> : 
    <div style="margin:5px 0px 12px">A name for the annotation that will appear as
    the key in the key=value pairs in the results.<br />If not defined, this will 
    default to the annotation filename for the first set of annotation added (e.g. "myPhenotypes.bed.gz" in the second example below if the short name was missing).</div></li>

    <li><b>File type</b> :
      <div style="margin:5px 0px 12px">
        <div style="float:left">
          <pre class="code sh_sh" style="margin:4px 0px">"bed", "gff", "gtf", "vcf" or "bigwig"</pre>
        </div>
        <div class="clear"></div>
      </div>
    </li>

    <li><b>Annotation type</b> :
      <div style="margin:5px 0px 12px">
        <div>
          <div style="float:left">
            <pre class="code sh_sh" style="margin:4px 0px">"exact" or "overlap"</pre>
          </div>
          <div style="float:left;padding:8px 10px;margin:4px 0px">
            (if left blank, assumed to be <b>overlap</b>) 
          </div>
          <div class="clear"></div>
        </div>
        <div>
          When using "<span style="color:blue">exact</span>" only annotations whose coordinates match exactly those of the
          variant will be reported. This would be suitable for position specific
          information such as conservation scores, allele frequencies or phenotype
          information. Using "<span style="color:blue">overlap</span>", any annotation that overlaps the variant
          by even 1bp will be reported.
        </div>
      </div>
    </li>
    
    <li><b>Force report coordinates</b> :
      <div style="margin:5px 0px 12px">
        <div>
          <div style="float:left">
            <pre class="code sh_sh" style="margin:4px 0px">"0" or "1"</pre>
          </div>
          <div style="float:left;padding:8px 10px;margin:4px 0px">
            (if left blank, assumed to be <b>0</b>)
          </div>
          <div class="clear"></div>
        </div>
        <div>If set to "1", this forces Ensembl VEP to output the
          coordinates of an overlapping custom feature instead of any found
          identifier (or value in the case of bigWig) field. If set to "0" (the
          default), Ensembl VEP will output the identifier field if one is found; if
          none is found, then the coordinates are used instead.
        </div>
      </div>
    </li>
      
    <li><b>VCF fields</b> :
      <div style="margin:5px 0px 12px">You can specify any info type (e.g. "AC") present in the INFO field of the custom input VCF or specify "FILTER" for the FILTER field, to add these as custom annotations:
        <ul>
          <li>If using "<span style="color:blue">exact</span>" annotation type, allele-specific annotation will be retrieved.</li>
          <li>The INFO field name will be prefixed with the short name, e.g.
              using short name "test", the INFO field "foo" will appear as "test_FOO" in the Ensembl VEP output. Similarly FILTER field will appear as "test_FILTER".</li>
          <li>In VCF files the custom annotations are added to the CSQ INFO field.</li>
          <li>Alleles in the input and VCF entry are trimmed in both directions in an attempt to match complex or poorly formatted entries.</li>
        </ul>
      </div>
    </li>
  </ul>

  <p>For example: </p>
  
  <pre class="code sh_sh">
    # BigWig file
    ./vep [...] --custom frequencies.bw,Frequency,bigwig,exact,0
    # BED file
    ./vep [...] --custom http://www.myserver.com/data/myPhenotypes.bed.gz,Phenotype,bed,exact,1
    # VCF file
    ./vep [...] --custom [[SPECIESDEFS::ENSEMBL_FTP_URL]]/data_files/homo_sapiens/GRCh37/variation_genotype/TOPMED_GRCh37.vcf.gz,,vcf,exact,0,TOPMED

    # For multiple custom files, use:
    ./vep [...] --custom clinvar.vcf.gz,ClinVar,vcf,exact,0,CLNSIG,CLNREVSTAT,CLNDN \
                --custom TOPMED_GRCh38_20180418.vcf.gz,topmed_20180418,vcf,exact,0,TOPMED \
                --custom UK10K_COHORT.20160215.sites.GRCh38.vcf.gz,uk10k,vcf,exact,0,AF_ALSPAC</pre>

  </div>

  <br/>
  <div class="navbar" onclick="show_hide('custom_options');" id="custom_options" style="padding-left:5px;margin-bottom:0px;cursor:pointer"><img id="img_custom_options" src="/i/16/minus-button.png" style="vertical-align:middle"/> <span style="position:absolute;left:28px;top:9px">Using key-value pairs in <kbd>--custom</kbd> with VEP [[SPECIESDEFS::ENSEMBL_VERSION]]</span></div>
  <div id="div_custom_options" style="display:inline">  
  <br />

  <p> Since Ensembl VEP 110, you can configure each custom file using a comma-separated
  list of key-value pairs: </p>
  
  <pre class="code sh_sh">
./vep [...] --custom <b>file=Filename</b>,<b>short_name=Short_name</b>,<b>format=File_type</b>,<b>type=Annotation_type</b>,<b>fields=VCF_fields</b></pre>
  
  <p> The order of the options is irrelevant and
  most options have sensible defaults as described below: </p>
  
  <table class="ss"><tbody>
    <tr>
      <th>Option</th>
      <th>Accepted values</th>
      <th>Description</th>
    </tr>
    <tr id="opt_file">
      <td><pre>file</pre></td>
      <td>String with valid path to file</td>
      <td>
        <b>(Required) Filename:</b>
        The path to the file. For Tabix indexed files, Ensembl VEP will check if both
        the file and the corresponding index (<kbd>.tbi</kbd>) exist.
        For remote files, Ensembl VEP will check that the tabix index is accessible on
        startup.
      </td>
    </tr>
    <tr id="opt_format">
      <td><pre>format</pre></td>
      <td><kbd>bed</kbd>, <kbd>gff</kbd>, <kbd>gtf</kbd>, <kbd>vcf</kbd> or <kbd>bigwig</kbd></td>
      <td>
        <b>(Required) File format</b> of <a href="#opt_file"><kbd>file</kbd></a>.
      </td>
    </tr>
    <tr id="opt_short_name">
      <td><pre>short_name</pre></td>
      <td>Annotation filename (default) or any string without commas</td>
      <td>
        <b>Short name:</b>
        A name for the annotation that will appear as
        the key in the key=value pairs in the results.
        
        If not defined, this will default to the annotation filename.
      </td>
    </tr>
    <tr id="opt_fields">
      <td><pre>fields</pre></td>
      <td></td>
      <td>
        <b>VCF fields:</b>
        Percentage (%) separated list of INFO fields to print (such as
        <kbd>AC</kbd>) present in the custom input VCF or specify
        <kbd>FILTER</kbd> for the FILTER field, to add these as custom
        annotations:
        <ul>
          <li>
            If using <kbd>exact</kbd> annotation type, allele-specific
            annotation will be retrieved.
          </li>
          <li>
            The INFO field name will be prefixed with the short name, e.g.
            using short name <kbd>test</kbd>, the INFO field <kbd>foo</kbd> will
            appear as <kbd>test_FOO</kbd> in the Ensembl VEP output. Similarly FILTER
            field will appear as <kbd>test_FILTER</kbd>.
          </li>
          <li>
            In VCF files the custom annotations are added to the CSQ INFO field.
          </li>
          <li>
            Alleles in the input and VCF entry are trimmed in both directions in
            an attempt to match complex or poorly formatted entries.
          </li>
        </ul>
      </td>
    </tr>
    <tr id="opt_type">
      <td><pre>type</pre></td>
      <td>
        <kbd>overlap</kbd> (default),
        <kbd>within</kbd>,
        <kbd>surrounding</kbd> or
        <kbd>exact</kbd>
      </td>
      <td>
        <b>Annotation type:</b>
        <ul>
          <li>
            <kbd>overlap</kbd>: reports any annotation that overlaps the variant
            by even 1 base pair.
          </li>
          <li>
            <kbd>within</kbd> (*): only reports annotations within the variant.
          </li>
          <li>
            <kbd>surrounding</kbd> (*): only reports annotations that completely
            surround the variant.
          </li>
          <li>
            <kbd>exact</kbd>: only reports annotations whose coordinates match
            exactly those of the variant. This is suitable for position-specific
            information such as conservation scores, allele frequencies or
            phenotype information.
          </li>
        </ul>
      </td>
    </tr>
    <tr id="opt_overlap_cutoff">
      <td><pre>overlap_cutoff</pre></td>
      <td>From <kbd>0</kbd> (default) to <kbd>100</kbd></td>
      <td>
        <b>Minimum percentage overlap (*)</b> between annotation and variant.
        See also <a href="#opt_reciprocal"><kbd>reciprocal</kbd></a>.
      </td>
    </tr>
    <tr id="opt_reciprocal">
      <td><pre>reciprocal</pre></td>
      <td><kbd>0</kbd> (default) or <kbd>1</kbd></td>
      <td>
        <b>Mode of calculating the overlap percentage (*):</b>
        <ul>
          <li><kbd>0</kbd>: percentage of annotation covered by variant</li>
          <li><kbd>1</kbd>: percentage of variant covered by annotation</li>
        </ul>
      </td>
    </tr>
    <tr id="opt_distance">
      <td><pre>distance</pre></td>
      <td><kbd>0</kbd> or a positive integer (disabled by default)</td>
      <td>Distance (in base pairs) to the ends of the overlapping feature (*).</td>
    </tr>
    <tr id="opt_coords">
      <td><pre>coords</pre></td>
      <td><kbd>0</kbd> (default) or <kbd>1</kbd></td>
      <td>
        <b>Force report coordinates:</b>
        <ul>
          <li>
            <kbd>0</kbd>: outputs the identifier field (or value in the case of
            <kbd>bigWig</kbd>) if available; otherwise, outputs coordinates
            instead.
          </li>
          <li>
            <kbd>1</kbd>: always outputs the coordinates of an overlapping
            custom feature.
          </li>
        </ul>
      </td>
    </tr>
    <tr id="opt_same_type">
      <td><pre>same_type</pre></td>
      <td><kbd>0</kbd> (default) or <kbd>1</kbd></td>
      <td>
        Only match identical variant classes (*). For instance, only match
        deletions with deletions. This is only available for VCF annotations.
      </td>
    </tr>
    <tr id="opt_num_records">
      <td><pre>num_records</pre></td>
      <td><kbd>50</kbd> (default), <kbd>all</kbd>, <kbd>0</kbd> or any positive integer</td>
      <td>
        <b>Number of matching records to display.</b>
        Any remaining records are represented with ellipsis (<kbd>...</kbd>).
        Use <kbd>num_records = all</kbd> to display all matching records and
        <kbd>num_records = 0</kbd> to only display <kbd>...</kbd> if
        there are matching records.
      </td>
    </tr>
    <tr id="opt_summary_stats">
      <td><pre>summary_stats</pre></td>
      <td>
        <kbd>none</kbd> (default), <kbd>min</kbd>, <kbd>mean</kbd>,
        <kbd>max</kbd>, <kbd>count</kbd> or <kbd>sum</kbd>
      </td>
      <td>
        <b>Summary statistics to display</b>. A percentage-separated list may be used to
        calculate multiple summary statistics, such as
        <kbd>min%mean%max%count%sum</kbd>.
      </td>
    </tr>
    <tr id="opt_gff_type">
      <td><pre>gff_type</pre></td>
      <td>
        <kbd>transcript</kbd> (default) or <kbd>gencode_promoter</kbd>
      </td>
      <td>
        <b>GFF feature type</b>. Generally GFF files are parsed to extract and form gene/transcript object. 
        Use <kbd>gencode_promoter</kbd> to extract <a href="https://www.gencodegenes.org/pages/promoter_windows">GENCODE promoter windows</a>.
      </td>
    </tr>
  </tbody></table>
  
  <p>
    When <kbd>format = vcf</kbd>, the features marked with (*) only work on
    structural variants.
  </p>

  <p>Examples:</p>
  
  <pre class="code sh_sh">
# BigWig file
./vep [...] --custom file=frequencies.bw,short_name=Frequency,format=bigwig,type=exact,coords=0
# BED file
./vep [...] --custom file=http://www.myserver.com/data/myPhenotypes.bed.gz,short_name=Phenotype,format=bed,type=exact,coords=1
# VCF file
./vep [...] --custom file=[[SPECIESDEFS::ENSEMBL_FTP_URL]]/data_files/homo_sapiens/GRCh37/variation_genotype/TOPMED_GRCh37.vcf.gz,format=vcf,type=exact,coords=0,fields=TOPMED
./vep [...] --custom file=gnomad_v2.1_sv.sites.vcf.gz,short_name=gnomad,fields=PC%EVIDENCE%SVTYPE,format=vcf,type=within,reciprocal=1,overlap_cutoff=80

# For multiple custom files, use:
./vep [...] --custom file=clinvar.vcf.gz,short_name=ClinVar,format=vcf,type=exact,coords=0,fields=CLNSIG%CLNREVSTAT%CLNDN \
            --custom file=TOPMED_GRCh38_20180418.vcf.gz,short_name=topmed_20180418,format=vcf,type=exact,coords=0,fields=TOPMED \
            --custom file=UK10K_COHORT.20160215.sites.GRCh38.vcf.gz,short_name=uk10k,format=vcf,type=exact,coords=0,fields=AF_ALSPAC</pre>
  </div>
  <hr/>
  <h2 id="custom_example">Example - ClinVar</h2>

  <p>We include the most recent public variant and phenotype data available in each Ensembl release, but some projects release data more frequently than we do.<br />
  If you want to have the very latest annotations, you can use the data files from your prefered projects (in any format listed in <a href="#custom_formats">Data formats</a>) and use them as a VEP custom annotation.</p>
  <p>For instance, you can annotate you variants with Ensembl VEP, using the the latest ClinVar data as custom annotation.<br />
  ClinVar provides VCF files on their FTP site: <a href="https://ftp.ncbi.nlm.nih.gov/pub/clinvar/vcf_GRCh37/" rel="external">GRCh37</a> and <a href="https://ftp.ncbi.nlm.nih.gov/pub/clinvar/vcf_GRCh38/" rel="external">GRCh38</a>.</p>
  <p>See below an example about how to use ClinVar VCF files as a Ensembl VEP custom annotation:</p>

  <ol>
    <li>Download the VCF files (you need the compressed VCF file and the index file), e.g.:
      <pre class="code sh_sh top-margin"># Compressed VCF file
curl -O https://ftp.ncbi.nlm.nih.gov/pub/clinvar/vcf_GRCh38/clinvar.vcf.gz
# Index file
curl -O https://ftp.ncbi.nlm.nih.gov/pub/clinvar/vcf_GRCh38/clinvar.vcf.gz.tbi</pre>
    </li>
    <li>Example of command you can use:
      <pre class="code sh_sh top-margin" style="margin-bottom:2px">./vep [...] --custom file=clinvar.vcf.gz,short_name=ClinVar,format=vcf,type=exact,coords=0,fields=CLNSIG%CLNREVSTAT%CLNDN

## Where the selected ClinVar INFO fields (from the ClinVar VCF file) are:
# - CLNSIG:     Clinical significance for this single variant
# - CLNREVSTAT: ClinVar review status for the Variation ID
# - CLNDN:      ClinVar's preferred disease name for the concept specified by disease identifiers in CLNDISDB
# Of course you can select the INFO fields you want in the ClinVar VCF file

# Quick example on GRCh38:
./vep --id "1  230710048 230710048 A/G 1" --species homo_sapiens -o /path/to/output/output.txt --cache --offline --assembly GRCh38 --custom file=/path/to/custom_files/clinvar.vcf.gz,short_name=ClinVar,format=vcf,type=exact,coords=0,fields=CLNSIG%CLNREVSTAT%CLNDN</pre>

      <div class="navbar" onclick="show_hide('custom_results');" id="custom_results" style="padding-left:5px;margin-bottom:0px;cursor:pointer"><img id="img_custom_results" src="/i/16/plus-button.png" style="vertical-align:middle"/> <span style="position:absolute;left:28px;top:9px">Results in the default VEP format</span></div>
      <div id="div_custom_results" style="display:none">
        <pre class="code sh_sh">## Column descriptions:
## Uploaded_variation : Identifier of uploaded variant
## Location : Location of variant in standard coordinate format (chr:start or chr:start-end)
## Allele : The variant allele used to calculate the consequence
## Gene : Stable ID of affected gene
## Feature : Stable ID of feature
## Feature_type : Type of feature - Transcript, RegulatoryFeature or MotifFeature
## Consequence : Consequence type
## cDNA_position : Relative position of base pair in cDNA sequence
## CDS_position : Relative position of base pair in coding sequence
## Protein_position : Relative position of amino acid in protein
## Amino_acids : Reference and variant amino acids
## Codons : Reference and variant codon sequence
## Existing_variation : Identifier(s) of co-located known variants
## Extra column keys:
## IMPACT : Subjective impact classification of consequence type
## DISTANCE : Shortest distance from variant to transcript
## STRAND : Strand of the feature (1/-1)
## FLAGS : Transcript quality flags
## SOURCE : Source of transcript
## ClinVar : /opt/vep/.vep/custom/clinvar.vcf.gz (exact)
## ClinVar_CLNSIG : CLNSIG field from /path/to/custom_files/clinvar.vcf.gz
## ClinVar_CLNREVSTAT : CLNREVSTAT field from /path/to/custom_files/clinvar.vcf.gz
## ClinVar_CLNDN : CLNDN field from /path/to/custom_files/clinvar.vcf.gz
#Uploaded_variation  Location     Allele  Gene             Feature          Feature_type  Consequence              ...  Extra
1_230710048_A/G      1:230710048  G       ENSG00000135744  ENST00000366667  Transcript    missense_variant         ...  IMPACT=MODERATE;STRAND=-1;ClinVar=18068;ClinVar_CLNDN=Hypertension,_essential,_susceptibility_to|Preeclampsia,_susceptibility_to|Renal_dysplasia|Susceptibility_to_progression_to_renal_failure_in_IgA_nephropathy|not_specified;ClinVar_CLNREVSTAT=criteria_provided,_multiple_submitters,_no_conflicts;ClinVar_CLNSIG=Benign;ClinVar_FILTER=.
1_230710048_A/G      1:230710048  G       ENSG00000244137  ENST00000412344  Transcript    downstream_gene_variant  ...  IMPACT=MODIFIER;DISTANCE=650;STRAND=-1;ClinVar=18068;ClinVar_CLNDN=Hypertension,_essential,_susceptibility_to|Preeclampsia,_susceptibility_to|Renal_dysplasia|Susceptibility_to_progression_to_renal_failure_in_IgA_nephropathy|not_specified;ClinVar_CLNREVSTAT=criteria_provided,_multiple_submitters,_no_conflicts;ClinVar_CLNSIG=Benign;ClinVar_FILTER=.</pre>
      </div>

      <div class="navbar" onclick="show_hide('custom_results_vcf');" id="custom_results_vcf" style="padding-left:5px;margin:2px 0px 0px;cursor:pointer"><img id="img_custom_results_vcf" src="/i/16/plus-button.png" style="vertical-align:middle"/> <span style="position:absolute;left:28px;top:9px">Results in VCF (adding the tag <b>--vcf</b> in the command line)</span></div>
      <div id="div_custom_results_vcf" style="display:none">
        <pre class="code sh_sh">##fileformat=VCFv4.1
##INFO=&lt;ID=CSQ,Number=.,Type=String,Description="Consequence annotations from Ensembl VEP. Format: Allele|Consequence|IMPACT|SYMBOL|Gene|Feature_type|Feature|BIOTYPE|EXON|INTRON|HGVSc|HGVSp|cDNA_position|CDS_position|Protein_position|Amino_acids|Codons|Existing_variation|DISTANCE|STRAND|FLAGS|SYMBOL_SOURCE|HGNC_ID|SOURCE|ClinVar|ClinVar_CLNSIG|ClinVar_CLNREVSTAT|ClinVar_CLNDN"&gt;
##INFO=&lt;ID=ClinVar,Number=.,Type=String,Description="/path/to/custom_files/clinvar.vcf.gz (exact)"&gt;
##INFO=&lt;ID=ClinVar_CLNSIG,Number=.,Type=String,Description="CLNSIG field from /path/to/custom_files/clinvar.vcf.gz"&gt;
##INFO=&lt;ID=ClinVar_CLNREVSTAT,Number=.,Type=String,Description="CLNREVSTAT field from /path/to/custom_files/clinvar.vcf.gz"&gt;
##INFO=&lt;ID=ClinVar_CLNDN,Number=.,Type=String,Description="CLNDN field from /path/to/custom_files/clinvar.vcf.gz"&gt;
#CHROM  POS        ID               REF  ALT  QUAL  FILTER  INFO
1       230710048  1_230710048_A/G  A    G    .     .       CSQ=G|missense_variant|MODERATE|AGT|ENSG00000135744|Transcript|ENST00000366667|protein_coding|2/5||||1018|803|268|M/T|aTg/aCg|||-1||HGNC|HGNC:333||18068|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Hypertension&_essential&_susceptibility_to&Preeclampsia&_susceptibility_to&Renal_dysplasia&Susceptibility_to_progression_to_renal_failure_in_IgA_nephropathy&not_specified,G|downstream_gene_variant|MODIFIER|AL512328.1|ENSG00000244137|Transcript|ENST00000412344|antisense|||||||||||650|-1||Clone_based_ensembl_gene|||18068|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Hypertension&_essential&_susceptibility_to&Preeclampsia&_susceptibility_to&Renal_dysplasia&Susceptibility_to_progression_to_renal_failure_in_IgA_nephropathy&amp;not_specified</pre>
    </li>
  </ol>

  <hr/>
  <h2 id="custom_remote">Using remote files</h2>
  
  <p>The tabix utility makes it possible to read annotation files from remote
  locations, for example over HTTP or FTP protocols.</p>

  <p>In order to do this, the
  .tbi index file is downloaded locally (to the current working directory)
  when Ensembl VEP is run. From this point on, only the portions of data requested
  by Ensembl VEP (i.e. those overlapping the variants in your input file) are
  downloaded.</p>
 
  <div style="width:auto" class="warning">

    <h3>Be aware</h3>

    <div class="message-pad">
      <p>1. It is still possible to cause problems with network traffic in this manner by requesting data for a
         large number of variants.</p>
      
      <p>2. Users with large amounts of data should download the annotation file locally rather than risk causing any issues!</p>
    </div>
  </div>  
 
  <p> bigWig files can also be used remotely in the same way as tabix-indexed
  files, although less stringent checks are carried out on Ensembl VEP startup. </p>

  <br/>
  <hr/>
  <h2 id="remote_custom_example">Example - phyloP and phastCons conservation scores</h2>

  <p> The <a href="https://genome.ucsc.edu" rel="external">UCSC Genome Browser</a> provides
  multiple alignment files with phyloP and phastCons conservation scores for different
  genomes in the BigWig (<kbd>.bw</kbd>) format. </p>

  <p> These files can be remotely used as Ensembl VEP custom annotations by simply
  pointing to their URL.
  For instance, to include phyloP or phastCons 100 way conservation scores found in
  the <a href="https://hgdownload.soe.ucsc.edu/downloads" rel="external">Downloads section</a>
  of the UCSC Genome Browser, you can use commands such as: </p>

  <pre class="code sh_sh top-margin"># Human GRCh38/hg38 phyloP100way scores
./vep [...] --custom file=http://hgdownload.soe.ucsc.edu/goldenPath/hg38/phyloP100way/hg38.phyloP100way.bw,short_name=phyloP100way,format=bigwig

# Human GRCh38/hg38 phastCons100way scores
./vep [...] --custom file=http://hgdownload.soe.ucsc.edu/goldenPath/hg38/phastCons100way/hg38.phastCons100way.bw,short_name=phastCons100way,format=bigwig</pre>

<hr/>
<h2 id="custom_remote">Multiple files split by chromosome</h2>

<p>Often large annotation files are split into several smaller files for each chromosome. In such cases you can run custom annotation
  on those files using a single <a href="vep_options.html#opt_custom">--custom</a> flag instead of separate ones for each file.</p>

<p>To do this, you need all the files to have the same name except for chromosome. When providing the <kbd>file</kbd> 
  option replace the chromosome name with <kbd>###CHR###</kbd> placeholder.</p>

<p>For example, if you have separate gnomAD frequency files for each 24 chromosomes you can run:</p>
<pre class="code sh_sh top-margin">
# Human GRCh38/hg38 gnomAD frequency - there are total 24 files for each chromosome
./vep [...] --custom file=gnomad.exomes.v4.1.sites.chr###CHR###.vcf.bgz,short_name=gnomade,fields=AF,format=vcf
</pre>
</div>

<div style="clear:both"><h1 style="page-break-after: always">&nbsp</h1></div><div>
  <div style="float:right"><img src="/img/vep_logo.png"/></div>
	
  <h1 id="top"><span style="color:#006;padding-right:15px">Ensembl Variant Effect Predictor</span><span style="color:#666"><img src="/i/16/tool.png"/> Plugins</span></h1>
  <hr/>
        
  <p> You can use plugin modules written in Perl to <b>extend, filter and manipulate</b> the Ensembl VEP output. </p>

  <p> To use plugins: </p>
  <ul>
    <li> Install them using the <a href="vep_download.html#installer">Ensembl VEP installer script</a>. You can quickly check installed plugins by running:
    <ul>
      <li> <pre class="code sh_sh">perl INSTALL.pl -a p -g list</pre> </li>
    </ul>
    </li>

    <li> Use Ensembl VEP in <a href="vep_download.html#docker">Docker</a> and <a href="vep_download.html#singularity">Singularity</a>. The plugins and their dependencies are available in the <a href="https://hub.docker.com/r/ensemblorg/ensembl-vep">Docker image</a>. </li>

  </ul>

  <br />	
  <h2 id="plugins_existing">Existing plugins</h2>
	
  <p> We have written plugins to implement new
      functionalities that we do not (yet) include in the variation API, and these
      are stored in a public github repository: </p>
	
  <p><a href="//github.com/Ensembl/VEP_plugins" rel="external">https://github.com/Ensembl/VEP_plugins</a></p>

  <p>Here is the list of the Ensembl VEP plugins available:</p>

  <!-- Plugins list - start -->

  <div style="margin:5px 0px 10px">
    <span style="padding-right:8px">Select categories:</span>
    <select id="select_category" onchange="javascritpt:show_hide_category()">
       <option value="all">All categories</option><option value="conservation">Conservation</option><option value="external_id">External ID</option><option value="frequency_data">Frequency data</option><option value="functional_effect">Functional effect</option><option value="gene_tolerance_to_change">Gene tolerance to change</option><option value="hgvs">HGVS</option><option value="look_up">Look up</option><option value="motif">Motif</option><option value="nearby_features">Nearby features</option><option value="pathogenicity_predictions">Pathogenicity predictions</option><option value="phenotype_data_and_citations">Phenotype data and citations</option><option value="protein_annotation">Protein annotation</option><option value="protein_data">Protein data</option><option value="regulatory_impact">Regulatory impact</option><option value="sequence">Sequence</option><option value="splicing_predictions">Splicing predictions</option><option value="structural_variant_data">Structural variant data</option><option value="transcript_annotation">Transcript annotation</option><option value="variant_data">Variant data</option><option value="visualisation">Visualisation</option>
    </select>
  </div>
  <table class="ss" style="table-layout: fixed; word-break: break-word;">
    <thead>
      <tr>
        <th>Plugin</th>
        <th style="width:50%">Description</th>
        <th>Category</th>
        <th>External libraries</th>
        <th>Developer</th>
      </tr>
    </thead>
    <tbody>
<tr id="AlphaMissense" class="bg1 plugin_row" data-category="pathogenicity_predictions"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/AlphaMissense.pm">AlphaMissense</a></div></td><td><p> This plugin for the Ensembl Variant Effect Predictor (VEP) annotates missense variants with the 
 pre-computed AlphaMissense pathogenicity scores. AlphaMissense is a deep learning model developed 
 by Google DeepMind that predicts the pathogenicity of single nucleotide missense variants.  <a class="button" href="#alphamissense" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('alphamissense');" id="a_alphamissense">more</a></p><div id="div_alphamissense" style="display:none;word-wrap:break-word;"><p> This plugin will add two annotations per missense variant:</p><p><ul><li><kbd>am_pathogenicity</kbd>, a continuous score between 0 and 1 which can be interpreted as the predicted &nbsp; probability of the variant being pathogenic. </p><p></li><li><kbd>am_class</kbd> is the classification of the variant into one of three discrete categories:  &nbsp;   <kbd>likely_pathogenic</kbd>, <kbd>likely_benign</kbd>, or <kbd>ambiguous</kbd>. These are derived using the following 
&nbsp;   thresholds of am_pathogenicity: 
&nbsp;     <kbd>likely_benign</kbd>     if <kbd>am_pathogenicity</kbd> &lt 0.34; 
&nbsp;     <kbd>likely_pathogenic</kbd> if <kbd>am_pathogenicity</kbd> &gt 0.564; 
&nbsp;     <kbd>ambiguous</kbd> otherwise. </p><p></li></ul><p> These thresholds were chosen to achieve 90% precision for both pathogenic and benign ClinVar variants. 
 Note that AlphaMissense was not trained on ClinVar variants. Variants labeled as <kbd>ambiguous</kbd> should be 
 treated as <kbd>unknown</kbd> or <kbd>uncertain</kbd> according to AlphaMissense. </p><p> This plugin is available for both GRCh37 (hg19) and GRCh38 (hg38) genome builds.</p><p> The prediction scores of AlphaMissense can be downloaded from 
 <a href="https://console.cloud.google.com/storage/browser/dm_alphamissense">https://console.cloud.google.com/storage/browser/dm_alphamissense</a> 
 (AlphaMissense Database Copyright (2023) DeepMind Technologies Limited). Data contained within the
 AlphaMissense Database is licensed under the Creative Commons Attribution 4.0 International License
 (CC-BY) (the “License”). You may obtain a copy of the License at:
 <a href="https://creativecommons.org/licenses/by/4.0/legalcode">https://creativecommons.org/licenses/by/4.0/legalcode</a>.
 Use of the AlphaMissense Database is subject to Google Cloud Platform Terms of Service  </p><p> Please cite the AlphaMissense publication alongside the Ensembl VEP
 if you use this resource: <a href="https://doi.org/10.1126/science.adg7492">https://doi.org/10.1126/science.adg7492</a></p><p> Disclaimer: The AlphaMissense Database and other information provided on or linked to this site is 
 for theoretical modelling only, caution should be exercised in use. It is provided "as-is" without 
 any warranty of any kind, whether express or implied. For clarity, no warranty is given that use of 
 the information shall not infringe the rights of any third party (and this disclaimer takes precedence 
 over any contrary provisions in the Google Cloud Platform Terms of Service). The information provided 
 is not intended to be a substitute for professional medical advice, diagnosis, or treatment, and does 
 not constitute medical or other professional advice.</p><p> Before running the plugin for the first time, you need to create a tabix index (requires tabix to be 
 installed).</p><p><pre class="code sh_sh">tabix -s 1 -b 2 -e 2 -f -S 1 AlphaMissense_hg38.tsv.gz
</pre><p><pre class="code sh_sh">tabix -s 1 -b 2 -e 2 -f -S 1 AlphaMissense_hg19.tsv.gz
</pre><p></p><p> Options are passed to the plugin as key=value pairs:<table class="ss"><thead><tr><th>Argument</th><th>Description</th></tr></thead><tbody><tr class="bg1"><td><pre>file</pre></td><td>(mandatory) Tabix-indexed AlphaMissense data <tr class="bg2"><td><pre>cols</pre></td><td>(optional) Colon-separated columns to print from                       AlphaMissense data; if set to <kbd>all</kbd>, all columns are printed
                      (default: <kbd>Missense_pathogenicity:Missense_class</kbd>)
<tr class="bg1"><td><pre>transcript_match</pre></td><td>Only print data if transcript identifiers match those from                       AlphaMissense data (default: 0)</p><p></td></tr></tbody></table><p> AlphaMissense predictions are matched to input data by genomic location and protein change by default.</p><p></ul><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv AlphaMissense.pm ~/.vep/Plugins</p><p># print AlphaMissense scores and predictions (default)
./vep -i variations.vcf --plugin AlphaMissense,file=/full/path/to/file.tsv.gz</p><p># print all AlphaMissense information
./vep -i variations.vcf --plugin AlphaMissense,file=/full/path/to/file.tsv.gz,cols=all</p><p># only report results for the transcripts in the AlphaMissense prediction
./vep -i variations.vcf --plugin AlphaMissense,file=/full/path/to/file.tsv.gz,transcript_match=1
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#1E90FF">Pathogenicity predictions</div></td><td>-</td><td>Ensembl</td></tr>
<tr id="AncestralAllele" class="bg2 plugin_row" data-category="conservation"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/AncestralAllele.pm">AncestralAllele</a></div></td><td><p> A VEP plugin that retrieves ancestral allele sequences from a FASTA file. <a class="button" href="#ancestralallele" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('ancestralallele');" id="a_ancestralallele">more</a></p><div id="div_ancestralallele" style="display:none;word-wrap:break-word;"><p> Ensembl produces FASTA file dumps of the ancestral sequences of key species.
<ul><li>Data files for GRCh37: <a href="https://ftp.ensembl.org/pub/release-75/fasta/ancestral_alleles/">https://ftp.ensembl.org/pub/release-75/fasta/ancestral_alleles/</a></li><li>Data files for GRCh38: <a href="https://ftp.ensembl.org/pub/current_fasta/ancestral_alleles/">https://ftp.ensembl.org/pub/current_fasta/ancestral_alleles/</a>
</li></ul><p> For optimal retrieval speed, you should pre-process the FASTA files into a single
 bgzipped file that can be accessed via <kbd>Bio::DB::HTS::Faidx</kbd> (installed by VEP's
 INSTALL.pl):</p><p><pre class="code sh_sh">wget <a href="https://ftp.ensembl.org/pub/current_fasta/ancestral_alleles/homo_sapiens_ancestor_GRCh38.tar.gz">https://ftp.ensembl.org/pub/current_fasta/ancestral_alleles/homo_sapiens_ancestor_GRCh38.tar.gz</a>
tar xfz homo_sapiens_ancestor_GRCh38.tar.gz
cat homo_sapiens_ancestor_GRCh38/*.fa | bgzip -c &gt homo_sapiens_ancestor_GRCh38.fa.gz
rm -rf homo_sapiens_ancestor_GRCh38/ homo_sapiens_ancestor_GRCh38.tar.gz
./vep -i variations.vcf --plugin AncestralAllele,homo_sapiens_ancestor_GRCh38.fa.gz
</pre><p> The plugin is also compatible with <kbd>Bio::DB::Fasta</kbd> and an uncompressed FASTA file.</p><p> Note the first time you run the plugin with a newly generated FASTA file it will
 spend some time indexing the file. DO NOT INTERRUPT THIS PROCESS, particularly
 if you do not have <kbd>Bio::DB::HTS</kbd> installed.</p><p> Special cases:
<ul><li><kbd>-</kbd> represents an insertion</li><li><kbd>?</kbd> indicates the chromosome could not be looked up in the FASTA
</ul><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv AncestralAllele.pm ~/.vep/Plugins
./vep -i variations.vcf --plugin AncestralAllele,homo_sapiens_ancestor_GRCh38.fa.gz
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#02599C">Conservation</div></td><td>-</td><td>Ensembl</td></tr>
<tr id="AVADA" class="bg1 plugin_row" data-category="phenotype_data_and_citations"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/AVADA.pm">AVADA</a></div></td><td><p>Automatic VAriant evidence DAtabase is a novel machine learning tool that uses natural language processing 
to automatically identify pathogenic genetic variant evidence in full-text primary literature about 
monogenic disease and convert it to genomic coordinates. <a class="button" href="#avada" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('avada');" id="a_avada">more</a></p><div id="div_avada" style="display:none;word-wrap:break-word;"><p>Please cite the AVADA publication alongside the Ensembl VEP if you use this resource:
<a href="https://pubmed.ncbi.nlm.nih.gov/31467448/">https://pubmed.ncbi.nlm.nih.gov/31467448/</a></p><p>NB: The plugin currently does not annotate for downstream_gene_variant and upstream_gene_variant.</p><p>Pre-requisites
<ol><li value="1">AVADA data is available for GRCh37 and can be downloaded from: <a href="http://bejerano.stanford.edu/AVADA/avada_v1.00_2016.vcf.gz">http://bejerano.stanford.edu/AVADA/avada_v1.00_2016.vcf.gz</a></p><p></li></ol><p><pre class="code sh_sh">wget <a href="http://bejerano.stanford.edu/AVADA/avada_v1.00_2016.vcf.gz">http://bejerano.stanford.edu/AVADA/avada_v1.00_2016.vcf.gz</a>
</pre><p><ol><li value="2">The file needs to be tabix indexed. You can do this by following commands:
</li></ol><p><pre class="code sh_sh">gzip -d avada_v1.00_2016.vcf.gz
bgzip avada_v1.00_2016.vcf 
tabix avada_v1.00_2016.vcf.gz
</pre><p><ol><li value="3">As you have already noticed, tabix utility must be installed in your path to use this plugin. 
</li></ol><p>The plugin can then be run to retrieve AVADA annotations. 
By default, the variants are matched with the HGNC gene symbol
<pre class="code sh_sh">./vep -i variations.vcf --plugin AVADA,file=path/to/file
</pre><p>The output always includes one of the following columns depending on the option passed:
<ul><li><kbd>AVADA_PMID</kbd>: PubMed ID evidence for the variant as reported by AVADA</li><li><kbd>AVADA_PMID_WITH_VARIANT</kbd>: PubMed ID evidence for the variant as reported by AVADA along with the original variant string</li><li><kbd>AVADA_PMID_WITH_FEATURE</kbd>: PubMed ID evidence for the variant as reported by AVADA along with feature id</li><li><kbd>AVADA_PMID_WITH_FEATURE_AND_VARIANT</kbd>: PubMed ID evidence for the variant as reported by AVADA along with feature id and original variant string
</li></ul><p>The plugin can optionally be run by specifying the feature to match with.</p><p>In order to match by HGNC gene symbol:
<pre class="code sh_sh">./vep -i variations.vcf --plugin AVADA,file=path/to/file,feature_match_by=gene_symbol 
</pre><p>In order to match by Ensembl gene identifier :
<pre class="code sh_sh">./vep -i variations.vcf --plugin AVADA,file=path/to/file,feature_match_by=ensembl_gene_id
</pre><p>In order to match by RefSeq identifier :
<pre class="code sh_sh">./vep -i variations.vcf --plugin AVADA,file=path/to/file,feature_match_by=refseq_id
</pre><p>The plugin can also be run to report the original variant string reported in the publication. 
<pre class="code sh_sh">./vep -i variations.vcf --plugin AVADA,file=path/to/file,original_variant_string=1
</pre><p>
</ul><p></ol><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">./vep -i variations.vcf --plugin AVADA,file=path/to/file
./vep -i variations.vcf --plugin AVADA,file=path/to/file,feature_match_by=&ltgene_symbol|ensembl_gene_id|refseq_id&gt
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#556688">Phenotype data and citations</div></td><td><a href="https://metacpan.org/pod/List::MoreUtils" rel="external">List::MoreUtils</a> qw(uniq)</td><td>Ensembl</td></tr>
<tr id="BayesDel" class="bg2 plugin_row" data-category="pathogenicity_predictions"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/BayesDel.pm">BayesDel</a></div></td><td><p> This is a plugin for the Ensembl Variant Effect Predictor (VEP) that
 adds the BayesDel scores to Ensembl VEP output. <a class="button" href="#bayesdel" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('bayesdel');" id="a_bayesdel">more</a></p><div id="div_bayesdel" style="display:none;word-wrap:break-word;"><p> BayesDel is a deleteriousness meta-score combining multiple deleteriousness predictors to create an overall score. It works for coding and non-coding variants,
 single nucleotide variants and small insertion/deletions.
 The range of the score is from -1.29334 to 0.75731.
 The higher the score, the more likely the variant is pathogenic.
 For more information please visit: <a href="https://fenglab.chpc.utah.edu/BayesDel/BayesDel.html">https://fenglab.chpc.utah.edu/BayesDel/BayesDel.html</a></p><p> Please cite the BayesDel publication alongside the Ensembl VEP if you use this resource:
 <a href="https://onlinelibrary.wiley.com/doi/full/10.1002/humu.23158">https://onlinelibrary.wiley.com/doi/full/10.1002/humu.23158</a></p><p> BayesDel pre-computed scores can be downloaded from
 <a href="https://drive.google.com/drive/folders/1K4LI6ZSsUGBhHoChUtegC8bgCt7hbQlA">https://drive.google.com/drive/folders/1K4LI6ZSsUGBhHoChUtegC8bgCt7hbQlA</a>
 Note: These files only contain pre-computed BayesDel scores for missense variants for assembly GRCh37.</p><p> For GRCh37:
<pre class="code sh_sh">tar zxvf BayesDel_170824_addAF.tgz
rm *.gz.tbi
gunzip *.gz
for f in BayesDel_170824_addAF_chr*; do grep -v "^#" $f &gt&gt BayesDel_170824_addAF.txt; done
cat BayesDel_170824_addAF.txt | sort -k1,1 -k2,2n &gt BayesDel_170824_addAF_sorted.txt
grep "^#" BayesDel_170824_addAF_chr1 &gt BayesDel_170824_addAF_all_scores.txt
cat BayesDel_170824_addAF_sorted.txt &gt&gt BayesDel_170824_addAF_all_scores.txt
bgzip BayesDel_170824_addAF_all_scores.txt
tabix -s 1 -b 2 -e 2 BayesDel_170824_addAF_all_scores.txt.gz
</pre><p> For GRCh38:
 Remap GRCh37 file</p><p> The tabix utility must be installed in your path to use this plugin.</p><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv BayesDel.pm ~/.vep/Plugins
./vep -i variations.vcf --plugin BayesDel,file=/path/to/BayesDel/BayesDel_170824_addAF_all_scores.txt.gz
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#1E90FF">Pathogenicity predictions</div></td><td>-</td><td>Ensembl</td></tr>
<tr id="Blosum62" class="bg1 plugin_row" data-category="conservation"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/Blosum62.pm">Blosum62</a></div></td><td><p> This is a plugin for the Ensembl Variant Effect Predictor (VEP) that
 looks up the BLOSUM 62 substitution matrix score for the reference
 and alternative amino acids predicted for a missense mutation. It adds
 one new entry to Ensembl VEP's Extra column, BLOSUM62 which is the 
 associated score.  <a class="button" href="#blosum62" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('blosum62');" id="a_blosum62">more</a></p><div id="div_blosum62" style="display:none;word-wrap:break-word;"><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv Blosum62.pm ~/.vep/Plugins
./vep -i variations.vcf --plugin Blosum62
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#02599C">Conservation</div></td><td>-</td><td>Ensembl</td></tr>
<tr id="CADD" class="bg2 plugin_row" data-category="pathogenicity_predictions"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/CADD.pm">CADD</a></div><div style="margin-top:6px"><small>Combined Annotation Dependent Depletion</small></div></td><td><p> A VEP plugin that retrieves CADD scores for variants from one or more
 tabix-indexed CADD data files. <a class="button" href="#cadd" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('cadd');" id="a_cadd">more</a></p><div id="div_cadd" style="display:none;word-wrap:break-word;"><p> Please cite the CADD publication alongside Ensembl VEP if you use this resource:
 <a href="https://www.ncbi.nlm.nih.gov/pubmed/24487276">https://www.ncbi.nlm.nih.gov/pubmed/24487276</a></p><p> The tabix utility must be installed in your path to use this plugin. </p><p> The CADD SNV and indels data files (and respective Tabix index files) can be downloaded from - 
 <a href="http://cadd.gs.washington.edu/download">http://cadd.gs.washington.edu/download</a></p><p> The CADD SV data files (and respective Tabix index files)  can be downloaded from -
 <a href="https://kircherlab.bihealth.org/download/CADD-SV/v1.1/">https://kircherlab.bihealth.org/download/CADD-SV/v1.1/</a></p><p> By default the plugin is designed to not annotate SV variant if a SNV and/or indels CADD
 annotation file is provided. Because it can results in too many lines matched from the annotation
 files and increase run time exponentially. You can override this behavior by providing 
 force_annotate=1 which will force the plugin to annotate with the expense of increasing runtime.</p><p> The plugin works with all versions of available CADD files. The plugin only
 reports scores and does not consider any additional annotations from a CADD
 file. It is therefore sufficient to use CADD files without the additional
 annotations. </p><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv CADD.pm ~/.vep/Plugins
./vep -i variations.vcf --plugin CADD,snv=/FULL_PATH_TO_CADD_FILE/whole_genome_SNVs.tsv.gz,indels=/FULL_PATH_TO_CADD_FILE/InDels.tsv.gz
./vep -i structural_variations.vcf --plugin CADD,sv=/FULL_PATH_TO_CADD_FILE/1000G_phase3_SVs.tsv.gz
./vep -i structural_variations.vcf --plugin CADD,snv=/FULL_PATH_TO_CADD_FILE/whole_genome_SNVs.tsv.gz,indels=/FULL_PATH_TO_CADD_FILE/InDels.tsv.gz,force_annotate=1
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#1E90FF">Pathogenicity predictions</div></td><td>-</td><td>Ensembl</td></tr>
<tr id="CAPICE" class="bg1 plugin_row" data-category="pathogenicity_predictions"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/CAPICE.pm">CAPICE</a></div></td><td><p> An Ensembl VEP plugin that retrieves CAPICE scores for variants from one or more
 tabix-indexed CAPICE data files, in order to predict their pathogenicity. <a class="button" href="#capice" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('capice');" id="a_capice">more</a></p><div id="div_capice" style="display:none;word-wrap:break-word;"><p> Please cite the CAPICE publication alongside the VEP if you use this resource:
 <a href="https://pubmed.ncbi.nlm.nih.gov/32831124/">https://pubmed.ncbi.nlm.nih.gov/32831124/</a></p><p> The tabix utility must be installed in your path to use this plugin. The CAPICE
 SNVs, InDels and respective index (TBI) files for GRCh37 can be downloaded from
 <a href="https://zenodo.org/record/3928295">https://zenodo.org/record/3928295</a></p><p> To filter results, please use filter_vep with the output file or standard
 output. Documentation on filter_vep is available at:
 <a href="https://www.ensembl.org/info/docs/tools/vep/script/vep_filter.html">https://www.ensembl.org/info/docs/tools/vep/script/vep_filter.html</a></p><p> For recommendations on threshold selection, please read the CAPICE publication.</p><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv CAPICE.pm ~/.vep/Plugins
# Download CAPICE SNVs, InDels and index (TBI) files to the same path
# - capice_v1.0_build37_indels.tsv.gz
# - capice_v1.0_build37_indels.tsv.gz.tbi
# - capice_v1.0_build37_snvs.tsv.gz
# - capice_v1.0_build37_snvs.tsv.gz.tbi
./vep -i variations.vcf --plugin CAPICE,snv=/FULL_PATH_TO_CAPICE_FILE/capice_v1.0_build37_snvs.tsv.gz,indels=/FULL_PATH_TO_CAPICE_FILE/capice_v1.0_build37_indels.tsv.gz
./filter_vep -i variant_effect_output.txt --filter "CAPICE_SCORE &gt= 0.02"
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#1E90FF">Pathogenicity predictions</div></td><td>-</td><td>Ensembl</td></tr>
<tr id="Carol" class="bg2 plugin_row" data-category="pathogenicity_predictions"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/Carol.pm">Carol</a></div></td><td><p> An Ensembl VEP plugin that calculates the Combined Annotation scoRing toOL (CAROL)
 score (1) for a missense mutation based on the pre-calculated SIFT (2) and
 PolyPhen-2 (3) scores from the Ensembl API (4). <a class="button" href="#carol" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('carol');" id="a_carol">more</a></p><div id="div_carol" style="display:none;word-wrap:break-word;"><p> It adds one new entry class to Ensembl VEP's Extra column, CAROL which is
 the calculated CAROL score. Note that this module is a perl reimplementation of 
 the original R script, available at: <a href="https://sanger.ac.uk/tool/carol/">https://sanger.ac.uk/tool/carol/</a></p><p> I believe that both versions implement the same algorithm, but if there are any
 discrepancies the R version should be treated as the reference implementation. 
 Bug reports are welcome.</p><p> References:</p><p><ol><li value="1">Lopes MC, Joyce C, Ritchie GRS, John SL, Cunningham F, Asimit J, Zeggini E.      A combined functional annotation score for non-synonymous variants
     Human Heredity 73(1):47-51 (2012)
     <a rel="external" href="https://doi.org/10.1159/000334984">doi:10.1159/000334984</a></p><p></li><li value="2">Kumar P, Henikoff S, Ng PC.     Predicting the effects of coding non-synonymous variants on protein function using the SIFT algorithm
     Nature Protocols 4(8):1073-1081 (2009)
     <a rel="external" href="https://doi.org/10.1038/nprot.2009.86">doi:10.1038/nprot.2009.86</a></p><p></li><li value="3">Adzhubei IA, Schmidt S, Peshkin L, Ramensky VE, Gerasimova A, Bork P, Kondrashov AS, Sunyaev SR.      A method and server for predicting damaging missense mutations
     Nature Methods 7(4):248-249 (2010)
     <a rel="external" href="https://doi.org/10.1038/nmeth0410-248">doi:10.1038/nmeth0410-248</a></p><p></li><li value="4">Flicek P, et al.     Ensembl 2012
     Nucleic Acids Research 40(D1):D84-D90 (2011)
     doi: 10.1093/nar/gkr991</p><p></ol><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv Carol.pm ~/.vep/Plugins
./vep -i variations.vcf --plugin Carol
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#1E90FF">Pathogenicity predictions</div></td><td><a href="https://metacpan.org/pod/Math::CDF" rel="external">Math::CDF</a> qw(pnorm qnorm)</td><td>Ensembl</td></tr>
<tr id="ClinPred" class="bg1 plugin_row" data-category="pathogenicity_predictions"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/ClinPred.pm">ClinPred</a></div></td><td><p> This is a plugin for the Ensembl Variant Effect Predictor (VEP) that adds pre-calculated scores from ClinPred.
 ClinPred is a prediction tool to identify disease-relevant nonsynonymous variants. <a class="button" href="#clinpred" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('clinpred');" id="a_clinpred">more</a></p><div id="div_clinpred" style="display:none;word-wrap:break-word;"><p> Please cite the ClinPred publication alongside Ensembl VEP if you use this resource:
 <a href="https://www.sciencedirect.com/science/article/pii/S0002929718302714">https://www.sciencedirect.com/science/article/pii/S0002929718302714</a></p><p> ClinPred scores can be downloaded from 
 <a href="https://sites.google.com/site/clinpred/download">https://sites.google.com/site/clinpred/download</a></p><p> The following steps are neccessary to tabix the ClinPred.txt.gz file before running the plugin:</p><p> For GRCh37:
<pre class="code sh_sh">gzip -d ClinPred.txt.gz # to unzip the text file 
cat ClinPred.txt | tr " " "\t" &gt ClinPred_tabbed.tsv # change to tab-delimited file 
sed -i '1s/.*/#&/' ClinPred_tabbed.tsv  # comment the first line
sed -i <kbd>1s/Chr/chr/</kbd> ClinPred_tabbed.tsv # convert Chr to chr
bgzip ClinPred_tabbed.tsv
tabix -f -s 1 -b 2 -e 2 ClinPred_tabbed.tsv.gz
</pre><p> For GRCh38:
<pre class="code sh_sh">gzip -d ClinPred_hg38.txt.gz # unzip the text file 
awk '($2 == "Start" || $2 ~ /^[0-9]+$/){print $0}' ClinPred_hg38.txt &gt "ClinPred_hg38_tabbed.tsv" # remove problematic lines
sed -i '1s/.*/#&/' ClinPred_hg38_tabbed.tsv # comment the first line
sed -i <kbd>1s/Chr/chr/</kbd> ClinPred_hg38_tabbed.tsv # convert Chr to chr
</pre><p> { head -n 1 ClinPred_hg38_tabbed.tsv; tail -n +2 ClinPred_hg38_tabbed.tsv | sort -k1,1V -k2,2V; } &gt ClinPred_hg38_sorted_tabbed.tsv # sort file by chromosome and position
<pre class="code sh_sh">bgzip ClinPred_hg38_sorted_tabbed.tsv
tabix -f -s 1 -b 2 -e 2 ClinPred_hg38_sorted_tabbed.tsv.gz
</pre><p> The tabix utility must be installed in your path to use this plugin.
 Check <a href="https://github.com/samtools/htslib.git">https://github.com/samtools/htslib.git</a> for instructions.</p><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv ClinPred.pm ~/.vep/Plugins
./vep -i variations.vcf --plugin ClinPred,file=ClinPred_tabbed.tsv.gz</p><p></pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#1E90FF">Pathogenicity predictions</div></td><td>-</td><td>Ensembl</td></tr>
<tr id="Condel" class="bg2 plugin_row" data-category="pathogenicity_predictions"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/Condel.pm">Condel</a></div></td><td><p> A VEP plugin that calculates the Consensus Deleteriousness (Condel) score (1)
 for a missense mutation based on the pre-calculated SIFT (2) and PolyPhen-2 (3)
 scores from the Ensembl API (4). <a class="button" href="#condel" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('condel');" id="a_condel">more</a></p><div id="div_condel" style="display:none;word-wrap:break-word;"><p> It adds one new entry class to Ensembl VEP's Extra column, Condel which is the
 calculated Condel score. This version of Condel was developed by the Biomedical
 Genomics Group  of the Universitat Pompeu Fabra, at the Barcelona Biomedical
 Research Park and available at <a href="https://bg.upf.edu/condel">https://bg.upf.edu/condel</a>. The code in this
 plugin is based on a script provided by this group and slightly reformatted to
 fit into the Ensembl API.</p><p> The plugin takes 3 command line arguments by this order:
<ol><li value="1">Path to a Condel configuration directory which contains cutoffs and the    distribution files, etc.
</li><li value="2">Output: output the Condel score (<kbd>s</kbd>), prediction (<kbd>p</kbd>) or both (<kbd>b</kbd>);    both (<kbd>b</kbd>) is the default.
</li><li value="3">Version of Condel to use: either 1 (original version) or 2 (newer version);    <kbd>2</kbd> is the default and is recommended to avoid false positive predictions
    from Condel in some circumstances.</p><p></li></ol><p> An example Condel configuration file and a set of distribution files can be
 found in the <kbd>config/Condel</kbd> directory in this repository. You should edit the 
 <kbd>config/Condel/config/condel_SP.conf</kbd> file and set the <kbd>condel.dir</kbd> parameter to
 the full path to the location of the <kbd>config/Condel</kbd> directory on your system.</p><p> References:</p><p><ol><li value="1">Gonzalez-Perez A, Lopez-Bigas N.      Improving the assessment of the outcome of non-synonymous SNVs with a Consensus deleteriousness score (Condel)
     Am J Hum Genet 88(4):440-449 (2011)
     <a rel="external" href="https://doi.org/10.1016/j.ajhg.2011.03.004">doi:10.1016/j.ajhg.2011.03.004</a></p><p></li><li value="2">Kumar P, Henikoff S, Ng PC.     Predicting the effects of coding non-synonymous variants on protein function using the SIFT algorithm
     Nature Protocols 4(8):1073-1081 (2009)
     <a rel="external" href="https://doi.org/10.1038/nprot.2009.86">doi:10.1038/nprot.2009.86</a></p><p></li><li value="3">Adzhubei IA, Schmidt S, Peshkin L, Ramensky VE, Gerasimova A, Bork P, Kondrashov AS, Sunyaev SR.      A method and server for predicting damaging missense mutations
     Nature Methods 7(4):248-249 (2010)
     <a rel="external" href="https://doi.org/10.1038/nmeth0410-248">doi:10.1038/nmeth0410-248</a></p><p></li><li value="4">Flicek P, et al.     Ensembl 2012
     Nucleic Acids Research (2011)
     <a rel="external" href="https://doi.org/10.1093/nar/gkr991">doi:10.1093/nar/gkr991</a></p><p></ol><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv Condel.pm ~/.vep/Plugins
./vep -i variations.vcf --plugin Condel,/path/to/config/Condel/config,b
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#1E90FF">Pathogenicity predictions</div></td><td>-</td><td>Ensembl</td></tr>
<tr id="Conservation" class="bg1 plugin_row" data-category="conservation"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/Conservation.pm">Conservation</a></div></td><td><p> This is a plugin for the Ensembl Variant Effect Predictor (VEP) that
 retrieves a conservation score from the Ensembl Compara databases
 for variant positions. You can specify the method link type and
 species sets as command line options, the default is to fetch GERP
 scores from the EPO 35 way mammalian alignment (please refer to the
 Compara documentation for more details of available analyses).  <a class="button" href="#conservation" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('conservation');" id="a_conservation">more</a></p><div id="div_conservation" style="display:none;word-wrap:break-word;"><p> If a variant affects multiple nucleotides the average score for the
 position will be returned, and for insertions the average score of
 the 2 flanking bases will be returned. If the MAX parameter is
 used, the maximum score of any of the affected bases will be reported
 instead.</p><p> The plugin uses the ensembl-compara API module (optional, see
 <a href="http://www.ensembl.org/info/docs/api/index.html">http://www.ensembl.org/info/docs/api/index.html</a>) or obtains data
 directly from BigWig files (optional, see
 <a href="https://ftp.ensembl.org/pub/current_compara/conservation_scores/">https://ftp.ensembl.org/pub/current_compara/conservation_scores/</a>)</p><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv Conservation.pm ~/.vep/Plugins</p><p>./vep -i variations.vcf --plugin Conservation,mammals
./vep -i variations.vcf --plugin Conservation,/path/to/bigwigfile.bw
./vep -i variations.vcf --plugin Conservation,/path/to/bigwigfile.bw,MAX
./vep -i variations.vcf --plugin Conservation,database,GERP_CONSERVATION_SCORE,mammals
./vep -i variations.vcf --plugin Conservation,database,GERP_CONSERVATION_SCORE,mammals,MAX
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#02599C">Conservation</div></td><td><a href="https://metacpan.org/pod/Net::FTP" rel="external">Net::FTP</a></td><td>Ensembl</td></tr>
<tr id="dbNSFP" class="bg2 plugin_row" data-category="pathogenicity_predictions"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/dbNSFP.pm">dbNSFP</a></div></td><td><p> An Ensembl VEP plugin that retrieves data for missense variants from a tabix-indexed
 dbNSFP file. <a class="button" href="#dbnsfp" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('dbnsfp');" id="a_dbnsfp">more</a></p><div id="div_dbnsfp" style="display:none;word-wrap:break-word;"><p> Please cite the dbNSFP publications alongside Ensembl VEP if you use this resource:
<ul><li>dbNSFP      <a href="https://www.ncbi.nlm.nih.gov/pubmed/21520341">https://www.ncbi.nlm.nih.gov/pubmed/21520341</a></li><li>dbNSFP v2.0 <a href="https://www.ncbi.nlm.nih.gov/pubmed/23843252">https://www.ncbi.nlm.nih.gov/pubmed/23843252</a></li><li>dbNSFP v3.0 <a href="https://www.ncbi.nlm.nih.gov/pubmed/26555599">https://www.ncbi.nlm.nih.gov/pubmed/26555599</a></li><li>dbNSFP v4   <a href="https://www.ncbi.nlm.nih.gov/pubmed/33261662">https://www.ncbi.nlm.nih.gov/pubmed/33261662</a> 
</li></ul><p> You must have the <kbd>Bio::DB::HTS</kbd> module or the tabix utility must be installed
 in your path to use this plugin.</p><p> About dbNSFP data files:
<ul><li>Downoad dbNSFP files from <a href="https://sites.google.com/site/jpopgen/dbNSFP">https://sites.google.com/site/jpopgen/dbNSFP</a>.</li><li>There are two distinct branches of the files provided for academic and&nbsp;     commercial usage. Please use the appropriate files for your use case.
</li><li>The file must be processed depending on dbNSFP release version and assembly&nbsp;     (see commands below). We recommend using <kbd>-T</kbd> option with the sort command
&nbsp;     to specify a temporary directory with sufficient space.
</li><li>The resulting file must be indexed with tabix before use by this plugin&nbsp;     (see commands below).</p><p></li></ul><p> For release 4.9c:
<pre class="code sh_sh">version=4.9c
wget <a href="https://dbnsfp.s3.amazonaws.com/dbNSFP4.9c.zip">https://dbnsfp.s3.amazonaws.com/dbNSFP${version}.zip</a>
unzip dbNSFP${version}.zip
zcat dbNSFP${version}_variant.chr1.gz | head -n1 &gt h
</pre><p> # GRCh38/hg38 data
<pre class="code sh_sh">zgrep -h -v ^#chr dbNSFP${version}_variant.chr* | sort -k1,1 -k2,2n - | cat h - | bgzip -c &gt dbNSFP${version}_grch38.gz
tabix -s 1 -b 2 -e 2 dbNSFP${version}_grch38.gz
</pre><p> # GRCh37/hg19 data
<pre class="code sh_sh">zgrep -h -v ^#chr dbNSFP${version}_variant.chr* | awk '$8 != "." ' | sort -k8,8 -k9,9n - | cat h - | bgzip -c &gt dbNSFP${version}_grch37.gz
tabix -s 8 -b 9 -e 9 dbNSFP${version}_grch37.gz
</pre><p> When running the plugin you must list at least one column to retrieve from the
 dbNSFP file, specified as parameters to the plugin, such as:</p><p><pre class="code sh_sh">--plugin dbNSFP,/path/to/dbNSFP.gz,LRT_score,GERP++_RS
</pre><p> You may include all columns with <kbd>ALL</kbd>; this fetches a large amount of data per
 variant:</p><p><pre class="code sh_sh">--plugin dbNSFP,/path/to/dbNSFP.gz,ALL
</pre><p> Tabix also allows the data file to be hosted on a remote server. This plugin is
 fully compatible with such a setup - simply use the URL of the remote file:</p><p><pre class="code sh_sh">--plugin dbNSFP,<a href="http://my.files.com/dbNSFP.gz">http://my.files.com/dbNSFP.gz</a>,col1,col2
</pre><p> The plugin replaces occurrences of <kbd>;</kbd> with <kbd>,</kbd> and <kbd>|</kbd> with <kbd>&</kbd>. However, some
 data field columns, e.g. <kbd>Interpro_domain</kbd>, use the replacement characters. We
 added a file with replacement logic for customising the required replacement
 of <kbd>;</kbd> and <kbd>|</kbd> in dbNSFP data columns. In addition to the default replacements
 (<kbd>;</kbd> to <kbd>,</kbd> and <kbd>|</kbd> to <kbd>&</kbd>) users can add customised replacements. Users can either modify
 the file <kbd>dbNSFP_replacement_logic</kbd> in the VEP_plugins directory or provide their own
 file as second argument when calling the plugin:</p><p><pre class="code sh_sh">--plugin dbNSFP,/path/to/dbNSFP.gz,/path/to/dbNSFP_replacement_logic,LRT_score,GERP++_RS
</pre><p> Note that transcript sequences referred to in dbNSFP may be out of sync with
 those in the latest release of Ensembl; this may lead to discrepancies with
 scores retrieved from other sources.</p><p> If the dbNSFP README file is found in the same directory as the data file,
 column descriptions will be read from this and incorporated into the Ensembl VEP output
 file header.</p><p> The plugin matches rows in the tabix-indexed dbNSFP file on:
<ul><li>genomic position</li><li>alt allele</li><li><kbd>aaref</kbd> - reference amino acid</li><li><kbd>aaalt</kbd> - alternative amino acid
</li></ul><p> To match only on the genomic position and the alt allele use <kbd>pep_match=0</kbd>:</p><p><pre class="code sh_sh">--plugin dbNSFP,/path/to/dbNSFP.gz,pep_match=0,col1,col2
</pre><p> Some fields contain multiple values, one per Ensembl transcript ID.
 By default all values are returned, separated by <kbd>;</kbd> in the default Ensembl VEP output format.
 To return values only for the matched Ensembl transcript ID use <kbd>transcript_match=1</kbd>.
 This behaviour only affects transcript-specific fields; non-transcript-specific fields
 are unaffected.</p><p><pre class="code sh_sh">--plugin dbNSFP,/path/to/dbNSFP.gz,transcript_match=1,col1,col2
</pre><p> NB 1: Using this flag may cause no value to return if the version of the Ensembl
 transcript set differs between Ensembl VEP and dbNSFP.</p><p> NB 2: MutationTaster entries are keyed on a different set of transcript IDs. Using
 the <kbd>transcript_match</kbd> flag with any MutationTaster field selected will have no effect
 i.e. all entries are returned. Information on corresponding transcript(s) for
 MutationTaster fields can be found using <a href="http://www.mutationtaster.org/ChrPos.html">http://www.mutationtaster.org/ChrPos.html</a></p><p></ul><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv dbNSFP.pm ~/.vep/Plugins
./vep -i variations.vcf --plugin dbNSFP,/path/to/dbNSFP.gz,col1,col2
./vep -i variations.vcf --plugin dbNSFP,'consequence=ALL',/path/to/dbNSFP.gz,col1,col2
./vep -i variations.vcf --plugin dbNSFP,'consequence=3_prime_UTR_variant&intron_variant',/path/to/dbNSFP.gz,col1,col2
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#1E90FF">Pathogenicity predictions</div></td><td><a href="https://metacpan.org/pod/File::Basename" rel="external">File::Basename</a> qw(basename)</td><td>Ensembl</td></tr>
<tr id="dbscSNV" class="bg1 plugin_row" data-category="splicing_predictions"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/dbscSNV.pm">dbscSNV</a></div></td><td><p>  An Ensembl VEP plugin that retrieves data for splicing variants from a tabix-indexed
  dbscSNV file. <a class="button" href="#dbscsnv" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('dbscsnv');" id="a_dbscsnv">more</a></p><div id="div_dbscsnv" style="display:none;word-wrap:break-word;"><p>  Please cite the dbscSNV publication alongside Ensembl VEP if you use this resource:
  <a href="http://nar.oxfordjournals.org/content/42/22/13534">http://nar.oxfordjournals.org/content/42/22/13534</a></p><p>  The Bio::DB::HTS perl library or tabix utility must be installed in your path
  to use this plugin. The dbscSNV data file can be downloaded from
  <a href="https://sites.google.com/site/jpopgen/dbNSFP">https://sites.google.com/site/jpopgen/dbNSFP</a>.</p><p>  The file must be processed and indexed by tabix before use by this plugin.
  dbscSNV1.1 has coordinates for both GRCh38 and GRCh37; the file must be
  processed differently according to the assembly you use.</p><p><pre class="code sh_sh">wget <a href="ftp://dbnsfp:dbnsfp@dbnsfp.softgenetics.com/dbscSNV1.1.zip">ftp://dbnsfp:dbnsfp@dbnsfp.softgenetics.com/dbscSNV1.1.zip</a>
unzip dbscSNV1.1.zip
head -n1 dbscSNV1.1.chr1 &gt h
</pre><p>  # GRCh38
<pre class="code sh_sh">cat dbscSNV1.1.chr* | grep -v ^chr | sort -k5,5 -k6,6n | cat h - | awk '$5 != "."' | bgzip -c &gt dbscSNV1.1_GRCh38.txt.gz
tabix -s 5 -b 6 -e 6 -c c dbscSNV1.1_GRCh38.txt.gz
</pre><p>  # GRCh37
<pre class="code sh_sh">cat dbscSNV1.1.chr* | grep -v ^chr | cat h - | bgzip -c &gt dbscSNV1.1_GRCh37.txt.gz
tabix -s 1 -b 2 -e 2 -c c dbscSNV1.1_GRCh37.txt.gz
</pre><p>  Note that in the last command we tell tabix that the header line starts with "c";
  this may change to the default of "#" in future versions of dbscSNV.</p><p>  Tabix also allows the data file to be hosted on a remote server. This plugin is
  fully compatible with such a setup - simply use the URL of the remote file:</p><p><pre class="code sh_sh">--plugin dbscSNV,<a href="http://my.files.com/dbscSNV.txt.gz">http://my.files.com/dbscSNV.txt.gz</a>
</pre><p>  Note that transcript sequences referred to in dbscSNV may be out of sync with
  those in the latest release of Ensembl; this may lead to discrepancies with
  scores retrieved from other sources.</p><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv dbscSNV.pm ~/.vep/Plugins
./vep -i variations.vcf --plugin dbscSNV,/path/to/dbscSNV1.1_GRCh38.txt.gz
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#FF0000">Splicing predictions</div></td><td>-</td><td>Ensembl</td></tr>
<tr id="DeNovo" class="bg2 plugin_row" data-category="variant_data"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/DeNovo.pm">DeNovo</a></div></td><td><p> An Ensembl  VEP plugin that identifies de novo variants in a VCF file.
 The plugin is not compatible with JSON output format. <a class="button" href="#denovo" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('denovo');" id="a_denovo">more</a></p><div id="div_denovo" style="display:none;word-wrap:break-word;"><p> Options are passed to the plugin as key=value pairs:<table class="ss"><thead><tr><th>Argument</th><th>Description</th></tr></thead><tbody>
<tr class="bg1"><td><pre>ped</pre></td><td>Path to PED file (mandatory)                       The file is tab or white-space delimited with five mandatory columns:
<ul><li>family ID</li><li>individual ID</li><li>paternal ID</li><li>maternal ID</li><li>sex</li><li>phenotype (optional)
<tr class="bg2"><td><pre></li></ul><p> report_dir</pre></td><td>Write files in report_dir (optional) 
<tr class="bg1"><td><pre>full_report</pre></td><td>Set to 1 to report all types of variants (optional)                       By default, the plugin only reports de novo variants.</p><p></td></tr></tbody></table><p>
 The plugin can then be run:
<pre class="code sh_sh">./vep -i variations.vcf --plugin DeNovo,ped=samples.ped
./vep -i variations.vcf --plugin DeNovo,ped=samples.ped,report_dir=path/to/dir
./vep -i variations.vcf --plugin DeNovo,ped=samples.ped,report_dir=path/to/dir,full_report=1
</pre><p>
</ul><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv DeNovo.pm ~/.vep/Plugins
./vep -i variations.vcf --plugin DeNovo,ped=samples.ped
./vep -i variations.vcf --plugin DeNovo,ped=samples.ped,full_report=1
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#B22222">Variant data</div></td><td><ul style="padding-left:1em"><li><a href="https://metacpan.org/pod/List::MoreUtils" rel="external">List::MoreUtils</a> qw(uniq)</li><li><a href="https://metacpan.org/pod/Cwd" rel="external">Cwd</a></li></ul></td><td>Ensembl</td></tr>
<tr id="DosageSensitivity" class="bg1 plugin_row" data-category="gene_tolerance_to_change"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/DosageSensitivity.pm">DosageSensitivity</a></div></td><td><p> An Ensembl VEP plugin that retrieves haploinsufficiency and triplosensitivity probability scores
 for affected genes from a dosage sensitivity catalogue published in paper -
 <a href="https://www.sciencedirect.com/science/article/pii/S0092867422007887">https://www.sciencedirect.com/science/article/pii/S0092867422007887</a> <a class="button" href="#dosagesensitivity" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('dosagesensitivity');" id="a_dosagesensitivity">more</a></p><div id="div_dosagesensitivity" style="display:none;word-wrap:break-word;"><p> Please cite the above publication alongside Ensembl VEP if you use this resource.</p><p> This plugin returns two scores:
<ul><li>pHaplo score gives the probability of a gene being haploinsufficient (deletion intolerant)</li><li>pTriplo score gives the probability of a gene being triploinsensitive (duplication intolerant)
</li></ul><p> Pre-requisites:
 You need the compressed tsv file containing the dosage sensitivity score. The file
 Collins_rCNV_2022.dosage_sensitivity_scores.tsv.gz can be downloaded from here - 
 <a href="https://zenodo.org/record/6347673/files/Collins_rCNV_2022.dosage_sensitivity_scores.tsv.gz">https://zenodo.org/record/6347673/files/Collins_rCNV_2022.dosage_sensitivity_scores.tsv.gz</a></p><p> Options are passed to the plugin as key=value pairs:<table class="ss"><thead><tr><th>Argument</th><th>Description</th></tr></thead><tbody>
<tr class="bg1"><td><pre>file</pre></td><td>(mandatory) compressed tsv file containing dosage sensitivity scores <tr class="bg2"><td><pre>cover</pre></td><td>set value to 1 (0 by default) to report scores only if the variant  covers the affected feature completely (e.g. - a CNV that duplicates the gene). 
 The feature is a gene if using --database otherwise it is a transcript.</p><p></ul><p></td></tr></tbody></table><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv DosageSensitivity.pm ~/.vep/Plugins
./vep -i variations.vcf --plugin DosageSensitivity,file=/FULL_PATH_TO/Collins_rCNV_2022.dosage_sensitivity_scores.tsv.gz
./vep -i variations.vcf --plugin DosageSensitivity,file=/FULL_PATH_TO/Collins_rCNV_2022.dosage_sensitivity_scores.tsv.gz,cover=1
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#00DD66">Gene tolerance to change</div></td><td>-</td><td>Ensembl</td></tr>
<tr id="Downstream" class="bg2 plugin_row" data-category="nearby_features"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/Downstream.pm">Downstream</a></div></td><td><p> This is a plugin for the Ensembl Variant Effect Predictor (VEP) that
 predicts the downstream effects of a frameshift variant on the protein
 sequence of a transcript. It provides the predicted downstream protein
 sequence (including any amino acids overlapped by the variant itself),
 and the change in length relative to the reference protein. <a class="button" href="#downstream" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('downstream');" id="a_downstream">more</a></p><div id="div_downstream" style="display:none;word-wrap:break-word;"><p> Note that changes in splicing are not predicted - only the existing
 translateable (i.e. spliced) sequence is used as a source of
 translation. Any variants with a splice site consequence type are
 ignored.</p><p> If Ensembl VEP is run in offline mode using the flag --offline, a FASTA file is required.
 See: <a href="https://www.ensembl.org/info/docs/tools/vep/script/vep_cache.html#fasta">https://www.ensembl.org/info/docs/tools/vep/script/vep_cache.html#fasta</a>
 Sequence may be incomplete without a FASTA file or database connection.</p><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv Downstream.pm ~/.vep/Plugins
./vep -i variations.vcf --plugin Downstream
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#E75480">Nearby features</div></td><td>-</td><td>Ensembl</td></tr>
<tr id="Draw" class="bg1 plugin_row" data-category="visualisation"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/Draw.pm">Draw</a></div></td><td><p> An Ensembl VEP plugin that draws pictures of the transcript model showing the
 variant location. <a class="button" href="#draw" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('draw');" id="a_draw">more</a></p><div id="div_draw" style="display:none;word-wrap:break-word;"><p> Takes five optional paramters:</p><p><ol><li value="1">File name stem for images</li><li value="2">Image width in pixels (default: 1000px)</li><li value="3">Image height in pixels (default: 100px)</li><li value="4">Transcript ID - only draw images for this transcript</li><li value="5">Variant ID - only draw images for this variant 
</li></ol><p> e.g.</p><p><pre class="code sh_sh">./vep -i variations.vcf --plugin Draw,myimg,2000,100
</pre><p> Images are written to <kbd>[file_stem]_[transcript_id]_[variant_id].png</kbd></p><p> Requires GD library installed to run.</p><p></ol><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv Draw.pm ~/.vep/Plugins
./vep -i variations.vcf --plugin Draw
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#008000">Visualisation</div></td><td><ul style="padding-left:1em"><li><a href="https://metacpan.org/pod/GD::Polygon" rel="external">GD::Polygon</a></li><li><a href="https://metacpan.org/pod/GD" rel="external">GD</a></li></ul></td><td>Ensembl</td></tr>
<tr id="Enformer" class="bg2 plugin_row" data-category="regulatory_impact"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/Enformer.pm">Enformer</a></div></td><td><p> This is a plugin for the Ensembl Variant Effect Predictor (VEP) that adds pre-calculated Enformer predictions of variant impact on chromatin and gene expression. <a class="button" href="#enformer" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('enformer');" id="a_enformer">more</a></p><div id="div_enformer" style="display:none;word-wrap:break-word;"><p> The predictions have been aggregated across all 896 spatial bins to generate 5313 features corresponding to track prediction changes in differnet assays and cell types.</p><p> This plugin is available for GRCh37 and GRCh38</p><p> Please cite the Enformer publication alongside Ensembl VEP if you use this resource:
 <a href="https://www.nature.com/articles/s41592-021-01252-x">https://www.nature.com/articles/s41592-021-01252-x</a></p><p> GRCh38 scores were lifted over using CrossMap from the Enformer scores available here - <a href="https://console.cloud.google.com/storage/browser/dm-enformer/variant-scores/1000-genomes/enformer">https://console.cloud.google.com/storage/browser/dm-enformer/variant-scores/1000-genomes/enformer</a></p><p> Enformer scores can be downloaded from <a href="https://ftp.ensembl.org/pub/current_variation/Enformer">https://ftp.ensembl.org/pub/current_variation/Enformer</a> for GRCh37 and GRCh38.</p><p> The plugin can then be run as default to retrieve SAD (SNP Activity Difference (SAD) and SAR (Same as SAD, by computing np.log2(1 + model(alternate_sequence)) - np.log2(1 + model(reference_sequence)) scores from Enforme :
<pre class="code sh_sh">./vep -i variations.vcf --assembly GRCh38 --plugin Enformer,file=/path/to/Enformer/data.vcf.gz
</pre><p> or run with option to only retrieve the SAD (SNP Activity Difference (SAD) scores - main variant effect score computed as model(alternate_sequence) - model(reference_sequence) score 
<pre class="code sh_sh">./vep -i variations.vcf --assembly GRCh38 --plugin Enformer,file=/path/to/Enformer/data.vcf.gz,SAD=1 
</pre><p> or run with option to only retrieve the SAR (Same as SAD, by computing np.log2(1 + model(alternate_sequence)) - np.log2(1 + model(reference_sequence)) score 
<pre class="code sh_sh">./vep -i variations.vcf --assembly GRCh38 --plugin Enformer,file=/path/to/Enformer/data.vcf.gz,SAR=1 
</pre><p> or run with option to also retrieve the principal component scores which are a reduced representation of a much bigger vector of the SAD and SAR after using principal component analysis (PCA)
<pre class="code sh_sh">./vep -i variations.vcf --assembly GRCh38 --plugin Enformer,file=/path/to/Enformer/data.vcf.gz,PC=1 
</pre><p>
 The tabix utility must be installed in your path to use this plugin.
 Check <a href="https://github.com/samtools/htslib.git">https://github.com/samtools/htslib.git</a> for instructions.</p><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv Enformer.pm ~/.vep/Plugins
./vep -i variations.vcf --plugin Enformer,file=Enformer_grch38.vcf.gz</p><p></pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#2200FF">Regulatory impact</div></td><td>-</td><td>Ensembl</td></tr>
<tr id="EVE" class="bg1 plugin_row" data-category="pathogenicity_predictions"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/EVE.pm">EVE</a></div></td><td><p> This is a plugin for the Ensembl Variant Effect Predictor (VEP) that
 adds information from EVE (evolutionary model of variant effect). <a class="button" href="#eve" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('eve');" id="a_eve">more</a></p><div id="div_eve" style="display:none;word-wrap:break-word;"><p> This plugin only report EVE scores for input variants
 and does not merge input lines to report on adjacent variants.
 It is only available for GRCh38.</p><p> Please cite EVE publication alongside Ensembl VEP if you use this resource:
 <a href="https://www.nature.com/articles/s41586-021-04043-8">https://www.nature.com/articles/s41586-021-04043-8</a></p><p><pre class="code sh_sh">###################################################
# Bash script to merge all VCFs from EVE dataset. #
###################################################
### BEGIN
# EVE input file can be downloaded from <a href="https://evemodel.org/api/proteins/bulk/download/">https://evemodel.org/api/proteins/bulk/download/</a> 
# Input: VCF files by protein (vcf_files_missense_mutations inside zip folder)
# Output: Compressed Merged VCF file (vcf.gz) + index file (.tbi)
DATA_FOLDER=<kbd>/&ltPATH-TO&gt/vcf_files_missense_mutations</kbd> # Fill this line
OUTPUT_FOLDER=<kbd>/&ltPATH-TO&gt/eve_plugin</kbd> # Fill this line
OUTPUT_NAME=<kbd>eve_merged.vcf</kbd> # Default output name
# Get header from first VCF
cat `ls ${DATA_FOLDER}/*vcf | head -n1` &gt header
# Get variants from all VCFs and add to a single-file
ls ${DATA_FOLDER}/*vcf | while read VCF; do grep -v '^#' ${VCF} &gt&gt variants; done
# Merge Header + Variants in a single file
cat header variants | \
awk '$1 ~ /^#/ {print $0;next} {print $0 | "sort -k1,1V -k2,2n"}' &gt ${OUTPUT_FOLDER}/${OUTPUT_NAME};
# Remove temporary files
rm header variants
# Compress and index
bgzip ${OUTPUT_FOLDER}/${OUTPUT_NAME};
# If not installed, use: sudo apt install tabix
tabix ${OUTPUT_FOLDER}/${OUTPUT_NAME}.gz;
### END
</pre>
<p><h2>Usage examples:</h2> <pre class="code sh_sh">
cp EVE.pm ${HOME}/.vep/Plugins
./vep -i variations.vcf --plugin EVE,file=/path/to/eve/data.vcf.gz # By default, Class75 is used.
./vep -i variations.vcf --plugin EVE,file=/path/to/eve/data.vcf.gz,class_number=60
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#1E90FF">Pathogenicity predictions</div></td><td>-</td><td>Ensembl</td></tr>
<tr id="FATHMM" class="bg2 plugin_row" data-category="pathogenicity_predictions"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/FATHMM.pm">FATHMM</a></div></td><td><p> An Ensembl VEP plugin that gets FATHMM scores and predictions for missense variants. <a class="button" href="#fathmm" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('fathmm');" id="a_fathmm">more</a></p><div id="div_fathmm" style="display:none;word-wrap:break-word;"><p> You will need the fathmm.py script and its dependencies (Python, Python
 MySQLdb). You should create a "config.ini" file in the same directory as the
 fathmm.py script with the database connection options. More information about
 how to set up FATHMM can be found on the FATHMM website at
 <a href="https://github.com/HAShihab/fathmm">https://github.com/HAShihab/fathmm</a></p><p> A typical installation could consist of:</p><p><pre class="code sh_sh">wget <a href="https://raw.github.com/HAShihab/fathmm/master/cgi-bin/fathmm.py">https://raw.github.com/HAShihab/fathmm/master/cgi-bin/fathmm.py</a>
wget <a href="http://fathmm.biocompute.org.uk/database/fathmm.v2.3.SQL.gz">http://fathmm.biocompute.org.uk/database/fathmm.v2.3.SQL.gz</a>
gunzip fathmm.v2.3.SQL.gz
mysql -h[host] -P[port] -u[user] -p[pass] -e"CREATE DATABASE fathmm"
mysql -h[host] -P[port] -u[user] -p[pass] -Dfathmm &lt fathmm.v2.3.SQL
echo -e "[DATABASE]\nHOST = [host]\nPORT = [port]\nUSER = [user]\nPASSWD = [pass]\nDB = fathmm\n" &gt config.ini
</pre><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv FATHMM.pm ~/.vep/Plugins
./vep -i variations.vcf --plugin FATHMM,"python2 /path/to/fathmm/fathmm.py"
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#1E90FF">Pathogenicity predictions</div></td><td>-</td><td>Ensembl</td></tr>
<tr id="FATHMM_MKL" class="bg1 plugin_row" data-category="pathogenicity_predictions"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/FATHMM_MKL.pm">FATHMM_MKL</a></div></td><td><p> An Ensembl VEP plugin that retrieves FATHMM-MKL scores for variants from a tabix-indexed
 FATHMM-MKL data file. <a class="button" href="#fathmm_mkl" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('fathmm_mkl');" id="a_fathmm_mkl">more</a></p><div id="div_fathmm_mkl" style="display:none;word-wrap:break-word;"><p> See <a href="https://github.com/HAShihab/fathmm-MKL">https://github.com/HAShihab/fathmm-MKL</a> for details.</p><p> NB: The currently available data file is for GRCh37 only.</p><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv FATHMM_MKL.pm ~/.vep/Plugins
./vep -i input.vcf --plugin FATHMM_MKL,fathmm-MKL_Current.tab.gz
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#1E90FF">Pathogenicity predictions</div></td><td>-</td><td>Ensembl</td></tr>
<tr id="FlagLRG" class="bg2 plugin_row" data-category="external_id"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/FlagLRG.pm">FlagLRG</a></div></td><td><p> A VEP plugin that retrieves the LRG ID matching either the RefSeq or Ensembl
 transcript IDs. <a class="button" href="#flaglrg" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('flaglrg');" id="a_flaglrg">more</a></p><div id="div_flaglrg" style="display:none;word-wrap:break-word;"><p> You can obtain the <kbd>list_LRGs_transcripts_xrefs.txt</kbd> using:
<pre class="code sh_sh">wget <a href="https://ftp.ebi.ac.uk/pub/databases/lrgex/list_LRGs_transcripts_xrefs.txt">https://ftp.ebi.ac.uk/pub/databases/lrgex/list_LRGs_transcripts_xrefs.txt</a>
</pre><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv FlagLRG.pm ~/.vep/Plugins
./vep -i variants.vcf --plugin FlagLRG,/path/to/list_LRGs_transcripts_xrefs.txt
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#333333">External ID</div></td><td><a href="https://metacpan.org/pod/Text::CSV" rel="external">Text::CSV</a></td><td>Stephen Kazakoff</td></tr>
<tr id="FunMotifs" class="bg1 plugin_row" data-category="motif"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/FunMotifs.pm">FunMotifs</a></div></td><td><p> This is a plugin for the Ensembl Variant Effect Predictor (VEP) that
 adds tissue-specific transcription factor motifs from FunMotifs to Ensembl VEP output. <a class="button" href="#funmotifs" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('funmotifs');" id="a_funmotifs">more</a></p><div id="div_funmotifs" style="display:none;word-wrap:break-word;"><p> Please cite the FunMotifs publication alongside Ensembl VEP if you use this resource. 
 The preprint can be found at: <a href="https://www.biorxiv.org/content/10.1101/683722v1">https://www.biorxiv.org/content/10.1101/683722v1</a></p><p> FunMotifs files can be downloaded from: <a href="http://bioinf.icm.uu.se:3838/funmotifs/">http://bioinf.icm.uu.se:3838/funmotifs/</a>
 At the time of writing, all BED files found through this link support GRCh37, 
 however other assemblies are supported by the plugin if an appropriate BED file
 is supplied.</p><p> The tabix utility must be installed in your path to use this plugin.</p><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv FunMotifs.pm ~/.vep/Plugins
./vep -i variations.vcf --plugin FunMotifs,/path/to/funmotifs/all_tissues.bed.gz,uterus
./vep -i variations.vcf --plugin FunMotifs,/path/to/funmotifs/blood.funmotifs_sorted.bed.gz,fscore,dnase_seq</p><p>Parameters Required:</p><p>[0] : FunMotifs BED file
[1]+ : List of columns to include within the Ensembl VEP output (e.g. fscore, skin, contactingdomain)</p><p></pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#DAA520">Motif</div></td><td>-</td><td>Ensembl</td></tr>
<tr id="G2P" class="bg2 plugin_row" data-category="phenotype_data_and_citations"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/G2P.pm">G2P</a></div><div style="margin-top:6px"><small>gene2phenotype</small></div></td><td><p> An Ensembl VEP plugin that uses G2P allelic requirements to assess variants in genes
 for potential phenotype involvement. <a class="button" href="#g2p" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('g2p');" id="a_g2p">more</a></p><div id="div_g2p" style="display:none;word-wrap:break-word;"><p> The plugin has multiple configuration options, though minimally requires only
 the CSV file of G2P data.
 This Plugin is available for GRCh38 and GRCh37.</p><p> For further information see:
 Thormann A, Halachev M, McLaren W, et al. Flexible and scalable diagnostic filtering of genomic variants using G2P with Ensembl VEP.
 Nature Communications. 2019 May;10(1):2373. <a rel="external" href="https://doi.org/10.1038/s41467-019-10016-3">doi:10.1038/s41467-019-10016-3</a>. PMID: 31147538; PMCID: PMC6542828.</p><p> Options are passed to the plugin as key=value pairs, (defaults in parentheses):<table class="ss"><thead><tr><th>Argument</th><th>Description</th></tr></thead><tbody>
<tr class="bg1"><td><pre>file</pre></td><td>Path to G2P data file. The file needs to be uncompressed. <ul><li>Download from <a href="https://www.ebi.ac.uk/gene2phenotype/downloads">https://www.ebi.ac.uk/gene2phenotype/downloads</a></li><li>Download from PanelApp  
<tr class="bg2"><td><pre></li></ul><p> variant_include_list</pre></td><td>A list of variants to include even if variants do not pass allele                          frequency filtering. The include list needs to be a sorted, bgzipped and
                         tabixed VCF file.</p><p><tr class="bg1"><td><pre>af_monoallelic</pre></td><td>maximum allele frequency for inclusion for monoallelic genes (0.0001) 
<tr class="bg2"><td><pre>af_biallelic</pre></td><td>maximum allele frequency for inclusion for biallelic genes (0.005) <tr class="bg1"><td><pre>confidence_levels</pre></td><td>Confidence levels include: definitive, strong, moderate, limited                          Former confidence terms are still supported: confirmed, probable, possible, both RD and IF.
                         Separate multiple values with <kbd>&</kbd>.
                         <a href="https://www.ebi.ac.uk/gene2phenotype/terminology">https://www.ebi.ac.uk/gene2phenotype/terminology</a>
                         Default levels are confirmed and probable.
<tr class="bg2"><td><pre>all_confidence_levels</pre></td><td>Set to 1 to include all confidence levels                          Setting the value to 1 will overwrite any confidence levels provided with the
                         confidence_levels option.
<tr class="bg1"><td><pre>af_from_vcf</pre></td><td>set value to 1 to include allele frequencies from VCF file.                          Specifiy the list of reference populations to include with <kbd>--af_from_vcf_keys</kbd>
<tr class="bg2"><td><pre>af_from_vcf_keys</pre></td><td>VCF collections used for annotating variant alleles with observed                          allele frequencies. Allele frequencies are retrieved from VCF files. If
                         af_from_vcf is set to 1 but no VCF collections are specified with <kbd>--af_from_vcf_keys</kbd>
                         all available VCF collections are included. 
                         Available VCF collections: <kbd>topmed</kbd>, <kbd>uk10k</kbd>, <kbd>gnomADe</kbd>, <kbd>gnomADe_r2.1.1</kbd>, <kbd>gnomADg</kbd>, <kbd>gnomADg_v3.1.2</kbd>.
                         Separate multiple values with <kbd>&</kbd>.
                         VCF collections contain the following populations: 
<ul><li><kbd>topmed</kbd> - TOPMed (available for GRCh37 and GRCh38).</li><li><kbd>uk10k</kbd> - ALSPAC, TWINSUK (available for GRCh37 and GRCh38).</li><li><kbd>gnomADe</kbd> & <kbd>gnomADe_r2.1.1</kbd> - gnomADe:AFR, gnomADe:ALL, gnomADe:AMR, gnomADe:ASJ, gnomADe:EAS, gnomADe:FIN, gnomADe:NFE, gnomADe:OTH, gnomADe:SAS (for GRCh37 and GRCh38 respectively).</li><li><kbd>gnomADg</kbd> & <kbd>gnomADg_v3.1.2</kbd> - gnomADg:AFR, gnomADg:ALL, gnomADg:AMR, gnomADg:ASJ, gnomADg:EAS, gnomADg:FIN, gnomADg:NFE, gnomADg:OTH (for GRCh37 and GRCh38 respectively).&nbsp;                         Need to use <kbd>af_from_vcf</kbd> parameter to use this option. 
<tr class="bg1"><td><pre>&nbsp; default_af</pre></td><td>default frequency of the input variant if no frequency data is &nbsp;                         found (0). This determines whether such variants are included;
&nbsp;                         the value of 0 forces variants with no frequency data to be
&nbsp;                         included as this is considered equivalent to having a frequency
&nbsp;                         of 0. Set to 1 (or any value higher than <kbd>af</kbd>) to exclude them.
<tr class="bg2"><td><pre>&nbsp; types</pre></td><td>SO consequence types to include. Separate multiple values with <kbd>&</kbd> &nbsp;                         (splice_donor_variant, splice_acceptor_variant, stop_gained,
&nbsp;                         frameshift_variant, stop_lost, initiator_codon_variant,
&nbsp;                         inframe_insertion, inframe_deletion,missense_variant,
&nbsp;                         coding_sequence_variant, start_lost,transcript_ablation,
&nbsp;                         transcript_amplification, protein_altering_variant)</p><p><tr class="bg1"><td><pre></li></ul><p>  log_dir</pre></td><td>write stats to log files in log_dir 
<tr class="bg2"><td><pre>txt_report</pre></td><td>write all G2P complete genes and attributes to txt file 
<tr class="bg1"><td><pre>html_report</pre></td><td>write all G2P complete genes and attributes to html file 
<tr class="bg2"><td><pre>filter_by_gene_symbol</pre></td><td>set to 1 if filter by gene symbol.                           Do not set if filtering by HGNC_id.
                          This option is set to 1 when using PanelApp files. </p><p><tr class="bg1"><td><pre>only_mane</pre></td><td>set to 1 to ignore transcripts that are not MANE                          N/B - Information may be lost if this option is used.</p><p></td></tr></tbody></table><p> For more information - <a href="https://www.ebi.ac.uk/gene2phenotype/g2p_vep_plugin">https://www.ebi.ac.uk/gene2phenotype/g2p_vep_plugin</a></p><p> Example:
<pre class="code sh_sh">--plugin G2P,file=G2P.csv,af_monoallelic=0.05,types=stop_gained&frameshift_variant
--plugin G2P,file=G2P.csv,af_monoallelic=0.05,af_from_vcf=1
--plugin G2P,file=G2P.csv,af_from_vcf=1,af_from_vcf_keys=<kbd>topmed&gnomADe_r2.1.1</kbd>
--plugin G2P,file=G2P.csv,af_from_vcf=1,af_from_vcf_keys=<kbd>topmed&gnomADe_r2.1.1</kbd>,confidence_levels='confirmed&probable&both RD and IF' 
--plugin G2P,file=G2P.csv
</pre><p></ul><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv G2P.pm ~/.vep/Plugins
./vep -i variations.vcf --plugin G2P,file=/path/to/G2P.csv
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#556688">Phenotype data and citations</div></td><td><ul style="padding-left:1em"><li><a href="https://metacpan.org/pod/List::Util" rel="external">List::Util</a> qw(any)</li><li><a href="https://metacpan.org/pod/Text::CSV" rel="external">Text::CSV</a></li><li><a href="https://metacpan.org/pod/Scalar::Util" rel="external">Scalar::Util</a> qw(looks_like_number)</li><li><a href="https://metacpan.org/pod/FileHandle" rel="external">FileHandle</a></li><li><a href="https://metacpan.org/pod/Cwd" rel="external">Cwd</a></li></ul></td><td>Ensembl</td></tr>
<tr id="GeneBe" class="bg1 plugin_row" data-category="variant_data"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/GeneBe.pm">GeneBe</a></div></td><td><p> A user-contributed VEP plugin that retrieves automatic ACMG variant classification data from
 <a href="https://genebe.net/">https://genebe.net/</a> <a class="button" href="#genebe" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('genebe');" id="a_genebe">more</a></p><div id="div_genebe" style="display:none;word-wrap:break-word;"><p> Please cite the GeneBe publication alongside Ensembl VEP if you use this resource:
 <a href="https://onlinelibrary.wiley.com/doi/10.1111/cge.14516">https://onlinelibrary.wiley.com/doi/10.1111/cge.14516</a> .</p><p> Please be advised that the GeneBe API is freely accessible for academic purposes only, with a limited
 number of queries per day, albeit at a high threshold. Kindly utilize this resource judiciously
 to ensure its availability for others. For further information, please visit <a href="https://genebe.net/about/api">https://genebe.net/about/api</a>.</p><p> In order to extend your daily limits please make an account on <a href="https://genebe.net/">https://genebe.net/</a> and use your username and API-key as follows:
<pre class="code sh_sh">./vep -i variations.vcf --plugin GeneBe,user=example@email.com,password=your_api_key
</pre><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv GeneBe.pm ~/.vep/Plugins
./vep -i variations.vcf --plugin GeneBe
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#B22222">Variant data</div></td><td><a href="https://metacpan.org/pod/JSON" rel="external">JSON</a></td><td><ul style="padding-left:1em"><li>Ensembl</li><li>Piotr Stawinski</li></ul></td></tr>
<tr id="GeneSplicer" class="bg2 plugin_row" data-category="splicing_predictions"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/GeneSplicer.pm">GeneSplicer</a></div></td><td><p> This is a plugin for the Ensembl Variant Effect Predictor (VEP) that
 runs GeneSplicer (<a href="https://ccb.jhu.edu/software/genesplicer/">https://ccb.jhu.edu/software/genesplicer/</a>) to get
 splice site predictions. <a class="button" href="#genesplicer" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('genesplicer');" id="a_genesplicer">more</a></p><div id="div_genesplicer" style="display:none;word-wrap:break-word;"><p> It evaluates a tract of sequence either side of and including the
 variant, both in reference and alternate states. The amount of
 sequence included either side defaults to 100bp, but can be modified
 by passing e.g. "context=50" as a parameter to the plugin.</p><p> You will need to download the GeneSplicer binary and data from
 <a href="ftp://ftp.ccb.jhu.edu/pub/software/genesplicer/GeneSplicer.tar.gz">ftp://ftp.ccb.jhu.edu/pub/software/genesplicer/GeneSplicer.tar.gz</a>. Extract the
 folder using:</p><p><pre class="code sh_sh">tar -xzf GeneSplicer.tar.gz
</pre><p> GeneSplicer comes with precompiled binaries for multiple systems. If the
 provided binaries do not run, compile <kbd>genesplicer</kbd> from source:</p><p><pre class="code sh_sh">cd $GS/sources
# if macOS, run this step
[[ $(uname -s) == "Darwin" ]] && perl -pi -e "s/^main  /int main  /" genesplicer.cpp
make
cd -
./vep [options] --plugin GeneSplicer,$GS/sources/genesplicer,$GS/human
</pre>
 Predicted splicing regions that overlap the variant are reported in the output
 with a <kbd>/</kbd>-separated string (e.g., <kbd>loss/acceptor/727006-727007/High/16.231924</kbd>)
 consisting of the following data by this order:</p><p><ol><li value="1">state (no_change, diff, gain, loss)</li><li value="2">type (donor, acceptor)</li><li value="3">coordinates (start-end)</li><li value="4">confidence (Low, Medium, High)</li><li value="5">score
</li></ol><p> If multiple sites are predicted, their reports are separated by ",".</p><p> For diff, the confidence and score for both the reference and alternate
 sequences is reported as REF-ALT, such as
 <kbd>diff/donor/621915-621914/Medium-Medium/7.020731-6.988368</kbd>.</p><p> Several key=value parameters can be modified in the the plugin string:<table class="ss"><thead><tr><th>Argument</th><th>Description</th></tr></thead><tbody>
<tr class="bg1"><td><pre>training</pre></td><td>(mandatory) directory to species-specific training data, such as               <kbd>GeneSplicer/human</kbd>
<tr class="bg2"><td><pre>binary</pre></td><td>path to <kbd>genesplicer</kbd> binary (default: <kbd>genesplicer</kbd>) <tr class="bg1"><td><pre>context</pre></td><td>change the amount of sequence added either side of               the variant (default: 100bp)
<tr class="bg2"><td><pre>tmpdir</pre></td><td>change the temporary directory used (default: <kbd>/tmp</kbd>) <tr class="bg1"><td><pre>cache_size</pre></td><td>change how many sequences' scores are cached in memory               (default: 50)</p><p></td></tr></tbody></table><p> Example:
<pre class="code sh_sh">--plugin GeneSplicer,binary=$GS/bin/linux/genesplicer,training=$GS/human,context=200,tmpdir=/mytmp
</pre><p> When using the Ensembl VEP Docker/Singularity, the <kbd>binary</kbd> argument can be ommitted, as
 the <kbd>genesplicer</kbd> command is exported in the $PATH variable and is thus
 automatically detected by the plugin:
<pre class="code sh_sh">--plugin GeneSplicer,training=$GS/human,context=200,tmpdir=/mytmp
</pre><p></ol><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv GeneSplicer.pm ~/.vep/Plugins
./vep -i variants.vcf --plugin GeneSplicer,binary=$GS/bin/linux/genesplicer,training=$GS/human
./vep -i variants.vcf --plugin GeneSplicer,binary=$GS/bin/linux/genesplicer,training=$GS/human,context=200,tmpdir=/mytmp</p><p># Ensembl VEP Docker/Singularity: if 'genesplicer' is a command available in $PATH,
# there is no need to specify the location of the binary
./vep -i variants.vcf --plugin GeneSplicer,training=$GS/human
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#FF0000">Splicing predictions</div></td><td><a href="https://metacpan.org/pod/Digest::MD5" rel="external">Digest::MD5</a> qw(md5_hex)</td><td>Ensembl</td></tr>
<tr id="Geno2MP" class="bg1 plugin_row" data-category="phenotype_data_and_citations"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/Geno2MP.pm">Geno2MP</a></div></td><td><p> An Ensembl VEP plugin that adds information from Geno2MP, a web-accessible database of
 rare variant genotypes linked to phenotypic information. <a class="button" href="#geno2mp" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('geno2mp');" id="a_geno2mp">more</a></p><div id="div_geno2mp" style="display:none;word-wrap:break-word;"><p> Parameters can be set using a key=value system:<table class="ss"><thead><tr><th>Argument</th><th>Description</th></tr></thead><tbody><tr class="bg1"><td><pre>file</pre></td><td>VCF file containing Geno2MP data <tr class="bg2"><td><pre>cols</pre></td><td>colon-delimited list of Geno2MP columns to return from INFO fields           (by default it only returns the column HPO_CT)
<tr class="bg1"><td><pre>url</pre></td><td>build and return URL to Geno2MP variant page (boolean; 0 by default);           the variant location in Geno2MP website is based on GRCh37 coordinates</p><p></td></tr></tbody></table><p> Please cite Geno2MP alongside Ensembl VEP if you use this resource:
 Geno2MP, NHGRI/NHLBI University of Washington-Center for Mendelian Genomics (UW-CMG), Seattle, WA
 (URL: <a href="http://geno2mp.gs.washington.edu">http://geno2mp.gs.washington.edu</a> [date (month, yr) accessed]).</p><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
cp Geno2MP.pm ${HOME}/.vep/Plugins
./vep -i variations.vcf --plugin Geno2MP,file=/path/to/Geno2MP/data.vcf.gz</p><p># Return more columns from Geno2MP VCF file
./vep -i variations.vcf --plugin Geno2MP,file=/path/to/Geno2MP/data.vcf.gz,cols=HPO_CT:FXN:nhomalt_male_aff:nhomalt_male_unaff</p><p># Build and return Geno2MP URL based on GRCh37 variant location
./vep -i variations.vcf --plugin Geno2MP,file=/path/to/Geno2MP/data.vcf.gz,url=1
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#556688">Phenotype data and citations</div></td><td>-</td><td>Ensembl</td></tr>
<tr id="gnomADc" class="bg2 plugin_row" data-category="frequency_data"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/gnomADc.pm">gnomADc</a></div></td><td><p> An Ensembl VEP plugin that retrieves gnomAD annotation from either the genome
 or exome coverage files, available here:
   <a href="https://gnomad.broadinstitute.org/downloads">https://gnomad.broadinstitute.org/downloads</a> <a class="button" href="#gnomadc" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('gnomadc');" id="a_gnomadc">more</a></p><div id="div_gnomadc" style="display:none;word-wrap:break-word;"><p> To download the gnomad coverage file in TSV format:
  for Assembly GRCh37: 
   gnomad genomes:
<pre class="code sh_sh">wget <a href="https://storage.googleapis.com/gcp-public-data--gnomad/release/2.1/coverage/genomes/gnomad.genomes.coverage.summary.tsv.bgz">https://storage.googleapis.com/gcp-public-data--gnomad/release/2.1/coverage/genomes/gnomad.genomes.coverage.summary.tsv.bgz</a> --no-check-certificate
</pre><p>   gnomad exomes:
<pre class="code sh_sh">wget <a href="https://storage.googleapis.com/gcp-public-data--gnomad/release/2.1/coverage/exomes/gnomad.exomes.coverage.summary.tsv.bgz">https://storage.googleapis.com/gcp-public-data--gnomad/release/2.1/coverage/exomes/gnomad.exomes.coverage.summary.tsv.bgz</a> --no-check-certificate
</pre><p>  for Assembly GRCh38: 
   gnomad genomes: 
<pre class="code sh_sh">wget <a href="https://storage.googleapis.com/gcp-public-data--gnomad/release/3.0.1/coverage/genomes/gnomad.genomes.r3.0.1.coverage.summary.tsv.bgz">https://storage.googleapis.com/gcp-public-data--gnomad/release/3.0.1/coverage/genomes/gnomad.genomes.r3.0.1.coverage.summary.tsv.bgz</a> --no-check-certificate 
</pre><p> Necessary before using the plugin
  for Assembly GRCh37:
   The following steps are necessary to tabix the gnomad genomes coverage file :
<pre class="code sh_sh">gunzip -c gnomad.genomes.coverage.summary.tsv.bgz | sed '1s/.*/#&/' &gt gnomad.genomes.tabbed.tsv
bgzip gnomad.genomes.tabbed.tsv
tabix -s 1 -b 2 -e 2 gnomad.genomes.tabbed.tsv.gz
</pre><p>   The following steps are neccessary to tabix the gnomad exomes coverage file :
<pre class="code sh_sh">gunzip -c gnomad.exomes.coverage.summary.tsv.bgz | sed '1s/.*/#&/' &gt gnomad.exomes.tabbed.tsv
bgzip gnomad.exomes.tabbed.tsv
tabix -s 1 -b 2 -e 2 gnomad.exomes.tabbed.tsv.gz
</pre><p> for Assembly GRCh38:
   The following steps are necessary to tabix the gnomad genomes coverage file :
<pre class="code sh_sh">gunzip -c gnomad.genomes.r3.0.1.coverage.summary.tsv.bgz | sed '1s/.*/#&/' &gt gnomad.genomesv3.tabbed.tsv
sed "1s/locus/chr\tpos/; s/:/\t/g" gnomad.genomesv3.tabbed.tsv &gt gnomad.ch.genomesv3.tabbed.tsv
bgzip gnomad.ch.genomesv3.tabbed.tsv
tabix -s 1 -b 2 -e 2 gnomad.ch.genomesv3.tabbed.tsv
</pre><p> This plugin also tries to be backwards compatible with older versions of the
 coverage summary files, including releases 2.0.1 and 2.0.2. These releases
 provide one coverage file per chromosome and these can be used "as-is"
 without requiring any preprocessing. </p><p> If you use this plugin, please see the terms and data information:
   <a href="https://gnomad.broadinstitute.org/terms">https://gnomad.broadinstitute.org/terms</a></p><p> You must have the Bio::DB::HTS module or the tabix utility must be installed
 in your path to use this plugin. 
<p><h2>Usage examples:</h2> <pre class="code sh_sh">mv gnomADc.pm ~/.vep/Plugins
./vep -i variations.vcf --plugin gnomADc,/path/to/gnomad.tsv.gz 
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#FF7F50">Frequency data</div></td><td><ul style="padding-left:1em"><li><a href="https://metacpan.org/pod/File::Spec" rel="external">File::Spec</a></li><li><a href="https://metacpan.org/pod/File::Basename" rel="external">File::Basename</a></li></ul></td><td>Stephen Kazakoff</td></tr>
<tr id="GO" class="bg1 plugin_row" data-category="phenotype_data_and_citations"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/GO.pm">GO</a></div><div style="margin-top:6px"><small>Gene Ontology</small></div></td><td><p> An Ensembl VEP plugin that retrieves Gene Ontology (GO) terms associated with
 transcripts (e.g. GRCh38) or their translations (e.g. GRCh37) using custom
 GFF annotation containing GO terms. <a class="button" href="#go" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('go');" id="a_go">more</a></p><div id="div_go" style="display:none;word-wrap:break-word;"><p> The custom GFF files are automatically created if the input file do not exist
 by querying the Ensembl core database, according to database version, species
 and assembly used in Ensembl VEP. Note that automatic retrieval fails if using the
 --offline option.</p><p> The GFF files containing the GO terms are saved to and loaded from the working
 directory by default. To change this, provide a directory path as an argument:</p><p><pre class="code sh_sh">--plugin GO,dir=${HOME}/go_terms
</pre><p> If your GFF file has a custom name, please provide the filename directly:</p><p><pre class="code sh_sh">--plugin GO,file=${HOME}/custom_go_terms.gff.gz
</pre><p> The GO terms can also be fetched by gene match (either gene Ensembl ID or
 gene symbol) instead:</p><p><pre class="code sh_sh">--plugin GO,match=gene
--plugin GO,match=gene_symbol
</pre><p> To create/use a custom GFF file, these programs must be installed in your path:
<ul><li>The GNU zgrep and GNU sort commands to create the GFF file.</li><li>The tabix and bgzip utilities to create and read the GFF file: check&nbsp;     <a href="https://github.com/samtools/htslib.git">https://github.com/samtools/htslib.git</a> for installation instructions.</p><p></li></ul><p> Alternatively, for compatibility purposes, the plugin allows to use a remote
 connection to the Ensembl API by using "remote" as a parameter. This method
 retrieves GO terms one by one at both the transcript and translation level.
 This is not compatible with any other parameters:</p><p><pre class="code sh_sh">--plugin GO,remote
</pre><p></ul><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv GO.pm ~/.vep/Plugins</p><p># automatically fetch GFF files with GO terms and annotate input with GO terms
# not compatible with --offline option
./vep -i variations.vcf --plugin GO</p><p># set directory used to write and read GFF files with GO terms
./vep -i variations.vcf --plugin GO,dir=${HOME}/go_terms</p><p># annotate input with GO terms from custom GFF file
./vep -i variations.vcf --plugin GO,file=${HOME}/custom_go_terms.gff.gz</p><p># annotate input based on gene identifiers instead of transcripts/translations
./vep -i variations.vcf --plugin GO,match=gene</p><p># use remote connection (available for compatibility purposes)
./vep -i variations.vcf --plugin GO,remote
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#556688">Phenotype data and citations</div></td><td>-</td><td>Ensembl</td></tr>
<tr id="GWAS" class="bg2 plugin_row" data-category="phenotype_data_and_citations"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/GWAS.pm">GWAS</a></div></td><td><p> An Ensembl VEP plugin that retrieves relevant NHGRI-EBI GWAS Catalog data given the file.  <a class="button" href="#gwas" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('gwas');" id="a_gwas">more</a></p><div id="div_gwas" style="display:none;word-wrap:break-word;"><p> This plugin supports both the curated data that is found in the download section of the NHGRI-EBI GWAS Catalog website and the 
 summary statistics file. By default the plugin assumes the file provided is the curated file but you can pass "type=sstate" 
 to say you want to annotate with a summary statistics file.</p><p> Please cite the following publication alongside Ensembl VEP if you use this resource:
 <a href="https://pubmed.ncbi.nlm.nih.gov/30445434/">https://pubmed.ncbi.nlm.nih.gov/30445434/</a></p><p> Pre-requisites:</p><p> For curated NHGRI-EBI GWAS Catalog file -
 GWAS files can be downloaded from - <a href="https://www.ebi.ac.uk/gwas/api/search/downloads/alternative">https://www.ebi.ac.uk/gwas/api/search/downloads/alternative</a></p><p> For summary statistics file -
 The plugin can process the harmonised version of the summary statistics file. Which can be downloaded from the FTP site - 
 <a href="http://ftp.ebi.ac.uk/pub/databases/gwas/summary_statistics">http://ftp.ebi.ac.uk/pub/databases/gwas/summary_statistics</a></p><p> They are under directory with related to their specific GCST id. For example -
 <a href="http://ftp.ebi.ac.uk/pub/databases/gwas/summary_statistics/GCST000001-GCST001000/GCST000028/harmonised/17463246-GCST000028-EFO_0001360.h.tsv.gz">http://ftp.ebi.ac.uk/pub/databases/gwas/summary_statistics/GCST000001-GCST001000/GCST000028/harmonised/17463246-GCST000028-EFO_0001360.h.tsv.gz</a></p><p> Please keep the filename format as it is because filename is parsed to get information.</p><p> When run for the first time for either type of file, the plugin will create a processed file that have genomic locations and indexed and 
 put it under the --dir location determined by Ensembl VEP. If db=1 option is used, depending on the file size it might take hour(s) to create 
 the processed file. Subsequent runs will be faster as the plugin will be using the already generated processed file. This option is not used by 
 default and the variant information is generally taken directly from the file provided.</p><p> Options are passed to the plugin as key=value pairs:<table class="ss"><thead><tr><th>Argument</th><th>Description</th></tr></thead><tbody>
<tr class="bg1"><td><pre>file</pre></td><td>(mandatory) Path to GWAS curated or summary statistics file <tr class="bg2"><td><pre>type</pre></td><td>type of the file. Valid values are "curated" and "sstate" (summary statistics). Default is "curated". <tr class="bg1"><td><pre>verbose</pre></td><td>display info level messages. Valid values are 0 or 1. Default is 0. <tr class="bg2"><td><pre>db</pre></td><td>get variant information from Ensembl database during creation of processed file. Valid values are 0 or 1. Default is 0 (variant             information is retrieved from curated file)</p><p></td></tr></tbody></table><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv GWAS.pm ~/.vep/Plugins
./vep -i variations.vcf --plugin GWAS,file=/FULL_PATH_TO/gwas_catalog_v1.0.2-associations_e107_r2022-09-14.tsv
./vep -i variations.vcf --plugin GWAS,type=sstate,file=/FULL_PATH_TO/17463246-GCST000028-EFO_0001360.h.tsv.gz
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#556688">Phenotype data and citations</div></td><td><ul style="padding-left:1em"><li><a href="https://metacpan.org/pod/Storable" rel="external">Storable</a> qw(dclone)</li><li><a href="https://metacpan.org/pod/File::Basename" rel="external">File::Basename</a></li></ul></td><td>Ensembl</td></tr>
<tr id="HGVSIntronOffset" class="bg1 plugin_row" data-category="hgvs"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/HGVSIntronOffset.pm">HGVSIntronOffset</a></div></td><td><p> An Ensembl VEP plugin for the Ensembl Variant Effect Predictor (VEP) that returns
 HGVS intron start and end offsets. To be used with --hgvs option. <a class="button" href="#hgvsintronoffset" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('hgvsintronoffset');" id="a_hgvsintronoffset">more</a></p><div id="div_hgvsintronoffset" style="display:none;word-wrap:break-word;"><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv HGVSIntronOffset.pm ~/.vep/Plugins
./vep -i variants.vcf --hgvs --plugin HGVSIntronOffset
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#9400D3">HGVS</div></td><td>-</td><td>Stephen Kazakoff</td></tr>
<tr id="IntAct" class="bg2 plugin_row" data-category="functional_effect"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/IntAct.pm">IntAct</a></div></td><td><p> A VEP plugin that retrieves molecular interaction data for variants as reprted by IntAct database. <a class="button" href="#intact" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('intact');" id="a_intact">more</a></p><div id="div_intact" style="display:none;word-wrap:break-word;"><p> Please cite the IntAct publication alongside Ensembl VEP if you use this resource:
 <a href="https://pubmed.ncbi.nlm.nih.gov/24234451/">https://pubmed.ncbi.nlm.nih.gov/24234451/</a></p><p> Pre-requisites:</p><p><ol><li value="1">IntAct files can be downloaded from - <a href="https://ftp.ebi.ac.uk/pub/databases/intact/current/various">https://ftp.ebi.ac.uk/pub/databases/intact/current/various</a></p><p></li><li value="2">The genomic location mapped file needs to be tabix indexed. You can  do this by following commands -</p><p></li></ol><p>  a) filter, sort and then zip
<pre class="code sh_sh">grep -v -e '^$' -e '^[#\-]' mutation_gc_map.txt | sed '1s/.*/#&/' | awk -F "\t" 'BEGIN { OFS="\t"} {if ($2 &gt $3) {a=$2; $2=$3; $3=a}; print $0 }' | sort -k1,1 -k2,2n -k3,3n | bgzip &gt mutation_gc_map.txt.gz
</pre><p>  b) create tabix indexed file -
<pre class="code sh_sh">tabix -s 1 -b 2 -e 3 -f mutation_gc_map.txt.gz
</pre><p><ol><li value="3">As you have already noticed, tabix utility must be installed in your path to use this plugin. 
</li></ol><p> Options are passed to the plugin as key=value pairs:<table class="ss"><thead><tr><th>Argument</th><th>Description</th></tr></thead><tbody>
<tr class="bg1"><td><pre>mapping_file</pre></td><td>(mandatory) Path to tabix-indexed genomic location mapped file <tr class="bg2"><td><pre>mutation_file</pre></td><td>(mandatory) Path to IntAct data file  
</td></tr></tbody></table><p> By default the output will always contain feature_type and interaction_ac from the IntAct data file. You can also add more fields using the following key=value options -<table class="ss"><thead><tr><th>Argument</th><th>Description</th></tr></thead><tbody><tr class="bg1"><td><pre>feature_ac</pre></td><td>Set value to 1 to include Feature AC in the output <tr class="bg2"><td><pre>feature_short_label</pre></td><td>Set value to 1 to include Feature short label in the output <tr class="bg1"><td><pre>feature_annotation</pre></td><td>Set value to 1 to include Feature annotation in the output <tr class="bg2"><td><pre>ap_ac</pre></td><td>Set value to 1 to include Affected protein AC in the output <tr class="bg1"><td><pre>interaction_participants</pre></td><td>Set value to 1 to include Interaction participants in the output <tr class="bg2"><td><pre>pmid</pre></td><td>Set value to 1 to include PubMedID in the output 
</td></tr></tbody></table><p> There are also two other key=value options for customizing the output - <table class="ss"><thead><tr><th>Argument</th><th>Description</th></tr></thead><tbody><tr class="bg1"><td><pre>all</pre></td><td>Set value to 1 to include all the fields <tr class="bg2"><td><pre>minimal</pre></td><td>Set value to 1 to overwrite default behavior and include only interaction_ac 				  in the output by default</p><p></td></tr></tbody></table><p> See what these options mean - <a href="https://www.ebi.ac.uk/intact/download/datasets#mutations">https://www.ebi.ac.uk/intact/download/datasets#mutations</a></p><p> Note that, interaction accession can be used to link to full details on the interaction website. For example, 
 where the Ensembl VEP output reports an interaction_ac of EBI-12501485, the URL would be : </p><p>   <a href="https://www.ebi.ac.uk/intact/details/interaction/EBI-12501485">https://www.ebi.ac.uk/intact/details/interaction/EBI-12501485</a></p><p></ol><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv IntAct.pm ~/.vep/Plugins
./vep -i variations.vcf --plugin IntAct,mutation_file=/FULL_PATH_TO_IntAct_FILE/mutations.tsv,mapping_file=/FULL_PATH_TO_IntAct_FILE/mutation_gc_map.txt.gz
./vep -i variations.vcf --plugin IntAct,mutation_file=/FULL_PATH_TO_IntAct_FILE/mutations.tsv,mapping_file=/FULL_PATH_TO_IntAct_FILE/mutation_gc_map.txt.gz,minimal=1
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#226600">Functional effect</div></td><td>-</td><td>Ensembl</td></tr>
<tr id="LD" class="bg1 plugin_row" data-category="variant_data"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/LD.pm">LD</a></div><div style="margin-top:6px"><small>Linkage Disequilibrium</small></div></td><td><p> A VEP plugin that finds variants in linkage disequilibrium with any overlapping
 existing variants from the Ensembl variation databases. <a class="button" href="#ld" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('ld');" id="a_ld">more</a></p><div id="div_ld" style="display:none;word-wrap:break-word;"><p> You can configure the population used to calculate the r2 value, and the
 r2 cutoff used by passing arguments to the plugin via the Ensembl VEP command line
 (separated by commas). This plugin adds a single new entry to the Extra column
 with a comma-separated list of linked variant IDs and the associated r2 values:
   LinkedVariants=rs123:0.879,rs234:0.943</p><p> If no arguments are supplied, the default population used is the CEU
 sample from the 1000 Genomes Project phase 3, and the default r2
 cutoff used is 0.8.</p><p> WARNING: Calculating LD is a relatively slow procedure, so this will 
 slow Ensembl VEP down considerably when running on large numbers of
 variants. Consider running vep followed by filter_vep to get a smaller
 input set:</p><p><pre class="code sh_sh">./vep -i input.vcf -cache -vcf -o input_vep.vcf
./filter_vep -i input_vep.vcf -filter "Consequence is missense_variant" &gt input_vep_filtered.vcf
./vep -i input_vep_filtered.vcf -cache -plugin LD
</pre><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv LD.pm ~/.vep/Plugins
./vep -i variations.vcf --plugin LD,1000GENOMES:phase_3:CEU,0.8
./vep -i variations.vcf --plugin LD,'populations=1000GENOMES:phase_3:CEU&1000GENOMES:phase_3:PUR&1000GENOMES:phase_3:STU',0.8</p><p></pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#B22222">Variant data</div></td><td>-</td><td>Ensembl</td></tr>
<tr id="LocalID" class="bg2 plugin_row" data-category="look_up"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/LocalID.pm">LocalID</a></div></td><td><p> The LocalID plugin allows you to use variant IDs as input without making a database connection. <a class="button" href="#localid" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('localid');" id="a_localid">more</a></p><div id="div_localid" style="display:none;word-wrap:break-word;"><p> Requires sqlite3.</p><p> A local sqlite3 database is used to look up variant IDs; this is generated either from Ensembl's
 public database (very slow, but includes synonyms), or from an Ensembl VEP cache file (faster, excludes
 synonyms).</p><p> NB this plugin is NOT compatible with the ensembl-tools variant_effect_predictor.pl version of Ensembl VEP.</p><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv LocalID.pm ~/.vep/Plugins</p><p>## first run create database</p><p># EITHER create from Ensembl variation database
# VERY slow but includes variant synonyms, if not required see next command
./vep -i variant_ids.txt --plugin LocalID,create_db=1 -safe</p><p># OR create from cache directory
# faster but does not include synonyms
# parameter passed to from_cache may be full path to cache e.g. $HOME/.vep/homo_sapiens/88_GRCh38
# cache may be tabix converted or in default state (http://www.ensembl.org/info/docs/tools/vep/script/vep_cache.html#convert)
./vep -i variant_ids.txt --plugin LocalID,create_db=1,from_cache=1 -safe</p><p># subsequent runs
./vep -i variant_ids.txt --plugin LocalID</p><p># db file can be specified with db=[file]
# default file name is $HOME/.vep/[species]_[version]_[assembly].variant_ids.sqlite3
./vep -i variant_ids.txt --plugin LocalID,db=my_db_file.txt
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#006400">Look up</div></td><td>-</td><td>Ensembl</td></tr>
<tr id="LOEUF" class="bg1 plugin_row" data-category="gene_tolerance_to_change"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/LOEUF.pm">LOEUF</a></div></td><td><p> This is a plugin for the Ensembl Variant Effect Predictor (VEP) that
 adds the LOEUF scores to Ensembl VEP output. LOEUF stands for the "loss-of-function 
 observed/expected upper bound fraction."  <a class="button" href="#loeuf" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('loeuf');" id="a_loeuf">more</a></p><div id="div_loeuf" style="display:none;word-wrap:break-word;"><p> The score can be added matching by either transcript or gene.
 When matched by gene: 
 If multiple transcripts are available for a gene, the most severe score is reported.</p><p> NB: The plugin currently does not add the score for downstream_gene_variant and upstream_gene_variant </p><p> Please cite the LOEUF publication alongside Ensembl VEP if you use this resource:
 <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7334197/">https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7334197/</a></p><p> LOEUF scores can be downloaded from
 GRCh37: <a href="https://gnomad.broadinstitute.org/downloads#v2-constraint">https://gnomad.broadinstitute.org/downloads#v2-constraint</a> (pLoF Metrics by Gene TSV)
 GRCh38: <a href="https://gnomad.broadinstitute.org/downloads#v4-constraint">https://gnomad.broadinstitute.org/downloads#v4-constraint</a> (Constraint metrics TSV)</p><p> For GRCh37:
 These files can be tabix-processed by:
<pre class="code sh_sh">zcat gnomad.v2.1.1.lof_metrics.by_gene.txt.bgz | (head -n 1 && tail -n +2  | sort -t$'\t' -k 76,76 -k 77,77n ) &gt loeuf_temp.tsv
sed '1s/.*/#&/' loeuf_temp.tsv &gt loeuf_dataset.tsv
bgzip loeuf_dataset.tsv
tabix -f -s 76 -b 77 -e 78 loeuf_dataset.tsv.gz
</pre><p> For GRCh38:
 The GRCh38 file does not have gene co-ordinates information. First you need to add the gene co-ordiates information.
 You can use the Ensembl Perl API to create a script and perform that - <a href="https://www.ensembl.org/info/docs/api/core/index.html">https://www.ensembl.org/info/docs/api/core/index.html</a>.
 After adding the start and end position of the genes at the last 2 columns you can process the file as follows:
<pre class="code sh_sh">cat gnomad.v4.1.constraint_metrics_with_coordinates.tsv | (head -n 1 && tail -n +2  | sort -t$'\t' -k 53,53 -k 56,56n ) &gt loeuf_grch38_temp.tsv
sed '1s/.*/#&/' loeuf_grch38_temp.tsv &gt loeuf_dataset_grch38.tsv
bgzip loeuf_dataset_grch38.tsv
tabix -f -s 53 -b 56 -e 57 loeuf_dataset_grch38.tsv.gz
</pre><p>
 The tabix utility must be installed in your path to use this plugin.</p><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv LOEUF.pm ~/.vep/Plugins
./vep -i variations.vcf --plugin LOEUF,file=/path/to/loeuf/data.tsv.gz,match_by=gene
./vep -i variations.vcf --plugin LOEUF,file=/path/to/loeuf/data.tsv.gz,match_by=transcript
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#00DD66">Gene tolerance to change</div></td><td><a href="https://metacpan.org/pod/Scalar::Util" rel="external">Scalar::Util</a> qw(looks_like_number)</td><td>Ensembl</td></tr>
<tr id="LoFtool" class="bg2 plugin_row" data-category="pathogenicity_predictions"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/LoFtool.pm">LoFtool</a></div><div style="margin-top:6px"><small>Loss-of-function</small></div></td><td><p>  Add LoFtool scores to Ensembl  VEP output. <a class="button" href="#loftool" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('loftool');" id="a_loftool">more</a></p><div id="div_loftool" style="display:none;word-wrap:break-word;"><p>  LoFtool provides a rank of genic intolerance and consequent
  susceptibility to disease based on the ratio of Loss-of-function (LoF)
  to synonymous mutations for each gene in 60,706 individuals from ExAC,
  adjusting for the gene de novo mutation rate and evolutionary protein
  conservation. The lower the LoFtool gene score percentile the most
  intolerant is the gene to functional variation. For more details please see
  (Fadista J et al. 2017), PMID:27563026.
  The authors would like to thank the Exome Aggregation Consortium and
  the groups that provided exome variant data for comparison. A full
  list of contributing groups can be found at <a href="http://exac.broadinstitute.org/about">http://exac.broadinstitute.org/about</a>.</p><p>  The LoFtool_scores.txt file is found alongside the plugin in the
  VEP_plugins GitHub repo.</p><p>  To use another scores file, add it as a parameter i.e.</p><p><pre class="code sh_sh">./vep -i variants.vcf --plugin LoFtool,scores_file.txt
</pre><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv LoFtool.pm ~/.vep/Plugins
mv LoFtool_scores.txt ~/.vep/Plugins
./vep -i variants.vcf --plugin LoFtool
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#1E90FF">Pathogenicity predictions</div></td><td><a href="https://metacpan.org/pod/DBI" rel="external">DBI</a></td><td>Ensembl</td></tr>
<tr id="LOVD" class="bg1 plugin_row" data-category="variant_data"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/LOVD.pm">LOVD</a></div><div style="margin-top:6px"><small>Leiden Open Variation Database</small></div></td><td><p> An Ensembl VEP plugin that retrieves LOVD variation data from <a href="http://www.lovd.nl/">http://www.lovd.nl/</a>. <a class="button" href="#lovd" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('lovd');" id="a_lovd">more</a></p><div id="div_lovd" style="display:none;word-wrap:break-word;"><p> Please be aware that LOVD is a public resource of curated variants, therefore
 please respect this resource and avoid intensive querying of their databases
 using this plugin, as it will impact the availability of this resource for
 others.</p><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv LOVD.pm ~/.vep/Plugins
./vep -i variations.vcf --plugin LOVD
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#B22222">Variant data</div></td><td><a href="https://metacpan.org/pod/LWP::UserAgent" rel="external">LWP::UserAgent</a></td><td>Ensembl</td></tr>
<tr id="Mastermind" class="bg2 plugin_row" data-category="phenotype_data_and_citations"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/Mastermind.pm">Mastermind</a></div></td><td><p> This is a plugin for the Ensembl Variant Effect Predictor (VEP) that
 uses the Mastermind Genomic Search Engine (<a href="https://www.genomenon.com/mastermind">https://www.genomenon.com/mastermind</a>)
 to report variants that have clinical evidence cited in the medical literature. 
 It is available for both GRCh37 and GRCh38. <a class="button" href="#mastermind" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('mastermind');" id="a_mastermind">more</a></p><div id="div_mastermind" style="display:none;word-wrap:break-word;"><p> Please cite the Mastermind publication alongside Ensembl VEP if you use this resource:
 <a href="https://www.frontiersin.org/article/10.3389/fgene.2020.577152">https://www.frontiersin.org/article/10.3389/fgene.2020.577152</a></p><p> Running options:
 The plugin has multiple parameters, the first one is expected to be the file name path 
 which can be followed by 3 optional flags.
 Default: the plugin matches the citation data with the specific mutation.
 Using first flag <kbd>1</kbd>: returns the citations for all mutations/transcripts.   
 Using the second flag <kbd>1</kbd>: only returns the Mastermind variant identifier(s).
 Using the third flag <kbd>1</kbd>: also returns the Mastermind URL.</p><p> Output: 
 The output includes three unique counts 'MMCNT1, MMCNT2, MMCNT3' and one identifier <kbd>MMID3</kbd>
 to be used to build an URL which shows all articles from MMCNT3.</p><p><ul><li><kbd>MMCNT1</kbd> is the count of Mastermind articles with cDNA matches for a specific variant;</li><li><kbd>MMCNT2</kbd> is the count of Mastermind articles with variants either explicitly matching at&nbsp;   the cDNA level or given only at protein level;
</li><li><kbd>MMCNT3</kbd> is the count of Mastermind articles including other DNA-level variants resulting&nbsp;   in the same amino acid change;
</li><li><kbd>MMID3</kbd> is the Mastermind variant identifier(s), as gene:key. Link to the Genomenon Mastermind Genomic Search Engine;
</li></ul><p> To build the URL, substitute the <kbd>gene:key</kbd> in the following link with the value from MMID3:
 <a href="https://mastermind.genomenon.com/detail?mutation=gene:key">https://mastermind.genomenon.com/detail?mutation=gene:key</a></p><p> If the third flag is used then the built URL is returned and it's identified by <kbd>URL</kbd>.</p><p> More information can be found at: <a href="https://www.genomenon.com/cvr/">https://www.genomenon.com/cvr/</a></p><p> The following steps are necessary before running this plugin:</p><p> Download and Registry (free): 
 <a href="https://www.genomenon.com/cvr/">https://www.genomenon.com/cvr/</a> </p><p> GRCh37 VCF:
<pre class="code sh_sh">unzip mastermind_cited_variants_reference-XXXX.XX.XX-grch37-vcf.zip
bgzip mastermind_cited_variants_reference-XXXX.XX.XX-GRCh37-vcf
tabix -p vcf mastermind_cited_variants_reference-XXXX.XX.XX.GRCh37-vcf.gz
</pre><p> GRCh38 VCF:
<pre class="code sh_sh">unzip mastermind_cited_variants_reference-XXXX.XX.XX-grch38-vcf.zip
bgzip mastermind_cited_variants_reference-XXXX.XX.XX-GRCh38-vcf
tabix -p vcf mastermind_cited_variants_reference-XXXX.XX.XX.GRCh38-vcf.gz
</pre><p> 
 The plugin can then be run as default:
<pre class="code sh_sh">./vep -i variations.vcf --plugin Mastermind,file=/path/to/mastermind_cited_variants_reference-XXXX.XX.XX.GRChXX-vcf.gz
</pre><p> or with an option to not filter by mutations (first flag): 
<pre class="code sh_sh">./vep -i variations.vcf --plugin Mastermind,file=/path/to/mastermind_cited_variants_reference-XXXX.XX.XX.GRChXX-vcf.gz,mutations=1 
</pre><p> or with an option to only return <kbd>MMID3</kbd> e.g. the Mastermind variant identifier as gene:key (second flag):
<pre class="code sh_sh">./vep -i variations.vcf --plugin Mastermind,file=/path/to/mastermind_cited_variants_reference-XXXX.XX.XX.GRChXX-vcf.gz,mutations=0,var_iden=1
</pre><p> or with an option to also return the Mastermind URL (third flag):
<pre class="code sh_sh">./vep -i variations.vcf --plugin Mastermind,file=/path/to/mastermind_cited_variants_reference-XXXX.XX.XX.GRChXX-vcf.gz,mutations=0,var_iden=0,url=1
</pre><p> Note: when running Ensembl VEP in offline mode Mastermind requires a fasta file (--fasta)</p><p></ul><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv Mastermind.pm ~/.vep/Plugins
./vep -i variations.vcf --plugin Mastermind,file=/path/to/data.vcf.gz
./vep -i variations.vcf --plugin Mastermind,file=/path/to/data.vcf.gz,mutations=1
./vep -i variations.vcf --plugin Mastermind,file=/path/to/data.vcf.gz,mutations=0,var_iden=1
./vep -i variations.vcf --plugin Mastermind,file=/path/to/data.vcf.gz,mutations=0,var_iden=0,url=1
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#556688">Phenotype data and citations</div></td><td>-</td><td>Ensembl</td></tr>
<tr id="MaveDB" class="bg1 plugin_row" data-category="functional_effect"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/MaveDB.pm">MaveDB</a></div></td><td><p> An Ensembl VEP plugin that retrieves data from MaveDB (<a href="https://www.mavedb.org">https://www.mavedb.org</a>), a
 database that contains multiplex assays of variant effect, including deep
 mutational scans and massively parallel report assays. <a class="button" href="#mavedb" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('mavedb');" id="a_mavedb">more</a></p><div id="div_mavedb" style="display:none;word-wrap:break-word;"><p> To run the MaveDB plugin, please download the following files containing
 MaveDB data for GRCh38 (we do not currently host data for other assemblies):
<ul><li><a href="https://ftp.ensembl.org/pub/current_variation/MaveDB/MaveDB_variants.tsv.gz">https://ftp.ensembl.org/pub/current_variation/MaveDB/MaveDB_variants.tsv.gz</a></li><li><a href="https://ftp.ensembl.org/pub/current_variation/MaveDB/MaveDB_variants.tsv.gz.tbi">https://ftp.ensembl.org/pub/current_variation/MaveDB/MaveDB_variants.tsv.gz.tbi</a>
</li></ul><p> Options are passed to the plugin as key=value pairs:<table class="ss"><thead><tr><th>Argument</th><th>Description</th></tr></thead><tbody><tr class="bg1"><td><pre>file</pre></td><td>(mandatory) Tabix-indexed MaveDB file <tr class="bg2"><td><pre>cols</pre></td><td>Colon-separated columns to print from MaveDB                               files; if set to <kbd>all</kbd>, all columns are
                              printed (default: <kbd>urn:score:nt:pro</kbd>)
<tr class="bg1"><td><pre>single_aminoacid_changes</pre></td><td>Return matches for single aminoacid changes only;                               if disabled, return all matches associated with a
                              genetic variant (default: 1)
<tr class="bg2"><td><pre>transcript_match</pre></td><td>Return results only if (Ensembl or RefSeq)                               transcript identifiers match (default: 1)</p><p></td></tr></tbody></table><p> Please cite the MaveDB publication alongside the VEP if you use this resource:
 <a href="https://doi.org/10.1186/s13059-019-1845-6">https://doi.org/10.1186/s13059-019-1845-6</a></p><p> The tabix utility must be installed in your path to use this plugin.</p><p></ul><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv MaveDB.pm ~/.vep/Plugins</p><p># print only scores for single aminoacid changes from MaveDB data (default)
./vep -i variations.vcf --plugin MaveDB,file=/full/path/to/data.csv.gz</p><p># print all scores associated with the genetic variant
./vep -i variations.vcf --plugin MaveDB,file=/full/path/to/data.csv.gz,single_aminoacid_changes=0</p><p># print all columns from MaveDB data
./vep -i variations.vcf --plugin MaveDB,file=/full/path/to/data.csv.gz,cols=all
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#226600">Functional effect</div></td><td><ul style="padding-left:1em"><li><a href="https://metacpan.org/pod/Bio::SeqUtils" rel="external">Bio::SeqUtils</a></li><li><a href="https://metacpan.org/pod/File::Basename" rel="external">File::Basename</a></li></ul></td><td>Ensembl</td></tr>
<tr id="MaxEntScan" class="bg2 plugin_row" data-category="splicing_predictions"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/MaxEntScan.pm">MaxEntScan</a></div></td><td><p> This is a plugin for the Ensembl Variant Effect Predictor (VEP) that
 runs MaxEntScan (<a href="http://hollywood.mit.edu/burgelab/maxent/Xmaxentscan_scoreseq.html">http://hollywood.mit.edu/burgelab/maxent/Xmaxentscan_scoreseq.html</a>)
 to get splice site predictions. <a class="button" href="#maxentscan" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('maxentscan');" id="a_maxentscan">more</a></p><div id="div_maxentscan" style="display:none;word-wrap:break-word;"><p> The plugin copies most of the code verbatim from the score5.pl and score3.pl
 scripts provided in the MaxEntScan download. To run the plugin you must get and
 unpack the archive from <a href="http://hollywood.mit.edu/burgelab/maxent/download/">http://hollywood.mit.edu/burgelab/maxent/download/</a>; the path
 to this unpacked directory is then the param you pass to the --plugin flag.</p><p> The plugin executes the logic from one of the scripts depending on which
 splice region the variant overlaps:</p><p><ul><li>score5.pl : last 3 bases of exon    --&gt first 6 bases of intron</li><li>score3.pl : last 20 bases of intron --&gt first 3 bases of exon
</li></ul><p> The plugin reports the reference, alternate and difference (REF - ALT) maximum
 entropy scores.</p><p> If <kbd>SWA</kbd> is specified as a command-line argument, a sliding window algorithm
 is applied to subsequences containing the reference and alternate alleles to
 identify k-mers with the highest donor and acceptor splice site scores. To assess
 the impact of variants, reference comparison scores are also provided. For SNVs,
 the comparison scores are derived from sequence in the same frame as the highest
 scoring k-mers containing the alternate allele. For all other variants, the
 comparison scores are derived from the highest scoring k-mers containing the
 reference allele. The difference between the reference comparison and alternate
 scores (SWA_REF_COMP - SWA_ALT) are also provided.</p><p> If <kbd>NCSS</kbd> is specified as a command-line argument, scores for the nearest
 upstream and downstream canonical splice sites are also included.</p><p> By default, only scores are reported. Add <kbd>verbose</kbd> to the list of command-
 line arguments to include the sequence output associated with those scores.</p><p></ul><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv MaxEntScan.pm ~/.vep/Plugins
./vep -i variants.vcf --plugin MaxEntScan,/path/to/maxentscan/fordownload
./vep -i variants.vcf --plugin MaxEntScan,/path/to/maxentscan/fordownload,SWA,NCSS
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#FF0000">Splicing predictions</div></td><td><a href="https://metacpan.org/pod/Digest::MD5" rel="external">Digest::MD5</a> qw(md5_hex)</td><td>Ensembl</td></tr>
<tr id="MPC" class="bg1 plugin_row" data-category="pathogenicity_predictions"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/MPC.pm">MPC</a></div><div style="margin-top:6px"><small>missense deleteriousness metric</small></div></td><td><p> An Ensembl VEP plugin that retrieves MPC scores for variants from a tabix-indexed MPC data file. <a class="button" href="#mpc" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('mpc');" id="a_mpc">more</a></p><div id="div_mpc" style="display:none;word-wrap:break-word;"><p> MPC is a missense deleteriousness metric based on the analysis of genic regions
 depleted of missense mutations in the Exome Agggregation Consortium (ExAC) data.</p><p> The MPC score is the product of work by Kaitlin Samocha (ks20@sanger.ac.uk).
 Publication currently in pre-print: Samocha et al bioRxiv 2017 (TBD)</p><p> The MPC score file is available to download from:</p><p> <a href="https://ftp.broadinstitute.org/pub/ExAC_release/release1/regional_missense_constraint/">https://ftp.broadinstitute.org/pub/ExAC_release/release1/regional_missense_constraint/</a></p><p> The data are currently mapped to GRCh37 only. Not all transcripts are included; see
 README in the above directory for exclusion criteria.</p><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv MPC.pm ~/.vep/Plugins
./vep -i variations.vcf --plugin MPC,fordist_constraint_official_mpc_values.txt.gz
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#1E90FF">Pathogenicity predictions</div></td><td>-</td><td>Ensembl</td></tr>
<tr id="MTR" class="bg2 plugin_row" data-category="pathogenicity_predictions"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/MTR.pm">MTR</a></div><div style="margin-top:6px"><small>Missense Tolerance Ratio</small></div></td><td><p>An Ensembl VEP plugin that retrieves Missense Tolerance Ratio (MTR) scores for
variants from a tabix-indexed flat file. <a class="button" href="#mtr" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('mtr');" id="a_mtr">more</a></p><div id="div_mtr" style="display:none;word-wrap:break-word;"><p>MTR scores quantify the amount of purifying selection acting
specifically on missense variants in a given window of protein-coding
sequence. It is estimated across a sliding window of 31 codons and uses
observed standing variation data from the WES component of the Exome
Aggregation Consortium Database (ExAC), version 2.0
(<a href="http://gnomad.broadinstitute.org">http://gnomad.broadinstitute.org</a>).</p><p>Please cite the MTR publication alongside Ensembl VEP if you use this resource:
<a href="http://genome.cshlp.org/content/27/10/1715">http://genome.cshlp.org/content/27/10/1715</a></p><p>The Bio::DB::HTS perl library or tabix utility must be installed in your path to use this plugin. 
MTR flat files can be downloaded from <a href="http://biosig.unimelb.edu.au/mtr-viewer/downloads">http://biosig.unimelb.edu.au/mtr-viewer/downloads</a>
The following steps are necessary before running the plugin 
<pre class="code sh_sh">gzip -d mtrflatfile_2.0.txt.gz # to unzip the text file 
cat mtrflatfile_2.0.txt | tr " " "\t" &gt mtrflatfile_2.00.tsv # to change the file to a tabbed delimited file 
sed '1s/.*/#&/'  mtrflatfile_2.00.tsv &gt mtrflatfile_2.0.tsv # to add # to the first line of the file 
bgzip mtrflatfile_2.0.tsv
tabix -f -s 1 -b 2 -e 2 mtrflatfile_2.0.tsv.gz
</pre><p>NB: Data are available for GRCh37 only</p><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv MTR.pm ~/.vep/Plugins
./vep -i variations.vcf --plugin MTR,mtrflatfile_2.0.tsv.gz
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#1E90FF">Pathogenicity predictions</div></td><td>-</td><td><ul style="padding-left:1em"><li>Slave Petrovski</li><li>Michael Silk</li></ul></td></tr>
<tr id="mutfunc" class="bg1 plugin_row" data-category="protein_annotation"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/mutfunc.pm">mutfunc</a></div></td><td><p> An Ensembl VEP plugin that retrieves data from mutfunc db predicting destabilization of protein structure, interaction interface, and motif. <a class="button" href="#mutfunc" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('mutfunc');" id="a_mutfunc">more</a></p><div id="div_mutfunc" style="display:none;word-wrap:break-word;"><p> Please cite the mutfunc publication alongside Ensembl VEP if you use this resource:
 <a href="http://msb.embopress.org/content/14/12/e8430">http://msb.embopress.org/content/14/12/e8430</a></p><p> Pre-requisites:</p><p><ol><li value="1">The data file. mutfunc SQLite db can be downloaded from -  <a href="https://ftp.ensembl.org/pub/current_variation/mutfunc/mutfunc_data.db">https://ftp.ensembl.org/pub/current_variation/mutfunc/mutfunc_data.db</a></p><p></li><li value="2">If you are using --offline please provide a FASTA file as this plugin requires the translation sequence to function.</p><p></li></ol><p> Options are passed to the plugin as key=value pairs:<table class="ss"><thead><tr><th>Argument</th><th>Description</th></tr></thead><tbody>
<tr class="bg1"><td><pre>db</pre></td><td>(mandatory) Path to SQLite database containing data for other analysis. <tr class="bg2"><td><pre>motif</pre></td><td>Select this option to have mutfunc motif analysis in the output <tr class="bg1"><td><pre>int</pre></td><td>Select this option to have mutfunc protein interection analysis in the output <tr class="bg2"><td><pre>mod</pre></td><td>Select this option to have mutfunc protein structure analysis in the output <tr class="bg1"><td><pre>exp</pre></td><td>Select this option to have mutfunc protein structure (experimental) analysis in the output <tr class="bg2"><td><pre>extended</pre></td><td>By default mutfunc outputs the most significant field for any analysis. Select this option to get more verbose output. 
</td></tr></tbody></table><p> By default all of the four type of analysis (motif, int, mod, and exp) data are available in the output. But if you want to have 
 some selected analysis and not all of them just select the relevant options.</p><p></ol><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv mutfunc.pm ~/.vep/Plugins
./vep -i variations.vcf --plugin mutfunc,motif=1,extended=1,db=/FULL_PATH_TO/mutfunc_data.db
./vep -i variations.vcf --plugin mutfunc,db=/FULL_PATH_TO/mutfunc_data.db
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#8844CC">Protein annotation</div></td><td><ul style="padding-left:1em"><li><a href="https://metacpan.org/pod/List::MoreUtils" rel="external">List::MoreUtils</a> qw(first_index)</li><li><a href="https://metacpan.org/pod/Compress::Zlib" rel="external">Compress::Zlib</a></li><li><a href="https://metacpan.org/pod/Digest::MD5" rel="external">Digest::MD5</a> qw(md5_hex)</li><li><a href="https://metacpan.org/pod/DBI" rel="external">DBI</a></li></ul></td><td>Ensembl</td></tr>
<tr id="NearestExonJB" class="bg2 plugin_row" data-category="nearby_features"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/NearestExonJB.pm">NearestExonJB</a></div></td><td><p> This is a plugin for the Ensembl Variant Effect Predictor (VEP) that
 finds the nearest exon junction boundary to a coding sequence variant. More than 
 one boundary may be reported if the boundaries are equidistant. <a class="button" href="#nearestexonjb" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('nearestexonjb');" id="a_nearestexonjb">more</a></p><div id="div_nearestexonjb" style="display:none;word-wrap:break-word;"><p> The plugin will report the Ensembl identifier of the exon, the distance to the
 exon boundary, the boundary type (start or end of exon) and the total
 length in nucleotides of the exon.</p><p> Various key=value parameters can be altered by passing them to the plugin command:<table class="ss"><thead><tr><th>Argument</th><th>Description</th></tr></thead><tbody>
<tr class="bg1"><td><pre>max_range</pre></td><td>maximum search range in bp (default: 10000) 
</td></tr></tbody></table><p> Parameters are passed e.g.:</p><p><pre class="code sh_sh">--plugin NearestExonJB,max_range=50000
</pre><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv NearestExonJB.pm ~/.vep/Plugins
./vep -i variations.vcf --cache --plugin NearestExonJB
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#E75480">Nearby features</div></td><td>-</td><td>Ensembl</td></tr>
<tr id="NearestGene" class="bg1 plugin_row" data-category="nearby_features"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/NearestGene.pm">NearestGene</a></div></td><td><p> This is a plugin for the Ensembl Variant Effect Predictor (VEP) that
 finds the nearest gene(s) to a non-genic variant. More than one gene
 may be reported if the genes overlap the variant or if genes are
 equidistant. <a class="button" href="#nearestgene" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('nearestgene');" id="a_nearestgene">more</a></p><div id="div_nearestgene" style="display:none;word-wrap:break-word;"><p> Various key=value parameters can be altered by passing them to the plugin command:<table class="ss"><thead><tr><th>Argument</th><th>Description</th></tr></thead><tbody>
<tr class="bg1"><td><pre>limit</pre></td><td>limit the number of genes returned (default: 1) <tr class="bg2"><td><pre>range</pre></td><td>initial search range in bp (default: 1000) <tr class="bg1"><td><pre>max_range</pre></td><td>maximum search range in bp (default: 10000) 
</td></tr></tbody></table><p> Parameters are passed e.g.:</p><p><pre class="code sh_sh">--plugin NearestGene,limit=3,max_range=50000
</pre><p> This plugin requires a database connection. It cannot be run with Ensembl VEP
 in offline mode i.e. using the --offline flag.</p><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv NearestGene.pm ~/.vep/Plugins
./vep -i variations.vcf --cache --plugin NearestGene
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#E75480">Nearby features</div></td><td>-</td><td>Ensembl</td></tr>
<tr id="neXtProt" class="bg2 plugin_row" data-category="protein_data"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/neXtProt.pm">neXtProt</a></div></td><td><p> This is a plugin for the Ensembl Variant Effect Predictor (VEP) that
 retrieves data for missense and stop gain variants from neXtProt, which is a comprehensive 
 human-centric discovery platform that offers integration of and navigation 
 through protein-related data for example, variant information, localization 
 and interactions  (<a href="https://www.nextprot.org/">https://www.nextprot.org/</a>). <a class="button" href="#nextprot" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('nextprot');" id="a_nextprot">more</a></p><div id="div_nextprot" style="display:none;word-wrap:break-word;"><p> Please cite the neXtProt publication alongside Ensembl VEP if you use this resource:
 <a href="https://doi.org/10.1093/nar/gkz995">https://doi.org/10.1093/nar/gkz995</a></p><p> This plugin is only suitable for small sets of variants as an additional 
 individual remote API query is run for each variant.</p><p> The neXtProt_headers.txt file is a requirement for running this plugin and is found alongside the plugin in the
 VEP_plugins GitHub repository. The file contains the RDF entities extracted from 
 	<a href="https://snorql.nextprot.org/">https://snorql.nextprot.org/</a></p><p> Running options:
 (Default) the data retrieved by default is the MatureProtein, NucleotidePhosphateBindingRegion,
 Variant, MiscellaneousRegion, TopologicalDomain and InteractingRegion.
 The plugin can also be run with other options to retrieve other data than the default.</p><p> Options are passed to the plugin as key=value pairs:<table class="ss"><thead><tr><th>Argument</th><th>Description</th></tr></thead><tbody><tr class="bg1"><td><pre>max_set</pre></td><td>Set value to 1 to return all available protein-related data                   (includes the default data)</p><p><tr class="bg2"><td><pre>return_values</pre></td><td>The set of data to be returned with different data separated by <kbd>&</kbd>.                   Use file <kbd>neXtProt_headers.txt</kbd> to check which data (labels) are available.
                  Example: --plugin neXtProt,return_values=<kbd>Domain&InteractingRegion</kbd></p><p><tr class="bg1"><td><pre>url</pre></td><td>Set value to 1 to include the URL to link to the neXtProt entry. 
<tr class="bg2"><td><pre>all_labels</pre></td><td>Set value to 1 to include all labels, even if data is not available. 
<tr class="bg1"><td><pre>position</pre></td><td>Set value to 1 to include the start and end position in the protein. 
</td></tr></tbody></table><p> (*) note: <kbd>max_set</kbd> and <kbd>return_values</kbd> cannot be used simultaneously.</p><p> Output:
  By default, the plugin only returns data that is available. Example (default behaviour):
<pre class="code sh_sh">neXtProt_MatureProtein=Rho guanine nucleotide exchange factor 10
</pre>
  The option <kbd>all_labels</kbd> returns a consistent set of the requested fields, using "-" where 
  values are not available. Same example as above:
<pre class="code sh_sh">neXtProt_MatureProtein=Rho guanine nucleotide exchange factor 10;
neXtProt_InteractingRegion=-;neXtProt_NucleotidePhosphateBindingRegion=-;neXtProt_Variant=-;
neXtProt_MiscellaneousRegion=-;neXtProt_TopologicalDomain=-;
</pre>
  Of notice, multiple values can be returned for the same label. In this case, the values will
  be separeted by <kbd>|</kbd> for tab and txt format, and <kbd>&</kbd> for VCF format. </p><p>  N/B: This plugin requires a connection to the Ensembl database, and can not be used in offline mode.</p><p> The plugin can then be run as default:
<pre class="code sh_sh">./vep -i variations.vcf --plugin neXtProt
</pre><p> or to return only the data specified by the user:
<pre class="code sh_sh">./vep -i variations.vcf --plugin neXtProt,return_values=<kbd>Domain&InteractingRegion</kbd>
</pre><p>
<p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv neXtProt.pm ~/.vep/Plugins
./vep -i variations.vcf --plugin neXtProt
./vep -i variations.vcf --plugin neXtProt,max_set=1
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#77FF55">Protein data</div></td><td><a href="https://metacpan.org/pod/JSON::XS" rel="external">JSON::XS</a></td><td>Ensembl</td></tr>
<tr id="NMD" class="bg1 plugin_row" data-category="transcript_annotation"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/NMD.pm">NMD</a></div></td><td><p> This is a plugin for the Ensembl Variant Effect Predictor (VEP) 
 that predicts if a variant allows the 
 transcript escape nonsense-mediated mRNA decay based on certain rules. <a class="button" href="#nmd" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('nmd');" id="a_nmd">more</a></p><div id="div_nmd" style="display:none;word-wrap:break-word;"><p> The rules are : </p><p><ol><li value="1">The variant location  falls in the last exon of the transcript.<pre class="code sh_sh">                                        vvvv
      ES...EE..I.ES...EE.I.ES....EE.I.ES....EE 
</pre> (ES= exon_start,EE = exon_end, I = intron, v = variant location)</p><p></li><li value="2">The variant location falls  50 bases upstream of the penultimate (second to the last ) exon.<pre class="code sh_sh">                               vvv
      ES...EE..I.ES...EE.I.ES....EE.I.ES....EE 
</pre> (ES= exon_start,EE = exon_end, I = intron, v = variant location)</p><p></li><li value="3">The variant falls in the first 100 coding bases in the transcript.<pre class="code sh_sh">         vvv
      ..ES...EE..I.ES...EE.I.ES....EE.I.ES....EE 
</pre> (ES= exon_start,EE = exon_end, I = intron, v = variant location)</p><p></li><li value="4">If the variant is in an intronless transcript, meaning only one exon exist in the transcript.  
</li></ol><p> The additional term NMD-escaping variant (nonsense-mediated mRNA decay escaping variants) will be added if the variant matches any of the rules.</p><p> REFERENCES :
<ul><li>Identifying Genes Whose Mutant Transcripts Cause Dominant Disease Traits by Potential Gain-of-Function Alleles (Coban-Akdemir, 2018)</li><li>The rules and impact of nonsense-mediated mRNA decay in human cancers (Lindeboom, 2016)
</li></ul><p>
</ul><p></ol><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv NMD.pm  ~/.vep/Plugins
./vep -i variations.vcf --plugin NMD
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#FF2288">Transcript annotation</div></td><td>-</td><td>Ensembl</td></tr>
<tr id="OpenTargets" class="bg2 plugin_row" data-category="variant_data"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/OpenTargets.pm">OpenTargets</a></div></td><td><p> An Ensembl VEP plugin that integrates data from Open Targets Genetics
 (<a href="https://genetics.opentargets.org">https://genetics.opentargets.org</a>), a tool that highlights variant-centric
 statistical evidence to allow both prioritisation of candidate causal variants
 at trait-associated loci and identification of potential drug targets. <a class="button" href="#opentargets" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('opentargets');" id="a_opentargets">more</a></p><div id="div_opentargets" style="display:none;word-wrap:break-word;"><p> Data from Open Targets Genetics includes locus-to-gene (L2G) scores to predict
 causal genes at GWAS loci.</p><p> The tabix utility must be installed in your path to use this plugin. The Open
 Targets Genetics file and respective index (TBI) file can be downloaded from:
 <a href="https://ftp.ebi.ac.uk/pub/databases/opentargets/genetics/latest/OTGenetics_VEP">https://ftp.ebi.ac.uk/pub/databases/opentargets/genetics/latest/OTGenetics_VEP</a></p><p> Options are passed to the plugin as key=value pairs:<table class="ss"><thead><tr><th>Argument</th><th>Description</th></tr></thead><tbody><tr class="bg1"><td><pre>file</pre></td><td>(mandatory) Tabix-indexed file from Open Targets Genetics <tr class="bg2"><td><pre>cols</pre></td><td>(optional) Colon-separated list of columns to return from the plugin           file (default: "l2g:geneId"); use <kbd>all</kbd> to print all data</p><p></td></tr></tbody></table><p> Please cite the Open Targets Genetics publication alongside Ensembl VEP if you use
 this resource: <a href="https://doi.org/10.1093/nar/gkaa84">https://doi.org/10.1093/nar/gkaa84</a></p><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv OpenTargets.pm ~/.vep/Plugins</p><p># print Open Targets Genetics scores and respective gene identifiers (default)
./vep -i variations.vcf --plugin OpenTargets,file=path/to/data.tsv.bz</p><p># print all information from Open Targets Genetics
./vep -i variations.vcf --plugin OpenTargets,file=path/to/data.tsv.bz,cols=all
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#B22222">Variant data</div></td><td><ul style="padding-left:1em"><li><a href="https://metacpan.org/pod/Bio::SeqUtils" rel="external">Bio::SeqUtils</a></li><li><a href="https://metacpan.org/pod/File::Basename" rel="external">File::Basename</a></li></ul></td><td>Ensembl</td></tr>
<tr id="Paralogues" class="bg1 plugin_row" data-category="variant_data"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/Paralogues.pm">Paralogues</a></div></td><td><p> An Ensembl VEP plugin that fetches variants overlapping the genomic coordinates of amino
 acids aligned between paralogue proteins. This is useful to predict the
 pathogenicity of variants in paralogue positions. <a class="button" href="#paralogues" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('paralogues');" id="a_paralogues">more</a></p><div id="div_paralogues" style="display:none;word-wrap:break-word;"><p> This plugin can determine paralogue regions for a variant based on:
<ol><li value="1">Pre-computed matches between genomic regions and paralogue variants.      For this approach, either download the file calculated using ClinVar variants and respective TBI from
      <a href="https://ftp.ensembl.org/pub/current_variation/Paralogues">https://ftp.ensembl.org/pub/current_variation/Paralogues</a> or create such matches file yourself. Details on how
      to create such <kbd>matches</kbd> file can be found below.
</li><li value="2">Ensembl paralogue annotation. These versatile annotations can look up      paralogue regions for all variants from any species with Ensembl
      paralogues, but take longer to process.</p><p></li></ol><p> After retrieving the paralogue regions, this plugin fetches variants
 overlapping those regions from one of the following sources (by this order):
<ol><li value="1">Custom VCF via the <kbd>vcf</kbd> parameter</li><li value="2">Ensembl VEP cache (in cache/offline mode)</li><li value="3">Ensembl API (in database mode)
</li></ol><p> To create a <kbd>matches</kbd> file based on a custom set of variants, run Ensembl VEP using
 `--plugin Paralogues,regions=1,min_perc_cov=0,min_perc_pos=0,clnsig=ignore`
 and the <kbd>--vcf</kbd> option. Afterwards, process the output of the Ensembl VEP command:
 `perl -e "use Paralogues; Paralogues::prepare_matches_file(<kbd>variant_effect_output.txt</kbd>)"`</p><p> Options are passed to the plugin as key=value pairs:<table class="ss"><thead><tr><th>Argument</th><th>Description</th></tr></thead><tbody><tr class="bg1"><td><pre>matches</pre></td><td>Tabix-indexed TSV file with pre-computed matches between                   genomic regions and paralogue variants (fastest method); this
                  option is incompatible with the <kbd>paralogues</kbd> and <kbd>vcf</kbd> options</p><p><tr class="bg2"><td><pre>dir</pre></td><td>Directory with paralogue annotation (the annotation is created                   in this folder if the paralogue annotation files do not exist)
<tr class="bg1"><td><pre>paralogues</pre></td><td>Tabix-indexed TSV file with paralogue annotation (if the file                   does not exist, the annotation is automatically created); if
                  set to <kbd>remote</kbd>, the annotation is fetched but not stored
<tr class="bg2"><td><pre>vcf</pre></td><td>Tabix-indexed VCF file to fetch variant information (if not                   used, variants are fetched from the Ensembl VEP cache in cache/offline
                  mode or Ensembl API in database mode)</p><p><tr class="bg1"><td><pre>fields</pre></td><td>Colon-separated list of information from paralogue variants to                   output (default: <kbd>identifier:alleles:clinical_significance</kbd>);
                  keyword <kbd>all</kbd> can be used to print all fields; available
                  fields include <kbd>identifier</kbd>, <kbd>chromosome</kbd>, <kbd>start</kbd>, <kbd>alleles</kbd>,
                  <kbd>perc_cov</kbd>, <kbd>perc_pos</kbd>, and <kbd>clinical_significance</kbd> (if
                  <kbd>clnsig_col</kbd> is defined for custom VCF); additional fields
                  are available depending on variant source:
<ul><li>VEP cache: <kbd>end</kbd> and <kbd>strand</kbd></li><li>Ensembl API: <kbd>end</kbd>, <kbd>strand</kbd>, <kbd>source</kbd>, <kbd>consequence</kbd> and&nbsp;                      <kbd>gene_symbol</kbd>
</li><li>Custom VCF: <kbd>quality</kbd>, <kbd>filter</kbd> and name of INFO fields</li><li>Matches file: check column names in file header
<tr class="bg2"><td><pre></li></ul><p>   clnsig</pre></td><td>Clinical significance term to filter variants (default:                   <kbd>pathogenic</kbd>); use <kbd>ignore</kbd> to fetch all paralogue variants,
                  regardless of clinical significance
<tr class="bg1"><td><pre>clnsig_match</pre></td><td>Type of match when filtering variants based on option                   <kbd>clnsig</kbd>: <kbd>partial</kbd> (default), <kbd>exact</kbd> or <kbd>regex</kbd>
<tr class="bg2"><td><pre>clnsig_col</pre></td><td>Column name containing clinical significance in custom VCF                   (required with <kbd>vcf</kbd> option and if <kbd>clnsig</kbd> is not <kbd>ignore</kbd>)</p><p><tr class="bg1"><td><pre>min_perc_cov</pre></td><td>Minimum alignment percentage of the peptide associated with                   the input variant (default: 0)
<tr class="bg2"><td><pre>min_perc_pos</pre></td><td>Minimum percentage of positivity (similarity) between both                   homologues (default: 50)
<tr class="bg1"><td><pre>regions</pre></td><td>Boolean value to return regions used to look up paralogue <pre class="code sh_sh">                  variants (default: 1)
</pre><p></td></tr></tbody></table><p> The tabix utility must be installed in your path to read the paralogue
 annotation, the custom VCF file and the matches file.</p><p></ul><p></ol><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv Paralogues.pm ~/.vep/Plugins</p><p># Find paralogue regions of all input variants using Ensembl paralogue annotation
# (automatically created if not in current directory) and fetch variants within
# those regions from the Ensembl VEP cache and whose clinical significance partially
# matches 'pathogenic'
./vep -i variations.vcf --cache --plugin Paralogues</p><p># Find paralogue regions of input variants using Ensembl paralogue annotation
# (automatically created if not in current directory) and fetch variants within
# those regions from a custom VCF file (regardless of their clinical significance)
./vep -i variations.vcf --cache --plugin Paralogues,vcf=/path/to/file.vcf,clnsig=ignore</p><p># Same using a custom VCF file but filtering for 'pathogenic' variants
./vep -i variations.vcf --cache --plugin Paralogues,vcf=/path/to/file.vcf,clnsig_col=CLNSIG</p><p># Same but output different fields
./vep -i variations.vcf --cache --plugin Paralogues,vcf=/path/to/file.vcf.gz,clnsig_col=CLNSIG,fields=identifier:alleles:CLNSIG:CLNVI:GENEINFO</p><p># Use a file with regions matched to paralogue variants -- fastest method;
# download 'matches' files from https://ftp.ensembl.org/pub/current_variation/Paralogues
./vep -i variations.vcf --cache --plugin Paralogues,matches=Paralogues.pm_homo_sapiens_113_GRCh38_clinvar_20240107.tsv.gz,clnsig=ignore</p><p># Same using a 'matches' file but filtering for 'pathogenic' variants (default)
./vep -i variations.vcf --cache --plugin Paralogues,matches=Paralogues.pm_homo_sapiens_113_GRCh38_clinvar_20240107.tsv.gz</p><p># Fetch all Ensembl variants in paralogue proteins using only the Ensembl API
# (requires database access)
./vep -i variations.vcf --database --plugin Paralogues,mode=remote,clnsig=ignore
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#B22222">Variant data</div></td><td><ul style="padding-left:1em"><li><a href="https://metacpan.org/pod/Compress::Zlib" rel="external">Compress::Zlib</a></li><li><a href="https://metacpan.org/pod/Bio::SimpleAlign" rel="external">Bio::SimpleAlign</a></li><li><a href="https://metacpan.org/pod/File::Spec" rel="external">File::Spec</a></li><li><a href="https://metacpan.org/pod/List::Util" rel="external">List::Util</a> qw(any)</li><li><a href="https://metacpan.org/pod/File::Basename" rel="external">File::Basename</a></li></ul></td><td>Ensembl</td></tr>
<tr id="PhenotypeOrthologous" class="bg2 plugin_row" data-category="phenotype_data_and_citations"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/PhenotypeOrthologous.pm">PhenotypeOrthologous</a></div></td><td><p> An Ensembl VEP plugin that retrieves phenotype information associated with orthologous genes from model organisms. <a class="button" href="#phenotypeorthologous" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('phenotypeorthologous');" id="a_phenotypeorthologous">more</a></p><div id="div_phenotypeorthologous" style="display:none;word-wrap:break-word;"><p> The plugin annotates human variants and reports orthologous information from rat and mouse. 
 The plugin is only available for GRCh38.</p><p> The PhenotypeOrthologous file can be downloaded from <a href="https://ftp.ensembl.org/pub/current_variation/PhenotypeOrthologous">https://ftp.ensembl.org/pub/current_variation/PhenotypeOrthologous</a></p><p> The plugin can be run: </p><p><pre class="code sh_sh">./vep -i variations.vcf --plugin PhenotypeOrthologous,file=PhenotypesOrthologous_homo_sapiens_112_GRCh38.gff3.gz
</pre><p>  The file option is mandatory to run this plugin </p><p>  To return only results for rat :
<pre class="code sh_sh">./vep -i variations.vcf --plugin PhenotypeOrthologous,file=PhenotypesOrthologous_homo_sapiens_112_GRCh38.gff3.gz,model=rat
</pre><p>  To return only results for mouse:
<pre class="code sh_sh">./vep -i variations.vcf --plugin PhenotypeOrthologous,file=PhenotypesOrthologous_homo_sapiens_112_GRCh38.gff3.gz,model=mouse
</pre><p> The tabix utility must be installed in your path to use this plugin.
 Check <a href="https://github.com/samtools/htslib.git">https://github.com/samtools/htslib.git</a> for instructions.</p><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv PhenotypeOrthologous.pm ~/.vep/Plugins
./vep -i variations.vcf --plugin PhenotypeOrthologous,file=PhenotypesOrthologous_homo_sapiens_112_GRCh38.gff3.gz
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#556688">Phenotype data and citations</div></td><td>-</td><td>Ensembl</td></tr>
<tr id="Phenotypes" class="bg1 plugin_row" data-category="phenotype_data_and_citations"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/Phenotypes.pm">Phenotypes</a></div></td><td><p> A VEP plugin that retrieves overlapping phenotype information. <a class="button" href="#phenotypes" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('phenotypes');" id="a_phenotypes">more</a></p><div id="div_phenotypes" style="display:none;word-wrap:break-word;"><p> On the first run for each new version/species/assembly will
 download a GFF-format dump to ~/.vep/Plugins/</p><p> Ensembl provides phenotype annotations mapped to a number of genomic
 feature types, including genes, variants and QTLs.</p><p> This plugin is best used with JSON output format; the output will be
 more verbose and include all available phenotype annotation data and
 metadata.</p><p> For other output formats, only a concatenated list of phenotype
 description strings is returned.</p><p> Several paramters can be set using a key=value system:<table class="ss"><thead><tr><th>Argument</th><th>Description</th></tr></thead><tbody>
<tr class="bg1"><td><pre>dir</pre></td><td>Path to directory where to look for phenotypes annotation. If                   the required file does not exist, the file is downloaded and
                  saved in the provided directory (download requires using
                  database or cache mode).</p><p><tr class="bg2"><td><pre>file</pre></td><td>File path to phenotypes annotation. If the file does                   not exist, the file is downloaded and saved with this name
                  (download requires using database or cache mode).</p><p></td></tr></tbody></table><p> exclude_sources: &-separated list of phenotype sources to exclude. By default,
                  HGMD-PUBLIC and COSMIC annotations are excluded. See
                  <a href="http://www.ensembl.org/info/genome/variation/phenotype/sources_phenotype_documentation.html">http://www.ensembl.org/info/genome/variation/phenotype/sources_phenotype_documentation.html</a></p><p> include_sources: &-separated list of phenotype sources to include. If defined,
                  exclude_sources is ignored.</p><p> exclude_types  : &-separated list of feature types to exclude: Gene, Variation,
                  QTL, StructuralVariation, SupportingStructuralVariation,
                  RegulatoryFeature. By default, StructuralVariation and 
                  SupportingStructuralVariation annotations are always excluded
                  (due to size issues) and Variation is excluded when annotating
                  structural variants; to get these annotations in all cases, use
                  include_types=StructuralVariation&SupportingStructuralVariation&Variation</p><p> include_types  : &-separated list of feature types to include. If defined,
                  exclude_types is ignored.</p><p> expand_right   : Cache size in bp. By default, annotations 100000bp (100kb)
                  downstream of the initial lookup are cached.</p><p> phenotype_feature : Boolean to report the gene/variation associated with the
                     phenotype (such as overlapping gene or structural
<pre class="code sh_sh">                     variation) and annotation source (default: 0)
</pre><p> cols           : &-separated list of column and/or attribute names to output
                  from the gff file. The output fields will be ordered in the same
                  way given in cols argument. (default: <kbd>phenotype</kbd> or <kbd>source,phenotype,id</kbd>
                  if you set phenotype_feature=1)</p><p> id_match       : Return results only if the identifiers matches with the 
<pre class="code sh_sh">                  variant or the gene depending on the type (default: 0)
</pre><p> Example:</p><p><pre class="code sh_sh">--plugin Phenotypes,file=${HOME}/phenotypes.gff.gz,include_types=Gene
--plugin Phenotypes,dir=${HOME},include_types=Gene
</pre><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv Phenotypes.pm ~/.vep/Plugins</p><p># Automatically download phenotype annotation files if needed and annotate
# variants with phenotypes
./vep -i variations.vcf --plugin Phenotypes</p><p># Fetch only gene-associated phenotypes
./vep -i variations.vcf --plugin Phenotypes,include_types=Gene</p><p># Set directory with phenotypes annotations (phenotype annotation file is
# automatically downloaded if not available in this directory)
./vep -i variations.vcf --plugin Phenotypes,dir=${HOME},include_types=Gene</p><p># Specify a file with phenotypes annotation (file is automatically
# downloaded and saved with this name if it does not exist)
./vep -i variations.vcf --plugin Phenotypes,file=${HOME}/phenotypes.gff.gz,include_types=Gene
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#556688">Phenotype data and citations</div></td><td>-</td><td>Ensembl</td></tr>
<tr id="pLI" class="bg2 plugin_row" data-category="gene_tolerance_to_change"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/pLI.pm">pLI</a></div></td><td><p>  An Ensembl VEP plugin that adds the probabililty of a gene being 
  loss-of-function intolerant (pLI) to the Ensembl VEP output. <a class="button" href="#pli" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('pli');" id="a_pli">more</a></p><div id="div_pli" style="display:none;word-wrap:break-word;"><p>  Lek et al. (2016) estimated pLI using the expectation-maximization 
  (EM) algorithm and data from 60,706 individuals from 
  ExAC (<a href="http://exac.broadinstitute.org">http://exac.broadinstitute.org</a>). The closer pLI is to 1, 
  the more likely the gene is loss-of-function (LoF) intolerant. </p><p>  Note: the pLI was calculated using a representative transcript and
  is reported by gene in the plugin.</p><p>  The data for the plugin is provided by Kaitlin Samocha and Daniel MacArthur. 
  See <a href="https://www.ncbi.nlm.nih.gov/pubmed/27535533">https://www.ncbi.nlm.nih.gov/pubmed/27535533</a> for a description 
  of the dataset and analysis.</p><p>  The pLI_values.txt file is found alongside the plugin in the 
  VEP_plugins GitHub repository. The file contains the fields gene and pLI 
  extracted from the file at </p><p>    <a href="https://ftp.broadinstitute.org/pub/ExAC_release/release0.3/functional_gene_constraint/fordist_cleaned_exac_r03_march16_z_pli_rec_null_data.txt">https://ftp.broadinstitute.org/pub/ExAC_release/release0.3/functional_gene_constraint/fordist_cleaned_exac_r03_march16_z_pli_rec_null_data.txt</a></p><p>  From this file, extract gene or transcipt pLI scores:
   To extract gene scores : 
<pre class="code sh_sh">awk '{print $2, $20 }'  fordist_cleaned_exac_r03_march16_z_pli_rec_null_data.txt &gt plI_gene.txt 
</pre><p>  NB: The gene scores file can also be found in the VEP_plugins directory.</p><p>   To extract transcript scores:
<pre class="code sh_sh">awk '{print $1, $20 }'  fordist_cleaned_exac_r03_march16_z_pli_rec_null_data.txt &gt plI_transcript.txt 
</pre><p>  NB: Using this file, No transcript score will be returned.</p><p>  To use another values file, add it as a parameter i.e.
<pre class="code sh_sh">./vep -i variants.vcf --plugin pLI,values_file.txt
./vep -i variants.vcf --plugin pLI,values_file.txt,transcript # to check for the transcript score.
</pre><p>  gnomAD v4 release expanded the scale of pLI score calculation. The file can be downloaded from -
    <a href="https://gnomad.broadinstitute.org/downloads#v4-constraint">https://gnomad.broadinstitute.org/downloads#v4-constraint</a> (Constraint metrics TSV)
  To use the data you can follow the same procedure as above but needs to change the column number to accordingly.</p><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv pLI.pm ~/.vep/Plugins
mv pLI_values.txt ~/.vep/Plugins
./vep -i variants.vcf --plugin pLI
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#00DD66">Gene tolerance to change</div></td><td><ul style="padding-left:1em"><li><a href="https://metacpan.org/pod/List::MoreUtils" rel="external">List::MoreUtils</a> qw/zip/</li><li><a href="https://metacpan.org/pod/DBI" rel="external">DBI</a></li></ul></td><td>Ensembl</td></tr>
<tr id="PolyPhen_SIFT" class="bg1 plugin_row" data-category="pathogenicity_predictions"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/PolyPhen_SIFT.pm">PolyPhen_SIFT</a></div></td><td><p> An Ensembl VEP plugin that retrieves PolyPhen and SIFT predictions from a
 locally constructed SQLite database. It can be used when your main
 source of the Ensembl VEP transcript annotation (e.g. a GFF file or GFF-based cache)
 does not contain these predictions. <a class="button" href="#polyphen_sift" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('polyphen_sift');" id="a_polyphen_sift">more</a></p><div id="div_polyphen_sift" style="display:none;word-wrap:break-word;"><p> You must create a SQLite database of the predictions or point to the SQLite
 database file already created. Compatible SQLite databases based on pangenome
 data are available at <a href="http://ftp.ensembl.org/pub/current_variation/pangenomes">http://ftp.ensembl.org/pub/current_variation/pangenomes</a></p><p> You may point to the file by adding parameter <kbd>db=[file]</kbd>. If the file is not
 in <kbd>HOME/.vep</kbd>, you can also use parameter <kbd>dir=[dir]</kbd> to indicate its path.</p><p><pre class="code sh_sh">--plugin PolyPhen_SIFT,db=[file]
--plugin PolyPhen_SIFT,db=[file],dir=[dir]
</pre><p> To create a SQLite database using PolyPhen/SIFT data from the Ensembl database,
 you must have an active database connection (i.e. not using <kbd>--offline</kbd>) and
 add parameter <kbd>create_db=1</kbd>. This will create a SQLite file named
 <kbd>[species].PolyPhen_SIFT.db</kbd>, placed in the directory specified by the <kbd>dir</kbd>
 parameter:</p><p><pre class="code sh_sh">--plugin PolyPhen_SIFT,create_db=1
--plugin PolyPhen_SIFT,create_db=1,dir=/some/specific/directory
</pre><p> *** NB: this will take some hours! ***</p><p> When creating a PolyPhen_SIFT by simply using <kbd>create_db=1</kbd>, you do not need to
 specify any parameters to load the appropriate file based on the species:</p><p><pre class="code sh_sh">--plugin PolyPhen_SIFT
</pre><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv PolyPhen_SIFT.pm ~/.vep/Plugins</p><p># Read default PolyPhen/SIFT SQLite file in $HOME/.vep
./vep -i variations.vcf -cache --plugin PolyPhen_SIFT</p><p># Read database with custom name and/or located in a custom directory
./vep -i variations.vcf -cache --plugin PolyPhen_SIFT,db=custom.db
./vep -i variations.vcf -cache --plugin PolyPhen_SIFT,dir=/some/custom/dir
./vep -i variations.vcf -cache --plugin PolyPhen_SIFT,db=custom.db,dir=/some/custom/dir</p><p># Create PolyPhen/SIFT SQLite file based on Ensembl database
./vep -i variations.vcf -cache --plugin PolyPhen_SIFT,create_db=1
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#1E90FF">Pathogenicity predictions</div></td><td><ul style="padding-left:1em"><li><a href="https://metacpan.org/pod/Digest::MD5" rel="external">Digest::MD5</a> qw(md5_hex)</li><li><a href="https://metacpan.org/pod/DBI" rel="external">DBI</a></li></ul></td><td>Ensembl</td></tr>
<tr id="PON_P2" class="bg2 plugin_row" data-category="pathogenicity_predictions"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/PON_P2.pm">PON_P2</a></div></td><td><p> This plugin for Ensembl Variant Effect Predictor (VEP) computes the predictions of PON-P2
 for amino acid substitutions in human proteins. <a class="button" href="#pon_p2" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('pon_p2');" id="a_pon_p2">more</a></p><div id="div_pon_p2" style="display:none;word-wrap:break-word;"><p> PON-P2 is developed and maintained by Protein Structure and Bioinformatics Group
 at Lund University and is available at <a href="http://structure.bmc.lu.se/PON-P2/">http://structure.bmc.lu.se/PON-P2/</a>.</p><p> If you use this data, please cite the following publication
 Niroula, A., Vihinen, M. Harmful somatic amino acid substitutions affect key pathways in cancers.
 BMC Med Genomics 8, 53 (2015). <a href="https://doi.org/10.1186/s12920-015-0125-x">https://doi.org/10.1186/s12920-015-0125-x</a></p><p> There are two ways to run the plugin:</p><p><ol><li value="1">To compute the predictions from the PON-P2 API, use python script <kbd>ponp2.py</kbd> (*)    and select the reference genome (acceptable values are: hg37 and hg38):
<pre class="code sh_sh">--plugin PON_P2,pyscript=/path/to/python/script/ponp2.py,hg=hg37
</pre><p>    (*) To run this mode, you will require a python script and its dependencies (Python,
    python suds). The python file can be downloaded from <a href="http://structure.bmc.lu.se/PON-P2/vep.html/">http://structure.bmc.lu.se/PON-P2/vep.html/</a>
    and the complete path to this file must be supplied while using this plugin.</p><p></li><li value="2">To fetch the predictions from a file containing pre-calculated predictions for somatic variations  please use the following key=value option (only available for GRCh37):<table class="ss"><thead><tr><th>Argument</th><th>Description</th></tr></thead><tbody>
</li></ol><p><tr class="bg1"><td><pre>file</pre></td><td>COSMIC text file with pre-calculated predictions downloaded from                   <a href="http://structure.bmc.lu.se/PON-P2/cancer30.html/">http://structure.bmc.lu.se/PON-P2/cancer30.html/</a></p><p></td></tr></tbody></table><p>   The following steps are necessary before using the file:
<pre class="code sh_sh">(head -n 1 COSMIC.txt && tail -n +2 COSMIC.txt | sort -t $'\t' -k1,1 -k2,2n) &gt cosmic_sorted.txt
sed -i 's/Chromosome/#Chromosome/' cosmic_sorted.txt
bgzip cosmic_sorted.txt
tabix -s 1 -b 2 -e 2 cosmic_sorted.txt.gz
</pre><p><pre class="code sh_sh">--plugin PON_P2,file=path/to/cosmic_sorted.txt.gz 
</pre><p></ol><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv PON_P2.pm ~/.vep/Plugins
./vep -i variations.vcf --plugin PON_P2,pyscript=/path/to/python/script/ponp2.py,hg=hg37
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#1E90FF">Pathogenicity predictions</div></td><td>-</td><td><ul style="padding-left:1em"><li>Abhishek Niroula</li><li>Mauno Vihinen</li></ul></td></tr>
<tr id="PostGAP" class="bg1 plugin_row" data-category="phenotype_data_and_citations"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/PostGAP.pm">PostGAP</a></div></td><td><p>A VEP plugin that retrieves data for variants from a tabix-indexed PostGAP file (1-based file). <a class="button" href="#postgap" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('postgap');" id="a_postgap">more</a></p><div id="div_postgap" style="display:none;word-wrap:break-word;"><p>Please refer to the PostGAP github and wiki for more information:
<a href="https://github.com/Ensembl/postgap">https://github.com/Ensembl/postgap</a>
<a href="https://github.com/Ensembl/postgap/wiki">https://github.com/Ensembl/postgap/wiki</a>
<a href="https://github.com/Ensembl/postgap/wiki/algorithm-pseudo-code">https://github.com/Ensembl/postgap/wiki/algorithm-pseudo-code</a></p><p>The Bio::DB::HTS perl library or tabix utility must be installed in your path
to use this plugin. The PostGAP data file can be downloaded from
<a href="https://storage.googleapis.com/postgap-data">https://storage.googleapis.com/postgap-data</a>.</p><p>The file must be processed and indexed by tabix before use by this plugin.
PostGAP has coordinates for both GRCh38 and GRCh37; the file must be
processed differently according to the assembly you use.</p><p><pre class="code sh_sh">wget <a href="https://storage.googleapis.com/postgap-data/postgap.txt.gz">https://storage.googleapis.com/postgap-data/postgap.txt.gz</a>
gunzip postgap.txt.gz
</pre><p># GRCh38
<pre class="code sh_sh">(grep ^"ld_snp_rsID" postgap.txt; grep -v ^"ld_snp_rsID" postgap.txt | sort -k4,4 -k5,5n ) | bgzip &gt postgap_GRCh38.txt.gz
tabix -s 4 -b 5 -e 5 -c l postgap_GRCh38.txt.gz
</pre><p># GRCh37
<pre class="code sh_sh">(grep ^"ld_snp_rsID" postgap.txt; grep -v ^"ld_snp_rsID" postgap.txt | sort -k2,2 -k3,3n ) | bgzip &gt postgap_GRCh37.txt.gz
tabix -s 2 -b 3 -e 3 -c l postgap_GRCh37.txt.gz
</pre><p>Note that in the last command we tell tabix that the header line starts with "l";
this may change to the default of "#" in future versions of PostGAP.</p><p>When running the plugin by default <kbd>disease_efo_id</kbd>, <kbd>disease_name</kbd>, <kbd>gene_id</kbd>
and <kbd>score</kbd> information is returned e.g.</p><p><pre class="code sh_sh">--plugin POSTGAP,/path/to/PostGap.gz
</pre><p>You may include all columns with ALL; this fetches a large amount of data per
variant!:</p><p><pre class="code sh_sh">--plugin POSTGAP,/path/to/PostGap.gz,ALL
</pre><p>You may want to select only a specific subset of additional information to be
reported, you can do this by specifying the columns as parameters to the plugin e.g.</p><p><pre class="code sh_sh">--plugin POSTGAP,/path/to/PostGap.gz,gwas_pmid,gwas_size
</pre><p>If a requested column is not found, the error message will report the
complete list of available columns in the POSTGAP file. For a brief description
of the available information please refer to the 'How do I use POSTGAP output?'
section in the POSTGAP wiki.</p><p>Tabix also allows the data file to be hosted on a remote server. This plugin is
fully compatible with such a setup - simply use the URL of the remote file:</p><p><pre class="code sh_sh">--plugin PostGAP,<a href="http://my.files.com/postgap.txt.gz">http://my.files.com/postgap.txt.gz</a>
</pre><p>Note that gene sequences referred to in PostGAP may be out of sync with
those in the latest release of Ensembl; this may lead to discrepancies with
scores retrieved from other sources.</p><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv PostGAP.pm ~/.vep/Plugins
./vep -i variations.vcf --plugin PostGAP,/path/to/PostGap.gz,col1,col2
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#556688">Phenotype data and citations</div></td><td>-</td><td>Ensembl</td></tr>
<tr id="PrimateAI" class="bg2 plugin_row" data-category="pathogenicity_predictions"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/PrimateAI.pm">PrimateAI</a></div></td><td><p>  The PrimateAI VEP plugin is designed to retrieve clinical impact scores of
  variants, as described in <a href="https://www.nature.com/articles/s41588-018-0167-z">https://www.nature.com/articles/s41588-018-0167-z</a>.
  Please consider citing the paper if using this plugin. <a class="button" href="#primateai" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('primateai');" id="a_primateai">more</a></p><div id="div_primateai" style="display:none;word-wrap:break-word;"><p>  In brief, common missense mutations in non-human primate species are usually
  benign in humans. Thousands of common variants from six non-human primate
  species were used to train a deep neural network to identify pathogenic 
  mutations in humans with a rare disease.</p><p>  This plugin uses files generated by the PrimateAI software, which is
  available from <a href="https://github.com/Illumina/PrimateAI">https://github.com/Illumina/PrimateAI</a>. The files containing
  predicted pathogenicity scores can be downloaded from
  <a href="https://basespace.illumina.com/s/yYGFdGih1rXL">https://basespace.illumina.com/s/yYGFdGih1rXL</a> (a free BaseSpace account may
  be required):
      PrimateAI_scores_v0.2.tsv.gz (for GRCh37/hg19)
      PrimateAI_scores_v0.2_hg38.tsv.gz (for GRCh38/hg38)</p><p>  Before running the plugin for the first time, the following steps must be 
  taken to format the downloaded files:</p><p><ol><li value="1"> Unzip the score files</li><li value="2"> Add '#' in front of the column description line</li><li value="3"> Remove any empty lines.</li><li value="4"> Sort the file by chromosome and position</li><li value="5"> Compress the file in .bgz format</li><li value="6"> Create tabix index (requires tabix to be installed).
</li></ol><p>  Command line examples for formatting input files:
<pre class="code sh_sh">gunzip -cf PrimateAI_scores_v0.2.tsv.gz | sed '12s/.*/#&/' | sed '/^$/d' | awk 'NR&lt12{print $0;next}{print $0 | "sort -k1,1 -k 2,2n -V"}' | bgzip &gt PrimateAI_scores_v0.2_GRCh37_sorted.tsv.bgz
tabix -s 1 -b 2 -e 2 PrimateAI_scores_v0.2_GRCh37_sorted.tsv.bgz
</pre><p><pre class="code sh_sh">gunzip -cf PrimateAI_scores_v0.2_hg38.tsv.gz | sed '12s/.*/#&/' | sed '/^$/d' | awk 'NR&lt12{print $0;next}{print $0 | "sort -k1,1 -k 2,2n -V"}' | bgzip &gt PrimateAI_scores_v0.2_GRCh38_sorted.tsv.bgz
tabix -s 1 -b 2 -e 2 PrimateAI_scores_v0.2_GRCh38_sorted.tsv.bgz
</pre><p></ol><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv PrimateAI.pm ~/.vep/Plugins</p><p>./vep -i variations.vcf --plugin PrimateAI,PrimateAI_scores_v0.2_GRCh37_sorted.tsv.bgz
./vep -i variations.vcf --plugin PrimateAI,PrimateAI_scores_v0.2_GRCh38_sorted.tsv.bgz
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#1E90FF">Pathogenicity predictions</div></td><td>-</td><td>Ensembl</td></tr>
<tr id="ProteinSeqs" class="bg1 plugin_row" data-category="sequence"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/ProteinSeqs.pm">ProteinSeqs</a></div></td><td><p> This is a plugin for the Ensembl Variant Effect Predictor (VEP) that
 prints out the reference and mutated protein sequences of any
 proteins found with non-synonymous mutations in the input file.  <a class="button" href="#proteinseqs" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('proteinseqs');" id="a_proteinseqs">more</a></p><div id="div_proteinseqs" style="display:none;word-wrap:break-word;"><p> You should supply the name of file where you want to store the
 reference protein sequences as the first argument, and a file to
 store the mutated sequences as the second argument.</p><p> Note that, for simplicity, where stop codons are gained the plugin 
 simply substitutes a '*' into the sequence and does not truncate the 
 protein. Where a stop codon is lost any new amino acids encoded by the 
 mutation are appended to the sequence, but the plugin does not attempt 
 to translate until the next downstream stop codon. Also, the protein
 sequence resulting from each mutation is printed separately, no attempt
 is made to apply multiple mutations to the same protein.</p><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv ProteinSeqs.pm ~/.vep/Plugins
./vep -i variations.vcf --plugin ProteinSeqs,reference.fa,mutated.fa
./vep -i variations.vcf --plugin ProteinSeqs,reference=reference.fa,mutated=mutated.fa
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#5F81A9">Sequence</div></td><td>-</td><td>Ensembl</td></tr>
<tr id="ReferenceQuality" class="bg2 plugin_row" data-category="sequence"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/ReferenceQuality.pm">ReferenceQuality</a></div></td><td><p> This is a plugin for the Ensembl Variant Effect Predictor (VEP) that
 reports on the quality of the reference genome using GRC data at the location of your variants.
 More information can be found at: <a href="https://www.ncbi.nlm.nih.gov/grc/human/issues">https://www.ncbi.nlm.nih.gov/grc/human/issues</a> <a class="button" href="#referencequality" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('referencequality');" id="a_referencequality">more</a></p><div id="div_referencequality" style="display:none;word-wrap:break-word;"><p> The following steps are necessary before running this plugin:</p><p> GRCh38:</p><p><pre class="code sh_sh">wget <a href="https://ftp.ncbi.nlm.nih.gov/pub/grc/human/GRC/GRCh38/MISC/annotated_clone_assembly_problems_GCF_000001405.38.gff3">https://ftp.ncbi.nlm.nih.gov/pub/grc/human/GRC/GRCh38/MISC/annotated_clone_assembly_problems_GCF_000001405.38.gff3</a>
wget <a href="https://ftp.ncbi.nlm.nih.gov/pub/grc/human/GRC/Issue_Mapping/GRCh38.p12_issues.gff3">https://ftp.ncbi.nlm.nih.gov/pub/grc/human/GRC/Issue_Mapping/GRCh38.p12_issues.gff3</a>
cat annotated_clone_assembly_problems_GCF_000001405.38.gff3 GRCh38.p12_issues.gff3 &gt GRCh38_quality_mergedfile.gff3
sort -k 1,1 -k 4,4n -k 5,5n GRCh38_quality_mergedfile.gff3 &gt sorted_GRCh38_quality_mergedfile.gff3
bgzip sorted_GRCh38_quality_mergedfile.gff3
tabix -p gff sorted_GRCh38_quality_mergedfile.gff3.gz
</pre><p> The plugin can then be run with:
<pre class="code sh_sh">./vep -i variations.vcf --plugin ReferenceQuality,sorted_GRCh38_quality_mergedfile.gff3.gz
</pre><p> GRCh37:</p><p><pre class="code sh_sh">wget <a href="https://ftp.ncbi.nlm.nih.gov/pub/grc/human/GRC/GRCh37/MISC/annotated_clone_assembly_problems_GCF_000001405.25.gff3">https://ftp.ncbi.nlm.nih.gov/pub/grc/human/GRC/GRCh37/MISC/annotated_clone_assembly_problems_GCF_000001405.25.gff3</a>
wget <a href="https://ftp.ncbi.nlm.nih.gov/pub/grc/human/GRC/Issue_Mapping/GRCh37.p13_issues.gff3">https://ftp.ncbi.nlm.nih.gov/pub/grc/human/GRC/Issue_Mapping/GRCh37.p13_issues.gff3</a>
cat annotated_clone_assembly_problems_GCF_000001405.25.gff3 GRCh37.p13_issues.gff3 &gt GRCh37_quality_mergedfile.gff3
sort -k 1,1 -k 4,4n -k 5,5n GRCh37_quality_mergedfile.gff3 &gt sorted_GRCh37_quality_mergedfile.gff3
bgzip sorted_GRCh37_quality_mergedfile.gff3
tabix -p gff sorted_GRCh37_quality_mergedfile.gff3.gz
</pre><p> The plugin can then be run with:
<pre class="code sh_sh">./vep -i variations.vcf --plugin ReferenceQuality,sorted_GRCh37_quality_mergedfile.gff3.gz
</pre><p> </p><p> The tabix utility must be installed in your path to use this plugin.</p><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv ReferenceQuality.pm ~/.vep/Plugins
./vep -i variations.vcf --plugin ReferenceQuality,/path/to/data.gff3.gz
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#5F81A9">Sequence</div></td><td>-</td><td>Ensembl</td></tr>
<tr id="REVEL" class="bg1 plugin_row" data-category="pathogenicity_predictions"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/REVEL.pm">REVEL</a></div></td><td><p> This is a plugin for the Ensembl Variant Effect Predictor (VEP) that
 adds the REVEL score for missense variants to the Ensembl VEP output. <a class="button" href="#revel" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('revel');" id="a_revel">more</a></p><div id="div_revel" style="display:none;word-wrap:break-word;"><p> Please cite the REVEL publication alongside Ensembl VEP if you use this resource:
 <a href="https://www.ncbi.nlm.nih.gov/pubmed/27666373">https://www.ncbi.nlm.nih.gov/pubmed/27666373</a></p><p> Running options:
 If available, the plugin will match the scores by transcript id (default).
 Using the flag <kbd>1</kbd> the plugin will not try to match by transcript id.</p><p> REVEL scores can be downloaded from: <a href="https://sites.google.com/site/revelgenomics/downloads">https://sites.google.com/site/revelgenomics/downloads</a></p><p> The plugin supports several REVEL file versions:
<ul><li>REVEL file version Dec 2017, which has 7 columns and only GRCh37 coordinates</li><li>REVEL file version Feb 2020, which has 8 columns with GRCh37 and GRCh38 coordinates</li><li>REVEL file version May 2021, which has 9 columns with GRCh37 and GRCh38 coordinates and a new column with transcript ids 
</li></ul><p> These files can be tabix-processed by:
<pre class="code sh_sh">unzip revel-v1.3_all_chromosomes.zip
cat revel_with_transcript_ids | tr "," "\t" &gt tabbed_revel.tsv
sed '1s/.*/#&/' tabbed_revel.tsv &gt new_tabbed_revel.tsv
bgzip new_tabbed_revel.tsv
</pre><p> for GRCh37:
<pre class="code sh_sh">tabix -f -s 1 -b 2 -e 2 new_tabbed_revel.tsv.gz
</pre><p> for GRCh38:
<pre class="code sh_sh">zcat new_tabbed_revel.tsv.gz | head -n1 &gt h
zgrep -h -v ^#chr new_tabbed_revel.tsv.gz | awk '$3 != "." ' | sort -k1,1 -k3,3n - | cat h - | bgzip -c &gt new_tabbed_revel_grch38.tsv.gz
tabix -f -s 1 -b 3 -e 3 new_tabbed_revel_grch38.tsv.gz
</pre><p> The plugin can then be run as default:
<pre class="code sh_sh">./vep -i variations.vcf --assembly GRCh38 --plugin REVEL,file=/path/to/revel/data.tsv.gz
</pre><p> or with the option to not match by transcript id:
<pre class="code sh_sh">./vep -i variations.vcf --assembly GRCh38 --plugin REVEL,file=/path/to/revel/data.tsv.gz,no_match=1
</pre><p> Requirements:
  The tabix utility must be installed in your path to use this plugin.
  The --assembly flag is required to use this plugin.</p><p></ul><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv REVEL.pm ~/.vep/Plugins
./vep -i variations.vcf --assembly GRCh37 --plugin REVEL,file=/path/to/revel/data.tsv.gz
./vep -i variations.vcf --assembly GRCh38 --plugin REVEL,file=/path/to/revel/data.tsv.gz
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#1E90FF">Pathogenicity predictions</div></td><td>-</td><td>Ensembl</td></tr>
<tr id="RiboseqORFs" class="bg2 plugin_row" data-category="transcript_annotation"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/RiboseqORFs.pm">RiboseqORFs</a></div></td><td><p> This is an Ensembl VEP plugin that uses a standardized catalog of human Ribo-seq ORFs to
 re-calculate consequences for variants located in these translated regions. <a class="button" href="#riboseqorfs" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('riboseqorfs');" id="a_riboseqorfs">more</a></p><div id="div_riboseqorfs" style="display:none;word-wrap:break-word;"><p> This plugin reports new consequences based on the evidence from the Ribo-seq
 ORF annotation and supporting publications. The human Ribo-seq ORF data can be
 downloaded from: <a href="https://ftp.ebi.ac.uk/pub/databases/gencode/riboseq_orfs/data">https://ftp.ebi.ac.uk/pub/databases/gencode/riboseq_orfs/data</a></p><p> After downloading the annotation, please bgzip and tabix it:
<pre class="code sh_sh">bgzip Ribo-seq_ORFs.bed
tabix Ribo-seq_ORFs.bed.gz
</pre><p> For optimal performance when running this plugin in Ensembl VEP, please use a FASTA
 file (<kbd>--fasta</kbd>). A FASTA file is always required in offline mode.</p><p> Please cite the publication for the Ribo-seq ORF annotation alongside Ensembl VEP
 if you use this resource: <a href="https://doi.org/10.1038/s41587-022-01369-0">https://doi.org/10.1038/s41587-022-01369-0</a></p><p> The tabix utility must be installed in your path to use this plugin.</p><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
./vep -i variations.vcf --plugin RiboseqORFs,file=/path/to/Ribo-seq_ORFs.bed.gz
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#FF2288">Transcript annotation</div></td><td>-</td><td>Ensembl</td></tr>
<tr id="SameCodon" class="bg1 plugin_row" data-category="variant_data"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/SameCodon.pm">SameCodon</a></div></td><td><p> An Ensembl VEP plugin that reports existing variants that fall in the same codon.
 This plugin requires a database connection, can not be run in offline mode <a class="button" href="#samecodon" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('samecodon');" id="a_samecodon">more</a></p><div id="div_samecodon" style="display:none;word-wrap:break-word;"><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv SameCodon.pm ~/.vep/Plugins
./vep -i variations.vcf --plugin SameCodon
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#B22222">Variant data</div></td><td>-</td><td>Ensembl</td></tr>
<tr id="satMutMPRA" class="bg2 plugin_row" data-category="phenotype_data_and_citations"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/satMutMPRA.pm">satMutMPRA</a></div></td><td><p>An Ensembl VEP plugin that retrieves data for variants from a tabix-indexed satMutMPRA file (1-based file).
The saturation mutagenesis-based massively parallel reporter assays (satMutMPRA) measures variant
effects on gene RNA expression for 21 regulatory elements (11 enhancers, 10 promoters). <a class="button" href="#satmutmpra" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('satmutmpra');" id="a_satmutmpra">more</a></p><div id="div_satmutmpra" style="display:none;word-wrap:break-word;"><p>The 20 disease-associated regulatory elements and one ultraconserved enhancer analysed in different cell lines are the following:
<ul><li>ten promoters (of TERT, LDLR, HBB, HBG, HNF4A, MSMB, PKLR, F9, FOXE1 and GP1BB) and</li><li>ten enhancers (of SORT1, ZRS, BCL11A, IRF4, IRF6, MYC (2x), RET, TCF7L2 and ZFAND3) and</li><li>one ultraconserved enhancer (UC88).
</li></ul><p>Please refer to the satMutMPRA web server and Kircher M et al. (2019) paper for more information:
<a href="https://mpra.gs.washington.edu/satMutMPRA/">https://mpra.gs.washington.edu/satMutMPRA/</a>
<a href="https://www.ncbi.nlm.nih.gov/pubmed/31395865">https://www.ncbi.nlm.nih.gov/pubmed/31395865</a></p><p>Parameters can be set using a key=value system:<table class="ss"><thead><tr><th>Argument</th><th>Description</th></tr></thead><tbody><tr class="bg1"><td><pre>file</pre></td><td>required - a tabix indexed file of the satMutMPRA data corresponding to desired assembly. 
<tr class="bg2"><td><pre>pvalue</pre></td><td>p-value threshold  (default: 0.00001) 
<tr class="bg1"><td><pre>cols</pre></td><td>colon delimited list of data types to be returned from the satMutMPRA data                 (default: <kbd>Value</kbd>, <kbd>P-Value</kbd>, and <kbd>Element</kbd>)</p><p><tr class="bg2"><td><pre>incl_repl</pre></td><td>include replicates (default: off): <ul><li>full replicate for LDLR promoter (LDLR.2) and SORT1 enhancer (SORT1.2)</li><li>a reversed sequence orientation for SORT1 (SORT1-flip)</li><li>other conditions: PKLR-48h, ZRSh-13h2, TERT-GAa, TERT-GBM, TERG-GSc
</td></tr></tbody></table><p></li></ul><p>The Bio::DB::HTS perl library or tabix utility must be installed in your path
to use this plugin. The satMutMPRA data file can be downloaded from
<a href="https://mpra.gs.washington.edu/satMutMPRA/">https://mpra.gs.washington.edu/satMutMPRA/</a></p><p>satMutMPRA data can be downloaded for both GRCh38 and GRCh37 from the web server (<a href="https://mpra.gs.washington.edu/satMutMPRA/">https://mpra.gs.washington.edu/satMutMPRA/</a>):
<kbd>Download</kbd> section, select <kbd>GRCh37</kbd> or <kbd>GRCh38</kbd> for 'Genome release' and 'Download All Elements'.</p><p>The file must be processed and indexed by tabix before use by this plugin.</p><p># GRCh38
<pre class="code sh_sh">(grep ^Chr GRCh38_ALL.tsv; grep -v ^Chr GRCh38_ALL.tsv | sort -k1,1 -k2,2n ) | bgzip &gt satMutMPRA_GRCh38_ALL.gz
tabix -s 1 -b 2 -e 2 -c C satMutMPRA_GRCh38_ALL.gz
</pre><p># GRCh37
<pre class="code sh_sh">(grep ^Chr GRCh37_ALL.tsv; grep -v ^Chr GRCh37_ALL.tsv | sort -k1,1 -k2,2n ) | bgzip &gt satMutMPRA_GRCh37_ALL.gz
tabix -s 1 -b 2 -e 2 -c C satMutMPRA_GRCh37_ALL.gz
</pre><p>
When running the plugin by default <kbd>Value</kbd>, <kbd>P-Value</kbd>, and <kbd>Element</kbd>
information is returned e.g.</p><p><pre class="code sh_sh">--plugin satMutMPRA,file=/path/to/satMutMPRA_GRCh38_ALL.gz
</pre><p>You may include all columns with ALL; this fetches all data per variant
(e.g. Tags, DNA, RNA, Value, P-Value, Element):</p><p><pre class="code sh_sh">--plugin satMutMPRA,file=/path/to/satMutMPRA_GRCh38_ALL.gz,cols=ALL
</pre><p>You may want to select only a specific subset of information to be
reported, you can do this by specifying the specific columns as parameters to the plugin e.g.</p><p><pre class="code sh_sh">--plugin satMutMPRA,file=/path/to/satMutMPRA_GRCh38_ALL.gz,cols=Tags:DNA
</pre><p>If a requested column is not found, the error message will report the
complete list of available columns in the satMutMPRA file. For a detailed description
of the available information please refer to the manuscript or online web server.</p><p>Tabix also allows the data file to be hosted on a remote server. This plugin is
fully compatible with such a setup - simply use the URL of the remote file:</p><p><pre class="code sh_sh">--plugin satMutMPRA,file=<a href="http://my.files.com/satMutMPRA.gz">http://my.files.com/satMutMPRA.gz</a>
</pre><p>Note that gene locations referred to in satMutMPRA may be out of sync with
those in the latest release of Ensembl; this may lead to discrepancies with
information retrieved from other sources.</p><p></ul><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv satMutMPRA.pm ~/.vep/Plugins
./vep -i variations.vcf --plugin satMutMPRA,file=/path/to/satMutMPRA_data.gz,cols=col1:col2
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#556688">Phenotype data and citations</div></td><td>-</td><td>Ensembl</td></tr>
<tr id="SingleLetterAA" class="bg1 plugin_row" data-category="hgvs"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/SingleLetterAA.pm">SingleLetterAA</a></div></td><td><p> This is a plugin for the Ensembl Variant Effect Predictor (VEP) that
 returns a HGVSp string with single amino acid letter codes <a class="button" href="#singleletteraa" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('singleletteraa');" id="a_singleletteraa">more</a></p><div id="div_singleletteraa" style="display:none;word-wrap:break-word;"><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv SingleLetterAA.pm ~/.vep/Plugins
./vep -i variations.vcf --plugin SingleLetterAA
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#9400D3">HGVS</div></td><td>-</td><td>Ensembl</td></tr>
<tr id="SpliceAI" class="bg2 plugin_row" data-category="splicing_predictions"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/SpliceAI.pm">SpliceAI</a></div></td><td><p> An Ensembl VEP plugin that retrieves pre-calculated annotations from SpliceAI.
 SpliceAI is a deep neural network, developed by Illumina, Inc 
 that predicts splice junctions from an arbitrary pre-mRNA transcript sequence. <a class="button" href="#spliceai" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('spliceai');" id="a_spliceai">more</a></p><div id="div_spliceai" style="display:none;word-wrap:break-word;"><p> Delta score of a variant, defined as the maximum of (DS_AG, DS_AL, DS_DG, DS_DL), 
 ranges from 0 to 1 and can be interpreted as the probability of the variant being 
 splice-altering. The author-suggested cutoffs are:
<ul><li>0.2 (high recall)</li><li>0.5 (recommended)</li><li>0.8 (high precision)
</li></ul><p> This plugin is available for both GRCh37 and GRCh38.</p><p> More information can be found at:
 <a href="https://pypi.org/project/spliceai/">https://pypi.org/project/spliceai/</a></p><p> Please cite the SpliceAI publication alongside Ensembl VEP if you use this resource:
 <a href="https://www.ncbi.nlm.nih.gov/pubmed/30661751">https://www.ncbi.nlm.nih.gov/pubmed/30661751</a></p><p> Running options:
<ol><li value="1">By default, this plugin appends all scores from SpliceAI files.</li><li value="2">Besides the pre-calculated scores, it can also be specified a score cutoff between 0 and 1.</p><p></li></ol><p> Output: 
  The output includes the gene symbol, delta scores (DS) and delta positions (DP)
  for acceptor gain (AG), acceptor loss (AL), donor gain (DG), and donor loss (DL).</p><p><ul><li>For tab the output contains one header <kbd>SpliceAI_pred</kbd> with all&nbsp;    the delta scores and positions. The format is:
&nbsp;      <kbd>SYMBOL|DS_AG|DS_AL|DS_DG|DS_DL|DP_AG|DP_AL|DP_DG|DP_DL</kbd></p><p></li><li>For JSON the output is a hash with the following format:&nbsp;    "spliceai":
&nbsp;      {"DP_DL":0,"DS_AL":0,"DP_AG":0,"DS_DL":0,"SYMBOL":"X","DS_AG":0,"DP_AL":0,"DP_DG":0,"DS_DG":0}</p><p></li><li>For VCF output the delta scores and positions are stored in different headers.&nbsp;    The values are <kbd>SpliceAI_pred_xx</kbd> being <kbd>xx</kbd> the score/position.
&nbsp;      Example: <kbd>SpliceAI_pred_DS_AG</kbd> is the delta score for acceptor gain.</p><p></li></ul><p>  Gene matching:
  SpliceAI can contain scores for multiple genes that overlap a variant,
  and Ensembl VEP can also predict consequences on multiple genes for a given variant.
  The plugin only returns SpliceAI scores for the gene symbols that match (if any).</p><p> If plugin is run with option 2, the output also contains a flag: <kbd>PASS</kbd> if delta score
 passes the cutoff, <kbd>FAIL</kbd> otherwise. </p><p> The following steps are necessary before running this plugin:</p><p> The files with the annotations for all possible substitutions (snv), 1 base insertions 
 and 1-4 base deletions (indel) within genes are available here:
 <a href="https://basespace.illumina.com/s/otSPW8hnhaZR">https://basespace.illumina.com/s/otSPW8hnhaZR</a></p><p> GRCh37:
<pre class="code sh_sh">tabix -p vcf spliceai_scores.raw.snv.hg37.vcf.gz
tabix -p vcf spliceai_scores.raw.indel.hg37.vcf.gz
</pre><p> GRCh38:
<pre class="code sh_sh">tabix -p vcf spliceai_scores.raw.snv.hg38.vcf.gz
tabix -p vcf spliceai_scores.raw.indel.hg38.vcf.gz
</pre><p> The plugin can then be run:
<pre class="code sh_sh">./vep -i variations.vcf --plugin SpliceAI,snv=/path/to/spliceai_scores.raw.snv.hg38.vcf.gz,indel=/path/to/spliceai_scores.raw.indel.hg38.vcf.gz
./vep -i variations.vcf --plugin SpliceAI,snv=/path/to/spliceai_scores.raw.snv.hg38.vcf.gz,indel=/path/to/spliceai_scores.raw.indel.hg38.vcf.gz,cutoff=0.5
</pre><p></ul><p></ol><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv SpliceAI.pm ~/.vep/Plugins
./vep -i variations.vcf --plugin SpliceAI,snv=/path/to/spliceai_snv_.vcf.gz,
indel=/path/to/spliceai_indel_.vcf.gz
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#FF0000">Splicing predictions</div></td><td><a href="https://metacpan.org/pod/List::Util" rel="external">List::Util</a> qw(max)</td><td>Ensembl</td></tr>
<tr id="SpliceRegion" class="bg1 plugin_row" data-category="splicing_predictions"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/SpliceRegion.pm">SpliceRegion</a></div></td><td><p> This is a plugin for the Ensembl Variant Effect Predictor (VEP) that
 provides more granular predictions of splicing effects. <a class="button" href="#spliceregion" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('spliceregion');" id="a_spliceregion">more</a></p><div id="div_spliceregion" style="display:none;word-wrap:break-word;"><p> Three additional terms may be added:</p><p> # splice_donor_5th_base_variant : variant falls in the 5th base after the splice donor junction (5' end of intron)</p><p><pre class="code sh_sh">             v
 ...EEEEEIIIIIIIIII...
</pre>
 (E = exon, I = intron, v = variant location)</p><p> # splice_donor_region_variant : variant falls in region between 3rd and 6th base after splice junction (5' end of intron)</p><p><pre class="code sh_sh">           vv vvv
 ...EEEEEIIIIIIIIII...
</pre>
 # splice_polypyrimidine_tract_variant : variant falls in polypyrimidine tract at 3' end of intron, between 17 and 3 bases from the end</p><p><pre class="code sh_sh">      vvvvvvvvvvvvvvv
 ...IIIIIIIIIIIIIIIIIIIIEEEEE...
</pre></p><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv SpliceRegion.pm ~/.vep/Plugins
./vep -i variations.vcf --plugin SpliceRegion</p><p>To only show the additional consequence extended_intronic_splice_region_variant, use: 
./vep -i variations.vcf --plugin SpliceRegion,Extended
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#FF0000">Splicing predictions</div></td><td>-</td><td>Ensembl</td></tr>
<tr id="SpliceVault" class="bg2 plugin_row" data-category="splicing_predictions"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/SpliceVault.pm">SpliceVault</a></div></td><td><p> An Ensembl VEP plugin that retrieves SpliceVault data to predict exon-skipping events
 and activated cryptic splice sites based on the most common mis-splicing events
 around a splice site. <a class="button" href="#splicevault" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('splicevault');" id="a_splicevault">more</a></p><div id="div_splicevault" style="display:none;word-wrap:break-word;"><p> This plugin returns the most common variant-associated mis-splicing events
 based on SpliceVault data. Each event includes the following information:
<ul><li>Type: exon skipping (ES), cryptic donor (CD) or cryptic acceptor (CA)</li><li>Transcript impact:</li><li>For ES, describes skipped exons, e.g. ES:2 represents exon 2 skipping and&nbsp;     ES:2-3 represents skipping of exon 2 and 3
</li><li>For CD/CA, describes the distance from the annotated splice-site to the&nbsp;     cryptic splice-site with reference to the transcript (distances to negative
&nbsp;     strand transcripts are reported according to the 5' to 3' distance)
</li><li>Percent of supporting samples: percent of samples supporting the event over&nbsp;   total samples where splicing occurs in that site (note this may be above 100%
&nbsp;   if the event is seen in more samples than annotated splicing)
</li><li>Frameshift: inframe or out-of-frame event
</li></ul><p> The plugin also returns information specific to each splice site:
<ul><li>Site position/type: genomic location and type (donor/acceptor) of the&nbsp;   splice-site predicted to be lost by SpliceAI. Cryptic positions are relative
&nbsp;   to this genomic coordinate.
</li><li>Out of frame events: fraction of the top events that cause a frameshift. As&nbsp;   per <a href="https://pubmed.ncbi.nlm.nih.gov/36747048">https://pubmed.ncbi.nlm.nih.gov/36747048</a>, sites with 3/4 or more in-frame
&nbsp;   events are likely to be splice-rescued and not loss-of-function (LoF).
</li><li>Site sample count and max depth: sample count for this splice site and max&nbsp;   number of reads in any single sample representing annotated splicing in
&nbsp;   Genotype-Tissue Expression (GTEx). This information allows to filter events
&nbsp;   based on a minimum number of samples or minimum depth in GTEx.
</li><li>SpliceAI delta score (provided by SpliceVault)
</li></ul><p> Please cite the SpliceVault publication alongside Ensembl VEP if you use this
 resource: <a href="https://pubmed.ncbi.nlm.nih.gov/36747048">https://pubmed.ncbi.nlm.nih.gov/36747048</a></p><p> The tabix utility must be installed in your path to use this plugin. The
 SpliceVault TSV and respective index (TBI) for GRCh38 can be downloaded from:
<ul><li><a href="https://ftp.ensembl.org/pub/current_variation/SpliceVault/SpliceVault_data_GRCh38.tsv.gz">https://ftp.ensembl.org/pub/current_variation/SpliceVault/SpliceVault_data_GRCh38.tsv.gz</a></li><li><a href="https://ftp.ensembl.org/pub/current_variation/SpliceVault/SpliceVault_data_GRCh38.tsv.gz.tbi">https://ftp.ensembl.org/pub/current_variation/SpliceVault/SpliceVault_data_GRCh38.tsv.gz.tbi</a>
</li></ul><p> To filter results, please use filter_vep with the output file or standard
 output. Documentation on filter_vep is available at:
 <a href="https://www.ensembl.org/info/docs/tools/vep/script/vep_filter.html">https://www.ensembl.org/info/docs/tools/vep/script/vep_filter.html</a></p><p></ul><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv SpliceVault.pm ~/.vep/Plugins</p><p>./vep -i variations.vcf --plugin SpliceVault,file=/path/to/SpliceVault_data_GRCh38.tsv.gz</p><p># Stringently select predicted loss-of-function (pLoF) splicing variants
./filter_vep -i variant_effect_output.txt --filter "SPLICEVAULT_OUT_OF_FRAME_EVENTS &gt= 3"
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#FF0000">Splicing predictions</div></td><td>-</td><td>Ensembl</td></tr>
<tr id="StructuralVariantOverlap" class="bg1 plugin_row" data-category="structural_variant_data"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/StructuralVariantOverlap.pm">StructuralVariantOverlap</a></div></td><td><p> A VEP plugin that retrieves information from overlapping structural variants. <a class="button" href="#structuralvariantoverlap" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('structuralvariantoverlap');" id="a_structuralvariantoverlap">more</a></p><div id="div_structuralvariantoverlap" style="display:none;word-wrap:break-word;"><p> Parameters can be set using a key=value system:<table class="ss"><thead><tr><th>Argument</th><th>Description</th></tr></thead><tbody>
<tr class="bg1"><td><pre>file</pre></td><td>required - a VCF file of reference data. 
<tr class="bg2"><td><pre>percentage</pre></td><td>percentage overlap between SVs (default: 80) <tr class="bg1"><td><pre>reciprocal</pre></td><td>calculate reciprocal overlap, options:  0 or 1. (default: 0)                   (overlap is expressed as % of input SV by default)</p><p><tr class="bg2"><td><pre>cols</pre></td><td>colon delimited list of data types to return from the INFO fields (only AF by default) 
<tr class="bg1"><td><pre>same_type</pre></td><td>1/0 only report SV of the same type (eg deletions for deletions, off by default) 
<tr class="bg2"><td><pre>distance</pre></td><td>the distance the ends of the overlapping SVs should be within. 
<tr class="bg1"><td><pre>match_type</pre></td><td>only report reference SV which lie within or completely surround the input SV                   options: within, surrounding</p><p><tr class="bg2"><td><pre>label</pre></td><td>annotation label that will appear in the output (default: "SV_overlap")                   Example- input: label=mydata, output: mydata_name=refSV,mydata_PC=80,mydata_AF=0.05</p><p></td></tr></tbody></table><p> Example reference data</p><p><ul><li>1000 Genomes Project:&nbsp;   <a href="https://ftp.1000genomes.ebi.ac.uk/vol1/ftp/phase3/integrated_sv_map/ALL.wgs.mergedSV.v8.20130502.svs.genotypes.vcf.gz">https://ftp.1000genomes.ebi.ac.uk/vol1/ftp/phase3/integrated_sv_map/ALL.wgs.mergedSV.v8.20130502.svs.genotypes.vcf.gz</a></p><p></li><li>gnomAD:&nbsp;   <a href="https://storage.googleapis.com/gcp-public-data--gnomad/papers/2019-sv/gnomad_v2.1_sv.sites.vcf.gz">https://storage.googleapis.com/gcp-public-data--gnomad/papers/2019-sv/gnomad_v2.1_sv.sites.vcf.gz</a></p><p></li></ul><p> Example:
<pre class="code sh_sh">./vep -i structvariants.vcf --plugin StructuralVariantOverlap,file=gnomad_v2_sv.sites.vcf.gz
</pre><p>
</ul><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv StructuralVariantOverlap.pm ~/.vep/Plugins
./vep -i structvariants.vcf --plugin StructuralVariantOverlap,file=gnomad_v2_sv.sites.vcf.gz
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#601212">Structural variant data</div></td><td>-</td><td>Ensembl</td></tr>
<tr id="SubsetVCF" class="bg2 plugin_row" data-category="variant_data"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/SubsetVCF.pm">SubsetVCF</a></div></td><td><p> An Ensembl VEP plugin to retrieve overlapping records from a given VCF file.
 Values for POS, ID, and ALT, are retrieved as well as values for any requested
 INFO field. Additionally, the allele number of the matching ALT is returned. <a class="button" href="#subsetvcf" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('subsetvcf');" id="a_subsetvcf">more</a></p><div id="div_subsetvcf" style="display:none;word-wrap:break-word;"><p> Though similar to using <kbd>--custom</kbd>, this plugin returns all ALTs for a given
 POS, as well as all associated INFO values.</p><p> By default, only VCF records with a filter value of "PASS" are returned, 
 however this behaviour can be changed via the <kbd>filter</kbd> option.</p><p> The plugin accepts the following key=value parameters:<table class="ss"><thead><tr><th>Argument</th><th>Description</th></tr></thead><tbody><tr class="bg1"><td><pre>name</pre></td><td>short name added used as a prefix (required) <tr class="bg2"><td><pre>file</pre></td><td>path to tabix-index vcf file (required) <tr class="bg1"><td><pre>filter</pre></td><td>only consider variants marked as <kbd>PASS</kbd>, 1 or 0 (default, 1) <tr class="bg2"><td><pre>fields</pre></td><td>info fields to be returned (default, not used) <ul><li>'%' can delimit multiple fields</li><li>'*' can be used as a wildcard 
</td></tr></tbody></table><p></li></ul><p> Returns:
<ul><li><kbd>&ltname&gt_POS</kbd>: POS field from VCF</li><li><kbd>&ltname&gt_REF</kbd>: REF field from VCF (minimised)</li><li><kbd>&ltname&gt_ALT</kbd>: ALT field from VCF (minimised)</li><li><kbd>&ltname&gt_alt_index</kbd>: Index of matching variant (zero-based)</li><li><kbd>&ltname&gt_&ltfield&gt</kbd>: List of requested info values
</ul><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
./vep -i variations.vcf --plugin SubsetVCF,file=filepath.vcf.gz,name=myvfc,fields=AC*%AN*
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#B22222">Variant data</div></td><td><ul style="padding-left:1em"><li><a href="https://metacpan.org/pod/Data::Dumper" rel="external">Data::Dumper</a></li><li><a href="https://metacpan.org/pod/Storable" rel="external">Storable</a> qw(dclone)</li></ul></td><td>Joseph A. Prinz</td></tr>
<tr id="TranscriptAnnotator" class="bg1 plugin_row" data-category="transcript_annotation"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/TranscriptAnnotator.pm">TranscriptAnnotator</a></div></td><td><p> An Ensembl VEP plugin that annotates variant-transcript pairs based on a given file: <a class="button" href="#transcriptannotator" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('transcriptannotator');" id="a_transcriptannotator">more</a></p><div id="div_transcriptannotator" style="display:none;word-wrap:break-word;"><p><pre class="code sh_sh">--plugin TranscriptAnnotator,file=${HOME}/file.tsv.gz
</pre><p> Example of a valid tab-separated annotation file:</p><p><pre class="code sh_sh">#Chrom  Pos       Ref  Alt  Transcript       SIFT_score  SIFT_pred    Comment
11      436154    A    G    NM_001347882.2   0.03        Deleterious  Bad
11      1887471   C    T    ENST00000421485  0.86        Tolerated    Good
</pre>
 Please bgzip and tabix the file with commands such as:</p><p><pre class="code sh_sh">bgzip file.txt
tabix -b2 -e2 file.txt.gz
</pre><p> Options are passed to the plugin as key=value pairs:<table class="ss"><thead><tr><th>Argument</th><th>Description</th></tr></thead><tbody>
<tr class="bg1"><td><pre>file</pre></td><td>(mandatory) Tabix-indexed file to parse. Must contain variant             location (chromosome, position, reference allele, alternative allele)
            and transcript ID as the first 5 columns. Accepted transcript IDs
            include those from Ensembl and RefSeq.
<tr class="bg2"><td><pre>cols</pre></td><td>Colon-delimited list with names of the columns to append. Column             names are based on the last header line. By default, all columns
            (except the first 5) are appended.
<tr class="bg1"><td><pre>prefix</pre></td><td>String to prefix the name of appended columns (default: basename of             the filename without extensions). Set to 0 to avoid any prefix.
<tr class="bg2"><td><pre>trim</pre></td><td>Trim whitespaces from both ends of each column (default: 1). 
</td></tr></tbody></table><p> The tabix and bgzip utilities must be installed in your path to read the
 tabix-indexed annotation file: check <a href="https://github.com/samtools/htslib.git">https://github.com/samtools/htslib.git</a> for
 installation instructions.</p><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv TranscriptAnnotator.pm ~/.vep/Plugins
./vep -i variations.vcf --plugin TranscriptAnnotator,file=/path/to/file.txt.gz
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#FF2288">Transcript annotation</div></td><td><a href="https://metacpan.org/pod/File::Basename" rel="external">File::Basename</a></td><td>Ensembl</td></tr>
<tr id="TSSDistance" class="bg2 plugin_row" data-category="nearby_features"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/TSSDistance.pm">TSSDistance</a></div></td><td><p> An Ensembl VEP plugin that calculates the distance from the transcription
 start site for upstream variants. <a class="button" href="#tssdistance" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('tssdistance');" id="a_tssdistance">more</a></p><div id="div_tssdistance" style="display:none;word-wrap:break-word;"><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv TSSDistance.pm ~/.vep/Plugins
./vep -i variations.vcf --plugin TSSDistance
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#E75480">Nearby features</div></td><td>-</td><td>Ensembl</td></tr>
<tr id="UTRAnnotator" class="bg1 plugin_row" data-category="transcript_annotation"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/UTRAnnotator.pm">UTRAnnotator</a></div></td><td><p>    An Ensembl VEP plugin that annotates the effect of 5' UTR variant especially for variant creating/disrupting upstream ORFs.
    Available for both GRCh37 and GRCh38. <a class="button" href="#utrannotator" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('utrannotator');" id="a_utrannotator">more</a></p><div id="div_utrannotator" style="display:none;word-wrap:break-word;"><p>    Options are passed to the plugin as key=value pairs:<table class="ss"><thead><tr><th>Argument</th><th>Description</th></tr></thead><tbody>
<tr class="bg1"><td><pre>file</pre></td><td>(Required) Path to UTRAnnotator data file: <ul><li>Download <kbd>uORF_5UTR_GRCh37_PUBLIC.txt</kbd> or <kbd>uORF_5UTR_GRCh38_PUBLIC.txt</kbd>&nbsp;                    from <a href="https://github.com/Ensembl/UTRannotator">https://github.com/Ensembl/UTRannotator</a>
</li><li>Download from <a href="http://sorfs.org">http://sorfs.org</a><tr class="bg2"><td><pre>&nbsp;    max_overlap</pre></td><td>(Optional) Maximum percentage of overlap between variant and &nbsp;                  UTR for UTR annotation (default: 100)</p><p></td></tr></tbody></table><p></li></ul><p>    Citation</p><p>    About the role of 5'UTR variants in human genetic disease:</p><p>    Whiffin, N., Karczewski, K.J., Zhang, X. et al. Characterising the loss-of-function impact of 5' untranslated region variants in 15,708 individuals. Nat Commun 11, 2523 (2020). <a href="https://doi.org/10.1038/s41467-019-10717-9">https://doi.org/10.1038/s41467-019-10717-9</a></p><p>    About UTRAnnotator:</p><p>    The original UTRAnnotator plugin is written by Xiaolei Zhang et al. Later adopted by Ensembl VEP plugins with some changes.
    You can find the original plugin here -
    <a href="https://github.com/ImperialCardioGenetics/UTRannotator">https://github.com/ImperialCardioGenetics/UTRannotator</a> </p><p>    Please cite the UTRannotator publication alongside the Ensembl VEP if you use this resource -
    Annotating high-impact 5'untranslated region variants with the UTRannotator Zhang, X., Wakeling, M.N., Ware, J.S, Whiffin, N. Bioinformatics; doi: <a href="https://academic.oup.com/bioinformatics/advance-article/doi/10.1093/bioinformatics/btaa783/5905476">https://academic.oup.com/bioinformatics/advance-article/doi/10.1093/bioinformatics/btaa783/5905476</a></p><p></ul><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv UTRAnnotator.pm ~/.vep/Plugins
vep -i variations.vcf --plugin UTRAnnotator,file=/path/to/uORF_starts_ends_GRCh38_PUBLIC.txt</p><p># skip annotation for variants with a 80% or higher overlap of the UTR
vep -i variations.vcf --plugin UTRAnnotator,file=/path/to/uORF_starts_ends_GRCh38_PUBLIC.txt,max_overlap=80
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#FF2288">Transcript annotation</div></td><td><ul style="padding-left:1em"><li><a href="https://metacpan.org/pod/List::Util" rel="external">List::Util</a> qw(min max)</li><li><a href="https://metacpan.org/pod/Scalar::Util" rel="external">Scalar::Util</a> qw(looks_like_number)</li></ul></td><td>Ensembl</td></tr>
<tr id="VARITY" class="bg2 plugin_row" data-category="pathogenicity_predictions"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/VARITY.pm">VARITY</a></div></td><td><p> This is a plugin for the Ensembl Variant Effect Predictor (VEP) that
 adds the pre-computed VARITY scores to predict pathogenicity of 
 rare missense variants to the Ensembl VEP output. <a class="button" href="#varity" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('varity');" id="a_varity">more</a></p><div id="div_varity" style="display:none;word-wrap:break-word;"><p> Please cite the VARITY publication alongside Ensembl VEP if you use this resource:
 <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8715197/">https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8715197/</a></p><p> Running options :</p><p> VARITY scores can be downloaded using
<pre class="code sh_sh">wget <a href="http://varity.varianteffect.org/downloads/varity_all_predictions.tar.gz">http://varity.varianteffect.org/downloads/varity_all_predictions.tar.gz</a>
</pre><p> The files can be tabix processed by :
<pre class="code sh_sh">tar -xzvf varity_all_predictions.tar.gz 
cat varity_all_predictions.txt | (head -n 1 && tail -n +2  | sort -t$'\t' -k 1,1 -k 2,2n) &gt varity_all_predictions_sorted.tsv
sed '1s/.*/#&/'  varity_all_predictions_sorted.tsv &gt varity_all_predictions.tsv  # to add a # in the first line of the file 
bgzip varity_all_predictions.tsv
tabix -f -s 1 -b 2 -e 2 varity_all_predictions.tsv.gz
</pre><p>
 Requirements:
 The tabix utility must be installed in your path to use this plugin.
 The --assembly flag is required to use this plugin.</p><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv VARITY.pm ~/.vep/Plugins
./vep -i variations.vcf --assembly GRCh37 --plugin VARITY,file=/path/to/varity_all_predictions.txt
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#1E90FF">Pathogenicity predictions</div></td><td>-</td><td>Ensembl</td></tr>
</tbody></table><!-- Plugins list - end -->
  
  <p> We hope that these will serve as useful examples for users implementing
	new plugins. If you have any questions about the system, or suggestions for
	enhancements please let us know on the 
  <a href="//lists.ensembl.org/mailman/listinfo/dev" rel="external">ensembl-dev</a> 
  mailing list.<br />
  We also encourage you to share any plugins you develop: we are happy to accept pull requests on the <a href="//github.com/Ensembl/VEP_plugins" rel="external">VEP_plugins</a> git repository. </p>
    

  <p> There are further published plugins available outside the Ensembl VEP repository including:
  <li> <a href="https://github.com/konradjk/loftee">LOFTEE </a> a Loss-Of-Function Transcript Effect Estimator (<a href="https://www.nature.com/articles/s41586-020-2308-7">Konrad Karczewski et al,2020</a>)


	
  <hr/>
  <h2 id="plugins_how">How it works</h2>
	
	<p>Plugins are run once Ensembl VEP has finished its analysis for each line of
	the output, but before anything is printed to the output file.</p>

	<p>When each
	plugin is called (using the <i>run</i> method) it is passed two data structures
	to use in its analysis; the first is a data structure containing all the
	data for the current line, and the second is a reference to a variation
	API object that represents the combination of a variant allele and an
	overlapping or nearby genomic feature (such as a transcript or regulatory
	region).</p>

	<p>This object provides access to all the relevant API objects that
	may be useful for further analysis by the plugin (such as the current
	VariationFeature and Transcript). Please refer to the <a
	href="/info/docs/Doxygen/variation-api/index.html"><img style="vertical-align:bottom" src="/i/16/documentation.png"/> Ensembl Variation API
	documentation</a> for more details.</p>

  <hr/>
  <h2 id="plugins_functionality">Functionality</h2>
	
	<p>We expect that most plugins will simply add information to the last
	column of the output file, the "Extra" column, and the plugin system assumes
	this in various places, but plugins are also free to alter the output line
	as desired.</p>
	
	<p>The only hard requirement for
	a plugin to work with Ensembl VEP is that it implements a number of required
	methods (such as <i>new</i> which should create and return an instance of this
	plugin, <i>get_header_info</i> which should return descriptions of the type of
	data this plugin produces to be included in Ensembl VEP output's header, and
	<i>run</i> which should actually perform the logic of the plugin).</p>

	<p>To make development of plugins easier, we suggest that users use the <a
	href="/info/docs/Doxygen/variation-api/classBio_1_1EnsEMBL_1_1Variation_1_1Utils_1_1BaseVepPlugin.html">Bio::EnsEMBL::Variation::Utils::BaseVepPlugin</a>
	module as their base class, which provides default implementations of all the
	necessary methods which can be overridden as required. Please refer to the
	documentation in this module for details of all required methods and for a
	simple example of a plugin implementation. </p>
	
	<hr/>
	<h2 id="plugins_filter">Filtering using plugins</h2>
	
	<p> A common use for plugins will be to filter the output in some way (for
	example to limit output lines to missense variants) and so we provide
	a simple mechanism to support this.</p>

	<p>The <i>run</i> method of a plugin is assumed
	to return a reference to a hash containing information to be included in the
	output, and if a plugin should not add any data to a particular line
	it should return an empty hashref. If a plugin should instead filter a
	line and exclude it from the output, it should return <i>undef</i> from its <i>run</i>
	method, this also means that no further plugins will be run on the line.</p>

	<p>If you are developing a filter plugin, we suggest that you use the <a
	href="/info/docs/Doxygen/variation-api/classBio_1_1EnsEMBL_1_1Variation_1_1Utils_1_1BaseVepFilterPlugin.html">Bio::EnsEMBL::Variation::Utils::BaseVepFilterPlugin</a>
	as your base class and then you need only override the <i>include_line</i> method
	to return true if you want to include this line, and false otherwise.
	Again, please refer to the documentation in this module for more details and an
	example	implementation of a missense filter. </p>
	
	<hr/>
	<h2 id="plugins_use">Using plugins</h2>
    
    <p> In order to run a plugin you need to include the plugin module in Perl's
    library path somehow; by default Ensembl VEP includes the <i>&#126;/.vep/Plugins</i>
    directory in the path, so this is a convenient place to store plugins, but
    you are also able to include modules by any other means (e.g using the
    <i>$PERL5LIB</i> environment variable in Unix-like systems).<br />You can then run a
    plugin using the <a href="vep_options.html#opt_plugin">--plugin</a> command line option, passing the name of
    the plugin module as the argument.</p>

    <p>For example, if your plugin is in a
    module called <i>MyPlugin.pm</i>, stored in <i>&#126;/.vep/Plugins</i>, you can run it
    with a command line like: </p>
	
	<pre class="code sh_sh">./vep -i input.vcf --plugin MyPlugin</pre>
	
	<p> You can pass arguments to the plugin's 'new' method by including them after
	the plugin name on the command line, separated by commas, e.g.: </p>
	
	<pre class="code sh_sh">./vep -i input.vcf --plugin MyPlugin,1,FOO</pre>
	
	<p> If your plugin inherits from BaseVepPlugin, you can then retrieve these
	parameters as a list from the <i>params</i> method. </p>
	
	<p> You can run multiple plugins by supplying multiple <a href="vep_options.html#opt_plugin">--plugin</a>
	arguments. Plugins are run serially in the order in which they are specified
	on the command line, so they can be run as a pipeline, with, for example, a
	later plugin filtering output based on the results from an earlier plugin.
	Note though that the first plugin to filter a line 'wins', and any later
	plugins won't get
	run on a filtered line. </p>

  <div id="nv_species">
    <div class="info" style="float:left">
      <h3>Plugin warnings</h3>
      <div class="message-pad">
        <p>By default, an Ensembl VEP run does not fail even when a plugin raises
        warnings or compilation errors. To avoid this, please use the
        <a href="vep_options.html#opt_safe">--safe</a> flag. This is
        particularly useful to ensure plugins run properly when using Ensembl VEP in
        pipelines.</p>
      </div>
    </div>
    <div style="clear:both"></div>
  </div>

	<hr/>
	<h2 id="plugins_intergenic">Intergenic variants</h2>
	
	<p> When a variant falls in an intergenic region, it will usually not have
	any consequence types called, and hence will not have any associated
	VariationFeatureOverlap objects. In this special case, Ensembl VEP creates a new
	VariationFeatureOverlap that overlaps a feature of type "Intergenic".</p>

	<p>To force your plugin to handle these, you must add "Intergenic" to the feature
	types that it will recognize; you do this by writing your own feature_types
	sub-routine:</p>
	
	<pre class="code sh_perl sh_sourceCode">sub feature_types {
    return ['Transcript', 'Intergenic'];
}</pre>
	
	<p> This will cause your plugin to handle any variation features that
	overlap transcripts or intergenic regions. To also include any regulatory
	features, you should use the generic type "Feature":</p>
	
	<pre class="code sh_perl sh_sourceCode">sub feature_types {
    return ['Feature', 'Intergenic'];
}</pre>
	
</div>

<div style="clear:both"><h1 style="page-break-after: always">&nbsp</h1></div>    
<div>
	
  <div style="float:right"><img src="/img/vep_logo.png"/></div>
	
  <h1 id="top"><span style="color:#006;padding-right:15px">Ensembl Variant Effect Predictor</span><span style="color:#666"><img src="/i/16/user.png"/> Examples and use cases</span></h1>
  <hr/>
  
  
  <h2 id="examples">Example commands</h2>
  
  <ul>
    <li>
      <p>Read input from <b>STDIN</b>, output to <b>STDOUT</b></p>
      <pre class="code sh_sh">./vep --cache -o stdout</pre>
    </li>
    <li>
      <p>Add <b>regulatory</b> region <b>consequences</b></p>
      <pre class="code sh_sh">./vep --cache -i variants.txt --regulatory</pre>
    </li>
    <li>
      <p>Input file <kbd>variants.vcf.txt</kbd>, input file <b>format VCF</b>, add <b>gene symbol</b> identifiers</p>
      <pre class="code sh_sh">./vep --cache -i variants.vcf.txt --format vcf --symbol</pre>
    </li>
    <li>
      <p><b>Filter</b> out <b>common variants</b> based on 1000 Genomes data</p>
      <pre class="code sh_sh">./vep --cache -i variants.txt --filter_common</pre>
    </li>
    <li>
      <p><b>Force overwrite</b> of output file <kbd>variants_output.txt</kbd>, check for existing
      <b>co-located variants</b>, output only <b>coding sequence</b> consequences, output
      <b>HGVS names</b></p>
      <pre class="code sh_sh">./vep --cache -i variants.txt -o variants_output.txt --force --check_existing --coding_only --hgvs</pre>
    </li>
    <li>
      <p>Specify <b>DB connection parameters</b> in registry file <kbd>ensembl.registry</kbd>, add <b>SIFT</b>
      score and prediction, <b>PolyPhen</b> prediction</p>
      <pre class="code sh_sh">./vep --database -i variants.txt --registry ensembl.registry --sift b --polyphen p</pre>
    </li>
    <li>
      <p>Connect to <b>Ensembl Genomes</b> db server for <i>Arabidopsis thaliana</i></p>
      <pre class="code sh_sh">./vep --database -i variants.txt --genomes --species arabidopsis_thaliana</pre>
    </li>
    <li>
      <p>Load config from <b>ini file</b>, run in <b>quiet mode</b></p>
      <pre class="code sh_sh">./vep --config vep.ini -i variants.txt -q</pre>
    </li>
    <li>
      <p>Use <b>cache</b> in <kbd>/home/vep/mycache/</kbd>, use <b>gzcat</b> instead of zcat</p>
      <pre class="code sh_sh">./vep --cache --dir /home/vep/mycache/ -i variants.txt --compress gzcat</pre>
    </li>
    <li>
      <p>Add custom position-based <b>phenotype</b> annotation from remote <b>BED file</b></p>
      <pre class="code sh_sh">./vep --cache -i variants.vcf --custom file=ftp://ftp.myhost.org/data/phenotypes.bed.gz,short_name=phenotype</pre>
    </li>
    <li>
      <p>Use the <b>plugin</b> named MyPlugin, output only the variation name, feature, consequence type and MyPluginOutput <b>fields</b></p>
      <pre class="code sh_sh">./vep --cache -i variants.vcf --plugin MyPlugin --fields Uploaded_variation,Feature,Consequence,MyPluginOutput</pre>
    </li>
    <li>
      <p>Right align variants before consequence calculation. For more information, see <a
         href="/info/docs/tools/vep/script/vep_other.html#shifting"
         target="_blank">here</a>.</p>
      <pre class="code sh_sh">./vep --cache -i variants.vcf --shift_3prime 1</pre>
    </li>
     <li>
      <p>Report uploaded allele before minimisation. For more information, see <a
         href="/info/docs/tools/vep/script/vep_options.html#opt_uploaded_allele"
         target="_blank">here</a>.</p>
      <pre class="code sh_sh">./vep --cache -i variants.vcf --uploaded_allele</pre>
    </li>
  </ul>
  
    
  <hr/>
  
  <h2 id="gnomad">gnomAD</h2>
  
  <p> <a href="http://gnomad.broadinstitute.org/" rel="external">gnomAD</a> exome frequency data is included in Ensembl VEP's cache files from release 90, replacing ExAC; use <a href="vep_options.html#opt_af_gnomade">--af_gnomade</a> to enable using this data. Ensembl VEP can also retrieve frequency data from the gnomAD genomes set or ExAC via Ensembl VEP's custom annotation functionality.<p>
  <p>For the latest gnomAD data, please visit <a href="https://gnomad.broadinstitute.org/downloads" rel="external">gnomAD downloads</a>.<p>
  
  <ol>
    <li>Ensembl VEP requires Bio::DB::HTS to read data from tabix-indexed VCFs - see <a href="vep_download.html#installer"><img style="vertical-align:bottom" src="/i/16/download.png"/> installation instructions</a></li>
    <li>The Ensembl FTP site hosts abridged VCF files for gnomAD additionally remapped to GRCh38 using <a href="http://crossmap.sourceforge.net/" rel="external">CrossMap</a>. It is possible for Ensembl VEP to read these files directly from their remote location, though for optimal performance the VCF and index should be downloaded to a local file system.
      <ul>
        <li>
          <b>GRCh38</b>
          <ul>
            <li>gnomAD genomes (r2.1, remapped with CrossMap):
              <a href="[[SPECIESDEFS::ENSEMBL_FTP_URL]]/data_files/homo_sapiens/GRCh38/variation_genotype/gnomad/r2.1/genomes/">[VCFs and tabix indexes]</a>
            </li>
            <li>gnomAD exomes (r2.1, remapped with CrossMap):
              <a href="[[SPECIESDEFS::ENSEMBL_FTP_URL]]/data_files/homo_sapiens/GRCh38/variation_genotype/gnomad/r2.1/exomes/">[VCFs and tabix indexes]</a>
            </li>
            <li>ExAC (v0.3, remapped using CrossMap):
              <a href="[[SPECIESDEFS::ENSEMBL_FTP_URL]]/data_files/homo_sapiens/GRCh38/variation_genotype/ExAC.0.3.GRCh38.vcf.gz">[VCF]</a>
              <a href="[[SPECIESDEFS::ENSEMBL_FTP_URL]]/data_files/homo_sapiens/GRCh38/variation_genotype/ExAC.0.3.GRCh38.vcf.gz.tbi">[tabix index]</a>
            </li>
          </ul>
        </li>

        <li>
          <b>GRCh37</b>
          <ul>
            <li>gnomAD genomes (r2.1):
              <a href="[[SPECIESDEFS::ENSEMBL_FTP_URL]]/data_files/homo_sapiens/GRCh37/variation_genotype/">[VCF and tabix indexes]</a>
            </li>
            <li>gnomAD exomes (r2.1):
              <a href="[[SPECIESDEFS::ENSEMBL_FTP_URL]]/data_files/homo_sapiens/GRCh37/variation_genotype/">[VCF and tabix indexes]</a>
            </li>
            <li>ExAC (v0.3):
              <a href="[[SPECIESDEFS::ENSEMBL_FTP_URL]]/data_files/homo_sapiens/GRCh37/variation_genotype/ExAC.0.3.GRCh37.vcf.gz">[VCF]</a>
              <a href="[[SPECIESDEFS::ENSEMBL_FTP_URL]]/data_files/homo_sapiens/GRCh37/variation_genotype/ExAC.0.3.GRCh37.vcf.gz.tbi">[tabix index]</a>
            </li>
          </ul>
        </li>

      </ul>
    </li>
    <li>
      <p>Run Ensembl VEP with the following command (using the GRCh38 input example) to get locations and continental-level allele frequencies:</p>
      <pre class="code sh_sh">./vep -i examples/homo_sapiens_GRCh38.vcf --cache \
--custom file=gnomad.genomes.r2.0.1.sites.GRCh38.noVEP.vcf.gz,short_name=gnomADg,format=vcf,type=exact,coords=0,fields=AF_AFR%AF_AMR%AF_ASJ%AF_EAS%AF_FIN%AF_NFE%AF_OTH</pre>
      <p>You will then see data under field names as described in the Ensembl VEP output header:</p>
      <pre class="code sh_sh">## gnomADg : gnomad.genomes.r2.0.1.sites.GRCh38.noVEP.vcf.gz (exact)
## gnomADg_AFR_AF : AFR_AF field from gnomad.genomes.r2.0.1.sites.GRCh38.noVEP.vcf.gz
## gnomADg_AMR_AF : AMR_AF field from gnomad.genomes.r2.0.1.sites.GRCh38.noVEP.vcf.gz
...</pre>
      where the gnomADg field contains the ID (or coordinates if no ID found) of the variant in the VCF file. Any of the fields in the gnomAD file INFO field can be added by appending them to the list in your Ensembl VEP command.
    </li>
  </ol>
    
  <hr/>
  
  <h2 id="gerp">Conservation scores</h2>
  
  <p> You can use the <a href="vep_custom.html">custom annotation</a>
  feature to add conservation scores to your output. For example, to add GERP
  scores, download the bigWig file from the list below, and run Ensembl VEP with the
  following flag:</p>
  
  <pre class="code sh_sh">./vep --cache -i example.vcf --custom file=All_hg19_RS.bw,short_name=GERP,format=bigwig</pre>
  
  <p> Example conservation score files: </p>
 
  <div>
    <div style="float:left"> 
      <b>Human (GRCh38)</b>
      <ul>
        <li><a href="http://hgdownload.soe.ucsc.edu/goldenPath/hg38/phastCons7way/" rel="external">phastCons 7-way</a> </li>
        <li><a href="http://hgdownload.soe.ucsc.edu/goldenPath/hg38/phastCons20way/" rel="external">phastCons 20-way</a> </li>
        <li><a href="http://hgdownload.soe.ucsc.edu/goldenPath/hg38/phastCons100way/" rel="external">phastCons 100-way</a> </li>
        <li><a href="http://hgdownload.soe.ucsc.edu/goldenPath/hg38/phyloP7way/" rel="external">phyloP 7-way</a> </li>
        <li><a href="http://hgdownload.soe.ucsc.edu/goldenPath/hg38/phyloP20way/" rel="external">phyloP 20-way</a> </li>
        <li><a href="http://hgdownload.soe.ucsc.edu/goldenPath/hg38/phyloP100way/" rel="external">phyloP 100-way</a> </li>
      </ul>
    </div>
    <div style="float:left;margin-left:40px">
      <b>Human (GRCh37)</b>
      <ul>
        <li><a href="http://hgdownload.soe.ucsc.edu/gbdb/hg19/bbi/All_hg19_RS.bw" rel="external">GERP</a> </li>
        <li><a href="http://hgdownload.soe.ucsc.edu/goldenPath/hg19/phastCons46way/" rel="external">phastCons 46-way</a> </li>
        <li><a href="http://hgdownload.soe.ucsc.edu/goldenPath/hg19/phastCons100way/" rel="external">phastCons 100-way</a> </li>
        <li><a href="http://hgdownload.soe.ucsc.edu/goldenPath/hg19/phyloP46way/" rel="external">phyloP 46-way</a> </li>
        <li><a href="http://hgdownload.soe.ucsc.edu/goldenPath/hg19/phyloP100way/" rel="external">phyloP 100-way</a> </li>
      </ul>
    </div>
    <div class="clear"></div>
  </div>  

  <p> All files provided by the UCSC genome browser - files for other species
  are available from their <a
  href="http://hgdownload.soe.ucsc.edu/goldenPath/" rel="external">FTP site</a>, though be sure
  to use the file corresponding to the <a
  href="/Homo_sapiens/Info/Annotation#assembly">correct assembly</a>. </p>
    
  <hr/>
  
  
  <h2 id="dbNSFP">dbNSFP</h2>

  <p> dbNSFP - <a href="http://www.ncbi.nlm.nih.gov/pubmed/21520341"
  rel="external">&quot;a lightweight database of human nonsynonymous SNPs and
  their functional predictions&quot;</a> - provides pathogenicity predictions
  from many tools (including SIFT, LRT, MutationTaster, FATHMM) across
  every possible missense substitution in the human proteome. </p>

  <p> Ensembl VEP plugins sometimes require data processed in specific ways as
  arguments. Any requirements and usage instructions for each plugin can be
  found in the <a href="vep_plugins.html">plugin documentation</a>. </p>

  <p> In the case of the dbNSFP.pm plugin, the data needs to be
  <a href="https://sites.google.com/site/jpopgen/dbNSFP" rel="external">downloaded</a>
  and then processed into a format that the plugin can use. Note that there are
  two distinct branches of the files provided for academic and commercial usage;
  please use the appropriate files for your use case. </p>
  
  <p> After downloading the file, you will need to process it so that tabix can
  index it correctly. This will take a while as the file is very large! Note
  that you will need the <a href="http://samtools.sourceforge.net/tabix.shtml"
  rel="external">tabix</a> utility in your path to use dbNSFP. </p>

  <pre class="code sh_sh">version=4.5c
unzip dbNSFP${version}.zip
zcat dbNSFP${version}_variant.chr1.gz | head -n1 > h

# GRCh38/hg38 data
zgrep -h -v "^#chr" dbNSFP${version}_variant.chr* | sort -k1,1 -k2,2n - | cat h - | bgzip -c > dbNSFP${version}_grch38.gz
tabix -s 1 -b 2 -e 2 dbNSFP${version}_grch38.gz

# GRCh37/hg19 data
zgrep -h -v "^#chr" dbNSFP${version}_variant.chr* | awk '$8 != "." ' | sort -k8,8 -k9,9n - | cat h - | bgzip -c > dbNSFP${version}_grch37.gz
tabix -s 8 -b 9 -e 9 dbNSFP${version}_grch37.gz</pre>

  <p> Then simply download the <a
  href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/dbNSFP.pm"
  rel="external">dbNSFP.pm plugin</a> and place it either in
  <b>$HOME/.vep/Plugins/</b> or a path in your <b>$PERL5LIB</b>. When you run
  Ensembl VEP with the plugin, you will need to select some of the columns that you
  wish to retrieve; to list them run Ensembl VEP with the plugin and the path to the
  dbNSFP file and no further parameters: </p>
  
  <pre class="code sh_sh">./vep --cache --force --plugin dbNSFP,dbNSFP4.5c_grch38.txt.gz
2014-04-04 11:27:05 - Read existing cache info
2014-04-04 11:27:05 - Auto-detected FASTA file in cache directory
2014-04-04 11:27:05 - Checking/creating FASTA index
2014-04-04 11:27:05 - Failed to instantiate plugin dbNSFP: ERROR: No columns selected to fetch. Available columns are:
#chr,pos(1-coor),ref,alt,aaref,aaalt,hg18_pos(1-coor),genename,Uniprot_acc,
Uniprot_id,Uniprot_aapos,Interpro_domain,cds_strand,refcodon,SLR_test_statistic,
codonpos,fold-degenerate,Ancestral_allele,Ensembl_geneid,Ensembl_transcriptid,
...</pre>
  
  <p> Note that some of these fields are replicates of those produced by the
  core Ensembl VEP code (e.g. <a href="vep_options.html#opt_sift">SIFT</a>, the <a
  href="vep_options.html#opt_af_1kg">1000 Genomes Project</a> frequencies) - you should use the
  Ensembl VEP options for frequency information rather than the annotations from
  dbNSFP as the dbNSFP file covers <b>only</b> missense substitutions. Other
  fields, such as the conservation scores, may be better served by using
  genome-wide files as described <a href="#gerp">above</a>. </p>
  
  <p> To select fields, just add them as a comma-separated list to your command
  line:</p>
  
  <pre class="code sh_sh">./vep --cache --force --plugin dbNSFP,dbNSFP4.5c_grch38.txt.gz,LRT_score,FATHM_score,MutationTaster_score</pre>
  
  <p> One final point to note is that the dbNSFP scores are frozen on a
  particular Ensembl release's transcript set; check the readme file on their
  download site to find out exactly which. While in the majority of cases
  protein sequences don't change between releases, in some circumstances the
  protein sequence used by Ensembl VEP in the latest release may differ from the
  sequence used to calculate the scores in dbNSFP. </p>
  
  <hr/>


  <h2 id="StructVar">Structural variants</h2>

  <p>Ensembl VEP can annotate structural variants (SV) with their predicted effect on other genomic features.
     For more information on SV input format, see <a href="/info/docs/tools/vep/vep_formats.html#sv">here</a>.</p>

  <h4>Prediction process</h4>
  <ul>
  <li> If the INFO keys <code>END</code> or <code>SVLEN</code> are present, the proportion of any overlapping feature covered by the variant is calculated</li>
  <li> The alternative allele (or <code>SVTYPE</code> in older VCF files) defines the type of structural variant; some types of structural variants are tested for specific consequences:</li>

  <table class="ss">
    <tr>
      <th>Structural variant type</th>
      <th>Abbreviation</th>
      <th>Specific consequences</th>
    </tr>
    <tr class="bg1">
      <td>Insertion</td>
      <td>INS</td>
      <td>Feature elongation</td>
    </tr>
    <tr class="bg2">
      <td>Deletion</td>
      <td>DEL</td>
      <td>Feature truncation</td>
    </tr>
    <tr class="bg1">
      <td>Duplication</td>
      <td>DUP</td>
      <td>Feature amplification/elongation</td>
    </tr>
    <tr class="bg2">
      <td>Inversion</td>
      <td>INV</td>
      <td><i>Not tested for any specific consequence</i></td>
    </tr>
    <tr class="bg1">
      <td>Copy number variation</td>
      <td>CNV</td>
      <td>Feature amplification/elongation (if copy number is 2) or truncation (if copy number is 0)</td>
    </tr>
    <tr class="bg2">
      <td>Breakpoint variant</td>
      <td>BND</td>
      <td>Feature truncation</td>
    </tr>
  </table>
  </ul>

  <h5> Insertions and deletions</h5>

  <ul>
    <li> Supports <a href="/info/docs/tools/vep/vep_formats.html#sv">mobile element insertions/deletions</a>, including ALU, HERV, LINE1 and SVA elements
    <ul>
      <li> Currently, mobile element variants are treated as any insertion/deletion
    </ul>
  </ul>

  <h5> Breakpoint variants</h5>

  <ul>
    <li> Supports chromosome synonyms in breakends (such as <code>chr4</code> and <code>NC_000004.12</code>)
    <li> Processes <a href="/info/docs/tools/vep/vep_formats.html#sv">single breakends and multiple, comma-separated alternative breakends</a>
    <li> Consequences are reported for each breakend; for instance, for a VCF input like <code>1 7936271 . N N[12:58877476[,N[X:10932343[</code>, it will report the consequences for each of the 3 breakends:
    <ul>
      <li> <code>N[12:58877476[</code>: consequences for the first alternative breakend near chr12:58877476
      <li> <code>N[X:10932343[</code>: consequences for the second alternative breakend near chrX:10932343
      <li> <code>N.</code>: consequences for the reference breakend near chr1:7936271 (represented as detailed in the <a href="https://samtools.github.io/hts-specs/VCFv4.4.pdf" rel="external">VCF 4.4 specification, section 5.4.9: Single breakends</a>)
    </ul>
    <li> In case of specific breakends not overlaping any reported Ensembl features (such as transcripts and regulatory regions), that specific breakend will <b>NOT</b> be presented in VEP output.
  </ul>

  <h4> Reported overlaps</h4>
  <ul>
  <li> Ensembl VEP calculates the length and proportion of each genomic feature overlapped by a structural variant
  <li> Use the <a href="vep_options.html#opt_overlaps">--overlaps</a> option to enable this when using VCF or tab format.
       (This is reported by default in standard Ensembl VEP and JSON format.)
  <li> The keys <code>bp_overlap</code> and <code>percentage_overlap</code> are used in JSON format and <code>OverlapBP</code> and <code>OverlapPC</code> in other formats.
  </ul>

  <h4> Plugin support</h4>

  <ul>
    <li> <a href="vep_plugins.html#CADD">CADD plugin</a>
    <li> <a href="vep_plugins.html#Conservation">Conservation plugin</a>
    <li> <a href="vep_plugins.html#NearestGene">NearestGene plugin</a>
    <li> <a href="vep_plugins.html#Phenotypes">Phenotypes plugin</a>
    <li> <a href="vep_plugins.html#StructuralVariantOverlap">StructuralVariantOverlap plugin</a>: please note that all features of this plugin have been ported to <a href="vep_custom.html">--custom annotation</a>, with additional improvements
    <li> <a href="vep_plugins.html#TSSDistance">TSSDistance plugin</a>
  </ul>

  <h4> Changing memory requirements</h4>
  <ul>
  <li> By default, Ensembl VEP does not annotate variants larger than 10M. If you are using the command
  line tool, you can use the <a href="vep_options.html#opt_max_sv_size">--max_sv_size</a> option to modify this.
    <ul>
      <li> This limit is not associated with breakpoint variants: each breakend in a breakpoint variant is analysed by Ensembl VEP as a single base (the alternative sequence is currently ignored).
    </ul>
  <li>By default, variants are analysed in batches of 5000. Using the <a href="vep_options.html#opt_buffer_size">--buffer_size</a>
  option to reduce this can reduce memory requirements, especially if your data is sparse.
  A smaller buffer size is essential when annotating structural variants with regulatory data.
  </ul>

  <hr/>


  
  <h2 id="pangenomes">Pangenome assemblies</h2>

  <p> Ensembl VEP is able to analyse variants in <b>any species or assembly</b> (even if not part of <a href="/info/about/species.html">Ensembl data</a>) by providing your own <a href="vep_cache.html#fasta">FASTA file</a> and <a href="vep_cache.html#gff" target="_blank">GFF/GTF annotation</a>: </p>

  <pre class="code sh_sh">./vep -i variants.txt -o variants_output.txt --gff data.gff.gz --fasta genome.fa.gz</pre>

  <p> We also provide data for other assemblies besides those supported in the current Ensembl and Ensembl Genomes sites. </p>

  <h3 id="HPRC">HPRC assemblies</h3>

  <p>The <a href="https://humanpangenome.org/" rel="external">Human Pangenome Reference Consortium (HPRC)</a> aims to sequence 350 individuals of diverse ancestries, producing a pangenome of 700 haplotypes by the end of 2024. The first publication (<i><a href="https://www.nature.com/articles/s41586-023-05896-x" rel="external">A draft human pangenome reference</a></i>) describes 47 phased, diploid assemblies from a cohort of genetically diverse individuals.</p>

  <p>The Ensembl VEP command-line tool (CLI) can annotate and filter variants called against the latest human assemblies, including the telomere-to-telomere assembly of the CHM13 cell line (T2T-CHM13). We have annotated genes on these human assemblies, based on Ensembl/<a href="https://www.gencodegenes.org/human/release_38.html" rel="external">GENCODE 38</a> genes and transcripts, via a new mapping pipeline as detailed in the Methods section of <i><a href="https://www.nature.com/articles/s41586-023-05896-x" rel="external">A draft human pangenome reference</a></i>. The links to download and visualise the human annotations for HPRC assemblies are summarised in the <a href="https://projects.ensembl.org/hprc/" rel="external">Ensembl HPRC data page</a>.</p>

  <h4 id="HPRC_run">Running Ensembl VEP with the human HPRC assemblies</h4>

  <p>Currently, Ensembl VEP can only be run with HPRC assemblies in offline mode, one assembly at a time. There are two ways to use Ensembl VEP with HPRC assemblies:</p>

  <ul>
    <li> Using <b>Ensembl VEP cache</b> with (recommended) <b>FASTA sequence</b> (the most efficient way) </li>
    <li> Using <b>GTF annotation</b> with (mandatory) <b>FASTA sequence</b> </li>
  </ul>

  <p>In the examples below, we demonstrate annotating variants on <b>T2T-CHM13v2.0</b> (<a href="https://ftp.ensembl.org/pub/rapid-release/species/Homo_sapiens/GCA_009914755.4/" rel="external">GCA_009914755.4</a> assembly). To create a sample VCF to use in the examples below, you can take the first 100 lines from the ClinVar VCF file mapped to T2T-CHM13:</p>

  <pre class="code sh_sh">clinvar=ftp://ftp.ensembl.org/pub/rapid-release/species/Homo_sapiens/GCA_009914755.4/ensembl/variation/2022_10/vcf/2024_07/clinvar_20240624_GCA_009914755.4.vcf.gz
tabix -h $clinvar 1 | head -n 100 > test.vcf</pre>

  <h5 id="cache">Ensembl VEP cache</h5>

  <p>The <a href="vep_cache.html#cache">cache</a> is a downloadable archive containing all transcript models for an assembly; it may also contain regulatory features and variant data.</p>

  <p>Let's start by downloading and extracting the cache to the default Ensembl VEP directory (available for each annotation by clicking in <b>VEP cache</b> in the <a href="https://projects.ensembl.org/hprc/" rel="external">Ensembl HPRC data page</a>). In the case of T2T-CHM13:</p>

  <pre class="code sh_sh">
cd $HOME/.vep
curl -O https://ftp.ensembl.org/pub/rapid-release/species/Homo_sapiens/GCA_009914755.4/ensembl/variation/2022_10/indexed_vep_cache/Homo_sapiens-GCA_009914755.4-2022_10.tar.gz
tar xzf Homo_sapiens-GCA_009914755.4-2022_10.tar.gz</pre>

  <p>This will create the folder <kbd>homo_sapiens_gca009914755v4/107_T2T-CHM13v2.0</kbd> with the gene data required to run Ensembl VEP. The name of this folder contains relevant information when running Ensembl VEP:</p>

  <ul>
    <li> Species: <kbd>homo_sapiens_gca009914755v4</kbd> </li>
    <li> Cache version: <kbd>107</kbd> </li>
    <li> Assembly: <kbd>T2T-CHM13v2.0</kbd> </li>
  </ul>

  <p> As well as molecular consequence predictions, many gene/transcript-based <a href="vep_options.html">options</a> are supported for HPRC assemblies: </p>

  <pre class="code sh_sh">vep -i test.vcf --offline \
    --species homo_sapiens_gca009914755v4 \
    --cache_version 107 \
    --fasta Homo_sapiens-GCA_009914755.4-softmasked.fa.gz \
    --domains --symbol --canonical --protein --biotype --uniprot --variant_class</pre>

  <p> We don't have other annotations, such as RefSeq transcripts or variant information in the cache. </p>

  <p> To run Ensembl VEP with the downloaded cache in offline mode, please specify the species (which here includes assembly name) and cache version:</p>

  <pre class="code sh_sh">vep -i test.vcf --offline --species homo_sapiens_gca009914755v4 --cache_version 107</pre>

  <h5 id="fasta">FASTA sequence</h5>

  <p>When using the cache, supplying the reference genomic sequence in a FASTA file is optional, but is required to enable the following options:</p>

  <ul>
    <li> Create HGVS notations (<a href="vep_options.html#opt_hgvs">--hgvs</a> and <a href="vep_options.html#opt_hgvsg">--hgvsg</a>) </li>
    <li> Check the reference sequence given in input data (<a href="vep_options.html#opt_check_ref">--check_ref</a>) </li>
  </ul>

  <p> Genomic FASTA files can be found in <a href="https://projects.ensembl.org/hprc/" rel="external">Ensembl HPRC data page</a> <b>> FTP dumps > ensembl > genome</b>. FASTA files need to be either uncompressed or compressed with <b>bgzip</b> (recommended) to be compatible with Ensembl VEP. For instance, to download a compressed FASTA file, uncompress it and then re-compress it with bgzip: </p>

  <pre class="code sh_sh">curl -O https://ftp.ensembl.org/pub/rapid-release/species/Homo_sapiens/GCA_009914755.4/ensembl/genome/Homo_sapiens-GCA_009914755.4-softmasked.fa.gz
gzip -d Homo_sapiens-GCA_009914755.4-softmasked.fa.gz
bgzip Homo_sapiens-GCA_009914755.4-softmasked.fa.gz</pre>

  <p> Afterwards, you can run Ensembl VEP using cache and the <a href="vep_options.html#opt_fasta">--fasta</a> flag: </p>

  <pre class="code sh_sh">vep -i test.vcf --offline \
    --species homo_sapiens_gca009914755v4 \
    --cache_version 107 \
    --fasta Homo_sapiens-GCA_009914755.4-softmasked.fa.gz</pre>

  <p> More information on using FASTA files is available <a href="vep_cache.html#fasta">here</a>. </p>

  <h5 id="gtf">GTF and GFF annotation</h5>

  <p> As an alternative to using cache files, Ensembl VEP can utilise gene information in appropriately indexed GTF or GFF files. GTF and GFF files can be downloaded from the annotation column in the <a href="https://projects.ensembl.org/hprc/" rel="external">Ensembl HPRC data page</a>. The data needs to be re-sorted in chromosomal order, compressed in <b>bgzip</b> and indexed with <b>tabix</b>. We present here the example for a GTF file: </p>

  <pre class="code sh_sh">curl -O https://ftp.ensembl.org/pub/rapid-release/species/Homo_sapiens/GCA_009914755.4/ensembl/geneset/2022_07/Homo_sapiens-GCA_009914755.4-2022_07-genes.gtf.gz
gzip -d Homo_sapiens-GCA_009914755.4-2022_07-genes.gtf.gz
grep -v "#" Homo_sapiens-GCA_009914755.4-2022_07-genes.gtf |\
  sort -k1,1 -k4,4n -k5,5n -t$'\t' |\
  bgzip -c > Homo_sapiens-GCA_009914755.4-2022_07-genes.gtf.gz
tabix Homo_sapiens-GCA_009914755.4-2022_07-genes.gtf.gz</pre>

  <p> FASTA files are <b>always</b> required when running HPRC data with GTF annotation, as the transcript sequences are not available in the GFF files. </p>

  <p> Afterwards, you can run Ensembl VEP using the GTF and FASTA files: </p>

  <pre class="code sh_sh">vep -i test.vcf \
      --gtf Homo_sapiens-GCA_009914755.4-2022_07-genes.gtf.gz \
      --fasta Homo_sapiens-GCA_009914755.4-softmasked.fa.gz</pre>

  <p> Check <a href="vep_cache.html#gff">here</a> for more information on using gene annotations in GTF and GFF files. </p>

  <h3 id="PolyPhen_SIFT"> Missense deleteriousness predictions </h3>

  <p> Although PolyPhen-2 and SIFT scores are not directly available for alternative assemblies by using <a href="vep_options.html#opt_polyphen">--polyphen</a> and <a href="vep_options.html#opt_sift">--sift</a>, they can be retrieved via the <a href="vep_plugins.html#PolyPhen_SIFT">PolyPhen_SIFT plugin</a>. </p>

  <p> Using our <a href="https://github.com/Ensembl/ensembl-variation/tree/main/nextflow/ProteinFunction" rel="external">ProteinFunction pipeline</a>, we ran <b>PolyPhen-2 2.2.3</b> and <b>SIFT 6.2.1</b> on the proteome sequences for GRCh38 and all HPRC assemblies (the protein FASTA files indicated in <a href="https://projects.ensembl.org/hprc/" rel="external">Ensembl HPRC data page</a>) and stored their results in a single SQLite file: <a href="https://ftp.ensembl.org/pub/current_variation/pangenomes/Human/homo_sapiens_pangenome_PolyPhen_SIFT_20240502.db" rel="external"><kbd>homo_sapiens_pangenome_PolyPhen_SIFT_20240502.db</kbd></a>.

  <p> Pre-computed scores and predictions can be retrieved by downloading this file and using the <b>PolyPhen_SIFT plugin</b>: </p>

  <pre class="code sh_sh">curl -O https://ftp.ensembl.org/pub/current_variation/pangenomes/Human/homo_sapiens_pangenome_PolyPhen_SIFT_20240502.db
vep -i test.vcf --offline \
    --species homo_sapiens_gca009914755v4 \
    --cache_version 107 \
    --fasta Homo_sapiens-GCA_009914755.4-softmasked.fa.gz \
    --plugin PolyPhen_SIFT,db=human_pangenomes.PolyPhen_SIFT.db</pre>

  <h3> Matched variant annotations (ClinVar, gnomAD and dbSNP) </h3>

  <p> We don't have variant data in the caches for the HPRC assemblies, but it can be integrated using the <a href="vep_custom.html">--custom</a> option with data files using the same assembly coordinates. We have lifted-over some key datasets, including ClinVar and gnomAD to the HPRC assemblies (downloadable from the VCF column in <a href="https://projects.ensembl.org/hprc/" rel="external">Ensembl HPRC data page</a>). </p>

  <pre class="code sh_sh"># Download ClinVar data and respective index (TBI)
curl -O https://ftp.ensembl.org/pub/rapid-release/species/Homo_sapiens/GCA_009914755.4/ensembl/variation/2022_10/vcf/2024_07/clinvar_20240624_GCA_009914755.4.vcf.gz
curl -O https://ftp.ensembl.org/pub/rapid-release/species/Homo_sapiens/GCA_009914755.4/ensembl/variation/2022_10/vcf/2024_07/clinvar_20240624_GCA_009914755.4.vcf.gz.tbi

# Run Ensembl VEP with ClinVar data
vep -i test.vcf --offline \
    --species homo_sapiens_gca009914755v4 --cache_version 107 \
    --fasta Homo_sapiens-GCA_009914755.4-softmasked.fa.gz \
    --custom file=clinvar_20240624_GCA_009914755.4.vcf.gz,short_name=ClinVar,format=vcf,type=exact,coords=0,fields=CLNSIG%CLNREVSTAT%CLNDN</pre>

  <h3> Additional annotations </h3>

  <p> Ensembl VEP plugins are a simple way to add new functionality to your analysis. Many require data that is only available for GRCh37 or GRCh38, but others, for example those based on gene attributes or on the fly analysis are compatible with the HGRC assemblies. </p>

  <p> Here follows Ensembl VEP plugins that are easily compatible with alternative human assemblies: </p>

  <style>
    tr:nth-child(odd) {background-color: #f0f0f0;}
    td {
      line-height: 1.25em;
      word-wrap: break-word;
    }
    th { word-wrap: break-word; }
    .info {width: inherit !important;}
  </style>

  <table class="ss" style="table-layout: fixed"><thead>
    <tr>
      <th>Plugin</th>
      <th>Description</th>
      <th>Plugin data</th>
      <th>Usage example</th>
    </tr></thead>
  <tbody>
    <tr>
      <td><a href="https://github.com/Ensembl/VEP_plugins/blob/main/Blosum62.pm"  rel="external">Blosum62</a></td>
      <td>Looks up the BLOSUM 62 substitution matrix score for the reference and alternative amino acids predicted for a missense mutation.</td>
      <td></td>
      <td><kbd>--plugin Blosum62</kbd></td>
    </tr>
    <tr>
      <td><a href="https://github.com/Ensembl/VEP_plugins/blob/main/DosageSensitivity.pm" rel="external">DosageSensitivity</a></td>
      <td>Retrieves haploinsufficiency and triplosensitivity probability scores<br>for affected genes (<a href="https://doi.org/10.1016/j.cell.2022.06.036">Collins <i>et al.</i>, 2022</a>).</td>
      <td><a href="https://zenodo.org/record/6347673/files/Collins_rCNV_2022.dosage_sensitivity_scores.tsv.gz"><kbd>Collins_rCNV_2022.dosage_sensitivity_scores.tsv.gz</kbd></a></td>
      <td><kbd>--plugin DosageSensitivity,file=Collins_rCNV_2022.dosage_sensitivity_scores.tsv.gz</kbd></td>
    </tr>
    <tr>
      <td><a href="https://github.com/Ensembl/VEP_plugins/blob/main/Downstream.pm" rel="external">Downstream</a></td>
      <td>Predicts downstream effects of a frameshift variant on the protein sequence of a transcript.</td>
      <td>Requires a FASTA file provided via the <a href="vep_options.html#opt_fasta">--fasta</a> option</td>
      <td><kbd>--plugin Downstream</kbd></td>
    </tr>
    <tr>
      <td><a href="https://github.com/Ensembl/VEP_plugins/blob/main/Draw.pm" rel="external">Draw</a></td>
      <td>Draws pictures of the transcript model showing the variant location.</td>
      <td></td>
      <td><kbd>--plugin Draw</kbd></td>
    </tr>
    <tr>
      <td><a href="https://github.com/Ensembl/VEP_plugins/blob/main/GeneSplicer.pm" rel="external">GeneSplicer</a></td>
      <td>Runs <a href="https://ccb.jhu.edu/software/genesplicer/">GeneSplicer</a> to get splice site predictions.</td>
      <td>Binary and training data for GeneSplicer (<a href="vep_plugins.html#GeneSplicer">plugin instructions</a>)</td>
      <td><kbd>--plugin GeneSplicer,binary=genesplicer/bin/linux/genesplicer,training=genesplicer/human</kbd></td>
    </tr>
    <tr>
      <td><a href="https://github.com/Ensembl/VEP_plugins/blob/main/GO.pm" rel="external">GO</a></td>
      <td>Retrieves Gene Ontology (GO) terms associated with genes (for HGRC assemblies, specifically) using custom GFF annotation containing GO terms.</td>
      <td><a href="https://projects.ensembl.org/hprc/">Ensembl HPRC data page</a> <b>&gt; FTP dumps &gt; ensembl &gt; variation &gt; [date] &gt; gff:</b>
      <ul>
        <li> <kbd>*_GO_plugin.gff.gz</kbd> </li>
        <li> <kbd>*_GO_plugin.gff.gz.tbi</kbd> </li>
      </ul>
      </td>
      <td><kbd>--plugin GO,file=homo_sapiens_gca009914755v4_110_VEP_GO_plugin.gff.gz</kbd></td>
    </tr>
    <tr>
      <td><a href="https://github.com/Ensembl/VEP_plugins/blob/main/HGVSIntronOffset.pm" rel="external">HGVSIntronOffset</a></td>
      <td>Returns HGVS intron start and end offsets. To be used with <a href="https://www.ensembl.org/info/docs/tools/vep/script/vep_options.html#opt_hgvs">--hgvs</a> option.</td>
      <td></td>
      <td><kbd>--plugin HGVSIntronOffset</kbd></td>
    </tr>
    <tr>
      <td><a href="https://github.com/Ensembl/VEP_plugins/blob/main/LoFtool.pm" rel="external">LoFtool</a></td>
      <td>Provides a rank of genic intolerance and consequent susceptibility to disease based on the ratio of Loss-of-function (LoF) to synonymous mutations for each gene.</td>
      <td></td>
      <td><kbd>--plugin LoFtool</kbd></td>
    </tr>
    <tr>
      <td><a href="https://github.com/Ensembl/VEP_plugins/blob/main/MaxEntScan.pm" rel="external">MaxEntScan</a></td>
      <td>Runs <a href="http://hollywood.mit.edu/burgelab/maxent/Xmaxentscan_scoreseq.html">MaxEntScan</a> to get splice site predictions.</td>
      <td>Extracted directory from <a href="http://hollywood.mit.edu/burgelab/maxent/download/fordownload.tar.gz">fordownload.tar.gz</td>
      <td><kbd>--plugin MaxEntScan,/path/to/fordownload</kbd></td>
    </tr>
    <tr>
      <td><a href="https://github.com/Ensembl/VEP_plugins/blob/main/NearestExonJB.pm" rel="external">NearestExonJB</a></td>
      <td>Finds the nearest exon junction boundary to a coding sequence variant.</td>
      <td></td>
      <td><kbd>--plugin NearestExonJB</kbd></td>
    </tr>
    <tr>
      <td><a href="https://github.com/Ensembl/VEP_plugins/blob/main/NMD.pm" rel="external">NMD</a></td>
      <td>Predicts if a variant allows the transcript to escape nonsense-mediated mRNA decay based on certain rules.</td>
      <td></td>
      <td><kbd>--plugin NMD</kbd></td>
    </tr>
    <tr>
      <td><a href="https://github.com/Ensembl/VEP_plugins/blob/main/Phenotypes.pm" rel="external">Phenotypes</a></td>
      <td>Retrieves overlapping phenotype information.</td>
      <td><a href="https://projects.ensembl.org/hprc/">Ensembl HPRC data page</a> <b>&gt; FTP dumps &gt; ensembl &gt; variation &gt; [date] &gt; gff:</b>
        <ul>
          <li> <kbd>*_phenotypes_plugin.gvf.gz</kbd> </li>
          <li> <kbd>*_phenotypes_plugin.gvf.gz.tbi</kbd> </li>
        </ul>
      </td>
      <td><kbd>--plugin Phenotypes,file=homo_sapiens_gca009914755v4_110_VEP_phenotypes_plugin.gvf.gz</kbd></td>
    </tr>
    <tr>
      <td><a href="https://github.com/Ensembl/VEP_plugins/blob/main/pLI.pm" rel="external">pLI</a></td>
      <td>Adds the probability of a gene being loss-of-function intolerant (pLI).</td>
      <td></td>
      <td><kbd>--plugin pLI</kbd></td>
    </tr>
    <tr>
      <td><a href="https://github.com/Ensembl/VEP_plugins/blob/main/PolyPhen_SIFT.pm" rel="external">PolyPhen_SIFT</a></td>
      <td>Retrieves PolyPhen and SIFT predictions from a SQLite database.</td>
      <td><a href="https://ftp.ensembl.org/pub/current_variation/pangenomes/Human/homo_sapiens_pangenome_PolyPhen_SIFT_20240502.db"><kbd>homo_sapiens_pangenome_PolyPhen_SIFT_20240502.db</kbd></a></td>
      <td><kbd>--plugin PolyPhen_SIFT,db=homo_sapiens_pangenome_PolyPhen_SIFT_20240502.db</kbd></td>
    </tr>
    <tr>
      <td><a href="https://github.com/Ensembl/VEP_plugins/blob/main/ProteinSeqs.pm" rel="external">ProteinSeqs</a></td>
      <td>Writes two files with the reference and mutated protein sequences of any proteins found with non-synonymous mutations in the input file.</td>
      <td></td>
      <td><kbd>--plugin ProteinSeqs</kbd></td>
    </tr>
    <tr>
      <td><a href="https://github.com/Ensembl/VEP_plugins/blob/main/SingleLetterAA.pm" rel="external">SingleLetterAA</a></td>
      <td>Returns HGVSp string with single amino acid letter codes.</td>
      <td></td>
      <td><kbd>--plugin SingleLetterAA</kbd></td>
    </tr>
    <tr>
      <td><a href="https://github.com/Ensembl/VEP_plugins/blob/main/SpliceRegion.pm" rel="external">SpliceRegion</a></td>
      <td>Provides more granular predictions of splicing effects.</td>
      <td></td>
      <td><kbd>--plugin SpliceRegion</kbd></td>
    </tr>
    <tr>
      <td><a href="https://github.com/Ensembl/VEP_plugins/blob/main/SubsetVCF.pm" rel="external">SubsetVCF</a></td>
      <td>Retrieves overlapping records from a given VCF file.</td>
      <td>A VCF file</td>
      <td><kbd>--plugin SubsetVCF,file=file.vcf.gz,name=myvfc</kbd></td>
    </tr>
    <tr>
      <td><a href="https://github.com/Ensembl/VEP_plugins/blob/main/TranscriptAnnotator.pm" rel="external">TranscriptAnnotator</a></td>
      <td>Annotates variant-transcript pairs based on a given file.</td>
      <td>Tab-separated annotation file (<a href="vep_plugins.html#TranscriptAnnotator">plugin instructions</a>)</td>
      <td><kbd>--plugin TranscriptAnnotator,file=annotation.txt.gz</kbd></td>
    </tr>
    <tr>
      <td><a href="https://github.com/Ensembl/VEP_plugins/blob/main/TSSDistance.pm" rel="external">TSSDistance</a></td>
      <td>Calculates the distance from the transcription start site for upstream variants.</td>
      <td></td>
      <td><kbd>--plugin TSSDistance</kbd></td>
    </tr>
  </tbody></table>

  <hr/>

  <h2 id="citations">Citations and Ensembl VEP users</h2>
  
  <p> Ensembl VEP is used by many organisations and projects: </p>
  
  <ul>
    <li> Ensembl VEP forms a part of <a href="http://variantstudio.software.illumina.com/" rel="external">Illumina's VariantStudio</a> software </li>
    <li> <a href="http://gemini.readthedocs.org/" rel="external">Gemini</a> is a framework for exploring genome variation that uses Ensembl VEP </li>
    <li> The <a href="https://www.deciphergenomics.org/" rel="external">DECIPHER project</a> uses Ensembl VEP to aid variant interpretation</li>
  </ul>
  
  <p> Other citations and use cases: </p>
  
  <ul>
    <li> <a href="http://bib.oxfordjournals.org/content/early/2014/03/12/bib.bbu008.full" rel="external">VAX</a> is a suite of plugins that expand Ensembl VEP functionality</li>
    <li> <a href="https://academic.oup.com/bioinformatics/article/30/23/3408/207567" rel="external">pViz</a> is a visualisation tool for Ensembl VEP results files </li>
    <li> <a href="https://genomemedicine.biomedcentral.com/articles/10.1186/gm543" rel="external">McCarthy <i>et al</i></a> compares Ensembl VEP to AnnoVar </li>
    <li> <a href="http://bib.oxfordjournals.org/content/early/2013/01/21/bib.bbs086.full" rel="external">Pabinger <i>et al</i></a> reviews variant analysis software, including Ensembl VEP</li>
    <li> Ensembl VEP is used to provide annotation for the <a href="http://gnomad.broadinstitute.org/" rel="external">gnomAD</a> project</li>
  </ul>
</div>

<div style="clear:both"><h1 style="page-break-after: always">&nbsp</h1></div><div>
  <div style="float:right;"><img src="/img/vep_logo.png"/></div>
	
  <h1 id="top"><span style="color:#006;padding-right:15px">Ensembl Variant Effect Predictor</span><span style="color:#666"><img src="/i/16/help.png"/> FAQ</span></h1>
  <hr/>
    
    <p> For any questions not covered here, please send an email to the Ensembl
    <a href="/info/about/contact/index.html">developer's mailing list</a>
    (public) or contact the
    <a href="/Help/Contact" class="popup">Ensembl Helpdesk</a> (private).
    Also you can report issues through our (public) Github repositories. For general Ensembl VEP issues you should use 
    <a href="https://github.com/Ensembl/ensembl-vep/issues">ensembl-vep</a> repository and for specific plugins you should use 
    <a href="https://github.com/Ensembl/VEP_plugins/issues">VEP_plugins</a> repository.
    </p>

<h2 id="general">General questions</h2>

<h3 id="indel-vcf">Q: Why has my insertion/deletion variant encoded in VCF
disappeared from the output? </h3>

<p> Ensembl treats unbalanced variants differently to VCF - your variant
hasn't disappeared, it may have just changed slightly! You can solve this by
giving your variants a unique identifier in the third column of the VCF file.
See <a href="vep_formats.html#vcf">here</a> for a full discussion.</p>

<p>&nbsp;</p>

<h3 id="no-co-located-vars"> Q: Why don't I see any co-located variants when
using species X? </h3>

<p> Not all species have variants and not all species that do are in the Ensembl variation resource
- see <a href="/info/genome/variation/species/species_data_types.html#source">this
  document</a> for details.
The <a href="script/vep_options.html#opt_custom"> custom </a>option can be used in the commandline interface to include more variant sets</p>
</p>

<p>&nbsp;</p>

<h3 id="multi-vars-mapped"> Q: Why do I see multiple known variants mapped to my input variant? </h3>

<p> Ensembl VEP compares your input to known variants from the Ensembl variation database.
In some cases one input variant can match multiple known variants:
<ul>
  <li>Germline variants from dbSNP and somatic mutations from COSMIC may be found at the same locus</li>
  <li>Some sources, e.g. HGMD, do not provide public access to allele-specific data, so an HGMD variant with unknown alleles may colocate with one from dbSNP with known alleles</li>
  <li>Multiple alternate alleles from your input may match different variants as they are described in dbSNP</li>
</ul>

See <a href="script/vep_other.html#colocated">here</a> for a full discussion.</p>

<p>&nbsp;</p>

<h3 id="no-freq"> Q: Ensembl VEP is not assigning a frequency to my input variant - why? </h3>

<p> Ensembl VEP's cache contains frequency data only for specific studies. See <a href="script/vep_other.html#colocated">here</a> for a full discussion. 
The <a href="script/vep_options.html#opt_custom"> custom </a>option can be used in the commandline interface to include more frequency sets</p>

<p>&nbsp;</p>

<h3 id="many-lines"> Q: Why do I see so many lines of output for each variant in my input? </h3>

<p> While it would be convenient to have a simple, one word answer to the
question "What is the consequence of this variant?", in reality biology is not
 this simple! Many genes have more than one transcript, so Ensembl VEP
provides a prediction for each transcript that a variant overlaps. Ensembl VEP
has options to help select results according to your requirements; the <a
href="script/vep_options.html#opt_canonical">--canonical</a> and <a
href="script/vep_options.html#mane">--mane</a> options indicate which
transcripts are canonical and belong to the human MANE set respectively, while <a
href="script/vep_options.html#opt_pick">--pick</a>, <a
href="script/vep_options.html#opt_per_gene">--per_gene</a>, <a
href="script/vep_options.html#opt_summary">--summary</a> and <a
href="script/vep_options.html#opt_most_severe">--most_severe</a> allow you to
give a more summary level assessment per variant.</p>

<p> Furthermore, several "compound" consequences are also possible - if, for
example, a variant falls in the final few bases of an exon, it may be considered
to affect a splicing site, in addition to possibly affecting the coding
sequence. </p>


<p>&nbsp;</p>

<h3 id="reduce-mem"> Q: How do I reduce Ensembl VEP's memory requirement?</h3>

<p> There are a number of ways to do this-
<ol>
  <li>Ensure your input file is sorted by location. This can greatly reduce memory
      requirements and runtime </li>
  <li>Consider reducing the buffer size. This reduces the number of variants annotated
      together in a batch and can be modified in both command line and web interfaces.
      Reducing buffer size may increase run time.</li>
  <li>Ensure you are only using the options you need, rather than --everything.
      Some data-rich options, such as regulatory annotation have an impact on memory use</li>
</ol>
</p>

<p>&nbsp;</p>


<h3 id="cite-vep"> Q: How to cite Ensembl VEP?</h3>

<p> If you use Ensembl VEP, please cite our
<a href="/info/docs/tools/vep/index.html#publication">latest publication</a>
to continue to support Ensembl VEP development. </p>

<hr/>

<h2 id="web">Ensembl VEP web interface questions</h2>

<h3 id="web-vep"> Q: How do I access the web version of the Ensembl Variant
Effect Predictor? </h3>

<p> You can find the Ensembl VEP web tool on the <a href="/info/docs/tools/index.html">Tools</a> page.</p>

<p>&nbsp;</p>


<h3 id="different-output"> Q: Why is the output I get for my input file
different when I use the Ensembl VEP web and command line interfaces? </h3>

<p> Ensure that you are passing equivalent arguments to the command line tool that you
are using in the web interface. If you are sure this is still a problem, please
report it on the <a href="http://lists.ensembl.org/mailman/listinfo/dev" rel="external">ensembl-dev</a> mailing list.</p>

<p>&nbsp;</p>


<h3 id="web-vep-tutorial"> Q: Is there a tutorial for the web tool?</h3>

<p>Yes, see our latest tutorial <a href="https://onlinelibrary.wiley.com/doi/10.1002/humu.24298" rel="external">
Annotating and prioritizing genomic variants using the Ensembl Variant Effect Predictor — A tutorial</a>
for more information on using the Ensembl VEP web interface.</p>
<hr/>

<h2 id="script">Ensembl VEP command line tool questions</h2>

<h3 id="faster-vep"> Q: How can I make Ensembl VEP run faster? </h3>

<p> There are a number of factors that influence how fast Ensembl VEP runs. Have a look at our <a href="script/vep_other.html#faster">handy guide</a> for tips on improving runtime. </p>

<p>&nbsp;</p>

<h3 id="different-input-var"> Q: Why am I not seeing the same variant from my input in the output? </h3>
<p> Since the Ensembl 110 release, Ensembl VEP by default will minimise the input allele for display in the output. To see the exact allele representation you provided, use the<a
href="script/vep_options.html#opt_uploaded_allele"> --uploaded_allele</a> option. </p>

<p>&nbsp;</p>

<h3 id="hgvs-n"> Q: Why do I see "N" as the reference allele in my HGVS strings? </h3>
<h3 id="sequence.pm"> Q: Why do I get errors related with Sequence.pm?</h3>

<pre class="code sh_sh">substr outside of string at /nfs/users/nfs_w/wm2/Perl/ensembl-variation/modules/Bio/EnsEMBL/Variation/Utils/Sequence.pm line 511.
Use of uninitialized value $ref_allele in string eq at /nfs/users/nfs_w/wm2/Perl/ensembl-variation/modules/Bio/EnsEMBL/Variation/Utils/Sequence.pm line 514.
Use of uninitialized value in concatenation (.) or string at /nfs/users/nfs_w/wm2/Perl/ensembl-variation/modules/Bio/EnsEMBL/Variation/Utils/Sequence.pm line 643.</pre>

<p> Both of these error types are usually seen when using a <a
href="script/vep_cache.html#fasta">FASTA file</a> for retrieving sequence.
There are a couple of steps you can take to try to remedy them:</p>

<ol>
  <li>
    The index alongside the FASTA can become corrupted. Delete
    [fastafile].index and re-run Ensembl VEP to regenerate it. By default
    this file is located in your $HOME/.vep/[species]/[version]_[assembly]
    directory.
  </li>

  <li>
    The FASTA file itself may have been corrupted during download;
    delete the fasta file and the index and re-download (you can use the
    <a href="script/vep_download.html#install">Ensembl VEP installer</a> to do this).
  </li>

  <li>
    Older versions of BioPerl (1.2.3 in particular is known to have this)
    cannot properly index large FASTA files. Make sure you are using a later
    (>=1.6) version of BioPerl. The <a href="script/vep_download.html#install">Ensembl VEP
    installer</a> installs 1.6.924 for you.
  </li>
</ol>

<p> If you still see problems after taking these steps, or if you were not
using a FASTA file in the first place, please <a href="#top">contact us</a>. </p>

<p>&nbsp;</p>

<h3 id="chr-not-found"> Q: Why are chromosomes not found in annotation sources or synonyms?</h3>

<pre class="code sh_sh">WARNING: Chromosome 21 not found in annotation sources or synonyms on line 160</pre>

<p> This can occur if the chromosome names differ between your input variant and any annotation source that you are using (cache, database, GFF/GTF file, FASTA file, custom annotation file). To circumvent this you may provide a <a href="script/vep_options.html#opt_synonyms">synonyms file</a>. A synonym file is included in Ensembl VEP's cache files, so if you have one of these for your species you can use it as follows:</p>

<pre class="code sh_sh">./vep -i input.vcf -cache -synonyms ~/.vep/homo_sapiens/[[SPECIESDEFS::ENSEMBL_VERSION]]_GRCh38/chr_synonyms.txt</pre>

<p> The file consists of lines containing pairs of tab-separated synonyms. Order is not important as synonyms can be used in both "directions". </p>

<p>&nbsp;</p>

<h3 id="feature_type_warnings"> Q: Why do I get feature_type warnings from my GFF/GTF file?</h3>

<pre class="code sh_sh sh_sourceCode"><span class="sh_variable">WARNING: </span>Ignoring 'five_prime_utr' feature_type from Homo_sapiens.GRCh38.111.gtf.gz GFF/GTF file. This feature_type is not supported in Ensembl VEP.</span></pre>

<p> This can occur if you are using GFF/GTF file and the file contains a type that is not supported by Ensembl VEP. Those lines are simply ignored. However, in cases where the transcript model is incomplete the full model may be ignored.</p>
  
<p>Please try to use supported feature types as mentioned <a href="https://www.ensembl.org/info/docs/tools/vep/script/vep_cache.html#gff">here</a></p>

<p>&nbsp;</p>


<h3 id="gnomad-frequences"> Q: Can I get gnomAD exomes and genomes frequencies in Ensembl VEP? </h3>

<p> Yes, see <a href="script/vep_example.html#gnomad">this guide</a>. </p>

<p>&nbsp;</p>

<h3 id="no-server-connection"> Q: Why do I have issues connecting to Ensembl databases? </h3>

<pre class="code sh_sh">
Could not connect to database homo_sapiens_core_63_37 as user anonymous using [DBI:mysql:database=homo_sapiens_core_63_37;host=ensembldb.ensembl.org;port=5306] as a locator:
Unknown MySQL server host 'ensembldb.ensembl.org' (2) at $HOME/src/ensembl/modules/Bio/EnsEMBL/DBSQL/DBConnection.pm line 290.

-------------------- EXCEPTION --------------------
MSG: Could not connect to database homo_sapiens_core_63_37 as user anonymous using [DBI:mysql:database=homo_sapiens_core_63_37;host=ensembldb.ensembl.org;port=5306] as a locator:
Unknown MySQL server host 'ensembldb.ensembl.org' (2)
</pre>

<p> If you select the database option rather than using a cache
Ensembl VEP will try to connect to the public MySQL
server at ensembldb.ensembl.org. Occasionally the server may break connection
with your process, which causes this error. This can happen when the server is
busy, or due to various network issues. Consider using the 
<a href="script/vep_cache.html#cache">caching system</a>. 
Using a cache and fasta file is the most effcient way to run Ensembl VEP
</p>

<p>&nbsp;</p>

<h3 id="vep-windows"> Q: Can I use Ensembl VEP on Windows? </h3>

<p> Yes - see the <a href="script/vep_download.html#windows">documentation</a> for a
few different ways to get the Ensembl VEP running on Windows. </p>

<p>&nbsp;</p>

<h3 id="vep-custom-assemblies"> Q: Can I use Ensembl VEP with species and assemblies which are not available in Ensembl? </h3>

<p> Yes - you can run Ensembl VEP on any species you have data for by providing a custom
gene annotation in <a href="script/vep_cache.html#gff">GFF/GTF</a> and genome sequence in
<a href="script/vep_cache.html#fasta">FASTA</a> file, like so:
<pre class="code sh_sh">./vep -i input.vcf --gff data.gff.gz --fasta genome.fa.gz</pre>
</p>

<p>&nbsp;</p>

<h3 id="vep-custom-assemblies"> Q: Can I use Ensembl VEP with T2T-CHM13 and other human assemblies? </h3>

<p> Yes - you can run Ensembl VEP using <a href="https://humanpangenome.org/" rel="external">Human Pangenome Reference Consortium (HPRC)</a> data by following the instructions on how to <a href="script/vep_example.html#pangenomes">use Ensembl VEP with HPRC assemblies</a>.

</p>

<p>&nbsp;</p>

<h3 id="download-sift-polyphen"> Q: Can I download all of the SIFT and/or PolyPhen predictions? </h3>

<p> The Ensembl Variation database and the human Ensembl VEP cache file contain
precalculated SIFT and PolyPhen-2 predictions for every possible amino acid change
in every translated protein product in Ensembl. Since these data are huge, we
store them in a <a
href="/info/genome/variation/prediction/protein_function.html#nsSNP_data_format">compressed format</a>. </p>

<p> There are different approaches to download SIFT/PolyPhen-2 data: </p>

<ul>
  <li> Using the <a href="script/vep_plugins.html#PolyPhen_SIFT">PolyPhen_SIFT plugin</a>: </li>
  <ul>
    <li> For any species with predictions in our Ensembl databases, the plugin is able to download the predictions data into a local SQLite database for offline use. PolyPhen predictions are only available for human data. </li>
    <li> We also provide a downloadble SQLite database containing PolyPhen/SIFT predictions based on <a href="https://humanpangenome.org" rel="external">Human Pangenome Reference Consortium (HPRC)</a> and GRCh38 assemblies. For more information, refer to <a href="script/vep_example.html#PolyPhen_SIFT">Missense deleteriousness predictions</a> in HPRC assemblies.</li>
  </ul>
  <li> Using our <b>Perl API</b>: </li>
  <ul>
    <li> Fetch a <a
href="/info/docs/Doxygen/variation-api/classBio_1_1EnsEMBL_1_1Variation_1_1ProteinFunctionPredictionMatrix.html"><kbd>ProteinFunctionPredictionMatrix</kbd></a> for your protein of interest and then call its
<a href="/info/docs/Doxygen/variation-api/classBio_1_1EnsEMBL_1_1Variation_1_1ProteinFunctionPredictionMatrix.html#af6346bcc5c4c28281e3c9a009ffcb15d"><kbd>get_prediction()</kbd></a> method to get the score for a particular position and amino
acid, looping over all possible amino acids for your position. </li>
    <li> You would need to work out which peptide position your codon maps to, but
there are methods in the <a
href="/info/docs/Doxygen/variation-api/classBio_1_1EnsEMBL_1_1Variation_1_1TranscriptVariation.html"><kbd>TranscriptVariation</kbd></a>
class that should help you (probably <a href="/info/docs/Doxygen/variation-api/classBio_1_1EnsEMBL_1_1Variation_1_1BaseTranscriptVariation.html#acfc84f951919f445624545afbd5da0a8"><kbd>translation_start()</kbd></a> and <a href="/info/docs/Doxygen/variation-api/classBio_1_1EnsEMBL_1_1Variation_1_1BaseTranscriptVariation.html#ae32d5894a6466c7bf0fa02e6d200d038"><kbd>translation_end()</kbd></a>).</li>
  </ul>
</ul>

</div>

<div style="clear:both"><h1 style="page-break-after: always">&nbsp</h1></div></body></html>
