<html>
<head>
<title>Other information</title>
<meta name="order" content="9" />
</head>
<body>
<div>
  
  <div style="float:right"><img src="/img/vep_logo.png"/></div>
  
  <h1 id="top"><span style="color:#006">Variant Effect Predictor</span> <img src="/i/16/info.png"> <span style="color:#666">Other information</span></h1>
  <hr/>
  
  <h2 id="faster">Getting VEP to run faster</h2>
  
  <p> Set up correctly, VEP is capable of processing around 3
  million variants in 30 minutes. There are a number of steps you can take to
  make sure your VEP installation is running as fast as possible: </p>
  
  <ol>
    <li>
      <p> Make sure you have the <a href="vep_download.html">latest version</a> of
      VEP and the Ensembl API. We regularly introduce optimisations, alongside the
      new features and bug fixes of a typical new release. </p>
    </li>
    
    <li>
      <p> Download a <a href="vep_cache.html">cache file</a> for your species. If
      you are using <a href="vep_options.html#opt_database">--database</a>, you
      should consider using <a href="vep_options.html#opt_database">--cache</a> or
      <a href="vep_options.html#opt_database">--offline</a> instead. Any time
      VEP has to access data from the database (even if you have a local copy),
      it will be slower than accessing data in the cache on your local file system. </p>
      
      <p> Enabling <a href="vep_cache.html#limitations">certain flags</a>
      forces VEP to access the database, and the script will warn you at
      startup that it will do this with e.g.:</p>
      
      <pre class="code">2011-06-16 16:24:51 - INFO: Database will be accessed when using --check_svs</pre>
      
      <p> Consider carefully whether you need to use these flags in your analysis.</p>
    </li>
    
    <li>
      <p> If you use <a href="vep_options.html#opt_check_existing">--check_existing</a>
      or any flags that invoke it (e.g. <a href="vep_options.html#opt_af">--af</a>,
      <a href="vep_options.html#opt_af_1kg">--af_1kg</a>, <a
      href="vep_options.html#opt_filter_common">--filter_common</a>, <a
      href="vep_options.html#opt_everything">--everything</a>), <a
      href="vep_cache.html#convert">tabix-convert</a> your cache file. Checking for known
      variants using a converted cache is >100% faster than using the default format. </p>
    </li>
    
    <li>
      <p> Download a <a href="vep_cache.html#fasta">FASTA file</a> if you use
      <a href="vep_options.html#opt_hgvs">--hgvs</a> or <a
      href="vep_options.html#opt_check_ref">--check_ref</a>. Again, this will prevent
      VEP accessing the database unnecessarily (in this case to retrieve genomic
      sequence). </p>
    </li>
    
    <li id="forking">
      <p> Using forking enables VEP to run multiple parallel "threads", with
      each thread processing a subset of your input. Most modern computers have
      more than one processor core, so running VEP with forking enabled can
      give huge speed increases (3-4x faster in most cases). Even computers with a
      single core will see speed benefits due to overheads associated with using
      object-oriented code in Perl. </p>
  
      <p> To use forking, you must choose a number of forks to use with the <a
      href="vep_options#opt_fork">--fork</a> flag. Most users should use 4 forks: </p>
  
      <pre class=code>./vep -i my_input.vcf -fork 4 -offline</pre>
  
      <p> but depending on various factors specific to your setup you may see
      faster performance with fewer or more forks. </p>
      
      <p> VEP users writing <a href="vep_plugins.html">plugins</a>
      should be aware that while the VEP code attempts to preserve the state of
      any plugin-specific cached data between separate forks, there may be
      situations where data is lost. If you find this is the case, you should
      disable forking in the new() method of your plugin by deleting the "fork"
      key from the $config hash. </p>
    </li>
    
    <li>
      <p> Make sure your cache and FASTA files are stored on the fastest file system
      or disk you have available. If you have a lot of memory in your machine, you can
      even pre-copy the files to memory using <a
      href="https://www.howtoforge.com/storing-files-directories-in-memory-with-tmpfs"
      rel="external">tmpfs</a>. </p>
    </li>
    
    <li>
      <p> Consider if you need to generate HGVS notations (<a
      href="vep_options.html#opt_hgvs">--hgvs</a>); this is a complex annotation step
      that can add ~50-80% to your runtime. Note also that --hgvs is switched on by
      <a href="vep_options.html#opt_everything">--everything</a>.
    </li>
    
    <li>
      <p> Install the <a
      href="http://search.cpan.org/~benbooth/Set-IntervalTree/lib/Set/IntervalTree.pm"
      rel="external">Set::Interval</a> tree perl package. This package speeds up
      VEP's internals by changing how overlaps between variants and transcript components
      are calculated. </p>
    </li>
    
    <li>
      <p> Install the <a href="https://github.com/Ensembl/ensembl-xs" rel="external">
      Ensembl::XS</a> package. This contains compiled versions of certain key
      subroutines used in VEP that will run faster than the default native Perl
      equivalents. Using this should improve runtime by 5-10%. </p>
    </li>
    
    <li>
      <p> Add the <a href="vep_options.html#opt_no_stats">--no_stats</a> flag.
      Calculating statistics adds some runtime to VEP and most users will not
      need them. </p>
    </li>
    
    <li>
      <p> VEP is optimised to run on input files that are sorted in
      chromosomal order. Unsorted files will still work, albeit more slowly. </p>
    </li>
    
    <li>
      <p> For very large files (for example those from whole-genome sequencing),
      VEP process can be easily parallelised by dividing your file into chunks
      (e.g. by chromosome). VEP will also work with tabix-indexed, bgzipped
      VCF files, and so the tabix utility could be used to divide the input file:
      </p>
  
      <pre class="code"> tabix -h variants.vcf.gz 12:1000000-20000000 | ./vep -cache -vcf </pre>
    </li>
  </ol>
  

  
  
  <hr/>
  <h2 id="assembly">Species with multiple assemblies</h2>
  
  <p> With the arrival of GRCh38, Ensembl now supports two different assembly
  versions for the human genome while users transition from GRCh37. We provide a
  VEP cache download on the latest software version
  ([[SPECIESDEFS::ENSEMBL_VERSION]]) for both assembly versions. </p>
  
  <p> The <a href="vep_download.html#installer">VEP installer</a> will install
  and set up the correct cache and FASTA file for your assembly of interest. If
  using the --AUTO functionality to install without prompts, remember to add the
  assembly version required using e.g. "--ASSEMBLY GRCh37". It is also possible
  to have concurrent installations of caches from both assemblies; just use the
  <a href="vep_options.html#opt_assembly">--assembly</a> to select the correct
  one when you run VEP. </p>
  
  <p> Once you have installed the relevant cache and FASTA file, you are then
  able to use VEP as normal. For those using GRCh37 and requiring database
  access in addition to the cache (for example, to look up variant identifiers
  using <a href="vep_options.html#opt_format">--format id</a>, see <a
  href="vep_cache.html#limitations">cache limitations</a>), the script will warn
  you that you must change the database port in order to connect to the correct
  database: </p>
  
  <pre class="code">ERROR: Cache assembly version (GRCh37) and database or selected assembly version (GRCh38) do not match

If using human GRCh37 add "--port 3337" to use the GRCh37 database, or --offline to avoid database connection entirely</pre>
  
  <p> For users looking to move their data between assemblies, Ensembl provides
  an assembly converter tool - if you've downloaded VEP, then you have it
  already! The script is found in the ensembl-tools/scripts/assembly_converter
  folder. There is also an <a href="/info/docs/tools/index.html">online version
  of the tool</a> available. Both UCSC (<a
  href="https://genome.ucsc.edu/util.html" rel="external">liftOver</a>) and NCBI
  (<a href="http://www.ncbi.nlm.nih.gov/genome/tools/remap"
  rel="external">Remap</a>) also provide tools for converting data between
  assemblies. </p>
  
  <hr/>
    
  <h2 id="pick">Summarising annotation</h2>
  
  <p> By default VEP is configured to provide annotation on every genomic
  feature that each input variant overlaps. This means that if a variant
  overlaps a gene with multiple alternate splicing variants (transcripts), then
  a block of annotation for each of these transcripts is reported in the output.
  In the <a href="../vep_formats.html#output">default VEP output format</a> each
  of these blocks is written on a single line of output; in
  <a href="../vep_formats.html#vcfout">VCF output format</a> the blocks are
  separated by commas in the INFO field. </p>
  
  <p> For many users, however, this depth of annotation is not required, and to
  this end VEP provides a number of options to reduce the amount of output
  produced. Which to choose depends on your motivations and requirements on the
  output. </p>
  
  <p> <b>NB:</b> Wherever possible we would discourage users from summarising
  data in this way. Summarising inevitably involves data loss, and invariably at
  some point this will lead to the loss of biologically relevant information. For
  example, if your variant overlaps both a regulatory feature and a transcript
  and you use one of the flags below, the overlap with the regulatory feature
  will be lost in your output, when in some cases this may be a clue to the
  "real" functional effect of your variant. For these reasons we encourage users
  to use one of the flagging options
  (<a href="vep_options.html#opt_flag_pick">--flag_pick</a>,
  <a href="vep_options.html#opt_flag_pick_allele">--flag_pick_allele</a> or
  <a href="vep_options.html#opt_flag_pick_allele_gene">--flag_pick_allele_gene</a>) and to
  <a href="vep_filter.html">post-filter results</a>. </p>
  
  <ul>
    <li><b><a href="vep_options.html#opt_pick">--pick</a></b>: this is the option
    we anticipate will be of use to most users. VEP chooses one block of
    annotation per variant, using an ordered set of criteria. This order may be
    customised using <a href="vep_options.html#opt_pick_order">--pick_order</a>.
      <ol>
        <li>canonical status of transcript</li>
        <li><a href="/Help/Glossary?id=521">APPRIS isoform annotation</a></li>
        <li><a href="/Help/Glossary?id=492">transcript support level</a></li>
        <li>biotype of transcript (protein_coding preferred)</li>
        <li>CCDS status of transcript</li>
        <li>consequence rank according to <a href="/info/genome/variation/predicted_data.html#consequences">this table</a></li>
        <li>translated, transcript or feature length (longer preferred)</li>
      </ol>
    </li>

    <p> Note that some categories may not be available for the species or cache
    version that you are using; in these cases the category will be skipped and
    the next in line used. </p>
    
    <li><b><a href="vep_options.html#opt_pick_allele">--pick_allele</a></b>: as above, but chooses one consequence block per variant allele. This can be useful for <a href="../vep_formats.html#vcf">VCF input files</a> with more than one ALT allele</li>
    
    <li><b><a href="vep_options.html#opt_per_gene">--per_gene</a></b>: as <a href="vep_options.html#opt_pick">--pick</a>, but chooses one annotation block per gene that the input variant overlaps</li>
    
    <li><b><a href="vep_options.html#opt_pick_allele_gene">--pick_allele_gene</a></b>: as above, but chooses one consequence block per variant allele and gene combination.</li>
    
    <li><b><a href="vep_options.html#opt_flag_pick">--flag_pick</a></b>: instead of choosing one block and removing the others, this option adds a flag &quot;PICK=1&quot; to picked annotation block, allowing users to easily filter on this later using VEP's <a href="vep_filter.html">filtering script</a></li>
    
    <li><b><a href="vep_options.html#opt_flag_pick_allele">--flag_pick_allele</a></b>: as above, but flags one block per allele</li>
    
    <li><b><a href="vep_options.html#opt_flag_pick_allele">--flag_pick_allele_gene</a></b>: as above, but flags one block per allele and gene combination</li>
    
    <li><b><a href="vep_options.html#opt_most_severe">--most_severe</a></b>: this flag reports only the consequence type of the block with the highest rank, according to <a href="/info/genome/variation/predicted_data.html#consequences">this table</a>. Feature-specific annotation is absent from the output using this flag, so use with caution!</li>
    
    <li><b><a href="vep_options.html#opt_summary">--summary</a></b>: this flag reports only a comma-separated list of the consequence types predicted for this variant. Feature-specific annotation is absent from the output using this flag, so use with caution!</li>
  </ul>
  
  <hr/>
    
  <h2 id="hgvs">HGVS notations</h2>

  <h4>Output</h4>

  <p> <a rel="external" href="http://www.hgvs.org/mutnomen/" target="_blank">HGVS</a>
  notations can be produced by VEP using the
  <a href="vep_options.html#opt_hgvs">--hgvs</a> flag. Coding (c.) and protein (p.)
  notations given against Ensembl identifiers use
  <a href="/info/genome/stable_ids/versions.html">versioned</a> identifiers that
  guarantee the identifier refers always to the same sequence. </p>

  <p> Genomic HGVS notations may be reported using <a href="vep_options.html#opt_hgvsg">--hgvsg</a>. Note that the named reference for HGVSg notations will be the chromosome name from the user input (as opposed to the officially recommended chromosome accession). </p>

  <p> HGVS notations for insertions or deletions are by default shifted 3-prime relative to the reported transcript or protein sequence in accordance with HGVS specifications. This may lead to discrepancies between the coordinates reported in the HGVS nomenclature and the coordinate columns reported by VEP. You may instruct VEP not to shift using <a href="vep_options.html#opt_shift_hgvs">--shift_hgvs 0</a>. </p>

  <h4>Input</h4>
  
  <p> VEP supports using HGVS notations as input. This feature is
  currently under development, and not all HGVS notation types are supported.
  Notations relative to genomic (g.) or coding (c.) sequences
  are currently fully supported; protein (p.) notations are supported in limited
  fashion due to the complexity involved in determining the multiple possible
  underlying genomic sequence changes that could produce a single protein
  change. The script will warn the user if it fails to parse a particular
  notation. </p>
  
  <p> By default VEP uses Ensembl transcripts as its reference for
  determining consequences, and hence also for HGVS notations. However, it is
  possible to parse HGVS notations that use RefSeq transcripts as the
  reference sequence by using the <a
  href="vep_options.html#opt_refseq">--refseq</a> flag when running the script.
  Such notations must include the version number of the transcript e.g. </p>
  
  <pre>NM_080794.3:c.1001C>T</pre>
  
  <p> where ".3" denotes that this is version 3 of the transcript NM_080794.
  <a href="#refseq">See below</a> for more details on how VEP can use
  RefSeq transcripts. </p>
  
    <hr/>
  <h2 id="refseq">RefSeq transcripts</h2>
  
  <p> Ensembl produces databases containing alignments of RefSeq
  transcript objects to the reference genome, named <a
  href="/info/docs/api/core/index.html#est">otherfeatures</a> databases.
  The otherfeatures databases are used to build the RefSeq cache, and merged
  with the standard Ensembl core database to produce the merged cache.
  These caches also contain alignments of CCDS
  transcripts and Ensembl EST sequences - they may be included in your
  analysis using <a href="vep_options.html#opt_all_refseq">--all_refseq</a>.</p>

  <p>By using the <a href="vep_options.html#opt_refseq">--refseq</a> flag when
  running VEP, these alternative transcripts will be used as the
  reference for predicting variant consequences. Gene IDs given in the output
  when using this option are generally NCBI GeneIDs. </p>

  <p>Users may wish to exclude predicted RefSeq transcripts (those with identifiers beginning with "XM_" or "XR_") by using <a href="vep_options.html#opt_exclude_predicted">--exclude_predicted</a>.</p>

  <h3 id="refseq_limitations">Identifiers and other data</h3>

  <p> VEP's RefSeq cache lacks many classes of data present in the Ensembl transcript cache. </p>

  <ul>
    <li>
      Included in the RefSeq cache
      <ul>
        <li>Gene symbol</li>
        <li>SIFT and PolyPhen predictions</li>
      </ul>
    </li>
    <li>
      <b>Not</b> included in the RefSeq cache
      <ul>
        <li>APPRIS annotation</li>
        <li>TSL annotation</li>
        <li>Uniprot identifiers</li>
        <li>CCDS identifiers</li>
        <li>Protein domains</li>
        <li>Gene-phenotype association data</li>
        <li>LRG genes</li>
      </ul>
    </li>
  </ul>

  <h3>Differences to the reference genome</h3>
  
  <p> Users should note that RefSeq sequences may differ from the reference genome
  sequence to which they are aligned. Ensembl's API (and hence VEP) constructs
  transcript models using the genomic reference sequence. Differences between the
  RefSeq sequence and the genomic sequence are not accounted for, so the genomic sequence
  will be used, meaning that some annotations produced by VEP on these transcripts may be inaccurate.
  Most differences occur in non-coding regions, typically in UTRs at either end of transcripts or in the addition of a poly-A tail,
  meaning minimal impact on VEP's annotations.</p>

  <p> For the GRCh38 VEP cache, each RefSeq transcript is annotated with the <a href="../vep_formats.html#refseq_match">REFSEQ_MATCH flag</a>
  indicating whether and how the RefSeq model differs from the underlying genome. Note that currently the REFSEQ_MATCH flag will not be set when using the GRCh37 cache.</p>

  <h3 id="refseq_bam">Correcting transcript models with BAM files</h3>

  <p><b>WARNING: Experimental feature!</b> NCBI have released BAM files that contain alignments of RefSeq transcripts to the genome.
  VEP can use these files to correct the models as read from the RefSeq or merged cache, as well as from RefSeq GFF files. </p>

  <pre class="code">./vep -cache -refseq -i variants.vcf -bam interim_GRCh38.p10_knownrefseq_alignments_2017-01-13.bam</pre>

  <p>BAM files are available from NCBI:</p>

  <ul>
    <li><a href="ftp://ftp.ncbi.nlm.nih.gov/genomes/Homo_sapiens/GRCh38.p10_interim_annotation/" rel="external">Human GRCh38.p10</a></li>
    <li><a href="ftp://ftp.ncbi.nlm.nih.gov/genomes/Homo_sapiens/GRCh37.p13_interim_annotation/" rel="external">Human GRCh37.p13</a></li>
  </ul>

  <p><b>NB:</b>The BAM index files (.bai) in this directory are required and will need to be renamed as the perl library used to parse the files expects the index to be named [indexed_bam_file].bai:</p>

  <pre class="code">mv interim_GRCh38.p10_knownrefseq_alignments_2017-01-13.bai interim_GRCh38.p10_knownrefseq_alignments_2017-01-13.bam.bai</pre>

  <p>VEP uses the sequence and alignment in the BAM to correct the RefSeq model. If the corrected model does not match the original RefSeq sequence in the BAM, the corrected model is discarded. The success or failure of the BAM edit is recorded in the BAM_EDIT field of the VEP output.</p>

  <p>Using a BAM causes VEP to change how alleles are interpreted from input variants.
  Input variants are typically encoded in VCFs that are called using the reference genome.
  This means that the alternate (ALT) allele as given in the VCF may correspond to the reference allele as found in the corrected RefSeq transcript model.
  VEP will account for this, using the corrected reference allele when calculating consequences, and the GIVEN_REF and USED_REF fields in the VEP output indicate any change made.
  Note that this may clash with any interpretation from using <a href="vep_options.html#opt_check_ref">--check_ref</a>, so it is recommended to avoid using this flag.</p>

  
</div>

</body>
</html>
