<html>
<head>
<title>Examples and use cases</title>
<meta name="order" content="8" />
</head>

<body>
    
<div>
	
  <div style="float:right"><img src="/img/vep_logo.png"/></div>
	
  <h1 id="top"><span style="color:#006;padding-right:15px">Ensembl Variant Effect Predictor</span><span style="color:#666"><img src="/i/16/user.png"/> Examples and use cases</span></h1>
  <hr/>
  
  
  <h2 id="examples">Example commands</h2>
  
  <ul>
    <li>
      <p>Read input from <b>STDIN</b>, output to <b>STDOUT</b></p>
      <pre class="code sh_sh">./vep --cache -o stdout</pre>
    </li>
    <li>
      <p>Add <b>regulatory</b> region <b>consequences</b></p>
      <pre class="code sh_sh">./vep --cache -i variants.txt --regulatory</pre>
    </li>
    <li>
      <p>Input file <kbd>variants.vcf.txt</kbd>, input file <b>format VCF</b>, add <b>gene symbol</b> identifiers</p>
      <pre class="code sh_sh">./vep --cache -i variants.vcf.txt --format vcf --symbol</pre>
    </li>
    <li>
      <p><b>Filter</b> out <b>common variants</b> based on 1000 Genomes data</p>
      <pre class="code sh_sh">./vep --cache -i variants.txt --filter_common</pre>
    </li>
    <li>
      <p><b>Force overwrite</b> of output file <kbd>variants_output.txt</kbd>, check for existing
      <b>co-located variants</b>, output only <b>coding sequence</b> consequences, output
      <b>HGVS names</b></p>
      <pre class="code sh_sh">./vep --cache -i variants.txt -o variants_output.txt --force --check_existing --coding_only --hgvs</pre>
    </li>
    <li>
      <p>Specify <b>DB connection parameters</b> in registry file <kbd>ensembl.registry</kbd>, add <b>SIFT</b>
      score and prediction, <b>PolyPhen</b> prediction</p>
      <pre class="code sh_sh">./vep --database -i variants.txt --registry ensembl.registry --sift b --polyphen p</pre>
    </li>
    <li>
      <p>Connect to <b>Ensembl Genomes</b> db server for <i>Arabidopsis thaliana</i></p>
      <pre class="code sh_sh">./vep --database -i variants.txt --genomes --species arabidopsis_thaliana</pre>
    </li>
    <li>
      <p>Load config from <b>ini file</b>, run in <b>quiet mode</b></p>
      <pre class="code sh_sh">./vep --config vep.ini -i variants.txt -q</pre>
    </li>
    <li>
      <p>Use <b>cache</b> in <kbd>/home/vep/mycache/</kbd>, use <b>gzcat</b> instead of zcat</p>
      <pre class="code sh_sh">./vep --cache --dir /home/vep/mycache/ -i variants.txt --compress gzcat</pre>
    </li>
    <li>
      <p>Add custom position-based <b>phenotype</b> annotation from remote <b>BED file</b></p>
      <pre class="code sh_sh">./vep --cache -i variants.vcf --custom file=ftp://ftp.myhost.org/data/phenotypes.bed.gz,short_name=phenotype</pre>
    </li>
    <li>
      <p>Use the <b>plugin</b> named MyPlugin, output only the variation name, feature, consequence type and MyPluginOutput <b>fields</b></p>
      <pre class="code sh_sh">./vep --cache -i variants.vcf --plugin MyPlugin --fields Uploaded_variation,Feature,Consequence,MyPluginOutput</pre>
    </li>
    <li>
      <p>Right align variants before consequence calculation. For more information, see <a
         href="/info/docs/tools/vep/script/vep_other.html#shifting"
         target="_blank">here</a>.</p>
      <pre class="code sh_sh">./vep --cache -i variants.vcf --shift_3prime 1</pre>
    </li>
     <li>
      <p>Report uploaded allele before minimisation. For more information, see <a
         href="/info/docs/tools/vep/script/vep_options.html#opt_uploaded_allele"
         target="_blank">here</a>.</p>
      <pre class="code sh_sh">./vep --cache -i variants.vcf --uploaded_allele</pre>
    </li>
  </ul>
  
    
  <hr/>
  
  <h2 id="gnomad">gnomAD</h2>
  
  <p> <a href="http://gnomad.broadinstitute.org/" rel="external">gnomAD</a> exome frequency data is included in Ensembl VEP's cache files from release 90, replacing ExAC; use <a href="vep_options.html#opt_af_gnomade">--af_gnomade</a> to enable using this data. Ensembl VEP can also retrieve frequency data from the gnomAD genomes set or ExAC via Ensembl VEP's custom annotation functionality.<p>
  <p>For the latest gnomAD data, please visit <a href="https://gnomad.broadinstitute.org/downloads" rel="external">gnomAD downloads</a>.<p>
  
  <ol>
    <li>Ensembl VEP requires Bio::DB::HTS to read data from tabix-indexed VCFs - see <a href="vep_download.html#installer"><img style="vertical-align:bottom" src="/i/16/download.png"/> installation instructions</a></li>
    <li>The Ensembl FTP site hosts abridged VCF files for gnomAD additionally remapped to GRCh38 using <a href="http://crossmap.sourceforge.net/" rel="external">CrossMap</a>. It is possible for Ensembl VEP to read these files directly from their remote location, though for optimal performance the VCF and index should be downloaded to a local file system.
      <ul>
        <li>
          <b>GRCh38</b>
          <ul>
            <li>gnomAD genomes (r2.1, remapped with CrossMap):
              <a href="[[SPECIESDEFS::ENSEMBL_FTP_URL]]/data_files/homo_sapiens/GRCh38/variation_genotype/gnomad/r2.1/genomes/">[VCFs and tabix indexes]</a>
            </li>
            <li>gnomAD exomes (r2.1, remapped with CrossMap):
              <a href="[[SPECIESDEFS::ENSEMBL_FTP_URL]]/data_files/homo_sapiens/GRCh38/variation_genotype/gnomad/r2.1/exomes/">[VCFs and tabix indexes]</a>
            </li>
            <li>ExAC (v0.3, remapped using CrossMap):
              <a href="[[SPECIESDEFS::ENSEMBL_FTP_URL]]/data_files/homo_sapiens/GRCh38/variation_genotype/ExAC.0.3.GRCh38.vcf.gz">[VCF]</a>
              <a href="[[SPECIESDEFS::ENSEMBL_FTP_URL]]/data_files/homo_sapiens/GRCh38/variation_genotype/ExAC.0.3.GRCh38.vcf.gz.tbi">[tabix index]</a>
            </li>
          </ul>
        </li>

        <li>
          <b>GRCh37</b>
          <ul>
            <li>gnomAD genomes (r2.1):
              <a href="[[SPECIESDEFS::ENSEMBL_FTP_URL]]/data_files/homo_sapiens/GRCh37/variation_genotype/">[VCF and tabix indexes]</a>
            </li>
            <li>gnomAD exomes (r2.1):
              <a href="[[SPECIESDEFS::ENSEMBL_FTP_URL]]/data_files/homo_sapiens/GRCh37/variation_genotype/">[VCF and tabix indexes]</a>
            </li>
            <li>ExAC (v0.3):
              <a href="[[SPECIESDEFS::ENSEMBL_FTP_URL]]/data_files/homo_sapiens/GRCh37/variation_genotype/ExAC.0.3.GRCh37.vcf.gz">[VCF]</a>
              <a href="[[SPECIESDEFS::ENSEMBL_FTP_URL]]/data_files/homo_sapiens/GRCh37/variation_genotype/ExAC.0.3.GRCh37.vcf.gz.tbi">[tabix index]</a>
            </li>
          </ul>
        </li>

      </ul>
    </li>
    <li>
      <p>Run Ensembl VEP with the following command (using the GRCh38 input example) to get locations and continental-level allele frequencies:</p>
      <pre class="code sh_sh">./vep -i examples/homo_sapiens_GRCh38.vcf --cache \
--custom file=gnomad.genomes.r2.0.1.sites.GRCh38.noVEP.vcf.gz,short_name=gnomADg,format=vcf,type=exact,coords=0,fields=AF_AFR%AF_AMR%AF_ASJ%AF_EAS%AF_FIN%AF_NFE%AF_OTH</pre>
      <p>You will then see data under field names as described in the Ensembl VEP output header:</p>
      <pre class="code sh_sh">## gnomADg : gnomad.genomes.r2.0.1.sites.GRCh38.noVEP.vcf.gz (exact)
## gnomADg_AFR_AF : AFR_AF field from gnomad.genomes.r2.0.1.sites.GRCh38.noVEP.vcf.gz
## gnomADg_AMR_AF : AMR_AF field from gnomad.genomes.r2.0.1.sites.GRCh38.noVEP.vcf.gz
...</pre>
      where the gnomADg field contains the ID (or coordinates if no ID found) of the variant in the VCF file. Any of the fields in the gnomAD file INFO field can be added by appending them to the list in your Ensembl VEP command.
    </li>
  </ol>
    
  <hr/>
  
  <h2 id="gerp">Conservation scores</h2>
  
  <p> You can use the <a href="vep_custom.html">custom annotation</a>
  feature to add conservation scores to your output. For example, to add GERP
  scores, download the bigWig file from the list below, and run Ensembl VEP with the
  following flag:</p>
  
  <pre class="code sh_sh">./vep --cache -i example.vcf --custom file=All_hg19_RS.bw,short_name=GERP,format=bigwig</pre>
  
  <p> Example conservation score files: </p>
 
  <div>
    <div style="float:left"> 
      <b>Human (GRCh38)</b>
      <ul>
        <li><a href="http://hgdownload.soe.ucsc.edu/goldenPath/hg38/phastCons7way/" rel="external">phastCons 7-way</a> </li>
        <li><a href="http://hgdownload.soe.ucsc.edu/goldenPath/hg38/phastCons20way/" rel="external">phastCons 20-way</a> </li>
        <li><a href="http://hgdownload.soe.ucsc.edu/goldenPath/hg38/phastCons100way/" rel="external">phastCons 100-way</a> </li>
        <li><a href="http://hgdownload.soe.ucsc.edu/goldenPath/hg38/phyloP7way/" rel="external">phyloP 7-way</a> </li>
        <li><a href="http://hgdownload.soe.ucsc.edu/goldenPath/hg38/phyloP20way/" rel="external">phyloP 20-way</a> </li>
        <li><a href="http://hgdownload.soe.ucsc.edu/goldenPath/hg38/phyloP100way/" rel="external">phyloP 100-way</a> </li>
      </ul>
    </div>
    <div style="float:left;margin-left:40px">
      <b>Human (GRCh37)</b>
      <ul>
        <li><a href="http://hgdownload.soe.ucsc.edu/gbdb/hg19/bbi/All_hg19_RS.bw" rel="external">GERP</a> </li>
        <li><a href="http://hgdownload.soe.ucsc.edu/goldenPath/hg19/phastCons46way/" rel="external">phastCons 46-way</a> </li>
        <li><a href="http://hgdownload.soe.ucsc.edu/goldenPath/hg19/phastCons100way/" rel="external">phastCons 100-way</a> </li>
        <li><a href="http://hgdownload.soe.ucsc.edu/goldenPath/hg19/phyloP46way/" rel="external">phyloP 46-way</a> </li>
        <li><a href="http://hgdownload.soe.ucsc.edu/goldenPath/hg19/phyloP100way/" rel="external">phyloP 100-way</a> </li>
      </ul>
    </div>
    <div class="clear"></div>
  </div>  

  <p> All files provided by the UCSC genome browser - files for other species
  are available from their <a
  href="http://hgdownload.soe.ucsc.edu/goldenPath/" rel="external">FTP site</a>, though be sure
  to use the file corresponding to the <a
  href="/Homo_sapiens/Info/Annotation#assembly">correct assembly</a>. </p>
    
  <hr/>
  
  
  <h2 id="dbNSFP">dbNSFP</h2>

  <p> dbNSFP - <a href="http://www.ncbi.nlm.nih.gov/pubmed/21520341"
  rel="external">&quot;a lightweight database of human nonsynonymous SNPs and
  their functional predictions&quot;</a> - provides pathogenicity predictions
  from many tools (including SIFT, LRT, MutationTaster, FATHMM) across
  every possible missense substitution in the human proteome. </p>

  <p> Ensembl VEP plugins sometimes require data processed in specific ways as
  arguments. Any requirements and usage instructions for each plugin can be
  found in the <a href="vep_plugins.html">plugin documentation</a>. </p>

  <p> In the case of the dbNSFP.pm plugin, the data needs to be
  <a href="https://sites.google.com/site/jpopgen/dbNSFP" rel="external">downloaded</a>
  and then processed into a format that the plugin can use. Note that there are
  two distinct branches of the files provided for academic and commercial usage;
  please use the appropriate files for your use case. </p>
  
  <p> After downloading the file, you will need to process it so that tabix can
  index it correctly. This will take a while as the file is very large! Note
  that you will need the <a href="http://samtools.sourceforge.net/tabix.shtml"
  rel="external">tabix</a> utility in your path to use dbNSFP. </p>

  <pre class="code sh_sh">version=4.5c
unzip dbNSFP${version}.zip
zcat dbNSFP${version}_variant.chr1.gz | head -n1 > h

# GRCh38/hg38 data
zgrep -h -v "^#chr" dbNSFP${version}_variant.chr* | sort -k1,1 -k2,2n - | cat h - | bgzip -c > dbNSFP${version}_grch38.gz
tabix -s 1 -b 2 -e 2 dbNSFP${version}_grch38.gz

# GRCh37/hg19 data
zgrep -h -v "^#chr" dbNSFP${version}_variant.chr* | awk '$8 != "." ' | sort -k8,8 -k9,9n - | cat h - | bgzip -c > dbNSFP${version}_grch37.gz
tabix -s 8 -b 9 -e 9 dbNSFP${version}_grch37.gz</pre>

  <p> Then simply download the <a
  href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/dbNSFP.pm"
  rel="external">dbNSFP.pm plugin</a> and place it either in
  <b>$HOME/.vep/Plugins/</b> or a path in your <b>$PERL5LIB</b>. When you run
  Ensembl VEP with the plugin, you will need to select some of the columns that you
  wish to retrieve; to list them run Ensembl VEP with the plugin and the path to the
  dbNSFP file and no further parameters: </p>
  
  <pre class="code sh_sh">./vep --cache --force --plugin dbNSFP,dbNSFP4.5c_grch38.txt.gz
2014-04-04 11:27:05 - Read existing cache info
2014-04-04 11:27:05 - Auto-detected FASTA file in cache directory
2014-04-04 11:27:05 - Checking/creating FASTA index
2014-04-04 11:27:05 - Failed to instantiate plugin dbNSFP: ERROR: No columns selected to fetch. Available columns are:
#chr,pos(1-coor),ref,alt,aaref,aaalt,hg18_pos(1-coor),genename,Uniprot_acc,
Uniprot_id,Uniprot_aapos,Interpro_domain,cds_strand,refcodon,SLR_test_statistic,
codonpos,fold-degenerate,Ancestral_allele,Ensembl_geneid,Ensembl_transcriptid,
...</pre>
  
  <p> Note that some of these fields are replicates of those produced by the
  core Ensembl VEP code (e.g. <a href="vep_options.html#opt_sift">SIFT</a>, the <a
  href="vep_options.html#opt_af_1kg">1000 Genomes Project</a> frequencies) - you should use the
  Ensembl VEP options for frequency information rather than the annotations from
  dbNSFP as the dbNSFP file covers <b>only</b> missense substitutions. Other
  fields, such as the conservation scores, may be better served by using
  genome-wide files as described <a href="#gerp">above</a>. </p>
  
  <p> To select fields, just add them as a comma-separated list to your command
  line:</p>
  
  <pre class="code sh_sh">./vep --cache --force --plugin dbNSFP,dbNSFP4.5c_grch38.txt.gz,LRT_score,FATHM_score,MutationTaster_score</pre>
  
  <p> One final point to note is that the dbNSFP scores are frozen on a
  particular Ensembl release's transcript set; check the readme file on their
  download site to find out exactly which. While in the majority of cases
  protein sequences don't change between releases, in some circumstances the
  protein sequence used by Ensembl VEP in the latest release may differ from the
  sequence used to calculate the scores in dbNSFP. </p>
  
  <hr/>


  <h2 id="StructVar">Structural variants</h2>

  <p>Ensembl VEP can annotate structural variants (SV) with their predicted effect on other genomic features.
     For more information on SV input format, see <a href="/info/docs/tools/vep/vep_formats.html#sv">here</a>.</p>

  <h4>Prediction process</h4>
  <ul>
  <li> If the INFO keys <code>END</code> or <code>SVLEN</code> are present, the proportion of any overlapping feature covered by the variant is calculated</li>
  <li> The alternative allele (or <code>SVTYPE</code> in older VCF files) defines the type of structural variant; some types of structural variants are tested for specific consequences:</li>

  <table class="ss">
    <tr>
      <th>Structural variant type</th>
      <th>Abbreviation</th>
      <th>Specific consequences</th>
    </tr>
    <tr class="bg1">
      <td>Insertion</td>
      <td>INS</td>
      <td>Feature elongation</td>
    </tr>
    <tr class="bg2">
      <td>Deletion</td>
      <td>DEL</td>
      <td>Feature truncation</td>
    </tr>
    <tr class="bg1">
      <td>Duplication</td>
      <td>DUP</td>
      <td>Feature amplification/elongation</td>
    </tr>
    <tr class="bg2">
      <td>Inversion</td>
      <td>INV</td>
      <td><i>Not tested for any specific consequence</i></td>
    </tr>
    <tr class="bg1">
      <td>Copy number variation</td>
      <td>CNV</td>
      <td>Feature amplification/elongation (if copy number is 2) or truncation (if copy number is 0)</td>
    </tr>
    <tr class="bg2">
      <td>Breakpoint variant</td>
      <td>BND</td>
      <td>Feature truncation</td>
    </tr>
  </table>
  </ul>

  <h5> Insertions and deletions</h5>

  <ul>
    <li> Supports <a href="/info/docs/tools/vep/vep_formats.html#sv">mobile element insertions/deletions</a>, including ALU, HERV, LINE1 and SVA elements
    <ul>
      <li> Currently, mobile element variants are treated as any insertion/deletion
    </ul>
  </ul>

  <h5> Breakpoint variants</h5>

  <ul>
    <li> Supports chromosome synonyms in breakends (such as <code>chr4</code> and <code>NC_000004.12</code>)
    <li> Processes <a href="/info/docs/tools/vep/vep_formats.html#sv">single breakends and multiple, comma-separated alternative breakends</a>
    <li> Consequences are reported for each breakend; for instance, for a VCF input like <code>1 7936271 . N N[12:58877476[,N[X:10932343[</code>, it will report the consequences for each of the 3 breakends:
    <ul>
      <li> <code>N[12:58877476[</code>: consequences for the first alternative breakend near chr12:58877476
      <li> <code>N[X:10932343[</code>: consequences for the second alternative breakend near chrX:10932343
      <li> <code>N.</code>: consequences for the reference breakend near chr1:7936271 (represented as detailed in the <a href="https://samtools.github.io/hts-specs/VCFv4.4.pdf" rel="external">VCF 4.4 specification, section 5.4.9: Single breakends</a>)
    </ul>
    <li> In case of specific breakends not overlaping any reported Ensembl features (such as transcripts and regulatory regions), that specific breakend will <b>NOT</b> be presented in VEP output.
  </ul>

  <h4> Reported overlaps</h4>
  <ul>
  <li> Ensembl VEP calculates the length and proportion of each genomic feature overlapped by a structural variant
  <li> Use the <a href="vep_options.html#opt_overlaps">--overlaps</a> option to enable this when using VCF or tab format.
       (This is reported by default in standard Ensembl VEP and JSON format.)
  <li> The keys <code>bp_overlap</code> and <code>percentage_overlap</code> are used in JSON format and <code>OverlapBP</code> and <code>OverlapPC</code> in other formats.
  </ul>

  <h4> Plugin support</h4>

  <ul>
    <li> <a href="vep_plugins.html#CADD">CADD plugin</a>
    <li> <a href="vep_plugins.html#Conservation">Conservation plugin</a>
    <li> <a href="vep_plugins.html#NearestGene">NearestGene plugin</a>
    <li> <a href="vep_plugins.html#Phenotypes">Phenotypes plugin</a>
    <li> <a href="vep_plugins.html#StructuralVariantOverlap">StructuralVariantOverlap plugin</a>: please note that all features of this plugin have been ported to <a href="vep_custom.html">--custom annotation</a>, with additional improvements
    <li> <a href="vep_plugins.html#TSSDistance">TSSDistance plugin</a>
  </ul>

  <h4> Changing memory requirements</h4>
  <ul>
  <li> By default, Ensembl VEP does not annotate variants larger than 10M. If you are using the command
  line tool, you can use the <a href="vep_options.html#opt_max_sv_size">--max_sv_size</a> option to modify this.
    <ul>
      <li> This limit is not associated with breakpoint variants: each breakend in a breakpoint variant is analysed by Ensembl VEP as a single base (the alternative sequence is currently ignored).
    </ul>
  <li>By default, variants are analysed in batches of 5000. Using the <a href="vep_options.html#opt_buffer_size">--buffer_size</a>
  option to reduce this can reduce memory requirements, especially if your data is sparse.
  A smaller buffer size is essential when annotating structural variants with regulatory data.
  </ul>

  <hr/>


  
  <h2 id="pangenomes">Pangenome assemblies</h2>

  <p> Ensembl VEP is able to analyse variants in <b>any species or assembly</b> (even if not part of <a href="/info/about/species.html">Ensembl data</a>) by providing your own <a href="vep_cache.html#fasta">FASTA file</a> and <a href="vep_cache.html#gff" target="_blank">GFF/GTF annotation</a>: </p>

  <pre class="code sh_sh">./vep -i variants.txt -o variants_output.txt --gff data.gff.gz --fasta genome.fa.gz</pre>

  <p> We also provide data for other assemblies besides those supported in the current Ensembl and Ensembl Genomes sites. </p>

  <h3 id="HPRC">HPRC assemblies</h3>

  <p>The <a href="https://humanpangenome.org/" rel="external">Human Pangenome Reference Consortium (HPRC)</a> aims to sequence 350 individuals of diverse ancestries, producing a pangenome of 700 haplotypes by the end of 2024. The first publication (<i><a href="https://www.nature.com/articles/s41586-023-05896-x" rel="external">A draft human pangenome reference</a></i>) describes 47 phased, diploid assemblies from a cohort of genetically diverse individuals.</p>

  <p>The Ensembl VEP command-line tool (CLI) can annotate and filter variants called against the latest human assemblies, including the telomere-to-telomere assembly of the CHM13 cell line (T2T-CHM13). We have annotated genes on these human assemblies, based on Ensembl/<a href="https://www.gencodegenes.org/human/release_38.html" rel="external">GENCODE 38</a> genes and transcripts, via a new mapping pipeline as detailed in the Methods section of <i><a href="https://www.nature.com/articles/s41586-023-05896-x" rel="external">A draft human pangenome reference</a></i>. The links to download and visualise the human annotations for HPRC assemblies are summarised in the <a href="https://projects.ensembl.org/hprc/" rel="external">Ensembl HPRC data page</a>.</p>

  <h4 id="HPRC_run">Running Ensembl VEP with the human HPRC assemblies</h4>

  <p>Currently, Ensembl VEP can only be run with HPRC assemblies in offline mode, one assembly at a time. There are two ways to use Ensembl VEP with HPRC assemblies:</p>

  <ul>
    <li> Using <b>Ensembl VEP cache</b> with (recommended) <b>FASTA sequence</b> (the most efficient way) </li>
    <li> Using <b>GTF annotation</b> with (mandatory) <b>FASTA sequence</b> </li>
  </ul>

  <p>In the examples below, we demonstrate annotating variants on <b>T2T-CHM13v2.0</b> (<a href="https://ftp.ensembl.org/pub/rapid-release/species/Homo_sapiens/GCA_009914755.4/" rel="external">GCA_009914755.4</a> assembly). To create a sample VCF to use in the examples below, you can take the first 100 lines from the ClinVar VCF file mapped to T2T-CHM13:</p>

  <pre class="code sh_sh">clinvar=ftp://ftp.ensembl.org/pub/rapid-release/species/Homo_sapiens/GCA_009914755.4/ensembl/variation/2022_10/vcf/2024_07/clinvar_20240624_GCA_009914755.4.vcf.gz
tabix -h $clinvar 1 | head -n 100 > test.vcf</pre>

  <h5 id="cache">Ensembl VEP cache</h5>

  <p>The <a href="vep_cache.html#cache">cache</a> is a downloadable archive containing all transcript models for an assembly; it may also contain regulatory features and variant data.</p>

  <p>Let's start by downloading and extracting the cache to the default Ensembl VEP directory (available for each annotation by clicking in <b>VEP cache</b> in the <a href="https://projects.ensembl.org/hprc/" rel="external">Ensembl HPRC data page</a>). In the case of T2T-CHM13:</p>

  <pre class="code sh_sh">
cd $HOME/.vep
curl -O https://ftp.ensembl.org/pub/rapid-release/species/Homo_sapiens/GCA_009914755.4/ensembl/variation/2022_10/indexed_vep_cache/Homo_sapiens-GCA_009914755.4-2022_10.tar.gz
tar xzf Homo_sapiens-GCA_009914755.4-2022_10.tar.gz</pre>

  <p>This will create the folder <kbd>homo_sapiens_gca009914755v4/107_T2T-CHM13v2.0</kbd> with the gene data required to run Ensembl VEP. The name of this folder contains relevant information when running Ensembl VEP:</p>

  <ul>
    <li> Species: <kbd>homo_sapiens_gca009914755v4</kbd> </li>
    <li> Cache version: <kbd>107</kbd> </li>
    <li> Assembly: <kbd>T2T-CHM13v2.0</kbd> </li>
  </ul>

  <p> As well as molecular consequence predictions, many gene/transcript-based <a href="vep_options.html">options</a> are supported for HPRC assemblies: </p>

  <pre class="code sh_sh">vep -i test.vcf --offline \
    --species homo_sapiens_gca009914755v4 \
    --cache_version 107 \
    --fasta Homo_sapiens-GCA_009914755.4-softmasked.fa.gz \
    --domains --symbol --canonical --protein --biotype --uniprot --variant_class</pre>

  <p> We don't have other annotations, such as RefSeq transcripts or variant information in the cache. </p>

  <p> To run Ensembl VEP with the downloaded cache in offline mode, please specify the species (which here includes assembly name) and cache version:</p>

  <pre class="code sh_sh">vep -i test.vcf --offline --species homo_sapiens_gca009914755v4 --cache_version 107</pre>

  <h5 id="fasta">FASTA sequence</h5>

  <p>When using the cache, supplying the reference genomic sequence in a FASTA file is optional, but is required to enable the following options:</p>

  <ul>
    <li> Create HGVS notations (<a href="vep_options.html#opt_hgvs">--hgvs</a> and <a href="vep_options.html#opt_hgvsg">--hgvsg</a>) </li>
    <li> Check the reference sequence given in input data (<a href="vep_options.html#opt_check_ref">--check_ref</a>) </li>
  </ul>

  <p> Genomic FASTA files can be found in <a href="https://projects.ensembl.org/hprc/" rel="external">Ensembl HPRC data page</a> <b>> FTP dumps > ensembl > genome</b>. FASTA files need to be either uncompressed or compressed with <b>bgzip</b> (recommended) to be compatible with Ensembl VEP. For instance, to download a compressed FASTA file, uncompress it and then re-compress it with bgzip: </p>

  <pre class="code sh_sh">curl -O https://ftp.ensembl.org/pub/rapid-release/species/Homo_sapiens/GCA_009914755.4/ensembl/genome/Homo_sapiens-GCA_009914755.4-softmasked.fa.gz
gzip -d Homo_sapiens-GCA_009914755.4-softmasked.fa.gz
bgzip Homo_sapiens-GCA_009914755.4-softmasked.fa.gz</pre>

  <p> Afterwards, you can run Ensembl VEP using cache and the <a href="vep_options.html#opt_fasta">--fasta</a> flag: </p>

  <pre class="code sh_sh">vep -i test.vcf --offline \
    --species homo_sapiens_gca009914755v4 \
    --cache_version 107 \
    --fasta Homo_sapiens-GCA_009914755.4-softmasked.fa.gz</pre>

  <p> More information on using FASTA files is available <a href="vep_cache.html#fasta">here</a>. </p>

  <h5 id="gtf">GTF and GFF annotation</h5>

  <p> As an alternative to using cache files, Ensembl VEP can utilise gene information in appropriately indexed GTF or GFF files. GTF and GFF files can be downloaded from the annotation column in the <a href="https://projects.ensembl.org/hprc/" rel="external">Ensembl HPRC data page</a>. The data needs to be re-sorted in chromosomal order, compressed in <b>bgzip</b> and indexed with <b>tabix</b>. We present here the example for a GTF file: </p>

  <pre class="code sh_sh">curl -O https://ftp.ensembl.org/pub/rapid-release/species/Homo_sapiens/GCA_009914755.4/ensembl/geneset/2022_07/Homo_sapiens-GCA_009914755.4-2022_07-genes.gtf.gz
gzip -d Homo_sapiens-GCA_009914755.4-2022_07-genes.gtf.gz
grep -v "#" Homo_sapiens-GCA_009914755.4-2022_07-genes.gtf |\
  sort -k1,1 -k4,4n -k5,5n -t$'\t' |\
  bgzip -c > Homo_sapiens-GCA_009914755.4-2022_07-genes.gtf.gz
tabix Homo_sapiens-GCA_009914755.4-2022_07-genes.gtf.gz</pre>

  <p> FASTA files are <b>always</b> required when running HPRC data with GTF annotation, as the transcript sequences are not available in the GFF files. </p>

  <p> Afterwards, you can run Ensembl VEP using the GTF and FASTA files: </p>

  <pre class="code sh_sh">vep -i test.vcf \
      --gtf Homo_sapiens-GCA_009914755.4-2022_07-genes.gtf.gz \
      --fasta Homo_sapiens-GCA_009914755.4-softmasked.fa.gz</pre>

  <p> Check <a href="vep_cache.html#gff">here</a> for more information on using gene annotations in GTF and GFF files. </p>

  <h3 id="PolyPhen_SIFT"> Missense deleteriousness predictions </h3>

  <p> Although PolyPhen-2 and SIFT scores are not directly available for alternative assemblies by using <a href="vep_options.html#opt_polyphen">--polyphen</a> and <a href="vep_options.html#opt_sift">--sift</a>, they can be retrieved via the <a href="vep_plugins.html#PolyPhen_SIFT">PolyPhen_SIFT plugin</a>. </p>

  <p> Using our <a href="https://github.com/Ensembl/ensembl-variation/tree/main/nextflow/ProteinFunction" rel="external">ProteinFunction pipeline</a>, we ran <b>PolyPhen-2 2.2.3</b> and <b>SIFT 6.2.1</b> on the proteome sequences for GRCh38 and all HPRC assemblies (the protein FASTA files indicated in <a href="https://projects.ensembl.org/hprc/" rel="external">Ensembl HPRC data page</a>) and stored their results in a single SQLite file: <a href="https://ftp.ensembl.org/pub/current_variation/pangenomes/Human/homo_sapiens_pangenome_PolyPhen_SIFT_20240502.db" rel="external"><kbd>homo_sapiens_pangenome_PolyPhen_SIFT_20240502.db</kbd></a>.

  <p> Pre-computed scores and predictions can be retrieved by downloading this file and using the <b>PolyPhen_SIFT plugin</b>: </p>

  <pre class="code sh_sh">curl -O https://ftp.ensembl.org/pub/current_variation/pangenomes/Human/homo_sapiens_pangenome_PolyPhen_SIFT_20240502.db
vep -i test.vcf --offline \
    --species homo_sapiens_gca009914755v4 \
    --cache_version 107 \
    --fasta Homo_sapiens-GCA_009914755.4-softmasked.fa.gz \
    --plugin PolyPhen_SIFT,db=human_pangenomes.PolyPhen_SIFT.db</pre>

  <h3> Matched variant annotations (ClinVar, gnomAD and dbSNP) </h3>

  <p> We don't have variant data in the caches for the HPRC assemblies, but it can be integrated using the <a href="vep_custom.html">--custom</a> option with data files using the same assembly coordinates. We have lifted-over some key datasets, including ClinVar and gnomAD to the HPRC assemblies (downloadable from the VCF column in <a href="https://projects.ensembl.org/hprc/" rel="external">Ensembl HPRC data page</a>). </p>

  <pre class="code sh_sh"># Download ClinVar data and respective index (TBI)
curl -O https://ftp.ensembl.org/pub/rapid-release/species/Homo_sapiens/GCA_009914755.4/ensembl/variation/2022_10/vcf/2024_07/clinvar_20240624_GCA_009914755.4.vcf.gz
curl -O https://ftp.ensembl.org/pub/rapid-release/species/Homo_sapiens/GCA_009914755.4/ensembl/variation/2022_10/vcf/2024_07/clinvar_20240624_GCA_009914755.4.vcf.gz.tbi

# Run Ensembl VEP with ClinVar data
vep -i test.vcf --offline \
    --species homo_sapiens_gca009914755v4 --cache_version 107 \
    --fasta Homo_sapiens-GCA_009914755.4-softmasked.fa.gz \
    --custom file=clinvar_20240624_GCA_009914755.4.vcf.gz,short_name=ClinVar,format=vcf,type=exact,coords=0,fields=CLNSIG%CLNREVSTAT%CLNDN</pre>

  <h3> Additional annotations </h3>

  <p> Ensembl VEP plugins are a simple way to add new functionality to your analysis. Many require data that is only available for GRCh37 or GRCh38, but others, for example those based on gene attributes or on the fly analysis are compatible with the HGRC assemblies. </p>

  <p> Here follows Ensembl VEP plugins that are easily compatible with alternative human assemblies: </p>

  <style>
    tr:nth-child(odd) {background-color: #f0f0f0;}
    td {
      line-height: 1.25em;
      word-wrap: break-word;
    }
    th { word-wrap: break-word; }
    .info {width: inherit !important;}
  </style>

  <table class="ss" style="table-layout: fixed"><thead>
    <tr>
      <th>Plugin</th>
      <th>Description</th>
      <th>Plugin data</th>
      <th>Usage example</th>
    </tr></thead>
  <tbody>
    <tr>
      <td><a href="https://github.com/Ensembl/VEP_plugins/blob/main/Blosum62.pm"  rel="external">Blosum62</a></td>
      <td>Looks up the BLOSUM 62 substitution matrix score for the reference and alternative amino acids predicted for a missense mutation.</td>
      <td></td>
      <td><kbd>--plugin Blosum62</kbd></td>
    </tr>
    <tr>
      <td><a href="https://github.com/Ensembl/VEP_plugins/blob/main/DosageSensitivity.pm" rel="external">DosageSensitivity</a></td>
      <td>Retrieves haploinsufficiency and triplosensitivity probability scores<br>for affected genes (<a href="https://doi.org/10.1016/j.cell.2022.06.036">Collins <i>et al.</i>, 2022</a>).</td>
      <td><a href="https://zenodo.org/record/6347673/files/Collins_rCNV_2022.dosage_sensitivity_scores.tsv.gz"><kbd>Collins_rCNV_2022.dosage_sensitivity_scores.tsv.gz</kbd></a></td>
      <td><kbd>--plugin DosageSensitivity,file=Collins_rCNV_2022.dosage_sensitivity_scores.tsv.gz</kbd></td>
    </tr>
    <tr>
      <td><a href="https://github.com/Ensembl/VEP_plugins/blob/main/Downstream.pm" rel="external">Downstream</a></td>
      <td>Predicts downstream effects of a frameshift variant on the protein sequence of a transcript.</td>
      <td>Requires a FASTA file provided via the <a href="vep_options.html#opt_fasta">--fasta</a> option</td>
      <td><kbd>--plugin Downstream</kbd></td>
    </tr>
    <tr>
      <td><a href="https://github.com/Ensembl/VEP_plugins/blob/main/Draw.pm" rel="external">Draw</a></td>
      <td>Draws pictures of the transcript model showing the variant location.</td>
      <td></td>
      <td><kbd>--plugin Draw</kbd></td>
    </tr>
    <tr>
      <td><a href="https://github.com/Ensembl/VEP_plugins/blob/main/GeneSplicer.pm" rel="external">GeneSplicer</a></td>
      <td>Runs <a href="https://ccb.jhu.edu/software/genesplicer/">GeneSplicer</a> to get splice site predictions.</td>
      <td>Binary and training data for GeneSplicer (<a href="vep_plugins.html#GeneSplicer">plugin instructions</a>)</td>
      <td><kbd>--plugin GeneSplicer,binary=genesplicer/bin/linux/genesplicer,training=genesplicer/human</kbd></td>
    </tr>
    <tr>
      <td><a href="https://github.com/Ensembl/VEP_plugins/blob/main/GO.pm" rel="external">GO</a></td>
      <td>Retrieves Gene Ontology (GO) terms associated with genes (for HGRC assemblies, specifically) using custom GFF annotation containing GO terms.</td>
      <td><a href="https://projects.ensembl.org/hprc/">Ensembl HPRC data page</a> <b>&gt; FTP dumps &gt; ensembl &gt; variation &gt; [date] &gt; gff:</b>
      <ul>
        <li> <kbd>*_GO_plugin.gff.gz</kbd> </li>
        <li> <kbd>*_GO_plugin.gff.gz.tbi</kbd> </li>
      </ul>
      </td>
      <td><kbd>--plugin GO,file=homo_sapiens_gca009914755v4_110_VEP_GO_plugin.gff.gz</kbd></td>
    </tr>
    <tr>
      <td><a href="https://github.com/Ensembl/VEP_plugins/blob/main/HGVSIntronOffset.pm" rel="external">HGVSIntronOffset</a></td>
      <td>Returns HGVS intron start and end offsets. To be used with <a href="https://www.ensembl.org/info/docs/tools/vep/script/vep_options.html#opt_hgvs">--hgvs</a> option.</td>
      <td></td>
      <td><kbd>--plugin HGVSIntronOffset</kbd></td>
    </tr>
    <tr>
      <td><a href="https://github.com/Ensembl/VEP_plugins/blob/main/LoFtool.pm" rel="external">LoFtool</a></td>
      <td>Provides a rank of genic intolerance and consequent susceptibility to disease based on the ratio of Loss-of-function (LoF) to synonymous mutations for each gene.</td>
      <td></td>
      <td><kbd>--plugin LoFtool</kbd></td>
    </tr>
    <tr>
      <td><a href="https://github.com/Ensembl/VEP_plugins/blob/main/MaxEntScan.pm" rel="external">MaxEntScan</a></td>
      <td>Runs <a href="http://hollywood.mit.edu/burgelab/maxent/Xmaxentscan_scoreseq.html">MaxEntScan</a> to get splice site predictions.</td>
      <td>Extracted directory from <a href="http://hollywood.mit.edu/burgelab/maxent/download/fordownload.tar.gz">fordownload.tar.gz</td>
      <td><kbd>--plugin MaxEntScan,/path/to/fordownload</kbd></td>
    </tr>
    <tr>
      <td><a href="https://github.com/Ensembl/VEP_plugins/blob/main/NearestExonJB.pm" rel="external">NearestExonJB</a></td>
      <td>Finds the nearest exon junction boundary to a coding sequence variant.</td>
      <td></td>
      <td><kbd>--plugin NearestExonJB</kbd></td>
    </tr>
    <tr>
      <td><a href="https://github.com/Ensembl/VEP_plugins/blob/main/NMD.pm" rel="external">NMD</a></td>
      <td>Predicts if a variant allows the transcript to escape nonsense-mediated mRNA decay based on certain rules.</td>
      <td></td>
      <td><kbd>--plugin NMD</kbd></td>
    </tr>
    <tr>
      <td><a href="https://github.com/Ensembl/VEP_plugins/blob/main/Phenotypes.pm" rel="external">Phenotypes</a></td>
      <td>Retrieves overlapping phenotype information.</td>
      <td><a href="https://projects.ensembl.org/hprc/">Ensembl HPRC data page</a> <b>&gt; FTP dumps &gt; ensembl &gt; variation &gt; [date] &gt; gff:</b>
        <ul>
          <li> <kbd>*_phenotypes_plugin.gvf.gz</kbd> </li>
          <li> <kbd>*_phenotypes_plugin.gvf.gz.tbi</kbd> </li>
        </ul>
      </td>
      <td><kbd>--plugin Phenotypes,file=homo_sapiens_gca009914755v4_110_VEP_phenotypes_plugin.gvf.gz</kbd></td>
    </tr>
    <tr>
      <td><a href="https://github.com/Ensembl/VEP_plugins/blob/main/pLI.pm" rel="external">pLI</a></td>
      <td>Adds the probability of a gene being loss-of-function intolerant (pLI).</td>
      <td></td>
      <td><kbd>--plugin pLI</kbd></td>
    </tr>
    <tr>
      <td><a href="https://github.com/Ensembl/VEP_plugins/blob/main/PolyPhen_SIFT.pm" rel="external">PolyPhen_SIFT</a></td>
      <td>Retrieves PolyPhen and SIFT predictions from a SQLite database.</td>
      <td><a href="https://ftp.ensembl.org/pub/current_variation/pangenomes/Human/homo_sapiens_pangenome_PolyPhen_SIFT_20240502.db"><kbd>homo_sapiens_pangenome_PolyPhen_SIFT_20240502.db</kbd></a></td>
      <td><kbd>--plugin PolyPhen_SIFT,db=homo_sapiens_pangenome_PolyPhen_SIFT_20240502.db</kbd></td>
    </tr>
    <tr>
      <td><a href="https://github.com/Ensembl/VEP_plugins/blob/main/ProteinSeqs.pm" rel="external">ProteinSeqs</a></td>
      <td>Writes two files with the reference and mutated protein sequences of any proteins found with non-synonymous mutations in the input file.</td>
      <td></td>
      <td><kbd>--plugin ProteinSeqs</kbd></td>
    </tr>
    <tr>
      <td><a href="https://github.com/Ensembl/VEP_plugins/blob/main/SingleLetterAA.pm" rel="external">SingleLetterAA</a></td>
      <td>Returns HGVSp string with single amino acid letter codes.</td>
      <td></td>
      <td><kbd>--plugin SingleLetterAA</kbd></td>
    </tr>
    <tr>
      <td><a href="https://github.com/Ensembl/VEP_plugins/blob/main/SpliceRegion.pm" rel="external">SpliceRegion</a></td>
      <td>Provides more granular predictions of splicing effects.</td>
      <td></td>
      <td><kbd>--plugin SpliceRegion</kbd></td>
    </tr>
    <tr>
      <td><a href="https://github.com/Ensembl/VEP_plugins/blob/main/SubsetVCF.pm" rel="external">SubsetVCF</a></td>
      <td>Retrieves overlapping records from a given VCF file.</td>
      <td>A VCF file</td>
      <td><kbd>--plugin SubsetVCF,file=file.vcf.gz,name=myvfc</kbd></td>
    </tr>
    <tr>
      <td><a href="https://github.com/Ensembl/VEP_plugins/blob/main/TranscriptAnnotator.pm" rel="external">TranscriptAnnotator</a></td>
      <td>Annotates variant-transcript pairs based on a given file.</td>
      <td>Tab-separated annotation file (<a href="vep_plugins.html#TranscriptAnnotator">plugin instructions</a>)</td>
      <td><kbd>--plugin TranscriptAnnotator,file=annotation.txt.gz</kbd></td>
    </tr>
    <tr>
      <td><a href="https://github.com/Ensembl/VEP_plugins/blob/main/TSSDistance.pm" rel="external">TSSDistance</a></td>
      <td>Calculates the distance from the transcription start site for upstream variants.</td>
      <td></td>
      <td><kbd>--plugin TSSDistance</kbd></td>
    </tr>
  </tbody></table>

  <hr/>

  <h2 id="citations">Citations and Ensembl VEP users</h2>
  
  <p> Ensembl VEP is used by many organisations and projects: </p>
  
  <ul>
    <li> Ensembl VEP forms a part of <a href="http://variantstudio.software.illumina.com/" rel="external">Illumina's VariantStudio</a> software </li>
    <li> <a href="http://gemini.readthedocs.org/" rel="external">Gemini</a> is a framework for exploring genome variation that uses Ensembl VEP </li>
    <li> The <a href="https://www.deciphergenomics.org/" rel="external">DECIPHER project</a> uses Ensembl VEP to aid variant interpretation</li>
  </ul>
  
  <p> Other citations and use cases: </p>
  
  <ul>
    <li> <a href="http://bib.oxfordjournals.org/content/early/2014/03/12/bib.bbu008.full" rel="external">VAX</a> is a suite of plugins that expand Ensembl VEP functionality</li>
    <li> <a href="https://academic.oup.com/bioinformatics/article/30/23/3408/207567" rel="external">pViz</a> is a visualisation tool for Ensembl VEP results files </li>
    <li> <a href="https://genomemedicine.biomedcentral.com/articles/10.1186/gm543" rel="external">McCarthy <i>et al</i></a> compares Ensembl VEP to AnnoVar </li>
    <li> <a href="http://bib.oxfordjournals.org/content/early/2013/01/21/bib.bbs086.full" rel="external">Pabinger <i>et al</i></a> reviews variant analysis software, including Ensembl VEP</li>
    <li> Ensembl VEP is used to provide annotation for the <a href="http://gnomad.broadinstitute.org/" rel="external">gnomAD</a> project</li>
  </ul>
</div>

</body>
</html>
