<html>
<head>
<title>Other information</title>
<meta name="order" content="9" />
<script type="text/javascript">

  // Function to show/hide divs
  function show_hide (param) {
    div   = document.getElementById('div_'+param);
    alink = document.getElementById('a_'+param);
    if (div.style.display=='inline') {
      div.style.display='none';
      alink.innerHTML='Show';
    }
    else {
      if (div.style.display=='none') {
        div.style.display='inline';
        alink.innerHTML='Hide';
      }
    }
  }
</script>
</head>
<body>
<div>
  
  <div style="float:right"><img src="/img/vep_logo.png"/></div>
  
  <h1 id="top"><span style="color:#006;padding-right:15px">Variant Effect Predictor</span><span style="color:#666"><img src="/i/16/info.png"/> Other information</span></h1>
  <hr/>
  
  <h2 id="faster">Getting VEP to run faster</h2>
  
  <p> Set up correctly, VEP is capable of processing around 3
  million variants in 30 minutes. There are a number of steps you can take to
  make sure your VEP installation is running as fast as possible: </p>
  
  <ol>
    <li>
      <p> Make sure you have the <img src="/i/16/download.png" style="vertical-align:bottom"/> <a href="vep_download.html#download">latest version</a> of
      VEP and the Ensembl API. We regularly introduce optimisations, alongside the
      new features and bug fixes of a typical new release. </p>
    </li>
    
    <li>
      <p> Download a <a href="vep_cache.html">cache file</a> for your species. If
      you are using <a href="vep_options.html#opt_database">--database</a>, you
      should consider using <a href="vep_options.html#opt_database">--cache</a> or
      <a href="vep_options.html#opt_database">--offline</a> instead. Any time
      VEP has to access data from the database (even if you have a local copy),
      it will be slower than accessing data in the cache on your local file system. </p>
      
      <p> Enabling <a href="vep_cache.html#limitations">certain flags</a>
      forces VEP to access the database, and you will be warned at
      startup that it will do this with e.g.:</p>
      
      <pre class="code sh_sh">2011-06-16 16:24:51 - INFO: Database will be accessed when using --check_svs</pre>
      
      <p> Consider carefully whether you need to use these flags in your analysis.</p>
    </li>
    
    <li>
      <p> If you use <a href="vep_options.html#opt_check_existing">--check_existing</a>
      or any flags that invoke it (e.g. <a href="vep_options.html#opt_af">--af</a>,
      <a href="vep_options.html#opt_af_1kg">--af_1kg</a>, <a
      href="vep_options.html#opt_filter_common">--filter_common</a>, <a
      href="vep_options.html#opt_everything">--everything</a>), <a
      href="vep_cache.html#convert">tabix-convert</a> your cache file. Checking for known
      variants using a converted cache is >100% faster than using the default format. </p>
    </li>
    
    <li>
      <p> Download a <a href="vep_cache.html#fasta">FASTA file</a> (and use the flag <a href="vep_options.html#opt_fasta">--fasta</a>) if you use
      <a href="vep_options.html#opt_hgvs">--hgvs</a> or <a
      href="vep_options.html#opt_check_ref">--check_ref</a>. Again, this will prevent
      VEP accessing the database unnecessarily (in this case to retrieve genomic
      sequence). </p>
    </li>
    
    <li id="forking">
      <p> Using forking enables VEP to run multiple parallel "threads", with
      each thread processing a subset of your input. Most modern computers have
      more than one processor core, so running VEP with forking enabled can
      give huge speed increases (3-4x faster in most cases). Even computers with a
      single core will see speed benefits due to overheads associated with using
      object-oriented code in Perl. </p>
  
      <p> To use forking, you must choose a number of forks to use with the <a
      href="vep_options.html#opt_fork">--fork</a> flag. We recommend using 4 forks: </p>
  
      <pre class="code sh_sh">./vep -i my_input.vcf --fork 4 --offline</pre>
  
      <p> but depending on various factors specific to your setup you may see
      faster performance with fewer or more forks. </p>
      
      <p> When writing <a href="vep_plugins.html">plugins</a>
      be aware that while the VEP code attempts to preserve the state of
      any plugin-specific cached data between separate forks, there may be
      situations where data is lost. If you find this is the case, you should
      disable forking in the new() method of your plugin by deleting the "fork"
      key from the $config hash. </p>
    </li>
    
    <li>
      <p> Make sure your cache and FASTA files are stored on the fastest file system
      or disk you have available. If you have a lot of memory in your machine, you can
      even pre-copy the files to memory using <a
      href="https://www.howtoforge.com/storing-files-directories-in-memory-with-tmpfs"
      rel="external">tmpfs</a>. </p>
    </li>
    
    <li>
      <p> Consider if you need to generate HGVS notations (<a
      href="vep_options.html#opt_hgvs">--hgvs</a>); this is a complex annotation step
      that can add ~50-80% to your runtime. Note also that --hgvs is switched on by
      <a href="vep_options.html#opt_everything">--everything</a>.
    </li>
    
    <li>
      <p> Install the <a
      href="http://search.cpan.org/~benbooth/Set-IntervalTree/lib/Set/IntervalTree.pm"
      rel="external">Set::IntervalTree</a> Perl package. This package speeds up
      VEP's internals by changing how overlaps between variants and transcript components
      are calculated. </p>
    </li>
    
    <li>
      <p> Install the <a href="https://github.com/Ensembl/ensembl-xs" rel="external">
      Ensembl::XS</a> package. This contains compiled versions of certain key
      subroutines used in VEP that will run faster than the default native Perl
      equivalents. Using this should improve runtime by 5-10%. </p>
    </li>
    
    <li>
      <p> Add the <a href="vep_options.html#opt_no_stats">--no_stats</a> flag.
      Calculating summary statistics increases VEP runtime, so can be switched off if not required
      </p>
    </li>
    
    <li>
      <p> VEP is optimised to run on input files that are sorted in
      chromosomal order. Unsorted files will still work, albeit more slowly. </p>
    </li>
    
    <li>
      <p> For very large files (for example those from whole-genome sequencing),
      VEP process can be easily parallelised by dividing your file into chunks
      (e.g. by chromosome). VEP will also work with tabix-indexed, bgzipped
      VCF files, and so the tabix utility could be used to divide the input file:
      </p>
  
      <pre class="code sh_sh"> tabix -h variants.vcf.gz 12:1000000-20000000 | ./vep --cache --vcf </pre>
    </li>
  </ol>
  

  
  
  <hr/>
  <h2 id="assembly">Species with multiple assemblies</h2>
  
  <p> Ensembl currently supports the two latest human assembly
  versions. We provide a VEP cache using the latest software version
  ([[SPECIESDEFS::ENSEMBL_VERSION]]) for both GRCh37 and GRCh38. </p>
  
  <p> The <a href="vep_download.html#installer">VEP installer</a> will install
  and set up the correct cache and FASTA file for your assembly of interest. If
  using the --AUTO functionality to install without prompts, remember to add the
  assembly version required using e.g. "--ASSEMBLY GRCh37". It is also possible
  to have concurrent installations of caches from both assemblies; just use the
  <a href="vep_options.html#opt_assembly">--assembly</a> to select the correct
  one when you run VEP. </p>
  
  <p> Once you have installed the relevant cache and FASTA file, you are then
  able to use VEP as normal. If you are using GRCh37 and require database
  access in addition to the cache (for example, to look up variant identifiers
  using <a href="vep_options.html#opt_format">--format id</a>, see <a
  href="vep_cache.html#limitations">cache limitations</a>), you will be warned
  you that you must change the database port in order to connect to the correct
  database: </p>
  
  <pre class="code sh_sh">ERROR: Cache assembly version (GRCh37) and database or selected assembly version (GRCh38) do not match

If using human GRCh37 add "--port 3337" to use the GRCh37 database, or --offline to avoid database connection entirely</pre>
  
  <p> If you have data you wish to map to a new assembly, you can use the Ensembl
  assembly converter tool - if you've downloaded VEP, then you have it
  already! The tool is found in the ensembl-tools/scripts/assembly_converter
  folder. There is also an <a href="/info/docs/tools/index.html">online version
  of the tool</a> available. Both UCSC (<a
  href="https://genome.ucsc.edu/util.html" rel="external">liftOver</a>) and NCBI
  (<a href="http://www.ncbi.nlm.nih.gov/genome/tools/remap"
  rel="external">Remap</a>) also provide tools for converting data between
  assemblies. </p>
  
  <hr/>
    
  <h2 id="pick">Summarising annotation</h2>
  
  <p> By default VEP is configured to provide annotation on every genomic
  feature that each input variant overlaps. This means that if a variant
  overlaps a gene with multiple alternate splicing variants (transcripts), then
  a block of annotation for each of these transcripts is reported in the output.
  In the <a href="../vep_formats.html#output">default VEP output format</a> each
  of these blocks is written on a single line of output; in
  <a href="../vep_formats.html#vcfout">VCF output format</a> the blocks are
  separated by commas in the INFO field. </p>
  
  <p> A number of options are provided to reduce the amount of output produced if this
  depth of annotation is not required.
  </p>

  <div style="width:auto" class="info">

    <h3 id="limitations">Note</h3>

    <div class="message-pad">
      <p>Wherever possible we would discourage you from summarising
      data in this way. Summarising inevitably involves data loss, and invariably at
      some point this will lead to the loss of biologically relevant information. For
      example, if your variant overlaps both a regulatory feature and a transcript
      and you use one of the flags below, the overlap with the regulatory feature
      will be lost in your output, when in some cases this may be a clue to the
      "real" functional effect of your variant. For these reasons we recommend
      using one of the flagging options
      (<a href="vep_options.html#opt_flag_pick">--flag_pick</a>,
      <a href="vep_options.html#opt_flag_pick_allele">--flag_pick_allele</a> or
      <a href="vep_options.html#opt_flag_pick_allele_gene">--flag_pick_allele_gene</a>) and to
      <img src="/i/16/search.png" style="vertical-align:bottom"/> <a href="vep_filter.html">post-filter results</a>.
    </div>
  </div>

  <br />
  <h4>Example</h4>

  <p>Input data (VCF - input.vcf)</p>
  <pre class="code sh_sh">##fileformat=VCFv4.2
#CHROM  POS  ID  REF  ALT
1  230710048  rs699  A  G
1  230710514  var_2  A  G,T</pre>

  <p>Example of VEP command and output (no "pick" option):</p>
  <pre class="code sh_sh">./vep --cache -i input.vcf -o output.txt

#Uploaded_variation  Location  Allele  Gene  Feature  Feature_type  Consequence  cDNA_position  CDS_position  Protein_position  Amino_acids  Codons  Existing_variation  Extra
rs699  1:230710048  G  ENSG00000135744  ENST00000366667  Transcript  missense_variant         1018  803  268  M/T  aTg/aCg  -  IMPACT=MODERATE;STRAND=-1
rs699  1:230710048  G  ENSG00000244137  ENST00000412344  Transcript  downstream_gene_variant     -    -    -    -        -  -  IMPACT=MODIFIER;DISTANCE=650;STRAND=-1
var_2  1:230710514  G  ENSG00000135744  ENST00000366667  Transcript  synonymous_variant        552  337  113    L  Ttg/Ctg  -  IMPACT=LOW;STRAND=-1
var_2  1:230710514  T  ENSG00000135744  ENST00000366667  Transcript  missense_variant          552  337  113  L/M  Ttg/Atg  -  IMPACT=MODERATE;STRAND=-1
var_2  1:230710514  G  ENSG00000244137  ENST00000412344  Transcript  downstream_gene_variant     -    -    -    -        -  -  IMPACT=MODIFIER;DISTANCE=184;STRAND=-1
var_2  1:230710514  T  ENSG00000244137  ENST00000412344  Transcript  downstream_gene_variant     -    -    -    -        -  -  IMPACT=MODIFIER;DISTANCE=184;STRAND=-1</pre>

  <br />
  <h4 id="pick_options">Options</h4>
  <ul>
    <li style="margin-bottom:20px">
      <b>--pick</b></p>
      <p>This is the option
      we anticipate will be of use. VEP chooses one block of
      annotation per variant, using an ordered set of criteria. This order may be
      customised using <a href="vep_options.html#opt_pick_order">--pick_order</a>.
      <ol>
        <li>canonical status of transcript</li>
        <li><a href="/Help/Glossary?id=521">APPRIS isoform annotation</a></li>
        <li><a href="/Help/Glossary?id=492">transcript support level</a></li>
        <li>biotype of transcript ("protein_coding" preferred)</li>
        <li>CCDS status of transcript</li>
        <li>consequence rank according to <a href="/info/genome/variation/prediction/predicted_data.html#consequences">this table</a></li>
        <li>translated, transcript or feature length (longer preferred)</li>
        <li>MANE transcript status</li>
      </ol>
  
      <div>    
        <div style="float:left" class="info">
          <h3 id="limitations">Note</h3>
          <div class="message-pad">
           <p>Some categories may not be available for the species or cache
            version that you are using; in these cases the category will be skipped and
            the next in line used.</p>
          </div>
        </div>
        <div class="clear"></div>
      </div>
      <p><a class="button" href="#pick_eg" onclick="show_hide('pick_eg');" id="a_pick_eg" style="font-weight:normal">Show</a>example of VEP command and output, with the "<b>--pick</b>" option.</p>
      <div id="div_pick_eg" style="display:none">
        <pre class="code sh_sh">./vep --cache -i input.vcf -o output.txt --pick

rs699  1:230710048  G  ENSG00000135744  ENST00000366667  Transcript  missense_variant  1018  803  268  M/T  aTg/aCg  -  IMPACT=MODERATE;STRAND=-1
var_2  1:230710514  T  ENSG00000135744  ENST00000366667  Transcript  missense_variant   552  337  113  L/M  Ttg/Atg  -  IMPACT=MODERATE;STRAND=-1</pre>
      </div>
    </li>

    <li style="margin-bottom:20px">
      <p><b>--pick_allele</b></p>
      <p>As above, but chooses one consequence block per variant allele. This can be useful for <a href="../vep_formats.html#vcf">VCF input files</a> with more than one ALT allele.</p>

      <p><a class="button" href="#pick_allele_eg" onclick="show_hide('pick_allele_eg');" id="a_pick_allele_eg" style="font-weight:normal">Show</a> example of VEP command and output, with the "<b>--pick_allele</b>" option.</p>
      <div id="div_pick_allele_eg" style="display:none">
        <pre class="code sh_sh">./vep --cache -i input.vcf -o output.txt --pick_allele

rs699  1:230710048  G  ENSG00000135744  ENST00000366667  Transcript  missense_variant   1018  803  268  M/T  aTg/aCg  -  IMPACT=MODERATE;STRAND=-1
var_2  1:230710514  T  ENSG00000135744  ENST00000366667  Transcript  missense_variant    552  337  113  L/M  Ttg/Atg  -  IMPACT=MODERATE;STRAND=-1
var_2  1:230710514  G  ENSG00000135744  ENST00000366667  Transcript  synonymous_variant  552  337  113    L  Ttg/Ctg  -  IMPACT=LOW;STRAND=-1</pre>
      </div>
    </li>

    <li style="margin-bottom:20px">
      <p><b>--per_gene</b></p>
      <p>As <a href="vep_options.html#opt_pick">--pick</a>, but chooses one annotation block per gene that the input variant overlaps.</p>

      <p><a class="button" href="#per_gene_eg" onclick="show_hide('per_gene_eg');" id="a_per_gene_eg" style="font-weight:normal">Show</a> example of VEP command and output, with the "<b>--per_gene</b>" option.</p>
      <div id="div_per_gene_eg" style="display:none">
        <pre class="code sh_sh">./vep --cache -i input.vcf -o output.txt --per_gene

rs699  1:230710048  G  ENSG00000135744  ENST00000366667  Transcript  missense_variant         1018  803  268  M/T  aTg/aCg  -  IMPACT=MODERATE;STRAND=-1
rs699  1:230710048  G  ENSG00000244137  ENST00000412344  Transcript  downstream_gene_variant     -    -    -    -        -  -  IMPACT=MODIFIER;DISTANCE=650;STRAND=-1
var_2  1:230710514  G  ENSG00000244137  ENST00000412344  Transcript  downstream_gene_variant     -    -    -    -        -  -  IMPACT=MODIFIER;DISTANCE=184;STRAND=-1
var_2  1:230710514  T  ENSG00000135744  ENST00000366667  Transcript  missense_variant          552  337  113  L/M  Ttg/Atg  -  IMPACT=MODERATE;STRAND=-1</pre>
      </div>
    </li>

    <li style="margin-bottom:20px">
      <p><b>--pick_allele_gene</b></p>
      <p>As above, but chooses one consequence block per variant allele and gene combination.</p>

      <p><a class="button" href="#pick_allele_gene_eg" onclick="show_hide('pick_allele_gene_eg');" id="a_pick_allele_gene_eg" style="font-weight:normal">Show</a> example of VEP command and output, with the "<b>--pick_allele_gene</b>" option.</p>
      <div id="div_pick_allele_gene_eg" style="display:none">
        <pre class="code sh_sh">./vep --cache -i input.vcf -o output.txt --pick_allele_gene

rs699  1:230710048  G  ENSG00000244137  ENST00000412344  Transcript  downstream_gene_variant     -    -    -    -        -  -  IMPACT=MODIFIER;DISTANCE=650;STRAND=-1
rs699  1:230710048  G  ENSG00000135744  ENST00000366667  Transcript  missense_variant         1018  803  268  M/T  aTg/aCg  -  IMPACT=MODERATE;STRAND=-1
var_2  1:230710514  T  ENSG00000244137  ENST00000412344  Transcript  downstream_gene_variant     -    -    -    -        -  -  IMPACT=MODIFIER;DISTANCE=184;STRAND=-1
var_2  1:230710514  T  ENSG00000135744  ENST00000366667  Transcript  missense_variant          552  337  113  L/M  Ttg/Atg  -  IMPACT=MODERATE;STRAND=-1
var_2  1:230710514  G  ENSG00000135744  ENST00000366667  Transcript  synonymous_variant        552  337  113    L  Ttg/Ctg  -  IMPACT=LOW;STRAND=-1
var_2  1:230710514  G  ENSG00000244137  ENST00000412344  Transcript  downstream_gene_variant     -    -    -    -        -  -  IMPACT=MODIFIER;DISTANCE=184;STRAND=-1</pre>
      </div>
    </li>

    <li style="margin-bottom:20px">
      <p><b>--flag_pick</b></p>
      <p>Instead of choosing one block and removing the others, this option adds a flag &quot;PICK=1&quot; to picked annotation block, allowing you to easily filter on this later using VEP's <a href="vep_filter.html">filtering tool</a>.</p>
    </li>

    <li style="margin-bottom:20px">
      <p><b>--flag_pick_allele</b></p>
      <p>As above, but flags one block per allele.</p>
    </li>

    <li style="margin-bottom:20px">
      <p><b>--flag_pick_allele_gene</b></p>
      <p>As above, but flags one block per allele and gene combination.</p>
    </li>

    <li style="margin-bottom:20px">
      <p><b>--most_severe</b></p>
      <p>This flag reports only the consequence type of the block with the highest rank, according to <a href="/info/genome/variation/prediction/predicted_data.html#consequences">this table</a>.</p>

      <p><a class="button" href="#most_severe_eg" onclick="show_hide('most_severe_eg');" id="a_most_severe_eg" style="font-weight:normal">Show</a> example of VEP command and output, with the "<b>--most_severe</b>" option.</p>
      <div id="div_most_severe_eg" style="display:none">
        <pre class="code sh_sh">./vep --cache -i input.vcf -o output.txt --most_severe

rs699  1:230710048  -  -  -  -  missense_variant  -  -  -  -  -  -  -
var_2  1:230710514  -  -  -  -  missense_variant  -  -  -  -  -  -  -</pre>
      </div>
    </li>

    <li style="margin-bottom:20px">
      <p><b>--summary</b></p>
      <p>This flag reports only a comma-separated list of the consequence types predicted for this variant.</p>

      <p><a class="button" href="#summary_eg" onclick="show_hide('summary_eg');" id="a_summary_eg" style="font-weight:normal">Show</a> example of VEP command and output, with the "<b>--summary</b>" option.</p>
      <div id="div_summary_eg" style="display:none">
        <pre class="code sh_sh">./vep --cache -i input.vcf -o output.txt --summary

rs699  1:230710048  -  -  -  -  missense_variant,downstream_gene_variant                     -  -  -  -  -  -  -
var_2  1:230710514  -  -  -  -  missense_variant,synonymous_variant,downstream_gene_variant  -  -  -  -  -  -  -</pre>
      </div>
    </li>
  </ul>
  
  <div>
    <div style="float:left" class="warning">
      <h3 id="limitations">Caution</h3>
      <div class="message-pad">
        <p>Feature-specific annotation is absent from the output using the flags <b>--most_severe</b> or <b>--summary</b>, so use them with caution!</p>
      </div>
    </div>
    <div class="clear"></div>
  </div>
  <br />

  <hr/>
    
  <h2 id="hgvs">HGVS notations</h2>

  <h4>Output</h4>

  <p> <a rel="external" href="//varnomen.hgvs.org">HGVS</a>
  notations can be produced by VEP using the
  <a href="vep_options.html#opt_hgvs">--hgvs</a> flag. Coding (c.) and protein (p.)
  notations given against Ensembl identifiers use
  <a href="/info/genome/stable_ids/index.html">versioned</a> identifiers that
  guarantee the identifier refers always to the same sequence. </p>

  <p> Genomic HGVS notations may be reported using <a href="vep_options.html#opt_hgvsg">--hgvsg</a>. Note that the named reference for HGVSg notations will be the chromosome name from the input (as opposed to the officially recommended chromosome accession). </p>

  <p> HGVS notations for insertions or deletions are by default shifted 3-prime relative to the reported transcript or protein sequence in accordance with HGVS specifications. This may lead to discrepancies between the coordinates reported in the HGVS nomenclature and the coordinate columns reported by VEP. You may instruct VEP not to shift using <a href="vep_options.html#opt_shift_hgvs">--shift_hgvs 0</a>. </p>

  <br />
  <h4>Input</h4>
  
  <p> VEP supports using HGVS notations as input. This feature is
  currently under development and not all HGVS notation types are supported.
  Notations relative to genomic (g.) or coding (c.) sequences
  are fully supported; protein (p.) notations are supported in limited
  fashion due to the complexity involved in determining the multiple possible
  underlying genomic sequence changes that could produce a single protein
  change. A warning will be given if a particular notation cannot be parsed.
  </p>
  
  <p> By default VEP uses Ensembl transcripts as the reference for
  determining consequences, and hence also for HGVS notations. However, it is
  possible to parse HGVS notations that use RefSeq transcripts as the
  reference sequence by using the <a
  href="vep_options.html#opt_refseq">--refseq</a> flag.
  Such notations must include the version number of the transcript e.g. </p>
 
  <pre class="code sh_sh" style="display:inline-block">NM_080794.3:c.1001C>T</pre>
  
  <p> where ".3" denotes that this is version 3 of the transcript NM_080794.
  <a href="#refseq">See below</a> for more details on how VEP can use
  RefSeq transcripts. </p>
  
  <hr/>
  <h2 id="refseq">RefSeq transcripts</h2>
  

  <p>If you prefer to exclude predicted RefSeq transcripts (those with identifiers beginning with "XM_" or "XR_") use <a href="vep_options.html#opt_exclude_predicted">--exclude_predicted</a>.</p>

  <br />
  <h3 id="refseq_limitations">Identifiers and other data</h3>

  <p> VEP's RefSeq cache lacks many classes of data present in the Ensembl transcript cache. </p>

  <ul>
    <li>
      Included in the RefSeq cache
      <ul>
        <li>Gene symbol</li>
        <li>SIFT and PolyPhen predictions</li>
      </ul>
    </li>
    <li>
      <b>Not</b> included in the RefSeq cache
      <ul>
        <li>APPRIS annotation</li>
        <li>TSL annotation</li>
        <li>UniProt identifiers</li>
        <li>CCDS identifiers</li>
        <li>Protein domains</li>
        <li>Gene-phenotype association data</li>
      </ul>
    </li>
  </ul>

  <br />
  <h3>Differences to the reference genome</h3>
  
  <p> RefSeq transcript sequences may differ from the genome
  sequence to which they are aligned. Ensembl's API (and hence VEP) constructs
  transcript models using the genomic reference sequence.
  These differences are accounted for using <a href="#refseq_bam">BAM-edited transcript models</a>.
  in human cache files from release 90 onwards.
  Prior to release 90 and in non-human species differences between the
  RefSeq sequence and the genomic sequence are not accounted for,
  so some annotations produced by VEP on these transcripts may be inaccurate.
  Most differences occur in non-coding regions, typically in UTRs at either end of transcripts or in the addition of a poly-A tail,
  causing minimal impact on annotation.</p>

  <p> For human VEP cache files, each RefSeq transcript is annotated with the <a href="../vep_formats.html#refseq_match">REFSEQ_MATCH</a> flag
  indicating whether and how the RefSeq model differs from the underlying genome. </p>

  <br />
  <h3 id="refseq_bam">Correcting transcript models with BAM files</h3>

  <p> NCBI have released BAM files that contain alignments of RefSeq transcripts to the genome. From release 90 onwards, these alignments
  have been incorporated and used to correct the transcript models in the human RefSeq and merged cache files. </p>

  <p>VEP's cache building process uses the sequence and alignment in the BAM to correct the RefSeq model. If the corrected model does not match the original RefSeq sequence in the BAM, the corrected model is discarded. The success or failure of the BAM edit is recorded in the BAM_EDIT field of the VEP output. Failed edits are extremely rare (&lt; 0.01% of transcripts), but any VEP annotations produced on transcripts with a failed edit status should be interpreted with extreme caution.</p>

  <p>Using BAM-edited transcripts causes VEP to change how alleles are interpreted from input variants.
  Input variants are typically encoded in VCFs that are called using the reference genome.
  This means that the alternate (ALT) allele as given in the VCF may correspond to the reference allele as found in the corrected RefSeq transcript model.
  VEP will account for this, using the corrected reference allele (by enabling <a href="vep_options.html#opt_use_transcript_ref">--use_transcript_ref</a>) when calculating consequences, and the GIVEN_REF and USED_REF fields in the VEP output indicate any change made. If the reference allele derived from the transcript matches any given alternate (ALT) allele, then no consequence data will be produced for this allele as it will be considered non-variant.
  Note that this process may also clash with any interpretation from using <a href="vep_options.html#opt_check_ref">--check_ref</a>, so it is recommended to avoid using this flag. </p>

  <p> To override the behaviour of <a href="vep_options.html#opt_use_transcript_ref">--use_transcript_ref</a> and force VEP to use your input reference allele instead of the one derived from the transcript,
  you may use <a href="vep_options.html#opt_use_given_ref">--use_given_ref</a>.</p>

  <p>VEP can also side-load BAM files at runtime to correct transcript models on-the-fly; this may be used, for example, with pre-release 90 cache file or a RefSeq GFF files. </p>

  <pre class="code sh_sh">./vep --cache --refseq -i variants.vcf --bam GRCh38.p12_knownrefseq_alignments.bam</pre>

  <p>BAM files are available from NCBI:</p>

  <ul>
    <li><a href="ftp://ftp.ncbi.nlm.nih.gov/genomes/H_sapiens/ARCHIVE/ANNOTATION_RELEASE.109/GRCh38_interim_annotation/" rel="external">Human GRCh38.p12</a></li>
    <li><a href="ftp://ftp.ncbi.nlm.nih.gov/genomes/Homo_sapiens/ARCHIVE/ANNOTATION_RELEASE.109/GRCh37.p13_interim_annotation/" rel="external">Human GRCh37.p13</a></li>
  </ul>

  <div>
    <div style="float:left" class="info">
      <h3 id="limitations">Note for the GRCh37 bam file</h3>
      <div class="message-pad">
        <p>The BAM index files (.bai) in this directory are required and will need to be renamed as the perl library used to parse the files expects the index to be named [indexed_bam_file].bai:</p>
        <pre class="code sh_sh">mv interim_GRCh37.p13_knownrefseq_alignments_2017-01-13.bai interim_GRCh37.p13_knownrefseq_alignments_2017-01-13.bam.bai</pre>
      </div>
    </div>
    <div class="clear"></div>
  </div>
  
  <hr/>
  <h2 id="colocated">Existing or colocated variants</h2>

  <p> Use the <a href="vep_options.html#opt_check_existing">--check_existing</a> flag to identify known variants colocated with input variant. VEP's known variant cache is derived from Ensembl's variation database and contains variants from dbSNP and <a href="/info/genome/variation/species/sources_documentation.html">other sources</a>. </p>

  <p> VEP by default uses a normalisation-based allele matching algorithm to identify known variants that match input variants. Since both input and known variants may have multiple alternate (ALT) or variant alleles, each pair of reference (REF) and ALT alleles are normalised and compared independently to arrive at potential matches. VCF permits multiple allele types to be encoded on the same line, while dbSNP assigns separate rsID identifiers to different allele types at the same locus. This means different alleles from the same input variant may be assigned different known variant identifiers. </p>

  <p><img src="VEP_allele_matching_algorithm.png"/><br/><span class="small"><b>Illustration of VEP's allele matching algorithm resolving one VCF line with multiple ALTs to three different variant types and coordinates</b></span></p>

  <p> Note that allele matching occurs independently of any allele transformations carried out by <a href="vep_options.html#opt_minimal">--minimal</a>; VEP will match to the same identifiers and frequency data regardless of whether the flag is used. </a>

  <p> For some data sources (COSMIC, HGMD), Ensembl is not licensed to redistribute allele-specific data, so VEP will report the existence of co-located variants with unknown alleles <b>without</b> carrying out allele matching. To disable this behaviour and exclude these variants, use the <a href="vep_options.html#opt_exclude_null_alleles">--exclude_null_alleles</a> flag. </p>

  <p> To disable allele matching completely and compare variant locations only, use <a href="vep_options.html#opt_no_check_alleles">--no_check_alleles</a>. </p>

  <br />
  <h3 id="freq">Frequency data</h3>

  <p> In addition to identifying known variants, VEP also reports allele frequencies for input alleles from major genotyping projects (<a href="vep_options.html#opt_af_1kg">1000 genomes</a>, <a href="vep_options.html#opt_af_esp">ESP</a> and <a href="vep_options.html#opt_af_gnomad">gnomAD</a>). VEP's cache currently contains only frequency data for alleles that have been submitted to dbSNP or are imported via <a href="/info/genome/variation/sources_documentation.html">another source</a> into the Ensembl variation database. This means that until gnomAD's full data set is submitted to dbSNP and incorporated into Ensembl, the frequency for some alleles may be missing from VEP's cache data. </p>

  <p> To access the full gnomAD data set, it is possible to use VEP's custom annotation feature to retrieve the frequency data directly from the gnomAD VCF files; see <a href="vep_example.html#gnomad">instructions here</a>. </p>
  
</div>

</body>
</html>
