<html>
<head>
  <title>Custom annotations</title>
  <meta name="order" content="6" />
  <script type="text/javascript">
    // Function to show/hide divs
    function show_hide (param) {
      var div = document.getElementById('div_'+param);
      var img = document.getElementById('img_'+param);
      var img_path = img.src;
      if (div.style.display=='inline') {
        var src_path = img_path.replace('minus','plus');
        img.src = src_path;
        div.style.display='none';
      }
      else {
        if (div.style.display=='none') {
          var src_path = img_path.replace('plus','minus');
          img.src = src_path;
          div.style.display='inline';
        }
      }
    }
  </script>
</head>
<body>
<div>
  
  <div style="float:right"><img src="/img/vep_logo.png"/></div>
  
  <h1 id="top"><span style="color:#006;padding-right:15px">Variant Effect Predictor</span><span style="color:#666"><img src="/i/16/page-user.png"> Custom annotations</span></h1>
  <hr/> 
        
  <p> VEP can integrate custom annotation from standard format
  files into your results by using the <a href="vep_options.html#opt_custom">--custom</a> flag. </p>
    
  <p> These files may be hosted locally or remotely, with no limit to the
  number or size of the files. The files must be indexed using the <a
  rel="external" href="http://samtools.sourceforge.net/tabix.shtml" target="_blank">tabix</a>
  utility (BED, GFF, GTF, VCF); bigWig files contain their own indices. </p>
  
  <p> Annotations typically appear as key=value pairs in the Extra column of the VEP
  output; they will also appear in the INFO column if using VCF format output.
  The value for a particular annotation is defined as the identifier for each
  feature; if not available, an identifier derived from the coordinates of the
  annotation is used. Annotations will appear in each line of output for the
  variant where multiple lines exist. </p>
  
  <h2 id="custom_formats">Data formats</h2>
  
  <p> VEP supports the following annotation formats: </p>
  
  <style> tr:nth-child(2n + 1) { background-color: #f0f0f0; } </style>
  <table class="ss"><tbody>
    <tr>
      <th>Format</th>
      <th>Type</th>
      <th>Description</th>
      <th>Notes</th>
    </tr>
    <tr id="opt_gff">
      <td><a href="vep_cache.html#gff">GFF<br>GTF</a></td>
      <td>Gene/transcript annotations</td>
      <td>Formats to describe genes and other genomic features &mdash; format specifications: <a rel="external" href="https://github.com/The-Sequence-Ontology/Specifications/blob/master/gff3.md">GFF3</a> and <a href="/info/website/upload/gff.html">GTF</a></td>
      <td>Requires a <b><a href="vep_cache.html#fasta">FASTA</a></b> file in offline mode or if the desired species or assembly is not part of the <a href="/info/about/species.html">Ensembl species list</a>.</td>
    </tr>
    <tr id="opt_vcf">
      <td><a rel="external" href="http://www.1000genomes.org/wiki/Analysis/Variant%20Call%20Format/vcf-variant-call-format-version-41" target="_blank">VCF</a></td>
      <td>Variant data</td>
      <td>A format used to describe genomic variants</td>
      <td>VEP uses the 3rd column as the identifier. INFO and FILTER fields from
          records may be added to the VEP output.</td>
    </tr>
    <tr id="opt_bed">
      <td><a href="/info/website/upload/bed.html" target="_blank">BED</a></td>
      <td>Basic/uninterpreted data</td>
      <td>A simple tab-delimited format containing 3-12 columns of data.
          The first 3 columns contain the coordinates of the feature.</td>
      <td>VEP uses the 4th column (if available) as the feature identifier.</td>
    </tr>
    <tr id="opt_bigWig">
      <td><a rel="external" href="http://genome.ucsc.edu/goldenPath/help/bigWig.html"
        target="_blank">bigWig</a></td>
      <td>Basic/uninterpreted data</td>
      <td>A format for storage of dense continuous data.</td>
      <td>VEP uses the value for the given position as the identifier.
          BigWig files contain their own indices, and do not need to be indexed by tabix.
          Requires <a href="vep_download.html#bigfile">Bio::DB::BigFile</a>.
      </td>
    </tr>
  </tbody></table>

  <p> Any other files can be easily converted to be compatible with VEP;
  the easiest format to produce is a BED-like file containing coordinates and
  an (optional) identifier: </p>
  
  <pre class="code sh_sh">chr1    10000    11000    Feature1
chr3    25000    26000    Feature2
chrX    99000    99001    Feature3</pre>
  
  <p> Chromosomes can be denoted by either e.g. "chr7" or "7", "chrX" or "X". </p>
  
    <hr/>
  <h2 id="custom_prep">Preparing files</h2>
  
  <p> Custom annotation files must be prepared in a particular way in order to
  work with tabix and therefore with VEP. Files must be stripped of comment lines,
  sorted in chromosome and position order, compressed using bgzip
  and finally indexed using tabix. Here are some examples of that process for:</p>
  
  <ul>
    <li><b>GFF file</b>
      <pre class="code sh_sh" style="margin:5px 0px 10px">grep -v "#" myData.gff | sort -k1,1 -k4,4n -k5,5n -t$'\t' | bgzip -c > myData.gff.gz
tabix -p gff myData.gff.gz</pre>
      <div>
        <div class="warning" style="float:left">
          <h3>Note</h3>
          <div class="message-pad">
            <p>VEP expects few extra fields in the 9th column of the GFF file. See the <a href="vep_cache.html#gff_req">extra fields list</a>.</p>
          </div>
        </div>
        <div style="clear:both"></div>
      </div>
    </li>
    <li><b>BED file</b>
  <pre class="code sh_sh" style="margin-top:5px">grep -v "#" myData.bed | sort -k1,1 -k2,2n -k3,3n -t$'\t' | bgzip -c > myData.bed.gz
tabix -p bed myData.bed.gz</pre>
  </ul>  

  <p> The tabix utility has several preset filetypes that it can process, and
  it can also process any arbitrary filetype containing at least a chromosome
  and position column. See the <a
  rel="external" href="http://samtools.sourceforge.net/tabix.shtml"
  target="_blank">documentation</a> for details. </p>
  
  <p> If you are going to use the file remotely (i.e. over HTTP or FTP
  protocol), you should ensure the file is world-readable on your server. </p>
  
  <hr/>
  <h2 id="custom_options">Options</h2>

<div class="navbar" onclick="show_hide('custom_compatibility');" id="custom_compatibility" style="padding-left:5px;margin-bottom:0px;cursor:pointer"><img id="img_custom_compatibility" src="/i/16/plus-button.png" style="vertical-align:middle"/> <span style="position:absolute;left:28px;top:9px">Using positional options in <kbd>--custom</kbd> with VEP 109 and earlier (compatible with VEP [[SPECIESDEFS::ENSEMBL_VERSION]])</span></div>
  <div id="div_custom_compatibility" style="display:none">

  <br/>
  <p> Each custom file that you configure VEP to use can be configured.
  Beyond the filepath, there are further options, each of which is specified
  in a comma-separated list, like this: </p>

  <pre class="code sh_sh">
./vep [...] --custom <b>Filename</b>,<b>Short_name</b>,<b>File_type</b>,<b>Annotation_type</b>,<b>Force_report_coordinates</b>,<b>VCF_fields</b></pre>

  <p> The options are as follows: </p>
  
  <ul>
    <li><b>Filename</b> :
    <div style="margin:5px 0px 12px">The path to the file. For tabix indexed files, the
    VEP will check that both the file and the corresponding .tbi file exist.
    For remote files, VEP will check that the tabix index is accessible on startup.</div></li>

    <li><b>Short name</b> : 
    <div style="margin:5px 0px 12px">A name for the annotation that will appear as
    the key in the key=value pairs in the results.<br />If not defined, this will 
    default to the annotation filename for the first set of annotation added (e.g. "myPhenotypes.bed.gz" in the second example below if the short name was missing).</div></li>

    <li><b>File type</b> :
      <div style="margin:5px 0px 12px">
        <div style="float:left">
          <pre class="code sh_sh" style="margin:4px 0px">"bed", "gff", "gtf", "vcf" or "bigwig"</pre>
        </div>
        <div class="clear"></div>
      </div>
    </li>

    <li><b>Annotation type</b> :
      <div style="margin:5px 0px 12px">
        <div>
          <div style="float:left">
            <pre class="code sh_sh" style="margin:4px 0px">"exact" or "overlap"</pre>
          </div>
          <div style="float:left;padding:8px 10px;margin:4px 0px">
            (if left blank, assumed to be <b>overlap</b>) 
          </div>
          <div class="clear"></div>
        </div>
        <div>
          When using "<span style="color:blue">exact</span>" only annotations whose coordinates match exactly those of the
          variant will be reported. This would be suitable for position specific
          information such as conservation scores, allele frequencies or phenotype
          information. Using "<span style="color:blue">overlap</span>", any annotation that overlaps the variant
          by even 1bp will be reported.
        </div>
      </div>
    </li>
    
    <li><b>Force report coordinates</b> :
      <div style="margin:5px 0px 12px">
        <div>
          <div style="float:left">
            <pre class="code sh_sh" style="margin:4px 0px">"0" or "1"</pre>
          </div>
          <div style="float:left;padding:8px 10px;margin:4px 0px">
            (if left blank, assumed to be <b>0</b>)
          </div>
          <div class="clear"></div>
        </div>
        <div>If set to "1", this forces VEP to output the
          coordinates of an overlapping custom feature instead of any found
          identifier (or value in the case of bigWig) field. If set to "0" (the
          default), VEP will output the identifier field if one is found; if
          none is found, then the coordinates are used instead.
        </div>
      </div>
    </li>
      
    <li><b>VCF fields</b> :
      <div style="margin:5px 0px 12px">You can specify any info type (e.g. "AC") present in the INFO field of the custom input VCF or specify "FILTER" for the FILTER field, to add these as custom annotations:
        <ul>
          <li>If using "<span style="color:blue">exact</span>" annotation type, allele-specific annotation will be retrieved.</li>
          <li>The INFO field name will be prefixed with the short name, e.g.
              using short name "test", the INFO field "foo" will appear as "test_FOO" in the VEP output. Similarly FILTER field will appear as "test_FILTER".</li>
          <li>In VCF files the custom annotations are added to the CSQ INFO field.</li>
          <li>Alleles in the input and VCF entry are trimmed in both directions in an attempt to match complex or poorly formatted entries.</li>
        </ul>
      </div>
    </li>
  </ul>

  <p>For example: </p>
  
  <pre class="code sh_sh">
    # BigWig file
    ./vep [...] --custom frequencies.bw,Frequency,bigwig,exact,0
    # BED file
    ./vep [...] --custom http://www.myserver.com/data/myPhenotypes.bed.gz,Phenotype,bed,exact,1
    # VCF file
    ./vep [...] --custom [[SPECIESDEFS::ENSEMBL_FTP_URL]]/data_files/homo_sapiens/GRCh37/variation_genotype/TOPMED_GRCh37.vcf.gz,,vcf,exact,0,TOPMED

    # For multiple custom files, use:
    ./vep [...] --custom clinvar.vcf.gz,ClinVar,vcf,exact,0,CLNSIG,CLNREVSTAT,CLNDN \
                --custom TOPMED_GRCh38_20180418.vcf.gz,topmed_20180418,vcf,exact,0,TOPMED \
                --custom UK10K_COHORT.20160215.sites.GRCh38.vcf.gz,uk10k,vcf,exact,0,AF_ALSPAC</pre>

  </div>

  <br/>
  <div class="navbar" onclick="show_hide('custom_options');" id="custom_options" style="padding-left:5px;margin-bottom:0px;cursor:pointer"><img id="img_custom_options" src="/i/16/minus-button.png" style="vertical-align:middle"/> <span style="position:absolute;left:28px;top:9px">Using key-value pairs in <kbd>--custom</kbd> with VEP [[SPECIESDEFS::ENSEMBL_VERSION]]</span></div>
  <div id="div_custom_options" style="display:inline">  
  <br />

  <p> Since VEP 110, you can configure each custom file using a comma-separated
  list of key-value pairs: </p>
  
  <pre class="code sh_sh">
./vep [...] --custom <b>file=Filename</b>,<b>short_name=Short_name</b>,<b>format=File_type</b>,<b>type=Annotation_type</b>,<b>fields=VCF_fields</b></pre>
  
  <p> The order of the options is irrelevant and
  most options have sensible defaults as described below: </p>
  
  <table class="ss"><tbody>
    <tr>
      <th>Option</th>
      <th>Accepted values</th>
      <th>Description</th>
    </tr>
    <tr id="opt_file">
      <td><pre>file</pre></td>
      <td>String with valid path to file</td>
      <td>
        <b>(Required) Filename:</b>
        The path to the file. For Tabix indexed files, VEP will check if both
        the file and the corresponding index (<kbd>.tbi</kbd>) exist.
        For remote files, VEP will check that the tabix index is accessible on
        startup.
      </td>
    </tr>
    <tr id="opt_format">
      <td><pre>format</pre></td>
      <td><kbd>bed</kbd>, <kbd>gff</kbd>, <kbd>gtf</kbd>, <kbd>vcf</kbd> or <kbd>bigwig</kbd></td>
      <td>
        <b>(Required) File format</b> of <a href="#opt_file"><kbd>file</kbd></a>.
      </td>
    </tr>
    <tr id="opt_short_name">
      <td><pre>short_name</pre></td>
      <td>Annotation filename (default) or any string without commas</td>
      <td>
        <b>Short name:</b>
        A name for the annotation that will appear as
        the key in the key=value pairs in the results.
        
        If not defined, this will default to the annotation filename.
      </td>
    </tr>
    <tr id="opt_fields">
      <td><pre>fields</pre></td>
      <td></td>
      <td>
        <b>VCF fields:</b>
        Percentage (%) separated list of INFO fields to print (such as
        <kbd>AC</kbd>) present in the custom input VCF or specify
        <kbd>FILTER</kbd> for the FILTER field, to add these as custom
        annotations:
        <ul>
          <li>
            If using <kbd>exact</kbd> annotation type, allele-specific
            annotation will be retrieved.
          </li>
          <li>
            The INFO field name will be prefixed with the short name, e.g.
            using short name <kbd>test</kbd>, the INFO field <kbd>foo</kbd> will
            appear as <kbd>test_FOO</kbd> in the VEP output. Similarly FILTER
            field will appear as <kbd>test_FILTER</kbd>.
          </li>
          <li>
            In VCF files the custom annotations are added to the CSQ INFO field.
          </li>
          <li>
            Alleles in the input and VCF entry are trimmed in both directions in
            an attempt to match complex or poorly formatted entries.
          </li>
        </ul>
      </td>
    </tr>
    <tr id="opt_type">
      <td><pre>type</pre></td>
      <td>
        <kbd>overlap</kbd> (default),
        <kbd>within</kbd>,
        <kbd>surrounding</kbd> or
        <kbd>exact</kbd>
      </td>
      <td>
        <b>Annotation type:</b>
        <ul>
          <li>
            <kbd>overlap</kbd>: reports any annotation that overlaps the variant
            by even 1 base pair.
          </li>
          <li>
            <kbd>within</kbd> (*): only reports annotations within the variant.
          </li>
          <li>
            <kbd>surrounding</kbd> (*): only reports annotations that completely
            surround the variant.
          </li>
          <li>
            <kbd>exact</kbd>: only reports annotations whose coordinates match
            exactly those of the variant. This is suitable for position-specific
            information such as conservation scores, allele frequencies or
            phenotype information.
          </li>
        </ul>
      </td>
    </tr>
    <tr id="opt_overlap_cutoff">
      <td><pre>overlap_cutoff</pre></td>
      <td>From <kbd>0</kbd> (default) to <kbd>100</kbd></td>
      <td>
        <b>Minimum percentage overlap (*)</b> between annotation and variant.
        See also <a href="#opt_reciprocal"><kbd>reciprocal</kbd></a>.
      </td>
    </tr>
    <tr id="opt_reciprocal">
      <td><pre>reciprocal</pre></td>
      <td><kbd>0</kbd> (default) or <kbd>1</kbd></td>
      <td>
        <b>Mode of calculating the overlap percentage (*):</b>
        <ul>
          <li><kbd>0</kbd>: percentage of annotation covered by variant</li>
          <li><kbd>1</kbd>: percentage of variant covered by annotation</li>
        </ul>
      </td>
    </tr>
    <tr id="opt_distance">
      <td><pre>distance</pre></td>
      <td><kbd>0</kbd> or a positive integer (disabled by default)</td>
      <td>Distance (in base pairs) to the ends of the overlapping feature (*).</td>
    </tr>
    <tr id="opt_coords">
      <td><pre>coords</pre></td>
      <td><kbd>0</kbd> (default) or <kbd>1</kbd></td>
      <td>
        <b>Force report coordinates:</b>
        <ul>
          <li>
            <kbd>0</kbd>: outputs the identifier field (or value in the case of
            <kbd>bigWig</kbd>) if available; otherwise, outputs coordinates
            instead.
          </li>
          <li>
            <kbd>1</kbd>: always outputs the coordinates of an overlapping
            custom feature.
          </li>
        </ul>
      </td>
    </tr>
    <tr id="opt_same_type">
      <td><pre>same_type</pre></td>
      <td><kbd>0</kbd> (default) or <kbd>1</kbd></td>
      <td>
        Only match identical variant classes (*). For instance, only match
        deletions with deletions. This is only available for VCF annotations.
      </td>
    </tr>
    <tr id="opt_num_records">
      <td><pre>num_records</pre></td>
      <td><kbd>50</kbd> (default), <kbd>all</kbd>, <kbd>0</kbd> or any positive integer</td>
      <td>
        <b>Number of matching records to display.</b>
        Any remaining records are represented with ellipsis (<kbd>...</kbd>).
        Use <kbd>num_records = all</kbd> to display all matching records and
        <kbd>num_records = 0</kbd> to only display <kbd>...</kbd> if
        there are matching records.
      </td>
    </tr>
    <tr id="opt_summary_stats">
      <td><pre>summary_stats</pre></td>
      <td>
        <kbd>none</kbd> (default), <kbd>min</kbd>, <kbd>mean</kbd>,
        <kbd>max</kbd>, <kbd>count</kbd> or <kbd>sum</kbd>
      </td>
      <td>
        <b>Summary statistics to display</b>. A percentage-separated list may be used to
        calculate multiple summary statistics, such as
        <kbd>min%mean%max%count%sum</kbd>.
      </td>
    </tr>
  </tbody></table>
  
  <p>
    When <kbd>format = vcf</kbd>, the features marked with (*) only work on
    structural variants.
  </p>

  <p>Examples:</p>
  
  <pre class="code sh_sh">
# BigWig file
./vep [...] --custom file=frequencies.bw,short_name=Frequency,format=bigwig,type=exact,coords=0
# BED file
./vep [...] --custom file=http://www.myserver.com/data/myPhenotypes.bed.gz,short_name=Phenotype,format=bed,type=exact,coords=1
# VCF file
./vep [...] --custom file=[[SPECIESDEFS::ENSEMBL_FTP_URL]]/data_files/homo_sapiens/GRCh37/variation_genotype/TOPMED_GRCh37.vcf.gz,format=vcf,type=exact,coords=0,fields=TOPMED
./vep [...] --custom file=gnomad_v2.1_sv.sites.vcf.gz,short_name=gnomad,fields=PC%EVIDENCE%SVTYPE,format=vcf,type=within,reciprocal=1,overlap_cutoff=80

# For multiple custom files, use:
./vep [...] --custom file=clinvar.vcf.gz,short_name=ClinVar,format=vcf,type=exact,coords=0,fields=CLNSIG%CLNREVSTAT%CLNDN \
            --custom file=TOPMED_GRCh38_20180418.vcf.gz,short_name=topmed_20180418,format=vcf,type=exact,coords=0,fields=TOPMED \
            --custom file=UK10K_COHORT.20160215.sites.GRCh38.vcf.gz,short_name=uk10k,format=vcf,type=exact,coords=0,fields=AF_ALSPAC</pre>
  </div>
  <hr/>
  <h2 id="custom_example">Example - ClinVar</h2>

  <p>We include the most recent public variant and phenotype data available in each Ensembl release, but some projects release data more frequently than we do.<br />
  If you want to have the very latest annotations, you can use the data files from your prefered projects (in any format listed in <a href="#custom_formats">Data formats</a>) and use them as a VEP custom annotation.</p>
  <p>For instance, you can annotate you variants with VEP, using the the latest ClinVar data as custom annotation.<br />
  ClinVar provides VCF files on their FTP site: <a href="https://ftp.ncbi.nlm.nih.gov/pub/clinvar/vcf_GRCh37/" rel="external">GRCh37</a> and <a href="https://ftp.ncbi.nlm.nih.gov/pub/clinvar/vcf_GRCh38/" rel="external">GRCh38</a>.</p>
  <p>See below an example about how to use ClinVar VCF files as a VEP custom annotation:</p>

  <ol>
    <li>Download the VCF files (you need the compressed VCF file and the index file), e.g.:
      <pre class="code sh_sh top-margin"># Compressed VCF file
curl -O https://ftp.ncbi.nlm.nih.gov/pub/clinvar/vcf_GRCh38/clinvar.vcf.gz
# Index file
curl -O https://ftp.ncbi.nlm.nih.gov/pub/clinvar/vcf_GRCh38/clinvar.vcf.gz.tbi</pre>
    </li>
    <li>Example of command you can use:
      <pre class="code sh_sh top-margin" style="margin-bottom:2px">./vep [...] --custom file=clinvar.vcf.gz,short_name=ClinVar,format=vcf,type=exact,coords=0,fields=CLNSIG%CLNREVSTAT%CLNDN

## Where the selected ClinVar INFO fields (from the ClinVar VCF file) are:
# - CLNSIG:     Clinical significance for this single variant
# - CLNREVSTAT: ClinVar review status for the Variation ID
# - CLNDN:      ClinVar's preferred disease name for the concept specified by disease identifiers in CLNDISDB
# Of course you can select the INFO fields you want in the ClinVar VCF file

# Quick example on GRCh38:
./vep --id "1  230710048 230710048 A/G 1" --species homo_sapiens -o /path/to/output/output.txt --cache --offline --assembly GRCh38 --custom file=/path/to/custom_files/clinvar.vcf.gz,short_name=ClinVar,format=vcf,type=exact,coords=0,fields=CLNSIG%CLNREVSTAT%CLNDN</pre>

      <div class="navbar" onclick="show_hide('custom_results');" id="custom_results" style="padding-left:5px;margin-bottom:0px;cursor:pointer"><img id="img_custom_results" src="/i/16/plus-button.png" style="vertical-align:middle"/> <span style="position:absolute;left:28px;top:9px">Results in the default VEP format</span></div>
      <div id="div_custom_results" style="display:none">
        <pre class="code sh_sh">## Column descriptions:
## Uploaded_variation : Identifier of uploaded variant
## Location : Location of variant in standard coordinate format (chr:start or chr:start-end)
## Allele : The variant allele used to calculate the consequence
## Gene : Stable ID of affected gene
## Feature : Stable ID of feature
## Feature_type : Type of feature - Transcript, RegulatoryFeature or MotifFeature
## Consequence : Consequence type
## cDNA_position : Relative position of base pair in cDNA sequence
## CDS_position : Relative position of base pair in coding sequence
## Protein_position : Relative position of amino acid in protein
## Amino_acids : Reference and variant amino acids
## Codons : Reference and variant codon sequence
## Existing_variation : Identifier(s) of co-located known variants
## Extra column keys:
## IMPACT : Subjective impact classification of consequence type
## DISTANCE : Shortest distance from variant to transcript
## STRAND : Strand of the feature (1/-1)
## FLAGS : Transcript quality flags
## SOURCE : Source of transcript
## ClinVar : /opt/vep/.vep/custom/clinvar.vcf.gz (exact)
## ClinVar_CLNSIG : CLNSIG field from /path/to/custom_files/clinvar.vcf.gz
## ClinVar_CLNREVSTAT : CLNREVSTAT field from /path/to/custom_files/clinvar.vcf.gz
## ClinVar_CLNDN : CLNDN field from /path/to/custom_files/clinvar.vcf.gz
#Uploaded_variation  Location     Allele  Gene             Feature          Feature_type  Consequence              ...  Extra
1_230710048_A/G      1:230710048  G       ENSG00000135744  ENST00000366667  Transcript    missense_variant         ...  IMPACT=MODERATE;STRAND=-1;ClinVar=18068;ClinVar_CLNDN=Hypertension,_essential,_susceptibility_to|Preeclampsia,_susceptibility_to|Renal_dysplasia|Susceptibility_to_progression_to_renal_failure_in_IgA_nephropathy|not_specified;ClinVar_CLNREVSTAT=criteria_provided,_multiple_submitters,_no_conflicts;ClinVar_CLNSIG=Benign;ClinVar_FILTER=.
1_230710048_A/G      1:230710048  G       ENSG00000244137  ENST00000412344  Transcript    downstream_gene_variant  ...  IMPACT=MODIFIER;DISTANCE=650;STRAND=-1;ClinVar=18068;ClinVar_CLNDN=Hypertension,_essential,_susceptibility_to|Preeclampsia,_susceptibility_to|Renal_dysplasia|Susceptibility_to_progression_to_renal_failure_in_IgA_nephropathy|not_specified;ClinVar_CLNREVSTAT=criteria_provided,_multiple_submitters,_no_conflicts;ClinVar_CLNSIG=Benign;ClinVar_FILTER=.</pre>
      </div>

      <div class="navbar" onclick="show_hide('custom_results_vcf');" id="custom_results_vcf" style="padding-left:5px;margin:2px 0px 0px;cursor:pointer"><img id="img_custom_results_vcf" src="/i/16/plus-button.png" style="vertical-align:middle"/> <span style="position:absolute;left:28px;top:9px">Results in VCF (adding the tag <b>--vcf</b> in the command line)</span></div>
      <div id="div_custom_results_vcf" style="display:none">
        <pre class="code sh_sh">##fileformat=VCFv4.1
##INFO=&lt;ID=CSQ,Number=.,Type=String,Description="Consequence annotations from Ensembl VEP. Format: Allele|Consequence|IMPACT|SYMBOL|Gene|Feature_type|Feature|BIOTYPE|EXON|INTRON|HGVSc|HGVSp|cDNA_position|CDS_position|Protein_position|Amino_acids|Codons|Existing_variation|DISTANCE|STRAND|FLAGS|SYMBOL_SOURCE|HGNC_ID|SOURCE|ClinVar|ClinVar_CLNSIG|ClinVar_CLNREVSTAT|ClinVar_CLNDN"&gt;
##INFO=&lt;ID=ClinVar,Number=.,Type=String,Description="/path/to/custom_files/clinvar.vcf.gz (exact)"&gt;
##INFO=&lt;ID=ClinVar_CLNSIG,Number=.,Type=String,Description="CLNSIG field from /path/to/custom_files/clinvar.vcf.gz"&gt;
##INFO=&lt;ID=ClinVar_CLNREVSTAT,Number=.,Type=String,Description="CLNREVSTAT field from /path/to/custom_files/clinvar.vcf.gz"&gt;
##INFO=&lt;ID=ClinVar_CLNDN,Number=.,Type=String,Description="CLNDN field from /path/to/custom_files/clinvar.vcf.gz"&gt;
#CHROM  POS        ID               REF  ALT  QUAL  FILTER  INFO
1       230710048  1_230710048_A/G  A    G    .     .       CSQ=G|missense_variant|MODERATE|AGT|ENSG00000135744|Transcript|ENST00000366667|protein_coding|2/5||||1018|803|268|M/T|aTg/aCg|||-1||HGNC|HGNC:333||18068|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Hypertension&_essential&_susceptibility_to&Preeclampsia&_susceptibility_to&Renal_dysplasia&Susceptibility_to_progression_to_renal_failure_in_IgA_nephropathy&not_specified,G|downstream_gene_variant|MODIFIER|AL512328.1|ENSG00000244137|Transcript|ENST00000412344|antisense|||||||||||650|-1||Clone_based_ensembl_gene|||18068|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Hypertension&_essential&_susceptibility_to&Preeclampsia&_susceptibility_to&Renal_dysplasia&Susceptibility_to_progression_to_renal_failure_in_IgA_nephropathy&amp;not_specified</pre>
    </li>
  </ol>


  <br />
  <hr/>
  <h2 id="custom_remote">Using remote files</h2>
  
  <p>The tabix utility makes it possible to read annotation files from remote
  locations, for example over HTTP or FTP protocols.</p>

  <p>In order to do this, the
  .tbi index file is downloaded locally (to the current working directory)
  when VEP is run. From this point on, only the portions of data requested
  by VEP (i.e. those overlapping the variants in your input file) are
  downloaded.</p>
 
  <div style="width:auto" class="warning">

    <h3>Be aware</h3>

    <div class="message-pad">
      <p>1. It is still possible to cause problems with network traffic in this manner by requesting data for a
         large number of variants.</p>
      
      <p>2. Users with large amounts of data should download the annotation file locally rather than risk causing any issues!</p>
    </div>
  </div>  
 
  <p> bigWig files can also be used remotely in the same way as tabix-indexed
  files, although less stringent checks are carried out on VEP startup. </p>
  
</div>

</body>
</html>
      
