<html>
<head>
<title>Running VEP</title>
<meta name="order" content="3" />
</head>

<body>
    
<style>
tr:nth-child(odd) {background-color: #f0f0f0;}
td {line-height: 1.25em;}
</style>

<div>
  
  <div style="float:right"><img src="/img/vep_logo.png"/></div>
  
  <h1 id="top"><span style="color:#006;padding-right:15px">Variant Effect Predictor</span><span style="color:#666"><img src="/i/16/setting.png"/> Running VEP</span></h1>
  <hr/>
  
  <p> VEP is run on the command line as follows (assuming you are in the ensembl-vep directory): </p>
  
  <div>
    <div style="float:left">
      <pre class="code sh_sh"> ./vep [options] </pre>
    </div>
    <div style="clear:both"></div>
  </div>
  
  <p> where [options] represent a set of flags and options.
  A basic set of flags can be listed using <a href="#opt_help">--help</a>: </p>

  <div>
    <div style="float:left">
      <pre class="code sh_sh"> ./vep --help </pre>
    </div>
    <div style="clear:both"></div>
  </div>
  
  <p> For optimum performance, download a cache file for your species of interest, using
  either the <a href="vep_download.html#installer"><img style="vertical-align:bottom" src="/i/16/download.png"/> installer </a> or by
  following the <a href="vep_cache.html#pre">VEP Cache documentation</a>, and run VEP with either
  the <a href="#opt_cache">--cache</a> or <a href="#opt_offline">--offline</a>
  option. </p>
  
  <p> It is possible to run VEP connecting to the public Ensembl database servers in place of a cache. 
      This can be adequate when annotating small files, but the database servers can become busy and slow.
      To enable this option, use
  <a href="#opt_database">--database</a> </p>
  
  <p> To run VEP with default options, use the following command: </p>
  
  <div>
      <div style="float:left">
      <pre class="code sh_sh"> ./vep --cache -i input.txt -o output.txt </pre>
    </div>
    <div style="clear:both"></div>
  </div>

  <p> where input.txt contains data in one of the compatible 
  <a href="../vep_formats.html#input"> input formats </a>
  and output.txt is the <a href="../vep_formats.html#output">output file</a> to be created.
  </p>
  
  <p> Options can be passed as the full string (e.g. <a
  href="vep_options.html#opt_format">--format</a>), or as the shortest unique
  string among the options (e.g. <a
  href="vep_options.html#opt_form">--form</a> for <a
  href="vep_options.html#opt_format">--format</a>, since there is another
  option <a
  href="vep_options.html#opt_force_overwrite">--force_overwrite</a>). </p>

  <p> You may use one or two hypen ("-") characters before each option name; <b>--cache</b> or <b>-cache</b>.</p>

  <p> Options can also be read from a configuration file - either passively
  stored as $HOME/.vep/vep.ini, or actively using <a
  href="#opt_config">--config</a>. </p>

  <div id="nv_species"> 
    <div class="info" style="float:left">
      <h3>Running VEP on non-vertebrate species</h3>
      <div class="message-pad">
        <p>To use VEP on non-vertebrate species, you need to use extra options:</p>
        <p>- <b>Database</b> access: add the option <a href="#opt_genomes">--genomes</a> to point to the correct database server.</p>
        <div class="left-margin">
          <div>
            <div style="float:left">
              <pre class="code sh_sh"> ./vep -i input.txt -o output.txt --species triticum_aestivum --database --genomes </pre>
            </div>
            <div style="clear:both"></div>
          </div>
          <p>Some of the non-vetebrate species databases (mainly bacteria and protists) are composed of a collection of species.
            For these databases the flag "<a href="#opt_is_multispecies">--is_multispecies</a> 1" is required.</p>

          <div>
            <div style="float:left">
              <pre class="code sh_sh"> ./vep -i input.txt -o output.txt --species protists_euglenozoa1 --database --genomes --is_multispecies 1 </pre>
            </div>
            <div style="clear:both"></div>
          </div>
        </div>

        <p>- <b>Cache</b> access: use the option "<a href="#opt_genomes">--cache_version</a> <i>eg_version</i>" where <i>eg_version</i> is the 
        <a href="http://ensemblgenomes.org/">Ensembl genomes</a> version number, which differs from the Ensembl vertebrates/VEP version numbers
        ([[SPECIESDEFS::ENSEMBL_VERSION]]).</p>
        <div class="left-margin">
          <div>
            <div style="float:left">
              <pre class="code sh_sh"> ./vep -i input.txt -o output.txt --species triticum_aestivum --cache --cache_version 42</pre>
            </div>
            <div style="clear:both"></div>
          </div>
          <p>More information about VEP cache are available <a href="vep_cache.html#cache">here</a>.</p>
        </div>
      </div>
    </div>
    <div style="clear:both"></div>
  </div>
  

  <br/> 
  <hr/>
  <h2 id="basic">Basic options</h2>
  
  <table class="ss">
    <tr>
      <th width="20%">Flag</th>
      <th width="8%">Alternate</th>
      <th>Description</th>
      <th width="10%">Incompatible with</th>
    </tr>
    <tr id="opt_help">
      <td><pre>--help</pre></td>
      <td>&nbsp;</td>
      <td>Display help message and quit</td>
      <td>&nbsp;</td>
    </tr>
    <tr id="opt_quiet">
      <td><pre>--quiet</pre></td>
      <td><pre>-q</pre></td>
      <td>Suppress warning messages.<i>Not used by
      default</i></td>
      <td><a href="#opt_verbose">--verbose</a></td> 
    </tr>
    <tr id="opt_verbose">
      <td><pre>--verbose</pre></td>
      <td><pre>-v</pre></td>
      <td>Print out a bit more information while running. <i>Not used by
      default</i></td>
      <td><a href="#opt_quiet">--quiet</a></td>
    </tr>    
    <tr id="opt_config">
      <td><pre>--config [filename]</pre></td>
      <td>&nbsp;</td>
      <td>
        <p>Load configuration options from a config file. The config file
        should consist of whitespace-separated pairs of option names and
        settings e.g.:</p>
  <pre class="code sh_sh">output_file   my_output.txt
species       mus_musculus
format        vcf
host          useastdb.ensembl.org</pre>
        A config file can also be implicitly read; save the file as
        <b>$HOME/.vep/vep.ini</b> (or equivalent directory if using
        <a href="vep_options.html#opt_dir">--dir</a>). Any options in this file will be overridden by
        those specified in a config file using <a href="vep_options.html#opt_config">--config</a>, and in
        turn by any options specified on the command line. You
        can create a quick version file of this by setting the flags as
        normal and running VEP in verbose (<b>-v</b>) mode. This
        will output lines that can be copied to a config file that can
        be loaded in on the next run using <a href="vep_options.html#opt_config">--config</a>. <i>Not used
        by default</i>
      </td>
      <td>&nbsp;</td>
    </tr>
    <tr id="opt_everything">
      <td><pre>--everything</pre></td>
      <td><pre>-e</pre></td>
      <td>Shortcut flag to switch on all of the following:
        <p>
          <a href="#opt_sift">--sift b</a>,
          <a href="#opt_polyphen">--polyphen b</a>,
          <a href="#opt_ccds">--ccds</a>,
          <a href="#opt_uniprot">--uniprot</a>,
          <a href="#opt_hgvs">--hgvs</a>,
          <a href="#opt_symbol">--symbol</a>,
          <a href="#opt_numbers">--numbers</a>,
          <a href="#opt_domains">--domains</a>,
          <a href="#opt_regulatory">--regulatory</a>,
          <a href="#opt_canonical">--canonical</a>,
          <a href="#opt_protein">--protein</a>,
          <a href="#opt_biotype">--biotype</a>,
          <a href="#opt_uniprot">--uniprot</a>,
          <a href="#opt_tsl">--tsl</a>,
          <a href="#opt_appris">--appris</a>,
          <a href="#opt_gene_phenotype">--gene_phenotype</a>
          <a href="#opt_af">--af</a>,
          <a href="#opt_af_1kg">--af_1kg</a>,
          <a href="#opt_af_esp">--af_esp</a>,
          <a href="#opt_af_gnomad">--af_gnomad</a>,
          <a href="#opt_max_af">--max_af</a>,
          <a href="#opt_pubmed">--pubmed</a>,
          <a href="#opt_var_synonyms">--var_synonyms</a>,
          <a href="#opt_variant_class">--variant_class</a>,
          <a href="#opt_mane">--mane</a>
        </ul>
      </td>
      <td>&nbsp;</td>
    </tr>
    <tr id="opt_species">
      <td><pre>--species [species]</pre></td>
      <td>&nbsp;</td>
      <td>
        Species for your data. This can be the latin name e.g.
        "homo_sapiens" or any Ensembl alias e.g. "mouse". Specifying the
        latin name can speed up initial database connection as the
        registry does not have to load all available database aliases on
        the server. <i>Default = "homo_sapiens"</i>
      </td>
      <td>&nbsp;</td>
    </tr>
    <tr id="opt_assembly">
      <td><pre>--assembly [name]</pre></td>
      <td><pre>-a</pre></td>
      <td>
        Select the assembly version to use if more than one available. If using
        the cache, you must have the appropriate assembly's cache file
        installed. If not specified and you have only 1 assembly version
        installed, this will be chosen by default. <i>Default = use found
        assembly version</i>
      </td>
      <td>&nbsp;</td>
    </tr>
    <tr id="opt_input_file">
      <td><pre>--input_file [filename]</pre></td>
      <td><pre>-i</pre></td>
      <td>Input file name. If not specified, VEP will attempt to read from STDIN. Can use compressed file (gzipped).</td>
      <td>&nbsp;</td>
    </tr>
    <tr id="opt_input_data">
      <td><pre>--input_data [string]</pre></td>
      <td><pre>--id</pre></td>
      <td><p>Raw input data as a string. May be used, for example, to input a single rsID
        or HGVS notation quickly to vep:</p>
        <pre class="code sh_sh">--input_data rs699</pre>
      </td>
      <td>&nbsp;</td>
    </tr>
    <tr id="opt_format">
      <td><pre>--format [format]</pre></td>
      <td>&nbsp;</td>
      <td>
        <p>
        <a href="../vep_formats.html#input">Input file format</a> - one of
        "ensembl", "vcf", "hgvs", "id", "region", "spdi".<br />
        By default, VEP auto-detects the input file format. Using this option you
        can specify the input file is Ensembl, VCF, IDs, HGVS, SPDI or region format.
        Can use compressed version (gzipped) of any file format listed above.
        <i>Auto-detects format by default</i>
        </p>
      </td>
      <td>&nbsp;</td>
    </tr>
    <tr id="opt_output_file">
      <td><pre>--output_file [filename]</pre></td>
      <td><pre>-o</pre></td>
      <td>
        Output file name. Results can write to STDOUT by specifying
        'STDOUT' as the output file name - this will force quiet mode.
        <i>Default = "variant_effect_output.txt"</i>
      </td>
      <td>&nbsp;</td>
    </tr>
    <tr id="opt_force_overwrite">
      <td><pre>--force_overwrite</pre></td>
      <td><pre>--force</pre></td>
      <td>
        By default, VEP will fail with an error if the output
        file already exists. You can force the overwrite of the existing
        file by using this flag. <i>Not used by default</i>
      </td>
      <td>&nbsp;</td>
    </tr>
    <tr id="opt_stats_file">
      <td><pre>--stats_file [filename]</pre></td>
      <td><pre>--sf</pre></td>
      <td>
        Summary stats file name. This is an <a
        href="../vep_formats.html#stats">HTML file</a> containing a summary
        of the VEP run - the file name must end ".htm" or ".html".
        <i>Default = "variant_effect_output.txt_summary.html"</i>
      </td>
      <td>&nbsp;</td>
    </tr>
    <tr id="opt_no_stats">
      <td><pre>--no_stats</pre></td>
      <td>&nbsp;</td>
      <td>
        Don't generate a stats file. Provides marginal gains in run time.
      </td>
      <td>&nbsp;</td>
    </tr>
    <tr id="opt_stats_text">
      <td><pre>--stats_text</pre></td>
      <td>&nbsp;</td>
      <td>
        Generate a plain text stats file in place of the HTML.
      </td>
      <td>&nbsp;</td>
    </tr>
    <tr id="opt_warning_file">
      <td><pre>--warning_file [filename]</pre></td>
      <td>&nbsp;</td>
      <td>
        File name to write warnings and errors to.
        <i>Default = STDERR (standard error)</i>
      </td>
      <td>&nbsp;</td>
    </tr>
    <tr id="opt_max_sv_size">
      <td><pre>--max_sv_size</pre></td>
      <td>&nbsp;</td>
      <td>
        Extend the maximum Structural Variant size VEP can process.
      </td>
      <td>&nbsp;</td>
    </tr>
    <tr id="opt_no_check_variants_order">
      <td><pre>--no_check_variants_order</pre></td>
      <td>&nbsp;</td>
      <td>
        Permit the use of unsorted input files. However running VEP on unsorted input files slows down the tool and requires more memory.
      </td>
      <td>&nbsp;</td>
    </tr>
    <tr id="opt_fork">
      <td><pre>--fork [num_forks]</pre></td>
      <td>&nbsp;</td>
      <td>
        Enable <a href="vep_other.html#faster">forking</a>, using the specified
        number of forks. Forking can dramatically improve runtime. <i>Not used by default</i>
      </td>
      <td>&nbsp;</td>
    </tr>
  </table>
  
  
  

  <br/>  
  <hr/>
    
  <h2 id="cacheopt">Cache options</h2>
  
  <table class="ss">
    <tr>
      <th width="20%">Flag</th>
      <th width="8%">Alternate</th>
      <th>Description</th>
      <th width="12%">Output fields</th>
      <th width="10%">Incompatible with</th>
    </tr>
    <tr id="opt_cache">
      <td><pre>--cache</pre></td>
      <td>&nbsp;</td>
      <td>
        Enables use of the <a href="vep_cache.html#cache">cache</a>. Add <a
        href="#opt_refseq">--refseq</a> or <a href="#opt_merged">--merged</a> to use the refseq or merged cache, (if installed).
      </td>
      <td>&nbsp;</td>
      <td>
        <div style="padding:2px 0px"><a href="#opt_database">--database</a></div>
      </td>
    </tr>
    <tr id="opt_dir">
      <td><pre>--dir [directory]</pre></td>
      <td>&nbsp;</td>
      <td>
        Specify the base cache/plugin directory to use. <i>Default =
        "$HOME/.vep/"</i>
      </td>
      <td colspan="2">&nbsp;</td>
    </tr>
    <tr id="opt_dir_cache">
      <td><pre>--dir_cache [directory]</pre></td>
      <td>&nbsp;</td>
      <td>
        Specify the cache directory to use. <i>Default =
        "$HOME/.vep/"</i>
      </td>
      <td colspan="2">&nbsp;</td>
    </tr>
    <tr id="opt_dir_plugins">
      <td><pre>--dir_plugins [directory]</pre></td>
      <td>&nbsp;</td>
      <td>
        Specify the plugin directory to use. <i>Default =
        "$HOME/.vep/"</i>
      </td>
      <td colspan="2">&nbsp;</td>
    </tr>
    <tr id="opt_offline">
      <td><pre>--offline</pre></td>
      <td>&nbsp;</td>
      <td>
        Enable <a href="vep_cache.html#offline">offline mode</a>. No database
        connections will be made, and a cache file or <a href="#opt_gff">GFF</a>/<a href="#opt_gtf">GTF</a> file is required for annotation. 
        Add <a href="#opt_refseq">--refseq</a> to use the refseq cache (if
        installed). <i>Not used by default</i>
      </td>
      <td>&nbsp;</td>
      <td>
        <div style="padding:2px 0px"><a href="#opt_database">--database</a></div>
        <div style="padding:2px 0px"><a href="#opt_check_svs">--check_svs</a></div>
      </td>
    </tr>
    <tr id="opt_fasta">
      <td><pre>--fasta [file|dir]</pre></td>
      <td><pre>--fa</pre></td>
      <td>
        Specify a FASTA file or a directory containing FASTA files to
        use to look up reference sequence. The first time you run VEP
        with this parameter an index will be built which can take
        a few minutes. This is required if fetching HGVS annotations (<a
        href="#opt_hgvs">--hgvs</a>) or checking reference sequences (<a
        href="#opt_check_ref">--check_ref</a>) in offline mode (<a
        href="#opt_offline">--offline</a>), and optional with some
        performance increase in cache mode (<a
        href="#opt_cache">--cache</a>). See <a
        href="vep_cache.html#fasta">documentation</a> for more details.
        <i>Not used by default</i>
      </td>
      <td colspan="2">&nbsp;</td>
    </tr>
    <tr id="opt_refseq">
      <td><pre>--refseq</pre></td>
      <td>&nbsp;</td>
      <td>
        <p> Specify this option if you have installed the RefSeq
        cache in order for VEP to pick up the alternate cache directory. This cache contains
        transcript objects corresponding to RefSeq transcripts.
        Consequence output will be given relative to these transcripts in place
        of the default Ensembl transcripts (see <a
        href="vep_other.html#refseq">documentation</a>)</p>
        </p>
      </td>
      <td>REFSEQ_MATCH, BAM_EDIT</td>
      <td>
        <div style="padding:2px 0px"><a href="#opt_gencode_basic">--gencode_basic</a></div>
        <div style="padding:2px 0px"><a href="#opt_merged">--merged</a></div>
      </td>
    </tr>
    <tr id="opt_merged">
      <td><pre>--merged</pre></td>
      <td>&nbsp;</td>
      <td>
        <p> Use the merged Ensembl and RefSeq cache. Consequences are flagged
        with the SOURCE of each transcript used. <p>
      </td>
      <td>REFSEQ_MATCH, BAM_EDIT, SOURCE</td>
      <td>
        <div style="padding:2px 0px"><a href="#opt_refseq">--refseq</a></div>
      </td>
    </tr>
    <tr id="opt_cache_version">
      <td><pre>--cache_version</pre></td>
      <td>&nbsp;</td>
      <td>
        Use a different cache version than the assumed default (the VEP
        version). This should be used with Ensembl Genomes caches since their
        version numbers do not match Ensembl versions. For example, the
        VEP/Ensembl version may be 88 and the Ensembl Genomes version 35. <i>Not
        used by default</i>
      </td>
      <td colspan="2">&nbsp;</td>
    </tr>
    <tr id="opt_show_cache_info">
      <td><pre>--show_cache_info</pre></td>
      <td>&nbsp;</td>
      <td>
        Show source version information for selected cache and quit
      </td>
      <td colspan="2">&nbsp;</td>
    </tr>
    <tr id="opt_buffer_size">
      <td><pre>--buffer_size [number]</pre></td>
      <td>&nbsp;</td>
      <td>
        Sets the internal buffer size, corresponding to the number of
        variants that are read in to memory simultaneously. Set this
        lower to use less memory at the expense of longer run time, and
        higher to use more memory with a faster run time. <i>Default =
        5000</i>
      </td>
      <td colspan="2">&nbsp;</td>
    </tr>
  </table>


  <br/>
  <hr/>  
  <h2 id="other">Other annotation sources</h2>
  
  <table class="ss">
    <tr>
      <th width="20%">Flag</th>
      <th width="10%">Alternate</th>
      <th>Description</th>
      <th width="15%">Output fields</th>
    </tr>
    <tr id="opt_plugin">
      <td><pre>--plugin [plugin name]</pre></td>
      <td>&nbsp;</td>
      <td>
        Use named plugin. Plugin modules should be installed in the
        Plugins subdirectory of the VEP cache directory (defaults to
        $HOME/.vep/). Multiple plugins can be used by supplying the
        <a href="vep_options.html#opt_plugin">--plugin</a> flag multiple times. See <a
        href="vep_plugins.html">plugin documentation</a>. <i>Not used by
        default</i>
      </td>
      <td>Plugin-dependent</td>
    </tr>
    <tr id="opt_custom">
      <td><pre>--custom [filename]</pre></td>
      <td>&nbsp;</td>
      <td>
        Add custom annotation to the output. Files must be tabix indexed
        or in the bigWig format. Multiple files can be specified by
        supplying the <a href="vep_options.html#opt_custom">--custom</a> flag multiple times. <a
        href="vep_custom.html">See here</a> for full details. <i>Not
        used by default</i>
      </td>
      <td>SOURCE, Custom file dependent</td>
    </tr>
    <tr id="opt_gff">
      <td><pre>--gff [filename]</pre></td>
      <td>&nbsp;</td>
      <td>
        Use <a href="vep_cache.html#gff">GFF transcript annotations</a> in [filename] as an annotation source. Requires a <a href="#opt_fasta">FASTA file</a> of genomic sequence.<i>Not
        used by default</i>
      </td>
      <td>SOURCE</td>
    </tr>
    <tr id="opt_gtf">
      <td><pre>--gtf [filename]</pre></td>
      <td>&nbsp;</td>
      <td>
        Use <a href="vep_cache.html#gff">GTF transcript annotations</a> in [filename] as an annotation source. Requires a <a href="#opt_fasta">FASTA file</a> of genomic sequence.<i>Not
        used by default</i>
      </td>
      <td>SOURCE</td>
    </tr>
    <tr id="opt_bam">
      <td><pre>--bam [filename]</pre></td>
      <td>&nbsp;</td>
      <td>
        <b>ADVANCED</b> Use BAM file of sequence alignments to correct transcript models not derived from reference genome sequence. Used to correct <a href="vep_other.html#refseq_bam">RefSeq transcript models</a>. Enables <a href="#opt_use_transcript_ref">--use_transcript_ref</a>; add <a href="#opt_use_given_ref">--use_given_ref</a> to override this behaviour.
        <i>Not used by default</i>
      </td>
      <td>BAM_EDIT</td>
    </tr>
    <tr id="opt_use_transcript_ref">
      <td><pre>--use_transcript_ref</pre></td>
      <td>&nbsp;</td>
      <td>
        By default VEP uses the reference allele provided in the input file to calculate consequences for the provided alternate allele(s). Use this flag to force VEP to replace the provided reference allele with sequence derived from the overlapped transcript. This is especially relevant when using the RefSeq cache, see <a href="vep_other.html#refseq_bam">documentation</a> for more details. The <a href="../vep_formats.html#output">GIVEN_REF and USED_REF fields</a> are set in the output to indicate any change.
        <i>Not used by default</i>
      </td>
      <td>GIVEN_REF, USED_REF</td>
    </tr>
    <tr id="opt_use_given_ref">
      <td><pre>--use_given_ref</pre></td>
      <td>&nbsp;</td>
      <td>
        Using <a href="#opt_bam">--bam</a> or a <a href="vep_other.html#refseq_bam">BAM-edited RefSeq cache</a> by default enables <a href="#opt_use_transcript_ref">--use_transcript_ref</a>; add this flag to override this behaviour and use the provided reference allele from the input.
        <i>Not used by default</i>
      </td>
      <td>&nbsp;</td>
    </tr>
    <tr id="opt_custom_multi_allelic">
      <td><pre>--custom_multi_allelic</pre></td>
      <td>&nbsp;</td>
      <td>
        By default, comma separated lists found within the INFO field of custom annotation VCFs are assumed to be allele specific. For example, a variant with allele_string A/G/C with associated custom annotation 'single,double,triple' will associate triple with C, double with G and single with A. This flag instructs VEP to return all annotations for all alleles.
        <i>Not used by default</i>
      </td>
      <td>&nbsp;</td>
    </tr>
  </table>
    
 
  <br/>
  <hr/>
  <h2 id="format">Output format options</h2>

  <table class="ss">
    <tr>
      <th width="20%">Flag</th>
      <th width="10%">Alternate</th>
      <th>Description</th>
      <th width="10%">Output fields</th>
      <th width="10%">Incompatible with</th>
    </tr>
    <tr id="opt_vcf">
      <td><pre>--vcf</pre></td>
      <td>&nbsp;</td>
      <td>
        <p> Writes output in <a href="../vep_formats.html#vcfout">VCF format</a>.
        Consequences are added in the INFO field of the VCF file, using the key
        "CSQ". Data fields are encoded separated by "&#124;"; the order of
        fields is written in the VCF header. Output fields in the "CSQ" INFO field can be selected by
        using <a href="#opt_fields">--fields</a>.</p>
        <p> If the input format
        was VCF, the file will remain unchanged save for the addition of the CSQ
        field (unless using any filtering).<br/><br/> Custom data added with <a
        href="#opt_custom">--custom</a> are added as separate fields, using the
        key specified for each data file.<br/><br/> Commas in fields are
        replaced with ampersands (&amp;) to preserve VCF format.</p>
        <i>Not used by default</i>
      </td>
      <td>&nbsp;</td>
      <td>
        <div style="padding:2px 0px"><a href="#opt_json">--json</a></div>
        <div style="padding:2px 0px"><a href="#opt_tab">--tab</a></div>
      </td>
    </tr>
    <tr id="opt_tab">
      <td><pre>--tab</pre></td>
      <td>&nbsp;</td>
      <td>
        Writes output in <a href="../vep_formats.html#tab">tab-delimited format</a>.
        <i>Not used by default</i>
      </td>
      <td>&nbsp;</td>
      <td>
        <div style="padding:2px 0px"><a href="#opt_json">--json</a></div>
        <div style="padding:2px 0px"><a href="#opt_vcf">--vcf</a></div>
      </td>
    </tr>
    <tr id="opt_json">
      <td><pre>--json</pre></td>
      <td>&nbsp;</td>
      <td>
        Writes output in <a href="../vep_formats.html#json">JSON format</a>.
        <i>Not used by default</i>
      </td>
      <td>&nbsp;</td>
      <td>
        <div style="padding:2px 0px"><a href="#opt_tab">--tab</a></div>
        <div style="padding:2px 0px"><a href="#opt_vcf">--vcf</a></div>
      </td>
    </tr>  
    <tr id="opt_compress_output">
      <td><pre>--compress_output [gzip|bgzip]</pre></td>
      <td>&nbsp;</td>
      <td>
        Writes output compressed using either gzip or bgzip.
        <i>Not used by default</i>
      </td>
      <td colspan="2">&nbsp;</td>
    </tr>
    <tr id="opt_fields">
      <td><pre>--fields [list]</pre></td>
      <td>&nbsp;</td>
      <td>
        <p>Configure the output format using a comma separated list of
        fields.<br />
        Can only be used with <a href="../vep_formats.html#tab">tab</a> (<a href="#opt_tab">--tab</a>) or <a href="../vep_formats.html#vcfout">VCF format</a> (<a href="#opt_vcf">--vcf</a>) output.<br />
        For the tab format output, the selected fields may be those present in the default <a href="../vep_formats.html#output">output columns</a>, or any of
        those that appear in the Extra column (including those added by plugins or custom annotations) if the appropriate output is available (e.g. use <a href="#opt_show_ref_allele">--show_ref_allele</a> to access 'REF_ALLELE'). Output remains tab-delimited.<br />
        For the VCF format output, the selected fields are those present within the "CSQ" INFO field.
        </p>
        <p>Example of command for the tab output:</p>
        <pre class="code sh_sh">--tab --fields "Uploaded_variation,Location,Allele,Gene"</pre>
        <p>Example of command for the VCF format output:</p>
        <pre class="code sh_sh">--vcf --fields "Allele,Consequence,Feature_type,Feature"</pre>
        <i>Not used by default</i>
      </td>
      <td colspan="2">&nbsp;</td>
    </tr>
    <tr id="opt_minimal">
      <td><pre>--minimal</pre></td>
      <td>&nbsp;</td>
      <td>
        Convert alleles to their most minimal representation before
        consequence calculation i.e. sequence that is identical between
        each pair of reference and alternate alleles is trimmed off from
        both ends, with coordinates adjusted accordingly.<br />
        Note this may
        lead to discrepancies between input coordinates and coordinates
        reported by VEP relative to transcript sequences; to avoid
        issues, use <a href="#opt_allele_number">--allele_number</a>
        and/or ensure that your input variants have unique identifiers.
        The MINIMISED flag is set in the VEP output where relevant.
        <i>Not used by default</i>
      </td>
      <td>MINIMISED</td>
      <td>
        <div style="padding:2px 0px"><a href="#opt_individual">--individual</a></div>
      </td> 
    </tr>
  </table>


 
  <br/> 
  <hr/>
  <h2 id="output">Output options</h2>
  
  <table class="ss">
    <tr>
      <th width="20%">Flag</th>
      <th width="8%">Alternate</th>
      <th>Description</th>
      <th width="10%">Output fields</th>
      <th width="10%">Incompatible with</th>
    </tr>
    <tr id="opt_variant_class">
      <td><pre>--variant_class</pre></td>
      <td>&nbsp;</td>
      <td>
        Output the Sequence Ontology <a
        href="/info/genome/variation/prediction/classification.html#classes">variant
        class</a>. <i>Not used by default</i>
      </td>
      <td>VARIANT_CLASS</td>
      <td>&nbsp;</td>
    </tr>
    <tr id="opt_sift">
      <td><pre>--sift [p<b>|</b>s<b>|</b>b]</pre></td>
      <td>&nbsp;</td>
      <td>
        <span style="color:red;">Species limited</span>
        <a rel="external" href="http://sift.bii.a-star.edu.sg/" target="_blank">SIFT</a>
        predicts whether an amino acid substitution affects protein
        function based on sequence homology and the physical properties
        of amino acids. VEP can output the <b>p</b>rediction term,
        <b>s</b>core or <b>b</b>oth. <i>Not used by default</i>
      </td>
      <td>SIFT</td>
      <td>
        <div style="padding:2px 0px"><a href="#opt_most_severe">--most_severe</a></div>
        <div style="padding:2px 0px"><a href="#opt_summary">--summary</a></div>
      </td>
    </tr>
    <tr id="opt_polyphen">
      <td><pre>--polyphen [p<b>|</b>s<b>|</b>b]</pre></td>
      <td>&nbsp;</td>
      <td>
        <span style="color:red;">Human only</span>
        <a rel="external" href="http://genetics.bwh.harvard.edu/pph2/"
        target="_blank">PolyPhen</a> is a tool which predicts possible
        impact of an amino acid substitution on the structure and
        function of a human protein using straightforward physical and
        comparative considerations. VEP can output the
        <b>p</b>rediction term, <b>s</b>core or <b>b</b>oth. VEP uses the
        humVar score by default - use <a href="#opt_humdiv">--humdiv</a> to
        retrieve the humDiv score. <i>Not used by default</i>
      </td>
      <td>PolyPhen</td>
      <td>
        <div style="padding:2px 0px"><a href="#opt_most_severe">--most_severe</a></div>
        <div style="padding:2px 0px"><a href="#opt_summary">--summary</a></div>
      </td>
    </tr>
    <tr id="opt_humdiv">
      <td><pre>--humdiv</pre></td>
      <td>&nbsp;</td>
      <td>
        <span style="color:red;">Human only</span>
        Retrieve the <a
        href="http://genetics.bwh.harvard.edu/pph2/dokuwiki/overview#prediction"
        rel="external">humDiv PolyPhen prediction</a> instead of the default
        humVar. <i>Not used by default</i>
      </td>
      <td>PolyPhen</td>
      <td>&nbsp;</td>
    </tr>
    <tr id="opt_nearest">
      <td><pre>--nearest [transcript|gene|symbol]</pre></td>
      <td>&nbsp;</td>
      <td>
        <p> Retrieve the transcript or gene with the nearest protein-coding transcription start site (TSS)
        to each input variant. Use "transcript" to retrieve the transcript
        stable ID, "gene" to retrieve the gene stable ID, or "symbol" to retrieve
        the gene symbol. Note that the nearest TSS may not belong to a transcript
        that overlaps the input variant, and more than one may be reported in the
        case where two are equidistant from the input coordinates.</p>

        <p> Currently only available when using a
        <a href="vep_cache.html#cache">cache</a> annotation source, and <a href="vep_download.html#additional">requires
        the Set::IntervalTree perl module</a>.</p>
        <i>Not used by default</i>
      </td>
      <td>NEAREST</td>
      <td>&nbsp;</td>
    </tr>
    <tr id="opt_distance">
      <td><pre>--distance [bp_distance(,downstream_distance_if_different)]</pre></td>
      <td>&nbsp;</td>
      <td>
        Modify the distance up and/or downstream between a variant and a transcript
        for which VEP will assign the upstream_gene_variant or downstream_gene_variant
        consequences. Giving one distance will modify both up- and downstream distances;
        prodiving two separated by commas will set the up- (5') and down- (3') stream
        distances respectively.
        <i>Default: 5000</i>
      </td>
      <td colspan="2">&nbsp;</td>
    </tr>
    <tr id="opt_overlaps">
      <td><pre>--overlaps</pre></td>
      <td>&nbsp;</td>
      <td>
        Report the proportion and length of a transcript overlapped by a structural variant in VCF format.
      </td>
      <td colspan="2">&nbsp;</td>
    </tr>
    <tr id="opt_gene_phenotype">
      <td><pre>--gene_phenotype</pre></td>
      <td>&nbsp;</td>
      <td>
        Indicates if the overlapped gene is associated with a phenotype, disease
        or trait. See <a
        href="/info/genome/variation/phenotype/sources_phenotype_documentation.html">list
        of phenotype sources</a>. <i>Not used by default</i>
      </td>
      <td>GENE_PHENO</td>
      <td>&nbsp;</td>
    </tr>
    <tr id="opt_regulatory">
      <td><pre>--regulatory</pre></td>
      <td>&nbsp;</td>
      <td>
        Look for overlaps with regulatory regions. VEP can also
        report if a variant falls in a high information position within a
        transcription factor binding site. Output lines have a Feature
        type of RegulatoryFeature or MotifFeature. <i>Not used by
        default</i>
      </td>
      <td>MOTIF_NAME, MOTIF_POS, HIGH_INF_POS, MOTIF_SCORE_CHANGE</td>
      <td>&nbsp;</td>
    </tr>
    <tr id="opt_cell_type">
      <td><pre>--cell_type</pre></td>
      <td>&nbsp;</td>
      <td>
        Report only regulatory regions that are found in the given cell
        type(s). Can be a single cell type or a comma-separated list.
        The functional type in each cell type is reported under
        CELL_TYPE in the output. To retrieve a list of cell types, use
        <a href="vep_options.html#opt_cell_type">--cell_type list</a>. <i>Not used by default</i>
      </td>
      <td>CELL_TYPE</td>
      <td>&nbsp;</td>
    </tr>
    <tr id="opt_individual">
      <td><pre>--individual [all|ind list]</pre></td>
      <td>&nbsp;</td>
      <td>
        Consider only alternate alleles present in the genotypes of the
        specified individual(s). May be a single individual, a
        comma-separated list or "all" to assess all individuals
        separately. Individual variant combinations homozygous for the
        given reference allele will not be reported. Each individual and
        variant combination is given on a separate line of output. Only
        works with VCF files containing individual genotype data;
        individual IDs are taken from column headers. <i>Not used by
        default</i>
      </td>
      <td>IND, ZYG</td>
      <td>
        <div style="padding:2px 0px"><a href="#opt_minimal">--minimal</a></div>
      </td>
    </tr>
    <tr id="opt_phased">
      <td><pre>--phased</pre></td>
      <td>&nbsp;</td>
      <td>
        Force VCF genotypes to be interpreted as phased. For use with
        plugins that depend on phased data. <i>Not used by
        default</i>
      </td>
      <td colspan="2">&nbsp;</td>
    </tr>
    <tr id="opt_allele_number">
      <td><pre>--allele_number</pre></td>
      <td>&nbsp;</td>
      <td>
        Identify allele number from VCF input, where 1 = first ALT
        allele, 2 = second ALT allele etc. Useful when using <a href="#opt_minimal">--minimal</a>
        <i>Not used by default</i>
      </td>
      <td>ALLELE_NUM</td>
      <td>&nbsp;</td>
    </tr>
    <tr id="opt_show_ref_allele">
      <td><pre>--show_ref_allele</pre></td>
      <td>&nbsp;</td>
      <td>
        Adds the reference allele in the output. Mainly useful for the VEP "default" and tab-delimited output formats.
        <i>Not used by default</i>
      </td>
      <td>REF_ALLELE</td>
      <td>&nbsp;</td>
    </tr> 
    <tr id="opt_total_length">
      <td><pre>--total_length</pre></td>
      <td>&nbsp;</td>
      <td>
        Give cDNA, CDS and protein positions as Position/Length. <i>Not
        used by default</i>
      </td>
      <td colspan="2">&nbsp;</td>
    </tr>
    <tr id="opt_numbers">
      <td><pre>--numbers</pre></td>
      <td>&nbsp;</td>
      <td>
        Adds affected exon and intron numbering to to output. Format is
        Number/Total. <i>Not used by default</i>
      </td>
      <td>EXON, INTRON</td>
      <td>
        <div style="padding:2px 0px"><a href="#opt_most_severe">--most_severe</a></div>
        <div style="padding:2px 0px"><a href="#opt_summary">--summary</a></div>
      </td>
    </tr>
    <tr id="opt_no_escape">
      <td><pre>--no_escape</pre></td>
      <td>&nbsp;</td>
      <td>
        Don't URI escape HGVS strings. <i>Default = escape</i>
      </td>
      <td colspan="2">&nbsp;</td>
    </tr>
    <tr id="opt_keep_csq">
      <td><pre>--keep_csq</pre></td>
      <td>&nbsp;</td>
      <td>
        Don't overwrite existing CSQ entry in
        <a href="../vep_formats.html#vcfout">VCF INFO field</a>. <i>Overwrites
        by default</i>
      </td>
      <td colspan="2">&nbsp;</td>
    </tr>
    <tr id="opt_vcf_info_field">
      <td><pre>--vcf_info_field [CSQ|ANN|(other)]</pre></td>
      <td>&nbsp;</td>
      <td>
        Change the name of the INFO key that VEP write the consequences to
        in its <a href="../vep_formats.html#vcfout">VCF output</a>. Use "ANN"
        for compatibility with other tools such as <a
        href="http://snpeff.sourceforge.net/" rel="external">snpEff</a>.
        <i>Default: CSQ</i>
      </td>
      <td colspan="2">&nbsp;</td>
    </tr>
    <tr id="opt_terms">
      <td><pre>--terms [SO|display|NCBI]</pre></td>
      <td><pre>-t</pre></td>
      <td>
        The type of consequence terms to output. The Ensembl terms are
        described <a
        href="/info/genome/variation/prediction/predicted_data.html#consequences"
        target="_blank">here</a>. The <a
        rel="external" href="http://www.sequenceontology.org/" target="_blank">Sequence
        Ontology</a> is a joint effort by genome annotation centres to
        standardise descriptions of biological sequences. <i>Default =
        "SO"</i>
      </td>
      <td colspan="2">&nbsp;</td>
    </tr>
    <tr id="opt_no_headers">
      <td><pre>--no_headers</pre></td>
      <td>&nbsp;</td>
      <td>
        Don't write header lines in output files. <i>Default = add headers</i>
      </td>
      <td colspan="2">&nbsp;</td>
    </tr>
    <tr id="opt_shift3prime">
      <td><pre>--shift_3prime [0<b>|</b>1]</pre></td>
      <td>&nbsp;</td>
      <td> Right aligns all variants relative to their associated transcripts prior to consequence calculation. <br/>
	   An example using this option can be found <a
        href="/info/docs/tools/vep/script/vep_other.html#shifting"
        target="_blank">here</a>.<br/> <i>Default = 0</i>
      </td>
      <td></td>
      <td>
        <div style="padding:2px 0px"><a href="#opt_shift_hgvs">--shift_hgvs</a></div>
      </td>
    </tr>
    <tr id="opt_shiftgenomic">
      <td><pre>--shift_genomic [0<b>|</b>1]</pre></td>
      <td>&nbsp;</td>
      <td> Right aligns all variants, including intergenic variants, before consequence calculation and updates the <i>Location</i> field. <br/>
           An example using this option can be found <a
        href="/info/docs/tools/vep/script/vep_other.html#shifting"
        target="_blank">here</a>.<br/> <i>Default = 0</i>
      </td>
      <td></td>
      <td>
        <div style="padding:2px 0px"><a href="#opt_shift_hgvs">--shift_hgvs</a></div>
      </td>
    </tr>
    <tr id="opt_shiftlength">
      <td><pre>--shift_length</pre></td>
      <td>&nbsp;</td>
      <td> Reports the distance each variant has been shifted when used in conjuction with <a href="#opt_shift3prime">--shift_3prime</a></div>
      </td>
      <td></td>
      <td>
      </td>
    </tr>
  </table>
    
  
  <br/> 
  <hr/>
  <h2 id="ident">Identifiers</h2>
  
  <table class="ss">
    <tr>
      <th width="20%">Flag</th>
      <th width="8%">Alternate</th>
      <th>Description</th>
      <th width="10%">Output fields</th>
      <th width="10%">Incompatible with</th>
    </tr>
    <tr id="opt_hgvs">
      <td><pre>--hgvs</pre></td>
      <td>&nbsp;</td>
      <td>
        Add <a rel="external" href="//varnomen.hgvs.org"
        target="_blank">HGVS</a> nomenclature based on Ensembl stable
        identifiers to the output. Both coding and protein sequence
        names are added where appropriate. To generate HGVS identifiers when
        using <a href="#opt_cache">--cache</a> or <a
        href="#opt_offline">--offline</a> you must use a FASTA file and <a
        href="#opt_fasta">--fasta</a>. HGVS notations given on Ensembl
        identifiers are <a href="/info/genome/stable_ids/index.html">
        versioned</a>. <i>Not used by default</i>
      </td>
      <td>HGVSc, HGVSp, HGVS_OFFSET</td>
      <td>&nbsp;</td>
    </tr>
    <tr id="opt_hgvsg">
      <td><pre>--hgvsg</pre></td>
      <td>&nbsp;</td>
      <td>
        Add genomic <a rel="external" href="//varnomen.hgvs.org"
        target="_blank">HGVS</a> nomenclature based on the input chromosome name.
        To generate HGVS identifiers when
        using <a href="#opt_cache">--cache</a> or <a
        href="#opt_offline">--offline</a> you must use a FASTA file and <a
        href="#opt_fasta">--fasta</a>. <i>Not used by default</i>
      </td>
      <td>HGVSg</td>
      <td>&nbsp;</td>
    </tr>
    <tr id="opt_spdi">
      <td><pre>--spdi</pre></td>
      <td>&nbsp;</td>
      <td>
      Add genomic <a rel="external" href="https://www.ncbi.nlm.nih.gov/variation/notation/",
      target="_blank">SPDI</a> notation. To generate SPDI when using <a href="#opt_cache">--cache</a> or <a
      href="#opt_offline">--offline</a> you must use a FASTA file and <a
      href="#opt_fasta">--fasta</a>. <i>Not used by default</i>
      </td>
      <td>SPDI</td>
      <td>&nbsp;</td>
    </tr>
    <tr id="opt_shift_hgvs">
      <td><pre>--shift_hgvs [0|1]</pre></td>
      <td>&nbsp;</td>
      <td>
        Enable or disable 3' shifting of HGVS notations. When enabled, this
        causes ambiguous insertions or deletions (typically in repetetive
        sequence tracts) to be "shifted" to their most 3' possible
        coordinates (relative to the transcript sequence and strand) before
        the HGVS notations are calculated; the flag HGVS_OFFSET is set to
        the number of bases by which the variant has shifted, relative to
        the input genomic coordinates. Disabling retains the original
        input coordinates of the variant. <i>Default: 1 (shift)</i>
      </td>
      <td colspan="2">&nbsp;</td>
    </tr>
    <tr id="opt_transcript_version">
      <td><pre>--transcript_version</pre></td>
      <td>&nbsp;</td>
      <td>
        Add version numbers to Ensembl transcript identifiers</i>
      </td>
      <td colspan="2">&nbsp;</td>
    </tr>
    <tr id="opt_protein">
      <td><pre>--protein</pre></td>
      <td>&nbsp;</td>
      <td>
        Add the Ensembl protein identifier to the output where
        appropriate. <i>Not used by default</i>
      </td>
      <td>ENSP</td>
      <td>
        <div style="padding:2px 0px"><a href="#opt_most_severe">--most_severe</a></div>
        <div style="padding:2px 0px"><a href="#opt_summary">--summary</a></div>
      </td>
    </tr>
    <tr id="opt_symbol">
      <td><pre>--symbol</pre></td>
      <td>&nbsp;</td>
      <td>
        Adds the gene symbol (e.g. HGNC) (where available) to the output.
        <i>Not used by default</i>
      </td>
      <td>SYMBOL, SYMBOL_SOURCE, HGNC_ID</td>
      <td>
        <div style="padding:2px 0px"><a href="#opt_most_severe">--most_severe</a></div>
        <div style="padding:2px 0px"><a href="#opt_summary">--summary</a></div>
      </td>
    </tr>
    <tr id="opt_ccds">
      <td><pre>--ccds</pre></td>
      <td>&nbsp;</td>
      <td>
        Adds the CCDS transcript identifer (where available) to the
        output. <i>Not used by default</i>
      </td>
      <td>CCDS</td>
      <td>
        <div style="padding:2px 0px"><a href="#opt_most_severe">--most_severe</a></div>
        <div style="padding:2px 0px"><a href="#opt_summary">--summary</a></div>
      </td>
    </tr>
    <tr id="opt_uniprot">
      <td><pre>--uniprot</pre></td>
      <td>&nbsp;</td>
      <td>
        Adds best match accessions for translated protein products from three <a
        href="http://www.ebi.ac.uk/uniprot" rel="external">UniProt</a>-related
        databases (SWISSPROT, TREMBL and UniParc) to the output. <i>Not used by
        default</i>
      </td>
      <td>SWISSPROT, TREMBL, UNIPARC</td>
      <td>
        <div style="padding:2px 0px"><a href="#opt_most_severe">--most_severe</a></div>
        <div style="padding:2px 0px"><a href="#opt_summary">--summary</a></div>
      </td>
    </tr>
    <tr id="opt_tsl">
      <td><pre>--tsl</pre></td>
      <td>&nbsp;</td>
      <td>
        Adds the <a href="/Help/Glossary?id=492">transcript support level</a>
        for this transcript to the output. <i>Not used by default</i>
        <div class="info" style="max-width:350px;margin:5px 0px 5px 10px">
          <h3>Note</h3>
          <div class="message-pad" style="padding:5px 10px;background-color:#FFF">
            Only available for human on the GRCh38 assembly
          </div>
        </div>
      </td>
      <td>TSL</td>
      <td>
        <div style="padding:2px 0px"><a href="#opt_most_severe">--most_severe</a></div>
        <div style="padding:2px 0px"><a href="#opt_summary">--summary</a></div>
      </td>
    </tr>
    <tr id="opt_appris">
      <td><pre>--appris</pre></td>
      <td>&nbsp;</td>
      <td>
        Adds the <a href="/Help/Glossary?id=521">APPRIS</a> isoform annotation
        for this transcript to the output. <i>Not used by default</i>
        <div class="info" style="max-width:350px;margin:5px 0px 5px 10px">
          <h3>Note</h3>
          <div class="message-pad" style="padding:5px 10px;background-color:#FFF">
            Only available for human on the GRCh38 assembly
          </div>
        </div>
      </td>
      <td>APPRIS</td>
      <td>
        <div style="padding:2px 0px"><a href="#opt_most_severe">--most_severe</a></div>
        <div style="padding:2px 0px"><a href="#opt_summary">--summary</a></div>
      </td>
    </tr>
    <tr id="opt_canonical">
      <td><pre>--canonical</pre></td>
      <td>&nbsp;</td>
      <td>
        Adds a flag indicating if the transcript is the canonical
        transcript for the gene. <i>Not used by default</i>
      </td>
      <td>CANONICAL</td>
      <td>
        <div style="padding:2px 0px"><a href="#opt_most_severe">--most_severe</a></div>
        <div style="padding:2px 0px"><a href="#opt_summary">--summary</a></div>
      </td>
    </tr>
    <tr id="opt_mane">
      <td><pre>--mane</pre></td>
      <td>&nbsp;</td>
      <td>
        Adds a flag indicating if the transcript is the <a href="/info/genome/genebuild/mane.html">MANE Select</a>
        transcript for the gene. <i>Not used by default</i>
        <div class="info" style="max-width:350px;margin:5px 0px 5px 10px">
          <h3>Note</h3>
          <div class="message-pad" style="padding:5px 10px;background-color:#FFF">
            Only available for human on the GRCh38 assembly
          </div>
        </div>
      </td>
      <td>MANE</td>
      <td>
        <div style="padding:2px 0px"><a href="#opt_most_severe">--most_severe</a></div>
        <div style="padding:2px 0px"><a href="#opt_summary">--summary</a></div>
      </td>
    </tr>
    <tr id="opt_biotype">
      <td><pre>--biotype</pre></td>
      <td>&nbsp;</td>
      <td>
        Adds the biotype of the transcript or regulatory feature. <i>Not used by
        default</i>
      </td>
      <td>BIOTYPE</td>
      <td>
        <div style="padding:2px 0px"><a href="#opt_most_severe">--most_severe</a></div>
        <div style="padding:2px 0px"><a href="#opt_summary">--summary</a></div>
      </td>
    </tr>
    <tr id="opt_domains">
      <td><pre>--domains</pre></td>
      <td>&nbsp;</td>
      <td>
        Adds names of overlapping protein domains to output. <i>Not used
        by default</i>
      </td>
      <td>DOMAINS</td>
      <td>
        <div style="padding:2px 0px"><a href="#opt_most_severe">--most_severe</a></div>
        <div style="padding:2px 0px"><a href="#opt_summary">--summary</a></div>
      </td>
    </tr>
    <tr id="opt_xref_refseq">
      <td><pre>--xref_refseq</pre></td>
      <td>&nbsp;</td>
      <td>
        Output aligned RefSeq mRNA identifier for transcript. <i>Not used by default</i>
        <div class="info" style="margin:5px 0px 5px 10px">
          <h3>Note</h3>
          <div class="message-pad" style="padding:5px 10px;background-color:#FFF">
            The RefSeq and Ensembl transcripts aligned in this way MAY NOT,
            AND FREQUENTLY WILL NOT, match exactly in sequence, exon
            structure and protein product
          </div>
        </div>
      </td>
      <td>RefSeq</td>
      <td>
        <div style="padding:2px 0px"><a href="#opt_most_severe">--most_severe</a></div>
        <div style="padding:2px 0px"><a href="#opt_summary">--summary</a></div>
      </td>
    </tr>
    <tr id="opt_synonyms">
      <td><pre>--synonyms [file]</pre></td>
      <td>&nbsp;</td>
      <td>
        Load a file of chromosome synonyms. File should be tab-delimited
        with the primary identifier in column 1 and the synonym in column 2.
        Synonyms allow different chromosome identifiers to be used in the
        input file and any annotation source (cache, database, GFF, custom file, FASTA file).
        <i>Not used by default</i>
      </td>
      <td>&nbsp;</td>
    </tr>
  </table>
    
  
  <br/>
  <hr/>
  <h2 id="existing">Co-located variants</h2>
  
  <table class="ss">
    <tr>
      <th width="20%">Flag</th>
      <th width="8%">Alternate</th>
      <th>Description</th>
      <th width="10%">Output fields</th>
      <th width="10%">Incompatible with</th>
    </tr>
    <tr id="opt_check_existing">
      <td><pre>--check_existing</pre></td>
      <td>&nbsp;</td>
      <td>
        Checks for the existence of known variants that are co-located
        with your input. By default the alleles are compared and variants on an allele-specific
        basis - to compare only coordinates,
        use <a href="#opt_no_check_alleles">--no_check_alleles</a>.
        <br/><br/>Some databases may contain variants with unknown (null) alleles and these
        are included by default; to exclude them use <a href="#opt_exclude_null_alleles">
        --exclude_null_alleles</a>.
        <br/><br/>See <a href="vep_other.html#colocated">this page</a> for more details.
        <br/><br/><i>Not used by default</i>
      </td>
      <td>Existing_variation, CLIN_SIG, SOMATIC, PHENO</td>
      <td>&nbsp;</td>
    </tr>
    <tr id="opt_check_svs">
      <td><pre>--check_svs</pre></td>
      <td>&nbsp;</td>
      <td>
        Checks for the existence of structural variants that overlap your
        input. Currently requires database access. <i>Not used by
        default</i>
      </td>
      <td>SV</td>
      <td><a href="#opt_offline">--offline</a></td>
    </tr>
    <tr id="opt_clin_sig_allele">
      <td><pre>--clin_sig_allele [1|0]</pre></td>
      <td>&nbsp;</td>
      <td>
        Return allele specific clinical significance. Setting this option to 0 will provide all known clinical significance values at the given locus.
        <i>Default: 1 (Provide allele-specific annotations)</i>
      </td>
      <td>CLIN_SIG</td>
      <td>&nbsp;</td>
    </tr>
    <tr id="opt_exclude_null_alleles">
      <td><pre>--exclude_null_alleles</pre></td>
      <td>&nbsp;</td>
      <td>
        Do not include variants with unknown alleles when checking for co-located variants.
        Our human database contains variants from HGMD and COSMIC for which the
        alleles are not publically available; by default these are included when using
        <a href="#opt_check_existing">--check_existing</a>, use this flag to exclude them.
        <i>Not used by default</i>
      </td>
      <td colspan="2">&nbsp;</td>
    </tr>
    <tr id="opt_no_check_alleles">
      <td><pre>--no_check_alleles</pre></td>
      <td>&nbsp;</td>
      <td>
        When checking for existing variants, by default VEP only reports a co-located
        variant if none of the input alleles are novel. For example,
        if your input variant has alleles A/G, and an existing co-located
        variant has alleles A/C, the co-located variant will not be
        reported.<br/><br/>Strand is also taken into account - in the
        same example, if the input variant has alleles T/G but on the
        negative strand, then the co-located variant <b>will</b> be
        reported since its alleles match the reverse complement of
        input variant.
        <br/><br/>Use this flag to disable this behaviour and compare using coordinates alone.
        <i>Not used by default</i>
      </td>
      <td colspan="2">&nbsp;</td>
    </tr>
    <tr id="opt_af">
      <td><pre>--af</pre></td>
      <td>&nbsp;</td>
      <td>
        Add the global allele frequency (AF) from 1000 Genomes
        Phase 3 data for any known co-located variant to the output. For this and all --af_* flags, the frequency reported is for the <b>input allele</b> only, not necessarily the non-reference or derived allele. <i>Not used
        by default</i>
      </td>
      <td>AF</td>
      <td>&nbsp;</td>
    </tr>
    <tr id="opt_max_af">
      <td><pre>--max_af</pre></td>
      <td>&nbsp;</td>
      <td>
        Report the highest allele frequency observed in any population from 1000 genomes, ESP or gnomAD. <i>Not used
        by default</i>
      </td>
      <td>MAX_AF, MAX_AF_POPS</td>
      <td>
        <div style="padding:2px 0px"><a href="#opt_database">--database</a></div>
      </td>
    </tr>
    <tr id="opt_af_1kg">
      <td><pre>--af_1kg</pre></td>
      <td>&nbsp;</td>
      <td>
        Add allele frequency from continental populations (AFR,AMR,EAS,EUR,SAS) of <a href="http://www.1000genomes.org/" rel="external">1000
        Genomes Phase 3</a> to the output. Must be used with <a
        href="#opt_cache">--cache</a>. <i>Not used by default</i>
      </td>
      <td>AFR_AF, AMR_AF, EAS_AF, EUR_AF, SAS_AF</td>
      <td>
        <div style="padding:2px 0px"><a href="#opt_database">--database</a></div>
      </td>
    </tr>
    <tr id="opt_af_esp">
      <td><pre>--af_esp</pre></td>
      <td>&nbsp;</td>
      <td>
        Include allele frequency from <a href="http://evs.gs.washington.edu/EVS/" rel="external">NHLBI-ESP</a> populations. Must be used with
        <a href="#opt_cache">--cache</a>. <i>Not used by default</i>
      </td>
      <td>AA_AF, EA_AF</td>
      <td>
        <div style="padding:2px 0px"><a href="#opt_database">--database</a></div>
      </td>
    </tr>
    <tr id="opt_af_gnomad">
      <td><pre>--af_gnomad</pre></td>
      <td>&nbsp;</td>
      <td>
        Include allele frequency from <a href="http://gnomad.broadinstitute.org/" rel="external">Genome Aggregation Database (gnomAD)</a> exome populations. Note only data from the gnomAD exomes are included; to retrieve data from the additional genomes data set,
        see <a href="vep_example.html#gnomad">this guide</a>. Must be used with
        <a href="#opt_cache">--cache</a> <i>Not used by default</i>
      </td>
      <td>gnomAD_AF, gnomAD_AFR_AF, gnomAD_AMR_AF, gnomAD_ASJ_AF, gnomAD_EAS_AF, gnomAD_FIN_AF, gnomAD_NFE_AF, gnomAD_OTH_AF, gnomAD_SAS_AF</td>
      <td>
        <div style="padding:2px 0px"><a href="#opt_database">--database</a></div>
      </td>
    </tr>
    <tr id="opt_af_exac">
      <td><pre>--af_exac</pre></td>
      <td>&nbsp;</td>
      <td>
        Include allele frequency from <a href="http://exac.broadinstitute.org/" rel="external">ExAC project</a> populations. Must be used with
        <a href="#opt_cache">--cache</a>. <i>Not used by default</i>
        
        <div class="info" style="max-width:800px;margin:5px 0px 5px 10px">
          <h3>Note</h3>
          <div class="message-pad" style="padding:5px 10px;background-color:#FFF">
            ExAC data has been superceded by <a href="#opt_af_gnomad">gnomAD</a>.
            This flag remains for those wishing to use older cache versions containing ExAC data.
          </div>
        </div>
      </td>
      <td>ExAC_AF, ExAC_Adj_AF, ExAC_AFR_AF, ExAC_AMR_AF, ExAC_EAS_AF, ExAC_FIN_AF, ExAC_NFE_AF, ExAC_OTH_AF, ExAC_SAS_AF</td>
      <td>
        <div style="padding:2px 0px"><a href="#opt_database">--database</a></div>
      </td>
    </tr>
    <tr id="opt_pubmed">
      <td><pre>--pubmed</pre></td>
      <td>&nbsp;</td>
      <td>
        Report Pubmed IDs for publications that cite existing variant.
        Must be used with <a href="#opt_cache">--cache</a>. <i>Not used
        by default</i>
      </td>
      <td>PUBMED</td>
      <td>
        <div style="padding:2px 0px"><a href="#opt_database">--database</a></div>
      </td>
    </tr>
    <tr id="opt_var_synonyms">
      <td><pre>--var_synonyms</pre></td>
      <td>&nbsp;</td>
      <td>
        Report known synonyms for co-located variants.
        Must be used with <a href="#opt_cache">--cache</a>. <i>Not used
        by default</i>
      </td>
      <td>VAR_SYNONYMS</td>
      <td>
        <div style="padding:2px 0px"><a href="#opt_database">--database</a></div>
      </td>
    </tr>
    <tr id="opt_failed">
      <td><pre>--failed [0|1]</pre></td>
      <td>&nbsp;</td>
      <td>
        When checking for co-located variants, by default VEP
        will exclude variants that have been flagged as failed. Set this
        flag to include such variants. <i>Default: 0 (exclude)</i>
      </td>
      <td colspan="2">&nbsp;</td>
    </tr>
  </table>
  
  
  <br/>
  <hr/> 
  <h2 id="filt">Filtering and QC options</h2>
    
    <p> <b>NOTE:</b> The filtering options here filter your results
    <b>before</b> they are written to your output file. Using VEP's <a
    href="vep_filter.html">filtering script</a>, it is possible to filter your
    results <b>after</b> VEP has run. This way you can retain all of the
    results and run multiple filter sets on the same results to find different
    data of interest. </p>
  
  <table class="ss">
    <tr>
      <th width="20%">Flag</th>
      <th width="8%">Alternate</th>
      <th>Description</th>
      <th width="10%">Output fields</th>
      <th width="10%">Incompatible with</th>
    </tr>
    <tr id="opt_gencode_basic">
      <td><pre>--gencode_basic</pre></td>
      <td>&nbsp;</td>
      <td>
        Limit your analysis to transcripts belonging to the GENCODE
        basic set. This set has fragmented or problematic transcripts removed. <i>Not used by default</i>
      </td>
      <td>&nbsp;</td>
      <td>
        <div style="padding:2px 0px"><a href="#opt_refseq">--refseq</a></div>
      </td>
    </tr>
    <tr id="opt_exclude_predicted">
      <td><pre>--exclude_predicted</pre></td>
      <td>&nbsp;</td>
      <td>
        When using the RefSeq or merged cache, exclude predicted transcripts
        (i.e. those with identifiers beginning with "XM_" or "XR_").
      </td>
      <td colspan="2">&nbsp;</td>
    </tr>
    <tr id="opt_transcript_filter">
      <td><pre>--transcript_filter</pre></td>
      <td>&nbsp;</td>
      <td>
        <p><b>ADVANCED</b> Filter transcripts according to any arbitrary set of rules. Uses similar notation to <a href="vep_filter.html">filter_vep</a>.</p>
        <p>You may filter on any key defined in the root of the transcript object; most commonly this will be "stable_id":</p>
        <pre class="code sh_sh">--transcript_filter "stable_id match N[MR]_"</pre>
      </td>
      <td colspan="2">&nbsp;</td>
    </tr>
    <tr id="opt_check_ref">
      <td><pre>--check_ref</pre></td>
      <td>&nbsp;</td>
      <td>
        Force VEP to check the supplied reference allele against
        the sequence stored in the Ensembl Core database or supplied <a href="#opt_fasta">FASTA file</a>. Lines that do
        not match are skipped. <i>Not used by default</i>
      </td>
      <td>&nbsp;</td>
      <td>
        <div style="padding:2px 0px"><a href="#opt_lookup_ref">--lookup_ref</a></div>
      </td>
    </tr>
    <tr id="opt_lookup_ref">
      <td><pre>--lookup_ref</pre></td>
      <td>&nbsp;</td>
      <td>
        Force overwrite the supplied reference allele with
        the sequence stored in the Ensembl Core database or supplied <a href="#opt_fasta">FASTA file</a>. <i>Not used by default</i>
      </td>
      <td>&nbsp;</td>
      <td>
        <div style="padding:2px 0px"><a href="#opt_check_ref">--check_ref</a></div>
      </td>
    </tr>
    <tr id="dont_skip">
      <td><pre>--dont_skip</pre></td>
      <td>&nbsp;</td>
      <td>
        Don't skip input variants that fail validation, e.g. those that
        fall on unrecognised sequences.<br />Combining <a href="#opt_check_ref">--check_ref</a> with <a href="#opt_dont_skip">--dont_skip</a> will add a CHECK_REF output field when the given reference does not match the underlying reference sequence.
      </td>
      <td colspan="2">CHECK_REF</td>
    </tr>
    <tr id="opt_allow_non_variant">
      <td><pre>--allow_non_variant</pre></td>
      <td>&nbsp;</td>
      <td>
        When using VCF format as input and output, by default VEP
        will skip non-variant lines of input (where the ALT allele is
        null). Enabling this option the lines will be printed in the VCF
        output with no consequence data added.
      </td>
      <td colspan="2">&nbsp;</td>
    </tr>
    <tr id="opt_chr">
      <td><pre>--chr [list]</pre></td>
      <td>&nbsp;</td>
      <td>
        Select a subset of chromosomes to analyse from your file. Any
        data not on this chromosome in the input will be skipped. The
        list can be comma separated, with "-" characters representing an
        interval.<br />For example, to include chromosomes 1, 2, 3, 10 and X
        you could use <a href="vep_options.html#opt_chr">--chr 1-3,10,X</a> <i>Not used by
        default</i>
      </td>
      <td colspan="2">&nbsp;</td>
    </tr>
    <tr id="opt_coding_only">
      <td><pre>--coding_only</pre></td>
      <td>&nbsp;</td>
      <td>
        Only return consequences that fall in the coding regions of
        transcripts. <i>Not used by default</i>
      </td>
      <td>&nbsp;</td>
      <td>
        <div style="padding:2px 0px"><a href="#opt_most_severe">--most_severe</a></div>
        <div style="padding:2px 0px"><a href="#opt_summary">--summary</a></div>
      </td>
    </tr>
    <tr id="opt_no_intergenic">
      <td><pre>--no_intergenic</pre></td>
      <td>&nbsp;</td>
      <td>
        Do not include intergenic consequences in the output. <i>Not
        used by default</i>
      </td>
      <td>&nbsp;</td>
      <td>
        <div style="padding:2px 0px"><a href="#opt_most_severe">--most_severe</a></div>
        <div style="padding:2px 0px"><a href="#opt_summary">--summary</a></div>
      </td>
    </tr>
    <tr id="opt_pick">
      <td><pre>--pick</pre></td>
      <td>&nbsp;</td>
      <td>
        Pick one line or block of consequence data per variant, including
        transcript-specific columns.<br />Consequences are chosen according to the
        criteria described <a href="vep_other.html#pick">here</a>, and the
        order the criteria are applied may be customised with
        <a href="#opt_pick_order">--pick_order</a>.
        This is the best method to use if you are interested only in
        one consequence per variant. <i>Not used by default</i>
      </td>
      <td>&nbsp;</td>
      <td>
        <div style="padding:2px 0px"><a href="#opt_most_severe">--most_severe</a></div>
        <div style="padding:2px 0px"><a href="#opt_summary">--summary</a></div>
      </td>
    </tr>
    <tr id="opt_pick_allele">
      <td><pre>--pick_allele</pre></td>
      <td>&nbsp;</td>
      <td>
        Like <a href="#opt_pick">--pick</a>, but chooses one line or block of
        consequence data per variant allele. Will only differ in behaviour from
        <b>--pick</b> when the input variant has multiple alternate alleles.
        <i>Not used by default</i>
      </td>
      <td>&nbsp;</td>
      <td>
        <div style="padding:2px 0px"><a href="#opt_most_severe">--most_severe</a></div>
        <div style="padding:2px 0px"><a href="#opt_summary">--summary</a></div>
      </td>
    </tr>
    <tr id="opt_per_gene">
      <td><pre>--per_gene</pre></td>
      <td>&nbsp;</td>
      <td>
        Output only the most severe consequence per gene. The transcript
        selected is arbitrary if more than one has the same predicted
        consequence. Uses the same ranking system as
        <a href="#opt_pick">--pick</a>. <i>Not used by default</i>
      </td>
      <td colspan="2">&nbsp;</td>
    </tr>
    <tr id="opt_pick_allele_gene">
      <td><pre>--pick_allele_gene</pre></td>
      <td>&nbsp;</td>
      <td>
        Like <a href="#opt_pick">--pick_allele</a>, but chooses one line or block of
        consequence data per variant allele <b>and</b> gene combination.
        <i>Not used by default</i>
      </td>
      <td colspan="2">&nbsp;</td>
    </tr>
    <tr id="opt_flag_pick">
      <td><pre>--flag_pick</pre></td>
      <td>&nbsp;</td>
      <td>
        As per <a href="#opt_pick">--pick</a>, but adds the PICK flag to the
        chosen block of consequence data and retains others. <i>Not used by
        default</i>
      </td>
      <td>PICK</td>
      <td>
        <div style="padding:2px 0px"><a href="#opt_most_severe">--most_severe</a></div>
        <div style="padding:2px 0px"><a href="#opt_summary">--summary</a></div>
      </td>
    </tr>
    <tr id="opt_flag_pick_allele">
      <td><pre>--flag_pick_allele</pre></td>
      <td>&nbsp;</td>
      <td>
        As per <a href="#opt_pick_allele">--pick_allele</a>, but adds the PICK
        flag to the chosen block of consequence data and retains others. <i>Not used by default</i>
      </td>
      <td>PICK</td>
      <td>
        <div style="padding:2px 0px"><a href="#opt_most_severe">--most_severe</a></div>
        <div style="padding:2px 0px"><a href="#opt_summary">--summary</a></div>
      </td>
    </tr>
    <tr id="opt_flag_pick_allele_gene">
      <td><pre>--flag_pick_allele_gene</pre></td>
      <td>&nbsp;</td>
      <td>
        As per <a href="#opt_pick_allele_gene">--pick_allele_gene</a>, but adds the PICK
        flag to the chosen block of consequence data and retains others. <i>Not used by default</i>
      </td>
      <td>PICK</td>
      <td>&nbsp;</td>
    </tr>
    <tr id="opt_pick_order">
      <td><pre>--pick_order [c1,c2,...,cN]</pre></td>
      <td>&nbsp;</td>
      <td>
        <p>Customise the order of criteria (and the list of criteria) applied when choosing a block of
        annotation data with one of the following options: <a href="#opt_pick">--pick</a>, <a href="#opt_pick_allele">--pick_allele</a>, <a href="#opt_per_gene">--per_gene</a>, <a href="#opt_pick_allele_gene">--pick_allele_gene</a>, <a href="#opt_flag_pick">--flag_pick</a>, <a href="#opt_flag_pick_allele">--flag_pick_allele</a>, <a href="#opt_flag_pick_allele_gene">--flag_pick_allele_gene</a>. See
        <a href="vep_other.html#pick_options">this page</a> for the default order.<br />
        Valid criteria are: [ <i>canonical appris tsl biotype ccds rank length mane</i> ]. e.g.:</p>
        <pre class="code sh_sh">--pick --pick_order tsl,appris,rank</pre>
      </td>
      <td colspan="2">&nbsp;</td>
    </tr>
    <tr id="opt_most_severe">
      <td><pre>--most_severe</pre></td>
      <td>&nbsp;</td>
      <td>
        Output only the most severe consequence per variant.
        Transcript-specific columns will be left blank. Consequence ranks are
        given in <a
        href="/info/genome/variation/prediction/predicted_data.html#consequences">this
        table</a>.<br />
        To include regulatory consequences, use the <b>--regulatory</b> option in combination with this flag.<br /> 
        <i>Not used by default</i>
      </td>
      <td>&nbsp;</td>
      <td>
        <div style="padding:2px 0px"><a href="#opt_appris">--appris</a></div>
        <div style="padding:2px 0px"><a href="#opt_biotype">--biotype</a></div>
        <div style="padding:2px 0px"><a href="#opt_canonical">--canonical</a></div>
        <div style="padding:2px 0px"><a href="#opt_ccds">--ccds</a></div>
        <div style="padding:2px 0px"><a href="#opt_coding_only">--coding_only</a></div>
        <div style="padding:2px 0px"><a href="#opt_domains">--domains</a></div>
        <div style="padding:2px 0px"><a href="#opt_flag_pick">--flag_pick</a></div>
        <div style="padding:2px 0px"><a href="#opt_flag_pick_allele">--flag_pick_allele</a></div>
        <div style="padding:2px 0px"><a href="#opt_no_intergenic">--no_intergenic</a></div>
        <div style="padding:2px 0px"><a href="#opt_numbers">--numbers</a></div>
        <div style="padding:2px 0px"><a href="#opt_pick">--pick</a></div>
        <div style="padding:2px 0px"><a href="#opt_pick_allele">--pick_allele</a></div>
        <div style="padding:2px 0px"><a href="#opt_polyphen">--polyphen</a></div>
        <div style="padding:2px 0px"><a href="#opt_protein">--protein</a></div>
        <div style="padding:2px 0px"><a href="#opt_sift">--sift</a></div>
        <div style="padding:2px 0px"><a href="#opt_summary">--summary</a></div>
        <div style="padding:2px 0px"><a href="#opt_symbol">--symbol</a></div>
        <div style="padding:2px 0px"><a href="#opt_tsl">--tsl</a></div>
        <div style="padding:2px 0px"><a href="#opt_uniprot">--uniprot</a></div>
        <div style="padding:2px 0px"><a href="#opt_xref_refseq">--xref_refseq</a></div>
      </td>
    </tr>
    <tr id="opt_summary">
      <td><pre>--summary</pre></td>
      <td>&nbsp;</td>
      <td>
        Output only a comma-separated list of all observed consequences
        per variant. Transcript-specific columns will be left blank.
        <i>Not used by default</i>
      </td>
      <td>&nbsp;</td>
      <td>
        <div style="padding:2px 0px"><a href="#opt_appris">--appris</a></div>
        <div style="padding:2px 0px"><a href="#opt_biotype">--biotype</a></div>
        <div style="padding:2px 0px"><a href="#opt_canonical">--canonical</a></div>
        <div style="padding:2px 0px"><a href="#opt_ccds">--ccds</a></div>
        <div style="padding:2px 0px"><a href="#opt_coding_only">--coding_only</a></div>
        <div style="padding:2px 0px"><a href="#opt_domains">--domains</a></div>
        <div style="padding:2px 0px"><a href="#opt_flag_pick">--flag_pick</a></div>
        <div style="padding:2px 0px"><a href="#opt_flag_pick_allele">--flag_pick_allele</a></div>
        <div style="padding:2px 0px"><a href="#opt_most_severe">--most_severe</a></div>
        <div style="padding:2px 0px"><a href="#opt_no_intergenic">--no_intergenic</a></div>
        <div style="padding:2px 0px"><a href="#opt_numbers">--numbers</a></div>
        <div style="padding:2px 0px"><a href="#opt_pick">--pick</a></div>
        <div style="padding:2px 0px"><a href="#opt_pick_allele">--pick_allele</a></div>
        <div style="padding:2px 0px"><a href="#opt_polyphen">--polyphen</a></div>
        <div style="padding:2px 0px"><a href="#opt_protein">--protein</a></div>
        <div style="padding:2px 0px"><a href="#opt_sift">--sift</a></div>
        <div style="padding:2px 0px"><a href="#opt_symbol">--symbol</a></div>
        <div style="padding:2px 0px"><a href="#opt_tsl">--tsl</a></div>
        <div style="padding:2px 0px"><a href="#opt_uniprot">--uniprot</a></div>
        <div style="padding:2px 0px"><a href="#opt_xref_refseq">--xref_refseq</a></div>
      </td>
    </tr>
    <tr id="opt_filter_common">
      <td><pre>--filter_common</pre></td>
      <td>&nbsp;</td>
      <td>
        Shortcut flag for the filters below - this will exclude variants
        that have a co-located existing variant with global AF &gt;
        0.01 (1%). May be modified using any of the following freq_*
        filters. <i>Not used by default</i>
      </td>
      <td>FREQS</td>
      <td>&nbsp;</td>
    </tr>
    <tr id="opt_check_frequency">
      <td><pre>--check_frequency</pre></td>
      <td>&nbsp;</td>
      <td>
        Turns on frequency filtering. Use this to include or exclude
        variants based on the frequency of co-located existing variants
        in the Ensembl Variation database. You must also specify all of the --freq_* flags below.
        Frequencies used in filtering are added to the output under the FREQS
        key in the Extra field. <i>Not used by default</i>
      </td>
      <td>FREQS</td>
      <td>&nbsp;</td>
    </tr>
    <tr id="opt_freq_pop">
      <td><pre>--freq_pop [pop]</pre></td>
      <td>&nbsp;</td>
      <td>
        Name of the population to use in frequency filter. This must be one of the following:
        <p>
          <table class="ss" style="width:auto">
            <thead>
              <tr>
                <th>Name</th><th>Description</th>
              </tr>
            </thead>
            <tbody>
              <tr class="bg1">
                <td>1KG_ALL</td><td>1000 genomes combined population (global)</td>
              </tr>
              <tr>
                <td>1KG_AFR</td><td>1000 genomes combined African population</td>
              </tr>
              <tr class="bg1">
                <td>1KG_AMR</td><td>1000 genomes combined American population</td>
              </tr>
              <tr>
                <td>1KG_EAS</td><td>1000 genomes combined East Asian population</td>
              </tr>
              <tr class="bg1">
                <td>1KG_EUR</td><td>1000 genomes combined European population</td>
              </tr>
              <tr>
                <td>1KG_SAS</td><td>1000 genomes combined South Asian population</td>
              </tr>

              <tr class="bg1">
                <td>AA</td><td>NHLBI-ESP African American</td>
              </tr>
              <tr>
                <td>EA</td><td>NHLBI-ESP European American</td>
              </tr>

              <tr class="bg1">
                <td>gnomAD</td><td>gnomAD combined population</td>
              </tr>
              <tr>
                <td>gnomAD_AFR</td><td>gnomAD African/African American population</td>
              </tr>
              <tr class="bg1">
                <td>gnomAD_AMR</td><td>gnomAD Latino population</td>
              </tr>
              <tr>
                <td>gnomAD_ASJ</td><td>gnomAD Ashkenazi Jewish population</td>
              </tr>
              <tr class="bg1">
                <td>gnomAD_EAS</td><td>gnomAD East Asian population</td>
              </tr>
              <tr>
                <td>gnomAD_FIN</td><td>gnomAD Finnish population</td>
              </tr>
              <tr class="bg1">
                <td>gnomAD_NFE</td><td>gnomAD non-Finnish European population</td>
              </tr>
              <tr>
                <td>gnomAD_OTH</td><td>gnomAD other population</td>
              </tr>
              <tr class="bg1">
                <td>gnomAD_SAS</td><td>gnomAD South Asian population</td>
              </tr>
            </tbody>
          </table>
        </p>
      </td>
      <td colspan="2">&nbsp;</td>
    </tr>
    <tr id="opt_freq_freq">
      <td><pre>--freq_freq [freq]</pre></td>
      <td>&nbsp;</td>
      <td>
        Allele frequency to use for filtering. Must be a float
        value between 0 and 1
      </td>
      <td colspan="2">&nbsp;</td>
    </tr>
    <tr id="opt_freq_gt_lt">
      <td><pre>--freq_gt_lt [gt<b>|</b>lt]</pre></td>
      <td>&nbsp;</td>
      <td>
        Specify whether the frequency of the co-located variant must be
        greater than (<b>gt</b>) or less than (<b>lt</b>) the value
        specified with <a href="#opt_freq_freq">--freq_freq</a>
      </td>
      <td colspan="2">&nbsp;</td>
    </tr>
    <tr id="opt_freq_filter">
      <td><pre>--freq_filter [exclude<b>|</b>include]</pre></td>
      <td>&nbsp;</td>
      <td>
        Specify whether to <b>exclude</b> or <b>include</b> only
        variants that pass the frequency filter
      </td>
      <td colspan="2">&nbsp;</td>
    </tr>    
  </table>
 

  <br/>
  <hr/>
  <h2 id="dbopt">Database options</h2>
  
  <table class="ss">
    <tr>
      <th width="20%">Flag</th>
      <th width="8%">Alternate</th>
      <th>Description</th>
      <th width="10%">Output fields</th>
      <th width="10%">Incompatible with</th>
    </tr>
    <tr id="opt_database">
      <td><pre>--database</pre></td>
      <td>&nbsp;</td>
      <td>Enable VEP to use local or remote databases.</td>
      <td>&nbsp;</td>
      <td>
        <div style="padding:2px 0px"><a href="#opt_af_1kg">--af_1kg</a></div>
        <div style="padding:2px 0px"><a href="#opt_af_esp">--af_esp</a></div>
        <div style="padding:2px 0px"><a href="#opt_af_exac">--af_exac</a></div>
        <div style="padding:2px 0px"><a href="#opt_af_gnomad">--af_gnomad</a></div>
        <div style="padding:2px 0px"><a href="#opt_cache">--cache</a></div>
        <div style="padding:2px 0px"><a href="#opt_max_af">--max_af</a></div>
        <div style="padding:2px 0px"><a href="#opt_offline">--offline</a></div>
        <div style="padding:2px 0px"><a href="#opt_pubmed">--pubmed</a></div>
        <div style="padding:2px 0px"><a href="#opt_var_synonyms">--var_synonyms</a></div>
      </td>
    </tr>
    <tr id="opt_host">
      <td><pre>--host [hostname]</pre></td>
      <td>&nbsp;</td>
      <td>
        Manually define the database host to connect to. Users in the US
        may find connection and transfer speeds quicker using our East
        coast mirror, useastdb.ensembl.org. <i>Default =
        "ensembldb.ensembl.org"</i>
      </td>
      <td colspan="2">&nbsp;</td>
    </tr>
    <tr id="opt_user">
      <td><pre>--user [username]</pre></td>
      <td><pre>-u</pre></td>
      <td>Manually define the database username. <i>Default =
      "anonymous"</i></td>
      <td colspan="2">&nbsp;</td>
    </tr>
    <tr id="opt_password">
      <td><pre>--password [password]</pre></td>
      <td><pre>--pass</pre></td>
      <td>Manually define the database password. <i>Not used by
      default</i></td>
      <td colspan="2">&nbsp;</td>
    </tr>
    <tr id="opt_port">
      <td><pre>--port [number]</pre></td>
      <td>&nbsp;</td>
      <td>Manually define the database port. <i>Default = 5306</i></td>
      <td colspan="2">&nbsp;</td>
    </tr>
    <tr id="opt_genomes">
      <td><pre>--genomes</pre></td>
      <td>&nbsp;</td>
      <td>
        Override the default connection settings with those for the
        Ensembl Genomes public MySQL server. Required when using any of
        the <a rel="external" href="https://www.ensemblgenomes.org/"
        target="_blank">Ensembl Genomes</a> species. <i>Not used by
        default</i>
      </td>
      <td colspan="2">&nbsp;</td>
    </tr>
    <tr id="opt_is_multispecies">
      <td><pre>--is_multispecies [0|1]</pre></td>
      <td>&nbsp;</td>
      <td>
        Some of the <a rel="external" href="https://www.ensemblgenomes.org/"
        target="_blank">Ensembl Genomes</a> databases (mainly bacteria and protists) are composed of a collection of close species.
        It updates the database connection settings (i.e. the database name) if the value is set to 1. <i>Default: 0</i>
      </td>
      <td colspan="2">&nbsp;</td>
    </tr>
    <tr id="opt_lrg">
      <td><pre>--lrg</pre></td>
      <td>&nbsp;</td>
      <td>
        Map input variants to LRG coordinates (or to chromosome coordinates if
        given in LRG coordinates), and provide consequences on both LRG and
        chromosomal transcripts. <i>Not used by default</i>
      </td>
      <td>&nbsp;</td>
      <td>
        <div style="padding:2px 0px"><a href="#opt_offline">--offline</a></div>
      </td>
    </tr>
    <tr id="opt_db_version">
      <td><pre>--db_version [number]</pre></td>
      <td>&nbsp;</td>
      <td>
        Force VEP to connect to a specific version of the Ensembl
        databases. Not recommended as there may be conflicts
        between software and database versions. <i>Not used by
        default</i>
      </td>
      <td colspan="2">&nbsp;</td>
    </tr>
    <tr id="opt_registry">
      <td><pre>--registry [filename]</pre></td>
      <td>&nbsp;</td>
      <td>
        Defining a registry file overwrites other connection settings
        and uses those found in the specified registry file to connect.
        <i>Not used by default</i>
      </td>
      <td colspan="2">&nbsp;</td>
    </tr>
  </table>
</div>

</body>
</html>
