<html>
<head>
<title>Annotation sources</title>
<meta name="order" content="4" />
<script type="text/javascript">

  // Function to show/hide divs
  function show_hide (param) {
    div   = document.getElementById('div_'+param);
    alink = document.getElementById('a_'+param);
    if (div.style.display=='inline') {
      div.style.display='none';
      alink.innerHTML='Show';
    }
    else {
      if (div.style.display=='none') {
        div.style.display='inline';
        alink.innerHTML='Hide';
      }
    }
  }
</script>
</head>

<body>
<style>
tr:nth-child(odd) {background-color: #f0f0f0;}
</style>

<div>

	<div style="float:right"><img src="/img/vep_logo.png"/></div>

	<h1 id="top"><span style="color:#006">Variant Effect Predictor</span> <img src="/i/16/database.png"> <span style="color:#666">Annotation sources</span></h1>
	<hr/>

	<p> VEP can use a variety of annotation sources to retrieve the transcript
	models used to predict consequence types. </p>

  <ul>
    <li><b><a href="#cache">Cache</a></b> - a downloadable file containing all transcript models, regulatory features and variant data for a species</li>
    <li><b><a href="#gff">GFF or GTF</a></b> - use transcript models defined in a tabix-indexed GFF or GTF file</li>
    <li><b><a href="#database">Database</a></b> - connect to a MySQL database server hosting Ensembl databases</li>
  </ul>

  <p> Data from VCF, BED and bigWig files can also be incorporated by VEP's <a href="vep_custom.html">custom annotation</a> feature.</p>

  <div style="background-color:#f3f9f4; color:#333; border-radius:5px; font-size:16px; border: 1px solid #91c89c; margin: 10px 0 10px 0; padding: 10px;">
    Using a cache is the most efficient way to use VEP; we would
    encourage you to use a cache wherever possible. Caches are easy to
    download and set up using the <a
    href="vep_download.html#installer">installer</a>. Follow the <a
    href="vep_tutorial.html">tutorial</a> for a simple guide.
  </div>

  <hr/>
	<h2 id="cache">Caches</h2>

  <p> Using a cache (<a href="vep_options.html#opt_cache">--cache</a>) is
  the fastest and most efficient way to use VEP, as in most cases
  only a single initial network connection is made and most data is read from
  local disk. Use <a href="#offline">offline</a> mode to eliminate all network
  connections for speed and/or privacy.</p>

	<h3 id="pre">Downloading caches</h3>

	<p> Ensembl creates cache files for every species for each Ensembl release.
  They can be
	automatically downloaded and configured using <a
	href="vep_download.html#installer">INSTALL.pl</a>. <p>

  <p> If interested in RefSeq transcripts you may download an alternate cache
  file (e.g. homo_sapiens_refseq), or a merged file of RefSeq and Ensembl
  transcripts (eg homo_sapiens_merged); remember to specify <a
  href="vep_options.html#opt_refseq">--refseq</a> or <a
  href="vep_options.html#opt_merged">--merged</a> when running VEP to use
  the relevant cache. See <a href="vep_other.html#refseq">documentation</a> for full details.</p>

  <h3 id="manual_cache">Manually downloading caches</h3>

  <p> It is also simple to download and set up caches without using the installer.
  By default, VEP searches for caches in $HOME/.vep; to use a different directory when running VEP, use <a href="vep_options.html#opt_dir_cache">--dir_cache</a>.</p>

  <pre class="code">cd $HOME/.vep
curl -O ftp://ftp.ensembl.org/pub/release-[[SPECIESDEFS::ENSEMBL_VERSION]]/variation/VEP/homo_sapiens_vep_[[SPECIESDEFS::ENSEMBL_VERSION]]_GRCh38.tar.gz
tar xzf homo_sapiens_vep_[[SPECIESDEFS::ENSEMBL_VERSION]]_GRCh38.tar.gz</pre>

    <p> FTP directories by species grouping: </p>

    <ul>
      <li>
        Ensembl
        <ul>
          <li><a href="ftp://ftp.ensembl.org/pub/current_variation/VEP/">Vertebrates</a></li>
        </ul>
      </li>
      <li>
        Ensembl Genomes. <b>NB:</b> When using Ensembl Genomes caches, you should use the <a
    href="vep_options.html#opt_cache_version">--cache_version</a> option to
    specify the relevant Ensembl Genomes version number as these differ from the
    concurrent Ensembl/VEP version numbers.
        <ul>
          <li><a href="ftp://ftp.ensemblgenomes.org/pub/bacteria/current/vep/">Bacteria</a></li>
          <li><a href="ftp://ftp.ensemblgenomes.org/pub/fungi/current/vep/">Fungi</a></li>
          <li><a href="ftp://ftp.ensemblgenomes.org/pub/metazoa/current/vep/">Metazoa</a></li>
          <li><a href="ftp://ftp.ensemblgenomes.org/pub/plants/current/vep/">Plants</a></li>
          <li><a href="ftp://ftp.ensemblgenomes.org/pub/protists/current/vep/">Protists</a></li>
        </ul>
      </li>
    </ul>

    <p>  </p>

  <h3 id="limitations">Limitations of the cache</h3>

  <p> The cache stores the following information: </p>

  <ul>
    <li>Transcript location, sequence, exons and other attributes</li>
    <li>Gene, protein, HGNC and other identifiers for each transcript (where
    applicable, <a href="vep_other.html#refseq_limitations">limitations apply to RefSeq caches</a>)</li>
    <li>Locations, alleles and frequencies of existing variants</li>
    <li>Regulatory regions</li>
    <li>Predictions and scores for SIFT, PolyPhen</li>
  </ul>

  <p> It does not store any information pertaining to, and therefore cannot be
  used for, the following: </p>

  <ul>
    <li>HGVS names (<a href="vep_options.html#opt_hgvs">--hgvs</a>, <a href="vep_options.html#opt_hgvsg">--hgvsg</a>) - to retrieve these you must additionally point
    to a FASTA file containing the reference sequence for your species
    (<a href="vep_options.html#opt_fasta">--fasta</a>)</li>
    <li>Using HGVS notation as input (<a href="vep_options.html#opt_format">--format hgvs</a>)</li>
    <li>Using variant identifiers as input (<a href="vep_options.html#opt_format">--format id</a>)</li>
    <li>Finding overlapping structural variants (<a href="vep_options.html#opt_check_sv">--check_sv</a>)</li>
  </ul>

    <p> Enabling one of these options with <a href="vep_options.html#opt_cache">--cache</a> will cause VEP
    to warn you in its status output with something like the following: </p>

  <pre class="code"> 2011-06-16 16:24:51 - INFO: Database will be accessed when using --hgvs </pre>

	<h3 id="convert">Convert with tabix</h3>

    <p> For those with Bio::DB::HTS (as set up by INSTALL.pl) or <a rel="external" href="https://github.com/samtools/htslib"
    target="_blank">tabix</a> installed on their systems, the speed of
    retrieving existing co-located variants can be greatly improved by
    converting the cache files using the supplied script, convert_cache.pl. This
    replaces the plain-text, chunked variant dumps with a single tabix-indexed
    file per chromosome. The script is simple to run: </p>

    <pre class="code">perl convert_cache.pl -species [species] -version [vep_version]</pre>

    <p> To convert all species and all versions, use "all": </p>

    <pre class="code">perl convert_cache.pl -species all -version all</pre>

    <p> A full description of the options can be seen using <b>--help</b>. When complete, VEP will
    automatically detect the converted cache and use this in place. </p>

    <p> Note that
    tabix and bgzip must be installed on your system to convert a cache. INSTALL.pl
    downloads these when setting up Bio::DB::HTS; to enable convert_cache.pl to find them, run:</p>

    <pre class="code">export PATH=${PATH}:${PWD}/htslib</pre>
    </p>

	<h3 id="offline">Data privacy and offline mode</h3>

	<p> When using the public database servers, VEP requests
	transcript and variation data that overlap the loci in your input file. As
	such, these coordinates are transmitted over the network to a public server,
	which may not be appropriate for those with sensitive or private data. Users
	should note that <b>only</b> the coordinates are transmitted to the server;
	no other information is sent. </p>

	<p> To run VEP in an offline mode that does not use any network connections, use the flag
    <a href="vep_options.html#opt_offline">--offline</a>. </p>

    <p> The <a href="#limitations">limitations</a> described above apply
    absolutely when using offline mode. For example, if you specify <a
    href="vep_options.html#opt_offline">--offline</a> and <a
    href="vep_options.html#opt_format">--format id</a>, VEP will report
    an error and refuse to run: </p>

    <pre class="code">ERROR: Cannot use ID format in offline mode</pre>

    <p> All other features, including the ability to use <a
    href="vep_custom.html">custom annotations</a> and <a
    href="vep_plugins.html">plugins</a>, are accessible in offline mode. </p>


  <hr/>
  <h2 id="gff">GFF/GTF files</h2>

  <p> VEP can use transcript annotations defined in <a href="https://github.com/The-Sequence-Ontology/Specifications/blob/master/gff3.md" rel="external">GFF</a> or <a href="http://enssand-01.internal.sanger.ac.uk:9099/info/website/upload/gff.html">GTF</a> files. The files must be bgzipped and indexed with tabix, and VEP requires a FASTA file containing the genomic sequence in order to generate transcript models. </p>

  <p> Your GFF or GTF file must be sorted in chromosomal order. VEP does not use header lines so it is safe to remove them.</p>

  <pre class="code">grep -v "#" data.gff | sort -k1,1 -k4,4n -k5,5n | bgzip -c > data.gff.gz
tabix -p gff data.gff.gz
./vep -i input.vcf -gff data.gff.gz -fasta genome.fa.gz</pre>

  <p> You may use any number of GFF/GTF files in this way, providing they refer to the same genome. You may also use them in concert with annotations from a cache or database source; annotations are distinguished by the SOURCE field in the VEP output: </p>

  <pre class="code">./vep -i input.vcf -cache -gff data.gff.gz -fasta genome.fa.gz</pre>

  <p> This functionality uses VEP's custom annotation feature, and the <a href="vep_options.html#opt_gff">--gff</a> flag is a shortcut to:</p>

  <pre class="code">--custom data.gff.gz,,gff</pre>

  <p> You should use the <a href="vep_custom.html#custom_options">longer form</a> if you wish to customise the name of the GFF as it appears in the SOURCE field and VEP output header.</p>

  <h3 id="gff_req">GFF format expectations</h3>

  <p> VEP has been tested on GFF files generated by Ensembl and NCBI (RefSeq). Due to inconsistency in the GFF specification and adherence to it, VEP may encounter problems parsing some GFF files. For the same reason, not all transcript biotypes defined in your GFF may be supported by VEP. VEP does not support GFF files with embedded FASTA sequence.</p>

  <p> The following entity types (3rd column in the GFF) are supported by VEP. Lines of other types will be ignored; if this leads to an incomplete transcript model, the whole transcript model may be discarded. [<a href="#gfftypes" onclick="show_hide('gfftypes');" id="a_gfftypes">Show supported types</a>]</p>

  <div id="div_gfftypes" style="display:none;">
    <ul>
      <li>aberrant_processed_transcript</li>
      <li>CDS</li>
      <li>C_gene_segment</li>
      <li>D_gene_segment</li>
      <li>exon</li>
      <li>gene</li>
      <li>J_gene_segment</li>
      <li>lincRNA</li>
      <li>lincRNA_gene</li>
      <li>miRNA</li>
      <li>miRNA_gene</li>
      <li>mRNA</li>
      <li>mt_gene</li>
      <li>ncRNA</li>
      <li>NMD_transcript_variant</li>
      <li>primary_transcript</li>
      <li>processed_pseudogene</li>
      <li>processed_transcript</li>
      <li>pseudogene</li>
      <li>pseudogenic_transcript</li>
      <li>RNA</li>
      <li>rRNA</li>
      <li>rRNA_gene</li>
      <li>snoRNA</li>
      <li>snoRNA_gene</li>
      <li>snRNA</li>
      <li>snRNA_gene</li>
      <li>supercontig</li>
      <li>transcript</li>
      <li>tRNA</li>
      <li>VD_gene_segment</li>
      <li>V_gene_segment</li>
    </ul>
  </div>

  <p> Entities in the GFF are expected to be linked using a key named "parent" or "Parent" in the attributes (9th) column of the GFF. Unlinked entities (i.e. those with no parents <b>or</b> children) are discarded. Sibling entities (those that share the same parent) may have overlapping coordinates, e.g. for exon and CDS entities. </p>

  <p> Transcripts require a Sequence Ontology biotype to be defined in order to be parsed by VEP. The simplest way to define this is using an attribute named "biotype" on the transcript entity. Other configurations are supported in order for VEP to be able to parse GFF files from NCBI and other sources. </p>

  <h3 id="gtf_req">GTF format expectations</h3>

  <p> The following GTF entity types will be parsed by VEP:</p>

  <ul>
    <li>cds (or CDS)</li>
    <li>stop_codon</li>
    <li>exon</li>
    <li>gene</li>
    <li>transcript</li>
  </ul>

  <p> Entities are linked by an attribute named for the parent entity type e.g. exon is linked to transcript by transcript_id, transcript is linked to gene by gene_id. </p>

  <p> Transcript biotypes are defined in attributes named "biotype", "transcript_biotype" or "transcript_type". If none of these exist, VEP will attempt to interpret the source field (2nd column) of the GTF as the biotype. </p>

  <h3 id="gff_synonyms">Chromosome synonyms</h3>

  <p> If the chromosome names used in your GFF/GTF differ from those used in the FASTA or your input VCF, you may see warnings like this when running VEP:</p>

  <pre class="code">WARNING: Chromosome 21 not found in annotation sources or synonyms on line 160</pre>

  <p> To circumvent this you may provide VEP with a <a href="vep_options.html#opt_synonyms">synonyms file</a>. A synonym file is included in VEP's cache files, so if you have one of these for your species you can use it as follows:</p>

  <pre class="code">./vep -i input.vcf -cache -gff data.gff.gz -fasta genome.fa.gz -synonyms ~/.vep/homo_sapiens/[[SPECIESDEFS::ENSEMBL_VERSION]]_GRCh38/chr_synonyms.txt</pre>

  <h3 id="gff_limitations">Limitations</h3>

  <p> Using a GFF or GTF file as VEP's annotation source limits access to some auxiliary information available when using a <a href="#cache">cache</a>. Currently most external reference data such as gene symbols, transcript identifiers and protein domains are inaccessible when using only a GFF/GTF file.</p>

  <p> VEP's flexibility does allow some annotation types to be replaced. The following table illustrates some examples and alternative means to retrieve equivalent data. </p>

  <table class="ss">
    <tr>
      <th width="30%">Data type</th>
      <th>Alternative</th>
    </tr>
    <tr>
      <td>SIFT and PolyPhen predictions (<a href="vep_options.html#opt_sift">--sift</a>, <a href="vep_options.html#opt_polyphen">--polyphen</a>)</td>
      <td>Use the <a href="https://github.com/Ensembl/VEP_plugins/blob/release/87/PolyPhen_SIFT.pm" rel="external">PolyPhen_SIFT VEP plugin</a></td>
    </tr>
    <tr>
      <td>Co-located variants (<a href="vep_options.html#opt_check_existing">--check_existing</a>, --af* flags)</td>
      <td>
        A couple of options are available:
        <ol>
          <li>Use a <a href="vep_custom.html">VCF with --custom</a> to retrieve variant IDs, frequency and other data</li>
          <li>Add <a href="vep_options.html#opt_cache">--cache</a> to use variants in the <a href="#cache">cache</a>. <b>*</b></li>
        </ol>
      </td>
    </td>
    <tr>
      <td>Regulatory consequences (<a href="vep_options.html#opt_regulatory">--regulatory</a>)</td>
      <td>
        Add <a href="vep_options.html#opt_cache">--cache</a> to use regulatory features in the <a href="#cache">cache</a>. <b>*</b>
      </td>
    </td>
  </table>

  <p><b>*</b> Note this will also instruct VEP to annotate input variants against transcript models retrieved from the cache <b>as well as</b> those from the GFF/GTF file. It is possible to use <a href="vep_options.html#opt_transcript_filter">--transcript_filter</a> to include only the transcripts from your GFF/GTF file:</p>

  <pre class="code">./vep -i input.vcf -cache -custom data.gff.gz,myGFF,gff -fasta genome.fa.gz -transcript_filter "_source_cache is myGFF"</pre>


  <hr/>
  <h2 id="fasta">FASTA files</h2>

    <p> By pointing VEP to a FASTA file (or directory containing several
    files), it is possible to retrieve reference sequence locally when using
    <a href="vep_options.html#opt_cache">--cache</a> or <a
    href="vep_options.html#opt_offline">--offline</a>. This enables VEP to
    retrieve HGVS notations (<a href="vep_options.html#opt_hgvs">--hgvs</a>),
    check the reference sequence given in input data (<a
    href="vep_options.html#opt_check_ref">--check_ref</a>), and construct
    transcript models from a GFF or GTF file without accessing a
    database. </p>

    <p> FASTA files can be set up using the <a
    href="vep_download.html#installer">installer</a>; files set up using the
    installer are automatically detected by VEP when using <a
    href="vep_options.html#opt_cache">--cache</a> or <a
    href="vep_options.html#opt_offline">--offline</a>; you should not need to
    use <a href="vep_options.html#opt_fasta">--fasta</a> to manually specify
    them. </p>

  <p> To enable this VEP uses one of two modules: </p>

  <ul>
    <li>
      The <a rel="external"
      href="https://github.com/Ensembl/Bio-DB-HTS">Bio::DB::HTS</a> Perl XS
      module with <a href="http://www.htslib.org">HTSlib</a>. This module
      uses compiled C code and can access compressed
      (bgzipped) or uncompressed FASTA files. It is set up by the VEP <a
      href="vep_download.html#installer">installer</a>.
    </li>
    <li>
      The <a rel="external"
      href="http://search.cpan.org/~cjfields/BioPerl-1.6.901/Bio/DB/Fasta.pm">Bio::DB::Fasta</a>
      module. This may be used on systems where installation of the Bio::DB::HTS
      module has not been possible. It can access only uncompressed FASTA
      files. It is also set up by the VEP installer
      and comes as part of the BioPerl package.
    </li>
  </ul>

  <p> The first time you run VEP
  with a specific FASTA file, an index will be built. This can take a few
  minutes, depending on the size of the FASTA file and the speed of your
  system. On subsequent runs the index does not need to be rebuilt (if the
  FASTA file has been modified, VEP will force a rebuild of the index).
  </p>

  <p> Ensembl provides suitable reference FASTA files as downloads from its
  FTP server. See the <a href="/info/data/ftp/index.html">Downloads</a> page
  for details. You should preferably use the installer as described above to
  fetch these files; manual instructions are provided for reference.
  In most cases it is best to download the single large
  "primary_assembly" file for your species. You should use the unmasked
  (without "_rm" or "_sm" in the name) sequences. Note that VEP requires
  that the file be either unzipped (Bio::DB::Fasta) or unzipped and then
  recompressed with bgzip (Bio::DB::HTS::Faidx) to run; when unzipped these files can be very
  large (25GB for human). An example set of commands for setting up the data
  for human follows: </p>

  <pre class="code">curl -O ftp://ftp.ensembl.org/pub/release-[[SPECIESDEFS::ENSEMBL_VERSION]]/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.primary_assembly.fa.gz
gzip -d Homo_sapiens.GRCh38.dna.primary_assembly.fa.gz
bgzip Homo_sapiens.GRCh38.dna.primary_assembly.fa
./vep -i input.vcf --offline --hgvs --fasta Homo_sapiens.GRCh38.dna.primary_assembly.fa.gz</pre>



	<hr/>
  <h2 id="database">Databases</h2>

  <p> VEP can use remote or local database servers to retrieve annotations. </p>

  <ul>
    <li>Using <a href="vep_options.html#opt_cache">--cache</a> (without <a href="vep_options.html#opt_offline">--offline</a>) uses the local cache on disk to fetch most annotations, but allows database connections for some features (see <a href="#limitations">cache limitations</a>)</li>
    <li>Using <a href="vep_options.html#opt_database">--database</a> tells VEP to retrieve <b>all</b> annotations from the database. <b>Please only use this for small input files or when using a local database server!</b></li>
  </ul>

	<h3 id="public">Public database servers</h3>

    <p> By default, VEP is configured to connect to Ensembl's public
    MySQL instance at ensembldb.ensembl.org. For users in the US (or for any
    user geographically closer to the East coast of the USA than to Ensembl's
    data centre in Cambridge, UK), a mirror server is available at
    useastdb.ensembl.org. To use the mirror, use the flag <code><a
    href="vep_options.html#opt_host">--host</a> useastdb.ensembl.org</code> </p>

    <p> Users of Ensembl Genomes species (e.g. plants, fungi, microbes) should
    use their public MySQL instance; the connection parameters for this can be
    automatically loaded by using the flag <a
    href="vep_options.html#opt_genomes">--genomes</a> </p>

	<p> Users with small data sets (100s of variants) should find using the
	default connection settings adequate. Those with larger data sets, or those
	who wish to use VEP in a batch manner, should consider one of the
	alternatives below. </p>


	<h3 id="local">Using a local database</h3>

    <p> It is possible to set up a local MySQL mirror with the databases for
    your species of interest installed. For instructions on installing a local
    mirror, see <a href="/info/docs/webcode/mirror/install/ensembl-data.html"
    target="_blank">here</a>. You will need a MySQL server that you can connect
    to from the machine where you will run VEP (this can be the same
    machine). For most of the functionality of VEP, you will only need the
    Core database (e.g. homo_sapiens_core_[[SPECIESDEFS::ENSEMBL_VERSION]]_38)
    installed. In order to find co-located variants or to use SIFT or
    PolyPhen, it is also necessary to install the relevant variation database
    (e.g. homo_sapiens_variation_[[SPECIESDEFS::ENSEMBL_VERSION]]_38). </p>

	<p> Note that unless you have custom data to insert in the database, in most
	cases it will be much more efficient to use a <a href="#cache">pre-built
	cache</a> in place of a local database. </p>

    <p> To connect to your mirror, you can either set the connection parameters
    using <a href="vep_options.html#opt_host">--host</a>, <a
    href="vep_options.html#opt_port">--port</a>, <a
    href="vep_options.html#opt_user">--user</a> and <a
    href="vep_options.html#opt_password">--password</a>, or use a registry file.
    Registry files contain all the connection parameters for your database, as
    well as any species aliases you wish to set up: </p>

	<pre class="code sh_perl sh_sourceCode">
use Bio::EnsEMBL::DBSQL::DBAdaptor;
use Bio::EnsEMBL::Variation::DBSQL::DBAdaptor;
use Bio::EnsEMBL::Registry;

Bio::EnsEMBL::DBSQL::DBAdaptor->new(
  '-species' => "Homo_sapiens",
  '-group'   => "core",
  '-port'    => 5306,
  '-host'    => 'ensembldb.ensembl.org',
  '-user'    => 'anonymous',
  '-pass'    => '',
  '-dbname'  => 'homo_sapiens_core_[[SPECIESDEFS::ENSEMBL_VERSION]]_38'
);

Bio::EnsEMBL::Variation::DBSQL::DBAdaptor->new(
  '-species' => "Homo_sapiens",
  '-group'   => "variation",
  '-port'    => 5306,
  '-host'    => 'ensembldb.ensembl.org',
  '-user'    => 'anonymous',
  '-pass'    => '',
  '-dbname'  => 'homo_sapiens_variation_[[SPECIESDEFS::ENSEMBL_VERSION]]_38'
);

Bio::EnsEMBL::Registry->add_alias("Homo_sapiens","human");
	</pre>

	<p> For more information on the registry and registry files, see <a
	href="/info/docs/api/registry.html" target="_blank">here</a>. </p>

	<hr/>
	<h2 id="technical">Cache - technical information</h2>

	<p> <span style="color:red;">ADVANCED</span> The cache consists of
	compressed files containing listrefs of serialised objects. These objects
	are initially created from the database as if using the Ensembl API
	normally. In order to reduce the size of the cache and allow the
	serialisation to occur, some changes are made to the objects before they are
	dumped to disk. This means that they will not behave in exactly the same way
	as an object retrieved from the database when writing, for example, a plugin
	that uses the cache. </p>

	<p>The following hash keys are deleted from each transcript object: </p>

	<ul>
		<li><b>analysis</b></li>
		<li><b>created_date</b></li>
		<li><b>dbentries</b> : this contains the external references retrieved
		when calling $transcript->get_all_DBEntries(); hence this call
		on a cached object will return no entries</li>
		<li><b>description</b></li>
		<li><b>display_xref</b></li>
		<li><b>edits_enabled</b></li>
		<li><b>external_db</b></li>
		<li><b>external_display_name</b></li>
		<li><b>external_name</b></li>
		<li><b>external_status</b></li>
		<li><b>is_current</b></li>
		<li><b>modified_date</b></li>
		<li><b>status</b></li>
		<li><b>transcript_mapper</b> : used to convert between genomic, cdna,
		cds and protein coordinates. A copy of this is cached separately
		by VEP as
		<pre class="code sh_perl sh_sourceCode">$transcript->{_variation_effect_feature_cache}->{mapper}</pre>
		</li>
	</ul>

	<p> As mentioned above, a special hash key "_variation_effect_feature_cache"
	is created on the transcript object and used to cache things used by VEP
	in predicting consequences, things which might otherwise have to be fetched
	from the database. Some of these are stored in place of equivalent keys that
	are deleted as described above. The following keys and data are stored: </p>

	<ul>
		<li><b>introns</b> : listref of intron objects for the transcript. The adaptor,
		analysis, dbID, next, prev and seqname keys are stripped from each
		intron object</li>
		<li><b>translateable_seq</b> : as returned by
		<pre class="code sh_perl sh_sourceCode">$transcript->translateable_seq</pre></li>
		<li><b>mapper</b> : transcript mapper as described above</li>
		<li><b>peptide</b> : the translated sequence as a string, as returned by
		<pre class="code sh_perl sh_sourceCode">$transcript->translate->seq</pre></li>
		<li><b>protein_features</b> : protein domains for the transcript's translation
		as returned by
		<pre class="code sh_perl sh_sourceCode">$transcript->translation->get_all_ProteinFeatures</pre>
		Each protein feature is stripped of all keys but: start, end, analysis,
		hseqname</li>
		<li><b>codon_table</b> : the codon table ID used to translate the transcript,
		as returned by
		<pre class="code sh_perl sh_sourceCode">$transcript->slice->get_all_Attributes('codon_table')->[0]</pre></li>
		<li><b>protein_function_predictions</b> : a hashref containing the keys "sift"
		and "polyphen"; each one contains a protein function prediction matrix
		as returned by e.g.
		<pre class="code sh_perl sh_sourceCode">$protein_function_prediction_matrix_adaptor->fetch_by_analysis_translation_md5('sift', md5_hex($transcript-{_variation_effect_feature_cache}->{peptide}))</pre></li>
	</ul>

	<p> Similarly, some further data is cached directly on the transcript object
	under the following keys: </p>

	<ul>
		<li><b>_gene</b> : gene object. This object has all keys but the following
		deleted: start, end, strand, stable_id</li>
		<li><b>_gene_symbol</b> : the gene symbol</li>
		<li><b>_ccds</b> : the CCDS identifier for the transcript</li>
		<li><b>_refseq</b> : the "NM" RefSeq mRNA identifier for the transcript</li>
		<li><b>_protein</b> : the Ensembl stable identifier of the translation</li>
    <li><b>_source_cache</b> : the source of the transcript object. Only defined in the merged cache (values: Ensembl, RefSeq) or when using a GFF/GTF file (value: short name or filename)</li>
	</ul>
</div>

</body>
</html>
