<html>
<head>
  <title>Plugins</title>
  <meta name="order" content="7" />
  <script type="text/javascript">
    // Function to show/hide divs
    function show_hide (param) {
      div   = document.getElementById('div_'+param);
      alink = document.getElementById('a_'+param);
      if (div.style.display=='inline') {
        div.style.display='none';
        alink.innerHTML= alink.innerHTML.replace("less","more");
      }
      else {
        if (div.style.display=='none') {
          div.style.display='inline';
          alink.innerHTML= alink.innerHTML.replace("more", "less");
        }
      }
    }
    // Function to show/hide plugins by their category
    function show_hide_category () {
      var category_id = $('select#select_category').val();
      if (category_id == 'all') {
        $('.plugin_row').show();
      }
      else {
        $('.plugin_row').hide();
        $("tr[data-category='"+category_id+"']").show();
      }
    }
  </script>
</head>
<body>
<div>
  <div style="float:right"><img src="/img/vep_logo.png"/></div>
	
  <h1 id="top"><span style="color:#006;padding-right:15px">Variant Effect Predictor</span><span style="color:#666"><img src="/i/16/tool.png"/> Plugins</span></h1>
  <hr/>
        
  <p> VEP can use plugin modules written in Perl to <b>add functionality</b> to
  the software.</p>
  <p>Plugins are a powerful way to <b>extend</b>, <b>filter</b> and <b>manipulate</b> the
  VEP output.<br />They can be installed using VEP's installer script, run the
  following command to get a list of available plugins: </p>

  <pre class="code sh_sh">perl INSTALL.pl -a p -g list</pre>

  <p> Alternatively, VEP plugins and their dependencies are available in the <a href="https://hub.docker.com/r/ensemblorg/ensembl-vep">Docker image</a>. Read how to use Ensembl VEP in <a href="vep_download.html#docker">Docker</a> and <a href="vep_download.html#singularity">Singularity</a>. </p>

  <p> Some plugins are also available to use via the <a href="/Tools/VEP">VEP web</a> and <a href="https://rest.ensembl.org/#VEP">REST</a> interfaces. </p>

  <br />	
  <h2 id="plugins_existing">Existing plugins</h2>
	
  <p> We have written several plugins that implement experimental
      functionalities that we do not (yet) include in the variation API, and these
      are stored in a public github repository: </p>
	
  <p><a href="//github.com/Ensembl/VEP_plugins" rel="external">https://github.com/Ensembl/VEP_plugins</a></p>

  <p>Here is the list of the VEP plugins available:</p>

  <!-- Plugins list - start -->

  <div style="margin:5px 0px 10px">
    <span style="padding-right:8px">Select categories:</span>
    <select id="select_category" onchange="javascritpt:show_hide_category()">
       <option value="all">All categories</option><option value="conservation">Conservation</option><option value="external_id">External ID</option><option value="frequency_data">Frequency data</option><option value="functional_effect">Functional effect</option><option value="hgvs">HGVS</option><option value="look_up">Look up</option><option value="motif">Motif</option><option value="nearby_features">Nearby features</option><option value="pathogenicity_predictions">Pathogenicity predictions</option><option value="phenotype_data_and_citations">Phenotype data and citations</option><option value="protein_annotation">Protein annotation</option><option value="protein_data">Protein data</option><option value="regulatory_impact">Regulatory impact</option><option value="sequence">Sequence</option><option value="splicing_predictions">Splicing predictions</option><option value="structural_variant_data">Structural variant data</option><option value="transcript_annotation">Transcript annotation</option><option value="variant_data">Variant data</option><option value="visualisation">Visualisation</option>
    </select>
  </div>   
  <table class="ss">
    <thead>
      <tr>
        <th>Plugin</th>
        <th style="max-width:50%">Description</th>
        <th>Category</th>
        <th>External libraries</th>
        <th>Developer</th>
      </tr>
    </thead>
    <tbody>
<tr class="bg1 plugin_row" data-category="pathogenicity_predictions"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/AlphaMissense.pm">AlphaMissense</a></div></td><td><p>This plugin for the Ensembl Variant Effect Predictor (VEP) annotates missense variants with the <br />pre-computed AlphaMissense pathogenicity scores. AlphaMissense is a deep learning model developed <br />by Google DeepMind that predicts the pathogenicity of single nucleotide missense variants.  <a class="button" href="#alphamissense" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('alphamissense');" id="a_alphamissense">more</a></p><div id="div_alphamissense" style="display:none;"><p>This plugin will add two annotations per missense variant:</p><p>'am_pathogenicity', a continuous score between 0 and 1 which can be interpreted as the predicted <br />probability of the variant being pathogenic. </p><p>'am_class' is the classification of the variant into one of three discrete categories:  <br />'Likely pathogenic', 'Likely benign', or 'ambiguous'. These are derived using the following <br />thresholds of am_pathogenicity: <br />'Likely benign' if am_pathogenicity < 0.34; <br />'Likely pathogenic' if am_pathogenicity > 0.564; <br />'ambiguous' otherwise. </p><p>These thresholds were chosen to achieve 90% precision for both pathogenic and benign ClinVar variants. <br />Note that AlphaMissense was not trained on ClinVar variants. Variants labeled as 'ambiguous' should be <br />treated as 'unknown' or 'uncertain' according to AlphaMissense. </p><p>This plugin is available for both GRCh37 (hg19) and GRCh38 (hg38) genome builds.</p><p>The prediction scores of AlphaMissense can be downloaded from <br /><a href="https://console.cloud.google.com/storage/browser/dm_alphamissense">https://console.cloud.google.com/storage/browser/dm_alphamissense</a> <br />(AlphaMissense Database Copyright (2023) DeepMind Technologies Limited).  Data contained within the <br />AlphaMissense Database is provided for non-commercial research use only under CC BY-NC-SA 4.0 license. <br />Use of the AlphaMissense Database is subject to Google Cloud Platform Terms of Service  </p><p>Please cite the AlphaMissense publication alongside the VEP<br />if you use this resource: <a href="https://doi.org/10.1126/science.adg7492">https://doi.org/10.1126/science.adg7492</a></p><p>Disclaimer: The AlphaMissense Database and other information provided on or linked to this site is <br />for theoretical modelling only, caution should be exercised in use. It is provided "as-is" without <br />any warranty of any kind, whether express or implied. For clarity, no warranty is given that use of <br />the information shall not infringe the rights of any third party (and this disclaimer takes precedence <br />over any contrary provisions in the Google Cloud Platform Terms of Service). The information provided <br />is not intended to be a substitute for professional medical advice, diagnosis, or treatment, and does <br />not constitute medical or other professional advice.</p><p>Before running the plugin for the first time, you need to create a tabix index (requires tabix to be <br />installed).</p><p><pre class="code sh_sh">tabix -s 1 -b 2 -e 2 -f -S 1 AlphaMissense_hg38.tsv.gz<br /></pre><pre class="code sh_sh">tabix -s 1 -b 2 -e 2 -f -S 1 AlphaMissense_hg19.tsv.gz<br /></pre></p><p>Options are passed to the plugin as key=value pairs:<br />file             : (mandatory) Tabix-indexed AlphaMissense data<br />cols             : (optional) Colon-separated columns to print from<br />AlphaMissense data; if set to 'all', all columns are printed<br />(default: Missense_pathogenicity:Missense_class)<br />transcript_match : Only print data if transcript identifiers match those from<br />AlphaMissense data (default: 0)</p><p>AlphaMissense predictions are matched to input data by genomic location and protein change by default.</p></div></td><td><div class="vdoc_dtype_count" style="float:left;padding:2px 6px;cursor:default;background-color:#1E90FF">Pathogenicity predictions</div></td><td>-</td><td>Ensembl</td></tr>
<tr class="bg2 plugin_row" data-category="conservation"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/AncestralAllele.pm">AncestralAllele</a></div></td><td><p>A VEP plugin that retrieves ancestral allele sequences from a FASTA file. <a class="button" href="#ancestralallele" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('ancestralallele');" id="a_ancestralallele">more</a></p><div id="div_ancestralallele" style="display:none;"><p>Ensembl produces FASTA file dumps of the ancestral sequences of key species.<br />Data files for GRCh37 are available from <a href="https://ftp.ensembl.org/pub/release-75/fasta/ancestral_alleles/">https://ftp.ensembl.org/pub/release-75/fasta/ancestral_alleles/</a><br />Data files for GRCh38 are available from <a href="https://ftp.ensembl.org/pub/current_fasta/ancestral_alleles/">https://ftp.ensembl.org/pub/current_fasta/ancestral_alleles/</a></p><p>For optimal retrieval speed, you should pre-process the FASTA files into a single<br />bgzipped file that can be accessed via Bio::DB::HTS::Faidx (installed by VEP's<br />INSTALL.pl):</p><p><pre class="code sh_sh">wget <a href="https://ftp.ensembl.org/pub/current_fasta/ancestral_alleles/homo_sapiens_ancestor_GRCh38.tar.gz">https://ftp.ensembl.org/pub/current_fasta/ancestral_alleles/homo_sapiens_ancestor_GRCh38.tar.gz</a><br />tar xfz homo_sapiens_ancestor_GRCh38.tar.gz<br />cat homo_sapiens_ancestor_GRCh38/*.fa | bgzip -c > homo_sapiens_ancestor_GRCh38.fa.gz<br />rm -rf homo_sapiens_ancestor_GRCh38/ homo_sapiens_ancestor_GRCh38.tar.gz<br />./vep -i variations.vcf --plugin AncestralAllele,homo_sapiens_ancestor_GRCh38.fa.gz<br /></pre>Data file is only available for GRCh38. <br />The plugin is also compatible with Bio::DB::Fasta and an uncompressed FASTA file.</p><p>Note the first time you run the plugin with a newly generated FASTA file it will<br />spend some time indexing the file. DO NOT INTERRUPT THIS PROCESS, particularly<br />if you do not have Bio::DB::HTS installed.</p><p>Special cases:<br />"-" represents an insertion<br />"?" indicates the chromosome could not be looked up in the FASTA</p></div></td><td><div class="vdoc_dtype_count" style="float:left;padding:2px 6px;cursor:default;background-color:#02599C">Conservation</div></td><td>-</td><td>Ensembl</td></tr>
<tr class="bg1 plugin_row" data-category="conservation"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/Blosum62.pm">Blosum62</a></div></td><td><p>This is a plugin for the Ensembl Variant Effect Predictor (VEP) that<br />looks up the BLOSUM 62 substitution matrix score for the reference<br />and alternative amino acids predicted for a missense mutation. It adds<br />one new entry to the VEP's Extra column, BLOSUM62 which is the <br />associated score. </p></td><td><div class="vdoc_dtype_count" style="float:left;padding:2px 6px;cursor:default;background-color:#02599C">Conservation</div></td><td>-</td><td>Ensembl</td></tr>
<tr class="bg2 plugin_row" data-category="pathogenicity_predictions"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/CADD.pm">CADD</a></div><div style="margin-top:6px"><small>Combined Annotation Dependent Depletion</small></div></td><td><p>A VEP plugin that retrieves CADD scores for variants from one or more<br />tabix-indexed CADD data files. <a class="button" href="#cadd" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('cadd');" id="a_cadd">more</a></p><div id="div_cadd" style="display:none;"><p>Please cite the CADD publication alongside the VEP if you use this resource:<br /><a href="https://www.ncbi.nlm.nih.gov/pubmed/24487276">https://www.ncbi.nlm.nih.gov/pubmed/24487276</a></p><p>The tabix utility must be installed in your path to use this plugin. </p><p>The CADD SNV and indels data files (and respective Tabix index files) can be downloaded from - <br /><a href="http://cadd.gs.washington.edu/download">http://cadd.gs.washington.edu/download</a></p><p>The CADD SV data files (and respective Tabix index files)  can be downloaded from -<br /><a href="https://kircherlab.bihealth.org/download/CADD-SV/v1.1/">https://kircherlab.bihealth.org/download/CADD-SV/v1.1/</a></p><p>The plugin works with all versions of available CADD files. The plugin only<br />reports scores and does not consider any additional annotations from a CADD<br />file. It is therefore sufficient to use CADD files without the additional<br />annotations. </p></div></td><td><div class="vdoc_dtype_count" style="float:left;padding:2px 6px;cursor:default;background-color:#1E90FF">Pathogenicity predictions</div></td><td>-</td><td>Ensembl</td></tr>
<tr class="bg1 plugin_row" data-category="pathogenicity_predictions"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/CAPICE.pm">CAPICE</a></div></td><td><p>A VEP plugin that retrieves CAPICE scores for variants from one or more<br />tabix-indexed CAPICE data files, in order to predict their pathogenicity. <a class="button" href="#capice" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('capice');" id="a_capice">more</a></p><div id="div_capice" style="display:none;"><p>Please cite the CAPICE publication alongside the VEP if you use this resource:<br /><a href="https://pubmed.ncbi.nlm.nih.gov/32831124/">https://pubmed.ncbi.nlm.nih.gov/32831124/</a></p><p>The tabix utility must be installed in your path to use this plugin. The CAPICE<br />SNVs, InDels and respective index (TBI) files for GRCh37 can be downloaded from<br /><a href="https://zenodo.org/record/3928295">https://zenodo.org/record/3928295</a></p><p>To filter results, please use filter_vep with the output file or standard<br />output. Documentation on filter_vep is available at:<br /><a href="https://www.ensembl.org/info/docs/tools/vep/script/vep_filter.html">https://www.ensembl.org/info/docs/tools/vep/script/vep_filter.html</a></p><p>For recommendations on threshold selection, please read the CAPICE publication.</p></div></td><td><div class="vdoc_dtype_count" style="float:left;padding:2px 6px;cursor:default;background-color:#1E90FF">Pathogenicity predictions</div></td><td>-</td><td>Ensembl</td></tr>
<tr class="bg2 plugin_row" data-category="pathogenicity_predictions"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/Carol.pm">Carol</a></div></td><td><p>A VEP plugin that calculates the Combined Annotation scoRing toOL (CAROL)<br />score (1) for a missense mutation based on the pre-calculated SIFT (2) and<br />PolyPhen-2 (3) scores from the Ensembl API (4). <a class="button" href="#carol" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('carol');" id="a_carol">more</a></p><div id="div_carol" style="display:none;"><p>It adds one new entry class to the VEP's Extra column, CAROL which is<br />the calculated CAROL score. Note that this module is a perl reimplementation of <br />the original R script, available at: <a href="https://sanger.ac.uk/tool/carol/">https://sanger.ac.uk/tool/carol/</a></p><p>I believe that both versions implement the same algorithm, but if there are any<br />discrepancies the R version should be treated as the reference implementation. <br />Bug reports are welcome.</p><p>References:</p><p>(1) Lopes MC, Joyce C, Ritchie GRS, John SL, Cunningham F, Asimit J, Zeggini E. <br />A combined functional annotation score for non-synonymous variants<br />Human Heredity 73(1):47-51 (2012)<br />doi:10.1159/000334984</p><p>(2) Kumar P, Henikoff S, Ng PC.<br />Predicting the effects of coding non-synonymous variants on protein function using the SIFT algorithm<br />Nature Protocols 4(8):1073-1081 (2009)<br />doi:10.1038/nprot.2009.86</p><p>(3) Adzhubei IA, Schmidt S, Peshkin L, Ramensky VE, Gerasimova A, Bork P, Kondrashov AS, Sunyaev SR. <br />A method and server for predicting damaging missense mutations<br />Nature Methods 7(4):248-249 (2010)<br />doi:10.1038/nmeth0410-248</p><p>(4) Flicek P, et al.<br />Ensembl 2012<br />Nucleic Acids Research 40(D1):D84-D90 (2011)<br />doi: 10.1093/nar/gkr991</p></div></td><td><div class="vdoc_dtype_count" style="float:left;padding:2px 6px;cursor:default;background-color:#1E90FF">Pathogenicity predictions</div></td><td><a href="https://metacpan.org/pod/Math::CDF" rel="external">Math::CDF</a> qw(pnorm qnorm)</td><td>Ensembl</td></tr>
<tr class="bg1 plugin_row" data-category="pathogenicity_predictions"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/ClinPred.pm">ClinPred</a></div></td><td><p>This is a plugin for the Ensembl Variant Effect Predictor (VEP) that adds pre-calculated scores from ClinPred.<br />ClinPred is a prediction tool to identify disease-relevant nonsynonymous variants. <a class="button" href="#clinpred" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('clinpred');" id="a_clinpred">more</a></p><div id="div_clinpred" style="display:none;"><p>This Plugin is only available for GRCh37</p><p>Please cite the ClinPred publication alongside the VEP if you use this resource:<br /><a href="https://www.sciencedirect.com/science/article/pii/S0002929718302714">https://www.sciencedirect.com/science/article/pii/S0002929718302714</a></p><p>ClinPred scores can be downloaded from <br /><a href="https://sites.google.com/site/clinpred/download">https://sites.google.com/site/clinpred/download</a></p><p>The following steps are neccessary to tabix the ClinPred.txt.gz file before running the plugin:<br /><pre class="code sh_sh">gzip -d ClinPred.txt.gz # to unzip the text file <br />cat ClinPred.txt | tr " " "\t" > ClinPred_tabbed.tsv # to change the file to a tabbed delimited file <br />sed '1s/.*/#&/'  ClinPred_tabbed.tsv > tabbed_ClinPred.tsv  # to add a # in the first line of the file <br />sed '1s/C/c/' tabbed_ClinPred.tsv > ClinPred_tabbed.tsv # to convert the Chr to chr<br />bgzip ClinPred_tabbed.tsv<br />tabix -f -s 1 -b 2 -e 2 ClinPred_tabbed.tsv.gz<br /></pre>The tabix utility must be installed in your path to use this plugin.<br />Check <a href="https://github.com/samtools/htslib.git">https://github.com/samtools/htslib.git</a> for instructions.</p></div></td><td><div class="vdoc_dtype_count" style="float:left;padding:2px 6px;cursor:default;background-color:#1E90FF">Pathogenicity predictions</div></td><td>-</td><td>Ensembl</td></tr>
<tr class="bg2 plugin_row" data-category="pathogenicity_predictions"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/Condel.pm">Condel</a></div></td><td><p>A VEP plugin that calculates the Consensus Deleteriousness (Condel) score (1)<br />for a missense mutation based on the pre-calculated SIFT (2) and PolyPhen-2 (3)<br />scores from the Ensembl API (4). <a class="button" href="#condel" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('condel');" id="a_condel">more</a></p><div id="div_condel" style="display:none;"><p>It adds one new entry class to the VEP's Extra column, Condel which is the<br />calculated Condel score. This version of Condel was developed by the Biomedical<br />Genomics Group  of the Universitat Pompeu Fabra, at the Barcelona Biomedical<br />Research Park and available at <a href="https://bg.upf.edu/condel">https://bg.upf.edu/condel</a>. The code in this<br />plugin is based on a script provided by this group and slightly reformatted to<br />fit into the Ensembl API.</p><p>The plugin takes 3 command line arguments, the first is the path to a Condel <br />configuration directory which contains cutoffs and the distribution files etc., <br />the second is either "s", "p", or "b" to output the Condel score, prediction or <br />both (the default is both), and the third argument is either 1 or 2 to use the <br />original version of Condel (1), or the newer version (2) - 2 is the default and <br />is recommended to avoid false positive predictions from Condel in some <br />circumstances.</p><p>An example Condel configuration file and a set of distribution files can be<br />found in the config/Condel directory in this repository. You should edit the <br />config/Condel/config/condel_SP.conf file and set the 'condel.dir' parameter to<br />the full path to the location of the config/Condel directory on your system.</p><p>References:</p><p>(1) Gonzalez-Perez A, Lopez-Bigas N. <br />Improving the assessment of the outcome of non-synonymous SNVs with a Consensus deleteriousness score (Condel)<br />Am J Hum Genet 88(4):440-449 (2011)<br />doi:10.1016/j.ajhg.2011.03.004</p><p>(2) Kumar P, Henikoff S, Ng PC.<br />Predicting the effects of coding non-synonymous variants on protein function using the SIFT algorithm<br />Nature Protocols 4(8):1073-1081 (2009)<br />doi:10.1038/nprot.2009.86</p><p>(3) Adzhubei IA, Schmidt S, Peshkin L, Ramensky VE, Gerasimova A, Bork P, Kondrashov AS, Sunyaev SR. <br />A method and server for predicting damaging missense mutations<br />Nature Methods 7(4):248-249 (2010)<br />doi:10.1038/nmeth0410-248</p><p>(4) Flicek P, et al.<br />Ensembl 2012<br />Nucleic Acids Research (2011)<br />doi: 10.1093/nar/gkr991</p></div></td><td><div class="vdoc_dtype_count" style="float:left;padding:2px 6px;cursor:default;background-color:#1E90FF">Pathogenicity predictions</div></td><td>-</td><td>Ensembl</td></tr>
<tr class="bg1 plugin_row" data-category="conservation"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/Conservation.pm">Conservation</a></div></td><td><p>This is a plugin for the Ensembl Variant Effect Predictor (VEP) that<br />retrieves a conservation score from the Ensembl Compara databases<br />for variant positions. You can specify the method link type and<br />species sets as command line options, the default is to fetch GERP<br />scores from the EPO 35 way mammalian alignment (please refer to the<br />Compara documentation for more details of available analyses).  <a class="button" href="#conservation" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('conservation');" id="a_conservation">more</a></p><div id="div_conservation" style="display:none;"><p>If a variant affects multiple nucleotides the average score for the<br />position will be returned, and for insertions the average score of<br />the 2 flanking bases will be returned.</p><p>The plugin uses the ensembl-compara API module (optional, see<br /><a href="http://www.ensembl.org/info/docs/api/index.html">http://www.ensembl.org/info/docs/api/index.html</a>) or obtains data<br />directly from BigWig files (optional, see<br /><a href="https://ftp.ensembl.org/pub/current_compara/conservation_scores/">https://ftp.ensembl.org/pub/current_compara/conservation_scores/</a>)</p></div></td><td><div class="vdoc_dtype_count" style="float:left;padding:2px 6px;cursor:default;background-color:#02599C">Conservation</div></td><td><a href="https://metacpan.org/pod/Net::FTP" rel="external">Net::FTP</a></td><td>Ensembl</td></tr>
<tr class="bg2 plugin_row" data-category="pathogenicity_predictions"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/dbNSFP.pm">dbNSFP</a></div></td><td><p>A VEP plugin that retrieves data for missense variants from a tabix-indexed<br />dbNSFP file. <a class="button" href="#dbnsfp" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('dbnsfp');" id="a_dbnsfp">more</a></p><div id="div_dbnsfp" style="display:none;"><p>Please cite the dbNSFP publications alongside the VEP if you use this resource:<br />dbNSFP      <a href="https://www.ncbi.nlm.nih.gov/pubmed/21520341">https://www.ncbi.nlm.nih.gov/pubmed/21520341</a><br />dbNSFP v2.0 <a href="https://www.ncbi.nlm.nih.gov/pubmed/23843252">https://www.ncbi.nlm.nih.gov/pubmed/23843252</a><br />dbNSFP v3.0 <a href="https://www.ncbi.nlm.nih.gov/pubmed/26555599">https://www.ncbi.nlm.nih.gov/pubmed/26555599</a><br />dbNSFP v4   <a href="https://www.ncbi.nlm.nih.gov/pubmed/33261662">https://www.ncbi.nlm.nih.gov/pubmed/33261662</a></p><p>You must have the Bio::DB::HTS module or the tabix utility must be installed<br />in your path to use this plugin. The dbNSFP data file can be downloaded from<br /><a href="https://sites.google.com/site/jpopgen/dbNSFP">https://sites.google.com/site/jpopgen/dbNSFP</a></p><p>The file must be processed and indexed with tabix before use by this plugin.<br />The file must be processed according to the dbNSFP release version and the assembly you use.<br />It is recommended to use the -T option with the sort command to specify a temporary directory with sufficient space.</p><p>For release 4.4a:<br /><pre class="code sh_sh">version=4.4a<br />wget ftp://dbnsfp:dbnsfp@dbnsfp.softgenetics.com/dbNSFP${version}.zip<br />unzip dbNSFP${version}.zip<br />zcat dbNSFP${version}_variant.chr1.gz | head -n1 > h<br /></pre># GRCh38/hg38 data<br /><pre class="code sh_sh">zgrep -h -v ^#chr dbNSFP${version}_variant.chr* | sort -k1,1 -k2,2n - | cat h - | bgzip -c > dbNSFP${version}_grch38.gz<br />tabix -s 1 -b 2 -e 2 dbNSFP${version}_grch38.gz<br /></pre># GRCh37/hg19 data<br /><pre class="code sh_sh">zgrep -h -v ^#chr dbNSFP${version}_variant.chr* | awk '$8 != "." ' | sort -k8,8 -k9,9n - | cat h - | bgzip -c > dbNSFP${version}_grch37.gz<br />tabix -s 8 -b 9 -e 9 dbNSFP${version}_grch37.gz<br /></pre>When running the plugin you must list at least one column to retrieve from the<br />dbNSFP file, specified as parameters to the plugin e.g.</p><p><pre class="code sh_sh">--plugin dbNSFP,/path/to/dbNSFP.gz,LRT_score,GERP++_RS<br /></pre>You may include all columns with ALL; this fetches a large amount of data per<br />variant!:</p><p><pre class="code sh_sh">--plugin dbNSFP,/path/to/dbNSFP.gz,ALL<br /></pre>Tabix also allows the data file to be hosted on a remote server. This plugin is<br />fully compatible with such a setup - simply use the URL of the remote file:</p><p><pre class="code sh_sh">--plugin dbNSFP,<a href="http://my.files.com/dbNSFP.gz">http://my.files.com/dbNSFP.gz</a>,col1,col2<br /></pre>The plugin replaces occurrences of ';' with ',' and '|' with '&'. However, some<br />data field columns, e.g. Interpro_domain, use the replacement characters. We<br />added a file with replacement logic for customising the required replacement<br />of ';' and '|' in dbNSFP data columns. In addition to the default replacements<br />(; to , and | to &) users can add customised replacements. Users can either modify<br />the file dbNSFP_replacement_logic in the VEP_plugins directory or provide their own<br />file as second argument when calling the plugin:</p><p><pre class="code sh_sh">--plugin dbNSFP,/path/to/dbNSFP.gz,/path/to/dbNSFP_replacement_logic,LRT_score,GERP++_RS<br /></pre>Note that transcript sequences referred to in dbNSFP may be out of sync with<br />those in the latest release of Ensembl; this may lead to discrepancies with<br />scores retrieved from other sources.</p><p>If the dbNSFP README file is found in the same directory as the data file,<br />column descriptions will be read from this and incorporated into the VEP output<br />file header.</p><p>The plugin matches rows in the tabix-indexed dbNSFP file on:</p><p>genomic position<br />alt allele<br />aaref - reference amino acid<br />aaalt - alternative amino acid</p><p>To match only on the genomic position and the alt allele use pep_match=0</p><p><pre class="code sh_sh">--plugin dbNSFP,/path/to/dbNSFP.gz,pep_match=0,col1,col2<br /></pre>Some fields contain multiple values, one per Ensembl transcript ID.<br />By default all values are returned, separated by ";" in the default VEP output format.<br />To return values only for the matched Ensembl transcript ID use transcript_match=1.<br />This behaviour only affects transcript-specific fields; non-transcript-specific fields<br />are unaffected.</p><p><pre class="code sh_sh">--plugin dbNSFP,/path/to/dbNSFP.gz,transcript_match=1,col1,col2<br /></pre>NB 1: Using this flag may cause no value to return if the version of the Ensembl<br />transcript set differs between VEP and dbNSFP.</p><p>NB 2: MutationTaster entries are keyed on a different set of transcript IDs. Using<br />the transcript_match flag with any MutationTaster field selected will have no effect<br />i.e. all entries are returned. Information on corresponding transcript(s) for<br />MutationTaster fields can be found using <a href="http://www.mutationtaster.org/ChrPos.html">http://www.mutationtaster.org/ChrPos.html</a></p></div></td><td><div class="vdoc_dtype_count" style="float:left;padding:2px 6px;cursor:default;background-color:#1E90FF">Pathogenicity predictions</div></td><td><a href="https://metacpan.org/pod/File::Basename" rel="external">File::Basename</a> qw(basename)</td><td>Ensembl</td></tr>
<tr class="bg1 plugin_row" data-category="splicing_predictions"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/dbscSNV.pm">dbscSNV</a></div></td><td><p>A VEP plugin that retrieves data for splicing variants from a tabix-indexed<br />dbscSNV file. <a class="button" href="#dbscsnv" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('dbscsnv');" id="a_dbscsnv">more</a></p><div id="div_dbscsnv" style="display:none;"><p>Please cite the dbscSNV publication alongside the VEP if you use this resource:<br /><a href="http://nar.oxfordjournals.org/content/42/22/13534">http://nar.oxfordjournals.org/content/42/22/13534</a></p><p>The Bio::DB::HTS perl library or tabix utility must be installed in your path<br />to use this plugin. The dbscSNV data file can be downloaded from<br /><a href="https://sites.google.com/site/jpopgen/dbNSFP">https://sites.google.com/site/jpopgen/dbNSFP</a>.</p><p>The file must be processed and indexed by tabix before use by this plugin.<br />dbscSNV1.1 has coordinates for both GRCh38 and GRCh37; the file must be<br />processed differently according to the assembly you use.</p><p><pre class="code sh_sh">wget ftp://dbnsfp:dbnsfp@dbnsfp.softgenetics.com/dbscSNV1.1.zip<br />unzip dbscSNV1.1.zip<br />head -n1 dbscSNV1.1.chr1 > h<br /></pre># GRCh38<br /><pre class="code sh_sh">cat dbscSNV1.1.chr* | grep -v ^chr | sort -k5,5 -k6,6n | cat h - | awk '$5 != "."' | bgzip -c > dbscSNV1.1_GRCh38.txt.gz<br />tabix -s 5 -b 6 -e 6 -c c dbscSNV1.1_GRCh38.txt.gz<br /></pre># GRCh37<br /><pre class="code sh_sh">cat dbscSNV1.1.chr* | grep -v ^chr | cat h - | bgzip -c > dbscSNV1.1_GRCh37.txt.gz<br />tabix -s 1 -b 2 -e 2 -c c dbscSNV1.1_GRCh37.txt.gz<br /></pre>Note that in the last command we tell tabix that the header line starts with "c";<br />this may change to the default of "#" in future versions of dbscSNV.</p><p>Tabix also allows the data file to be hosted on a remote server. This plugin is<br />fully compatible with such a setup - simply use the URL of the remote file:</p><p><pre class="code sh_sh">--plugin dbscSNV,<a href="http://my.files.com/dbscSNV.txt.gz">http://my.files.com/dbscSNV.txt.gz</a><br /></pre>Note that transcript sequences referred to in dbscSNV may be out of sync with<br />those in the latest release of Ensembl; this may lead to discrepancies with<br />scores retrieved from other sources.</p></div></td><td><div class="vdoc_dtype_count" style="float:left;padding:2px 6px;cursor:default;background-color:#FF0000">Splicing predictions</div></td><td>-</td><td>Ensembl</td></tr>
<tr class="bg2 plugin_row" data-category="phenotype_data_and_citations"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/DisGeNET.pm">DisGeNET</a></div></td><td><p>This is a plugin for the Ensembl Variant Effect Predictor (VEP) that<br />adds Variant-Disease-PMID associations from the DisGeNET database.<br />It is available for GRCh38. <a class="button" href="#disgenet" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('disgenet');" id="a_disgenet">more</a></p><div id="div_disgenet" style="display:none;"><p>Please cite the DisGeNET publication alongside the VEP if you use this resource:<br /><a href="https://academic.oup.com/nar/article/48/D1/D845/5611674">https://academic.oup.com/nar/article/48/D1/D845/5611674</a></p><p>Options are passed to the plugin as key=value pairs:</p><p>file           : Path to DisGeNET data file (mandatory).</p><p>disease        : Set value to 1 to include the diseases/phenotype names reporting<br />the Variant-PMID association (optional).</p><p>rsid           : Set value to 1 to include the dbSNP variant Identifier (optional).</p><p>filter_score   : Only reports citations with score greater or equal than input value (optional).</p><p>filter_source  : Only reports citations from input sources (optional).<br />Accepted sources are: UNIPROT, CLINVAR, GWASDB, GWASCAT, BEFREE<br />Separate multiple values with '&'.</p><p>Output:<br />Each element of the output includes:<br />- PMID of the publication reporting the Variant-Disease association (default)<br />- DisGeNET score for the Variant-Disease association (default)<br />- diseases/phenotype names (optional)<br />- dbSNP variant Identifier (optional)</p><p>The following steps are necessary before running this plugin (tested with DisGeNET export date 2020-05-26):<br />This plugin uses file 'all_variant_disease_pmid_associations.tsv.gz'<br />File can be downloaded from: <a href="https://www.disgenet.org/downloads">https://www.disgenet.org/downloads</a></p><p><pre class="code sh_sh">gunzip all_variant_disease_pmid_associations.tsv.gz<br />awk '($1 ~ /^snpId/ || $2 ~ /NA/) {next} {print $0}' all_variant_disease_pmid_associations.tsv > all_variant_disease_pmid_associations_clean.tsv<br />sort -t $'\t' -k2,2 -k3,3n all_variant_disease_pmid_associations_clean.tsv > all_variant_disease_pmid_associations_sorted.tsv<br />awk '{ gsub (/\t +/, "\t", $0); print}' all_variant_disease_pmid_associations_sorted.tsv > all_variant_disease_pmid_associations_final.tsv<br />bgzip all_variant_disease_pmid_associations_final.tsv<br />tabix -s 2 -b 3 -e 3 all_variant_disease_pmid_associations_final.tsv.gz<br /></pre>The plugin can then be run as default:<br /><pre class="code sh_sh">./vep -i variations.vcf --plugin DisGeNET,file=all_variant_disease_pmid_associations_final.tsv.gz<br /></pre>or with an option to include optional data or/and filters: <br /><pre class="code sh_sh">./vep -i variations.vcf --plugin DisGeNET,file=all_variant_disease_pmid_associations_final.tsv.gz,disease=1<br />./vep -i variations.vcf --plugin DisGeNET,file=all_variant_disease_pmid_associations_final.tsv.gz,disease=1,filter_source='GWASDB&GWASCAT'<br /></pre>Of notice: this plugin only matches the chromosome and the position in the<br />chromosome, the alleles are not taken into account to append the DisGeNET data.<br />The rsid is provided (optional) in the output in order to help to filter the relevant data.</p></div></td><td><div class="vdoc_dtype_count" style="float:left;padding:2px 6px;cursor:default;background-color:#9966EE">Phenotype data and citations</div></td><td><a href="https://metacpan.org/pod/List::MoreUtils" rel="external">List::MoreUtils</a> qw(uniq)</td><td>Ensembl</td></tr>
<tr class="bg1 plugin_row" data-category="functional_effect"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/DosageSensitivity.pm">DosageSensitivity</a></div></td><td><p>A VEP plugin that retrieves haploinsufficiency and triplosensitivity probability scores<br />for affected genes from a dosage sensitivity catalogue published in paper -<br /><a href="https://www.sciencedirect.com/science/article/pii/S0092867422007887">https://www.sciencedirect.com/science/article/pii/S0092867422007887</a> <a class="button" href="#dosagesensitivity" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('dosagesensitivity');" id="a_dosagesensitivity">more</a></p><div id="div_dosagesensitivity" style="display:none;"><p>Please cite the above publication alongside the VEP if you use this resource.</p><p>This plugin returns two scores:<br />- pHaplo score gives the probability of a gene being haploinsufficient (deletion intolerant)<br />- pTriplo score gives the probability of a gene being triploinsensitive (duplication intolerant)</p><p>Pre-requisites:<br />You need the compressed tsv file containing the dosage sensitivity score. The file<br />Collins_rCNV_2022.dosage_sensitivity_scores.tsv.gz can be downloaded from here - <br /><a href="https://zenodo.org/record/6347673/files/Collins_rCNV_2022.dosage_sensitivity_scores.tsv.gz">https://zenodo.org/record/6347673/files/Collins_rCNV_2022.dosage_sensitivity_scores.tsv.gz</a></p><p>Options are passed to the plugin as key=value pairs:</p><p>file   : (mandatory) compressed tsv file containing dosage sensitivity scores<br />cover  : set value to 1 (0 by default) to report scores only if the variant<br />covers the affected feature completely (e.g. - a CNV that duplicates the gene). <br />The feature is a gene if using --database otherwise it is a transcript.</p></div></td><td><div class="vdoc_dtype_count" style="float:left;padding:2px 6px;cursor:default;background-color:#443377">Functional effect</div></td><td>-</td><td>Ensembl</td></tr>
<tr class="bg2 plugin_row" data-category="nearby_features"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/Downstream.pm">Downstream</a></div></td><td><p>This is a plugin for the Ensembl Variant Effect Predictor (VEP) that<br />predicts the downstream effects of a frameshift variant on the protein<br />sequence of a transcript. It provides the predicted downstream protein<br />sequence (including any amino acids overlapped by the variant itself),<br />and the change in length relative to the reference protein. <a class="button" href="#downstream" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('downstream');" id="a_downstream">more</a></p><div id="div_downstream" style="display:none;"><p>Note that changes in splicing are not predicted - only the existing<br />translateable (i.e. spliced) sequence is used as a source of<br />translation. Any variants with a splice site consequence type are<br />ignored.</p><p>If VEP is run in offline mode using the flag --offline, a FASTA file is required.<br />See: <a href="https://www.ensembl.org/info/docs/tools/vep/script/vep_cache.html#fasta">https://www.ensembl.org/info/docs/tools/vep/script/vep_cache.html#fasta</a><br />Sequence may be incomplete without a FASTA file or database connection.</p></div></td><td><div class="vdoc_dtype_count" style="float:left;padding:2px 6px;cursor:default;background-color:#E75480">Nearby features</div></td><td>-</td><td>Ensembl</td></tr>
<tr class="bg1 plugin_row" data-category="visualisation"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/Draw.pm">Draw</a></div></td><td><p>A VEP plugin that draws pictures of the transcript model showing the<br />variant location. Can take five optional paramters: <a class="button" href="#draw" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('draw');" id="a_draw">more</a></p><div id="div_draw" style="display:none;"><p>1) File name stem for images<br />2) Image width in pixels (default: 1000px)<br />3) Image height in pixels (default: 100px)<br />4) Transcript ID - only draw images for this transcript<br />5) Variant ID - only draw images for this variant</p><p>e.g.</p><p><pre class="code sh_sh">./vep -i variations.vcf --plugin Draw,myimg,2000,100<br /></pre>Images are written to [file_stem]_[transcript_id]_[variant_id].png</p><p>Requires GD library installed to run.</p></div></td><td><div class="vdoc_dtype_count" style="float:left;padding:2px 6px;cursor:default;background-color:#008000">Visualisation</div></td><td><ul style="padding-left:1em"><li><a href="https://metacpan.org/pod/GD::Polygon" rel="external">GD::Polygon</a></li><li><a href="https://metacpan.org/pod/GD" rel="external">GD</a></li></ul></td><td>Ensembl</td></tr>
<tr class="bg2 plugin_row" data-category="regulatory_impact"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/Enformer.pm">Enformer</a></div></td><td><p>This is a plugin for the Ensembl Variant Effect Predictor (VEP) that adds pre-calculated Enformer predictions of variant impact on chromatin and gene expression. <a class="button" href="#enformer" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('enformer');" id="a_enformer">more</a></p><div id="div_enformer" style="display:none;"><p>The predictions have been aggregated across all 896 spatial bins to generate 5313 features corresponding to track prediction changes in differnet assays and cell types.</p><p>This plugin is available for GRCh37 and GRCh38</p><p>Please cite the Enformer publication alongside the VEP if you use this resource:<br /><a href="https://www.nature.com/articles/s41592-021-01252-x">https://www.nature.com/articles/s41592-021-01252-x</a></p><p>GRCh38 scores were lifted over using CrossMap from the Enformer scores available here - <a href="https://console.cloud.google.com/storage/browser/dm-enformer/variant-scores/1000-genomes/enformer">https://console.cloud.google.com/storage/browser/dm-enformer/variant-scores/1000-genomes/enformer</a></p><p>Enformer scores can be downloaded from <a href="https://ftp.ensembl.org/pub/current_variation/Enformer">https://ftp.ensembl.org/pub/current_variation/Enformer</a> for GRCh37 and GRCh38.</p><p>The plugin can then be run as default to retrieve SAD (SNP Activity Difference (SAD) and SAR (Same as SAD, by computing np.log2(1 + model(alternate_sequence)) - np.log2(1 + model(reference_sequence)) scores from Enforme :<br /><pre class="code sh_sh">./vep -i variations.vcf --assembly GRCh38 --plugin Enformer,file=/path/to/Enformer/data.vcf.gz<br /></pre>or run with option to only retrieve the SAD (SNP Activity Difference (SAD) scores - main variant effect score computed as model(alternate_sequence) - model(reference_sequence) score <br /><pre class="code sh_sh">./vep -i variations.vcf --assembly GRCh38 --plugin Enformer,file=/path/to/Enformer/data.vcf.gz,SAD=1 <br /></pre>or run with option to only retrieve the SAR (Same as SAD, by computing np.log2(1 + model(alternate_sequence)) - np.log2(1 + model(reference_sequence)) score <br /><pre class="code sh_sh">./vep -i variations.vcf --assembly GRCh38 --plugin Enformer,file=/path/to/Enformer/data.vcf.gz,SAR=1 <br /></pre>or run with option to also retrieve the principal component scores which are a reduced representation of a much bigger vector of the SAD and SAR after using principal component analysis (PCA)<br /><pre class="code sh_sh">./vep -i variations.vcf --assembly GRCh38 --plugin Enformer,file=/path/to/Enformer/data.vcf.gz,PC=1 <br /></pre><br />The tabix utility must be installed in your path to use this plugin.<br />Check <a href="https://github.com/samtools/htslib.git">https://github.com/samtools/htslib.git</a> for instructions.</p></div></td><td><div class="vdoc_dtype_count" style="float:left;padding:2px 6px;cursor:default;background-color:#DD7700">Regulatory impact</div></td><td>-</td><td>Ensembl</td></tr>
<tr class="bg1 plugin_row" data-category="pathogenicity_predictions"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/EVE.pm">EVE</a></div></td><td><p>This is a plugin for the Ensembl Variant Effect Predictor (VEP) that<br />adds information from EVE (evolutionary model of variant effect).<br />This plugin only report EVE scores for input variants<br />and does not merge input lines to report on adjacent variants.<br />It is only available for GRCh38. <a class="button" href="#eve" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('eve');" id="a_eve">more</a></p><div id="div_eve" style="display:none;"><p>Please cite EVE publication alongside the VEP if you use this resource:<br /><a href="https://www.nature.com/articles/s41586-021-04043-8">https://www.nature.com/articles/s41586-021-04043-8</a></p><p><pre class="code sh_sh">###################################################<br /># Bash script to merge all VCFs from EVE dataset. #<br />###################################################</p><p>### BEGIN</p><p># EVE input file can be downloaded from <a href="https://evemodel.org/api/proteins/bulk/download/">https://evemodel.org/api/proteins/bulk/download/</a> <br /># Input: VCF files by protein (vcf_files_missense_mutations inside zip folder)<br /># Output: Compressed Merged VCF file (vcf.gz) + index file (.tbi)</p><p>DATA_FOLDER='/<PATH-TO>/vcf_files_missense_mutations' # Fill this line<br />OUTPUT_FOLDER='/<PATH-TO>/eve_plugin' # Fill this line<br />OUTPUT_NAME='eve_merged.vcf' # Default output name</p><p># Get header from first VCF<br />cat `ls ${DATA_FOLDER}/*vcf | head -n1` > header</p><p># Get variants from all VCFs and add to a single-file<br />ls ${DATA_FOLDER}/*vcf | while read VCF; do grep -v '^#' ${VCF} >> variants; done</p><p># Merge Header + Variants in a single file<br />cat header variants | \<br />awk '$1 ~ /^#/ {print $0;next} {print $0 | "sort -k1,1V -k2,2n"}' > ${OUTPUT_FOLDER}/${OUTPUT_NAME};</p><p># Remove temporary files<br />rm header variants</p><p># Compress and index<br />bgzip ${OUTPUT_FOLDER}/${OUTPUT_NAME};</p><p># If not installed, use: sudo apt install tabix<br />tabix ${OUTPUT_FOLDER}/${OUTPUT_NAME}.gz;</p><p>### END<br /></pre></p></div></td><td><div class="vdoc_dtype_count" style="float:left;padding:2px 6px;cursor:default;background-color:#1E90FF">Pathogenicity predictions</div></td><td>-</td><td>Ensembl</td></tr>
<tr class="bg2 plugin_row" data-category="pathogenicity_predictions"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/FATHMM.pm">FATHMM</a></div></td><td><p>A VEP plugin that gets FATHMM scores and predictions for missense variants. <a class="button" href="#fathmm" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('fathmm');" id="a_fathmm">more</a></p><div id="div_fathmm" style="display:none;"><p>You will need the fathmm.py script and its dependencies (Python, Python<br />MySQLdb). You should create a "config.ini" file in the same directory as the<br />fathmm.py script with the database connection options. More information about<br />how to set up FATHMM can be found on the FATHMM website at<br /><a href="https://github.com/HAShihab/fathmm">https://github.com/HAShihab/fathmm</a></p><p>A typical installation could consist of:</p><p><pre class="code sh_sh">wget <a href="https://raw.github.com/HAShihab/fathmm/master/cgi-bin/fathmm.py">https://raw.github.com/HAShihab/fathmm/master/cgi-bin/fathmm.py</a><br />wget <a href="http://fathmm.biocompute.org.uk/database/fathmm.v2.3.SQL.gz">http://fathmm.biocompute.org.uk/database/fathmm.v2.3.SQL.gz</a><br />gunzip fathmm.v2.3.SQL.gz<br />mysql -h[host] -P[port] -u[user] -p[pass] -e"CREATE DATABASE fathmm"<br />mysql -h[host] -P[port] -u[user] -p[pass] -Dfathmm < fathmm.v2.3.SQL<br />echo -e "[DATABASE]\nHOST = [host]\nPORT = [port]\nUSER = [user]\nPASSWD = [pass]\nDB = fathmm\n" > config.ini<br /></pre></p></div></td><td><div class="vdoc_dtype_count" style="float:left;padding:2px 6px;cursor:default;background-color:#1E90FF">Pathogenicity predictions</div></td><td>-</td><td>Ensembl</td></tr>
<tr class="bg1 plugin_row" data-category="pathogenicity_predictions"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/FATHMM_MKL.pm">FATHMM_MKL</a></div></td><td><p>A VEP plugin that retrieves FATHMM-MKL scores for variants from a tabix-indexed<br />FATHMM-MKL data file. <a class="button" href="#fathmm_mkl" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('fathmm_mkl');" id="a_fathmm_mkl">more</a></p><div id="div_fathmm_mkl" style="display:none;"><p>See <a href="https://github.com/HAShihab/fathmm-MKL">https://github.com/HAShihab/fathmm-MKL</a> for details.</p><p>NB: The currently available data file is for GRCh37 only.</p></div></td><td><div class="vdoc_dtype_count" style="float:left;padding:2px 6px;cursor:default;background-color:#1E90FF">Pathogenicity predictions</div></td><td>-</td><td>Ensembl</td></tr>
<tr class="bg2 plugin_row" data-category="external_id"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/FlagLRG.pm">FlagLRG</a></div></td><td><p>A VEP plugin that retrieves the LRG ID matching either the RefSeq or Ensembl<br />transcript IDs. You can obtain the 'list_LRGs_transcripts_xrefs.txt' using: <a class="button" href="#flaglrg" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('flaglrg');" id="a_flaglrg">more</a></p><div id="div_flaglrg" style="display:none;"><p><pre class="code sh_sh">wget <a href="https://ftp.ebi.ac.uk/pub/databases/lrgex/list_LRGs_transcripts_xrefs.txt">https://ftp.ebi.ac.uk/pub/databases/lrgex/list_LRGs_transcripts_xrefs.txt</a><br /></pre></p></div></td><td><div class="vdoc_dtype_count" style="float:left;padding:2px 6px;cursor:default;background-color:#333333">External ID</div></td><td><a href="https://metacpan.org/pod/Text::CSV" rel="external">Text::CSV</a></td><td>Stephen Kazakoff</td></tr>
<tr class="bg1 plugin_row" data-category="motif"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/FunMotifs.pm">FunMotifs</a></div></td><td><p>This is a plugin for the Ensembl Variant Effect Predictor (VEP) that<br />adds tissue-specific transcription factor motifs from FunMotifs to VEP output. <a class="button" href="#funmotifs" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('funmotifs');" id="a_funmotifs">more</a></p><div id="div_funmotifs" style="display:none;"><p>Please cite the FunMotifs publication alongside the VEP if you use this resource. <br />The preprint can be found at: <a href="https://www.biorxiv.org/content/10.1101/683722v1">https://www.biorxiv.org/content/10.1101/683722v1</a></p><p>FunMotifs files can be downloaded from: <a href="http://bioinf.icm.uu.se:3838/funmotifs/">http://bioinf.icm.uu.se:3838/funmotifs/</a><br />At the time of writing, all BED files found through this link support GRCh37, <br />however other assemblies are supported by the plugin if an appropriate BED file<br />is supplied.</p><p>The tabix utility must be installed in your path to use this plugin.</p></div></td><td><div class="vdoc_dtype_count" style="float:left;padding:2px 6px;cursor:default;background-color:#DAA520">Motif</div></td><td>-</td><td>Ensembl</td></tr>
<tr class="bg2 plugin_row" data-category="phenotype_data_and_citations"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/G2P.pm">G2P</a></div><div style="margin-top:6px"><small>gene2phenotype</small></div></td><td><p>A VEP plugin that uses G2P allelic requirements to assess variants in genes<br />for potential phenotype involvement. <a class="button" href="#g2p" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('g2p');" id="a_g2p">more</a></p><div id="div_g2p" style="display:none;"><p>The plugin has multiple configuration options, though minimally requires only<br />the CSV file of G2P data.</p><p>For further information see:<br />Thormann A, Halachev M, McLaren W, et al. Flexible and scalable diagnostic filtering of genomic variants using G2P with Ensembl VEP.<br />Nature Communications. 2019 May;10(1):2373. DOI: 10.1038/s41467-019-10016-3. PMID: 31147538; PMCID: PMC6542828.</p><p>Options are passed to the plugin as key=value pairs, (defaults in parentheses):</p><p>file                  : Path to G2P data file. The file needs to be uncompressed.<br />- Download from <a href="https://www.ebi.ac.uk/gene2phenotype/downloads">https://www.ebi.ac.uk/gene2phenotype/downloads</a><br />- Download from PanelApp  </p><p>variant_include_list  : A list of variants to include even if variants do not pass allele<br />frequency filtering. The include list needs to be a sorted, bgzipped and<br />tabixed VCF file.</p><p>af_monoallelic        : maximum allele frequency for inclusion for monoallelic genes (0.0001)</p><p>af_biallelic          : maximum allele frequency for inclusion for biallelic genes (0.005)<br />confidence_levels     : Confidence levels include: definitive, strong, moderate, limited<br />Former confidence terms are still supported: confirmed, probable, possible, both RD and IF.<br />Separate multiple values with '&'.<br /><a href="https://www.ebi.ac.uk/gene2phenotype/terminology">https://www.ebi.ac.uk/gene2phenotype/terminology</a><br />Default levels are confirmed and probable.<br />all_confidence_levels : Set to 1 to include all confidence levels<br />Setting the value to 1 will overwrite any confidence levels provided with the<br />confidence_levels option.<br />af_from_vcf           : set value to 1 to include allele frequencies from VCF file. <br />Specifiy the list of reference populations to include with --af_from_vcf_keys    <br />af_from_vcf_keys      : VCF collections used for annotating variant alleles with observed<br />allele frequencies. Allele frequencies are retrieved from VCF files. If<br />af_from_vcf is set to 1 but no VCF collections are specified with --af_from_vcf_keys<br />all available VCF collections are included. <br />Available VCF collections: topmed, uk10k, gnomADe, gnomADe_r2.1.1, gnomADg, gnomADg_v3.1.2.<br />Separate multiple values with '&'.<br />VCF collections contain the following populations: <br />topmed : TOPMed (available for GRCh37 and GRCh38).<br />uk10k : ALSPAC, TWINSUK (available for GRCh37 and GRCh38).<br />gnomADe & gnomADe_r2.1.1 - gnomADe:AFR, gnomADe:ALL, gnomADe:AMR, gnomADe:ASJ, gnomADe:EAS, gnomADe:FIN, gnomADe:NFE, gnomADe:OTH, gnomADe:SAS (for GRCh37 and GRCh38 respectively).<br />gnomADg & gnomADg_v3.1.2 - gnomADg:AFR, gnomADg:ALL, gnomADg:AMR, gnomADg:ASJ, gnomADg:EAS, gnomADg:FIN, gnomADg:NFE, gnomADg:OTH (for GRCh37 and GRCh38 respectively).<br />Need to use af_from_vcf paramter to use this option. <br />default_af            : default frequency of the input variant if no frequency data is<br />found (0). This determines whether such variants are included;<br />the value of 0 forces variants with no frequency data to be<br />included as this is considered equivalent to having a frequency<br />of 0. Set to 1 (or any value higher than af) to exclude them.<br />types                 : SO consequence types to include. Separate multiple values with '&'<br />(splice_donor_variant,splice_acceptor_variant,stop_gained,<br />frameshift_variant,stop_lost,initiator_codon_variant,<br />inframe_insertion,inframe_deletion,missense_variant,<br />coding_sequence_variant,start_lost,transcript_ablation,<br />transcript_amplification,protein_altering_variant)</p><p>log_dir              : write stats to log files in log_dir </p><p>txt_report           : write all G2P complete genes and attributes to txt file</p><p>html_report          : write all G2P complete genes and attributes to html file</p><p>filter_by_gene_symbol: set to 1 if filter by gene symbol.<br />Do not set if filtering by HGNC_id.<br />This option is set to 1 when using PanelApp files. </p><p>only_mane            : set to 1 to ignore transcripts that are not MANE<br />N/B - Information may be lost if this option is used.</p><p>For more information - <a href="https://www.ebi.ac.uk/gene2phenotype/g2p_vep_plugin">https://www.ebi.ac.uk/gene2phenotype/g2p_vep_plugin</a><br />Example:</p><p><pre class="code sh_sh">--plugin G2P,file=G2P.csv,af_monoallelic=0.05,types=stop_gained&frameshift_variant<br />--plugin G2P,file=G2P.csv,af_monoallelic=0.05,af_from_vcf=1<br />--plugin G2P,file=G2P.csv,af_from_vcf=1,af_from_vcf_keys='topmed&gnomADe_r2.1.1'<br />--plugin G2P,file=G2P.csv,af_from_vcf=1,af_from_vcf_keys='topmed&gnomADe_r2.1.1',confidence_levels='confirmed&probable&both RD and IF' <br />--plugin G2P,file=G2P.csv<br /></pre></p></div></td><td><div class="vdoc_dtype_count" style="float:left;padding:2px 6px;cursor:default;background-color:#9966EE">Phenotype data and citations</div></td><td><ul style="padding-left:1em"><li><a href="https://metacpan.org/pod/Cwd" rel="external">Cwd</a></li><li><a href="https://metacpan.org/pod/FileHandle" rel="external">FileHandle</a></li><li><a href="https://metacpan.org/pod/List::Util" rel="external">List::Util</a> qw(any)</li><li><a href="https://metacpan.org/pod/Text::CSV" rel="external">Text::CSV</a></li><li><a href="https://metacpan.org/pod/Scalar::Util" rel="external">Scalar::Util</a> qw(looks_like_number)</li></ul></td><td>Ensembl</td></tr>
<tr class="bg1 plugin_row" data-category="splicing_predictions"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/GeneSplicer.pm">GeneSplicer</a></div></td><td><p>This is a plugin for the Ensembl Variant Effect Predictor (VEP) that<br />runs GeneSplicer (<a href="https://ccb.jhu.edu/software/genesplicer/">https://ccb.jhu.edu/software/genesplicer/</a>) to get<br />splice site predictions. <a class="button" href="#genesplicer" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('genesplicer');" id="a_genesplicer">more</a></p><div id="div_genesplicer" style="display:none;"><p>It evaluates a tract of sequence either side of and including the<br />variant, both in reference and alternate states. The amount of<br />sequence included either side defaults to 100bp, but can be modified<br />by passing e.g. "context=50" as a parameter to the plugin.</p><p>Any predicted splicing regions that overlap the variant are reported<br />in the output with one of four states: no_change, diff, gain, loss</p><p>There follows a "/"-separated string consisting of the following data:</p><p>1) type (donor, acceptor)<br />2) coordinates (start-end)<br />3) confidence (Low, Medium, High)<br />4) score</p><p>Example: loss/acceptor/727006-727007/High/16.231924</p><p>If multiple sites are predicted, their reports are separated by ",".</p><p>For diff, the confidence and score for both the reference and alternate<br />sequences is reported as REF-ALT.</p><p>Example: diff/donor/621915-621914/Medium-Medium/7.020731-6.988368</p><p>Several parameters can be modified by passing them to the plugin string:</p><p>context    : change the amount of sequence added either side of<br />the variant (default: 100bp)<br />tmpdir     : change the temporary directory used (default: /tmp)<br />cache_size : change how many sequences' scores are cached in memory<br />(default: 50)</p><p>Example: --plugin GeneSplicer,$GS/bin/linux/genesplicer,$GS/human,context=200,tmpdir=/mytmp</p><p>On some systems the binaries provided will not execute, but can be compiled from source:</p><p><pre class="code sh_sh">cd $GS/sources<br />make<br />cd -<br />./vep [options] --plugin GeneSplicer,$GS/sources/genesplicer,$GS/human<br /></pre>On Mac OSX the make step is known to fail; the genesplicer.cpp file requires modification:</p><p><pre class="code sh_sh">cd $GS/sources<br />perl -pi -e "s/^main  /int main  /" genesplicer.cpp<br />make<br /></pre></p></div></td><td><div class="vdoc_dtype_count" style="float:left;padding:2px 6px;cursor:default;background-color:#FF0000">Splicing predictions</div></td><td><a href="https://metacpan.org/pod/Digest::MD5" rel="external">Digest::MD5</a> qw(md5_hex)</td><td>Ensembl</td></tr>
<tr class="bg2 plugin_row" data-category="phenotype_data_and_citations"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/Geno2MP.pm">Geno2MP</a></div></td><td><p>A VEP plugin that adds information from Geno2MP, a web-accessible database of<br />rare variant genotypes linked to phenotypic information. <a class="button" href="#geno2mp" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('geno2mp');" id="a_geno2mp">more</a></p><div id="div_geno2mp" style="display:none;"><p>Parameters can be set using a key=value system:<br />file: VCF file containing Geno2MP data<br />cols: colon-delimited list of Geno2MP columns to return from INFO fields<br />(by default it only returns the column HPO_CT)<br />url: build and return URL to Geno2MP variant page (boolean; 0 by default);<br />the variant location in Geno2MP website is based on GRCh37 coordinates</p><p>Please cite Geno2MP alongside the VEP if you use this resource:<br />Geno2MP, NHGRI/NHLBI University of Washington-Center for Mendelian Genomics (UW-CMG), Seattle, WA<br />(URL: <a href="http://geno2mp.gs.washington.edu">http://geno2mp.gs.washington.edu</a> [date (month, yr) accessed]).</p></div></td><td><div class="vdoc_dtype_count" style="float:left;padding:2px 6px;cursor:default;background-color:#9966EE">Phenotype data and citations</div></td><td>-</td><td>Ensembl</td></tr>
<tr class="bg1 plugin_row" data-category="frequency_data"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/gnomADc.pm">gnomADc</a></div></td><td><p>A VEP plugin that retrieves gnomAD annotation from either the genome<br />or exome coverage files, available here:<br /><a href="https://gnomad.broadinstitute.org/downloads">https://gnomad.broadinstitute.org/downloads</a> <a class="button" href="#gnomadc" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('gnomadc');" id="a_gnomadc">more</a></p><div id="div_gnomadc" style="display:none;"><p>To download the gnomad coverage file in TSV format:<br />for Assembly GRCh37: <br />gnomad genomes:<br /><pre class="code sh_sh">wget <a href="https://storage.googleapis.com/gcp-public-data--gnomad/release/2.1/coverage/genomes/gnomad.genomes.coverage.summary.tsv.bgz">https://storage.googleapis.com/gcp-public-data--gnomad/release/2.1/coverage/genomes/gnomad.genomes.coverage.summary.tsv.bgz</a> --no-check-certificate<br /></pre>gnomad exomes:<br /><pre class="code sh_sh">wget <a href="https://storage.googleapis.com/gcp-public-data--gnomad/release/2.1/coverage/exomes/gnomad.exomes.coverage.summary.tsv.bgz">https://storage.googleapis.com/gcp-public-data--gnomad/release/2.1/coverage/exomes/gnomad.exomes.coverage.summary.tsv.bgz</a> --no-check-certificate<br /></pre>for Assembly GRCh38: <br />gnomad genomes: <br /><pre class="code sh_sh">wget <a href="https://storage.googleapis.com/gcp-public-data--gnomad/release/3.0.1/coverage/genomes/gnomad.genomes.r3.0.1.coverage.summary.tsv.bgz">https://storage.googleapis.com/gcp-public-data--gnomad/release/3.0.1/coverage/genomes/gnomad.genomes.r3.0.1.coverage.summary.tsv.bgz</a> --no-check-certificate <br /></pre>Necessary before using the plugin<br />for Assembly GRCh37:<br />The following steps are necessary to tabix the gnomad genomes coverage file :<br /><pre class="code sh_sh">gunzip -c gnomad.genomes.coverage.summary.tsv.bgz | sed '1s/.*/#&/' > gnomad.genomes.tabbed.tsv<br />bgzip gnomad.genomes.tabbed.tsv<br />tabix -s 1 -b 2 -e 2 gnomad.genomes.tabbed.tsv.gz<br /></pre>The following steps are neccessary to tabix the gnomad exomes coverage file :<br /><pre class="code sh_sh">gunzip -c gnomad.exomes.coverage.summary.tsv.bgz | sed '1s/.*/#&/' > gnomad.exomes.tabbed.tsv<br />bgzip gnomad.exomes.tabbed.tsv<br />tabix -s 1 -b 2 -e 2 gnomad.exomes.tabbed.tsv.gz<br /></pre>for Assembly GRCh38:<br />The following steps are necessary to tabix the gnomad genomes coverage file :<br /><pre class="code sh_sh">gunzip -c gnomad.genomes.r3.0.1.coverage.summary.tsv.bgz | sed '1s/.*/#&/' > gnomad.genomesv3.tabbed.tsv<br />sed "1s/locus/chr\tpos/; s/:/\t/g" gnomad.genomesv3.tabbed.tsv > gnomad.ch.genomesv3.tabbed.tsv<br />bgzip gnomad.ch.genomesv3.tabbed.tsv<br />tabix -s 1 -b 2 -e 2 gnomad.ch.genomesv3.tabbed.tsv<br /></pre>This plugin also tries to be backwards compatible with older versions of the<br />coverage summary files, including releases 2.0.1 and 2.0.2. These releases<br />provide one coverage file per chromosome and these can be used "as-is"<br />without requiring any preprocessing. </p><p>If you use this plugin, please see the terms and data information:<br /><a href="https://gnomad.broadinstitute.org/terms">https://gnomad.broadinstitute.org/terms</a></p><p>You must have the Bio::DB::HTS module or the tabix utility must be installed<br />in your path to use this plugin. </p></div></td><td><div class="vdoc_dtype_count" style="float:left;padding:2px 6px;cursor:default;background-color:#FF7F50">Frequency data</div></td><td><ul style="padding-left:1em"><li><a href="https://metacpan.org/pod/File::Spec" rel="external">File::Spec</a></li><li><a href="https://metacpan.org/pod/File::Basename" rel="external">File::Basename</a></li></ul></td><td>Stephen Kazakoff</td></tr>
<tr class="bg2 plugin_row" data-category="phenotype_data_and_citations"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/GO.pm">GO</a></div><div style="margin-top:6px"><small>Gene Ontology</small></div></td><td><p>A VEP plugin that retrieves Gene Ontology (GO) terms associated with<br />transcripts (e.g. GRCh38) or their translations (e.g. GRCh37) using custom<br />GFF annotation containing GO terms. <a class="button" href="#go" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('go');" id="a_go">more</a></p><div id="div_go" style="display:none;"><p>The custom GFF files are automatically created if the input file do not exist<br />by querying the Ensembl core database, according to database version, species<br />and assembly used in VEP. Note that automatic retrieval fails if using the<br />--offline option.</p><p>The GFF files containing the GO terms are saved to and loaded from the working<br />directory by default. To change this, provide a directory path as an argument:</p><p><pre class="code sh_sh">--plugin GO,dir=${HOME}/go_terms<br /></pre>If your GFF file has a custom name, please provide the filename directly:</p><p><pre class="code sh_sh">--plugin GO,file=${HOME}/custom_go_terms.gff.gz<br /></pre>The GO terms can also be fetched by gene match (either gene Ensembl ID or<br />gene symbol) instead:</p><p><pre class="code sh_sh">--plugin GO,match=gene<br />--plugin GO,match=gene_symbol<br /></pre>To create/use a custom GFF file, these programs must be installed in your path:<br />* The GNU zgrep and GNU sort commands to create the GFF file.<br />* The tabix and bgzip utilities to create and read the GFF file: check<br /><a href="https://github.com/samtools/htslib.git">https://github.com/samtools/htslib.git</a> for installation instructions.</p><p>Alternatively, for compatibility purposes, the plugin allows to use a remote<br />connection to the Ensembl API by using "remote" as a parameter. This method<br />retrieves GO terms one by one at both the transcript and translation level.<br />This is not compatible with any other parameters:</p><p><pre class="code sh_sh">--plugin GO,remote<br /></pre></p></div></td><td><div class="vdoc_dtype_count" style="float:left;padding:2px 6px;cursor:default;background-color:#9966EE">Phenotype data and citations</div></td><td>-</td><td>Ensembl</td></tr>
<tr class="bg1 plugin_row" data-category="phenotype_data_and_citations"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/GWAS.pm">GWAS</a></div></td><td><p>A VEP plugin that retrieves relevant NHGRI-EBI GWAS Catalog data given the file.  <a class="button" href="#gwas" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('gwas');" id="a_gwas">more</a></p><div id="div_gwas" style="display:none;"><p>This plugin supports both the curated data that is found in the download section of the NHGRI-EBI GWAS Catalog website and the <br />summary statistics file. By default the plugin assumes the file provided is the curated file but you can pass "type=sstate" <br />to say you want to annotate with a summary statistics file.</p><p>Please cite the following publication alongside the VEP if you use this resource:<br /><a href="https://pubmed.ncbi.nlm.nih.gov/30445434/">https://pubmed.ncbi.nlm.nih.gov/30445434/</a></p><p>Pre-requisites:</p><p>For curated NHGRI-EBI GWAS Catalog file -<br />GWAS files can be downloaded from - <a href="https://www.ebi.ac.uk/gwas/api/search/downloads/alternative">https://www.ebi.ac.uk/gwas/api/search/downloads/alternative</a></p><p>For summary statistics file -<br />The plugin can process the harmonised version of the summary statistics file. Which can be downloaded from the FTP site - <br /><a href="http://ftp.ebi.ac.uk/pub/databases/gwas/summary_statistics">http://ftp.ebi.ac.uk/pub/databases/gwas/summary_statistics</a></p><p>They are under directory with related to their specific GCST id. For example -<br /><a href="http://ftp.ebi.ac.uk/pub/databases/gwas/summary_statistics/GCST000001-GCST001000/GCST000028/harmonised/17463246-GCST000028-EFO_0001360.h.tsv.gz">http://ftp.ebi.ac.uk/pub/databases/gwas/summary_statistics/GCST000001-GCST001000/GCST000028/harmonised/17463246-GCST000028-EFO_0001360.h.tsv.gz</a></p><p>Please keep the filename format as it is because filename is parsed to get information.</p><p>When run for the first time the for either type of file plugin will create a processed file that have genomic locations and indexed and <br />put it under the --dir location determined by Ensembl VEP. It might take 1~2 hour(s) to create the processed file depending on the file size. <br />But subsequent runs will be faster as the plugin will be using the already generated processed file.</p><p>Options are passed to the plugin as key=value pairs:</p><p>file   : (mandatory) Path to GWAS curated or summary statistics file<br />type   : type of the file. Valid values are "curated" and "sstate".</p></div></td><td><div class="vdoc_dtype_count" style="float:left;padding:2px 6px;cursor:default;background-color:#9966EE">Phenotype data and citations</div></td><td><ul style="padding-left:1em"><li><a href="https://metacpan.org/pod/Storable" rel="external">Storable</a> qw(dclone)</li><li><a href="https://metacpan.org/pod/File::Basename" rel="external">File::Basename</a></li></ul></td><td>Ensembl</td></tr>
<tr class="bg2 plugin_row" data-category="hgvs"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/HGVSIntronOffset.pm">HGVSIntronOffset</a></div></td><td><p>A VEP plugin for the Ensembl Variant Effect Predictor (VEP) that returns<br />HGVS intron start and end offsets. To be used with --hgvs option.</p></td><td><div class="vdoc_dtype_count" style="float:left;padding:2px 6px;cursor:default;background-color:#9400D3">HGVS</div></td><td>-</td><td>Stephen Kazakoff</td></tr>
<tr class="bg1 plugin_row" data-category="functional_effect"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/IntAct.pm">IntAct</a></div></td><td><p>A VEP plugin that retrieves molecular interaction data for variants as reprted by IntAct database. <a class="button" href="#intact" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('intact');" id="a_intact">more</a></p><div id="div_intact" style="display:none;"><p>Please cite the IntAct publication alongside the VEP if you use this resource:<br /><a href="https://pubmed.ncbi.nlm.nih.gov/24234451/">https://pubmed.ncbi.nlm.nih.gov/24234451/</a></p><p>Pre-requisites:</p><p>1) IntAct files can be downloaded from -<br /><a href="https://ftp.ebi.ac.uk/pub/databases/intact/current/various">https://ftp.ebi.ac.uk/pub/databases/intact/current/various</a></p><p>2) The genomic location mapped file needs to be tabix indexed. You can <br />do this by following commands -</p><p>a) filter, sort and then zip<br />grep -v -e '^$' -e '^[#\-]' mutation_gc_map.txt | sed '1s/.*/#&/' | awk -F "\t" 'BEGIN { OFS="\t"} {if ($2 > $3) {a=$2; $2=$3; $3=a}; print $0 }' | sort -k1,1 -k2,2n -k3,3n | bgzip > mutation_gc_map.txt.gz</p><p>b) create tabix indexed file -<br /><pre class="code sh_sh">tabix -s 1 -b 2 -e 3 -f mutation_gc_map.txt.gz<br /></pre>3) As you have already noticed, tabix utility must be installed in your path to use this plugin.</p><p>Options are passed to the plugin as key=value pairs:</p><p>mapping_file			: (mandatory) Path to tabix-indexed genomic location mapped file<br />mutation_file			: (mandatory) Path to IntAct data file</p><p>By default the output will always contain feature_type and interaction_ac from the IntAct data file. You can also add more fields using the following options -<br />feature_ac			: Set value to 1 to include Feature AC in the output<br />feature_short_label		: Set value to 1 to include Feature short label in the output<br />feature_annotation		: Set value to 1 to include Feature annotation in the output<br />ap_ac				: Set value to 1 to include Affected protein AC in the output<br />interaction_participants	: Set value to 1 to include Interaction participants in the output<br />pmid				: Set value to 1 to include PubMedID in the output</p><p>There are also two other options for customizing the output - <br />all                            : Set value to 1 to include all the fields<br />minimal                        : Set value to 1 to overwrite default behavior and include only interaction_ac <br />in the output by default</p><p>See what this options mean - <a href="https://www.ebi.ac.uk/intact/download/datasets#mutations">https://www.ebi.ac.uk/intact/download/datasets#mutations</a></p><p>Note that, interaction accession can be used to link to full details on the interaction website. For example, <br />where the VEP output reports an interaction_ac of EBI-12501485, the URL would be : </p><p><a href="https://www.ebi.ac.uk/intact/details/interaction/EBI-12501485">https://www.ebi.ac.uk/intact/details/interaction/EBI-12501485</a></p></div></td><td><div class="vdoc_dtype_count" style="float:left;padding:2px 6px;cursor:default;background-color:#443377">Functional effect</div></td><td>-</td><td>Ensembl</td></tr>
<tr class="bg2 plugin_row" data-category="variant_data"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/LD.pm">LD</a></div><div style="margin-top:6px"><small>Linkage Disequilibrium</small></div></td><td><p>A VEP plugin that finds variants in linkage disequilibrium with any overlapping<br />existing variants from the Ensembl variation databases. <a class="button" href="#ld" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('ld');" id="a_ld">more</a></p><div id="div_ld" style="display:none;"><p>You can configure the population used to calculate the r2 value, and the<br />r2 cutoff used by passing arguments to the plugin via the VEP command line<br />(separated by commas). This plugin adds a single new entry to the Extra column<br />with a comma-separated list of linked variant IDs and the associated r2 values:<br />LinkedVariants=rs123:0.879,rs234:0.943</p><p>If no arguments are supplied, the default population used is the CEU<br />sample from the 1000 Genomes Project phase 3, and the default r2<br />cutoff used is 0.8.</p><p>WARNING: Calculating LD is a relatively slow procedure, so this will <br />slow VEP down considerably when running on large numbers of<br />variants. Consider running vep followed by filter_vep to get a smaller<br />input set:</p><p><pre class="code sh_sh">./vep -i input.vcf -cache -vcf -o input_vep.vcf<br />./filter_vep -i input_vep.vcf -filter "Consequence is missense_variant" > input_vep_filtered.vcf<br />./vep -i input_vep_filtered.vcf -cache -plugin LD<br /></pre></p></div></td><td><div class="vdoc_dtype_count" style="float:left;padding:2px 6px;cursor:default;background-color:#B22222">Variant data</div></td><td>-</td><td>Ensembl</td></tr>
<tr class="bg1 plugin_row" data-category="look_up"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/LocalID.pm">LocalID</a></div></td><td><p>The LocalID plugin allows you to use variant IDs as input without making a database connection. <a class="button" href="#localid" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('localid');" id="a_localid">more</a></p><div id="div_localid" style="display:none;"><p>Requires sqlite3.</p><p>A local sqlite3 database is used to look up variant IDs; this is generated either from Ensembl's<br />public database (very slow, but includes synonyms), or from a VEP cache file (faster, excludes<br />synonyms).</p><p>NB this plugin is NOT compatible with the ensembl-tools variant_effect_predictor.pl version of VEP.</p></div></td><td><div class="vdoc_dtype_count" style="float:left;padding:2px 6px;cursor:default;background-color:#006400">Look up</div></td><td>-</td><td>Ensembl</td></tr>
<tr class="bg2 plugin_row" data-category="pathogenicity_predictions"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/LOEUF.pm">LOEUF</a></div></td><td><p>This is a plugin for the Ensembl Variant Effect Predictor (VEP) that<br />adds the LOEUF scores to VEP output. LOEUF stands for the "loss-of-function <br />observed/expected upper bound fraction."  <a class="button" href="#loeuf" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('loeuf');" id="a_loeuf">more</a></p><div id="div_loeuf" style="display:none;"><p>The score can be added matching by either transcript or gene.<br />When matched by gene: <br />If multiple transcripts are available for a gene, the most severe score is reported.</p><p>NB: The plugin currently does not add the score for downstream_gene_variant and upstream_gene_variant </p><p>Please cite the LOEUF publication alongside the VEP if you use this resource:<br /><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7334197/">https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7334197/</a></p><p>LOEUF scores can be downloaded from<br /><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7334197/bin/41586_2020_2308_MOESM4_ESM.zip">https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7334197/bin/41586_2020_2308_MOESM4_ESM.zip</a></p><p>For GRCh37:<br />These files can be tabix-processed by:<br />Unzip 41586_2020_2308_MOESM4_ESM.zip<br /><pre class="code sh_sh">cd supplement<br />zcat supplementary_dataset_11_full_constraint_metrics.tsv.gz | (head -n 1 && tail -n +2  | sort -t$'\t' -k 76,76 -k 77,77n ) > loeuf_temp.tsv<br />sed '1s/.*/#&/' loeuf_temp.tsv > loeuf_dataset.tsv<br />bgzip loeuf_dataset.tsv<br />tabix -f -s 76 -b 77 -e 78 loeuf_dataset.tsv.gz<br /></pre>For GRCh38:<br />The LOEUF scores are available for assembly GRCh37, to be able to run the plugin for GRCh38<br />please remap the regions from file supplementary_dataset_11_full_constraint_metrics.tsv<br />After the remapping the file can be tabix-processed by:<br /><pre class="code sh_sh">cat supplementary_dataset_11_full_constraint_metrics_grch38.tsv | (head -n 1 && tail -n +2  | sort -t$'\t' -k 76,76 -k 77,77n ) > loeuf_grch38_temp.tsv<br />sed '1s/.*/#&/' loeuf_grch38_temp.tsv > loeuf_dataset_grch38.tsv<br />bgzip loeuf_dataset_grch38.tsv<br />tabix -f -s 76 -b 77 -e 78 loeuf_dataset_grch38.tsv.gz<br /></pre><br />The tabix utility must be installed in your path to use this plugin.</p></div></td><td><div class="vdoc_dtype_count" style="float:left;padding:2px 6px;cursor:default;background-color:#1E90FF">Pathogenicity predictions</div></td><td><a href="https://metacpan.org/pod/Scalar::Util" rel="external">Scalar::Util</a> qw(looks_like_number)</td><td>Ensembl</td></tr>
<tr class="bg1 plugin_row" data-category="pathogenicity_predictions"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/LoFtool.pm">LoFtool</a></div><div style="margin-top:6px"><small>Loss-of-function</small></div></td><td><p>Add LoFtool scores to the VEP output. <a class="button" href="#loftool" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('loftool');" id="a_loftool">more</a></p><div id="div_loftool" style="display:none;"><p>LoFtool provides a rank of genic intolerance and consequent<br />susceptibility to disease based on the ratio of Loss-of-function (LoF)<br />to synonymous mutations for each gene in 60,706 individuals from ExAC,<br />adjusting for the gene de novo mutation rate and evolutionary protein<br />conservation. The lower the LoFtool gene score percentile the most<br />intolerant is the gene to functional variation. For more details please see<br />(Fadista J et al. 2017), PMID:27563026.<br />The authors would like to thank the Exome Aggregation Consortium and<br />the groups that provided exome variant data for comparison. A full<br />list of contributing groups can be found at <a href="http://exac.broadinstitute.org/about">http://exac.broadinstitute.org/about</a>.</p><p>The LoFtool_scores.txt file is found alongside the plugin in the<br />VEP_plugins GitHub repo.</p><p>To use another scores file, add it as a parameter i.e.</p><p><pre class="code sh_sh">./vep -i variants.vcf --plugin LoFtool,scores_file.txt<br /></pre></p></div></td><td><div class="vdoc_dtype_count" style="float:left;padding:2px 6px;cursor:default;background-color:#1E90FF">Pathogenicity predictions</div></td><td><a href="https://metacpan.org/pod/DBI" rel="external">DBI</a></td><td>Ensembl</td></tr>
<tr class="bg2 plugin_row" data-category="variant_data"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/LOVD.pm">LOVD</a></div><div style="margin-top:6px"><small>Leiden Open Variation Database</small></div></td><td><p>A VEP plugin that retrieves LOVD variation data from <a href="http://www.lovd.nl/">http://www.lovd.nl/</a>. <a class="button" href="#lovd" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('lovd');" id="a_lovd">more</a></p><div id="div_lovd" style="display:none;"><p>Please be aware that LOVD is a public resource of curated variants, therefore<br />please respect this resource and avoid intensive querying of their databases<br />using this plugin, as it will impact the availability of this resource for<br />others.</p></div></td><td><div class="vdoc_dtype_count" style="float:left;padding:2px 6px;cursor:default;background-color:#B22222">Variant data</div></td><td><a href="https://metacpan.org/pod/LWP::UserAgent" rel="external">LWP::UserAgent</a></td><td>Ensembl</td></tr>
<tr class="bg1 plugin_row" data-category="phenotype_data_and_citations"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/Mastermind.pm">Mastermind</a></div></td><td><p>This is a plugin for the Ensembl Variant Effect Predictor (VEP) that<br />uses the Mastermind Genomic Search Engine (<a href="https://www.genomenon.com/mastermind">https://www.genomenon.com/mastermind</a>)<br />to report variants that have clinical evidence cited in the medical literature. <br />It is available for both GRCh37 and GRCh38. <a class="button" href="#mastermind" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('mastermind');" id="a_mastermind">more</a></p><div id="div_mastermind" style="display:none;"><p>Please cite the Mastermind publication alongside the VEP if you use this resource:<br /><a href="https://www.frontiersin.org/article/10.3389/fgene.2020.577152">https://www.frontiersin.org/article/10.3389/fgene.2020.577152</a></p><p>Running options:<br />The plugin has multiple parameters, the first one is expected to be the file name path <br />which can be followed by 3 optional flags.<br />Default: the plugin matches the citation data with the specific mutation.<br />Using first flag '1': returns the citations for all mutations/transcripts.   <br />Using the second flag '1': only returns the Mastermind variant identifier(s).<br />Using the third flag '1': also returns the Mastermind URL.</p><p>Output: <br />The output includes three unique counts 'MMCNT1, MMCNT2, MMCNT3' and one identifier 'MMID3'<br />to be used to build an URL which shows all articles from MMCNT3.</p><p>'MMCNT1' is the count of Mastermind articles with cDNA matches for a specific variant;<br />'MMCNT2' is the count of Mastermind articles with variants either explicitly matching at<br />the cDNA level or given only at protein level;<br />'MMCNT3' is the count of Mastermind articles including other DNA-level variants resulting<br />in the same amino acid change;<br />'MMID3' is the Mastermind variant identifier(s), as gene:key. Link to the Genomenon Mastermind Genomic Search Engine;</p><p>To build the URL, substitute the 'gene:key' in the following link with the value from MMID3:<br /><a href="https://mastermind.genomenon.com/detail?mutation=gene">https://mastermind.genomenon.com/detail?mutation=gene</a>:key</p><p>If the third flag is used then the built URL is returned and it's identified by 'URL'.</p><p>More information can be found at: <a href="https://www.genomenon.com/cvr/">https://www.genomenon.com/cvr/</a></p><p>The following steps are necessary before running this plugin:</p><p>Download and Registry (free): <br /><a href="https://www.genomenon.com/cvr/">https://www.genomenon.com/cvr/</a> </p><p>GRCh37 VCF:<br /><pre class="code sh_sh">unzip mastermind_cited_variants_reference-XXXX.XX.XX-grch37-vcf.zip<br />bgzip mastermind_cited_variants_reference-XXXX.XX.XX-GRCh37-vcf<br />tabix -p vcf mastermind_cited_variants_reference-XXXX.XX.XX.GRCh37-vcf.gz<br /></pre>GRCh38 VCF:<br /><pre class="code sh_sh">unzip mastermind_cited_variants_reference-XXXX.XX.XX-grch38-vcf.zip<br />bgzip mastermind_cited_variants_reference-XXXX.XX.XX-GRCh38-vcf<br />tabix -p vcf mastermind_cited_variants_reference-XXXX.XX.XX.GRCh38-vcf.gz<br /></pre> <br />The plugin can then be run as default:<br /><pre class="code sh_sh">./vep -i variations.vcf --plugin Mastermind,file=/path/to/mastermind_cited_variants_reference-XXXX.XX.XX.GRChXX-vcf.gz<br /></pre>or with an option to not filter by mutations (first flag): <br /><pre class="code sh_sh">./vep -i variations.vcf --plugin Mastermind,file=/path/to/mastermind_cited_variants_reference-XXXX.XX.XX.GRChXX-vcf.gz,mutations=1 <br /></pre>or with an option to only return 'MMID3' e.g. the Mastermind variant identifier as gene:key (second flag):<br /><pre class="code sh_sh">./vep -i variations.vcf --plugin Mastermind,file=/path/to/mastermind_cited_variants_reference-XXXX.XX.XX.GRChXX-vcf.gz,mutations=0,var_iden=1<br /></pre>or with an option to also return the Mastermind URL (third flag):<br /><pre class="code sh_sh">./vep -i variations.vcf --plugin Mastermind,file=/path/to/mastermind_cited_variants_reference-XXXX.XX.XX.GRChXX-vcf.gz,mutations=0,var_iden=0,url=1<br /></pre>Note: when running VEP in offline mode Mastermind requires a fasta file (--fasta)</p></div></td><td><div class="vdoc_dtype_count" style="float:left;padding:2px 6px;cursor:default;background-color:#9966EE">Phenotype data and citations</div></td><td>-</td><td>Ensembl</td></tr>
<tr class="bg2 plugin_row" data-category="functional_effect"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/MaveDB.pm">MaveDB</a></div></td><td><p>A VEP plugin that retrieves data from MaveDB (<a href="https://www.mavedb.org">https://www.mavedb.org</a>), a<br />database that contains multiplex assays of variant effect, including deep<br />mutational scans and massively parallel report assays. <a class="button" href="#mavedb" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('mavedb');" id="a_mavedb">more</a></p><div id="div_mavedb" style="display:none;"><p>To run the MaveDB plugin, please download the following file with MaveDB data<br />mapped to variants:<br /><a href="https://ftp.ensembl.org/pub/current_variation/MaveDB/MaveDB_variants.tsv.gz">https://ftp.ensembl.org/pub/current_variation/MaveDB/MaveDB_variants.tsv.gz</a></p><p>Options are passed to the plugin as key=value pairs:<br />file                     : (mandatory) Tabix-indexed MaveDB file<br />cols                     : Colon-separated columns to print from MaveDB<br />files; if set to 'all', all columns are<br />printed (default: 'urn:score:nt:pro')<br />single_aminoacid_changes : Return matches for single aminoacid changes only;<br />if disabled, return all matches associated with a<br />genetic variant (default: 1)<br />transcript_match         : Return results only if (Ensembl or RefSeq)<br />transcript identifiers match (default: 1)</p><p>Please cite the MaveDB publication alongside the VEP if you use this resource:<br /><a href="https://doi.org/10.1186/s13059-019-1845-6">https://doi.org/10.1186/s13059-019-1845-6</a></p><p>The tabix utility must be installed in your path to use this plugin.</p></div></td><td><div class="vdoc_dtype_count" style="float:left;padding:2px 6px;cursor:default;background-color:#443377">Functional effect</div></td><td><ul style="padding-left:1em"><li><a href="https://metacpan.org/pod/Bio::SeqUtils" rel="external">Bio::SeqUtils</a></li><li><a href="https://metacpan.org/pod/File::Basename" rel="external">File::Basename</a></li></ul></td><td>Ensembl</td></tr>
<tr class="bg1 plugin_row" data-category="splicing_predictions"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/MaxEntScan.pm">MaxEntScan</a></div></td><td><p>This is a plugin for the Ensembl Variant Effect Predictor (VEP) that<br />runs MaxEntScan (<a href="http://hollywood.mit.edu/burgelab/maxent/Xmaxentscan_scoreseq.html">http://hollywood.mit.edu/burgelab/maxent/Xmaxentscan_scoreseq.html</a>)<br />to get splice site predictions. <a class="button" href="#maxentscan" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('maxentscan');" id="a_maxentscan">more</a></p><div id="div_maxentscan" style="display:none;"><p>The plugin copies most of the code verbatim from the score5.pl and score3.pl<br />scripts provided in the MaxEntScan download. To run the plugin you must get and<br />unpack the archive from <a href="http://hollywood.mit.edu/burgelab/maxent/download/">http://hollywood.mit.edu/burgelab/maxent/download/</a>; the path<br />to this unpacked directory is then the param you pass to the --plugin flag.</p><p>The plugin executes the logic from one of the scripts depending on which<br />splice region the variant overlaps:</p><p>score5.pl : last 3 bases of exon    --> first 6 bases of intron<br />score3.pl : last 20 bases of intron --> first 3 bases of exon</p><p>The plugin reports the reference, alternate and difference (REF - ALT) maximum<br />entropy scores.</p><p>If 'SWA' is specified as a command-line argument, a sliding window algorithm<br />is applied to subsequences containing the reference and alternate alleles to<br />identify k-mers with the highest donor and acceptor splice site scores. To assess<br />the impact of variants, reference comparison scores are also provided. For SNVs,<br />the comparison scores are derived from sequence in the same frame as the highest<br />scoring k-mers containing the alternate allele. For all other variants, the<br />comparison scores are derived from the highest scoring k-mers containing the<br />reference allele. The difference between the reference comparison and alternate<br />scores (SWA_REF_COMP - SWA_ALT) are also provided.</p><p>If 'NCSS' is specified as a command-line argument, scores for the nearest<br />upstream and downstream canonical splice sites are also included.</p><p>By default, only scores are reported. Add 'verbose' to the list of command-<br />line arguments to include the sequence output associated with those scores.</p></div></td><td><div class="vdoc_dtype_count" style="float:left;padding:2px 6px;cursor:default;background-color:#FF0000">Splicing predictions</div></td><td><a href="https://metacpan.org/pod/Digest::MD5" rel="external">Digest::MD5</a> qw(md5_hex)</td><td>Ensembl</td></tr>
<tr class="bg2 plugin_row" data-category="pathogenicity_predictions"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/MPC.pm">MPC</a></div><div style="margin-top:6px"><small>missense deleteriousness metric</small></div></td><td><p>A VEP plugin that retrieves MPC scores for variants from a tabix-indexed MPC data file. <a class="button" href="#mpc" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('mpc');" id="a_mpc">more</a></p><div id="div_mpc" style="display:none;"><p>MPC is a missense deleteriousness metric based on the analysis of genic regions<br />depleted of missense mutations in the Exome Agggregation Consortium (ExAC) data.</p><p>The MPC score is the product of work by Kaitlin Samocha (ks20@sanger.ac.uk).<br />Publication currently in pre-print: Samocha et al bioRxiv 2017 (TBD)</p><p>The MPC score file is available to download from:</p><p><a href="https://ftp.broadinstitute.org/pub/ExAC_release/release1/regional_missense_constraint/">https://ftp.broadinstitute.org/pub/ExAC_release/release1/regional_missense_constraint/</a></p><p>The data are currently mapped to GRCh37 only. Not all transcripts are included; see<br />README in the above directory for exclusion criteria.</p></div></td><td><div class="vdoc_dtype_count" style="float:left;padding:2px 6px;cursor:default;background-color:#1E90FF">Pathogenicity predictions</div></td><td>-</td><td>Ensembl</td></tr>
<tr class="bg1 plugin_row" data-category="pathogenicity_predictions"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/MTR.pm">MTR</a></div><div style="margin-top:6px"><small>Missense Tolerance Ratio</small></div></td><td><p>A VEP plugin that retrieves Missense Tolerance Ratio (MTR) scores for<br />variants from a tabix-indexed flat file. <a class="button" href="#mtr" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('mtr');" id="a_mtr">more</a></p><div id="div_mtr" style="display:none;"><p>MTR scores quantify the amount of purifying selection acting<br />specifically on missense variants in a given window of protein-coding<br />sequence. It is estimated across a sliding window of 31 codons and uses<br />observed standing variation data from the WES component of the Exome<br />Aggregation Consortium Database (ExAC), version 2.0<br />(<a href="http://gnomad.broadinstitute.org">http://gnomad.broadinstitute.org</a>).</p><p>Please cite the MTR publication alongside the VEP if you use this resource:<br /><a href="http://genome.cshlp.org/content/27/10/1715">http://genome.cshlp.org/content/27/10/1715</a></p><p>The Bio::DB::HTS perl library or tabix utility must be installed in your path to use this plugin. <br />MTR flat files can be downloaded from <a href="http://biosig.unimelb.edu.au/mtr-viewer/downloads">http://biosig.unimelb.edu.au/mtr-viewer/downloads</a><br />The following steps are necessary before running the plugin <br /><pre class="code sh_sh">gzip -d mtrflatfile_2.0.txt.gz # to unzip the text file <br />cat mtrflatfile_2.0.txt | tr " " "\t" > mtrflatfile_2.00.tsv # to change the file to a tabbed delimited file <br />sed '1s/.*/#&/'  mtrflatfile_2.00.tsv > mtrflatfile_2.0.tsv # to add # to the first line of the file <br />bgzip mtrflatfile_2.0.tsv<br />tabix -f -s 1 -b 2 -e 2 mtrflatfile_2.0.tsv.gz<br /></pre>NB: Data are available for GRCh37 only</p></div></td><td><div class="vdoc_dtype_count" style="float:left;padding:2px 6px;cursor:default;background-color:#1E90FF">Pathogenicity predictions</div></td><td>-</td><td><ul style="padding-left:1em"><li>Slave Petrovski</li><li>Michael Silk</li></ul></td></tr>
<tr class="bg2 plugin_row" data-category="protein_annotation"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/mutfunc.pm">mutfunc</a></div></td><td><p>A VEP plugin that retrieves data from mutfunc db predicting destabilization of protein structure, interaction interface, and motif. <a class="button" href="#mutfunc" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('mutfunc');" id="a_mutfunc">more</a></p><div id="div_mutfunc" style="display:none;"><p>Please cite the mutfunc publication alongside the VEP if you use this resource:<br /><a href="http://msb.embopress.org/content/14/12/e8430">http://msb.embopress.org/content/14/12/e8430</a></p><p>Pre-requisites:</p><p>1) The data file. mutfunc SQLite db can be downloaded from - <br /><a href="https://ftp.ensembl.org/pub/current_variation/mutfunc/mutfunc_data.db">https://ftp.ensembl.org/pub/current_variation/mutfunc/mutfunc_data.db</a></p><p>Options are passed to the plugin as key=value pairs:</p><p>By default all the fields (motif, int, mod, and exp) are added in the output. But if you want to have some selected fields and not all of<br />them just select the relevant options. The default behavior will then go away outputting only the selected fields.</p><p>db			  : (mandatory) Path to SQLite database containing data for other analysis.<br />motif    : Select this option to have mutfunc motif analysis in the output<br />int      : Select this option to have mutfunc protein interection analysis in the output<br />mod      : Select this option to have mutfunc protein structure analysis in the output<br />exp      : Select this option to have mutfunc protein structure (experimental) analysis in the output<br />extended : By default mutfunc outputs the most significant field for any analysis. Select this option to get more verbose output.</p></div></td><td><div class="vdoc_dtype_count" style="float:left;padding:2px 6px;cursor:default;background-color:#0011DD">Protein annotation</div></td><td><ul style="padding-left:1em"><li><a href="https://metacpan.org/pod/DBI" rel="external">DBI</a></li><li><a href="https://metacpan.org/pod/Digest::MD5" rel="external">Digest::MD5</a> qw(md5_hex)</li><li><a href="https://metacpan.org/pod/Compress::Zlib" rel="external">Compress::Zlib</a></li><li><a href="https://metacpan.org/pod/List::MoreUtils" rel="external">List::MoreUtils</a> qw(first_index)</li></ul></td><td>Ensembl</td></tr>
<tr class="bg1 plugin_row" data-category="nearby_features"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/NearestExonJB.pm">NearestExonJB</a></div></td><td><p>This is a plugin for the Ensembl Variant Effect Predictor (VEP) that<br />finds the nearest exon junction boundary to a coding sequence variant. More than <br />one boundary may be reported if the boundaries are equidistant. <a class="button" href="#nearestexonjb" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('nearestexonjb');" id="a_nearestexonjb">more</a></p><div id="div_nearestexonjb" style="display:none;"><p>The plugin will report the Ensembl identifier of the exon, the distance to the<br />exon boundary, the boundary type (start or end of exon) and the total<br />length in nucleotides of the exon.</p><p>Various parameters can be altered by passing them to the plugin command:</p><p>- max_range : maximum search range in bp (default: 10000)</p><p>Parameters are passed e.g.:</p><p><pre class="code sh_sh">--plugin NearestExonJB,max_range=50000<br /></pre></p></div></td><td><div class="vdoc_dtype_count" style="float:left;padding:2px 6px;cursor:default;background-color:#E75480">Nearby features</div></td><td>-</td><td>Ensembl</td></tr>
<tr class="bg2 plugin_row" data-category="nearby_features"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/NearestGene.pm">NearestGene</a></div></td><td><p>This is a plugin for the Ensembl Variant Effect Predictor (VEP) that<br />finds the nearest gene(s) to a non-genic variant. More than one gene<br />may be reported if the genes overlap the variant or if genes are<br />equidistant. <a class="button" href="#nearestgene" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('nearestgene');" id="a_nearestgene">more</a></p><div id="div_nearestgene" style="display:none;"><p>Various parameters can be altered by passing them to the plugin command:</p><p>- limit     : limit the number of genes returned (default: 1)<br />- range     : initial search range in bp (default: 1000)<br />- max_range : maximum search range in bp (default: 10000)</p><p>Parameters are passed e.g.:</p><p><pre class="code sh_sh">--plugin NearestGene,limit=3,max_range=50000<br /></pre>This plugin requires a database connection. It cannot be run with VEP<br />in offline mode i.e. using the --offline flag.</p></div></td><td><div class="vdoc_dtype_count" style="float:left;padding:2px 6px;cursor:default;background-color:#E75480">Nearby features</div></td><td>-</td><td>Ensembl</td></tr>
<tr class="bg1 plugin_row" data-category="protein_data"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/neXtProt.pm">neXtProt</a></div></td><td><p>This is a plugin for the Ensembl Variant Effect Predictor (VEP) that<br />retrieves data for missense and stop gain variants from neXtProt, which is a comprehensive <br />human-centric discovery platform that offers integration of and navigation <br />through protein-related data for example, variant information, localization <br />and interactions  (<a href="https://www.nextprot.org/">https://www.nextprot.org/</a>). <a class="button" href="#nextprot" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('nextprot');" id="a_nextprot">more</a></p><div id="div_nextprot" style="display:none;"><p>Please cite the neXtProt publication alongside the VEP if you use this resource:<br /><a href="https://doi.org/10.1093/nar/gkz995">https://doi.org/10.1093/nar/gkz995</a></p><p>This plugin is only suitable for small sets of variants as an additional <br />individual remote API query is run for each variant.</p><p>The neXtProt_headers.txt file is a requirement for running this plugin and is found alongside the plugin in the<br />VEP_plugins GitHub repository. The file contains the RDF entities extracted from <br /><a href="https://snorql.nextprot.org/">https://snorql.nextprot.org/</a></p><p>Running options:<br />(Default) the data retrieved by default is the MatureProtein, NucleotidePhosphateBindingRegion,<br />Variant, MiscellaneousRegion, TopologicalDomain and InteractingRegion.<br />The plugin can also be run with other options to retrieve other data than the default.</p><p>Options are passed to the plugin as key=value pairs:<br />max_set        : Set value to 1 to return all available protein-related data <br />(includes the default data)</p><p>return_values  : The set of data to be returned with different data separated by '&'. <br />Use file 'neXtProt_headers.txt' to check which data (labels) are available.<br />Example: --plugin neXtProt,return_values='Domain&InteractingRegion'</p><p>url            : Set value to 1 to include the URL to link to the neXtProt entry.</p><p>all_labels     : Set value to 1 to include all labels, even if data is not available.</p><p>position       : Set value to 1 to include the start and end position in the protein.</p><p>* note: 'max_set' and 'return_values' cannot be used simultaneously.</p><p>Output:<br />By default, the plugin only returns data that is available. Example (default behaviour):<br />neXtProt_MatureProtein=Rho guanine nucleotide exchange factor 10</p><p>The option 'all_labels' returns a consistent set of the requested fields, using "-" where <br />values are not available. Same example as above:<br />neXtProt_MatureProtein=Rho guanine nucleotide exchange factor 10;<br />neXtProt_InteractingRegion=-;neXtProt_NucleotidePhosphateBindingRegion=-;neXtProt_Variant=-;<br />neXtProt_MiscellaneousRegion=-;neXtProt_TopologicalDomain=-;</p><p>Of notice, multiple values can be returned for the same label. In this case, the values will<br />be separeted by '|' for tab and txt format, and '&' for VCF format. </p><p>The plugin can then be run as default:<br /><pre class="code sh_sh">./vep -i variations.vcf --plugin neXtProt<br /></pre>or to return only the data specified by the user:<br /><pre class="code sh_sh">./vep -i variations.vcf --plugin neXtProt,return_values='Domain&InteractingRegion'<br /></pre></p></div></td><td><div class="vdoc_dtype_count" style="float:left;padding:2px 6px;cursor:default;background-color:#556622">Protein data</div></td><td><a href="https://metacpan.org/pod/JSON::XS" rel="external">JSON::XS</a></td><td>Ensembl</td></tr>
<tr class="bg2 plugin_row" data-category="transcript_annotation"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/NMD.pm">NMD</a></div></td><td><p>This is a plugin for the Ensembl Variant Effect Predictor (VEP) <br />that predicts if a variant allows the <br />transcript escape nonsense-mediated mRNA decay based on certain rules. <a class="button" href="#nmd" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('nmd');" id="a_nmd">more</a></p><div id="div_nmd" style="display:none;"><p>The rules are : </p><p>1. The variant location  falls in the last exon of the transcript.<br /><pre class="code sh_sh">                                        vvvv<br />      ES...EE..I.ES...EE.I.ES....EE.I.ES....EE <br /></pre>(ES= exon_start,EE = exon_end, I = intron, v = variant location)</p><p>2. The variant location falls  50 bases upstream of the penultimate (second to the last ) exon.<br /><pre class="code sh_sh">                               vvv<br />      ES...EE..I.ES...EE.I.ES....EE.I.ES....EE <br /></pre>(ES= exon_start,EE = exon_end, I = intron, v = variant location)</p><p>3. The variant falls in the first 100 coding bases in the transcript.<br /><pre class="code sh_sh">         vvv<br />      ..ES...EE..I.ES...EE.I.ES....EE.I.ES....EE <br /></pre>(ES= exon_start,EE = exon_end, I = intron, v = variant location)</p><p>4. If the variant is in an intronless transcript, meaning only one exon exist in the transcript. </p><p>The additional term NMD-escaping variant (nonsense-mediated mRNA decay escaping variants) will be added if the variant matches any of the rules.</p><p>REFERENCES :<br />Identifying Genes Whose Mutant Transcripts Cause Dominant Disease Traits by Potential Gain-of-Function Alleles (Coban-Akdemir, 2018)<br />The rules and impact of nonsense-mediated mRNA decay in human cancers (Lindeboom, 2016)</p></div></td><td><div class="vdoc_dtype_count" style="float:left;padding:2px 6px;cursor:default;background-color:#FF0000">Transcript annotation</div></td><td>-</td><td>Ensembl</td></tr>
<tr class="bg1 plugin_row" data-category="variant_data"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/OpenTargets.pm">OpenTargets</a></div></td><td><p>A VEP plugin that integrates data from Open Targets Genetics<br />(<a href="https://genetics.opentargets.org">https://genetics.opentargets.org</a>), a tool that highlights variant-centric<br />statistical evidence to allow both prioritisation of candidate causal variants<br />at trait-associated loci and identification of potential drug targets. <a class="button" href="#opentargets" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('opentargets');" id="a_opentargets">more</a></p><div id="div_opentargets" style="display:none;"><p>Data from Open Targets Genetics includes locus-to-gene (L2G) scores to predict<br />causal genes at GWAS loci.</p><p>Options are passed to the plugin as key=value pairs:<br />file : (mandatory) Tabix-indexed file from Open Targets Genetics<br />cols : (optional) Colon-separated list of columns to return from the plugin<br />file (default: "l2g:geneId"); use 'all' to print all data</p><p>Please cite the Open Targets Genetics publication alongside the VEP if you use<br />this resource: <a href="https://doi.org/10.1093/nar/gkaa84">https://doi.org/10.1093/nar/gkaa84</a></p><p>The tabix utility must be installed in your path to use this plugin.</p></div></td><td><div class="vdoc_dtype_count" style="float:left;padding:2px 6px;cursor:default;background-color:#B22222">Variant data</div></td><td><ul style="padding-left:1em"><li><a href="https://metacpan.org/pod/Bio::SeqUtils" rel="external">Bio::SeqUtils</a></li><li><a href="https://metacpan.org/pod/File::Basename" rel="external">File::Basename</a></li></ul></td><td>Ensembl</td></tr>
<tr class="bg2 plugin_row" data-category="phenotype_data_and_citations"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/Phenotypes.pm">Phenotypes</a></div></td><td><p>A VEP plugin that retrieves overlapping phenotype information. <a class="button" href="#phenotypes" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('phenotypes');" id="a_phenotypes">more</a></p><div id="div_phenotypes" style="display:none;"><p>On the first run for each new version/species/assembly will<br />download a GFF-format dump to ~/.vep/Plugins/</p><p>Ensembl provides phenotype annotations mapped to a number of genomic<br />feature types, including genes, variants and QTLs.</p><p>This plugin is best used with JSON output format; the output will be<br />more verbose and include all available phenotype annotation data and<br />metadata.</p><p>For other output formats, only a concatenated list of phenotype<br />description strings is returned.</p><p>Several paramters can be set using a key=value system:</p><p>dir            : provide a dir path, where either to create anew the species<br />specific file from the download or to look for an existing file</p><p>file           : provide a file path, either to create anew from the download<br />or to point to an existing file</p><p>exclude_sources: exclude sources of phenotype information. By default<br />HGMD and COSMIC annotations are excluded. See<br /><a href="http://www.ensembl.org/info/genome/variation/phenotype/sources_phenotype_documentation.html">http://www.ensembl.org/info/genome/variation/phenotype/sources_phenotype_documentation.html</a><br />Separate multiple values with '&'</p><p>include_sources: force include sources, as exclude_sources</p><p>exclude_types  : exclude types of features. By default StructuralVariation<br />and SupportingStructuralVariation annotations are excluded<br />due to their size. Separate multiple values with '&'.<br />Valid types: Gene, Variation, QTL, StructuralVariation,<br />SupportingStructuralVariation, RegulatoryFeature</p><p>include_types  : force include types, as exclude_types</p><p>expand_right   : sets cache size in bp. By default annotations 100000bp (100kb)<br />downstream of the initial lookup are cached</p><p>phenotype_feature : report the specific gene or variation the phenotype is<br />linked to, this can be an overlapping gene or structural variation,<br />and the source of the annotation (default 0)</p><p>Example:</p><p><pre class="code sh_sh">--plugin Phenotypes,file=${HOME}/phenotypes.gff.gz,include_types=Gene<br />--plugin Phenotypes,dir=${HOME},include_types=Gene<br /></pre></p></div></td><td><div class="vdoc_dtype_count" style="float:left;padding:2px 6px;cursor:default;background-color:#9966EE">Phenotype data and citations</div></td><td>-</td><td>Ensembl</td></tr>
<tr class="bg1 plugin_row" data-category="pathogenicity_predictions"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/pLI.pm">pLI</a></div></td><td><p>A VEP plugin that adds the probabililty of a gene being <br />loss-of-function intolerant (pLI) to the VEP output. <a class="button" href="#pli" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('pli');" id="a_pli">more</a></p><div id="div_pli" style="display:none;"><p>Lek et al. (2016) estimated pLI using the expectation-maximization <br />(EM) algorithm and data from 60,706 individuals from <br />ExAC (<a href="http://exac.broadinstitute.org">http://exac.broadinstitute.org</a>). The closer pLI is to 1, <br />the more likely the gene is loss-of-function (LoF) intolerant. </p><p>Note: the pLI was calculated using a representative transcript and<br />is reported by gene in the plugin.</p><p>The data for the plugin is provided by Kaitlin Samocha and Daniel MacArthur. <br />See <a href="https://www.ncbi.nlm.nih.gov/pubmed/27535533">https://www.ncbi.nlm.nih.gov/pubmed/27535533</a> for a description <br />of the dataset and analysis.</p><p>The pLI_values.txt file is found alongside the plugin in the <br />VEP_plugins GitHub repository. The file contains the fields gene and pLI <br />extracted from the file at </p><p><a href="https://ftp.broadinstitute.org/pub/ExAC_release/release0.3/functional_gene_constraint/">https://ftp.broadinstitute.org/pub/ExAC_release/release0.3/functional_gene_constraint/</a><br />fordist_cleaned_exac_r03_march16_z_pli_rec_null_data.txt</p><p>From this file, extract gene or transcipt pLI scores:<br />To extract gene scores : <br /><pre class="code sh_sh">awk '{print $2, $20 }'  fordist_cleaned_exac_r03_march16_z_pli_rec_null_data.txt > plI_gene.txt <br /></pre>NB: The gene scores file can also be found in the VEP_plugins directory.</p><p>To extract transcript scores:<br /><pre class="code sh_sh">awk '{print $1, $20 }'  fordist_cleaned_exac_r03_march16_z_pli_rec_null_data.txt > plI_transcript.txt <br /></pre>NB: Using this file, No transcript score will be returned.</p><p>To use another values file, add it as a parameter i.e.<br /><pre class="code sh_sh">./vep -i variants.vcf --plugin pLI,values_file.txt<br />./vep -i variants.vcf --plugin pLI,values_file.txt,transcript # to check for the transcript score.<br /></pre></p></div></td><td><div class="vdoc_dtype_count" style="float:left;padding:2px 6px;cursor:default;background-color:#1E90FF">Pathogenicity predictions</div></td><td><ul style="padding-left:1em"><li><a href="https://metacpan.org/pod/DBI" rel="external">DBI</a></li><li><a href="https://metacpan.org/pod/List::MoreUtils" rel="external">List::MoreUtils</a> qw/zip/</li></ul></td><td>Ensembl</td></tr>
<tr class="bg2 plugin_row" data-category="pathogenicity_predictions"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/PON_P2.pm">PON_P2</a></div></td><td><p>This plugin for Ensembl Variant Effect Predictor (VEP) computes the predictions of PON-P2<br />for amino acid substitutions in human proteins. PON-P2 is developed and maintained by<br />Protein Structure and Bioinformatics Group at Lund University and is available at <br /><a href="http://structure.bmc.lu.se/PON-P2/">http://structure.bmc.lu.se/PON-P2/</a>. <a class="button" href="#pon_p2" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('pon_p2');" id="a_pon_p2">more</a></p><div id="div_pon_p2" style="display:none;"><p>There are two ways to run the plugin:</p><p>1) To compute the predictions from the PON-P2 API please use the following options:</p><p>Option 1:       python script 'ponp2.py'*<br />Option 2:       the reference genome. Acceptable values are: hg37, hg38</p><p><pre class="code sh_sh">--plugin PON_P2,pyscript=/path/to/python/script/ponp2.py,hg=hg37<br /></pre>* To run this mode, you will require a python script and its dependencies (Python,<br />python suds). The python file can be downloaded from <a href="http://structure.bmc.lu.se/PON-P2/vep.html/">http://structure.bmc.lu.se/PON-P2/vep.html/</a><br />and the complete path to this file must be supplied while using this plugin.</p><p>2) To fetch the predictions from a file containing pre-calculated predictions for somatic variations <br />please use the following option (only available for GRCh37):</p><p>file:           COSMIC text file with pre-calculated predictions downloaded from<br /><a href="http://structure.bmc.lu.se/PON-P2/cancer30.html/">http://structure.bmc.lu.se/PON-P2/cancer30.html/</a></p><p>The following steps are necessary before using the file:<br />(head -n 1 COSMIC.txt && tail -n +2 COSMIC.txt | sort -t $'\t' -k1,1 -k2,2n) > cosmic_sorted.txt<br /><pre class="code sh_sh">sed -i 's/Chromosome/#Chromosome/' cosmic_sorted.txt<br />bgzip cosmic_sorted.txt<br />tabix -s 1 -b 2 -e 2 cosmic_sorted.txt.gz<br /></pre><pre class="code sh_sh">--plugin PON_P2,file=path/to/cosmic_sorted.txt.gz<br /></pre>If you use this data, please cite the following publication<br />Niroula, A., Vihinen, M. Harmful somatic amino acid substitutions affect key pathways in cancers.<br />BMC Med Genomics 8, 53 (2015). <a href="https://doi.org/10.1186/s12920-015-0125-x">https://doi.org/10.1186/s12920-015-0125-x</a></p></div></td><td><div class="vdoc_dtype_count" style="float:left;padding:2px 6px;cursor:default;background-color:#1E90FF">Pathogenicity predictions</div></td><td>-</td><td><ul style="padding-left:1em"><li>Abhishek Niroula</li><li>Mauno Vihinen</li></ul></td></tr>
<tr class="bg1 plugin_row" data-category="phenotype_data_and_citations"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/PostGAP.pm">PostGAP</a></div></td><td><p>A VEP plugin that retrieves data for variants from a tabix-indexed PostGAP file (1-based file). <a class="button" href="#postgap" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('postgap');" id="a_postgap">more</a></p><div id="div_postgap" style="display:none;"><p>Please refer to the PostGAP github and wiki for more information:<br /><a href="https://github.com/Ensembl/postgap">https://github.com/Ensembl/postgap</a><br /><a href="https://github.com/Ensembl/postgap/wiki">https://github.com/Ensembl/postgap/wiki</a><br /><a href="https://github.com/Ensembl/postgap/wiki/algorithm-pseudo-code">https://github.com/Ensembl/postgap/wiki/algorithm-pseudo-code</a></p><p>The Bio::DB::HTS perl library or tabix utility must be installed in your path<br />to use this plugin. The PostGAP data file can be downloaded from<br /><a href="https://storage.googleapis.com/postgap-data">https://storage.googleapis.com/postgap-data</a>.</p><p>The file must be processed and indexed by tabix before use by this plugin.<br />PostGAP has coordinates for both GRCh38 and GRCh37; the file must be<br />processed differently according to the assembly you use.</p><p><pre class="code sh_sh">wget <a href="https://storage.googleapis.com/postgap-data/postgap.txt.gz">https://storage.googleapis.com/postgap-data/postgap.txt.gz</a><br />gunzip postgap.txt.gz<br /></pre># GRCh38<br /><pre class="code sh_sh">(grep ^"ld_snp_rsID" postgap.txt; grep -v ^"ld_snp_rsID" postgap.txt | sort -k4,4 -k5,5n ) | bgzip > postgap_GRCh38.txt.gz<br />tabix -s 4 -b 5 -e 5 -c l postgap_GRCh38.txt.gz<br /></pre># GRCh37<br /><pre class="code sh_sh">(grep ^"ld_snp_rsID" postgap.txt; grep -v ^"ld_snp_rsID" postgap.txt | sort -k2,2 -k3,3n ) | bgzip > postgap_GRCh37.txt.gz<br />tabix -s 2 -b 3 -e 3 -c l postgap_GRCh37.txt.gz<br /></pre>Note that in the last command we tell tabix that the header line starts with "l";<br />this may change to the default of "#" in future versions of PostGAP.</p><p>When running the plugin by default 'disease_efo_id', 'disease_name', 'gene_id'<br />and 'score' information is returned e.g.</p><p><pre class="code sh_sh">--plugin POSTGAP,/path/to/PostGap.gz<br /></pre>You may include all columns with ALL; this fetches a large amount of data per<br />variant!:</p><p><pre class="code sh_sh">--plugin POSTGAP,/path/to/PostGap.gz,ALL<br /></pre>You may want to select only a specific subset of additional information to be<br />reported, you can do this by specifying the columns as parameters to the plugin e.g.</p><p><pre class="code sh_sh">--plugin POSTGAP,/path/to/PostGap.gz,gwas_pmid,gwas_size<br /></pre>If a requested column is not found, the error message will report the<br />complete list of available columns in the POSTGAP file. For a brief description<br />of the available information please refer to the 'How do I use POSTGAP output?'<br />section in the POSTGAP wiki.</p><p>Tabix also allows the data file to be hosted on a remote server. This plugin is<br />fully compatible with such a setup - simply use the URL of the remote file:</p><p><pre class="code sh_sh">--plugin PostGAP,<a href="http://my.files.com/postgap.txt.gz">http://my.files.com/postgap.txt.gz</a><br /></pre>Note that gene sequences referred to in PostGAP may be out of sync with<br />those in the latest release of Ensembl; this may lead to discrepancies with<br />scores retrieved from other sources.</p></div></td><td><div class="vdoc_dtype_count" style="float:left;padding:2px 6px;cursor:default;background-color:#9966EE">Phenotype data and citations</div></td><td>-</td><td>Ensembl</td></tr>
<tr class="bg2 plugin_row" data-category="pathogenicity_predictions"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/PrimateAI.pm">PrimateAI</a></div></td><td><p>The PrimateAI VEP plugin is designed to retrieve clinical impact scores of<br />variants, as described in <a href="https://www.nature.com/articles/s41588-018-0167-z">https://www.nature.com/articles/s41588-018-0167-z</a>.<br />Please consider citing the paper if using this plugin. <a class="button" href="#primateai" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('primateai');" id="a_primateai">more</a></p><div id="div_primateai" style="display:none;"><p>In brief, common missense mutations in non-human primate species are usually<br />benign in humans. Thousands of common variants from six non-human primate<br />species were used to train a deep neural network to identify pathogenic <br />mutations in humans with a rare disease.</p><p>This plugin uses files generated by the PrimateAI software, which is<br />available from <a href="https://github.com/Illumina/PrimateAI">https://github.com/Illumina/PrimateAI</a>. The files containing<br />predicted pathogenicity scores can be downloaded from<br /><a href="https://basespace.illumina.com/s/yYGFdGih1rXL">https://basespace.illumina.com/s/yYGFdGih1rXL</a> (a free BaseSpace account may<br />be required):<br />PrimateAI_scores_v0.2.tsv.gz (for GRCh37/hg19)<br />PrimateAI_scores_v0.2_hg38.tsv.gz (for GRCh38/hg38)</p><p>Before running the plugin for the first time, the following steps must be <br />taken to format the downloaded files:</p><p>1.  Unzip the score files<br />2.  Add '#' in front of the column description line<br />3.  Remove any empty lines.<br />4.  Sort the file by chromosome and position<br />5.  Compress the file in .bgz format<br />6.  Create tabix index (requires tabix to be installed).</p><p>Command line examples for formatting input files:<br /><pre class="code sh_sh">gunzip -cf PrimateAI_scores_v0.2.tsv.gz | sed '12s/.*/#&/' | sed '/^$/d' | awk 'NR<12{print $0;next}{print $0 | "sort -k1,1 -k 2,2n -V"}' | bgzip > PrimateAI_scores_v0.2_GRCh37_sorted.tsv.bgz<br />tabix -s 1 -b 2 -e 2 PrimateAI_scores_v0.2_GRCh37_sorted.tsv.bgz<br /></pre><pre class="code sh_sh">gunzip -cf PrimateAI_scores_v0.2_hg38.tsv.gz | sed '12s/.*/#&/' | sed '/^$/d' | awk 'NR<12{print $0;next}{print $0 | "sort -k1,1 -k 2,2n -V"}' | bgzip > PrimateAI_scores_v0.2_GRCh38_sorted.tsv.bgz<br />tabix -s 1 -b 2 -e 2 PrimateAI_scores_v0.2_GRCh38_sorted.tsv.bgz<br /></pre></p></div></td><td><div class="vdoc_dtype_count" style="float:left;padding:2px 6px;cursor:default;background-color:#1E90FF">Pathogenicity predictions</div></td><td>-</td><td>Ensembl</td></tr>
<tr class="bg1 plugin_row" data-category="sequence"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/ProteinSeqs.pm">ProteinSeqs</a></div></td><td><p>This is a plugin for the Ensembl Variant Effect Predictor (VEP) that<br />prints out the reference and mutated protein sequences of any<br />proteins found with non-synonymous mutations in the input file.  <a class="button" href="#proteinseqs" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('proteinseqs');" id="a_proteinseqs">more</a></p><div id="div_proteinseqs" style="display:none;"><p>You should supply the name of file where you want to store the<br />reference protein sequences as the first argument, and a file to<br />store the mutated sequences as the second argument.</p><p>Note that, for simplicity, where stop codons are gained the plugin <br />simply substitutes a '*' into the sequence and does not truncate the <br />protein. Where a stop codon is lost any new amino acids encoded by the <br />mutation are appended to the sequence, but the plugin does not attempt <br />to translate until the next downstream stop codon. Also, the protein<br />sequence resulting from each mutation is printed separately, no attempt<br />is made to apply multiple mutations to the same protein.</p></div></td><td><div class="vdoc_dtype_count" style="float:left;padding:2px 6px;cursor:default;background-color:#5F81A9">Sequence</div></td><td>-</td><td>Ensembl</td></tr>
<tr class="bg2 plugin_row" data-category="sequence"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/ReferenceQuality.pm">ReferenceQuality</a></div></td><td><p>This is a plugin for the Ensembl Variant Effect Predictor (VEP) that<br />reports on the quality of the reference genome using GRC data at the location of your variants.<br />More information can be found at: <a href="https://www.ncbi.nlm.nih.gov/grc/human/issues">https://www.ncbi.nlm.nih.gov/grc/human/issues</a> <a class="button" href="#referencequality" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('referencequality');" id="a_referencequality">more</a></p><div id="div_referencequality" style="display:none;"><p>The following steps are necessary before running this plugin:</p><p>GRCh38:</p><p><pre class="code sh_sh">wget <a href="https://ftp.ncbi.nlm.nih.gov/pub/grc/human/GRC/GRCh38/MISC/annotated_clone_assembly_problems_GCF_000001405.38.gff3">https://ftp.ncbi.nlm.nih.gov/pub/grc/human/GRC/GRCh38/MISC/annotated_clone_assembly_problems_GCF_000001405.38.gff3</a><br />wget <a href="https://ftp.ncbi.nlm.nih.gov/pub/grc/human/GRC/Issue_Mapping/GRCh38.p12_issues.gff3">https://ftp.ncbi.nlm.nih.gov/pub/grc/human/GRC/Issue_Mapping/GRCh38.p12_issues.gff3</a><br />cat annotated_clone_assembly_problems_GCF_000001405.38.gff3 GRCh38.p12_issues.gff3 > GRCh38_quality_mergedfile.gff3<br />sort -k 1,1 -k 4,4n -k 5,5n GRCh38_quality_mergedfile.gff3 > sorted_GRCh38_quality_mergedfile.gff3<br />bgzip sorted_GRCh38_quality_mergedfile.gff3<br />tabix -p gff sorted_GRCh38_quality_mergedfile.gff3.gz<br /></pre>The plugin can then be run with:<br /><pre class="code sh_sh">./vep -i variations.vcf --plugin ReferenceQuality,sorted_GRCh38_quality_mergedfile.gff3.gz<br /></pre>GRCh37:</p><p><pre class="code sh_sh">wget <a href="https://ftp.ncbi.nlm.nih.gov/pub/grc/human/GRC/GRCh37/MISC/annotated_clone_assembly_problems_GCF_000001405.25.gff3">https://ftp.ncbi.nlm.nih.gov/pub/grc/human/GRC/GRCh37/MISC/annotated_clone_assembly_problems_GCF_000001405.25.gff3</a><br />wget <a href="https://ftp.ncbi.nlm.nih.gov/pub/grc/human/GRC/Issue_Mapping/GRCh37.p13_issues.gff3">https://ftp.ncbi.nlm.nih.gov/pub/grc/human/GRC/Issue_Mapping/GRCh37.p13_issues.gff3</a><br />cat annotated_clone_assembly_problems_GCF_000001405.25.gff3 GRCh37.p13_issues.gff3 > GRCh37_quality_mergedfile.gff3<br />sort -k 1,1 -k 4,4n -k 5,5n GRCh37_quality_mergedfile.gff3 > sorted_GRCh37_quality_mergedfile.gff3<br />bgzip sorted_GRCh37_quality_mergedfile.gff3<br />tabix -p gff sorted_GRCh37_quality_mergedfile.gff3.gz<br /></pre>The plugin can then be run with:<br /><pre class="code sh_sh">./vep -i variations.vcf --plugin ReferenceQuality,sorted_GRCh37_quality_mergedfile.gff3.gz<br /></pre> </p><p>The tabix utility must be installed in your path to use this plugin.</p></div></td><td><div class="vdoc_dtype_count" style="float:left;padding:2px 6px;cursor:default;background-color:#5F81A9">Sequence</div></td><td>-</td><td>Ensembl</td></tr>
<tr class="bg1 plugin_row" data-category="pathogenicity_predictions"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/REVEL.pm">REVEL</a></div></td><td><p>This is a plugin for the Ensembl Variant Effect Predictor (VEP) that<br />adds the REVEL score for missense variants to VEP output. <a class="button" href="#revel" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('revel');" id="a_revel">more</a></p><div id="div_revel" style="display:none;"><p>Please cite the REVEL publication alongside the VEP if you use this resource:<br /><a href="https://www.ncbi.nlm.nih.gov/pubmed/27666373">https://www.ncbi.nlm.nih.gov/pubmed/27666373</a></p><p>Running options:<br />If available, the plugin will match the scores by transcript id (default).<br />Using the flag '1' the plugin will not try to match by transcript id.</p><p>REVEL scores can be downloaded from: <a href="https://sites.google.com/site/revelgenomics/downloads">https://sites.google.com/site/revelgenomics/downloads</a></p><p>The plugin supports several REVEL file versions:<br />- REVEL file version Dec 2017, which has 7 columns and only GRCh37 coordinates<br />- REVEL file version Feb 2020, which has 8 columns with GRCh37 and GRCh38 coordinates<br />- REVEL file version May 2021, which has 9 columns with GRCh37 and GRCh38 coordinates and a new column with transcript ids</p><p>These files can be tabix-processed by:<br /><pre class="code sh_sh">unzip revel-v1.3_all_chromosomes.zip<br />cat revel_with_transcript_ids | tr "," "\t" > tabbed_revel.tsv<br />sed '1s/.*/#&/' tabbed_revel.tsv > new_tabbed_revel.tsv<br />bgzip new_tabbed_revel.tsv<br /></pre>for GRCh37:<br /><pre class="code sh_sh">tabix -f -s 1 -b 2 -e 2 new_tabbed_revel.tsv.gz<br /></pre>for GRCh38:<br /><pre class="code sh_sh">zcat new_tabbed_revel.tsv.gz | head -n1 > h<br />zgrep -h -v ^#chr new_tabbed_revel.tsv.gz | awk '$3 != "." ' | sort -k1,1 -k3,3n - | cat h - | bgzip -c > new_tabbed_revel_grch38.tsv.gz<br />tabix -f -s 1 -b 3 -e 3 new_tabbed_revel_grch38.tsv.gz<br /></pre>The plugin can then be run as default:<br /><pre class="code sh_sh">./vep -i variations.vcf --assembly GRCh38 --plugin REVEL,file=/path/to/revel/data.tsv.gz<br /></pre>or with the option to not match by transcript id:<br /><pre class="code sh_sh">./vep -i variations.vcf --assembly GRCh38 --plugin REVEL,file=/path/to/revel/data.tsv.gz,no_match=1<br /></pre>Requirements:<br />The tabix utility must be installed in your path to use this plugin.<br />The --assembly flag is required to use this plugin.</p></div></td><td><div class="vdoc_dtype_count" style="float:left;padding:2px 6px;cursor:default;background-color:#1E90FF">Pathogenicity predictions</div></td><td>-</td><td>Ensembl</td></tr>
<tr class="bg2 plugin_row" data-category="variant_data"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/SameCodon.pm">SameCodon</a></div></td><td><p>A VEP plugin that reports existing variants that fall in the same codon.<br />This plugin requires a database connection, can not be run in offline mode</p></td><td><div class="vdoc_dtype_count" style="float:left;padding:2px 6px;cursor:default;background-color:#B22222">Variant data</div></td><td>-</td><td>Ensembl</td></tr>
<tr class="bg1 plugin_row" data-category="phenotype_data_and_citations"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/satMutMPRA.pm">satMutMPRA</a></div></td><td><p>A VEP plugin that retrieves data for variants from a tabix-indexed satMutMPRA file (1-based file).<br />The saturation mutagenesis-based massively parallel reporter assays (satMutMPRA) measures variant<br />effects on gene RNA expression for 21 regulatory elements (11 enhancers, 10 promoters). <a class="button" href="#satmutmpra" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('satmutmpra');" id="a_satmutmpra">more</a></p><div id="div_satmutmpra" style="display:none;"><p>The 20 disease-associated regulatory elements and one ultraconserved enhancer analysed in different cell lines are the following:<br />- ten promoters (of TERT, LDLR, HBB, HBG, HNF4A, MSMB, PKLR, F9, FOXE1 and GP1BB) and<br />- ten enhancers (of SORT1, ZRS, BCL11A, IRF4, IRF6, MYC (2x), RET, TCF7L2 and ZFAND3) and<br />- one ultraconserved enhancer (UC88).</p><p>Please refer to the satMutMPRA web server and Kircher M et al. (2019) paper for more information:<br /><a href="https://mpra.gs.washington.edu/satMutMPRA/">https://mpra.gs.washington.edu/satMutMPRA/</a><br /><a href="https://www.ncbi.nlm.nih.gov/pubmed/31395865">https://www.ncbi.nlm.nih.gov/pubmed/31395865</a></p><p>Parameters can be set using a key=value system:<br />file           : required - a tabix indexed file of the satMutMPRA data corresponding to desired assembly.</p><p>pvalue         : p-value threshold  (default: 0.00001)</p><p>cols           : colon delimited list of data types to be returned from the satMutMPRA data<br />(default: 'Value', 'P-Value', and 'Element')</p><p>incl_repl       : include replicates (default: off):<br />- full replicate for LDLR promoter (LDLR.2) and SORT1 enhancer (SORT1.2)<br />- a reversed sequence orientation for SORT1 (SORT1-flip)<br />- other conditions: PKLR-48h, ZRSh-13h2, TERT-GAa, TERT-GBM, TERG-GSc</p><p>The Bio::DB::HTS perl library or tabix utility must be installed in your path<br />to use this plugin. The satMutMPRA data file can be downloaded from<br /><a href="https://mpra.gs.washington.edu/satMutMPRA/">https://mpra.gs.washington.edu/satMutMPRA/</a></p><p>satMutMPRA data can be downloaded for both GRCh38 and GRCh37 from the web server (<a href="https://mpra.gs.washington.edu/satMutMPRA/">https://mpra.gs.washington.edu/satMutMPRA/</a>):<br />'Download' section, select 'GRCh37' or 'GRCh38' for 'Genome release' and 'Download All Elements'.</p><p>The file must be processed and indexed by tabix before use by this plugin.</p><p># GRCh38<br /><pre class="code sh_sh">(grep ^Chr GRCh38_ALL.tsv; grep -v ^Chr GRCh38_ALL.tsv | sort -k1,1 -k2,2n ) | bgzip > satMutMPRA_GRCh38_ALL.gz<br />tabix -s 1 -b 2 -e 2 -c C satMutMPRA_GRCh38_ALL.gz<br /></pre># GRCh37<br /><pre class="code sh_sh">(grep ^Chr GRCh37_ALL.tsv; grep -v ^Chr GRCh37_ALL.tsv | sort -k1,1 -k2,2n ) | bgzip > satMutMPRA_GRCh37_ALL.gz<br />tabix -s 1 -b 2 -e 2 -c C satMutMPRA_GRCh37_ALL.gz<br /></pre><br />When running the plugin by default 'Value', 'P-Value', and 'Element'<br />information is returned e.g.</p><p><pre class="code sh_sh">--plugin satMutMPRA,file=/path/to/satMutMPRA_GRCh38_ALL.gz<br /></pre>You may include all columns with ALL; this fetches all data per variant<br />(e.g. Tags, DNA, RNA, Value, P-Value, Element):</p><p><pre class="code sh_sh">--plugin satMutMPRA,file=/path/to/satMutMPRA_GRCh38_ALL.gz,cols=ALL<br /></pre>You may want to select only a specific subset of information to be<br />reported, you can do this by specifying the specific columns as parameters to the plugin e.g.</p><p><pre class="code sh_sh">--plugin satMutMPRA,file=/path/to/satMutMPRA_GRCh38_ALL.gz,cols=Tags:DNA<br /></pre>If a requested column is not found, the error message will report the<br />complete list of available columns in the satMutMPRA file. For a detailed description<br />of the available information please refer to the manuscript or online web server.</p><p>Tabix also allows the data file to be hosted on a remote server. This plugin is<br />fully compatible with such a setup - simply use the URL of the remote file:</p><p><pre class="code sh_sh">--plugin satMutMPRA,file=<a href="http://my.files.com/satMutMPRA.gz">http://my.files.com/satMutMPRA.gz</a><br /></pre>Note that gene locations referred to in satMutMPRA may be out of sync with<br />those in the latest release of Ensembl; this may lead to discrepancies with<br />information retrieved from other sources.</p></div></td><td><div class="vdoc_dtype_count" style="float:left;padding:2px 6px;cursor:default;background-color:#9966EE">Phenotype data and citations</div></td><td>-</td><td>Ensembl</td></tr>
<tr class="bg2 plugin_row" data-category="hgvs"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/SingleLetterAA.pm">SingleLetterAA</a></div></td><td><p>This is a plugin for the Ensembl Variant Effect Predictor (VEP) that<br />returns a HGVSp string with single amino acid letter codes</p></td><td><div class="vdoc_dtype_count" style="float:left;padding:2px 6px;cursor:default;background-color:#9400D3">HGVS</div></td><td>-</td><td>Ensembl</td></tr>
<tr class="bg1 plugin_row" data-category="splicing_predictions"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/SpliceAI.pm">SpliceAI</a></div></td><td><p>A VEP plugin that retrieves pre-calculated annotations from SpliceAI.<br />SpliceAI is a deep neural network, developed by Illumina, Inc <br />that predicts splice junctions from an arbitrary pre-mRNA transcript sequence. <a class="button" href="#spliceai" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('spliceai');" id="a_spliceai">more</a></p><div id="div_spliceai" style="display:none;"><p>Delta score of a variant, defined as the maximum of (DS_AG, DS_AL, DS_DG, DS_DL), <br />ranges from 0 to 1 and can be interpreted as the probability of the variant being <br />splice-altering. The author-suggested cutoffs are:<br />* 0.2 (high recall)<br />* 0.5 (recommended)<br />* 0.8 (high precision)</p><p>This plugin is available for both GRCh37 and GRCh38.</p><p>More information can be found at:<br /><a href="https://pypi.org/project/spliceai/">https://pypi.org/project/spliceai/</a></p><p>Please cite the SpliceAI publication alongside VEP if you use this resource:<br /><a href="https://www.ncbi.nlm.nih.gov/pubmed/30661751">https://www.ncbi.nlm.nih.gov/pubmed/30661751</a></p><p>Running options:<br />(Option 1) By default, this plugin appends all scores from SpliceAI files.<br />(Option 2) Besides the pre-calculated scores, it can also be specified a score<br />cutoff between 0 and 1.</p><p>Output: <br />The output includes the gene symbol, delta scores (DS) and delta positions (DP)<br />for acceptor gain (AG), acceptor loss (AL), donor gain (DG), and donor loss (DL).</p><p>For tab the output contains one header 'SpliceAI_pred' with all<br />the delta scores and positions. The format is:<br />SYMBOL|DS_AG|DS_AL|DS_DG|DS_DL|DP_AG|DP_AL|DP_DG|DP_DL</p><p>For JSON the output is a hash with the following format:<br />"spliceai":<br />{"DP_DL":0,"DS_AL":0,"DP_AG":0,"DS_DL":0,"SYMBOL":"X","DS_AG":0,"DP_AL":0,"DP_DG":0,"DS_DG":0}</p><p>For VCF output the delta scores and positions are stored in different headers.<br />The values are 'SpliceAI_pred_xx' being 'xx' the score/position.<br />Example: 'SpliceAI_pred_DS_AG' is the delta score for acceptor gain.</p><p>Gene matching:<br />SpliceAI can contain scores for multiple genes that overlap a variant,<br />and VEP can also predict consequences on multiple genes for a given variant.<br />The plugin only returns SpliceAI scores for the gene symbols that match (if any).</p><p>If plugin is run with option 2, the output also contains a flag: 'PASS' if delta score<br />passes the cutoff, 'FAIL' otherwise. </p><p>The following steps are necessary before running this plugin:</p><p>The files with the annotations for all possible substitutions (snv), 1 base insertions <br />and 1-4 base deletions (indel) within genes are available here:<br /><a href="https://basespace.illumina.com/s/otSPW8hnhaZR">https://basespace.illumina.com/s/otSPW8hnhaZR</a></p><p>GRCh37:<br /><pre class="code sh_sh">tabix -p vcf spliceai_scores.raw.snv.hg37.vcf.gz<br />tabix -p vcf spliceai_scores.raw.indel.hg37.vcf.gz<br /></pre>GRCh38:<br /><pre class="code sh_sh">tabix -p vcf spliceai_scores.raw.snv.hg38.vcf.gz<br />tabix -p vcf spliceai_scores.raw.indel.hg38.vcf.gz<br /></pre>The plugin can then be run:<br /><pre class="code sh_sh">./vep -i variations.vcf --plugin SpliceAI,snv=/path/to/spliceai_scores.raw.snv.hg38.vcf.gz,indel=/path/to/spliceai_scores.raw.indel.hg38.vcf.gz<br />./vep -i variations.vcf --plugin SpliceAI,snv=/path/to/spliceai_scores.raw.snv.hg38.vcf.gz,indel=/path/to/spliceai_scores.raw.indel.hg38.vcf.gz,cutoff=0.5<br /></pre></p></div></td><td><div class="vdoc_dtype_count" style="float:left;padding:2px 6px;cursor:default;background-color:#FF0000">Splicing predictions</div></td><td><a href="https://metacpan.org/pod/List::Util" rel="external">List::Util</a> qw(max)</td><td>Ensembl</td></tr>
<tr class="bg2 plugin_row" data-category="splicing_predictions"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/SpliceRegion.pm">SpliceRegion</a></div></td><td><p>This is a plugin for the Ensembl Variant Effect Predictor (VEP) that<br />provides more granular predictions of splicing effects. <a class="button" href="#spliceregion" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('spliceregion');" id="a_spliceregion">more</a></p><div id="div_spliceregion" style="display:none;"><p>Three additional terms may be added:</p><p># splice_donor_5th_base_variant : variant falls in the 5th base after the splice donor junction (5' end of intron)</p><p><pre class="code sh_sh">             v<br /> ...EEEEEIIIIIIIIII...<br /></pre><br />(E = exon, I = intron, v = variant location)</p><p># splice_donor_region_variant : variant falls in region between 3rd and 6th base after splice junction (5' end of intron)</p><p><pre class="code sh_sh">           vv vvv<br /> ...EEEEEIIIIIIIIII...<br /></pre><br /># splice_polypyrimidine_tract_variant : variant falls in polypyrimidine tract at 3' end of intron, between 17 and 3 bases from the end</p><p><pre class="code sh_sh">      vvvvvvvvvvvvvvv<br /> ...IIIIIIIIIIIIIIIIIIIIEEEEE...<br /></pre></p></div></td><td><div class="vdoc_dtype_count" style="float:left;padding:2px 6px;cursor:default;background-color:#FF0000">Splicing predictions</div></td><td>-</td><td>Ensembl</td></tr>
<tr class="bg1 plugin_row" data-category="structural_variant_data"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/StructuralVariantOverlap.pm">StructuralVariantOverlap</a></div></td><td><p>A VEP plugin that retrieves information from overlapping structural variants. <a class="button" href="#structuralvariantoverlap" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('structuralvariantoverlap');" id="a_structuralvariantoverlap">more</a></p><div id="div_structuralvariantoverlap" style="display:none;"><p>Parameters can be set using a key=value system:</p><p>file           : required - a VCF file of reference data. </p><p>percentage     : percentage overlap between SVs (default: 80)<br />reciprocal     : calculate reciprocal overlap, options:  0 or 1. (default: 0)<br />(overlap is expressed as % of input SV by default)</p><p>cols           : colon delimited list of data types to return from the INFO fields (only AF by default)</p><p>same_type      : 1/0 only report SV of the same type (eg deletions for deletions, off by default)</p><p>distance       : the distance the ends of the overlapping SVs should be within.</p><p>match_type     : only report reference SV which lie within or completely surround the input SV<br />options: within, surrounding</p><p>label          : annotation label that will appear in the output (default: "SV_overlap")<br />Example- input: label=mydata, output: mydata_name=refSV,mydata_PC=80,mydata_AF=0.05</p><p>Example reference data</p><p>1000 Genomes Project:<br /><a href="https://ftp.1000genomes.ebi.ac.uk/vol1/ftp/phase3/integrated_sv_map/ALL.wgs.mergedSV.v8.20130502.svs.genotypes.vcf.gz">https://ftp.1000genomes.ebi.ac.uk/vol1/ftp/phase3/integrated_sv_map/ALL.wgs.mergedSV.v8.20130502.svs.genotypes.vcf.gz</a></p><p>gnomAD:<br /><a href="https://storage.googleapis.com/gcp-public-data--gnomad/papers/2019-sv/gnomad_v2.1_sv.sites.vcf.gz">https://storage.googleapis.com/gcp-public-data--gnomad/papers/2019-sv/gnomad_v2.1_sv.sites.vcf.gz</a></p><p>Example:</p><p><pre class="code sh_sh">./vep -i structvariants.vcf --plugin StructuralVariantOverlap,file=gnomad_v2_sv.sites.vcf.gz<br /></pre></p></div></td><td><div class="vdoc_dtype_count" style="float:left;padding:2px 6px;cursor:default;background-color:#601212">Structural variant data</div></td><td>-</td><td>Ensembl</td></tr>
<tr class="bg2 plugin_row" data-category="variant_data"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/SubsetVCF.pm">SubsetVCF</a></div></td><td><p>A VEP plugin to retrieve overlapping records from a given VCF file.<br />Values for POS, ID, and ALT, are retrieved as well as values for any requested<br />INFO field. Additionally, the allele number of the matching ALT is returned. <a class="button" href="#subsetvcf" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('subsetvcf');" id="a_subsetvcf">more</a></p><div id="div_subsetvcf" style="display:none;"><p>Though similar to using '--custom', this plugin returns all ALTs for a given<br />POS, as well as all associated INFO values.</p><p>By default, only VCF records with a filter value of "PASS" are returned, <br />however this behaviour can be changed via the 'filter' option.</p><p>Parameters:<br />name: short name added used as a prefix (required)<br />file: path to tabix-index vcf file (required)<br />filter: only consider variants marked as 'PASS', 1 or 0 (default, 1)<br />fields: info fields to be returned (default, not used)<br />'%' can delimit multiple fields<br />'*' can be used as a wildcard </p><p>Returns:<br /><name>_POS: POS field from VCF<br /><name>_REF: REF field from VCF (minimised)<br /><name>_ALT: ALT field from VCF (minimised)<br /><name>_alt_index: Index of matching variant (zero-based)<br /><name>_<field>: List of requested info values</p></div></td><td><div class="vdoc_dtype_count" style="float:left;padding:2px 6px;cursor:default;background-color:#B22222">Variant data</div></td><td><ul style="padding-left:1em"><li><a href="https://metacpan.org/pod/Data::Dumper" rel="external">Data::Dumper</a></li><li><a href="https://metacpan.org/pod/Storable" rel="external">Storable</a> qw(dclone)</li></ul></td><td>Joseph A. Prinz</td></tr>
<tr class="bg1 plugin_row" data-category="transcript_annotation"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/TranscriptAnnotator.pm">TranscriptAnnotator</a></div></td><td><p>A VEP plugin that annotates variant-transcript pairs based on a given file: <a class="button" href="#transcriptannotator" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('transcriptannotator');" id="a_transcriptannotator">more</a></p><div id="div_transcriptannotator" style="display:none;"><p><pre class="code sh_sh">--plugin TranscriptAnnotator,file=${HOME}/file.tsv.gz<br /></pre>Example of a valid tab-separated annotation file:</p><p>#Chrom  Pos       Ref  Alt  Transcript       SIFT_score  SIFT_pred    Comment<br />11      436154    A    G    NM_001347882.2   0.03        Deleterious  Bad<br />11      1887471   C	  T    ENST00000421485  0.86        Tolerated    Good</p><p>Please bgzip and tabix the file with commands such as:</p><p><pre class="code sh_sh">bgzip file.txt<br />tabix -b2 -e2 file.txt.gz<br /></pre>Options are passed to the plugin as key=value pairs:</p><p>file:   (mandatory) Tabix-indexed file to parse. Must contain variant<br />location (chromosome, position, reference allele, alternative allele)<br />and transcript ID as the first 5 columns. Accepted transcript IDs<br />include those from Ensembl and RefSeq.<br />cols:   Colon-delimited list with names of the columns to append. Column<br />names are based on the last header line. By default, all columns<br />(except the first 5) are appended.<br />prefix: String to prefix the name of appended columns (default: basename of<br />the filename without extensions). Set to 0 to avoid any prefix.<br />trim:   Trim whitespaces from both ends of each column (default: 1).</p><p>The tabix and bgzip utilities must be installed in your path to read the<br />tabix-indexed annotation file: check <a href="https://github.com/samtools/htslib.git">https://github.com/samtools/htslib.git</a> for<br />installation instructions.</p></div></td><td><div class="vdoc_dtype_count" style="float:left;padding:2px 6px;cursor:default;background-color:#FF0000">Transcript annotation</div></td><td><a href="https://metacpan.org/pod/File::Basename" rel="external">File::Basename</a></td><td>Ensembl</td></tr>
<tr class="bg2 plugin_row" data-category="nearby_features"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/TSSDistance.pm">TSSDistance</a></div></td><td><p>A VEP plugin that calculates the distance from the transcription<br />start site for upstream variants.</p></td><td><div class="vdoc_dtype_count" style="float:left;padding:2px 6px;cursor:default;background-color:#E75480">Nearby features</div></td><td>-</td><td>Ensembl</td></tr>
<tr class="bg1 plugin_row" data-category="transcript_annotation"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/UTRAnnotator.pm">UTRAnnotator</a></div></td><td><p>A VEP plugin that annotates the effect of 5' UTR variant especially for variant creating/disrupting upstream ORFs.<br />Available for both GRCh37 and GRCh38. <a class="button" href="#utrannotator" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('utrannotator');" id="a_utrannotator">more</a></p><div id="div_utrannotator" style="display:none;"><p>Options are passed to the plugin as key=value pairs, (defaults in parentheses):</p><p>file                : Path to UTRAnnotator data file.<br />- Download from <a href="https://github.com/Ensembl/UTRannotator">https://github.com/Ensembl/UTRannotator</a><br />- Download from <a href="http://www.sorfs.org/">http://www.sorfs.org/</a></p><p>Citation</p><p>About the role of 5'UTR variants in human genetic disease:</p><p>Whiffin, N., Karczewski, K.J., Zhang, X. et al. Characterising the loss-of-function impact of 5’ untranslated region variants in 15,708 individuals. Nat Commun 11, 2523 (2020). <a href="https://doi.org/10.1038/s41467-019-10717-9">https://doi.org/10.1038/s41467-019-10717-9</a></p><p>About UTRAnnotator:</p><p>Annotating high-impact 5'untranslated region variants with the UTRannotator Zhang, X., Wakeling, M.N., Ware, J.S, Whiffin, N. Bioinformatics; doi: <a href="https://academic.oup.com/bioinformatics/advance-article/doi/10.1093/bioinformatics/btaa783/5905476">https://academic.oup.com/bioinformatics/advance-article/doi/10.1093/bioinformatics/btaa783/5905476</a></p></div></td><td><div class="vdoc_dtype_count" style="float:left;padding:2px 6px;cursor:default;background-color:#FF0000">Transcript annotation</div></td><td><a href="https://metacpan.org/pod/Scalar::Util" rel="external">Scalar::Util</a> qw(looks_like_number)</td><td>Ensembl</td></tr>
<tr class="bg2 plugin_row" data-category="pathogenicity_predictions"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/VARITY.pm">VARITY</a></div></td><td><p>This is a plugin for the Ensembl Variant Effect Predictor (VEP) that<br />adds the pre-computed VARITY scores to predict pathogenicity of <br />rare missense variants to VEP output. <a class="button" href="#varity" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('varity');" id="a_varity">more</a></p><div id="div_varity" style="display:none;"><p>Please cite the VARITY publication alongside the VEP if you use this resource:<br /><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8715197/">https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8715197/</a></p><p>Running options :</p><p>VARITY scores can be downloaded using wget <a href="http://varity.varianteffect.org/downloads/varity_all_predictions.tar.gz">http://varity.varianteffect.org/downloads/varity_all_predictions.tar.gz</a></p><p>The files can be tabix processed by :<br /><pre class="code sh_sh">tar -xzvf varity_all_predictions.tar.gz <br />cat varity_all_predictions.txt | (head -n 1 && tail -n +2  | sort -t$'\t' -k 1,1 -k 2,2n) > varity_all_predictions_sorted.tsv<br />sed '1s/.*/#&/'  varity_all_predictions_sorted.tsv > varity_all_predictions.tsv  # to add a # in the first line of the file <br />bgzip varity_all_predictions.tsv<br />tabix -f -s 1 -b 2 -e 2 varity_all_predictions.tsv.gz<br /></pre><br />Requirements:<br />The tabix utility must be installed in your path to use this plugin.<br />The --assembly flag is required to use this plugin.</p></div></td><td><div class="vdoc_dtype_count" style="float:left;padding:2px 6px;cursor:default;background-color:#1E90FF">Pathogenicity predictions</div></td><td>-</td><td>Ensembl</td></tr>
</tbody></table><!-- Plugins list - end -->
  
  <p> We hope that these will serve as useful examples for users implementing
	new plugins. If you have any questions about the system, or suggestions for
	enhancements please let us know on the 
  <a href="//lists.ensembl.org/mailman/listinfo/dev" rel="external">ensembl-dev</a> 
  mailing list.<br />
  We also encourage you to share any plugins you develop: we are happy to accept pull requests on the <a href="//github.com/Ensembl/VEP_plugins" rel="external">VEP_plugins</a> git repository. </p>
    

  <p> There are further published plugins available outside the VEP repository including:
  <li> <a href="https://github.com/konradjk/loftee">LOFTEE </a> a Loss-Of-Function Transcript Effect Estimator (<a href="https://www.nature.com/articles/s41586-020-2308-7">Konrad Karczewski et al,2020</a>)


	
  <hr/>
  <h2 id="plugins_how">How it works</h2>
	
	<p>Plugins are run once VEP has finished its analysis for each line of
	the output, but before anything is printed to the output file.<br />When each
	plugin is called (using the <i>run</i> method) it is passed two data structures
	to use in its analysis; the first is a data structure containing all the
	data for the current line, and the second is a reference to a variation
	API object that represents the combination of a variant allele and an
	overlapping or nearby genomic feature (such as a transcript or regulatory
	region).<br />This object provides access to all the relevant API objects that
	may be useful for further analysis by the plugin (such as the current
	VariationFeature and Transcript).<br />Please refer to the <a
	href="/info/docs/Doxygen/variation-api/index.html"><img style="vertical-align:bottom" src="/i/16/documentation.png"/> Ensembl Variation API
	documentation</a> for more details.</p>
	
  <hr/>
  <h2 id="plugins_functionality">Functionality</h2>
	
	<p>We expect that most plugins will simply add information to the last
	column of the output file, the "Extra" column, and the plugin system assumes
	this in various places, but plugins are also free to alter the output line
	as desired.</p>
	
	<p>The only hard requirement for
	a plugin to work with VEP is that it implements a number of required
	methods (such as <i>new</i> which should create and return an instance of this
	plugin, <i>get_header_info</i> which should return descriptions of the type of
	data this plugin produces to be included in VEP output's header, and
	<i>run</i> which should actually perform the logic of the plugin).<br />To make
	development of plugins easier, we suggest that users use the <a
	href="/info/docs/Doxygen/variation-api/classBio_1_1EnsEMBL_1_1Variation_1_1Utils_1_1BaseVepPlugin.html">Bio::EnsEMBL::Variation::Utils::BaseVepPlugin</a>
	module as their base class, which provides default implementations of all the
	necessary methods which can be overridden as required.<br />Please refer to the
	documentation in this module for details of all required methods and for a
	simple
	example of a plugin implementation. </p>
	
	<hr/>
	<h2 id="plugins_filter">Filtering using plugins</h2>
	
	<p> A common use for plugins will be to filter the output in some way (for
	example to limit output lines to missense variants) and so we provide
	a simple mechanism to support this.<br />The <i>run</i> method of a plugin is assumed
	to return a reference to a hash containing information to be included in the
	output, and if a plugin should not add any data to a particular line
	it should return an empty hashref. If a plugin should instead filter a
	line and exclude it from the output, it should return <i>undef</i> from its <i>run</i>
	method, this also means that no further plugins will be run on the line.<br />If
	you are developing a filter plugin, we suggest that you use the <a
	href="/info/docs/Doxygen/variation-api/classBio_1_1EnsEMBL_1_1Variation_1_1Utils_1_1BaseVepFilterPlugin.html">Bio::EnsEMBL::Variation::Utils::BaseVepFilterPlugin</a>
	as your base class and then you need only override the <i>include_line</i> method
	to return true if you want to include this line, and false otherwise.<br />Again,
	please refer to the documentation in this module for more details and an
	example	implementation of a missense filter. </p>
	
	<hr/>
	<h2 id="plugins_use">Using plugins</h2>
    
    <p> In order to run a plugin you need to include the plugin module in Perl's
    library path somehow; by default VEP includes the <i>&#126;/.vep/Plugins</i>
    directory in the path, so this is a convenient place to store plugins, but
    you are also able to include modules by any other means (e.g using the
    <i>$PERL5LIB</i> environment variable in Unix-like systems).<br />You can then run a
    plugin using the <a href="vep_options.html#opt_plugin">--plugin</a> command line option, passing the name of
    the plugin module as the argument.</p>

    <p>For example, if your plugin is in a
    module called <i>MyPlugin.pm</i>, stored in <i>&#126;/.vep/Plugins</i>, you can run it
    with a command line like: </p>
	
	<pre class="code sh_sh">./vep -i input.vcf --plugin MyPlugin</pre>
	
	<p> You can pass arguments to the plugin's 'new' method by including them after
	the plugin name on the command line, separated by commas, e.g.: </p>
	
	<pre class="code sh_sh">./vep -i input.vcf --plugin MyPlugin,1,FOO</pre>
	
	<p> If your plugin inherits from BaseVepPlugin, you can then retrieve these
	parameters as a list from the <i>params</i> method. </p>
	
	<p> You can run multiple plugins by supplying multiple <a href="vep_options.html#opt_plugin">--plugin</a>
	arguments. Plugins are run serially in the order in which they are specified
	on the command line, so they can be run as a pipeline, with, for example, a
	later plugin filtering output based on the results from an earlier plugin.
	Note though that the first plugin to filter a line 'wins', and any later
	plugins won't get
	run on a filtered line. </p>
	
	<hr/>
	<h2 id="plugins_intergenic">Intergenic variants</h2>
	
	<p> When a variant falls in an intergenic region, it will usually not have
	any consequence types called, and hence will not have any associated
	VariationFeatureOverlap objects. In this special case, VEP creates a new
	VariationFeatureOverlap that overlaps a feature of type "Intergenic".<br />To
	force your plugin to handle these, you must add "Intergenic" to the feature
	types that it will recognize; you do this by writing your own feature_types
	sub-routine:</p>
	
	<pre class="code sh_perl sh_sourceCode">sub feature_types {
    return ['Transcript', 'Intergenic'];
}</pre>
	
	<p> This will cause your plugin to handle any variation features that
	overlap transcripts or intergenic regions. To also include any regulatory
	features, you should use the generic type "Feature":</p>
	
	<pre class="code sh_perl sh_sourceCode">sub feature_types {
    return ['Feature', 'Intergenic'];
}</pre>
	
</div>

</body>
</html>
