<html>
<head>
  <title>Plugins</title>
  <meta name="order" content="7" />
  <script type="text/javascript">
    // Function to show/hide divs
    function show_hide (param) {
      div   = document.getElementById('div_'+param);
      alink = document.getElementById('a_'+param);
      if (div.style.display=='inline') {
        div.style.display='none';
        alink.innerHTML= alink.innerHTML.replace("less","more");
      }
      else {
        if (div.style.display=='none') {
          div.style.display='inline';
          alink.innerHTML= alink.innerHTML.replace("more", "less");
        }
      }
    }
    // Function to show/hide plugins by their category
    function show_hide_category () {
      var category_id = $('select#select_category').val();
      if (category_id == 'all') {
        $('.plugin_row').show();
      }
      else {
        $('.plugin_row').hide();
        $("tr[data-category='"+category_id+"']").show();
      }
    }
  </script>
</head>
<body>
<div>
  <div style="float:right"><img src="/img/vep_logo.png"/></div>
	
  <h1 id="top"><span style="color:#006;padding-right:15px">Ensembl Variant Effect Predictor</span><span style="color:#666"><img src="/i/16/tool.png"/> Plugins</span></h1>
  <hr/>
        
  <p> You can use plugin modules written in Perl to <b>extend, filter and manipulate</b> the Ensembl VEP output. </p>

  <p> To use plugins: </p>
  <ul>
    <li> Install them using the <a href="vep_download.html#installer">Ensembl VEP installer script</a>. You can quickly check installed plugins by running:
    <ul>
      <li> <pre class="code sh_sh">perl INSTALL.pl -a p -g list</pre> </li>
    </ul>
    </li>

    <li> Use Ensembl VEP in <a href="vep_download.html#docker">Docker</a> and <a href="vep_download.html#singularity">Singularity</a>. The plugins and their dependencies are available in the <a href="https://hub.docker.com/r/ensemblorg/ensembl-vep">Docker image</a>. </li>

  </ul>

  <br />	
  <h2 id="plugins_existing">Existing plugins</h2>
	
  <p> We have written plugins to implement new
      functionalities that we do not (yet) include in the variation API, and these
      are stored in a public github repository: </p>
	
  <p><a href="//github.com/Ensembl/VEP_plugins" rel="external">https://github.com/Ensembl/VEP_plugins</a></p>

  <p>Here is the list of the Ensembl VEP plugins available:</p>

  <!-- Plugins list - start -->

  <div style="margin:5px 0px 10px">
    <span style="padding-right:8px">Select categories:</span>
    <select id="select_category" onchange="javascritpt:show_hide_category()">
       <option value="all">All categories</option><option value="conservation">Conservation</option><option value="external_id">External ID</option><option value="frequency_data">Frequency data</option><option value="functional_effect">Functional effect</option><option value="gene_tolerance_to_change">Gene tolerance to change</option><option value="hgvs">HGVS</option><option value="look_up">Look up</option><option value="motif">Motif</option><option value="nearby_features">Nearby features</option><option value="pathogenicity_predictions">Pathogenicity predictions</option><option value="phenotype_data_and_citations">Phenotype data and citations</option><option value="protein_annotation">Protein annotation</option><option value="protein_data">Protein data</option><option value="regulatory_impact">Regulatory impact</option><option value="sequence">Sequence</option><option value="splicing_predictions">Splicing predictions</option><option value="structural_variant_data">Structural variant data</option><option value="transcript_annotation">Transcript annotation</option><option value="variant_data">Variant data</option><option value="visualisation">Visualisation</option>
    </select>
  </div>
  <table class="ss" style="table-layout: fixed; word-break: break-word;">
    <thead>
      <tr>
        <th>Plugin</th>
        <th style="width:50%">Description</th>
        <th>Category</th>
        <th>External libraries</th>
        <th>Developer</th>
      </tr>
    </thead>
    <tbody>
<tr id="AlphaMissense" class="bg1 plugin_row" data-category="pathogenicity_predictions"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/AlphaMissense.pm">AlphaMissense</a></div></td><td><p> This plugin for the Ensembl Variant Effect Predictor (VEP) annotates missense variants with the 
 pre-computed AlphaMissense pathogenicity scores. AlphaMissense is a deep learning model developed 
 by Google DeepMind that predicts the pathogenicity of single nucleotide missense variants.  <a class="button" href="#alphamissense" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('alphamissense');" id="a_alphamissense">more</a></p><div id="div_alphamissense" style="display:none;word-wrap:break-word;"><p> This plugin will add two annotations per missense variant:</p><p><ul><li><kbd>am_pathogenicity</kbd>, a continuous score between 0 and 1 which can be interpreted as the predicted &nbsp; probability of the variant being pathogenic. </p><p></li><li><kbd>am_class</kbd> is the classification of the variant into one of three discrete categories:  &nbsp;   <kbd>likely_pathogenic</kbd>, <kbd>likely_benign</kbd>, or <kbd>ambiguous</kbd>. These are derived using the following 
&nbsp;   thresholds of am_pathogenicity: 
&nbsp;     <kbd>likely_benign</kbd>     if <kbd>am_pathogenicity</kbd> &lt 0.34; 
&nbsp;     <kbd>likely_pathogenic</kbd> if <kbd>am_pathogenicity</kbd> &gt 0.564; 
&nbsp;     <kbd>ambiguous</kbd> otherwise. </p><p></li></ul><p> These thresholds were chosen to achieve 90% precision for both pathogenic and benign ClinVar variants. 
 Note that AlphaMissense was not trained on ClinVar variants. Variants labeled as <kbd>ambiguous</kbd> should be 
 treated as <kbd>unknown</kbd> or <kbd>uncertain</kbd> according to AlphaMissense. </p><p> This plugin is available for both GRCh37 (hg19) and GRCh38 (hg38) genome builds.</p><p> The prediction scores of AlphaMissense can be downloaded from 
 <a href="https://console.cloud.google.com/storage/browser/dm_alphamissense">https://console.cloud.google.com/storage/browser/dm_alphamissense</a> 
 (AlphaMissense Database Copyright (2023) DeepMind Technologies Limited). Data contained within the
 AlphaMissense Database is licensed under the Creative Commons Attribution 4.0 International License
 (CC-BY) (the “License”). You may obtain a copy of the License at:
 <a href="https://creativecommons.org/licenses/by/4.0/legalcode">https://creativecommons.org/licenses/by/4.0/legalcode</a>.
 Use of the AlphaMissense Database is subject to Google Cloud Platform Terms of Service  </p><p> Please cite the AlphaMissense publication alongside the Ensembl VEP
 if you use this resource: <a href="https://doi.org/10.1126/science.adg7492">https://doi.org/10.1126/science.adg7492</a></p><p> Disclaimer: The AlphaMissense Database and other information provided on or linked to this site is 
 for theoretical modelling only, caution should be exercised in use. It is provided "as-is" without 
 any warranty of any kind, whether express or implied. For clarity, no warranty is given that use of 
 the information shall not infringe the rights of any third party (and this disclaimer takes precedence 
 over any contrary provisions in the Google Cloud Platform Terms of Service). The information provided 
 is not intended to be a substitute for professional medical advice, diagnosis, or treatment, and does 
 not constitute medical or other professional advice.</p><p> Before running the plugin for the first time, you need to create a tabix index (requires tabix to be 
 installed).</p><p><pre class="code sh_sh">tabix -s 1 -b 2 -e 2 -f -S 1 AlphaMissense_hg38.tsv.gz
</pre><p><pre class="code sh_sh">tabix -s 1 -b 2 -e 2 -f -S 1 AlphaMissense_hg19.tsv.gz
</pre><p></p><p> Options are passed to the plugin as key=value pairs:<table class="ss"><thead><tr><th>Argument</th><th>Description</th></tr></thead><tbody><tr class="bg1"><td><pre>file</pre></td><td>(mandatory) Tabix-indexed AlphaMissense data <tr class="bg2"><td><pre>cols</pre></td><td>(optional) Colon-separated columns to print from                       AlphaMissense data; if set to <kbd>all</kbd>, all columns are printed
                      (default: <kbd>Missense_pathogenicity:Missense_class</kbd>)
<tr class="bg1"><td><pre>transcript_match</pre></td><td>Only print data if transcript identifiers match those from                       AlphaMissense data (default: 0)</p><p></td></tr></tbody></table><p> AlphaMissense predictions are matched to input data by genomic location and protein change by default.</p><p></ul><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv AlphaMissense.pm ~/.vep/Plugins</p><p># print AlphaMissense scores and predictions (default)
./vep -i variations.vcf --plugin AlphaMissense,file=/full/path/to/file.tsv.gz</p><p># print all AlphaMissense information
./vep -i variations.vcf --plugin AlphaMissense,file=/full/path/to/file.tsv.gz,cols=all</p><p># only report results for the transcripts in the AlphaMissense prediction
./vep -i variations.vcf --plugin AlphaMissense,file=/full/path/to/file.tsv.gz,transcript_match=1
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#1E90FF">Pathogenicity predictions</div></td><td>-</td><td>Ensembl</td></tr>
<tr id="AncestralAllele" class="bg2 plugin_row" data-category="conservation"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/AncestralAllele.pm">AncestralAllele</a></div></td><td><p> A VEP plugin that retrieves ancestral allele sequences from a FASTA file. <a class="button" href="#ancestralallele" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('ancestralallele');" id="a_ancestralallele">more</a></p><div id="div_ancestralallele" style="display:none;word-wrap:break-word;"><p> Ensembl produces FASTA file dumps of the ancestral sequences of key species.
<ul><li>Data files for GRCh37: <a href="https://ftp.ensembl.org/pub/release-75/fasta/ancestral_alleles/">https://ftp.ensembl.org/pub/release-75/fasta/ancestral_alleles/</a></li><li>Data files for GRCh38: <a href="https://ftp.ensembl.org/pub/current_fasta/ancestral_alleles/">https://ftp.ensembl.org/pub/current_fasta/ancestral_alleles/</a>
</li></ul><p> For optimal retrieval speed, you should pre-process the FASTA files into a single
 bgzipped file that can be accessed via <kbd>Bio::DB::HTS::Faidx</kbd> (installed by VEP's
 INSTALL.pl):</p><p><pre class="code sh_sh">wget <a href="https://ftp.ensembl.org/pub/current_fasta/ancestral_alleles/homo_sapiens_ancestor_GRCh38.tar.gz">https://ftp.ensembl.org/pub/current_fasta/ancestral_alleles/homo_sapiens_ancestor_GRCh38.tar.gz</a>
tar xfz homo_sapiens_ancestor_GRCh38.tar.gz
cat homo_sapiens_ancestor_GRCh38/*.fa | bgzip -c &gt homo_sapiens_ancestor_GRCh38.fa.gz
rm -rf homo_sapiens_ancestor_GRCh38/ homo_sapiens_ancestor_GRCh38.tar.gz
./vep -i variations.vcf --plugin AncestralAllele,homo_sapiens_ancestor_GRCh38.fa.gz
</pre><p> The plugin is also compatible with <kbd>Bio::DB::Fasta</kbd> and an uncompressed FASTA file.</p><p> Note the first time you run the plugin with a newly generated FASTA file it will
 spend some time indexing the file. DO NOT INTERRUPT THIS PROCESS, particularly
 if you do not have <kbd>Bio::DB::HTS</kbd> installed.</p><p> Special cases:
<ul><li><kbd>-</kbd> represents an insertion</li><li><kbd>?</kbd> indicates the chromosome could not be looked up in the FASTA
</ul><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv AncestralAllele.pm ~/.vep/Plugins
./vep -i variations.vcf --plugin AncestralAllele,homo_sapiens_ancestor_GRCh38.fa.gz
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#02599C">Conservation</div></td><td>-</td><td>Ensembl</td></tr>
<tr id="AVADA" class="bg1 plugin_row" data-category="phenotype_data_and_citations"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/AVADA.pm">AVADA</a></div></td><td><p>Automatic VAriant evidence DAtabase is a novel machine learning tool that uses natural language processing 
to automatically identify pathogenic genetic variant evidence in full-text primary literature about 
monogenic disease and convert it to genomic coordinates. <a class="button" href="#avada" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('avada');" id="a_avada">more</a></p><div id="div_avada" style="display:none;word-wrap:break-word;"><p>Please cite the AVADA publication alongside the Ensembl VEP if you use this resource:
<a href="https://pubmed.ncbi.nlm.nih.gov/31467448/">https://pubmed.ncbi.nlm.nih.gov/31467448/</a></p><p>NB: The plugin currently does not annotate for downstream_gene_variant and upstream_gene_variant.</p><p>Pre-requisites
<ol><li value="1">AVADA data is available for GRCh37 and can be downloaded from: <a href="http://bejerano.stanford.edu/AVADA/avada_v1.00_2016.vcf.gz">http://bejerano.stanford.edu/AVADA/avada_v1.00_2016.vcf.gz</a></p><p></li></ol><p><pre class="code sh_sh">wget <a href="http://bejerano.stanford.edu/AVADA/avada_v1.00_2016.vcf.gz">http://bejerano.stanford.edu/AVADA/avada_v1.00_2016.vcf.gz</a>
</pre><p><ol><li value="2">The file needs to be tabix indexed. You can do this by following commands:
</li></ol><p><pre class="code sh_sh">gzip -d avada_v1.00_2016.vcf.gz
bgzip avada_v1.00_2016.vcf 
tabix avada_v1.00_2016.vcf.gz
</pre><p><ol><li value="3">As you have already noticed, tabix utility must be installed in your path to use this plugin. 
</li></ol><p>The plugin can then be run to retrieve AVADA annotations. 
By default, the variants are matched with the HGNC gene symbol
<pre class="code sh_sh">./vep -i variations.vcf --plugin AVADA,file=path/to/file
</pre><p>The output always includes one of the following columns depending on the option passed:
<ul><li><kbd>AVADA_PMID</kbd>: PubMed ID evidence for the variant as reported by AVADA</li><li><kbd>AVADA_PMID_WITH_VARIANT</kbd>: PubMed ID evidence for the variant as reported by AVADA along with the original variant string</li><li><kbd>AVADA_PMID_WITH_FEATURE</kbd>: PubMed ID evidence for the variant as reported by AVADA along with feature id</li><li><kbd>AVADA_PMID_WITH_FEATURE_AND_VARIANT</kbd>: PubMed ID evidence for the variant as reported by AVADA along with feature id and original variant string
</li></ul><p>The plugin can optionally be run by specifying the feature to match with.</p><p>In order to match by HGNC gene symbol:
<pre class="code sh_sh">./vep -i variations.vcf --plugin AVADA,file=path/to/file,feature_match_by=gene_symbol 
</pre><p>In order to match by Ensembl gene identifier :
<pre class="code sh_sh">./vep -i variations.vcf --plugin AVADA,file=path/to/file,feature_match_by=ensembl_gene_id
</pre><p>In order to match by RefSeq identifier :
<pre class="code sh_sh">./vep -i variations.vcf --plugin AVADA,file=path/to/file,feature_match_by=refseq_id
</pre><p>The plugin can also be run to report the original variant string reported in the publication. 
<pre class="code sh_sh">./vep -i variations.vcf --plugin AVADA,file=path/to/file,original_variant_string=1
</pre><p>
</ul><p></ol><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">./vep -i variations.vcf --plugin AVADA,file=path/to/file
./vep -i variations.vcf --plugin AVADA,file=path/to/file,feature_match_by=&ltgene_symbol|ensembl_gene_id|refseq_id&gt
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#556688">Phenotype data and citations</div></td><td><a href="https://metacpan.org/pod/List::MoreUtils" rel="external">List::MoreUtils</a> qw(uniq)</td><td>Ensembl</td></tr>
<tr id="BayesDel" class="bg2 plugin_row" data-category="pathogenicity_predictions"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/BayesDel.pm">BayesDel</a></div></td><td><p> This is a plugin for the Ensembl Variant Effect Predictor (VEP) that
 adds the BayesDel scores to Ensembl VEP output. <a class="button" href="#bayesdel" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('bayesdel');" id="a_bayesdel">more</a></p><div id="div_bayesdel" style="display:none;word-wrap:break-word;"><p> BayesDel is a deleteriousness meta-score combining multiple deleteriousness predictors to create an overall score. It works for coding and non-coding variants,
 single nucleotide variants and small insertion/deletions.
 The range of the score is from -1.29334 to 0.75731.
 The higher the score, the more likely the variant is pathogenic.
 For more information please visit: <a href="https://fenglab.chpc.utah.edu/BayesDel/BayesDel.html">https://fenglab.chpc.utah.edu/BayesDel/BayesDel.html</a></p><p> Please cite the BayesDel publication alongside the Ensembl VEP if you use this resource:
 <a href="https://onlinelibrary.wiley.com/doi/full/10.1002/humu.23158">https://onlinelibrary.wiley.com/doi/full/10.1002/humu.23158</a></p><p> BayesDel pre-computed scores can be downloaded from
 <a href="https://drive.google.com/drive/folders/1K4LI6ZSsUGBhHoChUtegC8bgCt7hbQlA">https://drive.google.com/drive/folders/1K4LI6ZSsUGBhHoChUtegC8bgCt7hbQlA</a>
 Note: These files only contain pre-computed BayesDel scores for missense variants for assembly GRCh37.</p><p> For GRCh37:
<pre class="code sh_sh">tar zxvf BayesDel_170824_addAF.tgz
rm *.gz.tbi
gunzip *.gz
for f in BayesDel_170824_addAF_chr*; do grep -v "^#" $f &gt&gt BayesDel_170824_addAF.txt; done
cat BayesDel_170824_addAF.txt | sort -k1,1 -k2,2n &gt BayesDel_170824_addAF_sorted.txt
grep "^#" BayesDel_170824_addAF_chr1 &gt BayesDel_170824_addAF_all_scores.txt
cat BayesDel_170824_addAF_sorted.txt &gt&gt BayesDel_170824_addAF_all_scores.txt
bgzip BayesDel_170824_addAF_all_scores.txt
tabix -s 1 -b 2 -e 2 BayesDel_170824_addAF_all_scores.txt.gz
</pre><p> For GRCh38:
 Remap GRCh37 file</p><p> The tabix utility must be installed in your path to use this plugin.</p><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv BayesDel.pm ~/.vep/Plugins
./vep -i variations.vcf --plugin BayesDel,file=/path/to/BayesDel/BayesDel_170824_addAF_all_scores.txt.gz
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#1E90FF">Pathogenicity predictions</div></td><td>-</td><td>Ensembl</td></tr>
<tr id="Blosum62" class="bg1 plugin_row" data-category="conservation"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/Blosum62.pm">Blosum62</a></div></td><td><p> This is a plugin for the Ensembl Variant Effect Predictor (VEP) that
 looks up the BLOSUM 62 substitution matrix score for the reference
 and alternative amino acids predicted for a missense mutation. It adds
 one new entry to Ensembl VEP's Extra column, BLOSUM62 which is the 
 associated score.  <a class="button" href="#blosum62" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('blosum62');" id="a_blosum62">more</a></p><div id="div_blosum62" style="display:none;word-wrap:break-word;"><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv Blosum62.pm ~/.vep/Plugins
./vep -i variations.vcf --plugin Blosum62
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#02599C">Conservation</div></td><td>-</td><td>Ensembl</td></tr>
<tr id="CADD" class="bg2 plugin_row" data-category="pathogenicity_predictions"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/CADD.pm">CADD</a></div><div style="margin-top:6px"><small>Combined Annotation Dependent Depletion</small></div></td><td><p> A VEP plugin that retrieves CADD scores for variants from one or more
 tabix-indexed CADD data files. <a class="button" href="#cadd" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('cadd');" id="a_cadd">more</a></p><div id="div_cadd" style="display:none;word-wrap:break-word;"><p> Please cite the CADD publication alongside Ensembl VEP if you use this resource:
 <a href="https://www.ncbi.nlm.nih.gov/pubmed/24487276">https://www.ncbi.nlm.nih.gov/pubmed/24487276</a></p><p> The tabix utility must be installed in your path to use this plugin. </p><p> The CADD SNV and indels data files (and respective Tabix index files) can be downloaded from - 
 <a href="http://cadd.gs.washington.edu/download">http://cadd.gs.washington.edu/download</a></p><p> The CADD SV data files (and respective Tabix index files)  can be downloaded from -
 <a href="https://kircherlab.bihealth.org/download/CADD-SV/v1.1/">https://kircherlab.bihealth.org/download/CADD-SV/v1.1/</a></p><p> By default the plugin is designed to not annotate SV variant if a SNV and/or indels CADD
 annotation file is provided. Because it can results in too many lines matched from the annotation
 files and increase run time exponentially. You can override this behavior by providing 
 force_annotate=1 which will force the plugin to annotate with the expense of increasing runtime.</p><p> The plugin works with all versions of available CADD files. The plugin only
 reports scores and does not consider any additional annotations from a CADD
 file. It is therefore sufficient to use CADD files without the additional
 annotations. </p><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv CADD.pm ~/.vep/Plugins
./vep -i variations.vcf --plugin CADD,snv=/FULL_PATH_TO_CADD_FILE/whole_genome_SNVs.tsv.gz,indels=/FULL_PATH_TO_CADD_FILE/InDels.tsv.gz
./vep -i structural_variations.vcf --plugin CADD,sv=/FULL_PATH_TO_CADD_FILE/1000G_phase3_SVs.tsv.gz
./vep -i structural_variations.vcf --plugin CADD,snv=/FULL_PATH_TO_CADD_FILE/whole_genome_SNVs.tsv.gz,indels=/FULL_PATH_TO_CADD_FILE/InDels.tsv.gz,force_annotate=1
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#1E90FF">Pathogenicity predictions</div></td><td>-</td><td>Ensembl</td></tr>
<tr id="CAPICE" class="bg1 plugin_row" data-category="pathogenicity_predictions"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/CAPICE.pm">CAPICE</a></div></td><td><p> An Ensembl VEP plugin that retrieves CAPICE scores for variants from one or more
 tabix-indexed CAPICE data files, in order to predict their pathogenicity. <a class="button" href="#capice" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('capice');" id="a_capice">more</a></p><div id="div_capice" style="display:none;word-wrap:break-word;"><p> Please cite the CAPICE publication alongside the VEP if you use this resource:
 <a href="https://pubmed.ncbi.nlm.nih.gov/32831124/">https://pubmed.ncbi.nlm.nih.gov/32831124/</a></p><p> The tabix utility must be installed in your path to use this plugin. The CAPICE
 SNVs, InDels and respective index (TBI) files for GRCh37 can be downloaded from
 <a href="https://zenodo.org/record/3928295">https://zenodo.org/record/3928295</a></p><p> To filter results, please use filter_vep with the output file or standard
 output. Documentation on filter_vep is available at:
 <a href="https://www.ensembl.org/info/docs/tools/vep/script/vep_filter.html">https://www.ensembl.org/info/docs/tools/vep/script/vep_filter.html</a></p><p> For recommendations on threshold selection, please read the CAPICE publication.</p><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv CAPICE.pm ~/.vep/Plugins
# Download CAPICE SNVs, InDels and index (TBI) files to the same path
# - capice_v1.0_build37_indels.tsv.gz
# - capice_v1.0_build37_indels.tsv.gz.tbi
# - capice_v1.0_build37_snvs.tsv.gz
# - capice_v1.0_build37_snvs.tsv.gz.tbi
./vep -i variations.vcf --plugin CAPICE,snv=/FULL_PATH_TO_CAPICE_FILE/capice_v1.0_build37_snvs.tsv.gz,indels=/FULL_PATH_TO_CAPICE_FILE/capice_v1.0_build37_indels.tsv.gz
./filter_vep -i variant_effect_output.txt --filter "CAPICE_SCORE &gt= 0.02"
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#1E90FF">Pathogenicity predictions</div></td><td>-</td><td>Ensembl</td></tr>
<tr id="Carol" class="bg2 plugin_row" data-category="pathogenicity_predictions"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/Carol.pm">Carol</a></div></td><td><p> An Ensembl VEP plugin that calculates the Combined Annotation scoRing toOL (CAROL)
 score (1) for a missense mutation based on the pre-calculated SIFT (2) and
 PolyPhen-2 (3) scores from the Ensembl API (4). <a class="button" href="#carol" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('carol');" id="a_carol">more</a></p><div id="div_carol" style="display:none;word-wrap:break-word;"><p> It adds one new entry class to Ensembl VEP's Extra column, CAROL which is
 the calculated CAROL score. Note that this module is a perl reimplementation of 
 the original R script, available at: <a href="https://sanger.ac.uk/tool/carol/">https://sanger.ac.uk/tool/carol/</a></p><p> I believe that both versions implement the same algorithm, but if there are any
 discrepancies the R version should be treated as the reference implementation. 
 Bug reports are welcome.</p><p> References:</p><p><ol><li value="1">Lopes MC, Joyce C, Ritchie GRS, John SL, Cunningham F, Asimit J, Zeggini E.      A combined functional annotation score for non-synonymous variants
     Human Heredity 73(1):47-51 (2012)
     <a rel="external" href="https://doi.org/10.1159/000334984">doi:10.1159/000334984</a></p><p></li><li value="2">Kumar P, Henikoff S, Ng PC.     Predicting the effects of coding non-synonymous variants on protein function using the SIFT algorithm
     Nature Protocols 4(8):1073-1081 (2009)
     <a rel="external" href="https://doi.org/10.1038/nprot.2009.86">doi:10.1038/nprot.2009.86</a></p><p></li><li value="3">Adzhubei IA, Schmidt S, Peshkin L, Ramensky VE, Gerasimova A, Bork P, Kondrashov AS, Sunyaev SR.      A method and server for predicting damaging missense mutations
     Nature Methods 7(4):248-249 (2010)
     <a rel="external" href="https://doi.org/10.1038/nmeth0410-248">doi:10.1038/nmeth0410-248</a></p><p></li><li value="4">Flicek P, et al.     Ensembl 2012
     Nucleic Acids Research 40(D1):D84-D90 (2011)
     doi: 10.1093/nar/gkr991</p><p></ol><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv Carol.pm ~/.vep/Plugins
./vep -i variations.vcf --plugin Carol
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#1E90FF">Pathogenicity predictions</div></td><td><a href="https://metacpan.org/pod/Math::CDF" rel="external">Math::CDF</a> qw(pnorm qnorm)</td><td>Ensembl</td></tr>
<tr id="ClinPred" class="bg1 plugin_row" data-category="pathogenicity_predictions"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/ClinPred.pm">ClinPred</a></div></td><td><p> This is a plugin for the Ensembl Variant Effect Predictor (VEP) that adds pre-calculated scores from ClinPred.
 ClinPred is a prediction tool to identify disease-relevant nonsynonymous variants. <a class="button" href="#clinpred" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('clinpred');" id="a_clinpred">more</a></p><div id="div_clinpred" style="display:none;word-wrap:break-word;"><p> Please cite the ClinPred publication alongside Ensembl VEP if you use this resource:
 <a href="https://www.sciencedirect.com/science/article/pii/S0002929718302714">https://www.sciencedirect.com/science/article/pii/S0002929718302714</a></p><p> ClinPred scores can be downloaded from 
 <a href="https://sites.google.com/site/clinpred/download">https://sites.google.com/site/clinpred/download</a></p><p> The following steps are neccessary to tabix the ClinPred.txt.gz file before running the plugin:</p><p> For GRCh37:
<pre class="code sh_sh">gzip -d ClinPred.txt.gz # to unzip the text file 
cat ClinPred.txt | tr " " "\t" &gt ClinPred_tabbed.tsv # change to tab-delimited file 
sed -i '1s/.*/#&/' ClinPred_tabbed.tsv  # comment the first line
sed -i <kbd>1s/Chr/chr/</kbd> ClinPred_tabbed.tsv # convert Chr to chr
bgzip ClinPred_tabbed.tsv
tabix -f -s 1 -b 2 -e 2 ClinPred_tabbed.tsv.gz
</pre><p> For GRCh38:
<pre class="code sh_sh">gzip -d ClinPred_hg38.txt.gz # unzip the text file 
awk '($2 == "Start" || $2 ~ /^[0-9]+$/){print $0}' ClinPred_hg38.txt &gt "ClinPred_hg38_tabbed.tsv" # remove problematic lines
sed -i '1s/.*/#&/' ClinPred_hg38_tabbed.tsv # comment the first line
sed -i <kbd>1s/Chr/chr/</kbd> ClinPred_hg38_tabbed.tsv # convert Chr to chr
</pre><p> { head -n 1 ClinPred_hg38_tabbed.tsv; tail -n +2 ClinPred_hg38_tabbed.tsv | sort -k1,1V -k2,2V; } &gt ClinPred_hg38_sorted_tabbed.tsv # sort file by chromosome and position
<pre class="code sh_sh">bgzip ClinPred_hg38_sorted_tabbed.tsv
tabix -f -s 1 -b 2 -e 2 ClinPred_hg38_sorted_tabbed.tsv.gz
</pre><p> The tabix utility must be installed in your path to use this plugin.
 Check <a href="https://github.com/samtools/htslib.git">https://github.com/samtools/htslib.git</a> for instructions.</p><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv ClinPred.pm ~/.vep/Plugins
./vep -i variations.vcf --plugin ClinPred,file=ClinPred_tabbed.tsv.gz</p><p></pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#1E90FF">Pathogenicity predictions</div></td><td>-</td><td>Ensembl</td></tr>
<tr id="Condel" class="bg2 plugin_row" data-category="pathogenicity_predictions"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/Condel.pm">Condel</a></div></td><td><p> A VEP plugin that calculates the Consensus Deleteriousness (Condel) score (1)
 for a missense mutation based on the pre-calculated SIFT (2) and PolyPhen-2 (3)
 scores from the Ensembl API (4). <a class="button" href="#condel" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('condel');" id="a_condel">more</a></p><div id="div_condel" style="display:none;word-wrap:break-word;"><p> It adds one new entry class to Ensembl VEP's Extra column, Condel which is the
 calculated Condel score. This version of Condel was developed by the Biomedical
 Genomics Group  of the Universitat Pompeu Fabra, at the Barcelona Biomedical
 Research Park and available at <a href="https://bg.upf.edu/condel">https://bg.upf.edu/condel</a>. The code in this
 plugin is based on a script provided by this group and slightly reformatted to
 fit into the Ensembl API.</p><p> The plugin takes 3 command line arguments by this order:
<ol><li value="1">Path to a Condel configuration directory which contains cutoffs and the    distribution files, etc.
</li><li value="2">Output: output the Condel score (<kbd>s</kbd>), prediction (<kbd>p</kbd>) or both (<kbd>b</kbd>);    both (<kbd>b</kbd>) is the default.
</li><li value="3">Version of Condel to use: either 1 (original version) or 2 (newer version);    <kbd>2</kbd> is the default and is recommended to avoid false positive predictions
    from Condel in some circumstances.</p><p></li></ol><p> An example Condel configuration file and a set of distribution files can be
 found in the <kbd>config/Condel</kbd> directory in this repository. You should edit the 
 <kbd>config/Condel/config/condel_SP.conf</kbd> file and set the <kbd>condel.dir</kbd> parameter to
 the full path to the location of the <kbd>config/Condel</kbd> directory on your system.</p><p> References:</p><p><ol><li value="1">Gonzalez-Perez A, Lopez-Bigas N.      Improving the assessment of the outcome of non-synonymous SNVs with a Consensus deleteriousness score (Condel)
     Am J Hum Genet 88(4):440-449 (2011)
     <a rel="external" href="https://doi.org/10.1016/j.ajhg.2011.03.004">doi:10.1016/j.ajhg.2011.03.004</a></p><p></li><li value="2">Kumar P, Henikoff S, Ng PC.     Predicting the effects of coding non-synonymous variants on protein function using the SIFT algorithm
     Nature Protocols 4(8):1073-1081 (2009)
     <a rel="external" href="https://doi.org/10.1038/nprot.2009.86">doi:10.1038/nprot.2009.86</a></p><p></li><li value="3">Adzhubei IA, Schmidt S, Peshkin L, Ramensky VE, Gerasimova A, Bork P, Kondrashov AS, Sunyaev SR.      A method and server for predicting damaging missense mutations
     Nature Methods 7(4):248-249 (2010)
     <a rel="external" href="https://doi.org/10.1038/nmeth0410-248">doi:10.1038/nmeth0410-248</a></p><p></li><li value="4">Flicek P, et al.     Ensembl 2012
     Nucleic Acids Research (2011)
     <a rel="external" href="https://doi.org/10.1093/nar/gkr991">doi:10.1093/nar/gkr991</a></p><p></ol><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv Condel.pm ~/.vep/Plugins
./vep -i variations.vcf --plugin Condel,/path/to/config/Condel/config,b
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#1E90FF">Pathogenicity predictions</div></td><td>-</td><td>Ensembl</td></tr>
<tr id="Conservation" class="bg1 plugin_row" data-category="conservation"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/Conservation.pm">Conservation</a></div></td><td><p> This is a plugin for the Ensembl Variant Effect Predictor (VEP) that
 retrieves a conservation score from the Ensembl Compara databases
 for variant positions. You can specify the method link type and
 species sets as command line options, the default is to fetch GERP
 scores from the EPO 35 way mammalian alignment (please refer to the
 Compara documentation for more details of available analyses).  <a class="button" href="#conservation" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('conservation');" id="a_conservation">more</a></p><div id="div_conservation" style="display:none;word-wrap:break-word;"><p> If a variant affects multiple nucleotides the average score for the
 position will be returned, and for insertions the average score of
 the 2 flanking bases will be returned. If the MAX parameter is
 used, the maximum score of any of the affected bases will be reported
 instead.</p><p> The plugin uses the ensembl-compara API module (optional, see
 <a href="http://www.ensembl.org/info/docs/api/index.html">http://www.ensembl.org/info/docs/api/index.html</a>) or obtains data
 directly from BigWig files (optional, see
 <a href="https://ftp.ensembl.org/pub/current_compara/conservation_scores/">https://ftp.ensembl.org/pub/current_compara/conservation_scores/</a>)</p><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv Conservation.pm ~/.vep/Plugins</p><p>./vep -i variations.vcf --plugin Conservation,mammals
./vep -i variations.vcf --plugin Conservation,/path/to/bigwigfile.bw
./vep -i variations.vcf --plugin Conservation,/path/to/bigwigfile.bw,MAX
./vep -i variations.vcf --plugin Conservation,database,GERP_CONSERVATION_SCORE,mammals
./vep -i variations.vcf --plugin Conservation,database,GERP_CONSERVATION_SCORE,mammals,MAX
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#02599C">Conservation</div></td><td><a href="https://metacpan.org/pod/Net::FTP" rel="external">Net::FTP</a></td><td>Ensembl</td></tr>
<tr id="dbNSFP" class="bg2 plugin_row" data-category="pathogenicity_predictions"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/dbNSFP.pm">dbNSFP</a></div></td><td><p> An Ensembl VEP plugin that retrieves data for missense variants from a tabix-indexed
 dbNSFP file. <a class="button" href="#dbnsfp" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('dbnsfp');" id="a_dbnsfp">more</a></p><div id="div_dbnsfp" style="display:none;word-wrap:break-word;"><p> Please cite the dbNSFP publications alongside Ensembl VEP if you use this resource:
<ul><li>dbNSFP      <a href="https://www.ncbi.nlm.nih.gov/pubmed/21520341">https://www.ncbi.nlm.nih.gov/pubmed/21520341</a></li><li>dbNSFP v2.0 <a href="https://www.ncbi.nlm.nih.gov/pubmed/23843252">https://www.ncbi.nlm.nih.gov/pubmed/23843252</a></li><li>dbNSFP v3.0 <a href="https://www.ncbi.nlm.nih.gov/pubmed/26555599">https://www.ncbi.nlm.nih.gov/pubmed/26555599</a></li><li>dbNSFP v4   <a href="https://www.ncbi.nlm.nih.gov/pubmed/33261662">https://www.ncbi.nlm.nih.gov/pubmed/33261662</a> 
</li></ul><p> You must have the <kbd>Bio::DB::HTS</kbd> module or the tabix utility must be installed
 in your path to use this plugin.</p><p> About dbNSFP data files:
<ul><li>Downoad dbNSFP files from <a href="https://sites.google.com/site/jpopgen/dbNSFP">https://sites.google.com/site/jpopgen/dbNSFP</a>.</li><li>There are two distinct branches of the files provided for academic and&nbsp;     commercial usage. Please use the appropriate files for your use case.
</li><li>The file must be processed depending on dbNSFP release version and assembly&nbsp;     (see commands below). We recommend using <kbd>-T</kbd> option with the sort command
&nbsp;     to specify a temporary directory with sufficient space.
</li><li>The resulting file must be indexed with tabix before use by this plugin&nbsp;     (see commands below).</p><p></li></ul><p> For release 4.9c:
<pre class="code sh_sh">version=4.9c
wget <a href="https://dbnsfp.s3.amazonaws.com/dbNSFP4.9c.zip">https://dbnsfp.s3.amazonaws.com/dbNSFP${version}.zip</a>
unzip dbNSFP${version}.zip
zcat dbNSFP${version}_variant.chr1.gz | head -n1 &gt h
</pre><p> # GRCh38/hg38 data
<pre class="code sh_sh">zgrep -h -v ^#chr dbNSFP${version}_variant.chr* | sort -k1,1 -k2,2n - | cat h - | bgzip -c &gt dbNSFP${version}_grch38.gz
tabix -s 1 -b 2 -e 2 dbNSFP${version}_grch38.gz
</pre><p> # GRCh37/hg19 data
<pre class="code sh_sh">zgrep -h -v ^#chr dbNSFP${version}_variant.chr* | awk '$8 != "." ' | sort -k8,8 -k9,9n - | cat h - | bgzip -c &gt dbNSFP${version}_grch37.gz
tabix -s 8 -b 9 -e 9 dbNSFP${version}_grch37.gz
</pre><p> When running the plugin you must list at least one column to retrieve from the
 dbNSFP file, specified as parameters to the plugin, such as:</p><p><pre class="code sh_sh">--plugin dbNSFP,/path/to/dbNSFP.gz,LRT_score,GERP++_RS
</pre><p> You may include all columns with <kbd>ALL</kbd>; this fetches a large amount of data per
 variant:</p><p><pre class="code sh_sh">--plugin dbNSFP,/path/to/dbNSFP.gz,ALL
</pre><p> Tabix also allows the data file to be hosted on a remote server. This plugin is
 fully compatible with such a setup - simply use the URL of the remote file:</p><p><pre class="code sh_sh">--plugin dbNSFP,<a href="http://my.files.com/dbNSFP.gz">http://my.files.com/dbNSFP.gz</a>,col1,col2
</pre><p> The plugin replaces occurrences of <kbd>;</kbd> with <kbd>,</kbd> and <kbd>|</kbd> with <kbd>&</kbd>. However, some
 data field columns, e.g. <kbd>Interpro_domain</kbd>, use the replacement characters. We
 added a file with replacement logic for customising the required replacement
 of <kbd>;</kbd> and <kbd>|</kbd> in dbNSFP data columns. In addition to the default replacements
 (<kbd>;</kbd> to <kbd>,</kbd> and <kbd>|</kbd> to <kbd>&</kbd>) users can add customised replacements. Users can either modify
 the file <kbd>dbNSFP_replacement_logic</kbd> in the VEP_plugins directory or provide their own
 file as second argument when calling the plugin:</p><p><pre class="code sh_sh">--plugin dbNSFP,/path/to/dbNSFP.gz,/path/to/dbNSFP_replacement_logic,LRT_score,GERP++_RS
</pre><p> Note that transcript sequences referred to in dbNSFP may be out of sync with
 those in the latest release of Ensembl; this may lead to discrepancies with
 scores retrieved from other sources.</p><p> If the dbNSFP README file is found in the same directory as the data file,
 column descriptions will be read from this and incorporated into the Ensembl VEP output
 file header.</p><p> The plugin matches rows in the tabix-indexed dbNSFP file on:
<ul><li>genomic position</li><li>alt allele</li><li><kbd>aaref</kbd> - reference amino acid</li><li><kbd>aaalt</kbd> - alternative amino acid
</li></ul><p> To match only on the genomic position and the alt allele use <kbd>pep_match=0</kbd>:</p><p><pre class="code sh_sh">--plugin dbNSFP,/path/to/dbNSFP.gz,pep_match=0,col1,col2
</pre><p> Some fields contain multiple values, one per Ensembl transcript ID.
 By default all values are returned, separated by <kbd>;</kbd> in the default Ensembl VEP output format.
 To return values only for the matched Ensembl transcript ID use <kbd>transcript_match=1</kbd>.
 This behaviour only affects transcript-specific fields; non-transcript-specific fields
 are unaffected.</p><p><pre class="code sh_sh">--plugin dbNSFP,/path/to/dbNSFP.gz,transcript_match=1,col1,col2
</pre><p> NB 1: Using this flag may cause no value to return if the version of the Ensembl
 transcript set differs between Ensembl VEP and dbNSFP.</p><p> NB 2: MutationTaster entries are keyed on a different set of transcript IDs. Using
 the <kbd>transcript_match</kbd> flag with any MutationTaster field selected will have no effect
 i.e. all entries are returned. Information on corresponding transcript(s) for
 MutationTaster fields can be found using <a href="http://www.mutationtaster.org/ChrPos.html">http://www.mutationtaster.org/ChrPos.html</a></p><p></ul><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv dbNSFP.pm ~/.vep/Plugins
./vep -i variations.vcf --plugin dbNSFP,/path/to/dbNSFP.gz,col1,col2
./vep -i variations.vcf --plugin dbNSFP,'consequence=ALL',/path/to/dbNSFP.gz,col1,col2
./vep -i variations.vcf --plugin dbNSFP,'consequence=3_prime_UTR_variant&intron_variant',/path/to/dbNSFP.gz,col1,col2
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#1E90FF">Pathogenicity predictions</div></td><td><a href="https://metacpan.org/pod/File::Basename" rel="external">File::Basename</a> qw(basename)</td><td>Ensembl</td></tr>
<tr id="dbscSNV" class="bg1 plugin_row" data-category="splicing_predictions"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/dbscSNV.pm">dbscSNV</a></div></td><td><p>  An Ensembl VEP plugin that retrieves data for splicing variants from a tabix-indexed
  dbscSNV file. <a class="button" href="#dbscsnv" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('dbscsnv');" id="a_dbscsnv">more</a></p><div id="div_dbscsnv" style="display:none;word-wrap:break-word;"><p>  Please cite the dbscSNV publication alongside Ensembl VEP if you use this resource:
  <a href="http://nar.oxfordjournals.org/content/42/22/13534">http://nar.oxfordjournals.org/content/42/22/13534</a></p><p>  The Bio::DB::HTS perl library or tabix utility must be installed in your path
  to use this plugin. The dbscSNV data file can be downloaded from
  <a href="https://sites.google.com/site/jpopgen/dbNSFP">https://sites.google.com/site/jpopgen/dbNSFP</a>.</p><p>  The file must be processed and indexed by tabix before use by this plugin.
  dbscSNV1.1 has coordinates for both GRCh38 and GRCh37; the file must be
  processed differently according to the assembly you use.</p><p><pre class="code sh_sh">wget <a href="ftp://dbnsfp:dbnsfp@dbnsfp.softgenetics.com/dbscSNV1.1.zip">ftp://dbnsfp:dbnsfp@dbnsfp.softgenetics.com/dbscSNV1.1.zip</a>
unzip dbscSNV1.1.zip
head -n1 dbscSNV1.1.chr1 &gt h
</pre><p>  # GRCh38
<pre class="code sh_sh">cat dbscSNV1.1.chr* | grep -v ^chr | sort -k5,5 -k6,6n | cat h - | awk '$5 != "."' | bgzip -c &gt dbscSNV1.1_GRCh38.txt.gz
tabix -s 5 -b 6 -e 6 -c c dbscSNV1.1_GRCh38.txt.gz
</pre><p>  # GRCh37
<pre class="code sh_sh">cat dbscSNV1.1.chr* | grep -v ^chr | cat h - | bgzip -c &gt dbscSNV1.1_GRCh37.txt.gz
tabix -s 1 -b 2 -e 2 -c c dbscSNV1.1_GRCh37.txt.gz
</pre><p>  Note that in the last command we tell tabix that the header line starts with "c";
  this may change to the default of "#" in future versions of dbscSNV.</p><p>  Tabix also allows the data file to be hosted on a remote server. This plugin is
  fully compatible with such a setup - simply use the URL of the remote file:</p><p><pre class="code sh_sh">--plugin dbscSNV,<a href="http://my.files.com/dbscSNV.txt.gz">http://my.files.com/dbscSNV.txt.gz</a>
</pre><p>  Note that transcript sequences referred to in dbscSNV may be out of sync with
  those in the latest release of Ensembl; this may lead to discrepancies with
  scores retrieved from other sources.</p><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv dbscSNV.pm ~/.vep/Plugins
./vep -i variations.vcf --plugin dbscSNV,/path/to/dbscSNV1.1_GRCh38.txt.gz
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#FF0000">Splicing predictions</div></td><td>-</td><td>Ensembl</td></tr>
<tr id="DeNovo" class="bg2 plugin_row" data-category="variant_data"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/DeNovo.pm">DeNovo</a></div></td><td><p> An Ensembl  VEP plugin that identifies de novo variants in a VCF file.
 The plugin is not compatible with JSON output format. <a class="button" href="#denovo" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('denovo');" id="a_denovo">more</a></p><div id="div_denovo" style="display:none;word-wrap:break-word;"><p> Options are passed to the plugin as key=value pairs:<table class="ss"><thead><tr><th>Argument</th><th>Description</th></tr></thead><tbody>
<tr class="bg1"><td><pre>ped</pre></td><td>Path to PED file (mandatory)                       The file is tab or white-space delimited with five mandatory columns:
<ul><li>family ID</li><li>individual ID</li><li>paternal ID</li><li>maternal ID</li><li>sex</li><li>phenotype (optional)
<tr class="bg2"><td><pre></li></ul><p> report_dir</pre></td><td>Write files in report_dir (optional) 
<tr class="bg1"><td><pre>full_report</pre></td><td>Set to 1 to report all types of variants (optional)                       By default, the plugin only reports de novo variants.</p><p></td></tr></tbody></table><p>
 The plugin can then be run:
<pre class="code sh_sh">./vep -i variations.vcf --plugin DeNovo,ped=samples.ped
./vep -i variations.vcf --plugin DeNovo,ped=samples.ped,report_dir=path/to/dir
./vep -i variations.vcf --plugin DeNovo,ped=samples.ped,report_dir=path/to/dir,full_report=1
</pre><p>
</ul><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv DeNovo.pm ~/.vep/Plugins
./vep -i variations.vcf --plugin DeNovo,ped=samples.ped
./vep -i variations.vcf --plugin DeNovo,ped=samples.ped,full_report=1
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#B22222">Variant data</div></td><td><ul style="padding-left:1em"><li><a href="https://metacpan.org/pod/List::MoreUtils" rel="external">List::MoreUtils</a> qw(uniq)</li><li><a href="https://metacpan.org/pod/Cwd" rel="external">Cwd</a></li></ul></td><td>Ensembl</td></tr>
<tr id="DosageSensitivity" class="bg1 plugin_row" data-category="gene_tolerance_to_change"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/DosageSensitivity.pm">DosageSensitivity</a></div></td><td><p> An Ensembl VEP plugin that retrieves haploinsufficiency and triplosensitivity probability scores
 for affected genes from a dosage sensitivity catalogue published in paper -
 <a href="https://www.sciencedirect.com/science/article/pii/S0092867422007887">https://www.sciencedirect.com/science/article/pii/S0092867422007887</a> <a class="button" href="#dosagesensitivity" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('dosagesensitivity');" id="a_dosagesensitivity">more</a></p><div id="div_dosagesensitivity" style="display:none;word-wrap:break-word;"><p> Please cite the above publication alongside Ensembl VEP if you use this resource.</p><p> This plugin returns two scores:
<ul><li>pHaplo score gives the probability of a gene being haploinsufficient (deletion intolerant)</li><li>pTriplo score gives the probability of a gene being triploinsensitive (duplication intolerant)
</li></ul><p> Pre-requisites:
 You need the compressed tsv file containing the dosage sensitivity score. The file
 Collins_rCNV_2022.dosage_sensitivity_scores.tsv.gz can be downloaded from here - 
 <a href="https://zenodo.org/record/6347673/files/Collins_rCNV_2022.dosage_sensitivity_scores.tsv.gz">https://zenodo.org/record/6347673/files/Collins_rCNV_2022.dosage_sensitivity_scores.tsv.gz</a></p><p> Options are passed to the plugin as key=value pairs:<table class="ss"><thead><tr><th>Argument</th><th>Description</th></tr></thead><tbody>
<tr class="bg1"><td><pre>file</pre></td><td>(mandatory) compressed tsv file containing dosage sensitivity scores <tr class="bg2"><td><pre>cover</pre></td><td>set value to 1 (0 by default) to report scores only if the variant  covers the affected feature completely (e.g. - a CNV that duplicates the gene). 
 The feature is a gene if using --database otherwise it is a transcript.</p><p></ul><p></td></tr></tbody></table><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv DosageSensitivity.pm ~/.vep/Plugins
./vep -i variations.vcf --plugin DosageSensitivity,file=/FULL_PATH_TO/Collins_rCNV_2022.dosage_sensitivity_scores.tsv.gz
./vep -i variations.vcf --plugin DosageSensitivity,file=/FULL_PATH_TO/Collins_rCNV_2022.dosage_sensitivity_scores.tsv.gz,cover=1
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#00DD66">Gene tolerance to change</div></td><td>-</td><td>Ensembl</td></tr>
<tr id="Downstream" class="bg2 plugin_row" data-category="nearby_features"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/Downstream.pm">Downstream</a></div></td><td><p> This is a plugin for the Ensembl Variant Effect Predictor (VEP) that
 predicts the downstream effects of a frameshift variant on the protein
 sequence of a transcript. It provides the predicted downstream protein
 sequence (including any amino acids overlapped by the variant itself),
 and the change in length relative to the reference protein. <a class="button" href="#downstream" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('downstream');" id="a_downstream">more</a></p><div id="div_downstream" style="display:none;word-wrap:break-word;"><p> Note that changes in splicing are not predicted - only the existing
 translateable (i.e. spliced) sequence is used as a source of
 translation. Any variants with a splice site consequence type are
 ignored.</p><p> If Ensembl VEP is run in offline mode using the flag --offline, a FASTA file is required.
 See: <a href="https://www.ensembl.org/info/docs/tools/vep/script/vep_cache.html#fasta">https://www.ensembl.org/info/docs/tools/vep/script/vep_cache.html#fasta</a>
 Sequence may be incomplete without a FASTA file or database connection.</p><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv Downstream.pm ~/.vep/Plugins
./vep -i variations.vcf --plugin Downstream
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#E75480">Nearby features</div></td><td>-</td><td>Ensembl</td></tr>
<tr id="Draw" class="bg1 plugin_row" data-category="visualisation"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/Draw.pm">Draw</a></div></td><td><p> An Ensembl VEP plugin that draws pictures of the transcript model showing the
 variant location. <a class="button" href="#draw" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('draw');" id="a_draw">more</a></p><div id="div_draw" style="display:none;word-wrap:break-word;"><p> Takes five optional paramters:</p><p><ol><li value="1">File name stem for images</li><li value="2">Image width in pixels (default: 1000px)</li><li value="3">Image height in pixels (default: 100px)</li><li value="4">Transcript ID - only draw images for this transcript</li><li value="5">Variant ID - only draw images for this variant 
</li></ol><p> e.g.</p><p><pre class="code sh_sh">./vep -i variations.vcf --plugin Draw,myimg,2000,100
</pre><p> Images are written to <kbd>[file_stem]_[transcript_id]_[variant_id].png</kbd></p><p> Requires GD library installed to run.</p><p></ol><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv Draw.pm ~/.vep/Plugins
./vep -i variations.vcf --plugin Draw
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#008000">Visualisation</div></td><td><ul style="padding-left:1em"><li><a href="https://metacpan.org/pod/GD::Polygon" rel="external">GD::Polygon</a></li><li><a href="https://metacpan.org/pod/GD" rel="external">GD</a></li></ul></td><td>Ensembl</td></tr>
<tr id="Enformer" class="bg2 plugin_row" data-category="regulatory_impact"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/Enformer.pm">Enformer</a></div></td><td><p> This is a plugin for the Ensembl Variant Effect Predictor (VEP) that adds pre-calculated Enformer predictions of variant impact on chromatin and gene expression. <a class="button" href="#enformer" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('enformer');" id="a_enformer">more</a></p><div id="div_enformer" style="display:none;word-wrap:break-word;"><p> The predictions have been aggregated across all 896 spatial bins to generate 5313 features corresponding to track prediction changes in differnet assays and cell types.</p><p> This plugin is available for GRCh37 and GRCh38</p><p> Please cite the Enformer publication alongside Ensembl VEP if you use this resource:
 <a href="https://www.nature.com/articles/s41592-021-01252-x">https://www.nature.com/articles/s41592-021-01252-x</a></p><p> GRCh38 scores were lifted over using CrossMap from the Enformer scores available here - <a href="https://console.cloud.google.com/storage/browser/dm-enformer/variant-scores/1000-genomes/enformer">https://console.cloud.google.com/storage/browser/dm-enformer/variant-scores/1000-genomes/enformer</a></p><p> Enformer scores can be downloaded from <a href="https://ftp.ensembl.org/pub/current_variation/Enformer">https://ftp.ensembl.org/pub/current_variation/Enformer</a> for GRCh37 and GRCh38.</p><p> The plugin can then be run as default to retrieve SAD (SNP Activity Difference (SAD) and SAR (Same as SAD, by computing np.log2(1 + model(alternate_sequence)) - np.log2(1 + model(reference_sequence)) scores from Enforme :
<pre class="code sh_sh">./vep -i variations.vcf --assembly GRCh38 --plugin Enformer,file=/path/to/Enformer/data.vcf.gz
</pre><p> or run with option to only retrieve the SAD (SNP Activity Difference (SAD) scores - main variant effect score computed as model(alternate_sequence) - model(reference_sequence) score 
<pre class="code sh_sh">./vep -i variations.vcf --assembly GRCh38 --plugin Enformer,file=/path/to/Enformer/data.vcf.gz,SAD=1 
</pre><p> or run with option to only retrieve the SAR (Same as SAD, by computing np.log2(1 + model(alternate_sequence)) - np.log2(1 + model(reference_sequence)) score 
<pre class="code sh_sh">./vep -i variations.vcf --assembly GRCh38 --plugin Enformer,file=/path/to/Enformer/data.vcf.gz,SAR=1 
</pre><p> or run with option to also retrieve the principal component scores which are a reduced representation of a much bigger vector of the SAD and SAR after using principal component analysis (PCA)
<pre class="code sh_sh">./vep -i variations.vcf --assembly GRCh38 --plugin Enformer,file=/path/to/Enformer/data.vcf.gz,PC=1 
</pre><p>
 The tabix utility must be installed in your path to use this plugin.
 Check <a href="https://github.com/samtools/htslib.git">https://github.com/samtools/htslib.git</a> for instructions.</p><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv Enformer.pm ~/.vep/Plugins
./vep -i variations.vcf --plugin Enformer,file=Enformer_grch38.vcf.gz</p><p></pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#2200FF">Regulatory impact</div></td><td>-</td><td>Ensembl</td></tr>
<tr id="EVE" class="bg1 plugin_row" data-category="pathogenicity_predictions"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/EVE.pm">EVE</a></div></td><td><p> This is a plugin for the Ensembl Variant Effect Predictor (VEP) that
 adds information from EVE (evolutionary model of variant effect). <a class="button" href="#eve" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('eve');" id="a_eve">more</a></p><div id="div_eve" style="display:none;word-wrap:break-word;"><p> This plugin only report EVE scores for input variants
 and does not merge input lines to report on adjacent variants.
 It is only available for GRCh38.</p><p> Please cite EVE publication alongside Ensembl VEP if you use this resource:
 <a href="https://www.nature.com/articles/s41586-021-04043-8">https://www.nature.com/articles/s41586-021-04043-8</a></p><p><pre class="code sh_sh">###################################################
# Bash script to merge all VCFs from EVE dataset. #
###################################################
### BEGIN
# EVE input file can be downloaded from <a href="https://evemodel.org/api/proteins/bulk/download/">https://evemodel.org/api/proteins/bulk/download/</a> 
# Input: VCF files by protein (vcf_files_missense_mutations inside zip folder)
# Output: Compressed Merged VCF file (vcf.gz) + index file (.tbi)
DATA_FOLDER=<kbd>/&ltPATH-TO&gt/vcf_files_missense_mutations</kbd> # Fill this line
OUTPUT_FOLDER=<kbd>/&ltPATH-TO&gt/eve_plugin</kbd> # Fill this line
OUTPUT_NAME=<kbd>eve_merged.vcf</kbd> # Default output name
# Get header from first VCF
cat `ls ${DATA_FOLDER}/*vcf | head -n1` &gt header
# Get variants from all VCFs and add to a single-file
ls ${DATA_FOLDER}/*vcf | while read VCF; do grep -v '^#' ${VCF} &gt&gt variants; done
# Merge Header + Variants in a single file
cat header variants | \
awk '$1 ~ /^#/ {print $0;next} {print $0 | "sort -k1,1V -k2,2n"}' &gt ${OUTPUT_FOLDER}/${OUTPUT_NAME};
# Remove temporary files
rm header variants
# Compress and index
bgzip ${OUTPUT_FOLDER}/${OUTPUT_NAME};
# If not installed, use: sudo apt install tabix
tabix ${OUTPUT_FOLDER}/${OUTPUT_NAME}.gz;
### END
</pre>
<p><h2>Usage examples:</h2> <pre class="code sh_sh">
cp EVE.pm ${HOME}/.vep/Plugins
./vep -i variations.vcf --plugin EVE,file=/path/to/eve/data.vcf.gz # By default, Class75 is used.
./vep -i variations.vcf --plugin EVE,file=/path/to/eve/data.vcf.gz,class_number=60
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#1E90FF">Pathogenicity predictions</div></td><td>-</td><td>Ensembl</td></tr>
<tr id="FATHMM" class="bg2 plugin_row" data-category="pathogenicity_predictions"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/FATHMM.pm">FATHMM</a></div></td><td><p> An Ensembl VEP plugin that gets FATHMM scores and predictions for missense variants. <a class="button" href="#fathmm" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('fathmm');" id="a_fathmm">more</a></p><div id="div_fathmm" style="display:none;word-wrap:break-word;"><p> You will need the fathmm.py script and its dependencies (Python, Python
 MySQLdb). You should create a "config.ini" file in the same directory as the
 fathmm.py script with the database connection options. More information about
 how to set up FATHMM can be found on the FATHMM website at
 <a href="https://github.com/HAShihab/fathmm">https://github.com/HAShihab/fathmm</a></p><p> A typical installation could consist of:</p><p><pre class="code sh_sh">wget <a href="https://raw.github.com/HAShihab/fathmm/master/cgi-bin/fathmm.py">https://raw.github.com/HAShihab/fathmm/master/cgi-bin/fathmm.py</a>
wget <a href="http://fathmm.biocompute.org.uk/database/fathmm.v2.3.SQL.gz">http://fathmm.biocompute.org.uk/database/fathmm.v2.3.SQL.gz</a>
gunzip fathmm.v2.3.SQL.gz
mysql -h[host] -P[port] -u[user] -p[pass] -e"CREATE DATABASE fathmm"
mysql -h[host] -P[port] -u[user] -p[pass] -Dfathmm &lt fathmm.v2.3.SQL
echo -e "[DATABASE]\nHOST = [host]\nPORT = [port]\nUSER = [user]\nPASSWD = [pass]\nDB = fathmm\n" &gt config.ini
</pre><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv FATHMM.pm ~/.vep/Plugins
./vep -i variations.vcf --plugin FATHMM,"python2 /path/to/fathmm/fathmm.py"
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#1E90FF">Pathogenicity predictions</div></td><td>-</td><td>Ensembl</td></tr>
<tr id="FATHMM_MKL" class="bg1 plugin_row" data-category="pathogenicity_predictions"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/FATHMM_MKL.pm">FATHMM_MKL</a></div></td><td><p> An Ensembl VEP plugin that retrieves FATHMM-MKL scores for variants from a tabix-indexed
 FATHMM-MKL data file. <a class="button" href="#fathmm_mkl" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('fathmm_mkl');" id="a_fathmm_mkl">more</a></p><div id="div_fathmm_mkl" style="display:none;word-wrap:break-word;"><p> See <a href="https://github.com/HAShihab/fathmm-MKL">https://github.com/HAShihab/fathmm-MKL</a> for details.</p><p> NB: The currently available data file is for GRCh37 only.</p><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv FATHMM_MKL.pm ~/.vep/Plugins
./vep -i input.vcf --plugin FATHMM_MKL,fathmm-MKL_Current.tab.gz
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#1E90FF">Pathogenicity predictions</div></td><td>-</td><td>Ensembl</td></tr>
<tr id="FlagLRG" class="bg2 plugin_row" data-category="external_id"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/FlagLRG.pm">FlagLRG</a></div></td><td><p> A VEP plugin that retrieves the LRG ID matching either the RefSeq or Ensembl
 transcript IDs. <a class="button" href="#flaglrg" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('flaglrg');" id="a_flaglrg">more</a></p><div id="div_flaglrg" style="display:none;word-wrap:break-word;"><p> You can obtain the <kbd>list_LRGs_transcripts_xrefs.txt</kbd> using:
<pre class="code sh_sh">wget <a href="https://ftp.ebi.ac.uk/pub/databases/lrgex/list_LRGs_transcripts_xrefs.txt">https://ftp.ebi.ac.uk/pub/databases/lrgex/list_LRGs_transcripts_xrefs.txt</a>
</pre><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv FlagLRG.pm ~/.vep/Plugins
./vep -i variants.vcf --plugin FlagLRG,/path/to/list_LRGs_transcripts_xrefs.txt
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#333333">External ID</div></td><td><a href="https://metacpan.org/pod/Text::CSV" rel="external">Text::CSV</a></td><td>Stephen Kazakoff</td></tr>
<tr id="FunMotifs" class="bg1 plugin_row" data-category="motif"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/FunMotifs.pm">FunMotifs</a></div></td><td><p> This is a plugin for the Ensembl Variant Effect Predictor (VEP) that
 adds tissue-specific transcription factor motifs from FunMotifs to Ensembl VEP output. <a class="button" href="#funmotifs" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('funmotifs');" id="a_funmotifs">more</a></p><div id="div_funmotifs" style="display:none;word-wrap:break-word;"><p> Please cite the FunMotifs publication alongside Ensembl VEP if you use this resource. 
 The preprint can be found at: <a href="https://www.biorxiv.org/content/10.1101/683722v1">https://www.biorxiv.org/content/10.1101/683722v1</a></p><p> FunMotifs files can be downloaded from: <a href="http://bioinf.icm.uu.se:3838/funmotifs/">http://bioinf.icm.uu.se:3838/funmotifs/</a>
 At the time of writing, all BED files found through this link support GRCh37, 
 however other assemblies are supported by the plugin if an appropriate BED file
 is supplied.</p><p> The tabix utility must be installed in your path to use this plugin.</p><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv FunMotifs.pm ~/.vep/Plugins
./vep -i variations.vcf --plugin FunMotifs,/path/to/funmotifs/all_tissues.bed.gz,uterus
./vep -i variations.vcf --plugin FunMotifs,/path/to/funmotifs/blood.funmotifs_sorted.bed.gz,fscore,dnase_seq</p><p>Parameters Required:</p><p>[0] : FunMotifs BED file
[1]+ : List of columns to include within the Ensembl VEP output (e.g. fscore, skin, contactingdomain)</p><p></pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#DAA520">Motif</div></td><td>-</td><td>Ensembl</td></tr>
<tr id="G2P" class="bg2 plugin_row" data-category="phenotype_data_and_citations"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/G2P.pm">G2P</a></div><div style="margin-top:6px"><small>gene2phenotype</small></div></td><td><p> An Ensembl VEP plugin that uses G2P allelic requirements to assess variants in genes
 for potential phenotype involvement. <a class="button" href="#g2p" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('g2p');" id="a_g2p">more</a></p><div id="div_g2p" style="display:none;word-wrap:break-word;"><p> The plugin has multiple configuration options, though minimally requires only
 the CSV file of G2P data.
 This Plugin is available for GRCh38 and GRCh37.</p><p> For further information see:
 Thormann A, Halachev M, McLaren W, et al. Flexible and scalable diagnostic filtering of genomic variants using G2P with Ensembl VEP.
 Nature Communications. 2019 May;10(1):2373. <a rel="external" href="https://doi.org/10.1038/s41467-019-10016-3">doi:10.1038/s41467-019-10016-3</a>. PMID: 31147538; PMCID: PMC6542828.</p><p> Options are passed to the plugin as key=value pairs, (defaults in parentheses):<table class="ss"><thead><tr><th>Argument</th><th>Description</th></tr></thead><tbody>
<tr class="bg1"><td><pre>file</pre></td><td>Path to G2P data file. The file needs to be uncompressed. <ul><li>Download from <a href="https://www.ebi.ac.uk/gene2phenotype/downloads">https://www.ebi.ac.uk/gene2phenotype/downloads</a></li><li>Download from PanelApp  
<tr class="bg2"><td><pre></li></ul><p> variant_include_list</pre></td><td>A list of variants to include even if variants do not pass allele                          frequency filtering. The include list needs to be a sorted, bgzipped and
                         tabixed VCF file.</p><p><tr class="bg1"><td><pre>af_monoallelic</pre></td><td>maximum allele frequency for inclusion for monoallelic genes (0.0001) 
<tr class="bg2"><td><pre>af_biallelic</pre></td><td>maximum allele frequency for inclusion for biallelic genes (0.005) <tr class="bg1"><td><pre>confidence_levels</pre></td><td>Confidence levels include: definitive, strong, moderate, limited                          Former confidence terms are still supported: confirmed, probable, possible, both RD and IF.
                         Separate multiple values with <kbd>&</kbd>.
                         <a href="https://www.ebi.ac.uk/gene2phenotype/terminology">https://www.ebi.ac.uk/gene2phenotype/terminology</a>
                         Default levels are confirmed and probable.
<tr class="bg2"><td><pre>all_confidence_levels</pre></td><td>Set to 1 to include all confidence levels                          Setting the value to 1 will overwrite any confidence levels provided with the
                         confidence_levels option.
<tr class="bg1"><td><pre>af_from_vcf</pre></td><td>set value to 1 to include allele frequencies from VCF file.                          Specifiy the list of reference populations to include with <kbd>--af_from_vcf_keys</kbd>
<tr class="bg2"><td><pre>af_from_vcf_keys</pre></td><td>VCF collections used for annotating variant alleles with observed                          allele frequencies. Allele frequencies are retrieved from VCF files. If
                         af_from_vcf is set to 1 but no VCF collections are specified with <kbd>--af_from_vcf_keys</kbd>
                         all available VCF collections are included. 
                         Available VCF collections: <kbd>topmed</kbd>, <kbd>uk10k</kbd>, <kbd>gnomADe</kbd>, <kbd>gnomADe_r2.1.1</kbd>, <kbd>gnomADg</kbd>, <kbd>gnomADg_v3.1.2</kbd>.
                         Separate multiple values with <kbd>&</kbd>.
                         VCF collections contain the following populations: 
<ul><li><kbd>topmed</kbd> - TOPMed (available for GRCh37 and GRCh38).</li><li><kbd>uk10k</kbd> - ALSPAC, TWINSUK (available for GRCh37 and GRCh38).</li><li><kbd>gnomADe</kbd> & <kbd>gnomADe_r2.1.1</kbd> - gnomADe:AFR, gnomADe:ALL, gnomADe:AMR, gnomADe:ASJ, gnomADe:EAS, gnomADe:FIN, gnomADe:NFE, gnomADe:OTH, gnomADe:SAS (for GRCh37 and GRCh38 respectively).</li><li><kbd>gnomADg</kbd> & <kbd>gnomADg_v3.1.2</kbd> - gnomADg:AFR, gnomADg:ALL, gnomADg:AMR, gnomADg:ASJ, gnomADg:EAS, gnomADg:FIN, gnomADg:NFE, gnomADg:OTH (for GRCh37 and GRCh38 respectively).&nbsp;                         Need to use <kbd>af_from_vcf</kbd> parameter to use this option. 
<tr class="bg1"><td><pre>&nbsp; default_af</pre></td><td>default frequency of the input variant if no frequency data is &nbsp;                         found (0). This determines whether such variants are included;
&nbsp;                         the value of 0 forces variants with no frequency data to be
&nbsp;                         included as this is considered equivalent to having a frequency
&nbsp;                         of 0. Set to 1 (or any value higher than <kbd>af</kbd>) to exclude them.
<tr class="bg2"><td><pre>&nbsp; types</pre></td><td>SO consequence types to include. Separate multiple values with <kbd>&</kbd> &nbsp;                         (splice_donor_variant, splice_acceptor_variant, stop_gained,
&nbsp;                         frameshift_variant, stop_lost, initiator_codon_variant,
&nbsp;                         inframe_insertion, inframe_deletion,missense_variant,
&nbsp;                         coding_sequence_variant, start_lost,transcript_ablation,
&nbsp;                         transcript_amplification, protein_altering_variant)</p><p><tr class="bg1"><td><pre></li></ul><p>  log_dir</pre></td><td>write stats to log files in log_dir 
<tr class="bg2"><td><pre>txt_report</pre></td><td>write all G2P complete genes and attributes to txt file 
<tr class="bg1"><td><pre>html_report</pre></td><td>write all G2P complete genes and attributes to html file 
<tr class="bg2"><td><pre>filter_by_gene_symbol</pre></td><td>set to 1 if filter by gene symbol.                           Do not set if filtering by HGNC_id.
                          This option is set to 1 when using PanelApp files. </p><p><tr class="bg1"><td><pre>only_mane</pre></td><td>set to 1 to ignore transcripts that are not MANE                          N/B - Information may be lost if this option is used.</p><p></td></tr></tbody></table><p> For more information - <a href="https://www.ebi.ac.uk/gene2phenotype/g2p_vep_plugin">https://www.ebi.ac.uk/gene2phenotype/g2p_vep_plugin</a></p><p> Example:
<pre class="code sh_sh">--plugin G2P,file=G2P.csv,af_monoallelic=0.05,types=stop_gained&frameshift_variant
--plugin G2P,file=G2P.csv,af_monoallelic=0.05,af_from_vcf=1
--plugin G2P,file=G2P.csv,af_from_vcf=1,af_from_vcf_keys=<kbd>topmed&gnomADe_r2.1.1</kbd>
--plugin G2P,file=G2P.csv,af_from_vcf=1,af_from_vcf_keys=<kbd>topmed&gnomADe_r2.1.1</kbd>,confidence_levels='confirmed&probable&both RD and IF' 
--plugin G2P,file=G2P.csv
</pre><p></ul><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv G2P.pm ~/.vep/Plugins
./vep -i variations.vcf --plugin G2P,file=/path/to/G2P.csv
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#556688">Phenotype data and citations</div></td><td><ul style="padding-left:1em"><li><a href="https://metacpan.org/pod/List::Util" rel="external">List::Util</a> qw(any)</li><li><a href="https://metacpan.org/pod/Text::CSV" rel="external">Text::CSV</a></li><li><a href="https://metacpan.org/pod/Scalar::Util" rel="external">Scalar::Util</a> qw(looks_like_number)</li><li><a href="https://metacpan.org/pod/FileHandle" rel="external">FileHandle</a></li><li><a href="https://metacpan.org/pod/Cwd" rel="external">Cwd</a></li></ul></td><td>Ensembl</td></tr>
<tr id="GeneBe" class="bg1 plugin_row" data-category="variant_data"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/GeneBe.pm">GeneBe</a></div></td><td><p> A user-contributed VEP plugin that retrieves automatic ACMG variant classification data from
 <a href="https://genebe.net/">https://genebe.net/</a> <a class="button" href="#genebe" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('genebe');" id="a_genebe">more</a></p><div id="div_genebe" style="display:none;word-wrap:break-word;"><p> Please cite the GeneBe publication alongside Ensembl VEP if you use this resource:
 <a href="https://onlinelibrary.wiley.com/doi/10.1111/cge.14516">https://onlinelibrary.wiley.com/doi/10.1111/cge.14516</a> .</p><p> Please be advised that the GeneBe API is freely accessible for academic purposes only, with a limited
 number of queries per day, albeit at a high threshold. Kindly utilize this resource judiciously
 to ensure its availability for others. For further information, please visit <a href="https://genebe.net/about/api">https://genebe.net/about/api</a>.</p><p> In order to extend your daily limits please make an account on <a href="https://genebe.net/">https://genebe.net/</a> and use your username and API-key as follows:
<pre class="code sh_sh">./vep -i variations.vcf --plugin GeneBe,user=example@email.com,password=your_api_key
</pre><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv GeneBe.pm ~/.vep/Plugins
./vep -i variations.vcf --plugin GeneBe
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#B22222">Variant data</div></td><td><a href="https://metacpan.org/pod/JSON" rel="external">JSON</a></td><td><ul style="padding-left:1em"><li>Ensembl</li><li>Piotr Stawinski</li></ul></td></tr>
<tr id="GeneSplicer" class="bg2 plugin_row" data-category="splicing_predictions"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/GeneSplicer.pm">GeneSplicer</a></div></td><td><p> This is a plugin for the Ensembl Variant Effect Predictor (VEP) that
 runs GeneSplicer (<a href="https://ccb.jhu.edu/software/genesplicer/">https://ccb.jhu.edu/software/genesplicer/</a>) to get
 splice site predictions. <a class="button" href="#genesplicer" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('genesplicer');" id="a_genesplicer">more</a></p><div id="div_genesplicer" style="display:none;word-wrap:break-word;"><p> It evaluates a tract of sequence either side of and including the
 variant, both in reference and alternate states. The amount of
 sequence included either side defaults to 100bp, but can be modified
 by passing e.g. "context=50" as a parameter to the plugin.</p><p> You will need to download the GeneSplicer binary and data from
 <a href="ftp://ftp.ccb.jhu.edu/pub/software/genesplicer/GeneSplicer.tar.gz">ftp://ftp.ccb.jhu.edu/pub/software/genesplicer/GeneSplicer.tar.gz</a>. Extract the
 folder using:</p><p><pre class="code sh_sh">tar -xzf GeneSplicer.tar.gz
</pre><p> GeneSplicer comes with precompiled binaries for multiple systems. If the
 provided binaries do not run, compile <kbd>genesplicer</kbd> from source:</p><p><pre class="code sh_sh">cd $GS/sources
# if macOS, run this step
[[ $(uname -s) == "Darwin" ]] && perl -pi -e "s/^main  /int main  /" genesplicer.cpp
make
cd -
./vep [options] --plugin GeneSplicer,$GS/sources/genesplicer,$GS/human
</pre>
 Predicted splicing regions that overlap the variant are reported in the output
 with a <kbd>/</kbd>-separated string (e.g., <kbd>loss/acceptor/727006-727007/High/16.231924</kbd>)
 consisting of the following data by this order:</p><p><ol><li value="1">state (no_change, diff, gain, loss)</li><li value="2">type (donor, acceptor)</li><li value="3">coordinates (start-end)</li><li value="4">confidence (Low, Medium, High)</li><li value="5">score
</li></ol><p> If multiple sites are predicted, their reports are separated by ",".</p><p> For diff, the confidence and score for both the reference and alternate
 sequences is reported as REF-ALT, such as
 <kbd>diff/donor/621915-621914/Medium-Medium/7.020731-6.988368</kbd>.</p><p> Several key=value parameters can be modified in the the plugin string:<table class="ss"><thead><tr><th>Argument</th><th>Description</th></tr></thead><tbody>
<tr class="bg1"><td><pre>training</pre></td><td>(mandatory) directory to species-specific training data, such as               <kbd>GeneSplicer/human</kbd>
<tr class="bg2"><td><pre>binary</pre></td><td>path to <kbd>genesplicer</kbd> binary (default: <kbd>genesplicer</kbd>) <tr class="bg1"><td><pre>context</pre></td><td>change the amount of sequence added either side of               the variant (default: 100bp)
<tr class="bg2"><td><pre>tmpdir</pre></td><td>change the temporary directory used (default: <kbd>/tmp</kbd>) <tr class="bg1"><td><pre>cache_size</pre></td><td>change how many sequences' scores are cached in memory               (default: 50)</p><p></td></tr></tbody></table><p> Example:
<pre class="code sh_sh">--plugin GeneSplicer,binary=$GS/bin/linux/genesplicer,training=$GS/human,context=200,tmpdir=/mytmp
</pre><p> When using the Ensembl VEP Docker/Singularity, the <kbd>binary</kbd> argument can be ommitted, as
 the <kbd>genesplicer</kbd> command is exported in the $PATH variable and is thus
 automatically detected by the plugin:
<pre class="code sh_sh">--plugin GeneSplicer,training=$GS/human,context=200,tmpdir=/mytmp
</pre><p></ol><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv GeneSplicer.pm ~/.vep/Plugins
./vep -i variants.vcf --plugin GeneSplicer,binary=$GS/bin/linux/genesplicer,training=$GS/human
./vep -i variants.vcf --plugin GeneSplicer,binary=$GS/bin/linux/genesplicer,training=$GS/human,context=200,tmpdir=/mytmp</p><p># Ensembl VEP Docker/Singularity: if 'genesplicer' is a command available in $PATH,
# there is no need to specify the location of the binary
./vep -i variants.vcf --plugin GeneSplicer,training=$GS/human
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#FF0000">Splicing predictions</div></td><td><a href="https://metacpan.org/pod/Digest::MD5" rel="external">Digest::MD5</a> qw(md5_hex)</td><td>Ensembl</td></tr>
<tr id="Geno2MP" class="bg1 plugin_row" data-category="phenotype_data_and_citations"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/Geno2MP.pm">Geno2MP</a></div></td><td><p> An Ensembl VEP plugin that adds information from Geno2MP, a web-accessible database of
 rare variant genotypes linked to phenotypic information. <a class="button" href="#geno2mp" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('geno2mp');" id="a_geno2mp">more</a></p><div id="div_geno2mp" style="display:none;word-wrap:break-word;"><p> Parameters can be set using a key=value system:<table class="ss"><thead><tr><th>Argument</th><th>Description</th></tr></thead><tbody><tr class="bg1"><td><pre>file</pre></td><td>VCF file containing Geno2MP data <tr class="bg2"><td><pre>cols</pre></td><td>colon-delimited list of Geno2MP columns to return from INFO fields           (by default it only returns the column HPO_CT)
<tr class="bg1"><td><pre>url</pre></td><td>build and return URL to Geno2MP variant page (boolean; 0 by default);           the variant location in Geno2MP website is based on GRCh37 coordinates</p><p></td></tr></tbody></table><p> Please cite Geno2MP alongside Ensembl VEP if you use this resource:
 Geno2MP, NHGRI/NHLBI University of Washington-Center for Mendelian Genomics (UW-CMG), Seattle, WA
 (URL: <a href="http://geno2mp.gs.washington.edu">http://geno2mp.gs.washington.edu</a> [date (month, yr) accessed]).</p><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
cp Geno2MP.pm ${HOME}/.vep/Plugins
./vep -i variations.vcf --plugin Geno2MP,file=/path/to/Geno2MP/data.vcf.gz</p><p># Return more columns from Geno2MP VCF file
./vep -i variations.vcf --plugin Geno2MP,file=/path/to/Geno2MP/data.vcf.gz,cols=HPO_CT:FXN:nhomalt_male_aff:nhomalt_male_unaff</p><p># Build and return Geno2MP URL based on GRCh37 variant location
./vep -i variations.vcf --plugin Geno2MP,file=/path/to/Geno2MP/data.vcf.gz,url=1
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#556688">Phenotype data and citations</div></td><td>-</td><td>Ensembl</td></tr>
<tr id="gnomADc" class="bg2 plugin_row" data-category="frequency_data"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/gnomADc.pm">gnomADc</a></div></td><td><p> An Ensembl VEP plugin that retrieves gnomAD annotation from either the genome
 or exome coverage files, available here:
   <a href="https://gnomad.broadinstitute.org/downloads">https://gnomad.broadinstitute.org/downloads</a> <a class="button" href="#gnomadc" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('gnomadc');" id="a_gnomadc">more</a></p><div id="div_gnomadc" style="display:none;word-wrap:break-word;"><p> To download the gnomad coverage file in TSV format:
  for Assembly GRCh37: 
   gnomad genomes:
<pre class="code sh_sh">wget <a href="https://storage.googleapis.com/gcp-public-data--gnomad/release/2.1/coverage/genomes/gnomad.genomes.coverage.summary.tsv.bgz">https://storage.googleapis.com/gcp-public-data--gnomad/release/2.1/coverage/genomes/gnomad.genomes.coverage.summary.tsv.bgz</a> --no-check-certificate
</pre><p>   gnomad exomes:
<pre class="code sh_sh">wget <a href="https://storage.googleapis.com/gcp-public-data--gnomad/release/2.1/coverage/exomes/gnomad.exomes.coverage.summary.tsv.bgz">https://storage.googleapis.com/gcp-public-data--gnomad/release/2.1/coverage/exomes/gnomad.exomes.coverage.summary.tsv.bgz</a> --no-check-certificate
</pre><p>  for Assembly GRCh38: 
   gnomad genomes: 
<pre class="code sh_sh">wget <a href="https://storage.googleapis.com/gcp-public-data--gnomad/release/3.0.1/coverage/genomes/gnomad.genomes.r3.0.1.coverage.summary.tsv.bgz">https://storage.googleapis.com/gcp-public-data--gnomad/release/3.0.1/coverage/genomes/gnomad.genomes.r3.0.1.coverage.summary.tsv.bgz</a> --no-check-certificate 
</pre><p> Necessary before using the plugin
  for Assembly GRCh37:
   The following steps are necessary to tabix the gnomad genomes coverage file :
<pre class="code sh_sh">gunzip -c gnomad.genomes.coverage.summary.tsv.bgz | sed '1s/.*/#&/' &gt gnomad.genomes.tabbed.tsv
bgzip gnomad.genomes.tabbed.tsv
tabix -s 1 -b 2 -e 2 gnomad.genomes.tabbed.tsv.gz
</pre><p>   The following steps are neccessary to tabix the gnomad exomes coverage file :
<pre class="code sh_sh">gunzip -c gnomad.exomes.coverage.summary.tsv.bgz | sed '1s/.*/#&/' &gt gnomad.exomes.tabbed.tsv
bgzip gnomad.exomes.tabbed.tsv
tabix -s 1 -b 2 -e 2 gnomad.exomes.tabbed.tsv.gz
</pre><p> for Assembly GRCh38:
   The following steps are necessary to tabix the gnomad genomes coverage file :
<pre class="code sh_sh">gunzip -c gnomad.genomes.r3.0.1.coverage.summary.tsv.bgz | sed '1s/.*/#&/' &gt gnomad.genomesv3.tabbed.tsv
sed "1s/locus/chr\tpos/; s/:/\t/g" gnomad.genomesv3.tabbed.tsv &gt gnomad.ch.genomesv3.tabbed.tsv
bgzip gnomad.ch.genomesv3.tabbed.tsv
tabix -s 1 -b 2 -e 2 gnomad.ch.genomesv3.tabbed.tsv
</pre><p> This plugin also tries to be backwards compatible with older versions of the
 coverage summary files, including releases 2.0.1 and 2.0.2. These releases
 provide one coverage file per chromosome and these can be used "as-is"
 without requiring any preprocessing. </p><p> If you use this plugin, please see the terms and data information:
   <a href="https://gnomad.broadinstitute.org/terms">https://gnomad.broadinstitute.org/terms</a></p><p> You must have the Bio::DB::HTS module or the tabix utility must be installed
 in your path to use this plugin. 
<p><h2>Usage examples:</h2> <pre class="code sh_sh">mv gnomADc.pm ~/.vep/Plugins
./vep -i variations.vcf --plugin gnomADc,/path/to/gnomad.tsv.gz 
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#FF7F50">Frequency data</div></td><td><ul style="padding-left:1em"><li><a href="https://metacpan.org/pod/File::Spec" rel="external">File::Spec</a></li><li><a href="https://metacpan.org/pod/File::Basename" rel="external">File::Basename</a></li></ul></td><td>Stephen Kazakoff</td></tr>
<tr id="GO" class="bg1 plugin_row" data-category="phenotype_data_and_citations"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/GO.pm">GO</a></div><div style="margin-top:6px"><small>Gene Ontology</small></div></td><td><p> An Ensembl VEP plugin that retrieves Gene Ontology (GO) terms associated with
 transcripts (e.g. GRCh38) or their translations (e.g. GRCh37) using custom
 GFF annotation containing GO terms. <a class="button" href="#go" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('go');" id="a_go">more</a></p><div id="div_go" style="display:none;word-wrap:break-word;"><p> The custom GFF files are automatically created if the input file do not exist
 by querying the Ensembl core database, according to database version, species
 and assembly used in Ensembl VEP. Note that automatic retrieval fails if using the
 --offline option.</p><p> The GFF files containing the GO terms are saved to and loaded from the working
 directory by default. To change this, provide a directory path as an argument:</p><p><pre class="code sh_sh">--plugin GO,dir=${HOME}/go_terms
</pre><p> If your GFF file has a custom name, please provide the filename directly:</p><p><pre class="code sh_sh">--plugin GO,file=${HOME}/custom_go_terms.gff.gz
</pre><p> The GO terms can also be fetched by gene match (either gene Ensembl ID or
 gene symbol) instead:</p><p><pre class="code sh_sh">--plugin GO,match=gene
--plugin GO,match=gene_symbol
</pre><p> To create/use a custom GFF file, these programs must be installed in your path:
<ul><li>The GNU zgrep and GNU sort commands to create the GFF file.</li><li>The tabix and bgzip utilities to create and read the GFF file: check&nbsp;     <a href="https://github.com/samtools/htslib.git">https://github.com/samtools/htslib.git</a> for installation instructions.</p><p></li></ul><p> Alternatively, for compatibility purposes, the plugin allows to use a remote
 connection to the Ensembl API by using "remote" as a parameter. This method
 retrieves GO terms one by one at both the transcript and translation level.
 This is not compatible with any other parameters:</p><p><pre class="code sh_sh">--plugin GO,remote
</pre><p></ul><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv GO.pm ~/.vep/Plugins</p><p># automatically fetch GFF files with GO terms and annotate input with GO terms
# not compatible with --offline option
./vep -i variations.vcf --plugin GO</p><p># set directory used to write and read GFF files with GO terms
./vep -i variations.vcf --plugin GO,dir=${HOME}/go_terms</p><p># annotate input with GO terms from custom GFF file
./vep -i variations.vcf --plugin GO,file=${HOME}/custom_go_terms.gff.gz</p><p># annotate input based on gene identifiers instead of transcripts/translations
./vep -i variations.vcf --plugin GO,match=gene</p><p># use remote connection (available for compatibility purposes)
./vep -i variations.vcf --plugin GO,remote
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#556688">Phenotype data and citations</div></td><td>-</td><td>Ensembl</td></tr>
<tr id="GWAS" class="bg2 plugin_row" data-category="phenotype_data_and_citations"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/GWAS.pm">GWAS</a></div></td><td><p> An Ensembl VEP plugin that retrieves relevant NHGRI-EBI GWAS Catalog data given the file.  <a class="button" href="#gwas" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('gwas');" id="a_gwas">more</a></p><div id="div_gwas" style="display:none;word-wrap:break-word;"><p> This plugin supports both the curated data that is found in the download section of the NHGRI-EBI GWAS Catalog website and the 
 summary statistics file. By default the plugin assumes the file provided is the curated file but you can pass "type=sstate" 
 to say you want to annotate with a summary statistics file.</p><p> Please cite the following publication alongside Ensembl VEP if you use this resource:
 <a href="https://pubmed.ncbi.nlm.nih.gov/30445434/">https://pubmed.ncbi.nlm.nih.gov/30445434/</a></p><p> Pre-requisites:</p><p> For curated NHGRI-EBI GWAS Catalog file -
 GWAS files can be downloaded from - <a href="https://www.ebi.ac.uk/gwas/api/search/downloads/alternative">https://www.ebi.ac.uk/gwas/api/search/downloads/alternative</a></p><p> For summary statistics file -
 The plugin can process the harmonised version of the summary statistics file. Which can be downloaded from the FTP site - 
 <a href="http://ftp.ebi.ac.uk/pub/databases/gwas/summary_statistics">http://ftp.ebi.ac.uk/pub/databases/gwas/summary_statistics</a></p><p> They are under directory with related to their specific GCST id. For example -
 <a href="http://ftp.ebi.ac.uk/pub/databases/gwas/summary_statistics/GCST000001-GCST001000/GCST000028/harmonised/17463246-GCST000028-EFO_0001360.h.tsv.gz">http://ftp.ebi.ac.uk/pub/databases/gwas/summary_statistics/GCST000001-GCST001000/GCST000028/harmonised/17463246-GCST000028-EFO_0001360.h.tsv.gz</a></p><p> Please keep the filename format as it is because filename is parsed to get information.</p><p> When run for the first time for either type of file, the plugin will create a processed file that have genomic locations and indexed and 
 put it under the --dir location determined by Ensembl VEP. If db=1 option is used, depending on the file size it might take hour(s) to create 
 the processed file. Subsequent runs will be faster as the plugin will be using the already generated processed file. This option is not used by 
 default and the variant information is generally taken directly from the file provided.</p><p> Options are passed to the plugin as key=value pairs:<table class="ss"><thead><tr><th>Argument</th><th>Description</th></tr></thead><tbody>
<tr class="bg1"><td><pre>file</pre></td><td>(mandatory) Path to GWAS curated or summary statistics file <tr class="bg2"><td><pre>type</pre></td><td>type of the file. Valid values are "curated" and "sstate" (summary statistics). Default is "curated". <tr class="bg1"><td><pre>verbose</pre></td><td>display info level messages. Valid values are 0 or 1. Default is 0. <tr class="bg2"><td><pre>db</pre></td><td>get variant information from Ensembl database during creation of processed file. Valid values are 0 or 1. Default is 0 (variant             information is retrieved from curated file)</p><p></td></tr></tbody></table><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv GWAS.pm ~/.vep/Plugins
./vep -i variations.vcf --plugin GWAS,file=/FULL_PATH_TO/gwas_catalog_v1.0.2-associations_e107_r2022-09-14.tsv
./vep -i variations.vcf --plugin GWAS,type=sstate,file=/FULL_PATH_TO/17463246-GCST000028-EFO_0001360.h.tsv.gz
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#556688">Phenotype data and citations</div></td><td><ul style="padding-left:1em"><li><a href="https://metacpan.org/pod/Storable" rel="external">Storable</a> qw(dclone)</li><li><a href="https://metacpan.org/pod/File::Basename" rel="external">File::Basename</a></li></ul></td><td>Ensembl</td></tr>
<tr id="HGVSIntronOffset" class="bg1 plugin_row" data-category="hgvs"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/HGVSIntronOffset.pm">HGVSIntronOffset</a></div></td><td><p> An Ensembl VEP plugin for the Ensembl Variant Effect Predictor (VEP) that returns
 HGVS intron start and end offsets. To be used with --hgvs option. <a class="button" href="#hgvsintronoffset" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('hgvsintronoffset');" id="a_hgvsintronoffset">more</a></p><div id="div_hgvsintronoffset" style="display:none;word-wrap:break-word;"><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv HGVSIntronOffset.pm ~/.vep/Plugins
./vep -i variants.vcf --hgvs --plugin HGVSIntronOffset
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#9400D3">HGVS</div></td><td>-</td><td>Stephen Kazakoff</td></tr>
<tr id="IntAct" class="bg2 plugin_row" data-category="functional_effect"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/IntAct.pm">IntAct</a></div></td><td><p> A VEP plugin that retrieves molecular interaction data for variants as reprted by IntAct database. <a class="button" href="#intact" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('intact');" id="a_intact">more</a></p><div id="div_intact" style="display:none;word-wrap:break-word;"><p> Please cite the IntAct publication alongside Ensembl VEP if you use this resource:
 <a href="https://pubmed.ncbi.nlm.nih.gov/24234451/">https://pubmed.ncbi.nlm.nih.gov/24234451/</a></p><p> Pre-requisites:</p><p><ol><li value="1">IntAct files can be downloaded from - <a href="https://ftp.ebi.ac.uk/pub/databases/intact/current/various">https://ftp.ebi.ac.uk/pub/databases/intact/current/various</a></p><p></li><li value="2">The genomic location mapped file needs to be tabix indexed. You can  do this by following commands -</p><p></li></ol><p>  a) filter, sort and then zip
<pre class="code sh_sh">grep -v -e '^$' -e '^[#\-]' mutation_gc_map.txt | sed '1s/.*/#&/' | awk -F "\t" 'BEGIN { OFS="\t"} {if ($2 &gt $3) {a=$2; $2=$3; $3=a}; print $0 }' | sort -k1,1 -k2,2n -k3,3n | bgzip &gt mutation_gc_map.txt.gz
</pre><p>  b) create tabix indexed file -
<pre class="code sh_sh">tabix -s 1 -b 2 -e 3 -f mutation_gc_map.txt.gz
</pre><p><ol><li value="3">As you have already noticed, tabix utility must be installed in your path to use this plugin. 
</li></ol><p> Options are passed to the plugin as key=value pairs:<table class="ss"><thead><tr><th>Argument</th><th>Description</th></tr></thead><tbody>
<tr class="bg1"><td><pre>mapping_file</pre></td><td>(mandatory) Path to tabix-indexed genomic location mapped file <tr class="bg2"><td><pre>mutation_file</pre></td><td>(mandatory) Path to IntAct data file  
</td></tr></tbody></table><p> By default the output will always contain feature_type and interaction_ac from the IntAct data file. You can also add more fields using the following key=value options -<table class="ss"><thead><tr><th>Argument</th><th>Description</th></tr></thead><tbody><tr class="bg1"><td><pre>feature_ac</pre></td><td>Set value to 1 to include Feature AC in the output <tr class="bg2"><td><pre>feature_short_label</pre></td><td>Set value to 1 to include Feature short label in the output <tr class="bg1"><td><pre>feature_annotation</pre></td><td>Set value to 1 to include Feature annotation in the output <tr class="bg2"><td><pre>ap_ac</pre></td><td>Set value to 1 to include Affected protein AC in the output <tr class="bg1"><td><pre>interaction_participants</pre></td><td>Set value to 1 to include Interaction participants in the output <tr class="bg2"><td><pre>pmid</pre></td><td>Set value to 1 to include PubMedID in the output 
</td></tr></tbody></table><p> There are also two other key=value options for customizing the output - <table class="ss"><thead><tr><th>Argument</th><th>Description</th></tr></thead><tbody><tr class="bg1"><td><pre>all</pre></td><td>Set value to 1 to include all the fields <tr class="bg2"><td><pre>minimal</pre></td><td>Set value to 1 to overwrite default behavior and include only interaction_ac 				  in the output by default</p><p></td></tr></tbody></table><p> See what these options mean - <a href="https://www.ebi.ac.uk/intact/download/datasets#mutations">https://www.ebi.ac.uk/intact/download/datasets#mutations</a></p><p> Note that, interaction accession can be used to link to full details on the interaction website. For example, 
 where the Ensembl VEP output reports an interaction_ac of EBI-12501485, the URL would be : </p><p>   <a href="https://www.ebi.ac.uk/intact/details/interaction/EBI-12501485">https://www.ebi.ac.uk/intact/details/interaction/EBI-12501485</a></p><p></ol><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv IntAct.pm ~/.vep/Plugins
./vep -i variations.vcf --plugin IntAct,mutation_file=/FULL_PATH_TO_IntAct_FILE/mutations.tsv,mapping_file=/FULL_PATH_TO_IntAct_FILE/mutation_gc_map.txt.gz
./vep -i variations.vcf --plugin IntAct,mutation_file=/FULL_PATH_TO_IntAct_FILE/mutations.tsv,mapping_file=/FULL_PATH_TO_IntAct_FILE/mutation_gc_map.txt.gz,minimal=1
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#226600">Functional effect</div></td><td>-</td><td>Ensembl</td></tr>
<tr id="LD" class="bg1 plugin_row" data-category="variant_data"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/LD.pm">LD</a></div><div style="margin-top:6px"><small>Linkage Disequilibrium</small></div></td><td><p> A VEP plugin that finds variants in linkage disequilibrium with any overlapping
 existing variants from the Ensembl variation databases. <a class="button" href="#ld" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('ld');" id="a_ld">more</a></p><div id="div_ld" style="display:none;word-wrap:break-word;"><p> You can configure the population used to calculate the r2 value, and the
 r2 cutoff used by passing arguments to the plugin via the Ensembl VEP command line
 (separated by commas). This plugin adds a single new entry to the Extra column
 with a comma-separated list of linked variant IDs and the associated r2 values:
   LinkedVariants=rs123:0.879,rs234:0.943</p><p> If no arguments are supplied, the default population used is the CEU
 sample from the 1000 Genomes Project phase 3, and the default r2
 cutoff used is 0.8.</p><p> WARNING: Calculating LD is a relatively slow procedure, so this will 
 slow Ensembl VEP down considerably when running on large numbers of
 variants. Consider running vep followed by filter_vep to get a smaller
 input set:</p><p><pre class="code sh_sh">./vep -i input.vcf -cache -vcf -o input_vep.vcf
./filter_vep -i input_vep.vcf -filter "Consequence is missense_variant" &gt input_vep_filtered.vcf
./vep -i input_vep_filtered.vcf -cache -plugin LD
</pre><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv LD.pm ~/.vep/Plugins
./vep -i variations.vcf --plugin LD,1000GENOMES:phase_3:CEU,0.8
./vep -i variations.vcf --plugin LD,'populations=1000GENOMES:phase_3:CEU&1000GENOMES:phase_3:PUR&1000GENOMES:phase_3:STU',0.8</p><p></pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#B22222">Variant data</div></td><td>-</td><td>Ensembl</td></tr>
<tr id="LocalID" class="bg2 plugin_row" data-category="look_up"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/LocalID.pm">LocalID</a></div></td><td><p> The LocalID plugin allows you to use variant IDs as input without making a database connection. <a class="button" href="#localid" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('localid');" id="a_localid">more</a></p><div id="div_localid" style="display:none;word-wrap:break-word;"><p> Requires sqlite3.</p><p> A local sqlite3 database is used to look up variant IDs; this is generated either from Ensembl's
 public database (very slow, but includes synonyms), or from an Ensembl VEP cache file (faster, excludes
 synonyms).</p><p> NB this plugin is NOT compatible with the ensembl-tools variant_effect_predictor.pl version of Ensembl VEP.</p><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv LocalID.pm ~/.vep/Plugins</p><p>## first run create database</p><p># EITHER create from Ensembl variation database
# VERY slow but includes variant synonyms, if not required see next command
./vep -i variant_ids.txt --plugin LocalID,create_db=1 -safe</p><p># OR create from cache directory
# faster but does not include synonyms
# parameter passed to from_cache may be full path to cache e.g. $HOME/.vep/homo_sapiens/88_GRCh38
# cache may be tabix converted or in default state (http://www.ensembl.org/info/docs/tools/vep/script/vep_cache.html#convert)
./vep -i variant_ids.txt --plugin LocalID,create_db=1,from_cache=1 -safe</p><p># subsequent runs
./vep -i variant_ids.txt --plugin LocalID</p><p># db file can be specified with db=[file]
# default file name is $HOME/.vep/[species]_[version]_[assembly].variant_ids.sqlite3
./vep -i variant_ids.txt --plugin LocalID,db=my_db_file.txt
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#006400">Look up</div></td><td>-</td><td>Ensembl</td></tr>
<tr id="LOEUF" class="bg1 plugin_row" data-category="gene_tolerance_to_change"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/LOEUF.pm">LOEUF</a></div></td><td><p> This is a plugin for the Ensembl Variant Effect Predictor (VEP) that
 adds the LOEUF scores to Ensembl VEP output. LOEUF stands for the "loss-of-function 
 observed/expected upper bound fraction."  <a class="button" href="#loeuf" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('loeuf');" id="a_loeuf">more</a></p><div id="div_loeuf" style="display:none;word-wrap:break-word;"><p> The score can be added matching by either transcript or gene.
 When matched by gene: 
 If multiple transcripts are available for a gene, the most severe score is reported.</p><p> NB: The plugin currently does not add the score for downstream_gene_variant and upstream_gene_variant </p><p> Please cite the LOEUF publication alongside Ensembl VEP if you use this resource:
 <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7334197/">https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7334197/</a></p><p> LOEUF scores can be downloaded from
 GRCh37: <a href="https://gnomad.broadinstitute.org/downloads#v2-constraint">https://gnomad.broadinstitute.org/downloads#v2-constraint</a> (pLoF Metrics by Gene TSV)
 GRCh38: <a href="https://gnomad.broadinstitute.org/downloads#v4-constraint">https://gnomad.broadinstitute.org/downloads#v4-constraint</a> (Constraint metrics TSV)</p><p> For GRCh37:
 These files can be tabix-processed by:
<pre class="code sh_sh">zcat gnomad.v2.1.1.lof_metrics.by_gene.txt.bgz | (head -n 1 && tail -n +2  | sort -t$'\t' -k 76,76 -k 77,77n ) &gt loeuf_temp.tsv
sed '1s/.*/#&/' loeuf_temp.tsv &gt loeuf_dataset.tsv
bgzip loeuf_dataset.tsv
tabix -f -s 76 -b 77 -e 78 loeuf_dataset.tsv.gz
</pre><p> For GRCh38:
 The GRCh38 file does not have gene co-ordinates information. First you need to add the gene co-ordiates information.
 You can use the Ensembl Perl API to create a script and perform that - <a href="https://www.ensembl.org/info/docs/api/core/index.html">https://www.ensembl.org/info/docs/api/core/index.html</a>.
 After adding the start and end position of the genes at the last 2 columns you can process the file as follows:
<pre class="code sh_sh">cat gnomad.v4.1.constraint_metrics_with_coordinates.tsv | (head -n 1 && tail -n +2  | sort -t$'\t' -k 53,53 -k 56,56n ) &gt loeuf_grch38_temp.tsv
sed '1s/.*/#&/' loeuf_grch38_temp.tsv &gt loeuf_dataset_grch38.tsv
bgzip loeuf_dataset_grch38.tsv
tabix -f -s 53 -b 56 -e 57 loeuf_dataset_grch38.tsv.gz
</pre><p>
 The tabix utility must be installed in your path to use this plugin.</p><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv LOEUF.pm ~/.vep/Plugins
./vep -i variations.vcf --plugin LOEUF,file=/path/to/loeuf/data.tsv.gz,match_by=gene
./vep -i variations.vcf --plugin LOEUF,file=/path/to/loeuf/data.tsv.gz,match_by=transcript
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#00DD66">Gene tolerance to change</div></td><td><a href="https://metacpan.org/pod/Scalar::Util" rel="external">Scalar::Util</a> qw(looks_like_number)</td><td>Ensembl</td></tr>
<tr id="LoFtool" class="bg2 plugin_row" data-category="pathogenicity_predictions"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/LoFtool.pm">LoFtool</a></div><div style="margin-top:6px"><small>Loss-of-function</small></div></td><td><p>  Add LoFtool scores to Ensembl  VEP output. <a class="button" href="#loftool" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('loftool');" id="a_loftool">more</a></p><div id="div_loftool" style="display:none;word-wrap:break-word;"><p>  LoFtool provides a rank of genic intolerance and consequent
  susceptibility to disease based on the ratio of Loss-of-function (LoF)
  to synonymous mutations for each gene in 60,706 individuals from ExAC,
  adjusting for the gene de novo mutation rate and evolutionary protein
  conservation. The lower the LoFtool gene score percentile the most
  intolerant is the gene to functional variation. For more details please see
  (Fadista J et al. 2017), PMID:27563026.
  The authors would like to thank the Exome Aggregation Consortium and
  the groups that provided exome variant data for comparison. A full
  list of contributing groups can be found at <a href="http://exac.broadinstitute.org/about">http://exac.broadinstitute.org/about</a>.</p><p>  The LoFtool_scores.txt file is found alongside the plugin in the
  VEP_plugins GitHub repo.</p><p>  To use another scores file, add it as a parameter i.e.</p><p><pre class="code sh_sh">./vep -i variants.vcf --plugin LoFtool,scores_file.txt
</pre><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv LoFtool.pm ~/.vep/Plugins
mv LoFtool_scores.txt ~/.vep/Plugins
./vep -i variants.vcf --plugin LoFtool
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#1E90FF">Pathogenicity predictions</div></td><td><a href="https://metacpan.org/pod/DBI" rel="external">DBI</a></td><td>Ensembl</td></tr>
<tr id="LOVD" class="bg1 plugin_row" data-category="variant_data"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/LOVD.pm">LOVD</a></div><div style="margin-top:6px"><small>Leiden Open Variation Database</small></div></td><td><p> An Ensembl VEP plugin that retrieves LOVD variation data from <a href="http://www.lovd.nl/">http://www.lovd.nl/</a>. <a class="button" href="#lovd" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('lovd');" id="a_lovd">more</a></p><div id="div_lovd" style="display:none;word-wrap:break-word;"><p> Please be aware that LOVD is a public resource of curated variants, therefore
 please respect this resource and avoid intensive querying of their databases
 using this plugin, as it will impact the availability of this resource for
 others.</p><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv LOVD.pm ~/.vep/Plugins
./vep -i variations.vcf --plugin LOVD
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#B22222">Variant data</div></td><td><a href="https://metacpan.org/pod/LWP::UserAgent" rel="external">LWP::UserAgent</a></td><td>Ensembl</td></tr>
<tr id="Mastermind" class="bg2 plugin_row" data-category="phenotype_data_and_citations"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/Mastermind.pm">Mastermind</a></div></td><td><p> This is a plugin for the Ensembl Variant Effect Predictor (VEP) that
 uses the Mastermind Genomic Search Engine (<a href="https://www.genomenon.com/mastermind">https://www.genomenon.com/mastermind</a>)
 to report variants that have clinical evidence cited in the medical literature. 
 It is available for both GRCh37 and GRCh38. <a class="button" href="#mastermind" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('mastermind');" id="a_mastermind">more</a></p><div id="div_mastermind" style="display:none;word-wrap:break-word;"><p> Please cite the Mastermind publication alongside Ensembl VEP if you use this resource:
 <a href="https://www.frontiersin.org/article/10.3389/fgene.2020.577152">https://www.frontiersin.org/article/10.3389/fgene.2020.577152</a></p><p> Running options:
 The plugin has multiple parameters, the first one is expected to be the file name path 
 which can be followed by 3 optional flags.
 Default: the plugin matches the citation data with the specific mutation.
 Using first flag <kbd>1</kbd>: returns the citations for all mutations/transcripts.   
 Using the second flag <kbd>1</kbd>: only returns the Mastermind variant identifier(s).
 Using the third flag <kbd>1</kbd>: also returns the Mastermind URL.</p><p> Output: 
 The output includes three unique counts 'MMCNT1, MMCNT2, MMCNT3' and one identifier <kbd>MMID3</kbd>
 to be used to build an URL which shows all articles from MMCNT3.</p><p><ul><li><kbd>MMCNT1</kbd> is the count of Mastermind articles with cDNA matches for a specific variant;</li><li><kbd>MMCNT2</kbd> is the count of Mastermind articles with variants either explicitly matching at&nbsp;   the cDNA level or given only at protein level;
</li><li><kbd>MMCNT3</kbd> is the count of Mastermind articles including other DNA-level variants resulting&nbsp;   in the same amino acid change;
</li><li><kbd>MMID3</kbd> is the Mastermind variant identifier(s), as gene:key. Link to the Genomenon Mastermind Genomic Search Engine;
</li></ul><p> To build the URL, substitute the <kbd>gene:key</kbd> in the following link with the value from MMID3:
 <a href="https://mastermind.genomenon.com/detail?mutation=gene:key">https://mastermind.genomenon.com/detail?mutation=gene:key</a></p><p> If the third flag is used then the built URL is returned and it's identified by <kbd>URL</kbd>.</p><p> More information can be found at: <a href="https://www.genomenon.com/cvr/">https://www.genomenon.com/cvr/</a></p><p> The following steps are necessary before running this plugin:</p><p> Download and Registry (free): 
 <a href="https://www.genomenon.com/cvr/">https://www.genomenon.com/cvr/</a> </p><p> GRCh37 VCF:
<pre class="code sh_sh">unzip mastermind_cited_variants_reference-XXXX.XX.XX-grch37-vcf.zip
bgzip mastermind_cited_variants_reference-XXXX.XX.XX-GRCh37-vcf
tabix -p vcf mastermind_cited_variants_reference-XXXX.XX.XX.GRCh37-vcf.gz
</pre><p> GRCh38 VCF:
<pre class="code sh_sh">unzip mastermind_cited_variants_reference-XXXX.XX.XX-grch38-vcf.zip
bgzip mastermind_cited_variants_reference-XXXX.XX.XX-GRCh38-vcf
tabix -p vcf mastermind_cited_variants_reference-XXXX.XX.XX.GRCh38-vcf.gz
</pre><p> 
 The plugin can then be run as default:
<pre class="code sh_sh">./vep -i variations.vcf --plugin Mastermind,file=/path/to/mastermind_cited_variants_reference-XXXX.XX.XX.GRChXX-vcf.gz
</pre><p> or with an option to not filter by mutations (first flag): 
<pre class="code sh_sh">./vep -i variations.vcf --plugin Mastermind,file=/path/to/mastermind_cited_variants_reference-XXXX.XX.XX.GRChXX-vcf.gz,mutations=1 
</pre><p> or with an option to only return <kbd>MMID3</kbd> e.g. the Mastermind variant identifier as gene:key (second flag):
<pre class="code sh_sh">./vep -i variations.vcf --plugin Mastermind,file=/path/to/mastermind_cited_variants_reference-XXXX.XX.XX.GRChXX-vcf.gz,mutations=0,var_iden=1
</pre><p> or with an option to also return the Mastermind URL (third flag):
<pre class="code sh_sh">./vep -i variations.vcf --plugin Mastermind,file=/path/to/mastermind_cited_variants_reference-XXXX.XX.XX.GRChXX-vcf.gz,mutations=0,var_iden=0,url=1
</pre><p> Note: when running Ensembl VEP in offline mode Mastermind requires a fasta file (--fasta)</p><p></ul><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv Mastermind.pm ~/.vep/Plugins
./vep -i variations.vcf --plugin Mastermind,file=/path/to/data.vcf.gz
./vep -i variations.vcf --plugin Mastermind,file=/path/to/data.vcf.gz,mutations=1
./vep -i variations.vcf --plugin Mastermind,file=/path/to/data.vcf.gz,mutations=0,var_iden=1
./vep -i variations.vcf --plugin Mastermind,file=/path/to/data.vcf.gz,mutations=0,var_iden=0,url=1
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#556688">Phenotype data and citations</div></td><td>-</td><td>Ensembl</td></tr>
<tr id="MaveDB" class="bg1 plugin_row" data-category="functional_effect"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/MaveDB.pm">MaveDB</a></div></td><td><p> An Ensembl VEP plugin that retrieves data from MaveDB (<a href="https://www.mavedb.org">https://www.mavedb.org</a>), a
 database that contains multiplex assays of variant effect, including deep
 mutational scans and massively parallel report assays. <a class="button" href="#mavedb" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('mavedb');" id="a_mavedb">more</a></p><div id="div_mavedb" style="display:none;word-wrap:break-word;"><p> To run the MaveDB plugin, please download the following files containing
 MaveDB data for GRCh38 (we do not currently host data for other assemblies):
<ul><li><a href="https://ftp.ensembl.org/pub/current_variation/MaveDB/MaveDB_variants.tsv.gz">https://ftp.ensembl.org/pub/current_variation/MaveDB/MaveDB_variants.tsv.gz</a></li><li><a href="https://ftp.ensembl.org/pub/current_variation/MaveDB/MaveDB_variants.tsv.gz.tbi">https://ftp.ensembl.org/pub/current_variation/MaveDB/MaveDB_variants.tsv.gz.tbi</a>
</li></ul><p> Options are passed to the plugin as key=value pairs:<table class="ss"><thead><tr><th>Argument</th><th>Description</th></tr></thead><tbody><tr class="bg1"><td><pre>file</pre></td><td>(mandatory) Tabix-indexed MaveDB file <tr class="bg2"><td><pre>cols</pre></td><td>Colon-separated columns to print from MaveDB                               files; if set to <kbd>all</kbd>, all columns are
                              printed (default: <kbd>urn:score:nt:pro</kbd>)
<tr class="bg1"><td><pre>single_aminoacid_changes</pre></td><td>Return matches for single aminoacid changes only;                               if disabled, return all matches associated with a
                              genetic variant (default: 1)
<tr class="bg2"><td><pre>transcript_match</pre></td><td>Return results only if (Ensembl or RefSeq)                               transcript identifiers match (default: 1)</p><p></td></tr></tbody></table><p> Please cite the MaveDB publication alongside the VEP if you use this resource:
 <a href="https://doi.org/10.1186/s13059-019-1845-6">https://doi.org/10.1186/s13059-019-1845-6</a></p><p> The tabix utility must be installed in your path to use this plugin.</p><p></ul><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv MaveDB.pm ~/.vep/Plugins</p><p># print only scores for single aminoacid changes from MaveDB data (default)
./vep -i variations.vcf --plugin MaveDB,file=/full/path/to/data.csv.gz</p><p># print all scores associated with the genetic variant
./vep -i variations.vcf --plugin MaveDB,file=/full/path/to/data.csv.gz,single_aminoacid_changes=0</p><p># print all columns from MaveDB data
./vep -i variations.vcf --plugin MaveDB,file=/full/path/to/data.csv.gz,cols=all
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#226600">Functional effect</div></td><td><ul style="padding-left:1em"><li><a href="https://metacpan.org/pod/Bio::SeqUtils" rel="external">Bio::SeqUtils</a></li><li><a href="https://metacpan.org/pod/File::Basename" rel="external">File::Basename</a></li></ul></td><td>Ensembl</td></tr>
<tr id="MaxEntScan" class="bg2 plugin_row" data-category="splicing_predictions"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/MaxEntScan.pm">MaxEntScan</a></div></td><td><p> This is a plugin for the Ensembl Variant Effect Predictor (VEP) that
 runs MaxEntScan (<a href="http://hollywood.mit.edu/burgelab/maxent/Xmaxentscan_scoreseq.html">http://hollywood.mit.edu/burgelab/maxent/Xmaxentscan_scoreseq.html</a>)
 to get splice site predictions. <a class="button" href="#maxentscan" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('maxentscan');" id="a_maxentscan">more</a></p><div id="div_maxentscan" style="display:none;word-wrap:break-word;"><p> The plugin copies most of the code verbatim from the score5.pl and score3.pl
 scripts provided in the MaxEntScan download. To run the plugin you must get and
 unpack the archive from <a href="http://hollywood.mit.edu/burgelab/maxent/download/">http://hollywood.mit.edu/burgelab/maxent/download/</a>; the path
 to this unpacked directory is then the param you pass to the --plugin flag.</p><p> The plugin executes the logic from one of the scripts depending on which
 splice region the variant overlaps:</p><p><ul><li>score5.pl : last 3 bases of exon    --&gt first 6 bases of intron</li><li>score3.pl : last 20 bases of intron --&gt first 3 bases of exon
</li></ul><p> The plugin reports the reference, alternate and difference (REF - ALT) maximum
 entropy scores.</p><p> If <kbd>SWA</kbd> is specified as a command-line argument, a sliding window algorithm
 is applied to subsequences containing the reference and alternate alleles to
 identify k-mers with the highest donor and acceptor splice site scores. To assess
 the impact of variants, reference comparison scores are also provided. For SNVs,
 the comparison scores are derived from sequence in the same frame as the highest
 scoring k-mers containing the alternate allele. For all other variants, the
 comparison scores are derived from the highest scoring k-mers containing the
 reference allele. The difference between the reference comparison and alternate
 scores (SWA_REF_COMP - SWA_ALT) are also provided.</p><p> If <kbd>NCSS</kbd> is specified as a command-line argument, scores for the nearest
 upstream and downstream canonical splice sites are also included.</p><p> By default, only scores are reported. Add <kbd>verbose</kbd> to the list of command-
 line arguments to include the sequence output associated with those scores.</p><p></ul><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv MaxEntScan.pm ~/.vep/Plugins
./vep -i variants.vcf --plugin MaxEntScan,/path/to/maxentscan/fordownload
./vep -i variants.vcf --plugin MaxEntScan,/path/to/maxentscan/fordownload,SWA,NCSS
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#FF0000">Splicing predictions</div></td><td><a href="https://metacpan.org/pod/Digest::MD5" rel="external">Digest::MD5</a> qw(md5_hex)</td><td>Ensembl</td></tr>
<tr id="MPC" class="bg1 plugin_row" data-category="pathogenicity_predictions"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/MPC.pm">MPC</a></div><div style="margin-top:6px"><small>missense deleteriousness metric</small></div></td><td><p> An Ensembl VEP plugin that retrieves MPC scores for variants from a tabix-indexed MPC data file. <a class="button" href="#mpc" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('mpc');" id="a_mpc">more</a></p><div id="div_mpc" style="display:none;word-wrap:break-word;"><p> MPC is a missense deleteriousness metric based on the analysis of genic regions
 depleted of missense mutations in the Exome Agggregation Consortium (ExAC) data.</p><p> The MPC score is the product of work by Kaitlin Samocha (ks20@sanger.ac.uk).
 Publication currently in pre-print: Samocha et al bioRxiv 2017 (TBD)</p><p> The MPC score file is available to download from:</p><p> <a href="https://ftp.broadinstitute.org/pub/ExAC_release/release1/regional_missense_constraint/">https://ftp.broadinstitute.org/pub/ExAC_release/release1/regional_missense_constraint/</a></p><p> The data are currently mapped to GRCh37 only. Not all transcripts are included; see
 README in the above directory for exclusion criteria.</p><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv MPC.pm ~/.vep/Plugins
./vep -i variations.vcf --plugin MPC,fordist_constraint_official_mpc_values.txt.gz
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#1E90FF">Pathogenicity predictions</div></td><td>-</td><td>Ensembl</td></tr>
<tr id="MTR" class="bg2 plugin_row" data-category="pathogenicity_predictions"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/MTR.pm">MTR</a></div><div style="margin-top:6px"><small>Missense Tolerance Ratio</small></div></td><td><p>An Ensembl VEP plugin that retrieves Missense Tolerance Ratio (MTR) scores for
variants from a tabix-indexed flat file. <a class="button" href="#mtr" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('mtr');" id="a_mtr">more</a></p><div id="div_mtr" style="display:none;word-wrap:break-word;"><p>MTR scores quantify the amount of purifying selection acting
specifically on missense variants in a given window of protein-coding
sequence. It is estimated across a sliding window of 31 codons and uses
observed standing variation data from the WES component of the Exome
Aggregation Consortium Database (ExAC), version 2.0
(<a href="http://gnomad.broadinstitute.org">http://gnomad.broadinstitute.org</a>).</p><p>Please cite the MTR publication alongside Ensembl VEP if you use this resource:
<a href="http://genome.cshlp.org/content/27/10/1715">http://genome.cshlp.org/content/27/10/1715</a></p><p>The Bio::DB::HTS perl library or tabix utility must be installed in your path to use this plugin. 
MTR flat files can be downloaded from <a href="http://biosig.unimelb.edu.au/mtr-viewer/downloads">http://biosig.unimelb.edu.au/mtr-viewer/downloads</a>
The following steps are necessary before running the plugin 
<pre class="code sh_sh">gzip -d mtrflatfile_2.0.txt.gz # to unzip the text file 
cat mtrflatfile_2.0.txt | tr " " "\t" &gt mtrflatfile_2.00.tsv # to change the file to a tabbed delimited file 
sed '1s/.*/#&/'  mtrflatfile_2.00.tsv &gt mtrflatfile_2.0.tsv # to add # to the first line of the file 
bgzip mtrflatfile_2.0.tsv
tabix -f -s 1 -b 2 -e 2 mtrflatfile_2.0.tsv.gz
</pre><p>NB: Data are available for GRCh37 only</p><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv MTR.pm ~/.vep/Plugins
./vep -i variations.vcf --plugin MTR,mtrflatfile_2.0.tsv.gz
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#1E90FF">Pathogenicity predictions</div></td><td>-</td><td><ul style="padding-left:1em"><li>Slave Petrovski</li><li>Michael Silk</li></ul></td></tr>
<tr id="mutfunc" class="bg1 plugin_row" data-category="protein_annotation"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/mutfunc.pm">mutfunc</a></div></td><td><p> An Ensembl VEP plugin that retrieves data from mutfunc db predicting destabilization of protein structure, interaction interface, and motif. <a class="button" href="#mutfunc" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('mutfunc');" id="a_mutfunc">more</a></p><div id="div_mutfunc" style="display:none;word-wrap:break-word;"><p> Please cite the mutfunc publication alongside Ensembl VEP if you use this resource:
 <a href="http://msb.embopress.org/content/14/12/e8430">http://msb.embopress.org/content/14/12/e8430</a></p><p> Pre-requisites:</p><p><ol><li value="1">The data file. mutfunc SQLite db can be downloaded from -  <a href="https://ftp.ensembl.org/pub/current_variation/mutfunc/mutfunc_data.db">https://ftp.ensembl.org/pub/current_variation/mutfunc/mutfunc_data.db</a></p><p></li><li value="2">If you are using --offline please provide a FASTA file as this plugin requires the translation sequence to function.</p><p></li></ol><p> Options are passed to the plugin as key=value pairs:<table class="ss"><thead><tr><th>Argument</th><th>Description</th></tr></thead><tbody>
<tr class="bg1"><td><pre>db</pre></td><td>(mandatory) Path to SQLite database containing data for other analysis. <tr class="bg2"><td><pre>motif</pre></td><td>Select this option to have mutfunc motif analysis in the output <tr class="bg1"><td><pre>int</pre></td><td>Select this option to have mutfunc protein interection analysis in the output <tr class="bg2"><td><pre>mod</pre></td><td>Select this option to have mutfunc protein structure analysis in the output <tr class="bg1"><td><pre>exp</pre></td><td>Select this option to have mutfunc protein structure (experimental) analysis in the output <tr class="bg2"><td><pre>extended</pre></td><td>By default mutfunc outputs the most significant field for any analysis. Select this option to get more verbose output. 
</td></tr></tbody></table><p> By default all of the four type of analysis (motif, int, mod, and exp) data are available in the output. But if you want to have 
 some selected analysis and not all of them just select the relevant options.</p><p></ol><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv mutfunc.pm ~/.vep/Plugins
./vep -i variations.vcf --plugin mutfunc,motif=1,extended=1,db=/FULL_PATH_TO/mutfunc_data.db
./vep -i variations.vcf --plugin mutfunc,db=/FULL_PATH_TO/mutfunc_data.db
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#8844CC">Protein annotation</div></td><td><ul style="padding-left:1em"><li><a href="https://metacpan.org/pod/List::MoreUtils" rel="external">List::MoreUtils</a> qw(first_index)</li><li><a href="https://metacpan.org/pod/Compress::Zlib" rel="external">Compress::Zlib</a></li><li><a href="https://metacpan.org/pod/Digest::MD5" rel="external">Digest::MD5</a> qw(md5_hex)</li><li><a href="https://metacpan.org/pod/DBI" rel="external">DBI</a></li></ul></td><td>Ensembl</td></tr>
<tr id="NearestExonJB" class="bg2 plugin_row" data-category="nearby_features"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/NearestExonJB.pm">NearestExonJB</a></div></td><td><p> This is a plugin for the Ensembl Variant Effect Predictor (VEP) that
 finds the nearest exon junction boundary to a coding sequence variant. More than 
 one boundary may be reported if the boundaries are equidistant. <a class="button" href="#nearestexonjb" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('nearestexonjb');" id="a_nearestexonjb">more</a></p><div id="div_nearestexonjb" style="display:none;word-wrap:break-word;"><p> The plugin will report the Ensembl identifier of the exon, the distance to the
 exon boundary, the boundary type (start or end of exon) and the total
 length in nucleotides of the exon.</p><p> Various key=value parameters can be altered by passing them to the plugin command:<table class="ss"><thead><tr><th>Argument</th><th>Description</th></tr></thead><tbody>
<tr class="bg1"><td><pre>max_range</pre></td><td>maximum search range in bp (default: 10000) 
</td></tr></tbody></table><p> Parameters are passed e.g.:</p><p><pre class="code sh_sh">--plugin NearestExonJB,max_range=50000
</pre><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv NearestExonJB.pm ~/.vep/Plugins
./vep -i variations.vcf --cache --plugin NearestExonJB
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#E75480">Nearby features</div></td><td>-</td><td>Ensembl</td></tr>
<tr id="NearestGene" class="bg1 plugin_row" data-category="nearby_features"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/NearestGene.pm">NearestGene</a></div></td><td><p> This is a plugin for the Ensembl Variant Effect Predictor (VEP) that
 finds the nearest gene(s) to a non-genic variant. More than one gene
 may be reported if the genes overlap the variant or if genes are
 equidistant. <a class="button" href="#nearestgene" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('nearestgene');" id="a_nearestgene">more</a></p><div id="div_nearestgene" style="display:none;word-wrap:break-word;"><p> Various key=value parameters can be altered by passing them to the plugin command:<table class="ss"><thead><tr><th>Argument</th><th>Description</th></tr></thead><tbody>
<tr class="bg1"><td><pre>limit</pre></td><td>limit the number of genes returned (default: 1) <tr class="bg2"><td><pre>range</pre></td><td>initial search range in bp (default: 1000) <tr class="bg1"><td><pre>max_range</pre></td><td>maximum search range in bp (default: 10000) 
</td></tr></tbody></table><p> Parameters are passed e.g.:</p><p><pre class="code sh_sh">--plugin NearestGene,limit=3,max_range=50000
</pre><p> This plugin requires a database connection. It cannot be run with Ensembl VEP
 in offline mode i.e. using the --offline flag.</p><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv NearestGene.pm ~/.vep/Plugins
./vep -i variations.vcf --cache --plugin NearestGene
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#E75480">Nearby features</div></td><td>-</td><td>Ensembl</td></tr>
<tr id="neXtProt" class="bg2 plugin_row" data-category="protein_data"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/neXtProt.pm">neXtProt</a></div></td><td><p> This is a plugin for the Ensembl Variant Effect Predictor (VEP) that
 retrieves data for missense and stop gain variants from neXtProt, which is a comprehensive 
 human-centric discovery platform that offers integration of and navigation 
 through protein-related data for example, variant information, localization 
 and interactions  (<a href="https://www.nextprot.org/">https://www.nextprot.org/</a>). <a class="button" href="#nextprot" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('nextprot');" id="a_nextprot">more</a></p><div id="div_nextprot" style="display:none;word-wrap:break-word;"><p> Please cite the neXtProt publication alongside Ensembl VEP if you use this resource:
 <a href="https://doi.org/10.1093/nar/gkz995">https://doi.org/10.1093/nar/gkz995</a></p><p> This plugin is only suitable for small sets of variants as an additional 
 individual remote API query is run for each variant.</p><p> The neXtProt_headers.txt file is a requirement for running this plugin and is found alongside the plugin in the
 VEP_plugins GitHub repository. The file contains the RDF entities extracted from 
 	<a href="https://snorql.nextprot.org/">https://snorql.nextprot.org/</a></p><p> Running options:
 (Default) the data retrieved by default is the MatureProtein, NucleotidePhosphateBindingRegion,
 Variant, MiscellaneousRegion, TopologicalDomain and InteractingRegion.
 The plugin can also be run with other options to retrieve other data than the default.</p><p> Options are passed to the plugin as key=value pairs:<table class="ss"><thead><tr><th>Argument</th><th>Description</th></tr></thead><tbody><tr class="bg1"><td><pre>max_set</pre></td><td>Set value to 1 to return all available protein-related data                   (includes the default data)</p><p><tr class="bg2"><td><pre>return_values</pre></td><td>The set of data to be returned with different data separated by <kbd>&</kbd>.                   Use file <kbd>neXtProt_headers.txt</kbd> to check which data (labels) are available.
                  Example: --plugin neXtProt,return_values=<kbd>Domain&InteractingRegion</kbd></p><p><tr class="bg1"><td><pre>url</pre></td><td>Set value to 1 to include the URL to link to the neXtProt entry. 
<tr class="bg2"><td><pre>all_labels</pre></td><td>Set value to 1 to include all labels, even if data is not available. 
<tr class="bg1"><td><pre>position</pre></td><td>Set value to 1 to include the start and end position in the protein. 
</td></tr></tbody></table><p> (*) note: <kbd>max_set</kbd> and <kbd>return_values</kbd> cannot be used simultaneously.</p><p> Output:
  By default, the plugin only returns data that is available. Example (default behaviour):
<pre class="code sh_sh">neXtProt_MatureProtein=Rho guanine nucleotide exchange factor 10
</pre>
  The option <kbd>all_labels</kbd> returns a consistent set of the requested fields, using "-" where 
  values are not available. Same example as above:
<pre class="code sh_sh">neXtProt_MatureProtein=Rho guanine nucleotide exchange factor 10;
neXtProt_InteractingRegion=-;neXtProt_NucleotidePhosphateBindingRegion=-;neXtProt_Variant=-;
neXtProt_MiscellaneousRegion=-;neXtProt_TopologicalDomain=-;
</pre>
  Of notice, multiple values can be returned for the same label. In this case, the values will
  be separeted by <kbd>|</kbd> for tab and txt format, and <kbd>&</kbd> for VCF format. </p><p>  N/B: This plugin requires a connection to the Ensembl database, and can not be used in offline mode.</p><p> The plugin can then be run as default:
<pre class="code sh_sh">./vep -i variations.vcf --plugin neXtProt
</pre><p> or to return only the data specified by the user:
<pre class="code sh_sh">./vep -i variations.vcf --plugin neXtProt,return_values=<kbd>Domain&InteractingRegion</kbd>
</pre><p>
<p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv neXtProt.pm ~/.vep/Plugins
./vep -i variations.vcf --plugin neXtProt
./vep -i variations.vcf --plugin neXtProt,max_set=1
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#77FF55">Protein data</div></td><td><a href="https://metacpan.org/pod/JSON::XS" rel="external">JSON::XS</a></td><td>Ensembl</td></tr>
<tr id="NMD" class="bg1 plugin_row" data-category="transcript_annotation"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/NMD.pm">NMD</a></div></td><td><p> This is a plugin for the Ensembl Variant Effect Predictor (VEP) 
 that predicts if a variant allows the 
 transcript escape nonsense-mediated mRNA decay based on certain rules. <a class="button" href="#nmd" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('nmd');" id="a_nmd">more</a></p><div id="div_nmd" style="display:none;word-wrap:break-word;"><p> The rules are : </p><p><ol><li value="1">The variant location  falls in the last exon of the transcript.<pre class="code sh_sh">                                        vvvv
      ES...EE..I.ES...EE.I.ES....EE.I.ES....EE 
</pre> (ES= exon_start,EE = exon_end, I = intron, v = variant location)</p><p></li><li value="2">The variant location falls  50 bases upstream of the penultimate (second to the last ) exon.<pre class="code sh_sh">                               vvv
      ES...EE..I.ES...EE.I.ES....EE.I.ES....EE 
</pre> (ES= exon_start,EE = exon_end, I = intron, v = variant location)</p><p></li><li value="3">The variant falls in the first 100 coding bases in the transcript.<pre class="code sh_sh">         vvv
      ..ES...EE..I.ES...EE.I.ES....EE.I.ES....EE 
</pre> (ES= exon_start,EE = exon_end, I = intron, v = variant location)</p><p></li><li value="4">If the variant is in an intronless transcript, meaning only one exon exist in the transcript.  
</li></ol><p> The additional term NMD-escaping variant (nonsense-mediated mRNA decay escaping variants) will be added if the variant matches any of the rules.</p><p> REFERENCES :
<ul><li>Identifying Genes Whose Mutant Transcripts Cause Dominant Disease Traits by Potential Gain-of-Function Alleles (Coban-Akdemir, 2018)</li><li>The rules and impact of nonsense-mediated mRNA decay in human cancers (Lindeboom, 2016)
</li></ul><p>
</ul><p></ol><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv NMD.pm  ~/.vep/Plugins
./vep -i variations.vcf --plugin NMD
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#FF2288">Transcript annotation</div></td><td>-</td><td>Ensembl</td></tr>
<tr id="OpenTargets" class="bg2 plugin_row" data-category="variant_data"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/OpenTargets.pm">OpenTargets</a></div></td><td><p> An Ensembl VEP plugin that integrates data from Open Targets Genetics
 (<a href="https://genetics.opentargets.org">https://genetics.opentargets.org</a>), a tool that highlights variant-centric
 statistical evidence to allow both prioritisation of candidate causal variants
 at trait-associated loci and identification of potential drug targets. <a class="button" href="#opentargets" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('opentargets');" id="a_opentargets">more</a></p><div id="div_opentargets" style="display:none;word-wrap:break-word;"><p> Data from Open Targets Genetics includes locus-to-gene (L2G) scores to predict
 causal genes at GWAS loci.</p><p> The tabix utility must be installed in your path to use this plugin. The Open
 Targets Genetics file and respective index (TBI) file can be downloaded from:
 <a href="https://ftp.ebi.ac.uk/pub/databases/opentargets/genetics/latest/OTGenetics_VEP">https://ftp.ebi.ac.uk/pub/databases/opentargets/genetics/latest/OTGenetics_VEP</a></p><p> Options are passed to the plugin as key=value pairs:<table class="ss"><thead><tr><th>Argument</th><th>Description</th></tr></thead><tbody><tr class="bg1"><td><pre>file</pre></td><td>(mandatory) Tabix-indexed file from Open Targets Genetics <tr class="bg2"><td><pre>cols</pre></td><td>(optional) Colon-separated list of columns to return from the plugin           file (default: "l2g:geneId"); use <kbd>all</kbd> to print all data</p><p></td></tr></tbody></table><p> Please cite the Open Targets Genetics publication alongside Ensembl VEP if you use
 this resource: <a href="https://doi.org/10.1093/nar/gkaa84">https://doi.org/10.1093/nar/gkaa84</a></p><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv OpenTargets.pm ~/.vep/Plugins</p><p># print Open Targets Genetics scores and respective gene identifiers (default)
./vep -i variations.vcf --plugin OpenTargets,file=path/to/data.tsv.bz</p><p># print all information from Open Targets Genetics
./vep -i variations.vcf --plugin OpenTargets,file=path/to/data.tsv.bz,cols=all
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#B22222">Variant data</div></td><td><ul style="padding-left:1em"><li><a href="https://metacpan.org/pod/Bio::SeqUtils" rel="external">Bio::SeqUtils</a></li><li><a href="https://metacpan.org/pod/File::Basename" rel="external">File::Basename</a></li></ul></td><td>Ensembl</td></tr>
<tr id="Paralogues" class="bg1 plugin_row" data-category="variant_data"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/Paralogues.pm">Paralogues</a></div></td><td><p> An Ensembl VEP plugin that fetches variants overlapping the genomic coordinates of amino
 acids aligned between paralogue proteins. This is useful to predict the
 pathogenicity of variants in paralogue positions. <a class="button" href="#paralogues" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('paralogues');" id="a_paralogues">more</a></p><div id="div_paralogues" style="display:none;word-wrap:break-word;"><p> This plugin can determine paralogue regions for a variant based on:
<ol><li value="1">Pre-computed matches between genomic regions and paralogue variants.      For this approach, either download the file calculated using ClinVar variants and respective TBI from
      <a href="https://ftp.ensembl.org/pub/current_variation/Paralogues">https://ftp.ensembl.org/pub/current_variation/Paralogues</a> or create such matches file yourself. Details on how
      to create such <kbd>matches</kbd> file can be found below.
</li><li value="2">Ensembl paralogue annotation. These versatile annotations can look up      paralogue regions for all variants from any species with Ensembl
      paralogues, but take longer to process.</p><p></li></ol><p> After retrieving the paralogue regions, this plugin fetches variants
 overlapping those regions from one of the following sources (by this order):
<ol><li value="1">Custom VCF via the <kbd>vcf</kbd> parameter</li><li value="2">Ensembl VEP cache (in cache/offline mode)</li><li value="3">Ensembl API (in database mode)
</li></ol><p> To create a <kbd>matches</kbd> file based on a custom set of variants, run Ensembl VEP using
 `--plugin Paralogues,regions=1,min_perc_cov=0,min_perc_pos=0,clnsig=ignore`
 and the <kbd>--vcf</kbd> option. Afterwards, process the output of the Ensembl VEP command:
 `perl -e "use Paralogues; Paralogues::prepare_matches_file(<kbd>variant_effect_output.txt</kbd>)"`</p><p> Options are passed to the plugin as key=value pairs:<table class="ss"><thead><tr><th>Argument</th><th>Description</th></tr></thead><tbody><tr class="bg1"><td><pre>matches</pre></td><td>Tabix-indexed TSV file with pre-computed matches between                   genomic regions and paralogue variants (fastest method); this
                  option is incompatible with the <kbd>paralogues</kbd> and <kbd>vcf</kbd> options</p><p><tr class="bg2"><td><pre>dir</pre></td><td>Directory with paralogue annotation (the annotation is created                   in this folder if the paralogue annotation files do not exist)
<tr class="bg1"><td><pre>paralogues</pre></td><td>Tabix-indexed TSV file with paralogue annotation (if the file                   does not exist, the annotation is automatically created); if
                  set to <kbd>remote</kbd>, the annotation is fetched but not stored
<tr class="bg2"><td><pre>vcf</pre></td><td>Tabix-indexed VCF file to fetch variant information (if not                   used, variants are fetched from the Ensembl VEP cache in cache/offline
                  mode or Ensembl API in database mode)</p><p><tr class="bg1"><td><pre>fields</pre></td><td>Colon-separated list of information from paralogue variants to                   output (default: <kbd>identifier:alleles:clinical_significance</kbd>);
                  keyword <kbd>all</kbd> can be used to print all fields; available
                  fields include <kbd>identifier</kbd>, <kbd>chromosome</kbd>, <kbd>start</kbd>, <kbd>alleles</kbd>,
                  <kbd>perc_cov</kbd>, <kbd>perc_pos</kbd>, and <kbd>clinical_significance</kbd> (if
                  <kbd>clnsig_col</kbd> is defined for custom VCF); additional fields
                  are available depending on variant source:
<ul><li>VEP cache: <kbd>end</kbd> and <kbd>strand</kbd></li><li>Ensembl API: <kbd>end</kbd>, <kbd>strand</kbd>, <kbd>source</kbd>, <kbd>consequence</kbd> and&nbsp;                      <kbd>gene_symbol</kbd>
</li><li>Custom VCF: <kbd>quality</kbd>, <kbd>filter</kbd> and name of INFO fields</li><li>Matches file: check column names in file header
<tr class="bg2"><td><pre></li></ul><p>   clnsig</pre></td><td>Clinical significance term to filter variants (default:                   <kbd>pathogenic</kbd>); use <kbd>ignore</kbd> to fetch all paralogue variants,
                  regardless of clinical significance
<tr class="bg1"><td><pre>clnsig_match</pre></td><td>Type of match when filtering variants based on option                   <kbd>clnsig</kbd>: <kbd>partial</kbd> (default), <kbd>exact</kbd> or <kbd>regex</kbd>
<tr class="bg2"><td><pre>clnsig_col</pre></td><td>Column name containing clinical significance in custom VCF                   (required with <kbd>vcf</kbd> option and if <kbd>clnsig</kbd> is not <kbd>ignore</kbd>)</p><p><tr class="bg1"><td><pre>min_perc_cov</pre></td><td>Minimum alignment percentage of the peptide associated with                   the input variant (default: 0)
<tr class="bg2"><td><pre>min_perc_pos</pre></td><td>Minimum percentage of positivity (similarity) between both                   homologues (default: 50)
<tr class="bg1"><td><pre>regions</pre></td><td>Boolean value to return regions used to look up paralogue <pre class="code sh_sh">                  variants (default: 1)
</pre><p></td></tr></tbody></table><p> The tabix utility must be installed in your path to read the paralogue
 annotation, the custom VCF file and the matches file.</p><p></ul><p></ol><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv Paralogues.pm ~/.vep/Plugins</p><p># Find paralogue regions of all input variants using Ensembl paralogue annotation
# (automatically created if not in current directory) and fetch variants within
# those regions from the Ensembl VEP cache and whose clinical significance partially
# matches 'pathogenic'
./vep -i variations.vcf --cache --plugin Paralogues</p><p># Find paralogue regions of input variants using Ensembl paralogue annotation
# (automatically created if not in current directory) and fetch variants within
# those regions from a custom VCF file (regardless of their clinical significance)
./vep -i variations.vcf --cache --plugin Paralogues,vcf=/path/to/file.vcf,clnsig=ignore</p><p># Same using a custom VCF file but filtering for 'pathogenic' variants
./vep -i variations.vcf --cache --plugin Paralogues,vcf=/path/to/file.vcf,clnsig_col=CLNSIG</p><p># Same but output different fields
./vep -i variations.vcf --cache --plugin Paralogues,vcf=/path/to/file.vcf.gz,clnsig_col=CLNSIG,fields=identifier:alleles:CLNSIG:CLNVI:GENEINFO</p><p># Use a file with regions matched to paralogue variants -- fastest method;
# download 'matches' files from https://ftp.ensembl.org/pub/current_variation/Paralogues
./vep -i variations.vcf --cache --plugin Paralogues,matches=Paralogues.pm_homo_sapiens_113_GRCh38_clinvar_20240107.tsv.gz,clnsig=ignore</p><p># Same using a 'matches' file but filtering for 'pathogenic' variants (default)
./vep -i variations.vcf --cache --plugin Paralogues,matches=Paralogues.pm_homo_sapiens_113_GRCh38_clinvar_20240107.tsv.gz</p><p># Fetch all Ensembl variants in paralogue proteins using only the Ensembl API
# (requires database access)
./vep -i variations.vcf --database --plugin Paralogues,mode=remote,clnsig=ignore
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#B22222">Variant data</div></td><td><ul style="padding-left:1em"><li><a href="https://metacpan.org/pod/Compress::Zlib" rel="external">Compress::Zlib</a></li><li><a href="https://metacpan.org/pod/Bio::SimpleAlign" rel="external">Bio::SimpleAlign</a></li><li><a href="https://metacpan.org/pod/File::Spec" rel="external">File::Spec</a></li><li><a href="https://metacpan.org/pod/List::Util" rel="external">List::Util</a> qw(any)</li><li><a href="https://metacpan.org/pod/File::Basename" rel="external">File::Basename</a></li></ul></td><td>Ensembl</td></tr>
<tr id="PhenotypeOrthologous" class="bg2 plugin_row" data-category="phenotype_data_and_citations"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/PhenotypeOrthologous.pm">PhenotypeOrthologous</a></div></td><td><p> An Ensembl VEP plugin that retrieves phenotype information associated with orthologous genes from model organisms. <a class="button" href="#phenotypeorthologous" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('phenotypeorthologous');" id="a_phenotypeorthologous">more</a></p><div id="div_phenotypeorthologous" style="display:none;word-wrap:break-word;"><p> The plugin annotates human variants and reports orthologous information from rat and mouse. 
 The plugin is only available for GRCh38.</p><p> The PhenotypeOrthologous file can be downloaded from <a href="https://ftp.ensembl.org/pub/current_variation/PhenotypeOrthologous">https://ftp.ensembl.org/pub/current_variation/PhenotypeOrthologous</a></p><p> The plugin can be run: </p><p><pre class="code sh_sh">./vep -i variations.vcf --plugin PhenotypeOrthologous,file=PhenotypesOrthologous_homo_sapiens_112_GRCh38.gff3.gz
</pre><p>  The file option is mandatory to run this plugin </p><p>  To return only results for rat :
<pre class="code sh_sh">./vep -i variations.vcf --plugin PhenotypeOrthologous,file=PhenotypesOrthologous_homo_sapiens_112_GRCh38.gff3.gz,model=rat
</pre><p>  To return only results for mouse:
<pre class="code sh_sh">./vep -i variations.vcf --plugin PhenotypeOrthologous,file=PhenotypesOrthologous_homo_sapiens_112_GRCh38.gff3.gz,model=mouse
</pre><p> The tabix utility must be installed in your path to use this plugin.
 Check <a href="https://github.com/samtools/htslib.git">https://github.com/samtools/htslib.git</a> for instructions.</p><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv PhenotypeOrthologous.pm ~/.vep/Plugins
./vep -i variations.vcf --plugin PhenotypeOrthologous,file=PhenotypesOrthologous_homo_sapiens_112_GRCh38.gff3.gz
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#556688">Phenotype data and citations</div></td><td>-</td><td>Ensembl</td></tr>
<tr id="Phenotypes" class="bg1 plugin_row" data-category="phenotype_data_and_citations"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/Phenotypes.pm">Phenotypes</a></div></td><td><p> A VEP plugin that retrieves overlapping phenotype information. <a class="button" href="#phenotypes" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('phenotypes');" id="a_phenotypes">more</a></p><div id="div_phenotypes" style="display:none;word-wrap:break-word;"><p> On the first run for each new version/species/assembly will
 download a GFF-format dump to ~/.vep/Plugins/</p><p> Ensembl provides phenotype annotations mapped to a number of genomic
 feature types, including genes, variants and QTLs.</p><p> This plugin is best used with JSON output format; the output will be
 more verbose and include all available phenotype annotation data and
 metadata.</p><p> For other output formats, only a concatenated list of phenotype
 description strings is returned.</p><p> Several paramters can be set using a key=value system:<table class="ss"><thead><tr><th>Argument</th><th>Description</th></tr></thead><tbody>
<tr class="bg1"><td><pre>dir</pre></td><td>Path to directory where to look for phenotypes annotation. If                   the required file does not exist, the file is downloaded and
                  saved in the provided directory (download requires using
                  database or cache mode).</p><p><tr class="bg2"><td><pre>file</pre></td><td>File path to phenotypes annotation. If the file does                   not exist, the file is downloaded and saved with this name
                  (download requires using database or cache mode).</p><p></td></tr></tbody></table><p> exclude_sources: &-separated list of phenotype sources to exclude. By default,
                  HGMD-PUBLIC and COSMIC annotations are excluded. See
                  <a href="http://www.ensembl.org/info/genome/variation/phenotype/sources_phenotype_documentation.html">http://www.ensembl.org/info/genome/variation/phenotype/sources_phenotype_documentation.html</a></p><p> include_sources: &-separated list of phenotype sources to include. If defined,
                  exclude_sources is ignored.</p><p> exclude_types  : &-separated list of feature types to exclude: Gene, Variation,
                  QTL, StructuralVariation, SupportingStructuralVariation,
                  RegulatoryFeature. By default, StructuralVariation and 
                  SupportingStructuralVariation annotations are always excluded
                  (due to size issues) and Variation is excluded when annotating
                  structural variants; to get these annotations in all cases, use
                  include_types=StructuralVariation&SupportingStructuralVariation&Variation</p><p> include_types  : &-separated list of feature types to include. If defined,
                  exclude_types is ignored.</p><p> expand_right   : Cache size in bp. By default, annotations 100000bp (100kb)
                  downstream of the initial lookup are cached.</p><p> phenotype_feature : Boolean to report the gene/variation associated with the
                     phenotype (such as overlapping gene or structural
<pre class="code sh_sh">                     variation) and annotation source (default: 0)
</pre><p> cols           : &-separated list of column and/or attribute names to output
                  from the gff file. The output fields will be ordered in the same
                  way given in cols argument. (default: <kbd>phenotype</kbd> or <kbd>source,phenotype,id</kbd>
                  if you set phenotype_feature=1)</p><p> id_match       : Return results only if the identifiers matches with the 
<pre class="code sh_sh">                  variant or the gene depending on the type (default: 0)
</pre><p> Example:</p><p><pre class="code sh_sh">--plugin Phenotypes,file=${HOME}/phenotypes.gff.gz,include_types=Gene
--plugin Phenotypes,dir=${HOME},include_types=Gene
</pre><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv Phenotypes.pm ~/.vep/Plugins</p><p># Automatically download phenotype annotation files if needed and annotate
# variants with phenotypes
./vep -i variations.vcf --plugin Phenotypes</p><p># Fetch only gene-associated phenotypes
./vep -i variations.vcf --plugin Phenotypes,include_types=Gene</p><p># Set directory with phenotypes annotations (phenotype annotation file is
# automatically downloaded if not available in this directory)
./vep -i variations.vcf --plugin Phenotypes,dir=${HOME},include_types=Gene</p><p># Specify a file with phenotypes annotation (file is automatically
# downloaded and saved with this name if it does not exist)
./vep -i variations.vcf --plugin Phenotypes,file=${HOME}/phenotypes.gff.gz,include_types=Gene
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#556688">Phenotype data and citations</div></td><td>-</td><td>Ensembl</td></tr>
<tr id="pLI" class="bg2 plugin_row" data-category="gene_tolerance_to_change"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/pLI.pm">pLI</a></div></td><td><p>  An Ensembl VEP plugin that adds the probabililty of a gene being 
  loss-of-function intolerant (pLI) to the Ensembl VEP output. <a class="button" href="#pli" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('pli');" id="a_pli">more</a></p><div id="div_pli" style="display:none;word-wrap:break-word;"><p>  Lek et al. (2016) estimated pLI using the expectation-maximization 
  (EM) algorithm and data from 60,706 individuals from 
  ExAC (<a href="http://exac.broadinstitute.org">http://exac.broadinstitute.org</a>). The closer pLI is to 1, 
  the more likely the gene is loss-of-function (LoF) intolerant. </p><p>  Note: the pLI was calculated using a representative transcript and
  is reported by gene in the plugin.</p><p>  The data for the plugin is provided by Kaitlin Samocha and Daniel MacArthur. 
  See <a href="https://www.ncbi.nlm.nih.gov/pubmed/27535533">https://www.ncbi.nlm.nih.gov/pubmed/27535533</a> for a description 
  of the dataset and analysis.</p><p>  The pLI_values.txt file is found alongside the plugin in the 
  VEP_plugins GitHub repository. The file contains the fields gene and pLI 
  extracted from the file at </p><p>    <a href="https://ftp.broadinstitute.org/pub/ExAC_release/release0.3/functional_gene_constraint/fordist_cleaned_exac_r03_march16_z_pli_rec_null_data.txt">https://ftp.broadinstitute.org/pub/ExAC_release/release0.3/functional_gene_constraint/fordist_cleaned_exac_r03_march16_z_pli_rec_null_data.txt</a></p><p>  From this file, extract gene or transcipt pLI scores:
   To extract gene scores : 
<pre class="code sh_sh">awk '{print $2, $20 }'  fordist_cleaned_exac_r03_march16_z_pli_rec_null_data.txt &gt plI_gene.txt 
</pre><p>  NB: The gene scores file can also be found in the VEP_plugins directory.</p><p>   To extract transcript scores:
<pre class="code sh_sh">awk '{print $1, $20 }'  fordist_cleaned_exac_r03_march16_z_pli_rec_null_data.txt &gt plI_transcript.txt 
</pre><p>  NB: Using this file, No transcript score will be returned.</p><p>  To use another values file, add it as a parameter i.e.
<pre class="code sh_sh">./vep -i variants.vcf --plugin pLI,values_file.txt
./vep -i variants.vcf --plugin pLI,values_file.txt,transcript # to check for the transcript score.
</pre><p>  gnomAD v4 release expanded the scale of pLI score calculation. The file can be downloaded from -
    <a href="https://gnomad.broadinstitute.org/downloads#v4-constraint">https://gnomad.broadinstitute.org/downloads#v4-constraint</a> (Constraint metrics TSV)
  To use the data you can follow the same procedure as above but needs to change the column number to accordingly.</p><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv pLI.pm ~/.vep/Plugins
mv pLI_values.txt ~/.vep/Plugins
./vep -i variants.vcf --plugin pLI
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#00DD66">Gene tolerance to change</div></td><td><ul style="padding-left:1em"><li><a href="https://metacpan.org/pod/List::MoreUtils" rel="external">List::MoreUtils</a> qw/zip/</li><li><a href="https://metacpan.org/pod/DBI" rel="external">DBI</a></li></ul></td><td>Ensembl</td></tr>
<tr id="PolyPhen_SIFT" class="bg1 plugin_row" data-category="pathogenicity_predictions"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/PolyPhen_SIFT.pm">PolyPhen_SIFT</a></div></td><td><p> An Ensembl VEP plugin that retrieves PolyPhen and SIFT predictions from a
 locally constructed SQLite database. It can be used when your main
 source of the Ensembl VEP transcript annotation (e.g. a GFF file or GFF-based cache)
 does not contain these predictions. <a class="button" href="#polyphen_sift" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('polyphen_sift');" id="a_polyphen_sift">more</a></p><div id="div_polyphen_sift" style="display:none;word-wrap:break-word;"><p> You must create a SQLite database of the predictions or point to the SQLite
 database file already created. Compatible SQLite databases based on pangenome
 data are available at <a href="http://ftp.ensembl.org/pub/current_variation/pangenomes">http://ftp.ensembl.org/pub/current_variation/pangenomes</a></p><p> You may point to the file by adding parameter <kbd>db=[file]</kbd>. If the file is not
 in <kbd>HOME/.vep</kbd>, you can also use parameter <kbd>dir=[dir]</kbd> to indicate its path.</p><p><pre class="code sh_sh">--plugin PolyPhen_SIFT,db=[file]
--plugin PolyPhen_SIFT,db=[file],dir=[dir]
</pre><p> To create a SQLite database using PolyPhen/SIFT data from the Ensembl database,
 you must have an active database connection (i.e. not using <kbd>--offline</kbd>) and
 add parameter <kbd>create_db=1</kbd>. This will create a SQLite file named
 <kbd>[species].PolyPhen_SIFT.db</kbd>, placed in the directory specified by the <kbd>dir</kbd>
 parameter:</p><p><pre class="code sh_sh">--plugin PolyPhen_SIFT,create_db=1
--plugin PolyPhen_SIFT,create_db=1,dir=/some/specific/directory
</pre><p> *** NB: this will take some hours! ***</p><p> When creating a PolyPhen_SIFT by simply using <kbd>create_db=1</kbd>, you do not need to
 specify any parameters to load the appropriate file based on the species:</p><p><pre class="code sh_sh">--plugin PolyPhen_SIFT
</pre><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv PolyPhen_SIFT.pm ~/.vep/Plugins</p><p># Read default PolyPhen/SIFT SQLite file in $HOME/.vep
./vep -i variations.vcf -cache --plugin PolyPhen_SIFT</p><p># Read database with custom name and/or located in a custom directory
./vep -i variations.vcf -cache --plugin PolyPhen_SIFT,db=custom.db
./vep -i variations.vcf -cache --plugin PolyPhen_SIFT,dir=/some/custom/dir
./vep -i variations.vcf -cache --plugin PolyPhen_SIFT,db=custom.db,dir=/some/custom/dir</p><p># Create PolyPhen/SIFT SQLite file based on Ensembl database
./vep -i variations.vcf -cache --plugin PolyPhen_SIFT,create_db=1
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#1E90FF">Pathogenicity predictions</div></td><td><ul style="padding-left:1em"><li><a href="https://metacpan.org/pod/Digest::MD5" rel="external">Digest::MD5</a> qw(md5_hex)</li><li><a href="https://metacpan.org/pod/DBI" rel="external">DBI</a></li></ul></td><td>Ensembl</td></tr>
<tr id="PON_P2" class="bg2 plugin_row" data-category="pathogenicity_predictions"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/PON_P2.pm">PON_P2</a></div></td><td><p> This plugin for Ensembl Variant Effect Predictor (VEP) computes the predictions of PON-P2
 for amino acid substitutions in human proteins. <a class="button" href="#pon_p2" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('pon_p2');" id="a_pon_p2">more</a></p><div id="div_pon_p2" style="display:none;word-wrap:break-word;"><p> PON-P2 is developed and maintained by Protein Structure and Bioinformatics Group
 at Lund University and is available at <a href="http://structure.bmc.lu.se/PON-P2/">http://structure.bmc.lu.se/PON-P2/</a>.</p><p> If you use this data, please cite the following publication
 Niroula, A., Vihinen, M. Harmful somatic amino acid substitutions affect key pathways in cancers.
 BMC Med Genomics 8, 53 (2015). <a href="https://doi.org/10.1186/s12920-015-0125-x">https://doi.org/10.1186/s12920-015-0125-x</a></p><p> There are two ways to run the plugin:</p><p><ol><li value="1">To compute the predictions from the PON-P2 API, use python script <kbd>ponp2.py</kbd> (*)    and select the reference genome (acceptable values are: hg37 and hg38):
<pre class="code sh_sh">--plugin PON_P2,pyscript=/path/to/python/script/ponp2.py,hg=hg37
</pre><p>    (*) To run this mode, you will require a python script and its dependencies (Python,
    python suds). The python file can be downloaded from <a href="http://structure.bmc.lu.se/PON-P2/vep.html/">http://structure.bmc.lu.se/PON-P2/vep.html/</a>
    and the complete path to this file must be supplied while using this plugin.</p><p></li><li value="2">To fetch the predictions from a file containing pre-calculated predictions for somatic variations  please use the following key=value option (only available for GRCh37):<table class="ss"><thead><tr><th>Argument</th><th>Description</th></tr></thead><tbody>
</li></ol><p><tr class="bg1"><td><pre>file</pre></td><td>COSMIC text file with pre-calculated predictions downloaded from                   <a href="http://structure.bmc.lu.se/PON-P2/cancer30.html/">http://structure.bmc.lu.se/PON-P2/cancer30.html/</a></p><p></td></tr></tbody></table><p>   The following steps are necessary before using the file:
<pre class="code sh_sh">(head -n 1 COSMIC.txt && tail -n +2 COSMIC.txt | sort -t $'\t' -k1,1 -k2,2n) &gt cosmic_sorted.txt
sed -i 's/Chromosome/#Chromosome/' cosmic_sorted.txt
bgzip cosmic_sorted.txt
tabix -s 1 -b 2 -e 2 cosmic_sorted.txt.gz
</pre><p><pre class="code sh_sh">--plugin PON_P2,file=path/to/cosmic_sorted.txt.gz 
</pre><p></ol><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv PON_P2.pm ~/.vep/Plugins
./vep -i variations.vcf --plugin PON_P2,pyscript=/path/to/python/script/ponp2.py,hg=hg37
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#1E90FF">Pathogenicity predictions</div></td><td>-</td><td><ul style="padding-left:1em"><li>Abhishek Niroula</li><li>Mauno Vihinen</li></ul></td></tr>
<tr id="PostGAP" class="bg1 plugin_row" data-category="phenotype_data_and_citations"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/PostGAP.pm">PostGAP</a></div></td><td><p>A VEP plugin that retrieves data for variants from a tabix-indexed PostGAP file (1-based file). <a class="button" href="#postgap" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('postgap');" id="a_postgap">more</a></p><div id="div_postgap" style="display:none;word-wrap:break-word;"><p>Please refer to the PostGAP github and wiki for more information:
<a href="https://github.com/Ensembl/postgap">https://github.com/Ensembl/postgap</a>
<a href="https://github.com/Ensembl/postgap/wiki">https://github.com/Ensembl/postgap/wiki</a>
<a href="https://github.com/Ensembl/postgap/wiki/algorithm-pseudo-code">https://github.com/Ensembl/postgap/wiki/algorithm-pseudo-code</a></p><p>The Bio::DB::HTS perl library or tabix utility must be installed in your path
to use this plugin. The PostGAP data file can be downloaded from
<a href="https://storage.googleapis.com/postgap-data">https://storage.googleapis.com/postgap-data</a>.</p><p>The file must be processed and indexed by tabix before use by this plugin.
PostGAP has coordinates for both GRCh38 and GRCh37; the file must be
processed differently according to the assembly you use.</p><p><pre class="code sh_sh">wget <a href="https://storage.googleapis.com/postgap-data/postgap.txt.gz">https://storage.googleapis.com/postgap-data/postgap.txt.gz</a>
gunzip postgap.txt.gz
</pre><p># GRCh38
<pre class="code sh_sh">(grep ^"ld_snp_rsID" postgap.txt; grep -v ^"ld_snp_rsID" postgap.txt | sort -k4,4 -k5,5n ) | bgzip &gt postgap_GRCh38.txt.gz
tabix -s 4 -b 5 -e 5 -c l postgap_GRCh38.txt.gz
</pre><p># GRCh37
<pre class="code sh_sh">(grep ^"ld_snp_rsID" postgap.txt; grep -v ^"ld_snp_rsID" postgap.txt | sort -k2,2 -k3,3n ) | bgzip &gt postgap_GRCh37.txt.gz
tabix -s 2 -b 3 -e 3 -c l postgap_GRCh37.txt.gz
</pre><p>Note that in the last command we tell tabix that the header line starts with "l";
this may change to the default of "#" in future versions of PostGAP.</p><p>When running the plugin by default <kbd>disease_efo_id</kbd>, <kbd>disease_name</kbd>, <kbd>gene_id</kbd>
and <kbd>score</kbd> information is returned e.g.</p><p><pre class="code sh_sh">--plugin POSTGAP,/path/to/PostGap.gz
</pre><p>You may include all columns with ALL; this fetches a large amount of data per
variant!:</p><p><pre class="code sh_sh">--plugin POSTGAP,/path/to/PostGap.gz,ALL
</pre><p>You may want to select only a specific subset of additional information to be
reported, you can do this by specifying the columns as parameters to the plugin e.g.</p><p><pre class="code sh_sh">--plugin POSTGAP,/path/to/PostGap.gz,gwas_pmid,gwas_size
</pre><p>If a requested column is not found, the error message will report the
complete list of available columns in the POSTGAP file. For a brief description
of the available information please refer to the 'How do I use POSTGAP output?'
section in the POSTGAP wiki.</p><p>Tabix also allows the data file to be hosted on a remote server. This plugin is
fully compatible with such a setup - simply use the URL of the remote file:</p><p><pre class="code sh_sh">--plugin PostGAP,<a href="http://my.files.com/postgap.txt.gz">http://my.files.com/postgap.txt.gz</a>
</pre><p>Note that gene sequences referred to in PostGAP may be out of sync with
those in the latest release of Ensembl; this may lead to discrepancies with
scores retrieved from other sources.</p><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv PostGAP.pm ~/.vep/Plugins
./vep -i variations.vcf --plugin PostGAP,/path/to/PostGap.gz,col1,col2
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#556688">Phenotype data and citations</div></td><td>-</td><td>Ensembl</td></tr>
<tr id="PrimateAI" class="bg2 plugin_row" data-category="pathogenicity_predictions"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/PrimateAI.pm">PrimateAI</a></div></td><td><p>  The PrimateAI VEP plugin is designed to retrieve clinical impact scores of
  variants, as described in <a href="https://www.nature.com/articles/s41588-018-0167-z">https://www.nature.com/articles/s41588-018-0167-z</a>.
  Please consider citing the paper if using this plugin. <a class="button" href="#primateai" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('primateai');" id="a_primateai">more</a></p><div id="div_primateai" style="display:none;word-wrap:break-word;"><p>  In brief, common missense mutations in non-human primate species are usually
  benign in humans. Thousands of common variants from six non-human primate
  species were used to train a deep neural network to identify pathogenic 
  mutations in humans with a rare disease.</p><p>  This plugin uses files generated by the PrimateAI software, which is
  available from <a href="https://github.com/Illumina/PrimateAI">https://github.com/Illumina/PrimateAI</a>. The files containing
  predicted pathogenicity scores can be downloaded from
  <a href="https://basespace.illumina.com/s/yYGFdGih1rXL">https://basespace.illumina.com/s/yYGFdGih1rXL</a> (a free BaseSpace account may
  be required):
      PrimateAI_scores_v0.2.tsv.gz (for GRCh37/hg19)
      PrimateAI_scores_v0.2_hg38.tsv.gz (for GRCh38/hg38)</p><p>  Before running the plugin for the first time, the following steps must be 
  taken to format the downloaded files:</p><p><ol><li value="1"> Unzip the score files</li><li value="2"> Add '#' in front of the column description line</li><li value="3"> Remove any empty lines.</li><li value="4"> Sort the file by chromosome and position</li><li value="5"> Compress the file in .bgz format</li><li value="6"> Create tabix index (requires tabix to be installed).
</li></ol><p>  Command line examples for formatting input files:
<pre class="code sh_sh">gunzip -cf PrimateAI_scores_v0.2.tsv.gz | sed '12s/.*/#&/' | sed '/^$/d' | awk 'NR&lt12{print $0;next}{print $0 | "sort -k1,1 -k 2,2n -V"}' | bgzip &gt PrimateAI_scores_v0.2_GRCh37_sorted.tsv.bgz
tabix -s 1 -b 2 -e 2 PrimateAI_scores_v0.2_GRCh37_sorted.tsv.bgz
</pre><p><pre class="code sh_sh">gunzip -cf PrimateAI_scores_v0.2_hg38.tsv.gz | sed '12s/.*/#&/' | sed '/^$/d' | awk 'NR&lt12{print $0;next}{print $0 | "sort -k1,1 -k 2,2n -V"}' | bgzip &gt PrimateAI_scores_v0.2_GRCh38_sorted.tsv.bgz
tabix -s 1 -b 2 -e 2 PrimateAI_scores_v0.2_GRCh38_sorted.tsv.bgz
</pre><p></ol><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv PrimateAI.pm ~/.vep/Plugins</p><p>./vep -i variations.vcf --plugin PrimateAI,PrimateAI_scores_v0.2_GRCh37_sorted.tsv.bgz
./vep -i variations.vcf --plugin PrimateAI,PrimateAI_scores_v0.2_GRCh38_sorted.tsv.bgz
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#1E90FF">Pathogenicity predictions</div></td><td>-</td><td>Ensembl</td></tr>
<tr id="ProteinSeqs" class="bg1 plugin_row" data-category="sequence"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/ProteinSeqs.pm">ProteinSeqs</a></div></td><td><p> This is a plugin for the Ensembl Variant Effect Predictor (VEP) that
 prints out the reference and mutated protein sequences of any
 proteins found with non-synonymous mutations in the input file.  <a class="button" href="#proteinseqs" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('proteinseqs');" id="a_proteinseqs">more</a></p><div id="div_proteinseqs" style="display:none;word-wrap:break-word;"><p> You should supply the name of file where you want to store the
 reference protein sequences as the first argument, and a file to
 store the mutated sequences as the second argument.</p><p> Note that, for simplicity, where stop codons are gained the plugin 
 simply substitutes a '*' into the sequence and does not truncate the 
 protein. Where a stop codon is lost any new amino acids encoded by the 
 mutation are appended to the sequence, but the plugin does not attempt 
 to translate until the next downstream stop codon. Also, the protein
 sequence resulting from each mutation is printed separately, no attempt
 is made to apply multiple mutations to the same protein.</p><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv ProteinSeqs.pm ~/.vep/Plugins
./vep -i variations.vcf --plugin ProteinSeqs,reference.fa,mutated.fa
./vep -i variations.vcf --plugin ProteinSeqs,reference=reference.fa,mutated=mutated.fa
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#5F81A9">Sequence</div></td><td>-</td><td>Ensembl</td></tr>
<tr id="ReferenceQuality" class="bg2 plugin_row" data-category="sequence"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/ReferenceQuality.pm">ReferenceQuality</a></div></td><td><p> This is a plugin for the Ensembl Variant Effect Predictor (VEP) that
 reports on the quality of the reference genome using GRC data at the location of your variants.
 More information can be found at: <a href="https://www.ncbi.nlm.nih.gov/grc/human/issues">https://www.ncbi.nlm.nih.gov/grc/human/issues</a> <a class="button" href="#referencequality" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('referencequality');" id="a_referencequality">more</a></p><div id="div_referencequality" style="display:none;word-wrap:break-word;"><p> The following steps are necessary before running this plugin:</p><p> GRCh38:</p><p><pre class="code sh_sh">wget <a href="https://ftp.ncbi.nlm.nih.gov/pub/grc/human/GRC/GRCh38/MISC/annotated_clone_assembly_problems_GCF_000001405.38.gff3">https://ftp.ncbi.nlm.nih.gov/pub/grc/human/GRC/GRCh38/MISC/annotated_clone_assembly_problems_GCF_000001405.38.gff3</a>
wget <a href="https://ftp.ncbi.nlm.nih.gov/pub/grc/human/GRC/Issue_Mapping/GRCh38.p12_issues.gff3">https://ftp.ncbi.nlm.nih.gov/pub/grc/human/GRC/Issue_Mapping/GRCh38.p12_issues.gff3</a>
cat annotated_clone_assembly_problems_GCF_000001405.38.gff3 GRCh38.p12_issues.gff3 &gt GRCh38_quality_mergedfile.gff3
sort -k 1,1 -k 4,4n -k 5,5n GRCh38_quality_mergedfile.gff3 &gt sorted_GRCh38_quality_mergedfile.gff3
bgzip sorted_GRCh38_quality_mergedfile.gff3
tabix -p gff sorted_GRCh38_quality_mergedfile.gff3.gz
</pre><p> The plugin can then be run with:
<pre class="code sh_sh">./vep -i variations.vcf --plugin ReferenceQuality,sorted_GRCh38_quality_mergedfile.gff3.gz
</pre><p> GRCh37:</p><p><pre class="code sh_sh">wget <a href="https://ftp.ncbi.nlm.nih.gov/pub/grc/human/GRC/GRCh37/MISC/annotated_clone_assembly_problems_GCF_000001405.25.gff3">https://ftp.ncbi.nlm.nih.gov/pub/grc/human/GRC/GRCh37/MISC/annotated_clone_assembly_problems_GCF_000001405.25.gff3</a>
wget <a href="https://ftp.ncbi.nlm.nih.gov/pub/grc/human/GRC/Issue_Mapping/GRCh37.p13_issues.gff3">https://ftp.ncbi.nlm.nih.gov/pub/grc/human/GRC/Issue_Mapping/GRCh37.p13_issues.gff3</a>
cat annotated_clone_assembly_problems_GCF_000001405.25.gff3 GRCh37.p13_issues.gff3 &gt GRCh37_quality_mergedfile.gff3
sort -k 1,1 -k 4,4n -k 5,5n GRCh37_quality_mergedfile.gff3 &gt sorted_GRCh37_quality_mergedfile.gff3
bgzip sorted_GRCh37_quality_mergedfile.gff3
tabix -p gff sorted_GRCh37_quality_mergedfile.gff3.gz
</pre><p> The plugin can then be run with:
<pre class="code sh_sh">./vep -i variations.vcf --plugin ReferenceQuality,sorted_GRCh37_quality_mergedfile.gff3.gz
</pre><p> </p><p> The tabix utility must be installed in your path to use this plugin.</p><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv ReferenceQuality.pm ~/.vep/Plugins
./vep -i variations.vcf --plugin ReferenceQuality,/path/to/data.gff3.gz
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#5F81A9">Sequence</div></td><td>-</td><td>Ensembl</td></tr>
<tr id="REVEL" class="bg1 plugin_row" data-category="pathogenicity_predictions"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/REVEL.pm">REVEL</a></div></td><td><p> This is a plugin for the Ensembl Variant Effect Predictor (VEP) that
 adds the REVEL score for missense variants to the Ensembl VEP output. <a class="button" href="#revel" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('revel');" id="a_revel">more</a></p><div id="div_revel" style="display:none;word-wrap:break-word;"><p> Please cite the REVEL publication alongside Ensembl VEP if you use this resource:
 <a href="https://www.ncbi.nlm.nih.gov/pubmed/27666373">https://www.ncbi.nlm.nih.gov/pubmed/27666373</a></p><p> Running options:
 If available, the plugin will match the scores by transcript id (default).
 Using the flag <kbd>1</kbd> the plugin will not try to match by transcript id.</p><p> REVEL scores can be downloaded from: <a href="https://sites.google.com/site/revelgenomics/downloads">https://sites.google.com/site/revelgenomics/downloads</a></p><p> The plugin supports several REVEL file versions:
<ul><li>REVEL file version Dec 2017, which has 7 columns and only GRCh37 coordinates</li><li>REVEL file version Feb 2020, which has 8 columns with GRCh37 and GRCh38 coordinates</li><li>REVEL file version May 2021, which has 9 columns with GRCh37 and GRCh38 coordinates and a new column with transcript ids 
</li></ul><p> These files can be tabix-processed by:
<pre class="code sh_sh">unzip revel-v1.3_all_chromosomes.zip
cat revel_with_transcript_ids | tr "," "\t" &gt tabbed_revel.tsv
sed '1s/.*/#&/' tabbed_revel.tsv &gt new_tabbed_revel.tsv
bgzip new_tabbed_revel.tsv
</pre><p> for GRCh37:
<pre class="code sh_sh">tabix -f -s 1 -b 2 -e 2 new_tabbed_revel.tsv.gz
</pre><p> for GRCh38:
<pre class="code sh_sh">zcat new_tabbed_revel.tsv.gz | head -n1 &gt h
zgrep -h -v ^#chr new_tabbed_revel.tsv.gz | awk '$3 != "." ' | sort -k1,1 -k3,3n - | cat h - | bgzip -c &gt new_tabbed_revel_grch38.tsv.gz
tabix -f -s 1 -b 3 -e 3 new_tabbed_revel_grch38.tsv.gz
</pre><p> The plugin can then be run as default:
<pre class="code sh_sh">./vep -i variations.vcf --assembly GRCh38 --plugin REVEL,file=/path/to/revel/data.tsv.gz
</pre><p> or with the option to not match by transcript id:
<pre class="code sh_sh">./vep -i variations.vcf --assembly GRCh38 --plugin REVEL,file=/path/to/revel/data.tsv.gz,no_match=1
</pre><p> Requirements:
  The tabix utility must be installed in your path to use this plugin.
  The --assembly flag is required to use this plugin.</p><p></ul><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv REVEL.pm ~/.vep/Plugins
./vep -i variations.vcf --assembly GRCh37 --plugin REVEL,file=/path/to/revel/data.tsv.gz
./vep -i variations.vcf --assembly GRCh38 --plugin REVEL,file=/path/to/revel/data.tsv.gz
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#1E90FF">Pathogenicity predictions</div></td><td>-</td><td>Ensembl</td></tr>
<tr id="RiboseqORFs" class="bg2 plugin_row" data-category="transcript_annotation"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/RiboseqORFs.pm">RiboseqORFs</a></div></td><td><p> This is an Ensembl VEP plugin that uses a standardized catalog of human Ribo-seq ORFs to
 re-calculate consequences for variants located in these translated regions. <a class="button" href="#riboseqorfs" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('riboseqorfs');" id="a_riboseqorfs">more</a></p><div id="div_riboseqorfs" style="display:none;word-wrap:break-word;"><p> This plugin reports new consequences based on the evidence from the Ribo-seq
 ORF annotation and supporting publications. The human Ribo-seq ORF data can be
 downloaded from: <a href="https://ftp.ebi.ac.uk/pub/databases/gencode/riboseq_orfs/data">https://ftp.ebi.ac.uk/pub/databases/gencode/riboseq_orfs/data</a></p><p> After downloading the annotation, please bgzip and tabix it:
<pre class="code sh_sh">bgzip Ribo-seq_ORFs.bed
tabix Ribo-seq_ORFs.bed.gz
</pre><p> For optimal performance when running this plugin in Ensembl VEP, please use a FASTA
 file (<kbd>--fasta</kbd>). A FASTA file is always required in offline mode.</p><p> Please cite the publication for the Ribo-seq ORF annotation alongside Ensembl VEP
 if you use this resource: <a href="https://doi.org/10.1038/s41587-022-01369-0">https://doi.org/10.1038/s41587-022-01369-0</a></p><p> The tabix utility must be installed in your path to use this plugin.</p><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
./vep -i variations.vcf --plugin RiboseqORFs,file=/path/to/Ribo-seq_ORFs.bed.gz
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#FF2288">Transcript annotation</div></td><td>-</td><td>Ensembl</td></tr>
<tr id="SameCodon" class="bg1 plugin_row" data-category="variant_data"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/SameCodon.pm">SameCodon</a></div></td><td><p> An Ensembl VEP plugin that reports existing variants that fall in the same codon.
 This plugin requires a database connection, can not be run in offline mode <a class="button" href="#samecodon" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('samecodon');" id="a_samecodon">more</a></p><div id="div_samecodon" style="display:none;word-wrap:break-word;"><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv SameCodon.pm ~/.vep/Plugins
./vep -i variations.vcf --plugin SameCodon
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#B22222">Variant data</div></td><td>-</td><td>Ensembl</td></tr>
<tr id="satMutMPRA" class="bg2 plugin_row" data-category="phenotype_data_and_citations"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/satMutMPRA.pm">satMutMPRA</a></div></td><td><p>An Ensembl VEP plugin that retrieves data for variants from a tabix-indexed satMutMPRA file (1-based file).
The saturation mutagenesis-based massively parallel reporter assays (satMutMPRA) measures variant
effects on gene RNA expression for 21 regulatory elements (11 enhancers, 10 promoters). <a class="button" href="#satmutmpra" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('satmutmpra');" id="a_satmutmpra">more</a></p><div id="div_satmutmpra" style="display:none;word-wrap:break-word;"><p>The 20 disease-associated regulatory elements and one ultraconserved enhancer analysed in different cell lines are the following:
<ul><li>ten promoters (of TERT, LDLR, HBB, HBG, HNF4A, MSMB, PKLR, F9, FOXE1 and GP1BB) and</li><li>ten enhancers (of SORT1, ZRS, BCL11A, IRF4, IRF6, MYC (2x), RET, TCF7L2 and ZFAND3) and</li><li>one ultraconserved enhancer (UC88).
</li></ul><p>Please refer to the satMutMPRA web server and Kircher M et al. (2019) paper for more information:
<a href="https://mpra.gs.washington.edu/satMutMPRA/">https://mpra.gs.washington.edu/satMutMPRA/</a>
<a href="https://www.ncbi.nlm.nih.gov/pubmed/31395865">https://www.ncbi.nlm.nih.gov/pubmed/31395865</a></p><p>Parameters can be set using a key=value system:<table class="ss"><thead><tr><th>Argument</th><th>Description</th></tr></thead><tbody><tr class="bg1"><td><pre>file</pre></td><td>required - a tabix indexed file of the satMutMPRA data corresponding to desired assembly. 
<tr class="bg2"><td><pre>pvalue</pre></td><td>p-value threshold  (default: 0.00001) 
<tr class="bg1"><td><pre>cols</pre></td><td>colon delimited list of data types to be returned from the satMutMPRA data                 (default: <kbd>Value</kbd>, <kbd>P-Value</kbd>, and <kbd>Element</kbd>)</p><p><tr class="bg2"><td><pre>incl_repl</pre></td><td>include replicates (default: off): <ul><li>full replicate for LDLR promoter (LDLR.2) and SORT1 enhancer (SORT1.2)</li><li>a reversed sequence orientation for SORT1 (SORT1-flip)</li><li>other conditions: PKLR-48h, ZRSh-13h2, TERT-GAa, TERT-GBM, TERG-GSc
</td></tr></tbody></table><p></li></ul><p>The Bio::DB::HTS perl library or tabix utility must be installed in your path
to use this plugin. The satMutMPRA data file can be downloaded from
<a href="https://mpra.gs.washington.edu/satMutMPRA/">https://mpra.gs.washington.edu/satMutMPRA/</a></p><p>satMutMPRA data can be downloaded for both GRCh38 and GRCh37 from the web server (<a href="https://mpra.gs.washington.edu/satMutMPRA/">https://mpra.gs.washington.edu/satMutMPRA/</a>):
<kbd>Download</kbd> section, select <kbd>GRCh37</kbd> or <kbd>GRCh38</kbd> for 'Genome release' and 'Download All Elements'.</p><p>The file must be processed and indexed by tabix before use by this plugin.</p><p># GRCh38
<pre class="code sh_sh">(grep ^Chr GRCh38_ALL.tsv; grep -v ^Chr GRCh38_ALL.tsv | sort -k1,1 -k2,2n ) | bgzip &gt satMutMPRA_GRCh38_ALL.gz
tabix -s 1 -b 2 -e 2 -c C satMutMPRA_GRCh38_ALL.gz
</pre><p># GRCh37
<pre class="code sh_sh">(grep ^Chr GRCh37_ALL.tsv; grep -v ^Chr GRCh37_ALL.tsv | sort -k1,1 -k2,2n ) | bgzip &gt satMutMPRA_GRCh37_ALL.gz
tabix -s 1 -b 2 -e 2 -c C satMutMPRA_GRCh37_ALL.gz
</pre><p>
When running the plugin by default <kbd>Value</kbd>, <kbd>P-Value</kbd>, and <kbd>Element</kbd>
information is returned e.g.</p><p><pre class="code sh_sh">--plugin satMutMPRA,file=/path/to/satMutMPRA_GRCh38_ALL.gz
</pre><p>You may include all columns with ALL; this fetches all data per variant
(e.g. Tags, DNA, RNA, Value, P-Value, Element):</p><p><pre class="code sh_sh">--plugin satMutMPRA,file=/path/to/satMutMPRA_GRCh38_ALL.gz,cols=ALL
</pre><p>You may want to select only a specific subset of information to be
reported, you can do this by specifying the specific columns as parameters to the plugin e.g.</p><p><pre class="code sh_sh">--plugin satMutMPRA,file=/path/to/satMutMPRA_GRCh38_ALL.gz,cols=Tags:DNA
</pre><p>If a requested column is not found, the error message will report the
complete list of available columns in the satMutMPRA file. For a detailed description
of the available information please refer to the manuscript or online web server.</p><p>Tabix also allows the data file to be hosted on a remote server. This plugin is
fully compatible with such a setup - simply use the URL of the remote file:</p><p><pre class="code sh_sh">--plugin satMutMPRA,file=<a href="http://my.files.com/satMutMPRA.gz">http://my.files.com/satMutMPRA.gz</a>
</pre><p>Note that gene locations referred to in satMutMPRA may be out of sync with
those in the latest release of Ensembl; this may lead to discrepancies with
information retrieved from other sources.</p><p></ul><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv satMutMPRA.pm ~/.vep/Plugins
./vep -i variations.vcf --plugin satMutMPRA,file=/path/to/satMutMPRA_data.gz,cols=col1:col2
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#556688">Phenotype data and citations</div></td><td>-</td><td>Ensembl</td></tr>
<tr id="SingleLetterAA" class="bg1 plugin_row" data-category="hgvs"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/SingleLetterAA.pm">SingleLetterAA</a></div></td><td><p> This is a plugin for the Ensembl Variant Effect Predictor (VEP) that
 returns a HGVSp string with single amino acid letter codes <a class="button" href="#singleletteraa" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('singleletteraa');" id="a_singleletteraa">more</a></p><div id="div_singleletteraa" style="display:none;word-wrap:break-word;"><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv SingleLetterAA.pm ~/.vep/Plugins
./vep -i variations.vcf --plugin SingleLetterAA
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#9400D3">HGVS</div></td><td>-</td><td>Ensembl</td></tr>
<tr id="SpliceAI" class="bg2 plugin_row" data-category="splicing_predictions"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/SpliceAI.pm">SpliceAI</a></div></td><td><p> An Ensembl VEP plugin that retrieves pre-calculated annotations from SpliceAI.
 SpliceAI is a deep neural network, developed by Illumina, Inc 
 that predicts splice junctions from an arbitrary pre-mRNA transcript sequence. <a class="button" href="#spliceai" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('spliceai');" id="a_spliceai">more</a></p><div id="div_spliceai" style="display:none;word-wrap:break-word;"><p> Delta score of a variant, defined as the maximum of (DS_AG, DS_AL, DS_DG, DS_DL), 
 ranges from 0 to 1 and can be interpreted as the probability of the variant being 
 splice-altering. The author-suggested cutoffs are:
<ul><li>0.2 (high recall)</li><li>0.5 (recommended)</li><li>0.8 (high precision)
</li></ul><p> This plugin is available for both GRCh37 and GRCh38.</p><p> More information can be found at:
 <a href="https://pypi.org/project/spliceai/">https://pypi.org/project/spliceai/</a></p><p> Please cite the SpliceAI publication alongside Ensembl VEP if you use this resource:
 <a href="https://www.ncbi.nlm.nih.gov/pubmed/30661751">https://www.ncbi.nlm.nih.gov/pubmed/30661751</a></p><p> Running options:
<ol><li value="1">By default, this plugin appends all scores from SpliceAI files.</li><li value="2">Besides the pre-calculated scores, it can also be specified a score cutoff between 0 and 1.</p><p></li></ol><p> Output: 
  The output includes the gene symbol, delta scores (DS) and delta positions (DP)
  for acceptor gain (AG), acceptor loss (AL), donor gain (DG), and donor loss (DL).</p><p><ul><li>For tab the output contains one header <kbd>SpliceAI_pred</kbd> with all&nbsp;    the delta scores and positions. The format is:
&nbsp;      <kbd>SYMBOL|DS_AG|DS_AL|DS_DG|DS_DL|DP_AG|DP_AL|DP_DG|DP_DL</kbd></p><p></li><li>For JSON the output is a hash with the following format:&nbsp;    "spliceai":
&nbsp;      {"DP_DL":0,"DS_AL":0,"DP_AG":0,"DS_DL":0,"SYMBOL":"X","DS_AG":0,"DP_AL":0,"DP_DG":0,"DS_DG":0}</p><p></li><li>For VCF output the delta scores and positions are stored in different headers.&nbsp;    The values are <kbd>SpliceAI_pred_xx</kbd> being <kbd>xx</kbd> the score/position.
&nbsp;      Example: <kbd>SpliceAI_pred_DS_AG</kbd> is the delta score for acceptor gain.</p><p></li></ul><p>  Gene matching:
  SpliceAI can contain scores for multiple genes that overlap a variant,
  and Ensembl VEP can also predict consequences on multiple genes for a given variant.
  The plugin only returns SpliceAI scores for the gene symbols that match (if any).</p><p> If plugin is run with option 2, the output also contains a flag: <kbd>PASS</kbd> if delta score
 passes the cutoff, <kbd>FAIL</kbd> otherwise. </p><p> The following steps are necessary before running this plugin:</p><p> The files with the annotations for all possible substitutions (snv), 1 base insertions 
 and 1-4 base deletions (indel) within genes are available here:
 <a href="https://basespace.illumina.com/s/otSPW8hnhaZR">https://basespace.illumina.com/s/otSPW8hnhaZR</a></p><p> GRCh37:
<pre class="code sh_sh">tabix -p vcf spliceai_scores.raw.snv.hg37.vcf.gz
tabix -p vcf spliceai_scores.raw.indel.hg37.vcf.gz
</pre><p> GRCh38:
<pre class="code sh_sh">tabix -p vcf spliceai_scores.raw.snv.hg38.vcf.gz
tabix -p vcf spliceai_scores.raw.indel.hg38.vcf.gz
</pre><p> The plugin can then be run:
<pre class="code sh_sh">./vep -i variations.vcf --plugin SpliceAI,snv=/path/to/spliceai_scores.raw.snv.hg38.vcf.gz,indel=/path/to/spliceai_scores.raw.indel.hg38.vcf.gz
./vep -i variations.vcf --plugin SpliceAI,snv=/path/to/spliceai_scores.raw.snv.hg38.vcf.gz,indel=/path/to/spliceai_scores.raw.indel.hg38.vcf.gz,cutoff=0.5
</pre><p></ul><p></ol><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv SpliceAI.pm ~/.vep/Plugins
./vep -i variations.vcf --plugin SpliceAI,snv=/path/to/spliceai_snv_.vcf.gz,
indel=/path/to/spliceai_indel_.vcf.gz
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#FF0000">Splicing predictions</div></td><td><a href="https://metacpan.org/pod/List::Util" rel="external">List::Util</a> qw(max)</td><td>Ensembl</td></tr>
<tr id="SpliceRegion" class="bg1 plugin_row" data-category="splicing_predictions"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/SpliceRegion.pm">SpliceRegion</a></div></td><td><p> This is a plugin for the Ensembl Variant Effect Predictor (VEP) that
 provides more granular predictions of splicing effects. <a class="button" href="#spliceregion" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('spliceregion');" id="a_spliceregion">more</a></p><div id="div_spliceregion" style="display:none;word-wrap:break-word;"><p> Three additional terms may be added:</p><p> # splice_donor_5th_base_variant : variant falls in the 5th base after the splice donor junction (5' end of intron)</p><p><pre class="code sh_sh">             v
 ...EEEEEIIIIIIIIII...
</pre>
 (E = exon, I = intron, v = variant location)</p><p> # splice_donor_region_variant : variant falls in region between 3rd and 6th base after splice junction (5' end of intron)</p><p><pre class="code sh_sh">           vv vvv
 ...EEEEEIIIIIIIIII...
</pre>
 # splice_polypyrimidine_tract_variant : variant falls in polypyrimidine tract at 3' end of intron, between 17 and 3 bases from the end</p><p><pre class="code sh_sh">      vvvvvvvvvvvvvvv
 ...IIIIIIIIIIIIIIIIIIIIEEEEE...
</pre></p><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv SpliceRegion.pm ~/.vep/Plugins
./vep -i variations.vcf --plugin SpliceRegion</p><p>To only show the additional consequence extended_intronic_splice_region_variant, use: 
./vep -i variations.vcf --plugin SpliceRegion,Extended
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#FF0000">Splicing predictions</div></td><td>-</td><td>Ensembl</td></tr>
<tr id="SpliceVault" class="bg2 plugin_row" data-category="splicing_predictions"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/SpliceVault.pm">SpliceVault</a></div></td><td><p> An Ensembl VEP plugin that retrieves SpliceVault data to predict exon-skipping events
 and activated cryptic splice sites based on the most common mis-splicing events
 around a splice site. <a class="button" href="#splicevault" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('splicevault');" id="a_splicevault">more</a></p><div id="div_splicevault" style="display:none;word-wrap:break-word;"><p> This plugin returns the most common variant-associated mis-splicing events
 based on SpliceVault data. Each event includes the following information:
<ul><li>Type: exon skipping (ES), cryptic donor (CD) or cryptic acceptor (CA)</li><li>Transcript impact:</li><li>For ES, describes skipped exons, e.g. ES:2 represents exon 2 skipping and&nbsp;     ES:2-3 represents skipping of exon 2 and 3
</li><li>For CD/CA, describes the distance from the annotated splice-site to the&nbsp;     cryptic splice-site with reference to the transcript (distances to negative
&nbsp;     strand transcripts are reported according to the 5' to 3' distance)
</li><li>Percent of supporting samples: percent of samples supporting the event over&nbsp;   total samples where splicing occurs in that site (note this may be above 100%
&nbsp;   if the event is seen in more samples than annotated splicing)
</li><li>Frameshift: inframe or out-of-frame event
</li></ul><p> The plugin also returns information specific to each splice site:
<ul><li>Site position/type: genomic location and type (donor/acceptor) of the&nbsp;   splice-site predicted to be lost by SpliceAI. Cryptic positions are relative
&nbsp;   to this genomic coordinate.
</li><li>Out of frame events: fraction of the top events that cause a frameshift. As&nbsp;   per <a href="https://pubmed.ncbi.nlm.nih.gov/36747048">https://pubmed.ncbi.nlm.nih.gov/36747048</a>, sites with 3/4 or more in-frame
&nbsp;   events are likely to be splice-rescued and not loss-of-function (LoF).
</li><li>Site sample count and max depth: sample count for this splice site and max&nbsp;   number of reads in any single sample representing annotated splicing in
&nbsp;   Genotype-Tissue Expression (GTEx). This information allows to filter events
&nbsp;   based on a minimum number of samples or minimum depth in GTEx.
</li><li>SpliceAI delta score (provided by SpliceVault)
</li></ul><p> Please cite the SpliceVault publication alongside Ensembl VEP if you use this
 resource: <a href="https://pubmed.ncbi.nlm.nih.gov/36747048">https://pubmed.ncbi.nlm.nih.gov/36747048</a></p><p> The tabix utility must be installed in your path to use this plugin. The
 SpliceVault TSV and respective index (TBI) for GRCh38 can be downloaded from:
<ul><li><a href="https://ftp.ensembl.org/pub/current_variation/SpliceVault/SpliceVault_data_GRCh38.tsv.gz">https://ftp.ensembl.org/pub/current_variation/SpliceVault/SpliceVault_data_GRCh38.tsv.gz</a></li><li><a href="https://ftp.ensembl.org/pub/current_variation/SpliceVault/SpliceVault_data_GRCh38.tsv.gz.tbi">https://ftp.ensembl.org/pub/current_variation/SpliceVault/SpliceVault_data_GRCh38.tsv.gz.tbi</a>
</li></ul><p> To filter results, please use filter_vep with the output file or standard
 output. Documentation on filter_vep is available at:
 <a href="https://www.ensembl.org/info/docs/tools/vep/script/vep_filter.html">https://www.ensembl.org/info/docs/tools/vep/script/vep_filter.html</a></p><p></ul><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv SpliceVault.pm ~/.vep/Plugins</p><p>./vep -i variations.vcf --plugin SpliceVault,file=/path/to/SpliceVault_data_GRCh38.tsv.gz</p><p># Stringently select predicted loss-of-function (pLoF) splicing variants
./filter_vep -i variant_effect_output.txt --filter "SPLICEVAULT_OUT_OF_FRAME_EVENTS &gt= 3"
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#FF0000">Splicing predictions</div></td><td>-</td><td>Ensembl</td></tr>
<tr id="StructuralVariantOverlap" class="bg1 plugin_row" data-category="structural_variant_data"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/StructuralVariantOverlap.pm">StructuralVariantOverlap</a></div></td><td><p> A VEP plugin that retrieves information from overlapping structural variants. <a class="button" href="#structuralvariantoverlap" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('structuralvariantoverlap');" id="a_structuralvariantoverlap">more</a></p><div id="div_structuralvariantoverlap" style="display:none;word-wrap:break-word;"><p> Parameters can be set using a key=value system:<table class="ss"><thead><tr><th>Argument</th><th>Description</th></tr></thead><tbody>
<tr class="bg1"><td><pre>file</pre></td><td>required - a VCF file of reference data. 
<tr class="bg2"><td><pre>percentage</pre></td><td>percentage overlap between SVs (default: 80) <tr class="bg1"><td><pre>reciprocal</pre></td><td>calculate reciprocal overlap, options:  0 or 1. (default: 0)                   (overlap is expressed as % of input SV by default)</p><p><tr class="bg2"><td><pre>cols</pre></td><td>colon delimited list of data types to return from the INFO fields (only AF by default) 
<tr class="bg1"><td><pre>same_type</pre></td><td>1/0 only report SV of the same type (eg deletions for deletions, off by default) 
<tr class="bg2"><td><pre>distance</pre></td><td>the distance the ends of the overlapping SVs should be within. 
<tr class="bg1"><td><pre>match_type</pre></td><td>only report reference SV which lie within or completely surround the input SV                   options: within, surrounding</p><p><tr class="bg2"><td><pre>label</pre></td><td>annotation label that will appear in the output (default: "SV_overlap")                   Example- input: label=mydata, output: mydata_name=refSV,mydata_PC=80,mydata_AF=0.05</p><p></td></tr></tbody></table><p> Example reference data</p><p><ul><li>1000 Genomes Project:&nbsp;   <a href="https://ftp.1000genomes.ebi.ac.uk/vol1/ftp/phase3/integrated_sv_map/ALL.wgs.mergedSV.v8.20130502.svs.genotypes.vcf.gz">https://ftp.1000genomes.ebi.ac.uk/vol1/ftp/phase3/integrated_sv_map/ALL.wgs.mergedSV.v8.20130502.svs.genotypes.vcf.gz</a></p><p></li><li>gnomAD:&nbsp;   <a href="https://storage.googleapis.com/gcp-public-data--gnomad/papers/2019-sv/gnomad_v2.1_sv.sites.vcf.gz">https://storage.googleapis.com/gcp-public-data--gnomad/papers/2019-sv/gnomad_v2.1_sv.sites.vcf.gz</a></p><p></li></ul><p> Example:
<pre class="code sh_sh">./vep -i structvariants.vcf --plugin StructuralVariantOverlap,file=gnomad_v2_sv.sites.vcf.gz
</pre><p>
</ul><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv StructuralVariantOverlap.pm ~/.vep/Plugins
./vep -i structvariants.vcf --plugin StructuralVariantOverlap,file=gnomad_v2_sv.sites.vcf.gz
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#601212">Structural variant data</div></td><td>-</td><td>Ensembl</td></tr>
<tr id="SubsetVCF" class="bg2 plugin_row" data-category="variant_data"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/SubsetVCF.pm">SubsetVCF</a></div></td><td><p> An Ensembl VEP plugin to retrieve overlapping records from a given VCF file.
 Values for POS, ID, and ALT, are retrieved as well as values for any requested
 INFO field. Additionally, the allele number of the matching ALT is returned. <a class="button" href="#subsetvcf" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('subsetvcf');" id="a_subsetvcf">more</a></p><div id="div_subsetvcf" style="display:none;word-wrap:break-word;"><p> Though similar to using <kbd>--custom</kbd>, this plugin returns all ALTs for a given
 POS, as well as all associated INFO values.</p><p> By default, only VCF records with a filter value of "PASS" are returned, 
 however this behaviour can be changed via the <kbd>filter</kbd> option.</p><p> The plugin accepts the following key=value parameters:<table class="ss"><thead><tr><th>Argument</th><th>Description</th></tr></thead><tbody><tr class="bg1"><td><pre>name</pre></td><td>short name added used as a prefix (required) <tr class="bg2"><td><pre>file</pre></td><td>path to tabix-index vcf file (required) <tr class="bg1"><td><pre>filter</pre></td><td>only consider variants marked as <kbd>PASS</kbd>, 1 or 0 (default, 1) <tr class="bg2"><td><pre>fields</pre></td><td>info fields to be returned (default, not used) <ul><li>'%' can delimit multiple fields</li><li>'*' can be used as a wildcard 
</td></tr></tbody></table><p></li></ul><p> Returns:
<ul><li><kbd>&ltname&gt_POS</kbd>: POS field from VCF</li><li><kbd>&ltname&gt_REF</kbd>: REF field from VCF (minimised)</li><li><kbd>&ltname&gt_ALT</kbd>: ALT field from VCF (minimised)</li><li><kbd>&ltname&gt_alt_index</kbd>: Index of matching variant (zero-based)</li><li><kbd>&ltname&gt_&ltfield&gt</kbd>: List of requested info values
</ul><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
./vep -i variations.vcf --plugin SubsetVCF,file=filepath.vcf.gz,name=myvfc,fields=AC*%AN*
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#B22222">Variant data</div></td><td><ul style="padding-left:1em"><li><a href="https://metacpan.org/pod/Data::Dumper" rel="external">Data::Dumper</a></li><li><a href="https://metacpan.org/pod/Storable" rel="external">Storable</a> qw(dclone)</li></ul></td><td>Joseph A. Prinz</td></tr>
<tr id="TranscriptAnnotator" class="bg1 plugin_row" data-category="transcript_annotation"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/TranscriptAnnotator.pm">TranscriptAnnotator</a></div></td><td><p> An Ensembl VEP plugin that annotates variant-transcript pairs based on a given file: <a class="button" href="#transcriptannotator" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('transcriptannotator');" id="a_transcriptannotator">more</a></p><div id="div_transcriptannotator" style="display:none;word-wrap:break-word;"><p><pre class="code sh_sh">--plugin TranscriptAnnotator,file=${HOME}/file.tsv.gz
</pre><p> Example of a valid tab-separated annotation file:</p><p><pre class="code sh_sh">#Chrom  Pos       Ref  Alt  Transcript       SIFT_score  SIFT_pred    Comment
11      436154    A    G    NM_001347882.2   0.03        Deleterious  Bad
11      1887471   C    T    ENST00000421485  0.86        Tolerated    Good
</pre>
 Please bgzip and tabix the file with commands such as:</p><p><pre class="code sh_sh">bgzip file.txt
tabix -b2 -e2 file.txt.gz
</pre><p> Options are passed to the plugin as key=value pairs:<table class="ss"><thead><tr><th>Argument</th><th>Description</th></tr></thead><tbody>
<tr class="bg1"><td><pre>file</pre></td><td>(mandatory) Tabix-indexed file to parse. Must contain variant             location (chromosome, position, reference allele, alternative allele)
            and transcript ID as the first 5 columns. Accepted transcript IDs
            include those from Ensembl and RefSeq.
<tr class="bg2"><td><pre>cols</pre></td><td>Colon-delimited list with names of the columns to append. Column             names are based on the last header line. By default, all columns
            (except the first 5) are appended.
<tr class="bg1"><td><pre>prefix</pre></td><td>String to prefix the name of appended columns (default: basename of             the filename without extensions). Set to 0 to avoid any prefix.
<tr class="bg2"><td><pre>trim</pre></td><td>Trim whitespaces from both ends of each column (default: 1). 
</td></tr></tbody></table><p> The tabix and bgzip utilities must be installed in your path to read the
 tabix-indexed annotation file: check <a href="https://github.com/samtools/htslib.git">https://github.com/samtools/htslib.git</a> for
 installation instructions.</p><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv TranscriptAnnotator.pm ~/.vep/Plugins
./vep -i variations.vcf --plugin TranscriptAnnotator,file=/path/to/file.txt.gz
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#FF2288">Transcript annotation</div></td><td><a href="https://metacpan.org/pod/File::Basename" rel="external">File::Basename</a></td><td>Ensembl</td></tr>
<tr id="TSSDistance" class="bg2 plugin_row" data-category="nearby_features"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/TSSDistance.pm">TSSDistance</a></div></td><td><p> An Ensembl VEP plugin that calculates the distance from the transcription
 start site for upstream variants. <a class="button" href="#tssdistance" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('tssdistance');" id="a_tssdistance">more</a></p><div id="div_tssdistance" style="display:none;word-wrap:break-word;"><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv TSSDistance.pm ~/.vep/Plugins
./vep -i variations.vcf --plugin TSSDistance
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#E75480">Nearby features</div></td><td>-</td><td>Ensembl</td></tr>
<tr id="UTRAnnotator" class="bg1 plugin_row" data-category="transcript_annotation"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/UTRAnnotator.pm">UTRAnnotator</a></div></td><td><p>    An Ensembl VEP plugin that annotates the effect of 5' UTR variant especially for variant creating/disrupting upstream ORFs.
    Available for both GRCh37 and GRCh38. <a class="button" href="#utrannotator" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('utrannotator');" id="a_utrannotator">more</a></p><div id="div_utrannotator" style="display:none;word-wrap:break-word;"><p>    Options are passed to the plugin as key=value pairs:<table class="ss"><thead><tr><th>Argument</th><th>Description</th></tr></thead><tbody>
<tr class="bg1"><td><pre>file</pre></td><td>(Required) Path to UTRAnnotator data file: <ul><li>Download <kbd>uORF_5UTR_GRCh37_PUBLIC.txt</kbd> or <kbd>uORF_5UTR_GRCh38_PUBLIC.txt</kbd>&nbsp;                    from <a href="https://github.com/Ensembl/UTRannotator">https://github.com/Ensembl/UTRannotator</a>
</li><li>Download from <a href="http://sorfs.org">http://sorfs.org</a><tr class="bg2"><td><pre>&nbsp;    max_overlap</pre></td><td>(Optional) Maximum percentage of overlap between variant and &nbsp;                  UTR for UTR annotation (default: 100)</p><p></td></tr></tbody></table><p></li></ul><p>    Citation</p><p>    About the role of 5'UTR variants in human genetic disease:</p><p>    Whiffin, N., Karczewski, K.J., Zhang, X. et al. Characterising the loss-of-function impact of 5' untranslated region variants in 15,708 individuals. Nat Commun 11, 2523 (2020). <a href="https://doi.org/10.1038/s41467-019-10717-9">https://doi.org/10.1038/s41467-019-10717-9</a></p><p>    About UTRAnnotator:</p><p>    The original UTRAnnotator plugin is written by Xiaolei Zhang et al. Later adopted by Ensembl VEP plugins with some changes.
    You can find the original plugin here -
    <a href="https://github.com/ImperialCardioGenetics/UTRannotator">https://github.com/ImperialCardioGenetics/UTRannotator</a> </p><p>    Please cite the UTRannotator publication alongside the Ensembl VEP if you use this resource -
    Annotating high-impact 5'untranslated region variants with the UTRannotator Zhang, X., Wakeling, M.N., Ware, J.S, Whiffin, N. Bioinformatics; doi: <a href="https://academic.oup.com/bioinformatics/advance-article/doi/10.1093/bioinformatics/btaa783/5905476">https://academic.oup.com/bioinformatics/advance-article/doi/10.1093/bioinformatics/btaa783/5905476</a></p><p></ul><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv UTRAnnotator.pm ~/.vep/Plugins
vep -i variations.vcf --plugin UTRAnnotator,file=/path/to/uORF_starts_ends_GRCh38_PUBLIC.txt</p><p># skip annotation for variants with a 80% or higher overlap of the UTR
vep -i variations.vcf --plugin UTRAnnotator,file=/path/to/uORF_starts_ends_GRCh38_PUBLIC.txt,max_overlap=80
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#FF2288">Transcript annotation</div></td><td><ul style="padding-left:1em"><li><a href="https://metacpan.org/pod/List::Util" rel="external">List::Util</a> qw(min max)</li><li><a href="https://metacpan.org/pod/Scalar::Util" rel="external">Scalar::Util</a> qw(looks_like_number)</li></ul></td><td>Ensembl</td></tr>
<tr id="VARITY" class="bg2 plugin_row" data-category="pathogenicity_predictions"><td><div style="font-weight:bold"><a rel="external" href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/VARITY.pm">VARITY</a></div></td><td><p> This is a plugin for the Ensembl Variant Effect Predictor (VEP) that
 adds the pre-computed VARITY scores to predict pathogenicity of 
 rare missense variants to the Ensembl VEP output. <a class="button" href="#varity" style="padding:3px 8px 0px 8px !important;height:18px" onclick="show_hide('varity');" id="a_varity">more</a></p><div id="div_varity" style="display:none;word-wrap:break-word;"><p> Please cite the VARITY publication alongside Ensembl VEP if you use this resource:
 <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8715197/">https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8715197/</a></p><p> Running options :</p><p> VARITY scores can be downloaded using
<pre class="code sh_sh">wget <a href="http://varity.varianteffect.org/downloads/varity_all_predictions.tar.gz">http://varity.varianteffect.org/downloads/varity_all_predictions.tar.gz</a>
</pre><p> The files can be tabix processed by :
<pre class="code sh_sh">tar -xzvf varity_all_predictions.tar.gz 
cat varity_all_predictions.txt | (head -n 1 && tail -n +2  | sort -t$'\t' -k 1,1 -k 2,2n) &gt varity_all_predictions_sorted.tsv
sed '1s/.*/#&/'  varity_all_predictions_sorted.tsv &gt varity_all_predictions.tsv  # to add a # in the first line of the file 
bgzip varity_all_predictions.tsv
tabix -f -s 1 -b 2 -e 2 varity_all_predictions.tsv.gz
</pre><p>
 Requirements:
 The tabix utility must be installed in your path to use this plugin.
 The --assembly flag is required to use this plugin.</p><p><p><h2>Usage examples:</h2> <pre class="code sh_sh">
mv VARITY.pm ~/.vep/Plugins
./vep -i variations.vcf --assembly GRCh37 --plugin VARITY,file=/path/to/varity_all_predictions.txt
</pre></p></p></div></td><td><div class="vdoc_dtype_count" style="white-space:normal;float:left;padding:2px 6px;cursor:default;background-color:#1E90FF">Pathogenicity predictions</div></td><td>-</td><td>Ensembl</td></tr>
</tbody></table><!-- Plugins list - end -->
  
  <p> We hope that these will serve as useful examples for users implementing
	new plugins. If you have any questions about the system, or suggestions for
	enhancements please let us know on the 
  <a href="//lists.ensembl.org/mailman/listinfo/dev" rel="external">ensembl-dev</a> 
  mailing list.<br />
  We also encourage you to share any plugins you develop: we are happy to accept pull requests on the <a href="//github.com/Ensembl/VEP_plugins" rel="external">VEP_plugins</a> git repository. </p>
    

  <p> There are further published plugins available outside the Ensembl VEP repository including:
  <li> <a href="https://github.com/konradjk/loftee">LOFTEE </a> a Loss-Of-Function Transcript Effect Estimator (<a href="https://www.nature.com/articles/s41586-020-2308-7">Konrad Karczewski et al,2020</a>)


	
  <hr/>
  <h2 id="plugins_how">How it works</h2>
	
	<p>Plugins are run once Ensembl VEP has finished its analysis for each line of
	the output, but before anything is printed to the output file.</p>

	<p>When each
	plugin is called (using the <i>run</i> method) it is passed two data structures
	to use in its analysis; the first is a data structure containing all the
	data for the current line, and the second is a reference to a variation
	API object that represents the combination of a variant allele and an
	overlapping or nearby genomic feature (such as a transcript or regulatory
	region).</p>

	<p>This object provides access to all the relevant API objects that
	may be useful for further analysis by the plugin (such as the current
	VariationFeature and Transcript). Please refer to the <a
	href="/info/docs/Doxygen/variation-api/index.html"><img style="vertical-align:bottom" src="/i/16/documentation.png"/> Ensembl Variation API
	documentation</a> for more details.</p>

  <hr/>
  <h2 id="plugins_functionality">Functionality</h2>
	
	<p>We expect that most plugins will simply add information to the last
	column of the output file, the "Extra" column, and the plugin system assumes
	this in various places, but plugins are also free to alter the output line
	as desired.</p>
	
	<p>The only hard requirement for
	a plugin to work with Ensembl VEP is that it implements a number of required
	methods (such as <i>new</i> which should create and return an instance of this
	plugin, <i>get_header_info</i> which should return descriptions of the type of
	data this plugin produces to be included in Ensembl VEP output's header, and
	<i>run</i> which should actually perform the logic of the plugin).</p>

	<p>To make development of plugins easier, we suggest that users use the <a
	href="/info/docs/Doxygen/variation-api/classBio_1_1EnsEMBL_1_1Variation_1_1Utils_1_1BaseVepPlugin.html">Bio::EnsEMBL::Variation::Utils::BaseVepPlugin</a>
	module as their base class, which provides default implementations of all the
	necessary methods which can be overridden as required. Please refer to the
	documentation in this module for details of all required methods and for a
	simple example of a plugin implementation. </p>
	
	<hr/>
	<h2 id="plugins_filter">Filtering using plugins</h2>
	
	<p> A common use for plugins will be to filter the output in some way (for
	example to limit output lines to missense variants) and so we provide
	a simple mechanism to support this.</p>

	<p>The <i>run</i> method of a plugin is assumed
	to return a reference to a hash containing information to be included in the
	output, and if a plugin should not add any data to a particular line
	it should return an empty hashref. If a plugin should instead filter a
	line and exclude it from the output, it should return <i>undef</i> from its <i>run</i>
	method, this also means that no further plugins will be run on the line.</p>

	<p>If you are developing a filter plugin, we suggest that you use the <a
	href="/info/docs/Doxygen/variation-api/classBio_1_1EnsEMBL_1_1Variation_1_1Utils_1_1BaseVepFilterPlugin.html">Bio::EnsEMBL::Variation::Utils::BaseVepFilterPlugin</a>
	as your base class and then you need only override the <i>include_line</i> method
	to return true if you want to include this line, and false otherwise.
	Again, please refer to the documentation in this module for more details and an
	example	implementation of a missense filter. </p>
	
	<hr/>
	<h2 id="plugins_use">Using plugins</h2>
    
    <p> In order to run a plugin you need to include the plugin module in Perl's
    library path somehow; by default Ensembl VEP includes the <i>&#126;/.vep/Plugins</i>
    directory in the path, so this is a convenient place to store plugins, but
    you are also able to include modules by any other means (e.g using the
    <i>$PERL5LIB</i> environment variable in Unix-like systems).<br />You can then run a
    plugin using the <a href="vep_options.html#opt_plugin">--plugin</a> command line option, passing the name of
    the plugin module as the argument.</p>

    <p>For example, if your plugin is in a
    module called <i>MyPlugin.pm</i>, stored in <i>&#126;/.vep/Plugins</i>, you can run it
    with a command line like: </p>
	
	<pre class="code sh_sh">./vep -i input.vcf --plugin MyPlugin</pre>
	
	<p> You can pass arguments to the plugin's 'new' method by including them after
	the plugin name on the command line, separated by commas, e.g.: </p>
	
	<pre class="code sh_sh">./vep -i input.vcf --plugin MyPlugin,1,FOO</pre>
	
	<p> If your plugin inherits from BaseVepPlugin, you can then retrieve these
	parameters as a list from the <i>params</i> method. </p>
	
	<p> You can run multiple plugins by supplying multiple <a href="vep_options.html#opt_plugin">--plugin</a>
	arguments. Plugins are run serially in the order in which they are specified
	on the command line, so they can be run as a pipeline, with, for example, a
	later plugin filtering output based on the results from an earlier plugin.
	Note though that the first plugin to filter a line 'wins', and any later
	plugins won't get
	run on a filtered line. </p>

  <div id="nv_species">
    <div class="info" style="float:left">
      <h3>Plugin warnings</h3>
      <div class="message-pad">
        <p>By default, an Ensembl VEP run does not fail even when a plugin raises
        warnings or compilation errors. To avoid this, please use the
        <a href="vep_options.html#opt_safe">--safe</a> flag. This is
        particularly useful to ensure plugins run properly when using Ensembl VEP in
        pipelines.</p>
      </div>
    </div>
    <div style="clear:both"></div>
  </div>

	<hr/>
	<h2 id="plugins_intergenic">Intergenic variants</h2>
	
	<p> When a variant falls in an intergenic region, it will usually not have
	any consequence types called, and hence will not have any associated
	VariationFeatureOverlap objects. In this special case, Ensembl VEP creates a new
	VariationFeatureOverlap that overlaps a feature of type "Intergenic".</p>

	<p>To force your plugin to handle these, you must add "Intergenic" to the feature
	types that it will recognize; you do this by writing your own feature_types
	sub-routine:</p>
	
	<pre class="code sh_perl sh_sourceCode">sub feature_types {
    return ['Transcript', 'Intergenic'];
}</pre>
	
	<p> This will cause your plugin to handle any variation features that
	overlap transcripts or intergenic regions. To also include any regulatory
	features, you should use the generic type "Feature":</p>
	
	<pre class="code sh_perl sh_sourceCode">sub feature_types {
    return ['Feature', 'Intergenic'];
}</pre>
	
</div>

</body>
</html>
