<html>
<head>
<title>Examples and use cases</title>
<meta name="order" content="8" />
</head>

<body>
    
<div>
	
  <div style="float:right"><img src="/img/vep_logo.png"/></div>
	
  <h1 id="top"><span style="color:#006;padding-right:15px">Variant Effect Predictor</span><span style="color:#666"><img src="/i/16/user.png"/> Examples and use cases</span></h1>
  <hr/>
  
  
  <h2 id="examples">Example commands</h2>
  
  <ul>
    <li>
      <p>Read input from <b>STDIN</b>, output to <b>STDOUT</b></p>
      <pre class="code sh_sh">./vep --cache -o stdout</pre>
    </li>
    <li>
      <p>Add <b>regulatory</b> region <b>consequences</b></p>
      <pre class="code sh_sh">./vep --cache -i variants.txt --regulatory</pre>
    </li>
    <li>
      <p>Input file <kbd>variants.vcf.txt</kbd>, input file <b>format VCF</b>, add <b>gene symbol</b> identifiers</p>
      <pre class="code sh_sh">./vep --cache -i variants.vcf.txt --format vcf --symbol</pre>
    </li>
    <li>
      <p><b>Filter</b> out <b>common variants</b> based on 1000 Genomes data</p>
      <pre class="code sh_sh">./vep --cache -i variants.txt --filter_common</pre>
    </li>
    <li>
      <p><b>Force overwrite</b> of output file <kbd>variants_output.txt</kbd>, check for existing
      <b>co-located variants</b>, output only <b>coding sequence</b> consequences, output
      <b>HGVS names</b></p>
      <pre class="code sh_sh">./vep --cache -i variants.txt -o variants_output.txt --force --check_existing --coding_only --hgvs</pre>
    </li>
    <li>
      <p><b>Run for any species or assembly</b> (even if not part of <a href="/info/about/species.html">Ensembl data</a>) by providing your own <a href="vep_cache.html#fasta">FASTA file</a> and <a href="vep_cache.html#gff" target="_blank">GFF/GTF annotation</a></p>
      <pre class="code sh_sh">./vep -i variants.txt -o variants_output.txt --gff data.gff.gz --fasta genome.fa.gz</pre>
    </li>
    <li>
      <p>Specify <b>DB connection parameters</b> in registry file <kbd>ensembl.registry</kbd>, add <b>SIFT</b>
      score and prediction, <b>PolyPhen</b> prediction</p>
      <pre class="code sh_sh">./vep --database -i variants.txt --registry ensembl.registry --sift b --polyphen p</pre>
    </li>
    <li>
      <p>Connect to <b>Ensembl Genomes</b> db server for <i>Arabidopsis thaliana</i></p>
      <pre class="code sh_sh">./vep --database -i variants.txt --genomes --species arabidopsis_thaliana</pre>
    </li>
    <li>
      <p>Load config from <b>ini file</b>, run in <b>quiet mode</b></p>
      <pre class="code sh_sh">./vep --config vep.ini -i variants.txt -q</pre>
    </li>
    <li>
      <p>Use <b>cache</b> in <kbd>/home/vep/mycache/</kbd>, use <b>gzcat</b> instead of zcat</p>
      <pre class="code sh_sh">./vep --cache --dir /home/vep/mycache/ -i variants.txt --compress gzcat</pre>
    </li>
    <li>
      <p>Add custom position-based <b>phenotype</b> annotation from remote <b>BED file</b></p>
      <pre class="code sh_sh">./vep --cache -i variants.vcf --custom file=ftp://ftp.myhost.org/data/phenotypes.bed.gz,short_name=phenotype</pre>
    </li>
    <li>
      <p>Use the <b>plugin</b> named MyPlugin, output only the variation name, feature, consequence type and MyPluginOutput <b>fields</b></p>
      <pre class="code sh_sh">./vep --cache -i variants.vcf --plugin MyPlugin --fields Uploaded_variation,Feature,Consequence,MyPluginOutput</pre>
    </li>
    <li>
      <p>Right align variants before consequence calculation. For more information, see <a
         href="/info/docs/tools/vep/script/vep_other.html#shifting"
         target="_blank">here</a>.</p>
      <pre class="code sh_sh">./vep --cache -i variants.vcf --shift_3prime 1</pre>
    </li>
     <li>
      <p>Report uploaded allele before minimisation. For more information, see <a
         href="/info/docs/tools/vep/script/vep_options.html#opt_uploaded_allele"
         target="_blank">here</a>.</p>
      <pre class="code sh_sh">./vep --cache -i variants.vcf --uploaded_allele</pre>
    </li>
  </ul>
  
    
  <hr/>
  
  <h2 id="gnomad">gnomAD</h2>
  
  <p> <a href="http://gnomad.broadinstitute.org/" rel="external">gnomAD</a> exome frequency data is included in VEP's cache files from release 90, replacing ExAC; use <a href="vep_options.html#opt_af_gnomade">--af_gnomade</a> to enable using this data. VEP can also retrieve frequency data from the gnomAD genomes set or ExAC via VEP's custom annotation functionality.<p>
  <p>For the latest gnomAD data, please visit <a href="https://gnomad.broadinstitute.org/downloads" rel="external">gnomAD downloads</a>.<p>
  
  <ol>
    <li>VEP requires Bio::DB::HTS to read data from tabix-indexed VCFs - see <a href="vep_download.html#installer"><img style="vertical-align:bottom" src="/i/16/download.png"/> installation instructions</a></li>
    <li>Ensembl's FTP site hosts abridged VCF files for gnomAD and ExAC, additionally remapped to GRCh38 using <a href="http://crossmap.sourceforge.net/" rel="external">CrossMap</a>. It is possible for VEP to read these files directly from their remote location, though for optimal performance the VCF and index should be downloaded to a local file system.
      <ul>
        <li>
          <b>GRCh38</b>
          <ul>
            <li>gnomAD genomes (r2.1, remapped with CrossMap):
              <a href="[[SPECIESDEFS::ENSEMBL_FTP_URL]]/data_files/homo_sapiens/GRCh38/variation_genotype/gnomad/r2.1/genomes/">[VCFs and tabix indexes]</a>
            </li>
            <li>gnomAD exomes (r2.1, remapped with CrossMap):
              <a href="[[SPECIESDEFS::ENSEMBL_FTP_URL]]/data_files/homo_sapiens/GRCh38/variation_genotype/gnomad/r2.1/exomes/">[VCFs and tabix indexes]</a>
            </li>
            <li>ExAC (v0.3, remapped using CrossMap):
              <a href="[[SPECIESDEFS::ENSEMBL_FTP_URL]]/data_files/homo_sapiens/GRCh38/variation_genotype/ExAC.0.3.GRCh38.vcf.gz">[VCF]</a>
              <a href="[[SPECIESDEFS::ENSEMBL_FTP_URL]]/data_files/homo_sapiens/GRCh38/variation_genotype/ExAC.0.3.GRCh38.vcf.gz.tbi">[tabix index]</a>
            </li>
          </ul>
        </li>

        <li>
          <b>GRCh37</b>
          <ul>
            <li>gnomAD genomes (r2.1):
              <a href="[[SPECIESDEFS::ENSEMBL_FTP_URL]]/data_files/homo_sapiens/GRCh37/variation_genotype/">[VCF and tabix indexes]</a>
            </li>
            <li>gnomAD exomes (r2.1):
              <a href="[[SPECIESDEFS::ENSEMBL_FTP_URL]]/data_files/homo_sapiens/GRCh37/variation_genotype/">[VCF and tabix indexes]</a>
            </li>
            <li>ExAC (v0.3):
              <a href="[[SPECIESDEFS::ENSEMBL_FTP_URL]]/data_files/homo_sapiens/GRCh37/variation_genotype/ExAC.0.3.GRCh37.vcf.gz">[VCF]</a>
              <a href="[[SPECIESDEFS::ENSEMBL_FTP_URL]]/data_files/homo_sapiens/GRCh37/variation_genotype/ExAC.0.3.GRCh37.vcf.gz.tbi">[tabix index]</a>
            </li>
          </ul>
        </li>

      </ul>
    </li>
    <li>
      <p>Run VEP with the following command (using the GRCh38 input example) to get locations and continental-level allele frequencies:</p>
      <pre class="code sh_sh">./vep -i examples/homo_sapiens_GRCh38.vcf --cache \
--custom file=gnomad.genomes.r2.0.1.sites.GRCh38.noVEP.vcf.gz,short_name=gnomADg,format=vcf,type=exact,coords=0,fields=AF_AFR%AF_AMR%AF_ASJ%AF_EAS%AF_FIN%AF_NFE%AF_OTH</pre>
      <p>You will then see data under field names as described in the VEP output header:</p>
      <pre class="code sh_sh">## gnomADg : gnomad.genomes.r2.0.1.sites.GRCh38.noVEP.vcf.gz (exact)
## gnomADg_AFR_AF : AFR_AF field from gnomad.genomes.r2.0.1.sites.GRCh38.noVEP.vcf.gz
## gnomADg_AMR_AF : AMR_AF field from gnomad.genomes.r2.0.1.sites.GRCh38.noVEP.vcf.gz
...</pre>
      where the gnomADg field contains the ID (or coordinates if no ID found) of the variant in the VCF file. Any of the fields in the gnomAD file INFO field can be added by appending them to the list in your VEP command.
    </li>
  </ol>
    
  <hr/>
  
  <h2 id="gerp">Conservation scores</h2>
  
  <p> You can use VEP's <a href="vep_custom.html">custom annotation</a>
  feature to add conservation scores to your output. For example, to add GERP
  scores, download the bigWig file from the list below, and run VEP with the
  following flag:</p>
  
  <pre class="code sh_sh">./vep --cache -i example.vcf --custom file=All_hg19_RS.bw,short_name=GERP,format=bigwig</pre>
  
  <p> Example conservation score files: </p>
 
  <div>
    <div style="float:left"> 
      <b>Human (GRCh38)</b>
      <ul>
        <li><a href="http://hgdownload.soe.ucsc.edu/goldenPath/hg38/phastCons7way/" rel="external">phastCons 7-way</a> </li>
        <li><a href="http://hgdownload.soe.ucsc.edu/goldenPath/hg38/phastCons20way/" rel="external">phastCons 20-way</a> </li>
        <li><a href="http://hgdownload.soe.ucsc.edu/goldenPath/hg38/phastCons100way/" rel="external">phastCons 100-way</a> </li>
        <li><a href="http://hgdownload.soe.ucsc.edu/goldenPath/hg38/phyloP7way/" rel="external">phyloP 7-way</a> </li>
        <li><a href="http://hgdownload.soe.ucsc.edu/goldenPath/hg38/phyloP20way/" rel="external">phyloP 20-way</a> </li>
        <li><a href="http://hgdownload.soe.ucsc.edu/goldenPath/hg38/phyloP100way/" rel="external">phyloP 100-way</a> </li>
      </ul>
    </div>
    <div style="float:left;margin-left:40px">
      <b>Human (GRCh37)</b>
      <ul>
        <li><a href="http://hgdownload.soe.ucsc.edu/gbdb/hg19/bbi/All_hg19_RS.bw" rel="external">GERP</a> </li>
        <li><a href="http://hgdownload.soe.ucsc.edu/goldenPath/hg19/phastCons46way/" rel="external">phastCons 46-way</a> </li>
        <li><a href="http://hgdownload.soe.ucsc.edu/goldenPath/hg19/phastCons100way/" rel="external">phastCons 100-way</a> </li>
        <li><a href="http://hgdownload.soe.ucsc.edu/goldenPath/hg19/phyloP46way/" rel="external">phyloP 46-way</a> </li>
        <li><a href="http://hgdownload.soe.ucsc.edu/goldenPath/hg19/phyloP100way/" rel="external">phyloP 100-way</a> </li>
      </ul>
    </div>
    <div class="clear"></div>
  </div>  

  <p> All files provided by the UCSC genome browser - files for other species
  are available from their <a
  href="http://hgdownload.soe.ucsc.edu/goldenPath/" rel="external">FTP site</a>, though be sure
  to use the file corresponding to the <a
  href="/Homo_sapiens/Info/Annotation#assembly">correct assembly</a>. </p>
    
  <hr/>
  
  
  <h2 id="dbNSFP">dbNSFP</h2>
  
  <p> dbNSFP - <a href="http://www.ncbi.nlm.nih.gov/pubmed/21520341"
  rel="external">&quot;a lightweight database of human nonsynonymous SNPs and
  their functional predictions&quot;</a> - provides pathogenicity predictions
  from many tools (including SIFT, PolyPhen, LRT, MutationTaster, FATHMM) across
  every possible missense substitution in the human proteome. The data is
  available to <a href="https://sites.google.com/site/jpopgen/dbNSFP"
  rel="external">download</a>, and while it cannot be immediately used by the
  VEP it is simple to process the data into a format that the dbNSFP.pm plugin
  can use. </p>
  
  <p> After downloading the file, you will need to process it so that tabix can
  index it correctly. This will take a while as the file is very large! Note
  that you will need the <a href="http://samtools.sourceforge.net/tabix.shtml"
  rel="external">tabix</a> utility in your path to use dbNSFP.</p>

  <pre class="code sh_sh">unzip dbNSFP4.0b2a.zip
head -n1 dbNSFP4.0b2a_variant.chr1 > dbNSFP4.0b2a.txt
cat dbNSFP4.0b2a_variant.chr* | grep -v "#" >> dbNSFP4.0b2a.txt
rm dbNSFP4.0b2a_variant.chr*
bgzip dbNSFP4.0b2a.txt
tabix -s 1 -b 2 -e 2 dbNSFP4.0b2a.txt.gz</pre>
  
  <p> Then simply download the <a
  href="https://github.com/Ensembl/VEP_plugins/blob/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/dbNSFP.pm"
  rel="external">dbNSFP VEP plugin</a> and place it either in
  <b>$HOME/.vep/Plugins/</b> or a path in your <b>$PERL5LIB</b>. When you run
  VEP with the plugin, you will need to select some of the columns that you
  wish to retrieve; to list them run VEP with the plugin and the path to the
  dbNSFP file and no further parameters: </p>
  
  <pre class="code sh_sh">./vep --cache --force --plugin dbNSFP,dbNSFP4.0b2a.txt.gz
2014-04-04 11:27:05 - Read existing cache info
2014-04-04 11:27:05 - Auto-detected FASTA file in cache directory
2014-04-04 11:27:05 - Checking/creating FASTA index
2014-04-04 11:27:05 - Failed to instantiate plugin dbNSFP: ERROR: No columns selected to fetch. Available columns are:
#chr,pos(1-coor),ref,alt,aaref,aaalt,hg18_pos(1-coor),genename,Uniprot_acc,
Uniprot_id,Uniprot_aapos,Interpro_domain,cds_strand,refcodon,SLR_test_statistic,
codonpos,fold-degenerate,Ancestral_allele,Ensembl_geneid,Ensembl_transcriptid,
...</pre>
  
  <p> Note that some of these fields are replicates of those produced by the
  core VEP code (e.g. <a href="vep_options.html#opt_sift">SIFT</a>, <a
  href="vep_options.html#opt_polyphen">PolyPhen</a>, the <a
  href="vep_options.html#opt_af_1kg">1000 Genomes</a> and <a
  href="vep_options.html#opt_af_esp">ESP</a> frequencies) - you should use the
  options to enable these from the VEP code in place of the annotations from
  dbNSFP as the dbNSFP file covers <b>only</b> missense substitutions. Other
  fields, such as the conservation scores, may be better served by using
  genome-wide files as described <a href="#gerp">above</a>. </p>
  
  <p> To select fields, just add them as a comma-separated list to your command
  line:</p>
  
  <pre class="code sh_sh">./vep --cache --force --plugin dbNSFP,dbNSFP4.0b2a.txt.gz,LRT_score,FATHM_score,MutationTaster_score</pre>
  
  <p> One final point to note is that the dbNSFP scores are frozen on a
  particular Ensembl release's transcript set; check the readme file on their
  download site to find out exactly which. While in the majority of cases
  protein sequences don't change between releases, in some circumstances the
  protein sequence used by VEP in the latest release may differ from the
  sequence used to calculate the scores in dbNSFP. </p>
  
  <hr/>


  <h2 id="StructVar">Structural Variants</h2>

  <p>VEP can be used to annotate structural variants (SV) with their predicted effect on other genomic features.
     For more information on SV input format, see <a href="/info/docs/tools/vep/vep_formats.html#sv">here</a>.</p>

  <h4>Prediction process</h4>
  <ul>
  <li> The INFO keys 'END' or 'SVLEN' are present, the proportion of any overlapping feature covered by the variant is calculated
  <li> If the SVTYPE or ALT is 'DEL', the variant tested for feature ablation/ truncation
  <li> If the SVTYPE or ALT is 'DUP', the variant tested for feature amplification
  <li> If the SVTYPE or ALT is 'INS' or 'DUP', the variant tested for feature elongatation
  <li> SVTYPE is used in preference to ALT to derive the variant type of an SV with 'CN*' alleles
  </ul>

  <h4> Reported overlaps</h4>
  <ul>
  <li> VEP calculates the length and proportion of each genomic feature overlapped by a structural variant
  <li> Use the <a href="vep_options.html#opt_overlaps">--overlaps</a> option to enable this when using VCF or tab format.
       (This is reported by default in standard VEP and JSON format.)
  <li> The keys bp_overlap and percentage_overlap are used in JSON format and OverlapBP and OverlapPC in other formats.
  </ul>

  <h4> Changing memory requirements</h4>
  <ul>
  <li> By default, VEP does not annotate variants larger than 10M. If you are using the command
  line tool, you can use the <a href="vep_options.html#opt_max_sv_size">--max_sv_size</a> option to modify this.
  <li>By default, variants are analysed in batches of 5000. Using the <a href="vep_options.html#opt_buffer_size">--buffer_size</a>
  option to reduce this can reduce memory requirements, especially if your data is sparse.
  A smaller buffer size is essential when annotating structural variants with regulatory data.
  </ul>

  <hr/>


  
  <h2 id="citations">Citations and VEP users</h2>
  
  <p> VEP is used by many organisations and projects: </p>
  
  <ul>
    <li> VEP forms a part of <a href="http://variantstudio.software.illumina.com/" rel="external">Illumina's VariantStudio</a> software </li>
    <li> <a href="http://gemini.readthedocs.org/" rel="external">Gemini</a> is a framework for exploring genome variation that uses VEP </li>
    <li> The <a href="http://decipher.sanger.ac.uk/" rel="external">DECIPHER project</a> uses VEP in its analysis pipelines</li>
  </ul>
  
  <p> Other citations and use cases: </p>
  
  <ul>
    <li> <a href="http://bib.oxfordjournals.org/content/early/2014/03/12/bib.bbu008.full" rel="external">VAX</a> is a suite of plugins for VEP that expands its functionality</li>
    <li> <a href="http://research-pub.gene.com/pviz/app/vep/" rel="external">pViz</a> is a visualisation tool for VEP results files </li>
    <li> <a href="http://genomemedicine.com/content/6/3/26/abstract" rel="external">McCarthy <i>et al</i></a> compares VEP to AnnoVar </li>
    <li> <a href="http://bib.oxfordjournals.org/content/early/2013/01/21/bib.bbs086.full" rel="external">Pabinger <i>et al</i></a> reviews variant analysis software, including VEP</li>
    <li> VEP is used to provide annotation for the <a href="http://exac.broadinstitute.org/" rel="external">ExAC</a> and <a href="http://gnomad.broadinstitute.org/" rel="external">gnomAD</a> projects </li>
  </ul>
</div>

</body>
</html>
