<html>
<head>
<meta name="order" content="1" />
<title>Linking in to Ensembl</title>
</head>

<body>

<h1>Linking in to Ensembl</h1>

<div class="plain-box">
<h2>Attaching a file via URL</h2>
<p>You can quickly and easily attach any 
<a href="/info/website/upload/index.html#formats">supported file format</a> 
to Region in Detail using the following URL pattern:


<p><kbd>http:/www.ensembl.org/[name_of_species]/Location/View?r=[coordinates];attach=[url_of_your_file]</kbd></p>

<p>Note that if your file URL contains semicolons, ampersands (&amp;) or equals signs, these will need to be URL-encoded.</p>

<p>For attaching data to other images or pages, see the <a href="#advanced">advanced customisation tips</a> below.</p>

</div>

<div class="plain-box">
<h2>Linking on a stable ID</h2>
<p>If you simply wish to link to a gene, transcript, protein or gene tree page, 
particularly for automated links, you can do so using the following template:</p>

<p><kbd>[site]/id/[stable id]</kbd></p>

<p>For example, <a href="http://www.ensembl.org/id/ENSG00000139618">http://www.ensembl.org/id/ENSG00000139618</a>, or <a href="http://grch37.ensembl.org/id/ENSMUST00000103109">http://grch37.ensembl.org/id/ENSMUST00000103109</a>.</p>

<p>As you can see, you don't need to know the name of the species that the ID belongs to;
the website will work that out automatically based on the 3-6 letter prefix.</p>

<p>Note however that you do have to use Ensembl stable IDs; you cannot substitute a 
gene symbol such as FOXP2, as that is found in many species. To link on a gene symbol,
see the <a href="#standard_url">standard URL structure</a> section, below.</p>

<p><a href="/Help/Glossary?id=349">Find out more about stable IDs</a>.</p>

<p>Valid ID types are:</p>

<ul>
<li>Gene: ENS[species code]<b>G</b>[number]</li>
<li>Transcript: ENS[species code]<b>T</b>[number]</li>
<li>Protein: ENS[species code]<b>P</b>[number]</li>
<li>Gene tree: ENS[species code]<b>GT</b>[number]</li>
</ul>

<p>The only site this type of URL will not work with is 
<a href="http://pre.ensembl.org">Ensembl Pre</a> since, by their nature,
most preview species have no genebuild and thus no stable IDs.</p>
</div>

<div class="plain-box">
<h2 id="standard_url">More about Ensembl URLs</h2>

<p>
  URLs for Ensembl views take the format:
</p>

<p><kbd>
  http://www.ensembl.org/<em>species</em>/<em>section</em>/<em>page</em>?g=<em>gene</em>;t=<em>transcript</em>;r=<em>seq_region</em>:<em>start</em>-<em>end</em>
</kbd></p>

<h3>Location-based views</h3>

<p>
  Any combination of the gene, transcript and genomic region may be specified.
  For example, the following all take the user to the genomic region surrounding
  the human gene BRCA1:
</p>

<ul>
  <li><a href="http://www.ensembl.org/Homo_sapiens/Location/View?g=ENSG00000012048">http://www.ensembl.org/Homo_sapiens/Location/View?g=ENSG00000012048</a></li>
  <li><a href="http://www.ensembl.org/Homo_sapiens/Location/View?t=ENST00000309486">http://www.ensembl.org/Homo_sapiens/Location/View?t=ENST00000309486</a></li>
  <li><a href="http://www.ensembl.org/Homo_sapiens/Location/View?r=17:38449840-38530994">http://www.ensembl.org/Homo_sapiens/Location/View?r=17:38449840-38530994</a></li>
</ul>

<h3>Feature-based views</h3>

<p>Views based on a gene, transcript or variation need the parameter of that type in their URL - a location alone is insufficient:</p>

<ul>
<li><strong>RIGHT</strong>: <a href="http://www.ensembl.org/Homo_sapiens/Gene/Summary?g=ENSG00000012048">http://www.ensembl.org/Homo_sapiens/Gene/Summary?g=ENSG00000012048</a></li>
<li><strong>WRONG!</strong> http://www.ensembl.org/Homo_sapiens/Gene/Summary?r=17:38449840-38530994</li>
</ul>


<h2 id="advanced">Advanced link customisation</h2>

<p>
  Many of the customisation options available to users of the website in the
  <a href="/info/website/control_panel.html">Control Panel</a> can be
  manipulated using specially constructed URLs. This allows configuration
  settings to be shared with colleagues, for example to highlight a particular
  data track. This includes:
</p>

<ul>
  <li>Configuring options for a page</li>
  <li>Configuring a panel/image (changing the way tracks are rendered)</li>
  <li>Attaching external data via URL, without using the Control Panel</li>
</ul>

<p>
  These are accomplished in a similar, systematic way. Extra parameters are
  appended to the URL for the relevant page:
</p>

<p><kbd>
  context=key=value
</kbd></p>

<p>
  Or when setting multiple values:
</p>

<p><kbd>
  context=key1=value1,key2=value2,key2=value2
</kbd></p>

<p>
  Or when setting multiple different types of parameter:
</p>

<p><kbd>
  context1=key1=value1,key2=value2,key2=value2;context2=key3=value3
</kbd></p>

<h3>Global Page Settings</h3>

<p>
  Global page settings are changed using the <strong>config</strong> key. For
  example, to turn off the top image on Location &gt; Region in detail:
</p>

<p><kbd>
  config=show_top_panel=no
</kbd></p>

<p>
  Appending this to the URL for the <em>Region in detail</em> page, the final
  URL is:
</p>

<p><kbd>
  <a href="http://www.ensembl.org/Homo_sapiens/Location/View?r=1:1000-2000;config=show_top_panel=no">http://www.ensembl.org/Homo_sapiens/Location/View?r=1:1000-2000;config=show_top_panel=no</a>
</kbd></p>

<p>
  To link directly to the <em>Exon Intron markup</em> panel (Transcript &gt;
  Exons), to show full introns, only 60bp flanking sequence AND turn the display
  to be 60bp wide:
</p>

<p><kbd>
  <a href="http://www.ensembl.org/Homo_sapiens/Transcript/Exons?t=ENST00000309255;config=flanking=60,seq_cols=60,fullseq=yes">http://www.ensembl.org/Homo_sapiens/Transcript/Exons?t=ENST00000309255;config=flanking=60,seq_cols=60,fullseq=yes</a>
</kbd></p>

<h3>Configuring Panels</h3>

<p>
  To change configuration for an individual panel add a parameter referring to
  the panel:
</p>

<ul>
  <li>
    Location &gt; Region in detail: <em>contigviewtop</em> and 
    <em>contigviewbottom</em>
  </li>
  <li>
    Location &gt; Region overview: <em>cytoview</em>
  </li>
</ul>

<p>
  Options are again specificed in a comma separated list, where the left hand
  side of each "=" is the name of the track, and the right hand side is the name
  of the "renderer" to use - the latter depends on the type of track.
</p>

<h4>Track names</h4>

<p>
  Track names are systematically named. The following example identifies a track
  with Ensembl transcripts from the core Ensembl database:
</p>

<p><kbd>
  transcript_core_ensembl
</kbd></p>

<h4>Renderers</h4>

<p>
  Renderers depend on the type of track:
</p>

<ul>
  <li>
    Transcripts: <em>transcript_label</em>, <em>transcript_nolabel</em>,
    <em>collapsed_label</em>, <em>collapsed_nolabel</em> or <em>off</em>
  </li>
  <li>
    Alignment features: <em>normal</em>, <em>half_height</em>, <em>stack</em>,
    <em>unlimited</em>, <em>ungrouped</em> or <em>off</em>
  </li>
</ul>

<h4>Example - configure a track</h4>

<p>
  To collapse the Ensembl genes track down to a single line per
  gene AND turn off the top panel in <em>Location &gt; Region in detail</em>:
</p>

<p><kbd>
  <a href="http://www.ensembl.org/Homo_sapiens/Location/View?g=ENSG00000012048;contigviewbottom=transcript_core_ensembl=collapsed_nolabel;config=show_top_panel=no">http://www.ensembl.org/Homo_sapiens/Location/View?g=ENSG00000012048;contigviewbottom=transcript_core_ensembl=collapsed_nolabel;config=show_top_panel=no</a>
</kbd></p>

<h3>Attaching External Data</h3>

<p>
  To attach URL data, follow the procedure above for
  configuring the appropriate panel, using special track names to identify the
  data source. These special names are of the format:
</p>

<p><kbd>
  url:source
</kbd></p>

<p>
  where <em>source</em> is the location of the external data.
</p>

<div class="info-box">
<p><strong>IMPORTANT:</strong> You <strong>must</strong> encode any = &amp; or ; characters in your URL for the attachment to work correctly.</p>
</div>

<h4>Renderers</h4>

<p>
  Renderers are the same as for alignment tracks (see above), or for wiggle/graph data, either 'off' or 'signal'.
</p>

<h4 id="attachurl">Examples - attach external data</h4>

<p>To attach an example data file by URL and display the features with the default renderer:</p>

<p><kbd>
  <a href="http://www.ensembl.org/Homo_sapiens/Location/View?g=ENSG00000130544;contigviewbottom=url:http://www.ensembl.org/info/website/upload/sample_files/example.bed">
    http://www.ensembl.org/Homo_sapiens/Location/View?g=ENSG00000130544;contigviewbottom=url:http://www.ensembl.org/info/website/upload/sample_files/example.bed
  </a>
</kbd></p>

<p>To attach an example data file by URL and display the features in a half_height style:</p>

<p><kbd>
  <a href="http://www.ensembl.org/Homo_sapiens/Location/View?g=ENSG00000130544;contigviewbottom=url:http://www.ensembl.org/info/website/upload/sample_files/example.bed=half-height">
    http://www.ensembl.org/Homo_sapiens/Location/View?g=ENSG00000130544;contigviewbottom=url:http://www.ensembl.org/info/website/upload/sample_files/example.bed=half_height
  </a>
</kbd></p>

<p><strong>Note on bedGraph format</strong> If attaching a bedGraph file, please add the parameter 'format=bedGraph' to the URL, e.g.</p>

<p><kbd>
  <a href="http://www.ensembl.org/Homo_sapiens/Location/View?g=ENSG00000012048;contigviewbottom=url:http://www.ensembl.org/info/website/upload/sample_files/example.bed=ungrouped">
    http://www.ensembl.org/Homo_sapiens/Location/View?g=ENSG00000012048;contigviewbottom=url:http://www.abcd.edu/myprojects/data.bed=signal;format=bedGraph
  </a>
</kbd></p>
</div>

</body>
</html>
