<html>

<head>
<meta name="order" content="5" />
  <meta name="navigation" content="Regulation">
  <title>Regulation</title>
</head>


<body lang="en-GB" dir="LTR">

<h1>Regulation Summary</h1>
<p>
Ensembl Regulation provides resources describing the mechanisms of gene regulation in human and mouse cells. In particular, it focuses on transcriptional and post-transcriptional mechanisms.
</p>

<h2 id="regbuild">Regulatory Features</h2>
<p>
Ensembl maintains a database of regions in the human and mouse genomes that could be involved in gene transcription regulation. These regions, called regulatory features, are inferred from <a href="regulation_sources.html">publicly available experimental data sets</a>, e.g.:
<ul>
	<li>Open chromatin assays (<a href="http://en.wikipedia.org/wiki/DNase-Seq" rel="external">DNase1-Seq</a> and <a href="http://en.wikipedia.org/wiki/faire-seq" rel="external">FAIRE-Seq)</a></li>
	<li>Histone modification assays (<a href="http://en.wikipedia.org/wiki/FAIRE-Seq" rel="external">ChIP-Seq</a>)</li>
	<li>Transcription factor binding assays (<a href="http://en.wikipedia.org/wiki/Chip-Seq" rel="external">ChIP-Seq</a>)</li>
	<li><a href="regulatory_build.html#tfbs">Transcription factor binding motifs</a>.</li>
</ul>
</p>
<p>
From this data, a process called the <a href="regulatory_build.html">Regulatory Build</a> computes MultiCell features, independently of cell-type. For each of 13 human and 5 mouse cell lines, each feature is annotated into one of the following categories:
<ul>
	<li>Predicted promoters</li>
	<li>Predicted promoter flanking regions</li>
	<li>Predicted enhancers</li>
	<li>CTCF binding sites</li>
	<li>Unannotated transcription factor binding sites</li>
	<li>Unannotated open chromatin regions</li>
</ul>
</p>

<p>
<h2 id="segmentation">Regulatory Segmentation</h2>
Ensembl also provides genome <a href="regulatory_segmentation.html">segmentation analyses</a> for six human cells lines from the <a href="http://encodeproject.org" rel="external">ENCODE</a> project, an analogous summary of a region's regulatory function.
</p>

<h2 id="microarray">Microarray Probe Mappings</h2>
<p>
Ensembl stores <a href="/info/genome/microarray_probe_set_mapping.html">microarray probe mappings</a> for several species and technologies, including:
<ul>
<li>Affymetrix: IVT &amp; ST arrays</li>
<li>Codelink: Whole genome arrays</li>
<li>Agilent: Whole genome and CGH arrays <!-- add Sureprint here for 66--></li>
<li>Illumina: Whole genome and Infinium methylation arrays</li>
<li>Phalanx</li>
</ul>
</p>

<h2 id="other">Other Regulatory Data</h2>
<p>
Ensembl Regulation databases also store data directly imported from external sources:
<ul>
<li>Micro RNA target predictions for human and mouse using <a rel="external" href="http://diana.imis.athena-innovation.gr/DianaTools/index.php?r=tarbase/index">Diana TarBase</a>.</li>
<li>Experimentally validated human enhancers from the <a rel="external" href="http://enhancer.lbl.gov/">VISTA Browser</a>.</li>
<li>Reduced Representation Bisulfite Sequencing (RRBS) assays on 44 cell lines and Whole Genome Bisulphite Sequencing (assays) on three cell lines (<a rel="external" href="http://encodeproject.org">ENCODE</a>).<br></li>
</ul>
</p>

<h2 id="displays">Displaying Regulatory Features</h2>
<p>
Regulation data can be accessed in the browser from various angles:
</p>

<h3><a href="/Homo_sapiens/Location/View?r=6:133017695-133161157">Location Tab</a></h3>

<p>
You can configure the Region in detail panel to display tracks linked to regulation. Watch our Region in detail video to find out how to add tracks. The Regulation tracks are grouped into subdivisions:
<ul>
	<li><b>Regulatory Features:</b> these are the regions computed by the Regulatory Build (see above). You can specify regulatory feature tracks for individual cell lines or summary MultiCell features.</li>
	<li><b>Open Chromatin and Transcription Factor Binding Sites (TFBS):</b> experimental ChIP-Seq assays measuring open chromatin (FAIRE or DNAse experiments) or transcription factor binding.</li>
	<li><b>Histones and Polymerases:</b> experimental ChIP-Seq assays measuring histone marks or Polymerases II and III binding.</li>
	<li><b>DNA Methylation:</b> RRBS and WGBS methylation tracks.</li>
	<li><b>Other regulatory regions:</b> tracks from external databases (e.g VISTA, cisRED, among others).</li>
</ul>
</p>
<p>
Species-specific microarray probe mappings can be visualised by turning on tracks from the separate <i>Oligo probes</i> section at the bottom of the configuration panel. 
</p>

<h3><a href="/Homo_sapiens/Gene/Summary?g=ENSG00000139618;r=13:32889611-32973805">Gene tab</a></h3>
<p>
Click on the Regulation link in the left hand side menu to view the <a href="/Homo_sapiens/Gene/Regulation?g=ENSG00000139618;r=13:32889611-32973805">regulatory features in the vicinity of your gene</a>. Note that even if a regulatory region is near or within a gene, it does not necessarily affect that gene. Conversely, distal enhancers can affect a gene.
</p>

<h3><a href="/Homo_sapiens/Regulation/Cell_line?db=funcgen;fdb=funcgen;r=17:46617590-46621119;rf=ENSR00001347721">Regulation tab</a></h3>
<p>
Clicking on a regulatory feature will open a Regulation tab with information about the evidence supporting that regulatory feature as well as cell-specific activity estimates. Different views can be selected:
<ul>
	<li><b>Summary:</b> displays both the MultiCell (in grey) and cell-specific regulatory features. The core evidence (see <a href="regulatory_build.html">the Regulatory Build</a> for further explanations) is displayed for MultiCell features by default but not for cell-specific features.</li>
	<li><b>Details by cell line:</b> this is the default view, where you get the generic MultiCell regulatory feature, cell-specific annotation and a default set of supporting evidence. You can display more evidence tracks by clicking on <i>Configure this page</i>.</li>
	<li><b>Feature Context:</b> this view displays the regulatory features in a wider context around the chosen regulatory feature.</li>
	<li><b>Evidence:</b> this view displays the complete list of supporting core evidence (transcription factor (TF) binding and open chromatin) for a regulatory feature.</li>
</ul>
</p>

<br>

<h2 id="access">Data Access</h2>
<p>
In addition to the browser, the regulation data in Ensembl can also be accessed through :
<ul>
	<li><a href="/info/data/biomart.html">BioMart (Ensembl Regulation database)</a>.</li>
	<li><a href="/info/docs/api/index.html">Perl APIs</a>, including the <a href="/info/docs/api/funcgen/index.html">Regulation API</a>. See the <a href="/info/docs/api/funcgen/regulation_tutorial.html">Regulation API tutorial</a> for more information.</li>
	<li><a href="/info/docs/api/funcgen/funcgen_schema.html">MySQL Ensembl funcgen database.</a></li>
</ul>
</p>

</body>
</html>
