<HTML>

<head>
<meta name="order" content="2" />
  <meta name="navigation" content="Regulation Sources">
  <title>Regulation Sources</title>
</head>


<BODY LANG="en-GB" DIR="LTR">

<h1>Datasets and Data Processing</h1>

<h2>Datasets</h2>
<p>
Regulatory features are generated by using a variety of genome-wide assays, such as ChIP-seq data on chromatin structure, epigenomic, and transcription factor binding. Metadata and links to the original data sources can be visualised via the the various experiment views:
<ul>
<li>Human</li>
<ul></li>
	<li><a href='/Homo_sapiens/Experiment/Sources?db=funcgen;ex=all;fdb=funcgen;r=17:46617590-46621119;rf=ENSR00001348194#ExperimentalMetaData'>All datasets (cell types and assays).</a></li>
	<li><a href='/Homo_sapiens/Experiment/Sources?db=funcgen;ex=project-Roadmap Epigenomics-;fdb=funcgen;r=17:46617590-46621119;rf=ENSR00001348194#ExperimentalMetaData'>Roadmap Epigenomics data.</a></li>
	<li><a href='/Homo_sapiens/Experiment/Sources?db=funcgen;ex=project-ENCODE-;fdb=funcgen;r=17:46617590-46621119;rf=ENSR00001348194#ExperimentalMetaData'>ENCODE data.</a></li>
  <li><a href='/Homo_sapiens/Experiment/Sources?db=funcgen;ex=project-BLUEPRINT-;fdb=funcgen;r=17:46617590-46621119#ExperimentalMetaData'>BLUEPRINT data.</a></li>

</ul></li>
<li>Mouse</li>
<ul></li>
	<li><a href='/Mus_musculus/Experiment/Sources?db=funcgen;ex=all;fdb=funcgen;r=17:46617590-46621119#ExperimentalMetaData'>All datasets (cell types andassays).</a></li>
	<li><a href='/Mus_musculus/Experiment/Sources?db=funcgen;ex=project-ENCODE-;fdb=funcgen;r=17:46617590-46621119#ExperimentalMetaData'>ENCODE data.</a></li>
</ul>
</ul>
</p>

<p>
At the bottom of these views, the results table also shows the mappings between transcription factors, Ensembl genes and Jaspar matrices that have been used in the regulatory build.
</p>

<h2>Read alignment</h2>
<p>
To maintain a standardised peak calling methodology, we start our analyses with raw reads from each experiment. We align reads (replicates are pooled) to the genome using <a rel='external' href='http://www.ncbi.nlm.nih.gov/pubmed/19451168'>BWA</a> (with default parameters). All matches to the mitochondrial DNA are filtered out to avoid alignment anomalies due to similarities with autosomal regions.
</p>

<h2>Peak calling</h2>
<p>
Peak calling is performed using two algorithms:
<ul>
  <li>SWEMBL (S. Wilder <i>et al.</i>, in preparation):
    <ul>
      <li>When replicates are available, permissive settings (-f 150 -R 0.0005 -d 150) are used, of which we choose the best peaks, so as to fit an IDR score of 0.01. This number of peaks is linearly adjusted to take into accounts replicate pairs with different numbers of peaks.</li>
      <li>Otherwise, different stringent settings are used for very tight peaks (e.g. DNAse1) (-f 150 -R 0.0025) and larger ones (-f 150 -R 0.015).</li>
    </ul>
  </li>
  <li><a rel='external' href='http://www.ncbi.nlm.nih.gov/pubmed?term=20371496'>CCAT (Xu <i>et al.</i>, 2010)</a>
    <ul>
      <li>The algorithm described by Xu et al, is specifically designed for peak calling of broad features, such as H3K36me3. The parameters were set empirically to (fragmentSize 200; slidingWinSize 150; movingStep 20; isStrandSensitiveMode 0; minCount 10; outputNum 100000; randomSeed 123456; minScore 4.0; bootstrapPass 50).</li>
    </ul>
  </li>
</ul>
</p>

<h2>Peak filtering</h2>
<p>
The <a rel='external' href='http://hgwdev.cse.ucsc.edu/cgi-bin/hgFileUi?db=hg19&g=wgEncodeMapability'>ENCODE DAC Blacklist regions</a> are then used to filter the resultant peaks. These regions are stored in the core database as 'encode_excluded' MiscFeatures.
</p>


</BODY>
</HTML>
