<html>
<head>
<meta name="order" content="1" />
<title>Predicted Data</title>
</head>

<body>
<b>
<a href="index.html">About Ensembl Variation</a> |
<a href="data_description.html">Data Description</a> | 
Predicted Data |
<a href="database.html">Database Description</a> | 
<a href="/info/docs/api/variation/index.html">Perl API</a> |
<a href="vep/index.html">Variant Effect Predictor</a>
</b>


<div>
  <div style="float:right;margin-left:8px;margin-top:2px;background-color:#F2F2F2;color:#333;border-radius:5px">
	  <div style="padding:5px;font-weight:bold;color:#FFF;background-color:#336;border-top-left-radius:5px;border-top-right-radius:5px">
		  <img src="/i/16/rev/info.png" style="vertical-align:top"/>
		  Page content:
		</div>
    <ul style="padding-left:25px;padding-right:5px">
      <li><a href="#consequences">Variation Consequences</a></li>
      <li><a href="#nsSNP">Protein function predictions</a>
			  <ul style="padding-left:25px">
				  <li><a href="#sift">SIFT</a></li>
					<li><a href="#polyphen">PolyPhen</a></li>
					<li><a href="#condel">Condel</a></li>
			  </ul>
			</li>
      <li><a href="#references">References</a></li>
	  </ul>
  </div>

  <br />
  <h1>Ensembl Variation - Predicted data</h1>

  <p>
  Ensembl imports variation data from a variety of different sources, as
  described on the <a href="data_description.html">Data description</a> page. Below we give more
  information about how Ensembl predicts the effects of variants. Ensembl calculates the:
  </p>
	
	<table>
    <tr><td>
		  <a href="/Homo_sapiens/Variation/Mappings?v=rs699"><img src="/i/96/var_gene_transcript.png" class="portal" alt="Variation consequences and predictions of a variant, by gene and
			transcript" title="Variation consequences and predictions of a variant, by gene and transcript"/></a>
		</td><td>	
	    <ul style="padding-left:5px">
	      <li>Consequence of variations in transcripts (synonymous, missense,...), e.g. 
		      <a href="/Homo_sapiens/Variation/Mappings?v=rs699#ENST00000366667_6797_G_tablePanel">rs699</a>
		    </li>
	      <li>Protein function prediction (SIFT, PolyPhen), e.g. 
		      <a href="/Homo_sapiens/Variation/Mappings?v=rs56404215">rs56404215</a>
		    </li>
	    </ul>
	  </td></tr>
		<tr><td>
		  <a href="/Homo_sapiens/Variation/HighLD?v=rs1333049"><img src="/i/96/var_linkage_disequilibrium.png" class="portal" alt="Linkage disequilibrium" title="Linkage disequilibrium"/></a>
		</td><td>	
	    <ul style="padding-left:5px">
	      <li>Linkage disequilibrium information, e.g. 
		      <a href="/Homo_sapiens/Variation/HighLD?v=rs1333049">rs1333049</a>
		    </li>
	      <li>Tagged variations</li>
      </ul>
		</td></tr>
	</table>		
	<br />
</div>
<div style="clear:both" />


<h2 id="consequences">Calculated variation consequences</h2>

<p>
For each variation that is mapped to the reference genome, we identify each
Ensembl transcript that overlap the variation. We then use a rule-based approach to
predict the effects that each allele of the variation may have on the transcript.
The set of consequence terms, defined by the <a rel="external"
href="http://www.sequenceontology.org">Sequence Ontology</a> (SO), that can be
currently assigned to each combination of an allele and a transcript is shown
in the table below. Note that each allele of each variation may have a different
effect in different transcripts. 
</p>

<p> 
This approach is applied to all germline variations and somatic mutations
stored in the Ensembl variation databases (though we do not yet currently calculate
consequences for structural variants). The resulting consequence type calls,
along with information determined as part of the process, such as the cDNA and
CDS coordinates, and the affected codons and amino acids in coding transcripts,
are stored in the variation database and displayed on the website. You can use 
this pipeline for your own data via the <a href="/info/docs/variation/vep">VEP</a>. 
</p>

<p>
Prior to release 68 we used our own internal terms to describe consequence
types and these continue to be available as an option on the website using 
'Configure this page', in the API and in the <a href="#VEP">VEP</a>. We refer
to these as 'Old Ensembl terms' and list these and their SO equivalents in 
the table below.
</p>

<p>
From release 68 however, we use <a rel="external"
href="http://www.sequenceontology.org">SO terms </a> by default. There is an equivalent 
SO term for each of our old Ensembl terms but in a few cases there is a more specific 
SO term available, as shown in the table below. If you have text files or VEP outputs 
with our old Ensembl terms, you can easily update these to using the SO terms by running the following 
<a href="http://cvs.sanger.ac.uk/cgi-bin/viewvc.cgi/ensembl-variation/scripts/misc/convert_ensembl_to_SO_consequences.pl?view=co&root=ensembl&pathrev=branch-ensembl-68">
script</a> e.g. <br />
</p>
<p>
<code style="border: 1px solid #CCC;padding:3px" >perl convert_ensembl_to_SO_consequences.pl input.txt > converted.txt</code>
</p>



<p>
The terms in the table are shown in order of severity as estimated by
Ensembl, and this ordering is used on the website summary views. This ordering
is necessarily subjective and API and VEP users can always get the full set of
consequences for each allele and make their own severity judgement.
</p>


<p>
See below a diagram showing the location of each display 
term relative to the transcript structure:
</p>
<a rel="external" href="consequences.jpg">
  <img src="consequences.jpg" style="width:700px;height:250px" alt="consequence diagram" />
</a>


<br /><br />

<table id="consequence_type_table" class="ss">
<tr>
	<th style="width:5px;padding-left:0px;padding-right:0px;text-align:center">*</th>
	<th>SO term</th>
	<th>SO description</th>
	<th>SO accession</th>
	<th>Old Ensembl term</th>
</tr>
<tr>
	<td style="padding:0px;margin:0px;background-color:#ff0000"></td>	<td>transcript_ablation</td>
	<td>A feature ablation whereby the deleted region includes a transcript feature</td>
	<td><a rel="external" href="http://www.sequenceontology.org/miso/current_release/term/SO:0001893">SO:0001893</a></td>
	<td>Transcript ablation</td>
</tr>
<tr class="bg2">
	<td style="padding:0px;margin:0px;background-color:#ff7f50"></td>	<td>splice_donor_variant</td>
	<td>A splice variant that changes the 2 base region at the 5' end of an intron</td>
	<td><a rel="external" href="http://www.sequenceontology.org/miso/current_release/term/SO:0001575">SO:0001575</a></td>
	<td rowspan="2">Essential splice site</td>
</tr>
<tr class="bg2">
	<td style="padding:0px;margin:0px;background-color:#ff7f50"></td>	<td>splice_acceptor_variant</td>
	<td>A splice variant that changes the 2 base region at the 3' end of an intron</td>
	<td><a rel="external" href="http://www.sequenceontology.org/miso/current_release/term/SO:0001574">SO:0001574</a></td>
</tr>
<tr>
	<td style="padding:0px;margin:0px;background-color:#ff0000"></td>	<td>stop_gained</td>
	<td>A sequence variant whereby at least one base of a codon is changed, resulting in a premature stop codon, leading to a shortened transcript</td>
	<td><a rel="external" href="http://www.sequenceontology.org/miso/current_release/term/SO:0001587">SO:0001587</a></td>
	<td>Stop gained</td>
</tr>
<tr class="bg2">
	<td style="padding:0px;margin:0px;background-color:#ff69b4"></td>	<td>frameshift_variant</td>
	<td>A sequence variant which causes a disruption of the translational reading frame, because the number of nucleotides inserted or deleted is not a multiple of three</td>
	<td><a rel="external" href="http://www.sequenceontology.org/miso/current_release/term/SO:0001589">SO:0001589</a></td>
	<td>Frameshift coding</td>
</tr>
<tr>
	<td style="padding:0px;margin:0px;background-color:#ff0000"></td>	<td>stop_lost</td>
	<td>A sequence variant where at least one base of the terminator codon (stop) is changed, resulting in an elongated transcript</td>
	<td><a rel="external" href="http://www.sequenceontology.org/miso/current_release/term/SO:0001578">SO:0001578</a></td>
	<td>Stop lost</td>
</tr>
<tr class="bg2">
	<td style="padding:0px;margin:0px;background-color:#ffd700"></td>	<td>initiator_codon_variant</td>
	<td>A codon variant that changes at least one base of the first codon of a transcript</td>
	<td><a rel="external" href="http://www.sequenceontology.org/miso/current_release/term/SO:0001582">SO:0001582</a></td>
	<td rowspan="4">Non synonymous coding</td>
</tr>
<tr class="bg2">
	<td style="padding:0px;margin:0px;background-color:#ff69b4"></td>	<td>inframe_insertion</td>
	<td>An inframe non synonymous variant that inserts bases into in the coding sequence</td>
	<td><a rel="external" href="http://www.sequenceontology.org/miso/current_release/term/SO:0001821">SO:0001821</a></td>
</tr>
<tr class="bg2">
	<td style="padding:0px;margin:0px;background-color:#ff69b4"></td>	<td>inframe_deletion</td>
	<td>An inframe non synonymous variant that deletes bases from the coding sequence</td>
	<td><a rel="external" href="http://www.sequenceontology.org/miso/current_release/term/SO:0001822">SO:0001822</a></td>
</tr>
<tr class="bg2">
	<td style="padding:0px;margin:0px;background-color:#ffd700"></td>	<td>missense_variant</td>
	<td>A sequence variant, that changes one or more bases, resulting in a different amino acid sequence but where the length is preserved</td>
	<td><a rel="external" href="http://www.sequenceontology.org/miso/current_release/term/SO:0001583">SO:0001583</a></td>
</tr>
<tr>
	<td style="padding:0px;margin:0px;background-color:#ff69b4"></td>	<td>transcript_amplification</td>
	<td>A feature amplification of a region containing a transcript</td>
	<td><a rel="external" href="http://www.sequenceontology.org/miso/current_release/term/SO:0001889">SO:0001889</a></td>
	<td>Transcript amplification</td>
</tr>
<tr class="bg2">
	<td style="padding:0px;margin:0px;background-color:#ff7f50"></td>	<td>splice_region_variant</td>
	<td>A sequence variant in which a change has occurred within the region of the splice site, either within 1-3 bases of the exon or 3-8 bases of the intron</td>
	<td><a rel="external" href="http://www.sequenceontology.org/miso/current_release/term/SO:0001630">SO:0001630</a></td>
	<td>Splice site</td>
</tr>
<tr>
	<td style="padding:0px;margin:0px;background-color:#ff00ff"></td>	<td>incomplete_terminal_codon_variant</td>
	<td>A sequence variant where at least one base of the final codon of an incompletely annotated transcript is changed</td>
	<td><a rel="external" href="http://www.sequenceontology.org/miso/current_release/term/SO:0001626">SO:0001626</a></td>
	<td>Partial codon</td>
</tr>
<tr class="bg2">
	<td style="padding:0px;margin:0px;background-color:#76ee00"></td>	<td>synonymous_variant</td>
	<td>A sequence variant where there is no resulting change to the encoded amino acid</td>
	<td><a rel="external" href="http://www.sequenceontology.org/miso/current_release/term/SO:0001819">SO:0001819</a></td>
	<td rowspan="2">Synonymous coding</td>
</tr>
<tr class="bg2">
	<td style="padding:0px;margin:0px;background-color:#76ee00"></td>	<td>stop_retained_variant</td>
	<td>A sequence variant where at least one base in the terminator codon is changed, but the terminator remains</td>
	<td><a rel="external" href="http://www.sequenceontology.org/miso/current_release/term/SO:0001567">SO:0001567</a></td>
</tr>
<tr>
	<td style="padding:0px;margin:0px;background-color:#458b00"></td>	<td>coding_sequence_variant</td>
	<td>A sequence variant that changes the coding sequence</td>
	<td><a rel="external" href="http://www.sequenceontology.org/miso/current_release/term/SO:0001580">SO:0001580</a></td>
	<td>Coding unknown</td>
</tr>
<tr class="bg2">
	<td style="padding:0px;margin:0px;background-color:#458b00"></td>	<td>mature_miRNA_variant</td>
	<td>A transcript variant located with the sequence of the mature miRNA</td>
	<td><a rel="external" href="http://www.sequenceontology.org/miso/current_release/term/SO:0001620">SO:0001620</a></td>
	<td>Within mature miRNA</td>
</tr>
<tr>
	<td style="padding:0px;margin:0px;background-color:#7ac5cd"></td>	<td>5_prime_UTR_variant</td>
	<td>A UTR variant of the 5' UTR</td>
	<td><a rel="external" href="http://www.sequenceontology.org/miso/current_release/term/SO:0001623">SO:0001623</a></td>
	<td>5prime UTR</td>
</tr>
<tr class="bg2">
	<td style="padding:0px;margin:0px;background-color:#7ac5cd"></td>	<td>3_prime_UTR_variant</td>
	<td>A UTR variant of the 3' UTR</td>
	<td><a rel="external" href="http://www.sequenceontology.org/miso/current_release/term/SO:0001624">SO:0001624</a></td>
	<td>3prime UTR</td>
</tr>
<tr>
	<td style="padding:0px;margin:0px;background-color:#02599c"></td>	<td>intron_variant</td>
	<td>A transcript variant occurring within an intron</td>
	<td><a rel="external" href="http://www.sequenceontology.org/miso/current_release/term/SO:0001627">SO:0001627</a></td>
	<td>Intronic</td>
</tr>
<tr class="bg2">
	<td style="padding:0px;margin:0px;background-color:#ff4500"></td>	<td>NMD_transcript_variant</td>
	<td>A variant in a transcript that is the target of NMD</td>
	<td><a rel="external" href="http://www.sequenceontology.org/miso/current_release/term/SO:0001621">SO:0001621</a></td>
	<td>NMD transcript</td>
</tr>
<tr>
	<td style="padding:0px;margin:0px;background-color:#32cd32"></td>	<td>non_coding_exon_variant</td>
	<td>A sequence variant that changes non-coding exon sequence</td>
	<td><a rel="external" href="http://www.sequenceontology.org/miso/current_release/term/SO:0001792">SO:0001792</a></td>
	<td rowspan="2">Within non coding gene</td>
</tr>
<tr>
	<td style="padding:0px;margin:0px;background-color:#32cd32"></td>	<td>nc_transcript_variant</td>
	<td>A transcript variant of a non coding RNA</td>
	<td><a rel="external" href="http://www.sequenceontology.org/miso/current_release/term/SO:0001619">SO:0001619</a></td>
</tr>
<tr class="bg2">
	<td style="padding:0px;margin:0px;background-color:#a2b5cd"></td>	<td>upstream_gene_variant</td>
	<td>A sequence variant located 5' of a gene</td>
	<td><a rel="external" href="http://www.sequenceontology.org/miso/current_release/term/SO:0001631">SO:0001631</a></td>
	<td>Upstream</td>
</tr>
<tr>
	<td style="padding:0px;margin:0px;background-color:#a2b5cd"></td>	<td>downstream_gene_variant</td>
	<td>A sequence variant located 3' of a gene</td>
	<td><a rel="external" href="http://www.sequenceontology.org/miso/current_release/term/SO:0001632">SO:0001632</a></td>
	<td>Downstream</td>
</tr>
<tr class="bg2">
    <td style="padding:0px;margin:0px;background-color:#a52a2a"></td>	<td>TFBS_ablation</td>
	<td>A feature ablation whereby the deleted region includes a transcription factor binding site</td>
	<td><a rel="external" href="http://www.sequenceontology.org/miso/current_release/term/SO:0001895">SO:0001895</a></td>
	<td>Tfbs ablation</td>
</tr>
<tr>
    <td style="padding:0px;margin:0px;background-color:#a52a2a"></td>	<td>TFBS_amplification</td>
	<td>A feature amplification of a region containing a transcription factor binding site</td>
	<td><a rel="external" href="http://www.sequenceontology.org/miso/current_release/term/SO:0001892">SO:0001892</a></td>
	<td>Tfbs amplification</td>
</tr>
<tr class="bg2">
    <td style="padding:0px;margin:0px;background-color:#a52a2a"></td>	<td>TF_binding_site_variant</td>
	<td>A sequence variant located within a transcription factor binding site</td>
	<td><a rel="external" href="http://www.sequenceontology.org/miso/current_release/term/SO:0001782">SO:0001782</a></td>
	<td rowspan="2">Regulatory region</td>
</tr>
<tr class="bg2">
    <td style="padding:0px;margin:0px;background-color:#a52a2a"></td>	<td>regulatory_region_variant</td>
	<td>A sequence variant located within a regulatory region</td>
	<td><a rel="external" href="http://www.sequenceontology.org/miso/current_release/term/SO:0001566">SO:0001566</a></td>
</tr>
<tr>
    <td style="padding:0px;margin:0px;background-color:#a52a2a"></td>	<td>regulatory_region_ablation</td>
	<td>A feature ablation whereby the deleted region includes a regulatory region</td>
	<td><a rel="external" href="http://www.sequenceontology.org/miso/current_release/term/SO:0001894">SO:0001894</a></td>
	<td>Regulatory region ablation</td>
</tr>
<tr class="bg2">
    <td style="padding:0px;margin:0px;background-color:#a52a2a"></td>	<td>regulatory_region_amplification</td>
	<td>A feature amplification of a region containing a regulatory region</td>
	<td><a rel="external" href="http://www.sequenceontology.org/miso/current_release/term/SO:0001891">SO:0001891</a></td>
	<td>Regulatory region amplification</td>
</tr>
<tr>
<td></td>	<td>feature_elongation</td>
	<td>A sequence variant that causes the extension of a genomic feature, with regard to the reference sequence</td>
	<td><a rel="external" href="http://www.sequenceontology.org/miso/current_release/term/SO:0001907">SO:0001907</a></td>
	<td>Feature elongation</td>
</tr>
<tr class="bg2">
<td></td>	<td>feature_truncation</td>
	<td>A sequence variant that causes the reduction of a genomic feature, with regard to the reference sequence</td>
	<td><a rel="external" href="http://www.sequenceontology.org/miso/current_release/term/SO:0001906">SO:0001906</a></td>
	<td>Feature truncation</td>
</tr>
<tr>
	<td style="padding:0px;margin:0px;background-color:#636363"></td>	<td>intergenic_variant</td>
	<td>A sequence variant located in the intergenic region, between genes</td>
	<td><a rel="external" href="http://www.sequenceontology.org/miso/current_release/term/SO:0001628">SO:0001628</a></td>
	<td>Intergenic</td>
</tr>
</table>
<p>* Corresponding colours for the Ensembl web displays.<p>



<br />
<h2 id="nsSNP">Protein function predictions</h2>

<p>
For human mutations that are predicted to result in an amino acid substitution we also 
provide SIFT and PolyPhen predictions for the effect of this substitution on
protein function. We compute the predictions for each of these tools for all possible 
single amino acid substitutions in the Ensembl human proteome. This means we can provide 
predictions for novel mutations for VEP and API users. We were able to compute predictions
from at least one tool for over 95% of the proteins in Ensembl. We also use the Condel tool
to provide a consensus prediction based on the SIFT and PolyPhen prediction scores.
</p>

<p>
These tools are developed by external groups and we provide a brief explanation of
the approach each takes below, and how we run it in Ensembl. For much more detail
please see the representative papers listed <a href="#references">below</a>, and the 
relevant publications available on each tool's website. We hope to be able to provide 
protein function predictions for species other than human in future releases.
</p>

<div style="border:1px solid #CCC;padding:4px">
<h3 id="sift">SIFT</h3>

<p>
<a rel="external" href="http://sift.bii.a-star.edu.sg">SIFT</a> predicts whether an amino acid 
substitution is likely to affect protein function based on sequence homology and the
physico-chemical similarity between the alternate amino acids. The data we provide for each amino
acid substitution is a score and a qualitative prediction (either 'tolerated' or 'deleterious'). 
The score is the normalized probability that the amino acid change is 
tolerated so scores nearer 0 are more likely to be deleterious.
The qualitative prediction is derived from this score such that substitutions with 
a score &lt; 0.05 are called 'deleterious' and all others are called 'tolerated'.
</p>

<p>
We ran <b>SIFT version 4.0.5</b> following the instructions from the authors and used SIFT
to choose homologous proteins rather than supplying them ourselves. We used all 
protein sequences available from UniProtKB (release 2012_01, both the SwissProt and TrEMBL sets) as 
the protein database. 
</p>
</div>
<br />

<div style="border:1px solid #CCC;padding:4px">
<h3 id="polyphen">PolyPhen</h3>

<p>
<a rel="external" href="http://genetics.bwh.harvard.edu/pph2/">PolyPhen-2</a> predicts
the effect of an amino acid substitution on the structure and function of a protein
using sequence homology, Pfam annotations, 3D structures from PDB where available, and
a number of other databases and tools (including DSSP, ncoils etc.). As with SIFT, for each amino
acid substitution where we have been able to calculate a prediction, we provide both a
qualitative prediction (one of 'probably damaging', 'possibly damaging', 'benign'
or 'unknown') and a score. The PolyPhen score represents the probability that a substitution 
is damaging, so values nearer 1 are more confidently predicted to be deleterious (note 
that this the opposite to SIFT). The qualitative prediction is based on the False 
Positive Rate of the classifier model used to make the predictions.
</p>

<p>
We ran <b>PolyPhen-2 version 2.2.2</b> (available 
<a rel="external" href="http://genetics.bwh.harvard.edu/pph2/dokuwiki/downloads">here</a>)
and again we followed all instructions from the authors, and used the UniProtKB
UniRef100 (release 2011_12) non-redundant protein set as the protein database, which was downloaded,
along with PDB structures, and annotations from Pfam and DSSP(snapshot 03-Jan-2012) in February 2012. When 
computing the predictions we store results for the classifier models trained on the HumDiv 
and HumVar datasets. Both result sets are available through the variation API which defaults to HumVar if no selection is made.</a>
 (Please refer to the PolyPhen  <a rel="external" href="http://genetics.bwh.harvard.edu/pph2/dokuwiki/overview">website</a> or publications for more details of the classification system).
<!-- HumDiv aims to discriminate variants that alter protein function, whereas HumVar aims to discriminate functions that cause human disease.-->
</p>
</div>
<br />

<div style="border:1px solid #CCC;padding:4px">
<h3 id="condel">Condel</h3>

<p>
<a rel="external" href="http://bg.upf.edu/group/projects/condel.php">Condel</a> is
a general method for calculating a consensus prediction from the output of tools
designed to predict the effect of an amino acid substitution. It does so by calculating
a weighted average score (WAS) of the scores of each component method. The Condel authors 
provided us with a version specialised for finding a consensus between SIFT and 
PolyPhen and we integrated this into a Variation Effect Predictor (VEP) 
<a rel="external" href="https://github.com/ensembl-variation/VEP_plugins">plugin</a>. 
Tests run by the authors 
on the HumVar dataset (a test set curated by the PolyPhen team), show that Condel 
can improve both the sensitivity and specificity of predictions compared to either 
SIFT or PolyPhen used alone (please contact the authors for details). The Condel 
score, along with a qualitative prediction (one of 'neutral' or 'deleterious'), 
are available in the VEP plugin. The Condel score is the consensus probability 
that a substitution is deleterious, so values nearer 1 are predicted with greater 
confidence to affect protein function.
</p>
</div>
<br />

<h4 id="nsSNP_data_format">Prediction data format</h4>

<p>
The SIFT and PolyPhen predictions are precomputed and stored in the variation
databases and predictions are accessible in the variation API by using the <i>sift_prediction</i>, 
<i>sift_score</i>, <i>polyphen_prediction</i> and <i>polyphen_score</i> methods on a 
<i>Bio::EnsEMBL::Variation::TranscriptVariationAllele</i> object. For users wanting
to access the complete set of predictions from the MySQL database or API, an explanation of
the format used is provided here.
</p>


<p>
The predictions and associated scores are stored as a matrix, with a column for 
each possible alternate amino acid and a row for each position in the translation. Each prediction
for a position and amino acid is stored as a 2-byte value which encodes both the qualitative
prediction and score encoded as described below. A simple example matrix is shown in the figure 
below, though note we only show the decoded score rather than the actual binary value stored
in the database.
</p>

<p>
<a rel="external" href="/img/protein_function_prediction_matrix.png">
<img src="/img/protein_function_prediction_matrix.png" style="width:500px;heigth:143px" alt="protein function encoding"/>
</a>
</p>

<p>
Prediction matrices can be fetched and manipulated in a user-friendly manner using the variation API, specifically
using the <i>ProteinFunctionPredictionMatrixAdaptor</i> which allows you to fetch a prediction matrix
using either a transcript or a translation stable ID. This adaptor returns a <i>ProteinFunctionPredictionMatrix</i>
object and you can use the <i>get_prediction</i> method to retrieve a prediction for a given position
and amino acid. Users who want to use entire matrices should use the <i>deserialize</i> method to decode
an entire binary formatted matrix into a simple Perl hash. Please refer to the <a
href="/info/docs/api/variation/index.html">API documentation</a> for both 
of these classes for more details. For users who require direct access to the MySQL database (for instance 
because they are accessing the database in a language other than Perl) we provide a description of the 
binary format used below.
</p>

<p>
Prediction matrices for each translation from each of SIFT and PolyPhen are stored in the 
<i>protein_function_predictions</i> table. The analysis used to calculate the predictions
is identified in the <i>analysis_attrib_id</i> column which refers to a value stored in the
<i>attrib</i> table, and the protein sequence the predictions
apply to is identified by the <i>translation_md5_id</i> column which refers to a hexadecimal 
MD5 hash of the sequence stored in the <i>translation_md5</i> table. The 
prediction matrices are stored in the <i>prediction_matrix</i> column as a gzip compressed binary string.
Once uncompressed, this string contains a 40 byte substring for each row in the matrix
concatenated together in position order. Each row is composed of 20 2-byte predictions, 
one for each possible alternative amino acid in alphabetical order, though note that the 
value for the amino acid that matches the reference amino acid is 
identified as a 2-byte value with all bits set, or <i>0xFFFF</i> in hexadecimal notation. 
The prediction itself is packed as a 16 bit little-endian 
("VAX" order, or "v" format if using the perl <i>pack</i> subroutine) unsigned short value. 
The top 2 bits of this short 
are used to encode the qualitative prediction (PolyPhen has 4 possible values, while SIFT has just
2), and the bottom 10 bits are used to encode the prediction score. To decode the qualitative 
prediction you should mask off all bits except the top 2, and shift the resulting short
right by 14 bits and treat this as an integer between 0 and 3. The corresponding prediction can then 
be looked up in the table below. To decode the prediction score you should mask off the 
top 6 bits and the resulting value can be treated as a number between 0 and 1000, which 
should be divided by 1000 to give a 3 decimal place score (casting to a floating point type
if necessary). Bits 11-14 are not used, except to encode the "same as reference" dummy prediction 
<i>0xFFFF</i>.
</p>

<img src="/img/protein_function_encoding.png" style="width:500px;heigth:143px" alt="protein function encoding"/>

<p>
A diagram of the encoding scheme is shown above. In this example from a polyphen prediction, 
the top prediction bits are <i>0b01</i> which in decimal is the number 1, which corresponds to 
"possibly damaging" in the table below. The score bits are <i>0b1110001010</i> which in decimal 
is the number 906, which when divided by 1000, gives a score of 0.906.
</p>

<table id="prediction_values" class="ss" style="width:50%;">
<tr>
	<th>Tool</th>
	<th>Numerical value</th>
	<th>Qualitative prediction</th>
</tr>
<tr>
	<td>PolyPhen</td>
	<td>0</td>
	<td>"probably damaging"</td>
</tr>
<tr class="bg2">
	<td>PolyPhen</td>
	<td>1</td>
	<td>"possibly damaging"</td>
</tr>
<tr>
	<td>PolyPhen</td>
	<td>2</td>
	<td>"benign"</td>
</tr>
<tr class="bg2">
	<td>PolyPhen</td>
	<td>3</td>
	<td>"unknown"</td>
</tr>
<tr>
  <td>SIFT</td>
	<td>0</td>
	<td>"tolerated"</td>
</tr>
<tr class="bg2">
	<td>SIFT</td>
	<td>1</td>
	<td>"deleterious"</td>
</tr>
</table>

<p>
To retrieve a prediction for a particular amino acid substitution at a given position
in a translation, first you must retrieve the binary matrix from the database and
uncompress it using gzip. You can calculate the offset into this string by multiplying 
the desired position (starting at 0) by 20 and then adding the index of the desired amino acid in an alphabetical ordering of
amino acids (also starting at 0), and then multiply this value by 2 to take into account the fact that each 
prediction uses 2 bytes. Each matrix also includes a 3 byte header used check if the data is corrupted etc. so
you will also need to add 3 to the calculated offset. The 2 bytes at the calculated position 
can then be extracted and the approach described above can be used to retrieve the prediction and score. 
A perl implementation of this scheme can be found in the <i>Bio::EnsEMBL::Variation::ProteinFunctionPredictionMatrix</i>
module in the variation API and should hopefully inform attempts to reimplement this 
scheme in other languages.
</p>

<p>
Condel predictions are very fast to compute and so are not precomputed and stored in the
database, instead we use the <i>get_condel_prediction</i> subroutine provided in the
<i>Bio::EnsEMBL::Variation::Utils::Condel</i> module to calculate the Condel prediction from
a precomputed SIFT and PolyPhen score on the fly.
</p>


<br />
<h2 id="references">References</h2>

<ul>
	<li>
		<p>Adzhubei IA, Schmidt S, Peshkin L, Ramensky VE, Gerasimova A, Bork P, Kondrashov AS, Sunyaev SR. <br />
		<strong>A method and server for predicting damaging missense mutations</strong><br />
		<em>Nature Methods</em> 7(4):248-249 (2010)<br />
		<a href="http://dx.doi.org/10.1038/nmeth0410-248" rel="external">doi:10.1038/nmeth0410-248</a></p> 
	</li>

	<li>
		<p>Kumar P, Henikoff S, Ng PC.<br />
		<strong>Predicting the effects of coding non-synonymous variants on protein function using the SIFT algorithm</strong><br />
		<em>Nature Protocols</em> 4(8):1073-1081 (2009)<br />
		<a href="http://dx.doi.org/10.1038/nprot.2009.86" rel="external">doi:10.1038/nprot.2009.86</a></p> 
	</li>

	<li>
		<p>Gonzalez-Perez A, Lopez-Bigas N.<br />
		<strong>Improving the assessment of the outcome of non-synonymous SNVs with a Consensus deleteriousness score (Condel)</strong><br />
		<em>Am J Hum Genet</em> 88(4):440-449 (2011)<br />
		<a href="http://dx.doi.org/10.1016/j.ajhg.2011.03.004" rel="external">doi:10.1016/j.ajhg.2011.03.004</a></p> 
	</li>
</ul>

</body>
</html>
