<html>
<head>
<title>Data formats</title>
<meta name="order" content="3" />
</head>

<body>
    
<style>
tr:nth-child(odd) {background-color: #f0f0f0;}
</style>

  <img src="/img/vep_logo.png" class="float-right" style="width:111px;height:50px" />	
	<h1 id="top">Variant Effect Predictor data formats</h1>

    <h2 id="input">Input</h2>
    
    <p> Both the web and script version of the VEP can use the same input
    formats. Formats can be auto-detected by the VEP script, but must be
    manually selected when using the web interface. The VEP can use <a
    href="#vcf">VCF</a>, <a href="#pileup">pileup</a> and <a href="#hgvs">HGVS
    notations</a> in addition to the <a href="#default">default</a> format</p>
    
    <h2 id="default">Default</h2>
    
    <p>The default format is a simple <strong>whitespace-separated</strong> format
    (columns may be separated by space or tab characters),
    containing five required columns plus an optional identifier column:
    </p>
    
    <ol>
        <li><strong>chromosome</strong> - just the name or number, with no 'chr'
        prefix</li>
        <li><strong>start</strong></li>
        <li><strong>end</strong></li>
        <li><strong>allele</strong> - pair of alleles separated by a '/', with the
        reference allele first</li>
        <li><strong>strand</strong> - defined as + (forward) or - (reverse).</li>
        <li><strong>identifier</strong> - this identifier will be used in the VEP's
        output. If not provided, the VEP will construct an identifier from the given
        coordinates and alleles.</li>
    </ol>
    
    <pre class="code">
1   881907    881906    -/C   +
5   140532    140532    T/C   +
12  1017956   1017956   T/A   +
2   946507    946507    G/C   +
14  19584687  19584687  C/T   -
19  66520     66520     G/A   +    var1
8   150029    150029    A/T   +    var2</pre>
    
    <p>An insertion (of any size) is indicated by start coordinate = end coordinate
    + 1. For example, an insertion of 'C' between nucleotides 12600 and 12601 on the
    forward strand of chromosome 8 is indicated as follows:</p>
    
    <pre class="code">
8   12601     12600     -/C   +</pre>
    <p>A deletion is indicated by the exact nucleotide coordinates. For example, a
    three base pair deletion of nucleotides 12600, 12601, and 12602 of the reverse
    strand of chromosome 8 will be:</p>
    
    <pre class="code">
8   12600     12602     CGT/- -</pre>
    
    
    <h2 id="vcf">VCF</h2>
    
    <p> The VEP also supports using <a
    rel="external" href="http://www.1000genomes.org/wiki/Analysis/vcf4.0">VCF (Variant Call Format)
    version 4.0</a>. This is a common format used by the 1000 genomes project, and
    can be produced as an output format by many variant calling tools. </p>
    
    <p> Users using VCF should note a peculiarity in the difference between how
    Ensembl and VCF describe unbalanced variants. For any unbalanced variant (i.e.
    insertion, deletion or unbalanced substitution), the VCF specification requires
    that the base immediately before the variant should be included in both the
    reference and variant alleles. This also affects the reported position i.e. the
    reported position will be one base before the actual site of the variant. </p>
    
    <p> In order to parse this correctly, the VEP needs to convert such variants
    into Ensembl-type coordinates, and it does this by removing the additional base
    and adjusting the coordinates accordingly. This means that if an identifier is
    not supplied for a variant (in the 3rd column of the VCF), then the identifier
    constructed and the position reported in the VEP's output file will differ from
    the input. </p>
    
    <p> This problem can be overcome by either: </p>
    <ol>
      <li> ensuring each variant has a unique
    identifier specified in the 3rd column of the VCF </li>
      <li> using VCF format as output (<a href="script/vep_options.html#opt_vcf">--vcf</a>) - this preserves the formatting of your input coordinates and alleles </li>
    </ol>
    
    <p> The following examples illustrate how VCF describes a variant and how it is
    handled internally by the VEP. Consider the following aligned sequences (for the
    purposes of discussion on chromosome 20): </p>
    
    <pre class="code">Ref: a t C g a // C is the reference base
1 : a t G g a // C base is a G in individual 1
2 : a t - g a // C base is deleted w.r.t. the reference in individual 2
3 : a t CAg a // A base is inserted w.r.t. the reference sequence in individual 3</pre>
    
    <h4> Individual 1 </h4>
    
    <p> The first individual shows a simple balanced substitution of G for C at base
    3. This is described in a compatible manner in VCF and Ensembl styles. Firstly,
    in VCF: </p>
    
    <pre class="code">20   3   .   C   G   .   PASS   .</pre>
    
    <p> And in Ensembl format: </p>
    
    <pre class="code"> 20   3   3   C/G   +</pre>
    
    <h4> Individual 2 </h4>
    
    <p> The second individual has the 3rd base deleted relative to the reference. In
    VCF, both the reference and variant allele columns must include the preceding
    base (T) and the reported position is that of the preceding base: </p>
    
    <pre class="code">20   2   .   TC   T   .   PASS   .</pre>
    
    <p> In Ensembl format, the preceding base is not included, and the start/end
    coordinates represent the region of the sequence deleted. A "-" character is
    used to indicate that the base is deleted in the variant sequence: </p>
    
    <pre class="code">20   3   3   C/-   +</pre>
    
    <p> The upshot of this is that while in the VCF input file the position of the
    variant is reported as 2, in the output file from the VEP the position will be
    reported as 3. If no identifier is provided in the third column of the VCF, then
    the constructed identifier will be: </p>
    
    <pre class="code">20_3_C/-</pre>
    
    <h4> Individual 3 </h4>
    
    <p> The third individual has an "A" inserted between the 3rd and 4th bases of
    the sequence relative to the reference. In VCF, as for the deletion, the base
    before the insertion is included in both the reference and variant allele
    columns, and the reported position is that of the preceding base: </p>
    
    <pre class="code">20   3   .   C   CA   .   PASS   .</pre>
    
    <p> In Ensembl format, again the preceding base is not included, and the
    start/end positions are "swapped" to indicate that this is an insertion.
    Similarly to a deletion, a "-" is used to indicate no sequence in the reference:
    </p>
    
    <pre class="code"> 20   4   3   -/A   +</pre>
    
    <p> Again, the output will appear different, and the constructed identifier may
    not be what is expected: </p>
    
    <pre class="code">20_3_-/A</pre>
    
    <p> The solution is to always add a unique identifer for each of your variants
    to the VCF file, or use VCF as your output format.</p>
    
    <h2 id="sv">Structural variants</h2>
    
    <p> The VEP can also call consequences on structural variants encoded in
    tab-delimited or VCF format. To recognise a variant as a structural variant, the
    allele string (or "SVTYPE" INFO field in VCF) must be set to one of the
    currently recognised values: </p>
    
    <ul>
        <li><b>INS</b> - insertion</li>
        <li><b>DEL</b> - deletion</li>
        <li><b>DUP</b> - duplication</li>
        <li><b>TDUP</b> - tandem duplication</li>
    </ul>
    
    <p> Examples of structural variants encoded in tab-delimited format: </p>
    
    <pre class="code">
1    160283    471362    DUP
1    1385015   1387562   DEL</pre>
    
    <p> Examples of structural variants encoded in VCF format: </p>
    
    <pre class="code">
#CHROM  POS     ID   REF  ALT    QUAL  FILTER  INFO                    FORMAT
1       160283  sv1  .    &lt;DUP&gt;  .     .       SVTYPE=DUP;END=471362   .
1       1385015 sv2  .    &lt;DEL&gt;  .     .       SVTYPE=DEL;END=1387562  .</pre>
    
    <p> See the <a
    rel="external" href="http://www.1000genomes.org/wiki/Analysis/Variant%20Call%20Format/VCF%20%28Variant%20Call%20Format%29%20version%204.0/encoding-structural-variants">VCF
    definition document</a> for more detail on how to describe
    structural variants in VCF format. </p>
    
    <h2 id="pileup">Pileup</h2>
    
    <p> The pileup format can also be used as input for the VEP. This is the output
    of the ssaha pileup package. </p>
    
    <h2 id="hgvs">HGVS identifiers</h2>
    
    <p> See <a
    rel="external" href="http://www.hgvs.org/mutnomen/">http://www.hgvs.org/mutnomen/</a> for
    details. These must be relative to genomic or Ensembl transcript coordinates. It
    also is possible to use RefSeq transcripts in both the web
    interface and the VEP script (see <a
    href="script/vep_other.html#hgvs">script documentation</a>).
     This works for RefSeq transcripts that align to the genome correctly. </p>
    
    <p> Examples: </p>
    
    <pre class="code">ENST00000207771.3:c.344+626A>T
ENST00000471631.1:c.28_33delTCGCGG
ENST00000285667.3:c.1047_1048insC
5:g.140532T>C</pre>
    
    <p> Examples using RefSeq identifiers (using <a
    href="script/vep_options.html#opt_refseq">--refseq</a> in the VEP script, or select
    the otherfeatures transcript database on the web interface and input type of
    HGVS): </p>
    
    <pre class="code">NM_153681.2:c.7C>T
NM_005239.4:c.190G>A
NM_001025204.1:c.336G>A</pre>
    
    <p> HGVS protein notations may also be used, provided that they
    unambiguously map to a single genomic change. Due to redundancy in the amino
    acid code, it is not always possible to work out the corresponding genomic
    sequence change for a given protein sequence change. The following example
    is for a permissable protein notation in dog <i>(Canis familiaris)</i>:</p>
    
    <pre class="code">ENSCAFP00000040171.1:p.Thr92Asn</pre>
    
    <p> HGVS notations may also be given in <a href="http://www.lrg-sequence.org/LRG"
    rel="external">LRG</a> coordinates: </p>
    
    <pre class="code">LRG_1t1:c.841G>T
LRG_1:g.10006G>T</pre>
    
    <h2 id="id">Variant identifiers</h2>
    
    <p> These should be e.g. dbSNP rsIDs, or any synonym for a variant present in
    the Ensembl Variation database. See <a
    href="/info/genome/variation/sources_documentation.html">here</a> for a list of
    identifier sources in Ensembl.</p>
    
    <hr/>
    <h2 id="output">Output</h2>
    
    <p> The default output format ("VEP" format when downloading from the web interface) is a 14 column tab-delimited file. The output
    columns are:</p>
    
    <ol>
        <li><strong>Uploaded variation</strong> - as chromosome_start_alleles</li>
        <li><strong>Location</strong> - in standard coordinate format (chr:start or chr:start-end)</li>
        <li><strong>Allele</strong> - the variant allele used to calculate the consequence</li>
        <li><strong>Gene</strong> - Ensembl stable ID of affected gene</li>
        <li><strong>Feature</strong> - Ensembl stable ID of feature</li>
        <li><strong>Feature type</strong> - type of feature. Currently one of Transcript, RegulatoryFeature, MotifFeature.</li>
        <li><strong>Consequence</strong> - <a href="/info/genome/variation/predicted_data.html#consequences">consequence type</a> of this variant</li>
        <li><strong>Position in cDNA</strong> - relative position of base pair in cDNA sequence</li>
        <li><strong>Position in CDS</strong> - relative position of base pair in coding sequence</li>
        <li><strong>Position in protein</strong> - relative position of amino acid in protein</li>
        <li><strong>Amino acid change</strong> - only given if the variant affects the protein-coding sequence</li>
        <li><strong>Codon change</strong> - the alternative codons with the variant base in upper case</li>
        <li><strong>Co-located variation</strong> - known identifier of existing variant</li>
        <li><strong>Extra</strong> - this column contains extra information as key=value pairs separated by ";". The keys are as follows:
            <ul>
                <li><i>IMPACT</i> - the impact modifier for the consequence type</li>
                <li><i>VARIANT_CLASS</i> - Sequence Ontology <a href="/info/genome/variation/data_description.html#classes">variant class</a></li>
                <li><i>SYMBOL</i> - the gene symbol</li>
                <li><i>SYMBOL_SOURCE</i> - the source of the gene symbol</li>
                <li><i>STRAND</i> - the DNA strand (1 or -1) on which the transcript/feature lies</li>
                <li><i>ENSP</i> - the Ensembl protein identifier of the affected transcript</li>
                <li>
                    <i>FLAGS</i> - transcript quality flags:
                    <ul>
                        <li><b>cds_start_NF:</b> CDS 5' incomplete</li>
                        <li><b>cds_end_NF:</b> CDS 3' incomplete</li>
                    </ul>
                </li>
                <li><i>SWISSPROT</i> - UniProtKB/Swiss-Prot accession of protein product
                <li><i>TREMBL</i> - UniProtKB/TrEMBL accession of protein product</li>
                <li><i>UNIPARC</i> - UniParc accession of protein product</li>
                <li><i>HGVSc</i> - the HGVS coding sequence name</li>
                <li><i>HGVSp</i> - the HGVS protein sequence name</li>
                <li><i>HGVS_OFFSET</i> - Indicates by how many bases the HGVS notations for this variant have been <a href="script/vep_options.html#opt_shift_hgvs">shifted</a></li>
                <li><i>SIFT</i> - the SIFT prediction and/or score, with both given as prediction(score)</li>
                <li><i>PolyPhen</i> - the PolyPhen prediction and/or score</li>
                <li><i>MOTIF_NAME</i> - the source and identifier of a transcription factor binding profile aligned at this position</li>
                <li><i>MOTIF_POS</i> - The relative position of the variation in the aligned TFBP</li>
                <li><i>HIGH_INF_POS</i> - a flag indicating if the variant falls in a high information position of a transcription factor binding profile (TFBP)</li>
                <li><i>MOTIF_SCORE_CHANGE</i> - The difference in motif score of the reference and variant sequences for the TFBP</li>
                <li><i>CELL_TYPE</i> - List of cell types and classifications for regulatory feature</li>
                <li><i>CANONICAL</i> - a flag indicating if the transcript is denoted as the canonical transcript for this gene</li>
                <li><i>CCDS</i> - the CCDS identifer for this transcript, where applicable</li>
                <li><i>INTRON</i> - the intron number (out of total number)</li>
                <li><i>EXON</i> - the exon number (out of total number)</li>
                <li><i>DOMAINS</i> - the source and identifer of any overlapping protein domains</li>
                <li><i>DISTANCE</i> - Shortest distance from variant to transcript</li>
                <li><i>IND</i> - individual name</li>
                <li><i>ZYG</i> - zygosity of individual genotype at this locus</li>
                <li><i>SV</i> - IDs of overlapping structural variants</li>
                <li><i>FREQS</i> - Frequencies of overlapping variants used in filtering</li>
                <li><i>GMAF</i> - Non-reference allele and frequency of existing variant in 1000 Genomes</li>
                <li><i>AFR_MAF</i> - Non-reference allele and frequency of existing variant in 1000 Genomes combined African population</li>
                <li><i>AMR_MAF</i> - Non-reference allele and frequency of existing variant in 1000 Genomes combined American population</li>
                <li><i>ASN_MAF</i> - Non-reference allele and frequency of existing variant in 1000 Genomes combined Asian population</li>
                <li><i>EUR_MAF</i> - Non-reference allele and frequency of existing variant in 1000 Genomes combined European population</li>
                <li><i>EAS_MAF</i> - Non-reference allele and frequency of existing variant in 1000 Genomes combined East Asian population</li>
                <li><i>SAS_MAF</i> - Non-reference allele and frequency of existing variant in 1000 Genomes combined South Asian population</li>
                <li><i>AA_MAF</i> - Non-reference allele and frequency of existing variant in NHLBI-ESP African American population</li>
                <li><i>EA_MAF</i> - Non-reference allele and frequency of existing variant in NHLBI-ESP European American population</li>
                <li><i>ExAC_MAF</i> - Frequency of existing variant in ExAC combined population</li>
                <li><i>ExAC_Adj_MAF</i> - Adjusted frequency of existing variant in ExAC combined population</li>
                <li><i>ExAC_AFR_MAF</i> - Frequency of existing variant in ExAC African/American population</li>
                <li><i>ExAC_AMR_MAF</i> - Frequency of existing variant in ExAC American population</li>
                <li><i>ExAC_EAS_MAF</i> - Frequency of existing variant in ExAC East Asian population</li>
                <li><i>ExAC_FIN_MAF</i> - Frequency of existing variant in ExAC Finnish population</li>
                <li><i>ExAC_NFE_MAF</i> - Frequency of existing variant in ExAC Non-Finnish European population</li>
                <li><i>ExAC_OTH_MAF</i> - Frequency of existing variant in ExAC combined other combined populations</li>
                <li><i>ExAC_SAS_MAF</i> - Frequency of existing variant in ExAC South Asian population</li>
                <li><i>CLIN_SIG</i> - ClinVar clinical significance of the dbSNP variant</li>
                <li><i>BIOTYPE</i> - Biotype of transcript or regulatory feature</li>
                <li><i>TSL</i> - Transcript support level</li>
                <li><i>PUBMED</i> - Pubmed ID(s) of publications that cite existing variant</li>
                <li><i>SOMATIC</i> - Somatic status of existing variant(s)</li>
                <li><i>PHENO</i> - Indicates if existing variant is associated with a phenotype, disease or trait</li>
                <li><i>GENE_PHENO</i> - Indicates if overlapped gene is associated with a phenotype, disease or trait</li>
                <li><i>ALLELE_NUM</i> - Allele number from input; 0 is reference, 1 is first alternate etc</li>
                <li><i>MINIMISED</i> - Alleles in this variant have been converted to minimal representation before consequence calculation</li>
                <li><i>PICK</i> - indicates if this block of consequence data was picked by <a href="script/vep_options.html#opt_flag_pick">--flag_pick</a> or <a href="script/vep_options.html#opt_flag_pick_allele">--flag_pick_allele</a></li>

                <li id="refseq_match"><i>REFSEQ_MATCH</i> - the RefSeq transcript match status; contains a number of flags indicating whether this RefSeq transcript matches the underlying reference sequence and/or an Ensembl transcript:
                  <ul>
                    <li><b>rseq_3p_mismatch:</b> signifies a mismatch between the underlying genomic sequence of the RefSeq transcript with the corresponding RefSeq mRNA sequence the model was built from. Specifically, there is a mismatch in the 3' UTR of the RefSeq model. </li>
                    <li><b>rseq_5p_mismatch:</b> signifies a mismatch between the underlying genomic sequence of the RefSeq transcript with the corresponding RefSeq mRNA sequence the model was built from. Specifically, there is a mismatch in the 5' UTR of the RefSeq model. </li>
                    <li><b>rseq_cds_mismatch:</b> signifies a mismatch between the underlying genomic sequence of the RefSeq transcript with the corresponding RefSeq mRNA sequence the model was built from. Specifically, there is a mismatch in the CDS of the RefSeq model. </li>
                    <li><b>rseq_ens_match_cds:</b> signifies that for the RefSeq transcript there is an overlapping Ensembl model that is identical across the CDS region only. A CDS match is defined as follows: the CDS and peptide sequences are identical and the genomic coordinates of every translatable exon match. Useful related attributes are: rseq_ens_match_wt and rseq_ens_no_match.</li>
                    <li><b>rseq_ens_match_wt:</b> signifies that for the RefSeq transcript there is an overlapping Ensembl model that is identical across the whole transcript. A whole transcript match is defined as follows: 1) In the case that both models are coding, the transcript, CDS and peptide sequences are all identical and the genomic coordinates of every exon match. 2) In the case that both transcripts are non-coding the transcript sequences and the genomic coordinates of every exon are identical. No comparison is made between a coding and a non-coding transcript. Useful related attributes are: rseq_ens_match_cds and rseq_ens_no_match.</li>
                    <li><b>rseq_ens_no_match:</b> signifies that for the RefSeq transcript there is no overlapping Ensembl model that is identical across either the whole transcript or the CDS. This is caused by differences between the transcript, CDS or peptide sequences or between the exon genomic coordinates. Useful related attributes are: rseq_ens_match_wt and rseq_ens_match_cds.</li>
                    <li><b>rseq_mrna_match:</b> signifies an exact match between the underlying genomic sequence of the RefSeq transcript with the corresponding RefSeq mRNA sequence the model was built from (based on a match between the transcript stable id and an accession in the RefSeq mRNA file). An exact match occurs when the underlying genomic sequence of the model can be perfectly aligned to the mRNA sequence post polyA clipping.</li>
                    <li><b>rseq_mrna_nonmatch:</b> signifies a non-match between the underlying genomic sequence of the RefSeq transcript with the corresponding RefSeq mRNA sequence the model was built from. A non-match is deemed to have occurred if the underlying genomic sequence does not have a perfect alignment to the mRNA sequence post polyA clipping. It can also signify that no comparison was possible as the model stable id may not have had a corresponding entry in the RefSeq mRNA file (sometimes happens when accessions are retired or changed). When a non-match occurs one or several of the following transcript attributes will also be present to provide more detail on the nature of the non-match: rseq_5p_mismatch, rseq_cds_mismatch, rseq_3p_mismatch, rseq_nctran_mismatch, rseq_no_comparison</li>
                    <li><b>rseq_nctran_mismatch:</b> signifies a mismatch between the underlying genomic sequence of the RefSeq transcript with the corresponding RefSeq mRNA sequence the model was built from. This is a comparison between the entire underlying genomic sequence of the RefSeq model to the mRNA in the case of RefSeq models that are non-coding. </li>
                    <li><b>rseq_no_comparison:</b> signifies that no alignment was carried out between the underlying genomic sequence of RefSeq model and a corresponding RefSeq mRNA. The reason for this is generally that no corresponding, unversioned accession was found in the RefSeq mRNA file for the transcript stable id. This sometimes happens when accessions are retired or replaced. A second possibility is that the sequences were too long and problematic to align (though this is rare).</li>
                  </ul>
                </li>
            </ul>
        </li>
    </ol>
    
    <p>Empty values are denoted by '-'. Further fields in the Extra column can be
    added by <a href="script/vep_plugins.html">plugins</a> or using <a
    href="script/vep_custom.html">custom annotations</a> in the VEP script. Output fields
    can be configured using the <a href="script/vep_options.html#opt_fields">--fields</a> flag when running the VEP script.
    </p>
    
    <pre class="code">
11_224088_C/A    11:224088   A  ENSG00000142082  ENST00000525319  Transcript         missense_variant           742  716  239  T/N  aCc/aAc  -  SIFT=deleterious(0);PolyPhen=unknown(0)
11_224088_C/A    11:224088   A  ENSG00000142082  ENST00000534381  Transcript         5_prime_UTR_variant        -    -    -    -    -        -  -
11_224088_C/A    11:224088   A  ENSG00000142082  ENST00000529055  Transcript         downstream_variant         -    -    -    -    -        -  -
11_224585_G/A    11:224585   A  ENSG00000142082  ENST00000529937  Transcript         intron_variant             -    -    -    -    -        -  HGVSc=ENST00000529937.1:c.136-346G>A
22_16084370_G/A  22:16084370 A  -                ENSR00000615113  RegulatoryFeature  regulatory_region_variant  -    -    -    -    -        -  -</pre>
    
    <p> The VEP script will also add a header to the output file. This contains
    information about the databases connected to, and also a key describing the
    key/value pairs used in the extra column. </p>
    
    <pre class="code">
## ENSEMBL VARIANT EFFECT PREDICTOR v[[SPECIESDEFS::ENSEMBL_VERSION]]
## Output produced at 2015-04-01 16:09:38
## Connected to homo_sapiens_core_[[SPECIESDEFS::ENSEMBL_VERSION]]_38 on ensembldb.ensembl.org
## Using API version [[SPECIESDEFS::ENSEMBL_VERSION]], DB version [[SPECIESDEFS::ENSEMBL_VERSION]]
## sift version sift5.2.2
## COSMIC version 71
## ESP version 20140509
## gencode version GENCODE 22
## HGMD-PUBLIC version 20142
## genebuild version 2014-07
## regbuild version 13.0
## assembly version GRCh38.p2
## dbSNP version 138
## ClinVar version 201410
## Extra column keys:
## IMPACT : Subjective impact classification
## DISTANCE : Shortest distance from variant to transcript
## STRAND : Strand of the feature (1/-1)</pre>
    
    <hr/>
    <h2 id="tab">Tab-delimited output</h2>

    <p> The <a href="script/vep_options.html#opt_tab">--tab</a> flag instructs the VEP
    to write output as a tab-delimited table. This differs from the default output format
    in that each individual field from the "Extra" field is written to a separate
    tab-delimited column. This makes the output more suitable for import into spreadsheet
    programs such as Excel. This is also the format used when selecting the "TXT" option
    on the VEP web interface. </p>

    <p> The choice and order of columns in the output may be configured using
    <a href="script/vep_options.html#opt_fields">--fields</a>. </p>
    
    <hr/>
    <h2 id="vcfout">VCF output</h2>
    
    <p> The VEP script can also generate VCF output using the <a href="script/vep_options.html#opt_vcf">--vcf</a> flag.
    Consequences are added in the INFO field of the VCF file, using the key
    "CSQ" (configure this using <a href="script/vep_options.html#opt_vcf_info_field">--vcf_info_field</a>). Data fields are encoded separated by "&#124;"; the order of fields is
    written in the VCF header. Output fields can be configured by using
    <a href="script/vep_options.html#opt_fields">--fields</a>. Unpopulated fields are
    represented by an empty string. </p>
    
    <p> VCFs produced by the VEP can be filtered by <a
    href="script/vep_filter.html">filter_vep.pl</a> in the same way as standard format
    output files. </p>
    
    <p> If the input format was VCF, the file will remain unchanged save for the
    addition of the CSQ field and the header (unless using any filtering). If an
    existing CSQ field is found, it will be replaced by the one added by the
    VEP (use <a href="script/vep_options.html#opt_keep_csq">--keep_csq</a> to
    preserve it). </p>
    
    <p> Custom data added with <a
    href="script/vep_options.html#opt_custom">--custom</a> are added as separate
    fields, using the key specified for each data file.</p>
    
    <p> Commas in fields are replaced with ampersands (&amp;) to preserve VCF
    format. </p>
    
    <pre class="code">##INFO=&lt;ID=CSQ,Number=.,Type=String,Description="Consequence annotations from Ensembl VEP. Format: Allele|Consequence|IMPACT|SYMBOL|Gene|Feature_type|Feature|BIOTYPE|EXON|INTRON|HGVSc|HGVSp|cDNA_position|CDS_position|Protein_position"&gt;
#CHROM  POS       ID          REF  ALT  QUAL  FILTER  INFO
21      26978790  rs75377686  T    C    .     .       CSQ=C|missense_variant|MODERATE|MRPL39|ENSG00000154719|Transcript|ENST00000419219|protein_coding|2/8||ENST00000419219.1:c.251A>G|ENSP00000404426.1:p.Asn84Ser|260|251|84</pre>
    
    
    <hr/>
    <h2 id="json">JSON output</h2>
    
    <p> The VEP can produce output in the form of serialised <a
    href="http://json.org/" rel="external">JSON</a> objects using the <a
    href="script/vep_options.html#opt_json">--json</a> flag. JSON is a
    serialisation format that can be parsed and processed easily by many
    packages and programming languages; it is used as the default output format
    for <a href="http://rest.ensembl.org" rel="external">Ensembl's REST
    server</a>. </p>
    
    <p> Each input variant is reported as a single JSON object which constitutes
    one line of the output file. The JSON object is structured somewhat
    differently to the other VEP output formats, in that per-variant fields
    (e.g. co-located existing variant details) are reported only once.
    Consequences are grouped under the feature type that they affect
    (Transcript, Regulatory Feature, etc). The original input line (e.g. from
    VCF input) is reported under the "input" key in order to aid aligning input
    with output.</p>
    
    <p> Here follows an example of JSON output (prettified and redacted for display here): </p>
    
    <pre class="code">{
  <span style="color:blue">"input"</span>: <span style="color:blue">"1 230845794 test1 A G . . ."</span>,
  <span style="color:blue">"id"</span>: <span style="color:blue">"test1"</span>,
  <span style="color:blue">"seq_region_name"</span>: <span style="color:blue">"1"</span>,
  <span style="color:blue">"start"</span>: <span style="color:green">230845794</span>,
  <span style="color:blue">"end"</span>: <span style="color:green">230845794</span>,
  <span style="color:blue">"strand"</span>: <span style="color:green">1</span>,
  <span style="color:blue">"allele_string"</span>: <span style="color:blue">"A/G"</span>,
  <span style="color:blue">"most_severe_consequence"</span>: <span style="color:blue">"missense_variant"</span>,
  <span style="color:blue">"colocated_variants"</span>: [
    {
      <span style="color:blue">"id"</span>: <span style="color:blue">"rs699"</span>,
      <span style="color:blue">"seq_region_name"</span>: <span style="color:blue">"1"</span>,
      <span style="color:blue">"start"</span>: <span style="color:green">230845794</span>,
      <span style="color:blue">"end"</span>: <span style="color:green">230845794</span>,
      <span style="color:blue">"strand"</span>: <span style="color:green">1</span>,
      <span style="color:blue">"allele_string"</span>: <span style="color:blue">"A/G"</span>,
      <span style="color:blue">"minor_allele"</span>: <span style="color:blue">"A"</span>,
      <span style="color:blue">"minor_allele_freq"</span>: <span style="color:green">0.3384</span>,
      <span style="color:blue">"afr_allele"</span>: <span style="color:blue">"A"</span>,
      <span style="color:blue">"afr_maf"</span>: <span style="color:green">0.13</span>,
      <span style="color:blue">"amr_allele"</span>: <span style="color:blue">"A"</span>,
      <span style="color:blue">"amr_maf"</span>: <span style="color:green">0.36</span>,
      <span style="color:blue">"asn_allele"</span>: <span style="color:blue">"A"</span>,
      <span style="color:blue">"asn_maf"</span>: <span style="color:green">0.16</span>,
      <span style="color:blue">"eur_allele"</span>: <span style="color:blue">"A"</span>,
      <span style="color:blue">"eur_maf"</span>: <span style="color:green">0.41</span>,
      <span style="color:blue">"pubmed"</span>: [
        <span style="color:green">18513389</span>,
        <span style="color:green">23716723</span>
      ]
    },
    {
      <span style="color:blue">"seq_region_name"</span>: <span style="color:blue">"1"</span>,
      <span style="color:blue">"strand"</span>: <span style="color:green">1</span>,
      <span style="color:blue">"id"</span>: <span style="color:blue">"COSM425562"</span>,
      <span style="color:blue">"allele_string"</span>: <span style="color:blue">"A/G"</span>,
      <span style="color:blue">"start"</span>: <span style="color:green">230845794</span>,
      <span style="color:blue">"end"</span>: <span style="color:green">230845794</span>
    }
  ],
  <span style="color:blue">"transcript_consequences"</span>: [
    {
      <span style="color:blue">"variant_allele"</span>: <span style="color:blue">"G"</span>,
      <span style="color:blue">"consequence_terms"</span>: [
        <span style="color:blue">"missense_variant"</span>
      ],
      <span style="color:blue">"gene_id"</span>: <span style="color:blue">"ENSG00000135744"</span>,
      <span style="color:blue">"gene_symbol"</span>: <span style="color:blue">"AGT"</span>,
      <span style="color:blue">"gene_symbol_source"</span>: <span style="color:blue">"HGNC"</span>,
      <span style="color:blue">"transcript_id"</span>: <span style="color:blue">"ENST00000366667"</span>,
      <span style="color:blue">"biotype"</span>: <span style="color:blue">"protein_coding"</span>,
      <span style="color:blue">"strand"</span>: <span style="color:green">-1</span>,
      <span style="color:blue">"cdna_start"</span>: <span style="color:green">1018</span>,
      <span style="color:blue">"cdna_end"</span>: <span style="color:green">1018</span>,
      <span style="color:blue">"cds_start"</span>: <span style="color:green">803</span>,
      <span style="color:blue">"cds_end"</span>: <span style="color:green">803</span>,
      <span style="color:blue">"protein_start"</span>: <span style="color:green">268</span>,
      <span style="color:blue">"protein_end"</span>: <span style="color:green">268</span>,
      <span style="color:blue">"codons"</span>: <span style="color:blue">"aTg/aCg"</span>,
      <span style="color:blue">"amino_acids"</span>: <span style="color:blue">"M/T"</span>,
      <span style="color:blue">"polyphen_prediction"</span>: <span style="color:blue">"benign"</span>,
      <span style="color:blue">"polyphen_score"</span>: <span style="color:green">0</span>,
      <span style="color:blue">"sift_prediction"</span>: <span style="color:blue">"tolerated"</span>,
      <span style="color:blue">"sift_score"</span>: <span style="color:green">1</span>,
      <span style="color:blue">"hgvsc"</span>: <span style="color:blue">"ENST00000366667.4:c.803T>C"</span>,
      <span style="color:blue">"hgvsp"</span>: <span style="color:blue">"ENSP00000355627.4:p.Met268Thr"</span>
    }
  ],
  <span style="color:blue">"regulatory_feature_consequences"</span>: [
    {
      <span style="color:blue">"variant_allele"</span>: <span style="color:blue">"G"</span>,
      <span style="color:blue">"consequence_terms"</span>: [
        <span style="color:blue">"regulatory_region_variant"</span>
      ],
      <span style="color:blue">"regulatory_feature_id"</span>: <span style="color:blue">"ENSR00001529861"
    }
  ]
}</pre>
    
    <p> In accordance with JSON conventions, all keys are lower-case. Some keys
    also have different names and structures to those found in the other VEP
    output formats: </p>
    
    <table class="ss">
      <tr>
        <th>Key</th>
        <th>JSON equivalent(s)</th>
        <th>Notes</th>
      </tr>
      <tr>
        <td>Consequence</td>
        <td>consequence_terms</td>
        <td></td>
      </tr>
      <tr>
        <td>Gene</td>
        <td>gene_id</td>
        <td></td>
      </tr>
      <tr>
        <td>Feature</td>
        <td>transcript_id, regulatory_feature_id, motif_feature_id</td>
        <td>Consequences are grouped under the feature type they affect</td>
      </tr>
      <tr>
        <td>ALLELE</td>
        <td>variant_allele</td>
        <td></td>
      </tr>
      <tr>
        <td>SYMBOL</td>
        <td>gene_symbol</td>
        <td></td>
      </tr>
      <tr>
        <td>SYMBOL_SOURCE</td>
        <td>gene_symbol_source</td>
        <td></td>
      </tr>
      <tr>
        <td>ENSP</td>
        <td>protein_id</td>
        <td></td>
      </tr>
      <tr>
        <td>OverlapBP</td>
        <td>bp_overlap</td>
        <td></td>
      </tr>
      <tr>
        <td>OverlapPC</td>
        <td>percentage_overlap</td>
        <td></td>
      </tr>
      <tr>
        <td>Uploaded_variation</td>
        <td>id</td>
        <td></td>
      </tr>
      <tr>
        <td>Location</td>
        <td>seq_region_name, start, end, strand</td>
        <td>The variant's location field is broken down into constituent coordinate parts for clarity. "seq_region_name" is used in place of "chr" or "chromosome" for consistency with other parts of Ensembl's REST API</td>
      </tr>
      <tr>
        <td>GMAF</td>
        <td>minor_allele, minor_allele_freq</td>
        <td></td>
      </tr>
      <tr>
        <td>*_maf</td>
        <td>*_allele, *_maf</td>
        <td></td>
      </tr>
      <tr>
        <td>cDNA_position</td>
        <td>cdna_start, cdna_end</td>
        <td></td>
      </tr>
      <tr>
        <td>CDS_position</td>
        <td>cds_start, cds_end</td>
        <td></td>
      </tr>
      <tr>
        <td>Protein_position</td>
        <td>protein_start, protein_end</td>
        <td></td>
      </tr>
      <tr>
        <td>SIFT</td>
        <td>sift_prediction, sift_score</td>
        <td></td>
      </tr>
      <tr>
        <td>PolyPhen</td>
        <td>polyphen_prediction, polyphen_score</td>
        <td></td>
      </tr>
    </table>
    
    <hr/>
    <h2 id="stats">Statistics</h2>
    
    <p> The VEP writes an HTML file containing statistics pertaining to the
    results of your job; it is named <b>[output_file]_summary.html</b> (with the
    default options the file will be named
    <b>variant_effect_output.txt_summary.html</b>). To view it you should open
    the file in your web browser. </p>
    
    <p> To prevent the VEP writing a stats file, use the flag <a
    href="script/vep_options.html#opt_no_stats">--no_stats</a>. To have the VEP write a
    machine-readable text file in place of the HTML, use <a
    href="script/vep_options.html#opt_stats_text">--stats_text</a>. To change the name
    of the stats file from the default, use <a
    href="script/vep_options.html#opt_stats_file">--stats_file [file]</a>. </p>
    
    <p> The page contains several sections: </p>
    
    <h4> General statistics </h4>
    
    <p> This section contains two tables. The first describes the cache and/or
    database used, the version of the VEP, species, command line parameters,
    input/output files and run time. The second table contains information about
    the number of variants, and the number of genes, transcripts and regulatory
    features overlapped by the input. </p>
    
    <h4> Charts and tables </h4>
    
    <p> There then follows several charts, most with accompanying tables. Tables
    and charts are interactive; clicking on a row to highlight it in the table
    will highlight the relevant segment in the chart, and vice versa. </p>
    
    <div class="tinted-box" style="width: 200px; padding-bottom: 5px; float: left; margin-right: 20px;">
        <a href="/img/vep_stats_1.png"><img style="width: 200px; border: 1px solid grey" src="/img/vep_stats_1.png">
        General statistics</a>
    </div>
    
    <div class="tinted-box" style="width: 200px; padding-bottom: 5px; float: left; margin-right: 20px;">
        <a href="/img/vep_stats_2.png"><img style="width: 200px; border: 1px solid grey" src="/img/vep_stats_2.png">
        Summary of called consequence types</a>
    </div>
    
    <div class="tinted-box" style="width: 200px; padding-bottom: 5px; float: left; margin-right: 20px;">
        <a href="/img/vep_stats_3.png"><img style="width: 200px; border: 1px solid grey" src="/img/vep_stats_3.png">
        Distribution of variants across chromosomes</a>
    </div>

</body>
</html>
