<html>
<head>
<title>Ensembl VEP FAQ</title>
<meta name="order" content="7" />
</head>
<body>
<div>
  <div style="float:right;"><img src="/img/vep_logo.png"/></div>
	
  <h1 id="top"><span style="color:#006;padding-right:15px">Ensembl Variant Effect Predictor</span><span style="color:#666"><img src="/i/16/help.png"/> FAQ</span></h1>
  <hr/>
    
    <p> For any questions not covered here, please send an email to the Ensembl
    <a href="/info/about/contact/index.html">developer's mailing list</a>
    (public) or contact the
    <a href="/Help/Contact" class="popup">Ensembl Helpdesk</a> (private).
    Also you can report issues through our (public) Github repositories. For general vep issues you should use 
    <a href="https://github.com/Ensembl/ensembl-vep/issues">ensembl-vep</a> repository and for specific plugins you should use 
    <a href="https://github.com/Ensembl/VEP_plugins/issues">VEP_plugins</a> repository.
    </p>

<h2 id="general">General questions</h2>

<h3 id="indel-vcf">Q: Why has my insertion/deletion variant encoded in VCF
disappeared from the output? </h3>

<p> Ensembl treats unbalanced variants differently to VCF - your variant
hasn't disappeared, it may have just changed slightly! You can solve this by
giving your variants a unique identifier in the third column of the VCF file.
See <a href="vep_formats.html#vcf">here</a> for a full discussion.</p>

<p>&nbsp;</p>

<h3 id="no-co-located-vars"> Q: Why don't I see any co-located variants when
using species X? </h3>

<p> Not all species have variants and not all species that do are in the Ensembl variation resource
- see <a href="/info/genome/variation/species/species_data_types.html#source">this
  document</a> for details.
The <a href="script/vep_options.html#opt_custom"> custom </a>option can be used in the commandline interface to include more variant sets</p>
</p>

<p>&nbsp;</p>

<h3 id="multi-vars-mapped"> Q: Why do I see multiple known variants mapped to my input variant? </h3>

<p> Ensembl VEP compares your input to known variants from the Ensembl variation database.
In some cases one input variant can match multiple known variants:
<ul>
  <li>Germline variants from dbSNP and somatic mutations from COSMIC may be found at the same locus</li>
  <li>Some sources, e.g. HGMD, do not provide public access to allele-specific data, so an HGMD variant with unknown alleles may colocate with one from dbSNP with known alleles</li>
  <li>Multiple alternate alleles from your input may match different variants as they are described in dbSNP</li>
</ul>

See <a href="script/vep_other.html#colocated">here</a> for a full discussion.</p>

<p>&nbsp;</p>

<h3 id="no-freq"> Q: Ensembl VEP is not assigning a frequency to my input variant - why? </h3>

<p> Ensembl VEP's cache contains frequency data only for specific studies. See <a href="script/vep_other.html#colocated">here</a> for a full discussion. 
The <a href="script/vep_options.html#opt_custom"> custom </a>option can be used in the commandline interface to include more frequency sets</p>

<p>&nbsp;</p>

<h3 id="many-lines"> Q: Why do I see so many lines of output for each variant in my input? </h3>

<p> While it would be convenient to have a simple, one word answer to the
question "What is the consequence of this variant?", in reality biology is not
 this simple! Many genes have more than one transcript, so Ensembl VEP
provides a prediction for each transcript that a variant overlaps. Ensembl VEP
has options to help select results according to your requirements; the <a
href="script/vep_options.html#opt_canonical">--canonical</a> and <a
href="script/vep_options.html#mane">--mane</a> options indicate which
transcripts are canonical and belong to the human MANE set respectively, while <a
href="script/vep_options.html#opt_pick">--pick</a>, <a
href="script/vep_options.html#opt_per_gene">--per_gene</a>, <a
href="script/vep_options.html#opt_summary">--summary</a> and <a
href="script/vep_options.html#opt_most_severe">--most_severe</a> allow you to
give a more summary level assessment per variant.</p>

<p> Furthermore, several "compound" consequences are also possible - if, for
example, a variant falls in the final few bases of an exon, it may be considered
to affect a splicing site, in addition to possibly affecting the coding
sequence. </p>


<p>&nbsp;</p>

<h3 id="reduce-mem"> Q: How do I reduce Ensembl VEP's memory requirement?</h3>

<p> There are a number of ways to do this-
<ol>
  <li>Ensure your input file is sorted by location. This can greatly reduce memory
      requirements and runtime </li>
  <li>Consider reducing the buffer size. This reduces the number of variants annotated
      together in a batch and can be modified in both command line and web interfaces.
      Reducing buffer size may increase run time.</li>
  <li>Ensure you are only using the options you need, rather than --everything.
      Some data-rich options, such as regulatory annotation have an impact on memory use</li>
</ol>
</p>

<p>&nbsp;</p>


<h3 id="cite-vep"> Q: How to cite Ensembl  VEP?</h3>

<p> If you use Ensembl VEP, please cite our
<a href="/info/docs/tools/vep/index.html#publication">latest publication</a>
to continue to support Ensembl VEP development. </p>

<hr/>

<h2 id="web">Ensembl VEP web interface questions</h2>

<h3 id="web-vep"> Q: How do I access the web version of the Ensembl Variant
Effect Predictor? </h3>

<p> You can find the Ensembl VEP web tool on the <a href="/info/docs/tools/index.html">Tools</a> page.</p>

<p>&nbsp;</p>


<h3 id="different-output"> Q: Why is the output I get for my input file
different when I use the Ensembl VEP web and command line interfaces? </h3>

<p> Ensure that you are passing equivalent arguments to the command line tool that you
are using in the web interface. If you are sure this is still a problem, please
report it on the <a href="http://lists.ensembl.org/mailman/listinfo/dev" rel="external">ensembl-dev</a> mailing list.</p>

<p>&nbsp;</p>


<h3 id="web-vep-tutorial"> Q: Is there a tutorial for the web tool?</h3>

<p>Yes, see our latest tutorial <a href="https://onlinelibrary.wiley.com/doi/10.1002/humu.24298" rel="external">
Annotating and prioritizing genomic variants using the Ensembl Variant Effect Predictor â€” A tutorial</a>
for more information on using the Ensembl VEP web interface.</p>
<hr/>

<h2 id="script">Ensembl VEP command line tool questions</h2>

<h3 id="faster-vep"> Q: How can I make Ensembl VEP run faster? </h3>

<p> There are a number of factors that influence how fast Ensembl VEP runs. Have a look at our <a href="script/vep_other.html#faster">handy guide</a> for tips on improving runtime. </p>

<p>&nbsp;</p>

<h3 id="different-input-var"> Q: Why am I not seeing the same variant from my input in the output? </h3>
<p> Since the Ensembl 110 release, Ensembl VEP by default will minimise the input allele for display in the output. To see the exact allele representation you provided, use the<a
href="script/vep_options.html#opt_uploaded_allele"> --uploaded_allele</a> option. </p>

<p>&nbsp;</p>

<h3 id="hgvs-n"> Q: Why do I see "N" as the reference allele in my HGVS strings? </h3>
<h3 id="sequence.pm"> Q: Why do I get errors related with Sequence.pm?</h3>

<pre class="code sh_sh">substr outside of string at /nfs/users/nfs_w/wm2/Perl/ensembl-variation/modules/Bio/EnsEMBL/Variation/Utils/Sequence.pm line 511.
Use of uninitialized value $ref_allele in string eq at /nfs/users/nfs_w/wm2/Perl/ensembl-variation/modules/Bio/EnsEMBL/Variation/Utils/Sequence.pm line 514.
Use of uninitialized value in concatenation (.) or string at /nfs/users/nfs_w/wm2/Perl/ensembl-variation/modules/Bio/EnsEMBL/Variation/Utils/Sequence.pm line 643.</pre>

<p> Both of these error types are usually seen when using a <a
href="script/vep_cache.html#fasta">FASTA file</a> for retrieving sequence.
There are a couple of steps you can take to try to remedy them:</p>

<ol>
  <li>
    The index alongside the FASTA can become corrupted. Delete
    [fastafile].index and re-run Ensembl VEP to regenerate it. By default
    this file is located in your $HOME/.vep/[species]/[version]_[assembly]
    directory.
  </li>

  <li>
    The FASTA file itself may have been corrupted during download;
    delete the fasta file and the index and re-download (you can use the
    <a href="script/vep_download.html#install">Ensembl VEP installer</a> to do this).
  </li>

  <li>
    Older versions of BioPerl (1.2.3 in particular is known to have this)
    cannot properly index large FASTA files. Make sure you are using a later
    (>=1.6) version of BioPerl. The <a href="script/vep_download.html#install">Ensembl VEP
    installer</a> installs 1.6.924 for you.
  </li>
</ol>

<p> If you still see problems after taking these steps, or if you were not
using a FASTA file in the first place, please <a href="#top">contact us</a>. </p>

<p>&nbsp;</p>

<h3 id="chr-not-found"> Q: Why are chromosomes not found in annotation sources or synonyms?</h3>

<pre class="code sh_sh">WARNING: Chromosome 21 not found in annotation sources or synonyms on line 160</pre>

<p> This can occur if the chromosome names differ between your input variant and any annotation source that you are using (cache, database, GFF/GTF file, FASTA file, custom annotation file). To circumvent this you may provide a <a href="script/vep_options.html#opt_synonyms">synonyms file</a>. A synonym file is included in Ensembl VEP's cache files, so if you have one of these for your species you can use it as follows:</p>

<pre class="code sh_sh">./vep -i input.vcf -cache -synonyms ~/.vep/homo_sapiens/[[SPECIESDEFS::ENSEMBL_VERSION]]_GRCh38/chr_synonyms.txt</pre>

<p> The file consists of lines containing pairs of tab-separated synonyms. Order is not important as synonyms can be used in both "directions". </p>

<p>&nbsp;</p>

<h3 id="feature_type_warnings"> Q: Why do I get feature_type warnings from my GFF/GTF file?</h3>

<pre class="code sh_sh sh_sourceCode"><span class="sh_variable">WARNING: </span>Ignoring 'five_prime_utr' feature_type from Homo_sapiens.GRCh38.111.gtf.gz GFF/GTF file. This feature_type is not supported in Ensembl VEP.</span></pre>

<p> This can occur if you are using GFF/GTF file and the file contains a type that is not supported by Ensembl VEP. Those lines are simply ignored. However, in cases where the transcript model is incomplete the full model may be ignored.</p>
  
<p>Please try to use supported feature types as mentioned <a href="https://www.ensembl.org/info/docs/tools/vep/script/vep_cache.html#gff">here</a></p>

<p>&nbsp;</p>


<h3 id="gnomad-frequences"> Q: Can I get gnomAD exomes and genomes frequencies in Ensembl VEP? </h3>

<p> Yes, see <a href="script/vep_example.html#gnomad">this guide</a>. </p>

<p>&nbsp;</p>

<h3 id="no-server-connection"> Q: Why do I have issues connecting to Ensembl databases? </h3>

<pre class="code sh_sh">
Could not connect to database homo_sapiens_core_63_37 as user anonymous using [DBI:mysql:database=homo_sapiens_core_63_37;host=ensembldb.ensembl.org;port=5306] as a locator:
Unknown MySQL server host 'ensembldb.ensembl.org' (2) at $HOME/src/ensembl/modules/Bio/EnsEMBL/DBSQL/DBConnection.pm line 290.

-------------------- EXCEPTION --------------------
MSG: Could not connect to database homo_sapiens_core_63_37 as user anonymous using [DBI:mysql:database=homo_sapiens_core_63_37;host=ensembldb.ensembl.org;port=5306] as a locator:
Unknown MySQL server host 'ensembldb.ensembl.org' (2)
</pre>

<p> If you select the database option rather than using a cache
Ensembl VEP will try to connect to the public MySQL
server at ensembldb.ensembl.org. Occasionally the server may break connection
with your process, which causes this error. This can happen when the server is
busy, or due to various network issues. Consider using the 
<a href="script/vep_cache.html#cache">caching system</a>. 
Using a cache and fasta file is the most effcient way to run Ensembl VEP
</p>

<p>&nbsp;</p>

<h3 id="vep-windows"> Q: Can I use Ensembl VEP on Windows? </h3>

<p> Yes - see the <a href="script/vep_download.html#windows">documentation</a> for a
few different ways to get the Ensembl VEP running on Windows. </p>

<p>&nbsp;</p>

<h3 id="vep-custom-assemblies"> Q: Can I use Ensembl VEP with species and assemblies which are not available in Ensembl? </h3>

<p> Yes - you can run Ensembl VEP on any species you have data for by providing a custom
gene annotation in <a href="script/vep_cache.html#gff">GFF/GTF</a> and genome sequence in
<a href="script/vep_cache.html#fasta">FASTA</a> file, like so:
<pre class="code sh_sh">./vep -i input.vcf --gff data.gff.gz --fasta genome.fa.gz</pre>
</p>

<p>&nbsp;</p>

<h3 id="vep-custom-assemblies"> Q: Can I use Ensembl VEP with T2T-CHM13 and other human assemblies? </h3>

<p> Yes - you can run Ensembl VEP using <a href="https://humanpangenome.org/" rel="external">Human Pangenome Reference Consortium (HPRC)</a> data by following the instructions on how to <a href="script/vep_example.html#pangenomes">use Ensembl VEP with HPRC assemblies</a>.

</p>

<p>&nbsp;</p>

<h3 id="download-sift-polyphen"> Q: Can I download all of the SIFT and/or PolyPhen predictions? </h3>

<p> The Ensembl Variation database and the human Ensembl VEP cache file contain
precalculated SIFT and PolyPhen-2 predictions for every possible amino acid change
in every translated protein product in Ensembl. Since these data are huge, we
store them in a <a
href="/info/genome/variation/prediction/protein_function.html#nsSNP_data_format">compressed format</a>. </p>

<p> There are different approaches to download SIFT/PolyPhen-2 data: </p>

<ul>
  <li> Using the <a href="script/vep_plugins.html#PolyPhen_SIFT">PolyPhen_SIFT plugin</a>: </li>
  <ul>
    <li> For any species with predictions in our Ensembl databases, the plugin is able to download the predictions data into a local SQLite database for offline use. PolyPhen predictions are only available for human data. </li>
    <li> We also provide a downloadble SQLite database containing PolyPhen/SIFT predictions based on <a href="https://humanpangenome.org" rel="external">Human Pangenome Reference Consortium (HPRC)</a> and GRCh38 assemblies. For more information, refer to <a href="script/vep_example.html#PolyPhen_SIFT">Missense deleteriousness predictions</a> in HPRC assemblies.</li>
  </ul>
  <li> Using our <b>Perl API</b>: </li>
  <ul>
    <li> Fetch a <a
href="/info/docs/Doxygen/variation-api/classBio_1_1EnsEMBL_1_1Variation_1_1ProteinFunctionPredictionMatrix.html"><kbd>ProteinFunctionPredictionMatrix</kbd></a> for your protein of interest and then call its
<a href="/info/docs/Doxygen/variation-api/classBio_1_1EnsEMBL_1_1Variation_1_1ProteinFunctionPredictionMatrix.html#af6346bcc5c4c28281e3c9a009ffcb15d"><kbd>get_prediction()</kbd></a> method to get the score for a particular position and amino
acid, looping over all possible amino acids for your position. </li>
    <li> You would need to work out which peptide position your codon maps to, but
there are methods in the <a
href="/info/docs/Doxygen/variation-api/classBio_1_1EnsEMBL_1_1Variation_1_1TranscriptVariation.html"><kbd>TranscriptVariation</kbd></a>
class that should help you (probably <a href="/info/docs/Doxygen/variation-api/classBio_1_1EnsEMBL_1_1Variation_1_1BaseTranscriptVariation.html#acfc84f951919f445624545afbd5da0a8"><kbd>translation_start()</kbd></a> and <a href="/info/docs/Doxygen/variation-api/classBio_1_1EnsEMBL_1_1Variation_1_1BaseTranscriptVariation.html#ae32d5894a6466c7bf0fa02e6d200d038"><kbd>translation_end()</kbd></a>).</li>
  </ul>
</ul>

</div>

</body>
</html>
