<html>
<head>
  <title>Data formats</title>
  <meta name="order" content="4" />
</head>

<body>
    
<style>
tr:nth-child(odd) {background-color: #f0f0f0;}
</style>

  <img src="/img/vep_logo.png" class="float-right" style="width:111px;height:50px" />	
 
  <h1 id="top"><span style="color:#006;padding-right:15px">Variant Effect Predictor</span><span style="color:#666"><img src="/i/16/copy.png"/> Data formats</span></h1>
  <hr />

    <!-- Input formats -->

    <h2 id="input">Input</h2>
    
    <p> Both the web and script version of VEP can use the same input
    formats. Formats can be auto-detected by the VEP script, but must be
    manually selected when using the web interface.</p>
    <p>VEP can use different input formats:</p>
    
    <table class="ss">
      <tr>
        <th>Format</th>
        <th><a href="#sv">Structural variant support</a></th>
        <th>Example</th>
      </tr>
      <tr>
        <td><a href="#default">Default VEP input</a></td>
        <td>Yes</td>
        <td><kbd>1   881907    881906    -/C   +</kbd></td>
      </tr>
      <tr>
        <td><a href="#vcf">VCF</a></td>
        <td>Yes</td>
        <td><kbd>1       65568      .       A    C              .     .       .     .</kbd></td>
      </tr>
      <tr>
        <td><a href="#hgvs">HGVS identifiers</a></td>
        <td></td>
        <td><kbd>ENST00000207771.3:c.344+626A>T</kbd></td>
      </tr>
      <tr>
        <td><a href="#id">Variant identifiers</a></td>
        <td></td>
        <td><kbd>rs699</kbd></td>
      </tr>
      <tr>
        <td><a href="#spdi">Genomic SPDI notation</a></td>
        <td></td>
        <td><kbd>NC_000016.10:68684738:G:A</kbd></td>
      </tr>
      <tr>
        <td><a href="#region">REST-style regions</a></td>
        <td>Yes</td>
        <td><kbd>14:19584687-19584687:-1/T</kbd></td>
      </tr>
    </table>

    <div>
      <div class="info" style="float:left">
        <h3>Note</h3>
        <div class="message-pad">
          <p>VEP can read compressed (gzipped) input file of any format listed above, e.g.:</p>
          <pre class="code sh_sh"> ./vep -i input.vcf.gz -o output.txt</pre>
        </div>
      </div>
      <div style="clear:both"></div>
    </div>
      

    <br />
    <hr /> 
    <h2 id="default"> Default VEP input</h2>
    
    <p>The default format is a simple <strong>whitespace-separated</strong> format
    (columns may be separated by space or tab characters),
    containing five required columns plus an optional identifier column:
    </p>
    
    <ol>
        <li><strong>chromosome</strong> - just the name or number, with no 'chr'
        prefix</li>
        <li><strong>start</strong></li>
        <li><strong>end</strong></li>
        <li><strong>allele</strong> - pair of alleles separated by a '/', with the
        reference allele first (or <a href="#sv">structural variant type)</a></li>
        <li><strong>strand</strong> - defined as + (forward) or - (reverse). The strand will only be used for VEP to know which alleles to use.</li>
        <li><strong>identifier</strong> - this identifier will be used in VEP's
        output. If not provided, VEP will construct an identifier from the given
        coordinates and alleles.</li>
    </ol>
    
    <pre class="code sh_sh">
1   881907    881906    -/C   +
2   946507    946507    G/C   +
5   140532    140532    T/C   +
8   150029    150029    A/T   +    var2
12  1017956   1017956   T/A   +
14  19584687  19584687  C/T   -
19  66520     66520     G/A   +    var1</pre>
    
    <p>An insertion (of any size) is indicated by start coordinate = end coordinate
    + 1. For example, an insertion of 'C' between nucleotides 12600 and 12601 on the
    forward strand of chromosome 8 is indicated as follows:</p>
    
    <pre class="code sh_sh">8   12601     12600     -/C   +</pre>
    <p>A deletion is indicated by the exact nucleotide coordinates. For example, a
    three base pair deletion of nucleotides 12600, 12601, and 12602 of the reverse
    strand of chromosome 8 will be:</p>
    
    <pre class="code sh_sh">8   12600     12602     CGT/- -</pre>
    
    <p>
      Structural variants are also supported by indicating a
      <a href="#sv">structural variant type</a> in the place of the allele:
    </p>
  
    <pre class="code sh_sh">
1    20000     30000     &lt;CN4&gt;  +  cn4
1    160283    471362    DUP    +  sv1
1    1385015   1387562   DEL    +  sv2
12   1017956   1017956   INV    +  inv1
21   25587759  25587769  &lt;CN0&gt;  +  del</pre>
    
    <br />
    <hr />
    <h2 id="vcf"> VCF</h2>
    
    <p> VEP also supports using <a
    rel="external" href="http://www.1000genomes.org/wiki/Analysis/vcf4.0">VCF (Variant Call Format)
    version 4.0</a>. This is a common format used by the 1000 genomes project, and
    can be produced as an output format by many variant calling tools: </p>

  <pre class="code sh_sh">
#CHROM  POS        ID      REF  ALT            QUAL  FILTER  INFO  FORMAT
1       65568      .       A    C              .     .       .     .
1       230710048  rs699   A    G              .     .       .     .
2       265023     .       C    T              .     .       .     .
3       319780     .       GA   G              .     .       .     .
20      3          .       C    CAAG,CAAGAAG   .     PASS    .     .
21      43762120   rs1300  T    A,C,G          .     .       .     .</pre>
    
    <p>Structural variants are also supported depending on
      <a href="#sv">structural variant type.</a></p>
    
    <p> Users using VCF should note a peculiarity in the difference between how
    Ensembl and VCF describe unbalanced variants. For any unbalanced variant (i.e.
    insertion, deletion or unbalanced substitution), the VCF specification requires
    that the base immediately before the variant should be included in both the
    reference and variant alleles. This also affects the reported position i.e. the
    reported position will be one base before the actual site of the variant. </p>
    
    <p> In order to parse this correctly, VEP needs to convert such variants
    into Ensembl-type coordinates, and it does this by removing the additional base
    and adjusting the coordinates accordingly. This means that if an identifier is
    not supplied for a variant (in the 3rd column of the VCF), then the identifier
    constructed and the position reported in VEP's output file will differ from
    the input. </p>
    
    <p> This problem can be overcome with the following: </p>
    <ol>
      <li> ensuring each variant has a unique
    identifier specified in the 3rd column of the VCF </li>
      <li> using VCF format as output (<a href="script/vep_options.html#opt_vcf">--vcf</a>) - this preserves the formatting of your input coordinates and alleles </li>
      <li> using <a href="script/vep_options.html#opt_minimal">--minimal</a> and <a href="script/vep_options.html#opt_allele_number">--allele_number</a> (see <a href="#complex_vcf">Complex VCF entries</a>).</li> 
    </ol>
    
    <p> The following examples illustrate how VCF describes a variant and how it is
    handled internally by VEP. Consider the following aligned sequences (for the
    purposes of discussion on chromosome 20): </p>
    
    <pre class="code sh_sh">Ref: a t C g a // C is the reference base
1 : a t G g a // C base is a G in individual 1
2 : a t - g a // C base is deleted w.r.t. the reference in individual 2
3 : a t CAg a // A base is inserted w.r.t. the reference sequence in individual 3</pre>
    
    <h4> Individual 1 </h4>
    
    <p> The first individual shows a simple balanced substitution of G for C at base
    3. This is described in a compatible manner in VCF and Ensembl styles. Firstly,
    in VCF: </p>
    
    <pre class="code sh_sh">20   3   .   C   G   .   PASS   .</pre>
    
    <p> And in Ensembl format: </p>
    
    <pre class="code sh_sh"> 20   3   3   C/G   +</pre>
    
    <h4> Individual 2 </h4>
    
    <p> The second individual has the 3rd base deleted relative to the reference. In
    VCF, both the reference and variant allele columns must include the preceding
    base (T) and the reported position is that of the preceding base: </p>
    
    <pre class="code sh_sh">20   2   .   TC   T   .   PASS   .</pre>
    
    <p> In Ensembl format, the preceding base is not included, and the start/end
    coordinates represent the region of the sequence deleted. A "-" character is
    used to indicate that the base is deleted in the variant sequence: </p>
    
    <pre class="code sh_sh">20   3   3   C/-   +</pre>
    
    <p> The upshot of this is that while in the VCF input file the position of the
    variant is reported as 2, in the output file from VEP the position will be
    reported as 3. If no identifier is provided in the third column of the VCF, then
    the constructed identifier will be: </p>
    
    <pre class="code sh_sh">20_3_C/-</pre>
    
    <h4> Individual 3 </h4>
    
    <p> The third individual has an "A" inserted between the 3rd and 4th bases of
    the sequence relative to the reference. In VCF, as for the deletion, the base
    before the insertion is included in both the reference and variant allele
    columns, and the reported position is that of the preceding base: </p>
    
    <pre class="code sh_sh">20   3   .   C   CA   .   PASS   .</pre>
    
    <p> In Ensembl format, again the preceding base is not included, and the
    start/end positions are "swapped" to indicate that this is an insertion.
    Similarly to a deletion, a "-" is used to indicate no sequence in the reference:
    </p>
    
    <pre class="code sh_sh"> 20   4   3   -/A   +</pre>
    
    <p> Again, the output will appear different, and the constructed identifier may
    not be what is expected: </p>
    
    <pre class="code sh_sh">20_3_-/A</pre>

    <p> Using VCF format output, or adding unique identifiers to the input (in the
    third VCF column), can mitigate this issue. </p>
   

    <br /> 
    <h3 id="complex_vcf"> Complex VCF entries </h3>

    <p> For VCF entries with multiple alternate alleles, VEP will only trim the
    leading base from alleles if <b>all</b> REF and ALT alleles start with the same
    base: </p>

    <pre class="code sh_sh">20   3   .   C   CAAG,CAAGAAG   .   PASS   .</pre>

    <p> This will be considered internally by VEP as equivalent to: </p>

    <pre class="code sh_sh">20   4   3   -/AAG/AAGAAG   +</pre>

    <p> Now consider the case where a single VCF line contains a representation
    of both a SNV and an insertion: </p>

    <pre class="code sh_sh">20   3   .   C   CAAAG,G   .   PASS   .</pre>

    <p> Here the input alleles will remain unchanged, and VEP will consider the first
    REF/ALT pair as a substitution of C for CAAG, and the second as a C/G SNV:</p>

    <pre class="code sh_sh">20   3   3   C/CAAG/G   +</pre>

    <p> To modify this behaviour, VEP script users may use <a href="script/vep_options.html#opt_minimal">--minimal</a>.
    This flag forces VEP to consider each REF/ALT pair independently, trimming
    identical leading and trailing bases from each as appropriate. Since this can
    lead to confusing output regarding coordinates etc, it is not the default
    behaviour. It is recommended to use the
    <a href="script/vep_options.html#opt_allele_number">--allele_number</a> flag to track
    the correspondence between alleles as input and how they appear in the output.</p>
    

    <br />
    <hr />
    <h3 id="sv"> Structural variant types</h3>
    
    <p>VEP can also call consequences on structural variants using the
      following formats:</p>
    <ul>
      <li><a href="#default">Default VEP input</a></li>
      <li><a href="#region">REST-style regions</a></li>
      <li> <a href="#vcf">VCF</a></li>
    </ul>
  
    <p> To recognise a
      variant as a structural variant, the allele string (or <code>SVTYPE</code>
      in the INFO column of the VCF format) must be set to one of the currently
      recognised values: </p>
    
    <ul>
        <li><b>INS</b> - insertion</li>
        <li><b>DEL</b> - deletion</li>
        <li><b>DUP</b> - duplication</li>
        <li><b>TDUP</b> - tandem duplication</li>
        <li><b>INV</b> - inversion</li>
        <li><b>CNV</b> - copy number variation</li>
        <ul>
          <li>
            The copy number value can be specified,
            such as <kbd>&ltCN0&gt</kbd> or <kbd>&ltCN=4&gt</kbd>
          </li>
        </ul>
        <li><b>BND</b> - breakend</li>
        <ul>
          <li>
            In VCF, breakend replacements are inserted into the <code>ALT</code>
            column and need to meet the
            <a rel="external" href="http://samtools.github.io/hts-specs/">HTS specifications</a>,
            such as <kbd>A[chr22:22893780[,CT[chrX:10932343[</kbd>.
          </li>
        </ul>
    </ul>
    
    <p> Examples of structural variants encoded in VCF format: </p>
    
    <pre class="code sh_sh">
#CHROM  POS        ID   REF  ALT               QUAL  FILTER  INFO                    FORMAT
1       160283     dup  .    &lt;DUP&gt;             .     .       SVTYPE=DUP;END=471362   .
1       1385015    del  .    &lt;DEL&gt;             .     .       SVTYPE=DEL;END=1387562  .
1       234919885  bnd  A    [chr1:17124942[A  .     .       SVTYPE=BND              .</pre>
    
    <p> See the <a
    rel="external" href="http://www.1000genomes.org/wiki/Analysis/Variant%20Call%20Format/VCF%20%28Variant%20Call%20Format%29%20version%204.0/encoding-structural-variants">VCF
    definition document</a> for more detail on how to describe
    structural variants in VCF format. </p>

   
    <br />
    <hr />
    <h2 id="hgvs"> HGVS identifiers</h2>
    
    <p> See <a rel="external" href="//varnomen.hgvs.org">https://varnomen.hgvs.org</a> for
    details. These must be relative to genomic or Ensembl transcript coordinates.</p>
    <p>It also is possible to use RefSeq transcripts in both the web
    interface and the VEP script (see <a href="script/vep_other.html#hgvs">script documentation</a>): 
    this works for RefSeq transcripts that align to the genome correctly.</p>
    
    <p> Examples: </p>
    
    <pre class="code sh_sh">ENST00000207771.3:c.344+626A>T
ENST00000471631.1:c.28_33delTCGCGG
ENST00000285667.3:c.1047_1048insC
5:g.140532T>C</pre>
    
    <p> Examples using RefSeq identifiers (using <a
    href="script/vep_options.html#opt_refseq">--refseq</a> in the VEP script, or select
    the otherfeatures transcript database on the web interface and input type of
    HGVS): </p>
    
    <pre class="code sh_sh">NM_153681.2:c.7C>T
NM_005239.4:c.190G>A
NM_001025204.1:c.336G>A</pre>
    
    <p> HGVS protein notations may also be used, provided that they
    unambiguously map to a single genomic change. Due to redundancy in the amino
    acid code, it is not always possible to work out the corresponding genomic
    sequence change for a given protein sequence change. The following example
    is for a permissable protein notation in dog <i>(Canis familiaris)</i>:</p>
    
    <pre class="code sh_sh">ENSCAFP00000040171.1:p.Thr92Asn</pre>
    
    <p> HGVS notations may also be given in <a href="https://www.lrg-sequence.org"
    rel="external">LRG</a> coordinates: </p>
    
    <pre class="code sh_sh">LRG_1t1:c.841G>T
LRG_1:g.10006G>T</pre>

    <h4>Ambiguous gene-based descriptions</h4>
    <p>It is possible to use ambiguous descriptions listing only gene symbol or UniProt accession and protein change (e.g. PHF21B:p.Tyr124Cys,  P01019:p.Ala268Val), as seen in the literature, though this is not recommended as it can produce multiple different variants at genomic level. The transcripts for a gene are considered in the following order:
      <ol>
        <li><a href="/info/genome/genebuild/mane.html">MANE Select transcript status</a></li>
        <li><a href="/info/genome/genebuild/mane.html">MANE Plus Clinical transcript status</a></li>
        <li>canonical status of transcript</li>
        <li><a href="/Help/Glossary?id=521">APPRIS isoform annotation</a></li>
        <li><a href="/Help/Glossary?id=492">transcript support level</a></li>
        <li>biotype of transcript ("protein_coding" preferred)</li>
        <li>CCDS status of transcript</li>
        <li>consequence rank according to <a href="/info/genome/variation/prediction/predicted_data.html#consequences">this table</a></li>
        <li>translated, transcript or feature length (longer preferred)</li>
      </ol>
      and the first compatible transcript is used to map variants to the genome for annotation.</p>

    <br />
    <hr />
    <h2 id="id"> Variant identifiers</h2>
    
    <p> These should be e.g. dbSNP rsIDs, or any synonym for a variant present in
    the Ensembl Variation database. See <a
    href="/info/genome/variation/species/sources_documentation.html">here</a> for a list of
    identifier sources in Ensembl.</p>
   
    <div>
      <div class="info" style="float:left">
        <h3>Note</h3>
        <div class="message-pad">
          <p>VEP is optimised for the analysis of variants in sorted VCF files or other position based formats, so input in these formats is recommended to variant names where possible. Using variant names can take 2-5 times longer.</p>
        </div>
      </div>
      <div style="clear:both"></div>
    </div> 

    <br />
    <hr />
    <h2 id="spdi"> Genomic SPDI notation</h2>

    <p> VEP can also support genomic SPDI notation which uses four fields delimited by colons S:P:D:I (Sequence:Position:Deletion:Insertion). In SPDI notation, the position refers to the base before the variant, not the base of the variant itsef.</p> 
    <p>See <a href="https://www.ncbi.nlm.nih.gov/variation/notation/">here</a> for details.</p>

    <p> Examples: </p>
    
    <pre class="code sh_sh">NC_000016.10:68684738:G:A
NC_000017.11:43092199:GCTTTT:
NC_000013.11:32315789::C
NC_000016.10:68644746:AA:GTA
16:68684738:2:AC</pre>


    <br />
    <hr />
    <h2 id="region"> REST-style regions</h2>
    <p>
      VEP's region REST endoint requires variants are described as
      <code>[chr]:[start]-[end]:[strand]/[allele]</code>.
    </p>
    <p>
      This follows the same conventions as the
      <a href="#default">default input format</a>, with the key difference being
      that this format does not require the reference (REF) allele to be
      included; VEP will look up the reference allele using either
      a provided FASTA file (preferred) or Ensembl core database. Strand is
      optional and defaults to 1 (forward strand).
    </p>

    <pre class="code sh_sh"># SNP
5:140532-140532:1/C

# SNP (reverse strand)
14:19584687-19584687:-1/T

# insertion
1:881907-881906:1/C

# 5bp deletion
2:946507-946511:1/-</pre>

  <p>
    Structural variants are also supported by indicating a
    <a href="#sv">structural variant type</a> in the place of the
    <code>[allele]</code>:
  </p>
  <pre class="code sh_sh"># structural variant: deletion
21:25587759-25587769/DEL
    
# structural variant: inversion
21:25587759-25587769/INV</pre>

    <!-- Output formats -->
    <br />
    <hr style="margin-bottom:0"/>
    <hr style="margin-top:0"/>
    <h2 id="output">Output</h2>

    <p>VEP can return the results in different formats:</p>
    <ul>
      <li><a href="#defaultout">Default VEP output</a></li>
      <li><a href="#tab">Tab-delimited output</a></li>
      <li><a href="#vcfout">VCF</a></li>
      <li><a href="#json">JSON output</a></li>
    </ul>
   
    <p>Along with the results VEP computes and returns some <a href="#stats">statistics</a>.</p>


    <br />
    <hr />
    <h2 id="defaultout"> Default VEP output</h2>
 
    <p> The default output format ("VEP" format when downloading from the web interface) is a 14 column tab-delimited file. Empty values are denoted by '-'. The output
    columns are:</p>
    
    <ol>
        <li><strong>Uploaded variation</strong> - as chromosome_start_alleles</li>
        <li><strong>Location</strong> - in standard coordinate format (chr:start or chr:start-end)</li>
        <li><strong>Allele</strong> - the variant allele used to calculate the consequence</li>
        <li><strong>Gene</strong> - Ensembl stable ID of affected gene</li>
        <li><strong>Feature</strong> - Ensembl stable ID of feature</li>
        <li><strong>Feature type</strong> - type of feature. Currently one of Transcript, RegulatoryFeature, MotifFeature.</li>
        <li><strong>Consequence</strong> - <a href="/info/genome/variation/prediction/predicted_data.html#consequences">consequence type</a> of this variant</li>
        <li><strong>Position in cDNA</strong> - relative position of base pair in cDNA sequence</li>
        <li><strong>Position in CDS</strong> - relative position of base pair in coding sequence</li>
        <li><strong>Position in protein</strong> - relative position of amino acid in protein</li>
        <li><strong>Amino acid change</strong> - only given if the variant affects the protein-coding sequence</li>
        <li><strong>Codon change</strong> - the alternative codons with the variant base in upper case</li>
        <li><strong>Co-located variation</strong> - identifier of any <a href="/info/docs/tools/vep/script/vep_other.html#colocated">existing variants</a>. Switch on with <a href="script/vep_options.html#opt_check_existing">--check_existing</a></li>
        <li><strong>Extra</strong> - this column contains extra information as key=value pairs separated by ";", see below.</li>
    </ol>


    <h3 style="margin:20px 0px 30px">Other output fields:</h3>

    <div id="div_other_fields" >
      <ul>
        <li><b>REF_ALLELE</b> - the reference allele (after minimisation)</li>
        <li><b>UPLOADED_ALLELE</b> - the uploaded allele string (before minimisation)</li>
        <li><b>IMPACT</b> - the impact modifier for the consequence type</li>
        <li><b>VARIANT_CLASS</b> - Sequence Ontology <a href="/info/genome/variation/prediction/classification.html#classes">variant class</a></li>
        <li><b>SYMBOL</b> - the gene symbol</li>
        <li><b>SYMBOL_SOURCE</b> - the source of the gene symbol</li>
        <li><b>STRAND</b> - the DNA strand (1 or -1) on which the transcript/feature lies</li>
        <li><b>ENSP</b> - the Ensembl protein identifier of the affected transcript</li>
        <li>
            <b>FLAGS</b> - transcript quality flags:
            <ul>
                <li><i>cds_start_NF:</i> CDS 5' incomplete</li>
                <li><i>cds_end_NF:</i> CDS 3' incomplete</li>
            </ul>
        </li>
        <li><b>SWISSPROT</b> - Best match UniProtKB/Swiss-Prot accession of protein product
        <li><b>TREMBL</b> - Best match UniProtKB/TrEMBL accession of protein product</li>
        <li><b>UNIPARC</b> - Best match UniParc accession of protein product</li>
        <li><b>HGVSc</b> - the HGVS coding sequence name</li>
        <li><b>HGVSp</b> - the HGVS protein sequence name</li>
        <li><b>HGVSg</b> - the HGVS genomic sequence name</li>
        <li><b>HGVS_OFFSET</b> - Indicates by how many bases the HGVS notations for this variant have been <a href="script/vep_options.html#opt_shift_hgvs">shifted</a>. Value must be greater than 0.</li>
        <li><b>NEAREST</b> - Identifier(s) of nearest transcription start site</li>
        <li><b>SIFT</b> - the SIFT prediction and/or score, with both given as prediction(score)</li>
        <li><b>PolyPhen</b> - the PolyPhen prediction and/or score</li>
        <li><b>MOTIF_NAME</b> - the source and identifier of a transcription factor binding profile aligned at this position</li>
        <li><b>MOTIF_POS</b> - The relative position of the variation in the aligned TFBP</li>
        <li><b>HIGH_INF_POS</b> - a flag indicating if the variant falls in a high information position of a transcription factor binding profile (TFBP)</li>
        <li><b>MOTIF_SCORE_CHANGE</b> - The difference in motif score of the reference and variant sequences for the TFBP</li>
        <li><b>CELL_TYPE</b> - List of cell types and classifications for regulatory feature</li>
        <li><b>CANONICAL</b> - a flag indicating if the transcript is denoted as the canonical transcript for this gene</li>
        <li><b>CCDS</b> - the CCDS identifer for this transcript, where applicable</li>
        <li><b>INTRON</b> - the intron number (out of total number)</li>
        <li><b>EXON</b> - the exon number (out of total number)</li>
        <li><b>DOMAINS</b> - the source and identifer of any overlapping protein domains</li>
        <li><b>DISTANCE</b> - Shortest distance from variant to transcript</li>
        <li><b>IND</b> - individual name</li>
        <li><b>ZYG</b> - zygosity of individual genotype at this locus</li>
        <li><b>SV</b> - IDs of overlapping structural variants</li>
        <li><b>FREQS</b> - Frequencies of overlapping variants used in filtering</li>
        <li><b>AF</b> - Frequency of existing variant in 1000 Genomes</li>
        <li><b>AFR_AF</b> - Frequency of existing variant in 1000 Genomes combined African population</li>
        <li><b>AMR_AF</b> - Frequency of existing variant in 1000 Genomes combined American population</li>
        <li><b>ASN_AF</b> - Frequency of existing variant in 1000 Genomes combined Asian population</li>
        <li><b>EUR_AF</b> - Frequency of existing variant in 1000 Genomes combined European population</li>
        <li><b>EAS_AF</b> - Frequency of existing variant in 1000 Genomes combined East Asian population</li>
        <li><b>SAS_AF</b> - Frequency of existing variant in 1000 Genomes combined South Asian population</li>
        <li><b>gnomADe_AF</b> - Frequency of existing variant in gnomAD exomes combined population</li>
        <li><b>gnomADe_AFR_AF</b> - Frequency of existing variant in gnomAD exomes African/American population</li>
        <li><b>gnomADe_AMR_AF</b> - Frequency of existing variant in gnomAD exomes American population</li>
        <li><b>gnomADe_ASJ_AF</b> - Frequency of existing variant in gnomAD exomes Ashkenazi Jewish population</li>
        <li><b>gnomADe_EAS_AF</b> - Frequency of existing variant in gnomAD exomes East Asian population</li>
        <li><b>gnomADe_FIN_AF</b> - Frequency of existing variant in gnomAD exomes Finnish population</li>
        <li><b>gnomADe_NFE_AF</b> - Frequency of existing variant in gnomAD exomes Non-Finnish European population</li>
        <li><b>gnomADe_OTH_AF</b> - Frequency of existing variant in gnomAD exomes combined other combined populations</li>
        <li><b>gnomADe_SAS_AF</b> - Frequency of existing variant in gnomAD exomes South Asian population</li>
        <li><b>gnomADg_AF</b> - Frequency of existing variant in gnomAD exomes combined population</li>
        <li><b>gnomADg_AFR_AF</b> - Frequency of existing variant in gnomAD genomes African/American population</li>
        <li><b>gnomADg_AMI_AF</b> - Frequency of existing variant in gnomAD genomes Amish population</li>
        <li><b>gnomADg_AMR_AF</b> - Frequency of existing variant in gnomAD genomes American population</li>
        <li><b>gnomADg_ASJ_AF</b> - Frequency of existing variant in gnomAD genomes Ashkenazi Jewish population</li>
        <li><b>gnomADg_EAS_AF</b> - Frequency of existing variant in gnomAD genomes East Asian population</li>
        <li><b>gnomADg_FIN_AF</b> - Frequency of existing variant in gnomAD genomes Finnish population</li>
        <li><b>gnomADg_MID_AF</b> - Frequency of existing variant in gnomAD genomes Mid-eastern population</li>
        <li><b>gnomADg_NFE_AF</b> - Frequency of existing variant in gnomAD genomes Non-Finnish European population</li>
        <li><b>gnomADg_OTH_AF</b> - Frequency of existing variant in gnomAD genomes combined other combined populations</li>
        <li><b>gnomADg_SAS_AF</b> - Frequency of existing variant in gnomAD genomes South Asian population</li>
        <li><b>MAX_AF</b> - Maximum observed allele frequency in 1000 Genomes, ESP and gnomAD</li>
        <li><b>MAX_AF_POPS</b> - Populations in which maximum allele frequency was observed</li>
        <li><b>CLIN_SIG</b> - ClinVar clinical significance of the dbSNP variant</li>
        <li><b>BIOTYPE</b> - Biotype of transcript or regulatory feature</li>
        <li><b>APPRIS</b> - Annotates alternatively spliced transcripts as primary or alternate based on a range of computational methods. NB: not available for GRCh37</li>
        <li><b>TSL</b> - Transcript support level. NB: not available for GRCh37</li>
        <li><b>PUBMED</b> - Pubmed ID(s) of publications that cite existing variant</li>
        <li><b>SOMATIC</b> - Somatic status of existing variant(s); multiple values correspond to multiple values in the Existing_variation field</li>
        <li><b>PHENO</b> - Indicates if existing variant is associated with a phenotype, disease or trait; multiple values correspond to multiple values in the Existing_variation field</li>
        <li><b>GENE_PHENO</b> - Indicates if overlapped gene is associated with a phenotype, disease or trait</li>
        <li><b>ALLELE_NUM</b> - Allele number from input; 0 is reference, 1 is first alternate etc</li>
        <li><b>MINIMISED</b> - Alleles in this variant have been converted to minimal representation before consequence calculation</li>
        <li><b>PICK</b> - indicates if this block of consequence data was picked by <a href="script/vep_options.html#opt_flag_pick">--flag_pick</a> or <a href="script/vep_options.html#opt_flag_pick_allele">--flag_pick_allele</a></li>
        <li><b>BAM_EDIT</b> - Indicates success or failure of edit using BAM file</li>
        <li><b>GIVEN_REF</b> - Reference allele from input</li>
        <li><b>USED_REF</b> - Reference allele as used to get consequences</li>

        <li id="refseq_match"><b>REFSEQ_MATCH</b> - the RefSeq transcript match status; contains a number of flags indicating whether this RefSeq transcript matches the underlying reference sequence and/or an Ensembl transcript (<a href="script/vep_other.html#refseq">more information</a>). 
          <ul>
            <li><i>rseq_3p_mismatch:</i> signifies a mismatch between the RefSeq transcript and the underlying primary genome assembly sequence. Specifically, there is a mismatch in the 3' UTR of the RefSeq model with respect to the primary genome assembly (e.g. GRCh37/GRCh38). </li>
            <li><i>rseq_5p_mismatch:</i> signifies a mismatch between the RefSeq transcript and the underlying primary genome assembly sequence. Specifically, there is a mismatch in the 5' UTR of the RefSeq model with respect to the primary genome assembly. </li>
            <li><i>rseq_cds_mismatch:</i> signifies a mismatch between the RefSeq transcript and the underlying primary genome assembly sequence. Specifically, there is a mismatch in the CDS of the RefSeq model with respect to the primary genome assembly. </li>
            <li><i>rseq_ens_match_cds:</i> signifies that for the RefSeq transcript there is an overlapping Ensembl model that is identical across the CDS region only. A CDS match is defined as follows: the CDS and peptide sequences are identical and the genomic coordinates of every translatable exon match. Useful related attributes are: rseq_ens_match_wt and rseq_ens_no_match.</li>
            <li><i>rseq_ens_match_wt:</i> signifies that for the RefSeq transcript there is an overlapping Ensembl model that is identical across the whole transcript. A whole transcript match is defined as follows: 1) In the case that both models are coding, the transcript, CDS and peptide sequences are all identical and the genomic coordinates of every exon match. 2) In the case that both transcripts are non-coding the transcript sequences and the genomic coordinates of every exon are identical. No comparison is made between a coding and a non-coding transcript. Useful related attributes are: rseq_ens_match_cds and rseq_ens_no_match.</li>
            <li><i>rseq_ens_no_match:</i> signifies that for the RefSeq transcript there is no overlapping Ensembl model that is identical across either the whole transcript or the CDS. This is caused by differences between the transcript, CDS or peptide sequences or between the exon genomic coordinates. Useful related attributes are: rseq_ens_match_wt and rseq_ens_match_cds.</li>
            <li><i>rseq_mrna_match:</i> signifies an exact match between the RefSeq transcript and the underlying primary genome assembly sequence (based on a match between the transcript stable id and an accession in the RefSeq mRNA file). An exact match occurs when the underlying genomic sequence of the model can be perfectly aligned to the mRNA sequence post polyA clipping.</li>
            <li><i>rseq_mrna_nonmatch:</i> signifies a non-match between the RefSeq transcript and the underlying primary genome assembly sequence. A non-match is deemed to have occurred if the underlying genomic sequence does not have a perfect alignment to the mRNA sequence post polyA clipping. It can also signify that no comparison was possible as the model stable id may not have had a corresponding entry in the RefSeq mRNA file (sometimes happens when accessions are retired or changed). When a non-match occurs one or several of the following transcript attributes will also be present to provide more detail on the nature of the non-match: rseq_5p_mismatch, rseq_cds_mismatch, rseq_3p_mismatch, rseq_nctran_mismatch, rseq_no_comparison</li>
            <li><i>rseq_nctran_mismatch:</i> signifies a mismatch between the RefSeq transcript and the underlying primary genome assembly sequence. This is a comparison between the entire underlying genomic sequence of the RefSeq model to the mRNA in the case of RefSeq models that are non-coding. </li>
            <li><i>rseq_no_comparison:</i> signifies that no alignment was carried out between the underlying primary genome assembly sequence and a corresponding RefSeq mRNA. The reason for this is generally that no corresponding, unversioned accession was found in the RefSeq mRNA file for the transcript stable id. This sometimes happens when accessions are retired or replaced. A second possibility is that the sequences were too long and problematic to align (though this is rare).</li>
          </ul>
        </li>
        <li><b>OverlapBP</b> - Number of base pairs overlapping with the corresponding structural variation feature</li>
        <li><b>OverlapPC</b> - Percentage of corresponding structural variation feature overlapped by the given input</li>
        <li><b>CHECK_REF</b> - Reports variants where the input reference does not match the expected reference</li>
        <li><b>AMBIGUITY</b> - IUPAC allele ambiguity code</li>
      </ul>
    </div>
   
    <p>Example of VEP default output format:</p> 
    <pre class="code sh_sh">
11_224088_C/A    11:224088   A  ENSG00000142082  ENST00000525319  Transcript         missense_variant           742  716  239  T/N  aCc/aAc  -  SIFT=deleterious(0);PolyPhen=unknown(0)
11_224088_C/A    11:224088   A  ENSG00000142082  ENST00000534381  Transcript         5_prime_UTR_variant        -    -    -    -    -        -  -
11_224088_C/A    11:224088   A  ENSG00000142082  ENST00000529055  Transcript         downstream_variant         -    -    -    -    -        -  -
11_224585_G/A    11:224585   A  ENSG00000142082  ENST00000529937  Transcript         intron_variant             -    -    -    -    -        -  HGVSc=ENST00000529937.1:c.136-346G>A
22_16084370_G/A  22:16084370 A  -                ENSR00000615113  RegulatoryFeature  regulatory_region_variant  -    -    -    -    -        -  -</pre>
    
    <p> The VEP script will also add a header to the output file. This contains
    information about the databases connected to, and also a key describing the
    key/value pairs used in the extra column. </p>
    
    <pre class="code sh_sh">
## ENSEMBL VARIANT EFFECT PREDICTOR v[[SPECIESDEFS::ENSEMBL_VERSION]].0
## Output produced at 2017-03-21 14:51:27
## Connected to homo_sapiens_core_[[SPECIESDEFS::ENSEMBL_VERSION]]_38 on ensembldb.ensembl.org
## Using cache in /homes/user/.vep/homo_sapiens/[[SPECIESDEFS::ENSEMBL_VERSION]]_GRCh38
## Using API version [[SPECIESDEFS::ENSEMBL_VERSION]], DB version [[SPECIESDEFS::ENSEMBL_VERSION]]
## polyphen version 2.2.2
## sift version sift5.2.2
## COSMIC version 78
## ESP version 20141103
## gencode version GENCODE 25
## genebuild version 2014-07
## HGMD-PUBLIC version 20162
## regbuild version 16
## assembly version GRCh38.p7
## ClinVar version 201610
## dbSNP version 147
## Column descriptions:
## Uploaded_variation : Identifier of uploaded variant
## Location : Location of variant in standard coordinate format (chr:start or chr:start-end)
## Allele : The variant allele used to calculate the consequence
## Gene : Stable ID of affected gene
## Feature : Stable ID of feature
## Feature_type : Type of feature - Transcript, RegulatoryFeature or MotifFeature
## Consequence : Consequence type
## cDNA_position : Relative position of base pair in cDNA sequence
## CDS_position : Relative position of base pair in coding sequence
## Protein_position : Relative position of amino acid in protein
## Amino_acids : Reference and variant amino acids
## Codons : Reference and variant codon sequence
## Existing_variation : Identifier(s) of co-located known variants
## Extra column keys:
## IMPACT : Subjective impact classification of consequence type
## DISTANCE : Shortest distance from variant to transcript
## STRAND : Strand of the feature (1/-1)
## FLAGS : Transcript quality flags</pre>


    <br />
    <hr />
    <h2 id="tab"> Tab-delimited output</h2>

    <p> The <a href="script/vep_options.html#opt_tab">--tab</a> flag instructs VEP
    to write output as a tab-delimited table.<br />
    <b>This differs from the default output format in that each individual field from the 
    "Extra" field is written to a separate tab-delimited column</b>.<br />
    This makes the output more suitable for import into spreadsheet programs such as Excel.<br />
    Furthermore the header is the same as the one for the VEP default output format and this is 
    also the format used when selecting the "TXT" option on the VEP web interface.</p>

    <p>Example of VEP tab-delimited output format:</p>
    <pre class="code sh_sh">
#Uploaded_variation  Location   Allele  Gene             Feature          Feature_type  Consequence                            cDNA_position  CDS_position  Protein_position  Amino_acids  Codons   Existing_variation  IMPACT    DISTANCE  STRAND  FLAGS 
11_224088_C/A        11:224088  A       ENSG00000142082  ENST00000525319  Transcript    missense_variant                       742            716           239               S/I          aGc/aTc  -                   MODERATE  -         -1      - 
11_224088_C/A        11:224088  A       ENSG00000142082  ENST00000534381  Transcript    downstream_gene_variant                -              -             -                 -            -        -                   MODIFIER  1674      -1      -
11_224088_C/A        11:224088  A       ENSG00000142082  ENST00000529055  Transcript    downstream_gene_variant                -              -             -                 -            -        -                   MODIFIER  134       -1      -
11_224585_G/A        11:224585  A       ENSG00000142082  ENST00000529937  Transcript    intron_variant,NMD_transcript_variant  -              -             -                 -            -        -                   MODIFIER  -         -1      -</pre>
    
    <div>
      <div class="info" style="float:left">
        <h3>Note</h3>
        <div class="message-pad">
          <p>The Existing_variation column is only populated if we use option <a href="script/vep_options.html#opt_check_existing">--check_existing</a> or any other option that switches it on (--af_1kg silently switches on --check_existing).</p>
        </div>
      </div>
      <div style="clear:both"></div>
    </div>

    <p> The choice and order of columns in the output may be configured using <a href="script/vep_options.html#opt_fields">--fields</a>. For instance:<br />
      <pre class="code sh_sh">./vep -i examples/homo_sapiens_GRCh38.vcf --cache --force_overwrite <b>--tab --fields "Uploaded variation,Location,Allele,Gene"</b></pre>
    </p>
    

    <br />
    <hr />
    <h2 id="vcfout"> VCF output</h2>
    
    <p> The VEP script can also generate VCF output using the <a href="script/vep_options.html#opt_vcf">--vcf</a> flag.</p>
    <p>Main information about the specificity of the VEP VCF output format:</p>
    <ul>
      <li>Consequences are added in the INFO field of the VCF file, using the key "<b>CSQ</b>" 
          (you can change it using <a href="script/vep_options.html#opt_vcf_info_field">--vcf_info_field</a>).</li>
      <li>Data fields are encoded separated by the character "<b>&#124;</b>" (pipe). The order of fields is written in the VCF header. 
          Unpopulated fields are represented by an empty string.</li>
      <li>Output fields in the "CSQ" INFO field can be configured by using <a href="script/vep_options.html#opt_fields">--fields</a>.</li>
      <li>Each prediction, for a given variant, is separated by the character "<b>,</b>" in the CSQ INFO field (e.g. when a variant overlaps more than 1 transcript)</li>
    </ul>
    <br />

    <p>
    Here is a list of the (default) fields you can find within the CSQ field:
    <pre class="code sh_sh">Allele|Consequence|IMPACT|SYMBOL|Gene|Feature_type|Feature|BIOTYPE|EXON|INTRON|HGVSc|HGVSp|cDNA_position|CDS_position|Protein_position|Amino_acids|Codons|Existing_variation|DISTANCE|STRAND|FLAGS|SYMBOL_SOURCE|HGNC_ID</pre>
    </p>

    <div>
      <div class="info" style="float:left">
        <h3>Note</h3>
        <div class="message-pad">
          <p>Some fields are not reported by default and need configuring including HGVS, Symbol and Existing_variation.</p>
        </div>
      </div>
      <div style="clear:both"></div>
    </div>

    <p>Example of VEP command using the <a href="script/vep_options.html#opt_vcf">--vcf</a> and <a href="script/vep_options.html#opt_fields">--fields</a> options:
    <pre class="code sh_sh">./vep -i examples/homo_sapiens_GRCh38.vcf --cache --force_overwrite <b>--vcf --fields "Allele,Consequence,Feature_type,Feature"</b></pre>
    
    <p> VCFs produced by VEP can be filtered by <a
    href="script/vep_filter.html">filter_vep.pl</a> in the same way as standard format
    output files. </p>
    
    <p> If the input format was VCF, the file will remain unchanged save for the
    addition of the CSQ field and the header (unless using any filtering). If an
    existing CSQ field is found, it will be replaced by the one added by the
    VEP (use <a href="script/vep_options.html#opt_keep_csq">--keep_csq</a> to
    preserve it). </p>
    
    <p> Custom data added with <a
    href="script/vep_options.html#opt_custom">--custom</a> are added as separate
    fields, using the key specified for each data file.</p>
    
    <p> Commas in fields are replaced with ampersands (&amp;) to preserve VCF
    format. </p>
    
    <pre class="code sh_sh">##INFO=&lt;ID=CSQ,Number=.,Type=String,Description="Consequence annotations from Ensembl VEP. Format: Allele|Consequence|IMPACT|SYMBOL|Gene|Feature_type|Feature|BIOTYPE|EXON|INTRON|HGVSc|HGVSp|cDNA_position|CDS_position|Protein_position"&gt;
#CHROM  POS       ID          REF  ALT  QUAL  FILTER  INFO
21      26978790  rs75377686  T    C    .     .       CSQ=C|missense_variant|MODERATE|MRPL39|ENSG00000154719|Transcript|ENST00000419219|protein_coding|2/8||ENST00000419219.1:c.251A>G|ENSP00000404426.1:p.Asn84Ser|260|251|84</pre>
    
    
    <br />
    <hr />
    <h2 id="json"> JSON output</h2>
    
    <p> VEP can produce output in the form of serialised <a
    href="http://json.org/" rel="external">JSON</a> objects using the <a
    href="script/vep_options.html#opt_json">--json</a> flag. JSON is a
    serialisation format that can be parsed and processed easily by many
    packages and programming languages; it is used as the default output format
    for <a href="http://rest.ensembl.org" rel="external">Ensembl's REST
    server</a>. </p>
    
    <p> Each input variant is reported as a single JSON object which constitutes
    one line of the output file. The JSON object is structured somewhat
    differently to the other VEP output formats, in that per-variant fields
    (e.g. co-located existing variant details) are reported only once.
    Consequences are grouped under the feature type that they affect
    (Transcript, Regulatory Feature, etc). The original input line (e.g. from
    VCF input) is reported under the "input" key in order to aid aligning input
    with output. When using a cache file, frequencies for co-located variants are reported by default (see
    <a href="./script/vep_options.html#opt_af_1kg">--af_1kg</a>,
    <a href="./script/vep_options.html#opt_af_gnomade">--af_gnomade</a>).
    </p>
    
    <p> Here follows an example of JSON output (prettified and redacted for display here): </p>
    
    <pre class="code sh_sh">{
  "input": "1 1918090 test1 A G . . .",
  "id": "test1",
  "seq_region_name": "1",
  "start": 1918090,
  "end": 1918090,
  "strand": 1,
  "allele_string": "A/G",
  "most_severe_consequence": "missense_variant",
  "colocated_variants": [
    {
      "id": "COSV57068665",
      "seq_region_name": "1",
      "start": 1918090,
      "end": 1918090,
      "strand": 1,
      "allele_string": "COSMIC_MUTATION"
    },
    {
      "id": "rs28640257",
      "seq_region_name": "1",
      "start": 1918090,
      "end": 1918090,
      "strand": 1,
      "allele_string": "A/G/T",
      "minor_allele": "G",
      "minor_allele_freq": 0.352,
      "frequencies": {
        "G": {
          "amr": 0.5072,
          "gnomade_sas": 0.369,
          "gnomade": 0.4541,
          "gnomade_oth": 0.4611,
          "gnomade_asj": 0.3909,
          "gnomade_nfe": 0.4944,
          "gnomade_afr": 0.103,
          "afr": 0.053,
          "gnomade_amr": 0.5641,
          "gnomade_fin": 0.474,
          "sas": 0.3906,
          "gnomade_eas": 0.4598,
          "eur": 0.4901,
          "eas": 0.4623
        }
      }
    }
  ],
  "transcript_consequences": [
    {
      "variant_allele": "G",
      "consequence_terms": [
        "missense_variant"
      ],
      "gene_id": "ENSG00000178821",
      "transcript_id": "ENST00000310991",
      "strand": -1,
      "cdna_start": 436,
      "cdna_end": 436,
      "cds_start": 422,
      "cds_end": 422,
      "protein_start": 141,
      "protein_end": 141,
      "codons": "aTg/aCg",
      "amino_acids": "M/T",
      "polyphen_prediction": "benign",
      "polyphen_score": 0.001,
      "sift_prediction": "tolerated",
      "sift_score": 0.22,
      "hgvsp": "ENSP00000311122.3:p.Met141Thr",
      "hgvsc": "ENST00000310991.8:c.422T>C"
      }
  ],
  "regulatory_feature_consequences": [
    {
      "variant_allele": "G",
      "consequence_terms": [
        "regulatory_region_variant"
      ],
      "regulatory_feature_id": "ENSR00000000255"
    }
  ]
}</pre>

    <p> In accordance with JSON conventions, all keys (except alleles) are lower-case. Some keys
    also have different names and structures to those found in the other VEP
    output formats: </p>
    
    <table class="ss">
      <tr>
        <th>Key</th>
        <th>JSON equivalent(s)</th>
        <th>Notes</th>
      </tr>
      <tr>
        <td>Consequence</td>
        <td>consequence_terms</td>
        <td></td>
      </tr>
      <tr>
        <td>Gene</td>
        <td>gene_id</td>
        <td></td>
      </tr>
      <tr>
        <td>Feature</td>
        <td>transcript_id, regulatory_feature_id, motif_feature_id</td>
        <td>Consequences are grouped under the feature type they affect</td>
      </tr>
      <tr>
        <td>ALLELE</td>
        <td>variant_allele</td>
        <td></td>
      </tr>
      <tr>
        <td>SYMBOL</td>
        <td>gene_symbol</td>
        <td></td>
      </tr>
      <tr>
        <td>SYMBOL_SOURCE</td>
        <td>gene_symbol_source</td>
        <td></td>
      </tr>
      <tr>
        <td>ENSP</td>
        <td>protein_id</td>
        <td></td>
      </tr>
      <tr>
        <td>OverlapBP</td>
        <td>bp_overlap</td>
        <td></td>
      </tr>
      <tr>
        <td>OverlapPC</td>
        <td>percentage_overlap</td>
        <td></td>
      </tr>
      <tr>
        <td>Uploaded_variation</td>
        <td>id</td>
        <td></td>
      </tr>
      <tr>
        <td>Location</td>
        <td>seq_region_name, start, end, strand</td>
        <td>The variant's location field is broken down into constituent coordinate parts for clarity. "seq_region_name" is used in place of "chr" or "chromosome" for consistency with other parts of Ensembl's REST API</td>
      </tr>
      <tr>
        <td>*_maf</td>
        <td>*_allele, *_maf</td>
        <td></td>
      </tr>
      <tr>
        <td>cDNA_position</td>
        <td>cdna_start, cdna_end</td>
        <td></td>
      </tr>
      <tr>
        <td>CDS_position</td>
        <td>cds_start, cds_end</td>
        <td></td>
      </tr>
      <tr>
        <td>Protein_position</td>
        <td>protein_start, protein_end</td>
        <td></td>
      </tr>
      <tr>
        <td>SIFT</td>
        <td>sift_prediction, sift_score</td>
        <td></td>
      </tr>
      <tr>
        <td>PolyPhen</td>
        <td>polyphen_prediction, polyphen_score</td>
        <td></td>
      </tr>
    </table>
   
    <br />
    <hr />
    <h2 id="stats"> Statistics</h2>
    
    <p> VEP writes an HTML file containing statistics pertaining to the
    results of your job; it is named <b>[output_file]_summary.html</b> (with the
    default options the file will be named
    <b>variant_effect_output.txt_summary.html</b>). To view it you should open
    the file in your web browser. </p>
    
    <p> To prevent VEP writing a stats file, use the flag <a
    href="script/vep_options.html#opt_no_stats">--no_stats</a>. To have VEP write a
    machine-readable text file in place of the HTML, use <a
    href="script/vep_options.html#opt_stats_text">--stats_text</a>. To change the name
    of the stats file from the default, use <a
    href="script/vep_options.html#opt_stats_file">--stats_file [file]</a>. </p>
    
    <p> The page contains several sections: </p>
    
    <h4> General statistics </h4>
    
    <p> This section contains two tables. The first describes the cache and/or
    database used, the version of VEP, species, command line parameters,
    input/output files and run time. The second table contains information about
    the number of variants, and the number of genes, transcripts and regulatory
    features overlapped by the input. </p>
    
    <h4> Charts and tables </h4>
    
    <p> There then follows several charts, most with accompanying tables. Tables
    and charts are interactive; clicking on a row to highlight it in the table
    will highlight the relevant segment in the chart, and vice versa. </p>
    
    <div class="tinted-box" style="width: 200px; padding-bottom: 5px; float: left; margin-right: 20px;">
        <a href="/img/vep_stats_1.png"><img style="width: 200px; border: 1px solid grey" src="/img/vep_stats_1.png">
        General statistics</a>
    </div>
    
    <div class="tinted-box" style="width: 200px; padding-bottom: 5px; float: left; margin-right: 20px;">
        <a href="/img/vep_stats_2.png"><img style="width: 200px; border: 1px solid grey" src="/img/vep_stats_2.png">
        Summary of called consequence types</a>
    </div>
    
    <div class="tinted-box" style="width: 200px; padding-bottom: 5px; float: left; margin-right: 20px;">
        <a href="/img/vep_stats_3.png"><img style="width: 200px; border: 1px solid grey" src="/img/vep_stats_3.png">
        Distribution of variants across chromosomes</a>
    </div>

</body>
</html>
