<html>
<head>
<title>VEP FAQ</title>
<meta name="order" content="7" />
</head>
<body>
<div>
  <div style="float:right;"><img src="/img/vep_logo.png"/></div>
	
  <h1 id="top"><span style="color:#006;padding-right:15px">Variant Effect Predictor</span><span style="color:#666"><img src="/i/16/help.png"/> FAQ</span></h1>
  <hr/>
    
    <p> For any questions not covered here, please send an email to the Ensembl
    <a href="/info/about/contact/index.html">developer's mailing list</a>
    (public) or contact the
    <a href="/Help/Contact" class="popup">Ensembl Helpdesk</a> (private).
    Also you can report issues through our (public) Github repositories. For general vep issues you should use 
    <a href="https://github.com/Ensembl/ensembl-vep/issues">ensembl-vep</a> repository and for specific plugins you should use 
    <a href="https://github.com/Ensembl/VEP_plugins/issues">VEP_plugins</a> repository.
    </p>

<h2 id="general">General questions</h2>

<p style="font-weight:bold;"> Q: Why has my insertion/deletion variant encoded in VCF
disappeared from the VEP output? </p>

<p> Ensembl treats unbalanced variants differently to VCF - your variant
hasn't disappeared, it may have just changed slightly! You can solve this by
giving your variants a unique identifier in the third column of the VCF file.
See <a href="vep_formats.html#vcf">here</a> for a full discussion.</p>

<p>&nbsp;</p>

<p style="font-weight:bold;"> Q: Why don't I see any co-located variants when
using species X? </p>

<p> Ensembl only has variation databases for a subset of all Ensembl species
- see <a href="/info/genome/variation/species/species_data_types.html#source">this
  document</a> for details.</p>

<p>&nbsp;</p>

<p style="font-weight:bold;"> Q: Why do I see multiple known variants mapped to my input variant? </p>

<p> VEP compares you input to known variants from the Ensembl variation database.
In some cases one input variant can match multiple known variants:
<ul>
  <li>Germline variants from dbSNP and somatic mutations from COSMIC may be found at the same locus</li>
  <li>Some sources, e.g. HGMD, do not provide public access to allele-specific data, so an HGMD variant with unknown alleles may colocate with one from dbSNP with known alleles</li>
  <li>Multiple alternate alleles from your input may match different variants as they are described in dbSNP</li>
</ul>

See <a href="script/vep_other.html#colocated">here</a> for a full discussion.</p>

<p>&nbsp;</p>

<p style="font-weight:bold;"> Q: VEP is not assigning a frequency to my input variant - why? </p>

<p> VEP's cache contains frequency data only for variants and alleles imported into Ensembl's variation database. See <a href="script/vep_other.html#colocated">here</a> for a full discussion. </p>

<p>&nbsp;</p>



<p style="font-weight:bold;"> Q: Why do I see so many lines of output for each variant in my input? </p>

<p> While it would be convenient to have a simple, one word answer to the
question "What is the consequence of this variant?", in reality biology is not
 this simple! Many genes have more than one transcript, so VEP
provides a prediction for each transcript that a variant overlaps. VEP
has options to help select results according to your requirements; the <a
href="script/vep_options.html#opt_canonical">--canonical</a> and <a
href="script/vep_options.html#opt_ccds">--ccds</a> options indicate which
transcripts are canonical and belong to the CCDS set respectively, while <a
href="script/vep_options.html#opt_pick">--pick</a>, <a
href="script/vep_options.html#opt_per_gene">--per_gene</a>, <a
href="script/vep_options.html#opt_summary">--summary</a> and <a
href="script/vep_options.html#opt_most_severe">--most_severe</a> allow you to
give a more summary level assessment per variant.</p>

<p> Furthermore, several "compound" consequences are also possible - if, for
example, a variant falls in the final few bases of an exon, it may be considered
to affect a splicing site, in addition to possibly affecting the coding
sequence. </p>


<p style="font-weight:bold;"> Q: How do I reduce VEP's memory requirement?</p>

<p> There are a number of ways to do this-
<ol>
  <li>Ensure your input file is sorted by location. This can greatly reduce memory
      requirements and runtime </li>
  <li>Consider reducing the buffer size. This reduces the number of variants annotated
      together in a batch and can be modified in both command line and web interfaces.
      Reducing buffer size may increase run time.</li>
  <li>Ensure you are only using the options you need, rather than --everything.
      Some data-rich options, such as regulatory annotation have an impact on memory use</li>
</ol>
</p>

<hr/>

<h2 id="web">Web VEP questions</h2>

<p style="font-weight:bold;"> Q: How do I access the web version of the Variant
Effect Predictor? </p>

<p> You can find the web VEP on the <a href="/info/docs/tools/index.html">Tools</a> page.</p>

<p>&nbsp;</p>


<p style="font-weight:bold;"> Q: Why is the output I get for my input file
different when I use the web VEP and command line VEP? </p>

<p> Ensure that you are passing equivalent arguments to the script that you
are using in the web version. If you are sure this is still a problem, please
report it on the <a href="http://lists.ensembl.org/mailman/listinfo/dev" rel="external">ensembl-dev</a> mailing list.</p>
<hr/>

<h2 id="script">Command line VEP questions</h2>

<p style="font-weight:bold;"> Q: How can I make VEP run faster? </p>

<p> There are a number of factors that influence how fast VEP runs. Have a look at our <a href="script/vep_other.html#faster">handy guide</a> for tips on improving VEP runtime. </p>

<p>&nbsp;</p>

<p style="font-weight:bold;"> Q: Why do I see "N" as the reference allele in my HGVS strings? </p>
<p style="font-weight:bold;"> Q: Why do I see the following error (or similar) in my VEP output?</p>
<pre class="code sh_sh">substr outside of string at /nfs/users/nfs_w/wm2/Perl/ensembl-variation/modules/Bio/EnsEMBL/Variation/Utils/Sequence.pm line 511.
Use of uninitialized value $ref_allele in string eq at /nfs/users/nfs_w/wm2/Perl/ensembl-variation/modules/Bio/EnsEMBL/Variation/Utils/Sequence.pm line 514.
Use of uninitialized value in concatenation (.) or string at /nfs/users/nfs_w/wm2/Perl/ensembl-variation/modules/Bio/EnsEMBL/Variation/Utils/Sequence.pm line 643.</pre>

<p> Both of these error types are usually seen when using a <a
href="script/vep_cache.html#fasta">FASTA file</a> for retrieving sequence.
There are a couple of steps you can take to try to remedy them:</p>

<ol>
  <li>
    The index alongside the FASTA can become corrupted. Delete
    [fastafile].index and re-run VEP to regenerate it. By default
    this file is located in your $HOME/.vep/[species]/[version]_[assembly]
    directory.
  </li>

  <li>
    The FASTA file itself may have been corrupted during download;
    delete the fasta file and the index and re-download (you can use the
    <a href="script/vep_download.html#install">VEP installer</a> to do this).
  </li>

  <li>
    Older versions of BioPerl (1.2.3 in particular is known to have this)
    cannot properly index large FASTA files. Make sure you are using a later
    (>=1.6) version of BioPerl. The <a href="script/vep_download.html#install">VEP
    installer</a> installs 1.6.924 for you.
  </li>
</ol>

<p> If you still see problems after taking these steps, or if you were not
using a FASTA file in the first place, please <a href="#top">contact us</a>. </p>

<p>&nbsp;</p>

<p style="font-weight:bold;"> Q: Why do I see the following warning?</p>

<pre class="code sh_sh">WARNING: Chromosome 21 not found in annotation sources or synonyms on line 160</pre>

<p> This can occur if the chromosome names differ between your input variant and any annotation source that you are using (cache, database, GFF/GTF file, FASTA file, custom annotation file). To circumvent this you may provide VEP with a <a href="script/vep_options.html#opt_synonyms">synonyms file</a>. A synonym file is included in VEP's cache files, so if you have one of these for your species you can use it as follows:</p>

<pre class="code sh_sh">./vep -i input.vcf -cache -synonyms ~/.vep/homo_sapiens/[[SPECIESDEFS::ENSEMBL_VERSION]]_GRCh38/chr_synonyms.txt</pre>

<p> The file consists of lines containing pairs of tab-separated synonyms. Order is not important as synonyms can be used in both "directions". </p>

<p>&nbsp;</p>

<p style="font-weight:bold;"> Q: Can I get gnomAD exomes and genomes frequencies in VEP? </p>

<p> Yes, see <a href="script/vep_example.html#gnomad">this guide</a>. </p>

<p>&nbsp;</p>

<p style="font-weight:bold;"> Q: Why do I see the following error? </p>

<pre class="code sh_sh">
Could not connect to database homo_sapiens_core_63_37 as user anonymous using [DBI:mysql:database=homo_sapiens_core_63_37;host=ensembldb.ensembl.org;port=5306] as a locator:
Unknown MySQL server host 'ensembldb.ensembl.org' (2) at $HOME/src/ensembl/modules/Bio/EnsEMBL/DBSQL/DBConnection.pm line 290.

-------------------- EXCEPTION --------------------
MSG: Could not connect to database homo_sapiens_core_63_37 as user anonymous using [DBI:mysql:database=homo_sapiens_core_63_37;host=ensembldb.ensembl.org;port=5306] as a locator:
Unknown MySQL server host 'ensembldb.ensembl.org' (2)
</pre>

<p> By default VEP is configured to connect to the public MySQL
server at ensembldb.ensembl.org. Occasionally the server may break connection
with your process, which causes this error. This can happen when the server is
busy, or due to various network issues. Consider using a <a
href="script/vep_cache.html#local">local copy of the database</a>, or the <a
href="script/vep_cache.html#cache">caching system</a>. </p>

<p>&nbsp;</p>

<p style="font-weight:bold;"> Q: Can I use VEP on Windows? </p>

<p> Yes - see the <a href="script/vep_download.html#windows">documentation</a> for a
few different ways to get the VEP running on Windows. </p>

<p>&nbsp;</p>

<p style="font-weight:bold;"> Q: Can I download all of the SIFT and/or PolyPhen predictions? </p>

<p> The Ensembl Variation database and the human VEP cache file contain
precalculated SIFT and PolyPhen-2 predictions for every possible amino acid change
in every translated protein product in Ensembl. Since these data are huge, we
store them in a compressed format. The best approach to extract them is to use
our Perl API.</p>

<p> The format in which the data are stored in our database is described <a
href="/info/genome/variation/prediction/protein_function.html#nsSNP_data_format">here</a></p>

<p> The simplest way to access these matrices is to use an API script to fetch a
ProteinFunctionPredictionMatrix for your protein of interest and then call its
'get_prediction' method to get the score for a particular position and amino
acid, looping over all possible amino acids for your position. There is some
detailed documentation on this class in the API documentation <a
href="/info/docs/Doxygen/variation-api/classBio_1_1EnsEMBL_1_1Variation_1_1ProteinFunctionPredictionMatrix.html">here</a>.</p>

<p>You would need to work out which peptide position your codon maps to, but
there are methods in the <a
href="/info/docs/Doxygen/variation-api/classBio_1_1EnsEMBL_1_1Variation_1_1TranscriptVariation.html">TranscriptVariation</a>
class that should help you (probably translation_start and translation_end).</p>

</div>

</body>
</html>
