<p>The naked mole-rat (Heterocephalus glaber), also known as the sand puppy or desert mole rat, is a burrowing rodent closely related to the blesmols, native to parts of East Africa, and is the only species currently classified in the genus Heterocephalus. The genus is classified as being in the family Bathyergidae or in its own family, Heterocephalidae. The naked mole-rat, the Damaraland mole-rat, and Mechow's mole-rat are eusocial mammals, the highest classification of sociality. It has a highly unusual set of physical traits that enable it to thrive in an otherwise harsh underground environment; it is the only mammalian thermoconformer, almost entirely ectothermic (cold-blooded) in how it regulates body temperature.</p>

<p>The gene annotation process was carried out using a combination of protein-to-genome alignments, annotation mapping from a suitable reference species and RNA-seq alignments (where RNA-seq data with appropriate meta data were publicly available).

For each candidate gene region, a selection process was applied to choose the most appropriate set of transcripts based on evolutionary distance, experimental evidence for the source data and quality of the alignments.
Small ncRNAs were obtained using a combination of BLAST and Infernal/RNAfold.

Pseudogenes were calculated by looking at genes with a large percentage of non-biological introns (introns of &lt;10bp), where the gene was covered in repeats, or where the gene was single exon and evidence of a functional multi-exon paralog was found elsewhere in the genome.

lincRNAs were generated via RNA-seq data where no evidence of protein homology or protein domains could be found in the transcript.</p>

<ul>
  <li><a href="/info/genome/genebuild/rodents_clade_gene_annotation.pdf">Detailed information on rodent clade  genebuild</a> (PDF)</li>
</ul>

<p>In accordance with the <a href="https://en.wikipedia.org/wiki/Fort_Lauderdale_Agreement">Fort Lauderdale Agreement </a>, please check the publication status of the genome/assembly before publishing any genome-wide analyses using these data.</p>
