<p>The annotation for the protein -coding genes was carried out using three main techniques:
<ul>
<li>Mouse lemur RNA-seq data from nine tissues.</li>
<li>Whole genome alignment against the human GRCh38 assembly followed by projection of GENCODE Basic protein-coding transcripts in regions of sufficient alignment conservation.</li>
<li>Splice-aware alignment of a subset of UniProt proteins to the Mmur_2.0 assembly.</li>
</ul>

<p>The subset of UniProt proteins used was our 'primates basic' set. This
      consisted of the proteins from the following clades and protein existence
      (PE) levels:
<ul>
<li>Human PE level 1 & 2 proteins.</li>
<li>Other primates PE level 1, 2 & 3 proteins.</li>
<li>Mouse PE level 1 & 2 proteins.</li>
<li>Other mammals PE level 1 & 2 proteins.</li>
<li>Other vertebrates PE level 1 & 2 proteins.</li>
</ul>

<p>UTRs were obtained (where possible) from the RNA-seq data and alignments of RefSeq mouse lemur cDNAs.</p>
<p>Small ncRNAs were obtained using a combination of BLAST and Infernal/RNAfold.</p>
<p>Pseudogenes were calculated by looking at genes with a large percentage of
      non-biological introns (introns of <10bp), where the gene was covered in
      repeats, or where the gene was single exon and evidence of a functional
      multi-exon paralog was found elsewhere in the genome.</p>

<p>The annotation process is described in the document below.</p>

<ul>
  <li><a href="/info/genome/genebuild/2016_09_08_mouselemur_genebuild.pdf">Detailed information on genebuild</a> (PDF)</li>
</ul>
