</p>
<p>The gene annotation process was carried out using a combination of protein-to-genome alignments, annotation mapping from a suitable reference species and RNA-seq alignments (where RNA-seq data with appropriate meta data were publicly available). For each candidate gene region, a selection process was applied to choose the most appropriate set of transcripts based on evolutionary distance, experimental evidence for the source data and quality of the alignments.</br>Small ncRNAs were obtained using a combination of BLAST and Infernal/RNAfold.</br> Pseudogenes were calculated by looking at genes with a large percentage of non-biological introns (introns of &lt;10bp), where the gene was covered in repeats, or where the gene was single exon and evidence of a functional multi-exon paralog was found elsewhere in the genome.</br> lincRNAs were generated via RNA-seq data where no evidence of protein homology or protein domains could be found in the transcript.</p>
<ul>
  <li><a href="/info/genome/genebuild/2020_08_tasmania_devil_gene_annotation.pdf">Detailed information on the genebuild</a> (PDF)</li>
</ul>
<p>In accordance with the <a href="https://en.wikipedia.org/wiki/Fort_Lauderdale_Agreement">Fort Lauderdale Agreement</a>, please check the publication status of the genome/assembly before publishing any genome-wide analyses using these data.</p>
<br>
<p><b>Data citation</b><br>
For use of these resources, please cite: <b>Stammnitz, MR et al. 2022, The evolution of two transmissible cancers in Tasmanian devils</b> (https://www.biorxiv.org/content/10.1101/2022.05.27.493404v1)<br></p>
