<p>
Sscrofa11.1 was annotated using the standard Ensembl automatic gene
annotation system, incorporating RNA-Seq data and PacBio long reads
provided by the SGSC.
The gene annotation process was carried out using a combination of
cDNA alignments, protein-to-genome alignments, PacBio long reads and
RNA-seq alignments. For each candidate gene region, a selection process
was applied to choose the most appropriate set of transcripts based on
evolutionary distance, experimental evidence for the source data and
quality of the alignments.
</br>
Small ncRNAs were obtained using a combination of BLAST and Infernal/RNAfold.
Pseudogenes were calculated by looking at genes with a large percentage
of non-biological introns (introns of &lt;10bp), where the gene was covered in
repeats, or where the gene was single exon and evidence of a functional
multi-exon paralog was found elsewhere in the genome. lincRNAs were generated
via transcriptomic data where no evidence of protein homology or protein
domains could be found in the transcript.
</p>

<p>
In accordance with the <a href="https://en.wikipedia.org/wiki/Fort_Lauderdale_Agreement" rel="external">Fort Lauderdale Agreement</a>,
please check the publication status of the genome/assembly before publishing any genome-wide analyses using these data.
</p>

<ul>
  <li><a href="/info/genome/genebuild/2022_05_Sus_scrofa_gene_annotation.pdf">Detailed information on pig genebuild</a> (PDF)</li>
</ul>
