<HTML>

<head>
  <meta name="order" content="3" />
  <meta name="navigation" content="Peak calling">
  <title>Peak calling</title>
</head>


<BODY LANG="en-GB" DIR="LTR">
  
  <h1>Peak calling</h1>

  <p>Peak calling is performed using <a href="https://github.com/jsh58/Genrich">Genrich</a>.</p>
  <p>Genrich supports biological replicates. Therefore, we first merge compatible technical replicates from the previous
    alignment steps to produce one alignment file per biosample.</p>
  <p>We use Genrich for calling peaks for ATAC-seq and ChIP-seq. The following parameters are in common:</p>
  <p style="font-size: small; margin-left: 15px">
    <code>-r</code>: Specifies PCR duplicates should be removed</br>
    <code>-e MT</code>: Reads aligning to the mitochondrial genome are excluded</br>
    <code>-E</code>: Regions to exclude from peak-calling (we exclude repeat masked regions)<br>
    <code>-s</code>: Keep secondary alignments with Alignment Score <code>(AS) >= bestAS - value</code>. We use a value
    of 20.<br>
  </p>
  <h2 style="font-size: small">ATAC-seq</h2>
  <p>In addition, ATAC-seq uses:</p>
  <p style="font-size: small; margin-left: 15px">
    <code>-j</code>: ATAC-seq mode<br>
    <code>-q</code>: We use a q-value of 0.1 as the threshold for peak-calling.<br>
  </p>
  <h2 style="font-size: small">ChIP-seq narrow peaks</h2>
  <p>Narrow peaks are ChIP-seq transcription factor targets and histone marks H3K27ac, H3K4me2, H3K4me3 and H3K9ac.</p>
  <p style="font-size: small; margin-left: 15px">
    <code>-y</code>: Keep unpaired alignments. This is necessary because some ChIP-seq experiments have single-end
    reads.
    This does not affect how paired alignments are processed since our alignment step for paired-end uses only
    properly paired reads.<br>
    <code>-q</code>: We use a q-value of 0.1 as the threshold for narrow peaks.<br>
  </p>
  <h2 style="font-size: small">ChIP-seq broad peaks</h2>
  <p>Broad peaks are ChIP-seq histone marks H3K27me3, H3K36me3 and H3K4me1. For these, we create a gapped peak track,
    which annotates broad peaks with peaks called with more restrictive parameters.</p>
  <p>Broad peaks are called with:</p>
  <p style="font-size: small; margin-left: 15px">
    <code>-p</code>: We use a p-value of 0.1 as the threshold for broad peaks.<br>
    <code>-g</code>: The maximum distance between significant sites is relaxed from 100 to 200 for broad peaks.<br>
    <code>-a</code>: The minimum AUC for a broad peak is increased from 200 to 800.<br>
  <p>
    Narrow peaks within these broad peaks are added based on:
  </p>
  <p style="font-size: small; margin-left: 15px">
    <code>-p</code>: We use a p-value of 0.05 as the threshold for narrow peaks within gapped peaks.<br>
  <p>
</BODY>

</HTML>